<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: MANUAL: apt-geno-qc (1.16.1)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="LICENSE.html"><span>License</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>MANUAL: apt-geno-qc (1.16.1) </h1>  </div>
</div>
<div class="contents">
<h2><a class="anchor" id="qccontents"></a>
Contents</h2>
<ul>
<li>
<a href="#intro">Introduction.</a> </li>
<li>
<a href="#quickstart">Quick Start - getting up and running.</a> </li>
<li>
<a href="#reportFile">The Report File - explanation of contents.</a> </li>
<li>
<a href="#options">Program Options - command line options.</a> </li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2><a class="anchor" id="chipqcIntr"></a>
Introduction</h2>
<p>apt-geno-qc is a program for making statistical assessments of experimental qualities from Affymetrix SNP microarrays using the model based algorithm DM.</p>
<p>For assessing Axiom&trade; arrays see <a href="VIGNETTE-Axiom-geno-qc.html">Axiom&trade; Single-Sample QC Analysis</a>.</p>
<p>As DM is a single sample model based algorithm it processes one CEL file a time in a multiple CEL file batch. Since DM requires MM existence apt-geno-qc only operates on SNPs tiled with mismatches, therefore, apt-geno-qc works on all SNPs on Mapping500K, but only works on a small subset of SNPs on SNP5 and SNP6 tiled for QC purposes (FQC SNPs).</p>
<p>The current option to output DM calls (see --dm-out) is experimental. This feature may go away or it may be moved into a separate program to generate DM calls.</p>
<p><a class="anchor" id="quickstart"></a> </p>
<h2><a class="anchor" id="QuickStart"></a>
Quick Start</h2>
<p>We illustrate the most basic way to run apt-geno-qc with an example. This example runs an analysis under the default parameter settings to generate a report file containing QC statistics for each CEL file with each specified method (or same method with different parameters). (Library files can be downloaded from <a href="http://www.affymetrix.com/estore/browse/level_one_category_template_one.jsp?category=35796">http://www.affymetrix.com/estore/browse/level_one_category_template_one.jsp?category=35796</a>)</p>
<p>An Axiom &trade; example: </p>
<div class="fragment"><pre class="fragment">
  apt-geno-qc \
    --analysis-files-path /library/file/path \
    --xml-file Axiom_GW_Hu_SNP.r4.apt-geno-qc.AxiomQC1.xml \
    --cel-files cel.txt \
    --out-file qc.txt
</pre></div><p> For more details on the use of apt-geno-qc with Axiom &trade; arrays see the vignette on <a href="VIGNETTE-Axiom-geno-qc.html">Axiom &trade; Single-Sample QC Analysis</a>.</p>
<p>A SNP6 example: </p>
<div class="fragment"><pre class="fragment">
apt-geno-qc \
  --cdf-file GenomeWideSNP_6.cdf \
  --qca-file GenomeWideSNP_6.r2.qca \
  --qcc-file GenomeWideSNP_6.r2.qcc \
  --chrX-probes GenomeWideSNP_6.chrXprobes \
  --chrY-probes GenomeWideSNP_6.chrYprobes \
  --cel-files cel_file_list.txt \
  --out-file results.txt
</pre></div><p>A SNP5 example: </p>
<div class="fragment"><pre class="fragment">
apt-geno-qc \
  --cdf-file GenomeWideSNP_5.cdf \
  --qca-file GenomeWideSNP_5.qca \
  --qcc-file GenomeWideSNP_5.qcc \
  --cel-files cel_file_list.txt \
  --out-file results.txt
</pre></div><p>A 500K example: </p>
<div class="fragment"><pre class="fragment">
apt-geno-qc \
  --cdf-file Mapping250K_Sty.cdf \
  --qca-file Mapping250K_Sty.qca \
  --qcc-file Mapping250K_Sty.qcc \
  --cel-files cel_file_list.txt \
  --out-file results.txt
</pre></div><p><a class="anchor" id="reportFile"></a> </p>
<h2><a class="anchor" id="reportFile"></a>
The Report File</h2>
<p>apt-geno-qc creates a report file with file name specified by --out-file option. The report file contains QC stats to measure the experimental quality for each CEL file. The format of the file is tab-delimited text with a header followed by a column header, then a line for each CEL file analyzed and a column for each method specified in the configuration file. The header contains information of which CDF file is used and what kind of methods are used for generating QC stats, and parameters associated with each method. The column entries are:</p>
<ol type="1">
<li>The CEL file name.</li>
<li>The stats for each method, variable number of columns equal to number of methods in the configuration file</li>
</ol>
<p><a class="anchor" id="options"></a> </p>
<h3><a class="anchor" id="manualOptions"></a>
Options:</h3>
<div class="fragment"><pre class="fragment">apt-geno-qc - a single-chip-based genotyping chip quality control tool,
it reports one or more metrics for each chip analyzed.  The metrics
computed are determined by one of the supplied input files (the qca file).

usage:
  apt-geno-qc --cdf-file my_chip.cdf --qcc-file my_chip.qcc \
    --qca-file my_chip.qca --out-file my_output.txt \
    --cel-files my_cel_files.txt

  apt-geno-qc -c my_chip.cdf -q my_chip.qcc -a my_chip.qca \
    --out-file my_output.txt --cel-files my_cel_files.txt


options:
 Common Options (not used by all programs)
   -h, --help                           Display program options and extra
                          documentation about possible analyses. See
                          -explain for information about a specific
                          operation. [default 'false'] 
   -v, --verbose How verbose to be with status messages 0 -
                          quiet, 1 - usual messages, 2 - more
                          messages. [default '1'] 
     --console-off Turn off the default messages to the 
                          console but not logging or sockets. 
                          [default 'false'] 
     --use-socket Host and port to print messages over in
                          localhost:port format [default ''] 
     --version Display version information. [default
                          'false'] 
   -f, --force Disable various checks including chip 
                          types. Consider using --chip-type option
                          rather than --force. [default 'false'] 
     --throw-exception Throw an exception rather than calling
                          exit() on error. Useful for debugging. This
                          option is intended for command line use
                          only. If you are wrapping an Engine and 
                          want exceptions thrown, then you should 
                          call Err::setThrowStatus(true) to ensure
                          that all Err::errAbort() calls result in an
                          exception. [default 'false'] 
     --analysis-files-path Search path for analysis library files. 
                          Will override AFFX_ANALYSIS_FILES_PATH
                          environment variable. [default ''] 
     --xml-file Input parameters in XML format (Will
                          override command line settings). [default
                          ''] 
     --temp-dir Directory for temporary files when working
                          off disk. Using network mounted drives is
                          not advised. When not set, the output 
                          folder will be used. The defaut is 
                          typically the output directory or the
                          current working directory. [default ''] 
   -o, --out-dir Directory for output files. Defaults to
                          current working directory. [default '.'] 
     --log-file The name of the log file. Generally 
                          defaults to the program name in the out-dir
                          folder. [default ''] 
 Engine Options (Not used on command line)
     --command-line The command line executed. [default ''] 
     --exec-guid The GUID for the process. [default ''] 
     --program-name The name of the program [default ''] 
     --program-company The company providing the program [default
                          ''] 
     --program-version The version of the program [default ''] 
     --program-cvs-id The CVS version of the program [default ''] 
     --version-to-report The version to report in the output files.
                          [default ''] 
     --free-mem-at-start How much physical memory was available when
                          the engine run started. [default '0'] 
     --meta-data-info Meta data in key=value pair that will be
                          output in headers. [default ''] 
 Input Options
   -c, --cdf-file File defining probe sets. [default ''] 
     --spf-file SPF File defining probe sets. [default ''] 
   -q, --qcc-file File defining QC probesets. [default ''] 
   -a, --qca-file File defining QC analysis methods. [default
                          ''] 
     --cel-files Text file specifying cel files to process,
                          one per line with the first line being
                          'cel_files'. [default ''] 
     --chrX-probes File containing probe_id (1-based) of 
                          probes on chrX. Used for copy number probe
                          chrX/Y ratio gender calling. [Experimental]
                          [default ''] 
     --chrY-probes File containing probe_id (1-based) of 
                          probes on chrY. Used for copy number probe
                          chrX/Y ratio gender calling. [Experimental]
                          [default ''] 
     --chrZ-probes File containing probe_id (1-based) of 
                          probes on chrZ. Used for copy number probe
                          chrZ/W ratio avian gender calling.
                          [Experimental] [default ''] 
     --chrW-probes File containing probe_id (1-based) of 
                          probes on chrW. Used for copy number probe
                          chrZ/W ratio avian gender calling.
                          [Experimental] [default ''] 
     --probe-class-file File containing probe_id (1-based) of 
                          probes and a 'class' designation. Used to
                          compute mean probe intensity by class for
                          report file. [default ''] 
     --target-sketch Sketch file. [default ''] 
     --channel-file Channel file. [default ''] 
     --reagent-kit-discriminator list of probeset names with pc1s and means
                          to use for classifying the reagent kits.
                          [default ''] 
 Output Options
     --out-file Name to use for the output file. [default
                          'apt-geno-qc.report.txt'] 
     --dm-out Folder to use for DM output. Enables DM
                          output. One per CEL file. [experimental]
                          [default ''] 
 Analysis Options
     --dm-het-mult DM Het Mult to use for DM output. [default
                          '1.25'] [default '1.25'] 
     --dm-thresh DM threshold to use for making no calls.
                          [default '0.33'] [default '0.33'] 
     --female-thresh Threshold for calling females when using
                          cn-probe-chrXY-ratio or 
                          cn-probe-chrZW-ratio method. [default
                          '0.48'] 
     --male-thresh Threshold for calling females when using
                          cn-probe-chrXY-ratio or 
                          cn-probe-chrZW-ratio method. [default
                          '0.71'] 
 Engine Options (Not used on command line)
     --cels Cel files to process. [default ''] 
     --time-start The time the engine run was started 
                          [default ''] 
     --time-end The time the engine run ended [default ''] 
     --time-run-minutes The run time in minutes. [default ''] 
     --analysis-guid The GUID for the analysis run. [default ''] 
</pre></div><p><a class="anchor" id="faq"></a> </p>
<h2><a class="anchor" id="manFaq"></a>
Frequently Asked Questions</h2>
<p>Q. What is a probe_id?</p>
<p>A. See the <a href="FAQ.html#probe_id_faq">FAQ item on probe IDs</a> for more info. </p>
</div>
<hr class="footer"/><address class="footer"><small>Generated for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
