<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Affymetrix Power Tools: MANUAL: apt-copynumber-onco-ssa (2.1.0)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="LICENSE.html"><span>License</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<h1>MANUAL: apt-copynumber-onco-ssa (2.1.0) </h1>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h2>Contents </h2>
<ul>
<li><a href="#intro">Introduction.</a></li>
<li><a href="#quickstart">Quick Start</a></li>
<li><a href="#options">Options</a></li>
<li><a href="#references">References</a></li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2>Introduction </h2>
<p><b>apt-copynumber-onco-ssa</b> implements the copynumber analysis pipeline for OncoScan arrays. Input for each OncoScan experiment consists of a pair of CEL files for AT and GC channels. For each experiment <code>apt-copynumber-onco-ssa</code> generates an <a href="oschp.html">OSCHP</a> file.</p>
<p><code>apt-copynumber-onco-ssa</code> supports two modes of operation: single sample mode and “with matched normal” mode. In the “with matched normal” mode a pair of cel files is identified as “normal” and a pair as “tumor” and the normal data is used to identify heterozygous markers for use by the algorithm when analyzing the tumor pair. In the “with matched normal” analysis the log2ratios are still calculated relative to the reference model file. (This is not a “paired analysis”.)</p>
<p><a class="anchor" id="quickstart"></a> </p>
<h2>Quick Start </h2>
<h3>Input/Output Files</h3>
<p>The following input files are required by apt-copynumber-onco-ssa:</p>
<table class="doxtable">
<tr>
<th>Name </th><th>File extension/type </th><th>Description  </th></tr>
<tr>
<td>reference-file </td><td>REF_MODEL (HDF5) </td><td>Reference data generated by apt-copynumber-onco-ref </td></tr>
<tr>
<td>snp-qc-snp-list </td><td>txt </td><td>SNP probes used in calculating SNP QC </td></tr>
<tr>
<td>annotation-file </td><td>annot.db (SQLite) </td><td>Annotation data for probes and probesets </td></tr>
<tr>
<td>x-probes-file </td><td>chrXprobes (text) </td><td>List of X-chromosome probes </td></tr>
<tr>
<td>y-probes-file </td><td>chrYprobes (text) </td><td>List of Y-chromosome probes </td></tr>
<tr>
<td>cel-pairs-file </td><td>tsv </td><td>Cel file pairs (AT and GC channel for each sample) for experiments </td></tr>
</table>
<p>The following output files are generated by apt-copynumber-onco:</p>
<table class="doxtable">
<tr>
<th>Name </th><th>File extension/type </th><th>Description  </th></tr>
<tr>
<td>OSCHP file </td><td>OSCHP (HDF5) </td><td>Copynumber and other analysis data for each experiment </td></tr>
<tr>
<td>report-file </td><td>text </td><td>Summary statistics for the experiments </td></tr>
<tr>
<td>sig-snp-report-file </td><td>text </td><td>Results of signature SNP checks for the experiments. </td></tr>
</table>
<h3>CEL files lists:</h3>
<p>The CEL file pairs are specified in a tab delimited text file. In single-sample mode this file will have either two or three columns, with two columns representing the AT-channel and the GC-channel CEL files. In normal/tumor pair mode, the file will have either four or five columns, with four columns representing the normal AT and GC-channel CEL files, and the tumor AT and GC-channel CEL files. The optional third or fifth column can be used to specify the name of the output OSCHP files. (If omitted an OSCHP file name is generated based on the CEL file names.)</p>
<p>The first row of the cel-pairs-file should contain the column names. Supported column names are:</p>
<pre class="fragment">  result                    - the OSCHP file name to be generated (optional)
  atchannelcel              - the AT channel CEL file
  gcchannelcel              - the GC channel CEL file
  atchannelmatchednormalcel - the AT channel CEL file for matched normal sample
  gcchannelmatchednormalcel - the GC channel CEL file for matched normal sample
</pre><p>In matched tumor/normal pair mode, each row in the cel-pairs-file will generated two OSCHP files: one for the tumor sample and one for the matched normal sample. These OSCHP files will be distinguished by adding the postfixes '<code>_N</code>' and '<code>_T</code>' to the file names.</p>
<p>See <a href="VIGNETTE-OncoScan-ssa.html">OncoScan single-sample analysis</a> for examples.</p>
<p><a class="anchor" id="options"></a> </p>
<h2>Options: </h2>
<pre class="fragment">
   'apt-copynumber-onco-ssa'
   2.1.0 svn://svn.affymetrix.com/apt2-copynumber/branches/APT_CN_2_1_0_BRANCH@915(1263) 2014-05-14-10:58

USER HELP ----------

  --version (flag) = Display the version info for this program.

  --user-help (flag) = Display help intended for end-users.
    Aliases: --help, -h

  --log-file (string8) = The name of the log file.  Defaults to 'PROGRAMNAME-NUMBER.log'.
    Aliases: --lf

  --console-add-select (string8) = Add selectors for console messages.
    e.g. to include all debug messages: --console-add-select debug
         to include all messages of level 3 or higher: --console-add-select '*:3'


  --console-add-neg-select (string8) = Add selectors to be excluded from console messages.
    e.g. to exclude all warning messages and errors summary: --console-add-neg-select WARNING,summary


  --arg-file (filename) = Read arguments from this file.
    File should be a valid XML file, and should contain elements of the form
    &lt;Parameter name="PARAMETER-NAME" analysis="NODE-NAME" currentValue="VALUE" /&gt;.

  --check-param (flag) = Stop the program after the command parameters have been checked.

  --out-dir (string8) = The output directory for result files.

  --temp-dir (string8) = The output directory for temporary files.

  --force (bool) = Force the files to process regardless of array type.

  --keep-intermediate-data (bool) = Dump data on itermediate calculations for testing purposes.

  --keep-combined-cels (bool) = Keep the combined cel file data.

  --snp-qc-snp-list (filename) = SNPs to use in calculating SNP QC.

  --reference-file (string8) = The reference input file.

  --y-target (float64) = The Y chromosome target log2 ratio value.
    default value: '-0.500000'

  --nddqc-min-count (int32) = Minimum number of normal diploid markers that must be found to be used as the basis for QC calculations.
    default value: '500'

  --set-analysis-name (string8) = Analysis name to use as prefix for output files.

  --file-name-prefix (string8) = OSCHP file name prefix

  --file-name-suffix (string8) = OSCHP file name suffix

  --brlmmp-HARD (int32) = brlmmp hardshell
    default value: '3'

  --brlmmp-SB (float64) = brlmmp shellbarrier
    default value: '0.450000'

  --brlmmp-CM (int32) = brlmmp callmethod (1=batch mode, 2=single sample mode)
    default value: '2'

  --brlmmp-bins (int32) = brlmmp bins
    default value: '100'

  --brlmmp-mix (int32) = brlmmp mix
    default value: '1'

  --brlmmp-bic (float64) = brlmmp bic
    default value: '2.000000'

  --brlmmp-CSepPen (float64) = brlmmp CSepPen
    default value: '0.000000'

  --brlmmp-CSepThr (float64) = brlmmp CSepThr
    default value: '16.000000'

  --brlmmp-lambda (float64) = brlmmp lambda
    default value: '1.000000'

  --brlmmp-wobble (float64) = brlmmp wobble
    default value: '0.050000'

  --brlmmp-copyqc (float64) = brlmmp copyqc. 0 = no test for copy qc
    default value: '0.000000'

  --brlmmp-copytype (int32) = brlmmp copytype. 0 = standard copy qc method
    default value: '0'

  --brlmmp-ocean (float64) = brlmmp ocean
    default value: '0.000000'

  --brlmmp-clustertype (int32) = brlmmp clustertype. 1 = standard 1-d clustering only
    default value: '1'

  --brlmmp-transform (string8) = brlmmp transform. (mva, rvt, ssf, ces, assf, ccs)
    default value: 'mva'

  --brlmmp-MS (float64) = brlmmp MS. confidence threshold
    default value: '0.050000'

  --brlmmp-hints (int32) = brlmmp hints
    default value: '0'

  --brlmmp-IsoHetY (float64) = brlmmp IsoHetY
    default value: '0.000000'

  --brlmmp-inflatePRA (float64) = brlmmp inflatePRA. 0 = no increase in uncertainty
    default value: '0.000000'

  --brlmmp-CP (float64) = brlmmp contradiction penalty. 0 = no penalty
    default value: '0.000000'

  --annotation-file (filename,required) = The annotation input file.

  --x-probes-file (filename) = The file that identifies the X chromosome probe ids.

  --y-probes-file (filename) = The file that identifies the Y chromosome probe ids.

  --cel-pairs-file (filename) = The file that identifies the cel file pairs to process.

  --cel-files (filename) = The file that identifies the cel files to process.

  --loh-files (filename) = The file that identifies the loh files to process.

  --fld-file (filename) = The string that identifies the pFLD values to load into the reference.

  --ap-baf-fld-weights (filename) = Name of file with FLD-derived weights.
    Aliases: --baf-fld-weights

  --agr-denominator-source (string8) = The denominator to use in calculating the antigenomic ratio.

  --agr-denominator-percentile (float64) = The percentile denominator to use in calculating the antigenomic ratio denominator.
    default value: '0.500000'

  --igender-female-threshold (float64) = The female threshold to use in the raw intensity gender call.
    default value: '0.500000'

  --igender-male-threshold (float64) = The male threshold to use in the raw intensity gender call.
    default value: '0.700000'

  --nddicov-min-nd-probe-count-per-bin (int32) = Override the icov-min-nd-probe-count-per-bin parameter. This is intended to be used in SSA mode.
    default value: '30'

  --nddicov-use-nd-signal-markers (string8) = Override the icov-use-nd-signal-markers parameter (on/off). This is intended to be used in SSA mode.
    default value: 'off'

  --dual-channel-normalization (bool) = Use dual channel normalization.

  --ad-outlier-trim (float64) = The trim value to use for alleleic difference outliers.
    default value: '3.000000'

  --ndd-weights-type (string8) = Normal diploid detection weigths types (FLD, replicates, equal).
    default value: 'FLD'

  --ndd-point-count (int32) = Normal diploid detection density point count.
    default value: '128'

  --ndd-bandwidth (float64) = Normal diploid detection density bandwidth.
    default value: '0.250000'

  --ndd-cutoff (float64) = Normal diploid detection density cutoff.
    default value: '0.050000'

  --ndd-step (int32) = Normal diploid detection moving window step.
    default value: '40'

  --ndd-window (int32) = Normal diploid detection moving window window size.
    default value: '350'

  --ndd-min-markers (int32) = Normal diploid detection moving window min markers.
    default value: '10'

  --ndd-log2-threshold (float64) = Normal diploid detection window median log2 ratio threshold.
    default value: '0.250000'

  --ndd-th1 (string8) = Normal diploid detection th1 (pair of values).
    default value: '0,0.45'

  --ndd-th2 (string8) = Normal diploid detection th2 (pair of values).
    default value: '0.85,1.25'

  --ndd-th3 (string8) = Normal diploid detection th3 (pair of values).
    default value: '0.7,1.3'

  --ndd-th4 (string8) = Normal diploid detection th4 (pair of values).
    default value: '0,0.3'

  --ndd-min-percent-nd (float64) = Minimum percent of normal-diploid autosomal probe sets required.
    default value: '0.000000'

  --ndd-min-number-nd (int32) = Minimum number of normal-diploid autosomal probe sets required.
    default value: '30'

  --ndd-med-l2r-low-percentile-ND (float64) = Normal diploid detection median log2 ratio low percentile.
    default value: '0.005000'

  --ndd-med-l2r-threshold-ND (float64) = Normal diploid detection median log2 ratio threshold.
    default value: '0.280000'

  --ndd-iqr-factor-coarse (float64) = Normal diploid detection IQR factor (coarse).
    default value: '5.000000'

  --ndd-iqr-factor-fine (float64) = Normal diploid detection IQR factor (fine).
    default value: '3.000000'

  --sig-gender-reference-chromosome (int32) = XY signal gender reference chromosome
    default value: '2'

  --sig-gender-xx-cutoff (float64) = XY signal gender XX cutoff
    default value: '0.800000'

  --sig-gender-xx-cutoff-high (float64) = XY signal gender XX cutoff high
    default value: '1.070000'

  --sig-gender-y-cutoff (float64) = XY signal gender Y cutoff
    default value: '0.650000'

  --hpf-mini-block-rows (int32) = Mini block rows.
    default value: '8'

  --hpf-mini-block-cols (int32) = Mini block cols.
    default value: '8'

  --hpf-global-smooth-weight (float64) = Global smooth weight.
    default value: '256.000000'

  --hpf-local-smooth-weight (float64) = Local smooth weight.
    default value: '64.000000'

  --hpf-converged (float64) = Converged.
    default value: '0.000100'

  --nddlrcov-min-nd-marker-count-per-bin (int32) = Override the lrcov-min-nd-marker-count-per-bin parameter. This is intended to be used in SSA mode.
    default value: '30'

  --nddlrcov-use-nd-lr-markers (string8) = Override the lrcov-use-nd-lr-markers parameter (on/off). This is intended to be used in SSA mode.
    default value: 'on'

  --wave-bandwidth (int32) = Wave correction bandwidth.
    default value: '101'

  --wave-bin-count (int32) = Wave correction bin count.
    default value: '25'

  --wave-count (int32) = Wave correction wave count.
    default value: '-1'

  --wave-smooth (bool) = Wave correction wave smooth.

  --median-autosome-median-normalization (bool) = Log2Ratio Median Autosmome Median Normalization

  --baf-transform (string8) = BAF transformation (linear, TBD).
    default value: 'linear'

  --baf-trim (float64) = BAF trim value. +/-
    default value: '10.000000'

  --ap-baf-step (int32) = The step to use in calculating allele peaks.
    default value: '20'
    Aliases: --baf-step

  --ap-baf-window (int32) = The window to use in calculating allele peaks.
    default value: '100'
    Aliases: --baf-window

  --ap-baf-point-count (int32) = The point count to use in calculating allele peaks.
    default value: '128'
    Aliases: --baf-point-count

  --ap-baf-bandwidth (float64) = The bandwidth to use in calculating allele peaks.
    default value: '0.250000'
    Aliases: --baf-bandwidth

  --ap-baf-cutoff (float64) = The cutoff to use in calculating allele peaks.
    default value: '0.050000'
    Aliases: --baf-cutoff

  --ap-baf-threshold (float64) = The clean threshold to use in calculating allele peaks.
    default value: '0.350000'
    Aliases: --baf-threshold

  --ap-baf-height-threshold (float64) = The height threshold factor to use in calculating allele peaks.
    default value: '0.000000'
    Aliases: --baf-height-threshold

  --ap-baf-height-threshold-bound (float64) = The bounds between which the height threshold is applied.
    default value: '1024.000000'
    Aliases: --baf-height-threshold-bound

  --ap-baf-symmetry (bool) = The symmetry value to use in calculating allele peaks.
    Aliases: --baf-symmetry

  --tuscan-m3-low-cost-band (float64) = assign trim-weight if BAF segment average is closer to 0.5 than this value
    default value: '0.030000'
    Aliases: --ascat-m3-low-cost-band

  --tuscan-nd-snp-qc-threshold (float64) = threshold value for NDSNPQC to trigger the alternate TuScan mode
    default value: '19.000000'
    Aliases: --ascat-nd-raw-snp-qc-threshold, --tuscan-nd-raw-snp-qc-threshold

  --tuscan-mapd-qc-threshold (float64) = threshold value for MAPD QC to trigger the alternate TuScan mode
    default value: '0.400000'
    Aliases: --ascat-mapd-qc-threshold

  --tuscan-run-once (bool) = do not rerun TuScan when this flag is set to true

  --tuscan-m1-shift (float64) = fraction of the BAF derivative first and second threshold crossing difference to offset the inflection point position by
    default value: '0.000000'
    Aliases: --ascat-m1-shift

  --tuscan-m1-trim (float64) = fraction of BAF length to cut off at both ends when searching for inflection point
    default value: '0.010000'
    Aliases: --ascat-m1-trim

  --tuscan-m1-threshold (float64) = threshold to pick the rightmost derivative maximum in hom/het separation
    default value: '0.001050'
    Aliases: --ascat-m1-threshold

  --tuscan-m1-use-raw-baf (bool) = use raw BAF values instead of cleaned BAFs in TuScan

  --tuscan-m2-lr-cost (float64) = float value for log-ratio cost weight.
    default value: '0.500000'
    Aliases: --ascat-m2-lr-cost

  --tuscan-m2-min-seg-size (int32) = integer value for minimum probe spacing.
    default value: '151'
    Aliases: --ascat-m2-min-seg-size

  --tuscan-m2-min-alt-seg-size (int32) = integer value for minimum probe spacing in alternate mode segmentation
    default value: '19'

  --tuscan-m2-min-fine-seg-size (int32) = integer value for minimum probe spacing in fine segmentation.
    default value: '19'
    Aliases: --ascat-m2-min-fine-seg-size

  --tuscan-m2-exclude-chr (string8) = comma-separated list of chromosomes to exclude from segmentation
    Aliases: --ascat-m2-exclude-chr

  --tuscan-consecutive-hom-count (int32) = force a segment break after a run of this many consecutive homs assuming no existing breakpoints within the vicinity of tuscan-no-break-vicinity
    default value: '100'

  --tuscan-no-break-vicinity (int32) = parameter describing the length needed for tuscan-consecutive-hom-count
    default value: '19'

  --lambda (float64) = cost of segment break
    default value: '50.000000'

  --tuscan-centromere-breakpoints (string8) = comma-separated list of pairs chromosome,genomic position denoting the centromere
    default value: '1,121535434,2,92326171,3,90504854,4,49660117,5,46405641,6,58830166,7,58054331,8,43838887,9,47367679,10,39254935,11,51644205,12,34856694,13,-1,14,-1,15,-1,16,35335801,17,22263006,18,15460898,19,24681782,20,26369569,21,-1,22,-1,24,58632012,25,10104553'

  --tuscan-centromere-min-seg-size (int32) = merge segments resulting from centromere breakpoints if they are smaller than this number
    default value: '19'

  --tuscan-m3-rho-min (float64) = minimum aberrant cell fraction value
    default value: '0.200000'
    Aliases: --ascat-m3-rho-min

  --tuscan-m3-rho-max (float64) = maximum aberrant cell fraction value
    default value: '1.000000'
    Aliases: --ascat-m3-rho-max

  --tuscan-m3-rho-N (int32) = number of aberrant cell fraction values between min and max
    default value: '100'
    Aliases: --ascat-m3-rho-N

  --tuscan-m3-psi-min (float64) = minimum tumor ploidy
    default value: '1.000000'
    Aliases: --ascat-m3-psi-min

  --tuscan-m3-psi-max (float64) = maximum tumor ploidy
    default value: '6.000000'
    Aliases: --ascat-m3-psi-max

  --tuscan-m3-psi-N (int32) = number of tumor ploidy values between min and max
    default value: '100'
    Aliases: --ascat-m3-psi-N

  --tuscan-m3-low-cost-weight (float64) = weight assigned if BAF segment average is within tuscan-m3-low-cost-band of 0.5
    default value: '0.010000'
    Aliases: --ascat-m3-low-cost-weight

  --tuscan-m3-max-percent-hom (float64) = maximum segment hom percentage to include it in the distance matrix and to avoid imputation
    default value: '85.000000'
    Aliases: --ascat-m3-max-percent-hom

  --tuscan-m3-min-count-het (int32) = minimum segment het count to include it in the distance matrix and to avoid imputation
    default value: '10'
    Aliases: --ascat-m3-min-count-het

  --tuscan-m3-ploidy-factor (float64) = multiplicative factor for ploidy-2 local minimum values
    default value: '3.000000'
    Aliases: --ascat-m3-ploidy-factor

  --tuscan-m3-ploidy-sigma (float64) = standard deviation of the Gaussian multiplicative factor for ploidy-2 local minimum values
    default value: '0.350000'
    Aliases: --ascat-m3-ploidy-sigma

  --alt-mode-bin-size (float64) = bin size to round copy numbers to in alternate processing modes
    default value: '0.333333'
    Aliases: --failure-mode-bin-size

  --nGoF-bin-size (float64) = bin size to round log2 ratios to when evaluating the nGoF4 metric
    default value: '0.500000'

  --MedL2Rcutoff (float64) = log ratio cutoff for fine segments to participate in the nGoF calculation
    default value: '0.080000'

  --MedBAFcutoff (float64) = BAF cutoff for coarse segments to participate in the nGoF4 calculation
    default value: '0.000000'

  --min-nGoF-length (int32) = minimum number of markers in log2-ratio-merged coarse segments to participate in the nGoF4 calculation
    default value: '1001'

  --MedL2Rcutoff-no-cn2 (float64) = log ratio cutoff for fine segments to participate in the nGoF calculation
    default value: '0.080000'

  --MedBAFcutoff-no-cn2 (float64) = BAF cutoff for coarse segments to participate in the nGoF4 calculation
    default value: '0.000000'

  --min-nGoF-length-no-cn2 (int32) = minimum number of markers in log2-ratio-merged coarse segments to participate in the nGoF4 calculation
    default value: '51'

  --nGoF4cutoff (float64) = trigger the alternate mode when nGoF4 is greater than this parameter
    default value: '1000.000000'

  --nGoF4cutoff-no-cn2 (float64) = trigger the alternate mode when nGoF4_noCN2 is greater than this parameter
    default value: '5000.000000'

  --gamma (float64) = the compaction factor.
    default value: '0.800000'

  --tuscan-round-ploidy (bool) = whether to round best ploidy before imputation

  --MinChrYthreshold (float64) = Use the modified imputation formula for chr X/Y in males if median l2r on Y is greater than this number
    default value: '-3.000000'

  --infMarkL2RMin (float64) = Informative marker L2R threshold in Module 3
    default value: '0.060000'

  --infMarkerThreshold (int32) = Total number of informative markers threshold in Module 3
    default value: '1000'

  --tuscan-infMarkTrimBand (float64) = 
    default value: '0.050000'

  --tuscan-fp-exponent (float64) = The exponent used in the false positive adjustment formula
    default value: '2.000000'

  --tuscan-fp-min-med-l2r (float64) = The minimum log2 ratio value for the false positive adjustment formula to apply
    default value: '0.100000'

  --tuscan-fp-max-med-l2r (float64) = The maximum log2 ratio value for the false positive adjustment formula to apply
    default value: '0.250000'

  --tuscan-use-fp-exponent-in-normal-mode (bool) = whether to apply the false positive adjustment formula in the normal mode

  --tuscan-maxCN2Weight (float64) = Parameter to balance the weights used in heat map calculation
    default value: '0.025000'

  --tuscan-maxNonCN2Increase (float64) = Parameter to cap the non-CN2 weights used in heat map calculation
    default value: '0.100000'

  --tuscan-FN_l2r_floor (float64) = False negative floor
    default value: '0.100000'

  --tuscan-FN_exp (float64) = False negative exponent
    default value: '2.460000'

  --tuscan-FN_low_l2r_threshold (float64) = 
    default value: '0.080000'

  --tuscan-FN_low_BAF_threshold (float64) = 
    default value: '0.480000'

  --tuscan-REL_l2r_weight (float64) = Fraction of log2 ratio reliability to use in the total reliability score
    default value: '0.500000'

  --tuscan-REL_CN2_gauss_P (float64) = Copy number 2 Gaussian amplitude for reliability score
    default value: '1.100000'

  --tuscan-REL_CN2_gauss_sigma (float64) = Copy number 2 Gaussian standard deviation for reliability score
    default value: '0.350000'

  --tuscan-REL_CN2_gauss_mu (float64) = Copy number 2 Gaussian mean for reliability score
    default value: '2.000000'

  --tuscan-REL_CN4_gauss_P (float64) = Copy number 4 Gaussian amplitude for reliability score
    default value: '1.023000'

  --tuscan-REL_CN4_gauss_sigma (float64) = Copy number 4 Gaussian standard deviation for reliability score
    default value: '0.200000'

  --tuscan-REL_CN4_gauss_mu (float64) = Copy number 4 Gaussian mean for reliability score
    default value: '4.000000'

  --tuscan-REL_threshold (float64) = Total reliability score threshold for engaging the alternate mode
    default value: '0.100000'

  --tuscan-m3-ploidy4-factor (float64) = 
    default value: '1.250000'

  --tuscan-m3-ploidy4-sigma (float64) = 
    default value: '0.200000'

  --tuscan-m3-ploidy4-mu (float64) = 
    default value: '4.000000'

  --tuscan-lowPloidy1 (float64) = 
    default value: '0.000000'

  --tuscan-lowPloidy2 (float64) = 
    default value: '1.760000'

  --tuscan-lowPloidyTB (float64) = 
    default value: '1.000000'

  --tuscan-lowPloidy_nGoF4 (float64) = 
    default value: '200.000000'

  --tuscan-hiPloidy1 (float64) = 
    default value: '2.300000'

  --tuscan-hiPloidy2 (float64) = 
    default value: '2.500000'

  --tuscan-hiPloidyTB (float64) = 
    default value: '1.000000'

  --tuscan-hiPloidy_nGoF4 (float64) = 
    default value: '200.000000'

  --tuscan-REL_threshold1 (float64) = 
    default value: '0.760000'

  --tuscan-REL_threshold2_hi (float64) = 
    default value: '0.900000'

  --tuscan-REL_threshold2_low (float64) = 
    default value: '0.700000'

  --tuscan-REL_maxMinimumIndex (int32) = 
    default value: '2'

  --tuscan-REL_noSearchIfminWeight (bool) = 

  --tuscan-report-internal-ploidy (bool) = 

  --shrink-lprec (vec_float32) = Precision of the likelihood for wavelet shrinkage.
    default value: '0.250000,2.000000'

  --outlier-shrink-lprec (vec_float32) = Precision of the likelihood for wavelet shrinkage outlier handling.
    default value: '0.025000,2.000000,8.000000'

  --shrink-converge (float64) = Convergence criteria for shrinkage.
    default value: '0.000001'

  --outlier-shrink-converge (float64) = Convergence criteria for shrinkage when downweighting outliers.
    default value: '0.000001'

  --shrink-downweight-maxiter (int32) = Maximum iterations for downweighting outliers during shrinkage.
    default value: '25'

  --outlier-shrink-large-stdev (float64) = Large standard deviation for downweighting outliers in shrinkage.
    default value: '125.000000'

  --cn-gender-cutoff (float64) = CN Gender cutoff
    default value: '0.500000'

  --cn-state-threshold (float64) = CN Gender threshold for counting CN state as nonzero
    default value: '0.000000'

  --tuscan-cn-bin (string8) = Bin ranges in the form '[0,6)[6,10)[10,20)[20,50)[50,10000]' where [6,10) indicates that values in the 2nd bin must be &gt;= 6 and &lt; 10
    default value: '[0,6)[6,10)[10,20)[20,50)[50,10000]'

  --tuscan-cn-bin-values (vec_string8) = Array of BinnedCN value strings. Special value: "as-is-2" means report the TotalCN value to 2 decimals for the bin. A value is associated with a bin by index. The number of values must match the number of bins in tuscan-cn-bin
    default value: 'as-is-2,6-10,10-20,20-50,50+'

  --kernel-sigma-span (float64) = Probes spanned by one stdev.
    default value: '50.000000'

  --sig-snp-min-call-rate (float64) = Threshhold for minimum allowable call rate per channel for signature snp probesets
    default value: '0.600000'

  --sig-snp-min-compare-rate (float64) = Threshhold for minimum percentage of paired probesets with valid AT and GC calls
    default value: '0.650000'

  --sig-snp-min-concordance (float64) = Threshhold for minimum percentage of matching valid genotype calls
    default value: '0.600000'

  --sig-snp-min-strength-sketch-rank (float64) = Signal strength threshold under which signature snps will be considered a NoCall when calculating concordance
    default value: '0.305000'

  --sig-snp-report-filename (string8) = Name of optional output text file for signature snp report.

  --rm-ap-mod-step (int32) = The step to use in calculating allele peaks.
    default value: '20'

  --rm-ap-mod-window (int32) = The window to use in calculating allele peaks.
    default value: '100'

  --rm-ap-mod-point-count (int32) = The point count to use in calculating allele peaks.
    default value: '128'

  --rm-ap-mod-bandwidth (float64) = The bandwidth to use in calculating allele peaks.
    default value: '0.250000'

  --rm-ap-mod-cutoff (float64) = The cutoff to use in calculating allele peaks.
    default value: '0.050000'

  --rm-ap-mod-threshold (float64) = The clean threshold to use in calculating allele peaks.
    default value: '0.350000'

  --rm-ap-mod-height-threshold (float64) = The height threshold factor to use in calculating allele peaks.
    default value: '0.000000'

  --rm-ap-mod-height-threshold-bound (float64) = The bounds between which the height threshold is applied.
    default value: '1024.000000'

  --rm-ap-mod-symmetry (bool) = The symmetry value to use in calculating allele peaks.

  --rm-ap-mod-fld-weights (filename) = Name of file with FLD-derived weights.

  --ap-ad-step (int32) = The step to use in calculating allele peaks.
    default value: '20'
    Aliases: --ap-step

  --ap-ad-window (int32) = The window to use in calculating allele peaks.
    default value: '100'
    Aliases: --ap-window

  --ap-ad-point-count (int32) = The point count to use in calculating allele peaks.
    default value: '128'
    Aliases: --ap-point-count

  --ap-ad-bandwidth (float64) = The bandwidth to use in calculating allele peaks.
    default value: '0.250000'
    Aliases: --ap-bandwidth

  --ap-ad-cutoff (float64) = The cutoff to use in calculating allele peaks.
    default value: '0.050000'
    Aliases: --ap-cutoff

  --ap-ad-threshold (float64) = The clean threshold to use in calculating allele peaks.
    default value: '0.350000'
    Aliases: --ap-threshold

  --ap-ad-height-threshold (float64) = The height threshold factor to use in calculating allele peaks.
    default value: '0.000000'
    Aliases: --ap-height-threshold

  --ap-ad-height-threshold-bound (float64) = The bounds between which the height threshold is applied.
    default value: '1024.000000'
    Aliases: --ap-height-threshold-bound

  --ap-ad-symmetry (bool) = The symmetry value to use in calculating allele peaks.
    Aliases: --ap-symmetry

  --ap-ad-fld-weights (filename) = Name of file with FLD-derived weights.
    Aliases: --ap-fld-weights

  --loh-error-rate (float64) = The error rate to use in LOH processing.
    default value: '0.050000'

  --loh-beta (float64) = The beta to use in LOH processing.
    default value: '0.001000'

  --loh-alpha (float64) = The alpha to use in LOH processing.
    default value: '0.010000'

  --loh-separation (int32) = The separation to use in LOH processing.
    default value: '1000000'

  --loh-min-marker-count (int32) = The minimum marker count to use in LOH processing.
    default value: '10'

  --loh-no-call-threshold (float64) = The no call threshold to use in LOH processing.
    default value: '0.050000'

  --loh-min-genomic-span (int32) = The minimum genomic span to use in LOH processing.
    default value: '1000000'

  --loh-ss-thresh (float64) = Smooth Signal threshold above which call is het
    default value: '2.500000'

  --loh-ss-begin-slope (float64) = Smooth Signal value above which hom/het call is a linear function of BAF and Smooth Signal
    default value: '1.450000'

  --loh-baf-upper-thresh (float64) = BAF value above which call can be hom
    default value: '0.700000'

  --loh-baf-lower-thresh (float64) = BAF value below which call can be hom
    default value: '0.300000'

  --loh-slope-thresh (float64) = Slope determining ratio of BAF to Smooth Signal used to determine hom/het when Smooth Signal is greater than loh-ss-begin-slope
    default value: '0.163636'

  --loh-alpha-cn-calibrate (float64) = IDK
    default value: '0.564278'

  --loh-alpha-X-cn-calibrate (float64) = IDK
    default value: '0.619453'

  --loh-alpha-Y-cn-calibrate (float64) = IDK
    default value: '0.494620'

  --loh-beta-cn-calibrate (float64) = IDK
    default value: '1.000000'

  --loh-beta-X-cn-calibrate (float64) = IDK
    default value: '1.000000'

  --loh-beta-Y-cn-calibrate (float64) = IDK
    default value: '1.000000'

  --report-file (string8) = Report file name.

  --text-output (bool) = Output data in ASCII text format.

  --oschp-output (bool) = Output data in HDF5 format.
    Aliases: --hdf5-output

  --mangle-probeset-names (bool) = Prefix probe set names written to the CYCHP file with S- or C-

  --meta-data-info (vec_string8) = Meta data in key=value pair format that will be written to the header.

  --waviness-block-size (int32) = marker count
    default value: '50'

  --waviness-genomic-span (int32) = genomic segment length
    default value: '0'

  --alpha-cn-calibrate (float64) = IDK
    default value: '0.564278'

  --alpha-X-cn-calibrate (float64) = IDK
    default value: '0.619453'

  --alpha-Y-cn-calibrate (float64) = IDK
    default value: '0.494620'

  --beta-cn-calibrate (float64) = IDK
    default value: '1.000000'

  --beta-X-cn-calibrate (float64) = IDK
    default value: '1.000000'

  --beta-Y-cn-calibrate (float64) = IDK
    default value: '1.000000'

  --@help       = print developer help

</pre><p><a class="anchor" id="faq"></a> </p>
<h2>Frequently Asked Questions </h2>
<p><a class="anchor" id="references"></a> </p>
<h2>References </h2>
<ul>
<li>BRLMM-P - <a href="http://www.affymetrix.com/support/technical/whitepapers/brlmmp_whitepaper.pdf">http://www.affymetrix.com/support/technical/whitepapers/brlmmp_whitepaper.pdf</a></li>
<li>PLIER - <a href="http://media.affymetrix.com/support/technical/technotes/plier_technote.pdf">http://media.affymetrix.com/support/technical/technotes/plier_technote.pdf</a> </li>
</ul>
</div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated for Affymetrix Power Tools by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
