<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: MANUAL: apt-summary-genotype (1.16.1)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li class="current"><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>MANUAL: apt-summary-genotype (1.16.1) </h1>  </div>
</div>
<div class="contents">
<h2><a class="anchor" id="aptSummaryGenotype"></a>
Contents</h2>
<p><a class="anchor" id="options"></a> </p>
<h3><a class="anchor" id="manualOptions"></a>
Options:</h3>
<div class="fragment"><pre class="fragment">apt-summary-genotype - Utility program to feed an allele summary file directly into
a genotype calling Quant method. Currenlty on brlmm-p is supported.

usage:
   ./apt-summary-genotype -p CM=1.bins=100.K=2.SB=0.003.MS=0.05 \
       -o results \
       -s brlmm-p.plier.summary.txt \
       --read-models-brlmmp GenomeWideSNP_5.models \
       --chrX-snps GenomeWideSNP_5.chrx \
       --read-genders genders.txt


options:
 Common Options (not used by all programs)
   -h, --help                           Display program options and extra
                          documentation about possible analyses. See
                          -explain for information about a specific
                          operation. [default 'false'] 
   -v, --verbose How verbose to be with status messages 0 -
                          quiet, 1 - usual messages, 2 - more
                          messages. [default '1'] 
     --console-off Turn off the default messages to the 
                          console but not logging or sockets. 
                          [default 'false'] 
     --use-socket Host and port to print messages over in
                          localhost:port format [default ''] 
     --version Display version information. [default
                          'false'] 
   -f, --force Disable various checks including chip 
                          types. Consider using --chip-type option
                          rather than --force. [default 'false'] 
     --throw-exception Throw an exception rather than calling
                          exit() on error. Useful for debugging. This
                          option is intended for command line use
                          only. If you are wrapping an Engine and 
                          want exceptions thrown, then you should 
                          call Err::setThrowStatus(true) to ensure
                          that all Err::errAbort() calls result in an
                          exception. [default 'false'] 
     --analysis-files-path Search path for analysis library files. 
                          Will override AFFX_ANALYSIS_FILES_PATH
                          environment variable. [default ''] 
     --xml-file Input parameters in XML format (Will
                          override command line settings). [default
                          ''] 
     --temp-dir Directory for temporary files when working
                          off disk. Using network mounted drives is
                          not advised. When not set, the output 
                          folder will be used. The defaut is 
                          typically the output directory or the
                          current working directory. [default ''] 
   -o, --out-dir Directory for output files. Defaults to
                          current working directory. [default '.'] 
     --log-file The name of the log file. Generally 
                          defaults to the program name in the out-dir
                          folder. [default ''] 
 Engine Options (Not used on command line)
     --command-line The command line executed. [default ''] 
     --exec-guid The GUID for the process. [default ''] 
     --program-name The name of the program [default ''] 
     --program-company The company providing the program [default
                          ''] 
     --program-version The version of the program [default ''] 
     --program-cvs-id The CVS version of the program [default ''] 
     --version-to-report The version to report in the output files.
                          [default ''] 
     --free-mem-at-start How much physical memory was available when
                          the engine run started. [default '0'] 
     --meta-data-info Meta data in key=value pair that will be
                          output in headers. [default ''] 
   -p, --algo-params A string specifying algo parameters. See
                          --explain option for acceptable parameters.
                          [default ''] 
     --file5-summaries Interleaved Allele summaries file in file5
                          format with summaries in /summaries group.
                          [default ''] 
   -s, --summaries Interleaved Allele summaries file. [default
                          ''] 
     --summaries-a Allele summarie file for A allele. [default
                          ''] 
     --summaries-b Allele summarie file for B allele. [default
                          ''] 
     --file5-output Should output be formatted in file5 (hdf5)
                          output. [default 'false'] 
     --chrX-snps File containing snps on chrX
                          (non-pseudoautosomal region). [default ''] 
     --special-snps File containing all snps of unusual copy
                          (chrX,mito,Y) [default ''] 
     --special-sample-snps File containing all sample-specific snps of
                          unusual copy [default ''] 
     --write-models Should we write snp specific models out for
                          analysis? [experimental] [default 'false'] 
     --read-models-brlmmp File to read precomputed BRLMM-P snp
                          specific models from. [default ''] 
     --read-genders Explicitly read genders from a file.
                          [default ''] 
     --genotypes File to read seed genotypes from. [default
                          ''] 
     --select-probes Should we evaluate model fit for clusters 
                          by probe-set/probe[experimental] [default
                          'false'] 
     --set-analysis-name Explicitly set the analysis name. This
                          affects output file names (ie prefix) and
                          various meta info. [default ''] 
     --time-start The time the engine run was started 
                          [default ''] 
     --time-end The time the engine run ended [default ''] 
     --time-run-minutes The run time in minutes. [default ''] 
     --analysis-guid The GUID for the analysis run. [default ''] 
</pre></div><p><a class="anchor" id="genderFile"></a> </p>
<h2><a class="anchor" id="genders"></a>
Genders File</h2>
<p>The genders file is a TSV file (tab separated) with a header line followed by data lines. The header must contain a "cel_files" column and a "gender" column. The values in the "cel_files" column must match up with the headers in the allele summary file. Here is an example:</p>
<div class="fragment"><pre class="fragment">
cel_files       gender
NA06985_GW5_C.CEL       female
NA06991_GW5_C.CEL       female
NA06993_GW5_C.CEL       male
NA06994_GW5_C.CEL       male
NA07000_GW5_C.CEL       female
</pre></div><p><a class="anchor" id="alleleSummaryFile"></a> </p>
<h2><a class="anchor" id="summaries"></a>
Allele Summaries File</h2>
<p>The allele summaries file is a TSV file (tab separated) with a header line followed by data lines. The header must contain a "probeset_id" column in the first column followed by a column for each sample. There are two rows for each SNP. The first row is the allele summaries (estimates) for the A allele. The second row is the summaries for the B allele. This is further indicated by the suffix "-A" and "-B" to the probeset ID in the first column. Here is an example:</p>
<div class="fragment"><pre class="fragment">
probeset_id     NA06985_GW5_C.CEL       NA06991_GW5_C.CEL       NA06993_GW5_C.CEL
SNP_A-1780520-A 221.77864       229.52002       200.68078
SNP_A-1780520-B 808.69278       910.90927       732.87915
SNP_A-1780618-A 369.83269       723.96987       530.21800
SNP_A-1780618-B 825.04576       764.62659       644.88729
</pre></div><p><a class="anchor" id="faq"></a> </p>
<h2><a class="anchor" id="manFaq"></a>
Frequently Asked Questions</h2>
<p>Q. What is the minimum input?</p>
<p>A. You can run the program with only an allele summary file. In this senario all genders are assumed unknown (which BRLMM-P treats as female) and all SNPs are considered diploid. </p>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:58 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
