<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: MANUAL: apt-midas (1.16.1)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li class="current"><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>MANUAL: apt-midas (1.16.1) </h1>  </div>
</div>
<div class="contents">
<p><code>apt-midas</code> is a program for computing an alternative splicing score based on the MiDAS method described in the Affymetrix <a href="http://www.affymetrix.com/support/technical/whitepapers.affx">Alternative Transcript Analysis Methods for Exon Arrays</a> whitepaper.</p>
<h2><a class="anchor" id="QuickStart"></a>
Quick Start</h2>
<p>Most users will just want to compute MiDAS p-values using:</p>
<div class="fragment"><pre class="fragment">
apt-midas -c celfiles.txt -g gene-plier.summary.txt -e exon-plier.summary.txt -m meta-probeset.full.txt -o apt-midas-results
</pre></div><p>Where <code>-c</code> specifies a cel group file, <code>-g</code> specifies a gene level signal estimate file, <code>-e</code> specifies an exon level signal estimate file, <code>-m</code> specifies a meta-probeset file to match the exon level data with the gene level data, and <code>-o</code> specifies an output directory.</p>
<h3><a class="anchor" id="manualOptions"></a>
Options:</h3>
<div class="fragment"><pre class="fragment">midas - Microarray Detection of Alternative Splicing.
Usage:
   midas --cel-files cels.txt -g gene.summary.txt -e exon.summary.txt -m metaprobeset.txt \
            -o out -pvalues -fstats -normalized -stabilize 8.0 -no-log-transform

options:
   -h, --help                       [default 'false'] 
     --cel-files File defining mapping between cel_files and
                      experimental group_id. [default ''] 
   -g, --genedata File containing gene-level summary data.
                      [default ''] 
   -e, --exondata File containing exon-level summary data.
                      [default ''] 
   -m, --metaprobeset File containing mapping (meta-probeset file)
                      between exon probesets and gene level 
                      probesets. [default ''] 
   -o, --out-dir Directory which will contain the midas output.
                      [default ''] 
   -pv, --pvalues Output p values. [default 'true'] 
   -f, --fstats Output F statistics. [default 'false'] 
   -n, --normalized Output normalized exon intensities. [default
                      'false'] 
   -s, --stabilize Stabilization factor for log data transforms.
                      [default '8.0'] 
   -nol, --no-logtrans Do not log transform gene and exon level 
                      summary data. [default 'false'] 
     --version Display version information. [default 'false'] 
     --keep-path Keep cel file path. [default 'false'] 
</pre></div><h3><a class="anchor" id="CEL"></a>
Group File</h3>
<p>The CEL group file is used to define which results are to be grouped together for the MiDAS ANOVA test. The file is a text file with at least two columns: cel_files, group_id. The cel_files column should contain values that correspond to the headers in the summary files. The group_id is an arbitrary value where rows with the same group_id value are grouped together. Here is an example:</p>
<div class="fragment"><pre class="fragment">
cel_files       group_id
brain-rep1.cel  brain
brain-rep2.cel  brain
brain-rep3.cel  brain
heart-rep1.cel  heart
heart-rep2.cel  heart
heart-rep3.cel  heart
</pre></div><h2><a class="anchor" id="manFaq"></a>
Frequently Asked Questions</h2>
<p>Q. What meta-probeset file(s) should I use.</p>
<p>A. A good starting point is to use the core or extended meta probeset file along with the <a class="el" href="classIterPlier.html" title="Runs plier() interatively using the &quot;best&quot; features in each iteration, currently the featur...">IterPlier</a> method to generate gene level signal estimates. You can then use the full meta probeset file with MiDAS. In other words, use the more conservative meta probeset list to generate gene level estimates, but then compute MiDAS p-values for all exons associated with the transcript clusters in your gene level signal estimate file regardless of how speculative the content is.</p>
<p>Q. What is the Log Stabilization Factor For?</p>
<p>A. This factor (default 8) acts to stabilize the variance and helps to prevent inflated p-values for low or un-expressed probesets. The default is based on using PLIER signal summaries. You may want to use a lower value, or not log transform, if you are using <a class="el" href="classRMA.html" title="Holds methods for computing the popular Robust Muliarray Average of Microarrays (RMA).">RMA</a> signal summaries.</p>
<p>Q. When would I want to not log transform?</p>
<p>A. The default is to log transform the signal estimates. This works well for PLIER input which is not already log transformed. If you are using <a class="el" href="classRMA.html" title="Holds methods for computing the popular Robust Muliarray Average of Microarrays (RMA).">RMA</a> signal estimates as input, then you probably want to turn off the log transformation. The Log Stabilization Factor is ignored when log transformation is turned off.</p>
<p>Q. What are the normalized intensities?</p>
<p>A. These are the exon by sample fitted values (ie values for samples in the same group for the same exon are identical). The values are also scaled to a max of 0 by subtracting the largest observed fitted value for a given exon over all the sample groups.</p>
<p>Q. In MIDAS result are the p values after multiple test correction?</p>
<p>A. No. No multiple testing correction is applied. </p>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:59 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
