<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: CHANGE LOG</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li class="current"><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>CHANGE LOG </h1>  </div>
</div>
<div class="contents">
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.16.1 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Fri June 23, 2014</dd></dl>
<p>The 1.16.1 release of APT addresses performance issues with the Ps_Classification application when used with an optional ps2snp mapping file.</p>
<p>Details:</p>
<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/SNPOL-357">SNPOL-357</a>] - Ps_Classification performance improvementswhen running with ps2snp-file option</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/SNPOL-359">SNPOL-359</a>] - Ps_Classification - added warning if Polyploid=TRUE and HomRO=TRUE</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/SNPOL-358">SNPOL-358</a>] - Ps_Classification - changed column snpid to affy_snp_id</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1212">APT-1212</a>] - Added new vignette for SNP6 arrays using Axiom analysis</li>
</ul>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.16.0 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Fri May 9, 2014</dd></dl>
<p>The 1.16.0 release of APT includes the following highlights:</p>
<ul>
<li>Support for automatic recentering for samples with ploidy not equal to 2 in the OncoScan analysis pipeline.</li>
<li>Support for probabilities file output in apt-probeset-genotype.</li>
<li>Addition of the applications Ps_Classification and Ps_Metrics (ported from the SNP Polisher R package).</li>
<li>Addition of the apt2-dset-util applications for converting between OSCHP and other HDF5 based file formats and text files.</li>
<li>Various performance and other improvements, minor features and minor bug fixes.</li>
</ul>
<p>Details below:</p>
<h4>New features and bug fixes </h4>
<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1114">APT-1114</a>] - Minor documentation corrections from 1.15.0.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-414">APT-414</a>] - Updated documentation for pgf, clf and bgp files to clarify row major instead of column major is the implementation.</li>
</ul>
<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1038">APT-1038</a>] - apt-copynumber-cyto - no warnings or errors when running Ref Model file creation with missing probe_tab file.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1184">APT-1184</a>] - apt-copynumber-cyto - mosaicism analysis could create overlapping MosaicGain segments on very noisy samples.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1187">APT-1187</a>] - apt-copynumber-cyto - mosaicism algorithm creating no-call segments with non-neutral copynumber and mixture on very noisy samples.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1186">APT-1186</a>] - apt-copynumber-cyto - Added option to remove Mosaic Segment Calculation from the Y chomosome only for CytoScan 750k and HD arrays (default true).</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1174">APT-1174</a>] - apt-probeset-genotype - added new output file option: AxiomGT1.probabilities.txt.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1125">APT-1125</a>] - apt-probeset-summarize added options to turn off probe caching for platforms with many duplicate probes (which caused performance issues with EC2).</li>
</ul>
<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-334">OSC-334</a>] - apt-copynumber-onco-ssa - Fixed short, low marker-count LOH segments being reported on certain samples in regions with small number of markers with large separation.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-335">OSC-335</a>] - apt-copynumber-onco-ssa - Fixed breakpoints for PAR region in chromosome X.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-369">OSC-369</a>] - apt-copynunber-onco-ssa - Fixed non-default calibration parameters being ignored by LOH and Cychp Reporter nodes.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-359">OSC-359</a>] - apt-copynumber-onco-ssa - Automated recentering/adjustment of log2 rations if rounded ploidy not equal to 2.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-373">OSC-373</a>] - apt-copynumber-onco-ssa - Added reliability score for TuScan algorithm.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-371">OSC-371</a>] - apt-copynumber-onco-ssa - Added breakpoints at centromeres in TuScan segmenter.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-393">OSC-393</a>] - apt-copynumber-onco-ssa - Added tuscan-initial-ploidy to OSCHP header and Report file</li>
</ul>
<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APTNG-365">APTNG-365</a>] - APT2 framework - Removed some blank lines in console output</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APTNG-367">APTNG-367</a>] - APT2 framework - Updated help text for --arg-file option</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APTNG-374">APTNG-374</a>] - APT2 framework - Added checks for badly formatted int and float parameters.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APTNG-368">APTNG-368</a>] - APT2 framework - Indicate required parameters in help text.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APTNG-369">APTNG-369</a>] - APT2 framework - Set sockets parameters as (hidden) developer parameters.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APTNG-370">APTNG-370</a>] - APT2 framework - Improved error messages for unrecognized parameters.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APTNG-353">APTNG-353</a>] - APT2 framework - Applications output usage info if called with no or bad parameters.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APTNG-337">APTNG-337</a>] - APT2 framework - Dropped unnecessary _&amp;size columns from HDF5 output.</li>
</ul>
<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APTNG-354">APTNG-354</a>] - apt2-dset-util - Added to release.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/SNPOL-258">SNPOL-258</a>] - Ps_Metrics and Ps_Classification added to release.</li>
</ul>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.15.2 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Fri Oct 30, 2013</dd></dl>
<p>This release adds support for the OncoScan array platforms through the addition of the APT2 application apt-copynumber-onco-ssa, apt-copynumber-onco-ref, apt-copynumber-onco-som-ssa and apt-copynumber-onco-som-ref. (Note only the binrary distributions include the new applications: a separte source code distribution will be made available in order to compile the applications from source.)</p>
<h4>New Features and Bug Fixes</h4>
<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1177">APT-1177</a>] - Added OncoScan applications.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1125">APT-1125</a>] - Added option to turn on duplicate probe caching manually</li>
</ul>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.15.1 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Mon Feb 25, 2013</dd></dl>
<p>This is a minor release addressing the following features:</p>
<ul>
<li>Adding mosaicism analysis to the apt-copynumber-cyto pipeline for the CytoScan family of arrays.</li>
<li>Adding saturation reports to the apt-geno-qc application.</li>
</ul>
<h4>New Features and Bug Fixes</h4>
<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1002">APT-1002</a>] - Mosaicism for Cytoscan pipeline</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1132">APT-1132</a>] - Added saturation report to apt-geno-qc and apt-cel-extract</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1131">APT-1131</a>] - Use a background value of 0 for probes that have no bg bins when using pm-gcbg.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1139">APT-1139</a>] - <a class="el" href="classAdd.html">Add</a> project file for apt-copynumber-gender for inclusion in Windows releases</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-414">APT-414</a>] - Updated documentation for CLF file format to address row_major vs col_major issue.</li>
</ul>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.15.0 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Thu May 24, 2012</dd></dl>
<p>This release features a number of improvements and changes. Highlights include:</p>
<ul>
<li>Migrating Windows binary releases to Visual Studio 2010</li>
<li>Support in Windows for file paths longer than 240 characters</li>
<li>Various improvements in Linux and OS X build procedures</li>
<li>Reorganization of source code repositories and regression data to make it easier for users to compile and test APT code on different platforms</li>
<li>Improvements in internal testing and reliability</li>
<li>Various performance improvements and new features</li>
</ul>
<h4>Bug Fixes</h4>
<ul>
<li>General<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-932">APT-932</a>] - Fixed problemswith HDF5 libraries compiling from source on ubuntu</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-939">APT-939</a>] - SQLite files read locking caused fatal errors - implemented retry for locked data</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-943">APT-943</a>] - Added missing apt-summary-genotype HTML documentation</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-966">APT-966</a>] - Fixed bugs in <a class="el" href="classFsPath.html#ae3f48d4132630e272a31487d778c5863" title="tries to make the entire path. (like mkdir -p)">FsPath::mkdirPath</a> and <a class="el" href="classFsPath.html#a9e3a97d238754ffe09b27bd528ab5149" title="true if is a file and is in HDF5 format. /// bool osIsHdf5() const;">FsPath::mkdir</a> including errIfExists being ignored</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1072">APT-1072</a>] - Fixed issue with SVN version not being set in APT compiled from source code distribution</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1080">APT-1080</a>] - Removed references to win32 from readme files</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1081">APT-1081</a>] - Fixed makefile to allow running "make check" without prior make (to make it consistent with "make integration" and "make regression" targets</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1087">APT-1087</a>] - Modified configure script to allow re-running to change regression data location</li>
</ul>
</li>
<li>Testing<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-948">APT-948</a>] - Negative test failures in unit tests not being caught</li>
</ul>
</li>
<li>Cyto<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-953">APT-953</a>] - apt-copynumber-cyto: Added check to the high-pass filter to process only those probesets with process_flag non-zero n0</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-958">APT-958</a>] - apt-copynumber-cyto: analysis-genotype was not run when no analysis steps included in command line, causing segmentation fault (intended behavior is to run all steps)</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1031">APT-1031</a>] - apt-copynumber-cyto: Added indication in log file to show if "allelic difference modulation correction" is turned on</li>
</ul>
</li>
<li>Genotyping<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-627">APT-627</a>] - apt-probeset-genotype: Fix crash when inbred file doesn&rsquo;t list all the CEL file names for the genotype batch run</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-938">APT-938</a>] - apt-probeset-genotype: Check for wrong header in probeset-ids file not working</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-945">APT-945</a>] - apt-probeset-genotype: Crash if there are insufficient valid calls on probeset - added error message</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-979">APT-979</a>] - apt-probeset-summarize: Error running a <a class="el" href="classRMA.html" title="Holds methods for computing the popular Robust Muliarray Average of Microarrays (RMA).">RMA</a> with no MPS or PS file on a WT exon array</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-984">APT-984</a>] - apt-probset-summarize: Does not handle unexpected exceptions during normalization gracefully.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1065">APT-1065</a>] - apt-probeset-genotype: Analysis-spec string in output files is reporting incorrect values for <a class="el" href="classArtifactReduction.html" title="ArtifactReduction for doing normalization.">ArtifactReduction</a> if defaults are not used (correct string is used in analysis)</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1096">APT-1096</a>] - apt-probeset-genotype: Modified help text to indicate --sqlite-output option not available on Windows</li>
</ul>
</li>
<li>DMET<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-970">APT-970</a>] - apt-dmet-genotype: Same file used to store reported and analyzed probe sets - changed to use different file names</li>
</ul>
</li>
</ul>
<h4>Improvement and New Features </h4>
<ul>
<li>General<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-650">APT-650</a>] - The string "amd64" should be "x86_64" in our environmen.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-678">APT-678</a>] - Change the APT build system for Windows to MSBuild</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-679">APT-679</a>] - Upgrade APT Solution and projects for VS2010</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-784">APT-784</a>] - Clean up unused values and other gcc warnings.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-944">APT-944</a>] - apt-calvin-equivalent fails to identify NaN - number differences</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-947">APT-947</a>] - Edit documents to reflect unix only applications.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-987">APT-987</a>] - apt-file5-equivalent: <a class="el" href="classAdd.html">Add</a> "--sign-allow-negation" parameter for command line access to <a class="el" href="classaffx_1_1File5__File.html#a0a44055c962238474b5974d89d7bcc92">affx::File5_File::equivalent()</a> argument "bAllowNegation"</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-904">APT-904</a>], [<a href="http://jira.affymetrix.com:8080/browse/APT-908">APT-908</a>], [<a href="http://jira.affymetrix.com:8080/browse/APT-913">APT-913</a>] - Static code analysis: Unitialized variables, unreachable code and other fixes</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-973">APT-973</a>] - Reorganize SVN repository for APT 1.15</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1012">APT-1012</a>] - Adopt Klocwork static analysis for APT</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1060">APT-1060</a>] - Rename apt/sdk/apt2 directory to avoid confusion with upcoming APT2 release</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1019">APT-1019</a>] - apt-cel-transform: refactor to use APT <a class="el" href="classBaseEngine.html" title="Base class for analysis engines.">BaseEngine</a> class</li>
</ul>
</li>
<li>Testing/Regression Data<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-680">APT-680</a>] - Remove symbolic links in regression data</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-952">APT-952</a>] - Remove confusing output including references to FATAL ERROR in negative unit tests that pass</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-976">APT-976</a>] - Allow test results to be moved to different location (outside of source code tree)</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-988">APT-988</a>] - Missing output for some integration and regression tests fixed</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1009">APT-1009</a>] - Reorganize regression data to allow selective customer downloads based on applications used</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-832">APT-832</a>] - <a class="el" href="classAdd.html">Add</a> a regression testcase that exercises sockets via the <a class="el" href="classSocketEngine.html" title="SocketEngine design notes:">SocketEngine</a></li>
</ul>
</li>
<li>Cyto<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-490">APT-490</a>] - apt-copynumber-cyto: AllelePeaks dump no-mans land and FLD filtering intermediate data for testing</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-886">APT-886</a>] - apt-copynumber-cyto: <a class="el" href="classAdd.html">Add</a> more intermediate results to assist with software verification</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-997">APT-997</a>] - apt-copynumber-cyto: clarify ref model annotation file in messaging</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1048">APT-1048</a>] - apt-copynumber-cyto: Improve performance of covariate log2 ratio adjuster</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-534">APT-534</a>] - apt-copynumber-cyto: <a class="el" href="classAdd.html">Add</a> support for processing flag for probe sets for CytoScan* arrays</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-897">APT-897</a>] - apt-copynumber-gender: Application to check genders of CELs to determine valid to run reference model generation reference model generation.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1085">APT-1085</a>] - apt-copynumber-cyto: Set algorithm version in CYCHP/DXCHP file to APT version instead of ChAS version</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-942">APT-942</a>] - apt-copynumber-cyto: Messaging to display major steps as an algorithm run progress for ChAS</li>
</ul>
</li>
<li>Genotyping<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-937">APT-937</a>] - apt-probeset-genotype: <a class="el" href="classAdd.html">Add</a> error message for non-genotype probeset in probeset subset list</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-954">APT-954</a>] - apt-probeset-summarize: Added the ability to use the probe set name from the PGF file rather than the probe set id in the output of CHP files</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-955">APT-955</a>] - apt-probeset-summarize: Do not error in DABG when no GC bin data exists.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-956">APT-956</a>] - apt-probeset-summarize: <a class="el" href="classAdd.html">Add</a> ability to subtract a given amount from the degrees of freedom for the chi squared test in DABG (required for miRNA analysis)</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1051">APT-1051</a>] - apt-probeset-summarize: Include Probeset Names (not IDs) in output TXT/CHP file</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-482">APT-482</a>] - apt-probeset-genotype: Suppress "affymetrix-algorithm-param-chip-view-md5sum" parameter when probeset file is not used</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1005">APT-1005</a>] - Create vignette for miRNA analysis</li>
</ul>
</li>
<li>DMET<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-998">APT-998</a>] - apt-md5-check: Create tool to verify MD5 values for DMET Console output.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1006">APT-1006</a>], [<a href="http://jira.affymetrix.com:8080/browse/APT-1030">APT-1030</a>] - apt-metabolizer: <a class="el" href="classAdd.html">Add</a> apt-metabolizer to linux/os x/Windows builds</li>
</ul>
</li>
</ul>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.14.4 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Fri Jan 26, 2012</dd></dl>
<p>This release focusses on improvements to the apt-copynumber-cyto pipeline for processing of CytoScanHD and CytoScan750K arrays. Highlights include:</p>
<ul>
<li>Support for dropped probesets expected in na33 annotation libraries.</li>
<li>Covariate adjusters to adjust for modulation in Allele Peaks data in certain rare samples.</li>
<li>Improved performance.</li>
</ul>
<h4>Bug fixes and new features</h4>
<ul>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-960">APT-960</a>] - Improve confusing help text for apt-copynumber-cyto</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-965">APT-965</a>] - Some <a class="el" href="classCNProbeSet.html" title="A class to store probe set data.">CNProbeSet</a> state is passed to the next probe set when a bad annotation is found.</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-981">APT-981</a>] - Implement Allele Peaks covariate adjuster.</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-1025">APT-1025</a>] - <a class="el" href="classAdd.html">Add</a> vignette for CytoScan750K support</li>
</ul>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.14.3.1 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Tue Nov 1, 2011</dd></dl>
<p>This is a minor release in support of Axiom avian arrays. Added support for ZW gender calling (ZW female, ZZ male).</p>
<h4>Improvements and New Features</h4>
<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-535">APT-535</a>] - apt-probeset-genotype - added support for gender calling on avian arrays. Added options --chrZprobes, --chrWprobes and --zw-gender-calling.</li>
</ul>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.14.3 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Fri July 15, 2011</dd></dl>
<p>The following highlights the changes enhancements in this version of the software.</p>
<ul>
<li>Support for CytoScanHD array analysis using apt-copynumber-cyto</li>
<li>Various new and improved algorithms for CytoScanHD analysis</li>
<li>Various performance improvements</li>
<li>apt-midas was restored to the release package</li>
</ul>
<p>Details below:</p>
<h4>Bug fixes</h4>
<ul>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-345">APT-345</a>] - <a class="el" href="classCNAnalysisMethodGaussianSmooth.html" title="The log2 ration gaussian smooth analysis method.">CNAnalysisMethodGaussianSmooth</a> - improved error messages and added a check for the minimum value of the smooth_bw parameter</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-434">APT-434</a>] - apt-probeset-summarize - improved checks and error messages for .bgp files lacking probes with necessary representative GC counts</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-517">APT-517</a>] - apt-probeset-genotype - added tests to verify the file format of priors files</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-727">APT-727</a>] - apt-copynumber-cyto - improved performance for SSA processing for CytoScanHD CEL files</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-728">APT-728</a>] - apt-copynumber-cyto - added "uncomputed gender" boolean flag to cychp file header to indicate when no gender calling was enabled</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-822">APT-822</a>] - apt-copynumber-cyto - updated the calvin file writer to not include empty segments tables</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-871">APT-871</a>] - apt-copynumber-cyto - standardized the naming of text files created by geno-qc</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-887">APT-887</a>] - apt-copynumber-cyto - fixed a bug causing LOH segments that extended across long gaps in marker coverage to be appropriately identified to gap boundaries</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-888">APT-888</a>] - apt-copynumber-cyto - fixed a bug where the LOH Segments algorithm did not always utilize respecting the minGenomicSpan parameter resulting in some LOH segments being miscalled</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-930">APT-930</a>] - Updated -help text in apt-copynumber-cyto</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-932">APT-932</a>] - Fixed problems compiling hdf5 libraries from source on Ubuntu</li>
</ul>
<h4>Improvements and New Features</h4>
<ul>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-40">APT-40</a>] - docs - The file format document has been updated</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-587">APT-587</a>] - apt-cdf-export - added the "--file-md5" option to compute the md5 of an entire file.</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-645">APT-645</a>] - CNReference api - updated the module to support CytoScanHD</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-763">APT-763</a>] - apt-copynumber-cyto - added a log message indicating if snp-qc-snp-list could not be read</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-789">APT-789</a>] - apt-copynumber-cyto - changed the default behaviour to require modules to be invoked explicitly. The following modules must now be explicitly called to use them as shown before:<ul>
<li>local-gc-background-correction-reference-method pdnn-reference-method</li>
<li>wave-correction-reference-method wave-correction-reference-method</li>
<li>local-gc-background-intensity-adjustment-method pdnn-intensity-adjustment-method</li>
<li>image-correction-intensity-adjustment-method high-pass-filter-intensity-adjustment-method</li>
<li>wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method</li>
</ul>
</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-810">APT-810</a>] - apt-copynumber-cyto - improved performance of wave correction</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-817">APT-817</a>] - apt-copynumber-cyto - Genotyping and LOH analysis on the X chromosome are now based on CN state and not on the presence/absence of the Y chromosome for the CytoScanHD pipeline</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-858">APT-858</a>] - apt-copynumber-cyto - made the additional-waves-reference-method module selected-qc parameter more generic</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-933">APT-933</a>] - Improved readme guidance for windows builds</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-534">APT-534</a>] - apt-copynumber-cyto - support for processing flags for CytoScanHD array annotations file</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-733">APT-733</a>] - apt-copynumber-cyto - implemented Generic Signal Covariate Adjuster</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-734">APT-734</a>] - apt-copynumber-cyto - implemented Generic log2 ratio Covariate Adjuster</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-752">APT-752</a>] - apt-copynumber-cyto - implemented high pass filtering</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-773">APT-773</a>] - apt-copynumber-cyto - implemented gradient correction applicable for all chips with copy number probes</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-776">APT-776</a>] - apt-copynumber-cyto - added Computed Call Rate to Report file and CYCHP header</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-778">APT-778</a>] - apt-copynumber-cyto - the proportion of the autosomes that exhibit LOH is now in the CYCHP header</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-785">APT-785</a>] - apt-copynumber-cyto - added support for separate HMM parameters for Autosomes, ChrX and ChrY</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-809">APT-809</a>] - apt-copynumber-cyto - added support for external covariates file</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-843">APT-843</a>] - apt-copynumber-cyto - changed name of the log2ratio-adjustment-high-pass-filter to log2ratio-adjustment-method</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-825">APT-825</a>] - apt-copynumber-cyto - implemented wavelet shrinkage estimator (signal restoration) for log2 ratios</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-884">APT-884</a>] - apt-copynumber-workflow - implemented a check to ensure the minimum number of cel files were provided for reference file creation</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-912">APT-912</a>] - Removed obsolete applications apt-copynumber-analysis, apt-copynumber-reference, and apt-copynumber-log2ratio from release. Restored apt-midas.</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-923">APT-923</a>] - Incorporated changes in Fusion SDK into APT package</li>
<li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-928">APT-928</a>] - Removed apt-dmet-copynumber from 1.14.3 release</li>
</ul>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.14.2 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Fri March 18, 2011</dd></dl>
<p>The following highlights the changes enhancements in this version of the software.</p>
<ul>
<li>Support for the new CDF file format version described in the section on the Fusion SDK Version 1.11.1 above.</li>
<li>Support for various arrays as indicated above in the section on the new features included in GTC 4.1.</li>
<li>New call rate metrics calculation using autosomal SNPs only</li>
<li>Support for 3000 SNPQC probesets during genotyping</li>
<li>New artifact reduction support in the genotyping workflow to support Axiom arrays with new design</li>
<li>Performance improvements</li>
</ul>
<p>Details below:</p>
<h4>Improvements and New Features</h4>
<ul>
<li>Genotyping<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-228">APT-228</a>] - apt-geno-qc aborts improved error reporting for corrupted cel files.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-252">APT-252</a>] - apt-probeset-genotype - fix description of --disk-cache option</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-104">APT-104</a>] - apt-probeset-genotype - Report sample call rate, sample percentage of AB calls (het_rate) and sample percentage of AA/AB calls (hom_rate)</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-148">APT-148</a>] - apt-snp-summary - break out snp summary table calculations in GTC into stand-alone app.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-159">APT-159</a>] - apt-data-subset - allow sample multichannel cel files for Axiom testing.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-213">APT-213</a>] - apt-probeset-genotype - More compact data output for genotyping for A5 and txt formats.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-314">APT-314</a>] - apt-probeset-genotype - API for reading cluster location information from models files.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-297">APT-297</a>] - apt-probeset-summarize - Remove mas5 signal and detection.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-467">APT-467</a>] - apt-probeset-genotype - Implemented new QC and overall Call Rate metrics for Axiom Genotyping Runs.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-501">APT-501</a>] - apt-probeset-genotype - Implemented new Artifact Reduction algorithms.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-518">APT-518</a>] - apt-probeset-genotype - add option to strip off allele suffix from probeset id when writing feature effects file</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-535">APT-535</a>] - apt-probeset-genotype - add gender flip to handle avian arrays (Z/W chromosomes).`</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-568">APT-568</a>] - apt-geno-qc Implemented reagent version 1 and 2 discriminator.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-659">APT-659</a>] - Improved genotyping performance change from GTC 4.0 / APT 1.12 to GTC 4.1 / APT 1.14.2</li>
</ul>
</li>
</ul>
<ul>
<li>Cyto<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-158">APT-158</a>] - Cannot specify waviness-seg-count-cutoff=0 and use-high-waviness-seg-count=false for the same run.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-238">APT-238</a>] - Cyto enhancement: write SCARS even when info is 0</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-339">APT-339</a>] - Replace guid value from .Report.txt file of the apt-copynumber-cyto application by something that keeps the same value for gold and test runs or delete it.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-380">APT-380</a>] - Improve handling of output directory creation for Windows</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-150">APT-150</a>] - Change apt-copynumber-cyto wave correction code to default to the number of waves found in the reference if none defined on the command line. Also report the number of waves used in the cychp file.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-183">APT-183</a>] - <a class="el" href="classAdd.html">Add</a> median log2 ratio per chromosome to chromosome summary table in cychp files.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-237">APT-237</a>] - <a class="el" href="classAdd.html">Add</a> ability to configure cychp file name</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-250">APT-250</a>] - Support for new parameter for ref mode file creation in cyto that indicates if CEL files based on ATP or an manual process.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-336">APT-336</a>] - <a class="el" href="classAdd.html">Add</a> Meta-data support to Reference Model file creation and chp files for Cyto</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-353">APT-353</a>] - Creation details in model files (genotyping specifically)</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-180">APT-180</a>] - Adding waveSmooth analysis string option for cyto</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-267">APT-267</a>] - Cyto enhancement memory issue for more large numbers of cel files</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-309">APT-309</a>] - For APS/APG, when processing cel files, print the name rather than a dot to the log file.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-275">APT-275</a>] - Warning message change when number of waves is not specified</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-266">APT-266</a>] - Change apt implementation of SCAR values so that no more nan values for the 1498 SNP markers with Information Content = 0</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-636">APT-636</a>] - Check probeset lists for errors and report meaningful message if found</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-740">APT-740</a>] - apt-geno-qc handling of avian array QC (Z/W chromosomes).</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-344">APT-344</a>] - LOH parameters reported in cychp file.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-563">APT-563</a>] - for apt-copynumber-cyto need to be able to specify sketch size (override default) for SNP6 data where --cyto2 is false</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-182">APT-182</a>] - Cyto image correction to use single blocking</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-516">APT-516</a>] - Introduction of copy number calling module for CytoScanHD.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-528">APT-528</a>] - Enable Cyto HMM call for CytoScanHD arrays under apt-copynumber-cyto</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-529">APT-529</a>] - Gender calling issues for CytoScanHD array from apt-copynumber-cyto</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-530">APT-530</a>] - Enable confidence score calculation in apt-copynumber-cyto for CytoScanHD array</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-531">APT-531</a>] - Enable LOH result from the Chromosome summary table in apt-copynumber-cyto for CytoScanHD</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-544">APT-544</a>] - <a class="el" href="classAdd.html">Add</a> processed allelic difference to apt-workflow-cyto.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-652">APT-652</a>] - Implement new QC metric for SNP markers</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-775">APT-775</a>] - Implement a quantile normaliztion methodology which respects differences between autosomes and chromosome X/Y</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-546">APT-546</a>] - <a class="el" href="classAdd.html">Add</a> command line control of BRLMM-P parameters in cyto</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-656">APT-656</a>] - SNP6 CN performance improvement from GTC 4.0/APT 1.12 to GTC 4.1/APT Head</li>
</ul>
</li>
</ul>
<ul>
<li>General<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-55">APT-55</a>] - apt-cel-transform: work with multi channel cel files</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-56">APT-56</a>] - apt-cel-transform: generate new GUID and propagate old in headers</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-57">APT-57</a>] - apt-cel-transform: include command line in cel header</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-121">APT-121</a>] - Reduce disk usage by recycling intesity marts</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-122">APT-122</a>] - Fix <a class="el" href="classDiskIntensityMart.html" title="Very simple version of an IntensityMart based directly on cel files for comparison and troubleshootin...">DiskIntensityMart</a> to allow arbitrary data names.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-127">APT-127</a>] - Remove <a class="el" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a> from <a class="el" href="classAnalysisStream.html" title="Utility class for centralizing analysis pathway.">AnalysisStream</a> function calls.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-129">APT-129</a>] - Store probe level data outside of <a class="el" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a>.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-140">APT-140</a>] - Remove references to raw_intensity_mean and raw_intensity_stddev from reports.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-155">APT-155</a>] - Write only analysis probe intensities instead of all intensities for final diskmart</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-156">APT-156</a>] - Initialize diskmart with CEL order instead of layout order</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-193">APT-193</a>] - <a class="el" href="classAdd.html">Add</a> a version of <a class="el" href="classIntensityMart.html#a0436f7d382a3614185e85705e1999267" title="Return all of the intensities for given chipIx.">IntensityMart::getCelData</a> that accepts a channel designation</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-206">APT-206</a>] - <a class="el" href="classAdd.html">Add</a> apt-copynumber-correlationÂ a new app to generate a correlation matrix of log2 ratios using cychp files as input.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-216">APT-216</a>] - The --analysis-files-path search path is Unix-style colon-separated and it gets confused by Windows absolute paths.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-220">APT-220</a>] - Improve version reporting</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-224">APT-224</a>] - Do not include NaNs in correlation calculations for Calvin::equivalent(...) function.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-253">APT-253</a>] - Allow hints for a subset of CEL files/samples in the hints file</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-262">APT-262</a>] - apt-cel-extract error message when --target-sketch-file is given, but not --analysis</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-279">APT-279</a>] - Replace calls to quickSort and binarySearch by STL equivalents.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-350">APT-350</a>] - Allow brlmm-p posteriors to be used as priors without edits</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-358">APT-358</a>] - Configure - add example of setting sdk_output_top</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-10">APT-10</a>] - Allow use of sparse-mart (in-memory mart) with chipstream code</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-138">APT-138</a>] - File5 - <a class="el" href="classAdd.html">Add</a> char/short datatypes</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-198">APT-198</a>] - <a class="el" href="classSummaryGenotypeEngine.html">SummaryGenotypeEngine</a> can input and output file5 format</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-212">APT-212</a>] - Output summaries, calls and confidences in sqlite format for Simon's group</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-234">APT-234</a>] - Support for providing metadata to write into the CHP file header</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-245">APT-245</a>] - <a class="el" href="classAdd.html">Add</a> capability of ignoring "Sets" in <a class="el" href="classCalvin.html">Calvin</a> comparison</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-293">APT-293</a>] - Engines listen and prints on sockets for Windows integration</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-273">APT-273</a>] - Handle long filenames on windows</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-284">APT-284</a>] - apt-cel-transformer - add --cel-files option.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-360">APT-360</a>] - Consolidate file functions from <a class="el" href="classUtil.html" title="Util.">Util</a> into calvin_files FileUtils</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-686">APT-686</a>] - Use environment variables/macros for documentation version numbers, etc.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-436">APT-436</a>] - Converted all of APT to use <a class="el" href="classFsPath.html" title="A class for working with richer filepaths than just a string.">FsPath</a> libraries. (Consistent fix for MAX_PATH issues).</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-246">APT-246</a>] - Tsv support for shorts</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-688">APT-688</a>] - Added CDF GUID to the CHP header</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-685">APT-685</a>] - Remove in-place updates of source code from doxygen make process</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-696">APT-696</a>] - Fix formatting of html docs</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-697">APT-697</a>] - Move help docs out of .cpp files into separate .dox files</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-699">APT-699</a>] - Clean-up/simplify make dox task</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-458">APT-458</a>] - apt-cdf-to-spf - modify to handle multi-channel cdf input (CDF file format version 4)</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-485">APT-485</a>] - calvinlite should check for other celfile versions.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-731">APT-731</a>] - calvinlite should be able to select keys to print.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-482">APT-482</a>] - Added md5 values to CHP file headers.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-548">APT-548</a>] - Replaced comman and tab delimited file access with file/TsvFile. (Removed util/AffxFile).</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-672">APT-672</a>] - apt-annotation-converter added for all platforms.</li>
</ul>
</li>
</ul>
<h4>Bugs</h4>
<ul>
<li>Genotyping<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-124">APT-124</a>] - Genotype chp check reports errors with confidences match instead of when they don't match</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-178">APT-178</a>] - Canary parameter help info printed on stdout when it should be on stderr</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-202">APT-202</a>] - <a class="el" href="classCalvin.html">Calvin</a> GenericFileWriter generates empty error message if it cannot open the file.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-225">APT-225</a>] - Uncaught exception in apt-probeset-genotype if cel file does not exist.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-261">APT-261</a>] - apt-probeset-summarize unable to remove temp dir</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-370">APT-370</a>] - Application apt-probeset-summarize memory issue for PM only array.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-388">APT-388</a>] - apt-probset-summarize fails when running Adapter Type Normalization analysis method - Allowing annotation-file to be specified for apt-probeset-summarize.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-398">APT-398</a>] - apt-probeset-genotype - added a minimum value to CSepThr.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-393">APT-393</a>] - Using SNP list for genotyping produces correct txt files but CHP file does not contain calls only header</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-415">APT-415</a>] - QC performance improvements.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-488">APT-488</a>] - Error when insufficient space to run genotyping</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-512">APT-512</a>] - Rat array genotyping fails using APT windows (head and 1.12), ok in Linux (head and 1.12)</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-550">APT-550</a>] - Reading a poorly formed models file causes the application to abruptly exit</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-632">APT-632</a>] - Added error if user tries to specify both --read-genders and --set-gender-method on the commandline.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-664">APT-664</a>] - SNP6 copynumber - Gender column in 4.1 changed to "Unknown" for all CNCHP files</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-682">APT-682</a>] - apt-probeset-genotype fails to run for axiom array when library files are in E or F drive (rather than C)</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-714">APT-714</a>] - apt-geno-qc memory leak when reading cel files fixed.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-722">APT-722</a>] - Memory ios_base::failbit set error shows when using .cel files on network paths.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-799">APT-799</a>] - snpQC call rates show as percentage instead of fraction.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-207">APT-207</a>] - APT High Pass Filtering never runs when PDNN Background Correction is turned off</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-445">APT-445</a>] - SpectClust::findNLargestEvals always returns "false" for "converged" (even if did converge)</li>
</ul>
</li>
</ul>
<ul>
<li>Cyto<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-154">APT-154</a>] - <a class="el" href="classCNReferenceMethodWaveCorrection.html" title="A class for calulating WaveCorrection.">CNReferenceMethodWaveCorrection</a> opens reference files as read/write when should be read only</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-162">APT-162</a>] - Cyto legacy bug fixes: util/AffyConv<a class="el" href="AffxConv_8h.html#a1f58abdd19fc60c5e0d100d9595adfc2" title="Converts a string with commas into a double.">getDouble</a>(...) - Rounding errors on floating point output.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-164">APT-164</a>] - Cyto legacy bug fixes: CNAnalysisMethodChipstream::getSampleId(...) - No longer parsing "_NA" from sample name.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-165">APT-165</a>] - Cyto legacy bug fixes: <a class="el" href="classCNAnalysisMethodChipstream.html" title="The Chipstream Analysis Method.">CNAnalysisMethodChipstream</a> Missing Last Genotype Call Override Sample</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-166">APT-166</a>] - Cyto legacy bug fixes: <a class="el" href="classCNAnalysisMethodLog2Ratio.html" title="The Log2Ratio analysis method.">CNAnalysisMethodLog2Ratio</a> Removed return when <a class="el" href="classMedian.html">Median</a> == 0. (Very rarely occuring bug, seen in some cancer samples only).</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-167">APT-167</a>] - Cyto legacy bug fixes: <a class="el" href="classCNAnalysisMethodReference.html" title="The Reference Analysis Method.">CNAnalysisMethodReference</a> PDNN <a class="el" href="classMedian.html">Median</a> Intensity</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-169">APT-169</a>] - Cyto legacy bug fixes: <a class="el" href="classCNLog2RatioAdjustmentMethodWaveCorrection.html" title="The WaveCorrection log2 ratio adjustment method.">CNLog2RatioAdjustmentMethodWaveCorrection</a> - Application always uses the maximum number of waves in CN reference regardless of what user specifies for wave-count.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-170">APT-170</a>] - Cyto legacy bug fixes: <a class="el" href="classCNProbeSet.html" title="A class to store probe set data.">CNProbeSet</a> Underflow in GC Binning.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-171">APT-171</a>] - Cyto legacy bug fixes: <a class="el" href="classCNReferenceMethodWaveCorrection.html" title="A class for calulating WaveCorrection.">CNReferenceMethodWaveCorrection</a>. Changed order of processing from <a class="el" href="classProbe.html" title="Represents an individual feature on a chip.">Probe</a> Set Name to Genomic Order.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-172">APT-172</a>] - Cyto legacy bug fixes: <a class="el" href="classCNSmoother.html">CNSmoother</a> Module - remove short segments in CN run time encoding.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-175">APT-175</a>] - CEL file index numbers are being printed to report files instead of CEL file names</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-241">APT-241</a>] - Cyto performance improvements.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-215">APT-215</a>] - Parameter --analysis-files-path is being ignored for the --snp-qc-snp-list file in certain cases</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-276">APT-276</a>] - Cyto CNAnalysis for reference model file uses wrong extension if not set in job file</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-294">APT-294</a>] - Improved CN/HMM</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-295">APT-295</a>] - Improve AllelePeaks detection</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-299">APT-299</a>] - Improve LOH calling for chromosomes 19 and 22</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-321">APT-321</a>] - Change in Behavior for --cels Argument: errAbort <a class="el" href="classCNCytoEngine.html" title="The engine for controlling the copynumber Cytos.">CNCytoEngine</a> if CEL files are specified both as command line arguments and cel-file or cels list(s).</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-354">APT-354</a>] - Added probe-file option to the analysis-files-path prepend list in <a class="el" href="classCNCytoEngine.html" title="The engine for controlling the copynumber Cytos.">CNCytoEngine</a></li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-391">APT-391</a>] - CN/SNP processing flags need to be handled properly - handling of flag 2.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-419">APT-419</a>] - apt-copynumber-cyto --explain unexpected should throw warning message.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-456">APT-456</a>] - Marker Names Unavailable for SCAR Values from APT - now output all markers used in the reference creation to the <a class="el" href="classProbeSet.html" title="Collection of atoms that measure target that should be absent or present at the same time...">ProbeSet</a> section of the cychp file.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-466">APT-466</a>] - apt-calvinlite-util does not display values of boolean header parameters, only an empty string.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-489">APT-489</a>] - Allele peaks data differences between Cyto SSA runs</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-502">APT-502</a>] - Marker count of Mosaicism counts the SNPs that do not have a log2ratio</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-519">APT-519</a>] - apt-copynumber-cyto vs workflow Chr X LOH differences for Female samples</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-520">APT-520</a>] - apt-copynumber-cyto vs workflow extra records in reference feature_response and snp_posteriors table</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-522">APT-522</a>] - Numerical differences between estimated waves from apt-copynumber-workflow and apt-copynumber-cyto</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-523">APT-523</a>] - AGR probe id was not generated in CN reference build for SNP6 data using apt-copynumber-cyto</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-525">APT-525</a>] - LOH calls problem comparing results from apt-copynumber-cyto and apt-copynumber-workflow</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-540">APT-540</a>] - Scrambled CNReferenceHeader Header Values in Cyto and Workflow</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-541">APT-541</a>] - cyto generates single-sample mode output with rounding differences vs workflow when wave correction enabled</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-615">APT-615</a>] - apt-copynumber-cyto AllelePeaks algorithm 6X duplicate rows in PekSets dataset</li>
</ul>
</li>
</ul>
<ul>
<li>General<ul>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-142">APT-142</a>] - File5 - size not correct after write_array.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-260">APT-260</a>] - apt-chp-to-txt documentation bug</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-281">APT-281</a>] - apt-file5-util: no help reported when no args on command line</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-337">APT-337</a>] - <a class="el" href="classUtil.html#abd09dcbcbe5d0dc6714bf008005fe0b7" title="Determine the free and total amount of memory in bytes on this machine.">Util::memInfo</a> reports swapAvail=0 on Darwin OS-X - Documented.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-363">APT-363</a>] - CHP file headers slow to read - improved.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-369">APT-369</a>] - Fix bug in apt-mas5 where .cel was included in output chp file name</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-374">APT-374</a>] - <a class="el" href="classTmpFileFactory.html" title="Allocate temporary filenames and arrange to remove them when done.">TmpFileFactory</a> ignoring user-defined file pre/suffixes</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-510">APT-510</a>] - APT will fail if batch name has "=" sign</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-612">APT-612</a>] - apt-calvinlite-util unable read short .chp file with missing data which <a class="el" href="classCalvin.html">Calvin</a> Viewer can read</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-634">APT-634</a>] - Fixed Windows 260 character MAX_PATH limit. Consolidated all file access methods and routines in util/Util and util/AffxFile respectively to util/Fs to support windows MAX_PATH.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-669">APT-669</a>] - Function SnpModelConverter::getTsvArrayType fails to get chip type of posterior models - added substring matching to check chip types.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-674">APT-674</a>] - Added exception handling for SQLite libraries.</li>
<li>[<a href="http://jira.affymetrix.com:8080/browse/APT-701">APT-701</a>] - <a class="el" href="classFsPath.html" title="A class for working with richer filepaths than just a string.">FsPath</a> constructor doesn't set internal path name.</li>
</ul>
</li>
</ul>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.12.a </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Tue Sept 23 2010<ul>
<li>Incremental release containing apt-geno-qc and apt-probeset-genotype only.</li>
</ul>
</dd></dl>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.12.0 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Monday Oct 12 2009<ul>
<li>Full Axiom &trade; support</li>
<li>Added support for multi channel cel files to apt-cel-extract</li>
</ul>
</dd></dl>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.11.0 (Internal Release Only) </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Sat Aug 29 2009<ul>
<li><b>Fixed bug in Windows x64 binary release where executables depended on debug runtime libraries</b></li>
<li><b>Fixed a bug in AdapterTypeNormalization where transformed intensities were not passed through to next chipstream when initializing.</b></li>
<li><b>Partial release</b> to support QC and genotype calling on the Axiom &trade; platform. We recomend Staying with 1.10.2 for other applications of APT until the next full release (1.12.0).</li>
<li><b>Fixed a problem with the segments in a CYCHP file. The segment_id should be a UINT, not a STRING.</b></li>
<li>Cytogenetics Genome Wide Array Support</li>
<li>Added apt-summary-genotype to windows binary releases</li>
<li>Added apt-mas5</li>
<li>Added support for SQLite annotation files to apt-copynumber-workflow</li>
<li>Added artifact reduction chipstream node.</li>
<li>Inbred tweaks to apt-probeset-genotype</li>
<li>Added waviness amplitude metric to cychp summary metrics</li>
<li>Added segments for CNNeutralLOH and NormalDiploid. [experimental]</li>
<li>Added antigenomic ratio to <a class="el" href="classCNExperiment.html" title="A class for storing CNExperiment data.">CNExperiment</a> [experimental]</li>
<li>Added Mosaicism. [experimental]</li>
<li>enable wave correction for SNP6 [experimental]</li>
<li>refactor how feature effects are handled. Now match by probe ID AND probeset ID</li>
<li>Switch Windows builds to use VS2008</li>
<li>Added XML output options for unit and regression tests</li>
<li>Added support for multiple intensity channels (eg multi color CEL files)</li>
<li>Refactor <a class="el" href="classBaseEngine.html" title="Base class for analysis engines.">BaseEngine</a> and handling of option and state.</li>
<li>Commandline programs which use a <a class="el" href="classBaseEngine.html" title="Base class for analysis engines.">BaseEngine</a> derived analysis, now accept a set of shared options</li>
<li>remove use of tmp chp file when outputing CHP files</li>
<li>removed in-memory mode for apt-probeset-genotype and apt-probeset-summarize</li>
<li>reconcilled differences in same options between apt-probeset-genotype and apt-probeset-summarize</li>
</ul>
</dd></dl>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.10.2 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Tue Jan 27 2009<ul>
<li>Added support for DMET Plus:<ul>
<li>apt-dmet-genotype will generate DMET CHP files from DMET CEL files. The CHP files contain genotyping and copynumber results.</li>
<li>apt-dmet-translation will generate DMET star allele translation reports from CHP files.</li>
</ul>
</li>
<li>Fix bug in apt-probeset-summarize --probeset-ids option when processing genotype probesets using the expr.genotype=true modifier in the analysis string.</li>
<li>Improve error handling when required columns in TSV files are missing.</li>
<li>Fix CNCHP header so that files can be loaded into GTC 3.x</li>
<li>Fixed problem in CN reference file generation so that APT generated reference files are backwards compatabile with GTC 3.x</li>
<li>reconcilled differences in same options between apt-probeset-genotype and apt-probeset-summarize</li>
</ul>
</dd></dl>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.10.2 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Tue Jan 27 2009<ul>
<li>Added support for DMET Plus:<ul>
<li>apt-dmet-genotype will generate DMET CHP files from DMET CEL files. The CHP files contain genotyping and copynumber results.</li>
<li>apt-dmet-translation will generate DMET star allele translation reports from CHP files.</li>
</ul>
</li>
<li>Fix bug in apt-probeset-summarize --probeset-ids option when processing genotype probesets using the expr.genotype=true modifier in the analysis string.</li>
<li>Improve error handling when required columns in TSV files are missing.</li>
<li>Fix CNCHP header so that files can be loaded into GTC 3.x</li>
<li>Fixed problem in CN reference file generation so that APT generated reference files are backwards compatabile with GTC 3.x</li>
<li>Fix quantile normalization code so that values outside a predefined target sketch are capped. Without this fix infinite values (and negative values) were possible.</li>
<li>Fix makeDir error in apt-geno-qc when output file specification (from --out-dir option) did not include a directory portion.</li>
<li>Accomidate genotype probesets which do not have equal numbers of allele A and B probes.</li>
<li>In apt-cel-extract, fix the ordering of probes in the on-disk intensity mart. Previous version was very slow due to non-optimal ordering of probe intensities on disk. Related to this fix is that the output order is different when using a probeset list. (It is now in the same order as the provided probeset list).</li>
<li>Fixed a bug where target sketches provided to apt-cel-transformer where assumed to be in a5 format.</li>
<li>Fixed a bug where apt-probeset-summarize would fail if an entire probeset was removed via the probe kill list when using a CDF file.</li>
<li>Added utility apt-engine-wrapper which allows one to directly call analysis engines and query for what options an engine will accept.</li>
<li>Added apt-file5-util which allows one to convert between a5 and text/tsv file formats.</li>
<li>Drop binaries for Solaris/SPARC</li>
<li>Change in OS-X binaries:<ul>
<li>dropped "un-tested" universal binaries</li>
<li>providing stand alone 64-bit PowerPC binaries</li>
<li>providing stand alone 64-bit Intel binaries</li>
</ul>
</li>
<li>Changed APT guids to be RFC-4122 format guids</li>
</ul>
</dd></dl>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.10.1 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Mon Nov 3 2008<ul>
<li><b>Fixed bug in MAPD calculation by copynumber workflow</b>. In 1.10.0, the MAPD values were computed before putting probesets in genome order. The implication is that MAPDs reported by 1.10.0 were inflated relative to the real MAPD values.</li>
<li><a class="el" href="classAdd.html">Add</a> genome GC content correction to copynumber workflow.</li>
<li><a class="el" href="classAdd.html">Add</a> a log file to apt-summary-genotype</li>
<li><a class="el" href="classAdd.html">Add</a> on-disk mode to apt-cel-extract (on by default)</li>
<li><a class="el" href="classAdd.html">Add</a> progress meters for chip layout file reading (ie CDF)</li>
<li>Fixed bug where block size and memory check code was still being run even when on-disk mode was set. The implication in 1.10.0 was that multiple iterations (library file reads) were being done unnecessarily. Results were still correct.</li>
<li><a class="el" href="classAdd.html">Add</a> copynumber binaries to windows installer (missing in 1.10.0 release)</li>
<li>Changed the code to only read lib files only once when doing one iteration in apt-probeset-genotype and apt-probeset-summarize.</li>
<li>Clean up of a5 options for apt-probeset-genotype</li>
<li><a class="el" href="classAdd.html">Add</a> a5 options to apt-probeset-summarize</li>
<li>Refactored the analysis engines to inherit from a common engine base class.</li>
</ul>
</dd></dl>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.10.0 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Wed Jul 9 2008<ul>
<li><b>Fixed bug where pm-sum method was reporting values equivalent to pm-only when using SPF file rather than a CDF file. This is in the context of using apt-probeset-summarize with genotyping array via the "expr" addition to the analysis string.</b></li>
<li><b>Fixed bug where incorrect "x" coordinate was reported in feature effect output files from apt-probeset-summarize and apt-probeset-genotype (all releases) and in apt-cel-extract (releases newer than 1.8.0) output. Incorrect values only show up when working with arrays where the number of rows and columns in the CEL file are not equal.</b></li>
<li><b>Fixed bug in PLIER where feature effects were updated after the last update to target responses</b></li>
<li>Added copynumber functionality for SNP6 (see apt-copynumber-workflow).</li>
<li>Added Canary copynumber calling for SNP6 (see apt-canary).</li>
<li>Complete refactor of apt-chp-to-txt. Now handles most chp file formats and types. Resequencing and Tag chp files are not handled by this tool.</li>
<li>Added Vignettes:<ul>
<li>SNP6 Copynumber analysis using individual components</li>
<li>Single sample genotyping (ie for QC purposes)</li>
</ul>
</li>
<li>Removed apt-cel-to-pq. Supplanted by apt-cel-extract.</li>
<li>Misc<ul>
<li>Added experimental HDF5 output file formats to apt-probeset-genotype</li>
<li>Added on-disk intensity mart. Set as the default for apt-probeset-summarize and apt-probeset-genotype.</li>
<li>Fixed memory capping on x64 windows and 64-bit OS-X.</li>
<li>Added the ability to apply pre-computed feature effects when using median polish in apt-probeset-summarize and apt-probeset-genotype.</li>
<li>Added an adaptor type normalization (used by copynumber workflow)</li>
<li>apt-probeset-genotype will now report signal estimates for copy number probesets when the --all-types and --summaries option is given</li>
</ul>
</li>
</ul>
</dd></dl>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.8.6 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Mon Feb 3 2008<ul>
<li><b>Fix bug in <a class="el" href="classIterPlier.html" title="Runs plier() interatively using the &quot;best&quot; features in each iteration, currently the featur...">IterPlier</a> where one would get a segfault if the number of chips exceeded 100.</b></li>
<li><b>Change from float to double in <a class="el" href="classQuantBirdseed.html">QuantBirdseed</a> methods to improve cross platform portability. This may result in slight differences in the confidence value and in rare cases a change between making a genotype call versus a no-call.</b></li>
<li><b>order of results in apt-probeset-genotype report file have changed due to some refactoring of the code.</b></li>
<li>added birdseed-v2</li>
<li>added ContrastQC qc metric to apt-geno-qc</li>
<li>fix memory estimation (don't cap at 2G on Windows (x64) and OS-X (ppc64, x86_64)</li>
<li>apt-cel-transformer: remove ability to change formats. add ability to handle agcc files</li>
<li>Improvements to command line option handling</li>
<li>Allow med-norm to be followed by other chip stream components when using a fixed target</li>
<li>added a target and doavg options to quant-norm</li>
<li>fix bug in --select-probes option of apt-probeset-genotype where probe IDs were not reported correctly</li>
</ul>
</dd></dl>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.8.5 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Wed Oct 31 2007</dd></dl>
<ul>
<li><b>Change to the DM Implementation</b> DM calls/confidences can be reported by apt-geno-qc and are generated and used by apt-probeset-genotype to generate seed calls for BRLMM on 500k data. Differences were observed between different platforms (particularly 64-bit vs 32-bit) in a small number of SNPs (ie 0-5 SNPs out of 250,000 on 500K Sty Chip). Code changes were made which rectify this, however you may see some some changes in DM results because of this.</li>
</ul>
<ul>
<li>Binary Releases<ul>
<li>Added 64-bit Windows Binaries (x64)</li>
</ul>
</li>
</ul>
<ul>
<li>New documentation<ul>
<li>Vignettes<ul>
<li>Masking Out Probes in apt-probeset-summarize</li>
<li>Use of APT to Analyze WT Based Exon and Gene Expression Arrays</li>
</ul>
</li>
<li>Added new top level pages and new navigation for end user (binary release) documentation<ul>
<li>Change Log</li>
<li>Vignettes</li>
<li>FAQs</li>
<li>License</li>
<li>File Formats</li>
<li>Platforms</li>
</ul>
</li>
<li>Updated Manuals<ul>
<li>Added new FAQ items to application specific manuals</li>
</ul>
</li>
</ul>
</li>
</ul>
<ul>
<li>New Applications:<ul>
<li>apt-cel-convert: a tool for converting between various CEL file formats.</li>
</ul>
</li>
</ul>
<ul>
<li>Changed Applications:<ul>
<li>Most Applications<ul>
<li>log messages on stderr rather than stdout</li>
</ul>
</li>
<li>apt-cel-transformer:<ul>
<li>support PGF/CLF and SPF in addition to CDF</li>
</ul>
</li>
<li>apt-cel-to-pq:<ul>
<li>support AGCC CEL files</li>
</ul>
</li>
<li>apt-cel-extract:<ul>
<li>added support for AGCC CEL files</li>
<li>complete rewrite of this application to use the core APT <a class="el" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a>, <a class="el" href="classChipStream.html" title="ChipStream - base class for objects that can initialize themselves one at a time and transform intens...">ChipStream</a>, and <a class="el" href="classSparseMart.html" title="SparseMart Class for representing only a subset of cel file intensity data in memory.">SparseMart</a> framework. As a result this is really a new tool as far as command line arguements and behavior of the application are concerned.</li>
<li>the tool can now scale to a large number of cel files (although you will need to set the --block-size manually)</li>
<li>block number is reported (critical for assigning genotype probes to alleles)</li>
<li>you can provide an analysis string (including pm-adjustor) to manipulate probe intensities</li>
<li>you can report just the pm values, the pm and background (ie mm, gcbg, ...) values together (ie 132,22), or the result of the pm minus background</li>
<li>you can provide both a probeset list and probe list (intersection will be extracted)</li>
<li>probeset and probe lists now work with CDF files</li>
<li>the probeset type is reported (useful for multi-type CDF files like SNP6)</li>
<li>the tool will work without any library files (with reduced functionality of course)</li>
<li>enable use of experimental spf library files</li>
</ul>
</li>
<li>apt-geno-qc:<ul>
<li>added --chrX-probe and --chrY-probe options which enable CN probe gender calling</li>
<li>added --probe-class-file option which will turn on the <a class="el" href="classCelStatListener.html" title="Class for objects that need to actually access real cel file data.">CelStatListener</a> and report probe intensity means by group</li>
<li>new column headers -- consistency with apt-probeset-genotype report.</li>
<li>new ordering of columns -- this will continue to vary over time. You should rely on column labels rather than column position to parse out specific information. Same holds true for apt-probeset-genotype report file</li>
<li>dropped --rpt/-r option, (replaced with --dm-out option)</li>
<li>Added --dm-out, --dm-thresh, and --dm-het-mult options for enabling DM calls and confidence output and setting DM options. Note that the output is now in TSV format.</li>
<li>complete refactor of the source code</li>
</ul>
</li>
<li>apt-probeset-summarize:<ul>
<li><b>Fixed a bug in MAS5 background correction routine when using pgf file.</b> In short, the number of probe sets was set to 0 when using a pgf file.</li>
<li>quality assessment metrics and QC probeset values are now written to the AGCC CHP file headers</li>
<li>probesets which failed to be quantified (ie certain controls) are no longer written to the AGCC CHP file. In previous versions such probesets would typically show up with a 0.0 quantification value.</li>
<li>changes to how the md5sum (gc background probes and normalization probes) is computed to ensure cross platform consistency</li>
<li>syncronized meta header data between apt-probeset-genotype and apt-probeset summarize. IE, there will be some changes to the meta info going into CHP files and other output files.</li>
<li>added support for SPF files</li>
<li>fixed bug where mas50-bg chipstream failed if there were additional chip streams after it</li>
<li>added additional boundary conditions in <a class="el" href="classRMA.html#a6ca4697cbf7e4e48fa7a3ee3e2e7f115" title="Implementation of the bw.nrd0() function in R to decide kernel bandwidth.">RMA::findBandWidth</a> in order to match behavior of BioConductor. May fix errors with certain chips with low overall intensity.</li>
<li>previous behavior was to not output mm_* quality metrics when bgrd probes were provided (via --bgp option). Now mm_* is reported if there are more than 5000 MM probes listed in the library file(s). Note that for the Human, Mouse, and Rat Exon 1.0 ST arrays, this will be dominated by the "genomic background" probes which are flagged as MM probes in the PGF file.</li>
<li>removed the path from the cel file names listed in the report file.</li>
<li>improved masking of probes (still experimental)<ul>
<li>--rm-probes to --kill-list (sync w/ apt-probeset-genotype)</li>
<li>take into account probeset when masking probes -- ie you can now mask out a probe that is in more than one probeset</li>
<li>masking file can have probe_id (1 base) or x/y position. if both are present it will check to see that they are consistent.</li>
<li>when probesets and meta probesets are completely masked out, a warning will be issued rather than having the software exit with an error message.</li>
</ul>
</li>
<li>added --mem-usage option to set memory use in MB as alternative to setting block-size</li>
<li>added --chip-type option. can be specified multiple times. This option will dictate the chip type in the output files and it will also specify what chip types the cel and lib files are checked against. Note that this will override the typcial check between chip types in lib files versus those in the cel files.</li>
<li>When doing CDF based analysis, chip types based on substrings ending with "." are also considered a match. IE, CDF file foo.bar.v1.r10.cdf will be detected as matching the possible chip types foo.bar.v1.r10, foo.bar.v1, foo.bar, and foo.</li>
<li>new --cc-md-chp-output option will generate experimental AGCC Multi Data CHP files.</li>
<li>added a --set-analysis-name option which will override the name computed from the analysis string. This option will affect the prefix used for output file names and the labels in certain meta fields and/or report file metrics.</li>
</ul>
</li>
<li>apt-probeset-genotype:<ul>
<li><b>fixed bug where chip-type in CHP file header was not set when using --force option</b></li>
<li><b>removed --txt-output option, which had a bug where calls were not thresholded in text output file for birdseed</b></li>
<li><b>fixed bug on solaris where use of birdseed method would cause a segfault</b></li>
<li>Added documentation on --write-models option of apt-probeset-genotype</li>
<li>fix reported analysis spec for all genotype calling methods. The genotype calling method (Quant Method) was not reported and in some cases incorrect algorithm parameters were reported.</li>
<li>for BRLMM-P, cluster_distance_mean metric is now correctly reported. Previously it was just set to zero.</li>
<li>fixed bug where -s option with --cc-chp-output would fail</li>
<li>added a number of additional metrics to the report <a href="file:">file:</a><ul>
<li>DM call rate (when DM listener is run -- ie BRLMM)</li>
<li>Various gender calls based on methods which are suitable to the chips/analysis being run</li>
<li>Various metrics underlying the gender calls</li>
</ul>
</li>
<li>added a --probe-class-file option which will read in a file grouping probes into classes for use by the <a class="el" href="classCelStatListener.html" title="Class for objects that need to actually access real cel file data.">CelStatListener</a>.</li>
<li>added --read-genders option to read genders from a file.</li>
<li>implement --set-gender-method option to force use of a particular gender method. Current options are:<ul>
<li>"cn-probe-chrXY-ratio"</li>
<li>"dm-chrX-het-rate"</li>
<li>"em-cluster-chrX-het-contrast"</li>
<li>"user-supplied"</li>
<li>"none"</li>
<li>"supplied-genotypes-chrX-het-rate"</li>
</ul>
</li>
<li>change chrXForce to --no-gender-force: ie run even if no suitable gender method is available</li>
<li>removed labelz (brlmm-p) KXY option which did nothing.</li>
<li>added birdseed-dev genotype calling method. "birdseed-v1" (alias for full analysis, and quant method name) will return the original birdseed implementation. "birdseed-dev" (alias for full analysis, and quant method name) will return the latest integrated method for birdseed from the Broad. This implementation will change over time. "birdseed" (alias for full analysis, and quant method name) will return the original birdseed-v1 implementation.</li>
<li>An option, --set-analysis-name, will be added to apt-probeset-genotype. This will allow one to set the analysis name irregardless of what analysis method is used. IE, "--set-analysis-name %birdseed" would allow birdseed-dev to masquerade as birdseed. Note that the analysis spec field would still enumerate the exact method used.</li>
<li>birdseed-dev has been updated to reflect birdseed version 2.6 from the Broad.</li>
<li>a column header for the cel file names is now present in report file output. <a class="el" href="classMeta.html" title="Helper class for meta probesets in apt-data-subset.cpp.">Meta</a> information is now included in the report file header. Additional column (call_rate) was added which matches that in the AGCC CHP file header. Also note that only the filename is now listed. The path, if present is removed.</li>
<li>changes to how the md5sum (gc background probes and normalization probes) is computed to ensure cross platform consistency</li>
<li>syncronized meta header data between apt-probeset-genotype and apt-probeset summarize. IE, there will be some changes to the meta info going into CHP files and other output files.</li>
<li>added support for SPF files</li>
<li>improved masking of probes (still experimental)<ul>
<li>see details under apt-probeset-summarize</li>
<li>currently cannot remove an entire probeset -- will result in an error. use probeset list with -s option instead.</li>
</ul>
</li>
<li>added --chip-type option. can be specified multiple times. This option will dictate the chip type in the output files and it will also specify what chip types the cel and lib files are checked against. Note that this will override the typcial check between chip types in lib files versus those in the cel files.</li>
<li>When doing CDF based analysis, chip types based on substrings ending with "." are also considered a match. IE, CDF file foo.bar.v1.r10.cdf will be detected as matching the possible chip types foo.bar.v1.r10, foo.bar.v1, foo.bar, and foo.</li>
<li>added a --set-analysis-name option which will override the name computed from the analysis string. This option will affect the prefix used for output file names and the labels in certain meta fields and/or report file metrics.</li>
</ul>
</li>
</ul>
</li>
</ul>
<ul>
<li>Source Code<ul>
<li>refactor how genders are handled. Now <a class="el" href="classGenderCalls.html" title="Class for determining gender from a cel set of cel files.">GenderCalls</a> interface which the various gender calling methods implement. A map of <a class="el" href="classGenderCalls.html" title="Class for determining gender from a cel set of cel files.">GenderCalls</a> is then generated and passed around in the genotype engine.<ul>
<li>refactor EM gender calling to implement <a class="el" href="classGenderCalls.html" title="Class for determining gender from a cel set of cel files.">GenderCalls</a> and <a class="el" href="classChipSummary.html" title="Class for objects that provide chip summary metrics.">ChipSummary</a>, hook <a class="el" href="classChipSummary.html" title="Class for objects that provide chip summary metrics.">ChipSummary</a> into reporters</li>
<li>refactor DM listener to implement <a class="el" href="classGenderCalls.html" title="Class for determining gender from a cel set of cel files.">GenderCalls</a> and <a class="el" href="classChipSummary.html" title="Class for objects that provide chip summary metrics.">ChipSummary</a>, hook <a class="el" href="classChipSummary.html" title="Class for objects that provide chip summary metrics.">ChipSummary</a> into reporters</li>
</ul>
</li>
<li>refactor <a class="el" href="classPgOptions.html">PgOptions</a> class to dynamicly allocate options. (See apt-probeset-genotype for example of the new interface.)</li>
<li>integrated birdseed 2.4 from the Broad<ul>
<li>QuantMethods: birdseed-v1, birdseed-dev, birdseed (which is birdseed-v1)</li>
<li><a class="el" href="classAnalysis.html">Analysis</a> Strings: birdseed-v1, birdseed-dev, birdseed (which is birdseed-v1)</li>
<li>refactor the CN probe gender calling method as a <a class="el" href="classCelListener.html" title="Class for objects that need to actually access real cel file data.">CelListener</a>, <a class="el" href="classGenderCalls.html" title="Class for determining gender from a cel set of cel files.">GenderCalls</a>, and <a class="el" href="classChipSummary.html" title="Class for objects that provide chip summary metrics.">ChipSummary</a>. Hook <a class="el" href="classChipSummary.html" title="Class for objects that provide chip summary metrics.">ChipSummary</a> into reporters.</li>
<li>change eps/epsilon alg params to use full number, not scientific notation which breaks the state propigation in analysis-spec</li>
<li>moved SpecialSnps into chipstream folder</li>
<li>fixed propigation of user set values into reported analysis spec for birdseed</li>
<li>current birdseed folder renamed birdseed-dev</li>
<li>created birdseed-v1 folder with previous birdseed implementation [GenderCaller and FitSNPGaussiansPriors3]</li>
<li>The code in birdseed-v1 moved into a C++ namespace "birdseed::v1".</li>
<li>The code in birdseed-dev moved into a C++ namespace "birdseed::dev".</li>
<li>New analysis spec alias: birdseed-v1, birdseed-dev. birdseed-dev will always point to the latest Broad recomendations. birdseed-v2 will be added when the birdseed v2 method is frozen.</li>
<li><a class="el" href="classAnalysis.html">Analysis</a> spec "birdseed" would remain as a pointer to "birdseed-v1".</li>
<li>birdseed correction-factor was locked at 1. can now be set by user.</li>
</ul>
</li>
<li>implement <a class="el" href="classFileGenders.html" title="Read gender assignments from file.">FileGenders</a> for reading genotype genders -- also checks against cel file names in opts.celFiles (previous stub did not).</li>
<li>removed unused legacy code in <a class="el" href="ProbesetGenotypeEngine_8cpp.html" title="Core routines for apt-probeset-genotype binaries. By separating the command line parsing form the com...">ProbesetGenotypeEngine.cpp</a></li>
<li>minor clean up of some memory de-allocation in main compute method</li>
<li>complete refactor of <a class="el" href="classGenoQC.html" title="Object for generating chip qc statistics from a Encapsulates all the work of generating chip qc stats...">GenoQC</a> to use <a class="el" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a> and CelListeners.</li>
<li>The regression tests have been reconfigured to allow for running the regression tests outside of Affymetrix. Regression test data is now provided as a separate download from the APT source code.</li>
<li>Refactored windows builds to use static libraries for core components (ie calvin, file, chipstream).</li>
<li>Lots more regression tests</li>
<li>Support for expression Chp file checking in <a class="el" href="ChpCheck_8h.html" title="Class for doing a check of two CHP files after regression run.">ChpCheck.h</a> and <a class="el" href="CalvinChpCheck_8h.html" title="Class for doing a check of two CHP files after regression run.">CalvinChpCheck.h</a></li>
<li>fix memory allocation in Pg::Options</li>
<li>More checks regarding GC count info when using gc-bg, pm-gcbg, and dabg</li>
<li>More use of <a class="el" href="classVerbose.html" title="Verbose.">Verbose</a> class for log output (rather than using cout or cerr directly.</li>
<li>More use of <a class="el" href="classErr.html" title="Err.">Err</a> class (rather than exit() or throwing exceptions)</li>
<li><a class="el" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a> and <a class="el" href="classVerbose.html#a34e1d936178741eeb78de93c5497e1b8" title="Print a warning message.">Verbose::warn</a> no default to cerr rather than cout.</li>
<li>Moved chip-type checks out of the command line wrappers and into the engine code. (ProbesetSummarize and ProbesetGenotype)<ul>
<li>generally you do not want to fill in the chipType and chipTypeSupplied members of the engine options class.</li>
<li>setting chipType will force all output to be reported as that chip type.</li>
<li>setting chipTypeSupplied will force chip type checking against that list of chip types. If chipType is not set, it will be set to the first entry in chipTypeSupplied.</li>
</ul>
</li>
<li>Refactored how chip summary metrics are propigated into output files. There is now a <a class="el" href="classChipSummary.html" title="Class for objects that provide chip summary metrics.">ChipSummary</a> interface which various classes can implement. These classes can then be registered with certain reporters (ie Run Report and AGCC CHP reporters) as sources of chip summary info.</li>
</ul>
</li>
</ul>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.8.0 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Tue May 29 2007</dd></dl>
<ul>
<li>New Applications:<ul>
<li>apt-geno-qc: a tool for computing single chip QC metrics for WGSA (500k, Genome Wide SNP5, and Genome Wide SNP6) genotyping chips</li>
</ul>
</li>
</ul>
<ul>
<li>Changed Applications:<ul>
<li>apt-probeset-summarize:<ul>
<li><b>Fixed overflow error in accumulator for DABG values (previous behavior affected probesets with lots of probes where the DABG p-value with go to 1 rather than 0.</b></li>
<li><b>fixed bug related to non-square CEL files</b> where the mapping between x/y coordinates and probe IDs was incorrect</li>
<li>atom_id field in the feature response and feature residual files now contain "0". <a class="el" href="classAtom.html" title="Represents a collection of probes that make up a unit for an intensity measurement of a particular se...">Atom</a> IDs are no longer retained in the analysis software in order to reduce the memory requirements. As a result, the information is not available for reporting into this output file.</li>
<li>added ability to treat genotyping and copy number probesets as expression probesets. See "--explain expr".</li>
<li>added pca-select and spect-select feature selection methods (intended for use in computing gene level estimates from WT Gene and Exon arrays). See "--explain pca-select" and "--explain spect-select".</li>
<li>enable use of meta probeset files (mps) with cdf files</li>
<li>added PM+MM (pm-sum) adjustor. See "--explain pm-sum".</li>
<li>added median and mean summarization methods. See "--explain median".</li>
</ul>
</li>
<li>apt-probeset-genotype:<ul>
<li><b>fixed bug related to non-square CEL files</b> where the mapping between x/y coordinates and probe IDs was incorrect</li>
<li>added birdseed method for genotyping calling (associated with SNP6)</li>
<li>added brlmm-p-plus method for genotyping calling (associated with SNP6)</li>
<li>added brlmm-p method for genotyping calling (associated with SNP5)</li>
</ul>
</li>
</ul>
</li>
</ul>
<ul>
<li>Source Code<ul>
<li>reduce memory requirements of apt-probeset-summarize and apt-probeset-genotype by using a new internal structures for chip layout information</li>
<li>added support for an experimental chip layout format (SPF)</li>
<li>new build system with multiple static libraries</li>
</ul>
</li>
</ul>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.6.0 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Fri Sep 29 2006</dd></dl>
<ul>
<li>General<ul>
<li>Upgraded to Newmat 10D</li>
<li>Added preliminary support for Command Console Files (CEL and CHP) note that not all applications support CC CEL and CHP files yet. See application specific notes below for which applications now support CC files.</li>
</ul>
</li>
</ul>
<ul>
<li><b>Change to the PLIER algorithm</b><ul>
<li>The PLIER source code contained in this release is replacing the separate stand alone PLIER SDK.</li>
<li>The PLIER code was changed by the addition of three new options:<ul>
<li>SafetyZero: Within the PLIER code zero values are now adjusted by a slight positive bias (default 0.000001) to prevent errors during log transformation. This is particularly relevant when feature response values are rounded to zero and feed into PLIER as precomputed feature effects.</li>
<li>NumericalTolerance: Sets how hard the PLIER algorithm will work to resolve small differences between results when using precomputed vs non-precomputed feature responses. (Default is 0.1)</li>
<li>FixPrecomputed: This option causes the PLIER code to recompute the target response after fitting the feature response. This improves concordance between PLIER results generated de novo versus those generated using a precomputed feature response.</li>
</ul>
</li>
<li>These changes improve the stability of PLIER estimates when comparing those generated using precomputed feature responses versus those generated using full PLIER model fit. Note that in this release of APT you may still see some significant differences in cases where the likelihood space is relatively flat combined with rounding errors caused by writing out floating point values to the text feature response file.</li>
</ul>
</li>
</ul>
<ul>
<li><b>Change sketch normalization default sketch size</b><ul>
<li>Sketch normalization (ie when you request quant-norm and do not specify a sketch size) will now default to a sketch size of the 1% of the features or 50,000 - which ever is higher. If there are fewer than 50,000 features, then all the features will be used (ie full quantile normalization).</li>
</ul>
</li>
</ul>
<ul>
<li>New Applications:<ul>
<li>apt-cel-extract: extract intensities from CEL files</li>
<li>apt-chp-to-txt: extract values from CHP files</li>
<li>apt-cel-to-pq: dump out mapping genotype probeset intensities as quartet</li>
<li>apt-dump-pgf: extract/dump out information in pgf and clf library files</li>
<li>apt-summary-vis: create IGB and BED visualization files from summary.txt files</li>
<li>apt-cdf-export: extract/dump out information in the cdf library files</li>
<li>apt-tsv-join: merge text files</li>
<li>apt-matrix-diff: compare two text files w/ matrix of values</li>
</ul>
</li>
</ul>
<ul>
<li>Changed Applications:<ul>
<li>apt-midas:<ul>
<li><b>Changed -c option to --cel-files for consistency with other APT apps.</b></li>
<li><b>Changed header in file grouping CEL files from cel_file to cel_files for consistency with other APT apps.</b></li>
</ul>
</li>
<li>apt-probeset-summarize:<ul>
<li>meta probeset files are now specified using a separate option, -m or --meta-probesets rather than by using the -s option as before.</li>
<li>change --chp-output option to --xda-chp-output</li>
<li>change output/input files to use probe_id rather than feature_id for improved consistency with terminology used in various library files</li>
<li>apt-probeset-summarize will now try and guess an appropriate block size given the amount of available memory to reduce the change of out of memory errors and disk swap</li>
<li>support for Comand Console (CC) CHP file output and CEL file input</li>
<li>rm-probe option for masking out probes</li>
<li>transposed report file contents to be consistent with apt-probeset-genotype</li>
<li>reports associated with each quantification method such that the use of multiple analysis parameters (-a) will result in multiple reports</li>
<li>reports are always generated -- removed --report option</li>
<li>added an avgdiff and sea methods</li>
<li>added md5sum of probes used for GC background correction</li>
<li>added md5sum of probes used for computing quantile normalization sketch</li>
<li>refactored memory allocation to improve memory use on Windows where the address space is more fragmented</li>
<li>reduced memory used by the quantile normalization (including sketch) code</li>
</ul>
</li>
</ul>
</li>
</ul>
<ul>
<li>apt-probeset-genotype:<ul>
<li>change --chp-output option to --xda-chp-output</li>
<li>rm-probe option for masking out probes</li>
<li>automatic blocking of probesets to improve usability on low memory machines</li>
<li>support for Comand Console (CC) CHP file output and CEL file input (CC genotype CHP file content is currently experimental and will change)</li>
<li>changed Unknown to UnknownGender</li>
<li>for xda CHP output, report median raw PM values for controls (in order to be consistent with CHP files generated by GTYPE)</li>
<li>the report file now had a column header for the cel file name, "cel_files"</li>
</ul>
</li>
</ul>
<ul>
<li>Binary Releases<ul>
<li>Added Mac OS-X Universal/Fat Binaries for i386/ppc/ppc64</li>
<li>Removed dependency on .NET for windows binaries</li>
</ul>
</li>
</ul>
<ul>
<li>Source Code<ul>
<li>draft file format documentation</li>
<li>refactor common code out of the application specific engines</li>
<li>refactor the QC reports to allow for probe and probeset level metrics</li>
<li>formalizing structure of options that are used by chipstream via <a class="el" href="classSelfDoc.html" title="Small interface to for algorithmic classes that can explain themselves.">SelfDoc</a> interface. There is now a <a class="el" href="classSelfDoc_1_1Opt.html" title="Description of one possible option/parameter for a self describing class.">SelfDoc::Opt</a> class which contains a name, description, default and range values for each available option.</li>
<li>include Command Console file parsing code</li>
<li>allow for serialization of engine parameters</li>
<li>clean up code to quell gcc warnings</li>
<li>enable building fat/universal binaries on OS-X</li>
<li>cancel engines is now done through the ProgressHandlers</li>
<li>refactor general enumeration and types not specific to any one class or method</li>
</ul>
</li>
</ul>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.4.0 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Wed Apr 26 2006</dd></dl>
<p>This is the first stand alone release of the Affymetrix Power Tools (APT). Previous versions were released as part of the Exon Array Computational Tool (ExACT).</p>
<ul>
<li>General<ul>
<li>The default APT license is now GPL. For commercial override contact <a href="mailto:devnet@affymetrix.com">devnet@affymetrix.com</a>.</li>
</ul>
</li>
</ul>
<ul>
<li>New Applications<ul>
<li>apt-cel-transformer: create new CEL files with various data transformations applied (ie normalization)</li>
<li>apt-snp-compare: simple tool for comparing matrix of SNP calls</li>
</ul>
</li>
</ul>
<ul>
<li>Changed Applications<ul>
<li>apt-probeset-genotype: implements the BRLMM genotype calling algorithm for mapping arrays</li>
<li>apt-probeset-summarize: generate probeset signal estimates from expression and exon arrays<ul>
<li>changed name to include apt prefix</li>
<li>change feature response and residual output (and input) files now have probeset_id, atom_id, feature_id (aka probe_id), x, and y values.</li>
<li>change default precision of output (0 places for most files, 5 digits for dabg p values)</li>
</ul>
</li>
<li>Broken out main probeset-genotype and probeset-summarize into a core engine and a separate thin command line wrapper.</li>
<li>Refactor status and log reporting to facilitate GUI integration</li>
<li>Migrate windows builds to Visual Studio 2005</li>
<li>New build system for *NIX systems</li>
<li>More doxygen high level and API documentation</li>
</ul>
</li>
</ul>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.2.1 (as part of ExACT 1.2.1) </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Mon Jan 30 2006</dd></dl>
<p>Minor changes on the C++ side to support re-applying a multi-chip PLIER analysis to new chips.</p>
<ul>
<li>Changed Applications<ul>
<li>probeset-summarize<ul>
<li>added option to use pre-computed normalization sketches and precomputed feature responses</li>
<li>added option to save sketches and precomputed feature responses</li>
<li><b>WARNING, because of the way the PLIER algorithm alternates optimizing the signal estimate and the feature responses, using precomputed feature responses will result in a slightly different answer when the same CEL files are used.</b></li>
</ul>
</li>
</ul>
</li>
</ul>
<dl class="version"><dt><b>Version:</b></dt><dd>apt-1.2.0 (as part of ExACT 1.2.0) </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>Mon Jan 02 2006</dd></dl>
<p>This was essentially the first release of APT. Most of the C++ code was experimental and thus subject to change. You may try the <a href="mailto:devnet@affymetrix.com">devnet@affymetrix.com</a> mail address or the DevNet ExACT message forum on Affymetrix.com for unofficial support.</p>
<ul>
<li>New Applications<ul>
<li>midas: compute the MiDAS F stat and P values for alternative splicing, the MiDAS method is described in a whitepaper available <a href="http://www.affymetrix.com/support/technical/byproduct.affx?product=exact">here</a>.</li>
<li>apt-probeset-summarize: generate probeset signal estimates from expression and exon arrays<ul>
<li>does normalization and signal summarization in a single step</li>
<li>includes <a class="el" href="classRMA.html" title="Holds methods for computing the popular Robust Muliarray Average of Microarrays (RMA).">RMA</a>, IterPLIER, and PLIER</li>
<li>experimental</li>
</ul>
</li>
</ul>
</li>
</ul>
<ul>
<li>Source Code<ul>
<li>Added C++ DABG implementation. See the source code under sdk/dabg.</li>
<li>Added experimental C++ implementation of the MiDAS altsplice detection method. See the source code under sdk/midas.</li>
<li>Added experimental C++ <a class="el" href="classChipStream.html" title="ChipStream - base class for objects that can initialize themselves one at a time and transform intens...">ChipStream</a> code. This code provides a framework to apply multiple transformations in a multichip analysis. See code under sdk/chipstream.<ul>
<li>A command line program using <a class="el" href="classChipStream.html" title="ChipStream - base class for objects that can initialize themselves one at a time and transform intens...">ChipStream</a>, probeset-summarize, is also provided. This program can do normalization and probe summarization (ie PLIER, <a class="el" href="classRMA.html" title="Holds methods for computing the popular Robust Muliarray Average of Microarrays (RMA).">RMA</a>, ...) in a single go and can do so with either a CDF file or a PGF/CLF files. See code under sdk/chipstream/probeset-summarize.</li>
</ul>
</li>
<li>Added a separate build system for the C++ code. The makefile will build a simple Affymetrix SDK static library and the chip stream (probeset-summarize) and midas executables. If you have CPP Unit installed, some basic unit testing will also be done.</li>
</ul>
</li>
</ul>
<p><b></p>
<pre>Affymetrix Power Tools (APT) Release 1.16.1</pre><p></b> </p>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:56 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
