<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: GenotypeTableModel Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&nbsp;List</span></a></li>
      <li><a href="classes.html"><span>Class&nbsp;Index</span></a></li>
      <li><a href="hierarchy.html"><span>Class&nbsp;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&nbsp;Members</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pub-attribs">Public Attributes</a>  </div>
  <div class="headertitle">
<h1>GenotypeTableModel Class Reference</h1>  </div>
</div>
<div class="contents">
<!-- doxytag: class="GenotypeTableModel" --><!-- doxytag: inherits="TranslationInputStreamTableModel" --><div class="dynheader">
Inheritance diagram for GenotypeTableModel:</div>
<div class="dyncontent">
 <div class="center">
  <img src="classGenotypeTableModel.png" usemap="#GenotypeTableModel_map" alt=""/>
  <map id="GenotypeTableModel_map" name="GenotypeTableModel_map">
<area href="classTranslationInputStreamTableModel.html" alt="TranslationInputStreamTableModel" shape="rect" coords="0,0,206,24"/>
</map>
</div>

<p><a href="classGenotypeTableModel-members.html">List of all members.</a></p>
<table class="memberdecls">
<tr><td colspan="2"><h2><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classGenotypeTableModel.html#ad46742dcc88af3643d6f1c4190ebf022">GenotypeTableModel</a> (const <a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp;rte, const std::string &amp;genoFileName, <a class="el" href="classGenotypeOverrideTableModel.html">GenotypeOverrideTableModel</a> *gotm=NULL)</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><a class="el" href="classGenotypeTableModel.html#ad46742dcc88af3643d6f1c4190ebf022" title="GenotypeTableModel::GenotypeTableModel Synopsis:">GenotypeTableModel::GenotypeTableModel</a> Synopsis:  <a href="#ad46742dcc88af3643d6f1c4190ebf022"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classGenotypeTableModel.html#a69c312c1684ea7269f7adfb0c5b9f1a0">GenotypeTableModel</a> (const <a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp;rte, <a class="el" href="classGenotypeOverrideTableModel.html">GenotypeOverrideTableModel</a> *gotm=NULL)</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><a class="el" href="classGenotypeTableModel.html#ad46742dcc88af3643d6f1c4190ebf022" title="GenotypeTableModel::GenotypeTableModel Synopsis:">GenotypeTableModel::GenotypeTableModel</a> Synopsis.  <a href="#a69c312c1684ea7269f7adfb0c5b9f1a0"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">void&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classGenotypeTableModel.html#a8ab297d27c47f4fee538d77b5bdaca95">describeVerboseExperimentGene</a> (const <a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp;rte, ADT_VERBOSE_ENUM overrideLevel=ADT_VERBOSE_NULL)</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><a class="el" href="classGenotypeTableModel.html#a8ab297d27c47f4fee538d77b5bdaca95" title="GenotypeTableModel::describeVerboseExperimentGene Synopsis:">GenotypeTableModel::describeVerboseExperimentGene</a> Synopsis:  <a href="#a8ab297d27c47f4fee538d77b5bdaca95"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a7e0e907849d13a6f285b1411deedf876"></a><!-- doxytag: member="GenotypeTableModel::getGeneName" ref="a7e0e907849d13a6f285b1411deedf876" args="()" -->
std::string&nbsp;</td><td class="memItemRight" valign="bottom"><b>getGeneName</b> ()</td></tr>
<tr><td class="memItemLeft" align="right" valign="top">int&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classGenotypeTableModel.html#a82ce9bc2679a496b3592fddb14224f48">getGeneRow</a> (const std::string &amp;gene, int row)</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><a class="el" href="classGenotypeTableModel.html#a82ce9bc2679a496b3592fddb14224f48" title="GenotypeTableModel::getGeneRow: Synopsis:">GenotypeTableModel::getGeneRow</a>: Synopsis:  <a href="#a82ce9bc2679a496b3592fddb14224f48"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">int&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classGenotypeTableModel.html#ad413a9b5e58f44ef33a42a641806cbed">getGeneRowSize</a> (const std::string &amp;gene)</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">getGeneRowSize: Synopsis:  <a href="#ad413a9b5e58f44ef33a42a641806cbed"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">int&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classGenotypeTableModel.html#ae393d55ec62a42dd5c04b68d7183d6b9">getNextExperiment</a> (<a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp;rte, <a class="el" href="classTranslationTableModel.html">TranslationTableModel</a> &amp;ttm, const std::map&lt; std::string, std::string &gt; &amp;geneExperimentCopyNumberCall)</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><a class="el" href="classGenotypeTableModel.html#ae393d55ec62a42dd5c04b68d7183d6b9" title="GenotypeTableModel::getNextExperiment: Synopsis:">GenotypeTableModel::getNextExperiment</a>: Synopsis:  <a href="#ae393d55ec62a42dd5c04b68d7183d6b9"></a><br/></td></tr>
<tr><td colspan="2"><h2><a name="pub-attribs"></a>
Public Attributes</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="adbeecd6e1d33cd754aa765e8aef18f02"></a><!-- doxytag: member="GenotypeTableModel::m_experiments" ref="adbeecd6e1d33cd754aa765e8aef18f02" args="" -->
std::set&lt; std::string &gt;&nbsp;</td><td class="memItemRight" valign="bottom"><b>m_experiments</b></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a77fe10ab01582e8875f749e9f31a34d5"></a><!-- doxytag: member="GenotypeTableModel::m_experimentGenes" ref="a77fe10ab01582e8875f749e9f31a34d5" args="" -->
std::set&lt; std::string &gt;&nbsp;</td><td class="memItemRight" valign="bottom"><b>m_experimentGenes</b></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a04fe6ea7edf18ced261944358b7868a6"></a><!-- doxytag: member="GenotypeTableModel::m_experimentName" ref="a04fe6ea7edf18ced261944358b7868a6" args="" -->
std::string&nbsp;</td><td class="memItemRight" valign="bottom"><b>m_experimentName</b></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aa4d3d17577fe5086184ec3afe0dc23a5"></a><!-- doxytag: member="GenotypeTableModel::m_geneCopyNumber" ref="aa4d3d17577fe5086184ec3afe0dc23a5" args="" -->
std::map&lt; std::string, int &gt;&nbsp;</td><td class="memItemRight" valign="bottom"><b>m_geneCopyNumber</b></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a6b3f5cb9941eb3294095ed44f98a3b30"></a><!-- doxytag: member="GenotypeTableModel::m_geneName" ref="a6b3f5cb9941eb3294095ed44f98a3b30" args="" -->
std::string&nbsp;</td><td class="memItemRight" valign="bottom"><b>m_geneName</b></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a9c5d9c2e23113cc99d5c9a2ea76387d1"></a><!-- doxytag: member="GenotypeTableModel::m_geneRowIndex" ref="a9c5d9c2e23113cc99d5c9a2ea76387d1" args="" -->
std::map&lt; std::string, <br class="typebreak"/>
std::vector&lt; int &gt; &gt;&nbsp;</td><td class="memItemRight" valign="bottom"><b>m_geneRowIndex</b></td></tr>
</table>
<hr/><a name="_details"></a><h2>Detailed Description</h2>

<p>Definition at line <a class="el" href="GenotypeTableModel_8h_source.html#l00040">40</a> of file <a class="el" href="GenotypeTableModel_8h_source.html">GenotypeTableModel.h</a>.</p>
<hr/><h2>Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="ad46742dcc88af3643d6f1c4190ebf022"></a><!-- doxytag: member="GenotypeTableModel::GenotypeTableModel" ref="ad46742dcc88af3643d6f1c4190ebf022" args="(const RunTimeEnvironment &amp;rte, const std::string &amp;genoFileName, GenotypeOverrideTableModel *gotm=NULL)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">GenotypeTableModel::GenotypeTableModel </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>rte</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const std::string &amp;&nbsp;</td>
          <td class="paramname"> <em>genoFileName</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classGenotypeOverrideTableModel.html">GenotypeOverrideTableModel</a> *&nbsp;</td>
          <td class="paramname"> <em>gotm</em> = <code>NULL</code></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p><a class="el" href="classGenotypeTableModel.html#ad46742dcc88af3643d6f1c4190ebf022" title="GenotypeTableModel::GenotypeTableModel Synopsis:">GenotypeTableModel::GenotypeTableModel</a> Synopsis: </p>
<p>DMET2: TsvFile constructor.</p>
<p>This is a single instance class where only one of these should ever be in existence:</p>
<p>genotype data file -&gt; genotype table model.</p>
<p>A quick glance at the column headers defined for this table shows that the schema for this table looks nothing like what's in a CHP file header.</p>
<p>A decision was made early on that the only data to be used for DMET3 translation with the CHP file would be a subset of the DMET2 data. Therefore the DMET2 table format was left in place.</p>
<p>This table is the only object to stream data. This model then is the closest to being a data access object. The transfer object is simply the rows (m_rows) and the "readNextExperiment" accessor is the data access method. This is unlike other "data access objects" using the "TranslationInputTsvTableModel" where all data is read into memory at construction.</p>
<p>The business object that consumes this data access / transfer object is the "TranslationTable" business object. The "TranslationTable" object creates chromatid call sets from the genotype data transfered.</p>
<p>When this object is constructed, no data is read in. To access data use the "readNextExperiment" accessor and the next experiment's worth of data will be read in, albiet from a DMET2 genotype short report file or a DMET3 CHP file.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>rte</em>&nbsp;</td><td>- the single instance <a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> which contains options </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>genoFileName</em>&nbsp;</td><td>- the input file to read in. </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>gotm</em>&nbsp;</td><td>- the single instance of the genotype override file table model </td></tr>
  </table>
  </dd>
</dl>

<p>Definition at line <a class="el" href="GenotypeTableModel_8cpp_source.html#l00100">100</a> of file <a class="el" href="GenotypeTableModel_8cpp_source.html">GenotypeTableModel.cpp</a>.</p>

<p>References <a class="el" href="GenotypeTableModel_8cpp_source.html#l00100">GenotypeTableModel()</a>.</p>

<p>Referenced by <a class="el" href="GenotypeTableModel_8cpp_source.html#l00100">GenotypeTableModel()</a>.</p>

</div>
</div>
<a class="anchor" id="a69c312c1684ea7269f7adfb0c5b9f1a0"></a><!-- doxytag: member="GenotypeTableModel::GenotypeTableModel" ref="a69c312c1684ea7269f7adfb0c5b9f1a0" args="(const RunTimeEnvironment &amp;rte, GenotypeOverrideTableModel *gotm=NULL)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">GenotypeTableModel::GenotypeTableModel </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>rte</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classGenotypeOverrideTableModel.html">GenotypeOverrideTableModel</a> *&nbsp;</td>
          <td class="paramname"> <em>gotm</em> = <code>NULL</code></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p><a class="el" href="classGenotypeTableModel.html#ad46742dcc88af3643d6f1c4190ebf022" title="GenotypeTableModel::GenotypeTableModel Synopsis:">GenotypeTableModel::GenotypeTableModel</a> Synopsis. </p>
<p>DMET3 CHP constructor to coalesce the <a class="el" href="classGenotypeTableModel.html">GenotypeTableModel</a> with the DMET2 files. See the TSV constructor for a complete discussion.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>rte</em>&nbsp;</td><td>- <a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> which contains options </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>gotm</em>&nbsp;</td><td>- single instance of the genotype override table model </td></tr>
  </table>
  </dd>
</dl>

<p>Definition at line <a class="el" href="GenotypeTableModel_8cpp_source.html#l00124">124</a> of file <a class="el" href="GenotypeTableModel_8cpp_source.html">GenotypeTableModel.cpp</a>.</p>

<p>References <a class="el" href="GenotypeTableModel_8cpp_source.html#l00100">GenotypeTableModel()</a>.</p>

</div>
</div>
<hr/><h2>Member Function Documentation</h2>
<a class="anchor" id="a8ab297d27c47f4fee538d77b5bdaca95"></a><!-- doxytag: member="GenotypeTableModel::describeVerboseExperimentGene" ref="a8ab297d27c47f4fee538d77b5bdaca95" args="(const RunTimeEnvironment &amp;rte, ADT_VERBOSE_ENUM overrideLevel=ADT_VERBOSE_NULL)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void GenotypeTableModel::describeVerboseExperimentGene </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>rte</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">ADT_VERBOSE_ENUM&nbsp;</td>
          <td class="paramname"> <em>overrideLevel</em> = <code>ADT_VERBOSE_NULL</code></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p><a class="el" href="classGenotypeTableModel.html#a8ab297d27c47f4fee538d77b5bdaca95" title="GenotypeTableModel::describeVerboseExperimentGene Synopsis:">GenotypeTableModel::describeVerboseExperimentGene</a> Synopsis: </p>
<p>Run time debug routine that can be invoked with various verbose levels. ADT_VERBOSE_INPUT_FILES is the default level.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>rte</em>&nbsp;</td><td>- run time environment </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>overrideLevel</em>&nbsp;</td><td>- ADT_VERBOSE_INPUT_FILES by default, otherwise the ADT_VERBOSE_ENUM level to output </td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>- void </dd></dl>

<p>Definition at line <a class="el" href="GenotypeTableModel_8cpp_source.html#l00148">148</a> of file <a class="el" href="GenotypeTableModel_8cpp_source.html">GenotypeTableModel.cpp</a>.</p>

<p>References <a class="el" href="TranslationInputStreamTableModel_8cpp_source.html#l00262">TranslationInputStreamTableModel::getRowAsString()</a>, and <a class="el" href="Verbose_8cpp_source.html#l00259">Verbose::out()</a>.</p>

</div>
</div>
<a class="anchor" id="a82ce9bc2679a496b3592fddb14224f48"></a><!-- doxytag: member="GenotypeTableModel::getGeneRow" ref="a82ce9bc2679a496b3592fddb14224f48" args="(const std::string &amp;gene, int row)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int GenotypeTableModel::getGeneRow </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&nbsp;</td>
          <td class="paramname"> <em>gene</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&nbsp;</td>
          <td class="paramname"> <em>row</em></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p><a class="el" href="classGenotypeTableModel.html#a82ce9bc2679a496b3592fddb14224f48" title="GenotypeTableModel::getGeneRow: Synopsis:">GenotypeTableModel::getGeneRow</a>: Synopsis: </p>
<p>Returns index std::map from gene row to experiment row index.</p>
<dl class="return"><dt><b>Returns:</b></dt><dd>- the row integer, -1 if the gene or row do not exist. </dd></dl>

<p>Definition at line <a class="el" href="GenotypeTableModel_8cpp_source.html#l00220">220</a> of file <a class="el" href="GenotypeTableModel_8cpp_source.html">GenotypeTableModel.cpp</a>.</p>

<p>Referenced by <a class="el" href="CallSet_8cpp_source.html#l00093">CallSet::CallSet()</a>.</p>

</div>
</div>
<a class="anchor" id="ad413a9b5e58f44ef33a42a641806cbed"></a><!-- doxytag: member="GenotypeTableModel::getGeneRowSize" ref="ad413a9b5e58f44ef33a42a641806cbed" args="(const std::string &amp;gene)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int GenotypeTableModel::getGeneRowSize </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&nbsp;</td>
          <td class="paramname"> <em>gene</em></td>
          <td>&nbsp;)&nbsp;</td>
          <td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>getGeneRowSize: Synopsis: </p>
<p>Returns the number of rows in the experiment m_rows table for one gene. Rows are guaranteed to be in gene order.</p>
<p>gene - the gene std::string to test.</p>
<dl class="return"><dt><b>Returns:</b></dt><dd>- The number of gene related rows, 0 if the gene doesn't exist. </dd></dl>

<p>Definition at line <a class="el" href="GenotypeTableModel_8cpp_source.html#l00197">197</a> of file <a class="el" href="GenotypeTableModel_8cpp_source.html">GenotypeTableModel.cpp</a>.</p>

<p>Referenced by <a class="el" href="CallSet_8cpp_source.html#l00093">CallSet::CallSet()</a>.</p>

</div>
</div>
<a class="anchor" id="ae393d55ec62a42dd5c04b68d7183d6b9"></a><!-- doxytag: member="GenotypeTableModel::getNextExperiment" ref="ae393d55ec62a42dd5c04b68d7183d6b9" args="(RunTimeEnvironment &amp;rte, TranslationTableModel &amp;ttm, const std::map&lt; std::string, std::string &gt; &amp;geneExperimentCopyNumberCall)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int GenotypeTableModel::getNextExperiment </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>rte</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classTranslationTableModel.html">TranslationTableModel</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>ttm</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const std::map&lt; std::string, std::string &gt; &amp;&nbsp;</td>
          <td class="paramname"> <em>geneExperimentCopyNumberCall</em></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p><a class="el" href="classGenotypeTableModel.html#ae393d55ec62a42dd5c04b68d7183d6b9" title="GenotypeTableModel::getNextExperiment: Synopsis:">GenotypeTableModel::getNextExperiment</a>: Synopsis: </p>
<p>This API is simply a wrapper for TranslationInputStreamTableModel-&gt;readNextExperiment.</p>
<p>Records read in are then scanned again for validation and reformating.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>rte</em>&nbsp;</td><td>- the single instance <a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>ttm</em>&nbsp;</td><td>- the single instance translation table model </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>geneExperimentCopyNumberCall</em>&nbsp;</td><td>- the copy "allele" designation per gene</td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>- the number of records read. </dd></dl>

<p>Definition at line <a class="el" href="GenotypeTableModel_8cpp_source.html#l00342">342</a> of file <a class="el" href="GenotypeTableModel_8cpp_source.html">GenotypeTableModel.cpp</a>.</p>

<p>References <a class="el" href="TranslationInputStreamTableModel_8cpp_source.html#l00301">TranslationInputStreamTableModel::readNextExperiment()</a>.</p>

<p>Referenced by <a class="el" href="TranslationEngine_8cpp_source.html#l00266">TranslationEngine::runImp()</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following files:<ul>
<li>translation/<a class="el" href="GenotypeTableModel_8h_source.html">GenotypeTableModel.h</a></li>
<li>translation/<a class="el" href="GenotypeTableModel_8cpp_source.html">GenotypeTableModel.cpp</a></li>
</ul>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:38:14 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
