<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: GeneCall Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&nbsp;List</span></a></li>
      <li><a href="classes.html"><span>Class&nbsp;Index</span></a></li>
      <li><a href="hierarchy.html"><span>Class&nbsp;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&nbsp;Members</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pub-attribs">Public Attributes</a>  </div>
  <div class="headertitle">
<h1>GeneCall Class Reference</h1>  </div>
</div>
<div class="contents">
<!-- doxytag: class="GeneCall" -->
<p><a href="classGeneCall-members.html">List of all members.</a></p>
<table class="memberdecls">
<tr><td colspan="2"><h2><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classGeneCall.html#acbd850ffa3532b1f6afc0712a48f0dde">GeneCall</a> (const <a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp;rte, class <a class="el" href="classTranslationTableModel.html">TranslationTableModel</a> &amp;ttm, const int headerRow, bool isHaplotypeCall)</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><a class="el" href="classGeneCall.html#acbd850ffa3532b1f6afc0712a48f0dde" title="GeneCall::GeneCall Synopsis:">GeneCall::GeneCall</a> Synopsis:  <a href="#acbd850ffa3532b1f6afc0712a48f0dde"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">bool&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classGeneCall.html#afdad68791a3d92cb1ccd78bcebff0c60">addCallElementToAlleleCallSet</a> (class <a class="el" href="classTranslationTableModel.html">TranslationTableModel</a> &amp;ttm, const int headerRow, const int row, const int column, bool allowMultiAllelic, bool rowIsHaplotype)</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><a class="el" href="classGeneCall.html#afdad68791a3d92cb1ccd78bcebff0c60" title="GeneCall::addCallElementToAlleleCallSet Synopsis:">GeneCall::addCallElementToAlleleCallSet</a> Synopsis:  <a href="#afdad68791a3d92cb1ccd78bcebff0c60"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">void&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classGeneCall.html#a804c2c06e1c5717f669c7d2dee212214">describeVerbose</a> (const <a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp;rte, ADT_VERBOSE_ENUM level=ADT_VERBOSE_NULL)</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><a class="el" href="classGeneCall.html#a804c2c06e1c5717f669c7d2dee212214" title="GeneCall::describeVerbose Synopsis:">GeneCall::describeVerbose</a> Synopsis:  <a href="#a804c2c06e1c5717f669c7d2dee212214"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">bool&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classGeneCall.html#ac9276bb5f94a5916eefc26aacdbbc027">existsAlleleCallSet</a> (const std::string &amp;alleleName)</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><a class="el" href="classGeneCall.html#ac9276bb5f94a5916eefc26aacdbbc027" title="GeneCall::existsAlleleCallSet Synopsis:">GeneCall::existsAlleleCallSet</a> Synopsis:  <a href="#ac9276bb5f94a5916eefc26aacdbbc027"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="el" href="classCallSet.html">CallSet</a> &amp;&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classGeneCall.html#a30757e06c17f4315d3686a5341f4b547">getAlleleCallSet</a> (unsigned int index)</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><a class="el" href="classGeneCall.html#a30757e06c17f4315d3686a5341f4b547" title="GeneCall::getAlleleCallSet Synopsis:">GeneCall::getAlleleCallSet</a> Synopsis:  <a href="#a30757e06c17f4315d3686a5341f4b547"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a05e36165b1aabc2becac362bbe93701e"></a><!-- doxytag: member="GeneCall::getReferenceCallSet" ref="a05e36165b1aabc2becac362bbe93701e" args="()" -->
<a class="el" href="classCallSet.html">CallSet</a> &amp;&nbsp;</td><td class="memItemRight" valign="bottom"><b>getReferenceCallSet</b> ()</td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a8d1efed2f0381f74c22107bf82e9e89a"></a><!-- doxytag: member="GeneCall::getVariantCallSet" ref="a8d1efed2f0381f74c22107bf82e9e89a" args="()" -->
<a class="el" href="classCallSet.html">CallSet</a> &amp;&nbsp;</td><td class="memItemRight" valign="bottom"><b>getVariantCallSet</b> ()</td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class <a class="el" href="classCallResults.html">CallResults</a>&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classGeneCall.html#a70871515298b6d8a54199225120c986a">translateExperimentCall</a> (<a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp;rte, const <a class="el" href="classCallSet.html">CallSet</a> &amp;completeSet, <a class="el" href="classGenotypeTableModel.html">GenotypeTableModel</a> &amp;gtm, <a class="el" href="classCallSet.html">CallSet</a> &amp;chromatid1, <a class="el" href="classCallSet.html">CallSet</a> &amp;chromatid2, const std::string &amp;copyNumberCall, const <a class="el" href="classCallSet.html">CallSet</a> &amp;copyNumberZeroCallSet)</td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Translating an experiment call means checking for allele names that correspond in the DMET3 translation table.  <a href="#a70871515298b6d8a54199225120c986a"></a><br/></td></tr>
<tr><td colspan="2"><h2><a name="pub-attribs"></a>
Public Attributes</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aceaa8ac546659c188edb8d209bb80a9f"></a><!-- doxytag: member="GeneCall::m_alleleSet" ref="aceaa8ac546659c188edb8d209bb80a9f" args="" -->
std::vector&lt; <a class="el" href="classCallSet.html">CallSet</a> &gt;&nbsp;</td><td class="memItemRight" valign="bottom"><b>m_alleleSet</b></td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="af0010bffab7859e0b7d9aeba22b21551"></a><!-- doxytag: member="GeneCall::m_gene" ref="af0010bffab7859e0b7d9aeba22b21551" args="" -->
std::string&nbsp;</td><td class="memItemRight" valign="bottom"><b>m_gene</b></td></tr>
</table>
<hr/><a name="_details"></a><h2>Detailed Description</h2>

<p>Definition at line <a class="el" href="GeneCall_8h_source.html#l00035">35</a> of file <a class="el" href="GeneCall_8h_source.html">GeneCall.h</a>.</p>
<hr/><h2>Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="acbd850ffa3532b1f6afc0712a48f0dde"></a><!-- doxytag: member="GeneCall::GeneCall" ref="acbd850ffa3532b1f6afc0712a48f0dde" args="(const RunTimeEnvironment &amp;rte, class TranslationTableModel &amp;ttm, const int headerRow, bool isHaplotypeCall)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">GeneCall::GeneCall </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>rte</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">class <a class="el" href="classTranslationTableModel.html">TranslationTableModel</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>ttm</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const int&nbsp;</td>
          <td class="paramname"> <em>headerRow</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">bool&nbsp;</td>
          <td class="paramname"> <em>isHaplotypeCall</em></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p><a class="el" href="classGeneCall.html#acbd850ffa3532b1f6afc0712a48f0dde" title="GeneCall::GeneCall Synopsis:">GeneCall::GeneCall</a> Synopsis: </p>
<p>Translation Table constructer. The translation table row order is per marker. The column order starting with Reference, Variant and then A1 - ANN is what is being defined here as a call. The translation table is rows of markes with columns of calls.</p>
<p>Each gene has a number of calls designated. The <a class="el" href="classGeneCall.html">GeneCall</a> object represents the entire set of calls for a gene. The <a class="el" href="classGeneCall.html">GeneCall</a> is a set of sets, where the sets are CallSets. The CallSets are sets of CallElements where the CallElements represent the probset id with reference and variant alleles.</p>
<p>A <a class="el" href="classGeneCall.html">GeneCall</a> then represents the set of columns A1 - A50 (ANN) of the translation table for a particular gene.</p>
<p>The columns are chopped up at the Gene level. Within a gene, the column is subdivided by haplotype or marker. A column, like A1, then will have one <a class="el" href="classGeneCall.html">GeneCall</a> per marker where the "Haplotype" column is "N". For those markers of the gene where "Haplotype" column is "Y" then exactly one <a class="el" href="classGeneCall.html">GeneCall</a> is created.</p>
<p>NOTE: While in general the "Haplotype" column is "Y" records are grouped together for a particular gene in the translation table, this is not always the case. <a class="el" href="classProbe.html" title="Represents an individual feature on a chip.">Probe</a> sets with "Haplotype" column of "Y" can be interpersed with the "Haplotype" column "N" probesets.</p>
<p>For a particular gene there is ever only one haplotype group. That group is designated by the set of markers for the gene with "Haplotype" column indicator of "Y". A gene haplotype call (<a class="el" href="classGeneCall.html">GeneCall</a>) then corresponds to a column for all those rows of a gene in the translation table within a haplotype group.</p>
<p>This constructor takes in the header row for a particular gene and then instantiates all the CallSets for all the columns.</p>
<p>The CallSets for each call is subsequently filled with the probe set ids, reference and variant alleles when the translation table is scanned.</p>
<p>In other words, multiple data structures are being created when the translation table is being scanned. To prevent scanning the table multiple times this constructor simply instantiates the empty <a class="el" href="classCallSet.html">CallSet</a> per call and the callee (<a class="el" href="classTranslationTable.html">TranslationTable</a>) is responsible for filling in the CallSets per call as the subsequent rows are being scanned.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>rte</em>&nbsp;</td><td>- <a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> which contains options </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>ttm</em>&nbsp;</td><td>- <a class="el" href="classTranslationTableModel.html">TranslationTableModel</a> with the meta data about the call. </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>headerRow</em>&nbsp;</td><td>- the row in the ttm that has the meta data. </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>isHaplotypeCall</em>&nbsp;</td><td>- if the data row has been predetermined to have "Y" in the "Haplotype" column. </td></tr>
  </table>
  </dd>
</dl>

<p>Definition at line <a class="el" href="GeneCall_8cpp_source.html#l00100">100</a> of file <a class="el" href="GeneCall_8cpp_source.html">GeneCall.cpp</a>.</p>

<p>References <a class="el" href="Err_8h_source.html#l00054">APT_ERR_ASSERT</a>, <a class="el" href="TranslationInputTsvTableModel_8cpp_source.html#l00383">TranslationInputTsvTableModel::columnCount()</a>, <a class="el" href="GeneCall_8cpp_source.html#l00365">existsAlleleCallSet()</a>, <a class="el" href="TranslationTableModel_8cpp_source.html#l00487">TranslationTableModel::getColumnIndex()</a>, <a class="el" href="TranslationTableModel_8cpp_source.html#l00674">TranslationTableModel::getHeaderRowColumnName()</a>, <a class="el" href="TranslationInputTsvTableModel_8cpp_source.html#l00350">TranslationInputTsvTableModel::getRowAsString()</a>, <a class="el" href="Verbose_8cpp_source.html#l00259">Verbose::out()</a>, and <a class="el" href="Verbose_8cpp_source.html#l00277">Verbose::warn()</a>.</p>

</div>
</div>
<hr/><h2>Member Function Documentation</h2>
<a class="anchor" id="afdad68791a3d92cb1ccd78bcebff0c60"></a><!-- doxytag: member="GeneCall::addCallElementToAlleleCallSet" ref="afdad68791a3d92cb1ccd78bcebff0c60" args="(class TranslationTableModel &amp;ttm, const int headerRow, const int row, const int column, bool allowMultiAllelic, bool rowIsHaplotype)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">bool GeneCall::addCallElementToAlleleCallSet </td>
          <td>(</td>
          <td class="paramtype">class <a class="el" href="classTranslationTableModel.html">TranslationTableModel</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>ttm</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const int&nbsp;</td>
          <td class="paramname"> <em>headerRow</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const int&nbsp;</td>
          <td class="paramname"> <em>row</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const int&nbsp;</td>
          <td class="paramname"> <em>column</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">bool&nbsp;</td>
          <td class="paramname"> <em>allowMultiAllelic</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">bool&nbsp;</td>
          <td class="paramname"> <em>rowIsHaplotype</em></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p><a class="el" href="classGeneCall.html#afdad68791a3d92cb1ccd78bcebff0c60" title="GeneCall::addCallElementToAlleleCallSet Synopsis:">GeneCall::addCallElementToAlleleCallSet</a> Synopsis: </p>
<p>Initialize a <a class="el" href="classCallElement.html">CallElement</a> from the translation table (and not an experiment data in the Genotype data (CHP)). To do this we need to identify the <a class="el" href="classProbeSet.html" title="Collection of atoms that measure target that should be absent or present at the same time...">ProbeSet</a> and the reference and variant base. The variant base will be either in the column that resides between A1-A29 or in the "Variant" column itself if no such animal exists.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>ttm</em>&nbsp;</td><td>- <a class="el" href="classTranslationTableModel.html">TranslationTableModel</a> with the meta data about the call. </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>headerRow</em>&nbsp;</td><td>- the row in the ttm that has the meta data. </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>row</em>&nbsp;</td><td>- the row in the <a class="el" href="classTranslationTableModel.html">TranslationTableModel</a> in question. </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>column</em>&nbsp;</td><td>- the column in the <a class="el" href="classTranslationTableModel.html">TranslationTableModel</a> in question. </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>allowMultiAllelic</em>&nbsp;</td><td>- probeSet may appear multiple times. </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>rowIsHaplotype</em>&nbsp;</td><td>- "Haplotype" column has "Y"</td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd>- true upon successfull add. </dd></dl>

<p>Definition at line <a class="el" href="GeneCall_8cpp_source.html#l00205">205</a> of file <a class="el" href="GeneCall_8cpp_source.html">GeneCall.cpp</a>.</p>

<p>References <a class="el" href="Err_8h_source.html#l00054">APT_ERR_ASSERT</a>, <a class="el" href="TranslationInputTsvTableModel_8cpp_source.html#l00383">TranslationInputTsvTableModel::columnCount()</a>, <a class="el" href="TranslationTableModel_8cpp_source.html#l00487">TranslationTableModel::getColumnIndex()</a>, <a class="el" href="TranslationTableModel_8cpp_source.html#l00674">TranslationTableModel::getHeaderRowColumnName()</a>, and <a class="el" href="Verbose_8cpp_source.html#l00259">Verbose::out()</a>.</p>

</div>
</div>
<a class="anchor" id="a804c2c06e1c5717f669c7d2dee212214"></a><!-- doxytag: member="GeneCall::describeVerbose" ref="a804c2c06e1c5717f669c7d2dee212214" args="(const RunTimeEnvironment &amp;rte, ADT_VERBOSE_ENUM level=ADT_VERBOSE_NULL)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void GeneCall::describeVerbose </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>rte</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">ADT_VERBOSE_ENUM&nbsp;</td>
          <td class="paramname"> <em>level</em> = <code>ADT_VERBOSE_NULL</code></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p><a class="el" href="classGeneCall.html#a804c2c06e1c5717f669c7d2dee212214" title="GeneCall::describeVerbose Synopsis:">GeneCall::describeVerbose</a> Synopsis: </p>
<p>Run time debug routine that can be invoked with various verbose levels. ADT_VERBOSE_INPUT_FILES is the default level.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>rte</em>&nbsp;</td><td>- run time environment </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>level</em>&nbsp;</td><td>- ADT_VERBOSE_INPUT_FILES by default, otherwise the ADT_VERBOSE_ENUM level to output </td></tr>
  </table>
  </dd>
</dl>

<p>Definition at line <a class="el" href="GeneCall_8cpp_source.html#l00317">317</a> of file <a class="el" href="GeneCall_8cpp_source.html">GeneCall.cpp</a>.</p>

<p>References <a class="el" href="Verbose_8cpp_source.html#l00259">Verbose::out()</a>.</p>

</div>
</div>
<a class="anchor" id="ac9276bb5f94a5916eefc26aacdbbc027"></a><!-- doxytag: member="GeneCall::existsAlleleCallSet" ref="ac9276bb5f94a5916eefc26aacdbbc027" args="(const std::string &amp;alleleName)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">bool GeneCall::existsAlleleCallSet </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&nbsp;</td>
          <td class="paramname"> <em>alleleName</em></td>
          <td>&nbsp;)&nbsp;</td>
          <td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p><a class="el" href="classGeneCall.html#ac9276bb5f94a5916eefc26aacdbbc027" title="GeneCall::existsAlleleCallSet Synopsis:">GeneCall::existsAlleleCallSet</a> Synopsis: </p>
<p>Returns true if a <a class="el" href="classCallSet.html">CallSet</a> in m_alleleSet has the passed in allele name.</p>
<p>alleleName - the call name to check (i.e. "Ref", "*4") </p>

<p>Definition at line <a class="el" href="GeneCall_8cpp_source.html#l00365">365</a> of file <a class="el" href="GeneCall_8cpp_source.html">GeneCall.cpp</a>.</p>

<p>Referenced by <a class="el" href="GeneCall_8cpp_source.html#l00100">GeneCall()</a>.</p>

</div>
</div>
<a class="anchor" id="a30757e06c17f4315d3686a5341f4b547"></a><!-- doxytag: member="GeneCall::getAlleleCallSet" ref="a30757e06c17f4315d3686a5341f4b547" args="(unsigned int index)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classCallSet.html">CallSet</a> &amp; GeneCall::getAlleleCallSet </td>
          <td>(</td>
          <td class="paramtype">unsigned int&nbsp;</td>
          <td class="paramname"> <em>index</em></td>
          <td>&nbsp;)&nbsp;</td>
          <td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p><a class="el" href="classGeneCall.html#a30757e06c17f4315d3686a5341f4b547" title="GeneCall::getAlleleCallSet Synopsis:">GeneCall::getAlleleCallSet</a> Synopsis: </p>
<p>Convienance function to return either the Reference or Variant allele set.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>index</em>&nbsp;</td><td>- between 0 and num m_alleleSet.size()</td></tr>
  </table>
  </dd>
</dl>
<dl class="return"><dt><b>Returns:</b></dt><dd><a class="el" href="classCallSet.html">CallSet</a> reference corresponding to the index. </dd></dl>

<p>Definition at line <a class="el" href="GeneCall_8cpp_source.html#l00391">391</a> of file <a class="el" href="GeneCall_8cpp_source.html">GeneCall.cpp</a>.</p>

<p>References <a class="el" href="Err_8h_source.html#l00054">APT_ERR_ASSERT</a>.</p>

</div>
</div>
<a class="anchor" id="a70871515298b6d8a54199225120c986a"></a><!-- doxytag: member="GeneCall::translateExperimentCall" ref="a70871515298b6d8a54199225120c986a" args="(RunTimeEnvironment &amp;rte, const CallSet &amp;completeSet, GenotypeTableModel &amp;gtm, CallSet &amp;chromatid1, CallSet &amp;chromatid2, const std::string &amp;copyNumberCall, const CallSet &amp;copyNumberZeroCallSet)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classCallResults.html">CallResults</a> GeneCall::translateExperimentCall </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>rte</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const <a class="el" href="classCallSet.html">CallSet</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>completeSet</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classGenotypeTableModel.html">GenotypeTableModel</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>gtm</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classCallSet.html">CallSet</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>chromatid1</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classCallSet.html">CallSet</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>chromatid2</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const std::string &amp;&nbsp;</td>
          <td class="paramname"> <em>dmet2CopyNumberCall</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const <a class="el" href="classCallSet.html">CallSet</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>geneCopyNumberZeroCallSet</em></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Translating an experiment call means checking for allele names that correspond in the DMET3 translation table. </p>
<p>For reporting purposes all input experiment records from the Genotype CHIP or Short Report files are to have a corresponding record somewhere in the reporting, i.e. be included in the result set. There only exception are for unrecognized genes. An unrecognized gene is simply skipped over. Records for unrecognized genes will not be in the experiment results.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>rte</em>&nbsp;</td><td>- <a class="el" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> which contains options </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>ttm</em>&nbsp;</td><td>- <a class="el" href="classTranslationTableModel.html">TranslationTableModel</a> with the meta data about the call. </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>headerRow</em>&nbsp;</td><td>- the row in the ttm that has the meta data. </td></tr>
  </table>
  </dd>
</dl>

<p>Definition at line <a class="el" href="GeneCall_8cpp_source.html#l00797">797</a> of file <a class="el" href="GeneCall_8cpp_source.html">GeneCall.cpp</a>.</p>

<p>References <a class="el" href="CallResults_8cpp_source.html#l00093">CallResults::appendAlleleCall()</a>, <a class="el" href="Err_8h_source.html#l00050">APT_ERR_ABORT</a>, <a class="el" href="CallSet_8cpp_source.html#l00264">CallSet::buildSecondSet()</a>, <a class="el" href="CallSet_8cpp_source.html#l00448">CallSet::describeVerbose()</a>, <a class="el" href="CallSet_8h_source.html#l00088">CallSet::getCallType()</a>, <a class="el" href="CallSet_8cpp_source.html#l00608">CallSet::match()</a>, <a class="el" href="CallResults_8cpp_source.html#l00214">CallResults::orderCalls()</a>, and <a class="el" href="Verbose_8cpp_source.html#l00259">Verbose::out()</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following files:<ul>
<li>translation/<a class="el" href="GeneCall_8h_source.html">GeneCall.h</a></li>
<li>translation/<a class="el" href="GeneCall_8cpp_source.html">GeneCall.cpp</a></li>
</ul>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:38:13 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
