<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: copynumber/CNAnalysisMethodAllelePeaks.cpp Source File</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li class="current"><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="tabs2">
    <ul class="tablist">
      <li><a href="files.html"><span>File&nbsp;List</span></a></li>
      <li><a href="globals.html"><span>File&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath">
    <ul>
      <li><a class="el" href="dir_2a6c9f730113b04fdf7a766d4dbe6d8e.html">copynumber</a>      </li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>CNAnalysisMethodAllelePeaks.cpp</h1>  </div>
</div>
<div class="contents">
<a href="CNAnalysisMethodAllelePeaks_8cpp.html">Go to the documentation of this file.</a><div class="fragment"><pre class="fragment"><a name="l00001"></a>00001 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00002"></a>00002 <span class="comment"></span><span class="comment">//</span>
<a name="l00003"></a>00003 <span class="comment">// Copyright (C) 2005 Affymetrix, Inc.</span>
<a name="l00004"></a>00004 <span class="comment">//</span>
<a name="l00005"></a>00005 <span class="comment">// This program is free software; you can redistribute it and/or modify</span>
<a name="l00006"></a>00006 <span class="comment">// it under the terms of the GNU General Public License (version 2) as</span>
<a name="l00007"></a>00007 <span class="comment">// published by the Free Software Foundation.</span>
<a name="l00008"></a>00008 <span class="comment">//</span>
<a name="l00009"></a>00009 <span class="comment">// This program is distributed in the hope that it will be useful,</span>
<a name="l00010"></a>00010 <span class="comment">// but WITHOUT ANY WARRANTY; without even the implied warranty of</span>
<a name="l00011"></a>00011 <span class="comment">// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU</span>
<a name="l00012"></a>00012 <span class="comment">// General Public License for more details.</span>
<a name="l00013"></a>00013 <span class="comment">//</span>
<a name="l00014"></a>00014 <span class="comment">// You should have received a copy of the GNU General Public License</span>
<a name="l00015"></a>00015 <span class="comment">// along with this program;if not, write to the</span>
<a name="l00016"></a>00016 <span class="comment">//</span>
<a name="l00017"></a>00017 <span class="comment">// Free Software Foundation, Inc.,</span>
<a name="l00018"></a>00018 <span class="comment">// 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA</span>
<a name="l00019"></a>00019 <span class="comment">//</span><span class="comment"></span>
<a name="l00020"></a>00020 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00021"></a>00021 <span class="comment"></span><span class="comment"></span>
<a name="l00022"></a>00022 <span class="comment">/**</span>
<a name="l00023"></a>00023 <span class="comment"> * @file CNAnalysisMethodAllelePeaks.cpp</span>
<a name="l00024"></a>00024 <span class="comment"> *</span>
<a name="l00025"></a>00025 <span class="comment"> * @brief This file contains the CNAnalysisMethodAllelePeaks class members.</span>
<a name="l00026"></a>00026 <span class="comment"> */</span>
<a name="l00027"></a>00027 
<a name="l00028"></a>00028 <span class="comment">//</span>
<a name="l00029"></a>00029 <span class="preprocessor">#include &quot;<a class="code" href="CNAnalysisMethodAllelePeaks_8h.html" title="This header contains the CNAnalysisMethodAllelePeaks class definition.">copynumber/CNAnalysisMethodAllelePeaks.h</a>&quot;</span>
<a name="l00030"></a>00030 <span class="comment">//</span>
<a name="l00031"></a>00031 <span class="preprocessor">#include &quot;chipstream/HomHiLoCelListener.h&quot;</span>
<a name="l00032"></a>00032 <span class="preprocessor">#include &quot;<a class="code" href="TsvFile_8h.html" title="Headers for the TsvFile classes.">file/TsvFile/TsvFile.h</a>&quot;</span>
<a name="l00033"></a>00033 <span class="preprocessor">#include &quot;file5/File5.h&quot;</span>
<a name="l00034"></a>00034 <span class="preprocessor">#include &quot;file5/File5_File.h&quot;</span>
<a name="l00035"></a>00035 <span class="preprocessor">#include &quot;<a class="code" href="Fs_8h.html" title="///">util/Fs.h</a>&quot;</span>
<a name="l00036"></a>00036 <span class="comment">//</span>
<a name="l00037"></a>00037 <span class="keyword">using namespace </span>std;
<a name="l00038"></a>00038 
<a name="l00039"></a>00039 <span class="comment"></span>
<a name="l00040"></a>00040 <span class="comment">/**</span>
<a name="l00041"></a>00041 <span class="comment"> * @brief Supply a little how/what/why about the algorithms this</span>
<a name="l00042"></a>00042 <span class="comment"> * class performs and what parameters it takes.</span>
<a name="l00043"></a>00043 <span class="comment"> * @return SelfDoc</span>
<a name="l00044"></a>00044 <span class="comment"> */</span>
<a name="l00045"></a><a class="code" href="classCNAnalysisMethodAllelePeaks.html#a98154f4795c290a086101b777704e493">00045</a> <a class="code" href="classSelfDoc.html" title="Small interface to for algorithmic classes that can explain themselves.">SelfDoc</a> <a class="code" href="classCNAnalysisMethodAllelePeaks.html#a98154f4795c290a086101b777704e493" title="Supply a little how/what/why about the algorithms this class performs and what parameters it takes...">CNAnalysisMethodAllelePeaks::explainSelf</a>()
<a name="l00046"></a>00046 {
<a name="l00047"></a>00047     <a class="code" href="classCNAnalysisMethodAllelePeaks.html" title="The Allele Peaks reporter method.">CNAnalysisMethodAllelePeaks</a> obj;
<a name="l00048"></a>00048     <a class="code" href="classSelfDoc.html" title="Small interface to for algorithmic classes that can explain themselves.">SelfDoc</a> doc;
<a name="l00049"></a>00049     doc.<a class="code" href="classSelfDoc.html#a51db8ea9abfa958f92fb5bc601ad68dc" title="Setter method for name appearing in doc.">setDocName</a>(obj.getType());
<a name="l00050"></a>00050     doc.<a class="code" href="classSelfDoc.html#acd4638b447701d1a642effb6b8d2a396" title="Setter method for description appearing in doc.">setDocDescription</a>(obj.getDescription());
<a name="l00051"></a>00051     doc.<a class="code" href="classSelfDoc.html#aa43e6721e35792413711e07bda90861d" title="Setter method for parameters and their documentation.">setDocOptions</a>(obj.<a class="code" href="classCNAnalysisMethodAllelePeaks.html#a4df7175e7bbaeaa3a562a1ba25438099" title="Default Getter method for parameters and their documentation.">getDefaultDocOptions</a>());
<a name="l00052"></a>00052     <span class="keywordflow">return</span> doc;
<a name="l00053"></a>00053 }
<a name="l00054"></a>00054 
<a name="l00055"></a>00055 <span class="comment"></span>
<a name="l00056"></a>00056 <span class="comment">/**</span>
<a name="l00057"></a>00057 <span class="comment"> * @brief Default Getter method for parameters and their documentation.</span>
<a name="l00058"></a>00058 <span class="comment"> * @return map of parameters and their descriptions.</span>
<a name="l00059"></a>00059 <span class="comment"> */</span>
<a name="l00060"></a><a class="code" href="classCNAnalysisMethodAllelePeaks.html#a4df7175e7bbaeaa3a562a1ba25438099">00060</a> std::vector&lt;SelfDoc::Opt&gt; <a class="code" href="classCNAnalysisMethodAllelePeaks.html#a4df7175e7bbaeaa3a562a1ba25438099" title="Default Getter method for parameters and their documentation.">CNAnalysisMethodAllelePeaks::getDefaultDocOptions</a>()
<a name="l00061"></a>00061 {
<a name="l00062"></a>00062   std::vector&lt;SelfDoc::Opt&gt; opts;
<a name="l00063"></a>00063 
<a name="l00064"></a>00064   <span class="comment">// SelfDoc::Opt(name, type, value, default, min, max, description)</span>
<a name="l00065"></a>00065   <a class="code" href="classSelfDoc_1_1Opt.html" title="Description of one possible option/parameter for a self describing class.">SelfDoc::Opt</a> Step = {<span class="stringliteral">&quot;step&quot;</span>, SelfDoc::Opt::Integer, <span class="stringliteral">&quot;50&quot;</span>, <span class="stringliteral">&quot;50&quot;</span>, <span class="stringliteral">&quot;1&quot;</span>, <span class="stringliteral">&quot;NA&quot;</span>, <span class="stringliteral">&quot;Allele Peaks step size.&quot;</span>};
<a name="l00066"></a>00066   opts.push_back(Step);
<a name="l00067"></a>00067 
<a name="l00068"></a>00068   <a class="code" href="classSelfDoc_1_1Opt.html" title="Description of one possible option/parameter for a self describing class.">SelfDoc::Opt</a> Window = {<span class="stringliteral">&quot;window&quot;</span>, SelfDoc::Opt::Integer, <span class="stringliteral">&quot;250&quot;</span>, <span class="stringliteral">&quot;250&quot;</span>, <span class="stringliteral">&quot;30&quot;</span>, <span class="stringliteral">&quot;NA&quot;</span>, <span class="stringliteral">&quot;Allele Peaks window.&quot;</span>};
<a name="l00069"></a>00069   opts.push_back(Window);
<a name="l00070"></a>00070 
<a name="l00071"></a>00071   <a class="code" href="classSelfDoc_1_1Opt.html" title="Description of one possible option/parameter for a self describing class.">SelfDoc::Opt</a> PointCount = {<span class="stringliteral">&quot;point-count&quot;</span>, SelfDoc::Opt::Integer, <span class="stringliteral">&quot;128&quot;</span>, <span class="stringliteral">&quot;128&quot;</span>, <span class="stringliteral">&quot;30&quot;</span>, <span class="stringliteral">&quot;1024&quot;</span>, <span class="stringliteral">&quot;Allele Peaks number of points.&quot;</span>};
<a name="l00072"></a>00072   opts.push_back(PointCount);
<a name="l00073"></a>00073 
<a name="l00074"></a>00074   <a class="code" href="classSelfDoc_1_1Opt.html" title="Description of one possible option/parameter for a self describing class.">SelfDoc::Opt</a> Bandwidth = {<span class="stringliteral">&quot;bandwidth&quot;</span>, SelfDoc::Opt::Double, <span class="stringliteral">&quot;0.45&quot;</span>, <span class="stringliteral">&quot;0.45&quot;</span>, <span class="stringliteral">&quot;0&quot;</span>, <span class="stringliteral">&quot;1&quot;</span>, <span class="stringliteral">&quot;Allele Peaks bandwidth.&quot;</span>};
<a name="l00075"></a>00075   opts.push_back(Bandwidth);
<a name="l00076"></a>00076 
<a name="l00077"></a>00077   <a class="code" href="classSelfDoc_1_1Opt.html" title="Description of one possible option/parameter for a self describing class.">SelfDoc::Opt</a> Cutoff = {<span class="stringliteral">&quot;cutoff&quot;</span>, SelfDoc::Opt::Double, <span class="stringliteral">&quot;0.05&quot;</span>, <span class="stringliteral">&quot;0.05&quot;</span>, <span class="stringliteral">&quot;0&quot;</span>, <span class="stringliteral">&quot;0.5&quot;</span>, <span class="stringliteral">&quot;Allele Peaks cutoff.&quot;</span>};
<a name="l00078"></a>00078   opts.push_back(Cutoff);
<a name="l00079"></a>00079 
<a name="l00080"></a>00080   <a class="code" href="classSelfDoc_1_1Opt.html" title="Description of one possible option/parameter for a self describing class.">SelfDoc::Opt</a> CleanThreshold = {<span class="stringliteral">&quot;clean-threshold&quot;</span>, SelfDoc::Opt::Double, <span class="stringliteral">&quot;0.25&quot;</span>, <span class="stringliteral">&quot;0.25&quot;</span>, <span class="stringliteral">&quot;0&quot;</span>, <span class="stringliteral">&quot;1.0&quot;</span>, <span class="stringliteral">&quot;Allele Peaks clean threshold.&quot;</span>};
<a name="l00081"></a>00081   opts.push_back(CleanThreshold);
<a name="l00082"></a>00082 
<a name="l00083"></a>00083   <a class="code" href="classSelfDoc_1_1Opt.html" title="Description of one possible option/parameter for a self describing class.">SelfDoc::Opt</a> Symmetry = {<span class="stringliteral">&quot;symmetry&quot;</span>, SelfDoc::Opt::Boolean, <span class="stringliteral">&quot;true&quot;</span>, <span class="stringliteral">&quot;true&quot;</span>, <span class="stringliteral">&quot;NA&quot;</span>, <span class="stringliteral">&quot;NA&quot;</span>, <span class="stringliteral">&quot;Allele Peaks SCAR mirror flag.&quot;</span>};
<a name="l00084"></a>00084   opts.push_back(Symmetry);
<a name="l00085"></a>00085 
<a name="l00086"></a>00086   <span class="keywordflow">return</span> opts;
<a name="l00087"></a>00087 }
<a name="l00088"></a>00088 
<a name="l00089"></a>00089 <span class="comment"></span>
<a name="l00090"></a>00090 <span class="comment">/**</span>
<a name="l00091"></a>00091 <span class="comment"> * @brief This static function should be overridden by child classes</span>
<a name="l00092"></a>00092 <span class="comment"> * to return an object of the correct type initialized correctly</span>
<a name="l00093"></a>00093 <span class="comment"> * with the parameters in the string, string map. All objects</span>
<a name="l00094"></a>00094 <span class="comment"> * created this way should be deleted when finished using.</span>
<a name="l00095"></a>00095 <span class="comment"> *</span>
<a name="l00096"></a>00096 <span class="comment"> * @param param - Map of key/value pairs to initialize the object.</span>
<a name="l00097"></a>00097 <span class="comment"> *</span>
<a name="l00098"></a>00098 <span class="comment"> * @return Pointer toCreate object, this should be sub casted as necessary.</span>
<a name="l00099"></a>00099 <span class="comment"> */</span>
<a name="l00100"></a><a class="code" href="classCNAnalysisMethodAllelePeaks.html#ac09eac5dd90eabd7c2eed00a5843f91b">00100</a> <a class="code" href="classSelfCreate.html" title="Small interface for functions that know how to make an instance of themselves given a map of key...">SelfCreate</a>* <a class="code" href="classCNAnalysisMethodAllelePeaks.html#ac09eac5dd90eabd7c2eed00a5843f91b" title="This static function should be overridden by child classes to return an object of the correct type in...">CNAnalysisMethodAllelePeaks::newObject</a>(std::map&lt;std::string,std::string&gt;&amp; params)
<a name="l00101"></a>00101 {
<a name="l00102"></a>00102     <a class="code" href="classSelfDoc.html" title="Small interface to for algorithmic classes that can explain themselves.">SelfDoc</a> doc = <a class="code" href="classCNAnalysisMethodAllelePeaks.html#a98154f4795c290a086101b777704e493" title="Supply a little how/what/why about the algorithms this class performs and what parameters it takes...">explainSelf</a>();
<a name="l00103"></a>00103     std::vector&lt;SelfDoc::Opt&gt; opts = <a class="code" href="classCNAnalysisMethodAllelePeaks.html#a4df7175e7bbaeaa3a562a1ba25438099" title="Default Getter method for parameters and their documentation.">getDefaultDocOptions</a>();
<a name="l00104"></a>00104     <a class="code" href="classCNAnalysisMethodAllelePeaks.html" title="The Allele Peaks reporter method.">CNAnalysisMethodAllelePeaks</a>* pMethod = <span class="keyword">new</span> <a class="code" href="classCNAnalysisMethodAllelePeaks.html#a766aafd60fb07ba126fb2dd51fc00752" title="Constructor.">CNAnalysisMethodAllelePeaks</a>();
<a name="l00105"></a>00105     std::string strPrefix = getPrefix();
<a name="l00106"></a>00106 
<a name="l00107"></a>00107     pMethod-&gt;m_iStep = <a class="code" href="classCNAnalysisMethod.html#a72e009aa8390f61ca119d5204db6115c" title="Setup an int parameter for this analysis method, including min amd max validation.">setupIntParameter</a>(<span class="stringliteral">&quot;step&quot;</span>, strPrefix, params, doc, opts);
<a name="l00108"></a>00108     pMethod-&gt;m_iWindow = <a class="code" href="classCNAnalysisMethod.html#a72e009aa8390f61ca119d5204db6115c" title="Setup an int parameter for this analysis method, including min amd max validation.">setupIntParameter</a>(<span class="stringliteral">&quot;window&quot;</span>, strPrefix, params, doc, opts);
<a name="l00109"></a>00109     pMethod-&gt;m_iPointCount = <a class="code" href="classCNAnalysisMethod.html#a72e009aa8390f61ca119d5204db6115c" title="Setup an int parameter for this analysis method, including min amd max validation.">setupIntParameter</a>(<span class="stringliteral">&quot;point-count&quot;</span>, strPrefix, params, doc, opts);
<a name="l00110"></a>00110     pMethod-&gt;m_dBandwidth = <a class="code" href="classCNAnalysisMethod.html#a866c7f2fb9d785c9716277a9395f1701" title="Setup a double parameter for this analysis method, including min amd max validation.">setupDoubleParameter</a>(<span class="stringliteral">&quot;bandwidth&quot;</span>, strPrefix, params, doc, opts);
<a name="l00111"></a>00111     pMethod-&gt;m_dCutoff = <a class="code" href="classCNAnalysisMethod.html#a866c7f2fb9d785c9716277a9395f1701" title="Setup a double parameter for this analysis method, including min amd max validation.">setupDoubleParameter</a>(<span class="stringliteral">&quot;cutoff&quot;</span>, strPrefix, params, doc, opts);
<a name="l00112"></a>00112     pMethod-&gt;m_dCleanThreshold = <a class="code" href="classCNAnalysisMethod.html#a866c7f2fb9d785c9716277a9395f1701" title="Setup a double parameter for this analysis method, including min amd max validation.">setupDoubleParameter</a>(<span class="stringliteral">&quot;clean-threshold&quot;</span>, strPrefix, params, doc, opts);
<a name="l00113"></a>00113     pMethod-&gt;m_bSymmetry = <a class="code" href="classCNAnalysisMethod.html#ab0a31500922c8cb3c81dbd688ab0aadb" title="Setup a bool parameter for this analysis method.">setupBoolParameter</a>(<span class="stringliteral">&quot;symmetry&quot;</span>, strPrefix, params, doc, opts);
<a name="l00114"></a>00114 
<a name="l00115"></a>00115     <span class="keywordflow">return</span> pMethod;
<a name="l00116"></a>00116 }
<a name="l00117"></a>00117 
<a name="l00118"></a>00118 <span class="comment"></span>
<a name="l00119"></a>00119 <span class="comment">/**</span>
<a name="l00120"></a>00120 <span class="comment"> * @brief Constructor</span>
<a name="l00121"></a>00121 <span class="comment"> */</span>
<a name="l00122"></a><a class="code" href="classCNAnalysisMethodAllelePeaks.html#a766aafd60fb07ba126fb2dd51fc00752">00122</a> <a class="code" href="classCNAnalysisMethodAllelePeaks.html#a766aafd60fb07ba126fb2dd51fc00752" title="Constructor.">CNAnalysisMethodAllelePeaks::CNAnalysisMethodAllelePeaks</a>()
<a name="l00123"></a>00123 {
<a name="l00124"></a>00124     m_bSNPAlreadyLoaded = <span class="keyword">false</span>;
<a name="l00125"></a>00125     m_bLocalProbeSetsDetermined = <span class="keyword">false</span>;
<a name="l00126"></a>00126 }
<a name="l00127"></a>00127 
<a name="l00128"></a>00128 
<a name="l00129"></a>00129 <span class="keywordtype">bool</span> CNAnalysisMethodAllelePeaks::loadSnpReferenceFile(<span class="keyword">const</span> <a class="code" href="classAffxString.html" title="A class derived from std::string.">AffxString</a>&amp; strFileName)
<a name="l00130"></a>00130 {
<a name="l00131"></a>00131     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(3, <span class="stringliteral">&quot;CNAnalysisMethodAllelePeaks::loadSnpReferenceFile&quot;</span>);
<a name="l00132"></a>00132     <span class="keywordtype">bool</span> bSuccessful = <span class="keyword">false</span>;
<a name="l00133"></a>00133 
<a name="l00134"></a>00134     <span class="keywordflow">if</span> (affx::File5_File::isHdf5file(strFileName)) {
<a name="l00135"></a>00135         <span class="keywordflow">try</span> {
<a name="l00136"></a>00136             <a class="code" href="classaffx_1_1File5__File.html">affx::File5_File</a> file5;
<a name="l00137"></a>00137             file5.open(strFileName, affx::FILE5_OPEN_RO);
<a name="l00138"></a>00138             <a class="code" href="classaffx_1_1File5__Group.html">affx::File5_Group</a>* group5 = file5.openGroup(<span class="stringliteral">&quot;SNPReference&quot;</span>, affx::FILE5_OPEN);
<a name="l00139"></a>00139 
<a name="l00140"></a>00140             <span class="keywordflow">if</span>(!group5-&gt;<a class="code" href="classaffx_1_1File5__Group.html#ab294feebc2fdab56881e2fb01619099b" title="Is there an object (dataset,group,link,etc...) with this name? ///.">name_exists</a>(<span class="stringliteral">&quot;FLDCutoff3peaks&quot;</span>) || !group5-&gt;<a class="code" href="classaffx_1_1File5__Group.html#ab294feebc2fdab56881e2fb01619099b" title="Is there an object (dataset,group,link,etc...) with this name? ///.">name_exists</a>(<span class="stringliteral">&quot;FLDCutoff4peaks&quot;</span>)) {
<a name="l00141"></a>00141                 <span class="keywordflow">return</span> <span class="keyword">false</span>;
<a name="l00142"></a>00142             }
<a name="l00143"></a>00143 
<a name="l00144"></a>00144             <a class="code" href="classaffx_1_1File5__Tsv.html">affx::File5_Tsv</a>* tsv5;
<a name="l00145"></a>00145 
<a name="l00146"></a>00146             tsv5 = group5-&gt;openTsv(<span class="stringliteral">&quot;FLDCutoff3peaks&quot;</span>, affx::FILE5_OPEN);
<a name="l00147"></a>00147             <span class="keywordtype">double</span> dThreePeakFLD_X = 0.0;
<a name="l00148"></a>00148             <span class="keywordtype">double</span> dThreePeakFLD_Y = 0.0;
<a name="l00149"></a>00149             <span class="keywordflow">while</span> (tsv5-&gt;nextLine() == affx::FILE5_OK)
<a name="l00150"></a>00150             {
<a name="l00151"></a>00151                 tsv5-&gt;get(0, 0, &amp;dThreePeakFLD_X);  m_vThreePeakFLD_X.push_back(dThreePeakFLD_X);
<a name="l00152"></a>00152                 tsv5-&gt;get(0, 1, &amp;dThreePeakFLD_Y);  m_vThreePeakFLD_Y.push_back(dThreePeakFLD_Y);
<a name="l00153"></a>00153             }
<a name="l00154"></a>00154             tsv5-&gt;close();
<a name="l00155"></a>00155             <span class="keyword">delete</span> tsv5;
<a name="l00156"></a>00156 
<a name="l00157"></a>00157             tsv5 = group5-&gt;openTsv(<span class="stringliteral">&quot;FLDCutoff4peaks&quot;</span>, affx::FILE5_OPEN);
<a name="l00158"></a>00158             <span class="keywordtype">double</span> dFourPeakFLD_X = 0.0;
<a name="l00159"></a>00159             <span class="keywordtype">double</span> dFourPeakFLD_Y = 0.0;
<a name="l00160"></a>00160             <span class="keywordflow">while</span> (tsv5-&gt;nextLine() == affx::FILE5_OK)
<a name="l00161"></a>00161             {
<a name="l00162"></a>00162                 tsv5-&gt;get(0, 0, &amp;dFourPeakFLD_X);  m_vFourPeakFLD_X.push_back(dFourPeakFLD_X);
<a name="l00163"></a>00163                 tsv5-&gt;get(0, 1, &amp;dFourPeakFLD_Y);  m_vFourPeakFLD_Y.push_back(dFourPeakFLD_Y);
<a name="l00164"></a>00164             }
<a name="l00165"></a>00165             tsv5-&gt;close();
<a name="l00166"></a>00166             <span class="keyword">delete</span> tsv5;
<a name="l00167"></a>00167 
<a name="l00168"></a>00168             <span class="keywordflow">if</span>(!group5-&gt;<a class="code" href="classaffx_1_1File5__Group.html#ab294feebc2fdab56881e2fb01619099b" title="Is there an object (dataset,group,link,etc...) with this name? ///.">name_exists</a>(<span class="stringliteral">&quot;ShrinkFactor3peaks&quot;</span>) || !group5-&gt;<a class="code" href="classaffx_1_1File5__Group.html#ab294feebc2fdab56881e2fb01619099b" title="Is there an object (dataset,group,link,etc...) with this name? ///.">name_exists</a>(<span class="stringliteral">&quot;ShrinkFactor4peaks&quot;</span>)) {
<a name="l00169"></a>00169                 <span class="keywordflow">return</span> <span class="keyword">false</span>;
<a name="l00170"></a>00170             }
<a name="l00171"></a>00171 
<a name="l00172"></a>00172             tsv5 = group5-&gt;openTsv(<span class="stringliteral">&quot;ShrinkFactor3peaks&quot;</span>, affx::FILE5_OPEN);
<a name="l00173"></a>00173             <span class="keywordtype">double</span> dThreePeakShrink_X = 0.0;
<a name="l00174"></a>00174             <span class="keywordtype">double</span> dThreePeakShrink_Y = 0.0;
<a name="l00175"></a>00175             <span class="keywordflow">while</span> (tsv5-&gt;nextLine() == affx::FILE5_OK)
<a name="l00176"></a>00176             {
<a name="l00177"></a>00177                 tsv5-&gt;get(0, 0, &amp;dThreePeakShrink_X);  m_vThreePeakShrink_X.push_back(dThreePeakShrink_X);
<a name="l00178"></a>00178                 tsv5-&gt;get(0, 1, &amp;dThreePeakShrink_Y);  m_vThreePeakShrink_Y.push_back(dThreePeakShrink_Y);
<a name="l00179"></a>00179             }
<a name="l00180"></a>00180             tsv5-&gt;close();
<a name="l00181"></a>00181             <span class="keyword">delete</span> tsv5;
<a name="l00182"></a>00182 
<a name="l00183"></a>00183             tsv5 = group5-&gt;openTsv(<span class="stringliteral">&quot;ShrinkFactor4peaks&quot;</span>, affx::FILE5_OPEN);
<a name="l00184"></a>00184             <span class="keywordtype">double</span> dFourPeakShrink_X = 0.0;
<a name="l00185"></a>00185             <span class="keywordtype">double</span> dFourPeakShrink_Y = 0.0;
<a name="l00186"></a>00186             <span class="keywordflow">while</span> (tsv5-&gt;nextLine() == affx::FILE5_OK)
<a name="l00187"></a>00187             {
<a name="l00188"></a>00188                 tsv5-&gt;get(0, 0, &amp;dFourPeakShrink_X);  m_vFourPeakShrink_X.push_back(dFourPeakShrink_X);
<a name="l00189"></a>00189                 tsv5-&gt;get(0, 1, &amp;dFourPeakShrink_Y);  m_vFourPeakShrink_Y.push_back(dFourPeakShrink_Y);
<a name="l00190"></a>00190             }
<a name="l00191"></a>00191             tsv5-&gt;close();
<a name="l00192"></a>00192             <span class="keyword">delete</span> tsv5;
<a name="l00193"></a>00193 
<a name="l00194"></a>00194             group5-&gt;close();
<a name="l00195"></a>00195             <span class="keyword">delete</span> group5;
<a name="l00196"></a>00196             file5.close();
<a name="l00197"></a>00197             bSuccessful = <span class="keyword">true</span>;
<a name="l00198"></a>00198 
<a name="l00199"></a>00199         } <span class="keywordflow">catch</span>(...) {<span class="keywordflow">throw</span>(<a class="code" href="classExcept.html" title="General purpose exception error for error handling.">Except</a>(<span class="stringliteral">&quot;Cannot open file: &quot;</span> + strFileName));}
<a name="l00200"></a>00200     }
<a name="l00201"></a>00201     <span class="keywordflow">return</span> bSuccessful;
<a name="l00202"></a>00202 }
<a name="l00203"></a>00203 
<a name="l00204"></a>00204 
<a name="l00205"></a>00205 <span class="keywordtype">void</span> CNAnalysisMethodAllelePeaks::filterCutOff( <a class="code" href="classCNProbeSetArray.html" title="A vector of ProbeSet Pointers.">CNProbeSetArray</a>* probeSets,
<a name="l00206"></a>00206                                                 <span class="keywordtype">int</span> chrStart,
<a name="l00207"></a>00207                                                 <span class="keywordtype">int</span> chrEnd,
<a name="l00208"></a>00208                                                 <span class="keywordtype">float</span> cutoff3,
<a name="l00209"></a>00209                                                 <span class="keywordtype">float</span> cutoff4,
<a name="l00210"></a>00210                                                 vector&lt;bool&gt;&amp; filteredProbeSets
<a name="l00211"></a>00211                                                 )
<a name="l00212"></a>00212 {
<a name="l00213"></a>00213     <span class="comment">// Filter based on the FLD cutoff functions</span>
<a name="l00214"></a>00214     <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = chrStart; i &lt;= chrEnd; i++) {
<a name="l00215"></a>00215       <span class="keywordflow">if</span> (((probeSets-&gt;getAt(i)-&gt;getMaxPeaks() &lt;= 3)  &amp;&amp;
<a name="l00216"></a>00216            (probeSets-&gt;getAt(i)-&gt;getFLD() &gt; cutoff3))  ||
<a name="l00217"></a>00217           ((probeSets-&gt;getAt(i)-&gt;getMaxPeaks() &gt;= 4)  &amp;&amp;
<a name="l00218"></a>00218            (probeSets-&gt;getAt(i)-&gt;getFLD() &gt; cutoff4)))
<a name="l00219"></a>00219         {
<a name="l00220"></a>00220             filteredProbeSets[i] = <span class="keyword">true</span>;
<a name="l00221"></a>00221         }
<a name="l00222"></a>00222     }
<a name="l00223"></a>00223 }
<a name="l00224"></a>00224 
<a name="l00225"></a>00225 <span class="keywordtype">void</span> CNAnalysisMethodAllelePeaks::computeSNPQC(<a class="code" href="classCNExperiment.html" title="A class for storing CNExperiment data.">CNExperiment</a>&amp; objExperiment, <a class="code" href="classCNProbeSetArray.html" title="A vector of ProbeSet Pointers.">CNProbeSetArray</a>&amp; vProbeSets)
<a name="l00226"></a>00226 {
<a name="l00227"></a>00227     <span class="comment">// Calculate SNPQC</span>
<a name="l00228"></a>00228     <span class="keywordflow">if</span> (!m_pEngine-&gt;<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;cyto2&quot;</span>)) {<span class="keywordflow">return</span>;}
<a name="l00229"></a>00229     <span class="keywordflow">if</span> (!m_pEngine-&gt;<a class="code" href="classOptions.html#a400b67224caf01af079f1709e0e868f4" title="Check to see if an option is defined.">isOptDefined</a>(<span class="stringliteral">&quot;snp-qc-use-contrast&quot;</span>)) {<span class="keywordflow">return</span>;}
<a name="l00230"></a>00230     <span class="keywordflow">if</span> (!m_pEngine-&gt;<a class="code" href="classOptions.html#a400b67224caf01af079f1709e0e868f4" title="Check to see if an option is defined.">isOptDefined</a>(<span class="stringliteral">&quot;snp-qc-k&quot;</span>)) {<span class="keywordflow">return</span>;}
<a name="l00231"></a>00231     <span class="keywordflow">if</span> (!m_pEngine-&gt;<a class="code" href="classOptions.html#a400b67224caf01af079f1709e0e868f4" title="Check to see if an option is defined.">isOptDefined</a>(<span class="stringliteral">&quot;snp-qc-em-threshold&quot;</span>)) {<span class="keywordflow">return</span>;}
<a name="l00232"></a>00232     <span class="keywordflow">if</span> (!m_pEngine-&gt;<a class="code" href="classOptions.html#a400b67224caf01af079f1709e0e868f4" title="Check to see if an option is defined.">isOptDefined</a>(<span class="stringliteral">&quot;snp-qc-bin-size&quot;</span>)) {<span class="keywordflow">return</span>;}
<a name="l00233"></a>00233     <a class="code" href="classAffxArray.html">AffxArray&lt;AffxString&gt;</a> arSNPIds;
<a name="l00234"></a>00234     <span class="keywordflow">if</span> ((m_pEngine-&gt;<a class="code" href="classOptions.html#a400b67224caf01af079f1709e0e868f4" title="Check to see if an option is defined.">isOptDefined</a>(<span class="stringliteral">&quot;snp-qc-snp-list&quot;</span>)) &amp;&amp; (m_pEngine-&gt;<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;snp-qc-snp-list&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>))
<a name="l00235"></a>00235     {
<a name="l00236"></a>00236       <a class="code" href="classaffx_1_1TsvFile.html" title="A class for reading and writing Tab Seperated Value (TSV) files. /// See the TsvFile format document ...">affx::TsvFile</a> tsv;
<a name="l00237"></a>00237       std::string snp;
<a name="l00238"></a>00238       tsv.<a class="code" href="classaffx_1_1TsvFile.html#a0b08c1eea21d4b6a893f57df16521a46" title="remove whitespace from value?">m_optAutoTrim</a> = <span class="keyword">true</span>;
<a name="l00239"></a>00239       <span class="keywordflow">if</span> (tsv.<a class="code" href="classaffx_1_1TsvFile.html#a28ad46dc5cadab63edc56cb5f36a0efe" title="Opens a file as a table.">openTable</a>(m_pEngine-&gt;<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;snp-qc-snp-list&quot;</span>)) == <a class="code" href="namespaceaffx.html#ad00847b6f358e7269f493808ee6bd395a05b4d7241e83bd24a318f4caa49b284e" title="No Error.">affx::TSV_OK</a>) {
<a name="l00240"></a>00240         <span class="keywordflow">while</span>( tsv.<a class="code" href="classaffx_1_1TsvFile.html#a2022895d082d1d635e7f59fccab12afb" title="Skip to the next level of data which matches seek_clvl ///.">nextLevel</a>(0) == affx::TSV_OK ) {
<a name="l00241"></a>00241           tsv.<a class="code" href="classaffx_1_1TsvFile.html#a7a81092ce134e8c9116a1a5bcfe287d1" title="Get the string value of a field ///.">get</a>(0,0, snp);
<a name="l00242"></a>00242           arSNPIds.add(<span class="keyword">new</span> <a class="code" href="classAffxString.html" title="A class derived from std::string.">AffxString</a>(snp) );
<a name="l00243"></a>00243         }
<a name="l00244"></a>00244         tsv.<a class="code" href="classaffx_1_1TsvFile.html#a6a397412f48d477d0e87b00f914d06f7" title="Closes the file and clears bindings and other info.">clear</a>();
<a name="l00245"></a>00245       }
<a name="l00246"></a>00246     }
<a name="l00247"></a>00247     arSNPIds.quickSort(0);
<a name="l00248"></a>00248     std::vector&lt;double&gt; vData;
<a name="l00249"></a>00249 
<a name="l00250"></a>00250     <span class="keywordflow">if</span> (m_pEngine-&gt;<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;snp-qc-use-contrast&quot;</span>)) {
<a name="l00251"></a>00251         <span class="keywordflow">if</span> (arSNPIds.empty()) {
<a name="l00252"></a>00252             <span class="keywordflow">for</span> (<span class="keywordtype">int</span> iIndex = 0; iIndex &lt; vProbeSets.getCount(); iIndex++) {
<a name="l00253"></a>00253                 <a class="code" href="classCNProbeSet.html" title="A class to store probe set data.">CNProbeSet</a>* pobjProbeSet = vProbeSets.at(iIndex);
<a name="l00254"></a>00254                 <span class="keywordflow">if</span> (pobjProbeSet-&gt;processAsSNP()) {
<a name="l00255"></a>00255                     vData.push_back(pobjProbeSet-&gt;getIntensityContrast(m_pEngine-&gt;<a class="code" href="classOptions.html#a8b119de0194389e5e91d09520df211d9" title="Get the integer value of an option.">getOptDouble</a>(<span class="stringliteral">&quot;snp-qc-k&quot;</span>)));
<a name="l00256"></a>00256                 }
<a name="l00257"></a>00257             }
<a name="l00258"></a>00258         } <span class="keywordflow">else</span> {
<a name="l00259"></a>00259             <span class="keywordflow">for</span> (<span class="keywordtype">int</span> iIndex = 0; iIndex &lt; vProbeSets.getCount(); iIndex++) {
<a name="l00260"></a>00260                 <a class="code" href="classCNProbeSet.html" title="A class to store probe set data.">CNProbeSet</a>* pobjProbeSet = vProbeSets.at(iIndex);
<a name="l00261"></a>00261                 <span class="keywordflow">if</span> (pobjProbeSet-&gt;processAsSNP()) {
<a name="l00262"></a>00262                     <a class="code" href="classAffxString.html" title="A class derived from std::string.">AffxString</a> str = pobjProbeSet-&gt;getProbeSetName();
<a name="l00263"></a>00263                     <span class="keywordflow">if</span> (arSNPIds.binarySearch(str, 0) != -1) {
<a name="l00264"></a>00264                         vData.push_back(pobjProbeSet-&gt;getIntensityContrast(m_pEngine-&gt;<a class="code" href="classOptions.html#a8b119de0194389e5e91d09520df211d9" title="Get the integer value of an option.">getOptDouble</a>(<span class="stringliteral">&quot;snp-qc-k&quot;</span>)));
<a name="l00265"></a>00265                     }
<a name="l00266"></a>00266                 }
<a name="l00267"></a>00267             }
<a name="l00268"></a>00268         }
<a name="l00269"></a>00269     } <span class="keywordflow">else</span> {
<a name="l00270"></a>00270         <span class="keywordflow">if</span> (arSNPIds.empty()) {
<a name="l00271"></a>00271             <span class="keywordflow">for</span> (<span class="keywordtype">int</span> iIndex = 0; iIndex &lt; vProbeSets.getCount(); iIndex++) {
<a name="l00272"></a>00272                 <a class="code" href="classCNProbeSet.html" title="A class to store probe set data.">CNProbeSet</a>* pobjProbeSet = vProbeSets.at(iIndex);
<a name="l00273"></a>00273                 <span class="keywordflow">if</span> (pobjProbeSet-&gt;getInformation() != 0.0 &amp;&amp; pobjProbeSet-&gt;processAsSNP()) {
<a name="l00274"></a>00274                     vData.push_back(pobjProbeSet-&gt;getSCAR());
<a name="l00275"></a>00275                 }
<a name="l00276"></a>00276             }
<a name="l00277"></a>00277         } <span class="keywordflow">else</span> {
<a name="l00278"></a>00278             <span class="keywordflow">for</span> (<span class="keywordtype">int</span> iIndex = 0; iIndex &lt; vProbeSets.getCount(); iIndex++) {
<a name="l00279"></a>00279                 <a class="code" href="classCNProbeSet.html" title="A class to store probe set data.">CNProbeSet</a>* pobjProbeSet = vProbeSets.at(iIndex);
<a name="l00280"></a>00280                 <span class="keywordflow">if</span> (pobjProbeSet-&gt;getInformation() != 0.0 &amp;&amp; pobjProbeSet-&gt;processAsSNP()) {
<a name="l00281"></a>00281                     <a class="code" href="classAffxString.html" title="A class derived from std::string.">AffxString</a> str = pobjProbeSet-&gt;getProbeSetName();
<a name="l00282"></a>00282                     <span class="keywordflow">if</span> (arSNPIds.binarySearch(str, 0) != -1) {
<a name="l00283"></a>00283                         vData.push_back(pobjProbeSet-&gt;getSCAR());
<a name="l00284"></a>00284                     }
<a name="l00285"></a>00285                 }
<a name="l00286"></a>00286             }
<a name="l00287"></a>00287         }
<a name="l00288"></a>00288     }
<a name="l00289"></a>00289     arSNPIds.deleteAll();
<a name="l00290"></a>00290     <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; (i &lt; vData.size()); i++)
<a name="l00291"></a>00291     {
<a name="l00292"></a>00292         <span class="keywordflow">if</span> (vData[i] != vData[i])
<a name="l00293"></a>00293         {
<a name="l00294"></a>00294             vData[i] = 0;
<a name="l00295"></a>00295         }
<a name="l00296"></a>00296     }
<a name="l00297"></a>00297     objExperiment.setSNPQC(<a class="code" href="classHomHiLoCelListener.html#a66126b831b4a4027a7088f379e057101" title="Process another std::vector&amp;lt;double&amp;gt; worth of data.">HomHiLoCelListener::computeStatistic</a>(vData, m_pEngine-&gt;<a class="code" href="classOptions.html#a8b119de0194389e5e91d09520df211d9" title="Get the integer value of an option.">getOptDouble</a>(<span class="stringliteral">&quot;snp-qc-em-threshold&quot;</span>), m_pEngine-&gt;<a class="code" href="classOptions.html#a8b119de0194389e5e91d09520df211d9" title="Get the integer value of an option.">getOptDouble</a>(<span class="stringliteral">&quot;snp-qc-bin-size&quot;</span>)));
<a name="l00298"></a>00298     objExperiment.setIsSNPQCset(<span class="keyword">true</span>);
<a name="l00299"></a>00299 
<a name="l00300"></a>00300     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;*&quot;</span>);
<a name="l00301"></a>00301     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;CNQC  (MAPD) = &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(objExperiment.getMadDiffCN()));
<a name="l00302"></a>00302     <span class="keywordflow">if</span> (m_pEngine-&gt;<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;snp-qc-use-contrast&quot;</span>))
<a name="l00303"></a>00303     {
<a name="l00304"></a>00304         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;SNPQC (Contrast) = &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(objExperiment.getSNPQC()));
<a name="l00305"></a>00305     }
<a name="l00306"></a>00306     <span class="keywordflow">else</span>
<a name="l00307"></a>00307     {
<a name="l00308"></a>00308         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;SNPQC (PVQC) = &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(objExperiment.getSNPQC()));
<a name="l00309"></a>00309     }
<a name="l00310"></a>00310     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;*&quot;</span>);
<a name="l00311"></a>00311 }
<a name="l00312"></a>00312 
<a name="l00313"></a>00313 <span class="comment"></span>
<a name="l00314"></a>00314 <span class="comment">/**</span>
<a name="l00315"></a>00315 <span class="comment"> * @brief Run the analysis.</span>
<a name="l00316"></a>00316 <span class="comment"> */</span>
<a name="l00317"></a><a class="code" href="classCNAnalysisMethodAllelePeaks.html#a8c67a1fd34862a9de4aa97cc98c3075d">00317</a> <span class="keywordtype">void</span> <a class="code" href="classCNAnalysisMethodAllelePeaks.html#a8c67a1fd34862a9de4aa97cc98c3075d" title="Run the analysis.">CNAnalysisMethodAllelePeaks::run</a>()
<a name="l00318"></a>00318 {
<a name="l00319"></a>00319     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;CNAnalysisMethodAllelePeaks::run(...) start&quot;</span>);
<a name="l00320"></a>00320     isSetup();
<a name="l00321"></a>00321 
<a name="l00322"></a>00322     resetAllelePeakInitialValues();
<a name="l00323"></a>00323     determineLocalProbeSets();
<a name="l00324"></a>00324 
<a name="l00325"></a>00325     <span class="keywordflow">if</span>(!m_bSNPAlreadyLoaded) {<span class="comment"></span>
<a name="l00326"></a>00326 <span class="comment">        ///@todo  Add error messaging if no snp-reference file is input. Also exit run gracefully.</span>
<a name="l00327"></a>00327 <span class="comment"></span>        <a class="code" href="classAffxString.html" title="A class derived from std::string.">AffxString</a> strFileName = m_pEngine-&gt;<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;snp-reference-input-file&quot;</span>);
<a name="l00328"></a>00328         loadSnpReferenceFile(strFileName);
<a name="l00329"></a>00329         m_bSNPAlreadyLoaded = <span class="keyword">true</span>;
<a name="l00330"></a>00330     }
<a name="l00331"></a>00331     <span class="comment">// Compute SNPQC</span>
<a name="l00332"></a>00332     <span class="keywordflow">if</span> (!getExperiment()-&gt;isSNPQCset()) {
<a name="l00333"></a>00333         computeSNPQC(*getExperiment(), *getProbeSets());
<a name="l00334"></a>00334     }
<a name="l00335"></a>00335     <span class="keywordtype">int</span> iStepSize = m_iStep;
<a name="l00336"></a>00336     <span class="keywordflow">if</span> (iStepSize &gt; m_iWindow) {
<a name="l00337"></a>00337         <a class="code" href="classVerbose.html#a34e1d936178741eeb78de93c5497e1b8" title="Print a warning message.">Verbose::warn</a>(0, <span class="stringliteral">&quot;Step size (&quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(m_iStep) + <span class="stringliteral">&quot;) exceeds window size (&quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(m_iWindow) +
<a name="l00338"></a>00338                       <span class="stringliteral">&quot;). Using window size instead.&quot;</span>);
<a name="l00339"></a>00339         iStepSize = m_iWindow;
<a name="l00340"></a>00340     }
<a name="l00341"></a>00341     fillSnpChrBounds();
<a name="l00342"></a>00342 
<a name="l00343"></a>00343     shrinkToPeaks(getProbeSets());
<a name="l00344"></a>00344 
<a name="l00345"></a>00345     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;CNAnalysisMethodAllelePeaks::run(...) end&quot;</span>);
<a name="l00346"></a>00346 }
<a name="l00347"></a>00347 
<a name="l00348"></a>00348 <span class="comment"></span>
<a name="l00349"></a>00349 <span class="comment">/**</span>
<a name="l00350"></a>00350 <span class="comment">  * Find chromosome bounds for SNP probe sets</span>
<a name="l00351"></a>00351 <span class="comment">  */</span>
<a name="l00352"></a>00352 <span class="keywordtype">void</span> CNAnalysisMethodAllelePeaks::fillSnpChrBounds()
<a name="l00353"></a>00353 {
<a name="l00354"></a>00354     fillChrBoundsImpl(getProbeSets(), m_chrBounds);
<a name="l00355"></a>00355 }
<a name="l00356"></a>00356 
<a name="l00357"></a>00357 
<a name="l00358"></a>00358 <span class="keywordtype">void</span> CNAnalysisMethodAllelePeaks::determineLocalProbeSets()
<a name="l00359"></a>00359 {
<a name="l00360"></a>00360     <span class="keywordflow">if</span>(m_bLocalProbeSetsDetermined)
<a name="l00361"></a>00361     {
<a name="l00362"></a>00362             <span class="keywordflow">return</span>;
<a name="l00363"></a>00363     }
<a name="l00364"></a>00364     <span class="keyword">const</span> <span class="keywordtype">bool</span> isCytoScanHD = m_pEngine-&gt;<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;cytoscan-hd&quot;</span>);
<a name="l00365"></a>00365     <span class="keyword">const</span> <span class="keywordtype">bool</span> isCyto2 = m_pEngine-&gt;<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;cyto2&quot;</span>);
<a name="l00366"></a>00366 
<a name="l00367"></a>00367     <span class="keywordtype">int</span> iNumberOfProbeSets=CNAnalysisMethod::getProbeSets()-&gt;getCount();
<a name="l00368"></a>00368     <span class="keywordflow">for</span> (<span class="keywordtype">int</span> iIndex = 0; iIndex&lt;iNumberOfProbeSets; iIndex++)
<a name="l00369"></a>00369     {
<a name="l00370"></a>00370         <a class="code" href="classCNProbeSet.html" title="A class to store probe set data.">CNProbeSet</a>* p = CNAnalysisMethod::getProbeSets()-&gt;getAt(iIndex);
<a name="l00371"></a>00371         <span class="keywordflow">if</span> (isCyto2 &amp;&amp; !p-&gt;getValidSCARExists())
<a name="l00372"></a>00372         {
<a name="l00373"></a>00373             <span class="keywordflow">continue</span>;
<a name="l00374"></a>00374         }
<a name="l00375"></a>00375                 <span class="keywordflow">if</span> (isCytoScanHD)
<a name="l00376"></a>00376                 {
<a name="l00377"></a>00377                         <span class="keywordflow">if</span>(p-&gt;processAsVisualization())
<a name="l00378"></a>00378                         {
<a name="l00379"></a>00379                                 getProbeSets()-&gt;add(p);
<a name="l00380"></a>00380                         }
<a name="l00381"></a>00381                 }
<a name="l00382"></a>00382                 <span class="keywordflow">else</span>
<a name="l00383"></a>00383                 {
<a name="l00384"></a>00384                         <span class="keywordflow">if</span>(p-&gt;processAsSNP())
<a name="l00385"></a>00385                         {
<a name="l00386"></a>00386                                 getProbeSets()-&gt;add(p);
<a name="l00387"></a>00387                         }
<a name="l00388"></a>00388                 }
<a name="l00389"></a>00389     }
<a name="l00390"></a>00390     m_bLocalProbeSetsDetermined=<span class="keyword">true</span>;
<a name="l00391"></a>00391 }
<a name="l00392"></a>00392 
<a name="l00393"></a>00393 
<a name="l00394"></a>00394 <a class="code" href="classCNProbeSetArray.html" title="A vector of ProbeSet Pointers.">CNProbeSetArray</a>* CNAnalysisMethodAllelePeaks::getProbeSets()
<a name="l00395"></a>00395 {
<a name="l00396"></a>00396         <span class="keywordflow">return</span> &amp;m_vLocalProbeSets;
<a name="l00397"></a>00397 }
<a name="l00398"></a>00398 
<a name="l00399"></a>00399 
<a name="l00400"></a>00400 <span class="keywordtype">void</span> CNAnalysisMethodAllelePeaks::filterNoMansLand(
<a name="l00401"></a>00401                                             <a class="code" href="classCNProbeSetArray.html" title="A vector of ProbeSet Pointers.">CNProbeSetArray</a>* probeSets,
<a name="l00402"></a>00402                                             vector&lt;bool&gt;&amp; filteredProbeSets,
<a name="l00403"></a>00403                                             <span class="keyword">const</span> vector&lt;pair&lt;int, int&gt; &gt;&amp; stepWindowBds,
<a name="l00404"></a>00404                                             <span class="keyword">const</span> vector&lt;vector&lt;double&gt; &gt;&amp; peaks
<a name="l00405"></a>00405                                             )
<a name="l00406"></a>00406 {
<a name="l00407"></a>00407     filterNoMansLandImpl&lt;CNAnalysisMethodAllelePeaks&gt;(
<a name="l00408"></a>00408                                                 probeSets,
<a name="l00409"></a>00409                                                 filteredProbeSets,
<a name="l00410"></a>00410                                                 stepWindowBds,
<a name="l00411"></a>00411                                                 peaks
<a name="l00412"></a>00412                                                 );
<a name="l00413"></a>00413 }
<a name="l00414"></a>00414 
<a name="l00415"></a>00415 
<a name="l00416"></a>00416 <span class="keywordtype">void</span> CNAnalysisMethodAllelePeaks::filterShrinkTowardPeaks(
<a name="l00417"></a>00417                                             <a class="code" href="classCNProbeSetArray.html" title="A vector of ProbeSet Pointers.">CNProbeSetArray</a>* probeSets,
<a name="l00418"></a>00418                                             vector&lt;bool&gt;&amp; filteredProbeSets,
<a name="l00419"></a>00419                                             <span class="keyword">const</span> vector&lt;pair&lt;int, int&gt; &gt;&amp; stepWindowBds,
<a name="l00420"></a>00420                                             <span class="keyword">const</span> vector&lt;vector&lt;double&gt; &gt;&amp; peaks,
<a name="l00421"></a>00421                                             <span class="keywordtype">float</span> shrinkFactor3,
<a name="l00422"></a>00422                                             <span class="keywordtype">float</span> shrinkFactor4,
<a name="l00423"></a>00423                                             vector&lt;pair&lt;int, double&gt; &gt;&amp; closestPeak,
<a name="l00424"></a>00424                                             vector&lt;pair&lt;int, double&gt; &gt;&amp; processedValues,
<a name="l00425"></a>00425                                             <span class="keywordtype">bool</span> saveAllelePeaksFlag
<a name="l00426"></a>00426                                             )
<a name="l00427"></a>00427 {
<a name="l00428"></a>00428     filterShrinkTowardPeaksImpl&lt;CNAnalysisMethodAllelePeaks&gt;(
<a name="l00429"></a>00429                                                 probeSets,
<a name="l00430"></a>00430                                                 filteredProbeSets,
<a name="l00431"></a>00431                                                 stepWindowBds,
<a name="l00432"></a>00432                                                 peaks,
<a name="l00433"></a>00433                                                 shrinkFactor3,
<a name="l00434"></a>00434                                                 shrinkFactor4,
<a name="l00435"></a>00435                                                 closestPeak,
<a name="l00436"></a>00436                                                 processedValues,
<a name="l00437"></a>00437                                                 saveAllelePeaksFlag
<a name="l00438"></a>00438                                                 );
<a name="l00439"></a>00439 }
</pre></div></div>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:38 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
