<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: chipstream/ Directory Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="navpath">
    <ul>
      <li><a class="el" href="dir_f48957cb46a44244b1a86c58bd157ee8.html">chipstream</a>      </li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>chipstream Directory Reference</h1>  </div>
</div>
<div class="contents">
<table class="memberdecls">
<tr><td colspan="2"><h2><a name="subdirs"></a>
Directories</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_46738a0cf4b69b2f6eb4c11fdc1c04f4.html">apt-cdf-to-spf</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_cb3c961afe8ab9349d3cdc29f846e7b5.html">apt-cel-extract</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_681243c61bc271b14d95d9e1fae8024a.html">apt-cel-transformer</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_032d72cd7a749843e32596a3bd5fa6c1.html">apt-convert-priors</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_f072a6fd7a0b8f959160dcf3ee1a4440.html">apt-convert-spf</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_1d819e5957f521c72c2159e489f328a9.html">apt-data-step</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_574face15954922462dd987847f94809.html">apt-data-subset</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_ad67da846e1ef490485c3aa44bf9fad4.html">apt-export-impute</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_f7cff11851daf76703b64f8466c3021f.html">apt-geno-qc</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_6850d6a3b3c57d21532387521e20fa7c.html">apt-genotype-report</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_750765ddaafbe3f0cf22bcddc8bf919e.html">apt-genotype-translation</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_303b3419630c07f7f9b92fd5094d27cc.html">apt-impute</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_cd0180e6efafb8a914b55f9858472dad.html">apt-list-summary</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_c15acdb83ac636cc0d5aa24c168c6d88.html">apt-probeset-genotype</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_3102b3045564bcc54ee8e95fd19e9539.html">apt-probeset-summarize</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_560b4ce93f4e15d9ec02a919d0aa22e0.html">apt-sample-snp-list</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_8a8b4897b72e48bc657a980c29739dbf.html">apt-snp-compare</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_b8a75f4e2e3edd382a33de1f8595553e.html">apt-snp-model-converter</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_45371eb9318bb5ce1719f4b19ab3e1d0.html">apt-snp-summary</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_89b9ddcb07e471fe0272bc2e39323086.html">apt-summary-genotype</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_c4932328f18bf4db5e1fdd4bff2a351e.html">apt-summary-normalization</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">directory &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="dir_5052177d5827f91b960bd4f67876fdba.html">example</a></td></tr>
<tr><td colspan="2"><h2><a name="files"></a>
Files</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="AdapterTypeNormTran_8cpp.html">AdapterTypeNormTran.cpp</a> <a href="AdapterTypeNormTran_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing adapter type normalization. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="AdapterTypeNormTran_8h.html">AdapterTypeNormTran.h</a> <a href="AdapterTypeNormTran_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing adapter type normalization. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>AdjHomHiLoCelListener.cpp</b> <a href="AdjHomHiLoCelListener_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>AdjHomHiLoCelListener.h</b> <a href="AdjHomHiLoCelListener_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="AnalysisInfo_8h.html">AnalysisInfo.h</a> <a href="AnalysisInfo_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Information about a particular analysis run. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="AnalysisStream_8cpp.html">AnalysisStream.cpp</a> <a href="AnalysisStream_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="AnalysisStream_8h.html">AnalysisStream.h</a> <a href="AnalysisStream_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Object for managing chipstreams, perfect match adjusters and summarization methods. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>AnalysisStreamExpPcaSel.cpp</b> <a href="AnalysisStreamExpPcaSel_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="AnalysisStreamExpPcaSel_8h.html">AnalysisStreamExpPcaSel.h</a> <a href="AnalysisStreamExpPcaSel_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>An analysis stream that takes the expression data and does a pca selection on the probes before passing them to the quantification method. Inspired by Jim's Corimbia algorithm. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>AnalysisStreamExpression.cpp</b> <a href="AnalysisStreamExpression_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="AnalysisStreamExpression_8h.html">AnalysisStreamExpression.h</a> <a href="AnalysisStreamExpression_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p><a class="el" href="classAnalysisStream.html" title="Utility class for centralizing analysis pathway.">AnalysisStream</a> class specialized for quantification methods that are used for mRNA expression genechips like the ubiquitous U133 human chip. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="AnalysisStreamExpSpectSel_8cpp.html">AnalysisStreamExpSpectSel.cpp</a> <a href="AnalysisStreamExpSpectSel_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="AnalysisStreamExpSpectSel_8h.html">AnalysisStreamExpSpectSel.h</a> <a href="AnalysisStreamExpSpectSel_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p><a class="el" href="classAnalysisStream.html" title="Utility class for centralizing analysis pathway.">AnalysisStream</a> that selects features to cluster based on spectral clustering. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="AnalysisStreamFactory_8cpp.html">AnalysisStreamFactory.cpp</a> <a href="AnalysisStreamFactory_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Object for making new analysis streams. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="AnalysisStreamFactory_8h.html">AnalysisStreamFactory.h</a> <a href="AnalysisStreamFactory_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Object for making new analysis streams. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="AnalysisStreamGType_8cpp.html">AnalysisStreamGType.cpp</a> <a href="AnalysisStreamGType_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p><a class="el" href="classAnalysisStream.html" title="Utility class for centralizing analysis pathway.">AnalysisStream</a> class specialized for quantification methods that are used for determining genotype on SNP chips. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="AnalysisStreamGType_8h.html">AnalysisStreamGType.h</a> <a href="AnalysisStreamGType_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p><a class="el" href="classAnalysisStream.html" title="Utility class for centralizing analysis pathway.">AnalysisStream</a> class specialized for quantification methods that are used for determining genotype on SNP chips. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="AptTypes_8h.html">AptTypes.h</a> <a href="AptTypes_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>These types are common in the APT code. Collect them here for use by the rest of the code. No classes or functions here, just types. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="ArtifactReduction_8cpp.html">ArtifactReduction.cpp</a> <a href="ArtifactReduction_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing normalization. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="ArtifactReduction_8h.html">ArtifactReduction.h</a> <a href="ArtifactReduction_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing normalization. Can do sketch and full quantile (just set sketch to chip size) and supports bioconductor compatibility. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="BioTypes_8cpp.html">BioTypes.cpp</a> <a href="BioTypes_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>/// </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="BioTypes_8h.html">BioTypes.h</a> <a href="BioTypes_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Central location enumerations and other types that aren't particular to a single class or application. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>CelListener.h</b> <a href="CelListener_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>CelReader.cpp</b> <a href="CelReader_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="CelReader_8h.html">CelReader.h</a> <a href="CelReader_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for reading cel files and passing the data to memory representation and analysis objects. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>CelStatListener.h</b> <a href="CelStatListener_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>ChipLayout.cpp</b> <a href="ChipLayout_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>ChipLayout.h</b> <a href="ChipLayout_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="ChipStream_8cpp.html">ChipStream.cpp</a> <a href="ChipStream_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Base class for objects that can initialize themselves one at a time and transform intensity data. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="ChipStream_8h.html">ChipStream.h</a> <a href="ChipStream_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Base class for objects that can initialize themselves one at a time and transform intensity data. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>ChipStreamDataTransform.cpp</b> <a href="ChipStreamDataTransform_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="ChipStreamDataTransform_8h.html">ChipStreamDataTransform.h</a> <a href="ChipStreamDataTransform_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Wrapper for <a class="el" href="classChipStream.html" title="ChipStream - base class for objects that can initialize themselves one at a time and transform intens...">ChipStream</a> objects that makes them into <a class="el" href="classDataTransform.html" title="Base class for objects that will manipulate data and produce a new set.">DataTransform</a> objects. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="ChipStreamFactory_8cpp.html">ChipStreamFactory.cpp</a> <a href="ChipStreamFactory_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="ChipStreamFactory_8h.html">ChipStreamFactory.h</a> <a href="ChipStreamFactory_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Factory class for making chip streams based on a string representation. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>ChipSummary.cpp</b> <a href="ChipSummary_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>ChipSummary.h</b> <a href="ChipSummary_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>ClusterZ.cpp</b> <a href="ClusterZ_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="ClusterZ_8h.html">ClusterZ.h</a> <a href="ClusterZ_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>CnProbeGenderCelListener.cpp</b> <a href="CnProbeGenderCelListener_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>CnProbeGenderCelListener.h</b> <a href="CnProbeGenderCelListener_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>CumulativeStats.h</b> <a href="CumulativeStats_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>DataStore.cpp</b> <a href="DataStore_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>DataStore.h</b> <a href="DataStore_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="DataTransform_8h.html">DataTransform.h</a> <a href="DataTransform_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Base class for objects that will manipulate data and produce a new set. For example statistics might be calculated, background might be subtracted, normalization might be performed, or data might be reordered. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="DiskIntensityMart_8cpp.html">DiskIntensityMart.cpp</a> <a href="DiskIntensityMart_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for encapsulating microarray data and storing it on disk in a reasonable way to reduce memory impact. Key idea is to reorganize the data on disk in the order which it will be used by the application to avoid lots of seeks. Hopefully with a small number of seeks the performance impact of on-disk stoarage will be minimal. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="DiskIntensityMart_8h.html">DiskIntensityMart.h</a> <a href="DiskIntensityMart_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for encapsulating microarray data and storing it on disk in a reasonable way to reduce memory impact. Key idea is to reorganize the data on disk in the order which it will be used by the application to avoid lots of seeks. Hopefully with a small number of seeks the performance impact of on-disk stoarage will be minimal. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="DmListener_8cpp.html">DmListener.cpp</a> <a href="DmListener_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for using DM calls as cell files become available. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="DmListener_8h.html">DmListener.h</a> <a href="DmListener_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for using DM calls as cell files become available. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>DTFactory.cpp</b> <a href="DTFactory_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="DTFactory_8h.html">DTFactory.h</a> <a href="DTFactory_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Wrapper class for all of our <a class="el" href="classDataTransform.html" title="Base class for objects that will manipulate data and produce a new set.">DataTransform</a> factories. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>EmGenderCelListener.cpp</b> <a href="EmGenderCelListener_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>EmGenderCelListener.h</b> <a href="EmGenderCelListener_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="EngineUtil_8cpp.html">EngineUtil.cpp</a> <a href="EngineUtil_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Some common functions that application executables make use of. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="EngineUtil_8h.html">EngineUtil.h</a> <a href="EngineUtil_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Some common functions that application executables make use of. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>FileGenders.cpp</b> <a href="FileGenders_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>FileGenders.h</b> <a href="FileGenders_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>FileInbred.cpp</b> <a href="FileInbred_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>FileInbred.h</b> <a href="FileInbred_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>FileListFilterEngine.cpp</b> <a href="FileListFilterEngine_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="FileListFilterEngine_8h.html">FileListFilterEngine.h</a> <a href="FileListFilterEngine_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>FileRemovalEngine.cpp</b> <a href="FileRemovalEngine_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="FileRemovalEngine_8h.html">FileRemovalEngine.h</a> <a href="FileRemovalEngine_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="GcAdjust_8cpp.html">GcAdjust.cpp</a> <a href="GcAdjust_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing an adjustment based on the median intensity of probes with similar GC content. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="GcAdjust_8h.html">GcAdjust.h</a> <a href="GcAdjust_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing an adjustment based on the median intensity of probes with similar GC content. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="GcBg_8cpp.html">GcBg.cpp</a> <a href="GcBg_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing an background adjustment on all probes based on the median intensity of probes with similar GC content. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="GcBg_8h.html">GcBg.h</a> <a href="GcBg_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing an background adjustment on all probes based on the median intensity of probes with similar GC content. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>GenderCalls.h</b> <a href="GenderCalls_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>GenderCelListener.h</b> <a href="GenderCelListener_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="GenoSeed_8cpp.html">GenoSeed.cpp</a> <a href="GenoSeed_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Abstract interface for a genotyping oracle used for genotyping seeds in brlmm. Originally seed calls came from DM, but now they are a subject of much research. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="GenoSeed_8h.html">GenoSeed.h</a> <a href="GenoSeed_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Abstract interface for a genotyping oracle used for genotyping seeds in brlmm. Originally seed calls came from DM, but now they are a subject of much research. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="GenoSeedTxt_8cpp.html">GenoSeedTxt.cpp</a> <a href="GenoSeedTxt_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Implements interface for a genotyping oracle used for genotyping seeds in brlmm. In this case the genotype seed calls are simply read from a text file. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="GenoSeedTxt_8h.html">GenoSeedTxt.h</a> <a href="GenoSeedTxt_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Implements interface for a genotyping oracle used for genotyping seeds in brlmm. In this case the genotype seed calls are simply read from a text file. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>GenotypeInfo.h</b> <a href="GenotypeInfo_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="GenoUtility_8cpp.html">GenoUtility.cpp</a> <a href="GenoUtility_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="GenoUtility_8h.html">GenoUtility.h</a> <a href="GenoUtility_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>HomHiLoCelListener.cpp</b> <a href="HomHiLoCelListener_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>HomHiLoCelListener.h</b> <a href="HomHiLoCelListener_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>IdxGroup.cpp</b> <a href="IdxGroup_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="IdxGroup_8h.html">IdxGroup.h</a> <a href="IdxGroup_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Named partitions of numbers (presumably indexes). </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>InbredStatus.h</b> <a href="InbredStatus_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="IntensityMart_8cpp.html">IntensityMart.cpp</a> <a href="IntensityMart_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Base class for objects that will dispense data on a [feature,chip] level. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="IntensityMart_8h.html">IntensityMart.h</a> <a href="IntensityMart_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Base class for ojbects that will dispense data on a [feature,chip] level. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="IntensityReader_8h.html">IntensityReader.h</a> <a href="IntensityReader_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Interface for an object that reads data from disk into memory. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>IntensityReporter.cpp</b> <a href="IntensityReporter_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="IntensityReporter_8h.html">IntensityReporter.h</a> <a href="IntensityReporter_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing adapter type normalization. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="IntensityTransformer_8h.html">IntensityTransformer.h</a> <a href="IntensityTransformer_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Abstract base class for objects that transform intensity data. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>KitAOCelListener.cpp</b> <a href="KitAOCelListener_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>KitAOCelListener.h</b> <a href="KitAOCelListener_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>KitAODb.cpp</b> <a href="KitAODb_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>KitAODb.h</b> <a href="KitAODb_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>MatrixUtil.cpp</b> <a href="MatrixUtil_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>MatrixUtil.h</b> <a href="MatrixUtil_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="MedNormTran_8cpp.html">MedNormTran.cpp</a> <a href="MedNormTran_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing median, or average, normalization. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="MedNormTran_8h.html">MedNormTran.h</a> <a href="MedNormTran_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing median, or average, normalization. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>MetaProbeset.cpp</b> <a href="MetaProbeset_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>MetaProbeset.h</b> <a href="MetaProbeset_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="MmAdjust_8cpp.html">MmAdjust.cpp</a> <a href="MmAdjust_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for supplying the mismatch intensity as pm adjustment. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="MmAdjust_8h.html">MmAdjust.h</a> <a href="MmAdjust_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for supplying the mismatch intensity as pm adjustment. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>MultiChannelHomHiLoCelListener.cpp</b> <a href="MultiChannelHomHiLoCelListener_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>MultiChannelHomHiLoCelListener.h</b> <a href="MultiChannelHomHiLoCelListener_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>MultiChannelNonOverlapCelListener.cpp</b> <a href="MultiChannelNonOverlapCelListener_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>MultiChannelNonOverlapCelListener.h</b> <a href="MultiChannelNonOverlapCelListener_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="MultiQuantMethodListener_8cpp.html">MultiQuantMethodListener.cpp</a> <a href="MultiQuantMethodListener_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for reporting results from multiple quantification methods. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="MultiQuantMethodListener_8h.html">MultiQuantMethodListener.h</a> <a href="MultiQuantMethodListener_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for reporting results from multiple quantification methods. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="PmAdjuster_8h.html">PmAdjuster.h</a> <a href="PmAdjuster_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Interface for determining a change based on intensity of perfect match probe. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="PmAdjusterFactory_8cpp.html">PmAdjusterFactory.cpp</a> <a href="PmAdjusterFactory_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Factory class for making chip streams based on a string representation. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="PmAdjusterFactory_8h.html">PmAdjusterFactory.h</a> <a href="PmAdjusterFactory_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Factory class for making chip streams based on a string representation. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>PmAdjustTransform.cpp</b> <a href="PmAdjustTransform_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>PmAdjustTransform.h</b> <a href="PmAdjustTransform_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="PmOnlyAdjust_8h.html">PmOnlyAdjust.h</a> <a href="PmOnlyAdjust_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Pm adjustment class that does no modification. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="PmSumAdjust_8cpp.html">PmSumAdjust.cpp</a> <a href="PmSumAdjust_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing an additive adjustment based on a probe that may be hybridizing to another allele. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="PmSumAdjust_8h.html">PmSumAdjust.h</a> <a href="PmSumAdjust_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing an additive adjustment based on a probe that may be hybridizing to another allele. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>ProbeList.h</b> <a href="ProbeList_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>ProbeListFactory.cpp</b> <a href="ProbeListFactory_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>ProbeListFactory.h</b> <a href="ProbeListFactory_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>ProbeListStl.cpp</b> <a href="ProbeListStl_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="ProbeListStl_8h.html">ProbeListStl.h</a> <a href="ProbeListStl_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>/// </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="ProbeSet_8cpp.html">ProbeSet.cpp</a> <a href="ProbeSet_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Classes and functions for dealing with probe sets. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="ProbeSet_8h.html">ProbeSet.h</a> <a href="ProbeSet_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Container class for probe sets. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>PsBoard.cpp</b> <a href="PsBoard_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>PsBoard.h</b> <a href="PsBoard_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QCAnalysisOptions.cpp</b> <a href="QCAnalysisOptions_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QCAnalysisOptions.h</b> <a href="QCAnalysisOptions_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QCProbesetOptions.cpp</b> <a href="QCProbesetOptions_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QCProbesetOptions.h</b> <a href="QCProbesetOptions_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantAvgDiff_8cpp.html">QuantAvgDiff.cpp</a> <a href="QuantAvgDiff_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing MAS 4 average difference signal estimation. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantAvgDiff_8h.html">QuantAvgDiff.h</a> <a href="QuantAvgDiff_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing MAS 4 average difference signal estimation. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantBirdseed.cpp</b> <a href="QuantBirdseed_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantBirdseed.h</b> <a href="QuantBirdseed_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantBirdseedDev.cpp</b> <a href="QuantBirdseedDev_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantBirdseedDev.h</b> <a href="QuantBirdseedDev_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantBirdseedLegacy.cpp</b> <a href="QuantBirdseedLegacy_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantBirdseedLegacy.h</b> <a href="QuantBirdseedLegacy_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantBirdseedv1.cpp</b> <a href="QuantBirdseedv1_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantBirdseedv1.h</b> <a href="QuantBirdseedv1_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantBirdseedv2.cpp</b> <a href="QuantBirdseedv2_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantBirdseedv2.h</b> <a href="QuantBirdseedv2_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantBRLMM_8cpp.html">QuantBRLMM.cpp</a> <a href="QuantBRLMM_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantBRLMM_8h.html">QuantBRLMM.h</a> <a href="QuantBRLMM_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantDabg_8cpp.html">QuantDabg.cpp</a> <a href="QuantDabg_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing Detected Above BackGround (DABG) analysis. Idea is to use the intensity of probes in a probeset to calculate a p-value for the entire probeset. The individual intensities are compared to a set of background probes. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantDabg_8h.html">QuantDabg.h</a> <a href="QuantDabg_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing Detected Above BackGround (DABG) analysis. Idea is to use the intensity of probes in a probeset to calculate a p-value for the entire probeset. The individual intensities are compared to a set of background probes. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantDataTransform.h</b> <a href="QuantDataTransform_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantExprMethod_8h.html">QuantExprMethod.h</a> <a href="QuantExprMethod_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Interface for computing quantification summaries from PM intensities grouped into probe set groups. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantGTypeMethod_8cpp.html">QuantGTypeMethod.cpp</a> <a href="QuantGTypeMethod_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantGTypeMethod_8h.html">QuantGTypeMethod.h</a> <a href="QuantGTypeMethod_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Interface for methods computing SNP genotyping calls. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantIterPlier_8cpp.html">QuantIterPlier.cpp</a> <a href="QuantIterPlier_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing probe set quantification estimate by iteratively calling PLIER with the probes that best correlate with signal estimate. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantIterPlier_8h.html">QuantIterPlier.h</a> <a href="QuantIterPlier_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing probe set quantification estimate by iteratively calling PLIER with the probes that best correlate with signal estimate. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantLabelZ_8cpp.html">QuantLabelZ.cpp</a> <a href="QuantLabelZ_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantLabelZ_8h.html">QuantLabelZ.h</a> <a href="QuantLabelZ_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantLabelZIO.cpp</b> <a href="QuantLabelZIO_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantLabelZIO.h</b> <a href="QuantLabelZIO_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantLabelZMulti_8cpp.html">QuantLabelZMulti.cpp</a> <a href="QuantLabelZMulti_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantLabelZMulti_8h.html">QuantLabelZMulti.h</a> <a href="QuantLabelZMulti_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Interface for methods computing SNP genotyping calls. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMedian_8cpp.html">QuantMedian.cpp</a> <a href="QuantMedian_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing probe set quantification using median. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMedian_8h.html">QuantMedian.h</a> <a href="QuantMedian_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing probe set quantification using median as does MEDIAN. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethod_8cpp.html">QuantMethod.cpp</a> <a href="QuantMethod_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Abstract interface for a quantification method. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethod_8h.html">QuantMethod.h</a> <a href="QuantMethod_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Abstract interface for a quantification method. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodExprCCCHPReport_8cpp.html">QuantMethodExprCCCHPReport.cpp</a> <a href="QuantMethodExprCCCHPReport_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for reporting results of quantification methods. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodExprCCCHPReport_8h.html">QuantMethodExprCCCHPReport.h</a> <a href="QuantMethodExprCCCHPReport_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for reporting results of quantification methods. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodExprChipSummary_8cpp.html">QuantMethodExprChipSummary.cpp</a> <a href="QuantMethodExprChipSummary_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Summarize and output basic statistics on the performance of a particular summarization or detection method across a number of cel files. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodExprChipSummary_8h.html">QuantMethodExprChipSummary.h</a> <a href="QuantMethodExprChipSummary_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Summarize basic statistics on the performance of a particular summarization or detection method across a number of cel files. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodExprCHPReport_8cpp.html">QuantMethodExprCHPReport.cpp</a> <a href="QuantMethodExprCHPReport_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for reporting results of quantification methods. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodExprCHPReport_8h.html">QuantMethodExprCHPReport.h</a> <a href="QuantMethodExprCHPReport_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for reporting results of quantification methods. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodExprReport_8cpp.html">QuantMethodExprReport.cpp</a> <a href="QuantMethodExprReport_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for reporting results of quantification methods. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodExprReport_8h.html">QuantMethodExprReport.h</a> <a href="QuantMethodExprReport_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for reporting results of quantification methods. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodFactory_8cpp.html">QuantMethodFactory.cpp</a> <a href="QuantMethodFactory_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodFactory_8h.html">QuantMethodFactory.h</a> <a href="QuantMethodFactory_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Factory class for making chip streams based on a string representation. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodGTypeChipSummary_8h.html">QuantMethodGTypeChipSummary.h</a> <a href="QuantMethodGTypeChipSummary_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for accumulating some run statistics from a genotyping run. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodGTypeCHPReport_8cpp.html">QuantMethodGTypeCHPReport.cpp</a> <a href="QuantMethodGTypeCHPReport_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Reporter for genotyping probe sets that outputs chp files. There is one CHP file produced for each cel file and the order of the CHP file is guarenteed to be the same as the cdf file. It is very important that the number and order of the probeset results is the exact same as the CDF file. There have been a number of high impact bugs on this front and it is important to be very careful. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodGTypeCHPReport_8h.html">QuantMethodGTypeCHPReport.h</a> <a href="QuantMethodGTypeCHPReport_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Reporter for genotyping probe sets that outputs chp files. There is one GCOS XDA CHP file produced for each cel file and the order of the GCOS XDA CHP file is guaranteed to be the same as the cdf file. It is very important that the number and order of the probeset results is the exact same as the CDF file. There have been a number of high impact bugs on this front and it is important to be very careful. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodGTypeExprChipSummary_8cpp.html">QuantMethodGTypeExprChipSummary.cpp</a> <a href="QuantMethodGTypeExprChipSummary_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for accumulating some run statistics from a genotyping run from the expression quant method. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodGTypeExprChipSummary_8h.html">QuantMethodGTypeExprChipSummary.h</a> <a href="QuantMethodGTypeExprChipSummary_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for accumulating some run statistics from a genotyping run from the expression quant method. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantMethodGTypeReport.cpp</b> <a href="QuantMethodGTypeReport_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodGTypeReport_8h.html">QuantMethodGTypeReport.h</a> <a href="QuantMethodGTypeReport_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Reporter for outputting genotype results. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodMultiDataCCCHPReport_8cpp.html">QuantMethodMultiDataCCCHPReport.cpp</a> <a href="QuantMethodMultiDataCCCHPReport_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Reporter for genotyping probe sets that outputs CC CHP files. There is one CC CHP file produced for each cel file and the order of the CC CHP file is guaranteed to be the same as the cdf file. It is very important that the number and order of the probeset results is the exact same as the CDF file. There have been a number of high impact bugs on this front and it is important to be very careful. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodMultiDataCCCHPReport_8h.html">QuantMethodMultiDataCCCHPReport.h</a> <a href="QuantMethodMultiDataCCCHPReport_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Reporter for genotyping probe sets that outputs chp files. There is one CC CHP file produced for each cel file and the order of the CC CHP file is guarenteed to be the same as the cdf file. It is very important that the number and order of the probeset results is the exact same as the CDF file. There have been a number of high impact bugs on this front and it is important to be very careful. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodReport_8cpp.html">QuantMethodReport.cpp</a> <a href="QuantMethodReport_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for reporting results of quantification methods. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodReport_8h.html">QuantMethodReport.h</a> <a href="QuantMethodReport_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for reporting results of quantification methods. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantMethodReportListener.cpp</b> <a href="QuantMethodReportListener_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodReportListener_8h.html">QuantMethodReportListener.h</a> <a href="QuantMethodReportListener_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Listens for reporting signals from <a class="el" href="classQuantMethod.html" title="QuantMethod - Interface for computing quantification summaries from PM intensities grouped into probe...">QuantMethod</a> and then passes them off to another coordinating object. This is needed as the analysis stream "owns" the reporters and will destroy them when the stream is destroyed. This is a problem if you want multiple analysis streams reporting into the same thing. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodRunReport_8cpp.html">QuantMethodRunReport.cpp</a> <a href="QuantMethodRunReport_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Summarize and output basic statistics on the performance of a particular summarization or detection method across a number of cel files. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodRunReport_8h.html">QuantMethodRunReport.h</a> <a href="QuantMethodRunReport_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Summarize and output basic statistics on the performance of a particular summarization or detection method across a number of cel files. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodSamplerReport_8h.html">QuantMethodSamplerReport.h</a> <a href="QuantMethodSamplerReport_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for reporting qc stats to a file. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantMethodSnpqcChipSummary.h</b> <a href="QuantMethodSnpqcChipSummary_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantMethodSqlExprReport.cpp</b> <a href="QuantMethodSqlExprReport_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantMethodSqlExprReport.h</b> <a href="QuantMethodSqlExprReport_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantMethodSqlGTypeReport.cpp</b> <a href="QuantMethodSqlGTypeReport_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodSqlGTypeReport_8h.html">QuantMethodSqlGTypeReport.h</a> <a href="QuantMethodSqlGTypeReport_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for reporting results of quantification methods into an sql database. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>QuantMethodTransform.cpp</b> <a href="QuantMethodTransform_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantMethodTransform_8h.html">QuantMethodTransform.h</a> <a href="QuantMethodTransform_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p><a class="el" href="classDataTransform.html" title="Base class for objects that will manipulate data and produce a new set.">DataTransform</a> wrapper for quantification methods. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantPlier_8cpp.html">QuantPlier.cpp</a> <a href="QuantPlier_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing probe set quantification using the PLIER (<a class="el" href="classProbe.html" title="Represents an individual feature on a chip.">Probe</a> Logarithmic Error Intensity Estimate) method. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantPlier_8h.html">QuantPlier.h</a> <a href="QuantPlier_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing probe set quantification using the PLIER (<a class="el" href="classProbe.html" title="Represents an individual feature on a chip.">Probe</a> Logarithmic Error Intensity Estimate) method. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantPlierBase_8cpp.html">QuantPlierBase.cpp</a> <a href="QuantPlierBase_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Base class for doing probe set quantification using either the PLIER (<a class="el" href="classProbe.html" title="Represents an individual feature on a chip.">Probe</a> Logarithmic Error Intensity Estimate) method or the ITER-PLIER method, which iteratively calls PLIER with the probes that best correlate with signal estimate. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantPlierBase_8h.html">QuantPlierBase.h</a> <a href="QuantPlierBase_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Base class for doing probe set quantification using either the PLIER (<a class="el" href="classProbe.html" title="Represents an individual feature on a chip.">Probe</a> Logarithmic Error Intensity Estimate) method or the ITER-PLIER method, which iteratively calls PLIER with the probes that best correlate with signal estimate. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantPlierParams_8h.html">QuantPlierParams.h</a> <a href="QuantPlierParams_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for plier paramaters. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantRma_8cpp.html">QuantRma.cpp</a> <a href="QuantRma_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing probe set quantification using median polish as does <a class="el" href="classRMA.html" title="Holds methods for computing the popular Robust Muliarray Average of Microarrays (RMA).">RMA</a>. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantRma_8h.html">QuantRma.h</a> <a href="QuantRma_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing probe set quantification using median polish as does <a class="el" href="classRMA.html" title="Holds methods for computing the popular Robust Muliarray Average of Microarrays (RMA).">RMA</a>. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantSea_8cpp.html">QuantSea.cpp</a> <a href="QuantSea_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing probe set quantification using the SEA, or Simplified Expression <a class="el" href="classAnalysis.html">Analysis</a>, method. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="QuantSea_8h.html">QuantSea.h</a> <a href="QuantSea_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing probe set quantification using the SEA, or Simplified Expression <a class="el" href="classAnalysis.html">Analysis</a>, method. This is a single chip method which is a subset of the PLIER (<a class="el" href="classProbe.html" title="Represents an individual feature on a chip.">Probe</a> Logarithmic Error Intensity Estimate) method. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="RmaBgTran_8cpp.html">RmaBgTran.cpp</a> <a href="RmaBgTran_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing <a class="el" href="classRMA.html" title="Holds methods for computing the popular Robust Muliarray Average of Microarrays (RMA).">RMA</a> style background subtraction. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="RmaBgTran_8h.html">RmaBgTran.h</a> <a href="RmaBgTran_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing <a class="el" href="classRMA.html" title="Holds methods for computing the popular Robust Muliarray Average of Microarrays (RMA).">RMA</a> style background subtraction. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>SelectFilesEngine.cpp</b> <a href="SelectFilesEngine_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="SelectFilesEngine_8h.html">SelectFilesEngine.h</a> <a href="SelectFilesEngine_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="SelfCreate_8cpp.html">SelfCreate.cpp</a> <a href="SelfCreate_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Small interface for functions that know how to make an instance of themselves given a map of key, value parameters. Idea is that you can call this class's static methods generically and dynamic_cast appropriately. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="SelfCreate_8h.html">SelfCreate.h</a> <a href="SelfCreate_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Small interface for functions that know how to make an instance of themselves given a map of key, value parameters. Idea is that you can call this class's static methods generically and cast appropriately. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>SelfDoc.cpp</b> <a href="SelfDoc_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="SelfDoc_8h.html">SelfDoc.h</a> <a href="SelfDoc_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Small interface to for algorithmic classes that can explain themselves. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>SignalBackgroundCelListener.cpp</b> <a href="SignalBackgroundCelListener_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>SignalBackgroundCelListener.h</b> <a href="SignalBackgroundCelListener_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="SketchQuantNormTran_8cpp.html">SketchQuantNormTran.cpp</a> <a href="SketchQuantNormTran_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing normalization. Can do sketch and full quantile (just set sketch to chip size) and supports bioconductor compatibility. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="SketchQuantNormTran_8h.html">SketchQuantNormTran.h</a> <a href="SketchQuantNormTran_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for doing normalization. Can do sketch and full quantile (just set sketch to chip size) and supports bioconductor compatibility. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>SnpClusterStore.cpp</b> <a href="SnpClusterStore_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="SnpClusterStore_8h.html">SnpClusterStore.h</a> <a href="SnpClusterStore_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for reading the posterior from snp clustering. Currently supports only Axiom but goal is to support all GTC Affymetrix Genotyping algorithms. Goal is to provide relatively quick random access to snp posteriors for display in cluster plots. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>SnpModelConverter.cpp</b> <a href="SnpModelConverter_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="SnpModelConverter_8h.html">SnpModelConverter.h</a> <a href="SnpModelConverter_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class collecting utility methods for reading various formats of snp models like:</p>
<ul>
<li>Brlmm-p (aka labelz) used for snp5, snp6 and Axiom</li>
<li>Birdseed priors used for snp 6</li>
<li>Birdseed posteriors from snp 6. </li>
</ul>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>SnpModelDb.cpp</b> <a href="SnpModelDb_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="SnpModelDb_8h.html">SnpModelDb.h</a> <a href="SnpModelDb_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for reading and writing snp clustering models (eg gaussian elipses of mean and variance). Uses sqlite as a background to provide convenient fast random access to large numbers of snps (e.g. Millions) with minimal memory impact. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="SparseMart_8cpp.html">SparseMart.cpp</a> <a href="SparseMart_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for representing cel file intensity data in memory. Can save a lot of the space if not all of a cel file needs to be remembered for downstream analysis. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="SparseMart_8h.html">SparseMart.h</a> <a href="SparseMart_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Class for representing cel file intensity data in memory. Can save a lot of the space if not all of a cel file needs to be remembered for downstream analysis. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="SpecialSnps_8cpp.html">SpecialSnps.cpp</a> <a href="SpecialSnps_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="SpecialSnps_8h.html">SpecialSnps.h</a> <a href="SpecialSnps_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>SpfReader.cpp</b> <a href="SpfReader_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>SpfReader.h</b> <a href="SpfReader_8h_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="SummaryStats_8cpp.html">SummaryStats.cpp</a> <a href="SummaryStats_8cpp_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Utility class for keeping track of summary statistics. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="SummaryStats_8h.html">SummaryStats.h</a> <a href="SummaryStats_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>Utility class for keeping track of summary statistics. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>TableReader.cpp</b> <a href="TableReader_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="TableReader_8h.html">TableReader.h</a> <a href="TableReader_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p><a class="el" href="classIntensityMart.html" title="IntensityMart.">IntensityMart</a> object for reading intensity files for text files. </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>TsvReport.cpp</b> <a href="TsvReport_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="TsvReport_8h.html">TsvReport.h</a> <a href="TsvReport_8h_source.html">[code]</a></td></tr>

<p><tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight"><p>/// </p>
<br/></td></tr>
</p>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>VariabilityScoreCelListener.cpp</b> <a href="VariabilityScoreCelListener_8cpp_source.html">[code]</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">file &nbsp;</td><td class="memItemRight" valign="bottom"><b>VariabilityScoreCelListener.h</b> <a href="VariabilityScoreCelListener_8h_source.html">[code]</a></td></tr>
</table>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:38:39 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
