<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: translation/ExperimentGeneResults.cpp Source File</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li class="current"><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="tabs2">
    <ul class="tablist">
      <li><a href="files.html"><span>File&nbsp;List</span></a></li>
      <li><a href="globals.html"><span>File&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath">
    <ul>
      <li><a class="el" href="dir_1668076c0e919fc49032c4fb42e3bb3d.html">translation</a>      </li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>ExperimentGeneResults.cpp</h1>  </div>
</div>
<div class="contents">
<a href="ExperimentGeneResults_8cpp.html">Go to the documentation of this file.</a><div class="fragment"><pre class="fragment"><a name="l00001"></a>00001 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00002"></a>00002 <span class="comment"></span><span class="comment">//</span>
<a name="l00003"></a>00003 <span class="comment">// Copyright (C) 2008 Affymetrix, Inc.</span>
<a name="l00004"></a>00004 <span class="comment">//</span>
<a name="l00005"></a>00005 <span class="comment">// This program is free software; you can redistribute it and/or modify </span>
<a name="l00006"></a>00006 <span class="comment">// it under the terms of the GNU General Public License (version 2) as </span>
<a name="l00007"></a>00007 <span class="comment">// published by the Free Software Foundation.</span>
<a name="l00008"></a>00008 <span class="comment">// </span>
<a name="l00009"></a>00009 <span class="comment">// This program is distributed in the hope that it will be useful, </span>
<a name="l00010"></a>00010 <span class="comment">// but WITHOUT ANY WARRANTY; without even the implied warranty of </span>
<a name="l00011"></a>00011 <span class="comment">// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU </span>
<a name="l00012"></a>00012 <span class="comment">// General Public License for more details.</span>
<a name="l00013"></a>00013 <span class="comment">// </span>
<a name="l00014"></a>00014 <span class="comment">// You should have received a copy of the GNU General Public License </span>
<a name="l00015"></a>00015 <span class="comment">// along with this program;if not, write to the </span>
<a name="l00016"></a>00016 <span class="comment">// </span>
<a name="l00017"></a>00017 <span class="comment">// Free Software Foundation, Inc., </span>
<a name="l00018"></a>00018 <span class="comment">// 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA</span>
<a name="l00019"></a>00019 <span class="comment">//</span><span class="comment"></span>
<a name="l00020"></a>00020 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00021"></a>00021 <span class="comment"></span><span class="comment">/**</span>
<a name="l00022"></a>00022 <span class="comment"> * @file   ExperimentGeneResults.cpp</span>
<a name="l00023"></a>00023 <span class="comment"> * @author Mybrid Spalding</span>
<a name="l00024"></a>00024 <span class="comment"> * @date   Tue Mar 18 14:42:35 PDT 2008</span>
<a name="l00025"></a>00025 <span class="comment"> * @brief  CallResults wrapper class for all gene translated experiment results to be consumed by various report objects.</span>
<a name="l00026"></a>00026 <span class="comment"> */</span>
<a name="l00027"></a>00027 
<a name="l00028"></a>00028 <span class="preprocessor">#include &quot;<a class="code" href="ExperimentGeneResults_8h.html" title="CallResults wrapper class for all gene translated experiment results to be consumed by various report...">translation/ExperimentGeneResults.h</a>&quot;</span>
<a name="l00029"></a>00029 <span class="comment">//</span>
<a name="l00030"></a>00030 <span class="preprocessor">#include &quot;<a class="code" href="GeneCall_8h.html" title="Business object that is the heart of the matter, class with the algorithm that translates experiment ...">translation/GeneCall.h</a>&quot;</span>
<a name="l00031"></a>00031 <span class="comment">//</span>
<a name="l00032"></a>00032 
<a name="l00033"></a>00033 <span class="keyword">using namespace </span>std;
<a name="l00034"></a>00034 
<a name="l00035"></a>00035 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00036"></a>00036 <span class="comment">/**</span>
<a name="l00037"></a>00037 <span class="comment"> * ExperimentGeneResults::ExperimentGeneResults</span>
<a name="l00038"></a>00038 <span class="comment"> * Synopsis:</span>
<a name="l00039"></a>00039 <span class="comment"> *</span>
<a name="l00040"></a>00040 <span class="comment"> * Main constructor that will simply contain CallResults in a list</span>
<a name="l00041"></a>00041 <span class="comment"> * built by using &quot;appendCallResults&quot;.</span>
<a name="l00042"></a>00042 <span class="comment"> *</span>
<a name="l00043"></a>00043 <span class="comment"> *</span>
<a name="l00044"></a>00044 <span class="comment"> * @param geneName        - the gene name</span>
<a name="l00045"></a>00045 <span class="comment"> * @param experimentName  - the corresponding experiment</span>
<a name="l00046"></a>00046 <span class="comment"> * @param zeroCopyHaplotype - a gene can only ever have one &quot;allele&quot; name for the zero copy number designation, and this is it.</span>
<a name="l00047"></a>00047 <span class="comment"> *</span>
<a name="l00048"></a>00048 <span class="comment"> */</span>
<a name="l00049"></a>00049 <span class="comment">/*****************************************************************************/</span>
<a name="l00050"></a><a class="code" href="classExperimentGeneResults.html#a3721951d4c71233e7c553e1d993872ff">00050</a> ExperimentGeneResults::ExperimentGeneResults(<span class="keyword">const</span> std::string &amp; geneName, <span class="keyword">const</span> std::string &amp; experimentName, <span class="keyword">const</span> std::string &amp; zeroCopyHaplotype):
<a name="l00051"></a>00051     m_experimentName(experimentName), m_geneName(geneName), m_geneZeroCopyHaplotype(zeroCopyHaplotype)
<a name="l00052"></a>00052 {
<a name="l00053"></a>00053   m_markerCallCount = 0;
<a name="l00054"></a>00054 
<a name="l00055"></a>00055 }
<a name="l00056"></a>00056 <span class="comment">// end ExperimentGeneResults::ExperimentGeneResults</span>
<a name="l00057"></a>00057 <span class="comment">/*****************************************************************************/</span>
<a name="l00058"></a>00058 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00059"></a>00059 <span class="comment">/**</span>
<a name="l00060"></a>00060 <span class="comment"> * ExperimentGeneResults::appendCallResults</span>
<a name="l00061"></a>00061 <span class="comment"> * Synopsis:</span>
<a name="l00062"></a>00062 <span class="comment"> *</span>
<a name="l00063"></a>00063 <span class="comment"> * Append the CallResults for a pair of chromatids to the std::vector of</span>
<a name="l00064"></a>00064 <span class="comment"> * CallResults. Does some book keeping as well by building indexes</span>
<a name="l00065"></a>00065 <span class="comment"> * in place to be used later at report time.</span>
<a name="l00066"></a>00066 <span class="comment"> *</span>
<a name="l00067"></a>00067 <span class="comment"> *</span>
<a name="l00068"></a>00068 <span class="comment"> * @param newCallResults - The results to append.</span>
<a name="l00069"></a>00069 <span class="comment"> * @param geneCall      - The variantSet and reference CallSet are needed</span>
<a name="l00070"></a>00070 <span class="comment"> *                         for multi-allelic reporting on variant bases.</span>
<a name="l00071"></a>00071 <span class="comment"> *</span>
<a name="l00072"></a>00072 <span class="comment"> */</span>
<a name="l00073"></a>00073 <span class="comment">/*****************************************************************************/</span>
<a name="l00074"></a><a class="code" href="classExperimentGeneResults.html#a13722310b26cf978fd3ab2c4325c798e">00074</a> <span class="keywordtype">void</span> <a class="code" href="classExperimentGeneResults.html#a13722310b26cf978fd3ab2c4325c798e" title="ExperimentGeneResults::appendCallResults Synopsis:">ExperimentGeneResults::appendCallResults</a>(<span class="keyword">const</span> <a class="code" href="classCallResults.html">CallResults</a> &amp; newResults,
<a name="l00075"></a>00075     <a class="code" href="classGeneCall.html">GeneCall</a> &amp; geneCall)
<a name="l00076"></a>00076 {
<a name="l00077"></a>00077 
<a name="l00078"></a>00078   m_callResults.push_back(newResults);
<a name="l00079"></a>00079   m_markerCallCount += newResults.m_markerCallCount;
<a name="l00080"></a>00080 
<a name="l00081"></a>00081   std::map&lt;std::string, CallElement&gt;::iterator iCEit;
<a name="l00082"></a>00082   std::map&lt;std::string, std::vector&lt;std::string&gt; &gt;::iterator jit;
<a name="l00083"></a>00083   std::map&lt;std::string, CallElement&gt;::iterator kCEit;
<a name="l00084"></a>00084 
<a name="l00085"></a>00085   <span class="keywordflow">if</span> (newResults.getCallType() == ADT_CALL_TYPE_MARKER) {
<a name="l00086"></a>00086     m_probeSetToMarkerCallResults[ newResults.getMarkerProbeSet()]
<a name="l00087"></a>00087     = m_callResults.size()  - 1;
<a name="l00088"></a>00088   }
<a name="l00089"></a>00089 
<a name="l00090"></a>00090   <span class="keywordflow">for</span> (iCEit = geneCall.getVariantCallSet().m_ceSet.begin();
<a name="l00091"></a>00091        iCEit != geneCall.getVariantCallSet().m_ceSet.end();
<a name="l00092"></a>00092        iCEit++) {
<a name="l00093"></a>00093 
<a name="l00094"></a>00094     jit = m_probeSetVariants.find(iCEit-&gt;first);
<a name="l00095"></a>00095 
<a name="l00096"></a>00096     <span class="keywordflow">if</span> (jit ==  m_probeSetVariants.end()) {
<a name="l00097"></a>00097       <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; iCEit-&gt;second.m_bases.size(); i++) {
<a name="l00098"></a>00098         m_probeSetVariants[iCEit-&gt;first].push_back(iCEit-&gt;second.m_bases[i]);
<a name="l00099"></a>00099       }
<a name="l00100"></a>00100     } <span class="keywordflow">else</span> {
<a name="l00101"></a>00101       <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; iCEit-&gt;second.m_bases.size(); i++) {
<a name="l00102"></a>00102         std::vector&lt;std::string&gt;::iterator kit;
<a name="l00103"></a>00103         <span class="keywordtype">bool</span> okToAdd = <span class="keyword">true</span>;
<a name="l00104"></a>00104 
<a name="l00105"></a>00105         <span class="keywordflow">for</span> (kit = m_probeSetVariants[iCEit-&gt;first].begin();
<a name="l00106"></a>00106              (kit != m_probeSetVariants[iCEit-&gt;first].end()) &amp;&amp;
<a name="l00107"></a>00107              okToAdd;
<a name="l00108"></a>00108              kit++) {
<a name="l00109"></a>00109           <span class="keywordflow">if</span> (*kit == iCEit-&gt;second.m_bases[i]) {
<a name="l00110"></a>00110             okToAdd = <span class="keyword">false</span>;
<a name="l00111"></a>00111           }
<a name="l00112"></a>00112         }
<a name="l00113"></a>00113         <span class="keywordflow">if</span> (okToAdd) {
<a name="l00114"></a>00114           m_probeSetVariants[iCEit-&gt;first].push_back(iCEit-&gt;second.m_bases[i]);
<a name="l00115"></a>00115         }
<a name="l00116"></a>00116       }
<a name="l00117"></a>00117     }
<a name="l00118"></a>00118   }
<a name="l00119"></a>00119 
<a name="l00120"></a>00120   <span class="keywordflow">for</span> (kCEit = geneCall.getReferenceCallSet().m_ceSet.begin();
<a name="l00121"></a>00121        kCEit != geneCall.getReferenceCallSet().m_ceSet.end();
<a name="l00122"></a>00122        kCEit++) {
<a name="l00123"></a>00123 
<a name="l00124"></a>00124     jit = m_probeSetReference.find(kCEit-&gt;first);
<a name="l00125"></a>00125 
<a name="l00126"></a>00126     <span class="keywordflow">if</span> (jit ==  m_probeSetReference.end()) {
<a name="l00127"></a>00127       <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; kCEit-&gt;second.m_bases.size(); i++) {
<a name="l00128"></a>00128         m_probeSetReference[kCEit-&gt;first].push_back(kCEit-&gt;second.m_bases[i]);
<a name="l00129"></a>00129       }
<a name="l00130"></a>00130     } <span class="keywordflow">else</span> {
<a name="l00131"></a>00131       <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; kCEit-&gt;second.m_bases.size(); i++) {
<a name="l00132"></a>00132         std::vector&lt;std::string&gt;::iterator kit;
<a name="l00133"></a>00133         <span class="keywordtype">bool</span> okToAdd = <span class="keyword">true</span>;
<a name="l00134"></a>00134 
<a name="l00135"></a>00135         <span class="keywordflow">for</span> (kit = m_probeSetReference[kCEit-&gt;first].begin();
<a name="l00136"></a>00136              (kit != m_probeSetReference[kCEit-&gt;first].end()) &amp;&amp;
<a name="l00137"></a>00137              okToAdd;
<a name="l00138"></a>00138              kit++) {
<a name="l00139"></a>00139           <span class="keywordflow">if</span> (*kit == kCEit-&gt;second.m_bases[i]) {
<a name="l00140"></a>00140             okToAdd = <span class="keyword">false</span>;
<a name="l00141"></a>00141           }
<a name="l00142"></a>00142         }
<a name="l00143"></a>00143         <span class="keywordflow">if</span> (okToAdd) {
<a name="l00144"></a>00144           m_probeSetReference[kCEit-&gt;first].push_back(kCEit-&gt;second.m_bases[i]);
<a name="l00145"></a>00145         }
<a name="l00146"></a>00146       }
<a name="l00147"></a>00147     }
<a name="l00148"></a>00148   }
<a name="l00149"></a>00149 
<a name="l00150"></a>00150 
<a name="l00151"></a>00151   <span class="keywordflow">return</span>;
<a name="l00152"></a>00152 }
<a name="l00153"></a>00153 
<a name="l00154"></a>00154 <span class="comment">// end ExperimentGeneResults::appendCallResults</span>
<a name="l00155"></a>00155 <span class="comment">/*****************************************************************************/</span>
<a name="l00156"></a>00156 
</pre></div></div>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:48 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
