<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: File Format: Genotyping Quality Control (GQC) (1.16.1)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li class="current"><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>File Format: Genotyping Quality Control (GQC) (1.16.1) </h1>  </div>
</div>
<div class="contents">
<center><b>DRAFT</b></center><p>The GQC (Genotyping Quality Control) file format is used to represent per-chip quality control data using one or more types of analyses on a given array type.</p>
<p>The format is a specialization of version 1 of the <a class="el" href="file-format-tsv.html">TSV file format </a>. The format is defined for genotyping array only.</p>
<p>The file contains quality assessment metrics and is generated by apt-geno-qc. There is one row per CEL file analyzed.</p>
<h2><a class="anchor" id="gqc_specifications"></a>
Specifications</h2>
<ul>
<li>Required headers:<ul>
<li>format_version: indicates the version of the output file format.</li>
<li>content_version: used to track the version of the content of the file.</li>
<li>default_analysis_name: The value for this field must correspond to an entry in the 'analysis_name' column within the file. It is used to indicate to the application reading the Analysis-QCC file which one of the various ways of analyzing different probeset groups is to be treated as the primary metric for ranking and/or qualifying the performance of experiments.</li>
<li>library <a href="file:">file:</a> the library (CDF) file used in the analyses</li>
</ul>
</li>
<li>Optional headers<ul>
<li>gender information: the explanation of gender mapping for gender interpretation and parsing.</li>
<li>method-n: a comment line for analysis number n, one line per analysis. These comment lines record all the information used for the analyses for tracking purposes.</li>
</ul>
</li>
<li>Columns<ul>
<li>Per TSV format, order of columns is not guaranteed.</li>
<li>Per TSV format, additional columns may be present.</li>
<li>Column Chip (required): the first column is always named 'Chip' with chip cel files as values.</li>
<li>Gender (optional):</li>
<li>QC call rate (Nsp) (optional):</li>
<li>QC call rate (Nsp/Sty overlap) (optional):</li>
<li>QC call rate (Sty) (optional):</li>
<li>QC call rate (all) (optional):</li>
<li>other columns may be present (optional):</li>
</ul>
</li>
</ul>
<h2><a class="anchor" id="gqc_implementation"></a>
Parsing and Writing</h2>
<p>The official C++ parser used by affy can be found in APT under <a class="el" href="TsvFile_8h.html" title="Headers for the TsvFile classes.">sdk/file/TsvFile/TsvFile.h</a>. When possible, parsing and writing of GQC files should be done using this code.</p>
<h2><a class="anchor" id="gqc_notes"></a>
Notes</h2>
<p>Nothing at this time.</p>
<h2><a class="anchor" id="gqc_example1"></a>
Example 1 -- GQC file for 5 analysis run on Genome Wide SNP 5</h2>
<div class="fragment"><pre class="fragment">
#%format_version=1
#%content_version=1
#%default_analysis_name=QC call rate (all)
#%library file=data/GenomeWideEx_5/data/GenomeWideSNP_5.cdf
#gender information: male=0,female=1,unknown=-1
#method1:analysis_name=QC call rate (all);analysis=dm;group_name=all_qc;as.percentage=1;dm.cutoff=0.33;dm.hetMult=1.25;precision=4
#method2:analysis_name=QC call rate (Nsp);analysis=dm;group_name=nsp_qc;as.percentage=1;dm.cutoff=0.33;dm.hetMult=1.25;precision=4
#method3:analysis_name=QC call rate (Sty);analysis=dm;group_name=sty_qc;as.percentage=1;dm.cutoff=0.33;dm.hetMult=1.25;precision=4
#method4:analysis_name=QC call rate (Nsp/Sty overlap);analysis=dm;group_name=nsp_sty_qc;as.percentage=1;dm.cutoff=0.33;dm.hetMult=1.25;precision=4
#method5:analysis_name=Gender;analysis=gender;group_name=chrX;as.text=1;em.cutoff=0.5;em.thresh=0.05;gender.cutoff=0.1
Chip	Gender	QC call rate (Nsp)	QC call rate (Nsp/Sty overlap)	QC call rate (Sty)	QC call rate (all)
NA06985_GW5_C.CEL	female	91.13	94.45	84.86	91.63
NA06991_GW5_C.CEL	female	93.06	94.58	86.47	92.52
NA06993_GW5_C.CEL	male	92.16	94.09	84.54	91.63
</pre></div><h2><a class="anchor" id="gqc_related"></a>
Related Pages</h2>
<ul>
<li><a href="./file-format-tsv-notes.html">TsvFile Design Notes</a></li>
<li><a href="./file-format-tsv.html">TSV File Format</a> </li>
</ul>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:57 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
