<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: File Format: Quality Control Analysis (QCA) (1.16.1)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li class="current"><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>File Format: Quality Control <a class="el" href="classAnalysis.html">Analysis</a> (QCA) (1.16.1) </h1>  </div>
</div>
<div class="contents">
<center><b>DRAFT</b></center><p>The Quality Control <a class="el" href="classAnalysis.html">Analysis</a> (QCA) file format is used to define types of quality assessment analysis to perform on an array. The file can specify multiple analyses to run, and specifies for each which probesets should be used and how they should be analyzed.</p>
<p>The format is a specialization of version 1 of the <a class="el" href="file-format-tsv.html">TSV file format </a>. The format is defined for genotyping array only.</p>
<h2><a class="anchor" id="qca_specifications"></a>
Specifications</h2>
<ul>
<li>Required headers<ul>
<li>format_version: indicates the version of the QCA format.</li>
<li>content_version: used to track the version of the content of the file.</li>
<li>default_analysis_name: The value for this field must correspond to an entry in the 'analysis_name' column within the file. It is used to indicate to the application reading the Analysis-QCC file which one of the various ways of analyzing different probeset groups is to be treated as the primary metric for ranking and/or qualifying the performance of experiments.</li>
</ul>
</li>
<li>Optional headers: none</li>
<li>Columns<ul>
<li>per TSV format, order of columns is not guaranteed.</li>
<li>per TSV format, additional columns may be present.</li>
<li>analysis_name: This is the primary key for the rows and thus should take on an unique value for each row. It is intended to provide a brief descriptive identifer which indicates the probeset group and analysis method being specified in the row. This field may be used by some applications as a method identifier in subsequent processing.</li>
<li>group_name: This field identifies the group(s) of probesets to be analyzed and must it must also exist in the corresponding QCC file (see QCC file format spec). The value is a string with one group name per analysis, different analyses may use the same group name (in other words it is possible to specify that the same probesets be anlayzed in multiple different ways).</li>
<li>analysis: This field specifies which algorithm is used for corresponding analysis. The calling application must be able to recogize the name of the analysis and know how to translate it into the analysis method to be used. Examples of some algorithms available are dm (DM) and gender (GENDER). They must be either all lower case or all upper case. Valid a analysis algorithms must be specified, otherwise no analysis results can be generated.</li>
<li>options: This field specifies parameter(s) for the corresponding analysis algorithm. Each parameter is a tag-value pair separated by an equal sign. Multiple parameters can be specified by supplying tag-value pairs in a comma-separated list. Parameters tag-value pairs are case-sensitive, unrecognized parameters will be ignored and treated as missing. The order of parameters is unimportant. Example parameters for some current algorithm options are: + dm<ul>
<li>dm.cutoff: confidence threshold for making calls (default is 0.33)</li>
<li>dm.hetMult: het/hom tuning parameter added to log-likelihood of het calls (default is 1.25)</li>
<li>as.percantage: logic value of 0 or 1 to indicate whether a percentage or a fraction is reported (default is 1)</li>
<li>precision: the precision of the stats to be reported (default is 6) + gender</li>
<li>em.thresh: threshold of em's confidence score for clustering, the smaller the more stringent (default value is 0.05).</li>
<li>em.cutoff: the cutoff on percentage of SNPs classified with an em's confidence score less than em.thresh (default value is 0.5)</li>
<li>gender.cutoff: heterozygosity above which called gender will be female and below which called gender will be male. (default value is 0.1)</li>
<li>as.text: logic value of 0 or 1 to indicate whether numeric value of -1,0,1 or text of "unknown","male","female" is reported (default is 1)</li>
</ul>
</li>
</ul>
</li>
</ul>
<h2><a class="anchor" id="qca_implementation"></a>
Parsing and Writing</h2>
<p>The official C++ parser used by affy can be found in APT under <a class="el" href="TsvFile_8h.html" title="Headers for the TsvFile classes.">sdk/file/TsvFile/TsvFile.h</a>. When possible, parsing and writing of QCA files should be done using this code.</p>
<h2><a class="anchor" id="qca_notes"></a>
Notes</h2>
<p>Nothing at this time.</p>
<h2><a class="anchor" id="qca_example1"></a>
Example 1 -- GenomeWideSNP_5.qca</h2>
<p>A QCA file for a the Genome Wide SNP 5 array.</p>
<div class="fragment"><pre class="fragment">
#%format_version=1.0
#%content_version=1.0
#%default_analysis_name=QC call rate (all)
#
# There is a row for each analysis method on each SNP list.
#   analysis name is the name that will be associated with the analysis &amp; snp list combination
#   snp_list is the identifier for the list of SNPs
#   analysis is the name of the algorithm to run
#   options is a field (may be empty) to pass options to the algorithm.
#
analysis_name	group_name	analysis	options
# Standard analyses
QC call rate (all)	all_qc	dm	dm.cutoff=0.33,dm.hetMult=1.25,as.percentage=1,precision=4
QC call rate (Nsp)	nsp_qc	dm	dm.cutoff=0.33,dm.hetMult=1.25,as.percentage=1,precision=4
QC call rate (Sty)	sty_qc	dm	dm.cutoff=0.33,dm.hetMult=1.25,as.percentage=1,precision=4
QC call rate (Nsp/Sty overlap)	nsp_sty_qc	dm	dm.cutoff=0.33,dm.hetMult=1.25,as.percentage=1,precision=4
Gender	chrX	gender	em.cutoff=0.5,em.thresh=0.05,gender.cutoff=0.1,as.text=1
</pre></div><h2><a class="anchor" id="qca_related"></a>
Related Pages</h2>
<ul>
<li><a href="./file-format-tsv-notes.html">TsvFile Design Notes</a></li>
<li><a href="./file-format-tsv.html">TSV File Format</a> </li>
</ul>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:57 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
