<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: File Format: PGF (NON-OFFICIAL-RELEASE)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li class="current"><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>File Format: PGF (NON-OFFICIAL-RELEASE) </h1>  </div>
</div>
<div class="contents">
<center><b>DRAFT</b></center><p>The PGF (probe group file) provides information about what probes are contained within a probeset and information about the nature of the probes necessary for analysis. The current PGF file format (version 1) is only specified for expression style probesets.</p>
<p>The PGF file is based on version 2 of the <a class="el" href="file-format-tsv.html">TSV file format </a>.</p>
<h2><a class="anchor" id="specifications"></a>
Specifications</h2>
<ul>
<li>Required headers<ul>
<li>chip_type: indicates the chip type (as stored in the CEL file) which the pgf file describes. Multiple chip_type headers may be present indicating the pgf file can be used with any one of the listed chip types.</li>
<li>lib_set_name: indicates the name of a collection of related library files for a given chip. For example, PGF and CLF files intended to be used together should have the same lib_set_name. Only a single lib_set_name is allowed.</li>
<li>lib_set_version: indicates the version of a collection of related library files for a given chip. For example, PGF and CLF files intended to be used together should have the same lib_set_version. Only a single lib_set_version is allowed.</li>
<li>pgf_format_version: currently the only documented and supported version is 1.0.</li>
<li>header0: indicates the values in the pgf file for the top level data (probeset level). A pgf file probeset level entry must contain probeset_id. This field must be unique over all the probeset_ids in the PGF file. Optional, but typical fields include type and probeset_name.</li>
<li>header1: the second level of data in the PGF file (atom level). An atom refers to a particular collection of probes that are interrogating the same position. For expression arrays an atom is usually a probe pair (pm and mm probe pair) from an array like HG-U133 Plus 2.0 or a single pm probe from an array like Human Exon 1.0 ST which does not contain mismatch probes. The atom level data must have an atom_id field. atom_ids are expected to be unique within the PGF file. In other words, atoms cannot be present in more than one probeset or present multiple times in any given probeset.</li>
<li>header2: the third level of data (probe level) contains information about a particular probe in a particular probeset. A probe may be present in more than one probeset, but it may only appear once in a given probeset. Required fields include probe_id and type. Optional, but typical fields include gc_count, probe_length, interrogation_position, and probe_sequence. A probe_id is not required to be unique within the PGF file -- ie that probe may be used in another probeset.</li>
</ul>
</li>
<li>Optional headers<ul>
<li>create_date: timestamp for when the file was created</li>
<li>guid: a unique identifier for the fileA</li>
<li>other headers may be present</li>
</ul>
</li>
<li>Columns<ul>
<li>per TSV format, order of columns is not guarateed</li>
<li>per TSV format, additional columns may be present</li>
<li>Probeset Level (level 0)<ul>
<li>probeset_id (required): an integer id &gt;= 0 which is unique over all the probeset_ids in the pgf file</li>
<li>type (optional): provides classification information for the probeset. See Type section.</li>
<li>probeset_name (optional)</li>
</ul>
</li>
<li><a class="el" href="classAtom.html" title="Represents a collection of probes that make up a unit for an intensity measurement of a particular se...">Atom</a> Level (level 1)<ul>
<li>atom_id (required): an integer id &gt;= 0 which is unique over all the atom_ids in the pgf file</li>
<li>type (optional): provides classification information for the atom. See Type section.</li>
<li>exon_position (optional): the position of the probe interrogation position relative to the target sequence</li>
</ul>
</li>
<li><a class="el" href="classProbe.html" title="Represents an individual feature on a chip.">Probe</a> Level (level 2)<ul>
<li>probe_id (required): an integer id &gt;= 0 which is a foreign key into the CLF file; a specific probe may be present in more than one probeset and as such is not guarateed to be unique in the PGF file. Also note that the additional columns of information at the probe level may be context dependent. So for example a particular probe could potentially be a PM probe in one probeset and an MM probe in another. While unlikely, this is a possibility.</li>
<li>type (required): provides classification information for the probe. See the Type section.</li>
<li>gc_count (optional): the number of G and C bases in the probe</li>
<li>probe_length (optional): the length of the probe</li>
<li>interrogation_position (optional): the interrogation position of the probe (typically 13 for 25-mer PM/MM probes)</li>
<li>probe_sequence (optional): the sequence of the probe on the array in from array surface to solution. For most standard Affymetrix arrays at this writing, this would be in a 3' to 5' direction. So for a sense target (st) probe (see the type field for the probe level) you would need to complement the sequence in this field before looking for matches to transcript sequences; for an antisense target (at) you would need to reverse this sequence.</li>
</ul>
</li>
</ul>
</li>
</ul>
<h2><a class="anchor" id="types"></a>
Types</h2>
<p>Type columns in PGF files use the following string format to catagorize probesets, atoms, and probes:</p>
<div class="fragment"><pre class="fragment">
simple_type:=[a-z0-9\_\-]+
</pre></div><p>So an example simple type</p>
<div class="fragment"><pre class="fragment">
pm
mm
st
at
control
affx
spike
</pre></div><p>Furthermore, types can be nested. For example a particular spike may be from Affymetrix and is intended for use as a control. As a result you would combine the simple types to reflect this:</p>
<div class="fragment"><pre class="fragment">
control-&gt;affx-&gt;spike
</pre></div><p>Thus</p>
<div class="fragment"><pre class="fragment">
nested_type:=(simple_type|nested_type)-&gt;(simple_type)
</pre></div><p>Lastly, a given probeset, atom, or probe may belong to multiple independent types. For example, a probeset may be both a normalization control gene and part of the main design:</p>
<div class="fragment"><pre class="fragment">
normgene-&gt;exon:main
</pre></div><p>Thus</p>
<div class="fragment"><pre class="fragment">
compound_type:=(simple_type|nested_type|compound_type):(simple_type|nested_type)
</pre></div><p>Currently type values are not strongly enumerated. Values used in current commercial PGF files include:</p>
<ul>
<li>Probeset Type:<ul>
<li>main: probesets which are a part of the main design for which the array was designed</li>
<li>normgene-&gt;exon: probe sets against exon regions of a set of housekeeping genes</li>
<li>normgene-&gt;intron: probe sets against intron regions of a set of housekeeping genes</li>
<li>control-&gt;affx: standard Affymetrix spike control probeset (ie bacterial and polyA spikes)</li>
<li>control-&gt;bgp-&gt;antigenomic: antigenomic background probes</li>
<li>control-&gt;bgp-&gt;genomic: genomic background probes</li>
<li>control-&gt;spike-&gt;arabidopsis: arabidopsis spike control probesets</li>
<li>rescue-&gt;FLmRNA-&gt;unmapped: probesets against mRNA sequences which did not align to the genome</li>
<li>types removed in more recent PGF files:<ul>
<li>control-&gt;QC: manufacturing control probesets</li>
<li>control-&gt;chip: internal control probesets used for gridding and other internal functions</li>
</ul>
</li>
</ul>
</li>
<li><a class="el" href="classAtom.html" title="Represents a collection of probes that make up a unit for an intensity measurement of a particular se...">Atom</a> Type<ul>
<li>(none)</li>
</ul>
</li>
<li><a class="el" href="classProbe.html" title="Represents an individual feature on a chip.">Probe</a> Type<ul>
<li>pm: perfect match probe relative to intended target</li>
<li>mm: specific mismatch probe</li>
<li>st: a probe designed for sense target</li>
<li>at: a probe designed for antisense target</li>
<li>types removed in more recent PGF files:<ul>
<li>blank: a blank feature</li>
<li>generic: an internal control feature</li>
<li>jumbo-checkerboard: an internal gridding feature</li>
<li>thermo: an internal control feature</li>
<li>trigrid: an internal gridding feature</li>
</ul>
</li>
</ul>
</li>
</ul>
<h2><a class="anchor" id="implementation"></a>
Parsing and Writing</h2>
<p>The official C++ parser used by affy can be found in APT under <a class="el" href="PgfFile_8h.html" title="Headers for the PgfFile class.">sdk/file/TsvFile/PgfFile.h</a>. When possible, parsing and writing of PGF files should be done using this code.</p>
<h2><a class="anchor" id="notes"></a>
Notes</h2>
<p>Specific applications may require extra/optional columns in the PGF file. Thus a valid PGF file may fail for a particular application or analysis algorithm because the information needed by that application and/or algorithm is not contained in the PGF file.</p>
<p>It should be noted that there is no significance to the ordering of probes within atoms and atoms within probesets or even probesets within the PGF file.</p>
<p>IDs do not have to be unique between different levels. In other words, the ID space for probeset_ids is separate from the ID space for atom_ids and probe_ids.</p>
<h2><a class="anchor" id="pgfExample1"></a>
Example 1 -- Human Exon 1.0 ST PGF File Excerpt</h2>
<div class="fragment"><pre class="fragment">
#%chip_type=HuEx-1_0-st-v2
#%chip_type=HuEx-1_0-st-v1
#%chip_type=HuEx-1_0-st-ta1
#%lib_set_name=HuEx-1_0-st
#%lib_set_version=r2
#%create_date=Tue Sep 19 15:18:05 PDT 2006
#%guid=0000008635-1158704285-0183259307-0389325148-0127012107
#%pgf_format_version=1.0
#%header0=probeset_id type
#%header1= atom_id
#%header2=  probe_id type gc_count probe_length interrogation_position probe_sequence
2590411 main
 1
  5402769 pm:st 12 25 13 CGAAGTTGTTCATTTCCCCGAAGAC
 2
  4684894 pm:st 13 25 13 ATGAGGTCACGACGGTAGGACTAAC
 3
  3869021 pm:st 11 25 13 AGGAGTACAGGGTAAGATATGGTCT
 4
  3774604 pm:st 14 25 13 CCCCGAAGACCCTAAGATGAGGTCA
...
</pre></div><h2><a class="anchor" id="pgfExample2"></a>
Example 2 -- Human Genome U133 2.0 Plus PGF File Excerpt</h2>
<p>Here is a hypothetical example of a PGF file for an expression array with PM/MM pairs.</p>
<div class="fragment"><pre class="fragment">
#%pgf_format_version=1.0
#%chip_type=HG-U133_Plus_2
#%lib_set_name=HG-U133_Plus_2
#%lib_set_version=1
#%create_date=Tue Mar 29 16:48:05 2005
#%header0=probeset_id type probeset_name
#%header1= atom_id
#%header2=  probe_id type gc_count probe_length interrogation_position probe_sequence exon_position
1354897  204339_s_at
 1354898
  1221821 pm:target-&gt;at 13 25 13 ACAACGACCGTTCCGGAATCGACAT 1703
  1222985 mm:target-&gt;at 13 25 13 ACAACGACCGTTGCGGAATCGACAT 1703
 1354899
  788881 pm:target-&gt;at 8 25 13 TTACATCATACTTTCTTGTCTCTAG 1355
  790045 mm:target-&gt;at 8 25 13 TTACATCATACTATCTTGTCTCTAG 1355
 1354900
  516645 pm:target-&gt;at 12 25 13 GAATCTTATCACGAGTTCCACCTCC 1518
  517809 mm:target-&gt;at 12 25 13 GAATCTTATCACCAGTTCCACCTCC 1518
 1354901
  736948 pm:target-&gt;at 12 25 13 GAGTTCCCTCGATTCCATAAACGAG 1543
</pre></div><h2><a class="anchor" id="related"></a>
Related Pages</h2>
<ul>
<li><a href="./file-format-tsv-notes.html">TsvFile Design Notes</a></li>
<li><a href="./file-format-tsv.html">TSV File Format</a></li>
</ul>
<h2><a class="anchor" id="pgf_dev_notes"></a>
Developer Notes</h2>
<ul>
<li>If you are defining a file format which is based on TsvFile, use this as an example.</li>
<li>PgfFile is not subclassed as we dont want it to inherit methods from TsvFile. This prevents calling a method by accident.</li>
<li>m_tsv is a member so it appears when a PgfFile is made. (ie: minus messy memory management)</li>
<li>We make "tsv" public so if the programmer wants to work with the underlying tsv object may do so. </li>
</ul>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:58 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
