<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: translation/GenotypeTableModel.cpp Source File</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li class="current"><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="tabs2">
    <ul class="tablist">
      <li><a href="files.html"><span>File&nbsp;List</span></a></li>
      <li><a href="globals.html"><span>File&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath">
    <ul>
      <li><a class="el" href="dir_1668076c0e919fc49032c4fb42e3bb3d.html">translation</a>      </li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>GenotypeTableModel.cpp</h1>  </div>
</div>
<div class="contents">
<a href="GenotypeTableModel_8cpp.html">Go to the documentation of this file.</a><div class="fragment"><pre class="fragment"><a name="l00001"></a>00001 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00002"></a>00002 <span class="comment"></span><span class="comment">//</span>
<a name="l00003"></a>00003 <span class="comment">// Copyright (C) 2008 Affymetrix, Inc.</span>
<a name="l00004"></a>00004 <span class="comment">//</span>
<a name="l00005"></a>00005 <span class="comment">// This program is free software; you can redistribute it and/or modify </span>
<a name="l00006"></a>00006 <span class="comment">// it under the terms of the GNU General Public License (version 2) as </span>
<a name="l00007"></a>00007 <span class="comment">// published by the Free Software Foundation.</span>
<a name="l00008"></a>00008 <span class="comment">// </span>
<a name="l00009"></a>00009 <span class="comment">// This program is distributed in the hope that it will be useful, </span>
<a name="l00010"></a>00010 <span class="comment">// but WITHOUT ANY WARRANTY; without even the implied warranty of </span>
<a name="l00011"></a>00011 <span class="comment">// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU </span>
<a name="l00012"></a>00012 <span class="comment">// General Public License for more details.</span>
<a name="l00013"></a>00013 <span class="comment">// </span>
<a name="l00014"></a>00014 <span class="comment">// You should have received a copy of the GNU General Public License </span>
<a name="l00015"></a>00015 <span class="comment">// along with this program;if not, write to the </span>
<a name="l00016"></a>00016 <span class="comment">// </span>
<a name="l00017"></a>00017 <span class="comment">// Free Software Foundation, Inc., </span>
<a name="l00018"></a>00018 <span class="comment">// 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA</span>
<a name="l00019"></a>00019 <span class="comment">//</span><span class="comment"></span>
<a name="l00020"></a>00020 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00021"></a>00021 <span class="comment"></span><span class="comment"></span>
<a name="l00022"></a>00022 <span class="comment">/**</span>
<a name="l00023"></a>00023 <span class="comment"> * @file   GenotypeTableModel.cpp</span>
<a name="l00024"></a>00024 <span class="comment"> * @author Mybrid Spalding</span>
<a name="l00025"></a>00025 <span class="comment"> * @date   Tue Mar 18 10:15:23 PDT 2008</span>
<a name="l00026"></a>00026 <span class="comment"> * @brief  Genotype Data (CHP) class</span>
<a name="l00027"></a>00027 <span class="comment"> */</span>
<a name="l00028"></a>00028 
<a name="l00029"></a>00029 
<a name="l00030"></a>00030 <span class="preprocessor">#include &quot;<a class="code" href="GenotypeTableModel_8h.html" title="Genotype Data (CHP) class.">translation/GenotypeTableModel.h</a>&quot;</span>
<a name="l00031"></a>00031 <span class="comment">//</span>
<a name="l00032"></a>00032 <span class="preprocessor">#include &quot;<a class="code" href="CallElement_8h.html" title="Atomic class element for translation.">translation/CallElement.h</a>&quot;</span>
<a name="l00033"></a>00033 <span class="preprocessor">#include &quot;<a class="code" href="TranslationTableModel_8h.html" title="Single instance data model class that wraps the translation file in the TranslationInputTsvModel.">translation/TranslationTableModel.h</a>&quot;</span>
<a name="l00034"></a>00034 <span class="comment">//</span>
<a name="l00035"></a>00035 <span class="preprocessor">#include &quot;<a class="code" href="Err_8h.html" title="Some utility error functions, currently prints message and aborts().">util/Err.h</a>&quot;</span>  <span class="comment">// includes Verbose.h</span>
<a name="l00036"></a>00036 <span class="comment">//</span>
<a name="l00037"></a>00037 <span class="preprocessor">#include &lt;sstream&gt;</span>
<a name="l00038"></a>00038 <span class="comment">//</span>
<a name="l00039"></a>00039 
<a name="l00040"></a>00040 <span class="keyword">using namespace </span>affx;
<a name="l00041"></a>00041 <span class="keyword">using namespace </span>std;
<a name="l00042"></a>00042 
<a name="l00043"></a>00043 <span class="comment">//Sample Name Experiment Name Gene External Id ProbeSet Id Allele 1 Allele 2</span>
<a name="l00044"></a>00044 <span class="comment">//001-000505 96dev_27_001-000505 ABCB1 MDR1_A-44G 367417 A G</span>
<a name="l00045"></a>00045 <span class="keyword">const</span> <a class="code" href="classTittmColumnDefinition.html">TittmColumnDefinition</a> GTM_COLUMN_DEFINITIONS[] = { <span class="comment">// column_name, empty_ok, index, valid_regex</span>
<a name="l00046"></a>00046   { std::string(<span class="stringliteral">&quot;Sample Name&quot;</span>),     0,  0, std::string(<span class="stringliteral">&quot;.*&quot;</span>), NULL },
<a name="l00047"></a>00047   { std::string(<span class="stringliteral">&quot;Experiment Name&quot;</span>), 0,  1, std::string(<span class="stringliteral">&quot;.*&quot;</span>), NULL },
<a name="l00048"></a>00048   { std::string(<span class="stringliteral">&quot;Gene&quot;</span>),            0,  2, std::string(<span class="stringliteral">&quot;^[A-Z\\d]+$&quot;</span>), NULL},
<a name="l00049"></a>00049   { std::string(<span class="stringliteral">&quot;External Id&quot;</span>),     0,  3, std::string(<span class="stringliteral">&quot;.*&quot;</span>), NULL  },
<a name="l00050"></a>00050   { std::string(<span class="stringliteral">&quot;Assay Id&quot;</span>),        0,  4, std::string(<span class="stringliteral">&quot;^(?:[_\\-\\w\\d]*\\d+)$&quot;</span>), NULL },
<a name="l00051"></a>00051   { std::string(<span class="stringliteral">&quot;Allele 1&quot;</span>),        0,  5,
<a name="l00052"></a>00052     std::string(<span class="stringliteral">&quot;^(?:[ACGT0]+)|(?:INS|Ins|DEL|Del|NC|PRA|NoCall|PossibleRareAllele|\\-)$&quot;</span>), NULL },
<a name="l00053"></a>00053   { std::string(<span class="stringliteral">&quot;Allele 2&quot;</span>),        1,  6,
<a name="l00054"></a>00054     std::string(<span class="stringliteral">&quot;^(?:[ACGT0]+)|(?:INS|Ins|DEL|Del|NC|PRA|NoCall|PossibleRareAllele|\\-)$&quot;</span>), NULL },
<a name="l00055"></a>00055 };
<a name="l00056"></a>00056 
<a name="l00057"></a>00057 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00058"></a>00058 <span class="comment">/**</span>
<a name="l00059"></a>00059 <span class="comment"> * GenotypeTableModel::GenotypeTableModel</span>
<a name="l00060"></a>00060 <span class="comment"> * Synopsis:</span>
<a name="l00061"></a>00061 <span class="comment"> *</span>
<a name="l00062"></a>00062 <span class="comment"> * DMET2: TsvFile constructor.</span>
<a name="l00063"></a>00063 <span class="comment"> *</span>
<a name="l00064"></a>00064 <span class="comment"> * This is a single instance class where only one of these should ever</span>
<a name="l00065"></a>00065 <span class="comment"> * be in existence:</span>
<a name="l00066"></a>00066 <span class="comment"> *</span>
<a name="l00067"></a>00067 <span class="comment"> * genotype data file -&gt; genotype table model.</span>
<a name="l00068"></a>00068 <span class="comment"> *</span>
<a name="l00069"></a>00069 <span class="comment"> * A quick glance at the column headers defined for this table shows</span>
<a name="l00070"></a>00070 <span class="comment"> * that the schema for this table looks nothing like what&#39;s in a CHP</span>
<a name="l00071"></a>00071 <span class="comment"> * file header.</span>
<a name="l00072"></a>00072 <span class="comment"> *</span>
<a name="l00073"></a>00073 <span class="comment"> * A decision was made early on that the only data to be used for DMET3</span>
<a name="l00074"></a>00074 <span class="comment"> * translation with the CHP file would be a subset of the DMET2 data.</span>
<a name="l00075"></a>00075 <span class="comment"> * Therefore the DMET2 table format was left in place.</span>
<a name="l00076"></a>00076 <span class="comment"> *</span>
<a name="l00077"></a>00077 <span class="comment"> * This table is the only object to stream data. This model then is the</span>
<a name="l00078"></a>00078 <span class="comment"> * closest to being a data access object. The transfer object is simply the</span>
<a name="l00079"></a>00079 <span class="comment"> * rows (m_rows) and the &quot;readNextExperiment&quot; accessor is the data</span>
<a name="l00080"></a>00080 <span class="comment"> * access method. This is unlike other &quot;data access objects&quot; using the</span>
<a name="l00081"></a>00081 <span class="comment"> * &quot;TranslationInputTsvTableModel&quot; where all data is read into memory</span>
<a name="l00082"></a>00082 <span class="comment"> * at construction.</span>
<a name="l00083"></a>00083 <span class="comment"> *</span>
<a name="l00084"></a>00084 <span class="comment"> * The business object that consumes this data access / transfer object</span>
<a name="l00085"></a>00085 <span class="comment"> * is the &quot;TranslationTable&quot; business object. The &quot;TranslationTable&quot; object</span>
<a name="l00086"></a>00086 <span class="comment"> * creates chromatid call sets from the genotype data transfered.</span>
<a name="l00087"></a>00087 <span class="comment"> *</span>
<a name="l00088"></a>00088 <span class="comment"> * When this object is constructed, no data is read in. To access</span>
<a name="l00089"></a>00089 <span class="comment"> * data use the &quot;readNextExperiment&quot; accessor and the next</span>
<a name="l00090"></a>00090 <span class="comment"> * experiment&#39;s worth of data will be read in, albiet from</span>
<a name="l00091"></a>00091 <span class="comment"> * a DMET2 genotype short report file or a DMET3 CHP file.</span>
<a name="l00092"></a>00092 <span class="comment"> *</span>
<a name="l00093"></a>00093 <span class="comment"> *</span>
<a name="l00094"></a>00094 <span class="comment"> *</span>
<a name="l00095"></a>00095 <span class="comment"> * @param rte - the single instance RunTimeEnvironment which contains options</span>
<a name="l00096"></a>00096 <span class="comment"> * @param genoFileName - the input file to read in.</span>
<a name="l00097"></a>00097 <span class="comment"> * @param gotm - the single instance of the genotype override file table model</span>
<a name="l00098"></a>00098 <span class="comment"> */</span>
<a name="l00099"></a>00099 <span class="comment">/*****************************************************************************/</span>
<a name="l00100"></a><a class="code" href="classGenotypeTableModel.html#ad46742dcc88af3643d6f1c4190ebf022">00100</a> <a class="code" href="classGenotypeTableModel.html#ad46742dcc88af3643d6f1c4190ebf022" title="GenotypeTableModel::GenotypeTableModel Synopsis:">GenotypeTableModel::GenotypeTableModel</a>(<span class="keyword">const</span> <a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp;rte,  <span class="keyword">const</span> std::string &amp;genoFileName, <a class="code" href="classGenotypeOverrideTableModel.html">GenotypeOverrideTableModel</a> *gotm) :
<a name="l00101"></a>00101     <a class="code" href="classTranslationInputStreamTableModel.html">TranslationInputStreamTableModel</a>(rte, genoFileName, GTM_COLUMN_DEFINITIONS,
<a name="l00102"></a>00102                                      (sizeof(GTM_COLUMN_DEFINITIONS) / sizeof(GTM_COLUMN_DEFINITIONS[0])), true, GT_EXPERIMENT_INDEX, GT_GENES_INDEX, GT_PROBE_SET_INDEX, gotm)
<a name="l00103"></a>00103 {
<a name="l00104"></a>00104 
<a name="l00105"></a>00105 
<a name="l00106"></a>00106   <span class="keywordflow">return</span>;
<a name="l00107"></a>00107 
<a name="l00108"></a>00108 }
<a name="l00109"></a>00109 <span class="comment">// end GenotypeTableModel::GenotypeTableModel</span>
<a name="l00110"></a>00110 <span class="comment">/*****************************************************************************/</span>
<a name="l00111"></a>00111 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00112"></a>00112 <span class="comment">/**</span>
<a name="l00113"></a>00113 <span class="comment"> * GenotypeTableModel::GenotypeTableModel</span>
<a name="l00114"></a>00114 <span class="comment"> * Synopsis</span>
<a name="l00115"></a>00115 <span class="comment"> *</span>
<a name="l00116"></a>00116 <span class="comment"> * DMET3 CHP constructor to coalesce the GenotypeTableModel with the</span>
<a name="l00117"></a>00117 <span class="comment"> * DMET2 files. See the TSV constructor for a complete discussion.</span>
<a name="l00118"></a>00118 <span class="comment"> *</span>
<a name="l00119"></a>00119 <span class="comment"> * @param rte - RunTimeEnvironment which contains options</span>
<a name="l00120"></a>00120 <span class="comment"> * @param gotm - single instance of the genotype override table model</span>
<a name="l00121"></a>00121 <span class="comment"> *</span>
<a name="l00122"></a>00122 <span class="comment"> */</span>
<a name="l00123"></a>00123 <span class="comment">/*****************************************************************************/</span>
<a name="l00124"></a><a class="code" href="classGenotypeTableModel.html#a69c312c1684ea7269f7adfb0c5b9f1a0">00124</a> <a class="code" href="classGenotypeTableModel.html#ad46742dcc88af3643d6f1c4190ebf022" title="GenotypeTableModel::GenotypeTableModel Synopsis:">GenotypeTableModel::GenotypeTableModel</a>(<span class="keyword">const</span> <a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp;rte,  <a class="code" href="classGenotypeOverrideTableModel.html">GenotypeOverrideTableModel</a> *gotm) :
<a name="l00125"></a>00125     <a class="code" href="classTranslationInputStreamTableModel.html">TranslationInputStreamTableModel</a>(rte, GTM_COLUMN_DEFINITIONS,
<a name="l00126"></a>00126                                      (sizeof(GTM_COLUMN_DEFINITIONS) / sizeof(GTM_COLUMN_DEFINITIONS[0])), gotm)
<a name="l00127"></a>00127 {
<a name="l00128"></a>00128 
<a name="l00129"></a>00129   <span class="keywordflow">return</span>;
<a name="l00130"></a>00130 
<a name="l00131"></a>00131 }
<a name="l00132"></a>00132 <span class="comment">// end GenotypeTableModel::GenotypeTableModel</span>
<a name="l00133"></a>00133 <span class="comment">/*****************************************************************************/</span>
<a name="l00134"></a>00134 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00135"></a>00135 <span class="comment">/**</span>
<a name="l00136"></a>00136 <span class="comment"> * GenotypeTableModel::describeVerboseExperimentGene</span>
<a name="l00137"></a>00137 <span class="comment"> * Synopsis:</span>
<a name="l00138"></a>00138 <span class="comment"> *</span>
<a name="l00139"></a>00139 <span class="comment"> * Run time debug routine that can be invoked with various verbose levels.</span>
<a name="l00140"></a>00140 <span class="comment"> * ADT_VERBOSE_INPUT_FILES is the default level.</span>
<a name="l00141"></a>00141 <span class="comment"> *</span>
<a name="l00142"></a>00142 <span class="comment"> * @param rte       - run time environment</span>
<a name="l00143"></a>00143 <span class="comment"> * @param overrideLevel  - ADT_VERBOSE_INPUT_FILES by default,</span>
<a name="l00144"></a>00144 <span class="comment"> *                    otherwise the ADT_VERBOSE_ENUM level to output</span>
<a name="l00145"></a>00145 <span class="comment"> * @returns - void</span>
<a name="l00146"></a>00146 <span class="comment"> */</span>
<a name="l00147"></a>00147 <span class="comment">/*****************************************************************************/</span>
<a name="l00148"></a><a class="code" href="classGenotypeTableModel.html#a8ab297d27c47f4fee538d77b5bdaca95">00148</a> <span class="keywordtype">void</span> <a class="code" href="classGenotypeTableModel.html#a8ab297d27c47f4fee538d77b5bdaca95" title="GenotypeTableModel::describeVerboseExperimentGene Synopsis:">GenotypeTableModel::describeVerboseExperimentGene</a>(<span class="keyword">const</span> <a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp; rte, ADT_VERBOSE_ENUM overrideLevel)
<a name="l00149"></a>00149 {
<a name="l00150"></a>00150 
<a name="l00151"></a>00151   <span class="keywordflow">if</span> (!overrideLevel &amp;&amp; (rte.m_currentVerbosity &lt; ADT_VERBOSE_INPUT_FILES)) {
<a name="l00152"></a>00152     <span class="keywordflow">return</span>;
<a name="l00153"></a>00153   }
<a name="l00154"></a>00154 
<a name="l00155"></a>00155   std::stringstream msgSStr;
<a name="l00156"></a>00156 
<a name="l00157"></a>00157 
<a name="l00158"></a>00158   ADT_VERBOSE_ENUM outLevel = overrideLevel ? overrideLevel : rte.m_currentVerbosity;
<a name="l00159"></a>00159 
<a name="l00160"></a>00160   <span class="keywordflow">if</span> (m_geneName.empty()) {
<a name="l00161"></a>00161 
<a name="l00162"></a>00162     msgSStr &lt;&lt;  <span class="stringliteral">&quot;describeVerboseExperimentGene: no experiment gene data has been read yet.&quot;</span> &lt;&lt; endl;
<a name="l00163"></a>00163 
<a name="l00164"></a>00164   }
<a name="l00165"></a>00165 
<a name="l00166"></a>00166   <span class="keywordflow">else</span> {
<a name="l00167"></a>00167 
<a name="l00168"></a>00168     msgSStr &lt;&lt; <span class="stringliteral">&quot;describeVerboseExperimentGene: &quot;</span> &lt;&lt; m_experimentName;
<a name="l00169"></a>00169     msgSStr &lt;&lt; <span class="stringliteral">&quot; &quot;</span> &lt;&lt; m_geneName &lt;&lt; endl;
<a name="l00170"></a>00170 
<a name="l00171"></a>00171     <span class="keywordflow">for</span> (<span class="keywordtype">int</span> row = 0; row &lt; m_rows.size(); row++) {
<a name="l00172"></a>00172       msgSStr &lt;&lt; <a class="code" href="classTranslationInputStreamTableModel.html#a9dd29d23cbf5a0d0320a905f017fd4ea" title="TranslationInputStreamTableModel::getRowAsString Synopsis:">getRowAsString</a>(row) &lt;&lt; endl;
<a name="l00173"></a>00173 
<a name="l00174"></a>00174     }
<a name="l00175"></a>00175   }
<a name="l00176"></a>00176 
<a name="l00177"></a>00177   <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(outLevel, msgSStr.str());
<a name="l00178"></a>00178 
<a name="l00179"></a>00179   <span class="keywordflow">return</span>;
<a name="l00180"></a>00180 }
<a name="l00181"></a>00181 
<a name="l00182"></a>00182 <span class="comment">//end GenotypeTableModel::describeVerobseExperimentGene</span>
<a name="l00183"></a>00183 <span class="comment">/*****************************************************************************/</span>
<a name="l00184"></a>00184 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00185"></a>00185 <span class="comment">/**</span>
<a name="l00186"></a>00186 <span class="comment"> * getGeneRowSize:</span>
<a name="l00187"></a>00187 <span class="comment"> * Synopsis:</span>
<a name="l00188"></a>00188 <span class="comment"> *</span>
<a name="l00189"></a>00189 <span class="comment"> *   Returns the number of rows in the experiment m_rows table for one gene.</span>
<a name="l00190"></a>00190 <span class="comment"> * Rows are guaranteed to be in gene order.</span>
<a name="l00191"></a>00191 <span class="comment"> *</span>
<a name="l00192"></a>00192 <span class="comment"> * @params gene - the gene std::string to test.</span>
<a name="l00193"></a>00193 <span class="comment"> *</span>
<a name="l00194"></a>00194 <span class="comment"> * @return - The number of gene related rows, 0 if the gene doesn&#39;t exist.</span>
<a name="l00195"></a>00195 <span class="comment"> */</span>
<a name="l00196"></a>00196 <span class="comment">/*****************************************************************************/</span>
<a name="l00197"></a><a class="code" href="classGenotypeTableModel.html#ad413a9b5e58f44ef33a42a641806cbed">00197</a> <span class="keywordtype">int</span> <a class="code" href="classGenotypeTableModel.html#ad413a9b5e58f44ef33a42a641806cbed" title="getGeneRowSize: Synopsis:">GenotypeTableModel::getGeneRowSize</a>(<span class="keyword">const</span> std::string &amp; gene)
<a name="l00198"></a>00198 {
<a name="l00199"></a>00199 
<a name="l00200"></a>00200   <span class="keywordflow">if</span> (m_geneRowIndex.count(gene) == 0) {
<a name="l00201"></a>00201     <span class="keywordflow">return</span> 0;
<a name="l00202"></a>00202   }
<a name="l00203"></a>00203 
<a name="l00204"></a>00204   <span class="keywordflow">return</span> m_geneRowIndex[gene].size();
<a name="l00205"></a>00205 
<a name="l00206"></a>00206 }
<a name="l00207"></a>00207 <span class="comment">// end GenotypeTableModel::getGeneRowSize</span>
<a name="l00208"></a>00208 <span class="comment">/*****************************************************************************/</span>
<a name="l00209"></a>00209 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00210"></a>00210 <span class="comment">/**</span>
<a name="l00211"></a>00211 <span class="comment"> * GenotypeTableModel::getGeneRow:</span>
<a name="l00212"></a>00212 <span class="comment"> * Synopsis:</span>
<a name="l00213"></a>00213 <span class="comment"> *</span>
<a name="l00214"></a>00214 <span class="comment"> *   Returns index std::map from gene row to experiment row index.</span>
<a name="l00215"></a>00215 <span class="comment"> *</span>
<a name="l00216"></a>00216 <span class="comment"> *</span>
<a name="l00217"></a>00217 <span class="comment"> * @return - the row integer, -1 if the gene or row do not exist.</span>
<a name="l00218"></a>00218 <span class="comment"> */</span>
<a name="l00219"></a>00219 <span class="comment">/*****************************************************************************/</span>
<a name="l00220"></a><a class="code" href="classGenotypeTableModel.html#a82ce9bc2679a496b3592fddb14224f48">00220</a> <span class="keywordtype">int</span> <a class="code" href="classGenotypeTableModel.html#a82ce9bc2679a496b3592fddb14224f48" title="GenotypeTableModel::getGeneRow: Synopsis:">GenotypeTableModel::getGeneRow</a>(<span class="keyword">const</span> std::string &amp; gene, <span class="keywordtype">int</span> row)
<a name="l00221"></a>00221 {
<a name="l00222"></a>00222 
<a name="l00223"></a>00223   <span class="keywordflow">if</span> ((m_geneRowIndex.count(gene) == 0) ||
<a name="l00224"></a>00224       (m_geneRowIndex[gene].size() &lt;= row)) {
<a name="l00225"></a>00225     <span class="keywordflow">return</span> -1;
<a name="l00226"></a>00226   }
<a name="l00227"></a>00227 
<a name="l00228"></a>00228   <span class="keywordflow">return</span> m_geneRowIndex[gene][row];
<a name="l00229"></a>00229 
<a name="l00230"></a>00230 }
<a name="l00231"></a>00231 <span class="comment">// end GenotypeTableModel::getGeneRow</span>
<a name="l00232"></a>00232 <span class="comment">/*****************************************************************************/</span>
<a name="l00233"></a>00233 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00234"></a>00234 <span class="comment">/**</span>
<a name="l00235"></a>00235 <span class="comment"> * GenotypeTableModel::_validateGenotypes:</span>
<a name="l00236"></a>00236 <span class="comment"> * Synopsis:</span>
<a name="l00237"></a>00237 <span class="comment"> *</span>
<a name="l00238"></a>00238 <span class="comment"> * Helper function used one time by GenotypeTableModel::getNextExperiment.</span>
<a name="l00239"></a>00239 <span class="comment"> * Broken out for readability.</span>
<a name="l00240"></a>00240 <span class="comment"> *</span>
<a name="l00241"></a>00241 <span class="comment"> *   Validate that the experiment genotype corresponds to either the reference</span>
<a name="l00242"></a>00242 <span class="comment"> *  or one of the variant bases in the translation table.</span>
<a name="l00243"></a>00243 <span class="comment"> *</span>
<a name="l00244"></a>00244 <span class="comment"> *</span>
<a name="l00245"></a>00245 <span class="comment"> *</span>
<a name="l00246"></a>00246 <span class="comment"> *</span>
<a name="l00247"></a>00247 <span class="comment"> *</span>
<a name="l00248"></a>00248 <span class="comment"> * @param rte         - the single instance run time environment</span>
<a name="l00249"></a>00249 <span class="comment"> * @param ttm         - the single instance translation table model</span>
<a name="l00250"></a>00250 <span class="comment"> * @param probeSetId  - the probeSetId in question</span>
<a name="l00251"></a>00251 <span class="comment"> * @param base1       - allele</span>
<a name="l00252"></a>00252 <span class="comment"> * @param base2       - allele</span>
<a name="l00253"></a>00253 <span class="comment"> */</span>
<a name="l00254"></a>00254 <span class="comment">/*****************************************************************************/</span>
<a name="l00255"></a>00255 <span class="keyword">static</span> <span class="keywordtype">void</span> _validateGenotypes(<a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp; rte,
<a name="l00256"></a>00256                                <span class="keyword">const</span> <a class="code" href="classTranslationTableModel.html">TranslationTableModel</a> &amp; ttm,
<a name="l00257"></a>00257                                <span class="keyword">const</span> std::string &amp; probeSetId,
<a name="l00258"></a>00258                                <span class="keyword">const</span> std::string &amp; base1,
<a name="l00259"></a>00259                                <span class="keyword">const</span> std::string &amp; base2)
<a name="l00260"></a>00260 {
<a name="l00261"></a>00261 
<a name="l00262"></a>00262   std::map&lt; std::string, std::vector&lt; int &gt;  &gt;::const_iterator itSVI;
<a name="l00263"></a>00263 
<a name="l00264"></a>00264   itSVI = ttm.m_probeSetRowIndex.find(probeSetId);
<a name="l00265"></a>00265 
<a name="l00266"></a>00266   <span class="comment">// It&#39;s given that experiments will contain probes not in the</span>
<a name="l00267"></a>00267   <span class="comment">// translation table. Those are perfectly valid and dealt with</span>
<a name="l00268"></a>00268   <span class="comment">// during translation.</span>
<a name="l00269"></a>00269   <span class="keywordflow">if</span> (itSVI == ttm.m_probeSetRowIndex.end()) {
<a name="l00270"></a>00270     <span class="keywordflow">return</span>;
<a name="l00271"></a>00271   }
<a name="l00272"></a>00272 
<a name="l00273"></a>00273   <span class="keywordtype">bool</span> base1OK = <a class="code" href="classCallElement.html#aae6a8be2dfee570fe4ddd451cb8a74c3" title="CallElement::isWildCardBase Synopsis:">CallElement::isWildCardBase</a>(base1);
<a name="l00274"></a>00274   <span class="keywordtype">bool</span> base2OK = <a class="code" href="classCallElement.html#aae6a8be2dfee570fe4ddd451cb8a74c3" title="CallElement::isWildCardBase Synopsis:">CallElement::isWildCardBase</a>(base2);
<a name="l00275"></a>00275 
<a name="l00276"></a>00276   <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; (i &lt; itSVI-&gt;second.size()) &amp;&amp; (!base1OK || !base2OK); i++) {
<a name="l00277"></a>00277 
<a name="l00278"></a>00278     <span class="keywordtype">int</span> row = itSVI-&gt;second[i];
<a name="l00279"></a>00279 
<a name="l00280"></a>00280     <span class="keywordflow">if</span> (!base1OK) {
<a name="l00281"></a>00281 
<a name="l00282"></a>00282       <span class="keywordflow">if</span> ((ttm.getType() == ADT_TRANSLATION_TABLE_TYPE_DMET3) &amp;&amp; (base1 == <span class="stringliteral">&quot;0&quot;</span>)) {
<a name="l00283"></a>00283         base1OK = <span class="keyword">true</span>;
<a name="l00284"></a>00284       } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (ttm.m_rows[row][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_REFERENCE)] == base1) {
<a name="l00285"></a>00285         base1OK = <span class="keyword">true</span>;
<a name="l00286"></a>00286       } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (ttm.m_rows[row][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_VARIANT)] == base1) {
<a name="l00287"></a>00287         base1OK = <span class="keyword">true</span>;
<a name="l00288"></a>00288       }
<a name="l00289"></a>00289     }
<a name="l00290"></a>00290     <span class="keywordflow">if</span> (!base2OK) {
<a name="l00291"></a>00291       <span class="keywordflow">if</span> ((ttm.getType() == ADT_TRANSLATION_TABLE_TYPE_DMET3) &amp;&amp; (base2 == <span class="stringliteral">&quot;&quot;</span>)) {
<a name="l00292"></a>00292         base2OK = <span class="keyword">true</span>;
<a name="l00293"></a>00293       }
<a name="l00294"></a>00294       <span class="keywordflow">if</span> (ttm.m_rows[row][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_REFERENCE)] == base2) {
<a name="l00295"></a>00295         base2OK = <span class="keyword">true</span>;
<a name="l00296"></a>00296       } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (ttm.m_rows[row][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_VARIANT)] == base2) {
<a name="l00297"></a>00297         base2OK = <span class="keyword">true</span>;
<a name="l00298"></a>00298       }
<a name="l00299"></a>00299     }
<a name="l00300"></a>00300   }
<a name="l00301"></a>00301 
<a name="l00302"></a>00302   <span class="keywordflow">if</span> (!base1OK) {
<a name="l00303"></a>00303     std::stringstream msgSStr;
<a name="l00304"></a>00304     msgSStr &lt;&lt; probeSetId &lt;&lt; <span class="stringliteral">&quot; has unknown chromatid 1 base &quot;</span> &lt;&lt; base1 &lt;&lt; <span class="stringliteral">&quot; not found in translation file.&quot;</span>;
<a name="l00305"></a>00305     <span class="keywordflow">if</span> (rte.m_adtOpts.m_ignoreUnknownAlleles) {
<a name="l00306"></a>00306       <a class="code" href="classVerbose.html#a34e1d936178741eeb78de93c5497e1b8" title="Print a warning message.">Verbose::warn</a>(ADT_VERBOSE_EXCEPTION, msgSStr.str());
<a name="l00307"></a>00307     } <span class="keywordflow">else</span> {
<a name="l00308"></a>00308       <a class="code" href="Err_8h.html#a7e26a673d8901b5bc880dfab285f5309" title="Calls Err::apt_err_abort with the filename and linenumber set.">APT_ERR_ABORT</a>(msgSStr.str());
<a name="l00309"></a>00309     }
<a name="l00310"></a>00310   } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (!base2OK) {
<a name="l00311"></a>00311     std::stringstream msgSStr;
<a name="l00312"></a>00312     msgSStr &lt;&lt; probeSetId &lt;&lt; <span class="stringliteral">&quot; has unknown chromatid 2 base &quot;</span> &lt;&lt; base2 &lt;&lt; <span class="stringliteral">&quot; not found in translation file.&quot;</span>;
<a name="l00313"></a>00313     <span class="keywordflow">if</span> (rte.m_adtOpts.m_ignoreUnknownAlleles) {
<a name="l00314"></a>00314       <a class="code" href="classVerbose.html#a34e1d936178741eeb78de93c5497e1b8" title="Print a warning message.">Verbose::warn</a>(ADT_VERBOSE_EXCEPTION, msgSStr.str());
<a name="l00315"></a>00315     } <span class="keywordflow">else</span> {
<a name="l00316"></a>00316       <a class="code" href="Err_8h.html#a7e26a673d8901b5bc880dfab285f5309" title="Calls Err::apt_err_abort with the filename and linenumber set.">APT_ERR_ABORT</a>(msgSStr.str());
<a name="l00317"></a>00317     }
<a name="l00318"></a>00318   }
<a name="l00319"></a>00319 
<a name="l00320"></a>00320   <span class="keywordflow">return</span>;
<a name="l00321"></a>00321 }
<a name="l00322"></a>00322 <span class="comment">// end _validateGenotypes</span>
<a name="l00323"></a>00323 <span class="comment">/*****************************************************************************/</span>
<a name="l00324"></a>00324 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00325"></a>00325 <span class="comment">/**</span>
<a name="l00326"></a>00326 <span class="comment"> * GenotypeTableModel::getNextExperiment:</span>
<a name="l00327"></a>00327 <span class="comment"> * Synopsis:</span>
<a name="l00328"></a>00328 <span class="comment"> *</span>
<a name="l00329"></a>00329 <span class="comment"> *  This API is simply a wrapper for</span>
<a name="l00330"></a>00330 <span class="comment"> *  TranslationInputStreamTableModel-&gt;readNextExperiment.</span>
<a name="l00331"></a>00331 <span class="comment"> *</span>
<a name="l00332"></a>00332 <span class="comment"> *  Records read in are then scanned again for validation and reformating.</span>
<a name="l00333"></a>00333 <span class="comment"> *</span>
<a name="l00334"></a>00334 <span class="comment"> *</span>
<a name="l00335"></a>00335 <span class="comment"> * @param rte - the single instance RunTimeEnvironment</span>
<a name="l00336"></a>00336 <span class="comment"> * @param ttm - the single instance translation table model</span>
<a name="l00337"></a>00337 <span class="comment"> * @param geneExperimentCopyNumberCall - the copy &quot;allele&quot; designation per gene</span>
<a name="l00338"></a>00338 <span class="comment"> *</span>
<a name="l00339"></a>00339 <span class="comment"> * @return - the number of records read.</span>
<a name="l00340"></a>00340 <span class="comment"> */</span>
<a name="l00341"></a>00341 <span class="comment">/*****************************************************************************/</span>
<a name="l00342"></a><a class="code" href="classGenotypeTableModel.html#ae393d55ec62a42dd5c04b68d7183d6b9">00342</a> <span class="keywordtype">int</span> <a class="code" href="classGenotypeTableModel.html#ae393d55ec62a42dd5c04b68d7183d6b9" title="GenotypeTableModel::getNextExperiment: Synopsis:">GenotypeTableModel::getNextExperiment</a>(<a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp; rte,
<a name="l00343"></a>00343     <a class="code" href="classTranslationTableModel.html">TranslationTableModel</a> &amp; ttm,
<a name="l00344"></a>00344     <span class="keyword">const</span> std::map&lt;std::string, std::string&gt; &amp;geneExperimentCopyNumberCall)
<a name="l00345"></a>00345 {
<a name="l00346"></a>00346 
<a name="l00347"></a>00347 
<a name="l00348"></a>00348   <span class="keywordflow">if</span> (rte.m_adtOpts.m_profile) {
<a name="l00349"></a>00349     rte.m_profiles[<span class="stringliteral">&quot;readNextExperiment&quot;</span>]-&gt;begin();
<a name="l00350"></a>00350   }
<a name="l00351"></a>00351 
<a name="l00352"></a>00352   m_geneCopyNumber.clear();
<a name="l00353"></a>00353   
<a name="l00354"></a>00354   <span class="keywordtype">size_t</span> numRecords = <a class="code" href="classTranslationInputStreamTableModel.html#ad0b554465fe47987c3da766bd203a9eb" title="TranslationInputStreamTableModel::readNextExperiment: Synopsis:">readNextExperiment</a>(rte, ttm, geneExperimentCopyNumberCall, m_geneCopyNumber);
<a name="l00355"></a>00355 
<a name="l00356"></a>00356   <span class="keywordflow">if</span> (rte.m_adtOpts.m_profile) {
<a name="l00357"></a>00357     rte.m_profiles[<span class="stringliteral">&quot;readNextExperiment&quot;</span>]-&gt;end();
<a name="l00358"></a>00358   }
<a name="l00359"></a>00359 
<a name="l00360"></a>00360   m_geneRowIndex.clear();
<a name="l00361"></a>00361   m_experimentGenes.clear();
<a name="l00362"></a>00362 
<a name="l00363"></a>00363   <span class="keywordflow">if</span> (numRecords  &gt; 0) {
<a name="l00364"></a>00364 
<a name="l00365"></a>00365     <span class="keywordflow">for</span> (<span class="keywordtype">int</span> row = 0; row &lt; m_rows.size(); row++) {
<a name="l00366"></a>00366 
<a name="l00367"></a>00367       <span class="keywordflow">if</span> (row == 0) {
<a name="l00368"></a>00368         m_experimentName = m_rows[row][GT_EXPERIMENT_INDEX];
<a name="l00369"></a>00369         m_geneName       = m_rows[row][GT_GENES_INDEX];
<a name="l00370"></a>00370       }
<a name="l00371"></a>00371       m_geneRowIndex[m_rows[row][GT_GENES_INDEX]].push_back(row);
<a name="l00372"></a>00372 
<a name="l00373"></a>00373       <span class="keywordflow">if</span> (m_experimentGenes.count(m_rows[row][GT_GENES_INDEX]) == 0) {
<a name="l00374"></a>00374         m_experimentGenes.insert(m_rows[row][GT_GENES_INDEX]);
<a name="l00375"></a>00375       }
<a name="l00376"></a>00376 
<a name="l00377"></a>00377       <span class="keywordflow">if</span> (rte.m_adtOpts.m_dmet2Calling) {
<a name="l00378"></a>00378         <span class="keywordflow">if</span> (m_rows[row][GT_ALLELE1_INDEX] == <span class="stringliteral">&quot;Ins&quot;</span>) {
<a name="l00379"></a>00379           m_rows[row][GT_ALLELE1_INDEX] = <span class="stringliteral">&quot;INS&quot;</span>;
<a name="l00380"></a>00380         }
<a name="l00381"></a>00381         <span class="keywordflow">if</span> (m_rows[row][GT_ALLELE2_INDEX] == <span class="stringliteral">&quot;Ins&quot;</span>) {
<a name="l00382"></a>00382           m_rows[row][GT_ALLELE2_INDEX] = <span class="stringliteral">&quot;INS&quot;</span>;
<a name="l00383"></a>00383         }
<a name="l00384"></a>00384         <span class="keywordflow">if</span> (m_rows[row][GT_ALLELE1_INDEX] == <span class="stringliteral">&quot;Del&quot;</span>) {
<a name="l00385"></a>00385           m_rows[row][GT_ALLELE1_INDEX] = <span class="stringliteral">&quot;DEL&quot;</span>;
<a name="l00386"></a>00386         }
<a name="l00387"></a>00387         <span class="keywordflow">if</span> (m_rows[row][GT_ALLELE2_INDEX] == <span class="stringliteral">&quot;Del&quot;</span>) {
<a name="l00388"></a>00388           m_rows[row][GT_ALLELE2_INDEX] = <span class="stringliteral">&quot;DEL&quot;</span>;
<a name="l00389"></a>00389         }
<a name="l00390"></a>00390       }
<a name="l00391"></a>00391 
<a name="l00392"></a>00392       _validateGenotypes(rte, ttm, m_rows[row][GT_PROBE_SET_INDEX], m_rows[row][GT_ALLELE1_INDEX], m_rows[row][GT_ALLELE2_INDEX]);
<a name="l00393"></a>00393 
<a name="l00394"></a>00394 
<a name="l00395"></a>00395     }
<a name="l00396"></a>00396   } <span class="keywordflow">else</span> {
<a name="l00397"></a>00397     m_geneName = <span class="stringliteral">&quot;&quot;</span>;
<a name="l00398"></a>00398     m_experimentName = <span class="stringliteral">&quot;&quot;</span>;
<a name="l00399"></a>00399   }
<a name="l00400"></a>00400 
<a name="l00401"></a>00401   <span class="keywordflow">return</span>(numRecords);
<a name="l00402"></a>00402 
<a name="l00403"></a>00403 }
<a name="l00404"></a>00404 <span class="comment">// end GenotypeTableModel::getNextExperiment</span>
<a name="l00405"></a>00405 <span class="comment">/*****************************************************************************/</span>
</pre></div></div>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:48 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
