<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: chipstream/GenoUtility.cpp Source File</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li class="current"><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="tabs2">
    <ul class="tablist">
      <li><a href="files.html"><span>File&nbsp;List</span></a></li>
      <li><a href="globals.html"><span>File&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath">
    <ul>
      <li><a class="el" href="dir_f48957cb46a44244b1a86c58bd157ee8.html">chipstream</a>      </li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>GenoUtility.cpp</h1>  </div>
</div>
<div class="contents">
<a href="GenoUtility_8cpp.html">Go to the documentation of this file.</a><div class="fragment"><pre class="fragment"><a name="l00001"></a>00001 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00002"></a>00002 <span class="comment"></span><span class="comment">//</span>
<a name="l00003"></a>00003 <span class="comment">// Copyright (C) 2005 Affymetrix, Inc.</span>
<a name="l00004"></a>00004 <span class="comment">//</span>
<a name="l00005"></a>00005 <span class="comment">// This program is free software; you can redistribute it and/or modify </span>
<a name="l00006"></a>00006 <span class="comment">// it under the terms of the GNU General Public License (version 2) as </span>
<a name="l00007"></a>00007 <span class="comment">// published by the Free Software Foundation.</span>
<a name="l00008"></a>00008 <span class="comment">// </span>
<a name="l00009"></a>00009 <span class="comment">// This program is distributed in the hope that it will be useful, </span>
<a name="l00010"></a>00010 <span class="comment">// but WITHOUT ANY WARRANTY; without even the implied warranty of </span>
<a name="l00011"></a>00011 <span class="comment">// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU </span>
<a name="l00012"></a>00012 <span class="comment">// General Public License for more details.</span>
<a name="l00013"></a>00013 <span class="comment">// </span>
<a name="l00014"></a>00014 <span class="comment">// You should have received a copy of the GNU General Public License </span>
<a name="l00015"></a>00015 <span class="comment">// along with this program;if not, write to the </span>
<a name="l00016"></a>00016 <span class="comment">// </span>
<a name="l00017"></a>00017 <span class="comment">// Free Software Foundation, Inc., </span>
<a name="l00018"></a>00018 <span class="comment">// 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA</span>
<a name="l00019"></a>00019 <span class="comment">//</span><span class="comment"></span>
<a name="l00020"></a>00020 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00021"></a>00021 <span class="comment"></span><span class="comment"></span>
<a name="l00022"></a>00022 <span class="comment">/**</span>
<a name="l00023"></a>00023 <span class="comment"> * @file   GenoUtility.cpp</span>
<a name="l00024"></a>00024 <span class="comment"> * @author Earl Hubbell</span>
<a name="l00025"></a>00025 <span class="comment"> * @date   Aug 2 2006</span>
<a name="l00026"></a>00026 <span class="comment"> *</span>
<a name="l00027"></a>00027 <span class="comment"> */</span>
<a name="l00028"></a>00028 
<a name="l00029"></a>00029 <span class="comment">//</span>
<a name="l00030"></a>00030 <span class="preprocessor">#include &quot;<a class="code" href="GenoUtility_8h.html">chipstream/GenoUtility.h</a>&quot;</span>
<a name="l00031"></a>00031 <span class="comment">//</span>
<a name="l00032"></a>00032 <span class="preprocessor">#include &quot;<a class="code" href="QuantMethodFactory_8h.html" title="Factory class for making chip streams based on a string representation.">chipstream/QuantMethodFactory.h</a>&quot;</span>
<a name="l00033"></a>00033 <span class="comment">//</span>
<a name="l00034"></a>00034 <span class="preprocessor">#include &quot;<a class="code" href="TsvFile_8h.html" title="Headers for the TsvFile classes.">file/TsvFile/TsvFile.h</a>&quot;</span>
<a name="l00035"></a>00035 <span class="preprocessor">#include &quot;<a class="code" href="Convert_8h.html" title="Utility functions for converting between types.">util/Convert.h</a>&quot;</span>
<a name="l00036"></a>00036 <span class="preprocessor">#include &quot;<a class="code" href="Fs_8h.html" title="///">util/Fs.h</a>&quot;</span>
<a name="l00037"></a>00037 <span class="preprocessor">#include &quot;<a class="code" href="Util_8h.html" title="General Utilities.">util/Util.h</a>&quot;</span>
<a name="l00038"></a>00038 <span class="preprocessor">#include &quot;<a class="code" href="Verbose_8h.html" title="Class for doing logging and some command line ui.">util/Verbose.h</a>&quot;</span>
<a name="l00039"></a>00039 
<a name="l00040"></a>00040 <span class="keyword">using namespace </span>affx;
<a name="l00041"></a>00041 <span class="comment"></span>
<a name="l00042"></a>00042 <span class="comment">/**</span>
<a name="l00043"></a>00043 <span class="comment"> * This is the engine that takes data and transforms it into clustering spaces</span>
<a name="l00044"></a>00044 <span class="comment"> *</span>
<a name="l00045"></a>00045 <span class="comment"> * @param aVal - a allele probe summary</span>
<a name="l00046"></a>00046 <span class="comment"> * @param bVal - b allele probe summary</span>
<a name="l00047"></a>00047 <span class="comment"> * @param trans - transformation of data</span>
<a name="l00048"></a>00048 <span class="comment"> * @param K - stretching parameter</span>
<a name="l00049"></a>00049 <span class="comment"> * @param xVal - output difference of a vs b, transformed</span>
<a name="l00050"></a>00050 <span class="comment"> * @param yVal - output total amount of stuff, transformed</span>
<a name="l00051"></a>00051 <span class="comment"> */</span>
<a name="l00052"></a><a class="code" href="GenoUtility_8h.html#aacfe80c3e8bd468402502f56a7704b15">00052</a> <span class="keywordtype">void</span> <a class="code" href="GenoUtility_8cpp.html#aacfe80c3e8bd468402502f56a7704b15" title="This is the engine that takes data and transforms it into clustering spaces.">valueForTransformation</a>(<span class="keywordtype">double</span> aVal, <span class="keywordtype">double</span> bVal, <span class="keyword">enum</span> <a class="code" href="GenoUtility_8h.html#a0176e2f61ce5ea7a3495a511cf46b2c1">Transformation</a> trans,
<a name="l00053"></a>00053                                          <span class="keywordtype">double</span> K, <span class="keywordtype">double</span> &amp;xVal, <span class="keywordtype">double</span> &amp;yVal) {
<a name="l00054"></a>00054   <span class="keywordtype">double</span> c = 0.001;
<a name="l00055"></a>00055   <span class="keywordtype">double</span> denom = 0;
<a name="l00056"></a>00056   <span class="keywordflow">switch</span> (trans) {
<a name="l00057"></a>00057   <span class="keywordflow">case</span> <a class="code" href="GenoUtility_8h.html#a0176e2f61ce5ea7a3495a511cf46b2c1a6db9488b39205827082e765aa0e87297" title="Minus Vs Average.">MvA</a>: <span class="comment">// Minus vs Average</span>
<a name="l00058"></a>00058     <span class="keywordflow">if</span>(aVal == 0) aVal = c;
<a name="l00059"></a>00059     <span class="keywordflow">if</span>(bVal == 0) bVal = c;
<a name="l00060"></a>00060         <span class="comment">//note: reversed to be consistent</span>
<a name="l00061"></a>00061         <span class="comment">// xVal always contains comparison between alleles</span>
<a name="l00062"></a>00062     yVal = (<a class="code" href="Util_8h.html#a18af743c2cec4baeee9ffb27999ddaad" title="Log base two.">log2</a>(bVal) + <a class="code" href="Util_8h.html#a18af743c2cec4baeee9ffb27999ddaad" title="Log base two.">log2</a>(aVal)) / 2;
<a name="l00063"></a>00063     xVal = <a class="code" href="Util_8h.html#a18af743c2cec4baeee9ffb27999ddaad" title="Log base two.">log2</a>(aVal) - <a class="code" href="Util_8h.html#a18af743c2cec4baeee9ffb27999ddaad" title="Log base two.">log2</a>(bVal);
<a name="l00064"></a>00064     <span class="keywordflow">break</span>;
<a name="l00065"></a>00065   <span class="keywordflow">case</span> <a class="code" href="GenoUtility_8h.html#a0176e2f61ce5ea7a3495a511cf46b2c1a1bab0eb6be5fad1c5e78f95647686374" title="R vs Theta.">RvT</a>: <span class="comment">// R vs Theta (polar)</span>
<a name="l00066"></a>00066     <span class="comment">// Check to make sure we&#39;re not got to take log of 0</span>
<a name="l00067"></a>00067     denom = bVal;
<a name="l00068"></a>00068     <span class="keywordflow">if</span>(bVal == 0) bVal = c;
<a name="l00069"></a>00069         <span class="comment">// center &quot;Theta&quot; at zero</span>
<a name="l00070"></a>00070     xVal = atan(aVal/ denom)-0.7853982;
<a name="l00071"></a>00071     yVal = log( sqrt((aVal * aVal) + (bVal * bVal) + c) );
<a name="l00072"></a>00072     <span class="keywordflow">break</span>;
<a name="l00073"></a>00073   <span class="keywordflow">case</span>  <a class="code" href="GenoUtility_8h.html#a0176e2f61ce5ea7a3495a511cf46b2c1a651cbbe4e01a316f0c0084eee693bb50" title="Contrast Extremes Stretch (nee south san francisco).">CES</a>:
<a name="l00074"></a>00074     <a class="code" href="GenoUtility_8cpp.html#ac842841ae05672e63157ac80b8a016f6" title="Contrast Extremes Stretch.">ContrastExtremesStretch</a>(K, aVal, bVal, xVal, yVal);
<a name="l00075"></a>00075     <span class="keywordflow">break</span>;
<a name="l00076"></a>00076   <span class="keywordflow">case</span> <a class="code" href="GenoUtility_8h.html#a0176e2f61ce5ea7a3495a511cf46b2c1a0af5537cdc9498e453464b9ac2e4f97f" title="Contrast Centers Stretch (nee alternative south san francisco).">CCS</a>:
<a name="l00077"></a>00077     <a class="code" href="GenoUtility_8cpp.html#a1e7d2233042fe7b5a9fc83b4f79b545c" title="Contrast Centers Stretch.">ContrastCentersStretch</a>(K, aVal, bVal, xVal, yVal);
<a name="l00078"></a>00078     <span class="keywordflow">break</span>;
<a name="l00079"></a>00079   <span class="keywordflow">default</span>:
<a name="l00080"></a>00080     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;GenoUtility: valueforTransformation Don&#39;t recognize transformation type: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(trans));
<a name="l00081"></a>00081   }
<a name="l00082"></a>00082 }
<a name="l00083"></a>00083 <span class="comment"></span>
<a name="l00084"></a>00084 <span class="comment">/**</span>
<a name="l00085"></a>00085 <span class="comment"> * Convert a (text insensitive) text repesentation of a transformation into</span>
<a name="l00086"></a>00086 <span class="comment"> * the enumration.</span>
<a name="l00087"></a>00087 <span class="comment"> * @param s - string specifying transformation.</span>
<a name="l00088"></a>00088 <span class="comment"> * @return - Enumeration version of transformation.</span>
<a name="l00089"></a>00089 <span class="comment"> */</span>
<a name="l00090"></a><a class="code" href="GenoUtility_8h.html#a4a4c4af98ccef1d6ff0482a9fec2c87a">00090</a> <span class="keyword">enum</span> <a class="code" href="GenoUtility_8h.html#a0176e2f61ce5ea7a3495a511cf46b2c1">Transformation</a> <a class="code" href="GenoUtility_8cpp.html#ab5c8b5877412aebfe6eb80db02bbc096" title="Convert a (text insensitive) text repesentation of a transformation into the enumration.">transformationForString</a>(<span class="keyword">const</span> std::string&amp; str) {
<a name="l00091"></a>00091   std::string lowerS = <a class="code" href="classUtil.html#ab73e26f5a353284899216123a494e3bf" title="Copy and downcase a 8bit string. ///.">Util::downcaseString</a>(str);
<a name="l00092"></a>00092 
<a name="l00093"></a>00093   <span class="keywordflow">if</span>(lowerS == <span class="stringliteral">&quot;mva&quot;</span>)
<a name="l00094"></a>00094     <span class="keywordflow">return</span> MvA;
<a name="l00095"></a>00095   <span class="keywordflow">else</span> <span class="keywordflow">if</span>(lowerS == <span class="stringliteral">&quot;rvt&quot;</span>)
<a name="l00096"></a>00096     <span class="keywordflow">return</span> RvT;
<a name="l00097"></a>00097   <span class="keywordflow">else</span> <span class="keywordflow">if</span>(lowerS == <span class="stringliteral">&quot;ssf&quot;</span> || lowerS == <span class="stringliteral">&quot;ces&quot;</span>)
<a name="l00098"></a>00098     <span class="keywordflow">return</span> CES;
<a name="l00099"></a>00099   <span class="keywordflow">else</span> <span class="keywordflow">if</span>(lowerS == <span class="stringliteral">&quot;assf&quot;</span> || lowerS == <span class="stringliteral">&quot;ccs&quot;</span>)
<a name="l00100"></a>00100     <span class="keywordflow">return</span> CCS;
<a name="l00101"></a>00101   <span class="keywordflow">else</span>
<a name="l00102"></a>00102     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;GenoUtility::transformationForString() - Don&#39;t recognize transformation: &quot;</span> + str);
<a name="l00103"></a>00103   <span class="keywordflow">return</span> MvA;
<a name="l00104"></a>00104 }
<a name="l00105"></a>00105 <span class="comment"></span>
<a name="l00106"></a>00106 <span class="comment">/**</span>
<a name="l00107"></a>00107 <span class="comment"> * Convert a transformation enum to string representation</span>
<a name="l00108"></a>00108 <span class="comment"> * @param t - tranformation enum value.</span>
<a name="l00109"></a>00109 <span class="comment"> * @return - text description of enumeration.</span>
<a name="l00110"></a>00110 <span class="comment"> */</span>
<a name="l00111"></a><a class="code" href="GenoUtility_8h.html#a119c3e97f150b0b9b91dea503b19c424">00111</a> std::string <a class="code" href="GenoUtility_8cpp.html#a119c3e97f150b0b9b91dea503b19c424" title="Convert a transformation enum to string representation.">stringForTransformation</a>(<span class="keyword">enum</span> <a class="code" href="GenoUtility_8h.html#a0176e2f61ce5ea7a3495a511cf46b2c1">Transformation</a> t) {
<a name="l00112"></a>00112   <span class="keywordflow">switch</span>(t) {
<a name="l00113"></a>00113   <span class="keywordflow">case</span> <a class="code" href="GenoUtility_8h.html#a0176e2f61ce5ea7a3495a511cf46b2c1a6db9488b39205827082e765aa0e87297" title="Minus Vs Average.">MvA</a> :
<a name="l00114"></a>00114     <span class="keywordflow">return</span> <span class="stringliteral">&quot;MvA&quot;</span>;
<a name="l00115"></a>00115     <span class="keywordflow">break</span>;
<a name="l00116"></a>00116   <span class="keywordflow">case</span> <a class="code" href="GenoUtility_8h.html#a0176e2f61ce5ea7a3495a511cf46b2c1a1bab0eb6be5fad1c5e78f95647686374" title="R vs Theta.">RvT</a> :
<a name="l00117"></a>00117     <span class="keywordflow">return</span> <span class="stringliteral">&quot;RvT&quot;</span>;
<a name="l00118"></a>00118     <span class="keywordflow">break</span>;
<a name="l00119"></a>00119   <span class="keywordflow">case</span> <a class="code" href="GenoUtility_8h.html#a0176e2f61ce5ea7a3495a511cf46b2c1a651cbbe4e01a316f0c0084eee693bb50" title="Contrast Extremes Stretch (nee south san francisco).">CES</a> :
<a name="l00120"></a>00120     <span class="keywordflow">return</span> <span class="stringliteral">&quot;CES&quot;</span>;
<a name="l00121"></a>00121     <span class="keywordflow">break</span>;
<a name="l00122"></a>00122   <span class="keywordflow">case</span> <a class="code" href="GenoUtility_8h.html#a0176e2f61ce5ea7a3495a511cf46b2c1a0af5537cdc9498e453464b9ac2e4f97f" title="Contrast Centers Stretch (nee alternative south san francisco).">CCS</a>:
<a name="l00123"></a>00123     <span class="keywordflow">return</span> <span class="stringliteral">&quot;CCS&quot;</span>;
<a name="l00124"></a>00124     <span class="keywordflow">break</span>;
<a name="l00125"></a>00125   <span class="keywordflow">default</span>:
<a name="l00126"></a>00126     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;GenoUtility::stringForTransformation() - Don&#39;t recognize type: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(t));
<a name="l00127"></a>00127   }
<a name="l00128"></a>00128   <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;GenoUtility::stringForTransformation() - Should never reach this point.&quot;</span>);
<a name="l00129"></a>00129   <span class="keywordflow">return</span> <span class="stringliteral">&quot;&quot;</span>;
<a name="l00130"></a>00130 }
<a name="l00131"></a>00131 <span class="comment"></span>
<a name="l00132"></a>00132 <span class="comment">/**</span>
<a name="l00133"></a>00133 <span class="comment"> * Return the calumn names for the values generated by a given transformation</span>
<a name="l00134"></a>00134 <span class="comment"> * @param t - tranformation enum value.</span>
<a name="l00135"></a>00135 <span class="comment"> * @return - a pair of column names</span>
<a name="l00136"></a>00136 <span class="comment"> */</span>
<a name="l00137"></a><a class="code" href="GenoUtility_8h.html#a8686c3ca8eabf2bb104f81c970d2ea2d">00137</a> std::pair&lt;std::string,std::string&gt; <a class="code" href="GenoUtility_8cpp.html#a8686c3ca8eabf2bb104f81c970d2ea2d" title="Return the calumn names for the values generated by a given transformation.">columnNamesForTransformation</a>(<span class="keyword">enum</span> <a class="code" href="GenoUtility_8h.html#a0176e2f61ce5ea7a3495a511cf46b2c1">Transformation</a> t) {
<a name="l00138"></a>00138   <span class="keywordflow">switch</span>(t) {
<a name="l00139"></a>00139   <span class="keywordflow">case</span> <a class="code" href="GenoUtility_8h.html#a0176e2f61ce5ea7a3495a511cf46b2c1a6db9488b39205827082e765aa0e87297" title="Minus Vs Average.">MvA</a> :
<a name="l00140"></a>00140     <span class="keywordflow">return</span> std::pair&lt;std::string,std::string&gt;(<span class="stringliteral">&quot;Log Ratio&quot;</span>,<span class="stringliteral">&quot;Strength&quot;</span>);
<a name="l00141"></a>00141     <span class="keywordflow">break</span>;
<a name="l00142"></a>00142   <span class="keywordflow">case</span> <a class="code" href="GenoUtility_8h.html#a0176e2f61ce5ea7a3495a511cf46b2c1a1bab0eb6be5fad1c5e78f95647686374" title="R vs Theta.">RvT</a> :
<a name="l00143"></a>00143     <span class="keywordflow">return</span> std::pair&lt;std::string,std::string&gt;(<span class="stringliteral">&quot;R&quot;</span>,<span class="stringliteral">&quot;Theta&quot;</span>);
<a name="l00144"></a>00144     <span class="keywordflow">break</span>;
<a name="l00145"></a>00145   <span class="keywordflow">case</span> <a class="code" href="GenoUtility_8h.html#a0176e2f61ce5ea7a3495a511cf46b2c1a651cbbe4e01a316f0c0084eee693bb50" title="Contrast Extremes Stretch (nee south san francisco).">CES</a> :
<a name="l00146"></a>00146     <span class="keywordflow">return</span> std::pair&lt;std::string,std::string&gt;(<span class="stringliteral">&quot;Contrast&quot;</span>,<span class="stringliteral">&quot;Strength&quot;</span>);
<a name="l00147"></a>00147     <span class="keywordflow">break</span>;
<a name="l00148"></a>00148   <span class="keywordflow">case</span> <a class="code" href="GenoUtility_8h.html#a0176e2f61ce5ea7a3495a511cf46b2c1a0af5537cdc9498e453464b9ac2e4f97f" title="Contrast Centers Stretch (nee alternative south san francisco).">CCS</a>:
<a name="l00149"></a>00149     <span class="keywordflow">return</span> std::pair&lt;std::string,std::string&gt;(<span class="stringliteral">&quot;Contrast&quot;</span>,<span class="stringliteral">&quot;Strength&quot;</span>);
<a name="l00150"></a>00150     <span class="keywordflow">break</span>;
<a name="l00151"></a>00151   <span class="keywordflow">default</span>:
<a name="l00152"></a>00152     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;GenoUtility::columnNamesForTransformation() - Don&#39;t recognize type: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(t));
<a name="l00153"></a>00153   }
<a name="l00154"></a>00154   <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;GenoUtility::columnNamesForTransformation() - Should never reach this point.&quot;</span>);
<a name="l00155"></a>00155   <span class="keywordflow">return</span> std::pair&lt;std::string,std::string&gt;(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;&quot;</span>);
<a name="l00156"></a>00156 }
<a name="l00157"></a>00157 <span class="comment"></span>
<a name="l00158"></a>00158 <span class="comment">/** transform the data vector */</span>
<a name="l00159"></a><a class="code" href="GenoUtility_8h.html#a1a2531d6033d567af0446274e94f74a4">00159</a> <span class="keywordtype">void</span> <a class="code" href="GenoUtility_8cpp.html#a1a2531d6033d567af0446274e94f74a4" title="transform the data vector">GenoUtility_transformData</a>(std::vector&lt;double&gt; &amp;aVals,
<a name="l00160"></a>00160                                std::vector&lt;double&gt; &amp;bVals,
<a name="l00161"></a>00161                                <a class="code" href="GenoUtility_8h.html#a0176e2f61ce5ea7a3495a511cf46b2c1">Transformation</a> m_Transform,
<a name="l00162"></a>00162                                <span class="keywordtype">double</span> m_K)
<a name="l00163"></a>00163 {
<a name="l00164"></a>00164     assert(aVals.size() == bVals.size());
<a name="l00165"></a>00165     <span class="keywordtype">double</span> x = 0, y = 0;
<a name="l00166"></a>00166     <span class="keywordflow">for</span>(<span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> i = 0; i &lt; aVals.size(); i++) {
<a name="l00167"></a>00167       <a class="code" href="GenoUtility_8cpp.html#aacfe80c3e8bd468402502f56a7704b15" title="This is the engine that takes data and transforms it into clustering spaces.">valueForTransformation</a>(aVals[i], bVals[i], m_Transform, m_K, x, y);
<a name="l00168"></a>00168       aVals[i] = x;
<a name="l00169"></a>00169       bVals[i] = y;
<a name="l00170"></a>00170     }
<a name="l00171"></a>00171   }
<a name="l00172"></a>00172 <span class="comment"></span>
<a name="l00173"></a>00173 <span class="comment">/**</span>
<a name="l00174"></a>00174 <span class="comment"> * Contrast Centers Stretch. In 2007 this is the current favorite for</span>
<a name="l00175"></a>00175 <span class="comment"> * algorithms, scales values between -1 and 1.</span>
<a name="l00176"></a>00176 <span class="comment"> * \f$ X = asinh(K * (A-B)/(A+B)) / asinh(K) \f$ &lt;br&gt;</span>
<a name="l00177"></a>00177 <span class="comment"> * \f$ Y = log_2(A + B) \f$ &lt;br&gt;</span>
<a name="l00178"></a>00178 <span class="comment"> *</span>
<a name="l00179"></a>00179 <span class="comment"> * @param K -</span>
<a name="l00180"></a>00180 <span class="comment"> * @param a - Value (usually summarized intensity) from A allele.</span>
<a name="l00181"></a>00181 <span class="comment"> * @param b - Value (usually summarized intensity) from B allele.</span>
<a name="l00182"></a>00182 <span class="comment"> * @param x -</span>
<a name="l00183"></a>00183 <span class="comment"> * @param y -</span>
<a name="l00184"></a>00184 <span class="comment"> */</span>
<a name="l00185"></a><a class="code" href="GenoUtility_8h.html#a1e7d2233042fe7b5a9fc83b4f79b545c">00185</a> <span class="keywordtype">void</span> <a class="code" href="GenoUtility_8cpp.html#a1e7d2233042fe7b5a9fc83b4f79b545c" title="Contrast Centers Stretch.">ContrastCentersStretch</a>(<span class="keywordtype">double</span> K, <span class="keywordtype">double</span> a, <span class="keywordtype">double</span> b, <span class="keywordtype">double</span> &amp;x, <span class="keywordtype">double</span> &amp;y) {
<a name="l00186"></a>00186   <span class="keywordtype">double</span> denom = 0;
<a name="l00187"></a>00187   <span class="keywordtype">double</span> r = 0.0;
<a name="l00188"></a>00188   denom = b + a;
<a name="l00189"></a>00189   <span class="keywordflow">if</span>(denom == 0) {
<a name="l00190"></a>00190     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;GenoUtility::ContrastCentersStretch() - Can&#39;t have zero sum signal.&quot;</span>);
<a name="l00191"></a>00191   }
<a name="l00192"></a>00192   r = K * (a - b)/denom;
<a name="l00193"></a>00193   r = log(r + sqrt(r*r+1));
<a name="l00194"></a>00194   x = r / log(K + sqrt(K*K+1));
<a name="l00195"></a>00195   y = <a class="code" href="Util_8h.html#a18af743c2cec4baeee9ffb27999ddaad" title="Log base two.">log2</a>(denom);
<a name="l00196"></a>00196 }
<a name="l00197"></a>00197 <span class="comment"></span>
<a name="l00198"></a>00198 <span class="comment">/**</span>
<a name="l00199"></a>00199 <span class="comment"> * Contrast Extremes Stretch.</span>
<a name="l00200"></a>00200 <span class="comment"> * \f$ X = sinh(K * (A-B)/(A+B)) / sinh(K) \f$ &lt;br&gt;</span>
<a name="l00201"></a>00201 <span class="comment"> * \f$ Y = log_2(A + B) \f$ &lt;br&gt;</span>
<a name="l00202"></a>00202 <span class="comment"> *</span>
<a name="l00203"></a>00203 <span class="comment"> * @param K -</span>
<a name="l00204"></a>00204 <span class="comment"> * @param a - Value (usually summarized intensity) from A allele.</span>
<a name="l00205"></a>00205 <span class="comment"> * @param b - Value (usually summarized intensity) from B allele.</span>
<a name="l00206"></a>00206 <span class="comment"> * @param x -</span>
<a name="l00207"></a>00207 <span class="comment"> * @param y -</span>
<a name="l00208"></a>00208 <span class="comment"> */</span>
<a name="l00209"></a><a class="code" href="GenoUtility_8h.html#ac842841ae05672e63157ac80b8a016f6">00209</a> <span class="keywordtype">void</span> <a class="code" href="GenoUtility_8cpp.html#ac842841ae05672e63157ac80b8a016f6" title="Contrast Extremes Stretch.">ContrastExtremesStretch</a>(<span class="keywordtype">double</span> K, <span class="keywordtype">double</span> a, <span class="keywordtype">double</span> b, <span class="keywordtype">double</span> &amp;x, <span class="keywordtype">double</span> &amp;y) {
<a name="l00210"></a>00210   <span class="keywordtype">double</span> denom = 0;
<a name="l00211"></a>00211   denom = b + a;
<a name="l00212"></a>00212   <span class="keywordflow">if</span>(denom == 0) {
<a name="l00213"></a>00213     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;GenoUtility::ContrastExtremesStretch() - Can&#39;t have zero sum signal.&quot;</span>);
<a name="l00214"></a>00214   }
<a name="l00215"></a>00215   x = sinh( K * (a - b)/denom) / sinh(K);
<a name="l00216"></a>00216   y = <a class="code" href="Util_8h.html#a18af743c2cec4baeee9ffb27999ddaad" title="Log base two.">log2</a>(denom);
<a name="l00217"></a>00217 }
<a name="l00218"></a>00218 <span class="comment"></span>
<a name="l00219"></a>00219 <span class="comment">/**</span>
<a name="l00220"></a>00220 <span class="comment"> * Calculate the median PM values for the A and B alleles given data</span>
<a name="l00221"></a>00221 <span class="comment"> * in cel file.</span>
<a name="l00222"></a>00222 <span class="comment"> *</span>
<a name="l00223"></a>00223 <span class="comment"> * @param cel - Cel file of data.</span>
<a name="l00224"></a>00224 <span class="comment"> * @param ps - ProbeSet for determining A and B alleles.</span>
<a name="l00225"></a>00225 <span class="comment"> * @param Amedian - Filled in with median PM A value.</span>
<a name="l00226"></a>00226 <span class="comment"> * @param Bmedian - Filled in with median PM B value.</span>
<a name="l00227"></a>00227 <span class="comment"> *</span>
<a name="l00228"></a>00228 <span class="comment"> * @return</span>
<a name="l00229"></a>00229 <span class="comment"> */</span>
<a name="l00230"></a><a class="code" href="GenoUtility_8h.html#ac3236f3bb0d22355aaa716877d56f881">00230</a> <span class="keywordtype">bool</span> <a class="code" href="GenoUtility_8cpp.html#ac3236f3bb0d22355aaa716877d56f881" title="Calculate the median PM values for the A and B alleles given data in cel file.">alleleMedians</a>(<a class="code" href="classaffymetrix__fusion__io_1_1FusionCELData.html">affymetrix_fusion_io::FusionCELData</a> *cel, <span class="keyword">const</span> <a class="code" href="classProbeSet.html" title="Collection of atoms that measure target that should be absent or present at the same time...">ProbeSet</a> *ps,
<a name="l00231"></a>00231                                       <span class="keywordtype">float</span> &amp;Amedian, <span class="keywordtype">float</span> &amp;Bmedian) {
<a name="l00232"></a>00232   vector&lt;float&gt; AValues, BValues;
<a name="l00233"></a>00233   <span class="comment">/* Genotype probe sets can have either 2 or 4 groups, for each</span>
<a name="l00234"></a>00234 <span class="comment">     type the even (0,2) are A alleles and odd (1,3) are B allele.s */</span>
<a name="l00235"></a>00235   <span class="keywordflow">if</span>(ps-&gt;<a class="code" href="classProbeSet.html#a6d8ed948634c42033bd0c35d2640b274" title="How many groups (or blocks) are there? Used extensively in genotyping chips.">numGroups</a> != 2 &amp;&amp; ps-&gt;<a class="code" href="classProbeSet.html#a6d8ed948634c42033bd0c35d2640b274" title="How many groups (or blocks) are there? Used extensively in genotyping chips.">numGroups</a> != 4) {
<a name="l00236"></a>00236     <span class="keywordflow">return</span> <span class="keyword">false</span>;
<a name="l00237"></a>00237   }
<a name="l00238"></a>00238 
<a name="l00239"></a>00239   <span class="keywordflow">if</span> (ps-&gt;<a class="code" href="classProbeSet.html#aa3e4170184169cc3bee6d958d74940ba" title="What type of probeset is this?">psType</a> != ProbeSet::GenoType)
<a name="l00240"></a>00240       <span class="keywordflow">return</span> <span class="keyword">false</span>;
<a name="l00241"></a>00241 
<a name="l00242"></a>00242   <span class="keywordtype">int</span> groupOffset = 0; <span class="comment">// atoms are stored in blocks for each allele, keep track of which block we&#39;re in.</span>
<a name="l00243"></a>00243   <span class="keywordflow">for</span>(<span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> groupIx = 0; groupIx &lt; ps-&gt;<a class="code" href="classProbeSet.html#a6d8ed948634c42033bd0c35d2640b274" title="How many groups (or blocks) are there? Used extensively in genotyping chips.">numGroups</a>; groupIx++) {
<a name="l00244"></a>00244         <span class="comment">/* Probe belongs to the A allele, put atoms in that probeset. */</span>
<a name="l00245"></a>00245         <span class="keywordflow">if</span>(groupIx == ProbeSet::Aforward || groupIx == ProbeSet::Areverse) {
<a name="l00246"></a>00246           <span class="keywordflow">for</span>(<span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> atomIx = groupOffset; atomIx &lt; ps-&gt;<a class="code" href="classProbeSet.html#a23b9df97be8aacc01ad21642b8c966b9" title="Number of atoms in each group, used to get the offsets into the atoms vector.">atomsPerGroup</a>[groupIx] + groupOffset; atomIx++) {
<a name="l00247"></a>00247         <a class="code" href="classAtom.html" title="Represents a collection of probes that make up a unit for an intensity measurement of a particular se...">Atom</a> *A = ps-&gt;<a class="code" href="classProbeSet.html#ab92eb06095441bf193a6fce3f758540b" title="Atoms that make up this probe set.">atoms</a>[atomIx];
<a name="l00248"></a>00248         <span class="keywordflow">for</span>(<span class="keywordtype">int</span> probeIx = 0; probeIx &lt; A-&gt;<a class="code" href="classAtom.html#a5953a4944a1e24b3f51a20ce92d8def2" title="Vector of probes that make up this atom. Order is important.">probes</a>.size(); probeIx++) {
<a name="l00249"></a>00249             <a class="code" href="classProbe.html" title="Represents an individual feature on a chip.">Probe</a> *p = A-&gt;<a class="code" href="classAtom.html#a5953a4944a1e24b3f51a20ce92d8def2" title="Vector of probes that make up this atom. Order is important.">probes</a>[probeIx];
<a name="l00250"></a>00250             <span class="keywordflow">if</span>(Probe::isPm(*p)) {
<a name="l00251"></a>00251                 <span class="keywordtype">float</span> intensity = cel-&gt;<a class="code" href="classaffymetrix__fusion__io_1_1FusionCELData.html#a76d2432b4075dbb9f00a1cf306a3a614">GetIntensity</a>(p-&gt;<a class="code" href="classProbe.html#a839c453922e807335dc6f4b385ed456c" title="Id of probe, in practice usually realated to position on array.">id</a>);
<a name="l00252"></a>00252                 AValues.push_back(intensity);
<a name="l00253"></a>00253             }
<a name="l00254"></a>00254         }
<a name="l00255"></a>00255           }
<a name="l00256"></a>00256         }
<a name="l00257"></a>00257         <span class="comment">/* Probe belongs to the B allele, put atoms in that probeset. */</span>
<a name="l00258"></a>00258         <span class="keywordflow">else</span> <span class="keywordflow">if</span>(groupIx == ProbeSet::Bforward || groupIx == ProbeSet::Breverse) {
<a name="l00259"></a>00259           <span class="keywordflow">for</span>(<span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> atomIx = groupOffset; atomIx &lt; ps-&gt;<a class="code" href="classProbeSet.html#a23b9df97be8aacc01ad21642b8c966b9" title="Number of atoms in each group, used to get the offsets into the atoms vector.">atomsPerGroup</a>[groupIx] + groupOffset; atomIx++) {
<a name="l00260"></a>00260         <a class="code" href="classAtom.html" title="Represents a collection of probes that make up a unit for an intensity measurement of a particular se...">Atom</a> *A = ps-&gt;<a class="code" href="classProbeSet.html#ab92eb06095441bf193a6fce3f758540b" title="Atoms that make up this probe set.">atoms</a>[atomIx];
<a name="l00261"></a>00261         <span class="keywordflow">for</span>(<span class="keywordtype">int</span> probeIx = 0; probeIx &lt; A-&gt;<a class="code" href="classAtom.html#a5953a4944a1e24b3f51a20ce92d8def2" title="Vector of probes that make up this atom. Order is important.">probes</a>.size(); probeIx++) {
<a name="l00262"></a>00262             <a class="code" href="classProbe.html" title="Represents an individual feature on a chip.">Probe</a> *p = A-&gt;<a class="code" href="classAtom.html#a5953a4944a1e24b3f51a20ce92d8def2" title="Vector of probes that make up this atom. Order is important.">probes</a>[probeIx];
<a name="l00263"></a>00263             <span class="keywordflow">if</span>(Probe::isPm(*p)) {
<a name="l00264"></a>00264                 <span class="keywordtype">float</span> intensity = cel-&gt;<a class="code" href="classaffymetrix__fusion__io_1_1FusionCELData.html#a76d2432b4075dbb9f00a1cf306a3a614">GetIntensity</a>(p-&gt;<a class="code" href="classProbe.html#a839c453922e807335dc6f4b385ed456c" title="Id of probe, in practice usually realated to position on array.">id</a>);
<a name="l00265"></a>00265                 BValues.push_back(intensity);
<a name="l00266"></a>00266             }
<a name="l00267"></a>00267         }
<a name="l00268"></a>00268           }
<a name="l00269"></a>00269         }
<a name="l00270"></a>00270         <span class="comment">/* Uh-oh, if it doesn&#39;t belong to A and it doesn&#39;t belong to B we&#39;re going to complain. */</span>
<a name="l00271"></a>00271         <span class="keywordflow">else</span> {
<a name="l00272"></a>00272           <span class="keywordflow">return</span> <span class="keyword">false</span>;
<a name="l00273"></a>00273         }
<a name="l00274"></a>00274         groupOffset += ps-&gt;<a class="code" href="classProbeSet.html#a23b9df97be8aacc01ad21642b8c966b9" title="Number of atoms in each group, used to get the offsets into the atoms vector.">atomsPerGroup</a>[groupIx];
<a name="l00275"></a>00275   }
<a name="l00276"></a>00276 
<a name="l00277"></a>00277   <span class="keywordflow">if</span>(AValues.empty() || BValues.empty())
<a name="l00278"></a>00278       <span class="keywordflow">return</span> <span class="keyword">false</span>;
<a name="l00279"></a>00279 
<a name="l00280"></a>00280   Amedian = median_in_place(AValues.begin(), AValues.end());
<a name="l00281"></a>00281   Bmedian = median_in_place(BValues.begin(), BValues.end());
<a name="l00282"></a>00282   <span class="keywordflow">return</span> <span class="keyword">true</span>;
<a name="l00283"></a>00283 }
<a name="l00284"></a>00284 
<a name="l00285"></a>00285 <span class="comment">/* mimic Eric&#39;s code */</span>
<a name="l00286"></a>00286 <span class="keywordtype">bool</span> probesetMean(<a class="code" href="classaffymetrix__fusion__io_1_1FusionCELData.html">affymetrix_fusion_io::FusionCELData</a> *cel, <span class="keyword">const</span> <a class="code" href="classProbeSet.html" title="Collection of atoms that measure target that should be absent or present at the same time...">ProbeSet</a> *ps, 
<a name="l00287"></a>00287                   <span class="keywordtype">float</span> &amp;mean) {
<a name="l00288"></a>00288   vector&lt;float&gt; Values;
<a name="l00289"></a>00289   <span class="comment">/* Genotype probe sets can have either 2 or 4 groups, for each</span>
<a name="l00290"></a>00290 <span class="comment">     type the even (0,2) are A alleles and odd (1,3) are B allele.s */</span>
<a name="l00291"></a>00291   <span class="comment">//if(ps-&gt;numGroups != 2 &amp;&amp; ps-&gt;numGroups != 4) {</span>
<a name="l00292"></a>00292   <span class="comment">//  return false;</span>
<a name="l00293"></a>00293   <span class="comment">//}</span>
<a name="l00294"></a>00294   
<a name="l00295"></a>00295   <span class="comment">//Verbose::out(4, &quot;there are &quot;+ ToStr(ps-&gt;numGroups) + &quot; groups&quot;);</span>
<a name="l00296"></a>00296   <a class="code" href="classErr.html#ad1a633f602fe60689cdc12d9ca1842e9" title="errAbort based version of assert with an error message.">Err::check</a>(ps-&gt;<a class="code" href="classProbeSet.html#a6d8ed948634c42033bd0c35d2640b274" title="How many groups (or blocks) are there? Used extensively in genotyping chips.">numGroups</a> == 1, 
<a name="l00297"></a>00297              <span class="stringliteral">&quot;alleleMinMax() - Thought for Falcon, there should just be 1 group.&quot;</span>);
<a name="l00298"></a>00298   <span class="keywordtype">int</span> atomsPer = ps-&gt;<a class="code" href="classProbeSet.html#ab92eb06095441bf193a6fce3f758540b" title="Atoms that make up this probe set.">atoms</a>.size();
<a name="l00299"></a>00299   <span class="comment">//Verbose::out(4, &quot;atoms per group is &quot;+ ToStr(atomsPer));</span>
<a name="l00300"></a>00300   <span class="comment">/* Each block will have atomsPer atoms in there. */</span>
<a name="l00301"></a>00301   <span class="keywordflow">for</span>(<span class="keywordtype">int</span> atomIx = 0; atomIx &lt; atomsPer; atomIx++) {
<a name="l00302"></a>00302     <a class="code" href="classAtom.html" title="Represents a collection of probes that make up a unit for an intensity measurement of a particular se...">Atom</a> *A = ps-&gt;<a class="code" href="classProbeSet.html#ab92eb06095441bf193a6fce3f758540b" title="Atoms that make up this probe set.">atoms</a>[atomIx];
<a name="l00303"></a>00303     <span class="keywordflow">for</span>(<span class="keywordtype">int</span> probeIx = 0; probeIx &lt; A-&gt;<a class="code" href="classAtom.html#a5953a4944a1e24b3f51a20ce92d8def2" title="Vector of probes that make up this atom. Order is important.">probes</a>.size(); probeIx++) {
<a name="l00304"></a>00304       <a class="code" href="classProbe.html" title="Represents an individual feature on a chip.">Probe</a> *p = A-&gt;<a class="code" href="classAtom.html#a5953a4944a1e24b3f51a20ce92d8def2" title="Vector of probes that make up this atom. Order is important.">probes</a>[probeIx];
<a name="l00305"></a>00305       <span class="keywordflow">if</span>(Probe::isPm(*p)) {
<a name="l00306"></a>00306         <span class="keywordtype">float</span> intensity = cel-&gt;<a class="code" href="classaffymetrix__fusion__io_1_1FusionCELData.html#a76d2432b4075dbb9f00a1cf306a3a614">GetIntensity</a>(p-&gt;<a class="code" href="classProbe.html#a839c453922e807335dc6f4b385ed456c" title="Id of probe, in practice usually realated to position on array.">id</a>);
<a name="l00307"></a>00307         Values.push_back(intensity);
<a name="l00308"></a>00308       }
<a name="l00309"></a>00309     }
<a name="l00310"></a>00310     
<a name="l00311"></a>00311   }
<a name="l00312"></a>00312   
<a name="l00313"></a>00313   <span class="keywordflow">if</span>(Values.empty()) 
<a name="l00314"></a>00314     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;alleleMinMax() - Empty value vectors in probeset: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(ps-&gt;<a class="code" href="classProbeSet.html#af10b2dac78a597ec736691302687c3f3" title="Name of probe set.">name</a>));
<a name="l00315"></a>00315    mean = average(Values.begin(), Values.end());
<a name="l00316"></a>00316   
<a name="l00317"></a>00317   <span class="keywordflow">return</span> <span class="keyword">true</span>;
<a name="l00318"></a>00318 }
<a name="l00319"></a>00319 
<a name="l00320"></a>00320 
<a name="l00321"></a>00321 <span class="keywordtype">bool</span> probesetStd(<a class="code" href="classaffymetrix__fusion__io_1_1FusionCELData.html">affymetrix_fusion_io::FusionCELData</a> *cel, <span class="keyword">const</span> <a class="code" href="classProbeSet.html" title="Collection of atoms that measure target that should be absent or present at the same time...">ProbeSet</a> *ps, 
<a name="l00322"></a>00322                   <span class="keywordtype">float</span> &amp;std) {
<a name="l00323"></a>00323   vector&lt;float&gt; Values;
<a name="l00324"></a>00324   <span class="comment">/* Genotype probe sets can have either 2 or 4 groups, for each</span>
<a name="l00325"></a>00325 <span class="comment">     type the even (0,2) are A alleles and odd (1,3) are B allele.s */</span>
<a name="l00326"></a>00326   <span class="comment">//if(ps-&gt;numGroups != 2 &amp;&amp; ps-&gt;numGroups != 4) {</span>
<a name="l00327"></a>00327   <span class="comment">//  return false;</span>
<a name="l00328"></a>00328   <span class="comment">//}</span>
<a name="l00329"></a>00329   
<a name="l00330"></a>00330   <span class="comment">//Verbose::out(4, &quot;there are &quot;+ ToStr(ps-&gt;numGroups) + &quot; groups&quot;);</span>
<a name="l00331"></a>00331   <a class="code" href="classErr.html#ad1a633f602fe60689cdc12d9ca1842e9" title="errAbort based version of assert with an error message.">Err::check</a>(ps-&gt;<a class="code" href="classProbeSet.html#a6d8ed948634c42033bd0c35d2640b274" title="How many groups (or blocks) are there? Used extensively in genotyping chips.">numGroups</a> == 1, 
<a name="l00332"></a>00332              <span class="stringliteral">&quot;alleleMinMax() - Thought for Falcon, there should just be 1 group.&quot;</span>);
<a name="l00333"></a>00333   <span class="keywordtype">int</span> atomsPer = ps-&gt;<a class="code" href="classProbeSet.html#ab92eb06095441bf193a6fce3f758540b" title="Atoms that make up this probe set.">atoms</a>.size();
<a name="l00334"></a>00334   <span class="keywordflow">for</span>(<span class="keywordtype">int</span> atomIx = 0; atomIx &lt; atomsPer; atomIx++) {
<a name="l00335"></a>00335     <a class="code" href="classAtom.html" title="Represents a collection of probes that make up a unit for an intensity measurement of a particular se...">Atom</a> *A = ps-&gt;<a class="code" href="classProbeSet.html#ab92eb06095441bf193a6fce3f758540b" title="Atoms that make up this probe set.">atoms</a>[atomIx];
<a name="l00336"></a>00336     <span class="keywordflow">for</span>(<span class="keywordtype">int</span> probeIx = 0; probeIx &lt; A-&gt;<a class="code" href="classAtom.html#a5953a4944a1e24b3f51a20ce92d8def2" title="Vector of probes that make up this atom. Order is important.">probes</a>.size(); probeIx++) {
<a name="l00337"></a>00337       <a class="code" href="classProbe.html" title="Represents an individual feature on a chip.">Probe</a> *p = A-&gt;<a class="code" href="classAtom.html#a5953a4944a1e24b3f51a20ce92d8def2" title="Vector of probes that make up this atom. Order is important.">probes</a>[probeIx];
<a name="l00338"></a>00338       <span class="keywordflow">if</span>(Probe::isPm(*p)) {
<a name="l00339"></a>00339         <span class="keywordtype">float</span> intensity = cel-&gt;<a class="code" href="classaffymetrix__fusion__io_1_1FusionCELData.html#a76d2432b4075dbb9f00a1cf306a3a614">GetIntensity</a>(p-&gt;<a class="code" href="classProbe.html#a839c453922e807335dc6f4b385ed456c" title="Id of probe, in practice usually realated to position on array.">id</a>);
<a name="l00340"></a>00340         Values.push_back(intensity);
<a name="l00341"></a>00341       }
<a name="l00342"></a>00342     }
<a name="l00343"></a>00343     
<a name="l00344"></a>00344   }
<a name="l00345"></a>00345   
<a name="l00346"></a>00346   <span class="keywordflow">if</span>(Values.empty()) 
<a name="l00347"></a>00347     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;alleleMinMax() - Empty value vectors in probeset: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(ps-&gt;<a class="code" href="classProbeSet.html#af10b2dac78a597ec736691302687c3f3" title="Name of probe set.">name</a>));
<a name="l00348"></a>00348    std = unbiased_stddev(Values.begin(), Values.end());
<a name="l00349"></a>00349   <span class="keywordflow">return</span> <span class="keyword">true</span>;
<a name="l00350"></a>00350 }
<a name="l00351"></a>00351 
<a name="l00352"></a>00352 
<a name="l00353"></a>00353 <span class="keywordtype">bool</span> probesetMeanStd(<a class="code" href="classaffymetrix__fusion__io_1_1FusionCELData.html">affymetrix_fusion_io::FusionCELData</a> *cel, <span class="keyword">const</span> <a class="code" href="classProbeSet.html" title="Collection of atoms that measure target that should be absent or present at the same time...">ProbeSet</a> *ps, 
<a name="l00354"></a>00354                      <span class="keywordtype">float</span> &amp;mean, <span class="keywordtype">float</span> &amp;std) {
<a name="l00355"></a>00355   vector&lt;float&gt; Values;
<a name="l00356"></a>00356   <span class="comment">/* Genotype probe sets can have either 2 or 4 groups, for each</span>
<a name="l00357"></a>00357 <span class="comment">     type the even (0,2) are A alleles and odd (1,3) are B allele.s */</span>
<a name="l00358"></a>00358   <span class="comment">//if(ps-&gt;numGroups != 2 &amp;&amp; ps-&gt;numGroups != 4) {</span>
<a name="l00359"></a>00359   <span class="comment">//  return false;</span>
<a name="l00360"></a>00360   <span class="comment">//}</span>
<a name="l00361"></a>00361   
<a name="l00362"></a>00362   <span class="comment">//Verbose::out(4, &quot;there are &quot;+ ToStr(ps-&gt;numGroups) + &quot; groups&quot;);</span>
<a name="l00363"></a>00363   <a class="code" href="classErr.html#ad1a633f602fe60689cdc12d9ca1842e9" title="errAbort based version of assert with an error message.">Err::check</a>(ps-&gt;<a class="code" href="classProbeSet.html#a6d8ed948634c42033bd0c35d2640b274" title="How many groups (or blocks) are there? Used extensively in genotyping chips.">numGroups</a> == 1, 
<a name="l00364"></a>00364              <span class="stringliteral">&quot;alleleMinMax() - Thought for Falcon, there should just be 1 group.&quot;</span>);
<a name="l00365"></a>00365   <span class="keywordtype">int</span> atomsPer = ps-&gt;<a class="code" href="classProbeSet.html#ab92eb06095441bf193a6fce3f758540b" title="Atoms that make up this probe set.">atoms</a>.size();
<a name="l00366"></a>00366   <span class="keywordflow">for</span>(<span class="keywordtype">int</span> atomIx = 0; atomIx &lt; atomsPer; atomIx++) {
<a name="l00367"></a>00367     <a class="code" href="classAtom.html" title="Represents a collection of probes that make up a unit for an intensity measurement of a particular se...">Atom</a> *A = ps-&gt;<a class="code" href="classProbeSet.html#ab92eb06095441bf193a6fce3f758540b" title="Atoms that make up this probe set.">atoms</a>[atomIx];
<a name="l00368"></a>00368     <span class="keywordflow">for</span>(<span class="keywordtype">int</span> probeIx = 0; probeIx &lt; A-&gt;<a class="code" href="classAtom.html#a5953a4944a1e24b3f51a20ce92d8def2" title="Vector of probes that make up this atom. Order is important.">probes</a>.size(); probeIx++) {
<a name="l00369"></a>00369       <a class="code" href="classProbe.html" title="Represents an individual feature on a chip.">Probe</a> *p = A-&gt;<a class="code" href="classAtom.html#a5953a4944a1e24b3f51a20ce92d8def2" title="Vector of probes that make up this atom. Order is important.">probes</a>[probeIx];
<a name="l00370"></a>00370       <span class="keywordflow">if</span>(Probe::isPm(*p)) {
<a name="l00371"></a>00371         <span class="keywordtype">float</span> intensity = cel-&gt;<a class="code" href="classaffymetrix__fusion__io_1_1FusionCELData.html#a76d2432b4075dbb9f00a1cf306a3a614">GetIntensity</a>(p-&gt;<a class="code" href="classProbe.html#a839c453922e807335dc6f4b385ed456c" title="Id of probe, in practice usually realated to position on array.">id</a>);
<a name="l00372"></a>00372         Values.push_back(intensity);
<a name="l00373"></a>00373       }
<a name="l00374"></a>00374     }
<a name="l00375"></a>00375     
<a name="l00376"></a>00376   }
<a name="l00377"></a>00377   
<a name="l00378"></a>00378   <span class="keywordflow">if</span>(Values.empty()) 
<a name="l00379"></a>00379     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;alleleMinMax() - Empty value vectors in probeset: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(ps-&gt;<a class="code" href="classProbeSet.html#af10b2dac78a597ec736691302687c3f3" title="Name of probe set.">name</a>));
<a name="l00380"></a>00380   std = unbiased_stddev(Values.begin(), Values.end());
<a name="l00381"></a>00381   mean = average(Values.begin(), Values.end());
<a name="l00382"></a>00382   <span class="keywordflow">return</span> <span class="keyword">true</span>;
<a name="l00383"></a>00383 }
<a name="l00384"></a>00384 
<a name="l00385"></a>00385 
<a name="l00386"></a>00386 
<a name="l00387"></a>00387 <span class="comment"></span>
<a name="l00388"></a>00388 <span class="comment">/**</span>
<a name="l00389"></a>00389 <span class="comment"> * Calculate the natural log ratio for two probeset signal</span>
<a name="l00390"></a>00390 <span class="comment"> *</span>
<a name="l00391"></a>00391 <span class="comment"> *@para signalA -- e.x. signal channel </span>
<a name="l00392"></a>00392 <span class="comment"> *@para signalB -- e.x. background channel</span>
<a name="l00393"></a>00393 <span class="comment"> *</span>
<a name="l00394"></a>00394 <span class="comment"> */</span>
<a name="l00395"></a><a class="code" href="GenoUtility_8h.html#a631afdfd53685cf4cc3f0cf46b448c5c">00395</a> <span class="keywordtype">double</span> <a class="code" href="GenoUtility_8cpp.html#a631afdfd53685cf4cc3f0cf46b448c5c" title="Calculate the natural log ratio for two probeset signal.">logRatio</a>(<span class="keywordtype">double</span> sigA, <span class="keywordtype">double</span> sigB){
<a name="l00396"></a>00396   sigB = sigB &lt;=0? 1 : sigB;
<a name="l00397"></a>00397   <span class="keywordflow">return</span>(log(sigA/sigB));
<a name="l00398"></a>00398 }
<a name="l00399"></a>00399 
<a name="l00400"></a>00400 <span class="comment"></span>
<a name="l00401"></a>00401 <span class="comment">/** </span>
<a name="l00402"></a>00402 <span class="comment"> * Open and read a target distribution from a text file. Must have a column</span>
<a name="l00403"></a>00403 <span class="comment"> * called &#39;intensities&#39; with distribution quantiles sorted from highest to </span>
<a name="l00404"></a>00404 <span class="comment"> * lowest.</span>
<a name="l00405"></a>00405 <span class="comment"> * </span>
<a name="l00406"></a>00406 <span class="comment"> * @param fileName - text file containing column of quantiles.</span>
<a name="l00407"></a>00407 <span class="comment"> * @para sketch - vector that stores the sketch.</span>
<a name="l00408"></a>00408 <span class="comment"> */</span>
<a name="l00409"></a>00409 
<a name="l00410"></a><a class="code" href="GenoUtility_8h.html#a208943f147d29a97c3aab8015aa707dc">00410</a> <span class="keywordtype">void</span>  <a class="code" href="GenoUtility_8cpp.html#a208943f147d29a97c3aab8015aa707dc" title="Open and read a target distribution from a text file.">loadTargetSketchFromFile</a>(<span class="keyword">const</span> <span class="keywordtype">char</span> *fileName, std::vector&lt;float&gt; &amp;sketch) {
<a name="l00411"></a>00411   <a class="code" href="classaffx_1_1TsvFile.html" title="A class for reading and writing Tab Seperated Value (TSV) files. /// See the TsvFile format document ...">TsvFile</a> tsv;
<a name="l00412"></a>00412   <span class="keywordtype">double</span> intensity;
<a name="l00413"></a>00413   <span class="comment">// file ok</span>
<a name="l00414"></a>00414   <span class="keywordtype">int</span> rc = tsv.<a class="code" href="classaffx_1_1TsvFile.html#a763ce0bbf19b4134d0445aa58e0f5011" title="Opens a file -- attempts to guess some defaults.">open</a>(fileName);
<a name="l00415"></a>00415   assert(rc==<a class="code" href="namespaceaffx.html#ad00847b6f358e7269f493808ee6bd395a05b4d7241e83bd24a318f4caa49b284e" title="No Error.">TSV_OK</a>);
<a name="l00416"></a>00416   <span class="comment">//</span>
<a name="l00417"></a>00417   tsv.bind(0,<span class="stringliteral">&quot;intensities&quot;</span>,&amp;intensity);
<a name="l00418"></a>00418   assert(tsv.<a class="code" href="classaffx_1_1TsvFile.html#a2dbd33415a815e221ae61cf3feac5500" title="Covert a column index to a column index ///.">cname2cidx</a>(0,<span class="stringliteral">&quot;intensities&quot;</span>)&gt;=0);
<a name="l00419"></a>00419 
<a name="l00420"></a>00420   sketch.clear();
<a name="l00421"></a>00421   <span class="keywordflow">while</span> (tsv.<a class="code" href="classaffx_1_1TsvFile.html#a2022895d082d1d635e7f59fccab12afb" title="Skip to the next level of data which matches seek_clvl ///.">nextLevel</a>(0)==TSV_OK) {
<a name="l00422"></a>00422     sketch.push_back(intensity);
<a name="l00423"></a>00423   }
<a name="l00424"></a>00424   tsv.<a class="code" href="classaffx_1_1TsvFile.html#ae7932be2156c04a562f327b53833aed7" title="Close the file.">close</a>();
<a name="l00425"></a>00425 }
<a name="l00426"></a>00426 
<a name="l00427"></a>00427 <span class="comment"></span>
<a name="l00428"></a>00428 <span class="comment">/** </span>
<a name="l00429"></a>00429 <span class="comment"> * Read in the precomputed genotypes from a tab delimited file. Must</span>
<a name="l00430"></a>00430 <span class="comment"> * be a column for each Celfile with same header name as celfile and an row</span>
<a name="l00431"></a>00431 <span class="comment"> * for every snp to be analyzed. There can be extra columns and order of columns</span>
<a name="l00432"></a>00432 <span class="comment"> * is not important as the resulting order will be the same as the colNames</span>
<a name="l00433"></a>00433 <span class="comment"> * vector. Steal from GenoSeedTxt. </span>
<a name="l00434"></a>00434 <span class="comment"> * @param fileName - text file containing the known genotype calls</span>
<a name="l00435"></a>00435 <span class="comment"> * @para colNames - the name of samples. </span>
<a name="l00436"></a>00436 <span class="comment"> * @para knownCalls -- the map that store knownCalls infomation for each SNP.</span>
<a name="l00437"></a>00437 <span class="comment"> */</span>
<a name="l00438"></a><a class="code" href="GenoUtility_8h.html#a0267cd994e2dec10ca3a6f1e4c16113c">00438</a> <span class="keywordtype">void</span> <a class="code" href="GenoUtility_8cpp.html#a0267cd994e2dec10ca3a6f1e4c16113c" title="Read in the precomputed genotypes from a tab delimited file.">loadSNPHints</a>(std::string fileName, 
<a name="l00439"></a>00439                   std::vector&lt;std::string&gt; &amp;colNames,
<a name="l00440"></a>00440                   std::map&lt;std::string,std::vector&lt;affx::GType&gt; &gt; &amp;knownCalls) {
<a name="l00441"></a>00441   <a class="code" href="classaffx_1_1TsvFile.html" title="A class for reading and writing Tab Seperated Value (TSV) files. /// See the TsvFile format document ...">affx::TsvFile</a> tsv;
<a name="l00442"></a>00442   vector&lt;GType&gt; calls(colNames.size());
<a name="l00443"></a>00443   vector&lt;int&gt; intCalls(colNames.size());
<a name="l00444"></a>00444   vector&lt;uint32_t&gt; hetChrxCalls(colNames.size(), 0);
<a name="l00445"></a>00445   vector&lt;uint32_t&gt; hetChrxPossibleCalls(colNames.size(), 0);
<a name="l00446"></a>00446   vector&lt;uint32_t&gt; madeCalls(colNames.size(), 0);
<a name="l00447"></a>00447   vector&lt;uint32_t&gt; possibleCalls(colNames.size(), 0);
<a name="l00448"></a>00448   
<a name="l00449"></a>00449   <span class="keywordtype">string</span> probeset;
<a name="l00450"></a>00450 
<a name="l00451"></a>00451   <span class="comment">/* Expecting one column for the id and one for each cel file. */</span>
<a name="l00452"></a>00452   tsv.bind(0,<span class="stringliteral">&quot;probeset_id&quot;</span>, &amp;probeset, <a class="code" href="namespaceaffx.html#a8f36853f1d3b6cfb79edebd82760691daf605fbe0e244c4a604d176991d8df2c6" title="warn if not bound">TSV_BIND_REQUIRED</a>);
<a name="l00453"></a>00453   <span class="keywordflow">for</span>(uint32_t i = 0; i &lt; colNames.size(); i++) {
<a name="l00454"></a>00454     <span class="keywordtype">string</span> s = <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(colNames[i]);
<a name="l00455"></a>00455     tsv.bind(0, s.c_str(), &amp;intCalls[i], TSV_BIND_REQUIRED);
<a name="l00456"></a>00456   }
<a name="l00457"></a>00457 
<a name="l00458"></a>00458   <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Reading in hints genotypes from &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(fileName));
<a name="l00459"></a>00459 
<a name="l00460"></a>00460   <span class="keywordflow">if</span>(tsv.<a class="code" href="classaffx_1_1TsvFile.html#a763ce0bbf19b4134d0445aa58e0f5011" title="Opens a file -- attempts to guess some defaults.">open</a>(fileName) != TSV_OK) {
<a name="l00461"></a>00461     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Couldn&#39;t open file: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(fileName) + <span class="stringliteral">&quot; to read.&quot;</span>);
<a name="l00462"></a>00462   }
<a name="l00463"></a>00463 
<a name="l00464"></a>00464   knownCalls.clear();
<a name="l00465"></a>00465 
<a name="l00466"></a>00466   <span class="keywordflow">while</span>(tsv.<a class="code" href="classaffx_1_1TsvFile.html#a2022895d082d1d635e7f59fccab12afb" title="Skip to the next level of data which matches seek_clvl ///.">nextLevel</a>(0) == TSV_OK) {
<a name="l00467"></a>00467     <span class="keywordflow">if</span>(knownCalls.find(probeset) != knownCalls.end()) {
<a name="l00468"></a>00468       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Found probeset: &quot;</span> + probeset + <span class="stringliteral">&quot; specified multiple times in file: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(fileName));
<a name="l00469"></a>00469     }
<a name="l00470"></a>00470     
<a name="l00471"></a>00471     <span class="comment">/* convert to char as tsv doesn&#39;t bind char. */</span>
<a name="l00472"></a>00472     <span class="keywordflow">for</span>(uint32_t i = 0; i &lt; intCalls.size(); i++) {
<a name="l00473"></a>00473       <span class="comment">// this converts -1 to NN, 0 to AA, 1 to AB, 2 to BB</span>
<a name="l00474"></a>00474       calls[i] = <a class="code" href="namespaceaffx.html#a8368f9d25c1badaa812816c14dc29ec0" title="Convert an integer from a file to an internal GType ///.">GType_from_int</a>(intCalls[i]);
<a name="l00475"></a>00475     }
<a name="l00476"></a>00476     knownCalls[probeset] = calls; <span class="comment">// contain no calls for a SNP.</span>
<a name="l00477"></a>00477   }
<a name="l00478"></a>00478   tsv.<a class="code" href="classaffx_1_1TsvFile.html#ae7932be2156c04a562f327b53833aed7" title="Close the file.">close</a>();
<a name="l00479"></a>00479 }
<a name="l00480"></a>00480 
<a name="l00481"></a>00481 
<a name="l00482"></a>00482 <span class="keywordtype">bool</span> probesetMinMax(<a class="code" href="classaffymetrix__fusion__io_1_1FusionCELData.html">affymetrix_fusion_io::FusionCELData</a> *cel, <span class="keyword">const</span> <a class="code" href="classProbeSet.html" title="Collection of atoms that measure target that should be absent or present at the same time...">ProbeSet</a> *ps, 
<a name="l00483"></a>00483                   <span class="keywordtype">float</span> &amp;Min, <span class="keywordtype">float</span> &amp;<a class="code" href="affy-base-types_8h.html#a4886a8f966a69949cefc46a6a3468006">Max</a>) {
<a name="l00484"></a>00484   Min = -1;
<a name="l00485"></a>00485   Max = -1;
<a name="l00486"></a>00486   vector&lt;float&gt; Values;
<a name="l00487"></a>00487   <span class="comment">/* Genotype probe sets can have either 2 or 4 groups, for each</span>
<a name="l00488"></a>00488 <span class="comment">     type the even (0,2) are A alleles and odd (1,3) are B allele.s */</span>
<a name="l00489"></a>00489   <span class="comment">//if(ps-&gt;numGroups != 2 &amp;&amp; ps-&gt;numGroups != 4) {</span>
<a name="l00490"></a>00490   <span class="comment">//  return false;</span>
<a name="l00491"></a>00491   <span class="comment">//}</span>
<a name="l00492"></a>00492   
<a name="l00493"></a>00493   <span class="comment">//Verbose::out(4, &quot;there are &quot;+ ToStr(ps-&gt;numGroups) + &quot; groups&quot;);</span>
<a name="l00494"></a>00494   <a class="code" href="classErr.html#ad1a633f602fe60689cdc12d9ca1842e9" title="errAbort based version of assert with an error message.">Err::check</a>(ps-&gt;<a class="code" href="classProbeSet.html#a6d8ed948634c42033bd0c35d2640b274" title="How many groups (or blocks) are there? Used extensively in genotyping chips.">numGroups</a> == 1, 
<a name="l00495"></a>00495              <span class="stringliteral">&quot;alleleMinMax() - Thought for Falcon, there should just be 1 group.&quot;</span>);
<a name="l00496"></a>00496   <span class="comment">//for(int psIx = 0; psIx &lt; ps-&gt;numGroups - 1; psIx ++) {</span>
<a name="l00497"></a>00497     <span class="comment">//Err::check(ps-&gt;atomsPerGroup[psIx] == ps-&gt;atomsPerGroup[psIx+1], </span>
<a name="l00498"></a>00498     <span class="comment">//&quot;alleleMinMax() - Thought groups were symmetrical.&quot;);</span>
<a name="l00499"></a>00499   <span class="keywordtype">int</span> atomsPer = ps-&gt;<a class="code" href="classProbeSet.html#ab92eb06095441bf193a6fce3f758540b" title="Atoms that make up this probe set.">atoms</a>.size();
<a name="l00500"></a>00500   <span class="comment">//Verbose::out(4, &quot;atoms per group is &quot;+ ToStr(atomsPer));</span>
<a name="l00501"></a>00501   <span class="comment">/* Each block will have atomsPer atoms in there. */</span>
<a name="l00502"></a>00502   <span class="keywordflow">for</span>(<span class="keywordtype">int</span> atomIx = 0; atomIx &lt; atomsPer; atomIx++) {
<a name="l00503"></a>00503     <a class="code" href="classAtom.html" title="Represents a collection of probes that make up a unit for an intensity measurement of a particular se...">Atom</a> *A = ps-&gt;<a class="code" href="classProbeSet.html#ab92eb06095441bf193a6fce3f758540b" title="Atoms that make up this probe set.">atoms</a>[atomIx];
<a name="l00504"></a>00504     <span class="keywordflow">for</span>(<span class="keywordtype">int</span> probeIx = 0; probeIx &lt; A-&gt;<a class="code" href="classAtom.html#a5953a4944a1e24b3f51a20ce92d8def2" title="Vector of probes that make up this atom. Order is important.">probes</a>.size(); probeIx++) {
<a name="l00505"></a>00505       <a class="code" href="classProbe.html" title="Represents an individual feature on a chip.">Probe</a> *p = A-&gt;<a class="code" href="classAtom.html#a5953a4944a1e24b3f51a20ce92d8def2" title="Vector of probes that make up this atom. Order is important.">probes</a>[probeIx];
<a name="l00506"></a>00506       <span class="keywordflow">if</span>(Probe::isPm(*p)) {
<a name="l00507"></a>00507         <span class="keywordtype">float</span> intensity = cel-&gt;<a class="code" href="classaffymetrix__fusion__io_1_1FusionCELData.html#a76d2432b4075dbb9f00a1cf306a3a614">GetIntensity</a>(p-&gt;<a class="code" href="classProbe.html#a839c453922e807335dc6f4b385ed456c" title="Id of probe, in practice usually realated to position on array.">id</a>);
<a name="l00508"></a>00508         Values.push_back(intensity);
<a name="l00509"></a>00509         <span class="keywordflow">if</span> (Min == -1 &amp;&amp; Max == -1){
<a name="l00510"></a>00510           Min = intensity;
<a name="l00511"></a>00511           Max = intensity;
<a name="l00512"></a>00512         }
<a name="l00513"></a>00513         <span class="keywordflow">else</span>{
<a name="l00514"></a>00514           <span class="keywordflow">if</span> (intensity &gt; Max) Max = intensity;
<a name="l00515"></a>00515           <span class="keywordflow">if</span> (intensity &lt; Min) Min = intensity;
<a name="l00516"></a>00516         }
<a name="l00517"></a>00517       }
<a name="l00518"></a>00518     }
<a name="l00519"></a>00519     
<a name="l00520"></a>00520   }
<a name="l00521"></a>00521   
<a name="l00522"></a>00522   <span class="comment">//}</span>
<a name="l00523"></a>00523   <span class="keywordflow">if</span>(Values.empty()) 
<a name="l00524"></a>00524     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;alleleMinMax() - Empty value vectors in probeset: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(ps-&gt;<a class="code" href="classProbeSet.html#af10b2dac78a597ec736691302687c3f3" title="Name of probe set.">name</a>));
<a name="l00525"></a>00525   <span class="comment">//float median = median_in_place(Values.begin(), Values.end());</span>
<a name="l00526"></a>00526   
<a name="l00527"></a>00527   <span class="keywordflow">return</span> <span class="keyword">true</span>;
<a name="l00528"></a>00528 }
</pre></div></div>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:35 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
