<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: birdseed-v1/GenotypeCaller.v1.cpp Source File</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li class="current"><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="tabs2">
    <ul class="tablist">
      <li><a href="files.html"><span>File&nbsp;List</span></a></li>
      <li><a href="globals.html"><span>File&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath">
    <ul>
      <li><a class="el" href="dir_bc639060591548de7d2f699ed08c7abe.html">birdseed-v1</a>      </li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>GenotypeCaller.v1.cpp</h1>  </div>
</div>
<div class="contents">
<a href="GenotypeCaller_8v1_8cpp.html">Go to the documentation of this file.</a><div class="fragment"><pre class="fragment"><a name="l00001"></a>00001 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00002"></a>00002 <span class="comment"></span><span class="comment">//</span>
<a name="l00003"></a>00003 <span class="comment">// Copyright (C) 2007 The Broad Institute and Affymetrix, Inc.</span>
<a name="l00004"></a>00004 <span class="comment">//</span>
<a name="l00005"></a>00005 <span class="comment">// This program is free software; you can redistribute it and/or modify </span>
<a name="l00006"></a>00006 <span class="comment">// it under the terms of the GNU General Public License (version 2) as </span>
<a name="l00007"></a>00007 <span class="comment">// published by the Free Software Foundation.</span>
<a name="l00008"></a>00008 <span class="comment">// </span>
<a name="l00009"></a>00009 <span class="comment">// This program is distributed in the hope that it will be useful, </span>
<a name="l00010"></a>00010 <span class="comment">// but WITHOUT ANY WARRANTY; without even the implied warranty of </span>
<a name="l00011"></a>00011 <span class="comment">// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU </span>
<a name="l00012"></a>00012 <span class="comment">// General Public License for more details.</span>
<a name="l00013"></a>00013 <span class="comment">// </span>
<a name="l00014"></a>00014 <span class="comment">// You should have received a copy of the GNU General Public License </span>
<a name="l00015"></a>00015 <span class="comment">// along with this program;if not, write to the </span>
<a name="l00016"></a>00016 <span class="comment">// </span>
<a name="l00017"></a>00017 <span class="comment">// Free Software Foundation, Inc., </span>
<a name="l00018"></a>00018 <span class="comment">// 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA</span>
<a name="l00019"></a>00019 <span class="comment">//</span><span class="comment"></span>
<a name="l00020"></a>00020 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00021"></a>00021 <span class="comment"></span><span class="comment"></span>
<a name="l00022"></a>00022 <span class="comment">/// @file GenotypeCaller.v1.cpp</span>
<a name="l00023"></a>00023 <span class="comment"></span>
<a name="l00024"></a>00024 <span class="comment">//</span>
<a name="l00025"></a>00025 <span class="preprocessor">#include &quot;birdseed-v1/FitSNPGaussiansPriors3.h&quot;</span>
<a name="l00026"></a>00026 <span class="preprocessor">#include &quot;birdseed-v1/GenotypeCaller.h&quot;</span>
<a name="l00027"></a>00027 <span class="comment">//</span>
<a name="l00028"></a>00028 <span class="preprocessor">#include &quot;broadutil/BroadException.h&quot;</span>
<a name="l00029"></a>00029 <span class="comment">//</span>
<a name="l00030"></a>00030 <span class="preprocessor">#include &lt;algorithm&gt;</span>
<a name="l00031"></a>00031 <span class="preprocessor">#include &lt;cstring&gt;</span>
<a name="l00032"></a>00032 <span class="preprocessor">#include &lt;iostream&gt;</span>
<a name="l00033"></a>00033 <span class="preprocessor">#include &lt;string.h&gt;</span>
<a name="l00034"></a>00034 <span class="comment">//</span>
<a name="l00035"></a>00035 
<a name="l00036"></a>00036 <span class="keyword">using namespace </span>std;
<a name="l00037"></a>00037 <span class="keyword">using namespace </span>birdseed::v1;
<a name="l00038"></a>00038 <span class="keyword">using namespace </span>birdseed::dev;
<a name="l00039"></a>00039 
<a name="l00040"></a>00040 <span class="keyword">static</span> <span class="keyword">const</span> <span class="keywordtype">double</span> MIN_VARIANCE = 1e-14;
<a name="l00041"></a>00041 
<a name="l00042"></a>00042 <span class="keyword">static</span> <span class="keywordtype">void</span> floorClusterWeights(<a class="code" href="classbirdseed_1_1dev_1_1Clusters.html">Clusters</a> *clusters, <span class="keywordtype">int</span> verbosity)
<a name="l00043"></a>00043 {
<a name="l00044"></a>00044     <span class="comment">// Make sure all weights are above minimum</span>
<a name="l00045"></a>00045     <span class="keywordflow">for</span> (<span class="keywordtype">size_t</span> i = 0; i &lt; clusters-&gt;weights.size(); ++i) {
<a name="l00046"></a>00046         clusters-&gt;weights[i] = max(clusters-&gt;weights[i], final_weight_min);
<a name="l00047"></a>00047     }
<a name="l00048"></a>00048     <span class="keywordflow">if</span> (verbosity &gt;= 2) {
<a name="l00049"></a>00049         cout &lt;&lt; <span class="stringliteral">&quot;After weight min: &quot;</span> &lt;&lt; clusters-&gt;tostring() &lt;&lt; <span class="stringliteral">&quot;\n&quot;</span>;
<a name="l00050"></a>00050     }
<a name="l00051"></a>00051 }
<a name="l00052"></a>00052 
<a name="l00053"></a>00053 
<a name="l00054"></a>00054 <span class="keyword">static</span> <a class="code" href="structbirdseed_1_1dev_1_1CallAndConfidence.html">CallAndConfidence</a> doCall(<span class="keyword">const</span> <a class="code" href="classbirdseed_1_1dev_1_1VarVector.html">PXI_ZJMatrix::ROW</a> &amp;pxi_zj,
<a name="l00055"></a>00055                                 <span class="keywordtype">bool</span> isDiploid,
<a name="l00056"></a>00056                                 <span class="keyword">const</span> <a class="code" href="classbirdseed_1_1dev_1_1FixedVector.html">IntensityMatrix::ROW</a> &amp;intensities,
<a name="l00057"></a>00057                                 <span class="keyword">const</span> <a class="code" href="classbirdseed_1_1dev_1_1Clusters.html">Clusters</a> &amp;clusters,
<a name="l00058"></a>00058                                 <span class="keyword">const</span> <a class="code" href="classbirdseed_1_1dev_1_1FixedMatrix.html">Matrix2x2</a> *B,
<a name="l00059"></a>00059                                 std::string priorName)
<a name="l00060"></a>00060 {
<a name="l00061"></a>00061     <span class="keywordtype">int</span> biggest = -1;
<a name="l00062"></a>00062     <span class="keywordtype">int</span> nextBiggest = -1;
<a name="l00063"></a>00063     <span class="keywordtype">double</span> biggestVal = -INFINITY;
<a name="l00064"></a>00064     <span class="keywordtype">double</span> nextBiggestVal = -INFINITY;
<a name="l00065"></a>00065     <span class="keywordflow">for</span> (<span class="keywordtype">size_t</span> i = 0; i &lt; pxi_zj.size(); ++i) {
<a name="l00066"></a>00066         <span class="keywordflow">if</span> (pxi_zj[i] &gt; biggestVal) {
<a name="l00067"></a>00067             nextBiggest = biggest;
<a name="l00068"></a>00068             nextBiggestVal = biggestVal;
<a name="l00069"></a>00069             biggest = i;
<a name="l00070"></a>00070             biggestVal = pxi_zj[i];
<a name="l00071"></a>00071         } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (pxi_zj[i] &gt; nextBiggestVal) {
<a name="l00072"></a>00072             nextBiggest = i;
<a name="l00073"></a>00073             nextBiggestVal = pxi_zj[i];
<a name="l00074"></a>00074         }
<a name="l00075"></a>00075     }
<a name="l00076"></a>00076 
<a name="l00077"></a>00077     <span class="keywordtype">double</span> pvalues[MAX_NUM_CLUSTERS];
<a name="l00078"></a>00078     memset(pvalues, 0, <span class="keyword">sizeof</span>(pvalues));
<a name="l00079"></a>00079 
<a name="l00080"></a>00080     <span class="comment">//</span>
<a name="l00081"></a>00081     <span class="keywordflow">if</span> ((biggest==-1)||(nextBiggest==-1)) {
<a name="l00082"></a>00082       <span class="comment">// dump pxi_zj</span>
<a name="l00083"></a>00083 <span class="comment">//      printf(&quot;pxi_zj[%d]=[&quot;,(int)index);</span>
<a name="l00084"></a>00084 <span class="comment">//      for (size_t i = 0; i &lt; pxi_zj.size(); ++i) {</span>
<a name="l00085"></a>00085 <span class="comment">//        printf(&quot;%s%f&quot;,((i!=0)?&quot;,&quot;:&quot;&quot;),pxi_zj[i]);</span>
<a name="l00086"></a>00086 <span class="comment">//      }</span>
<a name="l00087"></a>00087 <span class="comment">//      printf(&quot;]\n&quot;);</span>
<a name="l00088"></a>00088       <span class="comment">//</span>
<a name="l00089"></a>00089       <span class="keywordflow">return</span> <a class="code" href="structbirdseed_1_1dev_1_1CallAndConfidence.html">CallAndConfidence</a>(CallAndConfidence::NOCALL, 0.0, pvalues, isDiploid);
<a name="l00090"></a>00090     }
<a name="l00091"></a>00091 
<a name="l00092"></a>00092     <span class="keywordflow">for</span> (<span class="keywordtype">size_t</span> i = 0; i &lt; pxi_zj.size(); ++i) {
<a name="l00093"></a>00093         <span class="comment">// apt-probeset-genotype want a higher value to mean lower likelihood</span>
<a name="l00094"></a>00094         pvalues[i] = 1 - pxi_zj[i];
<a name="l00095"></a>00095     }
<a name="l00096"></a>00096 
<a name="l00097"></a>00097     <span class="keywordflow">if</span> (pxi_zj[biggest] &lt;= 0) {
<a name="l00098"></a>00098         cerr &lt;&lt; <span class="stringliteral">&quot;WARNING: pxi_zj[biggest] is not positive: &quot;</span> &lt;&lt; pxi_zj[biggest] &lt;&lt; <span class="stringliteral">&quot; for &quot;</span> &lt;&lt; priorName &lt;&lt; endl;
<a name="l00099"></a>00099     }
<a name="l00100"></a>00100     
<a name="l00101"></a>00101     <span class="keywordtype">double</span> confidence = (pxi_zj[biggest] &gt; 0? pxi_zj[nextBiggest] / pxi_zj[biggest]: 1.0);
<a name="l00102"></a>00102     <span class="keywordtype">double</span> confidenceRelativeToCluster =
<a name="l00103"></a>00103         calculateConfidenceRelativeToCluster(B[biggest],
<a name="l00104"></a>00104                                              intensities,
<a name="l00105"></a>00105                                              clusters.means[biggest]);
<a name="l00106"></a>00106     confidence = confidence * relative_distance_confidence_weight +
<a name="l00107"></a>00107         confidenceRelativeToCluster * (1.0 - relative_distance_confidence_weight);
<a name="l00108"></a>00108 
<a name="l00109"></a>00109     <span class="keywordflow">if</span> (isDiploid) {
<a name="l00110"></a>00110         <span class="keywordflow">switch</span> (biggest) {
<a name="l00111"></a>00111         <span class="keywordflow">case</span> Priors::AA_INDEX:
<a name="l00112"></a>00112             <span class="keywordflow">return</span> <a class="code" href="structbirdseed_1_1dev_1_1CallAndConfidence.html">CallAndConfidence</a>(<a class="code" href="namespaceaffx.html#a3886a33ccf4ba70052a915d878634c30a3e1bc1d31bc880685d5870951ffb5f17" title="Homozygous A version of allele. (0).">CallAndConfidence::AA</a>, confidence, pvalues, isDiploid);
<a name="l00113"></a>00113         <span class="keywordflow">case</span> Priors::AB_INDEX:
<a name="l00114"></a>00114             <span class="keywordflow">return</span> <a class="code" href="structbirdseed_1_1dev_1_1CallAndConfidence.html">CallAndConfidence</a>(<a class="code" href="namespaceaffx.html#a3886a33ccf4ba70052a915d878634c30a83b51b7ed2e4a1b4e23378773929465b" title="Heterozygous, one copy of both A and B. (1).">CallAndConfidence::AB</a>, confidence, pvalues, isDiploid);
<a name="l00115"></a>00115         <span class="keywordflow">case</span> Priors::BB_INDEX:
<a name="l00116"></a>00116             <span class="keywordflow">return</span> <a class="code" href="structbirdseed_1_1dev_1_1CallAndConfidence.html">CallAndConfidence</a>(<a class="code" href="namespaceaffx.html#a3886a33ccf4ba70052a915d878634c30a7c78281ed74a1b7cfe2ab49e6af25231" title="Homozygous B version of allele. (2).">CallAndConfidence::BB</a>, confidence, pvalues, isDiploid);
<a name="l00117"></a>00117         <span class="keywordflow">default</span>:
<a name="l00118"></a>00118             assert(<span class="keyword">false</span>);
<a name="l00119"></a>00119             <span class="keywordflow">return</span> <a class="code" href="structbirdseed_1_1dev_1_1CallAndConfidence.html">CallAndConfidence</a>(CallAndConfidence::NOCALL, 0.0, pvalues, isDiploid);
<a name="l00120"></a>00120         }
<a name="l00121"></a>00121     } <span class="keywordflow">else</span> {
<a name="l00122"></a>00122         <span class="keywordflow">switch</span> (biggest) {
<a name="l00123"></a>00123         <span class="keywordflow">case</span> Priors::A_INDEX:
<a name="l00124"></a>00124             <span class="keywordflow">return</span> <a class="code" href="structbirdseed_1_1dev_1_1CallAndConfidence.html">CallAndConfidence</a>(<a class="code" href="namespaceaffx.html#a3886a33ccf4ba70052a915d878634c30a3e1bc1d31bc880685d5870951ffb5f17" title="Homozygous A version of allele. (0).">CallAndConfidence::AA</a>, confidence, pvalues, isDiploid);
<a name="l00125"></a>00125         <span class="keywordflow">case</span> Priors::B_INDEX:
<a name="l00126"></a>00126             <span class="keywordflow">return</span> <a class="code" href="structbirdseed_1_1dev_1_1CallAndConfidence.html">CallAndConfidence</a>(<a class="code" href="namespaceaffx.html#a3886a33ccf4ba70052a915d878634c30a7c78281ed74a1b7cfe2ab49e6af25231" title="Homozygous B version of allele. (2).">CallAndConfidence::BB</a>, confidence, pvalues, isDiploid);
<a name="l00127"></a>00127         <span class="keywordflow">default</span>:
<a name="l00128"></a>00128             assert(<span class="keyword">false</span>);
<a name="l00129"></a>00129             <span class="keywordflow">return</span> <a class="code" href="structbirdseed_1_1dev_1_1CallAndConfidence.html">CallAndConfidence</a>(CallAndConfidence::NOCALL, 0.0, pvalues, isDiploid);
<a name="l00130"></a>00130         }
<a name="l00131"></a>00131     }
<a name="l00132"></a>00132 }
<a name="l00133"></a>00133 
<a name="l00134"></a>00134 <span class="keyword">static</span> <span class="keywordtype">void</span> doCalls(vector&lt;CallAndConfidence&gt; *calls,
<a name="l00135"></a>00135                     <span class="keyword">const</span> <a class="code" href="classbirdseed_1_1dev_1_1Clusters.html">Clusters</a> &amp;clusters,
<a name="l00136"></a>00136                     <span class="keyword">const</span> <a class="code" href="classbirdseed_1_1dev_1_1VarMatrix.html">IntensityMatrix</a> &amp;intensities,
<a name="l00137"></a>00137                     <span class="keywordtype">bool</span> isDiploid,
<a name="l00138"></a>00138                     <span class="keyword">const</span> <span class="keywordtype">string</span> &amp;priorName,
<a name="l00139"></a>00139                     <span class="keywordtype">size_t</span> verbosity)
<a name="l00140"></a>00140 {
<a name="l00141"></a>00141     <span class="keywordtype">size_t</span> numClusters = (isDiploid? MAX_NUM_CLUSTERS: MAX_NUM_CLUSTERS - 1);
<a name="l00142"></a>00142     <a class="code" href="classbirdseed_1_1dev_1_1VarVarMatrix.html">PXI_ZJMatrix</a> pxi_zj(intensities.numRows(), numClusters);
<a name="l00143"></a>00143     calculatePXI_ZJ(&amp;pxi_zj, clusters, intensities, numClusters);
<a name="l00144"></a>00144 
<a name="l00145"></a>00145     <a class="code" href="classbirdseed_1_1dev_1_1FixedMatrix.html">Matrix2x2</a> B[MAX_NUM_CLUSTERS];
<a name="l00146"></a>00146     <span class="keywordflow">for</span> (<span class="keywordtype">size_t</span> i = 0; i &lt; numClusters; ++i) {
<a name="l00147"></a>00147         B[i] = makeBMatrix(clusters.vars[i], clusters.covar);
<a name="l00148"></a>00148     }
<a name="l00149"></a>00149 
<a name="l00150"></a>00150     calls-&gt;reserve(intensities.numRows());
<a name="l00151"></a>00151     <span class="keywordflow">for</span> (<span class="keywordtype">size_t</span> i = 0; i &lt; intensities.numRows(); ++i) {
<a name="l00152"></a>00152         calls-&gt;push_back(doCall(pxi_zj[i], isDiploid, intensities[i], clusters, B, priorName));
<a name="l00153"></a>00153     }
<a name="l00154"></a>00154 }
<a name="l00155"></a>00155 
<a name="l00156"></a>00156 
<a name="l00157"></a>00157 <span class="keyword">static</span> <span class="keywordtype">void</span> adjustPriors(<a class="code" href="classbirdseed_1_1dev_1_1Priors.html">Priors</a> *priors, <span class="keyword">const</span> <a class="code" href="classbirdseed_1_1dev_1_1VarMatrix.html">IntensityMatrix</a> &amp;intensities, <span class="keywordtype">double</span> correctionFactor)
<a name="l00158"></a>00158 {
<a name="l00159"></a>00159     <span class="keywordtype">double</span> squared = correctionFactor * correctionFactor;
<a name="l00160"></a>00160 
<a name="l00161"></a>00161     <span class="keywordtype">bool</span> seenNonZeroObservations = <span class="keyword">false</span>;
<a name="l00162"></a>00162     <span class="keywordflow">for</span> (<span class="keywordtype">size_t</span> i = 0; i &lt; priors-&gt;getNumPriors(); ++i) {
<a name="l00163"></a>00163         priors-&gt;getPrior(i).m_mean = priors-&gt;getPrior(i).m_mean * correctionFactor;
<a name="l00164"></a>00164         priors-&gt;getPrior(i).m_covarMatrix = priors-&gt;getPrior(i).m_covarMatrix * squared;
<a name="l00165"></a>00165         priors-&gt;getPrior(i).m_covarMatrix[0][0] =
<a name="l00166"></a>00166             max(priors-&gt;getPrior(i).m_covarMatrix[0][0], MIN_VARIANCE);
<a name="l00167"></a>00167         priors-&gt;getPrior(i).m_covarMatrix[1][1] =
<a name="l00168"></a>00168             max(priors-&gt;getPrior(i).m_covarMatrix[1][1], MIN_VARIANCE);
<a name="l00169"></a>00169 
<a name="l00170"></a>00170         <span class="keywordflow">if</span> (priors-&gt;getPrior(i).m_numObservations &gt; 0) {
<a name="l00171"></a>00171             seenNonZeroObservations = <span class="keyword">true</span>;
<a name="l00172"></a>00172         }
<a name="l00173"></a>00173     }
<a name="l00174"></a>00174     <span class="keywordflow">if</span> (!seenNonZeroObservations) {
<a name="l00175"></a>00175         <span class="keywordflow">for</span> (<span class="keywordtype">size_t</span> i = 0; i &lt; priors-&gt;getNumPriors(); ++i) {
<a name="l00176"></a>00176             <span class="comment">// Don&#39;t allow numObservations to be zero, to avoid get NaNs from FP arithmetic</span>
<a name="l00177"></a>00177             priors-&gt;getPrior(i).m_numObservations = 1;
<a name="l00178"></a>00178         }
<a name="l00179"></a>00179     }
<a name="l00180"></a>00180 }
<a name="l00181"></a>00181 
<a name="l00182"></a>00182 birdseed::v1::BasicGenotypeCaller::BasicGenotypeCaller(<span class="keyword">const</span> <a class="code" href="classbirdseed_1_1dev_1_1VarMatrix.html">IntensityMatrix</a> &amp;intensities,
<a name="l00183"></a>00183                                          <span class="keyword">const</span> <a class="code" href="classbirdseed_1_1dev_1_1Priors.html">Priors</a> &amp;priors,
<a name="l00184"></a>00184                                          <span class="keywordtype">double</span> correctionFactor,
<a name="l00185"></a>00185                                          <span class="keyword">const</span> <span class="keywordtype">string</span> &amp;priorName,
<a name="l00186"></a>00186                                          <span class="keywordtype">int</span> verbosity,
<a name="l00187"></a>00187                                          std::ostream *clusterOstrm):
<a name="l00188"></a>00188     calls(),
<a name="l00189"></a>00189     index(0)
<a name="l00190"></a>00190 {
<a name="l00191"></a>00191     <span class="keywordflow">if</span> (intensities.numRows() &lt; MIN_SAMPLES_TO_CLUSTER) {
<a name="l00192"></a>00192         std::stringstream strm;
<a name="l00193"></a>00193         strm &lt;&lt; <span class="stringliteral">&quot;Not enough samples to clusters.  &quot;</span> &lt;&lt; MIN_SAMPLES_TO_CLUSTER &lt;&lt; <span class="stringliteral">&quot; are needed but there are only &quot;</span> &lt;&lt;
<a name="l00194"></a>00194             intensities.numRows() &lt;&lt; <span class="stringliteral">&quot;.&quot;</span>;
<a name="l00195"></a>00195         <span class="keywordflow">throw</span> <a class="code" href="classBroadException.html">BroadException</a>(strm.str().c_str(), __FILE__, __LINE__);
<a name="l00196"></a>00196     }
<a name="l00197"></a>00197     
<a name="l00198"></a>00198     setFitSNPGaussiansVerbosity(verbosity);
<a name="l00199"></a>00199     <span class="keywordflow">if</span> (correctionFactor &lt; 0) {
<a name="l00200"></a>00200         correctionFactor = calculateSNPSpecificCorrectionFactor(intensities, priors);
<a name="l00201"></a>00201         <span class="keywordflow">if</span> (verbosity &gt;= 2) {
<a name="l00202"></a>00202             cout.precision(10);
<a name="l00203"></a>00203             cout &lt;&lt; <span class="stringliteral">&quot;SNP-specific correction factor for &quot;</span> &lt;&lt; priorName &lt;&lt; <span class="stringliteral">&quot;=&quot;</span> &lt;&lt; correctionFactor &lt;&lt; endl;
<a name="l00204"></a>00204         }
<a name="l00205"></a>00205     }
<a name="l00206"></a>00206     <a class="code" href="classbirdseed_1_1dev_1_1Priors.html">Priors</a> adjustedPriors(priors);
<a name="l00207"></a>00207     adjustPriors(&amp;adjustedPriors, intensities, correctionFactor);
<a name="l00208"></a>00208     <span class="keywordflow">if</span> (verbosity &gt;= 2) {
<a name="l00209"></a>00209         cout &lt;&lt; <span class="stringliteral">&quot;Priors\n&quot;</span>;
<a name="l00210"></a>00210         <span class="keywordflow">for</span> (<span class="keywordtype">size_t</span> i = 0; i &lt; adjustedPriors.getNumPriors(); ++i) {
<a name="l00211"></a>00211             <span class="keyword">const</span> <a class="code" href="classbirdseed_1_1dev_1_1Prior.html">Prior</a> &amp;prior = adjustedPriors.getPrior(i);
<a name="l00212"></a>00212             cout &lt;&lt; prior.m_mean[0] &lt;&lt; <span class="stringliteral">&quot; &quot;</span> &lt;&lt; prior.m_mean[1] &lt;&lt; <span class="stringliteral">&quot; &quot;</span> &lt;&lt; prior.m_covarMatrix[0][0] &lt;&lt; <span class="stringliteral">&quot; &quot;</span>
<a name="l00213"></a>00213                  &lt;&lt; <span class="stringliteral">&quot; &quot;</span> &lt;&lt; prior.m_covarMatrix[1][0] &lt;&lt; <span class="stringliteral">&quot; &quot;</span> &lt;&lt; prior.m_covarMatrix[1][1] &lt;&lt; <span class="stringliteral">&quot; &quot;</span>
<a name="l00214"></a>00214                  &lt;&lt; prior.m_numObservations &lt;&lt; endl;
<a name="l00215"></a>00215         }
<a name="l00216"></a>00216     }
<a name="l00217"></a>00217     
<a name="l00218"></a>00218     <a class="code" href="classbirdseed_1_1dev_1_1Clusters.html">Clusters</a> clusters = FitSNPGaussianPriors3(intensities, adjustedPriors);
<a name="l00219"></a>00219     <span class="keywordflow">if</span> (clusterOstrm != NULL) {
<a name="l00220"></a>00220         (*clusterOstrm) &lt;&lt; priorName &lt;&lt; clusters.standardClusterString() &lt;&lt; endl;
<a name="l00221"></a>00221     }
<a name="l00222"></a>00222     floorClusterWeights(&amp;clusters, verbosity);
<a name="l00223"></a>00223     doCalls(&amp;calls, clusters, intensities, priors.isDiploid(), priorName, verbosity);
<a name="l00224"></a>00224 }
<a name="l00225"></a>00225 
<a name="l00226"></a>00226 <span class="keywordtype">double</span> birdseed::v1::DummyGenotypeCaller::pvalues[MAX_NUM_CLUSTERS] = {0.0, 0.0, 0.0};
<a name="l00227"></a>00227 
<a name="l00228"></a>00228 <a class="code" href="structbirdseed_1_1dev_1_1CallAndConfidence.html">CallAndConfidence</a> birdseed::v1::DummyGenotypeCaller::nextCall()
<a name="l00229"></a>00229 {
<a name="l00230"></a>00230     <span class="keywordflow">return</span> <a class="code" href="structbirdseed_1_1dev_1_1CallAndConfidence.html">CallAndConfidence</a>(CallAndConfidence::NOCALL, 1.0, pvalues, isDiploid);
<a name="l00231"></a>00231 }
<a name="l00232"></a>00232 
<a name="l00233"></a>00233 birdseed::v1::ForcedClusterGenotypeCaller::ForcedClusterGenotypeCaller(<span class="keyword">const</span> <a class="code" href="classbirdseed_1_1dev_1_1VarMatrix.html">IntensityMatrix</a> &amp;intensities,
<a name="l00234"></a>00234                                                          <span class="keyword">const</span> <a class="code" href="classbirdseed_1_1dev_1_1Clusters.html">Clusters</a> &amp;clusters,
<a name="l00235"></a>00235                                                          <span class="keywordtype">double</span> correctionFactor,
<a name="l00236"></a>00236                                                          <span class="keyword">const</span> std::string &amp;priorName,
<a name="l00237"></a>00237                                                          <span class="keywordtype">int</span> verbosity,
<a name="l00238"></a>00238                                                          std::ostream *clusterOstrm):
<a name="l00239"></a>00239     calls(),
<a name="l00240"></a>00240     index(0)
<a name="l00241"></a>00241 {
<a name="l00242"></a>00242     <span class="keywordtype">bool</span> isDiploid = clusters.k == MAX_NUM_CLUSTERS;
<a name="l00243"></a>00243     <a class="code" href="classbirdseed_1_1dev_1_1Clusters.html">Clusters</a> flooredClusters(clusters);
<a name="l00244"></a>00244     <span class="keywordflow">if</span> (clusterOstrm != NULL) {
<a name="l00245"></a>00245         (*clusterOstrm) &lt;&lt; priorName &lt;&lt; clusters.standardClusterString() &lt;&lt; endl;
<a name="l00246"></a>00246     }
<a name="l00247"></a>00247     floorClusterWeights(&amp;flooredClusters, verbosity);
<a name="l00248"></a>00248     doCalls(&amp;calls, flooredClusters, intensities, isDiploid, priorName, verbosity);
<a name="l00249"></a>00249 }
<a name="l00250"></a>00250 
<a name="l00251"></a>00251 <span class="comment">/******************************************************************/</span>
<a name="l00252"></a>00252 <span class="comment">/**************************[END OF GenotypeCaller.cpp]*************************/</span>
<a name="l00253"></a>00253 <span class="comment">/******************************************************************/</span>
<a name="l00254"></a>00254 
<a name="l00255"></a>00255 <span class="comment">/* Emacs configuration</span>
<a name="l00256"></a>00256 <span class="comment"> * Local Variables:</span>
<a name="l00257"></a>00257 <span class="comment"> * mode: C++</span>
<a name="l00258"></a>00258 <span class="comment"> * tab-width:4</span>
<a name="l00259"></a>00259 <span class="comment"> * End:</span>
<a name="l00260"></a>00260 <span class="comment"> */</span>
</pre></div></div>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:21 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
