<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: translation/MarkerExperimentReport.cpp Source File</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li class="current"><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="tabs2">
    <ul class="tablist">
      <li><a href="files.html"><span>File&nbsp;List</span></a></li>
      <li><a href="globals.html"><span>File&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath">
    <ul>
      <li><a class="el" href="dir_1668076c0e919fc49032c4fb42e3bb3d.html">translation</a>      </li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>MarkerExperimentReport.cpp</h1>  </div>
</div>
<div class="contents">
<a href="MarkerExperimentReport_8cpp.html">Go to the documentation of this file.</a><div class="fragment"><pre class="fragment"><a name="l00001"></a>00001 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00002"></a>00002 <span class="comment"></span><span class="comment">//</span>
<a name="l00003"></a>00003 <span class="comment">// Copyright (C) 2008 Affymetrix, Inc.</span>
<a name="l00004"></a>00004 <span class="comment">//</span>
<a name="l00005"></a>00005 <span class="comment">// This program is free software; you can redistribute it and/or modify </span>
<a name="l00006"></a>00006 <span class="comment">// it under the terms of the GNU General Public License (version 2) as </span>
<a name="l00007"></a>00007 <span class="comment">// published by the Free Software Foundation.</span>
<a name="l00008"></a>00008 <span class="comment">// </span>
<a name="l00009"></a>00009 <span class="comment">// This program is distributed in the hope that it will be useful, </span>
<a name="l00010"></a>00010 <span class="comment">// but WITHOUT ANY WARRANTY; without even the implied warranty of </span>
<a name="l00011"></a>00011 <span class="comment">// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU </span>
<a name="l00012"></a>00012 <span class="comment">// General Public License for more details.</span>
<a name="l00013"></a>00013 <span class="comment">// </span>
<a name="l00014"></a>00014 <span class="comment">// You should have received a copy of the GNU General Public License </span>
<a name="l00015"></a>00015 <span class="comment">// along with this program;if not, write to the </span>
<a name="l00016"></a>00016 <span class="comment">// </span>
<a name="l00017"></a>00017 <span class="comment">// Free Software Foundation, Inc., </span>
<a name="l00018"></a>00018 <span class="comment">// 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA</span>
<a name="l00019"></a>00019 <span class="comment">//</span><span class="comment"></span>
<a name="l00020"></a>00020 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00021"></a>00021 <span class="comment"></span><span class="comment">/**</span>
<a name="l00022"></a>00022 <span class="comment"> * @file   MarkerExperimentReport.cpp</span>
<a name="l00023"></a>00023 <span class="comment"> * @author Mybrid Spalding</span>
<a name="l00024"></a>00024 <span class="comment"> * @date   Tue Mar 18 09:02:01 PDT 2008</span>
<a name="l00025"></a>00025 <span class="comment"> * @brief  DEPRECATED DMET2: Class for the marker report as report files.</span>
<a name="l00026"></a>00026 <span class="comment"> */</span>
<a name="l00027"></a>00027 
<a name="l00028"></a>00028 <span class="comment">//</span>
<a name="l00029"></a>00029 <span class="preprocessor">#include &quot;<a class="code" href="MarkerExperimentReport_8h.html" title="DEPRECATED DMET2: Class for the marker report as report files.">translation/MarkerExperimentReport.h</a>&quot;</span>
<a name="l00030"></a>00030 <span class="comment">//</span>
<a name="l00031"></a>00031 <span class="preprocessor">#include &quot;<a class="code" href="ExperimentGeneResults_8h.html" title="CallResults wrapper class for all gene translated experiment results to be consumed by various report...">translation/ExperimentGeneResults.h</a>&quot;</span>
<a name="l00032"></a>00032 <span class="preprocessor">#include &quot;<a class="code" href="ExperimentResults_8h.html" title="Wrapper class for ExperimentGeneResults, all gene experiment results. There should ever only ever be ...">translation/ExperimentResults.h</a>&quot;</span>
<a name="l00033"></a>00033 <span class="preprocessor">#include &quot;<a class="code" href="RunTimeEnvironment_8h.html" title="A single instance class to hold the global runtime environment, similar shell enviorment variables...">translation/RunTimeEnvironment.h</a>&quot;</span>
<a name="l00034"></a>00034 <span class="preprocessor">#include &quot;<a class="code" href="TranslationTableModel_8h.html" title="Single instance data model class that wraps the translation file in the TranslationInputTsvModel.">translation/TranslationTableModel.h</a>&quot;</span>
<a name="l00035"></a>00035 <span class="comment">//</span>
<a name="l00036"></a>00036 <span class="preprocessor">#include &quot;<a class="code" href="TsvFile_8h.html" title="Headers for the TsvFile classes.">file/TsvFile/TsvFile.h</a>&quot;</span>
<a name="l00037"></a>00037 <span class="preprocessor">#include &quot;<a class="code" href="Err_8h.html" title="Some utility error functions, currently prints message and aborts().">util/Err.h</a>&quot;</span>
<a name="l00038"></a>00038 <span class="preprocessor">#include &quot;<a class="code" href="Fs_8h.html" title="///">util/Fs.h</a>&quot;</span>
<a name="l00039"></a>00039 <span class="preprocessor">#include &quot;<a class="code" href="Guid_8h.html" title="This file provides functions for creating globally unique identifiers.">util/Guid.h</a>&quot;</span>
<a name="l00040"></a>00040 <span class="preprocessor">#include &quot;<a class="code" href="Util_8h.html" title="General Utilities.">util/Util.h</a>&quot;</span>
<a name="l00041"></a>00041 <span class="preprocessor">#include &quot;<a class="code" href="Verbose_8h.html" title="Class for doing logging and some command line ui.">util/Verbose.h</a>&quot;</span>
<a name="l00042"></a>00042 <span class="comment">//</span>
<a name="l00043"></a>00043 <span class="preprocessor">#include &quot;pcrecpp.h&quot;</span>
<a name="l00044"></a>00044 
<a name="l00045"></a>00045 <span class="keyword">using namespace </span>std;
<a name="l00046"></a>00046 <span class="keyword">using namespace </span>affx;
<a name="l00047"></a>00047 
<a name="l00048"></a>00048 
<a name="l00049"></a>00049 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00050"></a>00050 <span class="comment">/**</span>
<a name="l00051"></a>00051 <span class="comment"> * MarkerRowComp::operato()</span>
<a name="l00052"></a>00052 <span class="comment"> *</span>
<a name="l00053"></a>00053 <span class="comment"> * Used in the STL std::sort for sorting MarkerRow objects.</span>
<a name="l00054"></a>00054 <span class="comment"> * Sort order is experiment -&gt; gene -&gt; probeSet.</span>
<a name="l00055"></a>00055 <span class="comment"> *</span>
<a name="l00056"></a>00056 <span class="comment"> *</span>
<a name="l00057"></a>00057 <span class="comment"> * @param  a,b - the two MarkerRow objects to compare.</span>
<a name="l00058"></a>00058 <span class="comment"> * @return true - if ( a &lt; b);</span>
<a name="l00059"></a>00059 <span class="comment"> *</span>
<a name="l00060"></a>00060 <span class="comment"> */</span>
<a name="l00061"></a>00061 <span class="comment">/*****************************************************************************/</span>
<a name="l00062"></a><a class="code" href="classMarkerRowComp.html#ac3070a092cfe868cf78d8fd2790ec7e1">00062</a> <span class="keywordtype">bool</span> <a class="code" href="classMarkerRowComp.html#ac3070a092cfe868cf78d8fd2790ec7e1" title="MarkerRowComp::operato().">MarkerRowComp::operator()</a>(<span class="keyword">const</span> <a class="code" href="classMarkerRow.html">MarkerRow</a> &amp;a,
<a name="l00063"></a>00063                                <span class="keyword">const</span> <a class="code" href="classMarkerRow.html">MarkerRow</a> &amp;b)<span class="keyword"> const</span>
<a name="l00064"></a>00064 <span class="keyword"></span>{
<a name="l00065"></a>00065 
<a name="l00066"></a>00066   <span class="keywordflow">if</span> (a.m_experiment != b.m_experiment) {
<a name="l00067"></a>00067     <span class="keywordflow">return</span>(a.m_experiment.compare(b.m_experiment) &lt;= 0);
<a name="l00068"></a>00068   }
<a name="l00069"></a>00069 
<a name="l00070"></a>00070   <span class="keywordflow">if</span> (a.m_gene != b.m_gene) {
<a name="l00071"></a>00071     <span class="keywordflow">return</span>(a.m_gene.compare(b.m_gene) &lt;= 0);
<a name="l00072"></a>00072   }
<a name="l00073"></a>00073 
<a name="l00074"></a>00074   <span class="keywordflow">return</span> (a.m_probeSet.compare(b.m_probeSet) &lt;= 0);
<a name="l00075"></a>00075 
<a name="l00076"></a>00076 }
<a name="l00077"></a>00077 <span class="comment">// end MarkerRowComp::operator</span>
<a name="l00078"></a>00078 <span class="comment">/*****************************************************************************/</span>
<a name="l00079"></a>00079 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00080"></a>00080 <span class="comment">/**</span>
<a name="l00081"></a>00081 <span class="comment"> * MarkerExperimentReport::close:</span>
<a name="l00082"></a>00082 <span class="comment"> * Synopsis:</span>
<a name="l00083"></a>00083 <span class="comment"> *</span>
<a name="l00084"></a>00084 <span class="comment"> *   Close the Marker report file.</span>
<a name="l00085"></a>00085 <span class="comment"> *</span>
<a name="l00086"></a>00086 <span class="comment"> * @return - void, closes report files.</span>
<a name="l00087"></a>00087 <span class="comment"> */</span>
<a name="l00088"></a>00088 <span class="comment">/*****************************************************************************/</span>
<a name="l00089"></a><a class="code" href="classMarkerExperimentReport.html#a283257ff02ecbc953a9145cce77eef38">00089</a> <span class="keywordtype">bool</span> <a class="code" href="classMarkerExperimentReport.html#a283257ff02ecbc953a9145cce77eef38" title="MarkerExperimentReport::close: Synopsis:">MarkerExperimentReport::close</a>(<span class="keywordtype">bool</span> abort)
<a name="l00090"></a>00090 {
<a name="l00091"></a>00091 
<a name="l00092"></a>00092   <span class="keywordflow">if</span> (! abort) {
<a name="l00093"></a>00093     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(ADT_VERBOSE_NORMAL, <span class="stringliteral">&quot;MarkerExperimentReport::generate called: &quot;</span>, <span class="keyword">false</span>);
<a name="l00094"></a>00094     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(ADT_VERBOSE_NORMAL,  m_reportName);
<a name="l00095"></a>00095   }
<a name="l00096"></a>00096 
<a name="l00097"></a>00097   <span class="keywordflow">if</span> (m_open_called) {
<a name="l00098"></a>00098     m_marker_tsv.<a class="code" href="classaffx_1_1TsvFile.html#ae7932be2156c04a562f327b53833aed7" title="Close the file.">close</a>();
<a name="l00099"></a>00099     m_open_called = <span class="keyword">false</span>;
<a name="l00100"></a>00100   }
<a name="l00101"></a>00101 
<a name="l00102"></a>00102   <span class="keywordflow">return</span> <span class="keyword">true</span>;
<a name="l00103"></a>00103 
<a name="l00104"></a>00104 }
<a name="l00105"></a>00105 <span class="comment">// end MarkerExperimentReport::close</span>
<a name="l00106"></a>00106 <span class="comment">/*****************************************************************************/</span>
<a name="l00107"></a>00107 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00108"></a>00108 <span class="comment">/**</span>
<a name="l00109"></a>00109 <span class="comment"> * _report_marker</span>
<a name="l00110"></a>00110 <span class="comment"> * Synopsis:</span>
<a name="l00111"></a>00111 <span class="comment"> *</span>
<a name="l00112"></a>00112 <span class="comment"> * Helper function for one time use by MarkerExperimentReport::generate.</span>
<a name="l00113"></a>00113 <span class="comment"> *</span>
<a name="l00114"></a>00114 <span class="comment"> * Write a tab delimited record to the output TSV file for marker data.</span>
<a name="l00115"></a>00115 <span class="comment"> *</span>
<a name="l00116"></a>00116 <span class="comment"> * @param marker_tsv - the TsvFile reference</span>
<a name="l00117"></a>00117 <span class="comment"> * @param mRow - the MarkerRow class defined just for this report.</span>
<a name="l00118"></a>00118 <span class="comment"> *</span>
<a name="l00119"></a>00119 <span class="comment"> */</span>
<a name="l00120"></a>00120 <span class="comment">/*****************************************************************************/</span>
<a name="l00121"></a>00121 <span class="keyword">static</span> <span class="keywordtype">void</span> _report_marker(<a class="code" href="classaffx_1_1TsvFile.html" title="A class for reading and writing Tab Seperated Value (TSV) files. /// See the TsvFile format document ...">affx::TsvFile</a> &amp; marker_tsv, <span class="keyword">const</span> <a class="code" href="classMarkerRow.html">MarkerRow</a> &amp; mRow)
<a name="l00122"></a>00122 {
<a name="l00123"></a>00123 
<a name="l00124"></a>00124   <span class="comment">// here we expect things to be in the order defined above.</span>
<a name="l00125"></a>00125   <span class="comment">// otherwise we could say: m_tsv.set(0,&quot;experiment&quot;,m_experiment);</span>
<a name="l00126"></a>00126   marker_tsv.set(0, 0,  mRow.m_experiment);
<a name="l00127"></a>00127   marker_tsv.set(0, 1,  mRow.m_gene);
<a name="l00128"></a>00128   marker_tsv.set(0, 2,  mRow.m_sample);
<a name="l00129"></a>00129   marker_tsv.set(0, 3,  mRow.m_functionalChange);
<a name="l00130"></a>00130   marker_tsv.set(0, 4,  mRow.m_externalId);
<a name="l00131"></a>00131   marker_tsv.set(0, 5,  mRow.m_probeSet);
<a name="l00132"></a>00132   marker_tsv.set(0, 6,  mRow.m_baseCall);
<a name="l00133"></a>00133   marker_tsv.set(0, 7,  mRow.m_referenceBase);
<a name="l00134"></a>00134   marker_tsv.set(0, 8,  mRow.m_variantBase);
<a name="l00135"></a>00135   marker_tsv.set(0, 9,  mRow.m_call);
<a name="l00136"></a>00136   marker_tsv.set(0, 10, mRow.m_haplotypeMarker);
<a name="l00137"></a>00137   marker_tsv.set(0, 11, mRow.m_changeForVariant);
<a name="l00138"></a>00138   marker_tsv.set(0, 12, mRow.m_variantCDNAChange);
<a name="l00139"></a>00139   marker_tsv.set(0, 13, mRow.m_variantDNAChange);
<a name="l00140"></a>00140   marker_tsv.set(0, 14, mRow.m_dbSNPId);
<a name="l00141"></a>00141   marker_tsv.set(0, 15, mRow.m_validated);
<a name="l00142"></a>00142   marker_tsv.set(0, 16, mRow.m_alleleDefiningMarker);
<a name="l00143"></a>00143   marker_tsv.set(0, 17, mRow.m_relevantAlleles);
<a name="l00144"></a>00144 
<a name="l00145"></a>00145   marker_tsv.<a class="code" href="classaffx_1_1TsvFile.html#a2a459be42bea9ea3f180773370ec0de7" title="Write a level of data to the file. ///.">writeLevel</a>(0);
<a name="l00146"></a>00146 
<a name="l00147"></a>00147   <span class="keywordflow">return</span>;
<a name="l00148"></a>00148 }
<a name="l00149"></a>00149 <span class="comment">// end _report_marker</span>
<a name="l00150"></a>00150 <span class="comment">/*****************************************************************************/</span>
<a name="l00151"></a>00151 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00152"></a>00152 <span class="comment">/**</span>
<a name="l00153"></a>00153 <span class="comment"> * MarkerExperimentReport::generate</span>
<a name="l00154"></a>00154 <span class="comment"> *</span>
<a name="l00155"></a>00155 <span class="comment"> * Create the report this object was made for.</span>
<a name="l00156"></a>00156 <span class="comment"> * As with all other report objects the data is generated one experiment</span>
<a name="l00157"></a>00157 <span class="comment"> * at a time.</span>
<a name="l00158"></a>00158 <span class="comment"> *</span>
<a name="l00159"></a>00159 <span class="comment"> *</span>
<a name="l00160"></a>00160 <span class="comment"> * @param rte - the single instnace RunTimeEnvironment which contains options</span>
<a name="l00161"></a>00161 <span class="comment"> * @param ttm - the single instance translation table model</span>
<a name="l00162"></a>00162 <span class="comment"> * @param er  - the container for all experiment results.</span>
<a name="l00163"></a>00163 <span class="comment"> *</span>
<a name="l00164"></a>00164 <span class="comment"> * @return true - when report was generated as expected.</span>
<a name="l00165"></a>00165 <span class="comment"> *</span>
<a name="l00166"></a>00166 <span class="comment"> */</span>
<a name="l00167"></a>00167 <span class="comment">/*****************************************************************************/</span>
<a name="l00168"></a><a class="code" href="classMarkerExperimentReport.html#ae09964d682424f296d8761520f36156a">00168</a> <span class="keywordtype">bool</span> <a class="code" href="classMarkerExperimentReport.html#ae09964d682424f296d8761520f36156a" title="MarkerExperimentReport::generate.">MarkerExperimentReport::generate</a>(<span class="keyword">class</span> <a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp; rte,
<a name="l00169"></a>00169                                       <span class="keyword">class</span> <a class="code" href="classTranslationTableModel.html">TranslationTableModel</a> &amp; ttm,
<a name="l00170"></a>00170                                       std::map&lt;std::string, ExperimentResults *&gt;&amp;er)
<a name="l00171"></a>00171 {
<a name="l00172"></a>00172 
<a name="l00173"></a>00173   <span class="keywordtype">bool</span> okReport = <span class="keyword">true</span>;
<a name="l00174"></a>00174 
<a name="l00175"></a>00175 
<a name="l00176"></a>00176   <span class="keywordflow">if</span> (! m_open_called) {
<a name="l00177"></a>00177     <a class="code" href="classMarkerExperimentReport.html#a5908d432cc5d9d39b995d157fc1a7689" title="MarkerExperimentReport::open Synopsis:">open</a>(rte);
<a name="l00178"></a>00178   }
<a name="l00179"></a>00179 
<a name="l00180"></a>00180 
<a name="l00181"></a>00181   <span class="comment">//std::sort(m_results.begin(), m_results.end());</span>
<a name="l00182"></a>00182   <span class="keywordtype">int</span> count = 0;
<a name="l00183"></a>00183 
<a name="l00184"></a>00184 
<a name="l00185"></a>00185   std::map&lt;std::string, ExperimentResults*&gt;::const_iterator itSER;
<a name="l00186"></a>00186 
<a name="l00187"></a>00187   <span class="comment">// For each experiment, recapitulate the translation table.</span>
<a name="l00188"></a>00188 
<a name="l00189"></a>00189   <span class="keywordflow">for</span> (itSER = er.begin(); (itSER != er.end()) &amp;&amp; okReport ;  itSER++) {
<a name="l00190"></a>00190 
<a name="l00191"></a>00191     <span class="comment">// IGNORE Experiments with no marker calls for DMET2.</span>
<a name="l00192"></a>00192     <span class="keywordflow">if</span> (rte.m_adtOpts.m_dmet2Calling &amp;&amp; (itSER-&gt;second-&gt;m_markerCallCount == 0)) {
<a name="l00193"></a>00193       <span class="keywordflow">continue</span>;
<a name="l00194"></a>00194     }
<a name="l00195"></a>00195 
<a name="l00196"></a>00196     std::vector&lt;MarkerRow&gt; mRows;
<a name="l00197"></a>00197     std::map&lt;std::string, int&gt; probeSetRow;
<a name="l00198"></a>00198 
<a name="l00199"></a>00199     <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 1; i &lt; ttm.size(); i++, count++) {
<a name="l00200"></a>00200 
<a name="l00201"></a>00201       <span class="comment">// Skip commented out rows as well as header rows.</span>
<a name="l00202"></a>00202       <span class="keywordflow">if</span> (ttm.<a class="code" href="classTranslationTableModel.html#a590375d07f5f5bf813275858d0466114" title="TranslationTableModel::ignoreRow Synopsis: A setter and getter for ignoring rows in the translation t...">ignoreRow</a>(i) || (!((ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_HAPLOTYPE)] == <span class="stringliteral">&quot;Y&quot;</span>) || (ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_HAPLOTYPE)] == <span class="stringliteral">&quot;N&quot;</span>)))) {
<a name="l00203"></a>00203         <span class="keywordflow">continue</span>;
<a name="l00204"></a>00204       }
<a name="l00205"></a>00205 
<a name="l00206"></a>00206 
<a name="l00207"></a>00207       std::string gene = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_GENE)];
<a name="l00208"></a>00208 
<a name="l00209"></a>00209       <a class="code" href="classExperimentGeneResults.html">ExperimentGeneResults</a> *egr = NULL;
<a name="l00210"></a>00210 
<a name="l00211"></a>00211       <span class="keywordflow">if</span> (itSER-&gt;second-&gt;m_geneResults.find(gene) != itSER-&gt;second-&gt;m_geneResults.end()) {
<a name="l00212"></a>00212         egr = itSER-&gt;second-&gt;m_geneResults[gene];
<a name="l00213"></a>00213       }
<a name="l00214"></a>00214 
<a name="l00215"></a>00215 
<a name="l00216"></a>00216       <a class="code" href="classMarkerRow.html">MarkerRow</a> mRowNew;
<a name="l00217"></a>00217       <a class="code" href="classMarkerRow.html">MarkerRow</a> * mRow = &amp;mRowNew;
<a name="l00218"></a>00218       std::string probeSet = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_PROBE_SET_ID)];
<a name="l00219"></a>00219 
<a name="l00220"></a>00220       <span class="keywordtype">bool</span> isMultiAllelicRow = probeSetRow.count(probeSet) &gt; 0;
<a name="l00221"></a>00221       <span class="keywordflow">if</span> (isMultiAllelicRow) {
<a name="l00222"></a>00222         <a class="code" href="Err_8h.html#ad024208f4ba6a4e75df4499ea1639a05" title="Calls Err::apt_err_assert with the filename and linenumber set. /// We want to avoid evaluating the m...">APT_ERR_ASSERT</a>(probeSetRow[probeSet] &lt; mRows.size(), <span class="stringliteral">&quot;&quot;</span>);
<a name="l00223"></a>00223         mRow = &amp;(mRows[probeSetRow[probeSet]]);
<a name="l00224"></a>00224       } <span class="keywordflow">else</span> {
<a name="l00225"></a>00225         probeSetRow[probeSet] = mRows.size();
<a name="l00226"></a>00226       }
<a name="l00227"></a>00227 
<a name="l00228"></a>00228       <span class="keywordtype">bool</span> isProbeSetPresent = <span class="keyword">false</span>;
<a name="l00229"></a>00229       <span class="keywordflow">if</span> (egr != NULL) {
<a name="l00230"></a>00230         isProbeSetPresent
<a name="l00231"></a>00231         = egr-&gt;m_probeSetToMarkerCallResults.count(probeSet) &gt; 0;
<a name="l00232"></a>00232       }
<a name="l00233"></a>00233 
<a name="l00234"></a>00234       <span class="keywordtype">int</span> markerCallResultsIndex = 0;
<a name="l00235"></a>00235       <span class="keywordtype">bool</span> hasAlleleCalls = <span class="keyword">false</span>;
<a name="l00236"></a>00236       <span class="keywordflow">if</span> (isProbeSetPresent) {
<a name="l00237"></a>00237         markerCallResultsIndex = egr-&gt;m_probeSetToMarkerCallResults[probeSet];
<a name="l00238"></a>00238         hasAlleleCalls =  egr-&gt;m_callResults[markerCallResultsIndex].m_alleleCalls.size() &gt; 0;
<a name="l00239"></a>00239       }
<a name="l00240"></a>00240 
<a name="l00241"></a>00241       <span class="comment">// EXPERIMENT</span>
<a name="l00242"></a>00242       mRow-&gt;m_experiment  = itSER-&gt;first;
<a name="l00243"></a>00243 
<a name="l00244"></a>00244       <span class="comment">// GENE</span>
<a name="l00245"></a>00245       mRow-&gt;m_gene       = gene;
<a name="l00246"></a>00246 
<a name="l00247"></a>00247       <span class="comment">// SAMPLE</span>
<a name="l00248"></a>00248       mRow-&gt;m_sample     = itSER-&gt;second-&gt;m_sample;
<a name="l00249"></a>00249 
<a name="l00250"></a>00250       <span class="comment">// FUNCTIONAL CHANGE</span>
<a name="l00251"></a>00251       mRow-&gt;m_functionalChange = <span class="stringliteral">&quot;&quot;</span>;
<a name="l00252"></a>00252 
<a name="l00253"></a>00253       <span class="comment">// EXTERNAL ID</span>
<a name="l00254"></a>00254       mRow-&gt;m_externalId = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_COMMON_NAME)];
<a name="l00255"></a>00255 
<a name="l00256"></a>00256       <span class="comment">// PROBE SET</span>
<a name="l00257"></a>00257       mRow-&gt;m_probeSet = probeSet;
<a name="l00258"></a>00258 
<a name="l00259"></a>00259       <span class="comment">// REFERENCE BASE</span>
<a name="l00260"></a>00260       mRow-&gt;m_referenceBase = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_REFERENCE)];
<a name="l00261"></a>00261       <span class="comment">// VARIANT BASE</span>
<a name="l00262"></a>00262       <span class="keywordflow">if</span> (isMultiAllelicRow) {
<a name="l00263"></a>00263         mRow-&gt;m_variantBase = mRow-&gt;m_variantBase + <span class="stringliteral">&quot;,&quot;</span> + ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_VARIANT)];
<a name="l00264"></a>00264       } <span class="keywordflow">else</span> {
<a name="l00265"></a>00265         mRow-&gt;m_variantBase = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_VARIANT)];
<a name="l00266"></a>00266       }
<a name="l00267"></a>00267 
<a name="l00268"></a>00268       <span class="comment">// CALL</span>
<a name="l00269"></a>00269       mRow-&gt;m_call = <span class="stringliteral">&quot;&quot;</span>;
<a name="l00270"></a>00270       <span class="keywordflow">if</span> (isProbeSetPresent &amp;&amp; (egr-&gt;m_callResults[markerCallResultsIndex].m_alleleCalls.size() &gt; 0)) {
<a name="l00271"></a>00271         <a class="code" href="Err_8h.html#ad024208f4ba6a4e75df4499ea1639a05" title="Calls Err::apt_err_assert with the filename and linenumber set. /// We want to avoid evaluating the m...">APT_ERR_ASSERT</a>(markerCallResultsIndex &lt; egr-&gt;m_callResults.size(), <span class="stringliteral">&quot;&quot;</span>);
<a name="l00272"></a>00272         <a class="code" href="Err_8h.html#ad024208f4ba6a4e75df4499ea1639a05" title="Calls Err::apt_err_assert with the filename and linenumber set. /// We want to avoid evaluating the m...">APT_ERR_ASSERT</a>(egr-&gt;m_callResults[markerCallResultsIndex].m_alleleCalls.size() &gt; 0, <span class="stringliteral">&quot;&quot;</span>);
<a name="l00273"></a>00273         mRow-&gt;m_call = egr-&gt;m_callResults[markerCallResultsIndex].m_alleleCalls[0].m_allele;
<a name="l00274"></a>00274       }
<a name="l00275"></a>00275 
<a name="l00276"></a>00276       <span class="comment">// BASE CALL</span>
<a name="l00277"></a>00277 
<a name="l00278"></a>00278       <span class="keywordflow">if</span> (!isMultiAllelicRow) {
<a name="l00279"></a>00279         mRow-&gt;m_baseCall = <span class="stringliteral">&quot;NotAvailable&quot;</span>;
<a name="l00280"></a>00280       }
<a name="l00281"></a>00281 
<a name="l00282"></a>00282       <span class="keywordflow">if</span> (isProbeSetPresent) {
<a name="l00283"></a>00283         <span class="keywordflow">if</span> ((egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_ceSet.size() &gt; 0) &amp;&amp; (egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_ceSet.size() &gt; 0) &amp;&amp; egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_ceSet.begin()-&gt;second.m_bases[0] == <span class="stringliteral">&quot;PRA&quot;</span>) {
<a name="l00284"></a>00284           mRow-&gt;m_baseCall = <span class="stringliteral">&quot;PossibleRareAllele&quot;</span>;
<a name="l00285"></a>00285         }
<a name="l00286"></a>00286 
<a name="l00287"></a>00287         <span class="keywordflow">if</span> (mRow-&gt;m_call == <span class="stringliteral">&quot;Ref/Ref&quot;</span>) {
<a name="l00288"></a>00288           mRow-&gt;m_baseCall = mRow-&gt;m_referenceBase + <span class="stringliteral">&quot;/&quot;</span> + mRow-&gt;m_referenceBase;
<a name="l00289"></a>00289         } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (mRow-&gt;m_call == <span class="stringliteral">&quot;Ref/Var&quot;</span>) {
<a name="l00290"></a>00290           std::string variance =  egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_ceSet.begin()-&gt;second.m_bases[0];
<a name="l00291"></a>00291           <span class="keywordflow">if</span> (variance ==  mRow-&gt;m_referenceBase) {
<a name="l00292"></a>00292             variance =  egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid2.m_ceSet.begin()-&gt;second.m_bases[0];
<a name="l00293"></a>00293           }
<a name="l00294"></a>00294           <span class="keywordflow">if</span> (variance == ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_VARIANT)]) {
<a name="l00295"></a>00295             mRow-&gt;m_baseCall = mRow-&gt;m_referenceBase + <span class="stringliteral">&quot;/&quot;</span> + variance;
<a name="l00296"></a>00296           }
<a name="l00297"></a>00297         } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (mRow-&gt;m_call == <span class="stringliteral">&quot;Var/Var&quot;</span>) {
<a name="l00298"></a>00298           std::string variance1 =  egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_ceSet.begin()-&gt;second.m_bases[0];
<a name="l00299"></a>00299           <span class="keywordflow">if</span> (variance1 == ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_VARIANT)]) {
<a name="l00300"></a>00300             std::string variance2 =  egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid2.m_ceSet.begin()-&gt;second.m_bases[0];
<a name="l00301"></a>00301             mRow-&gt;m_baseCall = variance1 + <span class="stringliteral">&quot;/&quot;</span> + variance2 ;
<a name="l00302"></a>00302           }
<a name="l00303"></a>00303         } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_ceSet.begin()-&gt;second.m_bases[0] == <span class="stringliteral">&quot;NC&quot;</span>) {
<a name="l00304"></a>00304           mRow-&gt;m_baseCall = <span class="stringliteral">&quot;NoCall&quot;</span>;
<a name="l00305"></a>00305         }
<a name="l00306"></a>00306       }
<a name="l00307"></a>00307 
<a name="l00308"></a>00308       <span class="comment">// HAPLOTYPE MARKER</span>
<a name="l00309"></a>00309       mRow-&gt;m_haplotypeMarker = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_HAPLOTYPE)];
<a name="l00310"></a>00310 
<a name="l00311"></a>00311       <span class="comment">// CHANGE FOR VARIANT</span>
<a name="l00312"></a>00312       <span class="keywordflow">if</span> (isMultiAllelicRow) {
<a name="l00313"></a>00313         <span class="keywordflow">if</span> (mRow-&gt;m_changeForVariant.empty()) {
<a name="l00314"></a>00314           mRow-&gt;m_changeForVariant = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_CHANGE)];
<a name="l00315"></a>00315         } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (! ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_CHANGE)].empty() &amp;&amp; (mRow-&gt;m_changeForVariant != ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_CHANGE)])) {
<a name="l00316"></a>00316 
<a name="l00317"></a>00317           mRow-&gt;m_changeForVariant = mRow-&gt;m_changeForVariant + <span class="stringliteral">&quot;,&quot;</span> + ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_CHANGE)];
<a name="l00318"></a>00318         }
<a name="l00319"></a>00319 
<a name="l00320"></a>00320       } <span class="keywordflow">else</span> {
<a name="l00321"></a>00321         mRow-&gt;m_changeForVariant = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_CHANGE)];
<a name="l00322"></a>00322       }
<a name="l00323"></a>00323 
<a name="l00324"></a>00324       <span class="comment">// VARIANT CDNA CHANGE</span>
<a name="l00325"></a>00325       <span class="keywordflow">if</span> (isMultiAllelicRow) {
<a name="l00326"></a>00326         <span class="keywordflow">if</span> (mRow-&gt;m_variantCDNAChange.empty()) {
<a name="l00327"></a>00327           mRow-&gt;m_variantCDNAChange = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_CDNA)];
<a name="l00328"></a>00328         } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (!ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_CDNA)].empty()) {
<a name="l00329"></a>00329           mRow-&gt;m_variantCDNAChange = mRow-&gt;m_variantCDNAChange + <span class="stringliteral">&quot;,&quot;</span> + ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_CDNA)];
<a name="l00330"></a>00330         }
<a name="l00331"></a>00331       } <span class="keywordflow">else</span>  {
<a name="l00332"></a>00332         mRow-&gt;m_variantCDNAChange = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_CDNA)];
<a name="l00333"></a>00333       }
<a name="l00334"></a>00334 
<a name="l00335"></a>00335       <span class="comment">// VARIANT DNA CHANGE</span>
<a name="l00336"></a>00336       <span class="keywordflow">if</span> (isMultiAllelicRow) {
<a name="l00337"></a>00337         <span class="keywordflow">if</span> (mRow-&gt;m_variantDNAChange.empty()) {
<a name="l00338"></a>00338           mRow-&gt;m_variantDNAChange = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_GENOME_POSITION)];
<a name="l00339"></a>00339         } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (!ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_GENOME_POSITION)].empty()) {
<a name="l00340"></a>00340           mRow-&gt;m_variantDNAChange = mRow-&gt;m_variantDNAChange + <span class="stringliteral">&quot;,&quot;</span> + ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_GENOME_POSITION)];
<a name="l00341"></a>00341         }
<a name="l00342"></a>00342       } <span class="keywordflow">else</span> {
<a name="l00343"></a>00343         mRow-&gt;m_variantDNAChange = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_GENOME_POSITION)];
<a name="l00344"></a>00344       }
<a name="l00345"></a>00345 
<a name="l00346"></a>00346       <span class="comment">// DBSNP ID</span>
<a name="l00347"></a>00347       mRow-&gt;m_dbSNPId = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_DBSNP)];
<a name="l00348"></a>00348 
<a name="l00349"></a>00349       <span class="comment">// VALIDATED</span>
<a name="l00350"></a>00350       mRow-&gt;m_validated = ttm.m_rows[i][ADT_DMET2_TT_VALIDATED];
<a name="l00351"></a>00351       <span class="comment">// ALLELE DEFINING MARKER</span>
<a name="l00352"></a>00352       <span class="keywordflow">if</span> (isMultiAllelicRow) {
<a name="l00353"></a>00353         <span class="keywordflow">if</span> (mRow-&gt;m_alleleDefiningMarker.empty()) {
<a name="l00354"></a>00354           mRow-&gt;m_alleleDefiningMarker = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_DEFINING)];
<a name="l00355"></a>00355         } <span class="keywordflow">else</span>  <span class="keywordflow">if</span> (!ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_DEFINING)].empty() &amp;&amp; (mRow-&gt;m_alleleDefiningMarker != ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_DEFINING)])) {
<a name="l00356"></a>00356           mRow-&gt;m_alleleDefiningMarker = mRow-&gt;m_alleleDefiningMarker + <span class="stringliteral">&quot;,&quot;</span> + ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_DEFINING)];
<a name="l00357"></a>00357         }
<a name="l00358"></a>00358       } <span class="keywordflow">else</span> {
<a name="l00359"></a>00359         mRow-&gt;m_alleleDefiningMarker = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_DEFINING)];
<a name="l00360"></a>00360       }
<a name="l00361"></a>00361 
<a name="l00362"></a>00362       <span class="comment">// RELEVANT ALLELES</span>
<a name="l00363"></a>00363       <span class="keywordtype">int</span> headerRow = 0;
<a name="l00364"></a>00364 
<a name="l00365"></a>00365       <span class="keywordflow">for</span> (headerRow = i - 1; headerRow &gt;= 0; headerRow--) {
<a name="l00366"></a>00366         <span class="keywordflow">if</span> (headerRow == 0) <span class="keywordflow">break</span>;
<a name="l00367"></a>00367         <span class="keywordflow">if</span> (ttm.m_rows[headerRow][0] != mRow-&gt;m_gene)  {
<a name="l00368"></a>00368           headerRow++;
<a name="l00369"></a>00369           <span class="keywordflow">break</span>;
<a name="l00370"></a>00370         }
<a name="l00371"></a>00371       }
<a name="l00372"></a>00372 
<a name="l00373"></a>00373       std::string relevantAlleles;
<a name="l00374"></a>00374       <span class="keywordtype">int</span> relevantAllelesIndex = ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_VARIANT);
<a name="l00375"></a>00375       <span class="keywordflow">for</span> (<span class="keywordtype">int</span> j = ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_ALLELE_START) + 1; j &lt; ttm.m_rows[i].size(); j++) {
<a name="l00376"></a>00376         <span class="keywordflow">if</span> ((mRow-&gt;m_haplotypeMarker == <span class="stringliteral">&quot;N&quot;</span>) || (ttm.m_rows[headerRow][j].c_str())[0] != <span class="charliteral">&#39;#&#39;</span>) {
<a name="l00377"></a>00377           <span class="keywordflow">if</span> (ttm.m_rows[i][j].empty()) {
<a name="l00378"></a>00378             <span class="keywordflow">if</span> (j == ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_ALLELE_START))  {
<a name="l00379"></a>00379               relevantAlleles = ttm.m_rows[headerRow][j] + <span class="stringliteral">&quot;,&quot;</span>;
<a name="l00380"></a>00380             }
<a name="l00381"></a>00381           } <span class="keywordflow">else</span>   {
<a name="l00382"></a>00382             relevantAllelesIndex = j;
<a name="l00383"></a>00383             relevantAlleles = relevantAlleles + ttm.m_rows[headerRow][j] + <span class="stringliteral">&quot;,&quot;</span>;
<a name="l00384"></a>00384           }
<a name="l00385"></a>00385         }
<a name="l00386"></a>00386       }
<a name="l00387"></a>00387       <span class="keywordflow">if</span> (!relevantAlleles.empty()) {
<a name="l00388"></a>00388         relevantAlleles.resize(relevantAlleles.size() - 1);
<a name="l00389"></a>00389       }
<a name="l00390"></a>00390       <span class="keywordflow">if</span> (isMultiAllelicRow) {
<a name="l00391"></a>00391         <span class="keywordflow">if</span> (! relevantAlleles.empty()) {
<a name="l00392"></a>00392           mRow-&gt;m_relevantAlleles = mRow-&gt;m_relevantAlleles  + <span class="stringliteral">&quot;,&quot;</span> + relevantAlleles;
<a name="l00393"></a>00393         }
<a name="l00394"></a>00394       } <span class="keywordflow">else</span> {
<a name="l00395"></a>00395         mRow-&gt;m_relevantAlleles = relevantAlleles;
<a name="l00396"></a>00396       }
<a name="l00397"></a>00397 
<a name="l00398"></a>00398       <span class="keywordflow">if</span> (isProbeSetPresent &amp;&amp; !relevantAlleles.empty() &amp;&amp; (mRow-&gt;m_haplotypeMarker == <span class="stringliteral">&quot;N&quot;</span>)  &amp;&amp; (ttm.m_rows[i][relevantAllelesIndex] ==  ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_VARIANT)])) {
<a name="l00399"></a>00399 
<a name="l00400"></a>00400         std::string variance1 =  egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_ceSet.begin()-&gt;second.m_bases[0];
<a name="l00401"></a>00401         std::string variance2 =  egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid2.m_ceSet.begin()-&gt;second.m_bases[0];
<a name="l00402"></a>00402         <span class="keywordflow">if</span> ((mRow-&gt;m_call == <span class="stringliteral">&quot;Var/Var&quot;</span>)) {
<a name="l00403"></a>00403 
<a name="l00404"></a>00404           <span class="keywordflow">if</span> (mRow-&gt;m_functionalChange.empty()) {
<a name="l00405"></a>00405             <span class="keywordflow">if</span> (mRow-&gt;m_relevantAlleles.find(<span class="stringliteral">&quot;,&quot;</span>) != std::string::npos) {
<a name="l00406"></a>00406               std::string firstAllele = mRow-&gt;m_relevantAlleles.substr(0, mRow-&gt;m_relevantAlleles.find(<span class="stringliteral">&quot;,&quot;</span>));
<a name="l00407"></a>00407               mRow-&gt;m_functionalChange = firstAllele + <span class="stringliteral">&quot;/&quot;</span> + firstAllele;
<a name="l00408"></a>00408             } <span class="keywordflow">else</span> {
<a name="l00409"></a>00409               mRow-&gt;m_functionalChange = mRow-&gt;m_relevantAlleles  + <span class="stringliteral">&quot;/&quot;</span> + mRow-&gt;m_relevantAlleles;
<a name="l00410"></a>00410             }
<a name="l00411"></a>00411           } <span class="keywordflow">else</span> {
<a name="l00412"></a>00412             mRow-&gt;m_functionalChange = mRow-&gt;m_functionalChange.substr(0, mRow-&gt;m_functionalChange.find(<span class="stringliteral">&quot;/&quot;</span>))  + <span class="stringliteral">&quot;/&quot;</span> + relevantAlleles;
<a name="l00413"></a>00413           }
<a name="l00414"></a>00414           <span class="keywordflow">if</span> ((variance1 != variance2) &amp;&amp;
<a name="l00415"></a>00415               (mRow-&gt;m_relevantAlleles.find(<span class="stringliteral">&quot;,&quot;</span>) != std::string::npos)) {
<a name="l00416"></a>00416             mRow-&gt;m_functionalChange = mRow-&gt;m_relevantAlleles;
<a name="l00417"></a>00417           }
<a name="l00418"></a>00418         } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (mRow-&gt;m_call == <span class="stringliteral">&quot;Ref/Var&quot;</span>) {
<a name="l00419"></a>00419           mRow-&gt;m_functionalChange = mRow-&gt;m_relevantAlleles;
<a name="l00420"></a>00420           <span class="keywordflow">if</span> (mRow-&gt;m_relevantAlleles.find(<span class="stringliteral">&quot;,&quot;</span>) != std::string::npos) {
<a name="l00421"></a>00421             mRow-&gt;m_functionalChange =  mRow-&gt;m_relevantAlleles.substr(0 , mRow-&gt;m_relevantAlleles.find(<span class="stringliteral">&quot;,&quot;</span>));
<a name="l00422"></a>00422           }
<a name="l00423"></a>00423 
<a name="l00424"></a>00424         }
<a name="l00425"></a>00425       }
<a name="l00426"></a>00426 
<a name="l00427"></a>00427       <span class="keywordflow">if</span> (!isMultiAllelicRow) {
<a name="l00428"></a>00428         mRows.push_back(*mRow);
<a name="l00429"></a>00429       }
<a name="l00430"></a>00430 
<a name="l00431"></a>00431 
<a name="l00432"></a>00432     } <span class="comment">//for each translation table model marker</span>
<a name="l00433"></a>00433 
<a name="l00434"></a>00434     <span class="keywordflow">for</span> (<span class="keywordtype">size_t</span> j = 0; j &lt; mRows.size(); j++) {
<a name="l00435"></a>00435       _report_marker(m_marker_tsv, mRows[j]);
<a name="l00436"></a>00436     }
<a name="l00437"></a>00437 
<a name="l00438"></a>00438   } <span class="comment">// for each gene, experiment result</span>
<a name="l00439"></a>00439 
<a name="l00440"></a>00440   <span class="keywordflow">return</span> okReport;
<a name="l00441"></a>00441 
<a name="l00442"></a>00442 }
<a name="l00443"></a>00443 <span class="comment">// end MarkerExperimentReport::generate</span>
<a name="l00444"></a>00444 <span class="comment">/*****************************************************************************/</span>
<a name="l00445"></a>00445 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00446"></a>00446 <span class="comment">/**</span>
<a name="l00447"></a>00447 <span class="comment"> * MarkerExperimentReport::getMarkerReportName</span>
<a name="l00448"></a>00448 <span class="comment"> * Synopsis:</span>
<a name="l00449"></a>00449 <span class="comment"> *</span>
<a name="l00450"></a>00450 <span class="comment"> * Create the Marker file name from various constant and runtime</span>
<a name="l00451"></a>00451 <span class="comment"> * environment settings.</span>
<a name="l00452"></a>00452 <span class="comment"> *</span>
<a name="l00453"></a>00453 <span class="comment"> *</span>
<a name="l00454"></a>00454 <span class="comment"> * @param rte - the single instance RunTimeEnvironment which contains options</span>
<a name="l00455"></a>00455 <span class="comment"> *</span>
<a name="l00456"></a>00456 <span class="comment"> * @return fileName - the name of the marker report file to use.</span>
<a name="l00457"></a>00457 <span class="comment"> *</span>
<a name="l00458"></a>00458 <span class="comment"> */</span>
<a name="l00459"></a>00459 <span class="comment">/*****************************************************************************/</span>
<a name="l00460"></a><a class="code" href="classMarkerExperimentReport.html#af330c283c562d19d2b96700c701abe51">00460</a> std::string <a class="code" href="classMarkerExperimentReport.html#af330c283c562d19d2b96700c701abe51" title="MarkerExperimentReport::getMarkerReportName Synopsis:">MarkerExperimentReport::getMarkerReportName</a>(<span class="keyword">const</span> <span class="keyword">class</span> <a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp; rte)
<a name="l00461"></a>00461 {
<a name="l00462"></a>00462 
<a name="l00463"></a>00463   std::string markerFileName;
<a name="l00464"></a>00464   <span class="keywordflow">if</span> (rte.m_adtOpts.m_inputGenoFile != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00465"></a>00465     markerFileName = <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(rte.m_adtOpts.m_inputGenoFile);
<a name="l00466"></a>00466     markerFileName = markerFileName.substr(0, markerFileName.find(<span class="charliteral">&#39;.&#39;</span>));
<a name="l00467"></a>00467   } <span class="keywordflow">else</span> {
<a name="l00468"></a>00468     markerFileName = name();
<a name="l00469"></a>00469   }
<a name="l00470"></a>00470 
<a name="l00471"></a>00471 
<a name="l00472"></a>00472   markerFileName = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(rte.m_adtOpts.m_outputDir,
<a name="l00473"></a>00473                             markerFileName+<span class="stringliteral">&quot;.dmet3_&quot;</span>+ADT_VERSION+<span class="stringliteral">&quot;.&quot;</span>+MARKER_FILE_EXT);
<a name="l00474"></a>00474 
<a name="l00475"></a>00475   m_reportName = markerFileName;
<a name="l00476"></a>00476 
<a name="l00477"></a>00477   <span class="keywordflow">return</span> markerFileName;
<a name="l00478"></a>00478 
<a name="l00479"></a>00479 }
<a name="l00480"></a>00480 <span class="comment">// end MarkerExperimentReport::getMarkerReportName</span>
<a name="l00481"></a>00481 <span class="comment">/*****************************************************************************/</span>
<a name="l00482"></a>00482 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00483"></a>00483 <span class="comment">/**</span>
<a name="l00484"></a>00484 <span class="comment"> * MarkerExperimentReport::open</span>
<a name="l00485"></a>00485 <span class="comment"> * Synopsis:</span>
<a name="l00486"></a>00486 <span class="comment"> *</span>
<a name="l00487"></a>00487 <span class="comment"> * Open the TSV file for output.</span>
<a name="l00488"></a>00488 <span class="comment"> *</span>
<a name="l00489"></a>00489 <span class="comment"> *</span>
<a name="l00490"></a>00490 <span class="comment"> * @param rte - the single instance run time environment.</span>
<a name="l00491"></a>00491 <span class="comment"> */</span>
<a name="l00492"></a>00492 <span class="comment">/*****************************************************************************/</span>
<a name="l00493"></a><a class="code" href="classMarkerExperimentReport.html#a5908d432cc5d9d39b995d157fc1a7689">00493</a> <span class="keywordtype">void</span> <a class="code" href="classMarkerExperimentReport.html#a5908d432cc5d9d39b995d157fc1a7689" title="MarkerExperimentReport::open Synopsis:">MarkerExperimentReport::open</a>(<span class="keyword">const</span> <span class="keyword">class</span> <a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp; rte)
<a name="l00494"></a>00494 {
<a name="l00495"></a>00495 
<a name="l00496"></a>00496   m_guid = <a class="code" href="classaffxutil_1_1Guid.html#a2929c982cb9f3b5f6c5d7d818a8e6ac4">affxutil::Guid::GenerateNewGuid</a>();
<a name="l00497"></a>00497 
<a name="l00498"></a>00498   <span class="comment">// describe our columns.</span>
<a name="l00499"></a>00499   <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; NUM_MARKER_TSV_COLUMNS; i++) {
<a name="l00500"></a>00500     m_marker_tsv.<a class="code" href="classaffx_1_1TsvFile.html#aadc22863a63b9c6588b0f2dc61875809" title="User methods.">defineColumn</a>(0, i, MARKER_TSV_COLUMNS[i]);
<a name="l00501"></a>00501   }
<a name="l00502"></a>00502 
<a name="l00503"></a>00503   m_marker_tsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot; For research use only. Not for diagnostic purposes.&quot;</span>);
<a name="l00504"></a>00504   m_marker_tsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;Version&quot;</span>, ADT_VERSION);
<a name="l00505"></a>00505   m_marker_tsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;Date&quot;</span>, <a class="code" href="classUtil.html#accf4f30492f3726aac3197a37b3f77cc" title="Get a timestamp string.">Util::getTimeStamp</a>());
<a name="l00506"></a>00506   m_marker_tsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;CopyNumberFile&quot;</span>, rte.m_adtOpts.m_inputCopyFile);
<a name="l00507"></a>00507   m_marker_tsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;TranslatTableFile&quot;</span>, rte.m_adtOpts.m_inputTTableFile);
<a name="l00508"></a>00508   m_marker_tsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;DMET3File&quot;</span>, rte.m_adtOpts.m_inputGenoFile);
<a name="l00509"></a>00509   m_marker_tsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;dmet3-report-guid&quot;</span>, m_guid);
<a name="l00510"></a>00510 
<a name="l00511"></a>00511   std::string markerName = <a class="code" href="classMarkerExperimentReport.html#af330c283c562d19d2b96700c701abe51" title="MarkerExperimentReport::getMarkerReportName Synopsis:">getMarkerReportName</a>(rte);
<a name="l00512"></a>00512 
<a name="l00513"></a>00513   m_marker_tsv.<a class="code" href="classaffx_1_1TsvFile.html#a1903489904b4e2f8f2a262b8f10cdb0e" title="Set the options up for &amp;quot;tsv&amp;quot; format and write the tsv-v1 header ///.">writeTsv_v1</a>(markerName);
<a name="l00514"></a>00514   m_open_called = <span class="keyword">true</span>;
<a name="l00515"></a>00515   <span class="keywordflow">return</span>;
<a name="l00516"></a>00516 
<a name="l00517"></a>00517 }
<a name="l00518"></a>00518 <span class="comment">// end MarkerExperimentReport::open</span>
<a name="l00519"></a>00519 <span class="comment">/*****************************************************************************/</span>
<a name="l00520"></a>00520 
</pre></div></div>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:48 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
