<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: midas/MidasEngine.cpp Source File</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li class="current"><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="tabs2">
    <ul class="tablist">
      <li><a href="files.html"><span>File&nbsp;List</span></a></li>
      <li><a href="globals.html"><span>File&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath">
    <ul>
      <li><a class="el" href="dir_e31b933a9bcc1adcf2933f070802d441.html">midas</a>      </li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>MidasEngine.cpp</h1>  </div>
</div>
<div class="contents">
<a href="MidasEngine_8cpp.html">Go to the documentation of this file.</a><div class="fragment"><pre class="fragment"><a name="l00001"></a>00001 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00002"></a>00002 <span class="comment"></span><span class="comment">//</span>
<a name="l00003"></a>00003 <span class="comment">// Copyright (C) 2005 Affymetrix, Inc.</span>
<a name="l00004"></a>00004 <span class="comment">//</span>
<a name="l00005"></a>00005 <span class="comment">// This program is free software; you can redistribute it and/or modify </span>
<a name="l00006"></a>00006 <span class="comment">// it under the terms of the GNU General Public License (version 2) as </span>
<a name="l00007"></a>00007 <span class="comment">// published by the Free Software Foundation.</span>
<a name="l00008"></a>00008 <span class="comment">// </span>
<a name="l00009"></a>00009 <span class="comment">// This program is distributed in the hope that it will be useful, </span>
<a name="l00010"></a>00010 <span class="comment">// but WITHOUT ANY WARRANTY; without even the implied warranty of </span>
<a name="l00011"></a>00011 <span class="comment">// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU </span>
<a name="l00012"></a>00012 <span class="comment">// General Public License for more details.</span>
<a name="l00013"></a>00013 <span class="comment">// </span>
<a name="l00014"></a>00014 <span class="comment">// You should have received a copy of the GNU General Public License </span>
<a name="l00015"></a>00015 <span class="comment">// along with this program;if not, write to the </span>
<a name="l00016"></a>00016 <span class="comment">// </span>
<a name="l00017"></a>00017 <span class="comment">// Free Software Foundation, Inc., </span>
<a name="l00018"></a>00018 <span class="comment">// 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA</span>
<a name="l00019"></a>00019 <span class="comment">//</span><span class="comment"></span>
<a name="l00020"></a>00020 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00021"></a>00021 <span class="comment"></span><span class="comment"></span>
<a name="l00022"></a>00022 <span class="comment">/// @file   MidasEngine.cpp</span>
<a name="l00023"></a>00023 <span class="comment">/// @brief  Iterator which reads input data files, calls</span>
<a name="l00024"></a>00024 <span class="comment">///         the MidasSpliceDetector for each probeset_id (gene) and</span>
<a name="l00025"></a>00025 <span class="comment">///         probeset_list_id (exon) data pair, writes output to file.</span>
<a name="l00026"></a>00026 <span class="comment"></span>
<a name="l00027"></a>00027 <span class="comment">//</span>
<a name="l00028"></a>00028 <span class="preprocessor">#include &quot;<a class="code" href="MidasEngine_8h.html" title="///">midas/MidasEngine.h</a>&quot;</span>
<a name="l00029"></a>00029 <span class="comment">//</span>
<a name="l00030"></a>00030 <span class="preprocessor">#include &quot;<a class="code" href="TsvFile_8h.html" title="Headers for the TsvFile classes.">file/TsvFile/TsvFile.h</a>&quot;</span>
<a name="l00031"></a>00031 <span class="preprocessor">#include &quot;<a class="code" href="affy-base-types_8h.html">portability/affy-base-types.h</a>&quot;</span>
<a name="l00032"></a>00032 <span class="preprocessor">#include &quot;<a class="code" href="Convert_8h.html" title="Utility functions for converting between types.">util/Convert.h</a>&quot;</span>
<a name="l00033"></a>00033 <span class="preprocessor">#include &quot;<a class="code" href="PgOptions_8h.html" title="Classes for reading program command line options.">util/PgOptions.h</a>&quot;</span>
<a name="l00034"></a>00034 <span class="comment">//</span>
<a name="l00035"></a>00035 <span class="preprocessor">#include &lt;cerrno&gt;</span>
<a name="l00036"></a>00036 <span class="preprocessor">#include &lt;iostream&gt;</span>
<a name="l00037"></a>00037 <span class="preprocessor">#include &lt;sstream&gt;</span>
<a name="l00038"></a>00038 <span class="preprocessor">#include &lt;stdexcept&gt;</span>
<a name="l00039"></a>00039 <span class="comment">//</span>
<a name="l00040"></a>00040 
<a name="l00041"></a>00041 <span class="keyword">using namespace </span>std;
<a name="l00042"></a>00042 <span class="keyword">using namespace </span>affx;
<a name="l00043"></a>00043 
<a name="l00044"></a><a class="code" href="MidasEngine_8h.html#a2735e2a39c4a48617b8dd1685dbf4294">00044</a> <span class="keywordtype">void</span> <a class="code" href="MidasEngine_8cpp.html#a2735e2a39c4a48617b8dd1685dbf4294" title="define the midas options.">define_midas_opts</a>(<a class="code" href="classPgOptions.html">PgOptions</a>* opts)
<a name="l00045"></a>00045 {
<a name="l00046"></a>00046   opts-&gt;<a class="code" href="classPgOptions.html#aa40409c034ecf67ad32694103782e97e" title="set the usage info for this program ///">setUsage</a>(<span class="stringliteral">&quot;midas - Microarray Detection of Alternative Splicing.\n&quot;</span>
<a name="l00047"></a>00047                  <span class="stringliteral">&quot;Usage:\n&quot;</span>
<a name="l00048"></a>00048                  <span class="stringliteral">&quot;   midas --cel-files cels.txt -g gene.summary.txt -e exon.summary.txt -m metaprobeset.txt \\\n&quot;</span>
<a name="l00049"></a>00049                  <span class="stringliteral">&quot;            -o out -pvalues -fstats -normalized -stabilize 8.0 -no-log-transform&quot;</span>);
<a name="l00050"></a>00050   
<a name="l00051"></a>00051   opts-&gt;<a class="code" href="classPgOptions.html#a7b3fbfff8c4496985f80bcfd49a59d61" title="define an option to be handled by this option parser. ///">defineOption</a>(<span class="stringliteral">&quot;h&quot;</span>, <span class="stringliteral">&quot;help&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00052"></a>00052                      <span class="stringliteral">&quot;&quot;</span>,
<a name="l00053"></a>00053                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00054"></a>00054   opts-&gt;<a class="code" href="classPgOptions.html#a7b3fbfff8c4496985f80bcfd49a59d61" title="define an option to be handled by this option parser. ///">defineOption</a>(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;cel-files&quot;</span>, PgOpt::STRING_OPT,
<a name="l00055"></a>00055                      <span class="stringliteral">&quot;File defining mapping between cel_files and experimental group_id.&quot;</span>,
<a name="l00056"></a>00056                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00057"></a>00057   opts-&gt;<a class="code" href="classPgOptions.html#a7b3fbfff8c4496985f80bcfd49a59d61" title="define an option to be handled by this option parser. ///">defineOption</a>(<span class="stringliteral">&quot;g&quot;</span>, <span class="stringliteral">&quot;genedata&quot;</span>, PgOpt::STRING_OPT,
<a name="l00058"></a>00058                      <span class="stringliteral">&quot;File containing gene-level summary data.&quot;</span>,
<a name="l00059"></a>00059                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00060"></a>00060   opts-&gt;<a class="code" href="classPgOptions.html#a7b3fbfff8c4496985f80bcfd49a59d61" title="define an option to be handled by this option parser. ///">defineOption</a>(<span class="stringliteral">&quot;e&quot;</span>, <span class="stringliteral">&quot;exondata&quot;</span>, PgOpt::STRING_OPT,
<a name="l00061"></a>00061                      <span class="stringliteral">&quot;File containing exon-level summary data.&quot;</span>,
<a name="l00062"></a>00062                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00063"></a>00063   opts-&gt;<a class="code" href="classPgOptions.html#a7b3fbfff8c4496985f80bcfd49a59d61" title="define an option to be handled by this option parser. ///">defineOption</a>(<span class="stringliteral">&quot;m&quot;</span>, <span class="stringliteral">&quot;metaprobeset&quot;</span>, PgOpt::STRING_OPT,
<a name="l00064"></a>00064                      <span class="stringliteral">&quot;File containing mapping (meta-probeset file) between exon probesets and gene level probesets.&quot;</span>,
<a name="l00065"></a>00065                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00066"></a>00066   opts-&gt;<a class="code" href="classPgOptions.html#a7b3fbfff8c4496985f80bcfd49a59d61" title="define an option to be handled by this option parser. ///">defineOption</a>(<span class="stringliteral">&quot;o&quot;</span>, <span class="stringliteral">&quot;out-dir&quot;</span>, PgOpt::STRING_OPT,
<a name="l00067"></a>00067                      <span class="stringliteral">&quot;Directory which will contain the midas output.&quot;</span>,
<a name="l00068"></a>00068                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00069"></a>00069   opts-&gt;<a class="code" href="classPgOptions.html#a7b3fbfff8c4496985f80bcfd49a59d61" title="define an option to be handled by this option parser. ///">defineOption</a>(<span class="stringliteral">&quot;pv&quot;</span>, <span class="stringliteral">&quot;pvalues&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00070"></a>00070                      <span class="stringliteral">&quot;Output p values.&quot;</span>,
<a name="l00071"></a>00071                      <span class="stringliteral">&quot;true&quot;</span>);
<a name="l00072"></a>00072   opts-&gt;<a class="code" href="classPgOptions.html#a7b3fbfff8c4496985f80bcfd49a59d61" title="define an option to be handled by this option parser. ///">defineOption</a>(<span class="stringliteral">&quot;f&quot;</span>, <span class="stringliteral">&quot;fstats&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00073"></a>00073                      <span class="stringliteral">&quot;Output F statistics.&quot;</span>,
<a name="l00074"></a>00074                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00075"></a>00075   opts-&gt;<a class="code" href="classPgOptions.html#a7b3fbfff8c4496985f80bcfd49a59d61" title="define an option to be handled by this option parser. ///">defineOption</a>(<span class="stringliteral">&quot;n&quot;</span>, <span class="stringliteral">&quot;normalized&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00076"></a>00076                      <span class="stringliteral">&quot;Output normalized exon intensities.&quot;</span>,
<a name="l00077"></a>00077                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00078"></a>00078   opts-&gt;<a class="code" href="classPgOptions.html#a7b3fbfff8c4496985f80bcfd49a59d61" title="define an option to be handled by this option parser. ///">defineOption</a>(<span class="stringliteral">&quot;s&quot;</span>, <span class="stringliteral">&quot;stabilize&quot;</span>, PgOpt::DOUBLE_OPT,
<a name="l00079"></a>00079                      <span class="stringliteral">&quot;Stabilization factor for log data transforms.&quot;</span>,
<a name="l00080"></a>00080                      <span class="stringliteral">&quot;8.0&quot;</span>);
<a name="l00081"></a>00081   opts-&gt;<a class="code" href="classPgOptions.html#a7b3fbfff8c4496985f80bcfd49a59d61" title="define an option to be handled by this option parser. ///">defineOption</a>(<span class="stringliteral">&quot;nol&quot;</span>, <span class="stringliteral">&quot;no-logtrans&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00082"></a>00082                      <span class="stringliteral">&quot;Do not log transform gene and exon level summary data.&quot;</span>,
<a name="l00083"></a>00083                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00084"></a>00084   opts-&gt;<a class="code" href="classPgOptions.html#a7b3fbfff8c4496985f80bcfd49a59d61" title="define an option to be handled by this option parser. ///">defineOption</a>(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;version&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00085"></a>00085                      <span class="stringliteral">&quot;Display version information.&quot;</span>,
<a name="l00086"></a>00086                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00087"></a>00087   opts-&gt;<a class="code" href="classPgOptions.html#a7b3fbfff8c4496985f80bcfd49a59d61" title="define an option to be handled by this option parser. ///">defineOption</a>(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;keep-path&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00088"></a>00088                      <span class="stringliteral">&quot;Keep cel file path.&quot;</span>,
<a name="l00089"></a>00089                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00090"></a>00090 }
<a name="l00091"></a>00091 <span class="comment"></span>
<a name="l00092"></a>00092 <span class="comment">/**</span>
<a name="l00093"></a>00093 <span class="comment"> *  \brief Constructor.</span>
<a name="l00094"></a>00094 <span class="comment"> *</span>
<a name="l00095"></a>00095 <span class="comment"> *  Input data files have already been opened by the caller -</span>
<a name="l00096"></a>00096 <span class="comment"> *  here we copy parameters, perform minor size and</span>
<a name="l00097"></a>00097 <span class="comment"> *  option checks.  Output file headers have also already</span>
<a name="l00098"></a>00098 <span class="comment"> *  been written - here we open to append.</span>
<a name="l00099"></a>00099 <span class="comment"> *</span>
<a name="l00100"></a>00100 <span class="comment"> *  @param celFiles List of cel_file names.</span>
<a name="l00101"></a>00101 <span class="comment"> *  @param groups List of group indices.</span>
<a name="l00102"></a>00102 <span class="comment"> *  @param metaFileName Meta probeset file name.</span>
<a name="l00103"></a>00103 <span class="comment"> *  @param geneDataFileName Gene data file name.</span>
<a name="l00104"></a>00104 <span class="comment"> *  @param exonDataFileName Exon data file name.</span>
<a name="l00105"></a>00105 <span class="comment"> *  @param pvaluesFileName Output file name for p values.</span>
<a name="l00106"></a>00106 <span class="comment"> *  @param fstatsFileName Output file name for F statistics.</span>
<a name="l00107"></a>00107 <span class="comment"> *  @param normalizedFileName Output file name for normalized exon signal.</span>
<a name="l00108"></a>00108 <span class="comment"> *  @param wantPvalues Caller requests p values.</span>
<a name="l00109"></a>00109 <span class="comment"> *  @param wantFstats Caller requests F statistics.</span>
<a name="l00110"></a>00110 <span class="comment"> *  @param wantNormalized Caller requests normalized exon signal</span>
<a name="l00111"></a>00111 <span class="comment"> *  @param logStabilize Amount to add to data before taking logarithm.</span>
<a name="l00112"></a>00112 <span class="comment"> *  @param noLogTransform Do not log transform data.</span>
<a name="l00113"></a>00113 <span class="comment"> *</span>
<a name="l00114"></a>00114 <span class="comment"> *  Errors: exception if parameter checks fail</span>
<a name="l00115"></a>00115 <span class="comment"> */</span>
<a name="l00116"></a><a class="code" href="classmidasEngine.html#a80271f4f94a477e9d4a0d67d068e4619">00116</a> <a class="code" href="classmidasEngine.html#a80271f4f94a477e9d4a0d67d068e4619" title="Constructor.">midasEngine::midasEngine</a>(<span class="keyword">const</span> std::vector&lt;std::string&gt;&amp; celFiles,
<a name="l00117"></a>00117                          <span class="keyword">const</span> std::vector&lt;int&gt;&amp; groups,
<a name="l00118"></a>00118                          std::string metaFileName,
<a name="l00119"></a>00119                          std::string geneDataFileName,
<a name="l00120"></a>00120                          std::string exonDataFileName,
<a name="l00121"></a>00121                          std::string pvaluesFileName,
<a name="l00122"></a>00122                          std::string fstatsFileName,
<a name="l00123"></a>00123                          std::string normalizedFileName,
<a name="l00124"></a>00124                          <span class="keywordtype">int</span> wantPvalues, 
<a name="l00125"></a>00125                          <span class="keywordtype">int</span> wantFstats,
<a name="l00126"></a>00126                          <span class="keywordtype">int</span> wantNormalized,
<a name="l00127"></a>00127                          <span class="keyword">const</span> <span class="keywordtype">float</span>&amp; logStabilize,
<a name="l00128"></a>00128                          <span class="keywordtype">int</span> noLogTransform)
<a name="l00129"></a>00129     : celFiles (celFiles), groups (groups), metaFileName (metaFileName),
<a name="l00130"></a>00130       geneDataFileName (geneDataFileName), exonDataFileName (exonDataFileName),
<a name="l00131"></a>00131       pvaluesFileName (pvaluesFileName), fstatsFileName (fstatsFileName),
<a name="l00132"></a>00132       normalizedFileName (normalizedFileName), wantPvalues (wantPvalues), wantFstats (wantFstats),
<a name="l00133"></a>00133       wantNormalized (wantNormalized), logStabilize (logStabilize), noLogTransform (noLogTransform),
<a name="l00134"></a>00134       spliceDetector (groups, wantPvalues, wantFstats, wantNormalized, logStabilize, noLogTransform),
<a name="l00135"></a>00135       numGroups (groups.size())
<a name="l00136"></a>00136 {
<a name="l00137"></a>00137   <span class="comment">// require at least one output option</span>
<a name="l00138"></a>00138   <span class="keywordflow">if</span> (! wantPvalues &amp;&amp; ! wantFstats &amp;&amp; ! wantNormalized)
<a name="l00139"></a>00139     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;At least one output option must be selected&quot;</span>);
<a name="l00140"></a>00140 
<a name="l00141"></a>00141   <span class="comment">// celFile list, group list must be of the same size</span>
<a name="l00142"></a>00142   <span class="keywordflow">if</span> (celFiles.size() != numGroups)
<a name="l00143"></a>00143     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Mismatch between size of cel_file names, group ids&quot;</span>);
<a name="l00144"></a>00144 }
<a name="l00145"></a>00145 <span class="comment"></span>
<a name="l00146"></a>00146 <span class="comment">/**</span>
<a name="l00147"></a>00147 <span class="comment"> *  \brief Destructor.</span>
<a name="l00148"></a>00148 <span class="comment"> *</span>
<a name="l00149"></a>00149 <span class="comment"> *  Delete each vector&lt;int&gt; in the map between probesetListId</span>
<a name="l00150"></a>00150 <span class="comment"> *  and its vector of probesetIds.</span>
<a name="l00151"></a>00151 <span class="comment">*/</span>
<a name="l00152"></a><a class="code" href="classmidasEngine.html#afc8c2780e9873ce920c1d858af135191">00152</a> <a class="code" href="classmidasEngine.html#afc8c2780e9873ce920c1d858af135191" title="Destructor.">midasEngine::~midasEngine</a>()
<a name="l00153"></a>00153 {
<a name="l00154"></a>00154   <span class="keyword">typedef</span> map&lt;int, vector&lt;int&gt;* &gt;::iterator mapVecIter_t;
<a name="l00155"></a>00155   mapVecIter_t mapPos;
<a name="l00156"></a>00156   <span class="keywordflow">for</span> (mapPos = probesetListIdToProbesetIdVectorMap.begin();
<a name="l00157"></a>00157       mapPos != probesetListIdToProbesetIdVectorMap.end(); ++mapPos)
<a name="l00158"></a>00158     <span class="keyword">delete</span> mapPos-&gt;second;
<a name="l00159"></a>00159 }
<a name="l00160"></a>00160 <span class="comment"></span>
<a name="l00161"></a>00161 <span class="comment">////////////////////</span>
<a name="l00162"></a>00162 <span class="comment"></span><span class="comment"></span>
<a name="l00163"></a>00163 <span class="comment">/**</span>
<a name="l00164"></a>00164 <span class="comment"> *  \brief Read meta</span>
<a name="l00165"></a>00165 <span class="comment"> *</span>
<a name="l00166"></a>00166 <span class="comment"> *  Read meta probeset file.  Generate map between probeset_list_ids</span>
<a name="l00167"></a>00167 <span class="comment"> *  (from probeset_list column) and vector of probeset_id</span>
<a name="l00168"></a>00168 <span class="comment"> *  (gene) ids (from probeset_id column).  The correspondence</span>
<a name="l00169"></a>00169 <span class="comment"> *  between probeset_list_id and probeset_id id is potentially</span>
<a name="l00170"></a>00170 <span class="comment"> *  one to many.</span>
<a name="l00171"></a>00171 <span class="comment"> *</span>
<a name="l00172"></a>00172 <span class="comment"> *  Errors: call errAbort if any type mismatch</span>
<a name="l00173"></a>00173 <span class="comment"> *  (require that all id strings can be converted to an int)</span>
<a name="l00174"></a>00174 <span class="comment"> *  or in the case of any error return from NextEntryConverted().</span>
<a name="l00175"></a>00175 <span class="comment">*/</span>
<a name="l00176"></a>00176 <span class="keywordtype">void</span> midasEngine::readMeta()
<a name="l00177"></a>00177 {
<a name="l00178"></a>00178   <span class="comment">// printf(&quot;readmeta: %s\n&quot;,metaFileName.c_str());</span>
<a name="l00179"></a>00179 
<a name="l00180"></a>00180   <span class="comment">// We open and close the file here</span>
<a name="l00181"></a>00181   <span class="keywordflow">if</span> (metaTsv.<a class="code" href="classaffx_1_1TsvFile.html#a763ce0bbf19b4134d0445aa58e0f5011" title="Opens a file -- attempts to guess some defaults.">open</a>(metaFileName)!=TSV_OK) {
<a name="l00182"></a>00182     <span class="keywordtype">string</span> msg = <span class="stringliteral">&quot;Problem opening file: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(metaFileName);
<a name="l00183"></a>00183     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(msg);
<a name="l00184"></a>00184   }
<a name="l00185"></a>00185   metaTsv.<a class="code" href="classaffx_1_1TsvFile.html#a3104f733113db8733c335bf425c89a7f" title="Obey the escapechar?">m_optEscapeOk</a>=<span class="keyword">false</span>;
<a name="l00186"></a>00186 
<a name="l00187"></a>00187   <span class="comment">//</span>
<a name="l00188"></a>00188   <span class="keyword">typedef</span> <span class="keyword">const</span> map&lt;int, vector&lt;int&gt;* &gt;::iterator cmapVecIter_t;
<a name="l00189"></a>00189   <span class="comment">//string stringOfProbesetListIds;</span>
<a name="l00190"></a>00190   <span class="keywordtype">string</span> probeset_list;
<a name="l00191"></a>00191   vector&lt;string&gt; probeset_list_vec;
<a name="l00192"></a>00192   int32_t probesetId;
<a name="l00193"></a>00193   <span class="keywordtype">int</span> row = 0;
<a name="l00194"></a>00194 
<a name="l00195"></a>00195   <span class="keywordflow">while</span> (metaTsv.<a class="code" href="classaffx_1_1TsvFile.html#a2022895d082d1d635e7f59fccab12afb" title="Skip to the next level of data which matches seek_clvl ///.">nextLevel</a>(0)==TSV_OK) {
<a name="l00196"></a>00196     ++row;
<a name="l00197"></a>00197     <span class="comment">//</span>
<a name="l00198"></a>00198     metaTsv.<a class="code" href="classaffx_1_1TsvFile.html#a7a81092ce134e8c9116a1a5bcfe287d1" title="Get the string value of a field ///.">get</a>(0,<span class="stringliteral">&quot;probeset_id&quot;</span>,probesetId);
<a name="l00199"></a>00199 
<a name="l00200"></a>00200     <span class="comment">// This is a space seperated list</span>
<a name="l00201"></a>00201     metaTsv.<a class="code" href="classaffx_1_1TsvFile.html#a7a81092ce134e8c9116a1a5bcfe287d1" title="Get the string value of a field ///.">get</a>(0,<span class="stringliteral">&quot;probeset_list&quot;</span>,probeset_list);
<a name="l00202"></a>00202     <span class="comment">// printf(&quot;readmeta: probeset_list=&#39;%s&#39;\n&quot;,probeset_list.c_str());</span>
<a name="l00203"></a>00203     <a class="code" href="namespaceaffx.html#ad5ecd67594ce204d9145778589740e90" title="Split a string into a vector ///.">affx::splitstr</a>(probeset_list,<span class="charliteral">&#39; &#39;</span>,probeset_list_vec);
<a name="l00204"></a>00204     <span class="comment">//</span>
<a name="l00205"></a>00205     <span class="keywordflow">for</span> (<span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> i=0;i&lt;probeset_list_vec.size();i++) {
<a name="l00206"></a>00206       <span class="keywordtype">int</span> probesetListId = -1;
<a name="l00207"></a>00207       <span class="keyword">const</span> <span class="keywordtype">char</span>* s_start=probeset_list_vec[i].c_str();
<a name="l00208"></a>00208       <span class="keywordtype">char</span>* s_end=NULL;
<a name="l00209"></a>00209       probesetListId=strtol(s_start,&amp;s_end,0);
<a name="l00210"></a>00210       <span class="comment">// error converting</span>
<a name="l00211"></a>00211       <span class="keywordflow">if</span> (s_start==s_end) {
<a name="l00212"></a>00212         <span class="keywordtype">string</span> msg = <span class="stringliteral">&quot;Unable to convert &#39;%s&#39;&quot;</span>+probeset_list_vec[i];
<a name="l00213"></a>00213         <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(msg);
<a name="l00214"></a>00214       }
<a name="l00215"></a>00215  
<a name="l00216"></a>00216       <span class="comment">// has this probesetListId been previously seen?</span>
<a name="l00217"></a>00217       cmapVecIter_t probesetListIdToProbesetIdVecPos
<a name="l00218"></a>00218         = probesetListIdToProbesetIdVectorMap.find (probesetListId);
<a name="l00219"></a>00219       <span class="keywordflow">if</span> (probesetListIdToProbesetIdVecPos == probesetListIdToProbesetIdVectorMap.end()) {
<a name="l00220"></a>00220         <span class="comment">// if not, store probesetId in a new vector</span>
<a name="l00221"></a>00221         vector&lt;int&gt;* pVectorInt = <span class="keyword">new</span> vector&lt;int&gt;;
<a name="l00222"></a>00222         (*pVectorInt).push_back (probesetId);
<a name="l00223"></a>00223         probesetListIdToProbesetIdVectorMap.insert (make_pair (probesetListId, pVectorInt) );
<a name="l00224"></a>00224       }
<a name="l00225"></a>00225       <span class="keywordflow">else</span> {
<a name="l00226"></a>00226         <span class="comment">// else a vector&lt;int&gt; already exists for this probesetListId - add to it</span>
<a name="l00227"></a>00227         <span class="comment">// for now, do not bother checking if this probesetId has already been associated</span>
<a name="l00228"></a>00228         <span class="comment">// with this probesetListId</span>
<a name="l00229"></a>00229         vector&lt;int&gt;* pVectorInt = probesetListIdToProbesetIdVecPos-&gt;second;
<a name="l00230"></a>00230         (*pVectorInt).push_back (probesetId);
<a name="l00231"></a>00231       }
<a name="l00232"></a>00232     }
<a name="l00233"></a>00233   }
<a name="l00234"></a>00234   <span class="comment">//</span>
<a name="l00235"></a>00235   metaTsv.<a class="code" href="classaffx_1_1TsvFile.html#ae7932be2156c04a562f327b53833aed7" title="Close the file.">close</a>();
<a name="l00236"></a>00236 }
<a name="l00237"></a>00237 <span class="comment"></span>
<a name="l00238"></a>00238 <span class="comment">/**</span>
<a name="l00239"></a>00239 <span class="comment"> *  \brief Read, process gene data</span>
<a name="l00240"></a>00240 <span class="comment"> *</span>
<a name="l00241"></a>00241 <span class="comment"> *  Store cel_file data as a vector of vector of floats.</span>
<a name="l00242"></a>00242 <span class="comment"> *  Fill in map with key gene (probeset) id, value gene data row.</span>
<a name="l00243"></a>00243 <span class="comment"> *</span>
<a name="l00244"></a>00244 <span class="comment"> *  Errors: Call errAbort if the gene (probeset) label is a duplicate,</span>
<a name="l00245"></a>00245 <span class="comment"> *  in the case of any error in reading the input file or converting data.</span>
<a name="l00246"></a>00246 <span class="comment">*/</span>
<a name="l00247"></a>00247 <span class="keywordtype">void</span> midasEngine::readGeneData ()
<a name="l00248"></a>00248 {
<a name="l00249"></a>00249   <span class="comment">//int rv;</span>
<a name="l00250"></a>00250   <span class="comment">// We open and close the file here</span>
<a name="l00251"></a>00251   <span class="keywordflow">if</span> (geneDataTsv.<a class="code" href="classaffx_1_1TsvFile.html#a763ce0bbf19b4134d0445aa58e0f5011" title="Opens a file -- attempts to guess some defaults.">open</a>(geneDataFileName)!=TSV_OK) {
<a name="l00252"></a>00252     <span class="keywordtype">string</span> msg = <span class="stringliteral">&quot;Problem opening file: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(geneDataFileName);
<a name="l00253"></a>00253     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(msg);
<a name="l00254"></a>00254   }
<a name="l00255"></a>00255   <span class="comment">// printf(&quot;engine: readgenedata: %s\n&quot;,geneDataFileName.c_str());</span>
<a name="l00256"></a>00256   <span class="comment">// The genedata file must have probeset_id as the first column. </span>
<a name="l00257"></a>00257   <span class="keywordflow">if</span> (geneDataTsv.<a class="code" href="classaffx_1_1TsvFile.html#a2dbd33415a815e221ae61cf3feac5500" title="Covert a column index to a column index ///.">cname2cidx</a>(0,<span class="stringliteral">&quot;probeset_id&quot;</span>)!=0) {
<a name="l00258"></a>00258     <span class="keywordtype">string</span> msg=<span class="stringliteral">&quot;probeset_id is not the 0th column.&quot;</span>;
<a name="l00259"></a>00259     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(msg);
<a name="l00260"></a>00260   }
<a name="l00261"></a>00261 
<a name="l00262"></a>00262   <span class="keywordtype">int</span> row=0;
<a name="l00263"></a>00263   <span class="keywordtype">int</span> geneDataProbesetId;
<a name="l00264"></a>00264 
<a name="l00265"></a>00265   <span class="keywordflow">while</span> (geneDataTsv.<a class="code" href="classaffx_1_1TsvFile.html#a2022895d082d1d635e7f59fccab12afb" title="Skip to the next level of data which matches seek_clvl ///.">nextLevel</a>(0)==TSV_OK) {
<a name="l00266"></a>00266     <span class="keywordflow">if</span> (geneDataTsv.<a class="code" href="classaffx_1_1TsvFile.html#a7a81092ce134e8c9116a1a5bcfe287d1" title="Get the string value of a field ///.">get</a>(0,<span class="stringliteral">&quot;probeset_id&quot;</span>,geneDataProbesetId)!=TSV_OK) {
<a name="l00267"></a>00267         <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Bad Conversion: probeset_id&quot;</span>);
<a name="l00268"></a>00268     }
<a name="l00269"></a>00269     <span class="comment">//</span>
<a name="l00270"></a>00270     <span class="keywordflow">if</span> (!geneIdToRowMap.insert(make_pair(geneDataProbesetId,row)).second) {
<a name="l00271"></a>00271       <span class="comment">// abort if a duplicate</span>
<a name="l00272"></a>00272       <span class="keywordtype">string</span> msg = <span class="stringliteral">&quot;Duplicate gene label &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a> (geneDataProbesetId)
<a name="l00273"></a>00273         + <span class="stringliteral">&quot;in file &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a> (geneDataFileName) + <span class="stringliteral">&quot; at row &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a> (row + 1);
<a name="l00274"></a>00274       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(msg);
<a name="l00275"></a>00275     }
<a name="l00276"></a>00276     <span class="comment">// printf(&quot;gdpid=%d &quot;,geneDataProbesetId); // dbg</span>
<a name="l00277"></a>00277     <span class="comment">// the number of data fields in gene data is the number of groups</span>
<a name="l00278"></a>00278     geneData.push_back(vector&lt;float&gt;(numGroups));
<a name="l00279"></a>00279     <span class="comment">//</span>
<a name="l00280"></a>00280     <span class="keywordflow">for</span> (<span class="keywordtype">int</span> col=1;col&lt;geneDataTsv.<a class="code" href="classaffx_1_1TsvFile.html#a94ad901d12ab05cce0f050f99665970f" title="The number of columns for this level ///.">getColumnCount</a>(0);col++) {
<a name="l00281"></a>00281       <span class="keywordflow">if</span> (geneDataTsv.<a class="code" href="classaffx_1_1TsvFile.html#a7a81092ce134e8c9116a1a5bcfe287d1" title="Get the string value of a field ///.">get</a>(0,col,geneData[row][col-1])!=TSV_OK) {
<a name="l00282"></a>00282           <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Bad Conversion: geneData&quot;</span>);
<a name="l00283"></a>00283       }
<a name="l00284"></a>00284       <span class="comment">// printf(&quot; %f&quot;,geneData[row][col-1]); //dbg</span>
<a name="l00285"></a>00285     }
<a name="l00286"></a>00286     row++;
<a name="l00287"></a>00287     <span class="comment">// printf(&quot;\n&quot;); //dbg</span>
<a name="l00288"></a>00288   }
<a name="l00289"></a>00289   geneDataTsv.<a class="code" href="classaffx_1_1TsvFile.html#ae7932be2156c04a562f327b53833aed7" title="Close the file.">close</a>();
<a name="l00290"></a>00290 }
<a name="l00291"></a>00291 <span class="comment"></span>
<a name="l00292"></a>00292 <span class="comment">////////////////////</span>
<a name="l00293"></a>00293 <span class="comment"></span>
<a name="l00294"></a>00294 <span class="keywordtype">void</span>
<a name="l00295"></a>00295 midasEngine::openOutput_Pvalues(std::string filename)
<a name="l00296"></a>00296 {
<a name="l00297"></a>00297   pvaluesTsv.<a class="code" href="classaffx_1_1TsvFile.html#a1b73ea9778aa74c6def832cdaf0ace83" title="Define a TSV file from a string of field names /.">defineFile</a>(PROBESET_LIST_ID <span class="stringliteral">&quot;\t&quot;</span> PROBESET_ID <span class="stringliteral">&quot;\t&quot;</span> <span class="stringliteral">&quot;pvalue&quot;</span>);
<a name="l00298"></a>00298   pvaluesTsv.<a class="code" href="classaffx_1_1TsvFile.html#a085eb57e2a7d91aa8e9e47e748e1a2b1" title="Add all the headers from another TSV to this one. ///.">addHeadersFrom</a>(metatagBucketTsv,<a class="code" href="namespaceaffx.html#a9a97e91d3f603e3c6377f88592ff5284a7b9a90c186868fa93a1fb93e23def373" title="return comments">TSV_ADD_ALL</a>);
<a name="l00299"></a>00299   pvaluesTsv.<a class="code" href="classaffx_1_1TsvFile.html#a1903489904b4e2f8f2a262b8f10cdb0e" title="Set the options up for &amp;quot;tsv&amp;quot; format and write the tsv-v1 header ///.">writeTsv_v1</a>(filename);
<a name="l00300"></a>00300 }
<a name="l00301"></a>00301 <span class="keywordtype">void</span>
<a name="l00302"></a>00302 midasEngine::openOutput_Fstats(std::string filename)
<a name="l00303"></a>00303 {
<a name="l00304"></a>00304   fstatsTsv.<a class="code" href="classaffx_1_1TsvFile.html#a1b73ea9778aa74c6def832cdaf0ace83" title="Define a TSV file from a string of field names /.">defineFile</a>(PROBESET_LIST_ID <span class="stringliteral">&quot;\t&quot;</span> PROBESET_ID <span class="stringliteral">&quot;\t&quot;</span> <span class="stringliteral">&quot;fstatistic&quot;</span>);
<a name="l00305"></a>00305   fstatsTsv.<a class="code" href="classaffx_1_1TsvFile.html#a085eb57e2a7d91aa8e9e47e748e1a2b1" title="Add all the headers from another TSV to this one. ///.">addHeadersFrom</a>(metatagBucketTsv,<a class="code" href="namespaceaffx.html#a9a97e91d3f603e3c6377f88592ff5284a7b9a90c186868fa93a1fb93e23def373" title="return comments">TSV_ADD_ALL</a>);
<a name="l00306"></a>00306   fstatsTsv.<a class="code" href="classaffx_1_1TsvFile.html#a1903489904b4e2f8f2a262b8f10cdb0e" title="Set the options up for &amp;quot;tsv&amp;quot; format and write the tsv-v1 header ///.">writeTsv_v1</a>(filename);
<a name="l00307"></a>00307 }
<a name="l00308"></a>00308 <span class="keywordtype">void</span>
<a name="l00309"></a>00309 midasEngine::openOutput_Normalized(std::string filename)
<a name="l00310"></a>00310 {
<a name="l00311"></a>00311   <span class="keywordtype">int</span> cidx=0;
<a name="l00312"></a>00312 
<a name="l00313"></a>00313   <span class="comment">// first two are fixed</span>
<a name="l00314"></a>00314   normalizedTsv.<a class="code" href="classaffx_1_1TsvFile.html#aadc22863a63b9c6588b0f2dc61875809" title="User methods.">defineColumn</a>(0,cidx++,PROBESET_LIST_ID);
<a name="l00315"></a>00315   normalizedTsv.<a class="code" href="classaffx_1_1TsvFile.html#aadc22863a63b9c6588b0f2dc61875809" title="User methods.">defineColumn</a>(0,cidx++,PROBESET_ID);
<a name="l00316"></a>00316   <span class="comment">// the gene data columns</span>
<a name="l00317"></a>00317   <span class="keywordflow">for</span> (<span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> i = 0; i &lt; numGroups; ++i) {
<a name="l00318"></a>00318     normalizedTsv.<a class="code" href="classaffx_1_1TsvFile.html#aadc22863a63b9c6588b0f2dc61875809" title="User methods.">defineColumn</a>(0,cidx++,celFiles[i]);
<a name="l00319"></a>00319   }
<a name="l00320"></a>00320   normalizedTsv.<a class="code" href="classaffx_1_1TsvFile.html#a085eb57e2a7d91aa8e9e47e748e1a2b1" title="Add all the headers from another TSV to this one. ///.">addHeadersFrom</a>(metatagBucketTsv,<a class="code" href="namespaceaffx.html#a9a97e91d3f603e3c6377f88592ff5284a7b9a90c186868fa93a1fb93e23def373" title="return comments">TSV_ADD_ALL</a>);
<a name="l00321"></a>00321   normalizedTsv.<a class="code" href="classaffx_1_1TsvFile.html#a1903489904b4e2f8f2a262b8f10cdb0e" title="Set the options up for &amp;quot;tsv&amp;quot; format and write the tsv-v1 header ///.">writeTsv_v1</a>(filename);
<a name="l00322"></a>00322 }
<a name="l00323"></a>00323 <span class="comment"></span>
<a name="l00324"></a>00324 <span class="comment">////////////////////</span>
<a name="l00325"></a>00325 <span class="comment"></span><span class="comment"></span>
<a name="l00326"></a>00326 <span class="comment">/**</span>
<a name="l00327"></a>00327 <span class="comment"> *  \brief Read, process exon data</span>
<a name="l00328"></a>00328 <span class="comment"> *</span>
<a name="l00329"></a>00329 <span class="comment"> *  Read exon data; for each row, access associated gene data using</span>
<a name="l00330"></a>00330 <span class="comment"> *  the previously constructed maps and write the requested output to file.</span>
<a name="l00331"></a>00331 <span class="comment"> *</span>
<a name="l00332"></a>00332 <span class="comment"> *  Errors: call errAbort for any i/o or format problem in the input file,</span>
<a name="l00333"></a>00333 <span class="comment"> *  if there is no matching gene id from the map.</span>
<a name="l00334"></a>00334 <span class="comment"> *  We also check for an error in the gene id to gene data row map - &quot;should&quot; not happen.</span>
<a name="l00335"></a>00335 <span class="comment"> *  Ofstreams are configured to throw for any file write error - no explicit checking</span>
<a name="l00336"></a>00336 <span class="comment"> *  is done here.</span>
<a name="l00337"></a>00337 <span class="comment">*/</span>
<a name="l00338"></a>00338 <span class="keywordtype">void</span> midasEngine::processExonData ()
<a name="l00339"></a>00339 {
<a name="l00340"></a>00340   <span class="keyword">typedef</span> <span class="keyword">const</span> map&lt;int, vector&lt;int&gt;* &gt;::iterator cmapVecIter_t;
<a name="l00341"></a>00341   <span class="keyword">typedef</span> <span class="keyword">const</span> map&lt;int, int&gt;::iterator mapIter_t;
<a name="l00342"></a>00342 
<a name="l00343"></a>00343   int32_t probesetListId;
<a name="l00344"></a>00344   <span class="keywordtype">float</span> fstat;
<a name="l00345"></a>00345   vector&lt;float&gt; normalizedSignal (numGroups);
<a name="l00346"></a>00346   vector&lt;float&gt; probesetListData (numGroups);
<a name="l00347"></a>00347   <span class="keywordtype">float</span> pvalue;
<a name="l00348"></a>00348   <span class="comment">//int rv;</span>
<a name="l00349"></a>00349 
<a name="l00350"></a>00350   <span class="keywordflow">if</span> (exonDataTsv.<a class="code" href="classaffx_1_1TsvFile.html#a763ce0bbf19b4134d0445aa58e0f5011" title="Opens a file -- attempts to guess some defaults.">open</a>(exonDataFileName)!=TSV_OK) {
<a name="l00351"></a>00351       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Cant open exon data: &quot;</span>+exonDataFileName);
<a name="l00352"></a>00352   }
<a name="l00353"></a>00353 
<a name="l00354"></a>00354   <span class="keywordtype">int</span> col_cnt=exonDataTsv.<a class="code" href="classaffx_1_1TsvFile.html#a94ad901d12ab05cce0f050f99665970f" title="The number of columns for this level ///.">getColumnCount</a>(0);
<a name="l00355"></a>00355 
<a name="l00356"></a>00356   <span class="comment">// open outputfiles</span>
<a name="l00357"></a>00357   <span class="keywordflow">if</span> (wantPvalues)
<a name="l00358"></a>00358     openOutput_Pvalues(pvaluesFileName);
<a name="l00359"></a>00359   <span class="keywordflow">if</span> (wantFstats)
<a name="l00360"></a>00360     openOutput_Fstats(fstatsFileName);
<a name="l00361"></a>00361   <span class="keywordflow">if</span> (wantNormalized)
<a name="l00362"></a>00362     openOutput_Normalized(normalizedFileName);
<a name="l00363"></a>00363 
<a name="l00364"></a>00364   <span class="keywordflow">while</span> (exonDataTsv.<a class="code" href="classaffx_1_1TsvFile.html#a2022895d082d1d635e7f59fccab12afb" title="Skip to the next level of data which matches seek_clvl ///.">nextLevel</a>(0)==TSV_OK) {
<a name="l00365"></a>00365     <span class="comment">// suck in data</span>
<a name="l00366"></a>00366     exonDataTsv.<a class="code" href="classaffx_1_1TsvFile.html#a7a81092ce134e8c9116a1a5bcfe287d1" title="Get the string value of a field ///.">get</a>(0,<span class="stringliteral">&quot;probeset_id&quot;</span>,probesetListId);
<a name="l00367"></a>00367     <span class="keywordflow">for</span> (<span class="keywordtype">int</span> col=1;col&lt;col_cnt;col++) {<span class="comment"></span>
<a name="l00368"></a>00368 <span class="comment">      /// @todo map names to columns</span>
<a name="l00369"></a>00369 <span class="comment"></span>      <span class="keywordflow">if</span> (exonDataTsv.<a class="code" href="classaffx_1_1TsvFile.html#a7a81092ce134e8c9116a1a5bcfe287d1" title="Get the string value of a field ///.">get</a>(0,col,probesetListData[col-1])!=TSV_OK) {
<a name="l00370"></a>00370         std::string bad_str;
<a name="l00371"></a>00371         exonDataTsv.<a class="code" href="classaffx_1_1TsvFile.html#a7a81092ce134e8c9116a1a5bcfe287d1" title="Get the string value of a field ///.">get</a>(0,col,bad_str);
<a name="l00372"></a>00372         <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Bad float conversion: &#39;&quot;</span>+bad_str+<span class="stringliteral">&quot;&#39;&quot;</span>);
<a name="l00373"></a>00373       }
<a name="l00374"></a>00374     }
<a name="l00375"></a>00375     
<a name="l00376"></a>00376     <span class="comment">// are any probeset (gene) ids associated with this probeset_list_id?</span>
<a name="l00377"></a>00377     cmapVecIter_t geneIdVecPos = probesetListIdToProbesetIdVectorMap.find (probesetListId);
<a name="l00378"></a>00378     <span class="comment">// skip it if not</span>
<a name="l00379"></a>00379     <span class="keywordflow">if</span> (geneIdVecPos==probesetListIdToProbesetIdVectorMap.end()) {
<a name="l00380"></a>00380       <span class="keywordflow">continue</span>;
<a name="l00381"></a>00381     }
<a name="l00382"></a>00382     vector&lt;int&gt;* pGeneIds = geneIdVecPos-&gt;second;
<a name="l00383"></a>00383     <span class="keyword">const</span> <span class="keywordtype">size_t</span> geneIdCount = (*pGeneIds).size();
<a name="l00384"></a>00384     <span class="keywordflow">for</span> (<span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> i = 0; i &lt; geneIdCount; ++i) {
<a name="l00385"></a>00385       <span class="keyword">const</span> <span class="keywordtype">int</span> geneId = (*pGeneIds)[i];
<a name="l00386"></a>00386       <span class="comment">// map gene id to gene data row</span>
<a name="l00387"></a>00387       mapIter_t geneRowIter = geneIdToRowMap.find (geneId);
<a name="l00388"></a>00388       <span class="comment">// skip if not</span>
<a name="l00389"></a>00389       <span class="keywordflow">if</span> (geneRowIter==geneIdToRowMap.end()) {
<a name="l00390"></a>00390         <span class="keywordflow">continue</span>;
<a name="l00391"></a>00391       }
<a name="l00392"></a>00392       <span class="keyword">const</span> <span class="keywordtype">int</span> geneRow = geneRowIter-&gt;second;
<a name="l00393"></a>00393 
<a name="l00394"></a>00394       <span class="comment">// run the detector</span>
<a name="l00395"></a>00395       spliceDetector.<a class="code" href="classmidasSpliceDetector.html#a713fe95a714de0919542e5e06efd6fb9" title="Calculates stats on single exon, gene data.">runMidasSingle</a>(probesetListData,geneData[geneRow],&amp;fstat,&amp;normalizedSignal,&amp;pvalue);
<a name="l00396"></a>00396       
<a name="l00397"></a>00397       <span class="comment">// dump the data</span>
<a name="l00398"></a>00398       <span class="keywordflow">if</span> (wantPvalues) {
<a name="l00399"></a>00399         pvaluesTsv.set(0,<span class="stringliteral">&quot;probeset_list_id&quot;</span>,probesetListId);
<a name="l00400"></a>00400         pvaluesTsv.set(0,<span class="stringliteral">&quot;probeset_id&quot;</span>     ,geneId);
<a name="l00401"></a>00401         pvaluesTsv.set(0,<span class="stringliteral">&quot;pvalue&quot;</span>          ,pvalue);
<a name="l00402"></a>00402         pvaluesTsv.<a class="code" href="classaffx_1_1TsvFile.html#a2a459be42bea9ea3f180773370ec0de7" title="Write a level of data to the file. ///.">writeLevel</a>(0);
<a name="l00403"></a>00403       }
<a name="l00404"></a>00404       <span class="keywordflow">if</span> (wantFstats) {
<a name="l00405"></a>00405         fstatsTsv.set(0,<span class="stringliteral">&quot;probeset_list_id&quot;</span>,probesetListId);
<a name="l00406"></a>00406         fstatsTsv.set(0,<span class="stringliteral">&quot;probeset_id&quot;</span>     ,geneId);
<a name="l00407"></a>00407         fstatsTsv.set(0,<span class="stringliteral">&quot;fstatistic&quot;</span>      ,fstat);
<a name="l00408"></a>00408         fstatsTsv.<a class="code" href="classaffx_1_1TsvFile.html#a2a459be42bea9ea3f180773370ec0de7" title="Write a level of data to the file. ///.">writeLevel</a>(0);
<a name="l00409"></a>00409       }
<a name="l00410"></a>00410       <span class="keywordflow">if</span> (wantNormalized) {
<a name="l00411"></a>00411         normalizedTsv.set(0,<span class="stringliteral">&quot;probeset_list_id&quot;</span>,probesetListId);
<a name="l00412"></a>00412         normalizedTsv.set(0,<span class="stringliteral">&quot;probeset_id&quot;</span>     ,geneId);
<a name="l00413"></a>00413         <span class="keywordflow">for</span> (<span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> g=0;g&lt;numGroups;g++) {
<a name="l00414"></a>00414           normalizedTsv.set(0,g+2,normalizedSignal[g]);
<a name="l00415"></a>00415         }
<a name="l00416"></a>00416         normalizedTsv.<a class="code" href="classaffx_1_1TsvFile.html#a2a459be42bea9ea3f180773370ec0de7" title="Write a level of data to the file. ///.">writeLevel</a>(0);
<a name="l00417"></a>00417       }
<a name="l00418"></a>00418     } <span class="comment">// for</span>
<a name="l00419"></a>00419   } <span class="comment">// while</span>
<a name="l00420"></a>00420   
<a name="l00421"></a>00421   exonDataTsv.<a class="code" href="classaffx_1_1TsvFile.html#ae7932be2156c04a562f327b53833aed7" title="Close the file.">close</a>();
<a name="l00422"></a>00422 
<a name="l00423"></a>00423   <span class="comment">// shut</span>
<a name="l00424"></a>00424   <span class="keywordflow">if</span> (wantPvalues)
<a name="l00425"></a>00425     pvaluesTsv.<a class="code" href="classaffx_1_1TsvFile.html#ae7932be2156c04a562f327b53833aed7" title="Close the file.">close</a>();
<a name="l00426"></a>00426   <span class="keywordflow">if</span> (wantFstats)
<a name="l00427"></a>00427     fstatsTsv.<a class="code" href="classaffx_1_1TsvFile.html#ae7932be2156c04a562f327b53833aed7" title="Close the file.">close</a>();
<a name="l00428"></a>00428   <span class="keywordflow">if</span> (wantNormalized)
<a name="l00429"></a>00429     normalizedTsv.<a class="code" href="classaffx_1_1TsvFile.html#ae7932be2156c04a562f327b53833aed7" title="Close the file.">close</a>();
<a name="l00430"></a>00430 }
<a name="l00431"></a>00431 <span class="comment"></span>
<a name="l00432"></a>00432 <span class="comment">/**</span>
<a name="l00433"></a>00433 <span class="comment"> *  \brief Process gene, exon data</span>
<a name="l00434"></a>00434 <span class="comment"> *</span>
<a name="l00435"></a>00435 <span class="comment"> *  Construct splice detector object, read in, process</span>
<a name="l00436"></a>00436 <span class="comment"> *  gene and exon data.</span>
<a name="l00437"></a>00437 <span class="comment"> *</span>
<a name="l00438"></a>00438 <span class="comment"> *  Errors: none detected here.</span>
<a name="l00439"></a>00439 <span class="comment">*/</span>
<a name="l00440"></a><a class="code" href="classmidasEngine.html#a23a9c4f7e6663cadaed70dd4521c2191">00440</a> <span class="keywordtype">void</span> <a class="code" href="classmidasEngine.html#a23a9c4f7e6663cadaed70dd4521c2191" title="Read, process data, write output.">midasEngine::analyze</a>()
<a name="l00441"></a>00441 {
<a name="l00442"></a>00442   readMeta();
<a name="l00443"></a>00443   readGeneData();                       <span class="comment">// read gene data</span>
<a name="l00444"></a>00444   processExonData ();                   <span class="comment">// process exon data</span>
<a name="l00445"></a>00445 }
</pre></div></div>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:47 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
