<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: chipstream/apt-data-step/ProbesetGenotypeStageEngine.cpp Source File</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li class="current"><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="tabs2">
    <ul class="tablist">
      <li><a href="files.html"><span>File&nbsp;List</span></a></li>
      <li><a href="globals.html"><span>File&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath">
    <ul>
      <li><a class="el" href="dir_f48957cb46a44244b1a86c58bd157ee8.html">chipstream</a>      </li>
      <li><a class="el" href="dir_1d819e5957f521c72c2159e489f328a9.html">apt-data-step</a>      </li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>ProbesetGenotypeStageEngine.cpp</h1>  </div>
</div>
<div class="contents">
<div class="fragment"><pre class="fragment"><a name="l00001"></a>00001 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00002"></a>00002 <span class="comment"></span><span class="comment">//</span>
<a name="l00003"></a>00003 <span class="comment">// Copyright (C) 2005 Affymetrix, Inc.</span>
<a name="l00004"></a>00004 <span class="comment">//</span>
<a name="l00005"></a>00005 <span class="comment">// This program is free software; you can redistribute it and/or modify </span>
<a name="l00006"></a>00006 <span class="comment">// it under the terms of the GNU General Public License (version 2) as </span>
<a name="l00007"></a>00007 <span class="comment">// published by the Free Software Foundation.</span>
<a name="l00008"></a>00008 <span class="comment">// </span>
<a name="l00009"></a>00009 <span class="comment">// This program is distributed in the hope that it will be useful, </span>
<a name="l00010"></a>00010 <span class="comment">// but WITHOUT ANY WARRANTY; without even the implied warranty of </span>
<a name="l00011"></a>00011 <span class="comment">// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU </span>
<a name="l00012"></a>00012 <span class="comment">// General Public License for more details.</span>
<a name="l00013"></a>00013 <span class="comment">// </span>
<a name="l00014"></a>00014 <span class="comment">// You should have received a copy of the GNU General Public License </span>
<a name="l00015"></a>00015 <span class="comment">// along with this program;if not, write to the </span>
<a name="l00016"></a>00016 <span class="comment">// </span>
<a name="l00017"></a>00017 <span class="comment">// Free Software Foundation, Inc., </span>
<a name="l00018"></a>00018 <span class="comment">// 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA</span>
<a name="l00019"></a>00019 <span class="comment">//</span><span class="comment"></span>
<a name="l00020"></a>00020 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00021"></a>00021 <span class="comment"></span>
<a name="l00022"></a>00022 <span class="comment">//</span>
<a name="l00023"></a>00023 <span class="preprocessor">#include &quot;<a class="code" href="ProbesetGenotypeStageEngine_8h.html" title="Core routines for probeset-genotype binaries. By separating the command line parsing form the computa...">chipstream/apt-data-step/ProbesetGenotypeStageEngine.h</a>&quot;</span>
<a name="l00024"></a>00024 <span class="comment">//</span>
<a name="l00025"></a>00025 <span class="preprocessor">#include &quot;<a class="code" href="AnalysisStream_8h.html" title="Object for managing chipstreams, perfect match adjusters and summarization methods.">chipstream/AnalysisStream.h</a>&quot;</span>
<a name="l00026"></a>00026 <span class="preprocessor">#include &quot;<a class="code" href="AnalysisStreamFactory_8h.html" title="Object for making new analysis streams.">chipstream/AnalysisStreamFactory.h</a>&quot;</span>
<a name="l00027"></a>00027 <span class="preprocessor">#include &quot;<a class="code" href="CelReader_8h.html" title="Class for reading cel files and passing the data to memory representation and analysis objects...">chipstream/CelReader.h</a>&quot;</span>
<a name="l00028"></a>00028 <span class="preprocessor">#include &quot;chipstream/CelStatListener.h&quot;</span>
<a name="l00029"></a>00029 <span class="preprocessor">#include &quot;<a class="code" href="ChipStream_8h.html" title="Base class for objects that can initialize themselves one at a time and transform intensity data...">chipstream/ChipStream.h</a>&quot;</span>
<a name="l00030"></a>00030 <span class="preprocessor">#include &quot;chipstream/CnProbeGenderCelListener.h&quot;</span>
<a name="l00031"></a>00031 <span class="preprocessor">#include &quot;<a class="code" href="IntensityMart_8h.html" title="Base class for ojbects that will dispense data on a [feature,chip] level.">chipstream/IntensityMart.h</a>&quot;</span>
<a name="l00032"></a>00032 <span class="preprocessor">#include &quot;<a class="code" href="DiskIntensityMart_8h.html" title="Class for encapsulating microarray data and storing it on disk in a reasonable way to reduce memory i...">chipstream/DiskIntensityMart.h</a>&quot;</span>
<a name="l00033"></a>00033 <span class="preprocessor">#include &quot;<a class="code" href="SparseMart_8h.html" title="Class for representing cel file intensity data in memory. Can save a lot of the space if not all of a...">chipstream/SparseMart.h</a>&quot;</span>
<a name="l00034"></a>00034 <span class="preprocessor">#include &quot;<a class="code" href="DmListener_8h.html" title="Class for using DM calls as cell files become available.">chipstream/DmListener.h</a>&quot;</span>
<a name="l00035"></a>00035 <span class="preprocessor">#include &quot;<a class="code" href="DTFactory_8h.html" title="Wrapper class for all of our DataTransform factories.">chipstream/DTFactory.h</a>&quot;</span>
<a name="l00036"></a>00036 <span class="preprocessor">#include &quot;chipstream/EmGenderCelListener.h&quot;</span>
<a name="l00037"></a>00037 <span class="preprocessor">#include &quot;<a class="code" href="EngineUtil_8h.html" title="Some common functions that application executables make use of.">chipstream/EngineUtil.h</a>&quot;</span>
<a name="l00038"></a>00038 <span class="preprocessor">#include &quot;chipstream/FileGenders.h&quot;</span>
<a name="l00039"></a>00039 <span class="preprocessor">#include &quot;chipstream/FileInbred.h&quot;</span>
<a name="l00040"></a>00040 <span class="preprocessor">#include &quot;<a class="code" href="GenoSeed_8h.html" title="Abstract interface for a genotyping oracle used for genotyping seeds in brlmm. Originally seed calls ...">chipstream/GenoSeed.h</a>&quot;</span>
<a name="l00041"></a>00041 <span class="preprocessor">#include &quot;<a class="code" href="GenoSeedTxt_8h.html" title="Implements interface for a genotyping oracle used for genotyping seeds in brlmm. In this case the gen...">chipstream/GenoSeedTxt.h</a>&quot;</span>
<a name="l00042"></a>00042 <span class="preprocessor">#include &quot;chipstream/GenotypeInfo.h&quot;</span>
<a name="l00043"></a>00043 <span class="preprocessor">#include &quot;chipstream/ProbeListFactory.h&quot;</span>
<a name="l00044"></a>00044 <span class="preprocessor">#include &quot;<a class="code" href="ProbeSet_8h.html" title="Container class for probe sets.">chipstream/ProbeSet.h</a>&quot;</span>
<a name="l00045"></a>00045 <span class="preprocessor">#include &quot;<a class="code" href="QuantBRLMM_8h.html">chipstream/QuantBRLMM.h</a>&quot;</span>
<a name="l00046"></a>00046 <span class="preprocessor">#include &quot;chipstream/QuantBirdseedDev.h&quot;</span>
<a name="l00047"></a>00047 <span class="preprocessor">#include &quot;chipstream/QuantBirdseedLegacy.h&quot;</span>
<a name="l00048"></a>00048 <span class="preprocessor">#include &quot;chipstream/QuantBirdseedv1.h&quot;</span>
<a name="l00049"></a>00049 <span class="preprocessor">#include &quot;chipstream/QuantBirdseedv2.h&quot;</span>
<a name="l00050"></a>00050 <span class="preprocessor">#include &quot;<a class="code" href="QuantExprMethod_8h.html" title="Interface for computing quantification summaries from PM intensities grouped into probe set groups...">chipstream/QuantExprMethod.h</a>&quot;</span>
<a name="l00051"></a>00051 <span class="preprocessor">#include &quot;<a class="code" href="QuantLabelZ_8h.html">chipstream/QuantLabelZ.h</a>&quot;</span>
<a name="l00052"></a>00052 <span class="preprocessor">#include &quot;<a class="code" href="QuantLabelZMulti_8h.html" title="Interface for methods computing SNP genotyping calls.">chipstream/QuantLabelZMulti.h</a>&quot;</span>
<a name="l00053"></a>00053 <span class="preprocessor">#include &quot;<a class="code" href="QuantMethod_8h.html" title="Abstract interface for a quantification method.">chipstream/QuantMethod.h</a>&quot;</span>
<a name="l00054"></a>00054 <span class="preprocessor">#include &quot;<a class="code" href="QuantMethodGTypeCHPReport_8h.html" title="Reporter for genotyping probe sets that outputs chp files. There is one GCOS XDA CHP file produced fo...">chipstream/QuantMethodGTypeCHPReport.h</a>&quot;</span>
<a name="l00055"></a>00055 <span class="preprocessor">#include &quot;<a class="code" href="QuantMethodGTypeChipSummary_8h.html" title="Class for accumulating some run statistics from a genotyping run.">chipstream/QuantMethodGTypeChipSummary.h</a>&quot;</span>
<a name="l00056"></a>00056 <span class="preprocessor">#include &quot;<a class="code" href="QuantMethodGTypeExprChipSummary_8h.html" title="Class for accumulating some run statistics from a genotyping run from the expression quant method...">chipstream/QuantMethodGTypeExprChipSummary.h</a>&quot;</span>
<a name="l00057"></a>00057 <span class="preprocessor">#include &quot;<a class="code" href="QuantMethodGTypeReport_8h.html" title="Reporter for outputting genotype results.">chipstream/QuantMethodGTypeReport.h</a>&quot;</span>
<a name="l00058"></a>00058 <span class="preprocessor">#include &quot;chipstream/apt-data-step/AnalysisStage.h&quot;</span>
<a name="l00059"></a>00059 
<a name="l00060"></a>00060 <span class="comment">/* Avoid compiling these into windows builds until GTC team has been </span>
<a name="l00061"></a>00061 <span class="comment">   informed and given time to update their builds as new libraries are requred. */</span>
<a name="l00062"></a>00062 <span class="preprocessor">#ifndef WIN32 </span>
<a name="l00063"></a>00063 <span class="preprocessor"></span><span class="preprocessor">#include &quot;<a class="code" href="QuantMethodSqlGTypeReport_8h.html" title="Class for reporting results of quantification methods into an sql database.">chipstream/QuantMethodSqlGTypeReport.h</a>&quot;</span>
<a name="l00064"></a>00064 <span class="preprocessor">#include &quot;chipstream/QuantMethodSqlExprReport.h&quot;</span>
<a name="l00065"></a>00065 <span class="preprocessor">#endif</span>
<a name="l00066"></a>00066 <span class="preprocessor"></span>
<a name="l00067"></a>00067 <span class="preprocessor">#include &quot;<a class="code" href="SnpModelConverter_8h.html" title="Class collecting utility methods for reading various formats of snp models like:Brlmm-p (aka labelz) ...">chipstream/SnpModelConverter.h</a>&quot;</span>
<a name="l00068"></a>00068 <span class="preprocessor">#include &quot;<a class="code" href="QuantMethodMultiDataCCCHPReport_8h.html" title="Reporter for genotyping probe sets that outputs chp files. There is one CC CHP file produced for each...">chipstream/QuantMethodMultiDataCCCHPReport.h</a>&quot;</span>
<a name="l00069"></a>00069 <span class="preprocessor">#include &quot;<a class="code" href="QuantMethodRunReport_8h.html" title="Summarize and output basic statistics on the performance of a particular summarization or detection m...">chipstream/QuantMethodRunReport.h</a>&quot;</span>
<a name="l00070"></a>00070 <span class="preprocessor">#include &quot;<a class="code" href="SpecialSnps_8h.html">chipstream/SpecialSnps.h</a>&quot;</span>
<a name="l00071"></a>00071 <span class="comment">//</span>
<a name="l00072"></a>00072 <span class="preprocessor">#include &quot;broadutil/BroadException.h&quot;</span>
<a name="l00073"></a>00073 <span class="preprocessor">#include &quot;broadutil/CelUtil.h&quot;</span>
<a name="l00074"></a>00074 <span class="preprocessor">#include &quot;<a class="code" href="ExceptionBase_8h.html">calvin_files/exception/src/ExceptionBase.h</a>&quot;</span>
<a name="l00075"></a>00075 <span class="preprocessor">#include &quot;<a class="code" href="FusionCDFData_8h.html">calvin_files/fusion/src/FusionCDFData.h</a>&quot;</span>
<a name="l00076"></a>00076 <span class="preprocessor">#include &quot;<a class="code" href="FusionCELData_8h.html">calvin_files/fusion/src/FusionCELData.h</a>&quot;</span>
<a name="l00077"></a>00077 <span class="preprocessor">#include &quot;<a class="code" href="StringUtils_8h.html">calvin_files/utils/src/StringUtils.h</a>&quot;</span>
<a name="l00078"></a>00078 <span class="preprocessor">#include &quot;<a class="code" href="TsvFile_8h.html" title="Headers for the TsvFile classes.">file/TsvFile/TsvFile.h</a>&quot;</span>
<a name="l00079"></a>00079 
<a name="l00080"></a>00080 <span class="preprocessor">#include &quot;<a class="code" href="BaseEngine_8h.html" title="base engine base class">util/BaseEngine.h</a>&quot;</span>
<a name="l00081"></a>00081 <span class="preprocessor">#include &quot;<a class="code" href="Err_8h.html" title="Some utility error functions, currently prints message and aborts().">util/Err.h</a>&quot;</span>
<a name="l00082"></a>00082 <span class="preprocessor">#include &quot;<a class="code" href="Fs_8h.html" title="///">util/Fs.h</a>&quot;</span>
<a name="l00083"></a>00083 <span class="preprocessor">#include &quot;<a class="code" href="PgOptions_8h.html" title="Classes for reading program command line options.">util/PgOptions.h</a>&quot;</span>
<a name="l00084"></a>00084 <span class="preprocessor">#include &quot;<a class="code" href="Util_8h.html" title="General Utilities.">util/Util.h</a>&quot;</span>
<a name="l00085"></a>00085 <span class="preprocessor">#include &quot;<a class="code" href="Verbose_8h.html" title="Class for doing logging and some command line ui.">util/Verbose.h</a>&quot;</span>
<a name="l00086"></a>00086 <span class="comment">//</span>
<a name="l00087"></a>00087 <span class="preprocessor">#include &lt;cstdio&gt;</span>
<a name="l00088"></a>00088 <span class="preprocessor">#include &lt;cstring&gt;</span>
<a name="l00089"></a>00089 <span class="preprocessor">#include &lt;ctime&gt;</span>
<a name="l00090"></a>00090 <span class="preprocessor">#include &lt;iostream&gt;</span>
<a name="l00091"></a>00091 <span class="preprocessor">#include &lt;map&gt;</span>
<a name="l00092"></a>00092 <span class="preprocessor">#include &lt;string&gt;</span>
<a name="l00093"></a>00093 <span class="preprocessor">#include &lt;vector&gt;</span>
<a name="l00094"></a>00094 
<a name="l00095"></a>00095 <span class="comment">//</span>
<a name="l00096"></a>00096 <span class="preprocessor">#ifndef WIN32</span>
<a name="l00097"></a>00097 <span class="preprocessor"></span><span class="preprocessor">#include &lt;unistd.h&gt;</span>
<a name="l00098"></a>00098 <span class="preprocessor">#endif </span><span class="comment">/* WIN32 */</span>
<a name="l00099"></a>00099 
<a name="l00100"></a>00100 <span class="keyword">using namespace </span>std;
<a name="l00101"></a>00101 <span class="keyword">using namespace </span>affx;
<a name="l00102"></a>00102 <span class="keyword">using namespace </span>affymetrix_calvin_exceptions;
<a name="l00103"></a>00103 
<a name="l00104"></a>00104 <a class="code" href="classProbesetGenotypeStageEngine_1_1Reg.html">ProbesetGenotypeStageEngine::Reg</a> <a class="code" href="classProbesetGenotypeStageEngine.html#a50d7a2c29002314c6f198ec78ac38c94">ProbesetGenotypeStageEngine::reg</a>;
<a name="l00105"></a>00105 
<a name="l00106"></a><a class="code" href="classProbesetGenotypeStageEngine.html#a9cf4dd3715bc5a74beb598f4b5333cc9">00106</a> <a class="code" href="classProbesetGenotypeStageEngine.html">ProbesetGenotypeStageEngine</a> * <a class="code" href="classProbesetGenotypeStageEngine.html#a9cf4dd3715bc5a74beb598f4b5333cc9">ProbesetGenotypeStageEngine::FromBase</a>(<a class="code" href="classBaseEngine.html" title="Base class for analysis engines.">BaseEngine</a> *engine)
<a name="l00107"></a>00107 {
<a name="l00108"></a>00108     <span class="keywordflow">if</span> (engine != NULL &amp;&amp; engine-&gt;getEngineName() == ProbesetGenotypeStageEngine::EngineName())
<a name="l00109"></a>00109         <span class="keywordflow">return</span> (<a class="code" href="classProbesetGenotypeStageEngine.html">ProbesetGenotypeStageEngine</a> *)engine;
<a name="l00110"></a>00110     <span class="keywordflow">return</span> NULL;
<a name="l00111"></a>00111 }
<a name="l00112"></a>00112 
<a name="l00113"></a><a class="code" href="classProbesetGenotypeStageEngine.html#a2b37addfac89752621645984a60173e4">00113</a> <a class="code" href="classProbesetGenotypeStageEngine.html#a2b37addfac89752621645984a60173e4" title="Constructor.">ProbesetGenotypeStageEngine::ProbesetGenotypeStageEngine</a>(
<a name="l00114"></a>00114         <a class="code" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a> *chipLayout,
<a name="l00115"></a>00115         std::vector&lt;const char *&gt; *probesetNames) {
<a name="l00116"></a>00116     m_a5_global_input_file = NULL;
<a name="l00117"></a>00117     m_a5_global_input_group = NULL;
<a name="l00118"></a>00118     m_a5_global_output_file = NULL;
<a name="l00119"></a>00119     m_a5_global_output_group = NULL;
<a name="l00120"></a>00120     defineStdMethods();
<a name="l00121"></a>00121     defineOptions();
<a name="l00122"></a>00122 
<a name="l00123"></a>00123     <span class="keywordflow">if</span>(chipLayout != NULL) {
<a name="l00124"></a>00124       <span class="keywordflow">if</span>(probesetNames == NULL)
<a name="l00125"></a>00125           <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Must specify probesetNames when providing a chipLayout to ProbesetGenotypeStageEngine constructor.&quot;</span>);
<a name="l00126"></a>00126       m_FreeChipLayout = <span class="keyword">false</span>;
<a name="l00127"></a>00127       m_ChipLayout = chipLayout;
<a name="l00128"></a>00128       m_ProbesetNames = *probesetNames;
<a name="l00129"></a>00129     } <span class="keywordflow">else</span> {
<a name="l00130"></a>00130       m_FreeChipLayout = <span class="keyword">true</span>;
<a name="l00131"></a>00131       m_ChipLayout = NULL;
<a name="l00132"></a>00132     }
<a name="l00133"></a>00133 }
<a name="l00134"></a>00134 
<a name="l00135"></a><a class="code" href="classProbesetGenotypeStageEngine.html#a5d543df52d60039be1c17d086304f847">00135</a> <a class="code" href="classProbesetGenotypeStageEngine.html#a5d543df52d60039be1c17d086304f847" title="Destructor.">ProbesetGenotypeStageEngine::~ProbesetGenotypeStageEngine</a>() {
<a name="l00136"></a>00136 }
<a name="l00137"></a>00137 
<a name="l00138"></a>00138 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::defineStdMethods() {
<a name="l00139"></a>00139     m_stdMethods[<span class="stringliteral">&quot;brlmm&quot;</span>] = <span class="stringliteral">&quot;quant-norm.sketch=50000,pm-only,brlmm.transform=ccs.K=4&quot;</span>;
<a name="l00140"></a>00140     m_stdMethods[<span class="stringliteral">&quot;brlmm-p&quot;</span>] = <span class="stringliteral">&quot;quant-norm.sketch=50000,pm-only,brlmm-p.CM=1.bins=100.K=2.SB=0.003.MS=0.05&quot;</span>;
<a name="l00141"></a>00141     m_stdMethods[<span class="stringliteral">&quot;brlmm-p.force&quot;</span>] = <span class="stringliteral">&quot;quant-norm.sketch=50000,pm-only,brlmm-p.CM=1.bins=100.K=2.SB=0.003.MS=1&quot;</span>;
<a name="l00142"></a>00142     m_stdMethods[<span class="stringliteral">&quot;brlmm-p-plus&quot;</span>] = <span class="stringliteral">&quot;quant-norm.sketch=50000,pm-only,brlmm-p.CM=1.bins=100.mix=1.bic=2.HARD=3.SB=0.45.KX=1.KH=1.5.KXX=0.5.KAH=-0.6.KHB=-0.6.transform=MVA.AAM=2.0.BBM=-2.0.AAV=0.06.BBV=0.06.ABV=0.06.copyqc=0.000001.wobble=0.05.MS=0.05&quot;</span>;
<a name="l00143"></a>00143     m_stdMethods[<span class="stringliteral">&quot;brlmm-p-plus.force&quot;</span>] = <span class="stringliteral">&quot;quant-norm.sketch=50000,pm-only,brlmm-p.CM=1.bins=100.mix=1.bic=2.HARD=3.SB=0.45.KX=1.KH=1.5.KXX=0.5.KAH=-0.6.KHB=-0.6.transform=MVA.AAM=2.0.BBM=-2.0.AAV=0.06.BBV=0.06.ABV=0.06.copyqc=0.000001.wobble=0.05.MS=1&quot;</span>;
<a name="l00144"></a>00144     m_stdMethods[<span class="stringliteral">&quot;birdseed&quot;</span>] = <span class="stringliteral">&quot;quant-norm.sketch=50000,pm-only,birdseed&quot;</span>;
<a name="l00145"></a>00145     m_stdMethods[<span class="stringliteral">&quot;birdseed.force&quot;</span>] = <span class="stringliteral">&quot;quant-norm.sketch=50000,pm-only,birdseed.conf-threshold=1&quot;</span>;
<a name="l00146"></a>00146     m_stdMethods[<span class="stringliteral">&quot;birdseed-v1&quot;</span>] = <span class="stringliteral">&quot;quant-norm.sketch=50000,pm-only,birdseed-v1&quot;</span>;
<a name="l00147"></a>00147     m_stdMethods[<span class="stringliteral">&quot;birdseed-v1.force&quot;</span>] = <span class="stringliteral">&quot;quant-norm.sketch=50000,pm-only,birdseed-v1.conf-threshold=1&quot;</span>;
<a name="l00148"></a>00148     m_stdMethods[<span class="stringliteral">&quot;birdseed-v2&quot;</span>] = <span class="stringliteral">&quot;quant-norm.sketch=50000.target=1000,pm-only,birdseed-v2&quot;</span>;
<a name="l00149"></a>00149     m_stdMethods[<span class="stringliteral">&quot;birdseed-v2.force&quot;</span>] = <span class="stringliteral">&quot;quant-norm.sketch=50000.target=1000,pm-only,birdseed-v2.conf-threshold=1&quot;</span>;
<a name="l00150"></a>00150     m_stdMethods[<span class="stringliteral">&quot;birdseed-dev&quot;</span>] = <span class="stringliteral">&quot;quant-norm.sketch=50000.target=1000,pm-only,birdseed-dev&quot;</span>;
<a name="l00151"></a>00151     m_stdMethods[<span class="stringliteral">&quot;birdseed-dev.force&quot;</span>] = <span class="stringliteral">&quot;quant-norm.sketch=50000.target=1000,pm-only,birdseed-dev.conf-threshold=1&quot;</span>;
<a name="l00152"></a>00152 }
<a name="l00153"></a>00153 
<a name="l00154"></a>00154 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::defineOptions() {
<a name="l00155"></a>00155 
<a name="l00156"></a>00156   defineOptionSection(<span class="stringliteral">&quot;Input Options&quot;</span>);
<a name="l00157"></a>00157   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;cel-files&quot;</span>, PgOpt::STRING_OPT,
<a name="l00158"></a>00158                      <span class="stringliteral">&quot;Text file specifying cel files to process, one per line with the first line being &#39;cel_files&#39;.&quot;</span>,
<a name="l00159"></a>00159                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00160"></a>00160   defineOption(<span class="stringliteral">&quot;c&quot;</span>, <span class="stringliteral">&quot;cdf-file&quot;</span>, PgOpt::STRING_OPT,
<a name="l00161"></a>00161                      <span class="stringliteral">&quot;File defining probe sets. Use either --cdf-file or --spf-file. &quot;</span>,
<a name="l00162"></a>00162                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00163"></a>00163   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;spf-file&quot;</span>, PgOpt::STRING_OPT,
<a name="l00164"></a>00164                      <span class="stringliteral">&quot;File defining probe sets in spf (simple probe format) which is like a text cdf file.&quot;</span>,
<a name="l00165"></a>00165                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00166"></a>00166   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;chrX-snps&quot;</span>, PgOpt::STRING_OPT,
<a name="l00167"></a>00167                     <span class="stringliteral">&quot;File containing snps on chrX (non-pseudoautosomal region).&quot;</span>,
<a name="l00168"></a>00168                     <span class="stringliteral">&quot;&quot;</span>);
<a name="l00169"></a>00169   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;special-snps&quot;</span>,PgOpt::STRING_OPT,
<a name="l00170"></a>00170                     <span class="stringliteral">&quot;File containing all snps of unusual copy (chrX,mito,Y)&quot;</span>,
<a name="l00171"></a>00171                     <span class="stringliteral">&quot;&quot;</span>);
<a name="l00172"></a>00172   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;chrX-probes&quot;</span>, PgOpt::STRING_OPT,
<a name="l00173"></a>00173                      <span class="stringliteral">&quot;File containing probe_id (1-based) of probes on chrX. &quot;</span>
<a name="l00174"></a>00174                      <span class="stringliteral">&quot;Used for copy number probe chrX/Y ratio gender calling. &quot;</span>
<a name="l00175"></a>00175                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00176"></a>00176                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00177"></a>00177   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;chrY-probes&quot;</span>, PgOpt::STRING_OPT,
<a name="l00178"></a>00178                      <span class="stringliteral">&quot;File containing probe_id (1-based) of probes on chrY. &quot;</span>
<a name="l00179"></a>00179                      <span class="stringliteral">&quot;Used for copy number probe chrX/Y ratio gender calling. &quot;</span>
<a name="l00180"></a>00180                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00181"></a>00181                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00182"></a>00182   defineOption(<span class="stringliteral">&quot;s&quot;</span>, <span class="stringliteral">&quot;probeset-ids&quot;</span>, PgOpt::STRING_OPT,
<a name="l00183"></a>00183                      <span class="stringliteral">&quot;Tab delimited file with column &#39;probeset_id&#39; specifying probesets to genotype.&quot;</span>,
<a name="l00184"></a>00184                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00185"></a>00185   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;probeset-ids-reported&quot;</span>, PgOpt::STRING_OPT,
<a name="l00186"></a>00186                      <span class="stringliteral">&quot;Tab delimited file with column &#39;probeset_id&#39; specifying probesets to report. &quot;</span>
<a name="l00187"></a>00187                      <span class="stringliteral">&quot;This should be a subset of those specified with --probeset-ids if that option is used.&quot;</span>,
<a name="l00188"></a>00188                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00189"></a>00189   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;probe-class-file&quot;</span>, PgOpt::STRING_OPT,
<a name="l00190"></a>00190                      <span class="stringliteral">&quot;File containing probe_id (1-based) of probes and a &#39;class&#39; designation. &quot;</span>
<a name="l00191"></a>00191                      <span class="stringliteral">&quot;Used to compute mean probe intensity by class for report file.&quot;</span>,
<a name="l00192"></a>00192                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00193"></a>00193   defOptMult(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;chip-type&quot;</span>, PgOpt::STRING_OPT,
<a name="l00194"></a>00194                      <span class="stringliteral">&quot;Chip types to check library and CEL files against. &quot;</span>
<a name="l00195"></a>00195                      <span class="stringliteral">&quot;Can be specified multiple times. &quot;</span>
<a name="l00196"></a>00196                      <span class="stringliteral">&quot;The first one is propigated as the chip type in the output files. &quot;</span>
<a name="l00197"></a>00197                      <span class="stringliteral">&quot;Warning, use of this option will override the usual check between chip types &quot;</span>
<a name="l00198"></a>00198                      <span class="stringliteral">&quot;found in the library files and cel files. You should use this option instead &quot;</span>
<a name="l00199"></a>00199                      <span class="stringliteral">&quot;of --force when possible. &quot;</span>,
<a name="l00200"></a>00200                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00201"></a>00201   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;annotation-file&quot;</span>, PgOpt::STRING_OPT,
<a name="l00202"></a>00202                      <span class="stringliteral">&quot;Annotation file.&quot;</span>,       
<a name="l00203"></a>00203                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00204"></a>00204   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;genotype-markers-cn-file&quot;</span>, PgOpt::STRING_OPT, 
<a name="l00205"></a>00205                      <span class="stringliteral">&quot;Tab delimited file with copy number calls for genotype probesets within copy number regions&quot;</span>, 
<a name="l00206"></a>00206                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00207"></a>00207   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;writeOldStyleFeatureEffectsFile&quot;</span> , PgOpt::BOOL_OPT, 
<a name="l00208"></a>00208                    <span class="stringliteral">&quot;Boolean value to determine whether or not old style feature effects files are written.&quot;</span>,
<a name="l00209"></a>00209                    <span class="stringliteral">&quot;false&quot;</span>); 
<a name="l00210"></a>00210   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;file5-compact&quot;</span> , PgOpt::BOOL_OPT, 
<a name="l00211"></a>00211                    <span class="stringliteral">&quot;Should we output results in a compact file5 output.&quot;</span>,
<a name="l00212"></a>00212                    <span class="stringliteral">&quot;false&quot;</span>); 
<a name="l00213"></a>00213 <span class="comment">/* Avoid compiling these into windows builds until GTC team has been </span>
<a name="l00214"></a>00214 <span class="comment">   informed and given time to update their builds as new libraries are requred. */</span>
<a name="l00215"></a>00215 <span class="preprocessor">#ifndef WIN32</span>
<a name="l00216"></a>00216 <span class="preprocessor"></span>  defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;sqlite-output&quot;</span> , PgOpt::BOOL_OPT, 
<a name="l00217"></a>00217                    <span class="stringliteral">&quot;Shoul output some results in sqlite3 format?&quot;</span>,
<a name="l00218"></a>00218                    <span class="stringliteral">&quot;false&quot;</span>); 
<a name="l00219"></a>00219 <span class="preprocessor">#endif</span>
<a name="l00220"></a>00220 <span class="preprocessor"></span>  defineOptionSection(<span class="stringliteral">&quot;Output Options&quot;</span>);
<a name="l00221"></a>00221   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;table-output&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00222"></a>00222                     <span class="stringliteral">&quot;Output matching matrices of tab delimited genotype calls and confidences.&quot;</span>,
<a name="l00223"></a>00223                     <span class="stringliteral">&quot;true&quot;</span>);
<a name="l00224"></a>00224   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;output-forced-calls&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00225"></a>00225                     <span class="stringliteral">&quot;Output a separate file with forced calls.&quot;</span>,
<a name="l00226"></a>00226                     <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00227"></a>00227   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;output-context&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00228"></a>00228                     <span class="stringliteral">&quot;Output a separate file with the allele context used. This is only relevant for marker type probesets which have multiple groups of probes for each allele based on the context of nearby SNPs.&quot;</span>,
<a name="l00229"></a>00229                     <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00230"></a>00230   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;cc-chp-output&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00231"></a>00231                     <span class="stringliteral">&quot;Output resulting calls in directory called &#39;cc-chp&#39; under out-dir. &quot;</span>
<a name="l00232"></a>00232                     <span class="stringliteral">&quot;This makes one AGCC Multi Data CHP file per cel file analyzed.&quot;</span>,
<a name="l00233"></a>00233                     <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00234"></a>00234   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;xda-chp-output&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00235"></a>00235                     <span class="stringliteral">&quot;Output resulting calls in directory called &#39;chp&#39; under out-dir. &quot;</span>
<a name="l00236"></a>00236                     <span class="stringliteral">&quot;This makes one GCOS XDA CHP file per cel file analyzed. &quot;</span>
<a name="l00237"></a>00237                     <span class="stringliteral">&quot;Note that this format is not supported beyond the Mapping500K chips, &quot;</span>
<a name="l00238"></a>00238                     <span class="stringliteral">&quot;for subsequent chips look at the CC CHP format instead.&quot;</span>,
<a name="l00239"></a>00239                     <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00240"></a>00240   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;cc-chp-out-dir&quot;</span>, PgOpt::STRING_OPT,
<a name="l00241"></a>00241                      <span class="stringliteral">&quot;Over-ride the default location for chp output.&quot;</span>,
<a name="l00242"></a>00242                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00243"></a>00243   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;xda-chp-out-dir&quot;</span>, PgOpt::STRING_OPT,
<a name="l00244"></a>00244                      <span class="stringliteral">&quot;Over-ride the default location for chp output.&quot;</span>,
<a name="l00245"></a>00245                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00246"></a>00246   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;summaries&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00247"></a>00247                      <span class="stringliteral">&quot;Output the summary values from the quantifcation method for each allele. &quot;</span>
<a name="l00248"></a>00248                      <span class="stringliteral">&quot;For brlmm-p this will also write a file of transformed summary values &quot;</span>
<a name="l00249"></a>00249                      <span class="stringliteral">&quot;in contrast space used in the clustering.&quot;</span>,
<a name="l00250"></a>00250                     <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00251"></a>00251   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;report-file&quot;</span>, PgOpt::STRING_OPT,
<a name="l00252"></a>00252                      <span class="stringliteral">&quot;Over-ride the default report file name.&quot;</span>,
<a name="l00253"></a>00253                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00254"></a>00254 
<a name="l00255"></a>00255   defineOptionSection(<span class="stringliteral">&quot;Analysis Options&quot;</span>);
<a name="l00256"></a>00256 
<a name="l00257"></a>00257   defOptMult(<span class="stringliteral">&quot;a&quot;</span>, <span class="stringliteral">&quot;analysis&quot;</span>, PgOpt::STRING_OPT,
<a name="l00258"></a>00258                    <span class="stringliteral">&quot;String representing analysis pathway desired. &quot;</span>
<a name="l00259"></a>00259                    <span class="stringliteral">&quot;For example: &#39;quant-norm.sketch=50000,pm-only,brlmm&#39;.&quot;</span>,
<a name="l00260"></a>00260                    <span class="stringliteral">&quot;brlmm&quot;</span>);
<a name="l00261"></a>00261   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;qmethod-spec&quot;</span>, PgOpt::STRING_OPT,
<a name="l00262"></a>00262                      <span class="stringliteral">&quot;Quantification Method to use for summarizing alleles.&quot;</span>,
<a name="l00263"></a>00263                      <span class="stringliteral">&quot;plier.optmethod=1&quot;</span>);
<a name="l00264"></a>00264   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;read-models-brlmm&quot;</span>, PgOpt::STRING_OPT,
<a name="l00265"></a>00265                      <span class="stringliteral">&quot;File to read precomputed BRLMM snp specific models from.&quot;</span>,
<a name="l00266"></a>00266                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00267"></a>00267   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;read-models-brlmmp&quot;</span>, PgOpt::STRING_OPT,
<a name="l00268"></a>00268                      <span class="stringliteral">&quot;File to read precomputed BRLMM-P snp specific models from.&quot;</span>,
<a name="l00269"></a>00269                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00270"></a>00270   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;brlmmp-models-non-sequential&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00271"></a>00271                      <span class="stringliteral">&quot;File to read precomputed BRLMM-P snp specific models from is not in sequential order.&quot;</span>,
<a name="l00272"></a>00272                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00273"></a>00273   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;read-models-birdseed&quot;</span>, PgOpt::STRING_OPT,
<a name="l00274"></a>00274                      <span class="stringliteral">&quot;File to read precomputed birdseed snp specific models from.&quot;</span>,
<a name="l00275"></a>00275                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00276"></a>00276   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;write-models&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00277"></a>00277                      <span class="stringliteral">&quot;Should we write snp specific models out for analysis? [experimental]&quot;</span>,
<a name="l00278"></a>00278                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00279"></a>00279   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;db-from-prior-models&quot;</span>, PgOpt::STRING_OPT,
<a name="l00280"></a>00280                      <span class="stringliteral">&quot;File to write prior snp models to for random access.&quot;</span>,
<a name="l00281"></a>00281                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00282"></a>00282   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;db-from-posterior-models&quot;</span>, PgOpt::STRING_OPT,
<a name="l00283"></a>00283                      <span class="stringliteral">&quot;File to write posterior snp models to for random access.&quot;</span>,
<a name="l00284"></a>00284                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00285"></a>00285   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;feat-effects&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00286"></a>00286                      <span class="stringliteral">&quot;Output feature effects when available.&quot;</span>,
<a name="l00287"></a>00287                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00288"></a>00288   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;use-feat-eff&quot;</span>, PgOpt::STRING_OPT,
<a name="l00289"></a>00289                     <span class="stringliteral">&quot;File defining a plier feature effect for each probe. &quot;</span>
<a name="l00290"></a>00290                      <span class="stringliteral">&quot;Note that precomputed effects should only be used for an appropriately similar analysis &quot;</span>
<a name="l00291"></a>00291                      <span class="stringliteral">&quot;(i.e. feature effects for pm-only may be different than for pm-mm).&quot;</span>,
<a name="l00292"></a>00292                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00293"></a>00293   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;feat-details&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00294"></a>00294                      <span class="stringliteral">&quot;Output the feature details (usually residuals) from the quantification method if available.&quot;</span>,
<a name="l00295"></a>00295                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00296"></a>00296   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;target-sketch&quot;</span>, PgOpt::STRING_OPT,
<a name="l00297"></a>00297                     <span class="stringliteral">&quot;File specifying a target distribution to use for quantile normalization.&quot;</span>,
<a name="l00298"></a>00298                     <span class="stringliteral">&quot;&quot;</span>);
<a name="l00299"></a>00299   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;write-sketch&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00300"></a>00300                     <span class="stringliteral">&quot;Write the quantile normalization distribution (or sketch) to a file for reuse with target-sketch option.&quot;</span>,
<a name="l00301"></a>00301                     <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00302"></a>00302   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;dm-thresh&quot;</span>, PgOpt::DOUBLE_OPT,
<a name="l00303"></a>00303                     <span class="stringliteral">&quot;Minimum DM p-value to seed clusters with.&quot;</span>,
<a name="l00304"></a>00304                      <span class="stringliteral">&quot;.17&quot;</span>);
<a name="l00305"></a>00305   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;reference-profile&quot;</span>, PgOpt::STRING_OPT,
<a name="l00306"></a>00306                     <span class="stringliteral">&quot;File specifying reference chip profile.&quot;</span>,
<a name="l00307"></a>00307                     <span class="stringliteral">&quot;&quot;</span>);
<a name="l00308"></a>00308   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;write-profile&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00309"></a>00309                     <span class="stringliteral">&quot;Write the reference chip profile to a file for reuse.&quot;</span>,
<a name="l00310"></a>00310                     <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00311"></a>00311   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;dm-hetmult&quot;</span>, PgOpt::DOUBLE_OPT,
<a name="l00312"></a>00312                      <span class="stringliteral">&quot;DM hetmultiplier to balance het/hom calls, additive to log likelihood.&quot;</span>,
<a name="l00313"></a>00313                      <span class="stringliteral">&quot;0&quot;</span>);
<a name="l00314"></a>00314   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;prior-size&quot;</span>, PgOpt::INT_OPT,
<a name="l00315"></a>00315                     <span class="stringliteral">&quot;How many probesets to use for determining prior.&quot;</span>,
<a name="l00316"></a>00316                     <span class="stringliteral">&quot;0&quot;</span>);
<a name="l00317"></a>00317   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;list-sample&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00318"></a>00318                     <span class="stringliteral">&quot;Only sample for prior from list specified via --probeset-ids, not entire chip.&quot;</span>,
<a name="l00319"></a>00319                     <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00320"></a>00320   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;read-priors-brlmm&quot;</span>, PgOpt::STRING_OPT,
<a name="l00321"></a>00321                     <span class="stringliteral">&quot;File to load BRLMM priors from. Prior format is tab separated id, center, var, and center.var.&quot;</span>,
<a name="l00322"></a>00322                     <span class="stringliteral">&quot;&quot;</span>);
<a name="l00323"></a>00323   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;write-prior&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00324"></a>00324                     <span class="stringliteral">&quot;Write prior out to file in output-dir.&quot;</span>,
<a name="l00325"></a>00325                     <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00326"></a>00326   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;norm-size&quot;</span>, PgOpt::INT_OPT,
<a name="l00327"></a>00327                     <span class="stringliteral">&quot;Do contrast normalization using a sample of this many snps (brlmm-p)&quot;</span>,
<a name="l00328"></a>00328                     <span class="stringliteral">&quot;0&quot;</span>);
<a name="l00329"></a>00329   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;write-norm&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00330"></a>00330                     <span class="stringliteral">&quot;Write covariate norm fcns to file&quot;</span>,
<a name="l00331"></a>00331                     <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00332"></a>00332   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;set-analysis-name&quot;</span>, PgOpt::STRING_OPT,
<a name="l00333"></a>00333                   <span class="stringliteral">&quot;Explicitly set the analysis name. This affects output file names (ie prefix) and various meta info.&quot;</span>,
<a name="l00334"></a>00334                     <span class="stringliteral">&quot;&quot;</span>);
<a name="l00335"></a>00335   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;include-quant-in-report-file-name&quot;</span>, PgOpt::STRING_OPT,
<a name="l00336"></a>00336                   <span class="stringliteral">&quot;Include the quant method name in the expression report files. &quot;</span>,
<a name="l00337"></a>00337                     <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00338"></a>00338 
<a name="l00339"></a>00339   defineOptionSection(<span class="stringliteral">&quot;Gender Options&quot;</span>);
<a name="l00340"></a>00340 
<a name="l00341"></a>00341   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;set-gender-method&quot;</span>, PgOpt::STRING_OPT,
<a name="l00342"></a>00342                     <span class="stringliteral">&quot;Explicitly force the use of a particular gender method for genotype calling. &quot;</span>
<a name="l00343"></a>00343                     <span class="stringliteral">&quot;Valid values include: cn-probe-chrXY-ratio, dm-chrX-het-rate, &quot;</span>
<a name="l00344"></a>00344                     <span class="stringliteral">&quot;em-cluster-chrX-het-contrast, user-supplied, and none. &quot;</span>
<a name="l00345"></a>00345                     <span class="stringliteral">&quot;If you are supplied seed genotype calls, you can also use &quot;</span>
<a name="l00346"></a>00346                     <span class="stringliteral">&quot;supplied-genotypes-chrX-het-rate. When not set, the default behavior depends on the analysis.&quot;</span>,
<a name="l00347"></a>00347                     <span class="stringliteral">&quot;&quot;</span>);
<a name="l00348"></a>00348   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;read-genders&quot;</span>, PgOpt::STRING_OPT,
<a name="l00349"></a>00349                     <span class="stringliteral">&quot;Explicitly read genders from a file.&quot;</span>,
<a name="l00350"></a>00350                     <span class="stringliteral">&quot;&quot;</span>);
<a name="l00351"></a>00351   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;read-inbred&quot;</span>, PgOpt::STRING_OPT,
<a name="l00352"></a>00352                     <span class="stringliteral">&quot;Read penalty for hets by level of inbreeding per sample.&quot;</span>,
<a name="l00353"></a>00353                     <span class="stringliteral">&quot;&quot;</span>);
<a name="l00354"></a>00354   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;no-gender-force&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00355"></a>00355                     <span class="stringliteral">&quot;Perform analysis even without a suitable gender method for genotype calling.&quot;</span>,
<a name="l00356"></a>00356                     <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00357"></a>00357   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;em-gender&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00358"></a>00358                     <span class="stringliteral">&quot;Enable EM Gender calling if special-snps or chrX-snp file is provided.&quot;</span>,
<a name="l00359"></a>00359                     <span class="stringliteral">&quot;true&quot;</span>);
<a name="l00360"></a>00360   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;female-thresh&quot;</span>, PgOpt::DOUBLE_OPT,
<a name="l00361"></a>00361                     <span class="stringliteral">&quot;Threshold for calling females when using cn-probe-chrXY-ratio method.&quot;</span>,
<a name="l00362"></a>00362                     <span class="stringliteral">&quot;0.48&quot;</span>);
<a name="l00363"></a>00363   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;male-thresh&quot;</span>, PgOpt::DOUBLE_OPT,
<a name="l00364"></a>00364                     <span class="stringliteral">&quot;Threshold for calling females when using cn-probe-chrXY-ratio method.&quot;</span>,
<a name="l00365"></a>00365                     <span class="stringliteral">&quot;0.71&quot;</span>);
<a name="l00366"></a>00366 
<a name="l00367"></a>00367   defineOptionSection(<span class="stringliteral">&quot;Misc Options&quot;</span>);
<a name="l00368"></a>00368   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;explain&quot;</span>, PgOpt::STRING_OPT,
<a name="l00369"></a>00369                     <span class="stringliteral">&quot;Explain a particular operation (i.e. --explain brlmm or --explain brlmm-p).&quot;</span>,
<a name="l00370"></a>00370                     <span class="stringliteral">&quot;&quot;</span>);
<a name="l00371"></a>00371 
<a name="l00372"></a>00372   defineOptionSection(<span class="stringliteral">&quot;Advanced Options&quot;</span>);
<a name="l00373"></a>00373   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;kill-list&quot;</span>, PgOpt::STRING_OPT,
<a name="l00374"></a>00374                     <span class="stringliteral">&quot;Do not use the probes specified in file for computing results. [experimental]&quot;</span>,
<a name="l00375"></a>00375                     <span class="stringliteral">&quot;&quot;</span>);
<a name="l00376"></a>00376   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;dm-out&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00377"></a>00377                     <span class="stringliteral">&quot;Output any initial seed calls used by BRLMM (seed default is DM calls). Only relevant for BRLMM.&quot;</span>,
<a name="l00378"></a>00378                     <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00379"></a>00379   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;all-types&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00380"></a>00380                     <span class="stringliteral">&quot;Try and analyze all probeset types rather than just genotyping. &quot;</span>
<a name="l00381"></a>00381                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00382"></a>00382                     <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00383"></a>00383   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;genotypes&quot;</span>, PgOpt::STRING_OPT,
<a name="l00384"></a>00384                      <span class="stringliteral">&quot;File to read seed genotypes from instead of using DM to generate. [experimental]&quot;</span>,
<a name="l00385"></a>00385                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00386"></a>00386   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;select-probes&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00387"></a>00387                     <span class="stringliteral">&quot;Output estimates of which probes are most accurate&quot;</span>,
<a name="l00388"></a>00388                     <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00389"></a>00389   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;call-coder-max-alleles&quot;</span>, PgOpt::INT_OPT,
<a name="l00390"></a>00390                     <span class="stringliteral">&quot;For encoding/decoding calls, the max number of alleles per marker to allow.&quot;</span>,
<a name="l00391"></a>00391                     <span class="stringliteral">&quot;6&quot;</span>);
<a name="l00392"></a>00392   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;call-coder-type&quot;</span>, PgOpt::STRING_OPT,
<a name="l00393"></a>00393                     <span class="stringliteral">&quot;The data size used to encode the call.&quot;</span>,
<a name="l00394"></a>00394                     <span class="stringliteral">&quot;UCHAR&quot;</span>);
<a name="l00395"></a>00395   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;call-coder-version&quot;</span>, PgOpt::STRING_OPT,
<a name="l00396"></a>00396                     <span class="stringliteral">&quot;The version of the encoder/decoder to use&quot;</span>,
<a name="l00397"></a>00397                     <span class="stringliteral">&quot;1.0&quot;</span>);
<a name="l00398"></a>00398 
<a name="l00399"></a>00399   defineOptionSection(<span class="stringliteral">&quot;Execution Control Options&quot;</span>);
<a name="l00400"></a>00400   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;use-disk&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00401"></a>00401                      <span class="stringliteral">&quot;Store CEL intensities to be analyzed on disk.&quot;</span>, <span class="stringliteral">&quot;true&quot;</span>);
<a name="l00402"></a>00402   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;disk-cache&quot;</span>, PgOpt::INT_OPT,
<a name="l00403"></a>00403                      <span class="stringliteral">&quot;Size of memory cache when working off disk in megabytes.&quot;</span>,
<a name="l00404"></a>00404                      <span class="stringliteral">&quot;50&quot;</span>);
<a name="l00405"></a>00405 
<a name="l00406"></a>00406   defineOptionSection(<span class="stringliteral">&quot;A5 output options&quot;</span>);
<a name="l00407"></a>00407 
<a name="l00408"></a>00408   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-global-file&quot;</span>,PgOpt::STRING_OPT,
<a name="l00409"></a>00409                      <span class="stringliteral">&quot;Filename for the A5 global output file. &quot;</span>
<a name="l00410"></a>00410                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00411"></a>00411                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00412"></a>00412   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-global-file-no-replace&quot;</span>,PgOpt::BOOL_OPT,
<a name="l00413"></a>00413                      <span class="stringliteral">&quot;Append or create rather than replace. &quot;</span>
<a name="l00414"></a>00414                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00415"></a>00415                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00416"></a>00416   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-group&quot;</span>,PgOpt::STRING_OPT,
<a name="l00417"></a>00417                      <span class="stringliteral">&quot;Group name where to put results in the A5 output files. &quot;</span>
<a name="l00418"></a>00418                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00419"></a>00419                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00420"></a>00420   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-calls&quot;</span>,PgOpt::BOOL_OPT,
<a name="l00421"></a>00421                      <span class="stringliteral">&quot;Output the genotype calls and confidences in A5 format. &quot;</span>
<a name="l00422"></a>00422                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00423"></a>00423                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00424"></a>00424   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-calls-use-global&quot;</span>,PgOpt::BOOL_OPT,
<a name="l00425"></a>00425                      <span class="stringliteral">&quot;Use the global A5 file for calls and confidences.&quot;</span>
<a name="l00426"></a>00426                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00427"></a>00427                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00428"></a>00428   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;a5-summaries&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00429"></a>00429                      <span class="stringliteral">&quot;Output the summary values from the quantifcation method for each allele in A5 format. &quot;</span>
<a name="l00430"></a>00430                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00431"></a>00431                     <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00432"></a>00432   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-summaries-use-global&quot;</span>,PgOpt::BOOL_OPT,
<a name="l00433"></a>00433                      <span class="stringliteral">&quot;Use the global A5 file for summaries. &quot;</span>
<a name="l00434"></a>00434                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00435"></a>00435                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00436"></a>00436   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-feature-effects&quot;</span>,PgOpt::BOOL_OPT,
<a name="l00437"></a>00437                      <span class="stringliteral">&quot;Output feature effects in A5 format. &quot;</span>
<a name="l00438"></a>00438                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00439"></a>00439                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00440"></a>00440   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-feature-effects-use-global&quot;</span>,PgOpt::BOOL_OPT,
<a name="l00441"></a>00441                      <span class="stringliteral">&quot;Use the global A5 file for feature effects.&quot;</span>
<a name="l00442"></a>00442                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00443"></a>00443                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00444"></a>00444   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-feature-details&quot;</span>,PgOpt::BOOL_OPT,
<a name="l00445"></a>00445                      <span class="stringliteral">&quot;Output feature level residuals in A5 format. &quot;</span>
<a name="l00446"></a>00446                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00447"></a>00447                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00448"></a>00448   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-feature-details-use-global&quot;</span>,PgOpt::BOOL_OPT,
<a name="l00449"></a>00449                      <span class="stringliteral">&quot;Use the global A5 file for residuals. &quot;</span>
<a name="l00450"></a>00450                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00451"></a>00451                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00452"></a>00452   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-sketch&quot;</span>,PgOpt::BOOL_OPT,
<a name="l00453"></a>00453                      <span class="stringliteral">&quot;Output normalization sketch in A5 format. &quot;</span>
<a name="l00454"></a>00454                      <span class="stringliteral">&quot;--write-sketch option will override this option. &quot;</span>
<a name="l00455"></a>00455                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00456"></a>00456                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00457"></a>00457   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-sketch-use-global&quot;</span>,PgOpt::BOOL_OPT,
<a name="l00458"></a>00458                      <span class="stringliteral">&quot;Put the sketch in the global A5 output file. &quot;</span>
<a name="l00459"></a>00459                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00460"></a>00460                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00461"></a>00461   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-write-models&quot;</span>,PgOpt::BOOL_OPT,
<a name="l00462"></a>00462                      <span class="stringliteral">&quot;Output genotype models/posteriors in A5 format. &quot;</span>
<a name="l00463"></a>00463                      <span class="stringliteral">&quot;--write-models option will override this option. &quot;</span>
<a name="l00464"></a>00464                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00465"></a>00465                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00466"></a>00466   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-write-models-use-global&quot;</span>,PgOpt::BOOL_OPT,
<a name="l00467"></a>00467                      <span class="stringliteral">&quot;Put the models in the global A5 output file. &quot;</span>
<a name="l00468"></a>00468                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00469"></a>00469                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00470"></a>00470   
<a name="l00471"></a>00471   defineOptionSection(<span class="stringliteral">&quot;A5 input options&quot;</span>);
<a name="l00472"></a>00472 
<a name="l00473"></a>00473   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-global-input-file&quot;</span>,PgOpt::STRING_OPT,
<a name="l00474"></a>00474                      <span class="stringliteral">&quot;Filename for the group in the global input file.&quot;</span>
<a name="l00475"></a>00475                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00476"></a>00476                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00477"></a>00477   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-input-group&quot;</span>,PgOpt::STRING_OPT,
<a name="l00478"></a>00478                      <span class="stringliteral">&quot;Group name for input. Defaults to --a5-group or if that &quot;</span>
<a name="l00479"></a>00479                      <span class="stringliteral">&quot;is not set, then &#39;/&#39;. &quot;</span>
<a name="l00480"></a>00480                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00481"></a>00481                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00482"></a>00482 
<a name="l00483"></a>00483   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-sketch-input-global&quot;</span>,PgOpt::BOOL_OPT,
<a name="l00484"></a>00484                      <span class="stringliteral">&quot;Read the sketch from the global A5 input file. &quot;</span>
<a name="l00485"></a>00485                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00486"></a>00486                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00487"></a>00487   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-sketch-input-file&quot;</span>,PgOpt::STRING_OPT,
<a name="l00488"></a>00488                      <span class="stringliteral">&quot;Read the sketch from the an A5 input file. &quot;</span>
<a name="l00489"></a>00489                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00490"></a>00490                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00491"></a>00491   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-sketch-input-group&quot;</span>,PgOpt::STRING_OPT,
<a name="l00492"></a>00492                      <span class="stringliteral">&quot;Group name to read the sketch from. Defaults to --a5-input-group. &quot;</span>
<a name="l00493"></a>00493                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00494"></a>00494                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00495"></a>00495   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-sketch-input-name&quot;</span>,PgOpt::STRING_OPT,
<a name="l00496"></a>00496                      <span class="stringliteral">&quot;The name of the data section. Defaults to &#39;target-sketch&#39;. &quot;</span>
<a name="l00497"></a>00497                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00498"></a>00498                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00499"></a>00499 
<a name="l00500"></a>00500   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-feature-effects-input-global&quot;</span>,PgOpt::BOOL_OPT,
<a name="l00501"></a>00501                      <span class="stringliteral">&quot;Read the feature effects global A5 input file. &quot;</span>
<a name="l00502"></a>00502                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00503"></a>00503                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00504"></a>00504   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-feature-effects-input-file&quot;</span>,PgOpt::STRING_OPT,
<a name="l00505"></a>00505                      <span class="stringliteral">&quot;Read the feature effects from the an A5 input file. &quot;</span>
<a name="l00506"></a>00506                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00507"></a>00507                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00508"></a>00508   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-feature-effects-input-group&quot;</span>,PgOpt::STRING_OPT,
<a name="l00509"></a>00509                      <span class="stringliteral">&quot;Group name to read the feature effects from. Defaults to --a5-input-group. &quot;</span>
<a name="l00510"></a>00510                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00511"></a>00511                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00512"></a>00512   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-feature-effects-input-name&quot;</span>,PgOpt::STRING_OPT,
<a name="l00513"></a>00513                      <span class="stringliteral">&quot;The name of the data section. Defaults to XXX.feature-response &quot;</span>
<a name="l00514"></a>00514                      <span class="stringliteral">&quot;where XXX is the analysis name and quant method. IE &#39;brlmm-p.plier&#39;. &quot;</span>
<a name="l00515"></a>00515                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00516"></a>00516                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00517"></a>00517 
<a name="l00518"></a>00518   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-models-input-global&quot;</span>,PgOpt::BOOL_OPT,
<a name="l00519"></a>00519                      <span class="stringliteral">&quot;Read the Models from the global A5 input file. &quot;</span>
<a name="l00520"></a>00520                      <span class="stringliteral">&quot;The tsv5 name must be &#39;XXX.snp-posteriors&#39;. &quot;</span> <span class="comment">// jhg</span>
<a name="l00521"></a>00521                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00522"></a>00522                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00523"></a>00523   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-models-input-file&quot;</span>,PgOpt::STRING_OPT,
<a name="l00524"></a>00524                      <span class="stringliteral">&quot;Read the models from the an A5 input file. &quot;</span>
<a name="l00525"></a>00525                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00526"></a>00526                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00527"></a>00527   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-models-input-group&quot;</span>,PgOpt::STRING_OPT,
<a name="l00528"></a>00528                      <span class="stringliteral">&quot;The group name where the models are located. Defaults to &quot;</span>
<a name="l00529"></a>00529                      <span class="stringliteral">&quot;the analysis name. &quot;</span>
<a name="l00530"></a>00530                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00531"></a>00531                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00532"></a>00532   defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;a5-models-input-name&quot;</span>,PgOpt::STRING_OPT,
<a name="l00533"></a>00533                      <span class="stringliteral">&quot;The name of the data section. Defaults to XXX.snp-posteriors &quot;</span>
<a name="l00534"></a>00534                      <span class="stringliteral">&quot;where XXX is the analysis name. IE &#39;brlmm-p&#39;. &quot;</span>
<a name="l00535"></a>00535                      <span class="stringliteral">&quot;[Experimental]&quot;</span>,
<a name="l00536"></a>00536                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00537"></a>00537 
<a name="l00538"></a>00538   defineOptionSection(<span class="stringliteral">&quot;Engine Options (Not used on command line)&quot;</span>);
<a name="l00539"></a>00539   defOptMult(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;cels&quot;</span>, PgOpt::STRING_OPT,
<a name="l00540"></a>00540                      <span class="stringliteral">&quot;Cel files to process.&quot;</span>,
<a name="l00541"></a>00541                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00542"></a>00542   defOptMult(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;result-files&quot;</span>, PgOpt::STRING_OPT,
<a name="l00543"></a>00543                      <span class="stringliteral">&quot;CHP file names to output. Must be paired with cels.&quot;</span>,
<a name="l00544"></a>00544                      <span class="stringliteral">&quot;&quot;</span>);
<a name="l00545"></a>00545 }
<a name="l00546"></a>00546 
<a name="l00547"></a>00547 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::defineStates() {
<a name="l00548"></a>00548   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;analysis-count&quot;</span>, PgOpt::INT_OPT,
<a name="l00549"></a>00549                      <span class="stringliteral">&quot;The number of user-specified analysis streams.&quot;</span>,
<a name="l00550"></a>00550                      <span class="stringliteral">&quot;-1&quot;</span>);
<a name="l00551"></a>00551   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;num-rows&quot;</span>, PgOpt::INT_OPT,
<a name="l00552"></a>00552                      <span class="stringliteral">&quot;The number of rows on the chip.&quot;</span>,
<a name="l00553"></a>00553                      <span class="stringliteral">&quot;-1&quot;</span>);
<a name="l00554"></a>00554   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;num-cols&quot;</span>, PgOpt::INT_OPT,
<a name="l00555"></a>00555                      <span class="stringliteral">&quot;The number of cols on the chip.&quot;</span>,
<a name="l00556"></a>00556                      <span class="stringliteral">&quot;-1&quot;</span>);
<a name="l00557"></a>00557   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;probe-count&quot;</span>, PgOpt::INT_OPT,
<a name="l00558"></a>00558                      <span class="stringliteral">&quot;The number of probes on the chip.&quot;</span>,
<a name="l00559"></a>00559                      <span class="stringliteral">&quot;-1&quot;</span>);
<a name="l00560"></a>00560   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;probeset-count&quot;</span>, PgOpt::INT_OPT,
<a name="l00561"></a>00561                      <span class="stringliteral">&quot;The number of probesets on the chip.&quot;</span>,
<a name="l00562"></a>00562                      <span class="stringliteral">&quot;-1&quot;</span>);
<a name="l00563"></a>00563   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;channel-count&quot;</span>, PgOpt::INT_OPT,
<a name="l00564"></a>00564                      <span class="stringliteral">&quot;The number of data channels in the CEL file.&quot;</span>,
<a name="l00565"></a>00565                      <span class="stringliteral">&quot;-1&quot;</span>);
<a name="l00566"></a>00566   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;need-gc&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00567"></a>00567                      <span class="stringliteral">&quot;Do we need GC probes for the analysis.&quot;</span>,
<a name="l00568"></a>00568                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00569"></a>00569   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;need-mismatch&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00570"></a>00570                      <span class="stringliteral">&quot;Do we need mismatch probes for the analysis.&quot;</span>,
<a name="l00571"></a>00571                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00572"></a>00572   defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;need-allelematch&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00573"></a>00573                      <span class="stringliteral">&quot;Do we need matching allele probes for the analysis.&quot;</span>,
<a name="l00574"></a>00574                      <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00575"></a>00575 }
<a name="l00576"></a>00576 <span class="comment"></span>
<a name="l00577"></a>00577 <span class="comment">/**</span>
<a name="l00578"></a>00578 <span class="comment"> * Check the special-snp-version field in birdseed prior file. Call</span>
<a name="l00579"></a>00579 <span class="comment"> * Err::errAbort() if mismatch.</span>
<a name="l00580"></a>00580 <span class="comment"> *</span>
<a name="l00581"></a>00581 <span class="comment"> * @param path           - The birdseed prior file name</span>
<a name="l00582"></a>00582 <span class="comment"> * @param specialSnpType - The chiptype expected</span>
<a name="l00583"></a>00583 <span class="comment"> */</span>
<a name="l00584"></a>00584 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::checkBirdseedTsvPriorFile(<span class="keyword">const</span> <span class="keywordtype">string</span> &amp;path, <span class="keyword">const</span> <span class="keywordtype">string</span> &amp;specialSnpType) {
<a name="l00585"></a>00585   <a class="code" href="classaffx_1_1TsvFile.html" title="A class for reading and writing Tab Seperated Value (TSV) files. /// See the TsvFile format document ...">affx::TsvFile</a> tsv;
<a name="l00586"></a>00586   <span class="keywordtype">string</span> chip;
<a name="l00587"></a>00587   <span class="keywordtype">int</span> specialSeen = 0;
<a name="l00588"></a>00588   <span class="keywordflow">if</span>(tsv.<a class="code" href="classaffx_1_1TsvFile.html#a763ce0bbf19b4134d0445aa58e0f5011" title="Opens a file -- attempts to guess some defaults.">open</a>(path) != TSV_OK)
<a name="l00589"></a>00589     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Couldn&#39;t open file: &#39;&quot;</span> + path + <span class="stringliteral">&quot;&#39; to read.&quot;</span>);
<a name="l00590"></a>00590   <span class="keywordtype">bool</span> specialSnpMatch = specialSnpType.empty(); <span class="comment">// treat empty as wildcard</span>
<a name="l00591"></a>00591   <span class="keywordflow">while</span>(tsv.<a class="code" href="classaffx_1_1TsvFile.html#adabcde206ca5a68ea3a983dfaca0d8f0" title="Advance to the next header matching key ///.">headersFindNext</a>(<span class="stringliteral">&quot;special-snp-version&quot;</span>, chip) == TSV_OK) {
<a name="l00592"></a>00592     specialSeen++;
<a name="l00593"></a>00593     <span class="keywordflow">if</span>(chip == specialSnpType)
<a name="l00594"></a>00594       specialSnpMatch = <span class="keyword">true</span>;
<a name="l00595"></a>00595   }
<a name="l00596"></a>00596   <span class="keywordflow">if</span>(!specialSnpMatch &amp;&amp; specialSeen &gt; 0) {
<a name="l00597"></a>00597     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>( <span class="stringliteral">&quot;Special Snp File: &#39;&quot;</span> + specialSnpType + <span class="stringliteral">&quot;&#39; not supported by prior file: &#39;&quot;</span> + path + <span class="stringliteral">&quot;&#39;.&quot;</span>);
<a name="l00598"></a>00598   }
<a name="l00599"></a>00599 }
<a name="l00600"></a>00600 <span class="comment"></span>
<a name="l00601"></a>00601 <span class="comment">/**</span>
<a name="l00602"></a>00602 <span class="comment"> * Make sure that our options are sane. Call Err::errAbort if not.</span>
<a name="l00603"></a>00603 <span class="comment"> */</span>
<a name="l00604"></a>00604 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::checkOptionsImp() {
<a name="l00605"></a>00605 
<a name="l00606"></a>00606   defineStates();
<a name="l00607"></a>00607 
<a name="l00608"></a>00608   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;cdf-file&quot;</span>);
<a name="l00609"></a>00609   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;spf-file&quot;</span>);
<a name="l00610"></a>00610   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;chrX-snps&quot;</span>);
<a name="l00611"></a>00611   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;special-snps&quot;</span>);
<a name="l00612"></a>00612   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;chrX-probes&quot;</span>);
<a name="l00613"></a>00613   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;chrY-probes&quot;</span>);
<a name="l00614"></a>00614   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;probeset-ids&quot;</span>);
<a name="l00615"></a>00615   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;probeset-ids-reported&quot;</span>);
<a name="l00616"></a>00616   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;probe-class-file&quot;</span>);
<a name="l00617"></a>00617   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;annotation-file&quot;</span>);
<a name="l00618"></a>00618   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;genotype-markers-cn-file&quot;</span>);
<a name="l00619"></a>00619   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;read-models-brlmm&quot;</span>);
<a name="l00620"></a>00620   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;read-models-brlmmp&quot;</span>);
<a name="l00621"></a>00621   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;read-models-birdseed&quot;</span>);
<a name="l00622"></a>00622   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;use-feat-eff&quot;</span>);
<a name="l00623"></a>00623   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;target-sketch&quot;</span>);
<a name="l00624"></a>00624   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;reference-profile&quot;</span>);
<a name="l00625"></a>00625   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;read-priors-brlmm&quot;</span>);
<a name="l00626"></a>00626   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;kill-list&quot;</span>);
<a name="l00627"></a>00627   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;a5-global-input-file&quot;</span>);
<a name="l00628"></a>00628   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;a5-sketch-input-file&quot;</span>);
<a name="l00629"></a>00629   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;a5-feature-effects-input-file&quot;</span>);
<a name="l00630"></a>00630   <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;a5-models-input-file&quot;</span>);
<a name="l00631"></a>00631 
<a name="l00632"></a>00632   <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;explain&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>) { explain(); exit(0); }
<a name="l00633"></a>00633 
<a name="l00634"></a>00634   <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;out-dir&quot;</span>) == <span class="stringliteral">&quot;&quot;</span>) {<a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Must specify an output directory.&quot;</span>);}
<a name="l00635"></a>00635   <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;temp-dir&quot;</span>) == <span class="stringliteral">&quot;&quot;</span>) { <a class="code" href="classOptions.html#aaf215b45e7097fcedae68c15073eb0f8" title="Set option value.">setOpt</a>(<span class="stringliteral">&quot;temp-dir&quot;</span>, <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;out-dir&quot;</span>), <span class="stringliteral">&quot;temp&quot;</span>)); }
<a name="l00636"></a>00636 
<a name="l00637"></a>00637   <span class="keywordtype">string</span> cdfFile = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;cdf-file&quot;</span>);
<a name="l00638"></a>00638   <span class="keywordtype">string</span> spfFile = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;spf-file&quot;</span>);
<a name="l00639"></a>00639   <span class="keywordtype">string</span> killList = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;kill-list&quot;</span>);
<a name="l00640"></a>00640   <span class="keywordtype">string</span> specialSnps = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;special-snps&quot;</span>);
<a name="l00641"></a>00641   <span class="keywordtype">string</span> chrXSnps = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chrX-snps&quot;</span>);
<a name="l00642"></a>00642   <span class="keywordtype">string</span> chrXProbes = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chrX-probes&quot;</span>);
<a name="l00643"></a>00643   <span class="keywordtype">string</span> chrYProbes = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chrY-probes&quot;</span>);
<a name="l00644"></a>00644   vector&lt;string&gt; analysis = <a class="code" href="classOptions.html#ab20d0083424d9d9a8b94da46f487f57a" title="Return a vector of values for a particular option.">getOptVector</a>(<span class="stringliteral">&quot;analysis&quot;</span>);
<a name="l00645"></a>00645 
<a name="l00646"></a>00646   <span class="keywordflow">if</span>(analysis.size() == 0) {
<a name="l00647"></a>00647       analysis.push_back(<span class="stringliteral">&quot;brlmm&quot;</span>);
<a name="l00648"></a>00648       <a class="code" href="classOptions.html#aaf215b45e7097fcedae68c15073eb0f8" title="Set option value.">setOpt</a>(<span class="stringliteral">&quot;analysis&quot;</span>,analysis);
<a name="l00649"></a>00649   }
<a name="l00650"></a>00650   <span class="keywordflow">if</span>(analysis.size() != 1) {
<a name="l00651"></a>00651       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Only one analysis is currently permitted.&quot;</span>);
<a name="l00652"></a>00652   }
<a name="l00653"></a>00653   <a class="code" href="classOptions.html#aaf215b45e7097fcedae68c15073eb0f8" title="Set option value.">setOpt</a>(<span class="stringliteral">&quot;analysis-count&quot;</span>, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(analysis.size()));
<a name="l00654"></a>00654 
<a name="l00655"></a>00655   <span class="comment">// Check that a global file was given if the user wants global a5 output</span>
<a name="l00656"></a>00656   <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-global-file&quot;</span>) == <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00657"></a>00657       vector&lt;string&gt; optionNames;
<a name="l00658"></a>00658       getOptionNames(optionNames);
<a name="l00659"></a>00659       <span class="keywordflow">for</span>(<span class="keywordtype">size_t</span> i=0; i&lt;optionNames.size(); i++) {
<a name="l00660"></a>00660           <span class="keywordtype">string</span> name = optionNames[i];
<a name="l00661"></a>00661           <span class="keywordflow">if</span>( name.find(<span class="stringliteral">&quot;use-global&quot;</span>) != name.npos ) {
<a name="l00662"></a>00662               <span class="keywordflow">if</span>( <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(name) ) {
<a name="l00663"></a>00663                   <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;--&quot;</span> + name + <span class="stringliteral">&quot; option given, but no global file. Must specify --a5-global-file.&quot;</span>);
<a name="l00664"></a>00664               }
<a name="l00665"></a>00665           }
<a name="l00666"></a>00666       }
<a name="l00667"></a>00667   }
<a name="l00668"></a>00668   <span class="comment">// Check that a global input file was given if the user wants global a5 input</span>
<a name="l00669"></a>00669   <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-global-input-file&quot;</span>) == <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00670"></a>00670       vector&lt;string&gt; optionNames;
<a name="l00671"></a>00671       getOptionNames(optionNames);
<a name="l00672"></a>00672       <span class="keywordflow">for</span>(<span class="keywordtype">size_t</span> i=0; i&lt;optionNames.size(); i++) {
<a name="l00673"></a>00673           <span class="keywordtype">string</span> name = optionNames[i];
<a name="l00674"></a>00674           <span class="keywordflow">if</span>( name.find(<span class="stringliteral">&quot;input-global&quot;</span>) != name.npos ) {
<a name="l00675"></a>00675               <span class="keywordflow">if</span>( <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(name) ){
<a name="l00676"></a>00676                   <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;--&quot;</span> + name + <span class="stringliteral">&quot; option given, but no global input file. Must specify --a5-global-input-file.&quot;</span>);
<a name="l00677"></a>00677               }
<a name="l00678"></a>00678           }
<a name="l00679"></a>00679       }
<a name="l00680"></a>00680   }
<a name="l00681"></a>00681   <span class="comment">// make sure global input and output files are different</span>
<a name="l00682"></a>00682   <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-global-input-file&quot;</span>) == <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-global-file&quot;</span>))
<a name="l00683"></a>00683       <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-global-input-file&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>)
<a name="l00684"></a>00684           <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Unable to use the same A5 file for global input and output.&quot;</span>);
<a name="l00685"></a>00685 
<a name="l00686"></a>00686   <span class="comment">/* Read in cel file list from other file if specified. */</span>
<a name="l00687"></a>00687   vector&lt;string&gt; celFiles;
<a name="l00688"></a>00688   EngineUtil::getCelFiles(celFiles, <span class="keyword">this</span>);
<a name="l00689"></a>00689   <span class="keywordflow">if</span>(celFiles.size() == 0)
<a name="l00690"></a>00690       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;No cel files specified.&quot;</span>);
<a name="l00691"></a>00691   <a class="code" href="classOptions.html#aaf215b45e7097fcedae68c15073eb0f8" title="Set option value.">setOpt</a>(<span class="stringliteral">&quot;cels&quot;</span>,celFiles);
<a name="l00692"></a>00692 
<a name="l00693"></a>00693 
<a name="l00694"></a>00694   vector&lt;string&gt; resultFiles = <a class="code" href="classOptions.html#ab20d0083424d9d9a8b94da46f487f57a" title="Return a vector of values for a particular option.">getOptVector</a>(<span class="stringliteral">&quot;result-files&quot;</span>);
<a name="l00695"></a>00695   <span class="keywordflow">if</span>(resultFiles.size() &gt; 0) {
<a name="l00696"></a>00696       <span class="keywordflow">if</span>(celFiles.size() != resultFiles.size())
<a name="l00697"></a>00697         <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;result-files option used but is not the same size as cel file listing&quot;</span>);
<a name="l00698"></a>00698       <span class="keywordtype">bool</span> ccOut = <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;cc-chp-output&quot;</span>);
<a name="l00699"></a>00699       <span class="keywordtype">bool</span> xdaOut = <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;xda-chp-output&quot;</span>);
<a name="l00700"></a>00700       <span class="keywordflow">if</span>(ccOut &amp;&amp; xdaOut) 
<a name="l00701"></a>00701           <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;cannot use multiple CHP output formats with the result-files option&quot;</span>);
<a name="l00702"></a>00702   }
<a name="l00703"></a>00703 
<a name="l00704"></a>00704   <span class="comment">// If we&#39;re doing birdseed or brlmm-p we can save space by not loading the mismatch and gc mappings.</span>
<a name="l00705"></a>00705   <span class="keywordtype">bool</span> needGc = <span class="keyword">false</span>, needMisMatch = <span class="keyword">false</span>, needAlleleMatch = <span class="keyword">false</span>;
<a name="l00706"></a>00706   <span class="keywordflow">for</span>(<span class="keywordtype">size_t</span> aIx = 0; aIx &lt; analysis.size(); aIx++) {<span class="comment"></span>
<a name="l00707"></a>00707 <span class="comment">      ///@todo would be nice to query AnalysisStreamFactory for this info</span>
<a name="l00708"></a>00708 <span class="comment"></span>      <span class="keywordtype">string</span> analysisName = analysis[aIx];
<a name="l00709"></a>00709       map&lt;string,string&gt;::iterator iter;
<a name="l00710"></a>00710       <span class="keywordflow">for</span>(iter = m_stdMethods.begin(); iter != m_stdMethods.end(); iter++) {
<a name="l00711"></a>00711           <span class="keywordflow">if</span>(iter-&gt;first == analysisName) {
<a name="l00712"></a>00712               analysisName = iter-&gt;second;
<a name="l00713"></a>00713           }
<a name="l00714"></a>00714       }
<a name="l00715"></a>00715       <span class="keywordflow">if</span>(analysis[aIx].find(<span class="stringliteral">&quot;pm-mm&quot;</span>) != string::npos)
<a name="l00716"></a>00716           needMisMatch = <span class="keyword">true</span>;
<a name="l00717"></a>00717       <span class="keywordflow">if</span>(analysis[aIx].find(<span class="stringliteral">&quot;pm-gcbg&quot;</span>) != string::npos)
<a name="l00718"></a>00718           needGc = <span class="keyword">true</span>;
<a name="l00719"></a>00719       <span class="keywordflow">if</span>(analysis[aIx].find(<span class="stringliteral">&quot;gc-bg&quot;</span>) != string::npos)
<a name="l00720"></a>00720           needGc = <span class="keyword">true</span>;
<a name="l00721"></a>00721       <span class="keywordflow">if</span>(analysis[aIx].find(<span class="stringliteral">&quot;pm-sum&quot;</span>) != string::npos)
<a name="l00722"></a>00722           needAlleleMatch = <span class="keyword">true</span>;
<a name="l00723"></a>00723 <span class="comment"></span>
<a name="l00724"></a>00724 <span class="comment">      ///@todo need to deal with DM listener, setting needMisMatch appropriately</span>
<a name="l00725"></a>00725 <span class="comment">      ///      this is a crude hack to get this. There may be cases where DM </span>
<a name="l00726"></a>00726 <span class="comment">      ///      listener is run even though brlmm was not selected (and cases where</span>
<a name="l00727"></a>00727 <span class="comment">      ///      brlmm is run but DM seeds are not used).</span>
<a name="l00728"></a>00728 <span class="comment"></span>      <span class="keywordflow">if</span>(analysis[aIx].find(<span class="stringliteral">&quot;,brlmm.&quot;</span>) != string::npos)
<a name="l00729"></a>00729           needMisMatch = <span class="keyword">true</span>;
<a name="l00730"></a>00730   }
<a name="l00731"></a>00731   <a class="code" href="classOptions.html#aaf215b45e7097fcedae68c15073eb0f8" title="Set option value.">setOpt</a>(<span class="stringliteral">&quot;need-gc&quot;</span>, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(needGc));
<a name="l00732"></a>00732   <a class="code" href="classOptions.html#aaf215b45e7097fcedae68c15073eb0f8" title="Set option value.">setOpt</a>(<span class="stringliteral">&quot;need-mismatch&quot;</span>,<a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(needMisMatch));
<a name="l00733"></a>00733   <a class="code" href="classOptions.html#aaf215b45e7097fcedae68c15073eb0f8" title="Set option value.">setOpt</a>(<span class="stringliteral">&quot;need-allelematch&quot;</span>,<a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(needAlleleMatch));
<a name="l00734"></a>00734 
<a name="l00735"></a>00735   <span class="keywordflow">if</span>(cdfFile == <span class="stringliteral">&quot;&quot;</span> &amp;&amp; spfFile == <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00736"></a>00736     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Must specify either a cdf file or spf (simple probe format) file.&quot;</span>);
<a name="l00737"></a>00737   }
<a name="l00738"></a>00738   <span class="keywordflow">if</span>(killList != <span class="stringliteral">&quot;&quot;</span> &amp;&amp; cdfFile == <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00739"></a>00739     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;&#39;--kill-list-file&#39; option only supported on cdf files.&quot;</span>);
<a name="l00740"></a>00740    }
<a name="l00741"></a>00741   <span class="keywordflow">if</span>((chrXSnps  == <span class="stringliteral">&quot;&quot;</span>) &amp;&amp; (specialSnps == <span class="stringliteral">&quot;&quot;</span>) &amp;&amp; ((chrXProbes == <span class="stringliteral">&quot;&quot;</span>) || (chrYProbes == <span class="stringliteral">&quot;&quot;</span>)) &amp;&amp; (<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-genders&quot;</span>) == <span class="stringliteral">&quot;&quot;</span>) &amp;&amp; !<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;no-gender-force&quot;</span>)) {
<a name="l00742"></a>00742     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Have to specify chrX snps, haploid snps, chrX/Y probe file, or a gender file in order to generate genders for genotype calling. Use the --no-gender-force option to force genotype calling.&quot;</span>);
<a name="l00743"></a>00743   }
<a name="l00744"></a>00744   <span class="keywordflow">if</span>(killList != <span class="stringliteral">&quot;&quot;</span> &amp;&amp; <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;xda-chp-output&quot;</span>)) {
<a name="l00745"></a>00745     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Can&#39;t specify a kill list and XDA CHP output at same time.&quot;</span>);
<a name="l00746"></a>00746   }
<a name="l00747"></a>00747   <span class="keywordflow">if</span>(killList != <span class="stringliteral">&quot;&quot;</span> &amp;&amp; <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;cc-chp-output&quot;</span>)) {
<a name="l00748"></a>00748     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Can&#39;t specify a kill list and CC CHP output at same time.&quot;</span>);
<a name="l00749"></a>00749   }
<a name="l00750"></a>00750   <span class="keywordflow">if</span> (chrXProbes != <span class="stringliteral">&quot;&quot;</span> &amp;&amp; chrYProbes == <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00751"></a>00751     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Must provide a chrY Probe File when providing a chrX Probe File.&quot;</span>);
<a name="l00752"></a>00752   }
<a name="l00753"></a>00753   <span class="keywordflow">if</span> (chrXProbes == <span class="stringliteral">&quot;&quot;</span> &amp;&amp; chrYProbes != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00754"></a>00754     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Must provide a chrX Probe File when providing a chrY Probe File.&quot;</span>);
<a name="l00755"></a>00755   }
<a name="l00756"></a>00756   <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;probeset-ids&quot;</span>) != <span class="stringliteral">&quot;&quot;</span> &amp;&amp; <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;xda-chp-output&quot;</span>)) {
<a name="l00757"></a>00757     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Can&#39;t specify a subset and XDA CHP output at same time. Must use all probesets for CHP file output.&quot;</span>);
<a name="l00758"></a>00758   }
<a name="l00759"></a>00759   <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;probeset-ids-reported&quot;</span>) != <span class="stringliteral">&quot;&quot;</span> &amp;&amp; <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;xda-chp-output&quot;</span>)) {
<a name="l00760"></a>00760     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Can&#39;t specify a reporting subset and XDA CHP output at same time. Must use all probesets for CHP file output.&quot;</span>);
<a name="l00761"></a>00761   }
<a name="l00762"></a>00762   <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;probeset-ids-reported&quot;</span>) != <span class="stringliteral">&quot;&quot;</span> &amp;&amp; <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;xda-chp-output&quot;</span>)) {
<a name="l00763"></a>00763     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Can&#39;t specify a reporting subset and AGCC CHP output at same time. Must use all probesets for CHP file output.&quot;</span>);
<a name="l00764"></a>00764   }
<a name="l00765"></a>00765 
<a name="l00766"></a>00766   <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;list-sample&quot;</span>) &amp;&amp; <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;probeset-ids&quot;</span>) == <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00767"></a>00767     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Can&#39;t set listSample without specifying a probeset-ids file.&quot;</span>);
<a name="l00768"></a>00768   }
<a name="l00769"></a>00769 
<a name="l00770"></a>00770   <span class="comment">/* Some sanity checks. */</span>
<a name="l00771"></a>00771   <span class="keywordflow">if</span>(analysis.empty())
<a name="l00772"></a>00772       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Must specify at least one analysis to preform.&quot;</span>);
<a name="l00773"></a>00773 
<a name="l00774"></a>00774   <span class="keywordflow">if</span> ((<a class="code" href="classOptions.html#afcfab86e21cac5bba43e0858bd428e62" title="Get the integer value of an option.">getOptInt</a>(<span class="stringliteral">&quot;prior-size&quot;</span>) == 0) &amp;&amp;
<a name="l00775"></a>00775       (<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-priors-brlmm&quot;</span>) == <span class="stringliteral">&quot;&quot;</span>)) {
<a name="l00776"></a>00776     <a class="code" href="classOptions.html#aaf215b45e7097fcedae68c15073eb0f8" title="Set option value.">setOpt</a>(<span class="stringliteral">&quot;prior-size&quot;</span>, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(10000)); <span class="comment">// our default value.</span>
<a name="l00777"></a>00777   } <span class="keywordflow">else</span> {
<a name="l00778"></a>00778     <a class="code" href="classOptions.html#aaf215b45e7097fcedae68c15073eb0f8" title="Set option value.">setOpt</a>(<span class="stringliteral">&quot;prior-size&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;prior-size&quot;</span>)); <span class="comment">// our default value.</span>
<a name="l00779"></a>00779   }
<a name="l00780"></a>00780   <span class="keywordtype">int</span> numProbesets = 0;
<a name="l00781"></a>00781   <span class="keywordflow">if</span>(cdfFile != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00782"></a>00782     <span class="comment">/* Check to make sure that we aren&#39;t using more probesets for prior than are available. */</span><span class="comment"></span>
<a name="l00783"></a>00783 <span class="comment">    ///@todo should have a similar check for spf input</span>
<a name="l00784"></a>00784 <span class="comment"></span>    <a class="code" href="classaffymetrix__fusion__io_1_1FusionCDFData.html">affymetrix_fusion_io::FusionCDFData</a> cdf;
<a name="l00785"></a>00785     cdf.<a class="code" href="classaffymetrix__fusion__io_1_1FusionCDFData.html#a992c651e317ec4e46907231c08fe78c2">SetFileName</a>(cdfFile.c_str());
<a name="l00786"></a>00786     <span class="keywordflow">if</span>(!cdf.<a class="code" href="classaffymetrix__fusion__io_1_1FusionCDFData.html#aa43618970a1b71d099d7bde8c2f10aa2">ReadHeader</a>()) {
<a name="l00787"></a>00787       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Can&#39;t read header for &quot;</span> + cdfFile + <span class="stringliteral">&quot;. Description: &quot;</span> + cdf.<a class="code" href="classaffymetrix__fusion__io_1_1FusionCDFData.html#a394dbd6ff71798988b781148e5bfdd23">GetError</a>());
<a name="l00788"></a>00788     }
<a name="l00789"></a>00789     <a class="code" href="classaffymetrix__fusion__io_1_1FusionCDFFileHeader.html">affymetrix_fusion_io::FusionCDFFileHeader</a> &amp;cdfHeader = cdf.<a class="code" href="classaffymetrix__fusion__io_1_1FusionCDFData.html#a1d5f050a3567d89ebf61bd37443ce6de">GetHeader</a>();
<a name="l00790"></a>00790     numProbesets = cdfHeader.<a class="code" href="classaffymetrix__fusion__io_1_1FusionCDFFileHeader.html#af93054275a681b3fa734fed489c9c46f">GetNumProbeSets</a>();
<a name="l00791"></a>00791     <span class="keywordflow">if</span> ((numProbesets &lt; <a class="code" href="classOptions.html#afcfab86e21cac5bba43e0858bd428e62" title="Get the integer value of an option.">getOptInt</a>(<span class="stringliteral">&quot;prior-size&quot;</span>)) &amp;&amp;
<a name="l00792"></a>00792         <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-priors-brlmm&quot;</span>) == <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00793"></a>00793       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Can&#39;t have size of prior: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(<a class="code" href="classOptions.html#afcfab86e21cac5bba43e0858bd428e62" title="Get the integer value of an option.">getOptInt</a>(<span class="stringliteral">&quot;prior-size&quot;</span>)) +
<a name="l00794"></a>00794                     <span class="stringliteral">&quot; larger than number of probesets: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(numProbesets));
<a name="l00795"></a>00795     }
<a name="l00796"></a>00796     cdf.<a class="code" href="classaffymetrix__fusion__io_1_1FusionCDFData.html#a0a2dab8bbd4c66cd2219e096eeb22f09">Close</a>();
<a name="l00797"></a>00797   }<span class="comment"></span>
<a name="l00798"></a>00798 <span class="comment">  /// @todo fix hack for discriminating snp5 vs snp6</span>
<a name="l00799"></a>00799 <span class="comment">  ///       aw: propose fix where we only allow xda chp output for</span>
<a name="l00800"></a>00800 <span class="comment">  ///           a specific set of chip types. Anything else is not</span>
<a name="l00801"></a>00801 <span class="comment">  ///           allowed.</span>
<a name="l00802"></a>00802 <span class="comment"></span>  <span class="comment">/* cws: If probesets are greater than 300K it has to be snp5</span>
<a name="l00803"></a>00803 <span class="comment">     or snp6 which shouldn&#39;t be used with xda format, has to be a</span>
<a name="l00804"></a>00804 <span class="comment">     better way to do this though... */</span>
<a name="l00805"></a>00805   <span class="keywordflow">if</span>(numProbesets &gt; 300000 &amp;&amp; <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;xda-chp-output&quot;</span>)) {
<a name="l00806"></a>00806     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Can&#39;t make XDA style probesets with newer chips, use --cc-chp-output instead to get new format chp files.&quot;</span>);
<a name="l00807"></a>00807   }
<a name="l00808"></a>00808 
<a name="l00809"></a>00809   <span class="comment">// special SNPs </span>
<a name="l00810"></a>00810   <span class="keywordtype">bool</span> needSpecialSnps = <span class="keyword">false</span>;
<a name="l00811"></a>00811   <span class="keywordflow">for</span>(<span class="keywordtype">size_t</span> aIx = 0; aIx &lt; analysis.size(); aIx++) {
<a name="l00812"></a>00812     <span class="keywordflow">if</span>(analysis[aIx].find(<span class="stringliteral">&quot;birdseed&quot;</span>) != string::npos) {
<a name="l00813"></a>00813       needSpecialSnps = <span class="keyword">true</span>;
<a name="l00814"></a>00814       <span class="keywordflow">if</span> ((<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-birdseed&quot;</span>) == <span class="stringliteral">&quot;&quot;</span>)) {
<a name="l00815"></a>00815         <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Birdseed requires models with &#39;--read-models-birdseed&#39;.&quot;</span>);
<a name="l00816"></a>00816       }
<a name="l00817"></a>00817     }
<a name="l00818"></a>00818   }
<a name="l00819"></a>00819   <span class="keywordflow">if</span> ((chrXSnps != <span class="stringliteral">&quot;&quot;</span>) &amp;&amp; needSpecialSnps)
<a name="l00820"></a>00820      <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Must use --special-snps rather than --chrX-snps when using birdseed &quot;</span>
<a name="l00821"></a>00821                    <span class="stringliteral">&quot;as birdseed expects chrY snps.&quot;</span>);
<a name="l00822"></a>00822   <span class="keywordflow">if</span> ((specialSnps == <span class="stringliteral">&quot;&quot;</span>) &amp;&amp; needSpecialSnps)
<a name="l00823"></a>00823     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Must specify &#39;--special-snps&#39; when special SNPS are needed.&quot;</span>);
<a name="l00824"></a>00824 
<a name="l00825"></a>00825   <span class="comment">// Check chip types</span>
<a name="l00826"></a>00826   vector&lt;string&gt; chipTypesInLayout;
<a name="l00827"></a>00827 
<a name="l00828"></a>00828   <span class="comment">/* Get the intial info about the chip and check cel files to make sure</span>
<a name="l00829"></a>00829 <span class="comment">     they match. */</span>
<a name="l00830"></a>00830   <a class="code" href="AptTypes_8h.html#a6b6d41ded05bd1136b17fa18be0d744f" title="For row and column indexes.">colrow_t</a> numRows = 0, numCols = 0;
<a name="l00831"></a>00831   <span class="keywordtype">int</span> probeCount = 0, probeSetCount=0, channelCount = 1;
<a name="l00832"></a>00832 
<a name="l00833"></a>00833   <span class="comment">// Get channel count from first CEL file.</span>
<a name="l00834"></a>00834   <a class="code" href="classaffymetrix__fusion__io_1_1FusionCELData.html">affymetrix_fusion_io::FusionCELData</a> cel;  
<a name="l00835"></a>00835   <span class="keywordflow">try</span> {
<a name="l00836"></a>00836     <span class="keywordflow">if</span>(!(Fs::fileExists(celFiles[0].c_str()))) {
<a name="l00837"></a>00837       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;File: &quot;</span> + celFiles[0] + <span class="stringliteral">&quot; doesn&#39;t appear to exist.&quot;</span>);
<a name="l00838"></a>00838       
<a name="l00839"></a>00839     }
<a name="l00840"></a>00840     cel.<a class="code" href="classaffymetrix__fusion__io_1_1FusionCELData.html#af43c5c7a95f9f535883c1ddda0bea057">SetFileName</a>(celFiles[0].c_str());
<a name="l00841"></a>00841     <span class="keywordflow">if</span>(!cel.<a class="code" href="classaffymetrix__fusion__io_1_1FusionCELData.html#afb44820c729a78caa986e598716c802c">Read</a>()) 
<a name="l00842"></a>00842       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;\nCan&#39;t read cel file: &quot;</span> + cel.<a class="code" href="classaffymetrix__fusion__io_1_1FusionCELData.html#acb10032210b46cb9b85c9dd03645acea">GetFileName</a>() + 
<a name="l00843"></a>00843                     <span class="stringliteral">&quot;\n&gt;&gt;&gt; Error reported: &quot;</span> + StringUtils::ConvertWCSToMBS(cel.<a class="code" href="classaffymetrix__fusion__io_1_1FusionCELData.html#a2f70ba19b2198059dc024db84b67c459">GetError</a>()));
<a name="l00844"></a>00844   }
<a name="l00845"></a>00845   <span class="keywordflow">catch</span>(<span class="keyword">const</span> <a class="code" href="classExcept.html" title="General purpose exception error for error handling.">Except</a> &amp;e) {
<a name="l00846"></a>00846     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Error opening cel file: &quot;</span> + celFiles[0] + <span class="stringliteral">&quot; error message is: &quot;</span> + e.<a class="code" href="classExcept.html#a6983a18bbb12fd78301d1ee77b2371ff" title="Standard what() call, just returns error message.">what</a>());
<a name="l00847"></a>00847   }
<a name="l00848"></a>00848   <span class="keywordflow">catch</span>(...) {
<a name="l00849"></a>00849     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Error opening cel file: &quot;</span> + celFiles[0]);
<a name="l00850"></a>00850   }
<a name="l00851"></a>00851 
<a name="l00852"></a>00852   <span class="comment">// If this is a AGCC CEL file, then get channels.  If this is a GCOS</span>
<a name="l00853"></a>00853   <span class="comment">// CEL file, then an empty vector is returned.</span>
<a name="l00854"></a>00854   std::vector&lt;std::wstring&gt; data_channels = cel.<a class="code" href="classaffymetrix__fusion__io_1_1FusionCELData.html#a36d1d74c33faee8550398e6f390a31e2">GetChannels</a>();
<a name="l00855"></a>00855   <span class="keywordflow">if</span> (!data_channels.empty()) {
<a name="l00856"></a>00856     channelCount = data_channels.size();
<a name="l00857"></a>00857   }
<a name="l00858"></a>00858 
<a name="l00859"></a>00859   <span class="keywordflow">if</span> (cdfFile != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00860"></a>00860     <a class="code" href="classEngineUtil.html#ade4ca714ab18a0ec35750acb9223cb4f" title="Open up a cdf file and get the chip type (i.e.">EngineUtil::getCdfChipType</a>(chipTypesInLayout, numRows, numCols, probeCount, probeSetCount, cdfFile);
<a name="l00861"></a>00861   }
<a name="l00862"></a>00862   <span class="keywordflow">else</span> <span class="keywordflow">if</span> (spfFile != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00863"></a>00863     <a class="code" href="classEngineUtil.html#a983c94e51a248918b98899f262f3e11a" title="Open up a spf (simple probe format) file and read the valid chiptypes (i.e.">EngineUtil::getSpfChipType</a>(chipTypesInLayout, numRows, numCols, probeCount, probeSetCount, spfFile);
<a name="l00864"></a>00864   }
<a name="l00865"></a>00865   <span class="keywordflow">else</span> {
<a name="l00866"></a>00866     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Must specify either a cdf file or spf (simple probe format) file.&quot;</span>);
<a name="l00867"></a>00867   }
<a name="l00868"></a>00868 
<a name="l00869"></a>00869   <a class="code" href="classOptions.html#aaf215b45e7097fcedae68c15073eb0f8" title="Set option value.">setOpt</a>(<span class="stringliteral">&quot;num-rows&quot;</span>, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(numRows));
<a name="l00870"></a>00870   <a class="code" href="classOptions.html#aaf215b45e7097fcedae68c15073eb0f8" title="Set option value.">setOpt</a>(<span class="stringliteral">&quot;num-cols&quot;</span>, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(numCols));
<a name="l00871"></a>00871   <a class="code" href="classOptions.html#aaf215b45e7097fcedae68c15073eb0f8" title="Set option value.">setOpt</a>(<span class="stringliteral">&quot;probe-count&quot;</span>, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(probeCount));
<a name="l00872"></a>00872   <a class="code" href="classOptions.html#aaf215b45e7097fcedae68c15073eb0f8" title="Set option value.">setOpt</a>(<span class="stringliteral">&quot;probeset-count&quot;</span>, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(probeSetCount));
<a name="l00873"></a>00873   <a class="code" href="classOptions.html#aaf215b45e7097fcedae68c15073eb0f8" title="Set option value.">setOpt</a>(<span class="stringliteral">&quot;channel-count&quot;</span>, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(channelCount));
<a name="l00874"></a>00874 
<a name="l00875"></a>00875   <span class="keywordflow">if</span>(chipTypesInLayout.empty() || chipTypesInLayout[0] == <span class="stringliteral">&quot;&quot;</span> || probeCount == 0) 
<a name="l00876"></a>00876       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Problem determining ChipType in file: &quot;</span> + 
<a name="l00877"></a>00877               ( cdfFile != <span class="stringliteral">&quot;&quot;</span> ? cdfFile : spfFile));
<a name="l00878"></a>00878 
<a name="l00879"></a>00879   <span class="comment">/* Did the user &quot;force&quot; a set of chip types via options? */</span>
<a name="l00880"></a>00880   vector&lt;string&gt; chipTypesSupplied = <a class="code" href="classOptions.html#ab20d0083424d9d9a8b94da46f487f57a" title="Return a vector of values for a particular option.">getOptVector</a>(<span class="stringliteral">&quot;chip-type&quot;</span>);
<a name="l00881"></a>00881 
<a name="l00882"></a>00882   <span class="comment">/* Figure out what chip type to report */</span>
<a name="l00883"></a>00883   <span class="keywordflow">if</span>(chipTypesSupplied.size() &gt; 0) {
<a name="l00884"></a>00884       <a class="code" href="classOptions.html#aaf215b45e7097fcedae68c15073eb0f8" title="Set option value.">setOpt</a>(<span class="stringliteral">&quot;chip-type&quot;</span>, chipTypesSupplied[0]);
<a name="l00885"></a>00885   } <span class="keywordflow">else</span> <span class="keywordflow">if</span>(chipTypesInLayout.size() &gt; 0) {
<a name="l00886"></a>00886       <a class="code" href="classOptions.html#aaf215b45e7097fcedae68c15073eb0f8" title="Set option value.">setOpt</a>(<span class="stringliteral">&quot;chip-type&quot;</span>, chipTypesInLayout[0]);
<a name="l00887"></a>00887   } <span class="keywordflow">else</span> {
<a name="l00888"></a>00888       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Unable to figure out a chip type.&quot;</span>);
<a name="l00889"></a>00889   }
<a name="l00890"></a>00890 
<a name="l00891"></a>00891   <span class="comment">/* Do Chip Type Check */</span>
<a name="l00892"></a>00892   <span class="keywordflow">if</span>(!<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;force&quot;</span>)) {
<a name="l00893"></a>00893     vector&lt;string&gt; chipTypesToCheck;
<a name="l00894"></a>00894 
<a name="l00895"></a>00895     <span class="keywordflow">if</span>(chipTypesSupplied.size() &gt; 0) {
<a name="l00896"></a>00896         chipTypesToCheck = chipTypesSupplied;
<a name="l00897"></a>00897         <a class="code" href="classEngineUtil.html#a6b5c169d7d42f87da86f31f675df557a" title="Check to make sure that there is at least one match between the two vectors of chip types...">EngineUtil::checkChipTypeVectors</a>(chipTypesSupplied, chipTypesInLayout);
<a name="l00898"></a>00898     } <span class="keywordflow">else</span> {
<a name="l00899"></a>00899         chipTypesToCheck = chipTypesInLayout;
<a name="l00900"></a>00900     }
<a name="l00901"></a>00901 
<a name="l00902"></a>00902     <a class="code" href="classEngineUtil.html#a952a57d44838690761d2f0bd28cd7a16" title="Check to make sure all the cel files are compatible with the chiptypes supported and number or probes...">EngineUtil::checkCelChipTypes</a>(chipTypesToCheck, probeCount, celFiles, numRows, numCols);
<a name="l00903"></a>00903 
<a name="l00904"></a>00904     <span class="comment">// Check model files</span>
<a name="l00905"></a>00905     <span class="comment">// We only check the first (primary) chip type.</span>
<a name="l00906"></a>00906     <span class="comment">// This ensures that model files for foo.cdf</span>
<a name="l00907"></a>00907     <span class="comment">// but not foo.bar.cdf will fail the check</span>
<a name="l00908"></a>00908     <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-birdseed&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00909"></a>00909         <a class="code" href="classEngineUtil.html#a44f506207f128da78d1c8ba67ab47de2" title="Check chip types in TSV file against those in vector.">EngineUtil::checkTsvFileChipType</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-birdseed&quot;</span>), chipTypesToCheck);
<a name="l00910"></a>00910         checkBirdseedTsvPriorFile(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-birdseed&quot;</span>), <span class="stringliteral">&quot;&quot;</span>);
<a name="l00911"></a>00911     }
<a name="l00912"></a>00912     <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-brlmm&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00913"></a>00913         <a class="code" href="classEngineUtil.html#a44f506207f128da78d1c8ba67ab47de2" title="Check chip types in TSV file against those in vector.">EngineUtil::checkTsvFileChipType</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-brlmm&quot;</span>), chipTypesToCheck);
<a name="l00914"></a>00914     }
<a name="l00915"></a>00915     <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-brlmmp&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00916"></a>00916         <a class="code" href="classEngineUtil.html#a44f506207f128da78d1c8ba67ab47de2" title="Check chip types in TSV file against those in vector.">EngineUtil::checkTsvFileChipType</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-brlmmp&quot;</span>), chipTypesToCheck);
<a name="l00917"></a>00917     }
<a name="l00918"></a>00918     
<a name="l00919"></a>00919     <span class="comment">// Check prior files</span><span class="comment"></span>
<a name="l00920"></a>00920 <span class="comment">    ///@todo check prior files chip type -- not tested</span>
<a name="l00921"></a>00921 <span class="comment"></span>    <span class="comment">//if (getOpt(&quot;read-priors-brlmm&quot;) != &quot;&quot;) {</span>
<a name="l00922"></a>00922     <span class="comment">//    EngineUtil:: checkTsvFileChipType(getOpt(&quot;read-priors-brlmm&quot;), chipTypesToCheck);</span>
<a name="l00923"></a>00923     <span class="comment">//}</span>
<a name="l00924"></a>00924     
<a name="l00925"></a>00925     <span class="comment">// Check special SNPs files</span>
<a name="l00926"></a>00926     <span class="keywordflow">if</span> (specialSnps != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00927"></a>00927         <a class="code" href="classEngineUtil.html#a44f506207f128da78d1c8ba67ab47de2" title="Check chip types in TSV file against those in vector.">EngineUtil::checkTsvFileChipType</a>(specialSnps, chipTypesToCheck);
<a name="l00928"></a>00928     }
<a name="l00929"></a>00929     <span class="keywordflow">if</span> (chrXSnps != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00930"></a>00930         <a class="code" href="classEngineUtil.html#a44f506207f128da78d1c8ba67ab47de2" title="Check chip types in TSV file against those in vector.">EngineUtil::checkTsvFileChipType</a>(chrXSnps, chipTypesToCheck);
<a name="l00931"></a>00931     }
<a name="l00932"></a>00932     
<a name="l00933"></a>00933     <span class="comment">// And other files</span>
<a name="l00934"></a>00934     <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;probe-class-file&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00935"></a>00935         <a class="code" href="classEngineUtil.html#a44f506207f128da78d1c8ba67ab47de2" title="Check chip types in TSV file against those in vector.">EngineUtil::checkTsvFileChipType</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;probe-class-file&quot;</span>), chipTypesToCheck);
<a name="l00936"></a>00936     }
<a name="l00937"></a>00937     <span class="keywordflow">if</span> (chrXProbes != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00938"></a>00938         <a class="code" href="classEngineUtil.html#a44f506207f128da78d1c8ba67ab47de2" title="Check chip types in TSV file against those in vector.">EngineUtil::checkTsvFileChipType</a>(chrXProbes, chipTypesToCheck);
<a name="l00939"></a>00939     }
<a name="l00940"></a>00940     <span class="keywordflow">if</span> (chrYProbes != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00941"></a>00941         <a class="code" href="classEngineUtil.html#a44f506207f128da78d1c8ba67ab47de2" title="Check chip types in TSV file against those in vector.">EngineUtil::checkTsvFileChipType</a>(chrYProbes, chipTypesToCheck);
<a name="l00942"></a>00942     }
<a name="l00943"></a>00943   }
<a name="l00944"></a>00944 
<a name="l00945"></a>00945   <span class="keywordflow">if</span>(!Fs::isWriteableDir(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;out-dir&quot;</span>)))
<a name="l00946"></a>00946     <span class="keywordflow">if</span>(<a class="code" href="classFs.html#aea474e8df7665f070c5334fa322bd283" title="make all path of directories ///">Fs::mkdirPath</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;out-dir&quot;</span>), <span class="keyword">false</span>) != APT_OK)
<a name="l00947"></a>00947       <a class="code" href="Err_8h.html#a7e26a673d8901b5bc880dfab285f5309" title="Calls Err::apt_err_abort with the filename and linenumber set.">APT_ERR_ABORT</a>(<span class="stringliteral">&quot;Can&#39;t make or write to directory: &quot;</span> + <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;out-dir&quot;</span>));
<a name="l00948"></a>00948   
<a name="l00949"></a>00949   makeTempDir(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;temp-dir&quot;</span>));
<a name="l00950"></a>00950 }
<a name="l00951"></a>00951 <span class="comment"></span>
<a name="l00952"></a>00952 <span class="comment">/**</span>
<a name="l00953"></a>00953 <span class="comment"> * @brief checks if there is enough disk space for diskmarts in the</span>
<a name="l00954"></a>00954 <span class="comment"> * temp-dir and for chp files in out-dir</span>
<a name="l00955"></a>00955 <span class="comment"> */</span>
<a name="l00956"></a>00956 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::checkDiskSpaceImp() {
<a name="l00957"></a>00957   int64_t out_disk_available = -1, scratch_disk_available = -1;
<a name="l00958"></a>00958   <span class="keywordtype">int</span> cel_count = <a class="code" href="classOptions.html#ab20d0083424d9d9a8b94da46f487f57a" title="Return a vector of values for a particular option.">getOptVector</a>(<span class="stringliteral">&quot;cels&quot;</span>).size();
<a name="l00959"></a>00959   <span class="keywordtype">int</span> probeset_count = <a class="code" href="classOptions.html#afcfab86e21cac5bba43e0858bd428e62" title="Get the integer value of an option.">getOptInt</a>(<span class="stringliteral">&quot;probeset-count&quot;</span>);
<a name="l00960"></a>00960   <span class="comment">//  int row_count = getOptInt(&quot;num-rows&quot;);</span>
<a name="l00961"></a>00961   <span class="comment">//  int col_count = getOptInt(&quot;num-cols&quot;);</span>
<a name="l00962"></a>00962   <span class="comment">//  int channel_count = getOptInt(&quot;channel-count&quot;);</span>
<a name="l00963"></a>00963   <span class="comment">//  int analysis_stream_count = getOptInt(&quot;analysis-count&quot;);</span>
<a name="l00964"></a>00964   
<a name="l00965"></a>00965   std::vector&lt;std::string&gt; result_files = <a class="code" href="classOptions.html#ab20d0083424d9d9a8b94da46f487f57a" title="Return a vector of values for a particular option.">getOptVector</a>(<span class="stringliteral">&quot;result-files&quot;</span>);
<a name="l00966"></a>00966 
<a name="l00967"></a>00967   <a class="code" href="classFsPath.html" title="A class for working with richer filepaths than just a string.">FsPath</a> out_path;
<a name="l00968"></a>00968   <span class="keywordflow">if</span> (!result_files.empty()) {
<a name="l00969"></a>00969     <a class="code" href="classFsPath.html" title="A class for working with richer filepaths than just a string.">FsPath</a> tmp_path(result_files[0]);
<a name="l00970"></a>00970     <span class="keywordflow">if</span> (tmp_path.getDirName()==<span class="stringliteral">&quot;&quot;</span>) {
<a name="l00971"></a>00971       out_path.<a class="code" href="classFsPath.html#aecd328162769610d039336a693bbe7a5" title="treat the entire path as a dir name.">setDirName</a>(tmp_path.getDirName());
<a name="l00972"></a>00972     }
<a name="l00973"></a>00973     <span class="comment">// @todo iterate through all of the result dirs to check available disk space</span>
<a name="l00974"></a>00974   }
<a name="l00975"></a>00975 
<a name="l00976"></a>00976   <span class="keywordflow">if</span> (out_path.<a class="code" href="classFsPath.html#ae496d12a3bc8b66643ffbf6874fbe9f9" title="True if it does not have a value.">empty</a>()) {
<a name="l00977"></a>00977     out_path.<a class="code" href="classFsPath.html#aecd328162769610d039336a693bbe7a5" title="treat the entire path as a dir name.">setDirName</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;cc-chp-out-dir&quot;</span>));
<a name="l00978"></a>00978   }
<a name="l00979"></a>00979   <span class="keywordflow">if</span> (out_path.<a class="code" href="classFsPath.html#ae496d12a3bc8b66643ffbf6874fbe9f9" title="True if it does not have a value.">empty</a>()) {
<a name="l00980"></a>00980     out_path.<a class="code" href="classFsPath.html#aecd328162769610d039336a693bbe7a5" title="treat the entire path as a dir name.">setDirName</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;out-dir&quot;</span>));
<a name="l00981"></a>00981   }
<a name="l00982"></a>00982 
<a name="l00983"></a>00983   out_disk_available = out_path.<a class="code" href="classFsPath.html#a952d6537de2702b23a84de68c068c4a1" title="rename the file to the new name">getFreeDiskSpace</a>();
<a name="l00984"></a>00984 
<a name="l00985"></a>00985   uint64_t out_disk_needed = 0;  
<a name="l00986"></a>00986   <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;cc-chp-output&quot;</span>)) {
<a name="l00987"></a>00987     <span class="comment">// @todo - why 35.49? CS</span>
<a name="l00988"></a>00988     out_disk_needed = <span class="keyword">static_cast&lt;</span>uint64_t<span class="keyword">&gt;</span>(probeset_count) * cel_count * 35.49;
<a name="l00989"></a>00989   }
<a name="l00990"></a>00990   
<a name="l00991"></a>00991   <span class="comment">// (1 + analysis_stream_count): one initial raw diskmart + one diskmart for</span>
<a name="l00992"></a>00992   <span class="comment">// each analysis stream</span>
<a name="l00993"></a>00993   uint64_t scratch_disk_needed = 0;
<a name="l00994"></a>00994 
<a name="l00995"></a>00995   <a class="code" href="classFsPath.html" title="A class for working with richer filepaths than just a string.">FsPath</a> tempdir_path;
<a name="l00996"></a>00996   tempdir_path.<a class="code" href="classFsPath.html#aecd328162769610d039336a693bbe7a5" title="treat the entire path as a dir name.">setDirName</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;temp-dir&quot;</span>));
<a name="l00997"></a>00997 
<a name="l00998"></a>00998   <span class="keywordtype">bool</span> same_vol = tempdir_path.<a class="code" href="classFsPath.html#a25d85a61c485800395a67a17df93cef6" title="are the paths on the same volume?">isSameVolume</a>(out_path);
<a name="l00999"></a>00999   <span class="keywordflow">if</span> (tempdir_path.<a class="code" href="classFsPath.html#ae496d12a3bc8b66643ffbf6874fbe9f9" title="True if it does not have a value.">empty</a>() ||
<a name="l01000"></a>01000       (same_vol==<span class="keyword">true</span>) ||
<a name="l01001"></a>01001       (tempdir_path == out_path)) {
<a name="l01002"></a>01002     out_disk_needed += scratch_disk_needed;
<a name="l01003"></a>01003   }
<a name="l01004"></a>01004   <span class="keywordflow">else</span> {
<a name="l01005"></a>01005     scratch_disk_available = tempdir_path.<a class="code" href="classFsPath.html#a952d6537de2702b23a84de68c068c4a1" title="rename the file to the new name">getFreeDiskSpace</a>();
<a name="l01006"></a>01006     <span class="keywordflow">if</span> (scratch_disk_needed &gt; 0 &amp;&amp; 
<a name="l01007"></a>01007         scratch_disk_available &gt;= 0 &amp;&amp;
<a name="l01008"></a>01008         scratch_disk_needed &gt;= scratch_disk_available) {
<a name="l01009"></a>01009       <span class="comment">// format in kb</span>
<a name="l01010"></a>01010       scratch_disk_needed = (scratch_disk_needed + 512) / 1024;
<a name="l01011"></a>01011       scratch_disk_available = (scratch_disk_available + 512) / 1024;
<a name="l01012"></a>01012       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;In &quot;</span> + tempdir_path.<a class="code" href="classFsPath.html#ad214f2acb541f27222d96c2a89626a00" title="a printable string which could be passed to our &amp;quot;setPath()&amp;quot; /// but it cant be opened in th...">asString</a>() + <span class="stringliteral">&quot;, need &quot;</span> +
<a name="l01013"></a>01013                     <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(scratch_disk_needed) + <span class="stringliteral">&quot;kb of scratch diskspace. &quot;</span>
<a name="l01014"></a>01014                     <span class="stringliteral">&quot;Only &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(scratch_disk_available) + <span class="stringliteral">&quot;kb available.&quot;</span>);
<a name="l01015"></a>01015     }
<a name="l01016"></a>01016   }
<a name="l01017"></a>01017 
<a name="l01018"></a>01018   <span class="keywordflow">if</span> (out_disk_needed &gt; 0 &amp;&amp;
<a name="l01019"></a>01019       out_disk_available &gt;= 0 &amp;&amp;
<a name="l01020"></a>01020       out_disk_needed &gt;= out_disk_available) {
<a name="l01021"></a>01021     <span class="comment">// format in kb</span>
<a name="l01022"></a>01022     out_disk_needed = (out_disk_needed + 512) / 1024;
<a name="l01023"></a>01023     out_disk_available = (out_disk_available + 512) / 1024;
<a name="l01024"></a>01024     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;In &quot;</span> + out_path.<a class="code" href="classFsPath.html#ad214f2acb541f27222d96c2a89626a00" title="a printable string which could be passed to our &amp;quot;setPath()&amp;quot; /// but it cant be opened in th...">asString</a>() + <span class="stringliteral">&quot;, need &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(out_disk_needed) + <span class="stringliteral">&quot;kb of diskspace. &quot;</span>
<a name="l01025"></a>01025                   <span class="stringliteral">&quot;Only &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(out_disk_available) + <span class="stringliteral">&quot;kb available.&quot;</span>);
<a name="l01026"></a>01026   }
<a name="l01027"></a>01027 }
<a name="l01028"></a>01028   
<a name="l01029"></a>01029 <span class="comment">/*   For copynumber chp files, the formula is */</span>
<a name="l01030"></a>01030 <span class="comment">/* 42.08 * num_probesets_processed */</span>
<a name="l01031"></a>01031 <span class="comment"></span>
<a name="l01032"></a>01032 <span class="comment">/**</span>
<a name="l01033"></a>01033 <span class="comment"> * @brief Get a list from a text file to select a subset of probe sets for</span>
<a name="l01034"></a>01034 <span class="comment"> * analysis.</span>
<a name="l01035"></a>01035 <span class="comment"> *</span>
<a name="l01036"></a>01036 <span class="comment"> * @param fileName - Path to file for reading probe subset from.</span>
<a name="l01037"></a>01037 <span class="comment"> * @param vNames - vector of probeset names</span>
<a name="l01038"></a>01038 <span class="comment"> * @param sNames - set of probeset names</span>
<a name="l01039"></a>01039 <span class="comment"> */</span>
<a name="l01040"></a>01040 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::loadPSFile(<span class="keyword">const</span> std::string &amp;fileName,
<a name="l01041"></a>01041                         std::vector&lt;const char *&gt; *vNames,
<a name="l01042"></a>01042                         std::set&lt;const char *, Util::ltstr&gt; *sNames) {
<a name="l01043"></a>01043   <a class="code" href="classaffx_1_1TsvFile.html" title="A class for reading and writing Tab Seperated Value (TSV) files. /// See the TsvFile format document ...">affx::TsvFile</a> tsv;
<a name="l01044"></a>01044   std::string probeset;
<a name="l01045"></a>01045   tsv.bind(0, <span class="stringliteral">&quot;probeset_id&quot;</span>, &amp;probeset, <a class="code" href="namespaceaffx.html#a8f36853f1d3b6cfb79edebd82760691daf605fbe0e244c4a604d176991d8df2c6" title="warn if not bound">TSV_BIND_REQUIRED</a>);
<a name="l01046"></a>01046   <span class="keywordflow">if</span>(tsv.<a class="code" href="classaffx_1_1TsvFile.html#a763ce0bbf19b4134d0445aa58e0f5011" title="Opens a file -- attempts to guess some defaults.">open</a>(fileName) != TSV_OK)
<a name="l01047"></a>01047     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Couldn&#39;t open file: &quot;</span> + fileName + <span class="stringliteral">&quot; to read.&quot;</span>);
<a name="l01048"></a>01048   <span class="keywordtype">int</span> psCount = 0;
<a name="l01049"></a>01049   <span class="keywordflow">while</span>(tsv.<a class="code" href="classaffx_1_1TsvFile.html#a2022895d082d1d635e7f59fccab12afb" title="Skip to the next level of data which matches seek_clvl ///.">nextLevel</a>(0) == TSV_OK) {
<a name="l01050"></a>01050     <span class="keywordflow">if</span>(vNames != NULL)
<a name="l01051"></a>01051         vNames-&gt;push_back(<a class="code" href="classUtil.html#a508b7aff9569ed929945d0d37eca384e" title="Create a copy of a string.">Util::cloneString</a>(probeset.c_str()));
<a name="l01052"></a>01052     <span class="keywordflow">if</span>(sNames != NULL) {
<a name="l01053"></a>01053         <span class="keywordflow">if</span>(vNames != NULL)
<a name="l01054"></a>01054             sNames-&gt;insert(vNames-&gt;at(psCount));
<a name="l01055"></a>01055         <span class="keywordflow">else</span>
<a name="l01056"></a>01056             sNames-&gt;insert(<a class="code" href="classUtil.html#a508b7aff9569ed929945d0d37eca384e" title="Create a copy of a string.">Util::cloneString</a>(probeset.c_str()));
<a name="l01057"></a>01057     }
<a name="l01058"></a>01058     psCount++;
<a name="l01059"></a>01059   }
<a name="l01060"></a>01060   tsv.<a class="code" href="classaffx_1_1TsvFile.html#ae7932be2156c04a562f327b53833aed7" title="Close the file.">close</a>();
<a name="l01061"></a>01061 }
<a name="l01062"></a>01062 <span class="comment"></span>
<a name="l01063"></a>01063 <span class="comment">/**</span>
<a name="l01064"></a>01064 <span class="comment"> * Read the design of the chip from the cdf or spf file.</span>
<a name="l01065"></a>01065 <span class="comment"> *</span>
<a name="l01066"></a>01066 <span class="comment"> * @param layout        - Object to be filled in.</span>
<a name="l01067"></a>01067 <span class="comment"> * @param probesetNames - Vector of probeset names to be filled in</span>
<a name="l01068"></a>01068 <span class="comment"> * @param killList      - Vector of probes to ignore</span>
<a name="l01069"></a>01069 <span class="comment"> * @param justStats     - Boolean indicating whether we want to keep probe layout info</span>
<a name="l01070"></a>01070 <span class="comment"> * @param  psTypesToLoad - What probeset types to load</span>
<a name="l01071"></a>01071 <span class="comment"> */</span>
<a name="l01072"></a>01072 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::loadLayout(<a class="code" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a> &amp;layout,
<a name="l01073"></a>01073                    std::vector&lt;const char *&gt; &amp;probesetNames,
<a name="l01074"></a>01074                    probeidmap_t &amp;killList,
<a name="l01075"></a>01075                    <span class="keywordtype">bool</span> justStats,
<a name="l01076"></a>01076                    std::set&lt;affxcdf::GeneChipProbeSetType&gt; &amp;psTypesToLoad) {
<a name="l01077"></a>01077 
<a name="l01078"></a>01078   std::set&lt;const char *, Util::ltstr&gt; probeSetsToLoad;
<a name="l01079"></a>01079   vector&lt;string&gt; requiredCols;
<a name="l01080"></a>01080 
<a name="l01081"></a>01081   <span class="keywordtype">string</span> cdfFile = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;cdf-file&quot;</span>);
<a name="l01082"></a>01082   <span class="keywordtype">string</span> spfFile = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;spf-file&quot;</span>);
<a name="l01083"></a>01083   <span class="keywordtype">int</span> computePrior = <a class="code" href="classOptions.html#afcfab86e21cac5bba43e0858bd428e62" title="Get the integer value of an option.">getOptInt</a>(<span class="stringliteral">&quot;prior-size&quot;</span>);
<a name="l01084"></a>01084   <span class="keywordtype">bool</span> listSample = <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;list-sample&quot;</span>);
<a name="l01085"></a>01085 <span class="comment"></span>
<a name="l01086"></a>01086 <span class="comment">  ///@todo populate probeSetsToLoad -- right now we are simply loading everything</span>
<a name="l01087"></a>01087 <span class="comment"></span>
<a name="l01088"></a>01088   <span class="comment">/* If we are sampling for a prior we need to load up all of the</span>
<a name="l01089"></a>01089 <span class="comment">     probesets, so empty the set which indicates this to openCdf(). */</span>
<a name="l01090"></a>01090   <span class="keywordflow">if</span>(computePrior &gt; 0 &amp;&amp; !listSample) {
<a name="l01091"></a>01091     probeSetsToLoad.clear();
<a name="l01092"></a>01092   }
<a name="l01093"></a>01093 
<a name="l01094"></a>01094   vector&lt;bool&gt; probeSubset;  <span class="comment">//leave empty as we simply load everything</span>
<a name="l01095"></a>01095 
<a name="l01096"></a>01096   <span class="keywordflow">try</span> {
<a name="l01097"></a>01097     <span class="keywordflow">if</span>(cdfFile != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l01098"></a>01098       <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(<span class="stringliteral">&quot;Opening layout file: &quot;</span>) + cdfFile);
<a name="l01099"></a>01099       <span class="keywordflow">if</span>(!layout.<a class="code" href="classChipLayout.html#aa9bdb0e181302d4953aab10ed847c7bf" title="Open a file and parse the probe sets.">openCdf</a>(       cdfFile, 
<a name="l01100"></a>01100                                 probeSetsToLoad, 
<a name="l01101"></a>01101                                 &amp;probesetNames,
<a name="l01102"></a>01102                                 probeSubset, 
<a name="l01103"></a>01103                                 <span class="stringliteral">&quot;&quot;</span>, 
<a name="l01104"></a>01104                                 killList, 
<a name="l01105"></a>01105                                 justStats, 
<a name="l01106"></a>01106                                 psTypesToLoad)) {
<a name="l01107"></a>01107         <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Couldn&#39;t open layout file: &quot;</span> + cdfFile);
<a name="l01108"></a>01108       }
<a name="l01109"></a>01109     }
<a name="l01110"></a>01110     <span class="keywordflow">else</span> <span class="keywordflow">if</span>(spfFile != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l01111"></a>01111       <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(<span class="stringliteral">&quot;Opening layout file: &quot;</span>) + spfFile);
<a name="l01112"></a>01112       layout.<a class="code" href="classChipLayout.html#af902ac26fbddd00db96825c4b9830379" title="Read in probelist data from a simple probe format file.">openSpf</a>(spfFile, probeSetsToLoad, &amp;probesetNames, probeSubset,
<a name="l01113"></a>01113                            <span class="stringliteral">&quot;&quot;</span>, justStats, psTypesToLoad);
<a name="l01114"></a>01114     }
<a name="l01115"></a>01115     <span class="keywordflow">else</span> {
<a name="l01116"></a>01116       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Must have either a cdf file or spf file.&quot;</span>);
<a name="l01117"></a>01117     }
<a name="l01118"></a>01118   }
<a name="l01119"></a>01119   <span class="keywordflow">catch</span>(<span class="keyword">const</span> <a class="code" href="classExcept.html" title="General purpose exception error for error handling.">Except</a> &amp;e) {
<a name="l01120"></a>01120     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(e.<a class="code" href="classExcept.html#a6983a18bbb12fd78301d1ee77b2371ff" title="Standard what() call, just returns error message.">what</a>());
<a name="l01121"></a>01121   }
<a name="l01122"></a>01122   <span class="keywordflow">catch</span>(...) {
<a name="l01123"></a>01123     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Unknown exception caught - terminating.&quot;</span>);
<a name="l01124"></a>01124   }
<a name="l01125"></a>01125   <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(2, <span class="stringliteral">&quot;Loaded &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(probesetNames.size()) + <span class="stringliteral">&quot; probesets.&quot;</span>);
<a name="l01126"></a>01126 }
<a name="l01127"></a>01127 <span class="comment"></span>
<a name="l01128"></a>01128 <span class="comment">/**</span>
<a name="l01129"></a>01129 <span class="comment"> * Open a text file and read in snps that are possibly haploid. Currently</span>
<a name="l01130"></a>01130 <span class="comment"> * only chrX snps are supported.</span>
<a name="l01131"></a>01131 <span class="comment"> *</span>
<a name="l01132"></a>01132 <span class="comment"> * @param SpecialSnps -</span>
<a name="l01133"></a>01133 <span class="comment"> * @param layout      -</span>
<a name="l01134"></a>01134 <span class="comment"> * @param permissive  -</span>
<a name="l01135"></a>01135 <span class="comment"> */</span>
<a name="l01136"></a>01136 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::fillInSpecialSnps(SpecialSnpMap &amp;SpecialSnps,
<a name="l01137"></a>01137                        <a class="code" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a> &amp;layout, <span class="keywordtype">bool</span> permissive) {
<a name="l01138"></a>01138   std::string specialType, chip;
<a name="l01139"></a>01139 
<a name="l01140"></a>01140   <span class="keywordtype">string</span> specialSnps = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;special-snps&quot;</span>);
<a name="l01141"></a>01141 
<a name="l01142"></a>01142   readSpecialSnpsFile(&amp;SpecialSnps, &amp;chip, &amp;specialType, specialSnps);
<a name="l01143"></a>01143 
<a name="l01144"></a>01144   <span class="comment">// check the chip type</span>
<a name="l01145"></a>01145   <a class="code" href="classaffx_1_1TsvFile.html" title="A class for reading and writing Tab Seperated Value (TSV) files. /// See the TsvFile format document ...">affx::TsvFile</a> tsv;
<a name="l01146"></a>01146   <span class="keywordtype">string</span> snpName;
<a name="l01147"></a>01147   tsv.bind(0, <span class="stringliteral">&quot;probeset_id&quot;</span>, &amp;snpName, <a class="code" href="namespaceaffx.html#a8f36853f1d3b6cfb79edebd82760691daf605fbe0e244c4a604d176991d8df2c6" title="warn if not bound">TSV_BIND_REQUIRED</a>);
<a name="l01148"></a>01148   <span class="comment">/* sanity checks */</span>
<a name="l01149"></a>01149   <span class="keywordflow">if</span>(tsv.<a class="code" href="classaffx_1_1TsvFile.html#a763ce0bbf19b4134d0445aa58e0f5011" title="Opens a file -- attempts to guess some defaults.">open</a>(specialSnps) != TSV_OK)
<a name="l01150"></a>01150     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Couldn&#39;t open file: &quot;</span> + specialSnps + <span class="stringliteral">&quot; to read.&quot;</span>);
<a name="l01151"></a>01151   tsv.<a class="code" href="classaffx_1_1TsvFile.html#ae7932be2156c04a562f327b53833aed7" title="Close the file.">close</a>();
<a name="l01152"></a>01152 
<a name="l01153"></a>01153   <span class="comment">// make sure the SNP in the specialSnps file is also in CDF</span>
<a name="l01154"></a>01154   <span class="keywordflow">if</span> (!permissive) {
<a name="l01155"></a>01155       <span class="keywordflow">for</span> (SpecialSnpMap::const_iterator it = SpecialSnps.begin();
<a name="l01156"></a>01156            it != SpecialSnps.end(); ++it) {
<a name="l01157"></a>01157           <span class="keyword">const</span> <span class="keywordtype">string</span> &amp;snpName = (*it).first;
<a name="l01158"></a>01158           <span class="keywordflow">if</span> (!layout.<a class="code" href="classChipLayout.html#af42fc829c0a3b795e674cfcbcb3dc671" title="Does the layout have a reference to the probeset with name psName? Must have name map filled to call ...">containsProbeSet</a>(snpName)) {
<a name="l01159"></a>01159               <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Can&#39;t find special snp: &quot;</span> + snpName + <span class="stringliteral">&quot; in cdf file. Wrong special snp file?&quot;</span>);
<a name="l01160"></a>01160           }
<a name="l01161"></a>01161       }
<a name="l01162"></a>01162   }
<a name="l01163"></a>01163 }
<a name="l01164"></a>01164 
<a name="l01165"></a>01165 
<a name="l01166"></a>01166 <span class="comment"></span>
<a name="l01167"></a>01167 <span class="comment">/**</span>
<a name="l01168"></a>01168 <span class="comment"> * Read the prior from a file or generate one from the data.</span>
<a name="l01169"></a>01169 <span class="comment"> *</span>
<a name="l01170"></a>01170 <span class="comment"> * @param outFile - File to output prior if desired.</span>
<a name="l01171"></a>01171 <span class="comment"> * @param sample - Vector of probesets to generate prior from, if using data.</span>
<a name="l01172"></a>01172 <span class="comment"> * @param layout - Grouping and location of probesets.</span>
<a name="l01173"></a>01173 <span class="comment"> * @param iMart - Intensity mart of data for generating prior.</span>
<a name="l01174"></a>01174 <span class="comment"> * @param chipStream - Vector of modification objects to pass raw data through.</span>
<a name="l01175"></a>01175 <span class="comment"> * @param pmAdjuster - Probe specific adjustment.</span>
<a name="l01176"></a>01176 <span class="comment"> * @param qBRLMM - Quantification method for calling snps.</span>
<a name="l01177"></a>01177 <span class="comment"> */</span>
<a name="l01178"></a>01178 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::setPrior(<span class="keywordtype">string</span> &amp;outFile, vector&lt;string&gt; &amp;sample,
<a name="l01179"></a>01179               <a class="code" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a> &amp;layout, <a class="code" href="classIntensityMart.html" title="IntensityMart.">IntensityMart</a> &amp;iMart,
<a name="l01180"></a>01180               vector&lt;ChipStream *&gt; &amp;chipStream, <a class="code" href="classPmAdjuster.html" title="Interface for determining a change based on intensity of perfect match.">PmAdjuster</a> &amp;pmAdjuster,
<a name="l01181"></a>01181               <a class="code" href="classQuantBRLMM.html" title="Implements a SNP genotype calling method motivated by Nusrat Rabbee&amp;#39;s RLMM (Robust Linear Model w...">QuantBRLMM</a> *qBRLMM) {
<a name="l01182"></a>01182   <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-priors-brlmm&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l01183"></a>01183     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Loading priors from: &quot;</span> + <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-priors-brlmm&quot;</span>) );
<a name="l01184"></a>01184     <a class="code" href="classClusterPrior.html">ClusterPrior</a> prior = <a class="code" href="classQuantBRLMM.html#a71f90338aa453da6b7c453a4fe44c1ab" title="Read the prior from the test file specified.">QuantBRLMM::loadPrior</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-priors-brlmm&quot;</span>));
<a name="l01185"></a>01185     <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;write-prior&quot;</span>)) {
<a name="l01186"></a>01186       <a class="code" href="classQuantBRLMM.html#a1a7e2033f6184a97ecef944e46c0aed5" title="Write out a text representation of the prior to the file specified.">QuantBRLMM::writePrior</a>(prior, outFile);
<a name="l01187"></a>01187     }
<a name="l01188"></a>01188     qBRLMM-&gt;<a class="code" href="classQuantBRLMM.html#ad6cb1cdd30699b802c7d81248da58da0" title="Specify the prior that will be used for predictions.">setPrior</a>(prior);
<a name="l01189"></a>01189   }
<a name="l01190"></a>01190   <span class="keywordflow">else</span> <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#afcfab86e21cac5bba43e0858bd428e62" title="Get the integer value of an option.">getOptInt</a>(<span class="stringliteral">&quot;prior-size&quot;</span>) &gt; 0) {
<a name="l01191"></a>01191     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Computing prior with: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(sample.size()) + <span class="stringliteral">&quot; initial snps.&quot;</span>);
<a name="l01192"></a>01192     <a class="code" href="classClusterPrior.html">ClusterPrior</a> prior = qBRLMM-&gt;<a class="code" href="classQuantBRLMM.html#afafae26ed3cd67f3e51ca8fd77cc7669" title="Construct a prior from a collection of probesets.">makePrior</a>(sample, layout, iMart, chipStream, pmAdjuster);
<a name="l01193"></a>01193     <span class="keywordflow">if</span>(prior.snpCount &lt; 10) {
<a name="l01194"></a>01194       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(<span class="stringliteral">&quot;Only &quot;</span>) + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(prior.snpCount) +
<a name="l01195"></a>01195                     <span class="stringliteral">&quot; SNP clusters had enough information to generate prior. Please try using more cel files.&quot;</span>);
<a name="l01196"></a>01196     }
<a name="l01197"></a>01197     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(prior.snpCount) + <span class="stringliteral">&quot; SNP clusters used to generate prior.&quot;</span>);
<a name="l01198"></a>01198     <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;write-prior&quot;</span>)) {
<a name="l01199"></a>01199       <a class="code" href="classQuantBRLMM.html#a1a7e2033f6184a97ecef944e46c0aed5" title="Write out a text representation of the prior to the file specified.">QuantBRLMM::writePrior</a>(prior, outFile);
<a name="l01200"></a>01200     }
<a name="l01201"></a>01201     qBRLMM-&gt;<a class="code" href="classQuantBRLMM.html#ad6cb1cdd30699b802c7d81248da58da0" title="Specify the prior that will be used for predictions.">setPrior</a>(prior);
<a name="l01202"></a>01202   }
<a name="l01203"></a>01203 }
<a name="l01204"></a>01204 <span class="comment"></span>
<a name="l01205"></a>01205 <span class="comment">/**</span>
<a name="l01206"></a>01206 <span class="comment"> * Make covariate normalizations from a file or generate one from the data.</span>
<a name="l01207"></a>01207 <span class="comment"> *</span>
<a name="l01208"></a>01208 <span class="comment"> * @param outFile - File to output normalization parameters if desired.</span>
<a name="l01209"></a>01209 <span class="comment"> * @param sample - Vector of probesets to generate normalizations from, if using data.</span>
<a name="l01210"></a>01210 <span class="comment"> * @param layout - Grouping and location of probesets.</span>
<a name="l01211"></a>01211 <span class="comment"> * @param iMart - Intensity mart of data for generating normalizers.</span>
<a name="l01212"></a>01212 <span class="comment"> * @param chipStream - Vector of modification objects to pass raw data through.</span>
<a name="l01213"></a>01213 <span class="comment"> * @param pmAdjuster - Probe specific adjustment.</span>
<a name="l01214"></a>01214 <span class="comment"> * @param qLabelZ - Quantification method for calling snps.</span>
<a name="l01215"></a>01215 <span class="comment"> */</span>
<a name="l01216"></a>01216 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::setNormLabelZ(<span class="keywordtype">string</span> &amp;outFile, vector&lt;std::string&gt; sample,
<a name="l01217"></a>01217                    <a class="code" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a> &amp;layout, <a class="code" href="classIntensityMart.html" title="IntensityMart.">IntensityMart</a> &amp;iMart,
<a name="l01218"></a>01218                    vector&lt;ChipStream *&gt; &amp;chipStream, <a class="code" href="classPmAdjuster.html" title="Interface for determining a change based on intensity of perfect match.">PmAdjuster</a> &amp;pmAdjuster,
<a name="l01219"></a>01219                    <a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a> *qLabelZ) {
<a name="l01220"></a>01220   <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#afcfab86e21cac5bba43e0858bd428e62" title="Get the integer value of an option.">getOptInt</a>(<span class="stringliteral">&quot;norm-size&quot;</span>) &gt; 0) {
<a name="l01221"></a>01221     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Computing normalizer with: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(sample.size()) + <span class="stringliteral">&quot; initial snps.&quot;</span>);
<a name="l01222"></a>01222     <span class="comment">// here&#39;s where we do the magic</span>
<a name="l01223"></a>01223     <span class="comment">// take the sample of snps</span>
<a name="l01224"></a>01224     <span class="comment">// build a normalizer map of normalization across samples</span>
<a name="l01225"></a>01225     <span class="comment">// 1 normalization predictor per sample</span>
<a name="l01226"></a>01226     vector&lt;NormalizationPredictor&gt; normMap = qLabelZ-&gt;<a class="code" href="classQuantLabelZ.html#a8e0d8c1489dd615f81feec9f094d2aeb" title="makes the normalization functions for all samples">makeNorm</a>(sample, layout, iMart,
<a name="l01227"></a>01227                                                                chipStream, pmAdjuster);
<a name="l01228"></a>01228     <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;write-norm&quot;</span>)) {
<a name="l01229"></a>01229       <a class="code" href="classQuantLabelZ.html#a12c40ec12ddbaabcd129750af201e5e8" title="writes out the contrast normalization functions">QuantLabelZ::writeNormMap</a>(normMap, outFile);
<a name="l01230"></a>01230     }
<a name="l01231"></a>01231     qLabelZ-&gt;<a class="code" href="classQuantLabelZ.html#a3b8e9682bc1b43ef80676f1eb9556d87" title="sets the normalization functions for all samples">setNorm</a>(normMap);
<a name="l01232"></a>01232   }
<a name="l01233"></a>01233 }
<a name="l01234"></a>01234 <span class="comment"></span>
<a name="l01235"></a>01235 <span class="comment">/**</span>
<a name="l01236"></a>01236 <span class="comment"> * Construct an object for output that contains the program an</span>
<a name="l01237"></a>01237 <span class="comment"> * algorithm information.</span>
<a name="l01238"></a>01238 <span class="comment"> * @param as - AnalysisStream to get information from.</span>
<a name="l01239"></a>01239 <span class="comment"> * @param layout - Layout to get information from.</span>
<a name="l01240"></a>01240 <span class="comment"> * @param probesetNames - List of probeset names to output</span>
<a name="l01241"></a>01241 <span class="comment"> *</span>
<a name="l01242"></a>01242 <span class="comment"> * @return - Information for making CHP file.</span>
<a name="l01243"></a>01243 <span class="comment"> */</span>
<a name="l01244"></a>01244 <a class="code" href="classAnalysisInfo.html" title="Information about a particular analysis run.">AnalysisInfo</a> ProbesetGenotypeStageEngine::makeAnalysisInfo(<a class="code" href="classAnalysisStream.html" title="Utility class for centralizing analysis pathway.">AnalysisStream</a> *as,
<a name="l01245"></a>01245                               <a class="code" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a> &amp;layout,
<a name="l01246"></a>01246                               <span class="keyword">const</span> std::vector&lt;const char*&gt; &amp;probesetNames,
<a name="l01247"></a>01247                               <a class="code" href="classGenderCalls.html" title="Class for determining gender from a cel set of cel files.">GenderCalls</a> *genders) {
<a name="l01248"></a>01248 
<a name="l01249"></a>01249   <a class="code" href="classAnalysisInfo.html" title="Information about a particular analysis run.">AnalysisInfo</a> info;
<a name="l01250"></a>01250   info.<a class="code" href="classAnalysisInfo.html#af50a44fcfb27fcf4aedbb5664857ff24" title="Length of the longest probeset name.">m_MaxPsNameLength</a> = layout.getMaxNameLength();
<a name="l01251"></a>01251   info.<a class="code" href="classAnalysisInfo.html#a302e3084afe43804e83c520b7beb463a" title="Number of expression probesets.">m_NumExpression</a> = layout.<a class="code" href="classChipLayout.html#acd976e4a3fa1141b5660ecab6f5c9b70" title="Number of expression probesets.">getNumExpressionPSets</a>();
<a name="l01252"></a>01252   info.<a class="code" href="classAnalysisInfo.html#a9a829523fde5f52310057a05b01ba601" title="Number of genotyping expression.">m_NumGenotyping</a> = layout.<a class="code" href="classChipLayout.html#acc2755565e72c8bcb52365c23572c216" title="Number of genotyping expression.">getNumGenotypeingPSets</a>();
<a name="l01253"></a>01253   info.<a class="code" href="classAnalysisInfo.html#a3fc5ed6fab19e4d4b155978fb6d6f3e3" title="How many snps to report in CHP file.">m_NumReporting</a> = layout.<a class="code" href="classChipLayout.html#a71df054a01d62ca33d6b8de8c8c5c3ca" title="How many probe sets in Pgf file?">getProbeSetCount</a>();
<a name="l01254"></a>01254   info.<a class="code" href="classAnalysisInfo.html#a540d9d55c05541e8fbe95474d6dbb7fb" title="How many rows are on the grid of the chip.">m_NumRows</a> = layout.<a class="code" href="classChipLayout.html#a5b5ed8dcd11bdfbbecfbce63e2d0b342" title="Get the number of x counts.">getXCount</a>();
<a name="l01255"></a>01255   info.<a class="code" href="classAnalysisInfo.html#a39da235062cb23d7392f29cca25d8da7" title="How many columns are the grid of the chip.">m_NumCols</a> = layout.<a class="code" href="classChipLayout.html#ae602f7c506b7ae9355844f06267ff528" title="Get the number of y counts.">getYCount</a>();
<a name="l01256"></a>01256   info.<a class="code" href="classAnalysisInfo.html#ac85efbbb6618cb87c39511af32c311a7" title="How many probesets are on the chip and in cdf file. This is /// critical as probesets must be the exa...">m_NumProbeSets</a> = probesetNames.size();
<a name="l01257"></a>01257   info.<a class="code" href="classAnalysisInfo.html#a1eaf6c81440945d17b192a9d7196e788" title="What type of chp file is this? For our purposes here it is genotyping.">m_ProbeSetType</a> = affxcdf::GenotypingProbeSetType;
<a name="l01258"></a>01258   fillInAnalysisInfo(info, as);
<a name="l01259"></a>01259 
<a name="l01260"></a>01260   <a class="code" href="classQuantGTypeMethod.html" title="Quantification methods used for making genotyping calls implement (currently just brlmm) this interfa...">QuantGTypeMethod</a> *method =
<a name="l01261"></a>01261       <span class="keyword">dynamic_cast&lt;</span><a class="code" href="classQuantGTypeMethod.html" title="Quantification methods used for making genotyping calls implement (currently just brlmm) this interfa...">QuantGTypeMethod</a> *<span class="keyword">&gt;</span>(as-&gt;<a class="code" href="classAnalysisStream.html#af5d7e7ca7a32a6f189290b5ef1888f64" title="Get the quantification method used.">getQuantMethod</a>());
<a name="l01262"></a>01262   <span class="keywordtype">float</span> min = method-&gt;<a class="code" href="classQuantGTypeMethod.html#af73b280fce51d3e4e324993356ceb543" title="Get the min value for a call.">getMinThresh</a>();
<a name="l01263"></a>01263   info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(<span class="stringliteral">&quot;min-thresh&quot;</span>,<a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(min));
<a name="l01264"></a>01264   <span class="keywordtype">float</span> max = method-&gt;<a class="code" href="classQuantGTypeMethod.html#abe1ebe82ba24b2a97130c0d8d7d9e4ce" title="Get the max value for a call.">getMaxThresh</a>();
<a name="l01265"></a>01265   info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(<span class="stringliteral">&quot;max-thresh&quot;</span>,<a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(max));
<a name="l01266"></a>01266 
<a name="l01267"></a>01267   info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(<span class="stringliteral">&quot;gender_method_used&quot;</span>,genders-&gt;getGenderName());
<a name="l01268"></a>01268 
<a name="l01269"></a>01269   <span class="comment">// this is a &quot;magic&quot; value that works with GCOS/GTYPE according to Richard</span>
<a name="l01270"></a>01270   info.<a class="code" href="classAnalysisInfo.html#aa32b7ba5f935bc64316e903980e9acbe" title="What is our id for GCOS?">m_ProgID</a> = <span class="stringliteral">&quot;GDMTAnalysis.BRMBaseCall.1&quot;</span>;
<a name="l01271"></a>01271 
<a name="l01272"></a>01272   <span class="comment">// If CHP Output, then we need list of probeset names to output</span>
<a name="l01273"></a>01273   <span class="comment">// Otherwise do not waste the memory</span>
<a name="l01274"></a>01274   <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;cc-chp-output&quot;</span>) || <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;xda-chp-output&quot;</span>)) 
<a name="l01275"></a>01275     info.<a class="code" href="classAnalysisInfo.html#a21ab2fbfe201742b20f48ac42a12a07e" title="Vector of probeset names and groups as they should appear in the chp file. /// this memory is owned e...">m_ProbesetNames</a> = probesetNames;
<a name="l01276"></a>01276 
<a name="l01277"></a>01277   <span class="comment">// Sanity check</span>
<a name="l01278"></a>01278   <a class="code" href="classErr.html#ad1a633f602fe60689cdc12d9ca1842e9" title="errAbort based version of assert with an error message.">Err::check</a>(info.m_ParamValues.size() == info.<a class="code" href="classAnalysisInfo.html#ad48864dcd9e2e22d9d74dfbbb00fff04" title="These next two vectors are matched pairs, one contains the /// keys and the other the values...">m_ParamNames</a>.size(),
<a name="l01279"></a>01279              <span class="stringliteral">&quot;AnalysisInfo - Names and values out of sync.&quot;</span>);
<a name="l01280"></a>01280   <span class="keywordflow">return</span> info;
<a name="l01281"></a>01281 }
<a name="l01282"></a>01282 
<a name="l01283"></a>01283 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::fillInAnalysisInfo(<a class="code" href="classAnalysisInfo.html" title="Information about a particular analysis run.">AnalysisInfo</a> &amp;info, <a class="code" href="classQuantMethod.html" title="QuantMethod - Interface for computing quantification summaries from PM intensities grouped into probe...">QuantMethod</a> *qMethod) {
<a name="l01284"></a>01284     info.<a class="code" href="classAnalysisInfo.html#a485fade346a6f248b915cbd0a6e1041c" title="What version of the algorithm is being used.">m_AlgVersion</a> = qMethod-&gt;<a class="code" href="classQuantMethod.html#a4799b1ab9adeff7ea3681938367ae670" title="What version of the expression algorithm are we implementing?">getVersion</a>();
<a name="l01285"></a>01285     <span class="comment">//    info.m_AlgName = as-&gt;getName();</span>
<a name="l01286"></a>01286     <span class="comment">//    info.addParam(prefix + &quot;analysis-guid&quot;, as-&gt;getGuid());</span>
<a name="l01287"></a>01287     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(<span class="stringliteral">&quot;quantification-name&quot;</span>, qMethod-&gt;<a class="code" href="classQuantMethod.html#ab1b819505891590c9f3e57d9cc67d4de" title="What is the name of the quantification method?">getType</a>());
<a name="l01288"></a>01288     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(<span class="stringliteral">&quot;quantification-version&quot;</span>, qMethod-&gt;<a class="code" href="classQuantMethod.html#a4799b1ab9adeff7ea3681938367ae670" title="What version of the expression algorithm are we implementing?">getVersion</a>());
<a name="l01289"></a>01289     <span class="comment">//    info.addParam(prefix + subprefix + &quot;analysis-name&quot;, as-&gt;getName());</span>
<a name="l01290"></a>01290 
<a name="l01291"></a>01291     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(<span class="stringliteral">&quot;quantification-scale&quot;</span>, <a class="code" href="classQuantMethod.html#a6a4921cca844ef99404228081863c8c4" title="Convert a Scale enumeration to a text description.">QuantMethod::scaleToTxt</a>(qMethod-&gt;<a class="code" href="classQuantMethod.html#a2284897bb0665c4cea9f8e030e20d796" title="What scale are the target effects on? Are they linear (like plier) or log2 (like rma).">getScale</a>()));
<a name="l01292"></a>01292     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(<span class="stringliteral">&quot;quantification-type&quot;</span>, <a class="code" href="classQuantMethod.html#aaab461044d36fd604004b47934b209d3" title="Convert a QuantType enumeration to a human readable string representation.">QuantMethod::quantTypeToTxt</a>(qMethod-&gt;<a class="code" href="classQuantMethod.html#a2889dd697efae0503edd00e4087e991b" title="What type of quantification method is this? Summarization like plier detection like dabg...">getQuantType</a>()));
<a name="l01293"></a>01293 
<a name="l01294"></a>01294 }
<a name="l01295"></a>01295 
<a name="l01296"></a>01296 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::fillInAnalysisInfo(<a class="code" href="classAnalysisInfo.html" title="Information about a particular analysis run.">AnalysisInfo</a> &amp;info, std::string prefix) {<span class="comment"></span>
<a name="l01297"></a>01297 <span class="comment">    ///@todo need to refactor to ensure consistency between meta info tracked in probeset-summarize</span>
<a name="l01298"></a>01298 <span class="comment">    ///      versus probeset-genotype</span>
<a name="l01299"></a>01299 <span class="comment">    ///@todo automatically pull state and options info?</span>
<a name="l01300"></a>01300 <span class="comment">    ///@todo sync up option name with label in chp header</span>
<a name="l01301"></a>01301 <span class="comment"></span>    <span class="comment">// FOR NOW -- PLEASE KEEP THIS IN SYNC WITH probeset-genotype</span>
<a name="l01302"></a>01302     vector&lt;string&gt; celFiles = <a class="code" href="classOptions.html#ab20d0083424d9d9a8b94da46f487f57a" title="Return a vector of values for a particular option.">getOptVector</a>(<span class="stringliteral">&quot;cels&quot;</span>);
<a name="l01303"></a>01303 
<a name="l01304"></a>01304     info.<a class="code" href="classAnalysisInfo.html#a2cc9986186213673fb4ca8decc70124e" title="What is the name of the program.">m_ProgramName</a> = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;program-name&quot;</span>);
<a name="l01305"></a>01305     info.<a class="code" href="classAnalysisInfo.html#a705a56532115d0c0c8f10644e633b64d" title="What program created this chp file.">m_ProgramVersion</a> = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;version-to-report&quot;</span>);
<a name="l01306"></a>01306     info.<a class="code" href="classAnalysisInfo.html#aa1f9eedaa9c2024dfde2af6a71155bb1" title="What company created this chp file.">m_ProgramCompany</a> = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;program-company&quot;</span>);
<a name="l01307"></a>01307     info.<a class="code" href="classAnalysisInfo.html#a89c1bfe7a31fccc70d06cb42f68c435e" title="What chip is this?">m_ChipType</a> = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chip-type&quot;</span>);
<a name="l01308"></a>01308     info.<a class="code" href="classAnalysisInfo.html#aa32b7ba5f935bc64316e903980e9acbe" title="What is our id for GCOS?">m_ProgID</a> = <span class="stringliteral">&quot;&quot;</span>;
<a name="l01309"></a>01309     info.<a class="code" href="classAnalysisInfo.html#ad906bb9bddd45bd2d3ed08ee3eff5fd3" title="GUID associated with a specific process.">m_ExecGuid</a> = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;exec-guid&quot;</span>);
<a name="l01310"></a>01310     info.<a class="code" href="classAnalysisInfo.html#ab9d3f400692f10270650ed0f02e8d2a0" title="GUID (global unique ID) associated with a specific analysis.">m_AnalysisGuid</a> = <a class="code" href="classaffxutil_1_1Guid.html#a2929c982cb9f3b5f6c5d7d818a8e6ac4">affxutil::Guid::GenerateNewGuid</a>();
<a name="l01311"></a>01311 
<a name="l01312"></a>01312     <span class="comment">// State and Execution info</span>
<a name="l01313"></a>01313     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(<span class="stringliteral">&quot;apt-engine&quot;</span>, <span class="stringliteral">&quot;ProbesetGenotypeStageEngine&quot;</span>);
<a name="l01314"></a>01314     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + <span class="stringliteral">&quot;program-name&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;program-name&quot;</span>));
<a name="l01315"></a>01315     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + <span class="stringliteral">&quot;command-line&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;command-line&quot;</span>));
<a name="l01316"></a>01316     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + <span class="stringliteral">&quot;exec-guid&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;exec-guid&quot;</span>));
<a name="l01317"></a>01317 
<a name="l01318"></a>01318     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + <span class="stringliteral">&quot;time-str&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;time-start&quot;</span>));
<a name="l01319"></a>01319     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + <span class="stringliteral">&quot;version&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;version-to-report&quot;</span>));
<a name="l01320"></a>01320     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + <span class="stringliteral">&quot;cvs-id&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;program-cvs-id&quot;</span>));
<a name="l01321"></a>01321     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + <span class="stringliteral">&quot;free-mem&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;free-mem-at-start&quot;</span>));<span class="comment"></span>
<a name="l01322"></a>01322 <span class="comment">    ///@todo add other state:</span>
<a name="l01323"></a>01323 <span class="comment">    ///@todo time-end state</span>
<a name="l01324"></a>01324 <span class="comment"></span>
<a name="l01325"></a>01325     <span class="comment">// Engine Options</span>
<a name="l01326"></a>01326     std::string subprefix = <span class="stringliteral">&quot;opt-&quot;</span>;
<a name="l01327"></a>01327     <span class="comment">// no do-ps-names</span>
<a name="l01328"></a>01328     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;chip-type&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chip-type&quot;</span>));
<a name="l01329"></a>01329     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;probe-count&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;probe-count&quot;</span>));
<a name="l01330"></a>01330     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;force&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;force&quot;</span>));
<a name="l01331"></a>01331     <span class="comment">// no precision</span>
<a name="l01332"></a>01332     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;out-dir&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;out-dir&quot;</span>));
<a name="l01333"></a>01333     <span class="comment">// cc-expr-chp-out-dir</span>
<a name="l01334"></a>01334     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;cc-md-chp-out-dir&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;cc-chp-out-dir&quot;</span>));
<a name="l01335"></a>01335     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;xda-chp-out-dir&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;xda-chp-out-dir&quot;</span>));
<a name="l01336"></a>01336     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;cdf-file&quot;</span>,       <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;cdf-file&quot;</span>)));
<a name="l01337"></a>01337     <span class="comment">// no spf-file </span>
<a name="l01338"></a>01338     <span class="comment">// no pgf-file</span>
<a name="l01339"></a>01339     <span class="comment">// no clf-file</span>
<a name="l01340"></a>01340     <span class="comment">// no bgp-file</span>
<a name="l01341"></a>01341     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;ps-list-file&quot;</span>,   <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;probeset-ids&quot;</span>)));
<a name="l01342"></a>01342     <span class="comment">// no meta-ps-file</span>
<a name="l01343"></a>01343     <span class="comment">// no-qc-groups-file</span>
<a name="l01344"></a>01344     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;kill-list&quot;</span>,      <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;kill-list&quot;</span>)));
<a name="l01345"></a>01345     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;temp-dir&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;temp-dir&quot;</span>));
<a name="l01346"></a>01346     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;use-disk&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;use-disk&quot;</span>));
<a name="l01347"></a>01347     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;diskCache&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;disk-cache&quot;</span>));
<a name="l01348"></a>01348     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;cel-count&quot;</span>, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(celFiles.size()));
<a name="l01349"></a>01349     <span class="keywordflow">for</span>(uint32_t i = 0; i &lt; celFiles.size(); i++) {
<a name="l01350"></a>01350       std::string paramName = prefix + subprefix + <span class="stringliteral">&quot;cel-&quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(i+1);
<a name="l01351"></a>01351       info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(paramName, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(celFiles[i]));
<a name="l01352"></a>01352     }
<a name="l01353"></a>01353 
<a name="l01354"></a>01354     <span class="comment">// Algorithm Options</span>
<a name="l01355"></a>01355     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;analysis-name&quot;</span>, info.<a class="code" href="classAnalysisInfo.html#a4e52df55ab7398557852b211c5a50d41" title="What algorithm are we reporting.">m_AlgName</a>);
<a name="l01356"></a>01356     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;set-analysis-name&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;set-analysis-name&quot;</span>));
<a name="l01357"></a>01357     <span class="comment">//    info.addParam(prefix + subprefix + &quot;set-analysis-name&quot;, getOpt(&quot;set-analysis-name&quot;));</span>
<a name="l01358"></a>01358 
<a name="l01359"></a>01359     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;feat-effect-file&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;use-feat-eff&quot;</span>)));
<a name="l01360"></a>01360     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;target-sketch-file&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;target-sketch&quot;</span>)));
<a name="l01361"></a>01361     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;reference-profile-file&quot;</span>,<a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;reference-profile&quot;</span>)));
<a name="l01362"></a>01362     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;do-residuals&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;feat-details&quot;</span>));
<a name="l01363"></a>01363     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;do-feature-effects&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;feat-effects&quot;</span>));
<a name="l01364"></a>01364     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;write-sketch&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;write-sketch&quot;</span>));
<a name="l01365"></a>01365     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;write-profile&quot;</span>,<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;write-profile&quot;</span>));
<a name="l01366"></a>01366 
<a name="l01367"></a>01367     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;a5-write-sketch&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-sketch&quot;</span>));
<a name="l01368"></a>01368     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;set-gender-method&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;set-gender-method&quot;</span>));
<a name="l01369"></a>01369     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;chrXProbeFile&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chrX-probes&quot;</span>)));
<a name="l01370"></a>01370     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;chrYProbeFile&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chrY-probes&quot;</span>)));
<a name="l01371"></a>01371     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;read-priors-brlmm&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-priors-brlmm&quot;</span>)));
<a name="l01372"></a>01372     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;initial-calls&quot;</span>, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(<span class="stringliteral">&quot;false&quot;</span>));
<a name="l01373"></a>01373     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;output-summaries&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;summaries&quot;</span>));
<a name="l01374"></a>01374     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;list-sample&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;list-sample&quot;</span>));
<a name="l01375"></a>01375     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;write-prior&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;write-prior&quot;</span>));
<a name="l01376"></a>01376     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;chrX-file&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chrX-snps&quot;</span>)));
<a name="l01377"></a>01377     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;chrX-snps&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chrX-snps&quot;</span>)));
<a name="l01378"></a>01378     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;no-gender-force&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;no-gender-force&quot;</span>));
<a name="l01379"></a>01379     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;special-snps&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;special-snps&quot;</span>)));
<a name="l01380"></a>01380     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;compute-prior&quot;</span>, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(<a class="code" href="classOptions.html#afcfab86e21cac5bba43e0858bd428e62" title="Get the integer value of an option.">getOptInt</a>(<span class="stringliteral">&quot;prior-size&quot;</span>)));
<a name="l01381"></a>01381     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;norm-size&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;norm-size&quot;</span>));
<a name="l01382"></a>01382     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;dm-thresh&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;dm-thresh&quot;</span>));
<a name="l01383"></a>01383     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;dm-het-mult&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;dm-hetmult&quot;</span>));
<a name="l01384"></a>01384     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;read-genders&quot;</span>,        <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-genders&quot;</span>)));
<a name="l01385"></a>01385     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;read-inbred&quot;</span>,         <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-inbred&quot;</span>)));
<a name="l01386"></a>01386     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;model-file-brlmm&quot;</span>   , <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-brlmm&quot;</span>)));
<a name="l01387"></a>01387     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;model-file-brlmmp&quot;</span>  , <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-brlmmp&quot;</span>)));
<a name="l01388"></a>01388     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;model-file-birdseed&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-birdseed&quot;</span>)));
<a name="l01389"></a>01389     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;genotypes-file&quot;</span>     , <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;genotypes&quot;</span>)));
<a name="l01390"></a>01390     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;dm-calls-out&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;dm-out&quot;</span>));
<a name="l01391"></a>01391     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;qmethod-spec&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;qmethod-spec&quot;</span>));
<a name="l01392"></a>01392     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;genotype-only&quot;</span>, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(!<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;all-types&quot;</span>)));
<a name="l01393"></a>01393 
<a name="l01394"></a>01394     <span class="comment">// Quantification Related Meta Info</span>
<a name="l01395"></a>01395     <span class="comment">// Add user defined meta data parameters</span>
<a name="l01396"></a>01396     vector&lt;pair&lt;string, string&gt; &gt; metaData = <a class="code" href="classBaseEngine.html#ac9fb139876b95877fb7412d0dc27cdd8" title="Get a vector of the key=value pairs specified via the &amp;quot;meta-data-info&amp;quot; parameters.">getMetaDataDescription</a>();
<a name="l01397"></a>01397     <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; metaData.size(); i++) {
<a name="l01398"></a>01398       pair&lt;string,string&gt; p = metaData[i];
<a name="l01399"></a>01399       info.<a class="code" href="classAnalysisInfo.html#a9482ae5e2f30e58b5a06d1f689edef59" title="Add client supplied key/value parameter pair to our information.">addClientMetaInfo</a>(p.first, p.second);
<a name="l01400"></a>01400     }
<a name="l01401"></a>01401 }
<a name="l01402"></a>01402 
<a name="l01403"></a>01403 <span class="comment"></span>
<a name="l01404"></a>01404 <span class="comment">/**</span>
<a name="l01405"></a>01405 <span class="comment"> * Construct an object for output that contains the program an</span>
<a name="l01406"></a>01406 <span class="comment"> * algorithm information.</span>
<a name="l01407"></a>01407 <span class="comment"> * @param as - AnalysisStream to get information from.</span>
<a name="l01408"></a>01408 <span class="comment"> * @param layout - Layout to get information from.</span>
<a name="l01409"></a>01409 <span class="comment"> * @param probesetNames - List of probeset names to output</span>
<a name="l01410"></a>01410 <span class="comment"> *</span>
<a name="l01411"></a>01411 <span class="comment"> * @return - Information for making CHP file.</span>
<a name="l01412"></a>01412 <span class="comment"> */</span>
<a name="l01413"></a>01413 <a class="code" href="classAnalysisInfo.html" title="Information about a particular analysis run.">AnalysisInfo</a> ProbesetGenotypeStageEngine::makeAnalysisInfo(
<a name="l01414"></a>01414                                                            <a class="code" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a> &amp;layout,
<a name="l01415"></a>01415                                                            <span class="keyword">const</span> std::vector&lt;const char*&gt; &amp;probesetNames,
<a name="l01416"></a>01416                                                            <a class="code" href="classGenderCalls.html" title="Class for determining gender from a cel set of cel files.">GenderCalls</a> *genders,
<a name="l01417"></a>01417                                                            <span class="keyword">const</span> std::string &amp;analysisName) {
<a name="l01418"></a>01418 
<a name="l01419"></a>01419   <a class="code" href="classAnalysisInfo.html" title="Information about a particular analysis run.">AnalysisInfo</a> info;
<a name="l01420"></a>01420   info.<a class="code" href="classAnalysisInfo.html#af50a44fcfb27fcf4aedbb5664857ff24" title="Length of the longest probeset name.">m_MaxPsNameLength</a> = layout.getMaxNameLength();
<a name="l01421"></a>01421   info.<a class="code" href="classAnalysisInfo.html#a302e3084afe43804e83c520b7beb463a" title="Number of expression probesets.">m_NumExpression</a> = layout.<a class="code" href="classChipLayout.html#acd976e4a3fa1141b5660ecab6f5c9b70" title="Number of expression probesets.">getNumExpressionPSets</a>();
<a name="l01422"></a>01422   info.<a class="code" href="classAnalysisInfo.html#a9a829523fde5f52310057a05b01ba601" title="Number of genotyping expression.">m_NumGenotyping</a> = layout.<a class="code" href="classChipLayout.html#acc2755565e72c8bcb52365c23572c216" title="Number of genotyping expression.">getNumGenotypeingPSets</a>();
<a name="l01423"></a>01423   info.<a class="code" href="classAnalysisInfo.html#a3fc5ed6fab19e4d4b155978fb6d6f3e3" title="How many snps to report in CHP file.">m_NumReporting</a> = layout.<a class="code" href="classChipLayout.html#acc2755565e72c8bcb52365c23572c216" title="Number of genotyping expression.">getNumGenotypeingPSets</a>();
<a name="l01424"></a>01424   info.<a class="code" href="classAnalysisInfo.html#a540d9d55c05541e8fbe95474d6dbb7fb" title="How many rows are on the grid of the chip.">m_NumRows</a> = layout.<a class="code" href="classChipLayout.html#a5b5ed8dcd11bdfbbecfbce63e2d0b342" title="Get the number of x counts.">getXCount</a>();
<a name="l01425"></a>01425   info.<a class="code" href="classAnalysisInfo.html#a39da235062cb23d7392f29cca25d8da7" title="How many columns are the grid of the chip.">m_NumCols</a> = layout.<a class="code" href="classChipLayout.html#ae602f7c506b7ae9355844f06267ff528" title="Get the number of y counts.">getYCount</a>();
<a name="l01426"></a>01426   info.<a class="code" href="classAnalysisInfo.html#a1c684133674d56f7588c24c56c6a4c75" title="Name of the analysis.">m_AnalysisName</a> = analysisName;
<a name="l01427"></a>01427   info.<a class="code" href="classAnalysisInfo.html#a4e52df55ab7398557852b211c5a50d41" title="What algorithm are we reporting.">m_AlgName</a> = analysisName;
<a name="l01428"></a>01428   info.<a class="code" href="classAnalysisInfo.html#ac85efbbb6618cb87c39511af32c311a7" title="How many probesets are on the chip and in cdf file. This is /// critical as probesets must be the exa...">m_NumProbeSets</a> = layout.<a class="code" href="classChipLayout.html#a71df054a01d62ca33d6b8de8c8c5c3ca" title="How many probe sets in Pgf file?">getProbeSetCount</a>(); <span class="comment">//probesetNames.size();</span>
<a name="l01429"></a>01429   info.<a class="code" href="classAnalysisInfo.html#a1eaf6c81440945d17b192a9d7196e788" title="What type of chp file is this? For our purposes here it is genotyping.">m_ProbeSetType</a> = affxcdf::GenotypingProbeSetType;
<a name="l01430"></a>01430   <span class="comment">// @todo refactor - what is output prefix?</span>
<a name="l01431"></a>01431   fillInAnalysisInfo(info, <span class="stringliteral">&quot;apt-&quot;</span>);
<a name="l01432"></a>01432 
<a name="l01433"></a>01433   <span class="comment">// @todo refactor - make this work</span>
<a name="l01434"></a>01434   <span class="comment">// QuantGTypeMethod *method =</span>
<a name="l01435"></a>01435   <span class="comment">//     dynamic_cast&lt;QuantGTypeMethod *&gt;(as-&gt;getQuantMethod());</span>
<a name="l01436"></a>01436   <span class="comment">// float min = method-&gt;getMinThresh();</span>
<a name="l01437"></a>01437   <span class="comment">// info.addParam(&quot;min-thresh&quot;,ToStr(min));</span>
<a name="l01438"></a>01438   <span class="comment">// float max = method-&gt;getMaxThresh();</span>
<a name="l01439"></a>01439   <span class="comment">// info.addParam(&quot;max-thresh&quot;,ToStr(max));</span>
<a name="l01440"></a>01440 
<a name="l01441"></a>01441   info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(<span class="stringliteral">&quot;gender_method_used&quot;</span>,genders-&gt;getGenderName());
<a name="l01442"></a>01442 
<a name="l01443"></a>01443   <span class="comment">// this is a &quot;magic&quot; value that works with GCOS/GTYPE according to Richard</span>
<a name="l01444"></a>01444   info.<a class="code" href="classAnalysisInfo.html#aa32b7ba5f935bc64316e903980e9acbe" title="What is our id for GCOS?">m_ProgID</a> = <span class="stringliteral">&quot;GDMTAnalysis.BRMBaseCall.1&quot;</span>;
<a name="l01445"></a>01445 
<a name="l01446"></a>01446   <span class="comment">// If CHP Output, then we need list of probeset names to output</span>
<a name="l01447"></a>01447   <span class="comment">// Otherwise do not waste the memory</span>
<a name="l01448"></a>01448   <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;cc-chp-output&quot;</span>) || <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;xda-chp-output&quot;</span>)) 
<a name="l01449"></a>01449     info.<a class="code" href="classAnalysisInfo.html#a21ab2fbfe201742b20f48ac42a12a07e" title="Vector of probeset names and groups as they should appear in the chp file. /// this memory is owned e...">m_ProbesetNames</a> = probesetNames;
<a name="l01450"></a>01450 
<a name="l01451"></a>01451   <span class="comment">// Sanity check</span>
<a name="l01452"></a>01452   <a class="code" href="classErr.html#ad1a633f602fe60689cdc12d9ca1842e9" title="errAbort based version of assert with an error message.">Err::check</a>(info.m_ParamValues.size() == info.<a class="code" href="classAnalysisInfo.html#ad48864dcd9e2e22d9d74dfbbb00fff04" title="These next two vectors are matched pairs, one contains the /// keys and the other the values...">m_ParamNames</a>.size(),
<a name="l01453"></a>01453              <span class="stringliteral">&quot;AnalysisInfo - Names and values out of sync.&quot;</span>);
<a name="l01454"></a>01454   <span class="keywordflow">return</span> info;
<a name="l01455"></a>01455 }
<a name="l01456"></a>01456 
<a name="l01457"></a>01457 <span class="comment"></span>
<a name="l01458"></a>01458 <span class="comment">/**</span>
<a name="l01459"></a>01459 <span class="comment"> * Use a string specification to create an analysis stream.</span>
<a name="l01460"></a>01460 <span class="comment"> *</span>
<a name="l01461"></a>01461 <span class="comment"> * @param analysis - vector to be filled in with analysis streams.</span>
<a name="l01462"></a>01462 <span class="comment"> * @param asFactory - factory for converting strings to streams.</span>
<a name="l01463"></a>01463 <span class="comment"> * @param layout - Probeset specifications.</span>
<a name="l01464"></a>01464 <span class="comment"> * @param probesetNames - Names probesets in order for chp files</span>
<a name="l01465"></a>01465 <span class="comment"> * @param numProbeSets -</span>
<a name="l01466"></a>01466 <span class="comment"> */</span>
<a name="l01467"></a>01467 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::createAnalysisFromSpec(
<a name="l01468"></a>01468                             std::vector&lt;AnalysisStream *&gt; &amp;analysisStreams,
<a name="l01469"></a>01469                             <a class="code" href="classAnalysisStreamFactory.html" title="Object for making new analysis streams from a text specification.">AnalysisStreamFactory</a> &amp;asFactory,
<a name="l01470"></a>01470                             <a class="code" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a> &amp;layout,
<a name="l01471"></a>01471                             std::vector&lt;const char*&gt; &amp;probesetNames,
<a name="l01472"></a>01472                             <span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> numProbeSets) {
<a name="l01473"></a>01473   <span class="comment">// *Warning* People really did a cut and paste number on this</span>
<a name="l01474"></a>01474   <span class="comment">// function and then things have evolved since then, be careful.</span>
<a name="l01475"></a>01475 
<a name="l01476"></a>01476   <span class="keywordtype">string</span> outDir = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;out-dir&quot;</span>);
<a name="l01477"></a>01477   <span class="keywordtype">bool</span> writeModels = <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;write-models&quot;</span>);
<a name="l01478"></a>01478   <span class="keywordtype">string</span> precompFeatureEff = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;use-feat-eff&quot;</span>);
<a name="l01479"></a>01479   <span class="keywordtype">string</span> qMethodSpec = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;qmethod-spec&quot;</span>);
<a name="l01480"></a>01480   vector&lt;string&gt; celFiles = <a class="code" href="classOptions.html#ab20d0083424d9d9a8b94da46f487f57a" title="Return a vector of values for a particular option.">getOptVector</a>(<span class="stringliteral">&quot;cels&quot;</span>);
<a name="l01481"></a>01481 
<a name="l01482"></a>01482   <span class="comment">// Create all the analysis paths requested.</span>
<a name="l01483"></a>01483   vector&lt;string&gt; analysisStrings = <a class="code" href="classOptions.html#ab20d0083424d9d9a8b94da46f487f57a" title="Return a vector of values for a particular option.">getOptVector</a>(<span class="stringliteral">&quot;analysis&quot;</span>);
<a name="l01484"></a>01484   <span class="keywordflow">for</span>(<span class="keywordtype">int</span> i = 0; i &lt; analysisStrings.size(); i++) {
<a name="l01485"></a>01485     <a class="code" href="classAnalysisStream.html" title="Utility class for centralizing analysis pathway.">AnalysisStream</a> *as = NULL;
<a name="l01486"></a>01486     <span class="keywordflow">if</span>(analysisStrings[i] == <span class="stringliteral">&quot;&quot;</span>)
<a name="l01487"></a>01487         <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Invalid (blank) analysis specification&quot;</span>);
<a name="l01488"></a>01488     as = asFactory.<a class="code" href="classAnalysisStreamFactory.html#a9f62371512e3164fd2a2df046dd69dba" title="Factory for creatng Genotype SNP chip AnalysisStream from a string description.">constructGTypeAnalysisStream</a>(analysisStrings[i],
<a name="l01489"></a>01489                                                 layout,
<a name="l01490"></a>01490                                                 m_stdMethods,
<a name="l01491"></a>01491                                                 <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;set-analysis-name&quot;</span>));
<a name="l01492"></a>01492     <a class="code" href="classQuantMethod.html" title="QuantMethod - Interface for computing quantification summaries from PM intensities grouped into probe...">QuantMethod</a> *method = as-&gt;<a class="code" href="classAnalysisStream.html#af5d7e7ca7a32a6f189290b5ef1888f64" title="Get the quantification method used.">getQuantMethod</a>();
<a name="l01493"></a>01493 
<a name="l01494"></a>01494     <span class="comment">// BRLMM method detected, do stuff</span>
<a name="l01495"></a>01495     <span class="keywordflow">if</span>(InstanceOf(method, <a class="code" href="classQuantBRLMM.html" title="Implements a SNP genotype calling method motivated by Nusrat Rabbee&amp;#39;s RLMM (Robust Linear Model w...">QuantBRLMM</a>)) {
<a name="l01496"></a>01496       <a class="code" href="classQuantBRLMM.html" title="Implements a SNP genotype calling method motivated by Nusrat Rabbee&amp;#39;s RLMM (Robust Linear Model w...">QuantBRLMM</a> *qBRLMM = NULL;
<a name="l01497"></a>01497       qBRLMM =  <span class="keyword">static_cast&lt;</span><a class="code" href="classQuantBRLMM.html" title="Implements a SNP genotype calling method motivated by Nusrat Rabbee&amp;#39;s RLMM (Robust Linear Model w...">QuantBRLMM</a> *<span class="keyword">&gt;</span>(method);
<a name="l01498"></a>01498 
<a name="l01499"></a>01499       <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-brlmm&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l01500"></a>01500         qBRLMM-&gt;<a class="code" href="classQuantBRLMM.html#a873e8c5e7d97ea648f749d1c410706f0" title="Load up the snp models from a text file.">readSnpModels</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-brlmm&quot;</span>));
<a name="l01501"></a>01501       }
<a name="l01502"></a>01502       <span class="comment">//</span>
<a name="l01503"></a>01503       {
<a name="l01504"></a>01504         <span class="keywordtype">string</span> outfile = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(outDir,as-&gt;<a class="code" href="classAnalysisStream.html#a79af545326dd23f70df846094f1573c5" title="What is the name of this particular analysis stream?">getName</a>() + <span class="stringliteral">&quot;.snp-models&quot;</span>);
<a name="l01505"></a>01505         <span class="keywordflow">if</span> (writeModels) {
<a name="l01506"></a>01506           qBRLMM-&gt;<a class="code" href="classQuantBRLMM.html#ad5e489c5edabd5fe6c0f7d1e34f777be" title="Tell brlmm to output snp models to file.">setSnpModelOut</a>(outfile,affx::TsvReport::FMT_TSV);
<a name="l01507"></a>01507         }
<a name="l01508"></a>01508         <span class="keywordflow">else</span> <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-write-models&quot;</span>)) {
<a name="l01509"></a>01509             <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-write-models-use-global&quot;</span>)) {
<a name="l01510"></a>01510                 <span class="keywordflow">if</span>(m_a5_global_output_group == NULL)
<a name="l01511"></a>01511                     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;--a5-write-models-use-global option given, but no global file. Must specify --a5-global-file.&quot;</span>);
<a name="l01512"></a>01512                 qBRLMM-&gt;<a class="code" href="classQuantBRLMM.html#ac85a70f66b43e43943f3da174d0378de" title="Output stream for printing snp specific models.">m_SnpModelsTsv</a>.setA5SharedGroup(m_a5_global_output_group);
<a name="l01513"></a>01513             }
<a name="l01514"></a>01514             qBRLMM-&gt;<a class="code" href="classQuantBRLMM.html#ad5e489c5edabd5fe6c0f7d1e34f777be" title="Tell brlmm to output snp models to file.">setSnpModelOut</a>(outfile,affx::TsvReport::FMT_A5);
<a name="l01515"></a>01515         }
<a name="l01516"></a>01516       }
<a name="l01517"></a>01517 
<a name="l01518"></a>01518       <span class="comment">// Add quantification method.</span>
<a name="l01519"></a>01519       <a class="code" href="classQuantMethodFactory.html" title="Factory class for making chip streams based on a string representation.">QuantMethodFactory</a> factory(QuantMethodFactory::Expression);
<a name="l01520"></a>01520       <span class="keywordflow">if</span>(precompFeatureEff != <span class="stringliteral">&quot;&quot;</span>)
<a name="l01521"></a>01521         factory.readPrecompFeatureEffectsFromFile(layout.<a class="code" href="classChipLayout.html#abedbffee7c6a978e3e5545d659a78ec7" title="Get the total number of probes on this chip.">getProbeCount</a>(), precompFeatureEff, layout);
<a name="l01522"></a>01522       <a class="code" href="classQuantExprMethod.html" title="QuantExprMethod - Interface for computing quantification summaries from PM intensities grouped into p...">QuantExprMethod</a> *eMethod = factory.quantExprMethodForString(qMethodSpec, layout,
<a name="l01523"></a>01523                                                                   QuantMethodFactory::Expression);
<a name="l01524"></a>01524       <span class="comment">// If we want to report allele specific summaries add a reporter for that.</span>
<a name="l01525"></a>01525       qBRLMM-&gt;setQuantExprMethod(eMethod);
<a name="l01526"></a>01526     } <span class="comment">// end of stuff for BRLMM detected</span>
<a name="l01527"></a>01527 
<a name="l01528"></a>01528     <span class="keywordflow">if</span>(InstanceOf(method, <a class="code" href="classQuantBirdseed.html">QuantBirdseed</a>)) {
<a name="l01529"></a>01529       <a class="code" href="classQuantBirdseed.html">QuantBirdseed</a> *qBirdseed = NULL;
<a name="l01530"></a>01530       qBirdseed =  <span class="keyword">static_cast&lt;</span><a class="code" href="classQuantBirdseed.html">QuantBirdseed</a> *<span class="keyword">&gt;</span>(method);
<a name="l01531"></a>01531 
<a name="l01532"></a>01532       <span class="comment">// Add quantification method.</span>
<a name="l01533"></a>01533       <a class="code" href="classQuantMethodFactory.html" title="Factory class for making chip streams based on a string representation.">QuantMethodFactory</a> factory(QuantMethodFactory::Expression);
<a name="l01534"></a>01534       <span class="keywordflow">if</span>(precompFeatureEff != <span class="stringliteral">&quot;&quot;</span>)
<a name="l01535"></a>01535         factory.readPrecompFeatureEffectsFromFile(layout.<a class="code" href="classChipLayout.html#abedbffee7c6a978e3e5545d659a78ec7" title="Get the total number of probes on this chip.">getProbeCount</a>(), precompFeatureEff, layout);
<a name="l01536"></a>01536       <a class="code" href="classQuantExprMethod.html" title="QuantExprMethod - Interface for computing quantification summaries from PM intensities grouped into p...">QuantExprMethod</a> *eMethod = factory.quantExprMethodForString(qMethodSpec, layout,
<a name="l01537"></a>01537                                                                   QuantMethodFactory::Expression);
<a name="l01538"></a>01538       qBirdseed-&gt;setQuantExprMethod(eMethod);
<a name="l01539"></a>01539       qBirdseed-&gt;setVerbosity(<a class="code" href="classOptions.html#afcfab86e21cac5bba43e0858bd428e62" title="Get the integer value of an option.">getOptInt</a>(<span class="stringliteral">&quot;verbose&quot;</span>));
<a name="l01540"></a>01540       <span class="comment">//</span>
<a name="l01541"></a>01541       <span class="keywordflow">if</span>(writeModels) {
<a name="l01542"></a>01542         <span class="keywordtype">string</span> outfile = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(outDir,as-&gt;<a class="code" href="classAnalysisStream.html#a79af545326dd23f70df846094f1573c5" title="What is the name of this particular analysis stream?">getName</a>()+<span class="stringliteral">&quot;.snp-models.txt&quot;</span>);
<a name="l01543"></a>01543         qBirdseed-&gt;setClusterOutFile(outfile);
<a name="l01544"></a>01544       }
<a name="l01545"></a>01545       <span class="keywordflow">else</span> <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-write-models&quot;</span>)) {<span class="comment"></span>
<a name="l01546"></a>01546 <span class="comment">          ///@todo implement models in A5 for birdseed</span>
<a name="l01547"></a>01547 <span class="comment"></span>          <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;--a5-write-models for birdseed not implemented&quot;</span>);
<a name="l01548"></a>01548       }
<a name="l01549"></a>01549     } <span class="comment">// end of stuff for Birdseed detected</span>
<a name="l01550"></a>01550 
<a name="l01551"></a>01551     <span class="comment">// LabelZ || LabelZMulti</span>
<a name="l01552"></a>01552     <span class="keywordflow">if</span>(InstanceOf(method, <a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a>)) {
<a name="l01553"></a>01553       <a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a> *qLabelZ = NULL;
<a name="l01554"></a>01554       qLabelZ =  <span class="keyword">static_cast&lt;</span><a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a> *<span class="keyword">&gt;</span>(method);
<a name="l01555"></a>01555       
<a name="l01556"></a>01556       <span class="comment">//</span>
<a name="l01557"></a>01557       <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-brlmmp&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l01558"></a>01558         qLabelZ-&gt;<a class="code" href="classQuantLabelZ.html#ab4a3970a1d381a0885d82ebc66ddb3ca" title="read in the snp specific priors from a file">readSnpPriorMap</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-brlmmp&quot;</span>));
<a name="l01559"></a>01559       }
<a name="l01560"></a>01560       <span class="keywordflow">else</span> <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-models-input-global&quot;</span>) || <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-models-input-file&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l01561"></a>01561         <span class="keywordtype">string</span> groupName = <span class="stringliteral">&quot;/&quot;</span>;
<a name="l01562"></a>01562         <span class="keywordtype">string</span> dataName = as-&gt;<a class="code" href="classAnalysisStream.html#a79af545326dd23f70df846094f1573c5" title="What is the name of this particular analysis stream?">getName</a>() + <span class="stringliteral">&quot;.snp-posteriors&quot;</span>;
<a name="l01563"></a>01563         <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-models-input-name&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>) 
<a name="l01564"></a>01564             dataName = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-models-input-name&quot;</span>);
<a name="l01565"></a>01565         <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-models-input-group&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>) 
<a name="l01566"></a>01566             groupName = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-models-input-group&quot;</span>);
<a name="l01567"></a>01567         <span class="keywordflow">else</span> <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-input-group&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>) 
<a name="l01568"></a>01568             groupName = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-input-group&quot;</span>);
<a name="l01569"></a>01569         <span class="keywordflow">else</span> <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-group&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>) 
<a name="l01570"></a>01570             groupName = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-group&quot;</span>);
<a name="l01571"></a>01571     
<a name="l01572"></a>01572         <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-models-input-global&quot;</span>)) {
<a name="l01573"></a>01573             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;Loading models from global A5 file, group &#39;&quot;</span> + groupName + <span class="stringliteral">&quot;&#39;, data &#39;&quot;</span> + dataName + <span class="stringliteral">&quot;&#39;&quot;</span>);
<a name="l01574"></a>01574             <span class="keywordflow">if</span>(m_a5_global_input_file == NULL)
<a name="l01575"></a>01575                 <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;--a5-models-input-global option given, but no global input file. Must specify --a5-global-file.&quot;</span>);
<a name="l01576"></a>01576             <a class="code" href="classaffx_1_1File5__Group.html">affx::File5_Group</a> *a5group = m_a5_global_input_file-&gt;openGroup(groupName,affx::FILE5_OPEN);
<a name="l01577"></a>01577             <a class="code" href="classaffx_1_1File5__Tsv.html">affx::File5_Tsv</a>* tsv5 = a5group-&gt;openTsv(dataName,affx::FILE5_OPEN);
<a name="l01578"></a>01578             qLabelZ-&gt;readSnpPriorMap_tsv5(tsv5);
<a name="l01579"></a>01579             tsv5-&gt;close();
<a name="l01580"></a>01580             <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(tsv5);
<a name="l01581"></a>01581             a5group-&gt;close();
<a name="l01582"></a>01582             <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(a5group);
<a name="l01583"></a>01583         } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-models-input-file&quot;</span>)!=<span class="stringliteral">&quot;&quot;</span>) {
<a name="l01584"></a>01584             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;Loading models from &#39;&quot;</span> + <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-models-input-file&quot;</span>) + <span class="stringliteral">&quot;&#39; A5 file, group &#39;&quot;</span> + groupName + <span class="stringliteral">&quot;&#39;, data &#39;&quot;</span> + dataName + <span class="stringliteral">&quot;&#39;&quot;</span>);
<a name="l01585"></a>01585             <a class="code" href="classaffx_1_1File5__File.html">affx::File5_File</a>  *a5file = <span class="keyword">new</span> <a class="code" href="classaffx_1_1File5__File.html">affx::File5_File</a>();
<a name="l01586"></a>01586             a5file-&gt;open(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-models-input-file&quot;</span>),affx::FILE5_OPEN|affx::FILE5_RO);
<a name="l01587"></a>01587             <a class="code" href="classaffx_1_1File5__Group.html">affx::File5_Group</a> *a5group = a5file-&gt;openGroup(groupName,affx::FILE5_OPEN);
<a name="l01588"></a>01588             <a class="code" href="classaffx_1_1File5__Tsv.html">affx::File5_Tsv</a>* tsv5 = a5group-&gt;openTsv(dataName,affx::FILE5_OPEN);
<a name="l01589"></a>01589             qLabelZ-&gt;readSnpPriorMap_tsv5(tsv5);
<a name="l01590"></a>01590             tsv5-&gt;close();
<a name="l01591"></a>01591             <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(tsv5);
<a name="l01592"></a>01592             a5group-&gt;close();
<a name="l01593"></a>01593             <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(a5group);
<a name="l01594"></a>01594             a5file-&gt;close();
<a name="l01595"></a>01595             <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(a5file);
<a name="l01596"></a>01596         }
<a name="l01597"></a>01597       }
<a name="l01598"></a>01598       <span class="comment">// block for scoping outfile</span>
<a name="l01599"></a>01599       {
<a name="l01600"></a>01600         <span class="keywordtype">string</span> outfile = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(outDir,as-&gt;<a class="code" href="classAnalysisStream.html#a79af545326dd23f70df846094f1573c5" title="What is the name of this particular analysis stream?">getName</a>() + <span class="stringliteral">&quot;.snp-posteriors&quot;</span>);
<a name="l01601"></a>01601         <span class="keywordflow">if</span>(writeModels) {
<a name="l01602"></a>01602           qLabelZ-&gt;<a class="code" href="classQuantLabelZ.html#a37af5d01565df9a913e6f93033a836d9" title="Pick the output version based on the file format. (txt=&amp;gt;v1,a5=&amp;gt;v2).">writeSnpPosteriorTsv</a>(outfile,affx::TsvReport::FMT_TSV);
<a name="l01603"></a>01603         }
<a name="l01604"></a>01604         <span class="keywordflow">else</span> <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-write-models&quot;</span>)) {
<a name="l01605"></a>01605             <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-write-models-use-global&quot;</span>)) {
<a name="l01606"></a>01606                 <span class="keywordflow">if</span>(m_a5_global_output_group == NULL)
<a name="l01607"></a>01607                     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;--a5-write-models-use-global option given, but no global file. Must specify --a5-global-file.&quot;</span>);
<a name="l01608"></a>01608                 qLabelZ-&gt;m_SnpPosteriorTsv.setA5SharedGroup(m_a5_global_output_group);
<a name="l01609"></a>01609             }
<a name="l01610"></a>01610             qLabelZ-&gt;<a class="code" href="classQuantLabelZ.html#a37af5d01565df9a913e6f93033a836d9" title="Pick the output version based on the file format. (txt=&amp;gt;v1,a5=&amp;gt;v2).">writeSnpPosteriorTsv</a>(outfile,affx::TsvReport::FMT_A5);
<a name="l01611"></a>01611         }
<a name="l01612"></a>01612       }
<a name="l01613"></a>01613       <span class="comment">// block for scoping outfile</span>
<a name="l01614"></a>01614       {
<a name="l01615"></a>01615         <span class="keywordtype">string</span> outfile = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(outDir,as-&gt;<a class="code" href="classAnalysisStream.html#a79af545326dd23f70df846094f1573c5" title="What is the name of this particular analysis stream?">getName</a>() + <span class="stringliteral">&quot;.normalized-summary&quot;</span>);
<a name="l01616"></a>01616         <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;summaries&quot;</span>)) {
<a name="l01617"></a>01617           qLabelZ-&gt;writeNormSummaryTsv(outfile, celFiles,affx::TsvReport::FMT_TSV);
<a name="l01618"></a>01618         }
<a name="l01619"></a>01619         <span class="keywordflow">else</span> <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-summaries&quot;</span>)) {
<a name="l01620"></a>01620             <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-summaries-use-global&quot;</span>)) {
<a name="l01621"></a>01621                 <span class="keywordflow">if</span>(m_a5_global_output_group == NULL) 
<a name="l01622"></a>01622                     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;--a5-summaries-use-global option given, but no global file. Must specify --a5-global-file.&quot;</span>);
<a name="l01623"></a>01623                 qLabelZ-&gt;m_NormSummaryTsv.setA5SharedGroup(m_a5_global_output_group);
<a name="l01624"></a>01624             } 
<a name="l01625"></a>01625             qLabelZ-&gt;writeNormSummaryTsv(outfile, celFiles,affx::TsvReport::FMT_A5);
<a name="l01626"></a>01626         }
<a name="l01627"></a>01627       }
<a name="l01628"></a>01628       <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;select-probes&quot;</span>))
<a name="l01629"></a>01629         {
<a name="l01630"></a>01630           <span class="keywordtype">string</span> outfile =<a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(outDir,as-&gt;<a class="code" href="classAnalysisStream.html#a79af545326dd23f70df846094f1573c5" title="What is the name of this particular analysis stream?">getName</a>() + <span class="stringliteral">&quot;.select-probes&quot;</span>);
<a name="l01631"></a>01631           qLabelZ-&gt;<a class="code" href="classQuantLabelZ.html#ad7cae32c6830efa7475fb948f2442682" title="Write header to the selection file so people don&amp;#39;t have to look up everything.">writeSnpProbeTsv</a>(outfile,affx::TsvReport::FMT_TSV);
<a name="l01632"></a>01632         }
<a name="l01633"></a>01633 
<a name="l01634"></a>01634       <span class="comment">// Add quantification method.</span>
<a name="l01635"></a>01635       <a class="code" href="classQuantMethodFactory.html" title="Factory class for making chip streams based on a string representation.">QuantMethodFactory</a> factory(QuantMethodFactory::Expression);
<a name="l01636"></a>01636       <a class="code" href="classQuantMethodFactory.html#a148bcae58f120aedb98a43ac358c146e">QuantMethodFactory::setupQuantMethodFactory</a>(
<a name="l01637"></a>01637             factory,
<a name="l01638"></a>01638             m_a5_global_input_file,
<a name="l01639"></a>01639             m_a5_global_input_group,
<a name="l01640"></a>01640             m_a5_global_output_file,
<a name="l01641"></a>01641             m_a5_global_output_group,
<a name="l01642"></a>01642             <a class="code" href="classOptions.html#afcfab86e21cac5bba43e0858bd428e62" title="Get the integer value of an option.">getOptInt</a>(<span class="stringliteral">&quot;probe-count&quot;</span>),
<a name="l01643"></a>01643             <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;out-dir&quot;</span>),
<a name="l01644"></a>01644             <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;use-feat-eff&quot;</span>),
<a name="l01645"></a>01645             <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-feature-effects-input-global&quot;</span>),
<a name="l01646"></a>01646             <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-feature-effects-input-file&quot;</span>),
<a name="l01647"></a>01647             <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-feature-effects-input-name&quot;</span>),
<a name="l01648"></a>01648             <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-feature-effects-input-group&quot;</span>),
<a name="l01649"></a>01649             <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-input-group&quot;</span>),
<a name="l01650"></a>01650             <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-group&quot;</span>),
<a name="l01651"></a>01651             <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;set-analysis-name&quot;</span>),
<a name="l01652"></a>01652             <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;qmethod-spec&quot;</span>),
<a name="l01653"></a>01653             layout
<a name="l01654"></a>01654       );
<a name="l01655"></a>01655       <a class="code" href="classQuantExprMethod.html" title="QuantExprMethod - Interface for computing quantification summaries from PM intensities grouped into p...">QuantExprMethod</a> *eMethod = factory.quantExprMethodForString(qMethodSpec, layout, QuantMethodFactory::Expression);
<a name="l01656"></a>01656       qLabelZ-&gt;<a class="code" href="classQuantLabelZ.html#ac8c2cc4ab6dc319d4c0bff9b3f299642" title="Set the quantification method to use for summarizing individual alleles.">setQuantExprMethod</a>(eMethod);
<a name="l01657"></a>01657     } <span class="comment">// end of stuff for LabelZ detected</span>
<a name="l01658"></a>01658 
<a name="l01659"></a>01659     analysisStreams.push_back(as);
<a name="l01660"></a>01660   }
<a name="l01661"></a>01661 }
<a name="l01662"></a>01662 <span class="comment"></span>
<a name="l01663"></a>01663 <span class="comment">/**</span>
<a name="l01664"></a>01664 <span class="comment"> * Get the intial calls from the DmListener and give them to analysis</span>
<a name="l01665"></a>01665 <span class="comment"> * methods that need them.</span>
<a name="l01666"></a>01666 <span class="comment"> *</span>
<a name="l01667"></a>01667 <span class="comment"> * @param layout - Probeset definitions.</span>
<a name="l01668"></a>01668 <span class="comment"> * @param dmCaller - Object that does the DM calls.</span>
<a name="l01669"></a>01669 <span class="comment"> * @param knownGenotypes - Known genotype object to fill in.</span>
<a name="l01670"></a>01670 <span class="comment"> * @param haploidSnps - Snps on chrX.</span>
<a name="l01671"></a>01671 <span class="comment"> * @param analysis - Analysis methods to be done.</span>
<a name="l01672"></a>01672 <span class="comment"> * @param precompGTypes -</span>
<a name="l01673"></a>01673 <span class="comment"> */</span>
<a name="l01674"></a>01674 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::setInitialCalls(<a class="code" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a> &amp;layout, <a class="code" href="classGenoSeed.html" title="Abstract interface for a genotyping oracle used for genotyping seeds in brlmm.">GenoSeed</a> &amp;seeds,
<a name="l01675"></a>01675                      std::map&lt;<span class="keyword">const</span> <span class="keywordtype">char</span> *,std::vector&lt;affx::GType&gt;, <a class="code" href="structUtil_1_1ltstr.html" title="Comparison object for use in map for char *.">Util::ltstr</a> &gt; *knownGenotypes,
<a name="l01676"></a>01676                      std::map&lt;std::string,bool&gt; &amp;haploidSnps,
<a name="l01677"></a>01677                      std::vector&lt;AnalysisStream *&gt; &amp;analysis,
<a name="l01678"></a>01678                      std::vector&lt;const char *&gt; toRunProbesets) {
<a name="l01679"></a>01679 
<a name="l01680"></a>01680   <a class="code" href="classQuantMethod.html" title="QuantMethod - Interface for computing quantification summaries from PM intensities grouped into probe...">QuantMethod</a> *qMethod = NULL;
<a name="l01681"></a>01681   <span class="keywordtype">int</span> genotypable_probeset_count = 0;
<a name="l01682"></a>01682 
<a name="l01683"></a>01683   <span class="comment">// Set the initial calls in DM</span>
<a name="l01684"></a>01684   <span class="keywordflow">for</span>(<span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> i = 0; i &lt; analysis.size(); i++) {
<a name="l01685"></a>01685     <a class="code" href="classAnalysisStream.html" title="Utility class for centralizing analysis pathway.">AnalysisStream</a> *as = analysis[i];
<a name="l01686"></a>01686     qMethod = as-&gt;<a class="code" href="classAnalysisStream.html#af5d7e7ca7a32a6f189290b5ef1888f64" title="Get the quantification method used.">getQuantMethod</a>();
<a name="l01687"></a>01687     <span class="keywordflow">if</span>(InstanceOf(qMethod, <a class="code" href="classQuantBRLMM.html" title="Implements a SNP genotype calling method motivated by Nusrat Rabbee&amp;#39;s RLMM (Robust Linear Model w...">QuantBRLMM</a>)) {
<a name="l01688"></a>01688       <span class="comment">// Get the initial genotype calls</span>
<a name="l01689"></a>01689       <span class="keywordflow">for</span>(uint32_t i = 0; i &lt; toRunProbesets.size(); i++) {
<a name="l01690"></a>01690           <span class="keyword">const</span> <span class="keywordtype">char</span> *name = toRunProbesets[i];
<a name="l01691"></a>01691         <a class="code" href="classProbeListPacked.html">ProbeListPacked</a> pList = layout.<a class="code" href="classChipLayout.html#ae465f92816b1c9983a45d1f73293cc7b" title="Find a probe set using the name as a key.">getProbeListByName</a>(name);
<a name="l01692"></a>01692         <span class="comment">// </span>
<a name="l01693"></a>01693         <span class="keywordflow">if</span> (pList.isNull()) {
<a name="l01694"></a>01694           <a class="code" href="classVerbose.html#a34e1d936178741eeb78de93c5497e1b8" title="Print a warning message.">Verbose::warn</a>(1,std::string(<span class="stringliteral">&quot;ProbesetGenotypeStageEngine::setInitialCalls(): missing probeset named: &#39;&quot;</span>)+name+<span class="stringliteral">&quot;&#39;&quot;</span>);
<a name="l01695"></a>01695           <span class="keywordflow">continue</span>;
<a name="l01696"></a>01696         }
<a name="l01697"></a>01697         <span class="comment">//</span>
<a name="l01698"></a>01698         <span class="keywordflow">if</span>((pList.get_type() == ProbeSet::GenoType &amp;&amp; (pList.block_cnt() == 2 || pList.block_cnt() == 4)) || pList.get_type() == ProbeSet::Marker || pList.get_type() == ProbeSet::MultichannelMarker) {
<a name="l01699"></a>01699             genotypable_probeset_count++;
<a name="l01700"></a>01700             vector&lt;GType&gt; callVec = seeds.<a class="code" href="classGenoSeed.html#ab3c2932ea7190946962b670a5a5ed259" title="Get the genotype calls for a particular probeset.">getGenoCalls</a>(name);
<a name="l01701"></a>01701             (*knownGenotypes)[name] = callVec;
<a name="l01702"></a>01702         }
<a name="l01703"></a>01703       }
<a name="l01704"></a>01704       <span class="comment">// Set the known genotypes here.</span>
<a name="l01705"></a>01705       <a class="code" href="classQuantBRLMM.html" title="Implements a SNP genotype calling method motivated by Nusrat Rabbee&amp;#39;s RLMM (Robust Linear Model w...">QuantBRLMM</a> *qBRLMM = NULL;
<a name="l01706"></a>01706       qBRLMM =  <span class="keyword">static_cast&lt;</span><a class="code" href="classQuantBRLMM.html" title="Implements a SNP genotype calling method motivated by Nusrat Rabbee&amp;#39;s RLMM (Robust Linear Model w...">QuantBRLMM</a> *<span class="keyword">&gt;</span>(qMethod);
<a name="l01707"></a>01707       qBRLMM-&gt;setHaploidSnps(haploidSnps);
<a name="l01708"></a>01708       qBRLMM-&gt;<a class="code" href="classQuantBRLMM.html#a72de5e66b4d6f1d6fc63426846d20950" title="Set the initial guesses of genotypes, usually from DM predictions.">setKnownGenoTypes</a>(*knownGenotypes);
<a name="l01709"></a>01709       <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Setting &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(knownGenotypes-&gt;size()) + <span class="stringliteral">&quot; seeds for &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(toRunProbesets.size()) + <span class="stringliteral">&quot; probesets.&quot;</span>);
<a name="l01710"></a>01710     }
<a name="l01711"></a>01711     <span class="keywordflow">else</span> <span class="keywordflow">if</span>(InstanceOf(qMethod,<a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a>)) {
<a name="l01712"></a>01712       <span class="comment">// Get the initial genotype calls</span>
<a name="l01713"></a>01713       <span class="keywordflow">for</span>(uint32_t i = 0; i &lt; toRunProbesets.size(); i++) {
<a name="l01714"></a>01714         <span class="keyword">const</span> <span class="keywordtype">char</span> *name = toRunProbesets[i];
<a name="l01715"></a>01715         <a class="code" href="classProbeListPacked.html">ProbeListPacked</a> pList = layout.<a class="code" href="classChipLayout.html#ae465f92816b1c9983a45d1f73293cc7b" title="Find a probe set using the name as a key.">getProbeListByName</a>(name);
<a name="l01716"></a>01716         <span class="comment">// didnt find it.</span>
<a name="l01717"></a>01717         <span class="keywordflow">if</span> (pList.isNull()) {
<a name="l01718"></a>01718           <a class="code" href="classVerbose.html#a34e1d936178741eeb78de93c5497e1b8" title="Print a warning message.">Verbose::warn</a>(1,std::string(<span class="stringliteral">&quot;ProbesetGenotypeStageEngine::setInitialCalls(): missing probeset named: &#39;&quot;</span>)+name+<span class="stringliteral">&quot;&#39;&quot;</span>);
<a name="l01719"></a>01719           <span class="keywordflow">continue</span>;
<a name="l01720"></a>01720         }
<a name="l01721"></a>01721         <span class="comment">//</span>
<a name="l01722"></a>01722         <span class="keywordflow">if</span>((pList.get_type() == ProbeSet::GenoType &amp;&amp; (pList.block_cnt() == 2 || pList.block_cnt() == 4)) || pList.get_type() == ProbeSet::Marker || pList.get_type() == ProbeSet::MultichannelMarker) {
<a name="l01723"></a>01723         <span class="comment">//if ((pList.get_type() == ProbeSet::GenoType || pList.get_type() == ProbeSet::Marker) &amp;&amp; pList.block_cnt() == 2) {</span>
<a name="l01724"></a>01724             genotypable_probeset_count++;
<a name="l01725"></a>01725             <span class="keywordflow">if</span> (seeds.<a class="code" href="classGenoSeed.html#a07a49a96380e19af763958c6f9a5fbdb" title="Check if a call exists for a given probeset.">checkGenoCallsName</a>(name)) {
<a name="l01726"></a>01726                 vector&lt;GType&gt; callVec = seeds.<a class="code" href="classGenoSeed.html#ab3c2932ea7190946962b670a5a5ed259" title="Get the genotype calls for a particular probeset.">getGenoCalls</a>(name);
<a name="l01727"></a>01727                 (*knownGenotypes)[name] = callVec;
<a name="l01728"></a>01728             }
<a name="l01729"></a>01729         }
<a name="l01730"></a>01730       }
<a name="l01731"></a>01731       <span class="comment">// Set the known genotypes here.</span>
<a name="l01732"></a>01732       <a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a> *qLabelZ = NULL;
<a name="l01733"></a>01733       qLabelZ = <span class="keyword">static_cast&lt;</span><a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a> *<span class="keyword">&gt;</span>(qMethod);
<a name="l01734"></a>01734       qLabelZ-&gt;<a class="code" href="classQuantLabelZ.html#a22b847bd3dff1573d18dc4f248ddd509" title="Set the initial guesses of genotypes, usually from DM predictions.">setKnownGenoTypes</a>(knownGenotypes);
<a name="l01735"></a>01735       <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Setting &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(knownGenotypes-&gt;size()) + <span class="stringliteral">&quot; seeds for &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(genotypable_probeset_count) + <span class="stringliteral">&quot; genotypable probesets.&quot;</span>);
<a name="l01736"></a>01736     }
<a name="l01737"></a>01737   }
<a name="l01738"></a>01738 }
<a name="l01739"></a>01739 
<a name="l01740"></a>01740 
<a name="l01741"></a>01741 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::checkSnpLists(<span class="keyword">const</span> map&lt;string,bool&gt; &amp;haploidSnps, <a class="code" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a> *layout) {
<a name="l01742"></a>01742   map&lt;string, bool&gt;::const_iterator hIter, hEnd = haploidSnps.end();
<a name="l01743"></a>01743   <span class="keywordflow">for</span>(hIter = haploidSnps.begin(); hIter != hEnd; hIter++) {
<a name="l01744"></a>01744     <span class="keywordflow">if</span>(!layout-&gt;<a class="code" href="classChipLayout.html#af42fc829c0a3b795e674cfcbcb3dc671" title="Does the layout have a reference to the probeset with name psName? Must have name map filled to call ...">containsProbeSet</a>(hIter-&gt;first)) {
<a name="l01745"></a>01745       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Can&#39;t find snp: &#39;&quot;</span> + hIter-&gt;first + <span class="stringliteral">&quot;&#39; in cdf file. Wrong chrx file?&quot;</span>);
<a name="l01746"></a>01746     }
<a name="l01747"></a>01747   }
<a name="l01748"></a>01748 }
<a name="l01749"></a>01749 <span class="comment"></span>
<a name="l01750"></a>01750 <span class="comment">/**</span>
<a name="l01751"></a>01751 <span class="comment"> * When selecting probes to sample from we have a few different things to consider:</span>
<a name="l01752"></a>01752 <span class="comment"> * - Are we selecting from a certain type of probeset (i.e. genotyping).</span>
<a name="l01753"></a>01753 <span class="comment"> * - Are we selecting from all probesets.</span>
<a name="l01754"></a>01754 <span class="comment"> * - are we selecting from a specific list (i.e. --list-sample)</span>
<a name="l01755"></a>01755 <span class="comment"> *</span>
<a name="l01756"></a>01756 <span class="comment"> * If we&#39;re selecting from a specific list then life is easy, just use the list.</span>
<a name="l01757"></a>01757 <span class="comment"> * otherwise we have to check to make sure that we&#39;re selecting from the right type.</span>
<a name="l01758"></a>01758 <span class="comment"> *</span>
<a name="l01759"></a>01759 <span class="comment"> * @param toSample - We&#39;ll return the list to sample in here.</span>
<a name="l01760"></a>01760 <span class="comment"> * @param layout - Layout has information about probesets types.</span>
<a name="l01761"></a>01761 <span class="comment"> * @param probesetNames - Names has names of all the probesets.</span>
<a name="l01762"></a>01762 <span class="comment"> * @param toRunProbesetNames - This is the list of probesets that will be run.</span>
<a name="l01763"></a>01763 <span class="comment"> * @param psTypesToSample - What types of probesets should we be sampling from.</span>
<a name="l01764"></a>01764 <span class="comment"> */</span>
<a name="l01765"></a>01765 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::makeToSampleVector(vector&lt;const char *&gt; &amp;toSample, 
<a name="l01766"></a>01766                         map&lt;std::string, int&gt; &amp;sampleMap,
<a name="l01767"></a>01767                         <a class="code" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a> *layout, <span class="keyword">const</span> vector&lt;const char *&gt; &amp;probesetNames,
<a name="l01768"></a>01768                         <span class="keyword">const</span> vector&lt;const char *&gt; &amp;toRunProbesetNames,
<a name="l01769"></a>01769                         std::set&lt;affxcdf::GeneChipProbeSetType&gt; &amp;psTypesToSample) {
<a name="l01770"></a>01770   <span class="keywordtype">int</span> i;
<a name="l01771"></a>01771   toSample.clear();
<a name="l01772"></a>01772   sampleMap.clear();
<a name="l01773"></a>01773   <span class="comment">/* if the user specified --list-sample life is easy, just do what they say. */</span>
<a name="l01774"></a>01774   <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;list-sample&quot;</span>)) {
<a name="l01775"></a>01775     toSample = toRunProbesetNames;
<a name="l01776"></a>01776     <span class="keywordflow">for</span> (i = 0; i &lt; toRunProbesetNames.size(); i++) {
<a name="l01777"></a>01777       sampleMap.insert(make_pair(<span class="keywordtype">string</span>(toRunProbesetNames[i]), i));
<a name="l01778"></a>01778     }
<a name="l01779"></a>01779   }
<a name="l01780"></a>01780   <span class="keywordflow">else</span> {
<a name="l01781"></a>01781     std::set&lt;affxcdf::GeneChipProbeSetType&gt;::const_iterator end = psTypesToSample.end();
<a name="l01782"></a>01782     <span class="keywordtype">int</span> pos = 0;
<a name="l01783"></a>01783     <span class="keywordflow">for</span>(i = 0; i &lt; probesetNames.size(); i++) {
<a name="l01784"></a>01784       <span class="keywordtype">bool</span> keep =  psTypesToSample.empty() ||
<a name="l01785"></a>01785         psTypesToSample.find((affxcdf::GeneChipProbeSetType)layout-&gt;getProbeSetType(i)) != end;
<a name="l01786"></a>01786       <span class="keywordflow">if</span>(keep) {
<a name="l01787"></a>01787         toSample.push_back(probesetNames[i]);
<a name="l01788"></a>01788         sampleMap.insert(make_pair(<span class="keywordtype">string</span>(probesetNames[i]), pos++));
<a name="l01789"></a>01789       }
<a name="l01790"></a>01790     }
<a name="l01791"></a>01791   }
<a name="l01792"></a>01792 }
<a name="l01793"></a>01793 <span class="comment"></span>
<a name="l01794"></a>01794 <span class="comment">/**</span>
<a name="l01795"></a>01795 <span class="comment"> * Figure out which snps are going to be used for a representation of all snps</span>
<a name="l01796"></a>01796 <span class="comment"> *</span>
<a name="l01797"></a>01797 <span class="comment"> * @param sampleSize -</span>
<a name="l01798"></a>01798 <span class="comment"> * @param layout - Description of chip format.</span>
<a name="l01799"></a>01799 <span class="comment"> * @param selectFrom -</span>
<a name="l01800"></a>01800 <span class="comment"> * @param SnpSampleNames - Vector to be filled in with pseudo random snp probeset names.</span>
<a name="l01801"></a>01801 <span class="comment"> */</span>
<a name="l01802"></a>01802 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::ObtainSnpSample(
<a name="l01803"></a>01803                      <a class="code" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a> *layout,
<a name="l01804"></a>01804                      std::vector&lt;const char *&gt; &amp;selectFrom,
<a name="l01805"></a>01805                      std::vector&lt;std::string&gt; &amp;SnpSampleNames) {
<a name="l01806"></a>01806   <span class="keywordtype">int</span> sampleSize = <a class="code" href="classOptions.html#afcfab86e21cac5bba43e0858bd428e62" title="Get the integer value of an option.">getOptInt</a>(<span class="stringliteral">&quot;prior-size&quot;</span>);
<a name="l01807"></a>01807   vector&lt;const char *&gt; selected; <span class="comment">// temporary for selection, no new memory allocated outside of stl</span>
<a name="l01808"></a>01808   <span class="keywordflow">if</span>(sampleSize &gt; selectFrom.size()) {
<a name="l01809"></a>01809     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(2, <span class="stringliteral">&quot;Sample size larger than number of probesets. Resetting to: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(selectFrom.size()));
<a name="l01810"></a>01810     sampleSize = selectFrom.size();
<a name="l01811"></a>01811     <a class="code" href="classOptions.html#aaf215b45e7097fcedae68c15073eb0f8" title="Set option value.">setOpt</a>(<span class="stringliteral">&quot;prior-size&quot;</span>, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(sampleSize));
<a name="l01812"></a>01812   }
<a name="l01813"></a>01813 
<a name="l01814"></a>01814   <span class="keywordflow">if</span>(sampleSize &lt; 100) {
<a name="l01815"></a>01815     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Warning!: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(sampleSize) +
<a name="l01816"></a>01816                  <span class="stringliteral">&quot; seems like a small number of SNPs to represent all!&quot;</span>);
<a name="l01817"></a>01817   }
<a name="l01818"></a>01818 
<a name="l01819"></a>01819   SnpSampleNames.reserve(sampleSize);
<a name="l01820"></a>01820   selected.resize(sampleSize);
<a name="l01821"></a>01821   <span class="comment">// time. Use the meaning of life, the universe and</span>
<a name="l01822"></a>01822   <span class="comment">// everything...</span>
<a name="l01823"></a>01823   reproducible_random_sample(selectFrom.begin(), selectFrom.end(),
<a name="l01824"></a>01824                              selected.begin(), sampleSize, 42);
<a name="l01825"></a>01825   <span class="keywordflow">for</span>(<span class="keywordtype">int</span> i = 0; i &lt; selected.size(); i++) {
<a name="l01826"></a>01826     SnpSampleNames.push_back(selected[i]);
<a name="l01827"></a>01827   }
<a name="l01828"></a>01828 }
<a name="l01829"></a>01829 <span class="comment"></span>
<a name="l01830"></a>01830 <span class="comment">/**</span>
<a name="l01831"></a>01831 <span class="comment"> * Sort probeset names according to given map</span>
<a name="l01832"></a>01832 <span class="comment"> * </span>
<a name="l01833"></a>01833 <span class="comment"> * @param probesetsToSort - vector of probeset names</span>
<a name="l01834"></a>01834 <span class="comment"> * @param mapOrder - map of probeset names to relative positions</span>
<a name="l01835"></a>01835 <span class="comment"> */</span>
<a name="l01836"></a>01836 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::sortProbesetsByMapOrder(vector&lt;std::string&gt;::iterator begin, vector&lt;std::string&gt;::iterator end, map&lt;std::string, int&gt; &amp;mapOrder) {
<a name="l01837"></a>01837   <span class="keywordflow">if</span> (begin &lt; end - 1) {
<a name="l01838"></a>01838     std::vector&lt;std::string&gt;::iterator right = end - 1;
<a name="l01839"></a>01839     std::vector&lt;std::string&gt;::iterator left = begin;
<a name="l01840"></a>01840     std::string temp;
<a name="l01841"></a>01841     <span class="keywordtype">int</span> pivot = mapOrder[*right];
<a name="l01842"></a>01842     
<a name="l01843"></a>01843     <span class="keywordflow">while</span> (left &lt; right) {
<a name="l01844"></a>01844       <span class="keywordflow">while</span> (left &lt; right &amp;&amp; mapOrder[*left] &lt;= pivot) {
<a name="l01845"></a>01845         left++;
<a name="l01846"></a>01846       }
<a name="l01847"></a>01847       <span class="keywordflow">while</span> (left &lt; right &amp;&amp; pivot &lt; mapOrder[*right]) {
<a name="l01848"></a>01848         right--;
<a name="l01849"></a>01849       }
<a name="l01850"></a>01850       
<a name="l01851"></a>01851       <span class="keywordflow">if</span> (left &lt; right) {
<a name="l01852"></a>01852         temp = *left;
<a name="l01853"></a>01853         *left = *right;
<a name="l01854"></a>01854         *right = temp;
<a name="l01855"></a>01855       }
<a name="l01856"></a>01856     }
<a name="l01857"></a>01857     
<a name="l01858"></a>01858     sortProbesetsByMapOrder(begin, right, mapOrder);
<a name="l01859"></a>01859     sortProbesetsByMapOrder(left, end, mapOrder);
<a name="l01860"></a>01860   }  
<a name="l01861"></a>01861 }
<a name="l01862"></a>01862 <span class="comment"></span>
<a name="l01863"></a>01863 <span class="comment">/**</span>
<a name="l01864"></a>01864 <span class="comment"> * Add required reporters to analysis.</span>
<a name="l01865"></a>01865 <span class="comment"> *</span>
<a name="l01866"></a>01866 <span class="comment"> * @param qMethod - QuantGTypeMethod</span>
<a name="l01867"></a>01867 <span class="comment"> * @param as  - AnalysisStream</span>
<a name="l01868"></a>01868 <span class="comment"> * @param gender - genders</span>
<a name="l01869"></a>01869 <span class="comment"> */</span>
<a name="l01870"></a>01870 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::addReporters(
<a name="l01871"></a>01871                   <a class="code" href="classQuantGTypeMethod.html" title="Quantification methods used for making genotyping calls implement (currently just brlmm) this interfa...">QuantGTypeMethod</a> *qMethod,
<a name="l01872"></a>01872                   <a class="code" href="classAnalysisStream.html" title="Utility class for centralizing analysis pathway.">AnalysisStream</a> *as,
<a name="l01873"></a>01873                   vector&lt;ChipSummary *&gt; &amp;chipSummaries,
<a name="l01874"></a>01874                   <a class="code" href="classGenderCalls.html" title="Class for determining gender from a cel set of cel files.">GenderCalls</a> *gender,
<a name="l01875"></a>01875                   std::set&lt;const char *, Util::ltstr&gt; *probeSetsToReport,
<a name="l01876"></a>01876                   SpecialSnpMap &amp;SpecialSnps) {
<a name="l01877"></a>01877 
<a name="l01878"></a>01878   <a class="code" href="classAnalysisInfo.html" title="Information about a particular analysis run.">AnalysisInfo</a> info = as-&gt;getInfo();
<a name="l01879"></a>01879   <span class="keywordtype">bool</span> doCompact = <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;file5-compact&quot;</span>);
<a name="l01880"></a>01880   <span class="keywordtype">string</span> outDir = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;out-dir&quot;</span>);
<a name="l01881"></a>01881   vector&lt;string&gt; celFiles = <a class="code" href="classOptions.html#ab20d0083424d9d9a8b94da46f487f57a" title="Return a vector of values for a particular option.">getOptVector</a>(<span class="stringliteral">&quot;cels&quot;</span>);
<a name="l01882"></a>01882 
<a name="l01883"></a>01883   <span class="comment">// Add Quant Listener to Get Summary Stats</span>
<a name="l01884"></a>01884   <a class="code" href="classQuantMethodGTypeChipSummary.html">QuantMethodGTypeChipSummary</a> *runSummary = <span class="keyword">new</span> <a class="code" href="classQuantMethodGTypeChipSummary.html">QuantMethodGTypeChipSummary</a>(qMethod-&gt;<a class="code" href="classQuantMethod.html#ab1b819505891590c9f3e57d9cc67d4de" title="What is the name of the quantification method?">getType</a>(), gender, SpecialSnps);
<a name="l01885"></a>01885   as-&gt;<a class="code" href="classAnalysisStream.html#a7d4b2b28a06660d16590d67b4613c886" title="Connect a reporter to an analysis stream.">addReporter</a>(runSummary);
<a name="l01886"></a>01886 
<a name="l01887"></a>01887   <span class="comment">// Add Expression Summary Stat</span>
<a name="l01888"></a>01888   <a class="code" href="classQuantMethodGTypeExprChipSummary.html">QuantMethodGTypeExprChipSummary</a> *exprSummary = <span class="keyword">new</span> <a class="code" href="classQuantMethodGTypeExprChipSummary.html">QuantMethodGTypeExprChipSummary</a>(qMethod-&gt;<a class="code" href="classQuantMethod.html#ab1b819505891590c9f3e57d9cc67d4de" title="What is the name of the quantification method?">getType</a>());
<a name="l01889"></a>01889   qMethod-&gt;<a class="code" href="classQuantGTypeMethod.html#a0acae5ee552ab83198c1f0a2f4ffd51e" title="Add an expression reporter to output the summary values for each allele, residuals, etc.">addExprReporter</a>(exprSummary);
<a name="l01890"></a>01890 <span class="comment"></span>
<a name="l01891"></a>01891 <span class="comment">  //// Add Chip Summary Report</span>
<a name="l01892"></a>01892 <span class="comment"></span>  <span class="comment">// TSV</span>
<a name="l01893"></a>01893   {
<a name="l01894"></a>01894     <a class="code" href="classQuantMethodRunReport.html" title="Summarize and output basic statistics on the performance of a particular summarization or detection m...">QuantMethodRunReport</a> *runReport = <span class="keyword">new</span> <a class="code" href="classQuantMethodRunReport.html" title="Summarize and output basic statistics on the performance of a particular summarization or detection m...">QuantMethodRunReport</a>(celFiles);
<a name="l01895"></a>01895     <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;report-file&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l01896"></a>01896         <span class="keywordtype">string</span> dir = <a class="code" href="classFs.html#a5308b59c6136f59d7f28947fcc0c6a71" title="just the dirname (will return &amp;quot;.&amp;quot; or &amp;quot;/&amp;quot; as well.) ///">Fs::dirname</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;report-file&quot;</span>));
<a name="l01897"></a>01897         <span class="keywordflow">if</span>(dir == <span class="stringliteral">&quot;&quot;</span>)
<a name="l01898"></a>01898             dir = outDir;
<a name="l01899"></a>01899         <span class="keywordtype">string</span> name = <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;report-file&quot;</span>));
<a name="l01900"></a>01900         runReport-&gt;setDirPath(dir);
<a name="l01901"></a>01901         runReport-&gt;setFilename(name);
<a name="l01902"></a>01902     } <span class="keywordflow">else</span> {
<a name="l01903"></a>01903         runReport-&gt;setDirPath(outDir);
<a name="l01904"></a>01904         runReport-&gt;setFilename(as-&gt;<a class="code" href="classAnalysisStream.html#a79af545326dd23f70df846094f1573c5" title="What is the name of this particular analysis stream?">getName</a>()+<span class="stringliteral">&quot;.report&quot;</span>);
<a name="l01905"></a>01905     }
<a name="l01906"></a>01906     runReport-&gt;setFormat(TsvReport::FMT_TSV); <span class="comment">// &lt;-- TSV (!)</span>
<a name="l01907"></a>01907     <span class="comment">//</span>
<a name="l01908"></a>01908     runReport-&gt;<a class="code" href="classQuantMethodRunReport.html#a29cbe5e522c6b3f7be86ae6ec6757ab1" title="Register a Chip Summary interface instance to pull metrics from.">registerChipSummary</a>(runSummary);
<a name="l01909"></a>01909     runReport-&gt;<a class="code" href="classQuantMethodRunReport.html#a29cbe5e522c6b3f7be86ae6ec6757ab1" title="Register a Chip Summary interface instance to pull metrics from.">registerChipSummary</a>(exprSummary);
<a name="l01910"></a>01910     <span class="keywordflow">for</span>(<span class="keywordtype">int</span> i = 0; i&lt;chipSummaries.size(); i++)
<a name="l01911"></a>01911       runReport-&gt;<a class="code" href="classQuantMethodRunReport.html#a29cbe5e522c6b3f7be86ae6ec6757ab1" title="Register a Chip Summary interface instance to pull metrics from.">registerChipSummary</a>(chipSummaries[i]);
<a name="l01912"></a>01912     as-&gt;<a class="code" href="classAnalysisStream.html#a7d4b2b28a06660d16590d67b4613c886" title="Connect a reporter to an analysis stream.">addReporter</a>(runReport);
<a name="l01913"></a>01913   }
<a name="l01914"></a>01914 
<a name="l01915"></a>01915   <span class="comment">// Add Text Reporter  (calls &amp; confs)</span>
<a name="l01916"></a>01916   <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;table-output&quot;</span>)) {
<a name="l01917"></a>01917     <a class="code" href="classQuantMethodGTypeReport.html">QuantMethodGTypeReport</a> *reporter = <span class="keyword">new</span> <a class="code" href="classQuantMethodGTypeReport.html">QuantMethodGTypeReport</a>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;output-forced-calls&quot;</span>),<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;output-context&quot;</span>));
<a name="l01918"></a>01918     <span class="comment">// @todo: hack -jhg</span>
<a name="l01919"></a>01919     reporter-&gt;setDirPath(outDir);
<a name="l01920"></a>01920     reporter-&gt;setFilename(<span class="stringliteral">&quot;QuantMethodGTypeReport-TXT&quot;</span>); <span class="comment">// not used -- for debugging.</span>
<a name="l01921"></a>01921     reporter-&gt;setFileprefix(as-&gt;<a class="code" href="classAnalysisStream.html#a79af545326dd23f70df846094f1573c5" title="What is the name of this particular analysis stream?">getName</a>());
<a name="l01922"></a>01922     reporter-&gt;setFormat(TsvReport::FMT_TSV);
<a name="l01923"></a>01923     <span class="comment">//</span>
<a name="l01924"></a>01924     reporter-&gt;setPrecision(5);
<a name="l01925"></a>01925     <span class="comment">//</span>
<a name="l01926"></a>01926     reporter-&gt;registerProbeSetsToReport(probeSetsToReport);
<a name="l01927"></a>01927     <span class="comment">//</span>
<a name="l01928"></a>01928     as-&gt;<a class="code" href="classAnalysisStream.html#a7d4b2b28a06660d16590d67b4613c886" title="Connect a reporter to an analysis stream.">addReporter</a>(reporter);
<a name="l01929"></a>01929   }
<a name="l01930"></a>01930 
<a name="l01931"></a>01931   <span class="comment">//</span>
<a name="l01932"></a>01932   <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-calls&quot;</span>)) {
<a name="l01933"></a>01933     <a class="code" href="classQuantMethodGTypeReport.html">QuantMethodGTypeReport</a> *reporter = <span class="keyword">new</span> <a class="code" href="classQuantMethodGTypeReport.html">QuantMethodGTypeReport</a>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;output-forced-calls&quot;</span>),<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;output-context&quot;</span>));
<a name="l01934"></a>01934     reporter-&gt;setDirPath(outDir);
<a name="l01935"></a>01935     reporter-&gt;setFileprefix(as-&gt;<a class="code" href="classAnalysisStream.html#a79af545326dd23f70df846094f1573c5" title="What is the name of this particular analysis stream?">getName</a>());
<a name="l01936"></a>01936     reporter-&gt;setFilename(<span class="stringliteral">&quot;QuantMethodGTypeReport-A5&quot;</span>); <span class="comment">// not used -- for debugging.</span>
<a name="l01937"></a>01937     <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-calls-use-global&quot;</span>)) {
<a name="l01938"></a>01938         <span class="keywordflow">if</span>(m_a5_global_output_group == NULL)
<a name="l01939"></a>01939             <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;--a5-calls-use-global option given, but no global file. Must specify --a5-global-file.&quot;</span>);
<a name="l01940"></a>01940         reporter-&gt;setA5SharedGroup(m_a5_global_output_group);
<a name="l01941"></a>01941     }
<a name="l01942"></a>01942     reporter-&gt;setFormat(affx::TsvReport::FMT_A5);
<a name="l01943"></a>01943     <span class="keywordflow">if</span>(doCompact) {
<a name="l01944"></a>01944       reporter-&gt;<a class="code" href="classQuantMethodGTypeReport.html#a9546b809be37fc59d33c14159c3ea42d" title="Use a compact version of the file5 format with structure of: /calls - with genotypes as single byte /...">setCompactFile5Format</a>(info.<a class="code" href="classAnalysisInfo.html#af50a44fcfb27fcf4aedbb5664857ff24" title="Length of the longest probeset name.">m_MaxPsNameLength</a>);
<a name="l01945"></a>01945     }
<a name="l01946"></a>01946     <span class="comment">//</span>
<a name="l01947"></a>01947     reporter-&gt;registerProbeSetsToReport(probeSetsToReport);
<a name="l01948"></a>01948     <span class="comment">//</span>
<a name="l01949"></a>01949     as-&gt;<a class="code" href="classAnalysisStream.html#a7d4b2b28a06660d16590d67b4613c886" title="Connect a reporter to an analysis stream.">addReporter</a>(reporter);
<a name="l01950"></a>01950   }
<a name="l01951"></a>01951 
<a name="l01952"></a>01952   <span class="comment">// Add AGCC CHP Reporter</span>
<a name="l01953"></a>01953   <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;cc-chp-output&quot;</span>)) {
<a name="l01954"></a>01954       <span class="keywordtype">string</span> ccchpDir;
<a name="l01955"></a>01955       <span class="comment">/* If chp directory not specified use  output/cc-chp/ by default. */</span>
<a name="l01956"></a>01956       <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;cc-chp-out-dir&quot;</span>) == <span class="stringliteral">&quot;&quot;</span>) {
<a name="l01957"></a>01957         ccchpDir = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(outDir,<span class="stringliteral">&quot;cc-chp&quot;</span>);
<a name="l01958"></a>01958       }
<a name="l01959"></a>01959       <span class="keywordflow">else</span> {
<a name="l01960"></a>01960         ccchpDir = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;cc-chp-out-dir&quot;</span>);
<a name="l01961"></a>01961       }
<a name="l01962"></a>01962       <a class="code" href="classUtil.html#a21a08a463a4b0b321b58979b0404e017" title="Chop off the last character if it is a path separator.">Util::chompLastIfSep</a>(ccchpDir);
<a name="l01963"></a>01963       <a class="code" href="classQuantMethodMultiDataCCCHPReport.html">QuantMethodMultiDataCCCHPReport</a> *ccchpReporter = <span class="keyword">new</span> <a class="code" href="classQuantMethodMultiDataCCCHPReport.html">QuantMethodMultiDataCCCHPReport</a>(ccchpDir);
<a name="l01964"></a>01964       vector&lt;string&gt; resultFiles = <a class="code" href="classOptions.html#ab20d0083424d9d9a8b94da46f487f57a" title="Return a vector of values for a particular option.">getOptVector</a>(<span class="stringliteral">&quot;result-files&quot;</span>);
<a name="l01965"></a>01965       <span class="keywordflow">if</span>(resultFiles.size() &gt; 0)
<a name="l01966"></a>01966           ccchpReporter-&gt;<a class="code" href="classQuantMethodMultiDataCCCHPReport.html#a3115e89542dcbf7822ddf5568a410098" title="Set the CHP filenames to use for output.">setChpFileNames</a>(resultFiles);
<a name="l01967"></a>01967       ccchpReporter-&gt;<a class="code" href="classQuantMethodMultiDataCCCHPReport.html#a7ab7a88764474376965312f92e6fa163" title="Register a Chip Summary interface instance to pull metrics from.">registerChipSummary</a>(runSummary);
<a name="l01968"></a>01968       ccchpReporter-&gt;<a class="code" href="classQuantMethodMultiDataCCCHPReport.html#a7ab7a88764474376965312f92e6fa163" title="Register a Chip Summary interface instance to pull metrics from.">registerChipSummary</a>(exprSummary);
<a name="l01969"></a>01969       <span class="keywordflow">for</span>(<span class="keywordtype">int</span> i = 0; i&lt;chipSummaries.size(); i++)
<a name="l01970"></a>01970         ccchpReporter-&gt;<a class="code" href="classQuantMethodMultiDataCCCHPReport.html#a7ab7a88764474376965312f92e6fa163" title="Register a Chip Summary interface instance to pull metrics from.">registerChipSummary</a>(chipSummaries[i]);
<a name="l01971"></a>01971       as-&gt;<a class="code" href="classAnalysisStream.html#a7d4b2b28a06660d16590d67b4613c886" title="Connect a reporter to an analysis stream.">addReporter</a>(ccchpReporter);
<a name="l01972"></a>01972   }
<a name="l01973"></a>01973 
<a name="l01974"></a>01974   <span class="comment">// Add XDA CHP Reporter</span>
<a name="l01975"></a>01975   <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;xda-chp-output&quot;</span>)) {
<a name="l01976"></a>01976       <span class="keywordtype">string</span> chpDir;
<a name="l01977"></a>01977       <span class="comment">/* If chp directory not specified use  output/chp/ by default. */</span>
<a name="l01978"></a>01978       <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;xda-chp-out-dir&quot;</span>) == <span class="stringliteral">&quot;&quot;</span>) {
<a name="l01979"></a>01979         chpDir = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(outDir,<span class="stringliteral">&quot;chp&quot;</span>);
<a name="l01980"></a>01980       }
<a name="l01981"></a>01981       <span class="keywordflow">else</span> {
<a name="l01982"></a>01982         chpDir = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;xda-chp-out-dir&quot;</span>);
<a name="l01983"></a>01983       }
<a name="l01984"></a>01984       <a class="code" href="classUtil.html#a21a08a463a4b0b321b58979b0404e017" title="Chop off the last character if it is a path separator.">Util::chompLastIfSep</a>(chpDir);
<a name="l01985"></a>01985       <a class="code" href="classQuantMethodGTypeCHPReport.html">QuantMethodGTypeCHPReport</a> *chpReporter = <span class="keyword">new</span> <a class="code" href="classQuantMethodGTypeCHPReport.html">QuantMethodGTypeCHPReport</a>(
<a name="l01986"></a>01986               info, chpDir, qMethod-&gt;<a class="code" href="classQuantMethod.html#ab1b819505891590c9f3e57d9cc67d4de" title="What is the name of the quantification method?">getType</a>());
<a name="l01987"></a>01987       vector&lt;string&gt; resultFiles = <a class="code" href="classOptions.html#ab20d0083424d9d9a8b94da46f487f57a" title="Return a vector of values for a particular option.">getOptVector</a>(<span class="stringliteral">&quot;result-files&quot;</span>);
<a name="l01988"></a>01988       <span class="keywordflow">if</span>(resultFiles.size() &gt; 0)
<a name="l01989"></a>01989           chpReporter-&gt;<a class="code" href="classQuantMethodGTypeCHPReport.html#a2319a59b04d96df777a22e0085182184" title="Set the CHP filenames to use for output.">setChpFileNames</a>(resultFiles);
<a name="l01990"></a>01990       as-&gt;<a class="code" href="classAnalysisStream.html#a7d4b2b28a06660d16590d67b4613c886" title="Connect a reporter to an analysis stream.">addReporter</a>(chpReporter);
<a name="l01991"></a>01991   }
<a name="l01992"></a>01992 <span class="comment">/* Avoid compiling these into windows builds until GTC team has been </span>
<a name="l01993"></a>01993 <span class="comment">   informed and given time to update their builds as new libraries are requred. */</span>
<a name="l01994"></a>01994 <span class="preprocessor">#ifndef WIN32</span>
<a name="l01995"></a>01995 <span class="preprocessor"></span>  <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;sqlite-output&quot;</span>)) {
<a name="l01996"></a>01996     <span class="keywordtype">string</span> databaseFile = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(outDir,as-&gt;<a class="code" href="classAnalysisStream.html#a79af545326dd23f70df846094f1573c5" title="What is the name of this particular analysis stream?">getName</a>() + <span class="stringliteral">&quot;.sqlite&quot;</span>);
<a name="l01997"></a>01997     <a class="code" href="classQuantMethodSqlGTypeReport.html" title="Class for reporting results of quantification methods.">QuantMethodSqlGTypeReport</a> *reporter = <span class="keyword">new</span> <a class="code" href="classQuantMethodSqlGTypeReport.html" title="Class for reporting results of quantification methods.">QuantMethodSqlGTypeReport</a>(databaseFile, <span class="stringliteral">&quot;call&quot;</span>, <span class="stringliteral">&quot;confidence&quot;</span>, info.<a class="code" href="classAnalysisInfo.html#af50a44fcfb27fcf4aedbb5664857ff24" title="Length of the longest probeset name.">m_MaxPsNameLength</a>);
<a name="l01998"></a>01998     as-&gt;<a class="code" href="classAnalysisStream.html#a7d4b2b28a06660d16590d67b4613c886" title="Connect a reporter to an analysis stream.">addReporter</a>(reporter);
<a name="l01999"></a>01999     <a class="code" href="classQuantMethodSqlExprReport.html" title="Class for reporting results of quantification methods.">QuantMethodSqlExprReport</a> *sumReporter = <span class="keyword">new</span> <a class="code" href="classQuantMethodSqlExprReport.html" title="Class for reporting results of quantification methods.">QuantMethodSqlExprReport</a>(databaseFile + <span class="stringliteral">&quot;.summary.db&quot;</span>, <span class="stringliteral">&quot;summary&quot;</span>, info.<a class="code" href="classAnalysisInfo.html#af50a44fcfb27fcf4aedbb5664857ff24" title="Length of the longest probeset name.">m_MaxPsNameLength</a>);
<a name="l02000"></a>02000     sumReporter-&gt;<a class="code" href="classQuantMethodSqlExprReport.html#a5819d40f952162fc1f466bb03eaea212" title="Print a message out to the stream.">addStdHeaders</a>(sumReporter, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;exec-guid&quot;</span>), as-&gt;getGuid(), <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;time-start&quot;</span>),
<a name="l02001"></a>02001                                <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;command-line&quot;</span>), <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;version-to-report&quot;</span>), info);
<a name="l02002"></a>02002     <span class="keywordflow">if</span>(InstanceOf(qMethod,<a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a>)) {
<a name="l02003"></a>02003       <span class="keyword">static_cast&lt;</span><a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a> *<span class="keyword">&gt;</span>(qMethod)-&gt;addExprReporter(sumReporter);
<a name="l02004"></a>02004     } 
<a name="l02005"></a>02005     <span class="keywordflow">else</span> <span class="keywordflow">if</span> (InstanceOf(qMethod,<a class="code" href="classQuantBRLMM.html" title="Implements a SNP genotype calling method motivated by Nusrat Rabbee&amp;#39;s RLMM (Robust Linear Model w...">QuantBRLMM</a>)) {
<a name="l02006"></a>02006       <span class="keyword">static_cast&lt;</span><a class="code" href="classQuantBRLMM.html" title="Implements a SNP genotype calling method motivated by Nusrat Rabbee&amp;#39;s RLMM (Robust Linear Model w...">QuantBRLMM</a> *<span class="keyword">&gt;</span>(qMethod)-&gt;addExprReporter(sumReporter);
<a name="l02007"></a>02007     } 
<a name="l02008"></a>02008     <span class="keywordflow">else</span> <span class="keywordflow">if</span> (InstanceOf(qMethod,<a class="code" href="classQuantBirdseed.html">QuantBirdseed</a>)) {
<a name="l02009"></a>02009       <span class="keyword">static_cast&lt;</span><a class="code" href="classQuantBirdseed.html">QuantBirdseed</a> *<span class="keyword">&gt;</span>(qMethod)-&gt;addExprReporter(sumReporter);
<a name="l02010"></a>02010     }
<a name="l02011"></a>02011                                                    
<a name="l02012"></a>02012   }
<a name="l02013"></a>02013 <span class="preprocessor">#endif</span>
<a name="l02014"></a>02014 <span class="preprocessor"></span>  <span class="comment">// Add txt expression reporter</span>
<a name="l02015"></a>02015   <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;summaries&quot;</span>) || <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;feat-effects&quot;</span>) || <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;feat-details&quot;</span>)) {
<a name="l02016"></a>02016     addReporters_Expression(affx::TsvReport::FMT_TSV,info, qMethod,as,chipSummaries,gender,probeSetsToReport, doCompact);
<a name="l02017"></a>02017   }
<a name="l02018"></a>02018   <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-summaries&quot;</span>) || <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-feature-effects&quot;</span>) || <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-feature-details&quot;</span>)) {
<a name="l02019"></a>02019     addReporters_Expression(affx::TsvReport::FMT_A5,info, qMethod,as,chipSummaries,gender,probeSetsToReport, doCompact);
<a name="l02020"></a>02020   }
<a name="l02021"></a>02021 }
<a name="l02022"></a>02022 <span class="comment"></span>
<a name="l02023"></a>02023 <span class="comment">/**</span>
<a name="l02024"></a>02024 <span class="comment"> * Add required reporters to analysis.</span>
<a name="l02025"></a>02025 <span class="comment"> *</span>
<a name="l02026"></a>02026 <span class="comment"> * @param qMethod - QuantGTypeMethod</span>
<a name="l02027"></a>02027 <span class="comment"> * @param as  - AnalysisStream</span>
<a name="l02028"></a>02028 <span class="comment"> * @param gender - genders</span>
<a name="l02029"></a>02029 <span class="comment"> */</span>
<a name="l02030"></a>02030 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::makeReporters(
<a name="l02031"></a>02031                                                 <span class="keyword">const</span> std::string &amp;qMethodName,
<a name="l02032"></a>02032                                                 std::vector&lt;QuantMethodReport *&gt; &amp;reporters,
<a name="l02033"></a>02033                                                 std::vector&lt;QuantMethodReport *&gt; &amp;exprReporters,
<a name="l02034"></a>02034                                                 <a class="code" href="classAnalysisInfo.html" title="Information about a particular analysis run.">AnalysisInfo</a> &amp;info,
<a name="l02035"></a>02035                                                 vector&lt;ChipSummary *&gt; &amp;chipSummaries,
<a name="l02036"></a>02036                                                 <a class="code" href="classGenderCalls.html" title="Class for determining gender from a cel set of cel files.">GenderCalls</a> *gender,
<a name="l02037"></a>02037                                                 std::set&lt;const char *, Util::ltstr&gt; *probeSetsToReport,
<a name="l02038"></a>02038                                                 SpecialSnpMap &amp;SpecialSnps) {
<a name="l02039"></a>02039 
<a name="l02040"></a>02040   <span class="keywordtype">bool</span> doCompact = <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;file5-compact&quot;</span>);
<a name="l02041"></a>02041   <span class="keywordtype">string</span> outDir = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;out-dir&quot;</span>);
<a name="l02042"></a>02042   vector&lt;string&gt; celFiles = <a class="code" href="classOptions.html#ab20d0083424d9d9a8b94da46f487f57a" title="Return a vector of values for a particular option.">getOptVector</a>(<span class="stringliteral">&quot;cels&quot;</span>);
<a name="l02043"></a>02043 
<a name="l02044"></a>02044   <span class="comment">// Add Quant Listener to Get Summary Stats</span>
<a name="l02045"></a>02045   <a class="code" href="classQuantMethodGTypeChipSummary.html">QuantMethodGTypeChipSummary</a> *runSummary = <span class="keyword">new</span> <a class="code" href="classQuantMethodGTypeChipSummary.html">QuantMethodGTypeChipSummary</a>(qMethodName, gender, SpecialSnps);
<a name="l02046"></a>02046   reporters.push_back(runSummary);
<a name="l02047"></a>02047 
<a name="l02048"></a>02048   <span class="comment">// Add Expression Summary Stat</span>
<a name="l02049"></a>02049   <a class="code" href="classQuantMethodGTypeExprChipSummary.html">QuantMethodGTypeExprChipSummary</a> *exprSummary = <span class="keyword">new</span> <a class="code" href="classQuantMethodGTypeExprChipSummary.html">QuantMethodGTypeExprChipSummary</a>(qMethodName);
<a name="l02050"></a>02050   exprReporters.push_back(exprSummary);
<a name="l02051"></a>02051 <span class="comment"></span>
<a name="l02052"></a>02052 <span class="comment">  //// Add Chip Summary Report</span>
<a name="l02053"></a>02053 <span class="comment"></span>  <span class="comment">// TSV</span>
<a name="l02054"></a>02054   {
<a name="l02055"></a>02055     <a class="code" href="classQuantMethodRunReport.html" title="Summarize and output basic statistics on the performance of a particular summarization or detection m...">QuantMethodRunReport</a> *runReport = <span class="keyword">new</span> <a class="code" href="classQuantMethodRunReport.html" title="Summarize and output basic statistics on the performance of a particular summarization or detection m...">QuantMethodRunReport</a>(celFiles);
<a name="l02056"></a>02056     <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;report-file&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l02057"></a>02057         <span class="keywordtype">string</span> dir = <a class="code" href="classFs.html#a5308b59c6136f59d7f28947fcc0c6a71" title="just the dirname (will return &amp;quot;.&amp;quot; or &amp;quot;/&amp;quot; as well.) ///">Fs::dirname</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;report-file&quot;</span>));
<a name="l02058"></a>02058         <span class="keywordflow">if</span>(dir == <span class="stringliteral">&quot;&quot;</span>)
<a name="l02059"></a>02059             dir = outDir;
<a name="l02060"></a>02060         <span class="keywordtype">string</span> name = <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;report-file&quot;</span>));
<a name="l02061"></a>02061         runReport-&gt;setDirPath(dir);
<a name="l02062"></a>02062         runReport-&gt;setFilename(name);
<a name="l02063"></a>02063     } <span class="keywordflow">else</span> {
<a name="l02064"></a>02064         runReport-&gt;setDirPath(outDir);
<a name="l02065"></a>02065         runReport-&gt;setFilename(info.<a class="code" href="classAnalysisInfo.html#a1c684133674d56f7588c24c56c6a4c75" title="Name of the analysis.">m_AnalysisName</a>+<span class="stringliteral">&quot;.report&quot;</span>);
<a name="l02066"></a>02066     }
<a name="l02067"></a>02067     runReport-&gt;setFormat(TsvReport::FMT_TSV); <span class="comment">// &lt;-- TSV (!)</span>
<a name="l02068"></a>02068     <span class="comment">//</span>
<a name="l02069"></a>02069     runReport-&gt;<a class="code" href="classQuantMethodRunReport.html#a29cbe5e522c6b3f7be86ae6ec6757ab1" title="Register a Chip Summary interface instance to pull metrics from.">registerChipSummary</a>(runSummary);
<a name="l02070"></a>02070     runReport-&gt;<a class="code" href="classQuantMethodRunReport.html#a29cbe5e522c6b3f7be86ae6ec6757ab1" title="Register a Chip Summary interface instance to pull metrics from.">registerChipSummary</a>(exprSummary);
<a name="l02071"></a>02071     <span class="keywordflow">for</span>(<span class="keywordtype">int</span> i = 0; i&lt;chipSummaries.size(); i++)
<a name="l02072"></a>02072       runReport-&gt;<a class="code" href="classQuantMethodRunReport.html#a29cbe5e522c6b3f7be86ae6ec6757ab1" title="Register a Chip Summary interface instance to pull metrics from.">registerChipSummary</a>(chipSummaries[i]);
<a name="l02073"></a>02073     reporters.push_back(runReport);
<a name="l02074"></a>02074   }
<a name="l02075"></a>02075 
<a name="l02076"></a>02076   <span class="comment">// Add Text Reporter  (calls &amp; confs)</span>
<a name="l02077"></a>02077   <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;table-output&quot;</span>)) {
<a name="l02078"></a>02078     <a class="code" href="classQuantMethodGTypeReport.html">QuantMethodGTypeReport</a> *reporter = <span class="keyword">new</span> <a class="code" href="classQuantMethodGTypeReport.html">QuantMethodGTypeReport</a>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;output-forced-calls&quot;</span>),<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;output-context&quot;</span>));
<a name="l02079"></a>02079     <span class="comment">// @todo: hack -jhg</span>
<a name="l02080"></a>02080     reporter-&gt;setDirPath(outDir);
<a name="l02081"></a>02081     reporter-&gt;setFilename(<span class="stringliteral">&quot;QuantMethodGTypeReport-TXT&quot;</span>); <span class="comment">// not used -- for debugging.</span>
<a name="l02082"></a>02082     reporter-&gt;setFileprefix(info.<a class="code" href="classAnalysisInfo.html#a1c684133674d56f7588c24c56c6a4c75" title="Name of the analysis.">m_AnalysisName</a>);
<a name="l02083"></a>02083     reporter-&gt;setFormat(TsvReport::FMT_TSV);
<a name="l02084"></a>02084     <span class="comment">//</span>
<a name="l02085"></a>02085     reporter-&gt;setPrecision(5);
<a name="l02086"></a>02086     <span class="comment">//</span>
<a name="l02087"></a>02087     reporter-&gt;registerProbeSetsToReport(probeSetsToReport);
<a name="l02088"></a>02088     <span class="comment">//</span>
<a name="l02089"></a>02089     reporters.push_back(reporter);
<a name="l02090"></a>02090   }
<a name="l02091"></a>02091 
<a name="l02092"></a>02092   <span class="comment">//</span>
<a name="l02093"></a>02093   <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-calls&quot;</span>)) {
<a name="l02094"></a>02094     <a class="code" href="classQuantMethodGTypeReport.html">QuantMethodGTypeReport</a> *reporter = <span class="keyword">new</span> <a class="code" href="classQuantMethodGTypeReport.html">QuantMethodGTypeReport</a>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;output-forced-calls&quot;</span>),<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;output-context&quot;</span>));
<a name="l02095"></a>02095     reporter-&gt;setDirPath(outDir);
<a name="l02096"></a>02096     reporter-&gt;setFileprefix(info.<a class="code" href="classAnalysisInfo.html#a1c684133674d56f7588c24c56c6a4c75" title="Name of the analysis.">m_AnalysisName</a>);
<a name="l02097"></a>02097     reporter-&gt;setFilename(<span class="stringliteral">&quot;QuantMethodGTypeReport-A5&quot;</span>); <span class="comment">// not used -- for debugging.</span>
<a name="l02098"></a>02098     <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-calls-use-global&quot;</span>)) {
<a name="l02099"></a>02099         <span class="keywordflow">if</span>(m_a5_global_output_group == NULL)
<a name="l02100"></a>02100             <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;--a5-calls-use-global option given, but no global file. Must specify --a5-global-file.&quot;</span>);
<a name="l02101"></a>02101         reporter-&gt;setA5SharedGroup(m_a5_global_output_group);
<a name="l02102"></a>02102     }
<a name="l02103"></a>02103     reporter-&gt;setFormat(affx::TsvReport::FMT_A5);
<a name="l02104"></a>02104     <span class="keywordflow">if</span>(doCompact) {
<a name="l02105"></a>02105       reporter-&gt;<a class="code" href="classQuantMethodGTypeReport.html#a9546b809be37fc59d33c14159c3ea42d" title="Use a compact version of the file5 format with structure of: /calls - with genotypes as single byte /...">setCompactFile5Format</a>(info.<a class="code" href="classAnalysisInfo.html#af50a44fcfb27fcf4aedbb5664857ff24" title="Length of the longest probeset name.">m_MaxPsNameLength</a>);
<a name="l02106"></a>02106     }
<a name="l02107"></a>02107     <span class="comment">//</span>
<a name="l02108"></a>02108     reporter-&gt;registerProbeSetsToReport(probeSetsToReport);
<a name="l02109"></a>02109     <span class="comment">//</span>
<a name="l02110"></a>02110     reporters.push_back(reporter);
<a name="l02111"></a>02111   }
<a name="l02112"></a>02112 
<a name="l02113"></a>02113   <span class="comment">// Add AGCC CHP Reporter</span>
<a name="l02114"></a>02114   <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;cc-chp-output&quot;</span>)) {
<a name="l02115"></a>02115       <span class="keywordtype">string</span> ccchpDir;
<a name="l02116"></a>02116       <span class="comment">/* If chp directory not specified use  output/cc-chp/ by default. */</span>
<a name="l02117"></a>02117       <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;cc-chp-out-dir&quot;</span>) == <span class="stringliteral">&quot;&quot;</span>) {
<a name="l02118"></a>02118         ccchpDir = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(outDir,<span class="stringliteral">&quot;cc-chp&quot;</span>);
<a name="l02119"></a>02119       }
<a name="l02120"></a>02120       <span class="keywordflow">else</span> {
<a name="l02121"></a>02121         ccchpDir = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;cc-chp-out-dir&quot;</span>);
<a name="l02122"></a>02122       }
<a name="l02123"></a>02123       <a class="code" href="classUtil.html#a21a08a463a4b0b321b58979b0404e017" title="Chop off the last character if it is a path separator.">Util::chompLastIfSep</a>(ccchpDir);
<a name="l02124"></a>02124       <a class="code" href="classQuantMethodMultiDataCCCHPReport.html">QuantMethodMultiDataCCCHPReport</a> *ccchpReporter = <span class="keyword">new</span> <a class="code" href="classQuantMethodMultiDataCCCHPReport.html">QuantMethodMultiDataCCCHPReport</a>(ccchpDir);
<a name="l02125"></a>02125       vector&lt;string&gt; resultFiles = <a class="code" href="classOptions.html#ab20d0083424d9d9a8b94da46f487f57a" title="Return a vector of values for a particular option.">getOptVector</a>(<span class="stringliteral">&quot;result-files&quot;</span>);
<a name="l02126"></a>02126       <span class="keywordflow">if</span>(resultFiles.size() &gt; 0)
<a name="l02127"></a>02127           ccchpReporter-&gt;<a class="code" href="classQuantMethodMultiDataCCCHPReport.html#a3115e89542dcbf7822ddf5568a410098" title="Set the CHP filenames to use for output.">setChpFileNames</a>(resultFiles);
<a name="l02128"></a>02128       ccchpReporter-&gt;<a class="code" href="classQuantMethodMultiDataCCCHPReport.html#a7ab7a88764474376965312f92e6fa163" title="Register a Chip Summary interface instance to pull metrics from.">registerChipSummary</a>(runSummary);
<a name="l02129"></a>02129       ccchpReporter-&gt;<a class="code" href="classQuantMethodMultiDataCCCHPReport.html#a7ab7a88764474376965312f92e6fa163" title="Register a Chip Summary interface instance to pull metrics from.">registerChipSummary</a>(exprSummary);
<a name="l02130"></a>02130       <span class="keywordflow">for</span>(<span class="keywordtype">int</span> i = 0; i&lt;chipSummaries.size(); i++)
<a name="l02131"></a>02131         ccchpReporter-&gt;<a class="code" href="classQuantMethodMultiDataCCCHPReport.html#a7ab7a88764474376965312f92e6fa163" title="Register a Chip Summary interface instance to pull metrics from.">registerChipSummary</a>(chipSummaries[i]);
<a name="l02132"></a>02132       reporters.push_back(ccchpReporter);
<a name="l02133"></a>02133   }
<a name="l02134"></a>02134 
<a name="l02135"></a>02135   <span class="comment">// Add XDA CHP Reporter</span>
<a name="l02136"></a>02136   <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;xda-chp-output&quot;</span>)) {
<a name="l02137"></a>02137       <span class="keywordtype">string</span> chpDir;
<a name="l02138"></a>02138       <span class="comment">/* If chp directory not specified use  output/chp/ by default. */</span>
<a name="l02139"></a>02139       <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;xda-chp-out-dir&quot;</span>) == <span class="stringliteral">&quot;&quot;</span>)
<a name="l02140"></a>02140         chpDir = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(outDir,<span class="stringliteral">&quot;chp&quot;</span>);
<a name="l02141"></a>02141       <span class="keywordflow">else</span>
<a name="l02142"></a>02142         chpDir = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;xda-chp-out-dir&quot;</span>);
<a name="l02143"></a>02143       <a class="code" href="classUtil.html#a21a08a463a4b0b321b58979b0404e017" title="Chop off the last character if it is a path separator.">Util::chompLastIfSep</a>(chpDir);
<a name="l02144"></a>02144       <a class="code" href="classQuantMethodGTypeCHPReport.html">QuantMethodGTypeCHPReport</a> *chpReporter = <span class="keyword">new</span> <a class="code" href="classQuantMethodGTypeCHPReport.html">QuantMethodGTypeCHPReport</a>(
<a name="l02145"></a>02145               info, chpDir, qMethodName);
<a name="l02146"></a>02146       vector&lt;string&gt; resultFiles = <a class="code" href="classOptions.html#ab20d0083424d9d9a8b94da46f487f57a" title="Return a vector of values for a particular option.">getOptVector</a>(<span class="stringliteral">&quot;result-files&quot;</span>);
<a name="l02147"></a>02147       <span class="keywordflow">if</span>(resultFiles.size() &gt; 0)
<a name="l02148"></a>02148           chpReporter-&gt;<a class="code" href="classQuantMethodGTypeCHPReport.html#a2319a59b04d96df777a22e0085182184" title="Set the CHP filenames to use for output.">setChpFileNames</a>(resultFiles);
<a name="l02149"></a>02149       reporters.push_back(chpReporter);
<a name="l02150"></a>02150   }
<a name="l02151"></a>02151 <span class="comment">/* Avoid compiling these into windows builds until GTC team has been </span>
<a name="l02152"></a>02152 <span class="comment">   informed and given time to update their builds as new libraries are requred. */</span>
<a name="l02153"></a>02153 <span class="preprocessor">#ifndef WIN32</span>
<a name="l02154"></a>02154 <span class="preprocessor"></span>  <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;sqlite-output&quot;</span>)) {
<a name="l02155"></a>02155     <span class="keywordtype">string</span> databaseFile = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(outDir,info.<a class="code" href="classAnalysisInfo.html#a1c684133674d56f7588c24c56c6a4c75" title="Name of the analysis.">m_AnalysisName</a>+<span class="stringliteral">&quot;.sqlite&quot;</span>);
<a name="l02156"></a>02156     <a class="code" href="classQuantMethodSqlGTypeReport.html" title="Class for reporting results of quantification methods.">QuantMethodSqlGTypeReport</a> *reporter = <span class="keyword">new</span> <a class="code" href="classQuantMethodSqlGTypeReport.html" title="Class for reporting results of quantification methods.">QuantMethodSqlGTypeReport</a>(databaseFile, <span class="stringliteral">&quot;call&quot;</span>, <span class="stringliteral">&quot;confidence&quot;</span>, info.<a class="code" href="classAnalysisInfo.html#af50a44fcfb27fcf4aedbb5664857ff24" title="Length of the longest probeset name.">m_MaxPsNameLength</a>);
<a name="l02157"></a>02157     reporters.push_back(reporter);
<a name="l02158"></a>02158     <a class="code" href="classQuantMethodSqlExprReport.html" title="Class for reporting results of quantification methods.">QuantMethodSqlExprReport</a> *sumReporter = <span class="keyword">new</span> <a class="code" href="classQuantMethodSqlExprReport.html" title="Class for reporting results of quantification methods.">QuantMethodSqlExprReport</a>(databaseFile + <span class="stringliteral">&quot;.summary.db&quot;</span>, <span class="stringliteral">&quot;summary&quot;</span>, info.<a class="code" href="classAnalysisInfo.html#af50a44fcfb27fcf4aedbb5664857ff24" title="Length of the longest probeset name.">m_MaxPsNameLength</a>);
<a name="l02159"></a>02159     sumReporter-&gt;<a class="code" href="classQuantMethodSqlExprReport.html#a5819d40f952162fc1f466bb03eaea212" title="Print a message out to the stream.">addStdHeaders</a>(sumReporter, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;exec-guid&quot;</span>), info.<a class="code" href="classAnalysisInfo.html#ab9d3f400692f10270650ed0f02e8d2a0" title="GUID (global unique ID) associated with a specific analysis.">m_AnalysisGuid</a>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;time-start&quot;</span>),
<a name="l02160"></a>02160                                <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;command-line&quot;</span>), <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;version-to-report&quot;</span>), info);
<a name="l02161"></a>02161     exprReporters.push_back(sumReporter);
<a name="l02162"></a>02162   }
<a name="l02163"></a>02163 <span class="preprocessor">#endif</span>
<a name="l02164"></a>02164 <span class="preprocessor"></span>  <span class="comment">// Add txt expression reporter</span>
<a name="l02165"></a>02165   <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;summaries&quot;</span>) || <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;feat-effects&quot;</span>) || <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;feat-details&quot;</span>)) {
<a name="l02166"></a>02166     addReporters_Expression(affx::TsvReport::FMT_TSV,info, exprReporters,chipSummaries,gender,probeSetsToReport, doCompact);
<a name="l02167"></a>02167   }
<a name="l02168"></a>02168   <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-summaries&quot;</span>) || <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-feature-effects&quot;</span>) || <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-feature-details&quot;</span>)) {
<a name="l02169"></a>02169     addReporters_Expression(affx::TsvReport::FMT_A5,info, exprReporters,chipSummaries,gender,probeSetsToReport, doCompact);
<a name="l02170"></a>02170   }
<a name="l02171"></a>02171 }
<a name="l02172"></a>02172 
<a name="l02173"></a>02173 <span class="comment"></span>
<a name="l02174"></a>02174 <span class="comment">/**</span>
<a name="l02175"></a>02175 <span class="comment"> * Any methods that must have seed genotype calls?  BRLMM</span>
<a name="l02176"></a>02176 <span class="comment"> *</span>
<a name="l02177"></a>02177 <span class="comment"> * @param analysis - which methods are called</span>
<a name="l02178"></a>02178 <span class="comment"> */</span>
<a name="l02179"></a>02179 <span class="keywordtype">int</span> ProbesetGenotypeStageEngine::DetectNeedForSeeds(vector&lt;AnalysisStream *&gt; analysis)
<a name="l02180"></a>02180 {
<a name="l02181"></a>02181   <span class="keywordtype">int</span> needSeeds=0;
<a name="l02182"></a>02182   <a class="code" href="classQuantMethod.html" title="QuantMethod - Interface for computing quantification summaries from PM intensities grouped into probe...">QuantMethod</a> *qMethod = NULL;
<a name="l02183"></a>02183   <span class="keywordflow">for</span>(<span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> asIx = 0; asIx &lt; analysis.size(); asIx++) {
<a name="l02184"></a>02184     <a class="code" href="classAnalysisStream.html" title="Utility class for centralizing analysis pathway.">AnalysisStream</a> *as = analysis[asIx];
<a name="l02185"></a>02185     qMethod = as-&gt;<a class="code" href="classAnalysisStream.html#af5d7e7ca7a32a6f189290b5ef1888f64" title="Get the quantification method used.">getQuantMethod</a>();
<a name="l02186"></a>02186     <span class="keywordflow">if</span>(InstanceOf(qMethod, <a class="code" href="classQuantBRLMM.html" title="Implements a SNP genotype calling method motivated by Nusrat Rabbee&amp;#39;s RLMM (Robust Linear Model w...">QuantBRLMM</a>))
<a name="l02187"></a>02187       needSeeds = 1;
<a name="l02188"></a>02188   }
<a name="l02189"></a>02189   <span class="keywordflow">return</span>(needSeeds);
<a name="l02190"></a>02190 }
<a name="l02191"></a>02191 
<a name="l02192"></a>02192 <span class="comment"></span>
<a name="l02193"></a>02193 <span class="comment">/**</span>
<a name="l02194"></a>02194 <span class="comment"> *  Guess what: does setup for analysis methods first time through the loop</span>
<a name="l02195"></a>02195 <span class="comment"> *  gets samples of snps for various purposes</span>
<a name="l02196"></a>02196 <span class="comment"> *  calls genders</span>
<a name="l02197"></a>02197 <span class="comment"> *  builds priors, normalization functions</span>
<a name="l02198"></a>02198 <span class="comment"> *</span>
<a name="l02199"></a>02199 <span class="comment"> * @param analysis - what methods need setting up</span>
<a name="l02200"></a>02200 <span class="comment"> * @param layout - this particular subset of the chip</span>
<a name="l02201"></a>02201 <span class="comment"> * @param iMart - intensities from cel files</span>
<a name="l02202"></a>02202 <span class="comment"> * @param priorSnpNames - names of snp sample for prior (BRLMM)</span>
<a name="l02203"></a>02203 <span class="comment"> * @param normSnpNames - names of snps sampled for normalization (LabelZ = BRLMM-p)</span>
<a name="l02204"></a>02204 <span class="comment"> * @param fileNames - names of celfiles</span>
<a name="l02205"></a>02205 <span class="comment"> * @param genders - genders to pass to quant methods</span>
<a name="l02206"></a>02206 <span class="comment"> * @param inbred - sample covariate to pass to quant methods, like genders</span>
<a name="l02207"></a>02207 <span class="comment"> * @param haploidSnps - chrX snps which get special treatment</span>
<a name="l02208"></a>02208 <span class="comment"> * @param specialSnps -</span>
<a name="l02209"></a>02209 <span class="comment"> */</span>
<a name="l02210"></a>02210 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::DoSetup(std::vector&lt;AnalysisStream *&gt; &amp;analysis,
<a name="l02211"></a>02211              <a class="code" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a> &amp;layout,
<a name="l02212"></a>02212              <a class="code" href="classIntensityMart.html" title="IntensityMart.">IntensityMart</a> &amp;iMart,
<a name="l02213"></a>02213 <span class="comment">//              IntensityMart &amp;priorIntensityMart,</span>
<a name="l02214"></a>02214 <span class="comment">//              IntensityMart &amp;normIntensityMart,</span>
<a name="l02215"></a>02215              vector&lt;string&gt; &amp;priorSnpNames,
<a name="l02216"></a>02216              vector&lt;string&gt; &amp;normSnpNames,
<a name="l02217"></a>02217              vector&lt;string&gt; &amp;fileNames,
<a name="l02218"></a>02218              <a class="code" href="classGenderCalls.html" title="Class for determining gender from a cel set of cel files.">GenderCalls</a> *genders,
<a name="l02219"></a>02219              <a class="code" href="classInbredStatus.html" title="Class for determining inbreeding covariate from a cel set of cel files.">InbredStatus</a> *inbred,
<a name="l02220"></a>02220              map&lt;string,bool&gt; &amp;haploidSnps,
<a name="l02221"></a>02221              SpecialSnpMap &amp;SpecialSnps,
<a name="l02222"></a>02222              CopyNumberMap &amp;copyNumberMap,
<a name="l02223"></a>02223              vector&lt;ChipSummary *&gt; &amp;chipSummaries,
<a name="l02224"></a>02224              <span class="keyword">const</span> vector&lt;const char *&gt; &amp;probesetNames,
<a name="l02225"></a>02225              std::set&lt;const char *, Util::ltstr&gt; *probeSetsToReport) {
<a name="l02226"></a>02226 
<a name="l02227"></a>02227   <a class="code" href="classQuantMethod.html" title="QuantMethod - Interface for computing quantification summaries from PM intensities grouped into probe...">QuantMethod</a> *qMethod = NULL;
<a name="l02228"></a>02228 
<a name="l02229"></a>02229   <span class="keywordtype">string</span> outDir = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;out-dir&quot;</span>);
<a name="l02230"></a>02230   vector&lt;string&gt; celFiles = <a class="code" href="classOptions.html#ab20d0083424d9d9a8b94da46f487f57a" title="Return a vector of values for a particular option.">getOptVector</a>(<span class="stringliteral">&quot;cels&quot;</span>);
<a name="l02231"></a>02231 
<a name="l02232"></a>02232   <span class="comment">// refactor out the hack that sets up everything</span>
<a name="l02233"></a>02233   <span class="comment">// now deal with the setup for streams</span>
<a name="l02234"></a>02234   <span class="comment">// including learning about samples/snps</span>
<a name="l02235"></a>02235 
<a name="l02236"></a>02236   <span class="keywordflow">for</span>(<span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> asIx = 0; asIx &lt; analysis.size(); asIx++) {
<a name="l02237"></a>02237     <a class="code" href="classAnalysisStream.html" title="Utility class for centralizing analysis pathway.">AnalysisStream</a> *as = analysis[asIx];
<a name="l02238"></a>02238     qMethod = as-&gt;<a class="code" href="classAnalysisStream.html#af5d7e7ca7a32a6f189290b5ef1888f64" title="Get the quantification method used.">getQuantMethod</a>();
<a name="l02239"></a>02239 
<a name="l02240"></a>02240     <a class="code" href="classAnalysisInfo.html" title="Information about a particular analysis run.">AnalysisInfo</a> info = makeAnalysisInfo(as, layout, probesetNames, genders);
<a name="l02241"></a>02241     as-&gt;setInfo(info);
<a name="l02242"></a>02242 
<a name="l02243"></a>02243     <span class="comment">// If we are learning a prior this is the time to do it.</span>
<a name="l02244"></a>02244     vector&lt;affx::Gender&gt; genderVec = genders-&gt;<a class="code" href="classGenderCalls.html#abb3474c5f29e2959ce730f539b73b006" title="Get the genders for the cel files that have been seen.">getGenders</a>();
<a name="l02245"></a>02245     vector&lt;double&gt; inbredVec = inbred-&gt;<a class="code" href="classInbredStatus.html#acbcac0e963a7c38b4da2fcf706ca9493" title="Get the inbreeding status for the cel files that have been seen.">getInbredStatus</a>();
<a name="l02246"></a>02246 
<a name="l02247"></a>02247     <span class="comment">// if(InstanceOf(qMethod, QuantBRLMM)) {</span>
<a name="l02248"></a>02248     <span class="comment">//   QuantBRLMM *qBRLMM = NULL;</span>
<a name="l02249"></a>02249     <span class="comment">//   qBRLMM =  static_cast&lt;QuantBRLMM *&gt;(qMethod);</span>
<a name="l02250"></a>02250     <span class="comment">//   qBRLMM-&gt;setGenders(genderVec);</span>
<a name="l02251"></a>02251     <span class="comment">//   qBRLMM-&gt;setHaploidSnps(haploidSnps);</span>
<a name="l02252"></a>02252     <span class="comment">//   /// @todo refactor dmCallsOut out of QuantBRLMM and into engine</span>
<a name="l02253"></a>02253     <span class="comment">//   if(getOptBool(&quot;dm-out&quot;)) {</span>
<a name="l02254"></a>02254     <span class="comment">//     qBRLMM-&gt;setDmCallsOut(outDir + PATH_SEPARATOR + &quot;dm.calls&quot;, celFiles);</span>
<a name="l02255"></a>02255     <span class="comment">//   }</span>
<a name="l02256"></a>02256     <span class="comment">//   addReporters(qBRLMM, as, chipSummaries, genders, probeSetsToReport);</span>
<a name="l02257"></a>02257     <span class="comment">//   string outPrefix = outDir + PATH_SEPARATOR + ToStr(as-&gt;getName()) + &quot;.prior.txt&quot;;</span>
<a name="l02258"></a>02258     <span class="comment">//   setPrior(outPrefix, priorSnpNames, layout, iMart,</span>
<a name="l02259"></a>02259     <span class="comment">//            *(as-&gt;getChipStream()), *(as-&gt;getPmAdjuster()), qBRLMM);</span>
<a name="l02260"></a>02260 
<a name="l02261"></a>02261     <span class="comment">// } // end BRLMM detected</span>
<a name="l02262"></a>02262 
<a name="l02263"></a>02263     <span class="comment">// if(InstanceOf(qMethod, QuantBirdseed)) {</span>
<a name="l02264"></a>02264     <span class="comment">//   QuantBirdseed *qBirdseed = NULL;</span>
<a name="l02265"></a>02265     <span class="comment">//   qBirdseed =  static_cast&lt;QuantBirdseed *&gt;(qMethod);</span>
<a name="l02266"></a>02266     <span class="comment">//   qBirdseed-&gt;setGenders(genderVec);</span>
<a name="l02267"></a>02267     <span class="comment">//   addReporters(qBirdseed, as, chipSummaries, genders, probeSetsToReport);</span>
<a name="l02268"></a>02268     <span class="comment">// } // end Birdseed detected</span>
<a name="l02269"></a>02269     
<a name="l02270"></a>02270     <span class="comment">//  Note that QuantLabelZMulti will also satisfy the &quot;InstanceOf&quot; check on next line.</span>
<a name="l02271"></a>02271     <span class="keywordflow">if</span> (InstanceOf(qMethod,<a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a>)){
<a name="l02272"></a>02272       <a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a> *qLabelZ = NULL;
<a name="l02273"></a>02273       qLabelZ = <span class="keyword">static_cast&lt;</span><a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a> *<span class="keyword">&gt;</span>(qMethod);
<a name="l02274"></a>02274       <span class="comment">// just report QC output</span>
<a name="l02275"></a>02275       <span class="keywordflow">if</span> (!SpecialSnps.empty())
<a name="l02276"></a>02276           qLabelZ-&gt;<a class="code" href="classQuantLabelZ.html#ac14724ad579f92836d1a6c779b9b396b" title="which snps are special">setSpecialSnps</a>(SpecialSnps);
<a name="l02277"></a>02277       <span class="keywordflow">if</span>(!haploidSnps.empty())
<a name="l02278"></a>02278         qLabelZ-&gt;<a class="code" href="classQuantLabelZ.html#a5afceb726f32320451aeb0c5a334a92c" title="which snps are haploid snps">setHaploidSnps</a>(haploidSnps);
<a name="l02279"></a>02279       qLabelZ-&gt;<a class="code" href="classQuantLabelZ.html#a75d2851d0da897d336e42010653fbb4e" title="sets the genders for all samples">setGenders</a>(genderVec);
<a name="l02280"></a>02280       <span class="comment">// add sample covariate inbreeding penalty here</span>
<a name="l02281"></a>02281       <span class="comment">// this is highly indirect at this point, but perhaps there&#39;s a general model for sample covariates</span>
<a name="l02282"></a>02282       qLabelZ-&gt;setInbredHetPenalty(inbredVec);
<a name="l02283"></a>02283       <span class="comment">// send genders to the summary reporter for all chip files</span>
<a name="l02284"></a>02284       addReporters(qLabelZ, as, chipSummaries, genders, probeSetsToReport, SpecialSnps);
<a name="l02285"></a>02285       <span class="comment">// set normalization if required</span>
<a name="l02286"></a>02286       <span class="comment">// that is, precompute the normalization parameters based on</span>
<a name="l02287"></a>02287       <span class="comment">// covariates for each sample</span>
<a name="l02288"></a>02288       <span class="keywordtype">string</span> outPrefix = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(outDir,as-&gt;<a class="code" href="classAnalysisStream.html#a79af545326dd23f70df846094f1573c5" title="What is the name of this particular analysis stream?">getName</a>()+<span class="stringliteral">&quot;.norm.txt&quot;</span>);
<a name="l02289"></a>02289       <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#afcfab86e21cac5bba43e0858bd428e62" title="Get the integer value of an option.">getOptInt</a>(<span class="stringliteral">&quot;norm-size&quot;</span>)&gt;0)
<a name="l02290"></a>02290         setNormLabelZ(outPrefix, normSnpNames, layout, iMart,
<a name="l02291"></a>02291                       *(as-&gt;getChipStream()), *(as-&gt;getPmAdjuster()), qLabelZ);
<a name="l02292"></a>02292 
<a name="l02293"></a>02293     } <span class="comment">// end labelZ detected</span>
<a name="l02294"></a>02294 
<a name="l02295"></a>02295     <span class="comment">// QuantLabelZMulti already partially set as it is also a QuantLabelZ</span>
<a name="l02296"></a>02296     <span class="keywordflow">if</span> (InstanceOf(qMethod,<a class="code" href="classQuantLabelZMulti.html">QuantLabelZMulti</a>)){
<a name="l02297"></a>02297       <a class="code" href="classQuantLabelZMulti.html">QuantLabelZMulti</a> *qLabelZMulti = NULL;
<a name="l02298"></a>02298       qLabelZMulti = <span class="keyword">dynamic_cast&lt;</span><a class="code" href="classQuantLabelZMulti.html">QuantLabelZMulti</a> *<span class="keyword">&gt;</span>(qMethod);<span class="comment"></span>
<a name="l02299"></a>02299 <span class="comment">      ///@todo we should really make this generic to all QuantMethods</span>
<a name="l02300"></a>02300 <span class="comment"></span>      qLabelZMulti-&gt;registerProbeSetsToReport(probeSetsToReport);
<a name="l02301"></a>02301       <span class="keywordflow">if</span>(!copyNumberMap.empty()) 
<a name="l02302"></a>02302         qLabelZMulti-&gt;<a class="code" href="classQuantLabelZMulti.html#a7fc54d4f9e1f64796c0a8380672deed2">setCopyNumberMap</a>(copyNumberMap);
<a name="l02303"></a>02303     }
<a name="l02304"></a>02304 
<a name="l02305"></a>02305     as-&gt;<a class="code" href="classAnalysisStream.html#acb4fca52e4470197e384151f349c7660" title="Write out the initial text header for a quantification method.">addStdHeaders</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;exec-guid&quot;</span>),
<a name="l02306"></a>02306                       <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;time-start&quot;</span>),
<a name="l02307"></a>02307                       <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;command-line&quot;</span>),
<a name="l02308"></a>02308                       <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;version-to-report&quot;</span>),
<a name="l02309"></a>02309                       info);
<a name="l02310"></a>02310     <span class="comment">// this opens the file</span>
<a name="l02311"></a>02311     as-&gt;prepare(iMart);
<a name="l02312"></a>02312   }
<a name="l02313"></a>02313 }
<a name="l02314"></a>02314 
<a name="l02315"></a>02315 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::setIterationData(<a class="code" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a> &amp;layout, 
<a name="l02316"></a>02316                       vector&lt;AnalysisStream *&gt; &amp;analysis,
<a name="l02317"></a>02317                       vector&lt;string&gt; &amp;priorSnpNames, vector&lt;string&gt; &amp;normSnpNames,
<a name="l02318"></a>02318                       vector&lt;const char *&gt; &amp;toRunProbesets,
<a name="l02319"></a>02319                       SpecialSnpMap &amp;specialSnps) {
<a name="l02320"></a>02320   <a class="code" href="classQuantMethod.html" title="QuantMethod - Interface for computing quantification summaries from PM intensities grouped into probe...">QuantMethod</a> *qMethod = NULL;
<a name="l02321"></a>02321   <span class="comment">// Set the initial data that doesn&#39;t need seeds</span>
<a name="l02322"></a>02322   <span class="keywordflow">for</span>(<span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> i = 0; i &lt; analysis.size(); i++) {
<a name="l02323"></a>02323     <a class="code" href="classAnalysisStream.html" title="Utility class for centralizing analysis pathway.">AnalysisStream</a> *as = analysis[i];
<a name="l02324"></a>02324     qMethod = as-&gt;<a class="code" href="classAnalysisStream.html#af5d7e7ca7a32a6f189290b5ef1888f64" title="Get the quantification method used.">getQuantMethod</a>();
<a name="l02325"></a>02325     <span class="keywordflow">if</span>(InstanceOf(qMethod, <a class="code" href="classQuantBirdseed.html">QuantBirdseed</a>)) {
<a name="l02326"></a>02326       <a class="code" href="classQuantBirdseed.html">QuantBirdseed</a> *qBirdseed = NULL;
<a name="l02327"></a>02327       qBirdseed =  <span class="keyword">static_cast&lt;</span><a class="code" href="classQuantBirdseed.html">QuantBirdseed</a> *<span class="keyword">&gt;</span>(qMethod);
<a name="l02328"></a>02328       <span class="comment">// Load up just the probeset priors we need for this iteration.</span>
<a name="l02329"></a>02329       <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-birdseed&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l02330"></a>02330         set&lt;const char *, Util::ltstr&gt; toLoad;
<a name="l02331"></a>02331         <span class="keywordflow">for</span>(<span class="keywordtype">int</span> i = 0; i &lt; priorSnpNames.size(); i++) {
<a name="l02332"></a>02332           toLoad.insert(priorSnpNames[i].c_str());
<a name="l02333"></a>02333         }
<a name="l02334"></a>02334         <span class="keywordflow">for</span>(<span class="keywordtype">int</span> i = 0; i &lt; normSnpNames.size(); i++) {
<a name="l02335"></a>02335           toLoad.insert(normSnpNames[i].c_str());
<a name="l02336"></a>02336         }
<a name="l02337"></a>02337         <span class="comment">// Always need to load up the snps to be processed this time through.</span>
<a name="l02338"></a>02338         <span class="keywordflow">for</span>(<span class="keywordtype">int</span> i = 0; i &lt; toRunProbesets.size(); i++) {
<a name="l02339"></a>02339           toLoad.insert(toRunProbesets[i]);
<a name="l02340"></a>02340         }
<a name="l02341"></a>02341         qBirdseed-&gt;setPriorFile(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-birdseed&quot;</span>), <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chip-type&quot;</span>), &amp;toLoad, specialSnps);
<a name="l02342"></a>02342       }
<a name="l02343"></a>02343     }
<a name="l02344"></a>02344   }
<a name="l02345"></a>02345 }
<a name="l02346"></a>02346 
<a name="l02347"></a>02347 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::fillInDesiredOrder(std::vector&lt;int&gt; &amp;desiredOrder) {
<a name="l02348"></a>02348     vector&lt;bool&gt; probeSeen(m_ChipLayout-&gt;<a class="code" href="classChipLayout.html#abedbffee7c6a978e3e5545d659a78ec7" title="Get the total number of probes on this chip.">getProbeCount</a>(), <span class="keyword">false</span>);
<a name="l02349"></a>02349     <span class="keywordtype">int</span> psCount = m_ChipLayout-&gt;<a class="code" href="classChipLayout.html#a71df054a01d62ca33d6b8de8c8c5c3ca" title="How many probe sets in Pgf file?">getProbeSetCount</a>();
<a name="l02350"></a>02350     <a class="code" href="Err_8h.html#ad024208f4ba6a4e75df4499ea1639a05" title="Calls Err::apt_err_assert with the filename and linenumber set. /// We want to avoid evaluating the m...">APT_ERR_ASSERT</a>(psCount &gt; 0, <span class="stringliteral">&quot;Must have positive number of probesets to count.&quot;</span>);
<a name="l02351"></a>02351     <span class="comment">// @todo - this should be a function rather than inline...</span>
<a name="l02352"></a>02352     <span class="keywordflow">for</span>(<span class="keywordtype">int</span> i = 0; i &lt; psCount ; i++) {
<a name="l02353"></a>02353         <a class="code" href="classProbeListPacked.html">ProbeListPacked</a> pl = m_ChipLayout-&gt;<a class="code" href="classChipLayout.html#a3127c7baedda03477f28da70bbfb686e" title="Get a probelist by the index in the layout.">getProbeListAtIndex</a>(i);
<a name="l02354"></a>02354         <span class="keywordflow">if</span> (!pl.isNull()) {
<a name="l02355"></a>02355             <span class="keywordflow">for</span>(<span class="keywordtype">int</span> pIx = 0; pIx &lt; pl.probe_cnt(); pIx++) {
<a name="l02356"></a>02356                 <span class="keywordflow">if</span>(!probeSeen[pl.get_probeId(pIx)]) {
<a name="l02357"></a>02357                     desiredOrder.push_back(pl.get_probeId(pIx));
<a name="l02358"></a>02358                     probeSeen[pl.get_probeId(pIx)]  = <span class="keyword">true</span>;
<a name="l02359"></a>02359                 }
<a name="l02360"></a>02360             }
<a name="l02361"></a>02361         }
<a name="l02362"></a>02362     }
<a name="l02363"></a>02363     
<a name="l02364"></a>02364     <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; probeSeen.size(); i++) {
<a name="l02365"></a>02365         <span class="keywordflow">if</span> (!probeSeen[i]) {
<a name="l02366"></a>02366             desiredOrder.push_back(i);
<a name="l02367"></a>02367         } 
<a name="l02368"></a>02368     }
<a name="l02369"></a>02369 
<a name="l02370"></a>02370 }
<a name="l02371"></a>02371 
<a name="l02372"></a>02372 <span class="keywordtype">void</span>  ProbesetGenotypeStageEngine::fillInSelectedGenders(<a class="code" href="classGenderCalls.html" title="Class for determining gender from a cel set of cel files.">GenderCalls</a> **selectedGenders, 
<a name="l02373"></a>02373                                                          std::string &amp;analysisName, 
<a name="l02374"></a>02374                                                          std::map&lt;std::string, GenderCalls *&gt; &amp;genderCallsMap,
<a name="l02375"></a>02375                                                          <a class="code" href="classNoGenderCalls.html" title="Instance of GenderCalls which provides unknown genders.">NoGenderCalls</a> *noGenderCalls) {
<a name="l02376"></a>02376 
<a name="l02377"></a>02377     <span class="keywordflow">if</span>(analysisName.find(<span class="stringliteral">&quot;birdseed-v2&quot;</span>) != string::npos || analysisName.find(<span class="stringliteral">&quot;birdseed-dev&quot;</span>) != string::npos) {
<a name="l02378"></a>02378         <span class="keywordflow">if</span>(genderCallsMap.find(<span class="stringliteral">&quot;cn-probe-chrXY-ratio&quot;</span>)!=genderCallsMap.end()) {
<a name="l02379"></a>02379             *selectedGenders = genderCallsMap[<span class="stringliteral">&quot;cn-probe-chrXY-ratio&quot;</span>];
<a name="l02380"></a>02380         } <span class="keywordflow">else</span> <span class="keywordflow">if</span>(genderCallsMap.find(<span class="stringliteral">&quot;em-cluster-chrX-het-contrast&quot;</span>)!=genderCallsMap.end()) {
<a name="l02381"></a>02381             *selectedGenders = genderCallsMap[<span class="stringliteral">&quot;em-cluster-chrX-het-contrast&quot;</span>];
<a name="l02382"></a>02382         }
<a name="l02383"></a>02383     } 
<a name="l02384"></a>02384     <span class="keywordflow">else</span> <span class="keywordflow">if</span>(analysisName.find(<span class="stringliteral">&quot;birdseed&quot;</span>) != string::npos) {
<a name="l02385"></a>02385         <span class="keywordflow">if</span>(genderCallsMap.find(<span class="stringliteral">&quot;em-cluster-chrX-het-contrast&quot;</span>)!=genderCallsMap.end()) {
<a name="l02386"></a>02386             *selectedGenders = genderCallsMap[<span class="stringliteral">&quot;em-cluster-chrX-het-contrast&quot;</span>];
<a name="l02387"></a>02387         }
<a name="l02388"></a>02388     }  
<a name="l02389"></a>02389     <span class="keywordflow">else</span> <span class="keywordflow">if</span>(analysisName.find(<span class="stringliteral">&quot;brlmm-p&quot;</span>) != string::npos) {
<a name="l02390"></a>02390         <span class="keywordflow">if</span>(genderCallsMap.find(<span class="stringliteral">&quot;cn-probe-chrXY-ratio&quot;</span>)!=genderCallsMap.end()) {
<a name="l02391"></a>02391             *selectedGenders = genderCallsMap[<span class="stringliteral">&quot;cn-probe-chrXY-ratio&quot;</span>];
<a name="l02392"></a>02392         } <span class="keywordflow">else</span> <span class="keywordflow">if</span>(genderCallsMap.find(<span class="stringliteral">&quot;em-cluster-chrX-het-contrast&quot;</span>)!=genderCallsMap.end()) {
<a name="l02393"></a>02393             *selectedGenders = genderCallsMap[<span class="stringliteral">&quot;em-cluster-chrX-het-contrast&quot;</span>];
<a name="l02394"></a>02394         }
<a name="l02395"></a>02395     }
<a name="l02396"></a>02396     <span class="keywordflow">else</span> <span class="keywordflow">if</span>(analysisName.find(<span class="stringliteral">&quot;brlmm&quot;</span>) != string::npos) {
<a name="l02397"></a>02397         <span class="keywordflow">if</span>(genderCallsMap.find(<span class="stringliteral">&quot;dm-chrX-het-rate&quot;</span>)!=genderCallsMap.end()) {
<a name="l02398"></a>02398             *selectedGenders = genderCallsMap[<span class="stringliteral">&quot;dm-chrX-het-rate&quot;</span>];
<a name="l02399"></a>02399         } <span class="keywordflow">else</span> <span class="keywordflow">if</span>(genderCallsMap.find(<span class="stringliteral">&quot;supplied-genotypes-chrX-het-rate&quot;</span>)!=genderCallsMap.end()) {
<a name="l02400"></a>02400             *selectedGenders = genderCallsMap[<span class="stringliteral">&quot;supplied-genotypes-chrX-het-rate&quot;</span>];
<a name="l02401"></a>02401         }
<a name="l02402"></a>02402     }
<a name="l02403"></a>02403     
<a name="l02404"></a>02404     <span class="comment">// user specified gender method trumps all</span>
<a name="l02405"></a>02405     <span class="keywordtype">string</span> genderMethod = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;set-gender-method&quot;</span>);
<a name="l02406"></a>02406     <span class="keywordflow">if</span>(genderMethod != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l02407"></a>02407         <span class="keywordflow">if</span>(genderMethod == <span class="stringliteral">&quot;none&quot;</span>) {
<a name="l02408"></a>02408             *selectedGenders = noGenderCalls;
<a name="l02409"></a>02409         } <span class="keywordflow">else</span> <span class="keywordflow">if</span>(genderCallsMap.find(genderMethod)!=genderCallsMap.end()) {
<a name="l02410"></a>02410             *selectedGenders = genderCallsMap[genderMethod];
<a name="l02411"></a>02411         } <span class="keywordflow">else</span> {
<a name="l02412"></a>02412             <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;No &quot;</span> + genderMethod + <span class="stringliteral">&quot; genders to use, but &quot;</span> + genderMethod + <span class="stringliteral">&quot; genders requested with --gender-method.&quot;</span>);
<a name="l02413"></a>02413         }
<a name="l02414"></a>02414     }
<a name="l02415"></a>02415 
<a name="l02416"></a>02416 }
<a name="l02417"></a>02417 
<a name="l02418"></a>02418 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::fillInChipLayout() {
<a name="l02419"></a>02419     std::set&lt;affxcdf::GeneChipProbeSetType&gt; psTypesToLoad; <span class="comment">// What types of probesets are we interested in analyzing.</span>
<a name="l02420"></a>02420     probeidmap_t killList;                 <span class="comment">// Probes in cdf that we don&#39;t want to use.</span>
<a name="l02421"></a>02421     <a class="code" href="AptTypes_8h.html#a6b6d41ded05bd1136b17fa18be0d744f" title="For row and column indexes.">colrow_t</a> numRows = <a class="code" href="classConvert.html#ac012fd4fbad9f297e7d6dddf4049d0a9" title="Make an unsigned int from a c string or die trying.">Convert::toUnsignedInt</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;num-rows&quot;</span>));
<a name="l02422"></a>02422     <a class="code" href="AptTypes_8h.html#a6b6d41ded05bd1136b17fa18be0d744f" title="For row and column indexes.">colrow_t</a> numCols = <a class="code" href="classConvert.html#ac012fd4fbad9f297e7d6dddf4049d0a9" title="Make an unsigned int from a c string or die trying.">Convert::toUnsignedInt</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;num-cols&quot;</span>));
<a name="l02423"></a>02423     <span class="keywordflow">if</span>(m_FreeChipLayout) {
<a name="l02424"></a>02424         
<a name="l02425"></a>02425         <span class="comment">// Does the user want to toss out some probes?</span>
<a name="l02426"></a>02426         <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;kill-list&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l02427"></a>02427             <a class="code" href="classChipLayout.html#a70bd6b3376288c01eec760251fc86ece" title="Reads in probes to kill.">ChipLayout::fillInKillList</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;kill-list&quot;</span>), killList, numRows, numCols);
<a name="l02428"></a>02428         }
<a name="l02429"></a>02429         <span class="keywordflow">if</span>(!<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;all-types&quot;</span>)) {
<a name="l02430"></a>02430             psTypesToLoad.insert(affxcdf::MarkerProbeSetType);
<a name="l02431"></a>02431             psTypesToLoad.insert(affxcdf::GenotypingProbeSetType);
<a name="l02432"></a>02432             psTypesToLoad.insert(affxcdf::MultichannelMarkerProbeSetType);
<a name="l02433"></a>02433         }
<a name="l02434"></a>02434         
<a name="l02435"></a>02435         <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(m_ChipLayout);
<a name="l02436"></a>02436         m_ChipLayout = <span class="keyword">new</span> <a class="code" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a>;
<a name="l02437"></a>02437         m_ChipLayout-&gt;setNeedMismatch(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;need-mismatch&quot;</span>));
<a name="l02438"></a>02438         m_ChipLayout-&gt;setNeedGc(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;need-gc&quot;</span>));
<a name="l02439"></a>02439         m_ChipLayout-&gt;setNeedPmAlleleMatch(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;need-allelematch&quot;</span>));
<a name="l02440"></a>02440         
<a name="l02441"></a>02441         loadLayout(  *m_ChipLayout, 
<a name="l02442"></a>02442                      m_ProbesetNames,    <span class="comment">// filled in and used later</span>
<a name="l02443"></a>02443                      killList, 
<a name="l02444"></a>02444                      <span class="keyword">false</span>, 
<a name="l02445"></a>02445                      psTypesToLoad
<a name="l02446"></a>02446                      );
<a name="l02447"></a>02447     }
<a name="l02448"></a>02448 }
<a name="l02449"></a>02449 <span class="comment"></span>
<a name="l02450"></a>02450 <span class="comment">/**</span>
<a name="l02451"></a>02451 <span class="comment">   This is the &quot;main()&quot; equivalent of the engine. This function has</span>
<a name="l02452"></a>02452 <span class="comment">   two main phases: 1) Creation an initialization of AnalysisStream</span>
<a name="l02453"></a>02453 <span class="comment">   and associated objects. 2) Looping through the list of cdf</span>
<a name="l02454"></a>02454 <span class="comment">   probesets to do the genotyping calls.</span>
<a name="l02455"></a>02455 <span class="comment"></span>
<a name="l02456"></a>02456 <span class="comment">   Phase 2 is definitely where most of the time and cpu cycles are</span>
<a name="l02457"></a>02457 <span class="comment">   spent. The list of things that happens for each iteration of phase</span>
<a name="l02458"></a>02458 <span class="comment">   2 is:</span>
<a name="l02459"></a>02459 <span class="comment">   - Read in cel files only keeping data needed for current iteration in RAM.</span>
<a name="l02460"></a>02460 <span class="comment">   - While reading cel files make DM calls.</span>
<a name="l02461"></a>02461 <span class="comment">   - Once cel files are read loop through probesets making BRLMM genotype</span>
<a name="l02462"></a>02462 <span class="comment">   calls.</span>
<a name="l02463"></a>02463 <span class="comment"></span>
<a name="l02464"></a>02464 <span class="comment">   Special things that happen in the first iteration of phase 2:</span>
<a name="l02465"></a>02465 <span class="comment">   - Calculate prior if desired.</span>
<a name="l02466"></a>02466 <span class="comment">   - Determine gender using DM calls if desired.</span>
<a name="l02467"></a>02467 <span class="comment">   - Learn chipstream parameters necessary.</span>
<a name="l02468"></a>02468 <span class="comment">   - Attach any reporters to the analysis streams.</span>
<a name="l02469"></a>02469 <span class="comment"></span>
<a name="l02470"></a>02470 <span class="comment">   Top of the call stack... This function is getting to be a real</span>
<a name="l02471"></a>02471 <span class="comment">   beast, time for some refactoring...</span>
<a name="l02472"></a>02472 <span class="comment">*/</span>
<a name="l02473"></a>02473 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::runImp() {
<a name="l02474"></a>02474   <span class="comment">// *Warning* This function has evolved into quite a behemoth with a</span>
<a name="l02475"></a>02475   <span class="comment">// series of feature requests and no time to refactor it, be</span>
<a name="l02476"></a>02476   <span class="comment">// careful with modifications.</span>
<a name="l02477"></a>02477   <span class="comment">// probeset names act as a common pool for a number of different</span>
<a name="l02478"></a>02478   <span class="comment">// things so we don&#39;t have to copy the relatively large (in RAM)</span>
<a name="l02479"></a>02479   <span class="comment">// probe names over and over. They are known to be used for the</span>
<a name="l02480"></a>02480   <span class="comment">// chp file writers, meta probesets, and prior selection but may</span>
<a name="l02481"></a>02481   <span class="comment">// also be used for other things as time goes on. Change or delete</span>
<a name="l02482"></a>02482   <span class="comment">// them prematurely at your own risk.</span>
<a name="l02483"></a>02483   std::vector&lt;const char *&gt; toRunProbesets; <span class="comment">/// names of probesets to compute/process</span>
<a name="l02484"></a>02484 <span class="comment"></span>  <span class="keywordtype">bool</span> needPrior = <span class="keyword">true</span>;                 <span class="comment">// Are we computing a prior or is it being read from file?</span>
<a name="l02485"></a>02485   vector&lt;string&gt; priorSnpNames;          <span class="comment">// Names of snps that will be used to estimate prior.</span>
<a name="l02486"></a>02486   <span class="comment">//  bool needNorm = false;                 // are we computing norm? - this variable is not used?</span>
<a name="l02487"></a>02487   vector&lt;string&gt; normSnpNames;           <span class="comment">// Names of snps that will be used to estimate normalizers.</span>
<a name="l02488"></a>02488   <a class="code" href="classIntensityMart.html" title="IntensityMart.">IntensityMart</a> *iMart = NULL;
<a name="l02489"></a>02489   map&lt;string,bool&gt; haploidSnps;          <span class="comment">// chrx snps, used for gender and estimated with different model</span>
<a name="l02490"></a>02490   SpecialSnpMap SpecialSnps;             <span class="comment">// all special snps</span>
<a name="l02491"></a>02491   CopyNumberMap copyNumberMap;           <span class="comment">// A map of names of markers to a vector of copy Numbers, indexed by sample. </span>
<a name="l02492"></a>02492   std::set&lt;affxcdf::GeneChipProbeSetType&gt; psTypesToLoad; <span class="comment">// What types of probesets are we interested in analyzing.</span>
<a name="l02493"></a>02493   std::set&lt;affxcdf::GeneChipProbeSetType&gt; psTypesToSample; <span class="comment">// What types of probesets are we interested in sampling.</span>
<a name="l02494"></a>02494   <a class="code" href="classCelStatListener.html" title="Class for objects that need to actually access real cel file data.">CelStatListener</a> *celStats = NULL;
<a name="l02495"></a>02495   std::set&lt;const char *, Util::ltstr&gt; probeSetsToReport; <span class="comment">// Probe sets to report on</span>
<a name="l02496"></a>02496 
<a name="l02497"></a>02497   <span class="comment">// seed and gender stuff</span>
<a name="l02498"></a>02498   <span class="comment">//  GenoSeed *seeds = NULL;                // The seed genotypes to use for determining gender and seeding brlmm</span>
<a name="l02499"></a>02499   <a class="code" href="classGenoSeedTxt.html" title="Implements interface for a genotyping oracle used for genotyping seeds in brlmm.">GenoSeedTxt</a> *precompSeeds = NULL;      <span class="comment">// Precomputed seeds that were read in.</span>
<a name="l02500"></a>02500   <a class="code" href="classDmListener.html" title="Class for using DM calls as cell files become available.">DmListener</a> *dmCaller = NULL;
<a name="l02501"></a>02501 
<a name="l02502"></a>02502   <span class="comment">// gender stuff</span>
<a name="l02503"></a>02503   map&lt;string, GenderCalls *&gt; genderCallsMap; <span class="comment">/// Map of gender call methods available</span>
<a name="l02504"></a>02504 <span class="comment"></span>  <a class="code" href="classFileGenders.html" title="Read gender assignments from file.">FileGenders</a> *fileGenderReader = NULL;
<a name="l02505"></a>02505   <a class="code" href="classEmGenderCelListener.html" title="Class for determining gender from a cel set of cel files.">EmGenderCelListener</a> *emCelListener = NULL;
<a name="l02506"></a>02506   <a class="code" href="classCnProbeGenderCelListener.html" title="Class for determining gender from a cel set of cel files.">CnProbeGenderCelListener</a> *cnProbeCelListener = NULL;
<a name="l02507"></a>02507   <a class="code" href="classNoGenderCalls.html" title="Instance of GenderCalls which provides unknown genders.">NoGenderCalls</a> *noGenderCalls = NULL;
<a name="l02508"></a>02508 
<a name="l02509"></a>02509   <span class="comment">// inbred sample covariate stuff</span>
<a name="l02510"></a>02510   map&lt;string, InbredStatus *&gt; InbredMap; <span class="comment">/// Map of Inbred status methods [only one currently, but who knows?]</span>
<a name="l02511"></a>02511 <span class="comment"></span>  <span class="comment">//  FileInbred *fileInbredReader = NULL;</span>
<a name="l02512"></a>02512   <a class="code" href="classNoInbredStatus.html" title="Instance of InbredStatus which provides null covariates.">NoInbredStatus</a> *noInbredReader = NULL;
<a name="l02513"></a>02513   <a class="code" href="classCelReader.html" title="Class for reading cel files and passing the data to memory representation and analysis objects...">CelReader</a> reader;
<a name="l02514"></a>02514   <a class="code" href="classPsBoard.html">PsBoard</a> *board = <span class="keyword">new</span> <a class="code" href="classPsBoard.html">PsBoard</a>(); 
<a name="l02515"></a>02515   board-&gt;setOptions(<span class="keyword">this</span>);
<a name="l02516"></a>02516   <span class="comment">// make this all one big try catch to make sure destructors are</span>
<a name="l02517"></a>02517   <span class="comment">// called if exceptions thrown.</span>
<a name="l02518"></a>02518   <span class="keywordflow">try</span> { <span class="comment">// outer try for handling the exception.</span>
<a name="l02519"></a>02519     <span class="keywordflow">try</span> { <span class="comment">// inner try for memory clean up.</span>
<a name="l02520"></a>02520         vector&lt;string&gt; celFiles = <a class="code" href="classOptions.html#ab20d0083424d9d9a8b94da46f487f57a" title="Return a vector of values for a particular option.">getOptVector</a>(<span class="stringliteral">&quot;cels&quot;</span>);
<a name="l02521"></a>02521         vector&lt;string&gt; analysisStrings = <a class="code" href="classOptions.html#ab20d0083424d9d9a8b94da46f487f57a" title="Return a vector of values for a particular option.">getOptVector</a>(<span class="stringliteral">&quot;analysis&quot;</span>);
<a name="l02522"></a>02522 
<a name="l02523"></a>02523         <span class="comment">// set up null covariates in case nothing is supplied</span>
<a name="l02524"></a>02524         <span class="comment">// sample covariates: gender, inbreeding status</span>
<a name="l02525"></a>02525         noGenderCalls = <span class="keyword">new</span> <a class="code" href="classNoGenderCalls.html" title="Instance of GenderCalls which provides unknown genders.">NoGenderCalls</a>(celFiles.size());
<a name="l02526"></a>02526         noInbredReader = <span class="keyword">new</span> <a class="code" href="classNoInbredStatus.html" title="Instance of InbredStatus which provides null covariates.">NoInbredStatus</a>(celFiles.size());
<a name="l02527"></a>02527 
<a name="l02528"></a>02528         <span class="comment">// mem tracking stuff</span>
<a name="l02529"></a>02529         std::vector&lt;ChipSummary *&gt; chipSummaries(0);  <span class="comment">// ChipSummaries to add to reporters</span>
<a name="l02530"></a>02530 
<a name="l02531"></a>02531         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Beginning analysis of &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(celFiles.size()) + <span class="stringliteral">&quot; cel files.&quot;</span>);
<a name="l02532"></a>02532 
<a name="l02533"></a>02533         <span class="comment">/* Are we doing a subset of all the probe sets? */</span>
<a name="l02534"></a>02534         <span class="keywordtype">string</span> psListFile = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;probeset-ids&quot;</span>);
<a name="l02535"></a>02535         <span class="keywordtype">string</span> readPriorsBrlmm = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-priors-brlmm&quot;</span>);
<a name="l02536"></a>02536         <span class="comment">//        int computePrior = getOptInt(&quot;prior-size&quot;);</span>
<a name="l02537"></a>02537         <span class="comment">//        bool listSample = getOptBool(&quot;list-sample&quot;);</span>
<a name="l02538"></a>02538         <span class="comment">/* Are we only reporting a subset of those processed */</span>
<a name="l02539"></a>02539         <span class="keywordtype">string</span> psReportListFile = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;probeset-ids-reported&quot;</span>);
<a name="l02540"></a>02540 
<a name="l02541"></a>02541         <span class="comment">// read in the layout of probesets on the chip.</span>
<a name="l02542"></a>02542         fillInChipLayout(); <span class="comment">// Fills in m_ChipLayout and m_ProbesetNames</span>
<a name="l02543"></a>02543 
<a name="l02544"></a>02544         vector&lt;string&gt; fileNames = celFiles;
<a name="l02545"></a>02545         board-&gt;set(<span class="stringliteral">&quot;chiplayout&quot;</span>, m_ChipLayout);
<a name="l02546"></a>02546         <span class="comment">// Load up marker by sample specific cn info</span>
<a name="l02547"></a>02547         <span class="keywordtype">string</span> genotype_markers_cn_file = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;genotype-markers-cn-file&quot;</span>);
<a name="l02548"></a>02548         <span class="keywordflow">if</span>(genotype_markers_cn_file != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l02549"></a>02549             <a class="code" href="SpecialSnps_8cpp.html#a04cdcf9ca41c9866bfa677d2c6fda24f">readCopyNumberFile</a>(copyNumberMap, genotype_markers_cn_file);
<a name="l02550"></a>02550         }
<a name="l02551"></a>02551 
<a name="l02552"></a>02552         <span class="comment">// We need to handle chrX snps separately for some algorithms, read in those snps if specified.</span>
<a name="l02553"></a>02553         <span class="comment">// special snps trumps chrX snps specification</span>
<a name="l02554"></a>02554         <span class="keywordflow">if</span> (<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;special-snps&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>){
<a name="l02555"></a>02555             fillInSpecialSnps(SpecialSnps, *m_ChipLayout, <span class="keyword">true</span>);
<a name="l02556"></a>02556             haploidFromSpecial(SpecialSnps,haploidSnps);
<a name="l02557"></a>02557         }        <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chrX-probes&quot;</span>) != <span class="stringliteral">&quot;&quot;</span>) { <span class="comment">// @todo</span><span class="comment"></span>
<a name="l02558"></a>02558 <span class="comment">            /// @todo should we require a minimum number of X and Y probes?</span>
<a name="l02559"></a>02559 <span class="comment"></span>            cnProbeCelListener = <span class="keyword">new</span> <a class="code" href="classCnProbeGenderCelListener.html" title="Class for determining gender from a cel set of cel files.">CnProbeGenderCelListener</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chrX-probes&quot;</span>), <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chrY-probes&quot;</span>), 
<a name="l02560"></a>02560                                                               <a class="code" href="classOptions.html#a8b119de0194389e5e91d09520df211d9" title="Get the integer value of an option.">getOptDouble</a>(<span class="stringliteral">&quot;female-thresh&quot;</span>), <a class="code" href="classOptions.html#a8b119de0194389e5e91d09520df211d9" title="Get the integer value of an option.">getOptDouble</a>(<span class="stringliteral">&quot;male-thresh&quot;</span>));
<a name="l02561"></a>02561             reader.<a class="code" href="classCelReader.html#a22dff93eacaa5d3d46e9b001b9771638" title="Register a cel listener to see the cel file when it is read.">registerCelListener</a>(cnProbeCelListener);
<a name="l02562"></a>02562             chipSummaries.push_back(cnProbeCelListener);
<a name="l02563"></a>02563             genderCallsMap[cnProbeCelListener-&gt;getGenderName()] = cnProbeCelListener;
<a name="l02564"></a>02564         }
<a name="l02565"></a>02565 
<a name="l02566"></a>02566         needPrior = <span class="keyword">false</span>;
<a name="l02567"></a>02567         <span class="keywordtype">string</span> diskDir = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;temp-dir&quot;</span>);
<a name="l02568"></a>02568 
<a name="l02569"></a>02569         <span class="comment">// @todo refactor create DataStore here...</span>
<a name="l02570"></a>02570         std::map&lt;const char *,std::vector&lt;affx::GType&gt;, <a class="code" href="structUtil_1_1ltstr.html" title="Comparison object for use in map for char *.">Util::ltstr</a> &gt; knownGenotypes;
<a name="l02571"></a>02571         map&lt;string,bool&gt;::iterator start, end = haploidSnps.end();
<a name="l02572"></a>02572         vector&lt;ProbeListPacked&gt; haploidProbeSets;
<a name="l02573"></a>02573         <span class="keywordflow">for</span>(start = haploidSnps.begin(); start != end; start++) {
<a name="l02574"></a>02574             <a class="code" href="classProbeListPacked.html">ProbeListPacked</a> ps = m_ChipLayout-&gt;<a class="code" href="classChipLayout.html#ae465f92816b1c9983a45d1f73293cc7b" title="Find a probe set using the name as a key.">getProbeListByName</a>(start-&gt;first);
<a name="l02575"></a>02575             <span class="keywordflow">if</span>(!ps.isNull()) {
<a name="l02576"></a>02576               haploidProbeSets.push_back(ps);
<a name="l02577"></a>02577             }
<a name="l02578"></a>02578             <span class="comment">//else</span>
<a name="l02579"></a>02579                 <span class="comment">// Verbose::out(1,&quot;Unable to find probeset &quot; + start-&gt;first + &quot; in layout for haploidSnps. Ignoring.&quot;);</span>
<a name="l02580"></a>02580         }
<a name="l02581"></a>02581         <span class="keywordflow">if</span>(!haploidProbeSets.empty()) {<span class="comment"></span>
<a name="l02582"></a>02582 <span class="comment">            /// @todo is 1 enough? probably not, should we require a minimum</span>
<a name="l02583"></a>02583 <span class="comment">            /// number of haploid probe sets?</span>
<a name="l02584"></a>02584 <span class="comment"></span>
<a name="l02585"></a>02585             <span class="comment">// Hack to turn this off for DMET_Plus</span>
<a name="l02586"></a>02586             <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;em-gender&quot;</span>)) {
<a name="l02587"></a>02587                 emCelListener = <span class="keyword">new</span> <a class="code" href="classEmGenderCelListener.html" title="Class for determining gender from a cel set of cel files.">EmGenderCelListener</a>(haploidProbeSets);
<a name="l02588"></a>02588                 reader.<a class="code" href="classCelReader.html#a22dff93eacaa5d3d46e9b001b9771638" title="Register a cel listener to see the cel file when it is read.">registerCelListener</a>(emCelListener);
<a name="l02589"></a>02589                 chipSummaries.push_back(emCelListener);
<a name="l02590"></a>02590                 genderCallsMap[emCelListener-&gt;getGenderName()] = emCelListener;
<a name="l02591"></a>02591             }
<a name="l02592"></a>02592         }
<a name="l02593"></a>02593 
<a name="l02594"></a>02594         <span class="comment">// @todo refactor </span>
<a name="l02595"></a>02595         <span class="comment">// 1) Add multiple channels for datastore</span>
<a name="l02596"></a>02596         <span class="comment">// 2) Add feature effects for multiple channels</span>
<a name="l02597"></a>02597         <span class="comment">// 3) SpfFile v4 reading and writing</span>
<a name="l02598"></a>02598         <span class="comment">// 4) Freez layout and work off disk...</span>
<a name="l02599"></a>02599         <span class="comment">// @todo refactor - setup data store here...</span>
<a name="l02600"></a>02600         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(2, <span class="stringliteral">&quot;Reading cel files.&quot;</span>);
<a name="l02601"></a>02601         <a class="code" href="classDataStore.html" title="Class that stores data about probes and probesets out of core but loads into RAM as necessary for com...">DataStore</a> *dStore = NULL;
<a name="l02602"></a>02602         <span class="keywordtype">int</span> stageIx = 0; 
<a name="l02603"></a>02603         <a class="code" href="classTmpFileFactory.html" title="Allocate temporary filenames and arrange to remove them when done.">TmpFileFactory</a> tmpFactory;
<a name="l02604"></a>02604         tmpFactory.<a class="code" href="classTmpFileFactory.html#a0c4ac0f09751f79adb32ddd8107e0ae4" title="Sets the directory in which the tmp files will be allocated. ///.">setTmpdir</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;temp-dir&quot;</span>));
<a name="l02605"></a>02605         <span class="comment">// string stageTemp = tmpFactory.genFilename_basic(ToStr(&quot;stage-&quot;) + ToStr(stageIx), &quot;.a5&quot;); </span>
<a name="l02606"></a>02606         <span class="keywordtype">string</span> stageTemp = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(diskDir,<a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(<span class="stringliteral">&quot;stage-&quot;</span>) + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(stageIx) + <span class="stringliteral">&quot;-temp.a5&quot;</span>);
<a name="l02607"></a>02607         dStore = <span class="keyword">new</span> <a class="code" href="classDataStore.html" title="Class that stores data about probes and probesets out of core but loads into RAM as necessary for com...">DataStore</a>(stageTemp);
<a name="l02608"></a>02608 
<a name="l02609"></a>02609         vector&lt;int&gt; desiredOrder;
<a name="l02610"></a>02610         fillInDesiredOrder(desiredOrder);
<a name="l02611"></a>02611 
<a name="l02612"></a>02612         dStore-&gt;<a class="code" href="classDataStore.html#a448b91bef58aadd3c8980110e4b53175" title="Set the probe order that things will be sorted on disk.">setProbeOrder</a>(desiredOrder);
<a name="l02613"></a>02613       <span class="comment">//  string probeInfoTemp = tmpFactory.genFilename_basic(ToStr(&quot;stage-probe-info&quot;) + ToStr(stageIx), &quot;.a5&quot;);</span>
<a name="l02614"></a>02614         <span class="keywordtype">string</span> probeInfoTemp = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(diskDir,<span class="stringliteral">&quot;stage-probe-info-temp.a5&quot;</span>); 
<a name="l02615"></a>02615         <a class="code" href="classDataStore.html" title="Class that stores data about probes and probesets out of core but loads into RAM as necessary for com...">DataStore</a> probeInfo(probeInfoTemp);
<a name="l02616"></a>02616         probeInfo.setProbeOrder(desiredOrder);
<a name="l02617"></a>02617         probeInfo.setValues(*m_ChipLayout, *board);
<a name="l02618"></a>02618         board-&gt;setProbeInfo(&amp;probeInfo);
<a name="l02619"></a>02619          
<a name="l02620"></a>02620         reader.<a class="code" href="classIntensityReader.html#a0124349ddc9af057e80f05dcd9557deb" title="Set the names of the files to read data from.">setFiles</a>(celFiles);
<a name="l02621"></a>02621         reader.<a class="code" href="classCelReader.html#a22dff93eacaa5d3d46e9b001b9771638" title="Register a cel listener to see the cel file when it is read.">registerCelListener</a>(dStore);
<a name="l02622"></a>02622         reader.<a class="code" href="classCelReader.html#a9966a326c023b7048e732db3c657913f" title="Do the heavy lifting of reading data from the cel files and passing to all the streams and intensity ...">readFiles</a>();
<a name="l02623"></a>02623 
<a name="l02624"></a>02624         <span class="comment">// Null covariate first</span>
<a name="l02625"></a>02625         <a class="code" href="classInbredStatus.html" title="Class for determining inbreeding covariate from a cel set of cel files.">InbredStatus</a> *selectedInbred = noInbredReader;
<a name="l02626"></a>02626         <a class="code" href="classGenderCalls.html" title="Class for determining gender from a cel set of cel files.">GenderCalls</a> *selectedGenders = noGenderCalls;        
<a name="l02627"></a>02627 
<a name="l02628"></a>02628         
<a name="l02629"></a>02629         <span class="keywordtype">string</span> analysisString = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;analysis&quot;</span>);
<a name="l02630"></a>02630         vector&lt;AnalysisStage&gt; stages = AnalysisStage::makeStages(analysisString, m_stdMethods);
<a name="l02631"></a>02631         <span class="keywordtype">string</span> analysisName =  <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;set-analysis-name&quot;</span>);
<a name="l02632"></a>02632         <span class="keywordflow">if</span> (analysisName.empty()) {
<a name="l02633"></a>02633             <span class="keywordflow">if</span>(m_stdMethods.find(analysisString) != m_stdMethods.end()) {
<a name="l02634"></a>02634                 analysisName = analysisString;
<a name="l02635"></a>02635             }
<a name="l02636"></a>02636             <span class="keywordflow">else</span> {
<a name="l02637"></a>02637                 analysisName = stages[0].name;
<a name="l02638"></a>02638                 <span class="keywordflow">for</span>(<span class="keywordtype">int</span> i = 1; i &lt; stages.size(); i++) {
<a name="l02639"></a>02639                     analysisName = analysisName + <span class="stringliteral">&quot;.&quot;</span> + stages[i].name;
<a name="l02640"></a>02640                 }
<a name="l02641"></a>02641             }
<a name="l02642"></a>02642         }
<a name="l02643"></a>02643         
<a name="l02644"></a>02644         fillInSelectedGenders(&amp;selectedGenders, analysisName, genderCallsMap, noGenderCalls);
<a name="l02645"></a>02645 
<a name="l02646"></a>02646         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;Using gender method &quot;</span> + selectedGenders-&gt;getGenderName() + <span class="stringliteral">&quot; for genotype calling.&quot;</span>);
<a name="l02647"></a>02647         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;Using inbred covariates &quot;</span> + selectedInbred-&gt;getInbredName() + <span class="stringliteral">&quot; for genotype calling.&quot;</span>);
<a name="l02648"></a>02648 
<a name="l02649"></a>02649         <span class="comment">// @todo refactor - setup blackboard calls, genders, etc.</span>
<a name="l02650"></a>02650         <a class="code" href="classGenotypeInfo.html">GenotypeInfo</a> gInfo;
<a name="l02651"></a>02651         gInfo.m_Genders = selectedGenders;
<a name="l02652"></a>02652         gInfo.m_Inbred = selectedInbred;
<a name="l02653"></a>02653         gInfo.m_HaploidSnps = &amp;haploidSnps;
<a name="l02654"></a>02654         gInfo.m_SpecialSnps = &amp;SpecialSnps;
<a name="l02655"></a>02655         gInfo.m_CopyNumberMap = &amp;copyNumberMap;
<a name="l02656"></a>02656         board-&gt;setGenotypeInfo(&amp;gInfo);
<a name="l02657"></a>02657 
<a name="l02658"></a>02658         <a class="code" href="classAnalysisInfo.html" title="Information about a particular analysis run.">AnalysisInfo</a> info = makeAnalysisInfo(*m_ChipLayout, toRunProbesets, selectedGenders, analysisName);
<a name="l02659"></a>02659         <span class="comment">//unsigned int dotMod = max(int(toRunProbesets.size()/40), 1);</span>
<a name="l02660"></a>02660 
<a name="l02661"></a>02661         board-&gt;setAnalysisInfo(&amp;info);
<a name="l02662"></a>02662         <a class="code" href="classDTFactory.html" title="Class for creating DataTransform objects by string description.">DTFactory</a> dtFactory;
<a name="l02663"></a>02663 
<a name="l02664"></a>02664         std::vector&lt;QuantMethodReport *&gt; reporters;
<a name="l02665"></a>02665         std::vector&lt;QuantMethodReport *&gt; exprReporters;
<a name="l02666"></a>02666         makeReporters(stages[stages.size() - 1].name, reporters, exprReporters, info, chipSummaries, selectedGenders, NULL, SpecialSnps);
<a name="l02667"></a>02667         board-&gt;setQuantGTypeReporters(&amp;reporters);
<a name="l02668"></a>02668         board-&gt;setQuantExpressionReporters(&amp;exprReporters);
<a name="l02669"></a>02669         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;Preprocessing cel file data&quot;</span>);
<a name="l02670"></a>02670         <span class="keywordflow">for</span>(<span class="keywordtype">int</span> i = 0; i &lt; stages.size(); i++) {
<a name="l02671"></a>02671 
<a name="l02672"></a>02672           <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Doing stage: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(i) + <span class="stringliteral">&quot; &quot;</span> + stages[i].name);
<a name="l02673"></a>02673           time_t startTime = time(NULL);
<a name="l02674"></a>02674           <span class="comment">//string nextStore = tmpFactory.genFilename_basic(ToStr(&quot;stage-&quot;) + ToStr(++stageIx), &quot;.a5&quot;); </span>
<a name="l02675"></a>02675           <span class="keywordtype">string</span> nextStore = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(diskDir,<a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(<span class="stringliteral">&quot;stage-&quot;</span>) + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(++stageIx) + <span class="stringliteral">&quot;-temp.a5&quot;</span>);
<a name="l02676"></a>02676           <a class="code" href="classDataStore.html" title="Class that stores data about probes and probesets out of core but loads into RAM as necessary for com...">DataStore</a> *out = <span class="keyword">new</span> <a class="code" href="classDataStore.html" title="Class that stores data about probes and probesets out of core but loads into RAM as necessary for com...">DataStore</a>(nextStore);
<a name="l02677"></a>02677           out-&gt;<a class="code" href="classDataStore.html#abf17dde5d58c057a8ee369a769adb233" title="Initialize values directly from another DataStore object.">initFromDataStore</a>(*dStore);
<a name="l02678"></a>02678           <span class="comment">//          runChipStreamStage(stages[i], *board, *dStore, *out);</span>
<a name="l02679"></a>02679           <a class="code" href="classDataTransform.html" title="Base class for objects that will manipulate data and produce a new set.">DataTransform</a> *transform = dtFactory.<a class="code" href="classDTFactory.html#a5848b314f31f403789bc7e98536f2413" title="Create a pointer to a new DataTransform object as described in the string specification.">dataTransformForString</a>(stages[i].spec, *board);
<a name="l02680"></a>02680           <span class="keywordtype">bool</span> updated = transform-&gt;<a class="code" href="classDataTransform.html#a48425d38589276aaa90e02202d4de00e" title="Core API.">transformData</a>(*board, *dStore, *out);
<a name="l02681"></a>02681           <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(transform);
<a name="l02682"></a>02682           <span class="keywordflow">if</span> (updated) {
<a name="l02683"></a>02683             <a class="code" href="classDataStore.html" title="Class that stores data about probes and probesets out of core but loads into RAM as necessary for com...">DataStore</a> *toFree = dStore;
<a name="l02684"></a>02684             dStore = out;
<a name="l02685"></a>02685             <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(toFree);
<a name="l02686"></a>02686           }
<a name="l02687"></a>02687           <span class="keywordflow">else</span> {
<a name="l02688"></a>02688             <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(out);
<a name="l02689"></a>02689           }
<a name="l02690"></a>02690           time_t endTime = time(NULL);
<a name="l02691"></a>02691           <span class="keywordtype">int</span> t = int(  (<span class="keywordtype">float</span>)(endTime - startTime) / 60.0 * 100); <span class="comment">// convert to minutes</span>
<a name="l02692"></a>02692           <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Stage finished (&quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>((<span class="keywordtype">float</span>)t/100) + <span class="stringliteral">&quot; min)&quot;</span>);
<a name="l02693"></a>02693         }
<a name="l02694"></a>02694         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;Done Preprocessing cel file data&quot;</span>);
<a name="l02695"></a>02695         <span class="keywordflow">if</span> (!<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;db-from-prior-models&quot;</span>).empty()) {
<a name="l02696"></a>02696             <a class="code" href="classSnpModelConverter.html">SnpModelConverter</a> conv;
<a name="l02697"></a>02697             <span class="keywordflow">if</span> (!<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-brlmmp&quot;</span>).empty()) {
<a name="l02698"></a>02698                 conv.convertToDbModel(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-brlmmp&quot;</span>), <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;db-from-prior-models&quot;</span>));
<a name="l02699"></a>02699             }
<a name="l02700"></a>02700             <span class="keywordflow">else</span> <span class="keywordflow">if</span> (!<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-birdseed&quot;</span>).empty()) {
<a name="l02701"></a>02701                 conv.convertToDbModel(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-birdseed&quot;</span>), <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;db-from-prior-models&quot;</span>));
<a name="l02702"></a>02702             }
<a name="l02703"></a>02703         }
<a name="l02704"></a>02704         <span class="keywordflow">if</span> (!<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;db-from-posterior-models&quot;</span>).empty()) {
<a name="l02705"></a>02705             <a class="code" href="classSnpModelConverter.html">SnpModelConverter</a> conv;
<a name="l02706"></a>02706             <span class="keywordtype">string</span> input;
<a name="l02707"></a>02707             board-&gt;get(<span class="stringliteral">&quot;model-file-written&quot;</span>, &amp;input);
<a name="l02708"></a>02708             <span class="keywordtype">string</span> file5Name = analysisName + <span class="stringliteral">&quot;.snp-posteriors&quot;</span>;
<a name="l02709"></a>02709             conv.convertToDbModel(input, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;db-from-posterior-models&quot;</span>), file5Name);
<a name="l02710"></a>02710         }
<a name="l02711"></a>02711         <span class="comment">// Tell analyses that we are done computing.</span>
<a name="l02712"></a>02712         <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(dStore);
<a name="l02713"></a>02713     } <span class="comment">// inner try end</span>
<a name="l02714"></a>02714     <span class="keywordflow">catch</span> (...) {
<a name="l02715"></a>02715       <span class="comment">// cleanup -- KEEP IN SYNC WITH MEMORY FREEZ BELOW WHEN NO EXCEPTION</span><span class="comment"></span>
<a name="l02716"></a>02716 <span class="comment">      ///@todo leaking probeSetToReport</span>
<a name="l02717"></a>02717 <span class="comment"></span>      <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(precompSeeds);
<a name="l02718"></a>02718       <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(dmCaller);
<a name="l02719"></a>02719       <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(emCelListener);
<a name="l02720"></a>02720       <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(cnProbeCelListener);
<a name="l02721"></a>02721       <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(fileGenderReader);
<a name="l02722"></a>02722       <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(noGenderCalls);
<a name="l02723"></a>02723       <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(noInbredReader);
<a name="l02724"></a>02724       <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(celStats);
<a name="l02725"></a>02725 
<a name="l02726"></a>02726       <span class="keywordflow">if</span>(m_FreeChipLayout) {
<a name="l02727"></a>02727         <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(m_ChipLayout);
<a name="l02728"></a>02728         <span class="keywordflow">for</span>(uint32_t i = 0; i &lt; m_ProbesetNames.size(); i++) {
<a name="l02729"></a>02729             <a class="code" href="Util_8h.html#aaf8a1260bc5853179000feabfb97b294" title="delete function that deletes an array and sets the pointer to NULL.">FreezArray</a>(m_ProbesetNames[i]);
<a name="l02730"></a>02730         }
<a name="l02731"></a>02731         m_ProbesetNames.clear();
<a name="l02732"></a>02732       }
<a name="l02733"></a>02733       <span class="keywordflow">if</span>(!<a class="code" href="classUtil.html#a0d2126d2d7027b17fdcba5f94e18567e" title="Check to see if two strings are the same.">Util::sameString</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;probeset-ids&quot;</span>),<span class="stringliteral">&quot;&quot;</span>)) {
<a name="l02734"></a>02734           <span class="keywordflow">for</span>(uint32_t i = 0; i &lt; toRunProbesets.size(); i++) {
<a name="l02735"></a>02735               <a class="code" href="Util_8h.html#aaf8a1260bc5853179000feabfb97b294" title="delete function that deletes an array and sets the pointer to NULL.">FreezArray</a>(toRunProbesets[i]);
<a name="l02736"></a>02736           }
<a name="l02737"></a>02737       }
<a name="l02738"></a>02738       toRunProbesets.clear();
<a name="l02739"></a>02739       <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(iMart);
<a name="l02740"></a>02740       closeGlobalA5();
<a name="l02741"></a>02741       removeTempDir(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;temp-dir&quot;</span>));
<a name="l02742"></a>02742 <span class="comment"></span>
<a name="l02743"></a>02743 <span class="comment">      /// @todo reporters and main chip summary generater not freed</span>
<a name="l02744"></a>02744 <span class="comment"></span>
<a name="l02745"></a>02745       <span class="comment">// re-throw the unchanged exception.</span>
<a name="l02746"></a>02746       <span class="keywordflow">throw</span>;
<a name="l02747"></a>02747     }
<a name="l02748"></a>02748   } <span class="comment">// outer try end</span>
<a name="l02749"></a>02749   <span class="comment">/* When things go wrong see if we can die gracefully here. */</span>
<a name="l02750"></a>02750   <span class="keywordflow">catch</span>(<a class="code" href="classExcept.html" title="General purpose exception error for error handling.">Except</a> &amp;e) {
<a name="l02751"></a>02751     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(0,<span class="stringliteral">&quot;&quot;</span>);
<a name="l02752"></a>02752     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(e.<a class="code" href="classExcept.html#a6983a18bbb12fd78301d1ee77b2371ff" title="Standard what() call, just returns error message.">what</a>());
<a name="l02753"></a>02753   }
<a name="l02754"></a>02754   <span class="keywordflow">catch</span>(<span class="keyword">const</span> std::bad_alloc &amp;e) {
<a name="l02755"></a>02755     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(0,<span class="stringliteral">&quot;&quot;</span>);
<a name="l02756"></a>02756     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Ran out of memory. &quot;</span>
<a name="l02757"></a>02757                   <span class="stringliteral">&quot;Try quitting other applications.&quot;</span>);
<a name="l02758"></a>02758   }
<a name="l02759"></a>02759   <span class="keywordflow">catch</span>(<a class="code" href="classaffymetrix__calvin__exceptions_1_1CalvinException.html">CalvinException</a> &amp;ce) {
<a name="l02760"></a>02760     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(0,<span class="stringliteral">&quot;&quot;</span>);
<a name="l02761"></a>02761     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Affymetrix GeneChip Command Console library has thrown an exception. &quot;</span>
<a name="l02762"></a>02762                   <span class="stringliteral">&quot;Description: &#39;&quot;</span> +StringUtils::ConvertWCSToMBS(ce.<a class="code" href="classaffymetrix__calvin__exceptions_1_1CalvinException.html#a24f3db249eaee80ba1010dcc9e4db3d2">Description</a>()) + <span class="stringliteral">&quot;&#39;&quot;</span>);
<a name="l02763"></a>02763   }
<a name="l02764"></a>02764   <span class="keywordflow">catch</span>(<a class="code" href="classBroadException.html">BroadException</a> &amp;e) {
<a name="l02765"></a>02765     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(0,<span class="stringliteral">&quot;&quot;</span>);
<a name="l02766"></a>02766     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Exception. msg: &#39;&quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(e.m_msg) + <span class="stringliteral">&quot;&#39; source file: &#39;&quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(e.m_sourcefile) +
<a name="l02767"></a>02767                   <span class="stringliteral">&quot;:&quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(e.m_sourceline) + <span class="stringliteral">&quot;&#39;&quot;</span>);
<a name="l02768"></a>02768   }
<a name="l02769"></a>02769   <span class="keywordflow">catch</span>(<span class="keyword">const</span> std::exception &amp;e) {
<a name="l02770"></a>02770     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(0,<span class="stringliteral">&quot;&quot;</span>);
<a name="l02771"></a>02771     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Exception caught. &quot;</span>
<a name="l02772"></a>02772                   <span class="stringliteral">&quot;Most problems with this program are memory related. &quot;</span>
<a name="l02773"></a>02773                   <span class="stringliteral">&quot;Message is: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(e.what()));
<a name="l02774"></a>02774   }
<a name="l02775"></a>02775   <span class="keywordflow">catch</span>(<span class="keyword">const</span> BaseException &amp;e) {
<a name="l02776"></a>02776     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(0,<span class="stringliteral">&quot;&quot;</span>);
<a name="l02777"></a>02777     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;newmat issue: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(e.what()));
<a name="l02778"></a>02778   }
<a name="l02779"></a>02779   <span class="keywordflow">catch</span>(...) {
<a name="l02780"></a>02780     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(0,<span class="stringliteral">&quot;&quot;</span>);
<a name="l02781"></a>02781     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Unknown exception caught &quot;</span>
<a name="l02782"></a>02782                   <span class="stringliteral">&quot;(most problems with this program are memory related).&quot;</span>);
<a name="l02783"></a>02783   }
<a name="l02784"></a>02784 
<a name="l02785"></a>02785   <span class="comment">// cleanup -- KEEP IN SYNC WITH MEMORY FREEZ ABOVE IN CATCH</span><span class="comment"></span>
<a name="l02786"></a>02786 <span class="comment">  ///@todo leaking probeSetToReport</span>
<a name="l02787"></a>02787 <span class="comment"></span>  <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(precompSeeds);
<a name="l02788"></a>02788   <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(dmCaller);
<a name="l02789"></a>02789   <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(emCelListener);
<a name="l02790"></a>02790   <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(cnProbeCelListener);
<a name="l02791"></a>02791   <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(fileGenderReader);
<a name="l02792"></a>02792   <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(noGenderCalls);
<a name="l02793"></a>02793   <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(noInbredReader);
<a name="l02794"></a>02794   <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(celStats);
<a name="l02795"></a>02795   <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(iMart);
<a name="l02796"></a>02796   
<a name="l02797"></a>02797   <span class="keywordflow">if</span>(m_FreeChipLayout) {
<a name="l02798"></a>02798      <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(m_ChipLayout);
<a name="l02799"></a>02799     <span class="keywordflow">for</span>(uint32_t i = 0; i &lt; m_ProbesetNames.size(); i++) {
<a name="l02800"></a>02800         <a class="code" href="Util_8h.html#aaf8a1260bc5853179000feabfb97b294" title="delete function that deletes an array and sets the pointer to NULL.">FreezArray</a>(m_ProbesetNames[i]);
<a name="l02801"></a>02801     }
<a name="l02802"></a>02802     m_ProbesetNames.clear();
<a name="l02803"></a>02803   }
<a name="l02804"></a>02804   <span class="keywordflow">if</span>(!<a class="code" href="classUtil.html#a0d2126d2d7027b17fdcba5f94e18567e" title="Check to see if two strings are the same.">Util::sameString</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;probeset-ids&quot;</span>),<span class="stringliteral">&quot;&quot;</span>)) {
<a name="l02805"></a>02805       <span class="keywordflow">for</span>(uint32_t i = 0; i &lt; toRunProbesets.size(); i++) {
<a name="l02806"></a>02806           <a class="code" href="Util_8h.html#aaf8a1260bc5853179000feabfb97b294" title="delete function that deletes an array and sets the pointer to NULL.">FreezArray</a>(toRunProbesets[i]);
<a name="l02807"></a>02807       }
<a name="l02808"></a>02808   }
<a name="l02809"></a>02809   toRunProbesets.clear();
<a name="l02810"></a>02810   closeGlobalA5();
<a name="l02811"></a>02811   removeTempDir(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;temp-dir&quot;</span>));
<a name="l02812"></a>02812   <span class="comment"></span>
<a name="l02813"></a>02813 <span class="comment">  /// @todo reporters and main chip summary generater not freed</span>
<a name="l02814"></a>02814 <span class="comment"></span>
<a name="l02815"></a>02815 }
<a name="l02816"></a>02816 
<a name="l02817"></a>02817 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::closeGlobalA5() {
<a name="l02818"></a>02818   <span class="comment">// A5 close global groups</span>
<a name="l02819"></a>02819   <span class="keywordflow">if</span> (m_a5_global_output_group!=NULL) {
<a name="l02820"></a>02820     m_a5_global_output_group-&gt;close();
<a name="l02821"></a>02821     <span class="keyword">delete</span> m_a5_global_output_group;
<a name="l02822"></a>02822     m_a5_global_output_group=NULL;
<a name="l02823"></a>02823   }
<a name="l02824"></a>02824   <span class="keywordflow">if</span> (m_a5_global_input_group!=NULL) {
<a name="l02825"></a>02825     m_a5_global_input_group-&gt;close();
<a name="l02826"></a>02826     <span class="keyword">delete</span> m_a5_global_input_group;
<a name="l02827"></a>02827     m_a5_global_input_group=NULL;
<a name="l02828"></a>02828   }
<a name="l02829"></a>02829 
<a name="l02830"></a>02830   <span class="comment">// A5 close global files</span>
<a name="l02831"></a>02831   <span class="keywordflow">if</span>(m_a5_global_output_file != m_a5_global_input_file) {
<a name="l02832"></a>02832     affx::TsvReport::closeA5File(m_a5_global_output_file);
<a name="l02833"></a>02833     affx::TsvReport::closeA5File(m_a5_global_input_file);
<a name="l02834"></a>02834   } <span class="keywordflow">else</span> {
<a name="l02835"></a>02835     affx::TsvReport::closeA5File(m_a5_global_output_file);
<a name="l02836"></a>02836     m_a5_global_input_file = NULL;
<a name="l02837"></a>02837   }
<a name="l02838"></a>02838 }
<a name="l02839"></a>02839 
<a name="l02840"></a>02840 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::printStandardMethods(std::ostream &amp;out) {
<a name="l02841"></a>02841       map&lt;string,string&gt;::iterator iter;
<a name="l02842"></a>02842       out &lt;&lt; endl &lt;&lt; <span class="stringliteral">&quot;Standard Methods:&quot;</span> &lt;&lt; endl;
<a name="l02843"></a>02843       <span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> maxLength = 0;
<a name="l02844"></a>02844       <span class="keywordflow">for</span>(iter = m_stdMethods.begin(); iter != m_stdMethods.end(); iter++) {
<a name="l02845"></a>02845         <span class="keywordflow">if</span>(iter-&gt;first.size() &gt; maxLength)
<a name="l02846"></a>02846           maxLength = iter-&gt;first.size();
<a name="l02847"></a>02847       }
<a name="l02848"></a>02848       <span class="keywordflow">for</span>(iter = m_stdMethods.begin(); iter != m_stdMethods.end(); iter++) {
<a name="l02849"></a>02849         <span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> currentLength = 0;
<a name="l02850"></a>02850         out &lt;&lt; <span class="stringliteral">&quot; &#39;&quot;</span> &lt;&lt; iter-&gt;first &lt;&lt; <span class="stringliteral">&quot;&#39; &quot;</span>;
<a name="l02851"></a>02851         currentLength = iter-&gt;first.size();
<a name="l02852"></a>02852         <span class="keywordflow">while</span>(currentLength &lt; maxLength + 1) {
<a name="l02853"></a>02853           out.put(<span class="charliteral">&#39; &#39;</span>);
<a name="l02854"></a>02854           currentLength++;
<a name="l02855"></a>02855         }
<a name="l02856"></a>02856         out &lt;&lt; iter-&gt;second &lt;&lt; endl;
<a name="l02857"></a>02857       }
<a name="l02858"></a>02858 }
<a name="l02859"></a>02859 
<a name="l02860"></a>02860 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::addReporters_Expression(affx::TsvReport::TsvReportFmt_t report_format,
<a name="l02861"></a>02861                             <a class="code" href="classAnalysisInfo.html" title="Information about a particular analysis run.">AnalysisInfo</a>&amp; info,
<a name="l02862"></a>02862                             <a class="code" href="classQuantGTypeMethod.html" title="Quantification methods used for making genotyping calls implement (currently just brlmm) this interfa...">QuantGTypeMethod</a> *qMethod,
<a name="l02863"></a>02863                             <a class="code" href="classAnalysisStream.html" title="Utility class for centralizing analysis pathway.">AnalysisStream</a> *as,
<a name="l02864"></a>02864                             vector&lt;ChipSummary *&gt; &amp;chipSummaries,
<a name="l02865"></a>02865                             <a class="code" href="classGenderCalls.html" title="Class for determining gender from a cel set of cel files.">GenderCalls</a> *gender,
<a name="l02866"></a>02866                             std::set&lt;const char *, Util::ltstr&gt; *probeSetsToReport,
<a name="l02867"></a>02867                                                      <span class="keywordtype">bool</span> doCompact) {
<a name="l02868"></a>02868   <span class="comment">//</span>
<a name="l02869"></a>02869   <span class="keywordflow">if</span>(InstanceOf(qMethod,<a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a>) || InstanceOf(qMethod,<a class="code" href="classQuantBRLMM.html" title="Implements a SNP genotype calling method motivated by Nusrat Rabbee&amp;#39;s RLMM (Robust Linear Model w...">QuantBRLMM</a>) || InstanceOf(qMethod,<a class="code" href="classQuantBirdseed.html">QuantBirdseed</a>)) {
<a name="l02870"></a>02870     <a class="code" href="classQuantExprMethod.html" title="QuantExprMethod - Interface for computing quantification summaries from PM intensities grouped into p...">QuantExprMethod</a> *eMethod = NULL;
<a name="l02871"></a>02871     <span class="keywordflow">if</span>(InstanceOf(qMethod,<a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a>)) {
<a name="l02872"></a>02872       eMethod = <span class="keyword">static_cast&lt;</span><a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a> *<span class="keyword">&gt;</span>(qMethod)-&gt;getQuantExprMethod();
<a name="l02873"></a>02873     } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (InstanceOf(qMethod,<a class="code" href="classQuantBRLMM.html" title="Implements a SNP genotype calling method motivated by Nusrat Rabbee&amp;#39;s RLMM (Robust Linear Model w...">QuantBRLMM</a>)) {
<a name="l02874"></a>02874       eMethod = <span class="keyword">static_cast&lt;</span><a class="code" href="classQuantBRLMM.html" title="Implements a SNP genotype calling method motivated by Nusrat Rabbee&amp;#39;s RLMM (Robust Linear Model w...">QuantBRLMM</a> *<span class="keyword">&gt;</span>(qMethod)-&gt;getQuantExprMethod();
<a name="l02875"></a>02875     } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (InstanceOf(qMethod,<a class="code" href="classQuantBirdseed.html">QuantBirdseed</a>)) {
<a name="l02876"></a>02876       eMethod = <span class="keyword">static_cast&lt;</span><a class="code" href="classQuantBirdseed.html">QuantBirdseed</a> *<span class="keyword">&gt;</span>(qMethod)-&gt;getQuantExprMethod();
<a name="l02877"></a>02877     }
<a name="l02878"></a>02878 
<a name="l02879"></a>02879     <a class="code" href="classQuantMethodExprReport.html" title="Class for reporting results of quantification methods.">QuantMethodExprReport</a> *report = <span class="keyword">new</span> <a class="code" href="classQuantMethodExprReport.html" title="Class for reporting results of quantification methods.">QuantMethodExprReport</a>(info.getNumCols());
<a name="l02880"></a>02880     report-&gt;m_qMethod=as-&gt;<a class="code" href="classAnalysisStream.html#af5d7e7ca7a32a6f189290b5ef1888f64" title="Get the quantification method used.">getQuantMethod</a>();
<a name="l02881"></a>02881     report-&gt;setIsHeaderBuffer(1);
<a name="l02882"></a>02882     report-&gt;setDirPath(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;out-dir&quot;</span>));
<a name="l02883"></a>02883     report-&gt;m_qMethod=eMethod;
<a name="l02884"></a>02884     report-&gt;setPrecision(5);
<a name="l02885"></a>02885     report-&gt;setWriteOldStyleFeatureEffectsFile(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;writeOldStyleFeatureEffectsFile&quot;</span>));
<a name="l02886"></a>02886 
<a name="l02887"></a>02887     <span class="comment">// We do not do this for the expression reporters as output</span>
<a name="l02888"></a>02888     <span class="comment">// is controlled by the quant method. If you set this then</span>
<a name="l02889"></a>02889     <span class="comment">// nothing will be reported as the probeset names reported</span>
<a name="l02890"></a>02890     <span class="comment">// have the extra allele and context stuff appended.</span>
<a name="l02891"></a>02891     <span class="comment">//report-&gt;registerProbeSetsToReport(probeSetsToReport);</span>
<a name="l02892"></a>02892     <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;include-quant-in-report-file-name&quot;</span>))
<a name="l02893"></a>02893         report-&gt;setFileprefix(as-&gt;<a class="code" href="classAnalysisStream.html#a79af545326dd23f70df846094f1573c5" title="What is the name of this particular analysis stream?">getName</a>()+<span class="stringliteral">&quot;.&quot;</span>+eMethod-&gt;<a class="code" href="classQuantExprMethod.html#a467749213636536eeab20b0100b2cadd" title="What is the name of the quantification method?">getType</a>());
<a name="l02894"></a>02894     <span class="keywordflow">else</span>
<a name="l02895"></a>02895         report-&gt;setFileprefix(as-&gt;<a class="code" href="classAnalysisStream.html#a79af545326dd23f70df846094f1573c5" title="What is the name of this particular analysis stream?">getName</a>());
<a name="l02896"></a>02896     report-&gt;setFilename(<span class="stringliteral">&quot;QuantMethodExprReport-DEBUG&quot;</span>); <span class="comment">// filename not used -- for debugging</span>
<a name="l02897"></a>02897 
<a name="l02898"></a>02898     <span class="keywordflow">if</span>(report_format == affx::TsvReport::FMT_TSV) {
<a name="l02899"></a>02899         report-&gt;m_DoSummary=<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;summaries&quot;</span>);
<a name="l02900"></a>02900         report-&gt;m_DoResiduals=<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;feat-details&quot;</span>);
<a name="l02901"></a>02901         report-&gt;m_DoFeatureEffects=<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;feat-effects&quot;</span>);
<a name="l02902"></a>02902     } <span class="keywordflow">else</span> {
<a name="l02903"></a>02903         report-&gt;m_DoSummary=<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-summaries&quot;</span>);
<a name="l02904"></a>02904         report-&gt;m_DoResiduals=<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-feature-details&quot;</span>);
<a name="l02905"></a>02905         report-&gt;m_DoFeatureEffects=<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-feature-effects&quot;</span>);
<a name="l02906"></a>02906         
<a name="l02907"></a>02907         <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-summaries-use-global&quot;</span>)) {
<a name="l02908"></a>02908             <span class="keywordflow">if</span>(m_a5_global_output_group == NULL)
<a name="l02909"></a>02909                 <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;--a5-summaries-use-global option given, but no global file. Must specify --a5-global-file.&quot;</span>);
<a name="l02910"></a>02910             report-&gt;m_summary_a5_group = m_a5_global_output_group;
<a name="l02911"></a>02911             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;Using global A5 file for allele summary output.&quot;</span>);
<a name="l02912"></a>02912         }
<a name="l02913"></a>02913     
<a name="l02914"></a>02914         <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-feature-details-use-global&quot;</span>)) {
<a name="l02915"></a>02915             <span class="keywordflow">if</span>(m_a5_global_output_group == NULL)
<a name="l02916"></a>02916                 <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;--a5-feature-details-use-global option given, but no global file. Must specify --a5-global-file.&quot;</span>);
<a name="l02917"></a>02917             report-&gt;m_residuals_a5_group = m_a5_global_output_group;
<a name="l02918"></a>02918             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;Using global A5 file for residuals output.&quot;</span>);
<a name="l02919"></a>02919         }
<a name="l02920"></a>02920     
<a name="l02921"></a>02921         <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-feature-effects-use-global&quot;</span>)) {
<a name="l02922"></a>02922             <span class="keywordflow">if</span>(m_a5_global_output_group == NULL)
<a name="l02923"></a>02923                 <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;--a5-feature-effects-use-global option given, but no global file. Must specify --a5-global-file.&quot;</span>);
<a name="l02924"></a>02924             report-&gt;m_feffects_a5_group = m_a5_global_output_group;
<a name="l02925"></a>02925             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;Using global A5 file for feature effects output.&quot;</span>);
<a name="l02926"></a>02926         }
<a name="l02927"></a>02927     }
<a name="l02928"></a>02928 
<a name="l02929"></a>02929     report-&gt;setFormat(report_format);
<a name="l02930"></a>02930     <span class="keywordflow">if</span>(doCompact &amp;&amp; report_format == affx::TsvReport::FMT_A5) {
<a name="l02931"></a>02931       report-&gt;setCompactFile5Format(info.<a class="code" href="classAnalysisInfo.html#af50a44fcfb27fcf4aedbb5664857ff24" title="Length of the longest probeset name.">m_MaxPsNameLength</a> + 2);
<a name="l02932"></a>02932     }
<a name="l02933"></a>02933     report-&gt;<a class="code" href="classQuantMethodReport.html#acc296969ce5774f4479fc7275de753f0" title="Print a message out to the stream.">addStdHeaders</a>(report,
<a name="l02934"></a>02934                           <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;exec-guid&quot;</span>),
<a name="l02935"></a>02935                           as-&gt;getGuid(),
<a name="l02936"></a>02936                           <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;time-start&quot;</span>),
<a name="l02937"></a>02937                           <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;command-line&quot;</span>),
<a name="l02938"></a>02938                           <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;version-to-report&quot;</span>),
<a name="l02939"></a>02939                           info);
<a name="l02940"></a>02940 
<a name="l02941"></a>02941     <span class="keywordflow">if</span>(InstanceOf(qMethod,<a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a>)) {
<a name="l02942"></a>02942       <span class="keyword">static_cast&lt;</span><a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a> *<span class="keyword">&gt;</span>(qMethod)-&gt;addExprReporter(report);
<a name="l02943"></a>02943     } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (InstanceOf(qMethod,<a class="code" href="classQuantBRLMM.html" title="Implements a SNP genotype calling method motivated by Nusrat Rabbee&amp;#39;s RLMM (Robust Linear Model w...">QuantBRLMM</a>)) {
<a name="l02944"></a>02944       <span class="keyword">static_cast&lt;</span><a class="code" href="classQuantBRLMM.html" title="Implements a SNP genotype calling method motivated by Nusrat Rabbee&amp;#39;s RLMM (Robust Linear Model w...">QuantBRLMM</a> *<span class="keyword">&gt;</span>(qMethod)-&gt;addExprReporter(report);
<a name="l02945"></a>02945     } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (InstanceOf(qMethod,<a class="code" href="classQuantBirdseed.html">QuantBirdseed</a>)) {
<a name="l02946"></a>02946       <span class="keyword">static_cast&lt;</span><a class="code" href="classQuantBirdseed.html">QuantBirdseed</a> *<span class="keyword">&gt;</span>(qMethod)-&gt;addExprReporter(report);
<a name="l02947"></a>02947     }
<a name="l02948"></a>02948   }
<a name="l02949"></a>02949 }
<a name="l02950"></a>02950 
<a name="l02951"></a>02951 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::addReporters_Expression(affx::TsvReport::TsvReportFmt_t report_format,
<a name="l02952"></a>02952                                                           <a class="code" href="classAnalysisInfo.html" title="Information about a particular analysis run.">AnalysisInfo</a>&amp; info,
<a name="l02953"></a>02953                                                           std::vector&lt;QuantMethodReport *&gt; &amp;exprReporters,
<a name="l02954"></a>02954                                                           vector&lt;ChipSummary *&gt; &amp;chipSummaries,
<a name="l02955"></a>02955                                                           <a class="code" href="classGenderCalls.html" title="Class for determining gender from a cel set of cel files.">GenderCalls</a> *gender,
<a name="l02956"></a>02956                                                           std::set&lt;const char *, Util::ltstr&gt; *probeSetsToReport,
<a name="l02957"></a>02957                                                           <span class="keywordtype">bool</span> doCompact) {
<a name="l02958"></a>02958   <span class="comment">//</span>
<a name="l02959"></a>02959 
<a name="l02960"></a>02960     <a class="code" href="classQuantMethodExprReport.html" title="Class for reporting results of quantification methods.">QuantMethodExprReport</a> *report = <span class="keyword">new</span> <a class="code" href="classQuantMethodExprReport.html" title="Class for reporting results of quantification methods.">QuantMethodExprReport</a>(info.getNumCols());
<a name="l02961"></a>02961     report-&gt;setIsHeaderBuffer(1);
<a name="l02962"></a>02962     report-&gt;setDirPath(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;out-dir&quot;</span>));
<a name="l02963"></a>02963     <span class="comment">//    report-&gt;m_qMethod=eMethod;</span>
<a name="l02964"></a>02964     report-&gt;setPrecision(5);
<a name="l02965"></a>02965     report-&gt;setWriteOldStyleFeatureEffectsFile(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;writeOldStyleFeatureEffectsFile&quot;</span>));
<a name="l02966"></a>02966 
<a name="l02967"></a>02967     <span class="comment">// We do not do this for the expression reporters as output</span>
<a name="l02968"></a>02968     <span class="comment">// is controlled by the quant method. If you set this then</span>
<a name="l02969"></a>02969     <span class="comment">// nothing will be reported as the probeset names reported</span>
<a name="l02970"></a>02970     <span class="comment">// have the extra allele and context stuff appended.</span>
<a name="l02971"></a>02971     <span class="comment">//report-&gt;registerProbeSetsToReport(probeSetsToReport);</span>
<a name="l02972"></a>02972     <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;include-quant-in-report-file-name&quot;</span>))
<a name="l02973"></a>02973       report-&gt;setFileprefix(info.<a class="code" href="classAnalysisInfo.html#a1c684133674d56f7588c24c56c6a4c75" title="Name of the analysis.">m_AnalysisName</a>+<span class="stringliteral">&quot;.emethod&quot;</span>);
<a name="l02974"></a>02974     <span class="keywordflow">else</span>
<a name="l02975"></a>02975         report-&gt;setFileprefix(info.<a class="code" href="classAnalysisInfo.html#a1c684133674d56f7588c24c56c6a4c75" title="Name of the analysis.">m_AnalysisName</a>);
<a name="l02976"></a>02976     report-&gt;setFilename(<span class="stringliteral">&quot;QuantMethodExprReport-DEBUG&quot;</span>); <span class="comment">// filename not used -- for debugging</span>
<a name="l02977"></a>02977 
<a name="l02978"></a>02978     <span class="keywordflow">if</span>(report_format == affx::TsvReport::FMT_TSV) {
<a name="l02979"></a>02979         report-&gt;m_DoSummary=<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;summaries&quot;</span>);
<a name="l02980"></a>02980         report-&gt;m_DoResiduals=<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;feat-details&quot;</span>);
<a name="l02981"></a>02981         report-&gt;m_DoFeatureEffects=<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;feat-effects&quot;</span>);
<a name="l02982"></a>02982     } <span class="keywordflow">else</span> {
<a name="l02983"></a>02983         report-&gt;m_DoSummary=<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-summaries&quot;</span>);
<a name="l02984"></a>02984         report-&gt;m_DoResiduals=<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-feature-details&quot;</span>);
<a name="l02985"></a>02985         report-&gt;m_DoFeatureEffects=<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-feature-effects&quot;</span>);
<a name="l02986"></a>02986         
<a name="l02987"></a>02987         <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-summaries-use-global&quot;</span>)) {
<a name="l02988"></a>02988             <span class="keywordflow">if</span>(m_a5_global_output_group == NULL)
<a name="l02989"></a>02989                 <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;--a5-summaries-use-global option given, but no global file. Must specify --a5-global-file.&quot;</span>);
<a name="l02990"></a>02990             report-&gt;m_summary_a5_group = m_a5_global_output_group;
<a name="l02991"></a>02991             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;Using global A5 file for allele summary output.&quot;</span>);
<a name="l02992"></a>02992         }
<a name="l02993"></a>02993     
<a name="l02994"></a>02994         <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-feature-details-use-global&quot;</span>)) {
<a name="l02995"></a>02995             <span class="keywordflow">if</span>(m_a5_global_output_group == NULL)
<a name="l02996"></a>02996                 <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;--a5-feature-details-use-global option given, but no global file. Must specify --a5-global-file.&quot;</span>);
<a name="l02997"></a>02997             report-&gt;m_residuals_a5_group = m_a5_global_output_group;
<a name="l02998"></a>02998             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;Using global A5 file for residuals output.&quot;</span>);
<a name="l02999"></a>02999         }
<a name="l03000"></a>03000     
<a name="l03001"></a>03001         <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;a5-feature-effects-use-global&quot;</span>)) {
<a name="l03002"></a>03002             <span class="keywordflow">if</span>(m_a5_global_output_group == NULL)
<a name="l03003"></a>03003                 <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;--a5-feature-effects-use-global option given, but no global file. Must specify --a5-global-file.&quot;</span>);
<a name="l03004"></a>03004             report-&gt;m_feffects_a5_group = m_a5_global_output_group;
<a name="l03005"></a>03005             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;Using global A5 file for feature effects output.&quot;</span>);
<a name="l03006"></a>03006         }
<a name="l03007"></a>03007     }
<a name="l03008"></a>03008 
<a name="l03009"></a>03009     report-&gt;setFormat(report_format);
<a name="l03010"></a>03010     <span class="keywordflow">if</span>(doCompact &amp;&amp; report_format == affx::TsvReport::FMT_A5) {
<a name="l03011"></a>03011       report-&gt;setCompactFile5Format(info.<a class="code" href="classAnalysisInfo.html#af50a44fcfb27fcf4aedbb5664857ff24" title="Length of the longest probeset name.">m_MaxPsNameLength</a> + 2);
<a name="l03012"></a>03012     }
<a name="l03013"></a>03013     report-&gt;<a class="code" href="classQuantMethodReport.html#acc296969ce5774f4479fc7275de753f0" title="Print a message out to the stream.">addStdHeaders</a>(report,
<a name="l03014"></a>03014                           <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;exec-guid&quot;</span>),
<a name="l03015"></a>03015                           info.<a class="code" href="classAnalysisInfo.html#ab9d3f400692f10270650ed0f02e8d2a0" title="GUID (global unique ID) associated with a specific analysis.">m_AnalysisGuid</a>,
<a name="l03016"></a>03016                           <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;time-start&quot;</span>),
<a name="l03017"></a>03017                           <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;command-line&quot;</span>),
<a name="l03018"></a>03018                           <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;version-to-report&quot;</span>),
<a name="l03019"></a>03019                           info);
<a name="l03020"></a>03020 
<a name="l03021"></a>03021   <span class="comment">//   if(InstanceOf(qMethod,QuantLabelZ)) {</span>
<a name="l03022"></a>03022   <span class="comment">//     static_cast&lt;QuantLabelZ *&gt;(qMethod)-&gt;addExprReporter(report);</span>
<a name="l03023"></a>03023   <span class="comment">//   } else if (InstanceOf(qMethod,QuantBRLMM)) {</span>
<a name="l03024"></a>03024   <span class="comment">//     static_cast&lt;QuantBRLMM *&gt;(qMethod)-&gt;addExprReporter(report);</span>
<a name="l03025"></a>03025   <span class="comment">//   } else if (InstanceOf(qMethod,QuantBirdseed)) {</span>
<a name="l03026"></a>03026   <span class="comment">//     static_cast&lt;QuantBirdseed *&gt;(qMethod)-&gt;addExprReporter(report);</span>
<a name="l03027"></a>03027   <span class="comment">//   }</span>
<a name="l03028"></a>03028   <span class="comment">// }</span>
<a name="l03029"></a>03029     exprReporters.push_back(report);
<a name="l03030"></a>03030 }
<a name="l03031"></a>03031 
<a name="l03032"></a>03032 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::fillInAnalysisInfo(<a class="code" href="classAnalysisInfo.html" title="Information about a particular analysis run.">AnalysisInfo</a> &amp;info, <a class="code" href="classAnalysisStream.html" title="Utility class for centralizing analysis pathway.">AnalysisStream</a> *as, std::string prefix) {<span class="comment"></span>
<a name="l03033"></a>03033 <span class="comment">    ///@todo need to refactor to ensure consistency between meta info tracked in probeset-summarize</span>
<a name="l03034"></a>03034 <span class="comment">    ///      versus probeset-genotype</span>
<a name="l03035"></a>03035 <span class="comment">    ///@todo automatically pull state and options info?</span>
<a name="l03036"></a>03036 <span class="comment">    ///@todo sync up option name with label in chp header</span>
<a name="l03037"></a>03037 <span class="comment"></span>    <span class="comment">// FOR NOW -- PLEASE KEEP THIS IN SYNC WITH probeset-genotype</span>
<a name="l03038"></a>03038     vector&lt;string&gt; celFiles = <a class="code" href="classOptions.html#ab20d0083424d9d9a8b94da46f487f57a" title="Return a vector of values for a particular option.">getOptVector</a>(<span class="stringliteral">&quot;cels&quot;</span>);
<a name="l03039"></a>03039     assert(as);
<a name="l03040"></a>03040     <a class="code" href="classQuantMethod.html" title="QuantMethod - Interface for computing quantification summaries from PM intensities grouped into probe...">QuantMethod</a> *qMethod = as-&gt;<a class="code" href="classAnalysisStream.html#af5d7e7ca7a32a6f189290b5ef1888f64" title="Get the quantification method used.">getQuantMethod</a>();
<a name="l03041"></a>03041     <a class="code" href="classErr.html#ad1a633f602fe60689cdc12d9ca1842e9" title="errAbort based version of assert with an error message.">Err::check</a>(qMethod != NULL, <span class="stringliteral">&quot;Options::fillInAnalysisInfo() - Must have quantification method.&quot;</span>);
<a name="l03042"></a>03042     info.<a class="code" href="classAnalysisInfo.html#a485fade346a6f248b915cbd0a6e1041c" title="What version of the algorithm is being used.">m_AlgVersion</a> = qMethod-&gt;<a class="code" href="classQuantMethod.html#a4799b1ab9adeff7ea3681938367ae670" title="What version of the expression algorithm are we implementing?">getVersion</a>();
<a name="l03043"></a>03043     info.<a class="code" href="classAnalysisInfo.html#a4e52df55ab7398557852b211c5a50d41" title="What algorithm are we reporting.">m_AlgName</a> = as-&gt;<a class="code" href="classAnalysisStream.html#a79af545326dd23f70df846094f1573c5" title="What is the name of this particular analysis stream?">getName</a>();
<a name="l03044"></a>03044     info.<a class="code" href="classAnalysisInfo.html#a2cc9986186213673fb4ca8decc70124e" title="What is the name of the program.">m_ProgramName</a> = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;program-name&quot;</span>);
<a name="l03045"></a>03045     info.<a class="code" href="classAnalysisInfo.html#a705a56532115d0c0c8f10644e633b64d" title="What program created this chp file.">m_ProgramVersion</a> = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;version-to-report&quot;</span>);
<a name="l03046"></a>03046     info.<a class="code" href="classAnalysisInfo.html#aa1f9eedaa9c2024dfde2af6a71155bb1" title="What company created this chp file.">m_ProgramCompany</a> = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;program-company&quot;</span>);
<a name="l03047"></a>03047     info.<a class="code" href="classAnalysisInfo.html#a89c1bfe7a31fccc70d06cb42f68c435e" title="What chip is this?">m_ChipType</a> = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chip-type&quot;</span>);
<a name="l03048"></a>03048     info.<a class="code" href="classAnalysisInfo.html#aa32b7ba5f935bc64316e903980e9acbe" title="What is our id for GCOS?">m_ProgID</a> = <span class="stringliteral">&quot;&quot;</span>;
<a name="l03049"></a>03049     info.<a class="code" href="classAnalysisInfo.html#ad906bb9bddd45bd2d3ed08ee3eff5fd3" title="GUID associated with a specific process.">m_ExecGuid</a> = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;exec-guid&quot;</span>);
<a name="l03050"></a>03050     info.<a class="code" href="classAnalysisInfo.html#ab9d3f400692f10270650ed0f02e8d2a0" title="GUID (global unique ID) associated with a specific analysis.">m_AnalysisGuid</a> = as-&gt;getGuid();
<a name="l03051"></a>03051 
<a name="l03052"></a>03052     <span class="comment">// State and Execution info</span>
<a name="l03053"></a>03053     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(<span class="stringliteral">&quot;apt-engine&quot;</span>, <span class="stringliteral">&quot;ProbesetGenotypeStageEngine&quot;</span>);
<a name="l03054"></a>03054     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + <span class="stringliteral">&quot;program-name&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;program-name&quot;</span>));
<a name="l03055"></a>03055     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + <span class="stringliteral">&quot;command-line&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;command-line&quot;</span>));
<a name="l03056"></a>03056     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + <span class="stringliteral">&quot;exec-guid&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;exec-guid&quot;</span>));
<a name="l03057"></a>03057     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + <span class="stringliteral">&quot;analysis-guid&quot;</span>, as-&gt;getGuid());
<a name="l03058"></a>03058     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + <span class="stringliteral">&quot;time-str&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;time-start&quot;</span>));
<a name="l03059"></a>03059     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + <span class="stringliteral">&quot;version&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;version-to-report&quot;</span>));
<a name="l03060"></a>03060     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + <span class="stringliteral">&quot;cvs-id&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;program-cvs-id&quot;</span>));
<a name="l03061"></a>03061     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + <span class="stringliteral">&quot;free-mem&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;free-mem-at-start&quot;</span>));<span class="comment"></span>
<a name="l03062"></a>03062 <span class="comment">    ///@todo add other state:</span>
<a name="l03063"></a>03063 <span class="comment">    ///@todo time-end state</span>
<a name="l03064"></a>03064 <span class="comment"></span>
<a name="l03065"></a>03065     <span class="comment">// Engine Options</span>
<a name="l03066"></a>03066     std::string subprefix = <span class="stringliteral">&quot;opt-&quot;</span>;
<a name="l03067"></a>03067     <span class="comment">// no do-ps-names</span>
<a name="l03068"></a>03068     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;chip-type&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chip-type&quot;</span>));
<a name="l03069"></a>03069     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;probe-count&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;probe-count&quot;</span>));
<a name="l03070"></a>03070     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;force&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;force&quot;</span>));
<a name="l03071"></a>03071     <span class="comment">// no precision</span>
<a name="l03072"></a>03072     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;out-dir&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;out-dir&quot;</span>));
<a name="l03073"></a>03073     <span class="comment">// cc-expr-chp-out-dir</span>
<a name="l03074"></a>03074     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;cc-md-chp-out-dir&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;cc-chp-out-dir&quot;</span>));
<a name="l03075"></a>03075     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;xda-chp-out-dir&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;xda-chp-out-dir&quot;</span>));
<a name="l03076"></a>03076     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;cdf-file&quot;</span>,       <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;cdf-file&quot;</span>)));
<a name="l03077"></a>03077     <span class="comment">// no spf-file </span>
<a name="l03078"></a>03078     <span class="comment">// no pgf-file</span>
<a name="l03079"></a>03079     <span class="comment">// no clf-file</span>
<a name="l03080"></a>03080     <span class="comment">// no bgp-file</span>
<a name="l03081"></a>03081     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;ps-list-file&quot;</span>,   <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;probeset-ids&quot;</span>)));
<a name="l03082"></a>03082     <span class="comment">// no meta-ps-file</span>
<a name="l03083"></a>03083     <span class="comment">// no-qc-groups-file</span>
<a name="l03084"></a>03084     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;kill-list&quot;</span>,      <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;kill-list&quot;</span>)));
<a name="l03085"></a>03085     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;temp-dir&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;temp-dir&quot;</span>));
<a name="l03086"></a>03086     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;use-disk&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;use-disk&quot;</span>));
<a name="l03087"></a>03087     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;diskCache&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;disk-cache&quot;</span>));
<a name="l03088"></a>03088     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;cel-count&quot;</span>, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(celFiles.size()));
<a name="l03089"></a>03089     <span class="keywordflow">for</span>(uint32_t i = 0; i &lt; celFiles.size(); i++) {
<a name="l03090"></a>03090       std::string paramName = prefix + subprefix + <span class="stringliteral">&quot;cel-&quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(i+1);
<a name="l03091"></a>03091       info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(paramName, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(celFiles[i]));
<a name="l03092"></a>03092     }
<a name="l03093"></a>03093 
<a name="l03094"></a>03094     <span class="comment">// Algorithm Options</span>
<a name="l03095"></a>03095     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;analysis-name&quot;</span>, as-&gt;<a class="code" href="classAnalysisStream.html#a79af545326dd23f70df846094f1573c5" title="What is the name of this particular analysis stream?">getName</a>());
<a name="l03096"></a>03096     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;set-analysis-name&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;set-analysis-name&quot;</span>));
<a name="l03097"></a>03097     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;analysis-spec&quot;</span>, as-&gt;<a class="code" href="classAnalysisStream.html#a024267de87ab9ad778d2ee2ccdff8022" title="Get a formal description of each object in the analysis stream including chipstream objects...">getAnalysisSpec</a>());
<a name="l03098"></a>03098     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;feat-effect-file&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;use-feat-eff&quot;</span>)));
<a name="l03099"></a>03099     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;target-sketch-file&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;target-sketch&quot;</span>)));
<a name="l03100"></a>03100     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;reference-profile-file&quot;</span>,<a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;reference-profile&quot;</span>)));
<a name="l03101"></a>03101     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;do-residuals&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;feat-details&quot;</span>));
<a name="l03102"></a>03102     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;do-feature-effects&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;feat-effects&quot;</span>));
<a name="l03103"></a>03103     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;write-sketch&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;write-sketch&quot;</span>));
<a name="l03104"></a>03104     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;write-profile&quot;</span>,<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;write-profile&quot;</span>));
<a name="l03105"></a>03105 
<a name="l03106"></a>03106     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;a5-write-sketch&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;a5-sketch&quot;</span>));
<a name="l03107"></a>03107     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;set-gender-method&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;set-gender-method&quot;</span>));
<a name="l03108"></a>03108     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;chrXProbeFile&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chrX-probes&quot;</span>)));
<a name="l03109"></a>03109     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;chrYProbeFile&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chrY-probes&quot;</span>)));
<a name="l03110"></a>03110     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;read-priors-brlmm&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-priors-brlmm&quot;</span>)));
<a name="l03111"></a>03111     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;initial-calls&quot;</span>, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(<span class="stringliteral">&quot;false&quot;</span>));
<a name="l03112"></a>03112     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;output-summaries&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;summaries&quot;</span>));
<a name="l03113"></a>03113     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;list-sample&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;list-sample&quot;</span>));
<a name="l03114"></a>03114     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;write-prior&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;write-prior&quot;</span>));
<a name="l03115"></a>03115     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;chrX-file&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chrX-snps&quot;</span>)));
<a name="l03116"></a>03116     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;chrX-snps&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chrX-snps&quot;</span>)));
<a name="l03117"></a>03117     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;no-gender-force&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;no-gender-force&quot;</span>));
<a name="l03118"></a>03118     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;special-snps&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;special-snps&quot;</span>)));
<a name="l03119"></a>03119     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;compute-prior&quot;</span>, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(<a class="code" href="classOptions.html#afcfab86e21cac5bba43e0858bd428e62" title="Get the integer value of an option.">getOptInt</a>(<span class="stringliteral">&quot;prior-size&quot;</span>)));
<a name="l03120"></a>03120     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;norm-size&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;norm-size&quot;</span>));
<a name="l03121"></a>03121     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;dm-thresh&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;dm-thresh&quot;</span>));
<a name="l03122"></a>03122     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;dm-het-mult&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;dm-hetmult&quot;</span>));
<a name="l03123"></a>03123     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;read-genders&quot;</span>,        <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-genders&quot;</span>)));
<a name="l03124"></a>03124     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;read-inbred&quot;</span>,         <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-inbred&quot;</span>)));
<a name="l03125"></a>03125     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;model-file-brlmm&quot;</span>   , <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-brlmm&quot;</span>)));
<a name="l03126"></a>03126     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;model-file-brlmmp&quot;</span>  , <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-brlmmp&quot;</span>)));
<a name="l03127"></a>03127     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;model-file-birdseed&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-birdseed&quot;</span>)));
<a name="l03128"></a>03128     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;genotypes-file&quot;</span>     , <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;genotypes&quot;</span>)));
<a name="l03129"></a>03129     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;dm-calls-out&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;dm-out&quot;</span>));
<a name="l03130"></a>03130     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;qmethod-spec&quot;</span>, <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;qmethod-spec&quot;</span>));
<a name="l03131"></a>03131     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(prefix + subprefix + <span class="stringliteral">&quot;genotype-only&quot;</span>, <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(!<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;all-types&quot;</span>)));
<a name="l03132"></a>03132 
<a name="l03133"></a>03133     <span class="comment">// Quantification Related Meta Info</span>
<a name="l03134"></a>03134     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(<span class="stringliteral">&quot;quantification-name&quot;</span>, qMethod-&gt;<a class="code" href="classQuantMethod.html#ab1b819505891590c9f3e57d9cc67d4de" title="What is the name of the quantification method?">getType</a>());
<a name="l03135"></a>03135     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(<span class="stringliteral">&quot;quantification-version&quot;</span>, qMethod-&gt;<a class="code" href="classQuantMethod.html#a4799b1ab9adeff7ea3681938367ae670" title="What version of the expression algorithm are we implementing?">getVersion</a>());
<a name="l03136"></a>03136     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(<span class="stringliteral">&quot;quantification-scale&quot;</span>, <a class="code" href="classQuantMethod.html#a6a4921cca844ef99404228081863c8c4" title="Convert a Scale enumeration to a text description.">QuantMethod::scaleToTxt</a>(qMethod-&gt;<a class="code" href="classQuantMethod.html#a2284897bb0665c4cea9f8e030e20d796" title="What scale are the target effects on? Are they linear (like plier) or log2 (like rma).">getScale</a>()));
<a name="l03137"></a>03137     info.<a class="code" href="classAnalysisInfo.html#af479563f1370004fffb4b14f67aefe23" title="Add a key/value parameter pair to our inforomation.">addParam</a>(<span class="stringliteral">&quot;quantification-type&quot;</span>, <a class="code" href="classQuantMethod.html#aaab461044d36fd604004b47934b209d3" title="Convert a QuantType enumeration to a human readable string representation.">QuantMethod::quantTypeToTxt</a>(qMethod-&gt;<a class="code" href="classQuantMethod.html#a2889dd697efae0503edd00e4087e991b" title="What type of quantification method is this? Summarization like plier detection like dabg...">getQuantType</a>()));
<a name="l03138"></a>03138 
<a name="l03139"></a>03139     <span class="comment">// Add user defined meta data parameters</span>
<a name="l03140"></a>03140     vector&lt;pair&lt;string, string&gt; &gt; metaData = <a class="code" href="classBaseEngine.html#ac9fb139876b95877fb7412d0dc27cdd8" title="Get a vector of the key=value pairs specified via the &amp;quot;meta-data-info&amp;quot; parameters.">getMetaDataDescription</a>();
<a name="l03141"></a>03141     <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; metaData.size(); i++) {
<a name="l03142"></a>03142       pair&lt;string,string&gt; p = metaData[i];
<a name="l03143"></a>03143       info.<a class="code" href="classAnalysisInfo.html#a9482ae5e2f30e58b5a06d1f689edef59" title="Add client supplied key/value parameter pair to our information.">addClientMetaInfo</a>(p.first, p.second);
<a name="l03144"></a>03144     }
<a name="l03145"></a>03145 }
<a name="l03146"></a>03146 
<a name="l03147"></a>03147 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::extraHelp() {
<a name="l03148"></a>03148     <a class="code" href="classChipStreamFactory.html" title="Factory class for making chip streams based on a string representation.">ChipStreamFactory</a> cFactory;
<a name="l03149"></a>03149     <a class="code" href="classPmAdjusterFactory.html" title="Factory class for making chip streams based on a string representation.">PmAdjusterFactory</a> aFactory;
<a name="l03150"></a>03150     <a class="code" href="classQuantMethodFactory.html" title="Factory class for making chip streams based on a string representation.">QuantMethodFactory</a> qFactory(QuantMethodFactory::Expression);
<a name="l03151"></a>03151     <a class="code" href="classAnalysisStreamFactory.html" title="Object for making new analysis streams from a text specification.">AnalysisStreamFactory</a> asFactory;
<a name="l03152"></a>03152 
<a name="l03153"></a>03153     printStandardMethods(cout);
<a name="l03154"></a>03154     <a class="code" href="classEngineUtil.html#ac60d1bb1a13c6bb184c42089ac962095" title="Print out a header and then a series of self documenting objects.">EngineUtil::printSelfDocs</a>(<span class="stringliteral">&quot;Data transformations:&quot;</span>, cFactory.<a class="code" href="classChipStreamFactory.html#a09751e33c85bfda7d8c8addee9cc1918" title="Get the vector of documentation objects that this factory is aware of.">getDocs</a>());
<a name="l03155"></a>03155     <a class="code" href="classEngineUtil.html#ac60d1bb1a13c6bb184c42089ac962095" title="Print out a header and then a series of self documenting objects.">EngineUtil::printSelfDocs</a>(<span class="stringliteral">&quot;Pm Intensity Adjustments:&quot;</span>, aFactory.<a class="code" href="classPmAdjusterFactory.html#a6f54e0030cc442973e3d95752f9c31cc" title="Get the vector of documentation objects that this factory is aware of.">getDocs</a>());
<a name="l03156"></a>03156     <a class="code" href="classEngineUtil.html#ac60d1bb1a13c6bb184c42089ac962095" title="Print out a header and then a series of self documenting objects.">EngineUtil::printSelfDocs</a>(<span class="stringliteral">&quot;Quantification Methods:&quot;</span>, qFactory.getDocs());
<a name="l03157"></a>03157     <a class="code" href="classEngineUtil.html#ac60d1bb1a13c6bb184c42089ac962095" title="Print out a header and then a series of self documenting objects.">EngineUtil::printSelfDocs</a>(<span class="stringliteral">&quot;Analysis Streams:&quot;</span>, asFactory.<a class="code" href="classAnalysisStreamFactory.html#ac7547e5f2c10200dc6bf49ba69e84405" title="Get the vector of documentation objects that this factory is aware of.">getDocs</a>());
<a name="l03158"></a>03158 }
<a name="l03159"></a>03159 
<a name="l03160"></a>03160 <span class="keywordtype">void</span> ProbesetGenotypeStageEngine::explain() {
<a name="l03161"></a>03161     <a class="code" href="classEngineUtil.html#a090e75dbffbe46910abd1fdc41c8b8c7" title="Print out a little ditty for the given type of self documenting object.">EngineUtil::explainParameter</a>(<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;explain&quot;</span>));
<a name="l03162"></a>03162 }
</pre></div></div>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:32 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
