<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: chipstream/QuantBRLMM.h Source File</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li class="current"><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="tabs2">
    <ul class="tablist">
      <li><a href="files.html"><span>File&nbsp;List</span></a></li>
      <li><a href="globals.html"><span>File&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath">
    <ul>
      <li><a class="el" href="dir_f48957cb46a44244b1a86c58bd157ee8.html">chipstream</a>      </li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>QuantBRLMM.h</h1>  </div>
</div>
<div class="contents">
<a href="QuantBRLMM_8h.html">Go to the documentation of this file.</a><div class="fragment"><pre class="fragment"><a name="l00001"></a>00001 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00002"></a>00002 <span class="comment"></span><span class="comment">//</span>
<a name="l00003"></a>00003 <span class="comment">// Copyright (C) 2005 Affymetrix, Inc.</span>
<a name="l00004"></a>00004 <span class="comment">//</span>
<a name="l00005"></a>00005 <span class="comment">// This program is free software; you can redistribute it and/or modify </span>
<a name="l00006"></a>00006 <span class="comment">// it under the terms of the GNU General Public License (version 2) as </span>
<a name="l00007"></a>00007 <span class="comment">// published by the Free Software Foundation.</span>
<a name="l00008"></a>00008 <span class="comment">// </span>
<a name="l00009"></a>00009 <span class="comment">// This program is distributed in the hope that it will be useful, </span>
<a name="l00010"></a>00010 <span class="comment">// but WITHOUT ANY WARRANTY; without even the implied warranty of </span>
<a name="l00011"></a>00011 <span class="comment">// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU </span>
<a name="l00012"></a>00012 <span class="comment">// General Public License for more details.</span>
<a name="l00013"></a>00013 <span class="comment">// </span>
<a name="l00014"></a>00014 <span class="comment">// You should have received a copy of the GNU General Public License </span>
<a name="l00015"></a>00015 <span class="comment">// along with this program;if not, write to the </span>
<a name="l00016"></a>00016 <span class="comment">// </span>
<a name="l00017"></a>00017 <span class="comment">// Free Software Foundation, Inc., </span>
<a name="l00018"></a>00018 <span class="comment">// 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA</span>
<a name="l00019"></a>00019 <span class="comment">//</span><span class="comment"></span>
<a name="l00020"></a>00020 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00021"></a>00021 <span class="comment"></span><span class="comment"></span>
<a name="l00022"></a>00022 <span class="comment">/**</span>
<a name="l00023"></a>00023 <span class="comment"> * @file   QuantBRLMM.h</span>
<a name="l00024"></a>00024 <span class="comment"> * @author Chuck Sugnet</span>
<a name="l00025"></a>00025 <span class="comment"> * @date   Wed Feb 22 09:50:58 2006</span>
<a name="l00026"></a>00026 <span class="comment"> *</span>
<a name="l00027"></a>00027 <span class="comment"> * Implements a SNP genotype calling method motivated by Nusrat Rabbee&#39;s</span>
<a name="l00028"></a>00028 <span class="comment"> * RLMM (Robust Linear Model with Mahalanobis Distance Classifier) that utilizes</span>
<a name="l00029"></a>00029 <span class="comment"> * a Baysian prior about the centers of SNP clusters. Idea is to produce a</span>
<a name="l00030"></a>00030 <span class="comment"> * summary value for all probes that measure same genotype (default using RMA to</span>
<a name="l00031"></a>00031 <span class="comment"> * summarize) and then classify the genotype based on the relative intensity of</span>
<a name="l00032"></a>00032 <span class="comment"> * those estimates. For example in genotype AA the A probesets should be</span>
<a name="l00033"></a>00033 <span class="comment"> * relatively bright and B probesets should be relatively dim. For genotype AB</span>
<a name="l00034"></a>00034 <span class="comment"> * both probesets should be lighting up. Exactly how much intensity is expected</span>
<a name="l00035"></a>00035 <span class="comment"> * is learned from a random set of SNPs in the data using high confidence DM</span>
<a name="l00036"></a>00036 <span class="comment"> * calls as reference.</span>
<a name="l00037"></a>00037 <span class="comment"> *</span>
<a name="l00038"></a>00038 <span class="comment"> * For further information about RLMM see:</span>
<a name="l00039"></a>00039 <span class="comment"> * http://www.stat.berkeley.edu/users/nrabbee/RLMM/</span>
<a name="l00040"></a>00040 <span class="comment"> *</span>
<a name="l00041"></a>00041 <span class="comment"> *</span>
<a name="l00042"></a>00042 <span class="comment"> */</span>
<a name="l00043"></a>00043 <span class="comment">/*</span>
<a name="l00044"></a>00044 <span class="comment">   We use floats to be compatible with R.</span>
<a name="l00045"></a>00045 <span class="comment"></span>
<a name="l00046"></a>00046 <span class="comment">   Questions:</span>
<a name="l00047"></a>00047 <span class="comment"></span>
<a name="l00048"></a>00048 <span class="comment">   - What happens in snp.clusterStats.R::clusterStats() when there are no</span>
<a name="l00049"></a>00049 <span class="comment">   genotypes observe? Looks like a bunch of NA&#39;s are passed back, what</span>
<a name="l00050"></a>00050 <span class="comment">   happens to them?</span>
<a name="l00051"></a>00051 <span class="comment">   - Couldn&#39;t calculate B summary values for SNP: AFFX-barcodeS</span>
<a name="l00052"></a>00052 <span class="comment">*/</span>
<a name="l00053"></a>00053 
<a name="l00054"></a>00054 <span class="preprocessor">#ifndef QUANTBRLMM_H</span>
<a name="l00055"></a>00055 <span class="preprocessor"></span><span class="preprocessor">#define QUANTBRLMM_H</span>
<a name="l00056"></a>00056 <span class="preprocessor"></span>
<a name="l00057"></a>00057 <span class="preprocessor">#define QUANTBRLMM_VERSION &quot;1.0&quot;</span>
<a name="l00058"></a>00058 <span class="preprocessor"></span>
<a name="l00059"></a>00059 <span class="comment">//</span>
<a name="l00060"></a>00060 <span class="preprocessor">#include &quot;<a class="code" href="BioTypes_8h.html" title="Central location enumerations and other types that aren&amp;#39;t particular to a single class or applica...">chipstream/BioTypes.h</a>&quot;</span>
<a name="l00061"></a>00061 <span class="preprocessor">#include &quot;<a class="code" href="ClusterZ_8h.html">chipstream/ClusterZ.h</a>&quot;</span>
<a name="l00062"></a>00062 <span class="preprocessor">#include &quot;<a class="code" href="ProbeSet_8h.html" title="Container class for probe sets.">chipstream/ProbeSet.h</a>&quot;</span>
<a name="l00063"></a>00063 <span class="preprocessor">#include &quot;<a class="code" href="QuantExprMethod_8h.html" title="Interface for computing quantification summaries from PM intensities grouped into probe set groups...">chipstream/QuantExprMethod.h</a>&quot;</span>
<a name="l00064"></a>00064 <span class="preprocessor">#include &quot;<a class="code" href="QuantGTypeMethod_8h.html" title="Interface for methods computing SNP genotyping calls.">chipstream/QuantGTypeMethod.h</a>&quot;</span>
<a name="l00065"></a>00065 <span class="preprocessor">#include &quot;<a class="code" href="QuantMethod_8h.html" title="Abstract interface for a quantification method.">chipstream/QuantMethod.h</a>&quot;</span>
<a name="l00066"></a>00066 <span class="preprocessor">#include &quot;<a class="code" href="TsvReport_8h.html" title="///">chipstream/TsvReport.h</a>&quot;</span>
<a name="l00067"></a>00067 <span class="comment">//</span>
<a name="l00068"></a>00068 <span class="preprocessor">#include &quot;<a class="code" href="stats_8h.html" title="///">stats/stats.h</a>&quot;</span>
<a name="l00069"></a>00069 <span class="comment">//</span>
<a name="l00070"></a>00070 <span class="preprocessor">#include &quot;newmat.h&quot;</span>
<a name="l00071"></a>00071 <span class="comment">//</span>
<a name="l00072"></a>00072 <span class="preprocessor">#include &lt;cfloat&gt;</span>
<a name="l00073"></a>00073 <span class="preprocessor">#include &lt;cstring&gt;</span>
<a name="l00074"></a>00074 <span class="preprocessor">#include &lt;string&gt;</span>
<a name="l00075"></a>00075 <span class="preprocessor">#include &lt;vector&gt;</span>
<a name="l00076"></a>00076 <span class="comment">//</span>
<a name="l00077"></a>00077 
<a name="l00078"></a>00078 
<a name="l00079"></a>00079 <span class="preprocessor">#if defined (WIN32)</span>
<a name="l00080"></a>00080 <span class="preprocessor"></span><span class="preprocessor">#define asinh(x) (log(x+sqrt(x*x+1)))</span>
<a name="l00081"></a>00081 <span class="preprocessor"></span><span class="preprocessor">#endif</span>
<a name="l00082"></a>00082 <span class="preprocessor"></span><span class="comment"></span>
<a name="l00083"></a>00083 <span class="comment">/// String describing quant method</span>
<a name="l00084"></a><a class="code" href="QuantBRLMM_8h.html#aa9aa554ff92b91869db108754ee24023">00084</a> <span class="comment"></span><span class="preprocessor">#define QUANTBRLMM &quot;brlmm&quot;</span>
<a name="l00085"></a>00085 <span class="preprocessor"></span><span class="comment"></span>
<a name="l00086"></a>00086 <span class="comment">/**</span>
<a name="l00087"></a>00087 <span class="comment"> * Implements a SNP genotype calling method motivated by Nusrat Rabbee&#39;s</span>
<a name="l00088"></a>00088 <span class="comment"> * RLMM (Robust Linear Model with Mahalanobis Distance Classifier) that utilizes</span>
<a name="l00089"></a>00089 <span class="comment"> * a Baysian prior about the centers of SNP clusters. Idea is to produce a</span>
<a name="l00090"></a>00090 <span class="comment"> * summary value for all probes that measure same genotype (default using RMA to</span>
<a name="l00091"></a>00091 <span class="comment"> * summarize) and then classify the genotype based on the relative intensity of</span>
<a name="l00092"></a>00092 <span class="comment"> * those estimates. For example in genotype AA the A probesets should be</span>
<a name="l00093"></a>00093 <span class="comment"> * relatively bright and B probesets should be relatively dim. For genotype AB</span>
<a name="l00094"></a>00094 <span class="comment"> * both probesets should be lighting up. Exactly how much intensity is expected</span>
<a name="l00095"></a>00095 <span class="comment"> * is learned from a random set of SNPs in the data using high confidence DM</span>
<a name="l00096"></a>00096 <span class="comment"> * calls as reference.</span>
<a name="l00097"></a>00097 <span class="comment"> *</span>
<a name="l00098"></a>00098 <span class="comment"> * For further information about RLMM see:</span>
<a name="l00099"></a>00099 <span class="comment"> * http://www.stat.berkeley.edu/users/nrabbee/RLMM/</span>
<a name="l00100"></a>00100 <span class="comment"> */</span>
<a name="l00101"></a><a class="code" href="classQuantBRLMM.html">00101</a> <span class="keyword">class </span><a class="code" href="classQuantBRLMM.html" title="Implements a SNP genotype calling method motivated by Nusrat Rabbee&amp;#39;s RLMM (Robust Linear Model w...">QuantBRLMM</a> : <span class="keyword">public</span> <a class="code" href="classQuantGTypeMethod.html" title="Quantification methods used for making genotyping calls implement (currently just brlmm) this interfa...">QuantGTypeMethod</a> {
<a name="l00102"></a>00102 
<a name="l00103"></a>00103 <span class="keyword">public</span>:
<a name="l00104"></a>00104 <span class="comment"></span>
<a name="l00105"></a>00105 <span class="comment">  /// What transformation of intensity estimates to feed into classifier.</span>
<a name="l00106"></a><a class="code" href="classQuantBRLMM.html#aa41baee052c062022a0ab6a999c21638">00106</a> <span class="comment"></span>  <span class="keyword">enum</span> <a class="code" href="classQuantBRLMM.html#aa41baee052c062022a0ab6a999c21638" title="What transformation of intensity estimates to feed into classifier.">Transformation</a> {
<a name="l00107"></a><a class="code" href="classQuantBRLMM.html#aa41baee052c062022a0ab6a999c21638aaf0dc6d5b83fda75ca175ce28504710c">00107</a>     MvA, <span class="comment">///&lt; Minus Vs Average</span>
<a name="l00108"></a><a class="code" href="classQuantBRLMM.html#aa41baee052c062022a0ab6a999c21638aea687c012cf0ba203c45713fd99a9673">00108</a> <span class="comment"></span>    RvT, <span class="comment">///&lt; R vs Theta</span>
<a name="l00109"></a><a class="code" href="classQuantBRLMM.html#aa41baee052c062022a0ab6a999c21638a2729e1dbbb52465172e689eada41e72a">00109</a> <span class="comment"></span>    CES, <span class="comment">///&lt; Contrast Extremes Stretch (previously known as south san francisco).</span>
<a name="l00110"></a><a class="code" href="classQuantBRLMM.html#aa41baee052c062022a0ab6a999c21638a2dea69397eccbeb17925a9d8361f9fb8">00110</a> <span class="comment"></span>    <a class="code" href="classQuantBRLMM.html#aa41baee052c062022a0ab6a999c21638a2dea69397eccbeb17925a9d8361f9fb8" title="Contrast Centers Stretch (previously known as alternative south san francisco).">CCS</a>  <span class="comment">///&lt; Contrast Centers Stretch (previously known as alternative south san francisco).</span>
<a name="l00111"></a>00111 <span class="comment"></span>  };
<a name="l00112"></a>00112 
<a name="l00113"></a><a class="code" href="classQuantBRLMM_1_1BrlmmParam.html">00113</a>   <span class="keyword">class </span><a class="code" href="classQuantBRLMM_1_1BrlmmParam.html">BrlmmParam</a> {
<a name="l00114"></a>00114 
<a name="l00115"></a>00115   <span class="keyword">public</span>:
<a name="l00116"></a>00116 
<a name="l00117"></a>00117     <a class="code" href="classQuantBRLMM_1_1BrlmmParam.html">BrlmmParam</a>();
<a name="l00118"></a>00118 
<a name="l00119"></a><a class="code" href="classQuantBRLMM_1_1BrlmmParam.html#a474238d1f98a341c21dc9c426e79036d">00119</a>     <span class="keywordtype">double</span> <a class="code" href="classQuantBRLMM_1_1BrlmmParam.html#a474238d1f98a341c21dc9c426e79036d" title="Number to balance het calls with to balance performance on het/hom calls.">m_HetMultiplier</a>;          <span class="comment">///&lt; Number to balance het calls with to balance performance on het/hom calls.</span>
<a name="l00120"></a><a class="code" href="classQuantBRLMM_1_1BrlmmParam.html#adf7bcd06c19218d14fe7d49b79b3c7b2">00120</a> <span class="comment"></span>    <span class="keywordtype">int</span> <a class="code" href="classQuantBRLMM_1_1BrlmmParam.html#adf7bcd06c19218d14fe7d49b79b3c7b2" title="How many iterations to do of BRLMM, i.e. boostrapping from initial calls.">m_Iterations</a>;                <span class="comment">///&lt; How many iterations to do of BRLMM, i.e. boostrapping from initial calls.</span>
<a name="l00121"></a><a class="code" href="classQuantBRLMM_1_1BrlmmParam.html#ad9919e718c6d119d94f2b824abce8d53">00121</a> <span class="comment"></span>    <span class="keywordtype">double</span> <a class="code" href="classQuantBRLMM_1_1BrlmmParam.html#ad9919e718c6d119d94f2b824abce8d53" title="What is our threshold (maximum value) for confidences befor /// calling &amp;#39;NN&amp;#39; in iterative mod...">m_IterationThreshold</a>;      <span class="comment">///&lt; What is our threshold (maximum value) for confidences befor</span>
<a name="l00122"></a>00122 <span class="comment"></span><span class="comment">                                     ///  calling &#39;NN&#39; in iterative mode?</span>
<a name="l00123"></a><a class="code" href="classQuantBRLMM_1_1BrlmmParam.html#a42c05834641c32011c0f5b6eac7ffe97">00123</a> <span class="comment"></span>    <span class="keywordtype">double</span> <a class="code" href="classQuantBRLMM_1_1BrlmmParam.html#a42c05834641c32011c0f5b6eac7ffe97" title="Scaling Parameter to use in transformations.">m_K</a>;                      <span class="comment">///&lt; Scaling Parameter to use in transformations.</span>
<a name="l00124"></a><a class="code" href="classQuantBRLMM_1_1BrlmmParam.html#a928f99fd333fed0c34013ce8d23acb49">00124</a> <span class="comment"></span>    <span class="keyword">enum</span> <a class="code" href="classQuantBRLMM.html#aa41baee052c062022a0ab6a999c21638" title="What transformation of intensity estimates to feed into classifier.">Transformation</a> <a class="code" href="classQuantBRLMM_1_1BrlmmParam.html#a928f99fd333fed0c34013ce8d23acb49" title="What transformation of initial data are we feeding into the classifier?">m_Transform</a>; <span class="comment">///&lt; What transformation of initial data are we feeding into the classifier?</span>
<a name="l00125"></a><a class="code" href="classQuantBRLMM_1_1BrlmmParam.html#a3d250342e8ec26cdfe1f162274d7bfab">00125</a> <span class="comment"></span>    <span class="keywordtype">int</span> <a class="code" href="classQuantBRLMM_1_1BrlmmParam.html#a3d250342e8ec26cdfe1f162274d7bfab" title="How much weight should the prior get?">m_PriorPseudoCount</a>;          <span class="comment">///&lt; How much weight should the prior get?</span>
<a name="l00126"></a><a class="code" href="classQuantBRLMM_1_1BrlmmParam.html#aa94a4cd0558a115b819dadc3501bff80">00126</a> <span class="comment"></span>    <span class="keywordtype">int</span> <a class="code" href="classQuantBRLMM_1_1BrlmmParam.html#aa94a4cd0558a115b819dadc3501bff80" title="Minimum examples for each genotype to be included in prior estimation.">m_MinCluster</a>;                <span class="comment">///&lt; Minimum examples for each genotype to be included in prior estimation.</span>
<a name="l00127"></a><a class="code" href="classQuantBRLMM_1_1BrlmmParam.html#aa9ebe7bbc8cb7de369bf2df9e795690c">00127</a> <span class="comment"></span>    <span class="keywordtype">bool</span> <a class="code" href="classQuantBRLMM_1_1BrlmmParam.html#aa9ebe7bbc8cb7de369bf2df9e795690c" title="Summary values fed into R are usually truncated, should we do /// that to be compatible? (used for regression testss).">m_LowPrecision</a>;             <span class="comment">///&lt; Summary values fed into R are usually truncated, should we do</span>
<a name="l00128"></a>00128 <span class="comment"></span><span class="comment">                                     /// that to be compatible? (used for regression testss)</span>
<a name="l00129"></a>00129 <span class="comment"></span>  };
<a name="l00130"></a>00130 
<a name="l00131"></a>00131 
<a name="l00132"></a>00132 <span class="comment"></span>
<a name="l00133"></a>00133 <span class="comment">  /** Constructor, currently creates prior estimates from pre-calculated data from R. */</span>
<a name="l00134"></a>00134   <a class="code" href="classQuantBRLMM.html#a703d4e7e27b3cb69e48518618b7c650a" title="Constructor, currently creates prior estimates from pre-calculated data from R.">QuantBRLMM</a>(<span class="keywordtype">double</span> hetMultiplier, <span class="keywordtype">int</span> iterations, <span class="keywordtype">double</span> iterationThreshold,
<a name="l00135"></a>00135              <span class="keywordtype">double</span> K, <span class="keyword">enum</span> <a class="code" href="classQuantBRLMM.html#aa41baee052c062022a0ab6a999c21638" title="What transformation of intensity estimates to feed into classifier.">Transformation</a> transform,
<a name="l00136"></a>00136              <span class="keywordtype">int</span> priorWeight, <span class="keywordtype">int</span> priorMinCall, <span class="keywordtype">bool</span> lowPrecision);
<a name="l00137"></a>00137 <span class="comment"></span>
<a name="l00138"></a>00138 <span class="comment">  /** Fill in our self documentation from current state. */</span>
<a name="l00139"></a>00139   <span class="keyword">static</span> <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#a53c159ab51bc8a716b63001410146e10" title="Fill in our self documentation from current state.">setupSelfDoc</a>(<a class="code" href="classSelfDoc.html" title="Small interface to for algorithmic classes that can explain themselves.">SelfDoc</a> &amp;doc);
<a name="l00140"></a>00140 <span class="comment"></span>
<a name="l00141"></a>00141 <span class="comment">  /** Destructor. */</span>
<a name="l00142"></a>00142   <a class="code" href="classQuantBRLMM.html#ad9a33be6eb00a6823576eac29302f5d1" title="Destructor.">~QuantBRLMM</a>();
<a name="l00143"></a>00143 <span class="comment"></span>
<a name="l00144"></a>00144 <span class="comment">  /**</span>
<a name="l00145"></a>00145 <span class="comment">   * Tell brlmm to output snp models to file.</span>
<a name="l00146"></a>00146 <span class="comment">   *</span>
<a name="l00147"></a>00147 <span class="comment">   * @param fileName - relative or full path of file to write models to.</span>
<a name="l00148"></a>00148 <span class="comment">   */</span>
<a name="l00149"></a>00149   <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#ad5e489c5edabd5fe6c0f7d1e34f777be" title="Tell brlmm to output snp models to file.">setSnpModelOut</a>(<span class="keyword">const</span> std::string&amp; fileName,affx::TsvReport::TsvReportFmt_t format);
<a name="l00150"></a>00150 <span class="comment"></span>
<a name="l00151"></a>00151 <span class="comment">  /**</span>
<a name="l00152"></a>00152 <span class="comment">   * Covert string representations into ClusterPrior</span>
<a name="l00153"></a>00153 <span class="comment">   *</span>
<a name="l00154"></a>00154 <span class="comment">   * @param id - identifying string.</span>
<a name="l00155"></a>00155 <span class="comment">   * @param center - centers.</span>
<a name="l00156"></a>00156 <span class="comment">   * @param var - variance of center variances.</span>
<a name="l00157"></a>00157 <span class="comment">   * @param covar - covariance of centers.</span>
<a name="l00158"></a>00158 <span class="comment">   *</span>
<a name="l00159"></a>00159 <span class="comment">   * @return - ClusterPrior.</span>
<a name="l00160"></a>00160 <span class="comment">   */</span>
<a name="l00161"></a>00161   <span class="keyword">static</span> <a class="code" href="classClusterPrior.html">ClusterPrior</a> <a class="code" href="classQuantBRLMM.html#a7b90f77e1f46e0f348ae56a83f03236d" title="Covert string representations into ClusterPrior.">clusterPriorFromStrings</a>(<span class="keyword">const</span> std::string&amp; fileName, std::string &amp;<span class="keywordtype">id</span>, std::string &amp;center,
<a name="l00162"></a>00162                                               std::string &amp;var, std::string &amp;covar);
<a name="l00163"></a>00163 <span class="comment"></span>
<a name="l00164"></a>00164 <span class="comment">  /**</span>
<a name="l00165"></a>00165 <span class="comment">   * Write out a text representation of the prior to the file specified.</span>
<a name="l00166"></a>00166 <span class="comment">   * @param prior - Prior to be written.</span>
<a name="l00167"></a>00167 <span class="comment">   * @param out - ostream to write to.</span>
<a name="l00168"></a>00168 <span class="comment">   */</span>
<a name="l00169"></a>00169   <span class="comment">//  static void writePriorOut(ClusterPrior &amp;prior,affx::TsvFile&amp; tsv,const );</span>
<a name="l00170"></a>00170   <span class="keyword">static</span> <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#abf28bce895fc18fbd73d5744262486d7" title="Write out a text representation of the prior to the file specified.">writePriorOut</a>(<a class="code" href="classClusterPrior.html">ClusterPrior</a> &amp;prior, std::ostream &amp;out,
<a name="l00171"></a>00171                             <span class="keyword">const</span> std::string&amp; fieldDelim, <span class="keyword">const</span> std::string&amp; sep);
<a name="l00172"></a>00172 <span class="comment"></span>
<a name="l00173"></a>00173 <span class="comment">  /**</span>
<a name="l00174"></a>00174 <span class="comment">   * Write out a text representation of the model to the file specified.</span>
<a name="l00175"></a>00175 <span class="comment">   * @param model - Model to be written.</span>
<a name="l00176"></a>00176 <span class="comment">   * @param out - ostream to write to.</span>
<a name="l00177"></a>00177 <span class="comment">   */</span>
<a name="l00178"></a>00178   <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#a008b914483156a796f217fae336ea644" title="Write out a text representation of the model to the file specified.">writeModelOut</a>(<a class="code" href="classClusterModel.html">ClusterModel</a> &amp;model,<a class="code" href="classaffx_1_1TsvReport.html" title="Unifies the presentation of file/TsvFile and file ///.">affx::TsvReport</a>&amp; tsv, <span class="keyword">const</span> std::string&amp; name,<span class="keywordtype">int</span> copies);
<a name="l00179"></a>00179 <span class="comment"></span>
<a name="l00180"></a>00180 <span class="comment">  /**</span>
<a name="l00181"></a>00181 <span class="comment">   * Write out a text representation of the prior to the file specified.</span>
<a name="l00182"></a>00182 <span class="comment">   * @param prior</span>
<a name="l00183"></a>00183 <span class="comment">   * @param fileName</span>
<a name="l00184"></a>00184 <span class="comment">   */</span>
<a name="l00185"></a>00185   <span class="keyword">static</span> <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#a1a7e2033f6184a97ecef944e46c0aed5" title="Write out a text representation of the prior to the file specified.">writePrior</a>(<a class="code" href="classClusterPrior.html">ClusterPrior</a> &amp;prior, <span class="keyword">const</span> std::string&amp; fileName);
<a name="l00186"></a>00186 <span class="comment"></span>
<a name="l00187"></a>00187 <span class="comment">  /**</span>
<a name="l00188"></a>00188 <span class="comment">   * Covert string representations into ClusterModel</span>
<a name="l00189"></a>00189 <span class="comment">   *</span>
<a name="l00190"></a>00190 <span class="comment">   * @param id - identifying string.</span>
<a name="l00191"></a>00191 <span class="comment">   * @param center - centers.</span>
<a name="l00192"></a>00192 <span class="comment">   * @param var - variance of center variances.</span>
<a name="l00193"></a>00193 <span class="comment">   *</span>
<a name="l00194"></a>00194 <span class="comment">   * @return - ClusterModel</span>
<a name="l00195"></a>00195 <span class="comment">   */</span>
<a name="l00196"></a>00196   <a class="code" href="classClusterModel.html">ClusterModel</a> <a class="code" href="classQuantBRLMM.html#a38aebf72436d249fef32c09cde6ed027" title="Covert string representations into ClusterModel.">clusterModelFromStrings</a>(<span class="keyword">const</span> std::string&amp; fileName, std::string &amp;<span class="keywordtype">id</span>,
<a name="l00197"></a>00197                                        std::string &amp;center, std::string &amp;var);
<a name="l00198"></a>00198 <span class="comment"></span>
<a name="l00199"></a>00199 <span class="comment">  /**</span>
<a name="l00200"></a>00200 <span class="comment">   * Read the prior from the test file specified.</span>
<a name="l00201"></a>00201 <span class="comment">   * @param fileName - Name of file to read from.</span>
<a name="l00202"></a>00202 <span class="comment">   * @return - The prior read in from file.</span>
<a name="l00203"></a>00203 <span class="comment">   */</span>
<a name="l00204"></a>00204   <span class="keyword">static</span> <a class="code" href="classClusterPrior.html">ClusterPrior</a> <a class="code" href="classQuantBRLMM.html#a71f90338aa453da6b7c453a4fe44c1ab" title="Read the prior from the test file specified.">loadPrior</a>(<span class="keyword">const</span> std::string&amp; fileName);
<a name="l00205"></a>00205 <span class="comment"></span>
<a name="l00206"></a>00206 <span class="comment">  /**</span>
<a name="l00207"></a>00207 <span class="comment">   * Generate a string representation of a snp model.</span>
<a name="l00208"></a>00208 <span class="comment">   * @return string - text description of snp clusters.</span>
<a name="l00209"></a>00209 <span class="comment">   */</span>
<a name="l00210"></a>00210   std::string <a class="code" href="classQuantBRLMM.html#a29ae28aefa68a869921c8f3a42c39110" title="Generate a string representation of a snp model.">getModelString</a>();
<a name="l00211"></a>00211 <span class="comment"></span>
<a name="l00212"></a>00212 <span class="comment">  /**</span>
<a name="l00213"></a>00213 <span class="comment">   * Load in a number of snp specific models.</span>
<a name="l00214"></a>00214 <span class="comment">   * @param snpModels - map of snp probeset names to models.</span>
<a name="l00215"></a>00215 <span class="comment">   * @param fileName - filename to read models from.</span>
<a name="l00216"></a>00216 <span class="comment">   */</span>
<a name="l00217"></a>00217   <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#a5f6db5e8a6818003a7e4629e555366a3" title="Load in a number of snp specific models.">loadSnpModels</a>(std::map&lt;std::string, ClusterModel&gt; &amp;snpModels,
<a name="l00218"></a>00218                      <span class="keyword">const</span> std::string&amp; fileName);
<a name="l00219"></a>00219 <span class="comment"></span>
<a name="l00220"></a>00220 <span class="comment">  /**</span>
<a name="l00221"></a>00221 <span class="comment">   * Load up the snp models from a text file.</span>
<a name="l00222"></a>00222 <span class="comment">   * @param fileName - text file to read models from.</span>
<a name="l00223"></a>00223 <span class="comment">   */</span>
<a name="l00224"></a>00224   <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#a873e8c5e7d97ea648f749d1c410706f0" title="Load up the snp models from a text file.">readSnpModels</a>(<span class="keyword">const</span> std::string&amp; fileName);
<a name="l00225"></a>00225 <span class="comment"></span>
<a name="l00226"></a>00226 <span class="comment">  /**</span>
<a name="l00227"></a>00227 <span class="comment">   * Load in a number of snp specific priors.</span>
<a name="l00228"></a>00228 <span class="comment">   * @param snpPriors - map of snp probeset names to priors.</span>
<a name="l00229"></a>00229 <span class="comment">   * @param fileName - filename to read priors from.</span>
<a name="l00230"></a>00230 <span class="comment">   */</span>
<a name="l00231"></a>00231   <span class="keyword">static</span> <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#a90060a28781e3d3d92bba20fba16c81f" title="Load in a number of snp specific priors.">loadSnpPriors</a>(std::map&lt;std::string, ClusterPrior&gt; &amp;snpPriors, <span class="keyword">const</span> std::string&amp; fileName);
<a name="l00232"></a>00232 <span class="comment"></span>
<a name="l00233"></a>00233 <span class="comment">  /**</span>
<a name="l00234"></a>00234 <span class="comment">   * @brief Set up the quantification method given all the data about the probe</span>
<a name="l00235"></a>00235 <span class="comment">   * set, chip layout and data.</span>
<a name="l00236"></a>00236 <span class="comment">   *</span>
<a name="l00237"></a>00237 <span class="comment">   * @param psGroup - Probes to be used for final estimate.</span>
<a name="l00238"></a>00238 <span class="comment">   * @param layout - Chip layout annotation.</span>
<a name="l00239"></a>00239 <span class="comment">   * @param iMart - Raw data from chips.</span>
<a name="l00240"></a>00240 <span class="comment">   * @param iTrans - Transformations to be applied to data before use.</span>
<a name="l00241"></a>00241 <span class="comment">   * @param pmAdjust - How to estimate background, or MM probe.</span>
<a name="l00242"></a>00242 <span class="comment">   * @return True if setup sucessful, false otherwise.</span>
<a name="l00243"></a>00243 <span class="comment">   */</span>
<a name="l00244"></a>00244   <span class="keywordtype">bool</span> <a class="code" href="classQuantBRLMM.html#a98c9763caf138fe3a960c0702e6dc723" title="Set up the quantification method given all the data about the probe set, chip layout and data...">setUp</a>(<a class="code" href="classProbeSetGroup.html" title="Group of probe sets that should be processed as a single large probe set.">ProbeSetGroup</a> &amp;psGroup, <span class="keyword">const</span> <a class="code" href="classIntensityMart.html" title="IntensityMart.">IntensityMart</a> &amp;iMart, 
<a name="l00245"></a>00245              std::vector&lt;ChipStream *&gt; &amp;iTrans, <a class="code" href="classPmAdjuster.html" title="Interface for determining a change based on intensity of perfect match.">PmAdjuster</a> &amp;pmAdjust);
<a name="l00246"></a>00246 
<a name="l00247"></a>00247   <span class="keyword">static</span> <span class="keywordtype">void</span> transformData(std::vector&lt;double&gt; &amp;aVals,
<a name="l00248"></a>00248                             std::vector&lt;double&gt; &amp;bVals,
<a name="l00249"></a>00249                             <a class="code" href="classQuantBRLMM_1_1BrlmmParam.html">QuantBRLMM::BrlmmParam</a> &amp;param);
<a name="l00250"></a>00250 
<a name="l00251"></a>00251   <a class="code" href="classClusterPrior.html">ClusterPrior</a> *priorForSnp();
<a name="l00252"></a>00252 <span class="comment"></span>
<a name="l00253"></a>00253 <span class="comment">  /**</span>
<a name="l00254"></a>00254 <span class="comment">   * @brief Do the heavy lifting of estimation.</span>
<a name="l00255"></a>00255 <span class="comment">   */</span>
<a name="l00256"></a>00256   <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#a14f9cb0971989fc0ee0d37f5242bdc2d" title="Do the heavy lifting of estimation.">computeEstimate</a>();
<a name="l00257"></a>00257 <span class="comment"></span>
<a name="l00258"></a>00258 <span class="comment">  /**</span>
<a name="l00259"></a>00259 <span class="comment">   * Remove the het cluster from a prior by setting center to be very</span>
<a name="l00260"></a>00260 <span class="comment">   * large negative number and removing covariance with other</span>
<a name="l00261"></a>00261 <span class="comment">   * clusters.</span>
<a name="l00262"></a>00262 <span class="comment">   * @param prior - Prior to remove het cluster from.</span>
<a name="l00263"></a>00263 <span class="comment">   * @param trans - transformation used for clustering.</span>
<a name="l00264"></a>00264 <span class="comment">   */</span>
<a name="l00265"></a>00265   <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#a761e1e30d718599d1ce3aad6853b8ffb" title="Remove the het cluster from a prior by setting center to be very large negative number and removing c...">removeHetFromPrior</a>(<a class="code" href="classClusterPrior.html">ClusterPrior</a> &amp;prior, <span class="keyword">enum</span> <a class="code" href="classQuantBRLMM.html#aa41baee052c062022a0ab6a999c21638" title="What transformation of intensity estimates to feed into classifier.">Transformation</a> trans);
<a name="l00266"></a>00266 <span class="comment"></span>
<a name="l00267"></a>00267 <span class="comment">  /**</span>
<a name="l00268"></a>00268 <span class="comment">   * This is the fix for the snps on chromosome X that don&#39;t appear</span>
<a name="l00269"></a>00269 <span class="comment">   * diploid in men (i.e. aren&#39;t in the pseudoautosomal region). The</span>
<a name="l00270"></a>00270 <span class="comment">   * concept is to fit and call the model separately for men and</span>
<a name="l00271"></a>00271 <span class="comment">   * women. Implementation is to set all the male initial calls to NN</span>
<a name="l00272"></a>00272 <span class="comment">   * when fitting for women. For fitting and calling men the female</span>
<a name="l00273"></a>00273 <span class="comment">   * samples are set to NN and further the prior is ajusted such that</span>
<a name="l00274"></a>00274 <span class="comment">   * het calls are impossible to get.</span>
<a name="l00275"></a>00275 <span class="comment">   */</span>
<a name="l00276"></a>00276   <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#a448772f9128f2987c4d8f4e3656961b2" title="This is the fix for the snps on chromosome X that don&amp;#39;t appear diploid in men (i...">callHaploidSnp</a>(<a class="code" href="classClusterPrior.html">ClusterPrior</a> &amp;prior);
<a name="l00277"></a>00277 
<a name="l00278"></a>00278   <span class="keyword">static</span> <span class="keywordtype">void</span> valueForTransformation(<span class="keywordtype">double</span> aVal, <span class="keywordtype">double</span> bVal, <span class="keyword">enum</span> <a class="code" href="classQuantBRLMM.html#aa41baee052c062022a0ab6a999c21638" title="What transformation of intensity estimates to feed into classifier.">QuantBRLMM::Transformation</a> trans,
<a name="l00279"></a>00279                                      <span class="keywordtype">double</span> K, <span class="keywordtype">double</span> &amp;xVal, <span class="keywordtype">double</span> &amp;yVal);
<a name="l00280"></a>00280 <span class="comment"></span>
<a name="l00281"></a>00281 <span class="comment">  /**</span>
<a name="l00282"></a>00282 <span class="comment">   * Transform the data in aValues and bValues using the method specified by trans.</span>
<a name="l00283"></a>00283 <span class="comment">   *</span>
<a name="l00284"></a>00284 <span class="comment">   * @param aValues - Summary values for A allele.</span>
<a name="l00285"></a>00285 <span class="comment">   * @param bValues - Summary values for B allele.</span>
<a name="l00286"></a>00286 <span class="comment">   * @param genoTypes - Genotypes for each sample.</span>
<a name="l00287"></a>00287 <span class="comment">   * @param Transformation - Type of transformation to use.</span>
<a name="l00288"></a>00288 <span class="comment">   * @param K - Scale parameter used in CCS and CES transformations.</span>
<a name="l00289"></a>00289 <span class="comment">   * @param xDim - two dimensional matrix for x dimension with rows</span>
<a name="l00290"></a>00290 <span class="comment">   * (index 0) being genotype (0,1,2) and columns being values observed.</span>
<a name="l00291"></a>00291 <span class="comment">   * @param yDim - two dimensional matrix for y dimension with rows</span>
<a name="l00292"></a>00292 <span class="comment">   * (index 0) being genotype (0,1,2) and columns being values observed.</span>
<a name="l00293"></a>00293 <span class="comment">   */</span>
<a name="l00294"></a>00294   <span class="keyword">static</span> <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#aae609c657d28f62ae7b923c73a93fc46" title="Transform the data in aValues and bValues using the method specified by trans.">fillInTransformedData</a>(std::vector&lt;double&gt; &amp;aValues, std::vector&lt;double&gt; &amp;bValues,
<a name="l00295"></a>00295                                     std::vector&lt;affx::GType&gt; &amp;genoTypes, <span class="keyword">enum</span> <a class="code" href="classQuantBRLMM.html#aa41baee052c062022a0ab6a999c21638" title="What transformation of intensity estimates to feed into classifier.">Transformation</a> trans,
<a name="l00296"></a>00296                                     <span class="keywordtype">double</span> K,
<a name="l00297"></a>00297                                     std::vector&lt;std::vector&lt;double&gt; &gt; &amp;xDim, std::vector&lt;std::vector&lt;double&gt; &gt; &amp;yDim);<span class="comment"></span>
<a name="l00298"></a>00298 <span class="comment">  /**</span>
<a name="l00299"></a>00299 <span class="comment">   * Specify the prior that will be used for predictions.</span>
<a name="l00300"></a>00300 <span class="comment">   *</span>
<a name="l00301"></a>00301 <span class="comment">   * @param prior - Prior data to use.</span>
<a name="l00302"></a>00302 <span class="comment">   */</span>
<a name="l00303"></a>00303   <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#ad6cb1cdd30699b802c7d81248da58da0" title="Specify the prior that will be used for predictions.">setPrior</a>(<span class="keyword">const</span> <a class="code" href="classClusterPrior.html">ClusterPrior</a> &amp;prior) ;
<a name="l00304"></a>00304 <span class="comment"></span>
<a name="l00305"></a>00305 <span class="comment">  /**</span>
<a name="l00306"></a>00306 <span class="comment">   * @brief What is the name of the quantification method?</span>
<a name="l00307"></a>00307 <span class="comment">   * @return name of adjuster.</span>
<a name="l00308"></a>00308 <span class="comment">   */</span>
<a name="l00309"></a>00309   std::string <a class="code" href="classQuantBRLMM.html#ac61886478a0d57b3c464015fa0ebc196" title="What is the name of the quantification method?">getType</a>();
<a name="l00310"></a>00310 
<a name="l00311"></a>00311 <span class="comment"></span>
<a name="l00312"></a>00312 <span class="comment">  /** Conversion utility for transforming vectors to column vectors. */</span>
<a name="l00313"></a>00313   <span class="keyword">static</span> ColumnVector <a class="code" href="classQuantBRLMM.html#a5205c4974d362c640198f75c0547982d" title="Conversion utility for transforming vectors to column vectors.">columnVectorFromVector</a>(std::vector&lt;double&gt; &amp;v);
<a name="l00314"></a>00314 <span class="comment"></span>
<a name="l00315"></a>00315 <span class="comment">  /** Conversion utility for transforming arrays to column vectors. */</span>
<a name="l00316"></a>00316   <span class="keyword">static</span> ColumnVector <a class="code" href="classQuantBRLMM.html#a1ad81844f338fefacfeee8e12677363a" title="Conversion utility for transforming arrays to column vectors.">columnVectorFromArray</a>(<span class="keywordtype">double</span> *array, <span class="keywordtype">int</span> count);
<a name="l00317"></a>00317 <span class="comment"></span>
<a name="l00318"></a>00318 <span class="comment">  /** Conversion utility for transforming arrays to matrices. */</span>
<a name="l00319"></a>00319   <span class="keyword">static</span> Matrix <a class="code" href="classQuantBRLMM.html#a1cb1a160c110e08d99999bf00d313d94" title="Conversion utility for transforming arrays to matrices.">matrixFromArray</a>(<span class="keywordtype">double</span> *array, <span class="keywordtype">int</span> count, <span class="keywordtype">int</span> nRow, <span class="keywordtype">int</span> nCol);
<a name="l00320"></a>00320 <span class="comment"></span>
<a name="l00321"></a>00321 <span class="comment">  /**</span>
<a name="l00322"></a>00322 <span class="comment">   * What version of the BRLMM algorithm are we implementing?</span>
<a name="l00323"></a>00323 <span class="comment">   * @return version string.</span>
<a name="l00324"></a>00324 <span class="comment">   */</span>
<a name="l00325"></a>00325   std::string <a class="code" href="classQuantBRLMM.html#a2110ed6b324c0399cc2467ea5da113a4" title="What version of the BRLMM algorithm are we implementing?">getVersion</a>();
<a name="l00326"></a>00326 <span class="comment"></span>
<a name="l00327"></a>00327 <span class="comment">  /** Make banded count matrix. */</span>
<a name="l00328"></a>00328   <span class="keyword">static</span> Matrix <a class="code" href="classQuantBRLMM.html#a76dda9af539b10df3326cf43c113b147" title="Make banded count matrix.">makeCountMat</a>(std::vector&lt;int&gt; &amp;counts);
<a name="l00329"></a>00329 <span class="comment"></span>
<a name="l00330"></a>00330 <span class="comment">  /** Reformat variance/covariance vector as a banded matrix. */</span>
<a name="l00331"></a>00331   <span class="keyword">static</span> Matrix <a class="code" href="classQuantBRLMM.html#a6a5480a106596136e420508b49b3b370" title="Reformat variance/covariance vector as a banded matrix.">formatAsBigVarMat</a>(ColumnVector v);
<a name="l00332"></a>00332 <span class="comment"></span>
<a name="l00333"></a>00333 <span class="comment">  /** Bayesian update of the mean vector, this is where the voodoo</span>
<a name="l00334"></a>00334 <span class="comment">      happens. Have to query Simon more about the motivation. */</span>
<a name="l00335"></a>00335   <span class="keyword">static</span> Matrix <a class="code" href="classQuantBRLMM.html#acf88e5de22303b2cfc1a929433875f14" title="Bayesian update of the mean vector, this is where the voodoo happens.">bayesianMeanUpdate</a>(ColumnVector &amp;newVar, std::vector&lt;int&gt; &amp;counts, ColumnVector &amp;means,
<a name="l00336"></a>00336                                    Matrix &amp;priorCenterVar, ColumnVector &amp;priorCenter);
<a name="l00337"></a>00337 <span class="comment"></span>
<a name="l00338"></a>00338 <span class="comment">  /// @todo should degFree be weight or counts rather than degrees of freedom?</span>
<a name="l00339"></a>00339 <span class="comment"></span>  <span class="keyword">static</span> ColumnVector <a class="code" href="classQuantBRLMM.html#a7ea1bd4cc10263c529ca483800fb3e09">shrinkVariance</a>(<a class="code" href="classClusterStats.html">ClusterStats</a> &amp;empirical, <a class="code" href="classClusterPrior.html">ClusterPrior</a> &amp;prior, <span class="keywordtype">int</span> degFree);
<a name="l00340"></a>00340 <span class="comment"></span>
<a name="l00341"></a>00341 <span class="comment">  /** Wrapper function for getting bayesian adjusted parameters given data. */</span>
<a name="l00342"></a>00342   <span class="keyword">static</span> <a class="code" href="classClusterModel.html">ClusterModel</a> <a class="code" href="classQuantBRLMM.html#af1acb3c6affa478feb02b2443d7d4962" title="Wrapper function for getting bayesian adjusted parameters given data.">fitModel</a>(std::vector&lt;double&gt; &amp;aValues, std::vector&lt;double&gt; &amp;bValues,
<a name="l00343"></a>00343                                std::vector&lt;std::vector&lt;double&gt; &gt; &amp;genoDist,
<a name="l00344"></a>00344                                std::vector&lt;affx::GType&gt; &amp;genoTypes, <a class="code" href="classClusterPrior.html">ClusterPrior</a> &amp;prior, <span class="keyword">const</span> std::string &amp;name,
<a name="l00345"></a>00345                                <a class="code" href="classQuantBRLMM_1_1BrlmmParam.html">BrlmmParam</a> &amp;param);
<a name="l00346"></a>00346 <span class="comment"></span>
<a name="l00347"></a>00347 <span class="comment">  /** Count up some elementary statistics about a particular snp over</span>
<a name="l00348"></a>00348 <span class="comment">      multiple experiments. */</span>
<a name="l00349"></a>00349   <span class="keyword">static</span> <a class="code" href="classClusterStats.html">ClusterStats</a> <a class="code" href="classQuantBRLMM.html#ad80f38b0695ffc08aef491b000deacb3" title="Count up some elementary statistics about a particular snp over multiple experiments.">computeClusterStats</a>(std::vector&lt;double&gt; &amp;aValues, std::vector&lt;double&gt; &amp;bValues,
<a name="l00350"></a>00350                                           std::vector&lt;affx::GType&gt; &amp;genoTypes, <span class="keyword">enum</span> <a class="code" href="classQuantBRLMM.html#aa41baee052c062022a0ab6a999c21638" title="What transformation of intensity estimates to feed into classifier.">Transformation</a> trans,
<a name="l00351"></a>00351                                           <span class="keywordtype">double</span> K);<span class="comment"></span>
<a name="l00352"></a>00352 <span class="comment">  /**</span>
<a name="l00353"></a>00353 <span class="comment">   * Compute the prior from a collection of snps and fill in the prior.</span>
<a name="l00354"></a>00354 <span class="comment">   *</span>
<a name="l00355"></a>00355 <span class="comment">   * @param snpStats - Stats for a relatively large number of snps</span>
<a name="l00356"></a>00356 <span class="comment">   * @param prior - Fill in the prior.</span>
<a name="l00357"></a>00357 <span class="comment">   */</span>
<a name="l00358"></a>00358   <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#afec1a0f13ad6d4586fae2c084d2e2e33" title="Compute the prior from a collection of snps and fill in the prior.">computeClusterPrior</a>(<span class="keyword">const</span> std::vector&lt;ClusterStats&gt; &amp;snpStats, <a class="code" href="classClusterPrior.html">ClusterPrior</a> &amp;prior);
<a name="l00359"></a>00359 <span class="comment"></span>
<a name="l00360"></a>00360 <span class="comment">  /** </span>
<a name="l00361"></a>00361 <span class="comment">   * Construct a prior from a collection of probesets. Note that to be part of</span>
<a name="l00362"></a>00362 <span class="comment">   * the prior a SNP must have at least 2 examples per AA, AB, and BB otherwise</span>
<a name="l00363"></a>00363 <span class="comment">   * the variance will be zero.</span>
<a name="l00364"></a>00364 <span class="comment">   * </span>
<a name="l00365"></a>00365 <span class="comment">   * @param probeSets - ProbeSets to be used in computing prior.</span>
<a name="l00366"></a>00366 <span class="comment">   * @param layout - description and chip and probesets.</span>
<a name="l00367"></a>00367 <span class="comment">   * @param iMart - Cel file data for probesets.</span>
<a name="l00368"></a>00368 <span class="comment">   * @param iTrans - Transformers to operate on raw data (i.e. normalization and bg subtraction).</span>
<a name="l00369"></a>00369 <span class="comment">   * @param pmAdjust - Individual probe adjustment like pm-mm or pm-only.</span>
<a name="l00370"></a>00370 <span class="comment">   * </span>
<a name="l00371"></a>00371 <span class="comment">   * @return - Prior constructed from summary of all the probesets.</span>
<a name="l00372"></a>00372 <span class="comment">   */</span>
<a name="l00373"></a>00373   <a class="code" href="classClusterPrior.html">ClusterPrior</a> <a class="code" href="classQuantBRLMM.html#afafae26ed3cd67f3e51ca8fd77cc7669" title="Construct a prior from a collection of probesets.">makePrior</a>(       std::vector&lt;string&gt; &amp;probeSetNames, 
<a name="l00374"></a>00374                                 <a class="code" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a> &amp;layout,
<a name="l00375"></a>00375                                 <a class="code" href="classIntensityMart.html" title="IntensityMart.">IntensityMart</a> &amp;iMart, 
<a name="l00376"></a>00376                                 std::vector&lt;ChipStream *&gt; &amp;iTrans,
<a name="l00377"></a>00377                                 <a class="code" href="classPmAdjuster.html" title="Interface for determining a change based on intensity of perfect match.">PmAdjuster</a> &amp;pmAdjust);
<a name="l00378"></a>00378 
<a name="l00379"></a>00379   <span class="comment">/* Make the inverse matrix from the variance and covariances. */</span>
<a name="l00380"></a>00380   <span class="keyword">static</span> Matrix makeVarInv(Matrix &amp;clusterVars, <span class="keywordtype">int</span> index);
<a name="l00381"></a>00381 
<a name="l00382"></a>00382   <span class="comment">/* Take the model derived in fitModel and use it to do predictions. */</span>
<a name="l00383"></a>00383   <span class="keyword">static</span> <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#ad0808c8bfeef3f05b4733ea08a91cd0d">brlmmCall</a>(        <span class="keyword">const</span> std::vector&lt;double&gt; &amp;aValues, 
<a name="l00384"></a>00384                                 <span class="keyword">const</span> std::vector&lt;double&gt; &amp;bValues,
<a name="l00385"></a>00385                                 <a class="code" href="classClusterModel.html">ClusterModel</a> &amp;model, std::vector&lt;affx::GType&gt; &amp;calls,
<a name="l00386"></a>00386                                 std::vector&lt;double&gt; &amp;confidences, 
<a name="l00387"></a>00387                                 std::vector&lt;std::vector&lt;double&gt; &gt; &amp;genoDist,  
<a name="l00388"></a>00388                                 <a class="code" href="classQuantBRLMM_1_1BrlmmParam.html">BrlmmParam</a> &amp;param);
<a name="l00389"></a>00389 
<a name="l00390"></a>00390   <span class="comment">/* Set the quantification method to use for summarizing individual alleles */</span>
<a name="l00391"></a>00391   <span class="keywordtype">void</span> setQuantExprMethod(<a class="code" href="classQuantExprMethod.html" title="QuantExprMethod - Interface for computing quantification summaries from PM intensities grouped into p...">QuantExprMethod</a> *qMethod);
<a name="l00392"></a>00392 
<a name="l00393"></a>00393   <span class="comment">/* Get the quantification method to use for summarizing individual alleles */</span>
<a name="l00394"></a>00394   <a class="code" href="classQuantExprMethod.html" title="QuantExprMethod - Interface for computing quantification summaries from PM intensities grouped into p...">QuantExprMethod</a> *getQuantExprMethod();
<a name="l00395"></a>00395 
<a name="l00396"></a>00396 <span class="comment"></span>
<a name="l00397"></a>00397 <span class="comment">  /** </span>
<a name="l00398"></a>00398 <span class="comment">   * Set the initial guesses of genotypes, usually from DM predictions.</span>
<a name="l00399"></a>00399 <span class="comment">   * </span>
<a name="l00400"></a>00400 <span class="comment">   * @param genotypes vector of genotypes AA=0, AB=1, AB=2, NN=3,</span>
<a name="l00401"></a>00401 <span class="comment">   */</span>
<a name="l00402"></a>00402   <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#a72de5e66b4d6f1d6fc63426846d20950" title="Set the initial guesses of genotypes, usually from DM predictions.">setKnownGenoTypes</a>(std::map&lt;<span class="keyword">const</span> <span class="keywordtype">char</span> *,std::vector&lt;affx::GType&gt;, <a class="code" href="structUtil_1_1ltstr.html" title="Comparison object for use in map for char *.">Util::ltstr</a>&gt; &amp;genotypes);
<a name="l00403"></a>00403 <span class="comment"></span>
<a name="l00404"></a>00404 <span class="comment">  /**</span>
<a name="l00405"></a>00405 <span class="comment">   * @brief Default Getter method for parameters and their documentation.</span>
<a name="l00406"></a>00406 <span class="comment">   * @return map of parameters and their descriptions.</span>
<a name="l00407"></a>00407 <span class="comment">   */</span>
<a name="l00408"></a>00408   <span class="keyword">static</span> std::vector&lt;SelfDoc::Opt&gt; <a class="code" href="classQuantBRLMM.html#af4303a14cab2be76711893a4082e8c19" title="Default Getter method for parameters and their documentation.">getDefaultDocOptions</a>();
<a name="l00409"></a>00409 <span class="comment"></span>
<a name="l00410"></a>00410 <span class="comment">  /**</span>
<a name="l00411"></a>00411 <span class="comment">   * @brief Supply a little how/what/why about the algorithms this</span>
<a name="l00412"></a>00412 <span class="comment">   * class performs and what parameters it takes.</span>
<a name="l00413"></a>00413 <span class="comment">   * @return SelfDoc</span>
<a name="l00414"></a>00414 <span class="comment">   */</span>
<a name="l00415"></a>00415   <span class="keyword">static</span> <a class="code" href="classSelfDoc.html" title="Small interface to for algorithmic classes that can explain themselves.">SelfDoc</a> <a class="code" href="classQuantBRLMM.html#a3632279e1824737d1d37242d2d25191e" title="Supply a little how/what/why about the algorithms this class performs and what parameters it takes...">explainSelf</a>();
<a name="l00416"></a>00416 <span class="comment"></span>
<a name="l00417"></a>00417 <span class="comment">  /**</span>
<a name="l00418"></a>00418 <span class="comment">   * Convert a (text insensitive) text repesentation of a transformation into</span>
<a name="l00419"></a>00419 <span class="comment">   * the enumration.</span>
<a name="l00420"></a>00420 <span class="comment">   * @param s - string specifying transformation.</span>
<a name="l00421"></a>00421 <span class="comment">   * @return - Enumeration version of transformation.</span>
<a name="l00422"></a>00422 <span class="comment">   */</span>
<a name="l00423"></a>00423   <span class="keyword">static</span> <span class="keyword">enum</span> <a class="code" href="classQuantBRLMM.html#aa41baee052c062022a0ab6a999c21638" title="What transformation of intensity estimates to feed into classifier.">Transformation</a> <a class="code" href="classQuantBRLMM.html#ac5880b4177534c9516f3d7243cb9aed7" title="Convert a (text insensitive) text repesentation of a transformation into the enumration.">transformationForString</a>(<span class="keyword">const</span> std::string &amp;s);
<a name="l00424"></a>00424 <span class="comment"></span>
<a name="l00425"></a>00425 <span class="comment">  /**</span>
<a name="l00426"></a>00426 <span class="comment">   * Convert a transformation enum to string representation</span>
<a name="l00427"></a>00427 <span class="comment">   * @param t - tranformation enum value.</span>
<a name="l00428"></a>00428 <span class="comment">   * @return - text description of enumeration.</span>
<a name="l00429"></a>00429 <span class="comment">   */</span>
<a name="l00430"></a>00430   <span class="keyword">static</span> std::string <a class="code" href="classQuantBRLMM.html#a03d17022504aa06d976b555fbd6e8ea7" title="Convert a transformation enum to string representation.">stringForTransformation</a>(<span class="keyword">enum</span> <a class="code" href="classQuantBRLMM.html#aa41baee052c062022a0ab6a999c21638" title="What transformation of intensity estimates to feed into classifier.">Transformation</a> t);
<a name="l00431"></a>00431 <span class="comment"></span>
<a name="l00432"></a>00432 <span class="comment">  /**</span>
<a name="l00433"></a>00433 <span class="comment">   * @brief This static function should be overridden by child classes</span>
<a name="l00434"></a>00434 <span class="comment">   * to return an object of the correct type initialized correctly</span>
<a name="l00435"></a>00435 <span class="comment">   * with the parameters in the string, string map. All objects</span>
<a name="l00436"></a>00436 <span class="comment">   * created this way should be deleted when finished using.</span>
<a name="l00437"></a>00437 <span class="comment">   *</span>
<a name="l00438"></a>00438 <span class="comment">   * @param param - Map of key/value pairs to initialize the object.</span>
<a name="l00439"></a>00439 <span class="comment">   *</span>
<a name="l00440"></a>00440 <span class="comment">   * @return Pointer toCreate object, this should be sub casted as necessary.</span>
<a name="l00441"></a>00441 <span class="comment">   */</span>
<a name="l00442"></a>00442   <span class="keyword">static</span> <a class="code" href="classSelfCreate.html" title="Small interface for functions that know how to make an instance of themselves given a map of key...">SelfCreate</a> *<a class="code" href="classQuantBRLMM.html#a26b9afc9130094403ebf4baff03a8a07" title="This static function should be overridden by child classes to return an object of the correct type in...">newObject</a>(std::map&lt;std::string,std::string&gt; &amp;param);
<a name="l00443"></a>00443 <span class="comment"></span>
<a name="l00444"></a>00444 <span class="comment">  /**</span>
<a name="l00445"></a>00445 <span class="comment">   * Clear out all of our results and values.</span>
<a name="l00446"></a>00446 <span class="comment">   */</span>
<a name="l00447"></a>00447   <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#a680eb2ad5f77c33e5f91a11d01b460cf" title="Clear out all of our results and values.">blankSelf</a>();
<a name="l00448"></a>00448 <span class="comment"></span>
<a name="l00449"></a>00449 <span class="comment">  /**</span>
<a name="l00450"></a>00450 <span class="comment">   * How many genotyping calls do we have? Also indicates how many</span>
<a name="l00451"></a>00451 <span class="comment">   * samples there are.</span>
<a name="l00452"></a>00452 <span class="comment">   *</span>
<a name="l00453"></a>00453 <span class="comment">   * @return Number of genotyping calls made.</span>
<a name="l00454"></a>00454 <span class="comment">   */</span>
<a name="l00455"></a>00455   <span class="keywordtype">size_t</span> <a class="code" href="classQuantBRLMM.html#ada9804ca9a3a11558d594813ad603e4b" title="How many genotyping calls do we have? Also indicates how many samples there are.">getNumCalls</a>();
<a name="l00456"></a>00456 <span class="comment"></span>
<a name="l00457"></a>00457 <span class="comment">  /**</span>
<a name="l00458"></a>00458 <span class="comment">   * Get our confidence value for a particular call in a particular sample.</span>
<a name="l00459"></a>00459 <span class="comment">   * @param index - sample of interest.</span>
<a name="l00460"></a>00460 <span class="comment">   * @return - Confidence in the predicted call.</span>
<a name="l00461"></a>00461 <span class="comment">   */</span>
<a name="l00462"></a>00462   <span class="keywordtype">double</span> <a class="code" href="classQuantBRLMM.html#a1b285d4ac1220d6a1e6b7dce90428b62" title="Get our confidence value for a particular call in a particular sample.">getConfidence</a>(<span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> index);
<a name="l00463"></a>00463 <span class="comment"></span>
<a name="l00464"></a>00464 <span class="comment">  /**</span>
<a name="l00465"></a>00465 <span class="comment">   * Get our confidence value for a particular call in a particular sample.</span>
<a name="l00466"></a>00466 <span class="comment">   * @param genoType -  Genotype of interest.</span>
<a name="l00467"></a>00467 <span class="comment">   * @param index - sample of interest.</span>
<a name="l00468"></a>00468 <span class="comment">   * @return - Confidence in the predicted call.</span>
<a name="l00469"></a>00469 <span class="comment">   */</span>
<a name="l00470"></a>00470   <span class="keywordtype">double</span> <a class="code" href="classQuantBRLMM.html#a8e5df36010d56b0782f0df8b4d6a1603" title="Get our confidence value for a particular call in a particular sample.">getConfGenotype</a>(affx::GType  genoType, <span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> index);
<a name="l00471"></a>00471 <span class="comment"></span>
<a name="l00472"></a>00472 <span class="comment">  /**</span>
<a name="l00473"></a>00473 <span class="comment">   * Get the summary value after transformation for the A and B alleles respectively.</span>
<a name="l00474"></a>00474 <span class="comment">   *</span>
<a name="l00475"></a>00475 <span class="comment">   * @param index - Which chip to get values for.</span>
<a name="l00476"></a>00476 <span class="comment">   * @param aValue - Filled in with transformed a value.</span>
<a name="l00477"></a>00477 <span class="comment">   * @param bValue - Filled in with transformed b value.</span>
<a name="l00478"></a>00478 <span class="comment">   */</span>
<a name="l00479"></a>00479   <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#ac0cd5a76fb8f17ab31f19ca35b3174f3" title="Get the summary value after transformation for the A and B alleles respectively.">getAlleleValues</a>(<span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> index, <span class="keywordtype">double</span> &amp;aValue, <span class="keywordtype">double</span> &amp;bValue);
<a name="l00480"></a>00480 <span class="comment"></span>
<a name="l00481"></a>00481 <span class="comment">  /**</span>
<a name="l00482"></a>00482 <span class="comment">   * Get the summary value names for the A and B alleles respectively.</span>
<a name="l00483"></a>00483 <span class="comment">   *</span>
<a name="l00484"></a>00484 <span class="comment">   * @param aName - Filled in with a value name.</span>
<a name="l00485"></a>00485 <span class="comment">   * @param bName - Filled in with b value name.</span>
<a name="l00486"></a>00486 <span class="comment">   */</span>
<a name="l00487"></a>00487   <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#a91dab0e7e2526025e0ade4628dd2c88b" title="Get the summary value names for the A and B alleles respectively.">getAlleleValueNames</a>(std::string &amp;aName, std::string &amp;bName);
<a name="l00488"></a>00488 <span class="comment"></span>
<a name="l00489"></a>00489 <span class="comment">  /**</span>
<a name="l00490"></a>00490 <span class="comment">   * Get the name of the probeset that these calls are being made for.</span>
<a name="l00491"></a>00491 <span class="comment">   * @return name of probeset.</span>
<a name="l00492"></a>00492 <span class="comment">   */</span>
<a name="l00493"></a>00493   <span class="keyword">const</span> std::string &amp;<a class="code" href="classQuantBRLMM.html#a35841cae2003df29b6968efb52e3a3a4" title="Get the name of the probeset that these calls are being made for.">getProbeSetName</a>();
<a name="l00494"></a>00494 <span class="comment"></span>
<a name="l00495"></a>00495 <span class="comment">  /**</span>
<a name="l00496"></a>00496 <span class="comment">   * Add an expression reporter to output the summary values for each</span>
<a name="l00497"></a>00497 <span class="comment">   * allele, residuals, etc.</span>
<a name="l00498"></a>00498 <span class="comment">   *</span>
<a name="l00499"></a>00499 <span class="comment">   * @param reporter - the object that will output all summary values.</span>
<a name="l00500"></a>00500 <span class="comment">   */</span>
<a name="l00501"></a>00501   <span class="keywordtype">void</span> <a class="code" href="classQuantBRLMM.html#a7ed6bc24b8df6012fb41b8be487acdc7" title="Add an expression reporter to output the summary values for each allele, residuals, etc.">addExprReporter</a>(<a class="code" href="classQuantMethodReport.html" title="Class for reporting results of quantification methods.">QuantMethodReport</a> *reporter);
<a name="l00502"></a>00502 <span class="comment"></span>
<a name="l00503"></a>00503 <span class="comment">  /**</span>
<a name="l00504"></a>00504 <span class="comment">   * Get the parameters used for this brlmm run.</span>
<a name="l00505"></a>00505 <span class="comment">   */</span>
<a name="l00506"></a>00506   <a class="code" href="classQuantBRLMM_1_1BrlmmParam.html">BrlmmParam</a> <a class="code" href="classQuantBRLMM.html#a6a3ae7a02c823637903a2d82d158b5b7" title="Get the parameters used for this brlmm run.">getParam</a>();
<a name="l00507"></a>00507 
<a name="l00508"></a>00508   <span class="keywordtype">void</span> setHaploidSnps(std::map&lt;std::string,bool&gt; &amp;haploidSnps);
<a name="l00509"></a>00509 
<a name="l00510"></a>00510   <span class="keywordtype">void</span> setSnpSpecificPriors(std::map&lt;std::string,ClusterPrior&gt; &amp;snpPriorMap);
<a name="l00511"></a>00511 
<a name="l00512"></a>00512   <span class="keywordtype">void</span> setGenders(std::vector&lt;affx::Gender&gt; &amp;genders);
<a name="l00513"></a>00513 
<a name="l00514"></a>00514   <span class="keywordtype">double</span> getMaxScore();
<a name="l00515"></a>00515   <span class="keywordtype">void</span> setMaxScore(<span class="keywordtype">float</span> v);
<a name="l00516"></a>00516 
<a name="l00517"></a>00517   <span class="keyword">typedef</span> std::map&lt;const char *,std::vector&lt;affx::GType&gt;, <a class="code" href="structUtil_1_1ltstr.html" title="Comparison object for use in map for char *.">Util::ltstr</a> &gt;  KnownGenoTypes_t;
<a name="l00518"></a>00518   <span class="keyword">typedef</span> KnownGenoTypes_t::iterator KnownGenoTypes_iter_t;
<a name="l00519"></a>00519 
<a name="l00520"></a>00520   <span class="keywordtype">double</span> <a class="code" href="classQuantBRLMM.html#a370cff001d055600645091956a385a5b" title="Get the min value for a call.">getMinThresh</a>();
<a name="l00521"></a>00521   <span class="keywordtype">double</span> <a class="code" href="classQuantBRLMM.html#a736c04fa1b48ef371480394af9a478c3" title="Get the max value for a call.">getMaxThresh</a>();
<a name="l00522"></a>00522 
<a name="l00523"></a>00523   <span class="keywordtype">void</span> setDmCallsOut(<span class="keyword">const</span> std::string&amp; fileName, std::vector&lt;std::string&gt; &amp;col_names);
<a name="l00524"></a>00524 
<a name="l00525"></a>00525   <span class="keyword">virtual</span> <span class="keywordtype">bool</span> <a class="code" href="classQuantBRLMM.html#ac6c6f4e8ffe27ec21902476b3d663fcc" title="Get set up for a run of reporting probesets.">prepare</a>(<span class="keyword">const</span> <a class="code" href="classIntensityMart.html" title="IntensityMart.">IntensityMart</a> &amp;iMart);
<a name="l00526"></a>00526 
<a name="l00527"></a>00527   <span class="keyword">virtual</span> <span class="keywordtype">bool</span> <a class="code" href="classQuantBRLMM.html#a4eea52e73038237e10b4328deb8aede1" title="No more probesets will be processed, this is a chance to finish outputting results and clean up...">finish</a>();
<a name="l00528"></a>00528 
<a name="l00529"></a>00529 <span class="keyword">private</span>:
<a name="l00530"></a>00530   std::vector&lt;affx::GType&gt; m_InitialCalls; <span class="comment">///&lt; Genotype calls were are supplied with for learning.</span>
<a name="l00531"></a>00531 <span class="comment"></span>  std::vector&lt;double&gt; m_Confidences; <span class="comment">///&lt; Our resulting confidences in those calls.</span>
<a name="l00532"></a>00532 <span class="comment"></span>  std::vector&lt;std::vector&lt;double&gt; &gt; m_Distances; <span class="comment">///&lt; Our resulting distances for AA, AB, and BB genotypes.</span>
<a name="l00533"></a>00533 <span class="comment"></span>  <a class="code" href="classQuantExprMethod.html" title="QuantExprMethod - Interface for computing quantification summaries from PM intensities grouped into p...">QuantExprMethod</a> *m_QuantMethod; <span class="comment">///&lt; Quantification method used for summarizing snps.</span>
<a name="l00534"></a>00534 <span class="comment"></span>  std::vector&lt;QuantMethodReport *&gt; m_Reporters;   <span class="comment">///&lt; Report object for outputting results</span>
<a name="l00535"></a>00535 <span class="comment"></span>  <a class="code" href="classClusterPrior.html">ClusterPrior</a> *m_Prior; <span class="comment">///&lt; Prior information about typical snp cluster layouts.</span>
<a name="l00536"></a>00536 <span class="comment"></span>  <a class="code" href="classClusterPrior.html">ClusterPrior</a> *m_CurrentPrior; <span class="comment">///&lt; Prior information about typical snp cluster layouts.</span>
<a name="l00537"></a>00537 <span class="comment"></span>  KnownGenoTypes_t m_KnownGenoTypes; <span class="comment">///&lt; Map from probeset names to the appopriate m_InitialCalls;</span>
<a name="l00538"></a>00538 <span class="comment"></span>  <a class="code" href="classQuantBRLMM_1_1BrlmmParam.html">BrlmmParam</a> m_Param; <span class="comment">///&lt; Various parameters used by the brlmm algorithm.</span>
<a name="l00539"></a>00539 <span class="comment"></span>  std::vector&lt;affx::Gender&gt; m_Genders;     <span class="comment">///&lt; What gender is each sample?</span>
<a name="l00540"></a>00540 <span class="comment"></span>  std::map&lt;std::string,bool&gt; m_HaploidSnps; <span class="comment">///&lt; Index of probesets that are in a haploid regions (usually X SNPs).</span>
<a name="l00541"></a>00541 <span class="comment"></span>  std::map&lt;std::string,ClusterPrior&gt; m_SnpPriorMap; <span class="comment">///&lt; Customized priors for different snps indexed by probeset name.</span>
<a name="l00542"></a>00542 <span class="comment"></span>
<a name="l00543"></a>00543   std::map&lt;std::string,ClusterModel&gt; m_SnpModelMap; <span class="comment">///&lt; Customized models for snps indexed by probeset name.</span>
<a name="l00544"></a>00544 <span class="comment"></span>  std::string m_ModelFileOutName; <span class="comment">///&lt; File to output snp specific models to if that is requested.</span>
<a name="l00545"></a>00545 <span class="comment"></span>  
<a name="l00546"></a>00546   <span class="keywordtype">double</span> m_BRLMMMaxScore;
<a name="l00547"></a>00547   <span class="comment">//</span>
<a name="l00548"></a>00548   <span class="comment">// std::string m_DmCallsFileName; ///&lt; File to output DM calls to if requested.</span>
<a name="l00549"></a>00549   <span class="comment">//</span>
<a name="l00550"></a>00550 <span class="keyword">public</span>:
<a name="l00551"></a><a class="code" href="classQuantBRLMM.html#ac85a70f66b43e43943f3da174d0378de">00551</a>   <a class="code" href="classaffx_1_1TsvReport.html" title="Unifies the presentation of file/TsvFile and file ///.">affx::TsvReport</a> <a class="code" href="classQuantBRLMM.html#ac85a70f66b43e43943f3da174d0378de" title="Output stream for printing snp specific models.">m_SnpModelsTsv</a>; <span class="comment">///&lt; Output stream for printing snp specific models</span>
<a name="l00552"></a><a class="code" href="classQuantBRLMM.html#a36e248207d3a69d8925365df52491fb9">00552</a> <span class="comment"></span>  <a class="code" href="classaffx_1_1TsvReport.html" title="Unifies the presentation of file/TsvFile and file ///.">affx::TsvReport</a> <a class="code" href="classQuantBRLMM.html#a36e248207d3a69d8925365df52491fb9" title="Output stream for printing snp specific models.">m_DmCallsTsv</a>; <span class="comment">///&lt; Output stream for printing snp specific models</span>
<a name="l00553"></a>00553 <span class="comment"></span>};
<a name="l00554"></a>00554 
<a name="l00555"></a>00555 <span class="preprocessor">#endif </span><span class="comment">/* QUANTBRLMM_H */</span>
</pre></div></div>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:36 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
