<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: chipstream/QuantMethodGTypeChipSummary.h Source File</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li class="current"><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="tabs2">
    <ul class="tablist">
      <li><a href="files.html"><span>File&nbsp;List</span></a></li>
      <li><a href="globals.html"><span>File&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath">
    <ul>
      <li><a class="el" href="dir_f48957cb46a44244b1a86c58bd157ee8.html">chipstream</a>      </li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>QuantMethodGTypeChipSummary.h</h1>  </div>
</div>
<div class="contents">
<a href="QuantMethodGTypeChipSummary_8h.html">Go to the documentation of this file.</a><div class="fragment"><pre class="fragment"><a name="l00001"></a>00001 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00002"></a>00002 <span class="comment"></span><span class="comment">//</span>
<a name="l00003"></a>00003 <span class="comment">// Copyright (C) 2007 Affymetrix, Inc.</span>
<a name="l00004"></a>00004 <span class="comment">//</span>
<a name="l00005"></a>00005 <span class="comment">// This program is free software; you can redistribute it and/or modify </span>
<a name="l00006"></a>00006 <span class="comment">// it under the terms of the GNU General Public License (version 2) as </span>
<a name="l00007"></a>00007 <span class="comment">// published by the Free Software Foundation.</span>
<a name="l00008"></a>00008 <span class="comment">// </span>
<a name="l00009"></a>00009 <span class="comment">// This program is distributed in the hope that it will be useful, </span>
<a name="l00010"></a>00010 <span class="comment">// but WITHOUT ANY WARRANTY; without even the implied warranty of </span>
<a name="l00011"></a>00011 <span class="comment">// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU </span>
<a name="l00012"></a>00012 <span class="comment">// General Public License for more details.</span>
<a name="l00013"></a>00013 <span class="comment">// </span>
<a name="l00014"></a>00014 <span class="comment">// You should have received a copy of the GNU General Public License </span>
<a name="l00015"></a>00015 <span class="comment">// along with this program;if not, write to the </span>
<a name="l00016"></a>00016 <span class="comment">// </span>
<a name="l00017"></a>00017 <span class="comment">// Free Software Foundation, Inc., </span>
<a name="l00018"></a>00018 <span class="comment">// 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA</span>
<a name="l00019"></a>00019 <span class="comment">//</span><span class="comment"></span>
<a name="l00020"></a>00020 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00021"></a>00021 <span class="comment"></span><span class="comment"></span>
<a name="l00022"></a>00022 <span class="comment">/**</span>
<a name="l00023"></a>00023 <span class="comment"> * @file   QuantMethodGTypeChipSummary.h</span>
<a name="l00024"></a>00024 <span class="comment"> * @author Alan Williams</span>
<a name="l00025"></a>00025 <span class="comment"> * </span>
<a name="l00026"></a>00026 <span class="comment"> * @brief  Class for accumulating some run statistics from a genotyping run.</span>
<a name="l00027"></a>00027 <span class="comment"> * </span>
<a name="l00028"></a>00028 <span class="comment"> */</span>
<a name="l00029"></a>00029 
<a name="l00030"></a>00030 <span class="preprocessor">#ifndef QUANTMETHODGTYPECHIPSUMMARY_H</span>
<a name="l00031"></a>00031 <span class="preprocessor"></span><span class="preprocessor">#define QUANTMETHODGTYPECHIPSUMMARY_H</span>
<a name="l00032"></a>00032 <span class="preprocessor"></span>
<a name="l00033"></a>00033 <span class="comment">//</span>
<a name="l00034"></a>00034 <span class="preprocessor">#include &quot;<a class="code" href="BioTypes_8h.html" title="Central location enumerations and other types that aren&amp;#39;t particular to a single class or applica...">chipstream/BioTypes.h</a>&quot;</span>
<a name="l00035"></a>00035 <span class="preprocessor">#include &quot;chipstream/ChipSummary.h&quot;</span>
<a name="l00036"></a>00036 <span class="preprocessor">#include &quot;chipstream/GenderCalls.h&quot;</span>
<a name="l00037"></a>00037 <span class="preprocessor">#include &quot;<a class="code" href="QuantGTypeMethod_8h.html" title="Interface for methods computing SNP genotyping calls.">chipstream/QuantGTypeMethod.h</a>&quot;</span>
<a name="l00038"></a>00038 <span class="preprocessor">#include &quot;<a class="code" href="QuantMethodReport_8h.html" title="Class for reporting results of quantification methods.">chipstream/QuantMethodReport.h</a>&quot;</span>
<a name="l00039"></a>00039 <span class="preprocessor">#include &quot;<a class="code" href="SpecialSnps_8h.html">chipstream/SpecialSnps.h</a>&quot;</span>
<a name="l00040"></a>00040 <span class="preprocessor">#include &quot;<a class="code" href="SummaryStats_8h.html" title="Utility class for keeping track of summary statistics.">chipstream/SummaryStats.h</a>&quot;</span>
<a name="l00041"></a>00041 <span class="comment">//</span>
<a name="l00042"></a>00042 <span class="preprocessor">#include &quot;<a class="code" href="stats_8h.html" title="///">stats/stats.h</a>&quot;</span>
<a name="l00043"></a>00043 <span class="preprocessor">#include &quot;<a class="code" href="Err_8h.html" title="Some utility error functions, currently prints message and aborts().">util/Err.h</a>&quot;</span>
<a name="l00044"></a>00044 <span class="preprocessor">#include &quot;<a class="code" href="Util_8h.html" title="General Utilities.">util/Util.h</a>&quot;</span>
<a name="l00045"></a>00045 <span class="comment">//</span>
<a name="l00046"></a>00046 <span class="preprocessor">#include &lt;cfloat&gt;</span>
<a name="l00047"></a>00047 <span class="preprocessor">#include &lt;iostream&gt;</span>
<a name="l00048"></a>00048 <span class="preprocessor">#include &lt;vector&gt;</span>
<a name="l00049"></a>00049 <span class="comment">//</span>
<a name="l00050"></a>00050 
<a name="l00051"></a><a class="code" href="classQuantMethodGTypeChipSummary.html">00051</a> <span class="keyword">class </span><a class="code" href="classQuantMethodGTypeChipSummary.html">QuantMethodGTypeChipSummary</a> : <span class="keyword">public</span> <a class="code" href="classQuantMethodReport.html" title="Class for reporting results of quantification methods.">QuantMethodReport</a>, <span class="keyword">public</span> <a class="code" href="classChipSummary.html" title="Class for objects that provide chip summary metrics.">ChipSummary</a> {
<a name="l00052"></a>00052 <span class="keyword">private</span>:
<a name="l00053"></a>00053   <span class="keyword">enum</span> CallType {
<a name="l00054"></a>00054       total = 0,
<a name="l00055"></a>00055       auto_total = 1,
<a name="l00056"></a>00056       hom   = 2,
<a name="l00057"></a>00057       auto_hom = 3,
<a name="l00058"></a>00058       het   = 4,
<a name="l00059"></a>00059       auto_het = 5,
<a name="l00060"></a>00060       nc    = 6,
<a name="l00061"></a>00061       auto_nc = 7,
<a name="l00062"></a>00062       pra   = 8,
<a name="l00063"></a>00063       cn0   = 9,
<a name="l00064"></a>00064       
<a name="l00065"></a>00065   };
<a name="l00066"></a>00066 
<a name="l00067"></a>00067   <span class="keyword">static</span> <span class="keyword">const</span> <span class="keywordtype">int</span> CallTypeSize = 10;
<a name="l00068"></a>00068 
<a name="l00069"></a>00069 
<a name="l00070"></a>00070 <span class="keyword">public</span>: 
<a name="l00071"></a>00071 
<a name="l00072"></a>00072   <a class="code" href="classQuantMethodGTypeChipSummary.html">QuantMethodGTypeChipSummary</a>(<span class="keyword">const</span> std::string&amp; analysisName,
<a name="l00073"></a>00073                               <a class="code" href="classGenderCalls.html" title="Class for determining gender from a cel set of cel files.">GenderCalls</a> *genders,
<a name="l00074"></a>00074                               SpecialSnpMap&amp; SpecialSnps,
<a name="l00075"></a>00075                               <span class="keywordtype">bool</span> alleleSummariesOnly = <span class="keyword">false</span>,
<a name="l00076"></a>00076                               <span class="keywordtype">double</span> minThreshold = -1 * DBL_MAX, 
<a name="l00077"></a>00077                               <span class="keywordtype">double</span> maxThreshold = DBL_MAX) {
<a name="l00078"></a>00078     <a class="code" href="classChipSummary.html#a81304a00c3b53c6ae1feb65d07d883fb" title="Set Valid State.">setValid</a>(<span class="keyword">false</span>);
<a name="l00079"></a>00079     m_AnalysisName = analysisName;
<a name="l00080"></a>00080     m_SpecialSnps = SpecialSnps;
<a name="l00081"></a>00081     m_MinThreshold = minThreshold;
<a name="l00082"></a>00082     m_MaxThreshold = maxThreshold;
<a name="l00083"></a>00083     m_Genders = genders;
<a name="l00084"></a>00084     m_AlleleSummariesOnly = alleleSummariesOnly;
<a name="l00085"></a>00085     <span class="comment">// set to reduce debugging warnings.</span>
<a name="l00086"></a>00086     setFormat(affx::TsvReport::FMT_TSV);
<a name="l00087"></a>00087     setFilename(<span class="stringliteral">&quot;QuantMethodGTypeChipSummary-DEBUG&quot;</span>); <span class="comment">// for debugging, not used.</span>
<a name="l00088"></a>00088 
<a name="l00089"></a>00089     <span class="keywordflow">if</span>(m_Genders-&gt;<a class="code" href="classGenderCalls.html#abb3474c5f29e2959ce730f539b73b006" title="Get the genders for the cel files that have been seen.">getGenders</a>().empty() == <span class="keyword">false</span>) {
<a name="l00090"></a>00090         <a class="code" href="classChipSummary.html#a8c573ca48c79d016640b01b9fa632294">declareMetric</a>(<span class="stringliteral">&quot;computed_gender&quot;</span>,ChipSummary::Metric::String);
<a name="l00091"></a>00091     }
<a name="l00092"></a>00092     <a class="code" href="classChipSummary.html#a8c573ca48c79d016640b01b9fa632294">declareMetric</a>(<span class="stringliteral">&quot;call_rate&quot;</span>,ChipSummary::Metric::Double);
<a name="l00093"></a>00093     <a class="code" href="classChipSummary.html#a8c573ca48c79d016640b01b9fa632294">declareMetric</a>(<span class="stringliteral">&quot;total_call_rate&quot;</span>,ChipSummary::Metric::Double);
<a name="l00094"></a>00094     <a class="code" href="classChipSummary.html#a8c573ca48c79d016640b01b9fa632294">declareMetric</a>(<span class="stringliteral">&quot;het_rate&quot;</span>,ChipSummary::Metric::Double);
<a name="l00095"></a>00095     <a class="code" href="classChipSummary.html#a8c573ca48c79d016640b01b9fa632294">declareMetric</a>(<span class="stringliteral">&quot;total_het_rate&quot;</span>,ChipSummary::Metric::Double);
<a name="l00096"></a>00096     <a class="code" href="classChipSummary.html#a8c573ca48c79d016640b01b9fa632294">declareMetric</a>(<span class="stringliteral">&quot;hom_rate&quot;</span>,ChipSummary::Metric::Double);
<a name="l00097"></a>00097     <a class="code" href="classChipSummary.html#a8c573ca48c79d016640b01b9fa632294">declareMetric</a>(<span class="stringliteral">&quot;total_hom_rate&quot;</span>,ChipSummary::Metric::Double);
<a name="l00098"></a>00098     <a class="code" href="classChipSummary.html#a8c573ca48c79d016640b01b9fa632294">declareMetric</a>(<span class="stringliteral">&quot;cluster_distance_mean&quot;</span>,ChipSummary::Metric::Double);
<a name="l00099"></a>00099     <a class="code" href="classChipSummary.html#a8c573ca48c79d016640b01b9fa632294">declareMetric</a>(<span class="stringliteral">&quot;cluster_distance_stdev&quot;</span>,ChipSummary::Metric::Double);
<a name="l00100"></a>00100   }
<a name="l00101"></a>00101 <span class="comment"></span>
<a name="l00102"></a>00102 <span class="comment">  /** </span>
<a name="l00103"></a>00103 <span class="comment">   * Get set up for a run of reporting probesets. Often used to open file</span>
<a name="l00104"></a>00104 <span class="comment">   * streams and print headers to files etc.</span>
<a name="l00105"></a>00105 <span class="comment">   * </span>
<a name="l00106"></a>00106 <span class="comment">   * @param qMethod - Quantification method to be used.</span>
<a name="l00107"></a>00107 <span class="comment">   * @param layout - Where the probesets, probes, etc are on the chip.</span>
<a name="l00108"></a>00108 <span class="comment">   * </span>
<a name="l00109"></a>00109 <span class="comment">   * @return true if success, false otherwise.</span>
<a name="l00110"></a>00110 <span class="comment">   */</span>
<a name="l00111"></a><a class="code" href="classQuantMethodGTypeChipSummary.html#ae5abc89349dbcfcf5e46d53e7f4d8f7f">00111</a>   <span class="keywordtype">bool</span> <a class="code" href="classQuantMethodGTypeChipSummary.html#ae5abc89349dbcfcf5e46d53e7f4d8f7f" title="Get set up for a run of reporting probesets.">prepare</a>(<a class="code" href="classQuantMethod.html" title="QuantMethod - Interface for computing quantification summaries from PM intensities grouped into probe...">QuantMethod</a> &amp;qMethod, <span class="keyword">const</span> <a class="code" href="classIntensityMart.html" title="IntensityMart.">IntensityMart</a> &amp;iMart) {
<a name="l00112"></a>00112     <span class="keywordtype">bool</span> allOk = <span class="keyword">true</span>;
<a name="l00113"></a>00113     <span class="keywordtype">int</span> chipCount = iMart.<a class="code" href="classIntensityMart.html#a4a6fdf2394b306e8effa666875e2865d" title="Get the total number of CELs/multi-CELs that this mart can supply.">getCelFileCount</a>();
<a name="l00114"></a>00114     <span class="keywordflow">for</span>(<span class="keywordtype">int</span> i=0; i&lt;chipCount; i++) {
<a name="l00115"></a>00115         std::vector&lt;ChipSummary::Metric&gt; metrics;
<a name="l00116"></a>00116         <a class="code" href="classChipSummary.html#a9dfa0a4049fef93ff9ec0d45f548c6dc" title="Stats for ChipSummary.">m_SummaryStats</a>.push_back(metrics);
<a name="l00117"></a>00117     }
<a name="l00118"></a>00118     m_Calls.resize(CallTypeSize);
<a name="l00119"></a>00119     <span class="keywordflow">for</span>(<span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> i = 0; i &lt; m_Calls.size(); i++) {
<a name="l00120"></a>00120       m_Calls[i].resize(chipCount);
<a name="l00121"></a>00121     }
<a name="l00122"></a>00122     m_MeanClosestDistance.resize(chipCount);
<a name="l00123"></a>00123     <span class="keywordflow">return</span> allOk;
<a name="l00124"></a>00124   }
<a name="l00125"></a>00125 <span class="comment"></span>
<a name="l00126"></a>00126 <span class="comment">  /** </span>
<a name="l00127"></a>00127 <span class="comment">   * After every probeset computation this function is called an is an opportunity</span>
<a name="l00128"></a>00128 <span class="comment">   * to query the quantification method for results, residuals, etc.</span>
<a name="l00129"></a>00129 <span class="comment">   * </span>
<a name="l00130"></a>00130 <span class="comment">   * @param psGroup - List of probesets from which probes were used.</span>
<a name="l00131"></a>00131 <span class="comment">   * @param qMethod - Quantification method with compute method called.</span>
<a name="l00132"></a>00132 <span class="comment">   * @param layout - Where the probesets, probes, etc are on the chip.</span>
<a name="l00133"></a>00133 <span class="comment">   * </span>
<a name="l00134"></a>00134 <span class="comment">   * @return true if success, false otherwise.</span>
<a name="l00135"></a>00135 <span class="comment">   */</span>
<a name="l00136"></a><a class="code" href="classQuantMethodGTypeChipSummary.html#ad0e98c8a204c646e59ff10e55fcb25b9">00136</a>   <span class="keywordtype">bool</span> <a class="code" href="classQuantMethodGTypeChipSummary.html#ad0e98c8a204c646e59ff10e55fcb25b9" title="After every probeset computation this function is called an is an opportunity to query the quantifica...">report</a>(<a class="code" href="classProbeSetGroup.html" title="Group of probe sets that should be processed as a single large probe set.">ProbeSetGroup</a> &amp;psGroup, <a class="code" href="classQuantMethod.html" title="QuantMethod - Interface for computing quantification summaries from PM intensities grouped into probe...">QuantMethod</a> &amp;qMethod,
<a name="l00137"></a>00137               <span class="keyword">const</span> <a class="code" href="classIntensityMart.html" title="IntensityMart.">IntensityMart</a> &amp;iMart, std::vector&lt;ChipStream *&gt; &amp;iTrans, 
<a name="l00138"></a>00138               <a class="code" href="classPmAdjuster.html" title="Interface for determining a change based on intensity of perfect match.">PmAdjuster</a> &amp;pmAdjust) {
<a name="l00139"></a>00139     <a class="code" href="classQuantGTypeMethod.html" title="Quantification methods used for making genotyping calls implement (currently just brlmm) this interfa...">QuantGTypeMethod</a> *gMethod = <span class="keyword">dynamic_cast&lt;</span><a class="code" href="classQuantGTypeMethod.html" title="Quantification methods used for making genotyping calls implement (currently just brlmm) this interfa...">QuantGTypeMethod</a> *<span class="keyword">&gt;</span>(&amp;qMethod);
<a name="l00140"></a>00140     <span class="keywordtype">bool</span> realCalls = <span class="keyword">true</span>;
<a name="l00141"></a>00141     <span class="keywordflow">if</span>(gMethod == NULL) {
<a name="l00142"></a>00142       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Can only use a QuantMethodGTypeReport with QuantGTypeMethods.&quot;</span>);
<a name="l00143"></a>00143     }
<a name="l00144"></a>00144     <a class="code" href="classErr.html#ad1a633f602fe60689cdc12d9ca1842e9" title="errAbort based version of assert with an error message.">Err::check</a>(gMethod-&gt;<a class="code" href="classQuantGTypeMethod.html#aa02713a7902dd54cc6edc3a51c62b3b1" title="How many genotyping calls do we have? Also indicates how many samples there are.">getNumCalls</a>() == m_Calls[0].size(), 
<a name="l00145"></a>00145                <span class="stringliteral">&quot;Error: QuantMethodGTypeChipSummary::report() - Wrong number of calls.&quot;</span>);
<a name="l00146"></a>00146     std::vector&lt;double&gt; distanceValues(gMethod-&gt;<a class="code" href="classQuantGTypeMethod.html#aa02713a7902dd54cc6edc3a51c62b3b1" title="How many genotyping calls do we have? Also indicates how many samples there are.">getNumCalls</a>());
<a name="l00147"></a>00147     <span class="comment">/* Keep track of the calls. */</span>
<a name="l00148"></a>00148     <span class="keyword">const</span> std::string probeSetName = <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(psGroup.<a class="code" href="classProbeSetGroup.html#a0eb8ea2dfda45864880e0495539753db" title="Name of probe set group.">name</a>);
<a name="l00149"></a>00149     <span class="keywordtype">bool</span> autosomal = !<a class="code" href="SpecialSnps_8cpp.html#a686b53128432db183ef4dc37252d3c4b" title="Determines if special snp is from chrX or chrY.">isHaploid</a>(m_SpecialSnps, probeSetName);
<a name="l00150"></a>00150     <span class="keywordflow">for</span>(<span class="keywordtype">unsigned</span> <span class="keywordtype">int</span> i = 0; i &lt; gMethod-&gt;<a class="code" href="classQuantGTypeMethod.html#aa02713a7902dd54cc6edc3a51c62b3b1" title="How many genotyping calls do we have? Also indicates how many samples there are.">getNumCalls</a>(); i++) {<span class="comment"></span>
<a name="l00151"></a>00151 <span class="comment">      ///@todo Nice to get away from notion of GTypeCall. This is a hack which works with QuantLabelZMulti</span>
<a name="l00152"></a>00152 <span class="comment">      ///      because we are only interested in if the call is a hom or het.</span>
<a name="l00153"></a>00153 <span class="comment"></span>      affx::GType forcedCall = gMethod-&gt;getGTypeForcedCall(i);
<a name="l00154"></a>00154       <span class="keywordtype">double</span> conf = gMethod-&gt;<a class="code" href="classQuantGTypeMethod.html#a0d242123ad6e0bfd6934d0ef0c0073cc" title="Get our confidence value for a particular call in a particular sample.">getConfidence</a>(i);
<a name="l00155"></a>00155       <span class="keywordtype">float</span> dist = gMethod-&gt;<a class="code" href="classQuantGTypeMethod.html#a5015e87ff9afae50eb5d926f0c587a29" title="Get our confidence value for a particular call in a particular sample.">getConfGenotype</a>(forcedCall, i);
<a name="l00156"></a>00156       <span class="keywordflow">if</span>(!<a class="code" href="classUtil.html#a66b3f27deb315aab108936d0830c6991" title="Wrapper for different version of isnan() on different systems.">Util::isFinite</a>(conf) || !<a class="code" href="classUtil.html#a66b3f27deb315aab108936d0830c6991" title="Wrapper for different version of isnan() on different systems.">Util::isFinite</a>(dist)) {
<a name="l00157"></a>00157         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Warning! - Non-finite (Nan/Inf) confidence produced in probeset: &#39;&quot;</span> + probeSetName + 
<a name="l00158"></a>00158                      <span class="stringliteral">&quot;&#39; at column: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(i));
<a name="l00159"></a>00159       }
<a name="l00160"></a>00160       <span class="keywordflow">if</span>(dist == FLT_MAX) 
<a name="l00161"></a>00161         realCalls = <span class="keyword">false</span>;
<a name="l00162"></a>00162       distanceValues[i] = dist;
<a name="l00163"></a>00163 
<a name="l00164"></a>00164 
<a name="l00165"></a>00165       <a class="code" href="classQuantLabelZMulti.html">QuantLabelZMulti</a> *qmm = <span class="keyword">dynamic_cast&lt;</span><a class="code" href="classQuantLabelZMulti.html">QuantLabelZMulti</a> *<span class="keyword">&gt;</span>(&amp;qMethod);
<a name="l00166"></a>00166       <span class="keywordflow">if</span>(qmm != NULL) {
<a name="l00167"></a>00167             <span class="keywordtype">int</span> call = qmm-&gt;<a class="code" href="classQuantLabelZMulti.html#a4d961567ab1e43868246c7300ef5b7c1" title="Gets the integer valued (not affx::GType) call for the probeset.">getCall</a>(i);
<a name="l00168"></a>00168             <a class="code" href="classGenoCallCoder.html" title="GenoCallCoder - Class for encoding and decoding numeric genotype call codes to and from allele string...">GenoCallCoder</a> *coder = qmm-&gt;m_coder;
<a name="l00169"></a>00169             <span class="keywordflow">if</span>(coder == NULL) 
<a name="l00170"></a>00170                 <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;GenoCallCoder is NULL&quot;</span>);
<a name="l00171"></a>00171             <span class="keywordflow">if</span>(coder-&gt;<a class="code" href="classGenoCallCoder.html#ab9ff7b3ee450048176b2ce5669a1899d" title="Decode from call code number to allele name string for given probeset id.">abstractAlleleToGenotypeCallNum</a>(<span class="stringliteral">&quot;NoCall&quot;</span>) == call) {
<a name="l00172"></a>00172                 m_Calls[nc][i]++;
<a name="l00173"></a>00173                 <span class="keywordflow">if</span> (autosomal) {
<a name="l00174"></a>00174                     m_Calls[auto_nc][i]++;
<a name="l00175"></a>00175                 }
<a name="l00176"></a>00176             } 
<a name="l00177"></a>00177             <span class="keywordflow">else</span> <span class="keywordflow">if</span>(coder-&gt;<a class="code" href="classGenoCallCoder.html#ab9ff7b3ee450048176b2ce5669a1899d" title="Decode from call code number to allele name string for given probeset id.">abstractAlleleToGenotypeCallNum</a>(<span class="stringliteral">&quot;PossibleRareAllele&quot;</span>) == call) {
<a name="l00178"></a>00178                 m_Calls[pra][i]++;
<a name="l00179"></a>00179             } 
<a name="l00180"></a>00180             <span class="keywordflow">else</span> <span class="keywordflow">if</span>(coder-&gt;<a class="code" href="classGenoCallCoder.html#ab9ff7b3ee450048176b2ce5669a1899d" title="Decode from call code number to allele name string for given probeset id.">abstractAlleleToGenotypeCallNum</a>(<span class="stringliteral">&quot;ZeroCopyNumber&quot;</span>) == call) {
<a name="l00181"></a>00181                 m_Calls[cn0][i]++;
<a name="l00182"></a>00182             } 
<a name="l00183"></a>00183             <span class="keywordflow">else</span> <span class="keywordflow">if</span>(coder-&gt;<a class="code" href="classGenoCallCoder.html#adc45f8788fe7571ab849403139e553d8" title="boolean indicating if call code corresponds to a homozygous call.">isHom</a>(call)) {
<a name="l00184"></a>00184                 m_Calls[hom][i]++;
<a name="l00185"></a>00185                 <span class="keywordflow">if</span> (autosomal) {
<a name="l00186"></a>00186                     m_Calls[auto_hom][i]++;
<a name="l00187"></a>00187                 }
<a name="l00188"></a>00188             } 
<a name="l00189"></a>00189             <span class="keywordflow">else</span> <span class="keywordflow">if</span>(coder-&gt;<a class="code" href="classGenoCallCoder.html#a0ec1b541ca46b796a0f4a44ee194651e" title="boolean indicating if call code corresponds to a heterozygous call">isHet</a>(call)) {
<a name="l00190"></a>00190                 m_Calls[het][i]++;
<a name="l00191"></a>00191                 <span class="keywordflow">if</span> (autosomal) {
<a name="l00192"></a>00192                     m_Calls[auto_het][i]++;
<a name="l00193"></a>00193                 }
<a name="l00194"></a>00194             } 
<a name="l00195"></a>00195             <span class="keywordflow">else</span> {
<a name="l00196"></a>00196                 <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;Unaccounted for call (&quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(call) + <span class="stringliteral">&quot;). Treating as no-call for summary stats.&quot;</span>);
<a name="l00197"></a>00197                 m_Calls[nc][i]++;
<a name="l00198"></a>00198 
<a name="l00199"></a>00199             }
<a name="l00200"></a>00200       } 
<a name="l00201"></a>00201       <span class="keywordflow">else</span> {
<a name="l00202"></a>00202             affx::GType call = gMethod-&gt;<a class="code" href="classQuantGTypeMethod.html#a4ef68832da9ea1ac365a098d2e47bf3d" title="Get the genotype call at specified index (sample).">getGTypeCall</a>(i);
<a name="l00203"></a>00203             <span class="keywordflow">if</span> (call == <a class="code" href="namespaceaffx.html#a3886a33ccf4ba70052a915d878634c30a3e1bc1d31bc880685d5870951ffb5f17" title="Homozygous A version of allele. (0).">affx::AA</a>) {
<a name="l00204"></a>00204                 m_Calls[hom][i]++;
<a name="l00205"></a>00205                 <span class="keywordflow">if</span> (autosomal) {
<a name="l00206"></a>00206                     m_Calls[auto_hom][i]++;
<a name="l00207"></a>00207                 }
<a name="l00208"></a>00208             } <span class="keywordflow">else</span> <span class="keywordflow">if</span>(call == <a class="code" href="namespaceaffx.html#a3886a33ccf4ba70052a915d878634c30a83b51b7ed2e4a1b4e23378773929465b" title="Heterozygous, one copy of both A and B. (1).">affx::AB</a>) {
<a name="l00209"></a>00209                 m_Calls[het][i]++;
<a name="l00210"></a>00210                 <span class="keywordflow">if</span> (autosomal) {
<a name="l00211"></a>00211                     m_Calls[auto_het][i]++;
<a name="l00212"></a>00212                 }
<a name="l00213"></a>00213             } <span class="keywordflow">else</span> <span class="keywordflow">if</span>(call == <a class="code" href="namespaceaffx.html#a3886a33ccf4ba70052a915d878634c30a7c78281ed74a1b7cfe2ab49e6af25231" title="Homozygous B version of allele. (2).">affx::BB</a>) {
<a name="l00214"></a>00214                 m_Calls[hom][i]++;
<a name="l00215"></a>00215                 <span class="keywordflow">if</span> (autosomal) {
<a name="l00216"></a>00216                     m_Calls[auto_hom][i]++;
<a name="l00217"></a>00217                 }
<a name="l00218"></a>00218             } <span class="keywordflow">else</span> <span class="keywordflow">if</span>(call == <a class="code" href="namespaceaffx.html#a3886a33ccf4ba70052a915d878634c30a1cef2fc4697bd5fc67116fbd1ee2027b" title="No call, unknown genotype. The internal is value is &amp;#39;3&amp;#39;; external is &amp;#39;-1&amp;#39;,.">affx::NN</a>) {
<a name="l00219"></a>00219                 m_Calls[nc][i]++;
<a name="l00220"></a>00220                 <span class="keywordflow">if</span> (autosomal) {
<a name="l00221"></a>00221                     m_Calls[auto_nc][i]++;
<a name="l00222"></a>00222                 }
<a name="l00223"></a>00223             } <span class="keywordflow">else</span> {
<a name="l00224"></a>00224                 <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;Unaccounted for call (&quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(call) + <span class="stringliteral">&quot;). Treating as no-call for summary stats.&quot;</span>);
<a name="l00225"></a>00225                 m_Calls[nc][i]++;
<a name="l00226"></a>00226                 <span class="keywordflow">if</span> (autosomal) {
<a name="l00227"></a>00227                     m_Calls[auto_nc][i]++;
<a name="l00228"></a>00228                 }
<a name="l00229"></a>00229             }
<a name="l00230"></a>00230       }
<a name="l00231"></a>00231       m_Calls[total][i]++;
<a name="l00232"></a>00232       <span class="keywordflow">if</span> (autosomal) {
<a name="l00233"></a>00233         m_Calls[auto_total][i]++;        
<a name="l00234"></a>00234       }
<a name="l00235"></a>00235     }
<a name="l00236"></a>00236     <span class="comment">/* if there were real predictions then keep track of them. */</span>
<a name="l00237"></a>00237     <span class="keywordflow">if</span>(realCalls) {
<a name="l00238"></a>00238       m_MeanClosestDistance.<a class="code" href="classSummaryStats.html#ad1721b36feb634af01eff7de2aa75156" title="Add a vector of data, keeping a running tab of the mean, and variance statistics.">addData</a>(distanceValues);
<a name="l00239"></a>00239     }
<a name="l00240"></a>00240 
<a name="l00241"></a>00241     <span class="keywordflow">return</span> <span class="keyword">true</span>;
<a name="l00242"></a>00242   }
<a name="l00243"></a>00243 <span class="comment"></span>
<a name="l00244"></a>00244 <span class="comment">  /** </span>
<a name="l00245"></a>00245 <span class="comment">   * If a probeset fails to compute for whatever reason then this method is</span>
<a name="l00246"></a>00246 <span class="comment">   * called rather than the normal report call above. By default does nothing.</span>
<a name="l00247"></a>00247 <span class="comment">   * </span>
<a name="l00248"></a>00248 <span class="comment">   * @param psGroup - List of probesets from which probes were used.</span>
<a name="l00249"></a>00249 <span class="comment">   * @param qMethod - Quantification method with compute method called.</span>
<a name="l00250"></a>00250 <span class="comment">   * @param layout - Where the probesets, probes, etc are on the chip.</span>
<a name="l00251"></a>00251 <span class="comment">   * </span>
<a name="l00252"></a>00252 <span class="comment">   * @return true if success, false otherwise.</span>
<a name="l00253"></a>00253 <span class="comment">   */</span>
<a name="l00254"></a><a class="code" href="classQuantMethodGTypeChipSummary.html#a301b73640e025183f0843c0a653cec0f">00254</a>    <span class="keywordtype">bool</span> <a class="code" href="classQuantMethodGTypeChipSummary.html#a301b73640e025183f0843c0a653cec0f" title="If a probeset fails to compute for whatever reason then this method is called rather than the normal ...">reportFailure</a>(<a class="code" href="classProbeSetGroup.html" title="Group of probe sets that should be processed as a single large probe set.">ProbeSetGroup</a> &amp;psGroup, <a class="code" href="classQuantMethod.html" title="QuantMethod - Interface for computing quantification summaries from PM intensities grouped into probe...">QuantMethod</a> &amp;qMethod, 
<a name="l00255"></a>00255                       <span class="keyword">const</span> <a class="code" href="classIntensityMart.html" title="IntensityMart.">IntensityMart</a> &amp;iMart, std::vector&lt;ChipStream *&gt; &amp;iTrans, 
<a name="l00256"></a>00256                       <a class="code" href="classPmAdjuster.html" title="Interface for determining a change based on intensity of perfect match.">PmAdjuster</a> &amp;pmAdjust) {
<a name="l00257"></a>00257 
<a name="l00258"></a>00258      <span class="comment">// do nothing.</span>
<a name="l00259"></a>00259      <span class="keywordflow">return</span> <span class="keyword">true</span>;
<a name="l00260"></a>00260    }
<a name="l00261"></a>00261 <span class="comment"></span>
<a name="l00262"></a>00262 <span class="comment">  /** </span>
<a name="l00263"></a>00263 <span class="comment">   * No more probesets will be processed, this is a chance to finish outputting</span>
<a name="l00264"></a>00264 <span class="comment">   * results and clean up.</span>
<a name="l00265"></a>00265 <span class="comment">   * @param qMethod - Quantification method that was used.</span>
<a name="l00266"></a>00266 <span class="comment">   * @return true if success, false otherwise.</span>
<a name="l00267"></a>00267 <span class="comment">   */</span>
<a name="l00268"></a><a class="code" href="classQuantMethodGTypeChipSummary.html#a78ec07b661d6fb54b150d7fd5ffba6f8">00268</a>   <span class="keywordtype">bool</span> <a class="code" href="classQuantMethodGTypeChipSummary.html#a78ec07b661d6fb54b150d7fd5ffba6f8" title="No more probesets will be processed, this is a chance to finish outputting results and clean up...">finish</a>(<a class="code" href="classQuantMethod.html" title="QuantMethod - Interface for computing quantification summaries from PM intensities grouped into probe...">QuantMethod</a> &amp;qMethod) {
<a name="l00269"></a>00269 
<a name="l00270"></a>00270     vector&lt;affx::Gender&gt; genders(0);
<a name="l00271"></a>00271     <span class="keywordflow">if</span>(m_Genders != NULL) {
<a name="l00272"></a>00272         genders = m_Genders-&gt;<a class="code" href="classGenderCalls.html#abb3474c5f29e2959ce730f539b73b006" title="Get the genders for the cel files that have been seen.">getGenders</a>();
<a name="l00273"></a>00273     }
<a name="l00274"></a>00274 
<a name="l00275"></a>00275     <span class="comment">// Load up other metrics computed in m_Stats</span>
<a name="l00276"></a>00276     <span class="keywordflow">for</span>(<span class="keywordtype">size_t</span> chip = 0; chip &lt; <a class="code" href="classChipSummary.html#a9dfa0a4049fef93ff9ec0d45f548c6dc" title="Stats for ChipSummary.">m_SummaryStats</a>.size(); chip++) {
<a name="l00277"></a>00277 <span class="comment"></span>
<a name="l00278"></a>00278 <span class="comment">        /// @todo should this be changed to m_AnalysisName + &quot;_gender_used&quot;? </span>
<a name="l00279"></a>00279 <span class="comment"></span>        <span class="keywordflow">if</span> (genders.empty() == <span class="keyword">false</span>) {
<a name="l00280"></a>00280             <a class="code" href="classChipSummary.html#a9dfa0a4049fef93ff9ec0d45f548c6dc" title="Stats for ChipSummary.">m_SummaryStats</a>[chip].push_back(<a class="code" href="classChipSummary_1_1Metric.html" title="Class for storing metrics.">ChipSummary::Metric</a>(<span class="stringliteral">&quot;computed_gender&quot;</span>, <a class="code" href="namespaceaffx.html#a91c5e36f4ed8c88d90f3aa3086ca95df" title="convert the BioType Gender to a string ///">affx::getGenderString</a>(genders[chip])));
<a name="l00281"></a>00281         }
<a name="l00282"></a>00282         <span class="comment">// was added -- but now removing to reduce redundancy. analysis can be had from the</span>
<a name="l00283"></a>00283         <span class="comment">// meta header info (which previously was not provided)</span>
<a name="l00284"></a>00284         <span class="comment">//m_SummaryStats[chip].push_back(ChipSummary::Metric(m_AnalysisName + &quot;_call_rate&quot;, getCallRate(chip)));</span>
<a name="l00285"></a>00285         <a class="code" href="classChipSummary.html#a9dfa0a4049fef93ff9ec0d45f548c6dc" title="Stats for ChipSummary.">m_SummaryStats</a>[chip].push_back(<a class="code" href="classChipSummary_1_1Metric.html" title="Class for storing metrics.">ChipSummary::Metric</a>(<span class="stringliteral">&quot;call_rate&quot;</span>, getAutoCallRate(chip)));
<a name="l00286"></a>00286         <a class="code" href="classChipSummary.html#a9dfa0a4049fef93ff9ec0d45f548c6dc" title="Stats for ChipSummary.">m_SummaryStats</a>[chip].push_back(<a class="code" href="classChipSummary_1_1Metric.html" title="Class for storing metrics.">ChipSummary::Metric</a>(<span class="stringliteral">&quot;total_call_rate&quot;</span>, getTotalCallRate(chip)));
<a name="l00287"></a>00287         <a class="code" href="classChipSummary.html#a9dfa0a4049fef93ff9ec0d45f548c6dc" title="Stats for ChipSummary.">m_SummaryStats</a>[chip].push_back(<a class="code" href="classChipSummary_1_1Metric.html" title="Class for storing metrics.">ChipSummary::Metric</a>(<span class="stringliteral">&quot;het_rate&quot;</span>, getAutoHetCallRate(chip)));
<a name="l00288"></a>00288         <a class="code" href="classChipSummary.html#a9dfa0a4049fef93ff9ec0d45f548c6dc" title="Stats for ChipSummary.">m_SummaryStats</a>[chip].push_back(<a class="code" href="classChipSummary_1_1Metric.html" title="Class for storing metrics.">ChipSummary::Metric</a>(<span class="stringliteral">&quot;total_het_rate&quot;</span>, getTotalHetCallRate(chip)));
<a name="l00289"></a>00289         <a class="code" href="classChipSummary.html#a9dfa0a4049fef93ff9ec0d45f548c6dc" title="Stats for ChipSummary.">m_SummaryStats</a>[chip].push_back(<a class="code" href="classChipSummary_1_1Metric.html" title="Class for storing metrics.">ChipSummary::Metric</a>(<span class="stringliteral">&quot;hom_rate&quot;</span>, getAutoHomCallRate(chip)));
<a name="l00290"></a>00290         <a class="code" href="classChipSummary.html#a9dfa0a4049fef93ff9ec0d45f548c6dc" title="Stats for ChipSummary.">m_SummaryStats</a>[chip].push_back(<a class="code" href="classChipSummary_1_1Metric.html" title="Class for storing metrics.">ChipSummary::Metric</a>(<span class="stringliteral">&quot;total_hom_rate&quot;</span>, getTotalHomCallRate(chip)));
<a name="l00291"></a>00291         <a class="code" href="classChipSummary.html#a9dfa0a4049fef93ff9ec0d45f548c6dc" title="Stats for ChipSummary.">m_SummaryStats</a>[chip].push_back(<a class="code" href="classChipSummary_1_1Metric.html" title="Class for storing metrics.">ChipSummary::Metric</a>(<span class="stringliteral">&quot;cluster_distance_mean&quot;</span>, m_MeanClosestDistance.getMean(chip)));
<a name="l00292"></a>00292         <span class="comment">// this little hack is here to suppress warning messages that would</span>
<a name="l00293"></a>00293         <span class="comment">// probably vex some users.</span>
<a name="l00294"></a>00294         <span class="keywordtype">double</span> distanceStdDev = (m_AlleleSummariesOnly ? 0 : m_MeanClosestDistance.getStdev(chip));
<a name="l00295"></a>00295         <a class="code" href="classChipSummary.html#a9dfa0a4049fef93ff9ec0d45f548c6dc" title="Stats for ChipSummary.">m_SummaryStats</a>[chip].push_back(<a class="code" href="classChipSummary_1_1Metric.html" title="Class for storing metrics.">ChipSummary::Metric</a>(<span class="stringliteral">&quot;cluster_distance_stdev&quot;</span>, distanceStdDev));
<a name="l00296"></a>00296     }
<a name="l00297"></a>00297 
<a name="l00298"></a>00298     <span class="comment">// @todo: this shouldnt be needed -jhg</span>
<a name="l00299"></a>00299     <span class="comment">// zero out the header buffering (We have headers pushed ontous which we never use.</span>
<a name="l00300"></a>00300     m_header_buffer.resize(0);
<a name="l00301"></a>00301     
<a name="l00302"></a>00302     <span class="comment">//</span>
<a name="l00303"></a>00303     <a class="code" href="classChipSummary.html#a81304a00c3b53c6ae1feb65d07d883fb" title="Set Valid State.">setValid</a>(<span class="keyword">true</span>);
<a name="l00304"></a>00304     <span class="keywordflow">return</span> <span class="keyword">true</span>;
<a name="l00305"></a>00305   }
<a name="l00306"></a>00306 
<a name="l00307"></a>00307 <span class="keyword">private</span>: 
<a name="l00308"></a>00308 <span class="comment"></span>
<a name="l00309"></a>00309 <span class="comment">  /** Get the autosomal call rate */</span>
<a name="l00310"></a>00310   <span class="keywordtype">double</span> getAutoCallRate(<span class="keywordtype">int</span> index) {
<a name="l00311"></a>00311     <span class="keywordflow">return</span> 100.0 * (m_Calls[auto_total][index] - m_Calls[auto_nc][index]) / m_Calls[auto_total][index];
<a name="l00312"></a>00312   }
<a name="l00313"></a>00313 <span class="comment"></span>
<a name="l00314"></a>00314 <span class="comment">  /** Get the total call rate */</span>
<a name="l00315"></a>00315   <span class="keywordtype">double</span> getTotalCallRate(<span class="keywordtype">int</span> index) {
<a name="l00316"></a>00316     <span class="keywordflow">return</span> 100.0 * (m_Calls[total][index] - m_Calls[nc][index]) / m_Calls[total][index];
<a name="l00317"></a>00317   }
<a name="l00318"></a>00318 <span class="comment"></span>
<a name="l00319"></a>00319 <span class="comment">  /** Get the autosomal het call rate */</span>
<a name="l00320"></a>00320   <span class="keywordtype">double</span> getAutoHetCallRate(<span class="keywordtype">int</span> index) {
<a name="l00321"></a>00321     <span class="keywordflow">return</span> 100.0 * (m_Calls[auto_het][index]) / (m_Calls[auto_total][index]);
<a name="l00322"></a>00322   }
<a name="l00323"></a>00323 <span class="comment"></span>
<a name="l00324"></a>00324 <span class="comment">  /** Get the total het call rate */</span>
<a name="l00325"></a>00325   <span class="keywordtype">double</span> getTotalHetCallRate(<span class="keywordtype">int</span> index) {
<a name="l00326"></a>00326     <span class="keywordflow">return</span> 100.0 * (m_Calls[het][index]) / (m_Calls[total][index]);
<a name="l00327"></a>00327   }
<a name="l00328"></a>00328 <span class="comment"></span>
<a name="l00329"></a>00329 <span class="comment">  /** Get the autosomal hom call rate */</span>
<a name="l00330"></a>00330   <span class="keywordtype">double</span> getAutoHomCallRate(<span class="keywordtype">int</span> index) {
<a name="l00331"></a>00331     <span class="keywordflow">return</span> 100.0 * (m_Calls[auto_hom][index]) / (m_Calls[auto_total][index]);
<a name="l00332"></a>00332   }
<a name="l00333"></a>00333 <span class="comment"></span>
<a name="l00334"></a>00334 <span class="comment">  /** Get the total hom call rate */</span>
<a name="l00335"></a>00335   <span class="keywordtype">double</span> getTotalHomCallRate(<span class="keywordtype">int</span> index) {
<a name="l00336"></a>00336     <span class="keywordflow">return</span> 100.0 * (m_Calls[hom][index]) / (m_Calls[total][index]);
<a name="l00337"></a>00337   }
<a name="l00338"></a>00338 
<a name="l00339"></a>00339 <span class="keyword">private</span>:<span class="comment"></span>
<a name="l00340"></a>00340 <span class="comment">  /// The name of our analysis.</span>
<a name="l00341"></a>00341 <span class="comment"></span>  std::string m_AnalysisName;<span class="comment"></span>
<a name="l00342"></a>00342 <span class="comment">  /// What level of confidence is needed to make a call?</span>
<a name="l00343"></a>00343 <span class="comment"></span>  <span class="keywordtype">double</span> m_MinThreshold;
<a name="l00344"></a>00344   <span class="keywordtype">double</span> m_MaxThreshold;<span class="comment"></span>
<a name="l00345"></a>00345 <span class="comment">  /// Keep track of call counts per chip. Indexes are CallType enum</span>
<a name="l00346"></a>00346 <span class="comment"></span>  std::vector&lt;std::vector&lt;int&gt; &gt; m_Calls;<span class="comment"></span>
<a name="l00347"></a>00347 <span class="comment">  /// Distance to nearest cluster center.</span>
<a name="l00348"></a>00348 <span class="comment"></span>  <a class="code" href="classSummaryStats.html">SummaryStats</a> m_MeanClosestDistance; <span class="comment"></span>
<a name="l00349"></a>00349 <span class="comment">  /// Gender of sample.</span>
<a name="l00350"></a>00350 <span class="comment"></span>  <a class="code" href="classGenderCalls.html" title="Class for determining gender from a cel set of cel files.">GenderCalls</a> *m_Genders;<span class="comment"></span>
<a name="l00351"></a>00351 <span class="comment">  /// map of haloid snp names</span>
<a name="l00352"></a>00352 <span class="comment"></span>  SpecialSnpMap m_SpecialSnps;
<a name="l00353"></a>00353   <span class="keywordtype">bool</span> m_AlleleSummariesOnly;
<a name="l00354"></a>00354 
<a name="l00355"></a>00355 <span class="comment">//   1) mean/stdev median absolute deviation</span>
<a name="l00356"></a>00356 <span class="comment">//   2) allele summary.</span>
<a name="l00357"></a>00357 
<a name="l00358"></a>00358 };
<a name="l00359"></a>00359 
<a name="l00360"></a>00360 <span class="preprocessor">#endif </span><span class="comment">/* QUANTMETHODGTYPECHIPSUMMARY_H */</span>
</pre></div></div>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:36 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
