<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: chipstream/apt-summary-genotype/SummaryGenotypeEngine.cpp Source File</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li class="current"><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="tabs2">
    <ul class="tablist">
      <li><a href="files.html"><span>File&nbsp;List</span></a></li>
      <li><a href="globals.html"><span>File&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath">
    <ul>
      <li><a class="el" href="dir_f48957cb46a44244b1a86c58bd157ee8.html">chipstream</a>      </li>
      <li><a class="el" href="dir_89b9ddcb07e471fe0272bc2e39323086.html">apt-summary-genotype</a>      </li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>SummaryGenotypeEngine.cpp</h1>  </div>
</div>
<div class="contents">
<div class="fragment"><pre class="fragment"><a name="l00001"></a>00001 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00002"></a>00002 <span class="comment"></span><span class="comment">//</span>
<a name="l00003"></a>00003 <span class="comment">// Copyright (C) 2009 Affymetrix, Inc.</span>
<a name="l00004"></a>00004 <span class="comment">//</span>
<a name="l00005"></a>00005 <span class="comment">// This program is free software; you can redistribute it and/or modify </span>
<a name="l00006"></a>00006 <span class="comment">// it under the terms of the GNU General Public License (version 2) as </span>
<a name="l00007"></a>00007 <span class="comment">// published by the Free Software Foundation.</span>
<a name="l00008"></a>00008 <span class="comment">// </span>
<a name="l00009"></a>00009 <span class="comment">// This program is distributed in the hope that it will be useful, </span>
<a name="l00010"></a>00010 <span class="comment">// but WITHOUT ANY WARRANTY; without even the implied warranty of </span>
<a name="l00011"></a>00011 <span class="comment">// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU </span>
<a name="l00012"></a>00012 <span class="comment">// General Public License for more details.</span>
<a name="l00013"></a>00013 <span class="comment">// </span>
<a name="l00014"></a>00014 <span class="comment">// You should have received a copy of the GNU General Public License </span>
<a name="l00015"></a>00015 <span class="comment">// along with this program;if not, write to the </span>
<a name="l00016"></a>00016 <span class="comment">// </span>
<a name="l00017"></a>00017 <span class="comment">// Free Software Foundation, Inc., </span>
<a name="l00018"></a>00018 <span class="comment">// 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA</span>
<a name="l00019"></a>00019 <span class="comment">//</span><span class="comment"></span>
<a name="l00020"></a>00020 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00021"></a>00021 <span class="comment"></span>
<a name="l00022"></a>00022 <span class="comment">//</span>
<a name="l00023"></a>00023 <span class="preprocessor">#include &quot;<a class="code" href="SummaryGenotypeEngine_8h.html" title="Core routines for summary-genotype binaries. By separating the command line parsing form the computat...">chipstream/apt-summary-genotype/SummaryGenotypeEngine.h</a>&quot;</span>
<a name="l00024"></a>00024 <span class="comment">//</span>
<a name="l00025"></a>00025 <span class="preprocessor">#include &quot;chipstream/ChipLayout.h&quot;</span>
<a name="l00026"></a>00026 <span class="preprocessor">#include &quot;<a class="code" href="EngineUtil_8h.html" title="Some common functions that application executables make use of.">chipstream/EngineUtil.h</a>&quot;</span>
<a name="l00027"></a>00027 <span class="preprocessor">#include &quot;chipstream/FileGenders.h&quot;</span>
<a name="l00028"></a>00028 <span class="preprocessor">#include &quot;<a class="code" href="GenoSeedTxt_8h.html" title="Implements interface for a genotyping oracle used for genotyping seeds in brlmm. In this case the gen...">chipstream/GenoSeedTxt.h</a>&quot;</span>
<a name="l00029"></a>00029 <span class="preprocessor">#include &quot;<a class="code" href="PmOnlyAdjust_8h.html" title="Pm adjustment class that does no modification.">chipstream/PmOnlyAdjust.h</a>&quot;</span>
<a name="l00030"></a>00030 <span class="preprocessor">#include &quot;<a class="code" href="QuantLabelZ_8h.html">chipstream/QuantLabelZ.h</a>&quot;</span>
<a name="l00031"></a>00031 <span class="preprocessor">#include &quot;<a class="code" href="QuantMethodFactory_8h.html" title="Factory class for making chip streams based on a string representation.">chipstream/QuantMethodFactory.h</a>&quot;</span>
<a name="l00032"></a>00032 <span class="preprocessor">#include &quot;<a class="code" href="QuantMethodGTypeReport_8h.html" title="Reporter for outputting genotype results.">chipstream/QuantMethodGTypeReport.h</a>&quot;</span>
<a name="l00033"></a>00033 <span class="preprocessor">#include &quot;<a class="code" href="SelfDoc_8h.html" title="Small interface to for algorithmic classes that can explain themselves.">chipstream/SelfDoc.h</a>&quot;</span>
<a name="l00034"></a>00034 <span class="preprocessor">#include &quot;<a class="code" href="SparseMart_8h.html" title="Class for representing cel file intensity data in memory. Can save a lot of the space if not all of a...">chipstream/SparseMart.h</a>&quot;</span>
<a name="l00035"></a>00035 <span class="preprocessor">#include &quot;<a class="code" href="SpecialSnps_8h.html">chipstream/SpecialSnps.h</a>&quot;</span>
<a name="l00036"></a>00036 <span class="comment">//</span>
<a name="l00037"></a>00037 <span class="preprocessor">#include &quot;<a class="code" href="Fs_8h.html" title="///">util/Fs.h</a>&quot;</span>
<a name="l00038"></a>00038 <span class="preprocessor">#include &quot;<a class="code" href="PgOptions_8h.html" title="Classes for reading program command line options.">util/PgOptions.h</a>&quot;</span>
<a name="l00039"></a>00039 <span class="comment">//</span>
<a name="l00040"></a>00040 <span class="preprocessor">#include &lt;cstdio&gt;</span>
<a name="l00041"></a>00041 <span class="preprocessor">#include &lt;cstring&gt;</span>
<a name="l00042"></a>00042 <span class="preprocessor">#include &lt;ctime&gt;</span>
<a name="l00043"></a>00043 <span class="preprocessor">#include &lt;iostream&gt;</span>
<a name="l00044"></a>00044 <span class="preprocessor">#include &lt;map&gt;</span>
<a name="l00045"></a>00045 <span class="preprocessor">#include &lt;string&gt;</span>
<a name="l00046"></a>00046 <span class="preprocessor">#include &lt;vector&gt;</span>
<a name="l00047"></a>00047 
<a name="l00048"></a>00048 <span class="comment">//</span>
<a name="l00049"></a>00049 <span class="preprocessor">#ifndef WIN32</span>
<a name="l00050"></a>00050 <span class="preprocessor"></span><span class="preprocessor">#include &lt;unistd.h&gt;</span>
<a name="l00051"></a>00051 <span class="preprocessor">#endif </span><span class="comment">/* WIN32 */</span>
<a name="l00052"></a>00052 
<a name="l00053"></a>00053 <span class="keyword">using namespace </span>std;
<a name="l00054"></a>00054 <span class="keyword">using namespace </span>affx;
<a name="l00055"></a>00055 <span class="keyword">using namespace </span>affymetrix_calvin_exceptions;
<a name="l00056"></a>00056 
<a name="l00057"></a>00057 
<a name="l00058"></a><a class="code" href="classSgIoHelper.html#ada8461b373fce9553e01f5f946dfdd54">00058</a> <a class="code" href="classSgIoHelper.html#ada8461b373fce9553e01f5f946dfdd54" title="Constructor.">SgIoHelper::SgIoHelper</a>(<span class="keyword">const</span> std::string &amp;summariesFile,
<a name="l00059"></a>00059                    <span class="keyword">const</span> std::string &amp;summariesFileA, <span class="keyword">const</span> std::string &amp;summariesFileB,
<a name="l00060"></a>00060                    <span class="keyword">const</span> std::string &amp;file5,
<a name="l00061"></a>00061                    <span class="keyword">const</span> std::string &amp;file5Group) {
<a name="l00062"></a>00062   f5SumGroup= NULL;
<a name="l00063"></a>00063   f5SumTsv = NULL;
<a name="l00064"></a>00064   <span class="keywordflow">if</span>(summariesFile != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00065"></a>00065     <span class="keywordtype">string</span> label;
<a name="l00066"></a>00066     <span class="keywordflow">if</span>(summaries.<a class="code" href="classaffx_1_1TsvFile.html#a763ce0bbf19b4134d0445aa58e0f5011" title="Opens a file -- attempts to guess some defaults.">open</a>(summariesFile) != TSV_OK)
<a name="l00067"></a>00067       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Couldn&#39;t open file &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(summariesFile) + <span class="stringliteral">&quot; to read.&quot;</span>);
<a name="l00068"></a>00068     summaries.<a class="code" href="classaffx_1_1TsvFile.html#a0958f996be8dea8c5e5d2a0bac36819b" title="Get the name of a column by its level and index.">cidx2cname</a>(0,0,label);
<a name="l00069"></a>00069     <span class="keywordflow">if</span>(label != <span class="stringliteral">&quot;probeset_id&quot;</span>)
<a name="l00070"></a>00070       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;probeset_id must be the first column in the summaries file&quot;</span>);
<a name="l00071"></a>00071     <span class="keywordtype">int</span> numCol = summaries.<a class="code" href="classaffx_1_1TsvFile.html#a94ad901d12ab05cce0f050f99665970f" title="The number of columns for this level ///.">getColumnCount</a>(0);
<a name="l00072"></a>00072     <span class="keywordflow">for</span>(<span class="keywordtype">int</span> cidx=1; cidx &lt; numCol; cidx++) {
<a name="l00073"></a>00073       <span class="keywordtype">string</span> name;
<a name="l00074"></a>00074       summaries.<a class="code" href="classaffx_1_1TsvFile.html#a0958f996be8dea8c5e5d2a0bac36819b" title="Get the name of a column by its level and index.">cidx2cname</a>(0,cidx,name);
<a name="l00075"></a>00075       <span class="comment">// We need *both* sets of column names.</span>
<a name="l00076"></a>00076       m_ColNames.push_back(name);
<a name="l00077"></a>00077     }
<a name="l00078"></a>00078     
<a name="l00079"></a>00079   } 
<a name="l00080"></a>00080   <span class="keywordflow">else</span> <span class="keywordflow">if</span>(summariesFileA != <span class="stringliteral">&quot;&quot;</span> &amp;&amp; summariesFileB != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00081"></a>00081     <span class="keywordtype">string</span> label;
<a name="l00082"></a>00082     <span class="keywordflow">if</span>(summariesA.<a class="code" href="classaffx_1_1TsvFile.html#a763ce0bbf19b4134d0445aa58e0f5011" title="Opens a file -- attempts to guess some defaults.">open</a>(summariesFileA) != TSV_OK)
<a name="l00083"></a>00083       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Couldn&#39;t open file &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(summariesFileA) + <span class="stringliteral">&quot; to read.&quot;</span>);
<a name="l00084"></a>00084     summariesA.<a class="code" href="classaffx_1_1TsvFile.html#a0958f996be8dea8c5e5d2a0bac36819b" title="Get the name of a column by its level and index.">cidx2cname</a>(0,0,label);
<a name="l00085"></a>00085     <span class="keywordflow">if</span>(label != <span class="stringliteral">&quot;probeset_id&quot;</span>)
<a name="l00086"></a>00086       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;probeset_id must be the first column in the summaries file&quot;</span>);
<a name="l00087"></a>00087     <span class="keywordflow">if</span>(summariesB.<a class="code" href="classaffx_1_1TsvFile.html#a763ce0bbf19b4134d0445aa58e0f5011" title="Opens a file -- attempts to guess some defaults.">open</a>(summariesFileB) != TSV_OK)
<a name="l00088"></a>00088       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Couldn&#39;t open file &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(summariesFileB) + <span class="stringliteral">&quot; to read.&quot;</span>);
<a name="l00089"></a>00089     summariesB.<a class="code" href="classaffx_1_1TsvFile.html#a0958f996be8dea8c5e5d2a0bac36819b" title="Get the name of a column by its level and index.">cidx2cname</a>(0,0,label);
<a name="l00090"></a>00090     <span class="keywordflow">if</span>(label != <span class="stringliteral">&quot;probeset_id&quot;</span>)
<a name="l00091"></a>00091       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;probeset_id must be the first column in the summaries file&quot;</span>);
<a name="l00092"></a>00092     <span class="keywordtype">int</span> numCol = summariesA.<a class="code" href="classaffx_1_1TsvFile.html#a94ad901d12ab05cce0f050f99665970f" title="The number of columns for this level ///.">getColumnCount</a>(0);
<a name="l00093"></a>00093     <span class="keywordtype">int</span> numColB = summariesB.<a class="code" href="classaffx_1_1TsvFile.html#a94ad901d12ab05cce0f050f99665970f" title="The number of columns for this level ///.">getColumnCount</a>(0);
<a name="l00094"></a>00094     <span class="keywordflow">if</span>(numCol != numColB)
<a name="l00095"></a>00095       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;allele A and allele B signal files do not have the same number of columns&quot;</span>);
<a name="l00096"></a>00096       
<a name="l00097"></a>00097     <span class="comment">//</span>
<a name="l00098"></a>00098     <span class="keywordflow">for</span>(<span class="keywordtype">int</span> cidx=1; cidx &lt; numCol; cidx++) {
<a name="l00099"></a>00099       <span class="keywordtype">string</span> name;
<a name="l00100"></a>00100       <span class="keywordtype">string</span> nameB;
<a name="l00101"></a>00101       summariesA.<a class="code" href="classaffx_1_1TsvFile.html#a0958f996be8dea8c5e5d2a0bac36819b" title="Get the name of a column by its level and index.">cidx2cname</a>(0,cidx,name);
<a name="l00102"></a>00102       summariesB.<a class="code" href="classaffx_1_1TsvFile.html#a0958f996be8dea8c5e5d2a0bac36819b" title="Get the name of a column by its level and index.">cidx2cname</a>(0,cidx,nameB);
<a name="l00103"></a>00103       <span class="keywordflow">if</span>(name != nameB)
<a name="l00104"></a>00104         <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Sample names do not match between allele A and B signal files.&quot;</span>);
<a name="l00105"></a>00105       <span class="comment">// We need *both* sets of column names.</span>
<a name="l00106"></a>00106       m_ColNames.push_back(name);
<a name="l00107"></a>00107     }
<a name="l00108"></a>00108       
<a name="l00109"></a>00109   } 
<a name="l00110"></a>00110   <span class="keywordflow">else</span> <span class="keywordflow">if</span>(file5 != <span class="stringliteral">&quot;&quot;</span> &amp;&amp; file5Group != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00111"></a>00111     <span class="keywordtype">int</span> status = f5Summaries.open(file5, affx::FILE5_OPEN_RO);
<a name="l00112"></a>00112     <span class="keywordflow">if</span>(status != FILE5_OK) {
<a name="l00113"></a>00113       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Can&#39;t open: &quot;</span> + file5);
<a name="l00114"></a>00114     }
<a name="l00115"></a>00115     f5SumGroup = f5Summaries.openGroup(file5Group, affx::FILE5_OPEN);
<a name="l00116"></a>00116     f5SumTsv = f5SumGroup-&gt;openTsv(<span class="stringliteral">&quot;summaries&quot;</span>, affx::FILE5_OPEN_RO);
<a name="l00117"></a>00117     <span class="keywordtype">int</span> columnCount = f5SumTsv-&gt;getColumnCount(0);
<a name="l00118"></a>00118     <span class="keywordtype">string</span> name;
<a name="l00119"></a>00119     <span class="keywordflow">for</span>(<span class="keywordtype">int</span> i = 1; i &lt; columnCount; i++) {
<a name="l00120"></a>00120       f5SumTsv-&gt;getColumnName(0,i,&amp;name);
<a name="l00121"></a>00121       m_ColNames.push_back(name);
<a name="l00122"></a>00122     }
<a name="l00123"></a>00123   }
<a name="l00124"></a>00124   <span class="keywordflow">else</span> {
<a name="l00125"></a>00125     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Must provide either an interleaved summary file or separate allele A and B summary files.&quot;</span>);
<a name="l00126"></a>00126   }
<a name="l00127"></a>00127 }
<a name="l00128"></a>00128 
<a name="l00129"></a><a class="code" href="classSgIoHelper.html#a5d5c5282731c23dd363325f764bbd9ad">00129</a> <a class="code" href="classSgIoHelper.html#a5d5c5282731c23dd363325f764bbd9ad" title="Destructor.">SgIoHelper::~SgIoHelper</a>() {
<a name="l00130"></a>00130   <span class="keywordflow">if</span>(f5SumTsv != NULL) {
<a name="l00131"></a>00131     f5SumTsv-&gt;close();
<a name="l00132"></a>00132     <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(f5SumTsv);
<a name="l00133"></a>00133   }
<a name="l00134"></a>00134   <span class="keywordflow">if</span>(f5SumGroup != NULL) {
<a name="l00135"></a>00135     f5SumGroup-&gt;close();
<a name="l00136"></a>00136     <a class="code" href="Util_8h.html#ad0787eda9709999bbd42dc11b21deb0d" title="delete function that deletes the pointer and sets it to NULL.">Freez</a>(f5SumGroup);
<a name="l00137"></a>00137   }
<a name="l00138"></a>00138   f5Summaries.close();
<a name="l00139"></a>00139 }
<a name="l00140"></a>00140 
<a name="l00141"></a>00141 
<a name="l00142"></a><a class="code" href="classSgIoHelper.html#a9ba39a7cfde128d845c2fdad58631be8">00142</a> <span class="keywordtype">void</span> <a class="code" href="classSgIoHelper.html#a9ba39a7cfde128d845c2fdad58631be8" title="Read in a single line from out summaries file and fill label and values.">SgIoHelper::getLine</a>(<span class="keywordtype">string</span> &amp;label, vector&lt;double&gt; &amp;vals, <a class="code" href="classaffx_1_1TsvFile.html" title="A class for reading and writing Tab Seperated Value (TSV) files. /// See the TsvFile format document ...">TsvFile</a> &amp;summaries, vector&lt;string&gt; &amp;colNames) {
<a name="l00143"></a>00143   vals.clear();
<a name="l00144"></a>00144   <span class="keywordflow">for</span>(<span class="keywordtype">int</span> i=0; i&lt;colNames.size(); i++) {
<a name="l00145"></a>00145     <span class="keywordtype">double</span> val;
<a name="l00146"></a>00146     <span class="keywordflow">if</span>(summaries.<a class="code" href="classaffx_1_1TsvFile.html#a7a81092ce134e8c9116a1a5bcfe287d1" title="Get the string value of a field ///.">get</a>(0,i+1,val)==TSV_OK)
<a name="l00147"></a>00147       vals.push_back(val);
<a name="l00148"></a>00148     <span class="keywordflow">else</span>
<a name="l00149"></a>00149         <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Unable to read column &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(i+1) + <span class="stringliteral">&quot; from summaries file.&quot;</span>);
<a name="l00150"></a>00150   }
<a name="l00151"></a>00151   
<a name="l00152"></a>00152   <span class="keywordflow">if</span>(summaries.<a class="code" href="classaffx_1_1TsvFile.html#a7a81092ce134e8c9116a1a5bcfe287d1" title="Get the string value of a field ///.">get</a>(0,0,label)!=TSV_OK)
<a name="l00153"></a>00153     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Unable ro read label from summaries file!&quot;</span>);
<a name="l00154"></a>00154 }
<a name="l00155"></a>00155 
<a name="l00156"></a><a class="code" href="classSgIoHelper.html#a47053a0e6dbbe864186cd720d597ec79">00156</a> <span class="keywordtype">void</span> <a class="code" href="classSgIoHelper.html#a47053a0e6dbbe864186cd720d597ec79" title="Read in a single line from out summaries file5 and fill label and values.">SgIoHelper::getFile5Line</a>(<span class="keywordtype">string</span> &amp;label, vector&lt;double&gt; &amp;vals, <a class="code" href="classaffx_1_1File5__Tsv.html">File5_Tsv</a> *summaries, vector&lt;string&gt; &amp;colNames) {
<a name="l00157"></a>00157   vals.clear();
<a name="l00158"></a>00158   <span class="keywordflow">for</span>(<span class="keywordtype">int</span> i=0; i&lt;colNames.size(); i++) {
<a name="l00159"></a>00159     <span class="keywordtype">double</span> val;
<a name="l00160"></a>00160     <span class="keywordflow">if</span>(summaries-&gt;get(0,i+1,&amp;val)==FILE5_OK)
<a name="l00161"></a>00161       vals.push_back(val);
<a name="l00162"></a>00162     <span class="keywordflow">else</span>
<a name="l00163"></a>00163       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Unable to read column &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(i+1) + <span class="stringliteral">&quot; from summaries file.&quot;</span>);
<a name="l00164"></a>00164   }
<a name="l00165"></a>00165   
<a name="l00166"></a>00166   <span class="keywordflow">if</span>(summaries-&gt;get(0,0,&amp;label) != FILE5_OK)
<a name="l00167"></a>00167     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Unable ro read label from summaries file!&quot;</span>);
<a name="l00168"></a>00168 }
<a name="l00169"></a>00169 
<a name="l00170"></a><a class="code" href="classSgIoHelper.html#aa6f292bd628e59ca57868943a0a3ca62">00170</a> <span class="keywordtype">bool</span> <a class="code" href="classSgIoHelper.html#aa6f292bd628e59ca57868943a0a3ca62" title="Read the next snp marker from the single interleved file5 summary file.">SgIoHelper::getNextMarkerTsvSingle</a>(vector&lt;string&gt; &amp;colNames, vector&lt;double&gt; &amp;aVals, 
<a name="l00171"></a>00171   vector&lt;double&gt; &amp;bVals, <span class="keywordtype">string</span> &amp;probesetName) {
<a name="l00172"></a>00172   <span class="keywordflow">if</span>(summaries.<a class="code" href="classaffx_1_1TsvFile.html#a2022895d082d1d635e7f59fccab12afb" title="Skip to the next level of data which matches seek_clvl ///.">nextLevel</a>(0)==TSV_OK) {
<a name="l00173"></a>00173     <span class="keywordtype">string</span> label, alleleName;
<a name="l00174"></a>00174     
<a name="l00175"></a>00175     <span class="comment">// parse the &quot;A&quot; allele</span>
<a name="l00176"></a>00176     <span class="keywordtype">string</span> aProbesetName, aProbesetAllele;
<a name="l00177"></a>00177     <a class="code" href="classSgIoHelper.html#a9ba39a7cfde128d845c2fdad58631be8" title="Read in a single line from out summaries file and fill label and values.">getLine</a>(label, aVals, summaries, colNames);
<a name="l00178"></a>00178     aProbesetName = label.substr(0, label.rfind(<span class="charliteral">&#39;-&#39;</span>));
<a name="l00179"></a>00179     alleleName = label.substr(label.rfind(<span class="charliteral">&#39;-&#39;</span>)+1, label.size());
<a name="l00180"></a>00180     <span class="keywordflow">if</span>(alleleName != <span class="stringliteral">&quot;A&quot;</span>)
<a name="l00181"></a>00181       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Expecting to see summaries for allele A first for &quot;</span> + aProbesetName);
<a name="l00182"></a>00182     
<a name="l00183"></a>00183     <span class="comment">// parse the &quot;B&quot; allele</span>
<a name="l00184"></a>00184     <span class="keywordflow">if</span>(!(summaries.<a class="code" href="classaffx_1_1TsvFile.html#a2022895d082d1d635e7f59fccab12afb" title="Skip to the next level of data which matches seek_clvl ///.">nextLevel</a>(0)==TSV_OK)) {
<a name="l00185"></a>00185       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Unable to read Allele B for probeset &quot;</span> + aProbesetName);
<a name="l00186"></a>00186     }
<a name="l00187"></a>00187     <span class="keywordtype">string</span> bProbesetName, bProbesetAllele;
<a name="l00188"></a>00188     <a class="code" href="classSgIoHelper.html#a9ba39a7cfde128d845c2fdad58631be8" title="Read in a single line from out summaries file and fill label and values.">getLine</a>(label, bVals, summaries, colNames);
<a name="l00189"></a>00189     bProbesetName = label.substr(0, label.rfind(<span class="charliteral">&#39;-&#39;</span>));
<a name="l00190"></a>00190     alleleName = label.substr(label.rfind(<span class="charliteral">&#39;-&#39;</span>)+1, label.size());
<a name="l00191"></a>00191     <span class="keywordflow">if</span>(alleleName != <span class="stringliteral">&quot;B&quot;</span>)
<a name="l00192"></a>00192       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Expecting to see summaries for allele B second for &quot;</span> + aProbesetName);
<a name="l00193"></a>00193     
<a name="l00194"></a>00194     <span class="comment">// check that we have the right pair</span>
<a name="l00195"></a>00195     <span class="keywordflow">if</span>(aProbesetName != bProbesetName)
<a name="l00196"></a>00196       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Allele summaries do not pair up: &quot;</span> + aProbesetName + <span class="stringliteral">&quot; != &quot;</span> + bProbesetName);
<a name="l00197"></a>00197     probesetName = aProbesetName;
<a name="l00198"></a>00198     
<a name="l00199"></a>00199     <span class="keywordflow">return</span> <span class="keyword">true</span>;
<a name="l00200"></a>00200   } 
<a name="l00201"></a>00201   <span class="keywordflow">return</span> <span class="keyword">false</span>;
<a name="l00202"></a>00202 }
<a name="l00203"></a>00203 
<a name="l00204"></a><a class="code" href="classSgIoHelper.html#a48083af830fe8c536bcb37838bd4e52b">00204</a> <span class="keywordtype">bool</span> <a class="code" href="classSgIoHelper.html#a48083af830fe8c536bcb37838bd4e52b" title="Read the next snp marker from the file5.">SgIoHelper::getNextMarkerFile5</a>(vector&lt;string&gt; &amp;colNames, vector&lt;double&gt; &amp;aVals, 
<a name="l00205"></a>00205                                   vector&lt;double&gt; &amp;bVals, <span class="keywordtype">string</span> &amp;probesetName) {
<a name="l00206"></a>00206   <span class="keywordflow">if</span>(f5SumTsv-&gt;nextLevel(0)==FILE5_OK) {
<a name="l00207"></a>00207     <span class="keywordtype">string</span> label, alleleName;
<a name="l00208"></a>00208     
<a name="l00209"></a>00209     <span class="comment">// parse the &quot;A&quot; allele</span>
<a name="l00210"></a>00210     <span class="keywordtype">string</span> aProbesetName, aProbesetAllele;
<a name="l00211"></a>00211     <a class="code" href="classSgIoHelper.html#a47053a0e6dbbe864186cd720d597ec79" title="Read in a single line from out summaries file5 and fill label and values.">getFile5Line</a>(label, aVals, f5SumTsv, colNames);
<a name="l00212"></a>00212     aProbesetName = label.substr(0, label.rfind(<span class="charliteral">&#39;-&#39;</span>));
<a name="l00213"></a>00213     alleleName = label.substr(label.rfind(<span class="charliteral">&#39;-&#39;</span>)+1, label.size());
<a name="l00214"></a>00214     <span class="keywordflow">if</span>(alleleName != <span class="stringliteral">&quot;A&quot;</span>)
<a name="l00215"></a>00215       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Expecting to see summaries for allele A first for &quot;</span> + aProbesetName);
<a name="l00216"></a>00216       
<a name="l00217"></a>00217     <span class="comment">// parse the &quot;B&quot; allele</span>
<a name="l00218"></a>00218     <span class="keywordflow">if</span>(!(f5SumTsv-&gt;nextLevel(0)==FILE5_OK)) {
<a name="l00219"></a>00219       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Unable to read Allele B for probeset &quot;</span> + aProbesetName);
<a name="l00220"></a>00220     }
<a name="l00221"></a>00221     <span class="keywordtype">string</span> bProbesetName, bProbesetAllele;
<a name="l00222"></a>00222     <a class="code" href="classSgIoHelper.html#a47053a0e6dbbe864186cd720d597ec79" title="Read in a single line from out summaries file5 and fill label and values.">getFile5Line</a>(label, bVals, f5SumTsv, colNames);
<a name="l00223"></a>00223     bProbesetName = label.substr(0, label.rfind(<span class="charliteral">&#39;-&#39;</span>));
<a name="l00224"></a>00224     alleleName = label.substr(label.rfind(<span class="charliteral">&#39;-&#39;</span>)+1, label.size());
<a name="l00225"></a>00225     <span class="keywordflow">if</span>(alleleName != <span class="stringliteral">&quot;B&quot;</span>)
<a name="l00226"></a>00226       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Expecting to see summaries for allele B second for &quot;</span> + aProbesetName);
<a name="l00227"></a>00227       
<a name="l00228"></a>00228     <span class="comment">// check that we have the right pair</span>
<a name="l00229"></a>00229     <span class="keywordflow">if</span>(aProbesetName != bProbesetName)
<a name="l00230"></a>00230       <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Allele summaries do not pair up: &quot;</span> + aProbesetName + <span class="stringliteral">&quot; != &quot;</span> + bProbesetName);
<a name="l00231"></a>00231     probesetName = aProbesetName;
<a name="l00232"></a>00232       
<a name="l00233"></a>00233     <span class="keywordflow">return</span> <span class="keyword">true</span>;
<a name="l00234"></a>00234   } 
<a name="l00235"></a>00235   <span class="keywordflow">return</span> <span class="keyword">false</span>;
<a name="l00236"></a>00236 }
<a name="l00237"></a>00237 
<a name="l00238"></a><a class="code" href="classSgIoHelper.html#a28cbaf1a92be7f3fb6ddfa90c5551e69">00238</a> <span class="keywordtype">bool</span> <a class="code" href="classSgIoHelper.html#a28cbaf1a92be7f3fb6ddfa90c5551e69" title="Read the next snp marker from the paired summary A &amp;amp; B files.">SgIoHelper::getNextMarkerTwoTsv</a>(vector&lt;string&gt; &amp;colNames, vector&lt;double&gt; &amp;aVals, 
<a name="l00239"></a>00239                                    vector&lt;double&gt; &amp;bVals, <span class="keywordtype">string</span> &amp;probesetName) {
<a name="l00240"></a>00240   <span class="keywordflow">if</span>(summariesA.<a class="code" href="classaffx_1_1TsvFile.html#a2022895d082d1d635e7f59fccab12afb" title="Skip to the next level of data which matches seek_clvl ///.">nextLevel</a>(0)!=TSV_OK)
<a name="l00241"></a>00241     <span class="keywordflow">return</span> <span class="keyword">false</span>;
<a name="l00242"></a>00242   <span class="keywordflow">if</span>(summariesB.<a class="code" href="classaffx_1_1TsvFile.html#a2022895d082d1d635e7f59fccab12afb" title="Skip to the next level of data which matches seek_clvl ///.">nextLevel</a>(0)!=TSV_OK)
<a name="l00243"></a>00243     <span class="keywordflow">return</span> <span class="keyword">false</span>;
<a name="l00244"></a>00244     
<a name="l00245"></a>00245   <span class="keywordtype">string</span> labelA, labelB;
<a name="l00246"></a>00246   <a class="code" href="classSgIoHelper.html#a9ba39a7cfde128d845c2fdad58631be8" title="Read in a single line from out summaries file and fill label and values.">getLine</a>(labelA, aVals, summariesA, colNames);
<a name="l00247"></a>00247   <a class="code" href="classSgIoHelper.html#a9ba39a7cfde128d845c2fdad58631be8" title="Read in a single line from out summaries file and fill label and values.">getLine</a>(labelB, bVals, summariesB, colNames);
<a name="l00248"></a>00248   <span class="keywordflow">if</span>(labelA != labelB)
<a name="l00249"></a>00249     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Probeset mismatch between allele signal files.&quot;</span>);
<a name="l00250"></a>00250   probesetName = labelA;
<a name="l00251"></a>00251   <span class="keywordflow">return</span> <span class="keyword">true</span>;
<a name="l00252"></a>00252 }
<a name="l00253"></a>00253 
<a name="l00254"></a><a class="code" href="classSgIoHelper.html#a6e91bab82eb0d030f0343d8b81c9e948">00254</a> <span class="keywordtype">bool</span> <a class="code" href="classSgIoHelper.html#a6e91bab82eb0d030f0343d8b81c9e948" title="Read the next snp marker.">SgIoHelper::getNextMarker</a>(vector&lt;string&gt; &amp;colNames, vector&lt;double&gt; &amp;aVals, 
<a name="l00255"></a>00255                    vector&lt;double&gt; &amp;bVals, <span class="keywordtype">string</span> &amp;probesetName) {
<a name="l00256"></a>00256   <span class="keywordflow">if</span>(summaries.is_open()) {
<a name="l00257"></a>00257     <span class="keywordflow">return</span> <a class="code" href="classSgIoHelper.html#aa6f292bd628e59ca57868943a0a3ca62" title="Read the next snp marker from the single interleved file5 summary file.">getNextMarkerTsvSingle</a>(colNames, aVals, bVals, probesetName);
<a name="l00258"></a>00258   } 
<a name="l00259"></a>00259   <span class="keywordflow">else</span> <span class="keywordflow">if</span>(summariesA.is_open() &amp;&amp; summariesB.is_open()) {
<a name="l00260"></a>00260     <span class="keywordflow">return</span> <a class="code" href="classSgIoHelper.html#a28cbaf1a92be7f3fb6ddfa90c5551e69" title="Read the next snp marker from the paired summary A &amp;amp; B files.">getNextMarkerTwoTsv</a>(colNames, aVals, bVals, probesetName);
<a name="l00261"></a>00261   }
<a name="l00262"></a>00262   <span class="keywordflow">else</span> <span class="keywordflow">if</span>(f5Summaries.is_open() &amp;&amp; f5SumTsv != NULL) {
<a name="l00263"></a>00263     <span class="keywordflow">return</span> <a class="code" href="classSgIoHelper.html#a48083af830fe8c536bcb37838bd4e52b" title="Read the next snp marker from the file5.">getNextMarkerFile5</a>(colNames, aVals, bVals, probesetName);
<a name="l00264"></a>00264   }
<a name="l00265"></a>00265   <span class="keywordflow">else</span> {
<a name="l00266"></a>00266     <a class="code" href="classErr.html#a121c0cbb1687c93444a1b5f7f6ea56af" title="Print the message supplied and abort program.">Err::errAbort</a>(<span class="stringliteral">&quot;Couldn&#39;t get next marker.&quot;</span>);
<a name="l00267"></a>00267   }
<a name="l00268"></a>00268   <span class="keywordflow">return</span> <span class="keyword">false</span>; <span class="comment">// Should never get here, just for compiler</span>
<a name="l00269"></a>00269 }
<a name="l00270"></a>00270 
<a name="l00271"></a>00271 
<a name="l00272"></a>00272 <a class="code" href="classSummaryGenotypeEngine_1_1Reg.html">SummaryGenotypeEngine::Reg</a> <a class="code" href="classSummaryGenotypeEngine.html#a9be8a8609608dfd04a2bdf21ee8dfcfb">SummaryGenotypeEngine::reg</a>;
<a name="l00273"></a>00273 
<a name="l00274"></a><a class="code" href="classSummaryGenotypeEngine.html#adb579494c2ccfed76de0995f1c28f940">00274</a> <a class="code" href="classSummaryGenotypeEngine.html">SummaryGenotypeEngine</a> * <a class="code" href="classSummaryGenotypeEngine.html#adb579494c2ccfed76de0995f1c28f940">SummaryGenotypeEngine::FromBase</a>(<a class="code" href="classBaseEngine.html" title="Base class for analysis engines.">BaseEngine</a> *engine)
<a name="l00275"></a>00275 {
<a name="l00276"></a>00276         <span class="keywordflow">if</span> (engine != NULL &amp;&amp; engine-&gt;getEngineName() == SummaryGenotypeEngine::EngineName())
<a name="l00277"></a>00277                 <span class="keywordflow">return</span> (<a class="code" href="classSummaryGenotypeEngine.html">SummaryGenotypeEngine</a> *)engine;
<a name="l00278"></a>00278         <span class="keywordflow">return</span> NULL;
<a name="l00279"></a>00279 }
<a name="l00280"></a>00280 
<a name="l00281"></a><a class="code" href="classSummaryGenotypeEngine.html#a0aa8c15fb7f3c1fe92a8b04fd245a4b9">00281</a> <a class="code" href="classSummaryGenotypeEngine.html#a0aa8c15fb7f3c1fe92a8b04fd245a4b9" title="Constructor.">SummaryGenotypeEngine::SummaryGenotypeEngine</a>() {
<a name="l00282"></a>00282     defineOptions();
<a name="l00283"></a>00283     defineStates();
<a name="l00284"></a>00284 }
<a name="l00285"></a>00285 
<a name="l00286"></a><a class="code" href="classSummaryGenotypeEngine.html#ab8b78f7da6b728a1ae5a3f626d8700b3">00286</a> <a class="code" href="classSummaryGenotypeEngine.html#ab8b78f7da6b728a1ae5a3f626d8700b3" title="Destructor.">SummaryGenotypeEngine::~SummaryGenotypeEngine</a>() {
<a name="l00287"></a>00287 }
<a name="l00288"></a>00288 
<a name="l00289"></a>00289 <span class="keywordtype">void</span> SummaryGenotypeEngine::defineOptions() {
<a name="l00290"></a>00290         defineOption(<span class="stringliteral">&quot;p&quot;</span>, <span class="stringliteral">&quot;algo-params&quot;</span>, PgOpt::STRING_OPT,
<a name="l00291"></a>00291                 <span class="stringliteral">&quot;A string specifying algo parameters. See --explain option for acceptable parameters.&quot;</span>,
<a name="l00292"></a>00292                 <span class="stringliteral">&quot;&quot;</span>);
<a name="l00293"></a>00293         defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;file5-summaries&quot;</span>, PgOpt::STRING_OPT,
<a name="l00294"></a>00294                 <span class="stringliteral">&quot;Interleaved Allele summaries file in file5 format with summaries in /summaries group.&quot;</span>,
<a name="l00295"></a>00295                 <span class="stringliteral">&quot;&quot;</span>);
<a name="l00296"></a>00296         defineOption(<span class="stringliteral">&quot;s&quot;</span>, <span class="stringliteral">&quot;summaries&quot;</span>, PgOpt::STRING_OPT,
<a name="l00297"></a>00297                 <span class="stringliteral">&quot;Interleaved Allele summaries file.&quot;</span>,
<a name="l00298"></a>00298                 <span class="stringliteral">&quot;&quot;</span>);
<a name="l00299"></a>00299         defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;summaries-a&quot;</span>, PgOpt::STRING_OPT,
<a name="l00300"></a>00300                 <span class="stringliteral">&quot;Allele summarie file for A allele.&quot;</span>,
<a name="l00301"></a>00301                 <span class="stringliteral">&quot;&quot;</span>);
<a name="l00302"></a>00302         defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;summaries-b&quot;</span>, PgOpt::STRING_OPT,
<a name="l00303"></a>00303                 <span class="stringliteral">&quot;Allele summarie file for B allele.&quot;</span>,
<a name="l00304"></a>00304                 <span class="stringliteral">&quot;&quot;</span>);
<a name="l00305"></a>00305         defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;file5-output&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00306"></a>00306                 <span class="stringliteral">&quot;Should output be formatted in file5 (hdf5) output.&quot;</span>,
<a name="l00307"></a>00307                 <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00308"></a>00308         defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;chrX-snps&quot;</span>, PgOpt::STRING_OPT,
<a name="l00309"></a>00309                 <span class="stringliteral">&quot;File containing snps on chrX (non-pseudoautosomal region).&quot;</span>,
<a name="l00310"></a>00310                 <span class="stringliteral">&quot;&quot;</span>);
<a name="l00311"></a>00311         defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;special-snps&quot;</span>,PgOpt::STRING_OPT,
<a name="l00312"></a>00312                 <span class="stringliteral">&quot;File containing all snps of unusual copy (chrX,mito,Y)&quot;</span>,
<a name="l00313"></a>00313                 <span class="stringliteral">&quot;&quot;</span>);
<a name="l00314"></a>00314         defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;special-sample-snps&quot;</span>,PgOpt::STRING_OPT,
<a name="l00315"></a>00315                 <span class="stringliteral">&quot;File containing all sample-specific snps of unusual copy&quot;</span>,
<a name="l00316"></a>00316                 <span class="stringliteral">&quot;&quot;</span>);
<a name="l00317"></a>00317         defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;write-models&quot;</span>, PgOpt::BOOL_OPT,
<a name="l00318"></a>00318                 <span class="stringliteral">&quot;Should we write snp specific models out for analysis? [experimental]&quot;</span>,
<a name="l00319"></a>00319                 <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00320"></a>00320         defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;read-models-brlmmp&quot;</span>, PgOpt::STRING_OPT,
<a name="l00321"></a>00321                 <span class="stringliteral">&quot;File to read precomputed BRLMM-P snp specific models from.&quot;</span>,
<a name="l00322"></a>00322                 <span class="stringliteral">&quot;&quot;</span>);
<a name="l00323"></a>00323         defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;read-genders&quot;</span>, PgOpt::STRING_OPT,
<a name="l00324"></a>00324                 <span class="stringliteral">&quot;Explicitly read genders from a file.&quot;</span>,
<a name="l00325"></a>00325                 <span class="stringliteral">&quot;&quot;</span>);
<a name="l00326"></a>00326         defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;genotypes&quot;</span>, PgOpt::STRING_OPT,
<a name="l00327"></a>00327                 <span class="stringliteral">&quot;File to read seed genotypes from.&quot;</span>,
<a name="l00328"></a>00328                 <span class="stringliteral">&quot;&quot;</span>);
<a name="l00329"></a>00329         defineOption(<span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;select-probes&quot;</span>,PgOpt::BOOL_OPT,
<a name="l00330"></a>00330                 <span class="stringliteral">&quot;Should we evaluate model fit for clusters by probe-set/probe[experimental]&quot;</span>,
<a name="l00331"></a>00331                 <span class="stringliteral">&quot;false&quot;</span>);
<a name="l00332"></a>00332         defineOption(<span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;set-analysis-name&quot;</span>, PgOpt::STRING_OPT,
<a name="l00333"></a>00333         <span class="stringliteral">&quot;Explicitly set the analysis name. This affects output file names (ie prefix) and various meta info.&quot;</span>,
<a name="l00334"></a>00334                 <span class="stringliteral">&quot;&quot;</span>);
<a name="l00335"></a>00335 }
<a name="l00336"></a>00336 
<a name="l00337"></a>00337 <span class="keywordtype">void</span> SummaryGenotypeEngine::defineStates() { }
<a name="l00338"></a>00338 
<a name="l00339"></a>00339 <span class="comment"></span>
<a name="l00340"></a>00340 <span class="comment">/**</span>
<a name="l00341"></a>00341 <span class="comment"> * Make sure that our options are sane. Call Err::errAbort if not.</span>
<a name="l00342"></a>00342 <span class="comment"> */</span>
<a name="l00343"></a>00343 <span class="keywordtype">void</span> SummaryGenotypeEngine::checkOptionsImp()
<a name="l00344"></a>00344 {
<a name="l00345"></a>00345     defineStates();
<a name="l00346"></a>00346 
<a name="l00347"></a>00347         <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;chrX-snps&quot;</span>);
<a name="l00348"></a>00348         <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;special-snps&quot;</span>);
<a name="l00349"></a>00349         <a class="code" href="classBaseEngine.html#a90a5cba41ed5a3fe1b2fb9dc2896c696" title="Swap the option value to include the full path if found.">setLibFileOpt</a>(<span class="stringliteral">&quot;read-models-brlmmp&quot;</span>);
<a name="l00350"></a>00350 
<a name="l00351"></a>00351         <span class="keywordtype">string</span> outDir = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;out-dir&quot;</span>);
<a name="l00352"></a>00352         <span class="keywordflow">if</span>(!Fs::isWriteableDir(outDir)) {
<a name="l00353"></a>00353           <span class="keywordflow">if</span>(<a class="code" href="classFs.html#aea474e8df7665f070c5334fa322bd283" title="make all path of directories ///">Fs::mkdirPath</a>(outDir, <span class="keyword">false</span>) != APT_OK) {
<a name="l00354"></a>00354             <a class="code" href="Err_8h.html#a7e26a673d8901b5bc880dfab285f5309" title="Calls Err::apt_err_abort with the filename and linenumber set.">APT_ERR_ABORT</a>(<span class="stringliteral">&quot;Can&#39;t make or write to directory: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(outDir));
<a name="l00355"></a>00355           }
<a name="l00356"></a>00356         }
<a name="l00357"></a>00357 }
<a name="l00358"></a>00358 <span class="comment"></span>
<a name="l00359"></a>00359 <span class="comment">/**</span>
<a name="l00360"></a>00360 <span class="comment">   This is the &quot;main()&quot; equivalent of the engine. This function has</span>
<a name="l00361"></a>00361 <span class="comment">   two main phases: 1) Creation an initialization of AnalysisStream</span>
<a name="l00362"></a>00362 <span class="comment">   and associated objects. 2) Looping through the list of cdf</span>
<a name="l00363"></a>00363 <span class="comment">   probesets to do the genotyping calls.</span>
<a name="l00364"></a>00364 <span class="comment"></span>
<a name="l00365"></a>00365 <span class="comment">   Phase 2 is definitely where most of the time and cpu cycles are</span>
<a name="l00366"></a>00366 <span class="comment">   spent. The list of things that happens for each iteration of phase</span>
<a name="l00367"></a>00367 <span class="comment">   2 is:</span>
<a name="l00368"></a>00368 <span class="comment">   - Read in cel files only keeping data needed for current iteration in RAM.</span>
<a name="l00369"></a>00369 <span class="comment">   - While reading cel files make DM calls.</span>
<a name="l00370"></a>00370 <span class="comment">   - Once cel files are read loop through probesets making BRLMM genotype</span>
<a name="l00371"></a>00371 <span class="comment">   calls.</span>
<a name="l00372"></a>00372 <span class="comment"></span>
<a name="l00373"></a>00373 <span class="comment">   Special things that happen in the first iteration of phase 2:</span>
<a name="l00374"></a>00374 <span class="comment">   - Calculate prior if desired.</span>
<a name="l00375"></a>00375 <span class="comment">   - Determine gender using DM calls if desired.</span>
<a name="l00376"></a>00376 <span class="comment">   - Learn chipstream parameters necessary.</span>
<a name="l00377"></a>00377 <span class="comment">   - Attach any reporters to the analysis streams.</span>
<a name="l00378"></a>00378 <span class="comment"></span>
<a name="l00379"></a>00379 <span class="comment">   Top of the call stack... This function is getting to be a real</span>
<a name="l00380"></a>00380 <span class="comment">   beast, time for some refactoring...</span>
<a name="l00381"></a>00381 <span class="comment">*/</span>
<a name="l00382"></a>00382 <span class="keywordtype">void</span> SummaryGenotypeEngine::runImp()
<a name="l00383"></a>00383 {
<a name="l00384"></a>00384     <span class="comment">// dummy stuff needed to spoof full analysis stream</span>
<a name="l00385"></a>00385     <a class="code" href="classChipLayout.html" title="ChipLayout - Data structure to represent the probesets encoded in the CDF, SPF or PGF files which def...">ChipLayout</a> layout; 
<a name="l00386"></a>00386     std::vector&lt;probeidx_t&gt; analysisOrder;
<a name="l00387"></a>00387     std::vector&lt;ChipStream *&gt; iTrans;
<a name="l00388"></a>00388     <a class="code" href="classPmOnlyAdjust.html" title="Pm adjustment class that does no modification.">PmOnlyAdjust</a> pmAdjust;
<a name="l00389"></a>00389     <a class="code" href="classProbeSetGroup.html" title="Group of probe sets that should be processed as a single large probe set.">ProbeSetGroup</a> psGroup;
<a name="l00390"></a>00390 
<a name="l00391"></a>00391         <span class="keywordtype">string</span> outDir = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;out-dir&quot;</span>);
<a name="l00392"></a>00392         <a class="code" href="classUtil.html#a21a08a463a4b0b321b58979b0404e017" title="Chop off the last character if it is a path separator.">Util::chompLastIfSep</a>(outDir);
<a name="l00393"></a>00393         <span class="keywordtype">string</span> summariesFile = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;summaries&quot;</span>);
<a name="l00394"></a>00394         <span class="keywordtype">string</span> summariesFileA = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;summaries-a&quot;</span>);
<a name="l00395"></a>00395         <span class="keywordtype">string</span> summariesFileB = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;summaries-b&quot;</span>);
<a name="l00396"></a>00396         <span class="keywordtype">string</span> quantMethod = <span class="stringliteral">&quot;brlmm-p&quot;</span>;
<a name="l00397"></a>00397         <span class="keywordtype">string</span> algoParams = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;algo-params&quot;</span>);
<a name="l00398"></a>00398         <span class="keywordtype">bool</span> selectProbes = <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;select-probes&quot;</span>);
<a name="l00399"></a>00399         <span class="keywordtype">string</span> specialSnpsFile = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;special-snps&quot;</span>);
<a name="l00400"></a>00400         <span class="keywordtype">string</span> specialSampleSnpsFile = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;special-sample-snps&quot;</span>);
<a name="l00401"></a>00401         <span class="keywordtype">string</span> chrXSnps = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;chrX-snps&quot;</span>);
<a name="l00402"></a>00402         <span class="keywordtype">string</span> readModelsBrlmmp=<a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-models-brlmmp&quot;</span>);
<a name="l00403"></a>00403         <span class="keywordtype">bool</span> writeModels = <a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;write-models&quot;</span>);
<a name="l00404"></a>00404         <span class="keywordtype">string</span> genderFileIn = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;read-genders&quot;</span>);
<a name="l00405"></a>00405         <span class="keywordtype">string</span> genoTypesFile = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;genotypes&quot;</span>);
<a name="l00406"></a>00406         <span class="keywordtype">string</span> filePrefix = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;set-analysis-name&quot;</span>);
<a name="l00407"></a>00407         <span class="keywordflow">if</span> (filePrefix.empty() == <span class="keyword">true</span>)
<a name="l00408"></a>00408                 filePrefix = quantMethod;
<a name="l00409"></a>00409 
<a name="l00410"></a>00410     <span class="comment">// open up summaries file for input</span>
<a name="l00411"></a>00411     vector&lt;string&gt; colNames;
<a name="l00412"></a>00412     <span class="keywordtype">string</span> summaryFile = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;summaries&quot;</span>);
<a name="l00413"></a>00413     <span class="keywordtype">string</span> summaryAFile = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;summaries-a&quot;</span>);
<a name="l00414"></a>00414     <span class="keywordtype">string</span> summaryBFile = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;summaries-b&quot;</span>);
<a name="l00415"></a>00415     <span class="keywordtype">string</span> summaryA5File = <a class="code" href="classOptions.html#a0efc7e2a4a55297f0f642d1d8bc08b82" title="Get the value of an option.">getOpt</a>(<span class="stringliteral">&quot;file5-summaries&quot;</span>);
<a name="l00416"></a>00416     <a class="code" href="classSgIoHelper.html" title="Class to wrap up the complexity of the different possible input files to SummaryGenotypeEngine.">SgIoHelper</a> ioHelp(summaryFile, summaryAFile, summaryBFile, summaryA5File, <span class="stringliteral">&quot;/&quot;</span>);
<a name="l00417"></a>00417     colNames = ioHelp.getColNames();
<a name="l00418"></a>00418 
<a name="l00419"></a>00419     <span class="comment">// Setup dummy sparse mart to spoof the quant method</span>
<a name="l00420"></a>00420     <a class="code" href="classSparseMart.html" title="SparseMart Class for representing only a subset of cel file intensity data in memory.">SparseMart</a> iMart(analysisOrder, colNames, <span class="keyword">false</span>);
<a name="l00421"></a>00421 
<a name="l00422"></a>00422     <span class="comment">// Create Quant Method</span>
<a name="l00423"></a>00423     <span class="keywordtype">string</span> analysis = quantMethod;
<a name="l00424"></a>00424     <span class="keywordflow">if</span>(algoParams != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00425"></a>00425         analysis += <span class="stringliteral">&quot;.&quot;</span> + algoParams;
<a name="l00426"></a>00426     }
<a name="l00427"></a>00427     <a class="code" href="classQuantMethodFactory.html" title="Factory class for making chip streams based on a string representation.">QuantMethodFactory</a> quantMethodFactory(QuantMethodFactory::GenoType);
<a name="l00428"></a>00428     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;Creating QuantMethod for genotyping from: &quot;</span> + analysis);
<a name="l00429"></a>00429     <a class="code" href="classQuantGTypeMethod.html" title="Quantification methods used for making genotyping calls implement (currently just brlmm) this interfa...">QuantGTypeMethod</a> *qMethod = 
<a name="l00430"></a>00430         quantMethodFactory.quantGTypeMethodForString(analysis, layout, QuantMethodFactory::GenoType);
<a name="l00431"></a>00431     <a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a> *qBrlmmp = <span class="keyword">dynamic_cast&lt;</span><a class="code" href="classQuantLabelZ.html" title="This is the primary class wrapping the bayes_label_routine it handles bookkeeping, gender determination, etc.">QuantLabelZ</a> *<span class="keyword">&gt;</span>(qMethod);
<a name="l00432"></a>00432 
<a name="l00433"></a>00433     <span class="comment">// Create Report</span>
<a name="l00434"></a>00434     <a class="code" href="classQuantMethodGTypeReport.html">QuantMethodGTypeReport</a> *report = <span class="keyword">new</span> <a class="code" href="classQuantMethodGTypeReport.html">QuantMethodGTypeReport</a>();
<a name="l00435"></a>00435     report-&gt;setDirPath(outDir);
<a name="l00436"></a>00436     report-&gt;setFileprefix(filePrefix);
<a name="l00437"></a>00437     report-&gt;setFilename(<span class="stringliteral">&quot;QuantMethodGTypeReport-A5&quot;</span>); <span class="comment">// for debugging</span>
<a name="l00438"></a>00438     <span class="keywordflow">if</span>(<a class="code" href="classOptions.html#a3c98038375042f9b230b3e4e2151fe00" title="Get the boolean value of an option.">getOptBool</a>(<span class="stringliteral">&quot;file5-output&quot;</span>)) {
<a name="l00439"></a>00439       report-&gt;<a class="code" href="classQuantMethodGTypeReport.html#a9546b809be37fc59d33c14159c3ea42d" title="Use a compact version of the file5 format with structure of: /calls - with genotypes as single byte /...">setCompactFile5Format</a>(20);
<a name="l00440"></a>00440     }
<a name="l00441"></a>00441     <span class="keywordflow">else</span> {
<a name="l00442"></a>00442       report-&gt;setFormat(affx::TsvReport::FMT_TSV);
<a name="l00443"></a>00443       report-&gt;setPrecision(5);
<a name="l00444"></a>00444     }
<a name="l00445"></a>00445     <span class="comment">//    report-&gt;setFilename(&quot;QuantMethodGTypeReport-TXT&quot;); // for debugging</span>
<a name="l00446"></a>00446     <span class="comment">//    report-&gt;setFormat(affx::TsvReport::FMT_A5);</span>
<a name="l00447"></a>00447 
<a name="l00448"></a>00448     <span class="comment">//</span>
<a name="l00449"></a>00449 
<a name="l00450"></a>00450     <span class="comment">//</span>
<a name="l00451"></a>00451     <span class="keywordtype">string</span> qMethodGuid = <a class="code" href="classaffxutil_1_1Guid.html#a2929c982cb9f3b5f6c5d7d818a8e6ac4">affxutil::Guid::GenerateNewGuid</a>();
<a name="l00452"></a>00452     <span class="keywordtype">string</span> execGuid = <a class="code" href="classaffxutil_1_1Guid.html#a2929c982cb9f3b5f6c5d7d818a8e6ac4">affxutil::Guid::GenerateNewGuid</a>();
<a name="l00453"></a>00453     <span class="keywordtype">string</span> timeStr = <span class="stringliteral">&quot;&quot;</span>; <span class="comment">///@todo set time string</span>
<a name="l00454"></a>00454 <span class="comment"></span>    <span class="keywordtype">string</span> commandLine = <span class="stringliteral">&quot;&quot;</span>; <span class="comment">///@todo set command line string</span>
<a name="l00455"></a>00455 <span class="comment"></span>    <span class="keywordtype">string</span> execVersion = <span class="stringliteral">&quot;&quot;</span>; <span class="comment">///@todo set command line string</span>
<a name="l00456"></a>00456 <span class="comment"></span>    <a class="code" href="classAnalysisInfo.html" title="Information about a particular analysis run.">AnalysisInfo</a> info; <span class="comment">///@todo populate analysis info object</span>
<a name="l00457"></a>00457 <span class="comment"></span>    <span class="comment">//</span>
<a name="l00458"></a>00458     report-&gt;<a class="code" href="classQuantMethodReport.html#acc296969ce5774f4479fc7275de753f0" title="Print a message out to the stream.">addStdHeaders</a>((<a class="code" href="classQuantMethodReport.html" title="Class for reporting results of quantification methods.">QuantMethodReport</a> *)report, 
<a name="l00459"></a>00459                           execGuid,
<a name="l00460"></a>00460                           qMethodGuid,
<a name="l00461"></a>00461                           timeStr,
<a name="l00462"></a>00462                           commandLine, 
<a name="l00463"></a>00463                           execVersion,
<a name="l00464"></a>00464                           info);
<a name="l00465"></a>00465 
<a name="l00466"></a>00466     report-&gt;<a class="code" href="classQuantMethodReport.html#a4defc7fcba241175db8c7c37d42f790c" title="Get set up for a run of reporting probesets.">prepare</a>(*qMethod, iMart);
<a name="l00467"></a>00467     
<a name="l00468"></a>00468     <span class="comment">// setup special SNPs</span>
<a name="l00469"></a>00469     SpecialSnpMap specialSnps;
<a name="l00470"></a>00470     map&lt;string,bool&gt; haploidSnps;
<a name="l00471"></a>00471     <span class="keywordflow">if</span>(specialSnpsFile != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00472"></a>00472         <span class="keywordtype">string</span> chip, specialType;
<a name="l00473"></a>00473         readSpecialSnpsFile(&amp;specialSnps, &amp;chip, &amp;specialType, specialSnpsFile);
<a name="l00474"></a>00474         haploidFromSpecial(specialSnps,haploidSnps);
<a name="l00475"></a>00475     } <span class="keywordflow">else</span> <span class="keywordflow">if</span>(chrXSnps != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00476"></a>00476         <a class="code" href="SpecialSnps_8cpp.html#a24da823ec3912983e63d79ccdd48e510" title="Determine which version of a chrx snp have and open appropriately.">fillInHaploidSnps</a>(chrXSnps, haploidSnps, NULL, <span class="keyword">true</span>);
<a name="l00477"></a>00477         specialFromHaploid(specialSnps, haploidSnps);
<a name="l00478"></a>00478     }
<a name="l00479"></a>00479 
<a name="l00480"></a>00480     <span class="keywordflow">if</span>(!specialSnps.empty())
<a name="l00481"></a>00481         qBrlmmp-&gt;<a class="code" href="classQuantLabelZ.html#ac14724ad579f92836d1a6c779b9b396b" title="which snps are special">setSpecialSnps</a>(specialSnps);
<a name="l00482"></a>00482     <span class="keywordflow">else</span> <span class="keywordflow">if</span>(!haploidSnps.empty())
<a name="l00483"></a>00483         qBrlmmp-&gt;<a class="code" href="classQuantLabelZ.html#a5afceb726f32320451aeb0c5a334a92c" title="which snps are haploid snps">setHaploidSnps</a>(haploidSnps);
<a name="l00484"></a>00484 
<a name="l00485"></a>00485     SpecialSampleSnpMap specialSampleSnps;
<a name="l00486"></a>00486     SpecialSampleNameVector specialSampleNames;
<a name="l00487"></a>00487     <span class="keywordflow">if</span>(specialSampleSnpsFile != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00488"></a>00488         <span class="keywordtype">string</span> chip, specialType;
<a name="l00489"></a>00489         readSpecialSampleSnpsFile(&amp;specialSampleSnps, &amp;specialSampleNames,
<a name="l00490"></a>00490                         specialSampleSnpsFile);
<a name="l00491"></a>00491     }
<a name="l00492"></a>00492 
<a name="l00493"></a>00493     <span class="keywordflow">if</span>(!specialSampleSnps.empty())
<a name="l00494"></a>00494         qBrlmmp-&gt;<a class="code" href="classQuantLabelZ.html#a7e75ab7ac471019f5aebde93a21d58b1" title="which snps are special and sample-specific">setSpecialSampleSnps</a>(specialSampleSnps, specialSampleNames);
<a name="l00495"></a>00495 
<a name="l00496"></a>00496     <span class="comment">// setup genders</span>
<a name="l00497"></a>00497     vector&lt;Gender&gt; genders;
<a name="l00498"></a>00498     <span class="keywordflow">if</span>(genderFileIn != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00499"></a>00499         <a class="code" href="classFileGenders.html" title="Read gender assignments from file.">FileGenders</a> genderReader(genderFileIn,colNames);
<a name="l00500"></a>00500         genders = genderReader.getGenders();
<a name="l00501"></a>00501     } <span class="keywordflow">else</span> {
<a name="l00502"></a>00502         <span class="keywordflow">for</span>(<span class="keywordtype">int</span> i = 0; i &lt; colNames.size(); i++)
<a name="l00503"></a>00503             genders.push_back(<a class="code" href="namespaceaffx.html#a1c064510a1a9f30814f840025d73d669aa259d114b82ef05d3f1f9d3c81c84719" title="Don&amp;#39;t know genotype, use XX model.">UnknownGender</a>);
<a name="l00504"></a>00504     }
<a name="l00505"></a>00505     qBrlmmp-&gt;<a class="code" href="classQuantLabelZ.html#a75d2851d0da897d336e42010653fbb4e" title="sets the genders for all samples">setGenders</a>(genders);
<a name="l00506"></a>00506 
<a name="l00507"></a>00507     <span class="comment">// read in SNP models</span>
<a name="l00508"></a>00508     <span class="keywordflow">if</span>(readModelsBrlmmp!=<span class="stringliteral">&quot;&quot;</span>)
<a name="l00509"></a>00509         qBrlmmp-&gt;<a class="code" href="classQuantLabelZ.html#ab4a3970a1d381a0885d82ebc66ddb3ca" title="read in the snp specific priors from a file">readSnpPriorMap</a>(readModelsBrlmmp.c_str());
<a name="l00510"></a>00510 
<a name="l00511"></a>00511     <span class="comment">// write SNP models?</span>
<a name="l00512"></a>00512     <span class="keywordflow">if</span>(writeModels) {
<a name="l00513"></a>00513       <span class="keywordtype">string</span> outfile = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(outDir,filePrefix + <span class="stringliteral">&quot;.snp-posteriors&quot;</span>);
<a name="l00514"></a>00514       qBrlmmp-&gt;<a class="code" href="classQuantLabelZ.html#a37af5d01565df9a913e6f93033a836d9" title="Pick the output version based on the file format. (txt=&amp;gt;v1,a5=&amp;gt;v2).">writeSnpPosteriorTsv</a>(outfile,affx::TsvReport::FMT_TSV);
<a name="l00515"></a>00515     }
<a name="l00516"></a>00516 
<a name="l00517"></a>00517 
<a name="l00518"></a>00518         <span class="keywordflow">if</span> (selectProbes)
<a name="l00519"></a>00519       {
<a name="l00520"></a>00520         <span class="keywordtype">string</span> outfile = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(outDir,filePrefix + <span class="stringliteral">&quot;.select-probes.txt&quot;</span>);
<a name="l00521"></a>00521           qBrlmmp-&gt;<a class="code" href="classQuantLabelZ.html#ad7cae32c6830efa7475fb948f2442682" title="Write header to the selection file so people don&amp;#39;t have to look up everything.">writeSnpProbeTsv</a>(outfile,affx::TsvReport::FMT_TSV);
<a name="l00522"></a>00522       }
<a name="l00523"></a>00523         
<a name="l00524"></a>00524 
<a name="l00525"></a>00525     <span class="comment">// read in seed genotypes</span>
<a name="l00526"></a>00526     std::map&lt;const char *,std::vector&lt;affx::GType&gt;, <a class="code" href="structUtil_1_1ltstr.html" title="Comparison object for use in map for char *.">Util::ltstr</a> &gt; knownGenotypes;
<a name="l00527"></a>00527     <span class="keywordflow">if</span>(!genoTypesFile.empty()) {
<a name="l00528"></a>00528       std::vector&lt;std::string&gt; probesets;
<a name="l00529"></a>00529       <a class="code" href="classGenoSeedTxt.html" title="Implements interface for a genotyping oracle used for genotyping seeds in brlmm.">GenoSeedTxt</a> seeds(genoTypesFile.c_str(), layout, colNames, haploidSnps);
<a name="l00530"></a>00530       seeds.fillProbesets(probesets);
<a name="l00531"></a>00531 
<a name="l00532"></a>00532       <span class="keywordflow">for</span>(<span class="keywordtype">int</span> i=0; i&lt;probesets.size(); i++) {
<a name="l00533"></a>00533         <span class="keyword">const</span> <span class="keywordtype">char</span> *name = <a class="code" href="classUtil.html#a508b7aff9569ed929945d0d37eca384e" title="Create a copy of a string.">Util::cloneString</a>(probesets[i].c_str());
<a name="l00534"></a>00534         vector&lt;GType&gt; callVec = seeds.getGenoCalls(name);
<a name="l00535"></a>00535         knownGenotypes[name] = callVec;
<a name="l00536"></a>00536       }
<a name="l00537"></a>00537       qBrlmmp-&gt;<a class="code" href="classQuantLabelZ.html#a22b847bd3dff1573d18dc4f248ddd509" title="Set the initial guesses of genotypes, usually from DM predictions.">setKnownGenoTypes</a>(&amp;knownGenotypes);
<a name="l00538"></a>00538     }
<a name="l00539"></a>00539 
<a name="l00540"></a>00540     <span class="comment">// for each SNP -- two rows in summaries file</span>
<a name="l00541"></a>00541     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;Making calls...&quot;</span>);
<a name="l00542"></a>00542     vector&lt;double&gt; aVals,bVals;
<a name="l00543"></a>00543 
<a name="l00544"></a>00544     <span class="keywordtype">string</span> probesetName;
<a name="l00545"></a>00545     <span class="keywordflow">while</span>(ioHelp.getNextMarker( colNames, aVals, bVals, probesetName)) {
<a name="l00546"></a>00546 
<a name="l00547"></a>00547         <span class="comment">// set probeset name for report</span>
<a name="l00548"></a>00548         qBrlmmp-&gt;<a class="code" href="classQuantLabelZ.html#a4b257a271e340b468e41a22865b738e6" title="Set the probeset name.">setProbeSetName</a>(probesetName);
<a name="l00549"></a>00549 
<a name="l00550"></a>00550         <span class="comment">// fill in allele values</span>
<a name="l00551"></a>00551         qBrlmmp-&gt;<a class="code" href="classQuantLabelZ.html#adde341ea0e1f3e3e18ff6e26b105da1e" title="Set allele values.">setAlleleValues</a>(aVals,bVals);
<a name="l00552"></a>00552     
<a name="l00553"></a>00553         <span class="comment">// transform the data as specified</span>
<a name="l00554"></a>00554         qBrlmmp-&gt;<a class="code" href="classQuantLabelZ.html#a0ec9313e121ec2cc1304350a7bb4a909" title="Transform allele estimates.">transform</a>();
<a name="l00555"></a>00555    
<a name="l00556"></a>00556         <span class="comment">// load up seed genotypes</span>
<a name="l00557"></a>00557         qBrlmmp-&gt;<a class="code" href="classQuantLabelZ.html#aa0bb7cf31d9f77d47947cf91d2fa75e0" title="Load up the initial call for the requested SNP.">loadInitialCallsForSnp</a>(probesetName);
<a name="l00558"></a>00558 
<a name="l00559"></a>00559         <span class="comment">// make the call</span>
<a name="l00560"></a>00560         qBrlmmp-&gt;<a class="code" href="classQuantLabelZ.html#ad83c392dc97c1557e25bdd8b9b901ec9" title="Do the heavy lifting of estimation.">computeEstimate</a>();
<a name="l00561"></a>00561     
<a name="l00562"></a>00562         <span class="comment">// report results</span>
<a name="l00563"></a>00563         report-&gt;<a class="code" href="classQuantMethodReport.html#a6045850dff05e101ad1afed3c752e250" title="After every probeset computation this function is called an is an opportunity to query the quantifica...">report</a>(psGroup, *qMethod, iMart, iTrans, pmAdjust);
<a name="l00564"></a>00564     }
<a name="l00565"></a>00565     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;Done.&quot;</span>);
<a name="l00566"></a>00566 
<a name="l00567"></a>00567     report-&gt;<a class="code" href="classQuantMethodReport.html#a07a2e186a7bff8b6618f160d496dc0e4" title="No more probesets will be processed, this is a chance to finish outputting results and clean up...">finish</a>(*qMethod);
<a name="l00568"></a>00568     <span class="comment">// clean up</span>
<a name="l00569"></a>00569     <span class="keyword">delete</span> qBrlmmp;
<a name="l00570"></a>00570     <span class="keyword">delete</span> report;
<a name="l00571"></a>00571 }
</pre></div></div>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:33 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
