<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: copynumber/apt-copynumber-cyto/regression/test-regression-copynumber-cyto.cpp Source File</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li class="current"><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="tabs2">
    <ul class="tablist">
      <li><a href="files.html"><span>File&nbsp;List</span></a></li>
      <li><a href="globals.html"><span>File&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath">
    <ul>
      <li><a class="el" href="dir_2a6c9f730113b04fdf7a766d4dbe6d8e.html">copynumber</a>      </li>
      <li><a class="el" href="dir_663aed007a9da5e64ffeed6ccd36988b.html">apt-copynumber-cyto</a>      </li>
      <li><a class="el" href="dir_97eda535b6460410eca596eecf937241.html">regression</a>      </li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>test-regression-copynumber-cyto.cpp</h1>  </div>
</div>
<div class="contents">
<a href="test-regression-copynumber-cyto_8cpp.html">Go to the documentation of this file.</a><div class="fragment"><pre class="fragment"><a name="l00001"></a>00001 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00002"></a>00002 <span class="comment"></span><span class="comment">//</span>
<a name="l00003"></a>00003 <span class="comment">// Copyright (C) 2009 Affymetrix, Inc.</span>
<a name="l00004"></a>00004 <span class="comment">//</span>
<a name="l00005"></a>00005 <span class="comment">// This program is free software; you can redistribute it and/or modify </span>
<a name="l00006"></a>00006 <span class="comment">// it under the terms of the GNU General Public License (version 2) as </span>
<a name="l00007"></a>00007 <span class="comment">// published by the Free Software Foundation.</span>
<a name="l00008"></a>00008 <span class="comment">// </span>
<a name="l00009"></a>00009 <span class="comment">// This program is distributed in the hope that it will be useful, </span>
<a name="l00010"></a>00010 <span class="comment">// but WITHOUT ANY WARRANTY; without even the implied warranty of </span>
<a name="l00011"></a>00011 <span class="comment">// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU </span>
<a name="l00012"></a>00012 <span class="comment">// General Public License for more details.</span>
<a name="l00013"></a>00013 <span class="comment">// </span>
<a name="l00014"></a>00014 <span class="comment">// You should have received a copy of the GNU General Public License </span>
<a name="l00015"></a>00015 <span class="comment">// along with this program;if not, write to the </span>
<a name="l00016"></a>00016 <span class="comment">// </span>
<a name="l00017"></a>00017 <span class="comment">// Free Software Foundation, Inc., </span>
<a name="l00018"></a>00018 <span class="comment">// 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA</span>
<a name="l00019"></a>00019 <span class="comment">//</span><span class="comment"></span>
<a name="l00020"></a>00020 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00021"></a>00021 <span class="comment"></span><span class="comment"></span>
<a name="l00022"></a>00022 <span class="comment">/**</span>
<a name="l00023"></a>00023 <span class="comment"> * @file   test-regression-copynumber-cyto.cpp</span>
<a name="l00024"></a>00024 <span class="comment"> * @brief  Program for doing regression tests on cyto data.</span>
<a name="l00025"></a>00025 <span class="comment"> */</span>
<a name="l00026"></a>00026 
<a name="l00027"></a>00027 <span class="comment">// This has to be last. Otherwise windows compile fails</span>
<a name="l00028"></a>00028 <span class="preprocessor">#include &quot;<a class="code" href="Calvin_8h.html" title="This header contains the Calvin class definition.">calvin_files/utils/src/Calvin.h</a>&quot;</span>
<a name="l00029"></a>00029 <span class="preprocessor">#include &quot;copynumber/apt-copynumber-cyto/regression/File5EquivalentCheck.h&quot;</span>
<a name="l00030"></a>00030 <span class="preprocessor">#include &quot;file5/File5.h&quot;</span>
<a name="l00031"></a>00031 <span class="preprocessor">#include &quot;<a class="code" href="CychpCheck_8cpp.html" title="Class for doing a check of two matrix files after regression run.">util/CychpCheck.cpp</a>&quot;</span>
<a name="l00032"></a>00032 <span class="preprocessor">#include &quot;<a class="code" href="Fs_8h.html" title="///">util/Fs.h</a>&quot;</span>
<a name="l00033"></a>00033 <span class="preprocessor">#include &quot;<a class="code" href="FsTestDir_8h.html" title="///">util/FsTestDir.h</a>&quot;</span>
<a name="l00034"></a>00034 <span class="preprocessor">#include &quot;<a class="code" href="LogStream_8h.html" title="Output verbose messages to log file as we go along.">util/LogStream.h</a>&quot;</span>
<a name="l00035"></a>00035 <span class="preprocessor">#include &quot;<a class="code" href="RegressionCheck_8h.html" title="Virtual class for checks to be done after a regression test.">util/RegressionCheck.h</a>&quot;</span>
<a name="l00036"></a>00036 <span class="preprocessor">#include &quot;<a class="code" href="RegressionSuite_8h.html" title="Base class for holding basic options for regression data.">util/RegressionSuite.h</a>&quot;</span>
<a name="l00037"></a>00037 <span class="preprocessor">#include &quot;<a class="code" href="RegressionTest_8h.html" title="Utility class for helping to do regression testing.">util/RegressionTest.h</a>&quot;</span>
<a name="l00038"></a>00038 <span class="preprocessor">#include &quot;<a class="code" href="Util_8h.html" title="General Utilities.">util/Util.h</a>&quot;</span>
<a name="l00039"></a>00039 <span class="preprocessor">#include &quot;<a class="code" href="Verbose_8h.html" title="Class for doing logging and some command line ui.">util/Verbose.h</a>&quot;</span>
<a name="l00040"></a>00040 <span class="comment">//</span>
<a name="l00041"></a>00041 <span class="preprocessor">#include &lt;cstring&gt;</span>
<a name="l00042"></a>00042 <span class="preprocessor">#include &lt;string&gt;</span>
<a name="l00043"></a>00043 <span class="comment">//</span>
<a name="l00044"></a>00044 
<a name="l00045"></a>00045 
<a name="l00046"></a>00046 
<a name="l00047"></a>00047 <span class="keyword">using namespace </span>std;
<a name="l00048"></a>00048 
<a name="l00049"></a><a class="code" href="classtest__regression__copynumber__cyto.html">00049</a> <span class="keyword">class </span><a class="code" href="classtest__regression__copynumber__cyto.html">test_regression_copynumber_cyto</a> : <span class="keyword">public</span> <a class="code" href="classRegressionSuite.html">RegressionSuite</a>
<a name="l00050"></a>00050 {
<a name="l00051"></a>00051 <span class="keyword">public</span>:
<a name="l00052"></a>00052 
<a name="l00053"></a>00053 
<a name="l00054"></a>00054     <span class="keywordtype">int</span> m_numPassed;
<a name="l00055"></a>00055     <span class="keywordtype">int</span> m_numFailed;
<a name="l00056"></a>00056     std::string m_testDir;
<a name="l00057"></a>00057   
<a name="l00058"></a>00058     std::set&lt;std::string&gt; setIgnore;
<a name="l00059"></a>00059     std::map&lt;std::string, float&gt; mapEpsilon;
<a name="l00060"></a>00060 
<a name="l00061"></a>00061     std::vector&lt;std::string&gt; m_vCytogenetics_Array_SixFiles;
<a name="l00062"></a>00062     std::vector&lt;std::string&gt; m_vCytogenetics_SNP6_SixFiles;
<a name="l00063"></a>00063     std::vector&lt;std::string&gt; m_vCytoScan_SixFiles;
<a name="l00064"></a>00064     std::vector&lt;std::string&gt; m_vTwoSevenCancer;
<a name="l00065"></a>00065     std::vector&lt;std::string&gt; m_vCytoScan_ThreeFiles;
<a name="l00066"></a>00066 
<a name="l00067"></a>00067 
<a name="l00068"></a>00068 <a class="code" href="classtest__regression__copynumber__cyto.html">test_regression_copynumber_cyto</a>()
<a name="l00069"></a>00069 {
<a name="l00070"></a>00070         m_numPassed = 0;
<a name="l00071"></a>00071         m_numFailed = 0;
<a name="l00072"></a>00072           
<a name="l00073"></a>00073         <span class="comment">// Header paramters to ignore</span>
<a name="l00074"></a>00074         setIgnore.insert(<span class="stringliteral">&quot;FileCreationTime&quot;</span>);
<a name="l00075"></a>00075         setIgnore.insert(<span class="stringliteral">&quot;FileIdentifier&quot;</span>);
<a name="l00076"></a>00076         setIgnore.insert(<span class="stringliteral">&quot;program-version&quot;</span>);
<a name="l00077"></a>00077         setIgnore.insert(<span class="stringliteral">&quot;create_date&quot;</span>);
<a name="l00078"></a>00078         setIgnore.insert(<span class="stringliteral">&quot;create-date&quot;</span>);
<a name="l00079"></a>00079         setIgnore.insert(<span class="stringliteral">&quot;affymetrix-algorithm-param-option-verbose&quot;</span>);
<a name="l00080"></a>00080         setIgnore.insert(<span class="stringliteral">&quot;affymetrix-algorithm-param-option-exec-guid&quot;</span>);
<a name="l00081"></a>00081         setIgnore.insert(<span class="stringliteral">&quot;affymetrix-algorithm-param-option-program-cvs-id&quot;</span>);
<a name="l00082"></a>00082         setIgnore.insert(<span class="stringliteral">&quot;affymetrix-algorithm-param-option-version-to-report&quot;</span>);
<a name="l00083"></a>00083         setIgnore.insert(<span class="stringliteral">&quot;affymetrix-algorithm-param-option-command-line&quot;</span>);
<a name="l00084"></a>00084         setIgnore.insert(<span class="stringliteral">&quot;affymetrix-algorithm-param-option-mem-usage&quot;</span>);
<a name="l00085"></a>00085         setIgnore.insert(<span class="stringliteral">&quot;affymetrix-algorithm-param-option-run-probeset-genotype&quot;</span>);
<a name="l00086"></a>00086         setIgnore.insert(<span class="stringliteral">&quot;affymetrix-algorithm-param-option-cels&quot;</span>);
<a name="l00087"></a>00087         setIgnore.insert(<span class="stringliteral">&quot;affymetrix-algorithm-param-option-out-dir&quot;</span>);
<a name="l00088"></a>00088         setIgnore.insert(<span class="stringliteral">&quot;affymetrix-algorithm-param-state-time-start&quot;</span>);
<a name="l00089"></a>00089         setIgnore.insert(<span class="stringliteral">&quot;affymetrix-algorithm-param-state-free-mem-at-start&quot;</span>);
<a name="l00090"></a>00090         setIgnore.insert(<span class="stringliteral">&quot;affymetrix-algorithm-param-option-temp-dir&quot;</span>);
<a name="l00091"></a>00091 
<a name="l00092"></a>00092         <span class="comment">// Data Set columns to ignore (Specify as Group.Set.Column)</span>
<a name="l00093"></a>00093         <span class="comment">//        setIgnore.insert(&quot;AlgorithmData.MarkerABSignal.SCAR&quot;);</span>
<a name="l00094"></a>00094 
<a name="l00095"></a>00095         <span class="comment">// Data Set column epsilon override (Specify as Group.Set.Column)</span>
<a name="l00096"></a>00096         <span class="comment">//        mapEpsilon.insert(std::pair&lt;std::string, float&gt;(&quot;AlgorithmData.MarkerABSignal.SCAR&quot;, (float)0.0005));</span>
<a name="l00097"></a>00097 
<a name="l00098"></a>00098         <span class="comment">// File tags to check for equivalency</span>
<a name="l00099"></a>00099         m_vCytogenetics_Array_SixFiles.push_back(<span class="stringliteral">&quot;HapMap-As_NA18971_C02_01_NN_20081218&quot;</span>);
<a name="l00100"></a>00100         m_vCytogenetics_Array_SixFiles.push_back(<span class="stringliteral">&quot;HapMap-As_NA18943_C11_01_NN_20081218&quot;</span>);
<a name="l00101"></a>00101         m_vCytogenetics_Array_SixFiles.push_back(<span class="stringliteral">&quot;HapMap-Cs_NA10857_A10_01_NN_20081218&quot;</span>);
<a name="l00102"></a>00102         m_vCytogenetics_Array_SixFiles.push_back(<span class="stringliteral">&quot;HapMap-Cs_NA12003_D04_01_NN_20081218&quot;</span>);
<a name="l00103"></a>00103 <span class="preprocessor">#ifdef __LP64__</span>
<a name="l00104"></a>00104 <span class="preprocessor"></span>        m_vCytogenetics_Array_SixFiles.push_back(<span class="stringliteral">&quot;HapMap-As_NA18547_A08_01_NN_20081218&quot;</span>);
<a name="l00105"></a>00105         m_vCytogenetics_Array_SixFiles.push_back(<span class="stringliteral">&quot;HapMap-As_NA18603_B03_01_NN_20081218&quot;</span>);
<a name="l00106"></a>00106 <span class="preprocessor">#endif</span>
<a name="l00107"></a>00107 <span class="preprocessor"></span>
<a name="l00108"></a>00108         <span class="comment">// File names for runs of cyto on snp6 cels</span>
<a name="l00109"></a>00109         m_vCytogenetics_SNP6_SixFiles.push_back(<span class="stringliteral">&quot;NA12004_GW6_C&quot;</span>);
<a name="l00110"></a>00110         m_vCytogenetics_SNP6_SixFiles.push_back(<span class="stringliteral">&quot;NA12003_GW6_C&quot;</span>);
<a name="l00111"></a>00111         m_vCytogenetics_SNP6_SixFiles.push_back(<span class="stringliteral">&quot;NA11993_GW6_C&quot;</span>);
<a name="l00112"></a>00112         m_vCytogenetics_SNP6_SixFiles.push_back(<span class="stringliteral">&quot;NA11882_GW6_C&quot;</span>);
<a name="l00113"></a>00113         m_vCytogenetics_SNP6_SixFiles.push_back(<span class="stringliteral">&quot;NA11881_GW6_C&quot;</span>);
<a name="l00114"></a>00114         m_vCytogenetics_SNP6_SixFiles.push_back(<span class="stringliteral">&quot;NA10863_GW6_C&quot;</span>);
<a name="l00115"></a>00115 
<a name="l00116"></a>00116         m_vCytoScan_SixFiles.push_back(<span class="stringliteral">&quot;NA12874_A05_Baseline_CytoScanHD_VH_20101103&quot;</span>);
<a name="l00117"></a>00117         m_vCytoScan_SixFiles.push_back(<span class="stringliteral">&quot;NA12892_A03_Baseline_CytoScanHD_VH_20101103&quot;</span>);
<a name="l00118"></a>00118         m_vCytoScan_SixFiles.push_back(<span class="stringliteral">&quot;NA18505_A01_Baseline_CytoScanHD_VH_20101103&quot;</span>);
<a name="l00119"></a>00119         m_vCytoScan_SixFiles.push_back(<span class="stringliteral">&quot;NA18633_A02_Baseline_CytoScanHD_VH_20101103&quot;</span>);
<a name="l00120"></a>00120         m_vCytoScan_SixFiles.push_back(<span class="stringliteral">&quot;NA18854_A06_Baseline_CytoScanHD_VH_20101103&quot;</span>);
<a name="l00121"></a>00121         m_vCytoScan_SixFiles.push_back(<span class="stringliteral">&quot;NA19131_A04_Baseline_CytoScanHD_VH_20101103&quot;</span>);
<a name="l00122"></a>00122 
<a name="l00123"></a>00123         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;LT611013_F_01_Cyto_JC_20090122&quot;</span>);
<a name="l00124"></a>00124         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;LT611011_F_01_Cyto_JC_20090122&quot;</span>);
<a name="l00125"></a>00125         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;LN611013_F_01_Cyto_JC_20090122&quot;</span>);
<a name="l00126"></a>00126         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;LN611011_F_01_Cyto_JC_20090122&quot;</span>);
<a name="l00127"></a>00127         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;D-21165ColonRectumMalignantN2_F_01_Cyto_JC_20090122&quot;</span>);
<a name="l00128"></a>00128         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;D-21163ColonRectumNormalN2_F_01_Cyto_JC_20090122&quot;</span>);
<a name="l00129"></a>00129         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;D-21142KidneyMalignantN2_F_01_Cyto_JC_20090122&quot;</span>);
<a name="l00130"></a>00130         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;D-21140KidneyNormalN2_F_01_Cyto_JC_20090122&quot;</span>);
<a name="l00131"></a>00131         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;CN810126_F_01_Cyto_JC_20090122&quot;</span>);
<a name="l00132"></a>00132         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;CT810126_F_01_Cyto_JC_20090122&quot;</span>);
<a name="l00133"></a>00133         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;CancerA37938_F_01_Cyto_SN_20090116&quot;</span>);
<a name="l00134"></a>00134         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;CancerA37937_F_01_Cyto_SN_20090116&quot;</span>);
<a name="l00135"></a>00135         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;CancerA37923_F_01_Cyto_SN_20090116&quot;</span>);
<a name="l00136"></a>00136         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;CancerA37922_F_01_Cyto_SN_20090116&quot;</span>);
<a name="l00137"></a>00137         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;Cancer08000082_F_01_Cyto_SN_20090115&quot;</span>);
<a name="l00138"></a>00138         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;Cancer07017684_F_01_Cyto_SN_20090115&quot;</span>);
<a name="l00139"></a>00139         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;Cancer07017424_F_01_Cyto_SN_20090115&quot;</span>);
<a name="l00140"></a>00140         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;Cancer07015875_F_01_Cyto_SN_20090115&quot;</span>);
<a name="l00141"></a>00141         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;Cancer07014885_F_01_Cyto_SN_20090115&quot;</span>);
<a name="l00142"></a>00142         m_vTwoSevenCancer.push_back(<span class="stringliteral">&quot;Cancer07002728_F_01_Cyto_SN_20090115&quot;</span>);
<a name="l00143"></a>00143 
<a name="l00144"></a>00144         m_vCytoScan_ThreeFiles.push_back(<span class="stringliteral">&quot;Ref103_A1_Alpha_CytoScan_DF_20101512&quot;</span>);
<a name="l00145"></a>00145         m_vCytoScan_ThreeFiles.push_back(<span class="stringliteral">&quot;NA16595_B6_Alpha_CytoScan_JS_20101512&quot;</span>);
<a name="l00146"></a>00146         m_vCytoScan_ThreeFiles.push_back(<span class="stringliteral">&quot;3x_A4_Alpha_CytoScan_DF_20101512&quot;</span>);
<a name="l00147"></a>00147 
<a name="l00148"></a>00148 }
<a name="l00149"></a>00149 
<a name="l00150"></a>00150 
<a name="l00151"></a>00151 
<a name="l00152"></a>00152 <span class="comment">/*</span>
<a name="l00153"></a>00153 <span class="comment">     Just a few comments.</span>
<a name="l00154"></a>00154 <span class="comment">     1)  The single_na30 testcase has the new Allele Peaks functionality invoked and has as inputs the appropriatly updated snp-reference file.</span>
<a name="l00155"></a>00155 <span class="comment">         It uses as input a CN reference that was generated using the na30 annotations file. The testcase itself runs on the na30.1</span>
<a name="l00156"></a>00156 <span class="comment">         annotations file.  This tests the CN flag functionality. At present the snp-reference file does not have a section that forces</span>
<a name="l00157"></a>00157 <span class="comment">         the use of the new LOH algorithm.</span>
<a name="l00158"></a>00158 <span class="comment">     2)  The single_newLOH testcase tests the new LOH algorithms. The snp-reference does have a section for the new AP algorithm.  No specific</span>
<a name="l00159"></a>00159 <span class="comment">         testing of CN flags was attempted for this testcase.</span>
<a name="l00160"></a>00160 <span class="comment">     3)  The single testcase is the original testcase using the na28 annotations file.  It has input of an updated AP section since the</span>
<a name="l00161"></a>00161 <span class="comment">         AP algorithm does not have a fall back to the old style if the appropriate section does not exist in the snp-reference.  The</span>
<a name="l00162"></a>00162 <span class="comment">         idea is to keep this testcase around to validate backwards compatibility.</span>
<a name="l00163"></a>00163 <span class="comment">     4)  The batch testcase creates and uses CN/SNP reference files.  It uses na28.  Keep it around to validate backwards compatibility.</span>
<a name="l00164"></a>00164 <span class="comment">         The idea being that the changes to the CN flags functionality to accomodate different annotation files in the create/use</span>
<a name="l00165"></a>00165 <span class="comment">         portions should not mess up the situation where the annotations files are the same for the two operations. I did have to</span>
<a name="l00166"></a>00166 <span class="comment">         update some of the golden cychp files, since the original algorithm wrote values for all ProbeSets in the Probeset section</span>
<a name="l00167"></a>00167 <span class="comment">         even those with processFlag=0.  This was not a problem in the original code since these were completely filtered out befor</span>
<a name="l00168"></a>00168 <span class="comment">         any analysis began.</span>
<a name="l00169"></a>00169 <span class="comment">     5)  The batch_newLOH testcase tests the create/use functionality when the new LOH algorithm is invoked.  Tests that the snp-reference</span>
<a name="l00170"></a>00170 <span class="comment">         file that is created gets all the new LOH information needed.</span>
<a name="l00171"></a>00171 <span class="comment">*/</span>
<a name="l00172"></a>00172 
<a name="l00173"></a>00173 <span class="keywordtype">void</span> run()
<a name="l00174"></a>00174 {
<a name="l00175"></a>00175 
<a name="l00176"></a>00176         single_na30();
<a name="l00177"></a>00177         batch_newLOH();
<a name="l00178"></a>00178 <span class="preprocessor">#if( defined(_WIN64))</span>
<a name="l00179"></a>00179 <span class="preprocessor"></span>        CytoScanBatch();
<a name="l00180"></a>00180         CytoScanUse();
<a name="l00181"></a>00181         CytoScanCreate();
<a name="l00182"></a>00182 <span class="preprocessor">#endif</span>
<a name="l00183"></a>00183 <span class="preprocessor"></span>
<a name="l00184"></a>00184 
<a name="l00185"></a>00185 
<a name="l00186"></a>00186 <span class="preprocessor">#if ( (!defined(__MACH__)) &amp;&amp; (defined(__LP64__)) )</span>
<a name="l00187"></a>00187 <span class="preprocessor"></span>        CytoScanBatch();
<a name="l00188"></a>00188         CytoScanUse();
<a name="l00189"></a>00189         CytoScanCreate();
<a name="l00190"></a>00190         twoSevenMCreate();
<a name="l00191"></a>00191         twoSevenCancerNA31();
<a name="l00192"></a>00192         twoSevenCancer();
<a name="l00193"></a>00193         batch_snp6_dual();
<a name="l00194"></a>00194         batch_snp6();
<a name="l00195"></a>00195         batch_snp6_dual_noAdapter();
<a name="l00196"></a>00196         batch_snp6_noAdapter();
<a name="l00197"></a>00197 <span class="preprocessor">#endif</span>
<a name="l00198"></a>00198 <span class="preprocessor"></span>        single_newLOH();
<a name="l00199"></a>00199         batch();
<a name="l00200"></a>00200         single();
<a name="l00201"></a>00201 
<a name="l00202"></a>00202 <span class="preprocessor">#if ( (!defined(_WIN32)) || (defined(_WIN64)))</span>
<a name="l00203"></a>00203 <span class="preprocessor"></span>        CytoScanBadAnno();
<a name="l00204"></a>00204 <span class="preprocessor">#else</span>
<a name="l00205"></a>00205 <span class="preprocessor"></span>        <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;APT-994: test-regression-copynumber-cyto: CytoScanBadAnno commented out for Windows 32 bit&quot;</span>);
<a name="l00206"></a>00206 <span class="preprocessor">#endif</span>
<a name="l00207"></a>00207 <span class="preprocessor"></span>            
<a name="l00208"></a>00208 }
<a name="l00209"></a>00209       <span class="comment">// This test is identical to the batch testcase except for the input snp reference file. The new snp reference has sections which cause the</span>
<a name="l00210"></a>00210       <span class="comment">// invocation of new loh algorithms. In the creation step, no differences should be found between the CN reference file generated in</span>
<a name="l00211"></a>00211       <span class="comment">// the batch ran and the present test case. Thus in the comparison below the CN reference file&#39;s generated in batch and batch_newLOH are</span>
<a name="l00212"></a>00212       <span class="comment">// compared to the same gold CN reference file. The snp reference will be different in this run than in the original batch run. So the</span>
<a name="l00213"></a>00213       <span class="comment">// the batch and batch_newLOH have separate gold snp reference files.</span>
<a name="l00214"></a>00214       <span class="comment">// In the use step for the new_LOH run, since the loh algorithm has different inputs (from the new snp-reference file), the cychp files</span>
<a name="l00215"></a>00215       <span class="comment">// generated in batch and batch_newLOH will be different. Thus batch and batch_newLOH have separate gold cychp files to compare to</span>
<a name="l00216"></a>00216       <span class="comment">// during the use portion of their runs.</span>
<a name="l00217"></a>00217 
<a name="l00218"></a>00218 
<a name="l00219"></a>00219 <span class="keywordtype">void</span> batch_newLOH()
<a name="l00220"></a>00220 {
<a name="l00221"></a>00221       <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on batch_newLOH part 1.&quot;</span>);
<a name="l00222"></a>00222       std::string outdir = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;batch_newLOH&quot;</span>);
<a name="l00223"></a>00223       <a class="code" href="classFs.html#a1206f6cf8f445461466f224e6cd1bc52" title="///">Fs::rmdirPath</a>(outdir, <span class="keyword">false</span>);
<a name="l00224"></a>00224       <a class="code" href="classFs.html#abf363360d88071c2f4a93a7e50d61c9f" title="Make one directory. ///.">Fs::mkdir</a>(outdir, <span class="keyword">false</span>);
<a name="l00225"></a>00225       vector&lt;RegressionCheck *&gt; checks;
<a name="l00226"></a>00226       <span class="keywordtype">string</span> command1 = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l00227"></a>00227 <span class="stringliteral">                    -v 3 --cyto2 true \</span>
<a name="l00228"></a>00228 <span class="stringliteral">                    --force true \</span>
<a name="l00229"></a>00229 <span class="stringliteral">                    --check-input-files true \</span>
<a name="l00230"></a>00230 <span class="stringliteral">                    --text-output true \</span>
<a name="l00231"></a>00231 <span class="stringliteral">                    --cnchp-output false \</span>
<a name="l00232"></a>00232 <span class="stringliteral">                    --cychp-output true \</span>
<a name="l00233"></a>00233 <span class="stringliteral">                    --set-analysis-name ca \</span>
<a name="l00234"></a>00234 <span class="stringliteral">                    --meta-data-info wittgenstein=bushido \</span>
<a name="l00235"></a>00235 <span class="stringliteral">                    --meta-data-info batchMode=Creation \</span>
<a name="l00236"></a>00236 <span class="stringliteral">                    --reference-output &quot;</span> + m_testDir + <span class="stringliteral">&quot;/batch_newLOH/Cytogenetics_Array.10_cels.REF_MODEL \</span>
<a name="l00237"></a>00237 <span class="stringliteral">                    --snp-reference-input-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.na28.batch_newLOH.snpref.a5 \</span>
<a name="l00238"></a>00238 <span class="stringliteral">                    --snp-reference-output-file &quot;</span> + m_testDir + <span class="stringliteral">&quot;/batch_newLOH/Cytogenetics_Array.10_cels.snpref.a5 \</span>
<a name="l00239"></a>00239 <span class="stringliteral">                    --cdf-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.CDF \</span>
<a name="l00240"></a>00240 <span class="stringliteral">                    --chrX-probes ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.chrXprobes \</span>
<a name="l00241"></a>00241 <span class="stringliteral">                    --chrY-probes ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.chrYprobes \</span>
<a name="l00242"></a>00242 <span class="stringliteral">                    --annotation-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.na28.annot.db \</span>
<a name="l00243"></a>00243 <span class="stringliteral">                    --probe-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.probe_tab \</span>
<a name="l00244"></a>00244 <span class="stringliteral">                    --snp-qc-snp-list ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.snplist.txt \</span>
<a name="l00245"></a>00245 <span class="stringliteral">                    --gender-override-file ../../../regression-data/data/lib/Cytogenetics_Array/HapMap-LOHReference-Full-Genders.txt \</span>
<a name="l00246"></a>00246 <span class="stringliteral">                    --genotype-call-override-file ../../../regression-data/data/lib/Cytogenetics_Array/HapMapCalls_LOHReferenceSet-FullArray.txt \</span>
<a name="l00247"></a>00247 <span class="stringliteral">                    --cel-files ../../../regression-data/data/cel/Cytogenetics_Array/CELFileList_5F5M.txt \</span>
<a name="l00248"></a>00248 <span class="stringliteral">                    --out-dir &quot;</span> + m_testDir + <span class="stringliteral">&quot;/batch_newLOH \</span>
<a name="l00249"></a>00249 <span class="stringliteral">                    --male-gender-ratio-cutoff=1.3 \</span>
<a name="l00250"></a>00250 <span class="stringliteral">                    --female-gender-ratio-cutoff=1.0 \</span>
<a name="l00251"></a>00251 <span class="stringliteral">                    --xx-cutoff=0.8 \</span>
<a name="l00252"></a>00252 <span class="stringliteral">                    --xx-cutoff-high=1.07 \</span>
<a name="l00253"></a>00253 <span class="stringliteral">                    --y-cutoff=0.65 \</span>
<a name="l00254"></a>00254 <span class="stringliteral">                    --keep-intermediate-data false \</span>
<a name="l00255"></a>00255 <span class="stringliteral">                    --analysis log2-ratio-cyto2.gc-correction=true.median-autosome-median-normalization=true.median-smooth-marker-count=5.trim-high=2.0.trim-low=-2.5 \</span>
<a name="l00256"></a>00256 <span class="stringliteral">                    --analysis kernel-smooth.sigma_span=50 \</span>
<a name="l00257"></a>00257 <span class="stringliteral">                    --analysis cn-cyto2.diagonal-weight=0.995.mapd-weight=0.22.min-segment-size=5.hmm-confidence-weight=0.6 \</span>
<a name="l00258"></a>00258 <span class="stringliteral">                    --analysis cn-cyto2-gender.cutoff=0.5 \</span>
<a name="l00259"></a>00259 <span class="stringliteral">                    --analysis cn-segment \</span>
<a name="l00260"></a>00260 <span class="stringliteral">                    --analysis loh-cyto2.lohCNSegSeparation=1000000.minInformation=100.lambdaCritical=8.0 \</span>
<a name="l00261"></a>00261 <span class="stringliteral">                    --analysis mosaicism.marker-bandwidth=6000.confidence-window=251 \</span>
<a name="l00262"></a>00262 <span class="stringliteral">                    --analysis allele-peaks.step=100.window=250.point-count=128.bandwidth=0.45.cutoff=0.05 \</span>
<a name="l00263"></a>00263 <span class="stringliteral">                    --local-gc-background-correction-reference-method pdnn-reference-method \</span>
<a name="l00264"></a>00264 <span class="stringliteral">                    --wave-correction-reference-method wave-correction-reference-method.trim=2.0.percentile=0.75.demean=false \</span>
<a name="l00265"></a>00265 <span class="stringliteral">                    --local-gc-background-intensity-adjustment-method pdnn-intensity-adjustment-method.predicted-intensity-bin-count=20.gc-bin-count=20.residual-trim=2.0 \</span>
<a name="l00266"></a>00266 <span class="stringliteral">                    --image-correction-intensity-adjustment-method high-pass-filter-intensity-adjustment-method.data-block-rows=320.data-block-cols=2015.mini-block-rows=8.mini-block-cols=8.global-smooth-weight=256.local-smooth-weight=64.converged=0.0001 \</span>
<a name="l00267"></a>00267 <span class="stringliteral">                    --wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method.bandwidth=101.bin-count=25.wave-count=3 \</span>
<a name="l00268"></a>00268 <span class="stringliteral">            &quot;</span>;
<a name="l00269"></a>00269 
<a name="l00270"></a>00270         std::set&lt;std::string&gt; setLocalIgnore;
<a name="l00271"></a>00271         std::string strFileName1 = <span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/batch_newLOH/Cytogenetics_Array.10_cels.snpref.a5&quot;</span>;
<a name="l00272"></a>00272         std::string strFileName2 = m_testDir + <span class="stringliteral">&quot;/batch_newLOH/Cytogenetics_Array.10_cels.snpref.a5&quot;</span>;
<a name="l00273"></a>00273 
<a name="l00274"></a>00274         <span class="comment">// See comments at the beginning of this testcase as to why we compare against the gold batch CN reference file. (And why there is no</span>
<a name="l00275"></a>00275         <span class="comment">// separate gold batch_newLOH CN reference file.)</span>
<a name="l00276"></a>00276 
<a name="l00277"></a>00277 
<a name="l00278"></a>00278         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;SNPReference&quot;</span>, <span class="stringliteral">&quot;EMSNParameters&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l00279"></a>00279         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;SNPReference&quot;</span>, <span class="stringliteral">&quot;EMSNArray&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l00280"></a>00280         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;SNPReference&quot;</span>, <span class="stringliteral">&quot;SNPReference&quot;</span>, setLocalIgnore, 0.0003, 1.0));
<a name="l00281"></a>00281 
<a name="l00282"></a>00282         strFileName1 = <span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/batch/Cytogenetics_Array.10_cels.REF_MODEL&quot;</span>;
<a name="l00283"></a>00283         strFileName2 = m_testDir + <span class="stringliteral">&quot;/batch_newLOH/Cytogenetics_Array.10_cels.REF_MODEL&quot;</span>;
<a name="l00284"></a>00284 
<a name="l00285"></a>00285         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;AntigenomicProbes&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l00286"></a>00286         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;MedianSignals&quot;</span>, setLocalIgnore, 0.5, 0.999999));
<a name="l00287"></a>00287         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;ProbeEffects&quot;</span>, setLocalIgnore, 0.0001, 0.999999));
<a name="l00288"></a>00288         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;SNPPosteriors&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l00289"></a>00289         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;SketchCN&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l00290"></a>00290         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;SketchSNP&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l00291"></a>00291         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;WaveCorrection&quot;</span>, setLocalIgnore, 0.0001, 0.99999));
<a name="l00292"></a>00292 
<a name="l00293"></a>00293 
<a name="l00294"></a>00294 
<a name="l00295"></a>00295         <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test(<span class="stringliteral">&quot;batch_newLOH part 1&quot;</span>, command1.c_str(), checks);
<a name="l00296"></a>00296         test.setSuite(*<span class="keyword">this</span>,  outdir, outdir + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir + <span class="stringliteral">&quot;/valgrind.log&quot;</span>);
<a name="l00297"></a>00297 
<a name="l00298"></a>00298         <span class="keywordtype">bool</span> bPassed = test.pass();
<a name="l00299"></a>00299         <span class="keywordflow">if</span>(!bPassed) {
<a name="l00300"></a>00300             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: batch_newLOH part1. &quot;</span> + test.getErrorMsg());
<a name="l00301"></a>00301         }
<a name="l00302"></a>00302         <span class="keywordflow">else</span>
<a name="l00303"></a>00303         {
<a name="l00304"></a>00304             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: batch_newLOH part1. &quot;</span> + test.getErrorMsg());    
<a name="l00305"></a>00305         }
<a name="l00306"></a>00306         {
<a name="l00307"></a>00307             m_numPassed += test.getNumPassed();
<a name="l00308"></a>00308             m_numFailed += test.getNumFailed();
<a name="l00309"></a>00309         }
<a name="l00310"></a>00310 
<a name="l00311"></a>00311         <span class="keywordflow">if</span> (bPassed)
<a name="l00312"></a>00312         {
<a name="l00313"></a>00313             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on batch_newLOH part 2.&quot;</span>); 
<a name="l00314"></a>00314             <span class="keywordtype">string</span> outdir = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir , <span class="stringliteral">&quot;batch_newLOH&quot;</span>);
<a name="l00315"></a>00315 
<a name="l00316"></a>00316             std::string  goldDirectory=<span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/batch_newLOH&quot;</span>;
<a name="l00317"></a>00317             std::string  newDirectory= <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;batch_newLOH&quot;</span>);
<a name="l00318"></a>00318             <span class="keywordtype">string</span> infix = <span class="stringliteral">&quot;ca&quot;</span>;
<a name="l00319"></a>00319 
<a name="l00320"></a>00320             std::map&lt;std::string, float&gt; mapEpsilon;
<a name="l00321"></a>00321 
<a name="l00322"></a>00322             std::set&lt;std::string&gt; setSetIgnore;
<a name="l00323"></a>00323 
<a name="l00324"></a>00324 
<a name="l00325"></a>00325 <span class="preprocessor">#ifndef __LP64__</span>
<a name="l00326"></a>00326 <span class="preprocessor"></span>            {
<a name="l00327"></a>00327                 std::string ignoreString = <span class="stringliteral">&quot;Segments.Mosaicism&quot;</span>;
<a name="l00328"></a>00328                 setSetIgnore.insert(ignoreString);
<a name="l00329"></a>00329                 std::string ignoreString2 = <span class="stringliteral">&quot;ProbeSets.AllelePeaks&quot;</span>;
<a name="l00330"></a>00330                 setSetIgnore.insert(ignoreString2);
<a name="l00331"></a>00331             }
<a name="l00332"></a>00332 <span class="preprocessor">#endif</span>
<a name="l00333"></a>00333 <span class="preprocessor"></span><span class="preprocessor">#ifdef __APPLE__</span>
<a name="l00334"></a>00334 <span class="preprocessor"></span>            {
<a name="l00335"></a>00335                 std::string ignoreString = <span class="stringliteral">&quot;Segments.Mosaicism&quot;</span>;
<a name="l00336"></a>00336                 setSetIgnore.insert(ignoreString);
<a name="l00337"></a>00337                 std::string ignoreString2 = <span class="stringliteral">&quot;Chromosomes.Summary&quot;</span>;
<a name="l00338"></a>00338                 setSetIgnore.insert(ignoreString2);
<a name="l00339"></a>00339                 std::string ignoreString3 = <span class="stringliteral">&quot;ProbeSets.AllelePeaks&quot;</span>;
<a name="l00340"></a>00340                 setSetIgnore.insert(ignoreString3);
<a name="l00341"></a>00341             }
<a name="l00342"></a>00342 <span class="preprocessor">#endif</span>
<a name="l00343"></a>00343 <span class="preprocessor"></span>
<a name="l00344"></a>00344 
<a name="l00345"></a>00345             vector&lt;RegressionCheck *&gt; checks2;
<a name="l00346"></a>00346             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classCychpCheck.html" title="Class for testing that two cychp files are the same +/- some epsilon.">CychpCheck</a>(
<a name="l00347"></a>00347                                                 m_vCytogenetics_Array_SixFiles,
<a name="l00348"></a>00348                                                 goldDirectory,
<a name="l00349"></a>00349                                                 newDirectory,
<a name="l00350"></a>00350                                                 infix,
<a name="l00351"></a>00351                                                 setIgnore,
<a name="l00352"></a>00352                                                 setSetIgnore,
<a name="l00353"></a>00353                                                 mapEpsilon,
<a name="l00354"></a>00354                                                 0.0001,
<a name="l00355"></a>00355                                                 0.9999989,
<a name="l00356"></a>00356                                                 <span class="keyword">false</span>));
<a name="l00357"></a>00357 
<a name="l00358"></a>00358 
<a name="l00359"></a>00359 
<a name="l00360"></a>00360             <span class="keywordtype">string</span> command2 = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l00361"></a>00361 <span class="stringliteral">                        -v 3 --cyto2 true \</span>
<a name="l00362"></a>00362 <span class="stringliteral">                        --force true \</span>
<a name="l00363"></a>00363 <span class="stringliteral">                        --text-output true \</span>
<a name="l00364"></a>00364 <span class="stringliteral">                        --cnchp-output false \</span>
<a name="l00365"></a>00365 <span class="stringliteral">                        --cychp-output true \</span>
<a name="l00366"></a>00366 <span class="stringliteral">                        --set-analysis-name ca \</span>
<a name="l00367"></a>00367 <span class="stringliteral">                        --meta-data-info wittgenstein=bushido \</span>
<a name="l00368"></a>00368 <span class="stringliteral">                        --meta-data-info batchMode=Use \</span>
<a name="l00369"></a>00369 <span class="stringliteral">                        --reference-input &quot;</span> + m_testDir + <span class="stringliteral">&quot;/batch_newLOH/Cytogenetics_Array.10_cels.REF_MODEL \</span>
<a name="l00370"></a>00370 <span class="stringliteral">                        --snp-reference-input-file &quot;</span> + m_testDir + <span class="stringliteral">&quot;/batch_newLOH/Cytogenetics_Array.10_cels.snpref.a5 \</span>
<a name="l00371"></a>00371 <span class="stringliteral">                        --cdf-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.CDF \</span>
<a name="l00372"></a>00372 <span class="stringliteral">                        --chrX-probes ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.chrXprobes \</span>
<a name="l00373"></a>00373 <span class="stringliteral">                        --chrY-probes ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.chrYprobes \</span>
<a name="l00374"></a>00374 <span class="stringliteral">                        --annotation-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.na28.annot.db \</span>
<a name="l00375"></a>00375 <span class="stringliteral">                        --probe-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.probe_tab \</span>
<a name="l00376"></a>00376 <span class="stringliteral">                        --snp-qc-snp-list ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.snplist.txt \</span>
<a name="l00377"></a>00377 <span class="stringliteral">                        --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18547_A08_01_NN_20081218.CEL \</span>
<a name="l00378"></a>00378 <span class="stringliteral">                        --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18603_B03_01_NN_20081218.CEL \</span>
<a name="l00379"></a>00379 <span class="stringliteral">                        --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18943_C11_01_NN_20081218.CEL \</span>
<a name="l00380"></a>00380 <span class="stringliteral">                        --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18971_C02_01_NN_20081218.CEL \</span>
<a name="l00381"></a>00381 <span class="stringliteral">                        --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-Cs_NA10857_A10_01_NN_20081218.CEL \</span>
<a name="l00382"></a>00382 <span class="stringliteral">                        --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-Cs_NA12003_D04_01_NN_20081218.CEL \</span>
<a name="l00383"></a>00383 <span class="stringliteral">                        --out-dir &quot;</span> + m_testDir + <span class="stringliteral">&quot;/batch_newLOH \</span>
<a name="l00384"></a>00384 <span class="stringliteral">                        --male-gender-ratio-cutoff=1.3 \</span>
<a name="l00385"></a>00385 <span class="stringliteral">                        --female-gender-ratio-cutoff=1.0 \</span>
<a name="l00386"></a>00386 <span class="stringliteral">                        --xx-cutoff=0.8 \</span>
<a name="l00387"></a>00387 <span class="stringliteral">                        --xx-cutoff-high=1.07 \</span>
<a name="l00388"></a>00388 <span class="stringliteral">                        --y-cutoff=0.65 \</span>
<a name="l00389"></a>00389 <span class="stringliteral">                        --keep-intermediate-data false \</span>
<a name="l00390"></a>00390 <span class="stringliteral">                        --analysis log2-ratio-cyto2.gc-correction=true.median-autosome-median-normalization=true.median-smooth-marker-count=5.trim-high=2.0.trim-low=-2.5 \</span>
<a name="l00391"></a>00391 <span class="stringliteral">                        --analysis kernel-smooth.sigma_span=50 \</span>
<a name="l00392"></a>00392 <span class="stringliteral">                        --analysis cn-cyto2.diagonal-weight=0.995.mapd-weight=0.22.min-segment-size=5.hmm-confidence-weight=0.6 \</span>
<a name="l00393"></a>00393 <span class="stringliteral">                        --analysis cn-cyto2-gender.cutoff=0.5 \</span>
<a name="l00394"></a>00394 <span class="stringliteral">                        --analysis cn-segment \</span>
<a name="l00395"></a>00395 <span class="stringliteral">                        --analysis loh-cyto2.lohCNSegSeparation=1000000.minInformation=100.lambdaCritical=8.0 \</span>
<a name="l00396"></a>00396 <span class="stringliteral">                        --analysis mosaicism.marker-bandwidth=6000.confidence-window=251 \</span>
<a name="l00397"></a>00397 <span class="stringliteral">                        --analysis allele-peaks.step=100.window=250.point-count=128.bandwidth=0.45.cutoff=0.05 \</span>
<a name="l00398"></a>00398 <span class="stringliteral">                        --local-gc-background-correction-reference-method pdnn-reference-method \</span>
<a name="l00399"></a>00399 <span class="stringliteral">                        --wave-correction-reference-method wave-correction-reference-method.trim=2.0.percentile=0.75.demean=false \</span>
<a name="l00400"></a>00400 <span class="stringliteral">                        --local-gc-background-intensity-adjustment-method pdnn-intensity-adjustment-method.predicted-intensity-bin-count=20.gc-bin-count=20.residual-trim=2.0 \</span>
<a name="l00401"></a>00401 <span class="stringliteral">                        --image-correction-intensity-adjustment-method high-pass-filter-intensity-adjustment-method.data-block-rows=320.data-block-cols=2015.mini-block-rows=8.mini-block-cols=8.global-smooth-weight=256.local-smooth-weight=64.converged=0.0001 \</span>
<a name="l00402"></a>00402 <span class="stringliteral">                        --wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method.bandwidth=101.bin-count=25.wave-count=3 \</span>
<a name="l00403"></a>00403 <span class="stringliteral">                &quot;</span>;
<a name="l00404"></a>00404 
<a name="l00405"></a>00405 
<a name="l00406"></a>00406 
<a name="l00407"></a>00407             <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test2(<span class="stringliteral">&quot;batch_newLOH part 2&quot;</span>, command2.c_str(), checks2, <span class="keyword">false</span>);
<a name="l00408"></a>00408             test2.setSuite(*<span class="keyword">this</span>,  outdir, outdir + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir + <span class="stringliteral">&quot;/valgrind.log&quot;</span>);
<a name="l00409"></a>00409             <span class="keywordflow">if</span>(!test2.pass()) {
<a name="l00410"></a>00410                 <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: batch_newLOH part 2. &quot;</span> + test2.getErrorMsg());
<a name="l00411"></a>00411             }
<a name="l00412"></a>00412             <span class="keywordflow">else</span>
<a name="l00413"></a>00413             {
<a name="l00414"></a>00414                 <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: batch_newLOH part 2. &quot;</span> + test.getErrorMsg());   
<a name="l00415"></a>00415             }
<a name="l00416"></a>00416             {
<a name="l00417"></a>00417                 m_numPassed += test2.getNumPassed();
<a name="l00418"></a>00418                 m_numFailed += test2.getNumFailed();
<a name="l00419"></a>00419             }
<a name="l00420"></a>00420 
<a name="l00421"></a>00421         }
<a name="l00422"></a>00422 }
<a name="l00423"></a>00423 
<a name="l00424"></a>00424 
<a name="l00425"></a>00425 
<a name="l00426"></a>00426 
<a name="l00427"></a>00427 <span class="keywordtype">void</span> single()
<a name="l00428"></a>00428 {
<a name="l00429"></a>00429         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on single.&quot;</span>);
<a name="l00430"></a>00430         <span class="keywordtype">string</span> outdir = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;single&quot;</span>);
<a name="l00431"></a>00431         <a class="code" href="classFs.html#a1206f6cf8f445461466f224e6cd1bc52" title="///">Fs::rmdirPath</a>(outdir, <span class="keyword">false</span>);
<a name="l00432"></a>00432         <a class="code" href="classFs.html#abf363360d88071c2f4a93a7e50d61c9f" title="Make one directory. ///.">Fs::mkdir</a>(outdir, <span class="keyword">false</span>);
<a name="l00433"></a>00433 
<a name="l00434"></a>00434 
<a name="l00435"></a>00435         std::string  goldDirectory=<span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/single&quot;</span>;
<a name="l00436"></a>00436         std::string  newDirectory= <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;single&quot;</span>);
<a name="l00437"></a>00437         <span class="keywordtype">string</span> infix = <span class="stringliteral">&quot;ca&quot;</span>;
<a name="l00438"></a>00438 
<a name="l00439"></a>00439         std::map&lt;std::string, float&gt; mapEpsilon;
<a name="l00440"></a>00440 
<a name="l00441"></a>00441         std::set&lt;std::string&gt; setSetIgnore;
<a name="l00442"></a>00442 <span class="preprocessor">#ifndef __LP64__</span>
<a name="l00443"></a>00443 <span class="preprocessor"></span>        {
<a name="l00444"></a>00444             std::string ignoreString2 = <span class="stringliteral">&quot;ProbeSets.AllelePeaks&quot;</span>;
<a name="l00445"></a>00445             setSetIgnore.insert(ignoreString2);
<a name="l00446"></a>00446         }
<a name="l00447"></a>00447 <span class="preprocessor">#endif</span>
<a name="l00448"></a>00448 <span class="preprocessor"></span><span class="preprocessor">#ifdef __APPLE__</span>
<a name="l00449"></a>00449 <span class="preprocessor"></span>        {
<a name="l00450"></a>00450             std::string ignoreString2 = <span class="stringliteral">&quot;ProbeSets.AllelePeaks&quot;</span>;
<a name="l00451"></a>00451             setSetIgnore.insert(ignoreString2);
<a name="l00452"></a>00452         }
<a name="l00453"></a>00453 <span class="preprocessor">#endif</span>
<a name="l00454"></a>00454 <span class="preprocessor"></span>
<a name="l00455"></a>00455 
<a name="l00456"></a>00456         vector&lt;RegressionCheck *&gt; checks;
<a name="l00457"></a>00457         checks.push_back(<span class="keyword">new</span> <a class="code" href="classCychpCheck.html" title="Class for testing that two cychp files are the same +/- some epsilon.">CychpCheck</a>(
<a name="l00458"></a>00458                                                 m_vCytogenetics_Array_SixFiles,
<a name="l00459"></a>00459                                                 goldDirectory,
<a name="l00460"></a>00460                                                 newDirectory,
<a name="l00461"></a>00461                                                 infix,
<a name="l00462"></a>00462                                                 setIgnore,
<a name="l00463"></a>00463                                                 setSetIgnore,
<a name="l00464"></a>00464                                                 mapEpsilon,
<a name="l00465"></a>00465                                                 0.0001,
<a name="l00466"></a>00466                                                 0.9999989,
<a name="l00467"></a>00467                                                 <span class="keyword">false</span>));
<a name="l00468"></a>00468 
<a name="l00469"></a>00469         <span class="keywordtype">string</span> command = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l00470"></a>00470 <span class="stringliteral">                    -v 3 --cyto2 true           \</span>
<a name="l00471"></a>00471 <span class="stringliteral">                    --force true \</span>
<a name="l00472"></a>00472 <span class="stringliteral">                    --text-output true \</span>
<a name="l00473"></a>00473 <span class="stringliteral">                    --cnchp-output false \</span>
<a name="l00474"></a>00474 <span class="stringliteral">                    --cychp-output true \</span>
<a name="l00475"></a>00475 <span class="stringliteral">                    --set-analysis-name ca \</span>
<a name="l00476"></a>00476 <span class="stringliteral">                    --keep-intermediate-data false \</span>
<a name="l00477"></a>00477 <span class="stringliteral">                    --meta-data-info wittgenstein=bushido \</span>
<a name="l00478"></a>00478 <span class="stringliteral">                    --reference-input ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.na28.REF_MODEL \</span>
<a name="l00479"></a>00479 <span class="stringliteral">                    --snp-reference-input-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.na28.snpref.a5 \</span>
<a name="l00480"></a>00480 <span class="stringliteral">                    --cdf-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.CDF \</span>
<a name="l00481"></a>00481 <span class="stringliteral">                    --chrX-probes ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.chrXprobes \</span>
<a name="l00482"></a>00482 <span class="stringliteral">                    --chrY-probes ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.chrYprobes \</span>
<a name="l00483"></a>00483 <span class="stringliteral">                    --annotation-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.na28.annot.db \</span>
<a name="l00484"></a>00484 <span class="stringliteral">                    --probe-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.probe_tab \</span>
<a name="l00485"></a>00485 <span class="stringliteral">                    --snp-qc-snp-list ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.snplist.txt \</span>
<a name="l00486"></a>00486 <span class="stringliteral">                    --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18547_A08_01_NN_20081218.CEL \</span>
<a name="l00487"></a>00487 <span class="stringliteral">                    --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18603_B03_01_NN_20081218.CEL \</span>
<a name="l00488"></a>00488 <span class="stringliteral">                    --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18943_C11_01_NN_20081218.CEL \</span>
<a name="l00489"></a>00489 <span class="stringliteral">                    --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18971_C02_01_NN_20081218.CEL \</span>
<a name="l00490"></a>00490 <span class="stringliteral">                    --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-Cs_NA10857_A10_01_NN_20081218.CEL \</span>
<a name="l00491"></a>00491 <span class="stringliteral">                    --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-Cs_NA12003_D04_01_NN_20081218.CEL \</span>
<a name="l00492"></a>00492 <span class="stringliteral">                    --out-dir &quot;</span> + outdir + <span class="stringliteral">&quot;        \</span>
<a name="l00493"></a>00493 <span class="stringliteral">                    --male-gender-ratio-cutoff=1.3 \</span>
<a name="l00494"></a>00494 <span class="stringliteral">                    --female-gender-ratio-cutoff=1.0 \</span>
<a name="l00495"></a>00495 <span class="stringliteral">                    --xx-cutoff=0.8 \</span>
<a name="l00496"></a>00496 <span class="stringliteral">                    --xx-cutoff-high=1.07 \</span>
<a name="l00497"></a>00497 <span class="stringliteral">                    --y-cutoff=0.65 \</span>
<a name="l00498"></a>00498 <span class="stringliteral">                    --analysis log2-ratio-cyto2.gc-correction=true.median-autosome-median-normalization=true.median-smooth-marker-count=5.trim-high=2.0.trim-low=-2.5 \</span>
<a name="l00499"></a>00499 <span class="stringliteral">                    --analysis kernel-smooth.sigma_span=50 \</span>
<a name="l00500"></a>00500 <span class="stringliteral">                    --analysis cn-cyto2.diagonal-weight=0.995.mapd-weight=0.22.min-segment-size=5.hmm-confidence-weight=0.6 \</span>
<a name="l00501"></a>00501 <span class="stringliteral">                    --analysis cn-cyto2-gender.cutoff=0.5 \</span>
<a name="l00502"></a>00502 <span class="stringliteral">                    --analysis cn-segment \</span>
<a name="l00503"></a>00503 <span class="stringliteral">                    --analysis loh-cyto2.lohCNSegSeparation=1000000.minInformation=100.lambdaCritical=8.0 \</span>
<a name="l00504"></a>00504 <span class="stringliteral">                    --analysis mosaicism.marker-bandwidth=6000.confidence-window=251 \</span>
<a name="l00505"></a>00505 <span class="stringliteral">                    --analysis allele-peaks.step=100.window=250.point-count=128.bandwidth=0.45.cutoff=0.05 \</span>
<a name="l00506"></a>00506 <span class="stringliteral">                    --local-gc-background-correction-reference-method pdnn-reference-method \</span>
<a name="l00507"></a>00507 <span class="stringliteral">                    --wave-correction-reference-method wave-correction-reference-method.trim=2.0.percentile=0.75.demean=false \</span>
<a name="l00508"></a>00508 <span class="stringliteral">                    --local-gc-background-intensity-adjustment-method pdnn-intensity-adjustment-method.predicted-intensity-bin-count=20.gc-bin-count=20.residual-trim=2.0 \</span>
<a name="l00509"></a>00509 <span class="stringliteral">                    --image-correction-intensity-adjustment-method high-pass-filter-intensity-adjustment-method.data-block-rows=320.data-block-cols=2015.mini-block-rows=8.mini-block-cols=8.global-smooth-weight=256.local-smooth-weight=64.converged=0.0001 \</span>
<a name="l00510"></a>00510 <span class="stringliteral">                    --wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method.bandwidth=101.bin-count=25.wave-count=3 \</span>
<a name="l00511"></a>00511 <span class="stringliteral">            &quot;</span>;
<a name="l00512"></a>00512         <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test(<span class="stringliteral">&quot;single&quot;</span>, command.c_str(), checks, <span class="keyword">false</span>);
<a name="l00513"></a>00513         test.setSuite(*<span class="keyword">this</span>,  outdir, outdir + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir + <span class="stringliteral">&quot;/valgrind.log&quot;</span>);
<a name="l00514"></a>00514         <span class="keywordflow">if</span>(!test.pass()) {
<a name="l00515"></a>00515             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: single. &quot;</span> + test.getErrorMsg());
<a name="l00516"></a>00516         }
<a name="l00517"></a>00517         <span class="keywordflow">else</span>
<a name="l00518"></a>00518         {
<a name="l00519"></a>00519             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: single. &quot;</span> + test.getErrorMsg());   
<a name="l00520"></a>00520         }
<a name="l00521"></a>00521         {
<a name="l00522"></a>00522             m_numPassed += test.getNumPassed();
<a name="l00523"></a>00523             m_numFailed += test.getNumFailed();
<a name="l00524"></a>00524         }
<a name="l00525"></a>00525 
<a name="l00526"></a>00526 }
<a name="l00527"></a>00527 
<a name="l00528"></a>00528 
<a name="l00529"></a>00529 
<a name="l00530"></a>00530 
<a name="l00531"></a>00531 <span class="keywordtype">void</span> single_newLOH()
<a name="l00532"></a>00532 {
<a name="l00533"></a>00533         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on single_newLOH.&quot;</span>);
<a name="l00534"></a>00534         <span class="keywordtype">string</span> outdir = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;single_newLOH&quot;</span>);
<a name="l00535"></a>00535 
<a name="l00536"></a>00536         std::string  goldDirectory=<span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/single_newLOH&quot;</span>;
<a name="l00537"></a>00537         std::string  newDirectory= <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;single_newLOH&quot;</span>);
<a name="l00538"></a>00538         <span class="keywordtype">string</span> infix = <span class="stringliteral">&quot;ca&quot;</span>;
<a name="l00539"></a>00539 
<a name="l00540"></a>00540         std::map&lt;std::string, float&gt; mapEpsilon;
<a name="l00541"></a>00541 
<a name="l00542"></a>00542         <a class="code" href="classFs.html#a1206f6cf8f445461466f224e6cd1bc52" title="///">Fs::rmdirPath</a>(outdir, <span class="keyword">false</span>);
<a name="l00543"></a>00543         <a class="code" href="classFs.html#abf363360d88071c2f4a93a7e50d61c9f" title="Make one directory. ///.">Fs::mkdir</a>(outdir, <span class="keyword">false</span>);
<a name="l00544"></a>00544 
<a name="l00545"></a>00545 
<a name="l00546"></a>00546         std::set&lt;std::string&gt; setSetIgnore;
<a name="l00547"></a>00547 <span class="preprocessor">#ifndef __LP64__</span>
<a name="l00548"></a>00548 <span class="preprocessor"></span>        {
<a name="l00549"></a>00549             std::string ignoreString2 = <span class="stringliteral">&quot;ProbeSets.AllelePeaks&quot;</span>;
<a name="l00550"></a>00550             setSetIgnore.insert(ignoreString2);
<a name="l00551"></a>00551         }
<a name="l00552"></a>00552 <span class="preprocessor">#endif</span>
<a name="l00553"></a>00553 <span class="preprocessor"></span><span class="preprocessor">#ifdef __APPLE__</span>
<a name="l00554"></a>00554 <span class="preprocessor"></span>        {
<a name="l00555"></a>00555             std::string ignoreString2 = <span class="stringliteral">&quot;ProbeSets.AllelePeaks&quot;</span>;
<a name="l00556"></a>00556             setSetIgnore.insert(ignoreString2);
<a name="l00557"></a>00557         }
<a name="l00558"></a>00558 <span class="preprocessor">#endif</span>
<a name="l00559"></a>00559 <span class="preprocessor"></span>
<a name="l00560"></a>00560 
<a name="l00561"></a>00561         vector&lt;RegressionCheck *&gt; checks;
<a name="l00562"></a>00562         checks.push_back(<span class="keyword">new</span> <a class="code" href="classCychpCheck.html" title="Class for testing that two cychp files are the same +/- some epsilon.">CychpCheck</a>(
<a name="l00563"></a>00563                                                 m_vCytogenetics_Array_SixFiles,
<a name="l00564"></a>00564                                                 goldDirectory,
<a name="l00565"></a>00565                                                 newDirectory,
<a name="l00566"></a>00566                                                 infix,
<a name="l00567"></a>00567                                                 setIgnore,
<a name="l00568"></a>00568                                                 setSetIgnore,
<a name="l00569"></a>00569                                                 mapEpsilon,
<a name="l00570"></a>00570                                                 0.0001,
<a name="l00571"></a>00571                                                 0.9999989,
<a name="l00572"></a>00572                                                 <span class="keyword">false</span>));
<a name="l00573"></a>00573 
<a name="l00574"></a>00574 
<a name="l00575"></a>00575 
<a name="l00576"></a>00576         <span class="keywordtype">string</span> command = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l00577"></a>00577 <span class="stringliteral">                    -v 3 --cyto2 true           \</span>
<a name="l00578"></a>00578 <span class="stringliteral">                    --force true \</span>
<a name="l00579"></a>00579 <span class="stringliteral">                    --text-output true \</span>
<a name="l00580"></a>00580 <span class="stringliteral">                    --cnchp-output false \</span>
<a name="l00581"></a>00581 <span class="stringliteral">                    --cychp-output true \</span>
<a name="l00582"></a>00582 <span class="stringliteral">                    --set-analysis-name ca \</span>
<a name="l00583"></a>00583 <span class="stringliteral">                    --keep-intermediate-data false \</span>
<a name="l00584"></a>00584 <span class="stringliteral">                    --meta-data-info wittgenstein=bushido \</span>
<a name="l00585"></a>00585 <span class="stringliteral">                    --reference-input ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.na28.REF_MODEL \</span>
<a name="l00586"></a>00586 <span class="stringliteral">                    --snp-reference-input-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.na28.single_newLOH.snpref.a5 \</span>
<a name="l00587"></a>00587 <span class="stringliteral">                    --cdf-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.CDF \</span>
<a name="l00588"></a>00588 <span class="stringliteral">                    --chrX-probes ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.chrXprobes \</span>
<a name="l00589"></a>00589 <span class="stringliteral">                    --chrY-probes ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.chrYprobes \</span>
<a name="l00590"></a>00590 <span class="stringliteral">                    --annotation-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.na28.annot.db \</span>
<a name="l00591"></a>00591 <span class="stringliteral">                    --probe-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.probe_tab \</span>
<a name="l00592"></a>00592 <span class="stringliteral">                    --snp-qc-snp-list ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.snplist.txt \</span>
<a name="l00593"></a>00593 <span class="stringliteral">                    --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18547_A08_01_NN_20081218.CEL \</span>
<a name="l00594"></a>00594 <span class="stringliteral">                    --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18603_B03_01_NN_20081218.CEL \</span>
<a name="l00595"></a>00595 <span class="stringliteral">                    --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18943_C11_01_NN_20081218.CEL \</span>
<a name="l00596"></a>00596 <span class="stringliteral">                    --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18971_C02_01_NN_20081218.CEL \</span>
<a name="l00597"></a>00597 <span class="stringliteral">                    --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-Cs_NA10857_A10_01_NN_20081218.CEL \</span>
<a name="l00598"></a>00598 <span class="stringliteral">                    --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-Cs_NA12003_D04_01_NN_20081218.CEL \</span>
<a name="l00599"></a>00599 <span class="stringliteral">                                        --out-dir &quot;</span> + outdir + <span class="stringliteral">&quot;        \</span>
<a name="l00600"></a>00600 <span class="stringliteral">                    --male-gender-ratio-cutoff=1.3 \</span>
<a name="l00601"></a>00601 <span class="stringliteral">                    --female-gender-ratio-cutoff=1.0 \</span>
<a name="l00602"></a>00602 <span class="stringliteral">                    --xx-cutoff=0.8 \</span>
<a name="l00603"></a>00603 <span class="stringliteral">                    --xx-cutoff-high=1.07 \</span>
<a name="l00604"></a>00604 <span class="stringliteral">                    --y-cutoff=0.65 \</span>
<a name="l00605"></a>00605 <span class="stringliteral">                    --analysis log2-ratio-cyto2.gc-correction=true.median-autosome-median-normalization=true.median-smooth-marker-count=5.trim-high=2.0.trim-low=-2.5 \</span>
<a name="l00606"></a>00606 <span class="stringliteral">                    --analysis kernel-smooth.sigma_span=50 \</span>
<a name="l00607"></a>00607 <span class="stringliteral">                    --analysis cn-cyto2.diagonal-weight=0.995.mapd-weight=0.22.min-segment-size=5.hmm-confidence-weight=0.6 \</span>
<a name="l00608"></a>00608 <span class="stringliteral">                    --analysis cn-cyto2-gender.cutoff=0.5 \</span>
<a name="l00609"></a>00609 <span class="stringliteral">                    --analysis cn-segment \</span>
<a name="l00610"></a>00610 <span class="stringliteral">                    --analysis loh-cyto2.lohCNSegSeparation=1000000.minInformation=100.lambdaCritical=8.0 \</span>
<a name="l00611"></a>00611 <span class="stringliteral">                    --analysis mosaicism.marker-bandwidth=6000.confidence-window=251 \</span>
<a name="l00612"></a>00612 <span class="stringliteral">                    --analysis allele-peaks.step=100.window=250.point-count=128.bandwidth=0.45.cutoff=0.05 \</span>
<a name="l00613"></a>00613 <span class="stringliteral">                    --local-gc-background-correction-reference-method pdnn-reference-method \</span>
<a name="l00614"></a>00614 <span class="stringliteral">                    --wave-correction-reference-method wave-correction-reference-method.trim=2.0.percentile=0.75.demean=false \</span>
<a name="l00615"></a>00615 <span class="stringliteral">                    --local-gc-background-intensity-adjustment-method pdnn-intensity-adjustment-method.predicted-intensity-bin-count=20.gc-bin-count=20.residual-trim=2.0 \</span>
<a name="l00616"></a>00616 <span class="stringliteral">                    --image-correction-intensity-adjustment-method high-pass-filter-intensity-adjustment-method.data-block-rows=320.data-block-cols=2015.mini-block-rows=8.mini-block-cols=8.global-smooth-weight=256.local-smooth-weight=64.converged=0.0001 \</span>
<a name="l00617"></a>00617 <span class="stringliteral">                    --wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method.bandwidth=101.bin-count=25.wave-count=3 \</span>
<a name="l00618"></a>00618 <span class="stringliteral">            &quot;</span>;
<a name="l00619"></a>00619         <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test(<span class="stringliteral">&quot;single_newLOH&quot;</span>, command.c_str(), checks, <span class="keyword">false</span>);
<a name="l00620"></a>00620         test.setSuite(*<span class="keyword">this</span>,  outdir, outdir + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir + <span class="stringliteral">&quot;/valgrind.log&quot;</span>);
<a name="l00621"></a>00621         <span class="keywordflow">if</span>(!test.pass()) {
<a name="l00622"></a>00622             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in:  single_newLOH.&quot;</span> + test.getErrorMsg());
<a name="l00623"></a>00623         }
<a name="l00624"></a>00624         <span class="keywordflow">else</span>
<a name="l00625"></a>00625         {
<a name="l00626"></a>00626             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: single_newLOH. &quot;</span> + test.getErrorMsg());   
<a name="l00627"></a>00627         }
<a name="l00628"></a>00628         {
<a name="l00629"></a>00629             m_numPassed += test.getNumPassed();
<a name="l00630"></a>00630             m_numFailed += test.getNumFailed();
<a name="l00631"></a>00631         }
<a name="l00632"></a>00632 
<a name="l00633"></a>00633 }
<a name="l00634"></a>00634 
<a name="l00635"></a>00635 
<a name="l00636"></a>00636 
<a name="l00637"></a>00637 
<a name="l00638"></a>00638 
<a name="l00639"></a>00639 <span class="keywordtype">void</span> single_na30()
<a name="l00640"></a>00640 {
<a name="l00641"></a>00641         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on single_na30.&quot;</span>);
<a name="l00642"></a>00642         <span class="keywordtype">string</span> outdir = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;single_na30&quot;</span>);
<a name="l00643"></a>00643 
<a name="l00644"></a>00644         std::string  goldDirectory=<span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/single_na30&quot;</span>;
<a name="l00645"></a>00645         std::string  newDirectory= <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;single_na30&quot;</span>);
<a name="l00646"></a>00646         <span class="keywordtype">string</span> infix = <span class="stringliteral">&quot;ca&quot;</span>;
<a name="l00647"></a>00647 
<a name="l00648"></a>00648         std::map&lt;std::string, float&gt; mapEpsilon;
<a name="l00649"></a>00649 
<a name="l00650"></a>00650         <a class="code" href="classFs.html#a1206f6cf8f445461466f224e6cd1bc52" title="///">Fs::rmdirPath</a>(outdir, <span class="keyword">false</span>);
<a name="l00651"></a>00651         <a class="code" href="classFs.html#abf363360d88071c2f4a93a7e50d61c9f" title="Make one directory. ///.">Fs::mkdir</a>(outdir, <span class="keyword">false</span>);
<a name="l00652"></a>00652 
<a name="l00653"></a>00653         std::vector&lt;std::string&gt; vLocalToCompare;
<a name="l00654"></a>00654         std::set&lt;std::string&gt; setSetIgnore;
<a name="l00655"></a>00655 
<a name="l00656"></a>00656 
<a name="l00657"></a>00657 <span class="preprocessor">#ifdef _WIN32</span>
<a name="l00658"></a>00658 <span class="preprocessor"></span>        vLocalToCompare.clear();
<a name="l00659"></a>00659         std::vector&lt;std::string&gt;::const_iterator it;
<a name="l00660"></a>00660         <span class="keywordflow">for</span> ( it = m_vCytogenetics_Array_SixFiles.begin(); it != m_vCytogenetics_Array_SixFiles.end(); it++ ) {
<a name="l00661"></a>00661           <span class="keywordflow">if</span> ( *it != std::string(<span class="stringliteral">&quot;HapMap-As_NA18943_C11_01_NN_20081218&quot;</span> )) {
<a name="l00662"></a>00662             vLocalToCompare.push_back(*it);
<a name="l00663"></a>00663           }
<a name="l00664"></a>00664         }
<a name="l00665"></a>00665 <span class="preprocessor">#else</span>
<a name="l00666"></a>00666 <span class="preprocessor"></span>        vLocalToCompare =  m_vCytogenetics_Array_SixFiles;
<a name="l00667"></a>00667 <span class="preprocessor">#endif</span>
<a name="l00668"></a>00668 <span class="preprocessor"></span>        setSetIgnore.clear();
<a name="l00669"></a>00669 
<a name="l00670"></a>00670 <span class="preprocessor">#ifndef __LP64__</span>
<a name="l00671"></a>00671 <span class="preprocessor"></span>        {
<a name="l00672"></a>00672             std::string ignoreString2 = <span class="stringliteral">&quot;ProbeSets.AllelePeaks&quot;</span>;
<a name="l00673"></a>00673             setSetIgnore.insert(ignoreString2);
<a name="l00674"></a>00674         }
<a name="l00675"></a>00675 <span class="preprocessor">#endif</span>
<a name="l00676"></a>00676 <span class="preprocessor"></span><span class="preprocessor">#ifdef __APPLE__</span>
<a name="l00677"></a>00677 <span class="preprocessor"></span>        {
<a name="l00678"></a>00678             std::string ignoreString2 = <span class="stringliteral">&quot;ProbeSets.AllelePeaks&quot;</span>;
<a name="l00679"></a>00679             setSetIgnore.insert(ignoreString2);
<a name="l00680"></a>00680         }
<a name="l00681"></a>00681 <span class="preprocessor">#endif</span>
<a name="l00682"></a>00682 <span class="preprocessor"></span>
<a name="l00683"></a>00683 
<a name="l00684"></a>00684 
<a name="l00685"></a>00685         vector&lt;RegressionCheck *&gt; checks;
<a name="l00686"></a>00686         checks.push_back(<span class="keyword">new</span> <a class="code" href="classCychpCheck.html" title="Class for testing that two cychp files are the same +/- some epsilon.">CychpCheck</a>(
<a name="l00687"></a>00687                                                 vLocalToCompare,
<a name="l00688"></a>00688                                                 goldDirectory,
<a name="l00689"></a>00689                                                 newDirectory,
<a name="l00690"></a>00690                                                 infix,
<a name="l00691"></a>00691                                                 setIgnore,
<a name="l00692"></a>00692                                                 setSetIgnore,
<a name="l00693"></a>00693                                                 mapEpsilon,
<a name="l00694"></a>00694                                                 0.0001,
<a name="l00695"></a>00695                                                 0.9999989,
<a name="l00696"></a>00696                                                 <span class="keyword">false</span>));
<a name="l00697"></a>00697 
<a name="l00698"></a>00698 
<a name="l00699"></a>00699 
<a name="l00700"></a>00700         <span class="keywordtype">string</span> command = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l00701"></a>00701 <span class="stringliteral">                    -v 3 --cyto2 true           \</span>
<a name="l00702"></a>00702 <span class="stringliteral">                    --force true \</span>
<a name="l00703"></a>00703 <span class="stringliteral">                    --text-output true \</span>
<a name="l00704"></a>00704 <span class="stringliteral">                    --cnchp-output false \</span>
<a name="l00705"></a>00705 <span class="stringliteral">                    --cychp-output true \</span>
<a name="l00706"></a>00706 <span class="stringliteral">                    --set-analysis-name ca \</span>
<a name="l00707"></a>00707 <span class="stringliteral">                    --keep-intermediate-data false \</span>
<a name="l00708"></a>00708 <span class="stringliteral">                    --reference-input ../../../regression-data/data/lib/Cytogenetics_Array/NA30/Cytogenetics_Array.na30.v1.REF_MODEL \</span>
<a name="l00709"></a>00709 <span class="stringliteral">                    --snp-reference-input-file ../../../regression-data/data/lib/Cytogenetics_Array/NA30/Cytogenetics_Array.na30.v1.snpref.a5 \</span>
<a name="l00710"></a>00710 <span class="stringliteral">                    --cdf-file ../../../regression-data/data/lib/Cytogenetics_Array/NA30/Cytogenetics_Array.CDF \</span>
<a name="l00711"></a>00711 <span class="stringliteral">                    --chrX-probes ../../../regression-data/data/lib/Cytogenetics_Array/NA30/Cytogenetics_Array.chrXprobes \</span>
<a name="l00712"></a>00712 <span class="stringliteral">                    --chrY-probes ../../../regression-data/data/lib/Cytogenetics_Array/NA30/Cytogenetics_Array.v2.chrYprobes \</span>
<a name="l00713"></a>00713 <span class="stringliteral">                    --annotation-file ../../../regression-data/data/lib/Cytogenetics_Array/NA30/Cytogenetics_Array.na30.1.annot.db \</span>
<a name="l00714"></a>00714 <span class="stringliteral">                    --probe-file ../../../regression-data/data/lib/Cytogenetics_Array/NA30/Cytogenetics_Array.probe_tab \</span>
<a name="l00715"></a>00715 <span class="stringliteral">                    --snp-qc-snp-list ../../../regression-data/data/lib/Cytogenetics_Array/NA30/Cytogenetics_Array.snplist.txt \</span>
<a name="l00716"></a>00716 <span class="stringliteral">                    --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18547_A08_01_NN_20081218.CEL \</span>
<a name="l00717"></a>00717 <span class="stringliteral">                    --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18603_B03_01_NN_20081218.CEL \</span>
<a name="l00718"></a>00718 <span class="stringliteral">                    --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18943_C11_01_NN_20081218.CEL \</span>
<a name="l00719"></a>00719 <span class="stringliteral">                    --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18971_C02_01_NN_20081218.CEL \</span>
<a name="l00720"></a>00720 <span class="stringliteral">                    --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-Cs_NA10857_A10_01_NN_20081218.CEL \</span>
<a name="l00721"></a>00721 <span class="stringliteral">                    --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-Cs_NA12003_D04_01_NN_20081218.CEL \</span>
<a name="l00722"></a>00722 <span class="stringliteral">                                        --out-dir &quot;</span> + outdir + <span class="stringliteral">&quot;        \</span>
<a name="l00723"></a>00723 <span class="stringliteral">                    --male-gender-ratio-cutoff=1.3 \</span>
<a name="l00724"></a>00724 <span class="stringliteral">                    --female-gender-ratio-cutoff=1.0 \</span>
<a name="l00725"></a>00725 <span class="stringliteral">                    --xx-cutoff=0.8 \</span>
<a name="l00726"></a>00726 <span class="stringliteral">                    --xx-cutoff-high=1.07 \</span>
<a name="l00727"></a>00727 <span class="stringliteral">                    --y-cutoff=0.65 \</span>
<a name="l00728"></a>00728 <span class="stringliteral">                    --analysis log2-ratio-cyto2.gc-correction=true.median-autosome-median-normalization=true.median-smooth-marker-count=5.trim-high=2.0.trim-low=-2.5 \</span>
<a name="l00729"></a>00729 <span class="stringliteral">                    --analysis kernel-smooth.sigma_span=50 \</span>
<a name="l00730"></a>00730 <span class="stringliteral">                    --analysis cn-cyto2.diagonal-weight=0.995.mapd-weight=0.22.min-segment-size=5.hmm-confidence-weight=0.6 \</span>
<a name="l00731"></a>00731 <span class="stringliteral">                    --analysis cn-cyto2-gender.cutoff=0.5 \</span>
<a name="l00732"></a>00732 <span class="stringliteral">                    --analysis cn-segment \</span>
<a name="l00733"></a>00733 <span class="stringliteral">                    --analysis loh-cyto2.lohCNSegSeparation=1000000.minInformation=100.lambdaCritical=8.0 \</span>
<a name="l00734"></a>00734 <span class="stringliteral">                    --analysis mosaicism.marker-bandwidth=6000.confidence-window=251 \</span>
<a name="l00735"></a>00735 <span class="stringliteral">                    --analysis allele-peaks.step=100.window=250.point-count=128.bandwidth=0.45.cutoff=0.05 \</span>
<a name="l00736"></a>00736 <span class="stringliteral">                    --local-gc-background-correction-reference-method pdnn-reference-method \</span>
<a name="l00737"></a>00737 <span class="stringliteral">                    --wave-correction-reference-method wave-correction-reference-method.trim=2.0.percentile=0.75.demean=false \</span>
<a name="l00738"></a>00738 <span class="stringliteral">                    --local-gc-background-intensity-adjustment-method pdnn-intensity-adjustment-method \</span>
<a name="l00739"></a>00739 <span class="stringliteral">                    --image-correction-intensity-adjustment-method high-pass-filter-intensity-adjustment-method \</span>
<a name="l00740"></a>00740 <span class="stringliteral">                    --wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method \</span>
<a name="l00741"></a>00741 <span class="stringliteral">            &quot;</span>;
<a name="l00742"></a>00742         <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test(<span class="stringliteral">&quot;single_na30&quot;</span>,command.c_str(), checks, <span class="keyword">false</span>);
<a name="l00743"></a>00743         test.setSuite(*<span class="keyword">this</span>,  outdir, outdir + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir + <span class="stringliteral">&quot;/valgrind.log&quot;</span>);
<a name="l00744"></a>00744         <span class="keywordflow">if</span>(!test.pass()) {
<a name="l00745"></a>00745             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: single_na30&quot;</span> + test.getErrorMsg());
<a name="l00746"></a>00746         }
<a name="l00747"></a>00747         <span class="keywordflow">else</span>
<a name="l00748"></a>00748         {
<a name="l00749"></a>00749             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: single_na30 &quot;</span> + test.getErrorMsg());   
<a name="l00750"></a>00750         }
<a name="l00751"></a>00751 
<a name="l00752"></a>00752         {
<a name="l00753"></a>00753             m_numPassed += test.getNumPassed();
<a name="l00754"></a>00754             m_numFailed += test.getNumFailed();
<a name="l00755"></a>00755         }
<a name="l00756"></a>00756 
<a name="l00757"></a>00757 }
<a name="l00758"></a>00758 
<a name="l00759"></a>00759 
<a name="l00760"></a>00760 <span class="keywordtype">void</span> batch()
<a name="l00761"></a>00761 {
<a name="l00762"></a>00762         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on batch part 1.&quot;</span>);
<a name="l00763"></a>00763         <span class="keywordtype">string</span> outdir = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;batch&quot;</span>);
<a name="l00764"></a>00764         <a class="code" href="classFs.html#a1206f6cf8f445461466f224e6cd1bc52" title="///">Fs::rmdirPath</a>(outdir, <span class="keyword">false</span>);
<a name="l00765"></a>00765         <a class="code" href="classFs.html#abf363360d88071c2f4a93a7e50d61c9f" title="Make one directory. ///.">Fs::mkdir</a>(outdir, <span class="keyword">false</span>);
<a name="l00766"></a>00766         vector&lt;RegressionCheck *&gt; checks;
<a name="l00767"></a>00767 
<a name="l00768"></a>00768 
<a name="l00769"></a>00769         <span class="keywordtype">string</span> command = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l00770"></a>00770 <span class="stringliteral">                    -v 3 --cyto2 true \</span>
<a name="l00771"></a>00771 <span class="stringliteral">                    --force true \</span>
<a name="l00772"></a>00772 <span class="stringliteral">                    --check-input-files true \</span>
<a name="l00773"></a>00773 <span class="stringliteral">                    --text-output true \</span>
<a name="l00774"></a>00774 <span class="stringliteral">                    --cnchp-output false \</span>
<a name="l00775"></a>00775 <span class="stringliteral">                    --cychp-output true \</span>
<a name="l00776"></a>00776 <span class="stringliteral">                    --set-analysis-name ca \</span>
<a name="l00777"></a>00777 <span class="stringliteral">                    --keep-intermediate-data true \</span>
<a name="l00778"></a>00778 <span class="stringliteral">                    --meta-data-info wittgenstein=bushido \</span>
<a name="l00779"></a>00779 <span class="stringliteral">                    --meta-data-info batchMode=Creation \</span>
<a name="l00780"></a>00780 <span class="stringliteral">                    --reference-output &quot;</span> + m_testDir + <span class="stringliteral">&quot;/batch/Cytogenetics_Array.10_cels.REF_MODEL \</span>
<a name="l00781"></a>00781 <span class="stringliteral">                    --snp-reference-input-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.na28.snpref.a5 \</span>
<a name="l00782"></a>00782 <span class="stringliteral">                    --snp-reference-output-file &quot;</span> + m_testDir + <span class="stringliteral">&quot;/batch/Cytogenetics_Array.10_cels.snpref.a5 \</span>
<a name="l00783"></a>00783 <span class="stringliteral">                    --cdf-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.CDF \</span>
<a name="l00784"></a>00784 <span class="stringliteral">                    --chrX-probes ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.chrXprobes \</span>
<a name="l00785"></a>00785 <span class="stringliteral">                    --chrY-probes ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.chrYprobes \</span>
<a name="l00786"></a>00786 <span class="stringliteral">                    --annotation-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.na28.annot.db \</span>
<a name="l00787"></a>00787 <span class="stringliteral">                    --probe-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.probe_tab \</span>
<a name="l00788"></a>00788 <span class="stringliteral">                    --snp-qc-snp-list ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.snplist.txt \</span>
<a name="l00789"></a>00789 <span class="stringliteral">                    --gender-override-file ../../../regression-data/data/lib/Cytogenetics_Array/HapMap-LOHReference-Full-Genders.txt \</span>
<a name="l00790"></a>00790 <span class="stringliteral">                    --genotype-call-override-file ../../../regression-data/data/lib/Cytogenetics_Array/HapMapCalls_LOHReferenceSet-FullArray.txt \</span>
<a name="l00791"></a>00791 <span class="stringliteral">                    --cel-files ../../../regression-data/data/cel/Cytogenetics_Array/CELFileList_5F5M.txt \</span>
<a name="l00792"></a>00792 <span class="stringliteral">                    --out-dir &quot;</span> + outdir + <span class="stringliteral">&quot;        \</span>
<a name="l00793"></a>00793 <span class="stringliteral">                    --male-gender-ratio-cutoff=1.3 \</span>
<a name="l00794"></a>00794 <span class="stringliteral">                    --female-gender-ratio-cutoff=1.0 \</span>
<a name="l00795"></a>00795 <span class="stringliteral">                    --xx-cutoff=0.8 \</span>
<a name="l00796"></a>00796 <span class="stringliteral">                    --xx-cutoff-high=1.07 \</span>
<a name="l00797"></a>00797 <span class="stringliteral">                    --y-cutoff=0.65 \</span>
<a name="l00798"></a>00798 <span class="stringliteral">                    --analysis log2-ratio-cyto2.gc-correction=true.median-autosome-median-normalization=true.median-smooth-marker-count=5.trim-high=2.0.trim-low=-2.5 \</span>
<a name="l00799"></a>00799 <span class="stringliteral">                    --analysis kernel-smooth.sigma_span=50 \</span>
<a name="l00800"></a>00800 <span class="stringliteral">                    --analysis cn-cyto2.diagonal-weight=0.995.mapd-weight=0.22.min-segment-size=5.hmm-confidence-weight=0.6 \</span>
<a name="l00801"></a>00801 <span class="stringliteral">                    --analysis cn-cyto2-gender.cutoff=0.5 \</span>
<a name="l00802"></a>00802 <span class="stringliteral">                    --analysis cn-segment \</span>
<a name="l00803"></a>00803 <span class="stringliteral">                    --analysis loh-cyto2.lohCNSegSeparation=1000000.minInformation=100.lambdaCritical=8.0 \</span>
<a name="l00804"></a>00804 <span class="stringliteral">                    --analysis mosaicism.marker-bandwidth=6000.confidence-window=251 \</span>
<a name="l00805"></a>00805 <span class="stringliteral">                    --analysis allele-peaks.step=100.window=250.point-count=128.bandwidth=0.45.cutoff=0.05 \</span>
<a name="l00806"></a>00806 <span class="stringliteral">                    --local-gc-background-correction-reference-method pdnn-reference-method \</span>
<a name="l00807"></a>00807 <span class="stringliteral">                    --wave-correction-reference-method wave-correction-reference-method.trim=2.0.percentile=0.75.demean=false \</span>
<a name="l00808"></a>00808 <span class="stringliteral">                    --local-gc-background-intensity-adjustment-method pdnn-intensity-adjustment-method.predicted-intensity-bin-count=20.gc-bin-count=20.residual-trim=2.0 \</span>
<a name="l00809"></a>00809 <span class="stringliteral">                    --image-correction-intensity-adjustment-method high-pass-filter-intensity-adjustment-method.data-block-rows=320.data-block-cols=2015.mini-block-rows=8.mini-block-cols=8.global-smooth-weight=256.local-smooth-weight=64.converged=0.0001 \</span>
<a name="l00810"></a>00810 <span class="stringliteral">                    --wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method.bandwidth=101.bin-count=25.wave-count=3 \</span>
<a name="l00811"></a>00811 <span class="stringliteral">            &quot;</span>;
<a name="l00812"></a>00812 
<a name="l00813"></a>00813         <span class="keywordtype">bool</span> bPassed = <span class="keyword">true</span>;
<a name="l00814"></a>00814         std::string strFileName1 = <span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/batch/Cytogenetics_Array.10_cels.snpref.a5&quot;</span>;
<a name="l00815"></a>00815         std::string strFileName2 = m_testDir + <span class="stringliteral">&quot;/batch/Cytogenetics_Array.10_cels.snpref.a5&quot;</span>;
<a name="l00816"></a>00816         std::set&lt;std::string&gt; setLocalIgnore;
<a name="l00817"></a>00817 
<a name="l00818"></a>00818 
<a name="l00819"></a>00819         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;SNPReference&quot;</span>, <span class="stringliteral">&quot;EMSNParameters&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l00820"></a>00820         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;SNPReference&quot;</span>, <span class="stringliteral">&quot;EMSNArray&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l00821"></a>00821         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;SNPReference&quot;</span>, <span class="stringliteral">&quot;SNPReference&quot;</span>, setLocalIgnore, 0.0003, 1.0));
<a name="l00822"></a>00822 
<a name="l00823"></a>00823         strFileName1 = <span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/batch/Cytogenetics_Array.10_cels.REF_MODEL&quot;</span>;
<a name="l00824"></a>00824         strFileName2 = m_testDir + <span class="stringliteral">&quot;/batch/Cytogenetics_Array.10_cels.REF_MODEL&quot;</span>;
<a name="l00825"></a>00825         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;AntigenomicProbes&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l00826"></a>00826         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;MedianSignals&quot;</span>, setLocalIgnore, 0.5, 0.999999));
<a name="l00827"></a>00827         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;ProbeEffects&quot;</span>, setLocalIgnore, 0.0001, 0.999999));
<a name="l00828"></a>00828         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;SNPPosteriors&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l00829"></a>00829         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;SketchCN&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l00830"></a>00830         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;SketchSNP&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l00831"></a>00831         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;WaveCorrection&quot;</span>, setLocalIgnore, 0.0001, 0.99999));
<a name="l00832"></a>00832         <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test(<span class="stringliteral">&quot;batch part 1&quot;</span>, command.c_str(), checks);
<a name="l00833"></a>00833         test.setSuite(*<span class="keyword">this</span>,  outdir, outdir + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir + <span class="stringliteral">&quot;/valgrind.log&quot;</span>);
<a name="l00834"></a>00834         bPassed = test.pass();
<a name="l00835"></a>00835         <span class="keywordflow">if</span>(!bPassed) {
<a name="l00836"></a>00836             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: batch part 1 &quot;</span> + test.getErrorMsg());
<a name="l00837"></a>00837         }
<a name="l00838"></a>00838         <span class="keywordflow">else</span>
<a name="l00839"></a>00839         {
<a name="l00840"></a>00840             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: batch part 1. &quot;</span> + test.getErrorMsg());   
<a name="l00841"></a>00841         }
<a name="l00842"></a>00842 
<a name="l00843"></a>00843         {
<a name="l00844"></a>00844             m_numPassed += test.getNumPassed();
<a name="l00845"></a>00845             m_numFailed += test.getNumFailed();
<a name="l00846"></a>00846         }
<a name="l00847"></a>00847 
<a name="l00848"></a>00848 
<a name="l00849"></a>00849         <span class="keywordflow">if</span>(bPassed)
<a name="l00850"></a>00850         {
<a name="l00851"></a>00851                 <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on batch part 2.&quot;</span>);
<a name="l00852"></a>00852                 std::set&lt;std::string&gt; setSetIgnore;
<a name="l00853"></a>00853 <span class="preprocessor">#ifndef __LP64__</span>
<a name="l00854"></a>00854 <span class="preprocessor"></span>                {
<a name="l00855"></a>00855                     std::string ignoreString = <span class="stringliteral">&quot;Segments.Mosaicism&quot;</span>;
<a name="l00856"></a>00856                     setSetIgnore.insert(ignoreString);
<a name="l00857"></a>00857                     std::string ignoreString2 = <span class="stringliteral">&quot;ProbeSets.AllelePeaks&quot;</span>;
<a name="l00858"></a>00858                     setSetIgnore.insert(ignoreString2);
<a name="l00859"></a>00859                 }                
<a name="l00860"></a>00860 <span class="preprocessor">#endif</span>
<a name="l00861"></a>00861 <span class="preprocessor"></span><span class="preprocessor">#ifdef __APPLE__</span>
<a name="l00862"></a>00862 <span class="preprocessor"></span>                {
<a name="l00863"></a>00863                     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;^^^MaxSignal being filtered out.^^^&quot;</span>);
<a name="l00864"></a>00864                     std::string ignoreString = <span class="stringliteral">&quot;Segments.Mosaicism&quot;</span>;
<a name="l00865"></a>00865                     setSetIgnore.insert(ignoreString);
<a name="l00866"></a>00866                     std::string  ignoreString2 = <span class="stringliteral">&quot;Chromosomes.Summary&quot;</span>;
<a name="l00867"></a>00867                     setSetIgnore.insert(ignoreString2);
<a name="l00868"></a>00868                     std::string ignoreString3 = <span class="stringliteral">&quot;ProbeSets.AllelePeaks&quot;</span>;
<a name="l00869"></a>00869                     setSetIgnore.insert(ignoreString3);
<a name="l00870"></a>00870                 }
<a name="l00871"></a>00871 <span class="preprocessor">#endif</span>
<a name="l00872"></a>00872 <span class="preprocessor"></span>                std::string  goldDirectory=<span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/batch&quot;</span>;
<a name="l00873"></a>00873                 std::string  newDirectory=<a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;batch&quot;</span>);
<a name="l00874"></a>00874                 <span class="keywordtype">string</span> infix = <span class="stringliteral">&quot;ca&quot;</span>; 
<a name="l00875"></a>00875 
<a name="l00876"></a>00876                 std::map&lt;std::string, float&gt; mapEpsilon;
<a name="l00877"></a>00877 
<a name="l00878"></a>00878                 <span class="keywordtype">string</span> outdir2 = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;batch&quot;</span>);
<a name="l00879"></a>00879                 vector&lt;RegressionCheck *&gt; checks2;
<a name="l00880"></a>00880 
<a name="l00881"></a>00881                 checks2.push_back(<span class="keyword">new</span> <a class="code" href="classCychpCheck.html" title="Class for testing that two cychp files are the same +/- some epsilon.">CychpCheck</a>(
<a name="l00882"></a>00882                                                 m_vCytogenetics_Array_SixFiles,
<a name="l00883"></a>00883                                                 goldDirectory,
<a name="l00884"></a>00884                                                 newDirectory,
<a name="l00885"></a>00885                                                 infix,
<a name="l00886"></a>00886                                                 setIgnore,
<a name="l00887"></a>00887                                                 setSetIgnore,
<a name="l00888"></a>00888                                                 mapEpsilon,
<a name="l00889"></a>00889                                                 0.0001,
<a name="l00890"></a>00890                                                 0.9999989,
<a name="l00891"></a>00891                                                 <span class="keyword">false</span>));
<a name="l00892"></a>00892 
<a name="l00893"></a>00893 
<a name="l00894"></a>00894                 <span class="keywordtype">string</span> command2 = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l00895"></a>00895 <span class="stringliteral">                        -v 3 --cyto2 true                \</span>
<a name="l00896"></a>00896 <span class="stringliteral">                        --force true                     \</span>
<a name="l00897"></a>00897 <span class="stringliteral">                        --text-output true               \</span>
<a name="l00898"></a>00898 <span class="stringliteral">                        --cnchp-output false             \</span>
<a name="l00899"></a>00899 <span class="stringliteral">                        --cychp-output true              \</span>
<a name="l00900"></a>00900 <span class="stringliteral">                        --set-analysis-name ca                 \</span>
<a name="l00901"></a>00901 <span class="stringliteral">                         --meta-data-info wittgenstein=bushido          \</span>
<a name="l00902"></a>00902 <span class="stringliteral">                         --meta-data-info batchMode=Use                 \</span>
<a name="l00903"></a>00903 <span class="stringliteral">                        --reference-input &quot;</span> + m_testDir + <span class="stringliteral">&quot;/batch/Cytogenetics_Array.10_cels.REF_MODEL \</span>
<a name="l00904"></a>00904 <span class="stringliteral">                        --snp-reference-input-file &quot;</span> + m_testDir + <span class="stringliteral">&quot;/batch/Cytogenetics_Array.10_cels.snpref.a5 \</span>
<a name="l00905"></a>00905 <span class="stringliteral">                        --cdf-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.CDF \</span>
<a name="l00906"></a>00906 <span class="stringliteral">                        --chrX-probes ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.chrXprobes \</span>
<a name="l00907"></a>00907 <span class="stringliteral">                        --chrY-probes ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.chrYprobes \</span>
<a name="l00908"></a>00908 <span class="stringliteral">                        --annotation-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.na28.annot.db \</span>
<a name="l00909"></a>00909 <span class="stringliteral">                        --probe-file ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.probe_tab \</span>
<a name="l00910"></a>00910 <span class="stringliteral">                        --snp-qc-snp-list ../../../regression-data/data/lib/Cytogenetics_Array/Cytogenetics_Array.snplist.txt \</span>
<a name="l00911"></a>00911 <span class="stringliteral">                        --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18547_A08_01_NN_20081218.CEL \</span>
<a name="l00912"></a>00912 <span class="stringliteral">                        --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18603_B03_01_NN_20081218.CEL \</span>
<a name="l00913"></a>00913 <span class="stringliteral">                        --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18943_C11_01_NN_20081218.CEL \</span>
<a name="l00914"></a>00914 <span class="stringliteral">                        --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-As_NA18971_C02_01_NN_20081218.CEL \</span>
<a name="l00915"></a>00915 <span class="stringliteral">                        --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-Cs_NA10857_A10_01_NN_20081218.CEL \</span>
<a name="l00916"></a>00916 <span class="stringliteral">                        --cels ../../../regression-data/data/cel/Cytogenetics_Array/HapMap-Cs_NA12003_D04_01_NN_20081218.CEL \</span>
<a name="l00917"></a>00917 <span class="stringliteral">                        --out-dir &quot;</span> + outdir2 + <span class="stringliteral">&quot;        \</span>
<a name="l00918"></a>00918 <span class="stringliteral">                        --male-gender-ratio-cutoff=1.3 \</span>
<a name="l00919"></a>00919 <span class="stringliteral">                        --female-gender-ratio-cutoff=1.0 \</span>
<a name="l00920"></a>00920 <span class="stringliteral">                        --xx-cutoff=0.8 \</span>
<a name="l00921"></a>00921 <span class="stringliteral">                        --xx-cutoff-high=1.07 \</span>
<a name="l00922"></a>00922 <span class="stringliteral">                        --y-cutoff=0.65 \</span>
<a name="l00923"></a>00923 <span class="stringliteral">                        --keep-intermediate-data false \</span>
<a name="l00924"></a>00924 <span class="stringliteral">                        --analysis log2-ratio-cyto2.gc-correction=true.median-autosome-median-normalization=true.median-smooth-marker-count=5.trim-high=2.0.trim-low=-2.5 \</span>
<a name="l00925"></a>00925 <span class="stringliteral">                        --analysis kernel-smooth.sigma_span=50 \</span>
<a name="l00926"></a>00926 <span class="stringliteral">                        --analysis cn-cyto2.diagonal-weight=0.995.mapd-weight=0.22.min-segment-size=5.hmm-confidence-weight=0.6 \</span>
<a name="l00927"></a>00927 <span class="stringliteral">                        --analysis cn-cyto2-gender.cutoff=0.5 \</span>
<a name="l00928"></a>00928 <span class="stringliteral">                        --analysis cn-segment \</span>
<a name="l00929"></a>00929 <span class="stringliteral">                        --analysis loh-cyto2.lohCNSegSeparation=1000000.minInformation=100.lambdaCritical=8.0 \</span>
<a name="l00930"></a>00930 <span class="stringliteral">                        --analysis mosaicism.marker-bandwidth=6000.confidence-window=251 \</span>
<a name="l00931"></a>00931 <span class="stringliteral">                        --analysis allele-peaks.step=100.window=250.point-count=128.bandwidth=0.45.cutoff=0.05 \</span>
<a name="l00932"></a>00932 <span class="stringliteral">                        --local-gc-background-correction-reference-method pdnn-reference-method \</span>
<a name="l00933"></a>00933 <span class="stringliteral">                        --wave-correction-reference-method wave-correction-reference-method.trim=2.0.percentile=0.75.demean=false \</span>
<a name="l00934"></a>00934 <span class="stringliteral">                        --local-gc-background-intensity-adjustment-method pdnn-intensity-adjustment-method.predicted-intensity-bin-count=20.gc-bin-count=20.residual-trim=2.0 \</span>
<a name="l00935"></a>00935 <span class="stringliteral">                        --image-correction-intensity-adjustment-method high-pass-filter-intensity-adjustment-method.data-block-rows=320.data-block-cols=2015.mini-block-rows=8.mini-block-cols=8.global-smooth-weight=256.local-smooth-weight=64.converged=0.0001 \</span>
<a name="l00936"></a>00936 <span class="stringliteral">                        --wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method.bandwidth=101.bin-count=25.wave-count=3 \</span>
<a name="l00937"></a>00937 <span class="stringliteral">                &quot;</span>;
<a name="l00938"></a>00938                 <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test2(<span class="stringliteral">&quot;batch part 2&quot;</span>, command2.c_str(), checks2, <span class="keyword">false</span>);
<a name="l00939"></a>00939                 test2.setSuite(*<span class="keyword">this</span>,  outdir2, outdir2 + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir2 + <span class="stringliteral">&quot;/valgrind.batch2.log&quot;</span>);
<a name="l00940"></a>00940                 <span class="keywordtype">bool</span> bPassed = test2.pass();
<a name="l00941"></a>00941                 <span class="keywordflow">if</span>(!bPassed)
<a name="l00942"></a>00942                 {
<a name="l00943"></a>00943                     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: batch part2 &quot;</span> + test2.getErrorMsg());
<a name="l00944"></a>00944                 }
<a name="l00945"></a>00945                 <span class="keywordflow">else</span>
<a name="l00946"></a>00946                 {
<a name="l00947"></a>00947                     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: batch part 2. &quot;</span> + test.getErrorMsg());      
<a name="l00948"></a>00948                 }
<a name="l00949"></a>00949 
<a name="l00950"></a>00950                 {
<a name="l00951"></a>00951                     m_numPassed += test2.getNumPassed();
<a name="l00952"></a>00952                     m_numFailed += test2.getNumFailed();
<a name="l00953"></a>00953                 }
<a name="l00954"></a>00954 
<a name="l00955"></a>00955         }
<a name="l00956"></a>00956     
<a name="l00957"></a>00957 }
<a name="l00958"></a>00958 
<a name="l00959"></a>00959 
<a name="l00960"></a>00960 
<a name="l00961"></a>00961 <span class="keywordtype">void</span> CytoScanCreate()
<a name="l00962"></a>00962 { 
<a name="l00963"></a>00963         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on CytoScanCreate part 1.&quot;</span>);     
<a name="l00964"></a>00964         <span class="keywordtype">string</span> outdir = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir , <span class="stringliteral">&quot;CytoScanCreatePart1&quot;</span>);
<a name="l00965"></a>00965         <a class="code" href="classFs.html#a1206f6cf8f445461466f224e6cd1bc52" title="///">Fs::rmdirPath</a>(outdir, <span class="keyword">false</span>);
<a name="l00966"></a>00966         <a class="code" href="classFs.html#abf363360d88071c2f4a93a7e50d61c9f" title="Make one directory. ///.">Fs::mkdir</a>(outdir, <span class="keyword">false</span>);
<a name="l00967"></a>00967         vector&lt;RegressionCheck *&gt; checks;
<a name="l00968"></a>00968 
<a name="l00969"></a>00969         <span class="keywordtype">string</span> command = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l00970"></a>00970 <span class="stringliteral">                    -verbose 4 \</span>
<a name="l00971"></a>00971 <span class="stringliteral">                    --cyto2 false \</span>
<a name="l00972"></a>00972 <span class="stringliteral">                    --force true \</span>
<a name="l00973"></a>00973 <span class="stringliteral">                    --keep-intermediate-data false \</span>
<a name="l00974"></a>00974 <span class="stringliteral">                    --doDualNormalization true \</span>
<a name="l00975"></a>00975 <span class="stringliteral">                    --adapter-type-normalization false \</span>
<a name="l00976"></a>00976 <span class="stringliteral">                    --probe-file ../../../regression-data/data/lib/CytoScan/CytoScanHD_Array.probe_tab \</span>
<a name="l00977"></a>00977 <span class="stringliteral">                    --run-geno-qc true \</span>
<a name="l00978"></a>00978 <span class="stringliteral">                    --qca-file ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.r1.qca \</span>
<a name="l00979"></a>00979 <span class="stringliteral">                    --qcc-file ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.r1.qcc \</span>
<a name="l00980"></a>00980 <span class="stringliteral">                    --text-output true \</span>
<a name="l00981"></a>00981 <span class="stringliteral">                    --cnchp-output false \</span>
<a name="l00982"></a>00982 <span class="stringliteral">                    --cychp-output true \</span>
<a name="l00983"></a>00983 <span class="stringliteral">                    --reference-output &quot;</span> + m_testDir + <span class="stringliteral">&quot;/CytoScanCreatePart1/CNReferenceOutput.a5 \</span>
<a name="l00984"></a>00984 <span class="stringliteral">                    --cdf-file ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.cdf \</span>
<a name="l00985"></a>00985 <span class="stringliteral">                    --chrX-probes ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.chrXprobes\</span>
<a name="l00986"></a>00986 <span class="stringliteral">                    --chrY-probes ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.chrYprobes \</span>
<a name="l00987"></a>00987 <span class="stringliteral">                    --annotation-file ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.na31.annot.db \</span>
<a name="l00988"></a>00988 <span class="stringliteral">                    --male-gender-ratio-cutoff 1.5 \</span>
<a name="l00989"></a>00989 <span class="stringliteral">                    --female-gender-ratio-cutoff 0.9 \</span>
<a name="l00990"></a>00990 <span class="stringliteral">                    --xx-cutoff 0.61 \</span>
<a name="l00991"></a>00991 <span class="stringliteral">                    --xx-cutoff-high 0.95 \</span>
<a name="l00992"></a>00992 <span class="stringliteral">                    --y-cutoff 0.58 \</span>
<a name="l00993"></a>00993 <span class="stringliteral">                    --out-dir &quot;</span> + outdir + <span class="stringliteral">&quot; \</span>
<a name="l00994"></a>00994 <span class="stringliteral">                    --signal-adjustment-covariates covariate-signal-adjuster.order=fragment-adapter-type,fragment-length.bin-type=discrete,equally-populated.bin-count=0,100 \</span>
<a name="l00995"></a>00995 <span class="stringliteral">                    --lr-adjustment-covariates covariate-lr-adjuster.order=fragment-gc,probe-gc,median-signal.bin-type=equally-populated,discrete,equally-populated.bin-count=100,0,100.iqr-scaling=off,on,on.subtract-from-XY=on,off,on \</span>
<a name="l00996"></a>00996 <span class="stringliteral">                    --wave-correction-reference-method wave-correction-reference-method.trim=2.0.percentile=0.75.wave-count=6.demean=false \</span>
<a name="l00997"></a>00997 <span class="stringliteral">                    --local-gc-background-correction-reference-method none \</span>
<a name="l00998"></a>00998 <span class="stringliteral">                    --local-gc-background-intensity-adjustment-method none \</span>
<a name="l00999"></a>00999 <span class="stringliteral">                    --image-correction-intensity-adjustment-method none \</span>
<a name="l01000"></a>01000 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA12874_A05_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01001"></a>01001 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA18633_A02_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01002"></a>01002 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA19131_A04_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01003"></a>01003 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA12892_A03_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01004"></a>01004 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA18854_A06_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01005"></a>01005 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA18505_A01_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01006"></a>01006 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/3x_A4_Alpha_CytoScan_DF_20101512.CEL \</span>
<a name="l01007"></a>01007 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA16595_B6_Alpha_CytoScan_JS_20101512.CEL \</span>
<a name="l01008"></a>01008 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/Ref103_A1_Alpha_CytoScan_DF_20101512.CEL \</span>
<a name="l01009"></a>01009 <span class="stringliteral">         &quot;</span>;
<a name="l01010"></a>01010         <span class="comment">/*   This follows the pattern M M F F M F */</span>
<a name="l01011"></a>01011 
<a name="l01012"></a>01012         std::string strFileName1 = m_testDir + <span class="stringliteral">&quot;/CytoScanCreatePart1/CNReferenceOutput.a5&quot;</span>;
<a name="l01013"></a>01013         std::string strFileName2 = <span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/CytoScanCreate/CNReferenceOutput.a5&quot;</span>;
<a name="l01014"></a>01014         std::set&lt;std::string&gt; setLocalIgnore;
<a name="l01015"></a>01015 
<a name="l01016"></a>01016         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;AntigenomicProbes&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01017"></a>01017         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;CN5.plier.feature-response&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01018"></a>01018         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;CN5.snp-posteriors&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01019"></a>01019         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;target-sketch&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01020"></a>01020         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;Reference&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01021"></a>01021         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;WaveCorrection&quot;</span>, setLocalIgnore, 0.0001, 1.0, <span class="keyword">true</span>));
<a name="l01022"></a>01022         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;SketchSNP&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01023"></a>01023         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;SketchCN&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01024"></a>01024         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;Covariates&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01025"></a>01025 
<a name="l01026"></a>01026         <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test(<span class="stringliteral">&quot;CytoScanCreate part 1&quot;</span>, command.c_str(), checks);
<a name="l01027"></a>01027         test.setSuite(*<span class="keyword">this</span>,  outdir, outdir + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir + <span class="stringliteral">&quot;/valgrind.log&quot;</span>);
<a name="l01028"></a>01028         <span class="keywordtype">bool</span> bPassed = <span class="keyword">false</span>;
<a name="l01029"></a>01029         bPassed = test.pass();
<a name="l01030"></a>01030         <span class="keywordflow">if</span>(!bPassed) {
<a name="l01031"></a>01031             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: CytoScanCreate part 1. &quot;</span> + test.getErrorMsg());
<a name="l01032"></a>01032         }
<a name="l01033"></a>01033         <span class="keywordflow">else</span>
<a name="l01034"></a>01034         {
<a name="l01035"></a>01035             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: CytoScanCreate part 1. &quot;</span> + test.getErrorMsg());      
<a name="l01036"></a>01036         }
<a name="l01037"></a>01037 
<a name="l01038"></a>01038         {
<a name="l01039"></a>01039             m_numPassed += test.getNumPassed();
<a name="l01040"></a>01040             m_numFailed += test.getNumFailed();
<a name="l01041"></a>01041         }
<a name="l01042"></a>01042 
<a name="l01043"></a>01043         <span class="keywordflow">if</span> (bPassed)
<a name="l01044"></a>01044         {
<a name="l01045"></a>01045             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on CytoScanCrete part 2.&quot;</span>);
<a name="l01046"></a>01046             <span class="keywordtype">string</span> outdir2 = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;CytoScanCreatePart2&quot;</span>);
<a name="l01047"></a>01047             <a class="code" href="classFs.html#a1206f6cf8f445461466f224e6cd1bc52" title="///">Fs::rmdirPath</a>(outdir2, <span class="keyword">false</span>);
<a name="l01048"></a>01048             <a class="code" href="classFs.html#abf363360d88071c2f4a93a7e50d61c9f" title="Make one directory. ///.">Fs::mkdir</a>(outdir2, <span class="keyword">false</span>);
<a name="l01049"></a>01049             vector&lt;RegressionCheck *&gt; checks2;
<a name="l01050"></a>01050 
<a name="l01051"></a>01051             <span class="keywordtype">string</span> command2 = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l01052"></a>01052 <span class="stringliteral">                    -verbose 4 \</span>
<a name="l01053"></a>01053 <span class="stringliteral">                    --cyto2 false \</span>
<a name="l01054"></a>01054 <span class="stringliteral">                    --force true \</span>
<a name="l01055"></a>01055 <span class="stringliteral">                    --keep-intermediate-data false \</span>
<a name="l01056"></a>01056 <span class="stringliteral">                    --doDualNormalization true \</span>
<a name="l01057"></a>01057 <span class="stringliteral">                    --adapter-type-normalization false \</span>
<a name="l01058"></a>01058 <span class="stringliteral">                    --probe-file ../../../regression-data/data/lib/CytoScan/CytoScanHD_Array.probe_tab \</span>
<a name="l01059"></a>01059 <span class="stringliteral">                    --run-geno-qc true \</span>
<a name="l01060"></a>01060 <span class="stringliteral">                    --qca-file ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.r1.qca \</span>
<a name="l01061"></a>01061 <span class="stringliteral">                    --qcc-file ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.r1.qcc \</span>
<a name="l01062"></a>01062 <span class="stringliteral">                    --text-output true \</span>
<a name="l01063"></a>01063 <span class="stringliteral">                    --cnchp-output false \</span>
<a name="l01064"></a>01064 <span class="stringliteral">                    --cychp-output true \</span>
<a name="l01065"></a>01065 <span class="stringliteral">                    --reference-output &quot;</span> + m_testDir + <span class="stringliteral">&quot;/CytoScanCreatePart2/CNReferenceOutput.a5 \</span>
<a name="l01066"></a>01066 <span class="stringliteral">                    --cdf-file ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.cdf \</span>
<a name="l01067"></a>01067 <span class="stringliteral">                    --chrX-probes ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.chrXprobes \</span>
<a name="l01068"></a>01068 <span class="stringliteral">                    --chrY-probes ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.chrYprobes \</span>
<a name="l01069"></a>01069 <span class="stringliteral">                    --annotation-file ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.na31.annot.db \</span>
<a name="l01070"></a>01070 <span class="stringliteral">                    --male-gender-ratio-cutoff 1.5 \</span>
<a name="l01071"></a>01071 <span class="stringliteral">                    --female-gender-ratio-cutoff 0.9 \</span>
<a name="l01072"></a>01072 <span class="stringliteral">                    --xx-cutoff 0.61 \</span>
<a name="l01073"></a>01073 <span class="stringliteral">                    --xx-cutoff-high 0.95 \</span>
<a name="l01074"></a>01074 <span class="stringliteral">                    --y-cutoff 0.58 \</span>
<a name="l01075"></a>01075 <span class="stringliteral">                    --out-dir &quot;</span> + outdir2 + <span class="stringliteral">&quot; \</span>
<a name="l01076"></a>01076 <span class="stringliteral">                    --signal-adjustment-covariates covariate-signal-adjuster.order=fragment-adapter-type,fragment-length.bin-type=discrete,equally-populated.bin-count=0,100 \</span>
<a name="l01077"></a>01077 <span class="stringliteral">                    --lr-adjustment-covariates covariate-lr-adjuster.order=fragment-gc,probe-gc,median-signal.bin-type=equally-populated,discrete,equally-populated.bin-count=100,0,100.iqr-scaling=off,on,on.subtract-from-XY=on,off,on \</span>
<a name="l01078"></a>01078 <span class="stringliteral">                    --wave-correction-reference-method wave-correction-reference-method.trim=2.0.percentile=0.75.wave-count=6.demean=false \</span>
<a name="l01079"></a>01079 <span class="stringliteral">                    --local-gc-background-correction-reference-method none \</span>
<a name="l01080"></a>01080 <span class="stringliteral">                    --local-gc-background-intensity-adjustment-method none \</span>
<a name="l01081"></a>01081 <span class="stringliteral">                    --image-correction-intensity-adjustment-method none \</span>
<a name="l01082"></a>01082 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA12892_A03_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01083"></a>01083 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA12874_A05_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01084"></a>01084 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA18505_A01_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01085"></a>01085 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA18633_A02_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01086"></a>01086 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA19131_A04_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01087"></a>01087 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA18854_A06_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01088"></a>01088 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/Ref103_A1_Alpha_CytoScan_DF_20101512.CEL \</span>
<a name="l01089"></a>01089 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA16595_B6_Alpha_CytoScan_JS_20101512.CEL \</span>
<a name="l01090"></a>01090 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/3x_A4_Alpha_CytoScan_DF_20101512.CEL \</span>
<a name="l01091"></a>01091 <span class="stringliteral">         &quot;</span>;
<a name="l01092"></a>01092           <span class="comment">/* This follows the pattern F M F M F M */</span>
<a name="l01093"></a>01093 
<a name="l01094"></a>01094             <span class="keywordtype">bool</span> bPassed = <span class="keyword">false</span>;
<a name="l01095"></a>01095             std::string strFileName1 = m_testDir + <span class="stringliteral">&quot;/CytoScanCreatePart2/CNReferenceOutput.a5&quot;</span>;
<a name="l01096"></a>01096             std::string strFileName2 = <span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/CytoScanCreate/CNReferenceOutput.a5&quot;</span>;
<a name="l01097"></a>01097 
<a name="l01098"></a>01098             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;AntigenomicProbes&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01099"></a>01099             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;CN5.plier.feature-response&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01100"></a>01100             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;CN5.snp-posteriors&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01101"></a>01101             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;target-sketch&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01102"></a>01102             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;Reference&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01103"></a>01103             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;WaveCorrection&quot;</span>, setLocalIgnore, 0.0001, 1.0, <span class="keyword">true</span>));
<a name="l01104"></a>01104             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;SketchSNP&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01105"></a>01105             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;SketchCN&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01106"></a>01106             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;Covariates&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01107"></a>01107 
<a name="l01108"></a>01108 
<a name="l01109"></a>01109             strFileName1 = m_testDir + <span class="stringliteral">&quot;/CytoScanCreatePart1/CNReferenceOutput.a5&quot;</span>;
<a name="l01110"></a>01110             strFileName2 = m_testDir + <span class="stringliteral">&quot;/CytoScanCreatePart2/CNReferenceOutput.a5&quot;</span>;
<a name="l01111"></a>01111 
<a name="l01112"></a>01112             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;AntigenomicProbes&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01113"></a>01113             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;CN5.plier.feature-response&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01114"></a>01114             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;CN5.snp-posteriors&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01115"></a>01115             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;target-sketch&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01116"></a>01116             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;Reference&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01117"></a>01117             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;WaveCorrection&quot;</span>, setLocalIgnore, 0.0001, 1.0, <span class="keyword">true</span>));
<a name="l01118"></a>01118             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;SketchSNP&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01119"></a>01119             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;SketchCN&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01120"></a>01120             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;Covariates&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01121"></a>01121 
<a name="l01122"></a>01122 
<a name="l01123"></a>01123             <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test2(<span class="stringliteral">&quot;CytoScanCreate part 2&quot;</span>, command2.c_str(), checks2);
<a name="l01124"></a>01124             test2.setSuite(*<span class="keyword">this</span>,  outdir2, outdir2 + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir2 + <span class="stringliteral">&quot;/valgrind.log&quot;</span>);
<a name="l01125"></a>01125             bPassed = test2.pass();
<a name="l01126"></a>01126             <span class="keywordflow">if</span>(!bPassed) {
<a name="l01127"></a>01127                 <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: CytoScanCreate part 2. &quot;</span> + test2.getErrorMsg());
<a name="l01128"></a>01128             }
<a name="l01129"></a>01129             <span class="keywordflow">else</span>
<a name="l01130"></a>01130             {
<a name="l01131"></a>01131                 <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: CytoScanCreate part 2.. &quot;</span> + test.getErrorMsg());      
<a name="l01132"></a>01132             }
<a name="l01133"></a>01133 
<a name="l01134"></a>01134             {
<a name="l01135"></a>01135                 m_numPassed += test.getNumPassed();
<a name="l01136"></a>01136                 m_numFailed += test.getNumFailed();
<a name="l01137"></a>01137             }
<a name="l01138"></a>01138 
<a name="l01139"></a>01139         }
<a name="l01140"></a>01140 
<a name="l01141"></a>01141 }
<a name="l01142"></a>01142 
<a name="l01143"></a>01143 
<a name="l01144"></a>01144 
<a name="l01145"></a>01145 
<a name="l01146"></a>01146 <span class="keywordtype">void</span> twoSevenMCreate()
<a name="l01147"></a>01147 {
<a name="l01148"></a>01148         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on twoSevenMCreate.&quot;</span>);
<a name="l01149"></a>01149         <span class="keywordtype">string</span> outdir = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;twoSevenMCreate&quot;</span>);
<a name="l01150"></a>01150         <a class="code" href="classFs.html#a1206f6cf8f445461466f224e6cd1bc52" title="///">Fs::rmdirPath</a>(outdir, <span class="keyword">false</span>);
<a name="l01151"></a>01151         <a class="code" href="classFs.html#abf363360d88071c2f4a93a7e50d61c9f" title="Make one directory. ///.">Fs::mkdir</a>(outdir, <span class="keyword">false</span>);
<a name="l01152"></a>01152         vector&lt;RegressionCheck *&gt; checks;
<a name="l01153"></a>01153 
<a name="l01154"></a>01154         std::string strFileName1 = m_testDir + <span class="stringliteral">&quot;/twoSevenMCreate/CNReference.a5&quot;</span>;
<a name="l01155"></a>01155         std::string strFileName2 = <span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/twoSevenMCreate/CNReference.a5&quot;</span>;
<a name="l01156"></a>01156         std::set&lt;std::string&gt; setLocalIgnore;
<a name="l01157"></a>01157 
<a name="l01158"></a>01158         <span class="keywordtype">string</span> command = <span class="stringliteral">&quot;./apt-copynumber-cyto&quot;</span>
<a name="l01159"></a>01159                <span class="stringliteral">&quot; --out-dir &quot;</span> + m_testDir + <span class="stringliteral">&quot;/twoSevenMCreate&quot;</span>
<a name="l01160"></a>01160                <span class="stringliteral">&quot; --reference-output &quot;</span> + strFileName1 + 
<a name="l01161"></a>01161                <span class="stringliteral">&quot; --xml-file-append-only ../../../regression-data/data/lib/Cytogenetics_Array/NA30.2/twoSevenMCreate.xml &quot;</span>;
<a name="l01162"></a>01162 
<a name="l01163"></a>01163         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;AntigenomicProbes&quot;</span>, setLocalIgnore, 0.0, 1.0));
<a name="l01164"></a>01164         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;MedianSignals&quot;</span>, setLocalIgnore, 0.0, 1.0));
<a name="l01165"></a>01165         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;ProbeEffects&quot;</span>, setLocalIgnore, 0.0, 1.0));
<a name="l01166"></a>01166         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;SNPPosteriors&quot;</span>, setLocalIgnore, 0.0, 1.0));
<a name="l01167"></a>01167         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;SketchCN&quot;</span>, setLocalIgnore, 0.0, 1.0));
<a name="l01168"></a>01168         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;SketchSNP&quot;</span>, setLocalIgnore, 0.0, 1.0));
<a name="l01169"></a>01169         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;Cyto2&quot;</span>, <span class="stringliteral">&quot;WaveCorrection&quot;</span>, setLocalIgnore, 0.0, 1.0));
<a name="l01170"></a>01170         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;SketchCN&quot;</span>, setLocalIgnore, 0.0, 1.0));
<a name="l01171"></a>01171         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;Covariates&quot;</span>, setLocalIgnore, 0.0, 1.0));
<a name="l01172"></a>01172 
<a name="l01173"></a>01173 
<a name="l01174"></a>01174 
<a name="l01175"></a>01175         <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test1(<span class="stringliteral">&quot;twoSevenMCreate&quot;</span>, command.c_str(), checks, <span class="keyword">false</span>);
<a name="l01176"></a>01176         test1.setSuite(*<span class="keyword">this</span>,  outdir, outdir + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir + <span class="stringliteral">&quot;/valgrind.twoSevenMCreate.log&quot;</span>);
<a name="l01177"></a>01177 
<a name="l01178"></a>01178         <span class="keywordtype">bool</span> bPassed=test1.pass();
<a name="l01179"></a>01179         <span class="keywordflow">if</span>(!bPassed)
<a name="l01180"></a>01180         {
<a name="l01181"></a>01181             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: twoSevenMCreate&quot;</span> + test1.getErrorMsg());
<a name="l01182"></a>01182         }
<a name="l01183"></a>01183         <span class="keywordflow">else</span>
<a name="l01184"></a>01184         {
<a name="l01185"></a>01185             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: twoSevenMCreate &quot;</span> + test1.getErrorMsg());      
<a name="l01186"></a>01186         }
<a name="l01187"></a>01187 
<a name="l01188"></a>01188         {
<a name="l01189"></a>01189             m_numPassed += test1.getNumPassed();
<a name="l01190"></a>01190             m_numFailed += test1.getNumFailed();
<a name="l01191"></a>01191         }
<a name="l01192"></a>01192 }
<a name="l01193"></a>01193 
<a name="l01194"></a>01194 <span class="keywordtype">void</span> CytoScanBatch()
<a name="l01195"></a>01195 {
<a name="l01196"></a>01196         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on CytoScanBatch part 1.&quot;</span>);
<a name="l01197"></a>01197         <span class="keywordtype">string</span> outdir = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;CytoScanBatch&quot;</span>);
<a name="l01198"></a>01198         <a class="code" href="classFs.html#a1206f6cf8f445461466f224e6cd1bc52" title="///">Fs::rmdirPath</a>(outdir, <span class="keyword">false</span>);
<a name="l01199"></a>01199         <a class="code" href="classFs.html#abf363360d88071c2f4a93a7e50d61c9f" title="Make one directory. ///.">Fs::mkdir</a>(outdir, <span class="keyword">false</span>);
<a name="l01200"></a>01200         vector&lt;RegressionCheck *&gt; checks;
<a name="l01201"></a>01201 
<a name="l01202"></a>01202         <span class="keywordtype">string</span> command = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l01203"></a>01203 <span class="stringliteral">                    -verbose 4 \</span>
<a name="l01204"></a>01204 <span class="stringliteral">                    --cyto2 false \</span>
<a name="l01205"></a>01205 <span class="stringliteral">                    --force false \</span>
<a name="l01206"></a>01206 <span class="stringliteral">                    --keep-intermediate-data false \</span>
<a name="l01207"></a>01207 <span class="stringliteral">                    --doDualNormalization true \</span>
<a name="l01208"></a>01208 <span class="stringliteral">                    --adapter-type-normalization false \</span>
<a name="l01209"></a>01209 <span class="stringliteral">                    --probe-file ../../../regression-data/data/lib/CytoScan/CytoScanHD_Array.probe_tab \</span>
<a name="l01210"></a>01210 <span class="stringliteral">                    --run-geno-qc true \</span>
<a name="l01211"></a>01211 <span class="stringliteral">                    --qca-file ../../../regression-data/data/lib/CytoScan/CytoScanHD_Array.r1.qca \</span>
<a name="l01212"></a>01212 <span class="stringliteral">                    --qcc-file ../../../regression-data/data/lib/CytoScan/CytoScanHD_Array.r1.qcc \</span>
<a name="l01213"></a>01213 <span class="stringliteral">                    --text-output false \</span>
<a name="l01214"></a>01214 <span class="stringliteral">                    --cnchp-output false \</span>
<a name="l01215"></a>01215 <span class="stringliteral">                    --cychp-output true \</span>
<a name="l01216"></a>01216 <span class="stringliteral">                    --reference-output &quot;</span> + m_testDir + <span class="stringliteral">&quot;/CytoScanBatch/CNReferenceOutput.a5 \</span>
<a name="l01217"></a>01217 <span class="stringliteral">                    --cdf-file ../../../regression-data/data/lib/CytoScan/CytoScanHD_Array.cdf \</span>
<a name="l01218"></a>01218 <span class="stringliteral">                    --chrX-probes ../../../regression-data/data/lib/CytoScan/CytoScanHD_AllCNprobeshg19.chrXprobes \</span>
<a name="l01219"></a>01219 <span class="stringliteral">                    --chrY-probes ../../../regression-data/data/lib/CytoScan/CytoScanHD_AllCNprobeshg19.chrYprobes \</span>
<a name="l01220"></a>01220 <span class="stringliteral">                    --annotation-file ../../../regression-data/data/lib/CytoScan/CytoScanHD_Array.na31.1.annot.db \</span>
<a name="l01221"></a>01221 <span class="stringliteral">                    --male-gender-ratio-cutoff 1.5 \</span>
<a name="l01222"></a>01222 <span class="stringliteral">                    --female-gender-ratio-cutoff 0.9 \</span>
<a name="l01223"></a>01223 <span class="stringliteral">                    --xx-cutoff 0.61 \</span>
<a name="l01224"></a>01224 <span class="stringliteral">                    --xx-cutoff-high 0.95 \</span>
<a name="l01225"></a>01225 <span class="stringliteral">                    --y-cutoff 0.58 \</span>
<a name="l01226"></a>01226 <span class="stringliteral">                    --out-dir &quot;</span> + outdir + <span class="stringliteral">&quot; \</span>
<a name="l01227"></a>01227 <span class="stringliteral">                    --signal-adjustment-covariates covariate-signal-adjuster.order=fragment-adapter-type,fragment-length.bin-type=discrete,equally-populated.bin-count=0,100 \</span>
<a name="l01228"></a>01228 <span class="stringliteral">                    --lr-adjustment-covariates covariate-lr-adjuster.order=fragment-gc,probe-gc,median-signal.bin-type=equally-populated,discrete,equally-populated.bin-count=100,0,100.iqr-scaling=off,on,on.subtract-from-XY=on,off,on \</span>
<a name="l01229"></a>01229 <span class="stringliteral">                    --wave-correction-reference-method wave-correction-reference-method.trim=2.0.percentile=0.75.wave-count=6.demean=false \</span>
<a name="l01230"></a>01230 <span class="stringliteral">                    --local-gc-background-correction-reference-method none \</span>
<a name="l01231"></a>01231 <span class="stringliteral">                    --local-gc-background-intensity-adjustment-method none \</span>
<a name="l01232"></a>01232 <span class="stringliteral">                    --image-correction-intensity-adjustment-method none \</span>
<a name="l01233"></a>01233 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA12874_A05_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01234"></a>01234 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA12892_A03_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01235"></a>01235 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA18505_A01_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01236"></a>01236 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA18633_A02_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01237"></a>01237 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA18854_A06_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01238"></a>01238 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan/NA19131_A04_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01239"></a>01239 <span class="stringliteral">         &quot;</span>;
<a name="l01240"></a>01240 
<a name="l01241"></a>01241 
<a name="l01242"></a>01242         std::string strFileName1 = m_testDir + <span class="stringliteral">&quot;/CytoScanBatch/CNReferenceOutput.a5&quot;</span>;
<a name="l01243"></a>01243         std::string strFileName2 = <span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/CytoScanBatch/CNReferenceOutput.a5&quot;</span>;
<a name="l01244"></a>01244         std::set&lt;std::string&gt; setLocalIgnore;
<a name="l01245"></a>01245         <span class="keywordtype">double</span> epsilon = 1.0;
<a name="l01246"></a>01246 <span class="preprocessor">        #ifdef _WIN64</span>
<a name="l01247"></a>01247 <span class="preprocessor"></span>            epsilon = 0.0000000001; <span class="comment">// 1.0 ei-10 </span>
<a name="l01248"></a>01248 <span class="preprocessor">        #endif</span>
<a name="l01249"></a>01249 <span class="preprocessor"></span>        checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;AntigenomicProbes&quot;</span>, setLocalIgnore, epsilon, 1.0));
<a name="l01250"></a>01250         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;CN5.plier.feature-response&quot;</span>, setLocalIgnore, epsilon, 1.0));
<a name="l01251"></a>01251         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;CN5.snp-posteriors&quot;</span>, setLocalIgnore, epsilon, 1.0));
<a name="l01252"></a>01252         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;target-sketch&quot;</span>, setLocalIgnore, epsilon, 1.0));
<a name="l01253"></a>01253         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;Reference&quot;</span>, setLocalIgnore, epsilon, 1.0));
<a name="l01254"></a>01254         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;WaveCorrection&quot;</span>, setLocalIgnore, epsilon, 1.0));
<a name="l01255"></a>01255         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;SketchSNP&quot;</span>, setLocalIgnore, epsilon, 1.0));
<a name="l01256"></a>01256         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;SketchCN&quot;</span>, setLocalIgnore, epsilon, 1.0));
<a name="l01257"></a>01257         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;Covariates&quot;</span>, setLocalIgnore, epsilon, 1.0));
<a name="l01258"></a>01258 
<a name="l01259"></a>01259         <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test(<span class="stringliteral">&quot;CytoScanBatch part 1&quot;</span>, command.c_str(), checks);
<a name="l01260"></a>01260         test.setSuite(*<span class="keyword">this</span>,  outdir, outdir + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir + <span class="stringliteral">&quot;/valgrind.log&quot;</span>);
<a name="l01261"></a>01261         <span class="keywordtype">bool</span> bPassed = test.pass();
<a name="l01262"></a>01262 
<a name="l01263"></a>01263         <span class="keywordflow">if</span>(!bPassed) {
<a name="l01264"></a>01264             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: CytoScanBatch part 1: &quot;</span> + test.getErrorMsg());
<a name="l01265"></a>01265         }
<a name="l01266"></a>01266         <span class="keywordflow">else</span>
<a name="l01267"></a>01267         {
<a name="l01268"></a>01268             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: CytoScanBatch part 1 &quot;</span> + test.getErrorMsg());      
<a name="l01269"></a>01269         }
<a name="l01270"></a>01270 
<a name="l01271"></a>01271         {
<a name="l01272"></a>01272             m_numPassed += test.getNumPassed();
<a name="l01273"></a>01273             m_numFailed += test.getNumFailed();
<a name="l01274"></a>01274         }
<a name="l01275"></a>01275 
<a name="l01276"></a>01276 
<a name="l01277"></a>01277         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on CytoScanBatch part2.&quot;</span>);
<a name="l01278"></a>01278         std::string  goldDirectory=<span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/CytoScanBatch&quot;</span>;
<a name="l01279"></a>01279         std::string  newDirectory=<a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;CytoScanBatch/&quot;</span>);
<a name="l01280"></a>01280         std::string  infix=<span class="stringliteral">&quot;CytoScanHD&quot;</span>;
<a name="l01281"></a>01281 
<a name="l01282"></a>01282         std::set&lt;std::string&gt; setSetIgnore;
<a name="l01283"></a>01283 <span class="preprocessor">#ifdef _WIN64</span>
<a name="l01284"></a>01284 <span class="preprocessor"></span>            {
<a name="l01285"></a>01285                 std::string ignoreString = <span class="stringliteral">&quot;ProbeSets.AllelePeaks&quot;</span>;
<a name="l01286"></a>01286                 setSetIgnore.clear(); 
<a name="l01287"></a>01287                 setSetIgnore.insert(ignoreString);
<a name="l01288"></a>01288             }
<a name="l01289"></a>01289 <span class="preprocessor">#endif</span>
<a name="l01290"></a>01290 <span class="preprocessor"></span>
<a name="l01291"></a>01291         std::map&lt;std::string, float&gt; mapEpsilon;
<a name="l01292"></a>01292 
<a name="l01293"></a>01293         <span class="keywordtype">string</span> outdir2 = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;CytoScanBatch&quot;</span>);
<a name="l01294"></a>01294         vector&lt;RegressionCheck *&gt; checks2;
<a name="l01295"></a>01295 
<a name="l01296"></a>01296 
<a name="l01297"></a>01297 
<a name="l01298"></a>01298 
<a name="l01299"></a>01299         std::vector&lt;std::string&gt; vLocalCytoScan_SixFiles;
<a name="l01300"></a>01300         vLocalCytoScan_SixFiles.push_back(<span class="stringliteral">&quot;NA12874_A05_Baseline_CytoScanHD_VH_20101103&quot;</span>);
<a name="l01301"></a>01301         vLocalCytoScan_SixFiles.push_back(<span class="stringliteral">&quot;NA18505_A01_Baseline_CytoScanHD_VH_20101103&quot;</span>);
<a name="l01302"></a>01302         vLocalCytoScan_SixFiles.push_back(<span class="stringliteral">&quot;NA18633_A02_Baseline_CytoScanHD_VH_20101103&quot;</span>);
<a name="l01303"></a>01303         vLocalCytoScan_SixFiles.push_back(<span class="stringliteral">&quot;NA18854_A06_Baseline_CytoScanHD_VH_20101103&quot;</span>);
<a name="l01304"></a>01304         vLocalCytoScan_SixFiles.push_back(<span class="stringliteral">&quot;NA19131_A04_Baseline_CytoScanHD_VH_20101103&quot;</span>);
<a name="l01305"></a>01305 
<a name="l01306"></a>01306         <span class="comment">//  There is a failure of one segmenting difference in CN for this testcase under win64</span>
<a name="l01307"></a>01307         <span class="comment">//  So we comment it out for this OS and accept the difference for now.</span>
<a name="l01308"></a>01308 <span class="preprocessor">        #ifndef _WIN64</span>
<a name="l01309"></a>01309 <span class="preprocessor"></span>            vLocalCytoScan_SixFiles.push_back(<span class="stringliteral">&quot;NA12892_A03_Baseline_CytoScanHD_VH_20101103&quot;</span>);
<a name="l01310"></a>01310 <span class="preprocessor">        #endif</span>
<a name="l01311"></a>01311 <span class="preprocessor"></span>
<a name="l01312"></a>01312 
<a name="l01313"></a>01313         checks2.push_back(<span class="keyword">new</span> <a class="code" href="classCychpCheck.html" title="Class for testing that two cychp files are the same +/- some epsilon.">CychpCheck</a>(       vLocalCytoScan_SixFiles,
<a name="l01314"></a>01314                                                 goldDirectory,
<a name="l01315"></a>01315                                                 newDirectory,
<a name="l01316"></a>01316                                                 infix,
<a name="l01317"></a>01317                                                 setIgnore,
<a name="l01318"></a>01318                                                 setSetIgnore,
<a name="l01319"></a>01319                                                 mapEpsilon,
<a name="l01320"></a>01320                                                 0.0001,
<a name="l01321"></a>01321                                                 0.9999989,
<a name="l01322"></a>01322                                                 <span class="keyword">false</span>));
<a name="l01323"></a>01323 
<a name="l01324"></a>01324 
<a name="l01325"></a>01325 
<a name="l01326"></a>01326 
<a name="l01327"></a>01327         <span class="keywordtype">string</span> command2 = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l01328"></a>01328 <span class="stringliteral">                        --verbose 4 --cyto2 false \</span>
<a name="l01329"></a>01329 <span class="stringliteral">                        --doDualNormalization true \</span>
<a name="l01330"></a>01330 <span class="stringliteral">                        --keep-intermediate-data-local false \</span>
<a name="l01331"></a>01331 <span class="stringliteral">                        --run-geno-qc true \</span>
<a name="l01332"></a>01332 <span class="stringliteral">                        --qca-file ../../../regression-data/data/lib/CytoScan/CytoScanHD_Array.r1.qca \</span>
<a name="l01333"></a>01333 <span class="stringliteral">                        --qcc-file ../../../regression-data/data/lib/CytoScan/CytoScanHD_Array.r1.qcc \</span>
<a name="l01334"></a>01334 <span class="stringliteral">                        --snp-qc-use-contrast true \</span>
<a name="l01335"></a>01335 <span class="stringliteral">                        --snp-qc-snp-list ../../../regression-data/data/lib/CytoScan/CytoScanHD_Array.snplist.txt \</span>
<a name="l01336"></a>01336 <span class="stringliteral">                        --force false \</span>
<a name="l01337"></a>01337 <span class="stringliteral">                        --adapter-type-normalization false \</span>
<a name="l01338"></a>01338 <span class="stringliteral">                        --text-output true \</span>
<a name="l01339"></a>01339 <span class="stringliteral">                        --cnchp-output false \</span>
<a name="l01340"></a>01340 <span class="stringliteral">                        --cychp-output true \</span>
<a name="l01341"></a>01341 <span class="stringliteral">                        --set-analysis-name CytoScanHD \</span>
<a name="l01342"></a>01342 <span class="stringliteral">                        --cdf-file ../../../regression-data/data/lib/CytoScan/CytoScanHD_Array.cdf \</span>
<a name="l01343"></a>01343 <span class="stringliteral">                        --chrX-probes ../../../regression-data/data/lib/CytoScan/CytoScanHD_AllCNprobeshg19.chrXprobes \</span>
<a name="l01344"></a>01344 <span class="stringliteral">                        --chrY-probes ../../../regression-data/data/lib/CytoScan/CytoScanHD_AllCNprobeshg19.chrYprobes \</span>
<a name="l01345"></a>01345 <span class="stringliteral">                        --annotation-file ../../../regression-data/data/lib/CytoScan/CytoScanHD_Array.na31.1.annot.db \</span>
<a name="l01346"></a>01346 <span class="stringliteral">                        --reference-input &quot;</span> + m_testDir + <span class="stringliteral">&quot;/CytoScanBatch/CNReferenceOutput.a5 \</span>
<a name="l01347"></a>01347 <span class="stringliteral">                        --out-dir &quot;</span> + outdir2 + <span class="stringliteral">&quot; \</span>
<a name="l01348"></a>01348 <span class="stringliteral">                        --male-gender-ratio-cutoff 1.5 \</span>
<a name="l01349"></a>01349 <span class="stringliteral">                        --female-gender-ratio-cutoff 0.9 \</span>
<a name="l01350"></a>01350 <span class="stringliteral">                        --xx-cutoff 0.61 \</span>
<a name="l01351"></a>01351 <span class="stringliteral">                        --xx-cutoff-high 0.95 \</span>
<a name="l01352"></a>01352 <span class="stringliteral">                        --y-cutoff 0.58 \</span>
<a name="l01353"></a>01353 <span class="stringliteral">                        --log2ratio-adjustment-method log2ratio-adjustment-method-high-pass-filter \</span>
<a name="l01354"></a>01354 <span class="stringliteral">                        --local-gc-background-intensity-adjustment-method none \</span>
<a name="l01355"></a>01355 <span class="stringliteral">                        --image-correction-intensity-adjustment-method none \</span>
<a name="l01356"></a>01356 <span class="stringliteral">                        --wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method.bandwidth=101.bin-count=25.wave-count=6.wave-smooth=true \</span>
<a name="l01357"></a>01357 <span class="stringliteral">                        --analysis genotype \</span>
<a name="l01358"></a>01358 <span class="stringliteral">                        --analysis allelic-difference-CytoScan.outlier-trim=3.0.step=20.window=100.point-count=128.bandwidth=0.25.cutoff=0.05.clean-threshold=0.35.symmetry=true \</span>
<a name="l01359"></a>01359 <span class="stringliteral">                        --analysis gaussian-smooth.expSmoothSignal=1.smooth_sigma_multiplier=2.smooth_bw=50000 \</span>
<a name="l01360"></a>01360 <span class="stringliteral">                        --analysis log2-ratio.gc-correction=false.median-autosome-median-normalization=true.median-smooth-marker-count=5 \</span>
<a name="l01361"></a>01361 <span class="stringliteral">                        --analysis cn-cyto2.hmmCN_state=\&quot;\&#39;\&quot;0,1,2,3,4,5\&quot;\&#39;\&quot;.hmmCN_mu=\&quot;\&#39;\&quot;-2,-0.54,0,0.37,0.57,0.74\&quot;\&#39;\&quot;.hmmCN_sigma=\&quot;\&#39;\&quot;0.26,0.26,0.26,0.26,0.26,0.26\&quot;\&#39;\&quot;.diagonal-weight=0.999.mapd-weight=0.min-segment-size=5.hmm-confidence-weight=0.6 \</span>
<a name="l01362"></a>01362 <span class="stringliteral">                        --analysis cn-cyto2-gender.cutoff=0.5 \</span>
<a name="l01363"></a>01363 <span class="stringliteral">                        --analysis cn-segment \</span>
<a name="l01364"></a>01364 <span class="stringliteral">                        --analysis lohCytoScan.lohCS_errorrate=0.05.lohCS_beta=0.001.lohCS_alpha=0.01.lohCS_separation=1000000.lohCS_nMinMarkers=10.lohCS_NoCallThreshold=0.05.lohCS_minGenomicSpan=1000000 \</span>
<a name="l01365"></a>01365 <span class="stringliteral">                        --analysis loh-segment \</span>
<a name="l01366"></a>01366 <span class="stringliteral">                         ../../../regression-data/data/cel/CytoScan/NA12874_A05_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01367"></a>01367 <span class="stringliteral">                         ../../../regression-data/data/cel/CytoScan/NA12892_A03_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01368"></a>01368 <span class="stringliteral">                         ../../../regression-data/data/cel/CytoScan/NA18505_A01_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01369"></a>01369 <span class="stringliteral">                         ../../../regression-data/data/cel/CytoScan/NA18633_A02_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01370"></a>01370 <span class="stringliteral">                         ../../../regression-data/data/cel/CytoScan/NA18854_A06_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01371"></a>01371 <span class="stringliteral">                         ../../../regression-data/data/cel/CytoScan/NA19131_A04_Baseline_CytoScanHD_VH_20101103.CEL \</span>
<a name="l01372"></a>01372 <span class="stringliteral">             &quot;</span>;
<a name="l01373"></a>01373         <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test2(<span class="stringliteral">&quot;CytoScanBatch part 2&quot;</span>, command2.c_str(), checks2, <span class="keyword">false</span>);
<a name="l01374"></a>01374         test2.setSuite(*<span class="keyword">this</span>,  outdir2, outdir2 + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir2 + <span class="stringliteral">&quot;/valgrind.batch2.log&quot;</span>);
<a name="l01375"></a>01375         <span class="keywordflow">if</span>(!test2.pass()) {
<a name="l01376"></a>01376             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: CytoScanBatch part 2 &quot;</span> + test2.getErrorMsg());
<a name="l01377"></a>01377         }
<a name="l01378"></a>01378         <span class="keywordflow">else</span>
<a name="l01379"></a>01379         {
<a name="l01380"></a>01380             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: CytoScanBatch part 2 &quot;</span> + test.getErrorMsg());      
<a name="l01381"></a>01381         }
<a name="l01382"></a>01382 
<a name="l01383"></a>01383         {
<a name="l01384"></a>01384             m_numPassed += test2.getNumPassed();
<a name="l01385"></a>01385             m_numFailed += test2.getNumFailed();
<a name="l01386"></a>01386         }
<a name="l01387"></a>01387 
<a name="l01388"></a>01388 
<a name="l01389"></a>01389 }
<a name="l01390"></a>01390 
<a name="l01391"></a>01391 
<a name="l01392"></a>01392 
<a name="l01393"></a>01393 
<a name="l01394"></a>01394 <span class="keywordtype">void</span> twoSevenCancerNA31()
<a name="l01395"></a>01395 {
<a name="l01396"></a>01396         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on twoSevenCancerNA31.&quot;</span>);
<a name="l01397"></a>01397 
<a name="l01398"></a>01398         std::string  goldDirectory=<span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/twoSevenCancerNA31&quot;</span>;
<a name="l01399"></a>01399         std::string  newDirectory= <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir ,<span class="stringliteral">&quot;twoSevenCancerNA31&quot;</span>);
<a name="l01400"></a>01400         std::string  infix=<span class="stringliteral">&quot;cy2wg&quot;</span>;
<a name="l01401"></a>01401 
<a name="l01402"></a>01402         std::set&lt;std::string&gt; setSetIgnore;
<a name="l01403"></a>01403         setSetIgnore.clear();
<a name="l01404"></a>01404         std::map&lt;std::string, float&gt; mapEpsilon;
<a name="l01405"></a>01405 
<a name="l01406"></a>01406         <span class="keywordtype">string</span> outdir1 = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;twoSevenCancerNA31&quot;</span>);
<a name="l01407"></a>01407         <a class="code" href="classFs.html#a1206f6cf8f445461466f224e6cd1bc52" title="///">Fs::rmdirPath</a>(outdir1, <span class="keyword">false</span>);
<a name="l01408"></a>01408         <a class="code" href="classFs.html#abf363360d88071c2f4a93a7e50d61c9f" title="Make one directory. ///.">Fs::mkdir</a>(outdir1, <span class="keyword">false</span>);
<a name="l01409"></a>01409         vector&lt;RegressionCheck *&gt; checks1;
<a name="l01410"></a>01410 
<a name="l01411"></a>01411         checks1.push_back(<span class="keyword">new</span> <a class="code" href="classCychpCheck.html" title="Class for testing that two cychp files are the same +/- some epsilon.">CychpCheck</a>(       m_vTwoSevenCancer,
<a name="l01412"></a>01412                                                 goldDirectory,
<a name="l01413"></a>01413                                                 newDirectory,
<a name="l01414"></a>01414                                                 infix,
<a name="l01415"></a>01415                                                 setIgnore,
<a name="l01416"></a>01416                                                 setSetIgnore,
<a name="l01417"></a>01417                                                 mapEpsilon,
<a name="l01418"></a>01418                                                 0.0001,
<a name="l01419"></a>01419                                                 0.9999989,
<a name="l01420"></a>01420                                                 <span class="keyword">false</span>));
<a name="l01421"></a>01421 
<a name="l01422"></a>01422         <span class="keywordtype">string</span> command1 = <span class="stringliteral">&quot;./apt-copynumber-cyto &quot;</span>
<a name="l01423"></a>01423                 <span class="stringliteral">&quot; --out-dir &quot;</span> + m_testDir + <span class="stringliteral">&quot;/twoSevenCancerNA31&quot;</span>
<a name="l01424"></a>01424                 <span class="stringliteral">&quot; --xml-file-append-only ../../../regression-data/data/lib/Cytogenetics_Array/NA31/ChasInputNA31&quot;</span>;
<a name="l01425"></a>01425 
<a name="l01426"></a>01426 
<a name="l01427"></a>01427         <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test1(<span class="stringliteral">&quot;twoSevenCancerNA31&quot;</span>, command1.c_str(), checks1, <span class="keyword">false</span>);
<a name="l01428"></a>01428         test1.setSuite(*<span class="keyword">this</span>,  outdir1, outdir1 + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir1 + <span class="stringliteral">&quot;/valgrind.twoSevenCancerNA31.log&quot;</span>);
<a name="l01429"></a>01429 
<a name="l01430"></a>01430         <span class="keywordtype">bool</span> bPassed=test1.pass();
<a name="l01431"></a>01431         <span class="keywordflow">if</span>(!bPassed)
<a name="l01432"></a>01432         {
<a name="l01433"></a>01433             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: twoSevenCancerNA31 &quot;</span> + test1.getErrorMsg());
<a name="l01434"></a>01434         }
<a name="l01435"></a>01435         <span class="keywordflow">else</span>
<a name="l01436"></a>01436         {
<a name="l01437"></a>01437             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: twoSevenCancerNA31. &quot;</span> + test1.getErrorMsg());      
<a name="l01438"></a>01438         }
<a name="l01439"></a>01439 
<a name="l01440"></a>01440         {
<a name="l01441"></a>01441             m_numPassed += test1.getNumPassed();
<a name="l01442"></a>01442             m_numFailed += test1.getNumFailed();
<a name="l01443"></a>01443         }
<a name="l01444"></a>01444 
<a name="l01445"></a>01445 }
<a name="l01446"></a>01446 
<a name="l01447"></a>01447 
<a name="l01448"></a>01448 
<a name="l01449"></a>01449 
<a name="l01450"></a>01450 <span class="keywordtype">void</span> twoSevenCancer()
<a name="l01451"></a>01451 {
<a name="l01452"></a>01452         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on twoSevenCancer.&quot;</span>);
<a name="l01453"></a>01453         std::string  goldDirectory=<span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/twoSevenCancer&quot;</span>;
<a name="l01454"></a>01454         std::string  newDirectory= <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;twoSevenCancer&quot;</span>);
<a name="l01455"></a>01455         std::string  infix=<span class="stringliteral">&quot;cy2wg&quot;</span>;
<a name="l01456"></a>01456 
<a name="l01457"></a>01457         std::set&lt;std::string&gt; setSetIgnore;
<a name="l01458"></a>01458         setSetIgnore.clear();
<a name="l01459"></a>01459         std::map&lt;std::string, float&gt; mapEpsilon;
<a name="l01460"></a>01460 
<a name="l01461"></a>01461         <span class="keywordtype">string</span> outdir1 = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir , <span class="stringliteral">&quot;twoSevenCancer&quot;</span>);
<a name="l01462"></a>01462         <a class="code" href="classFs.html#a1206f6cf8f445461466f224e6cd1bc52" title="///">Fs::rmdirPath</a>(outdir1, <span class="keyword">false</span>);
<a name="l01463"></a>01463         <a class="code" href="classFs.html#abf363360d88071c2f4a93a7e50d61c9f" title="Make one directory. ///.">Fs::mkdir</a>(outdir1, <span class="keyword">false</span>);
<a name="l01464"></a>01464         vector&lt;RegressionCheck *&gt; checks1;
<a name="l01465"></a>01465 
<a name="l01466"></a>01466         checks1.push_back(<span class="keyword">new</span> <a class="code" href="classCychpCheck.html" title="Class for testing that two cychp files are the same +/- some epsilon.">CychpCheck</a>(       m_vTwoSevenCancer,
<a name="l01467"></a>01467                                                 goldDirectory,
<a name="l01468"></a>01468                                                 newDirectory,
<a name="l01469"></a>01469                                                 infix,
<a name="l01470"></a>01470                                                 setIgnore,
<a name="l01471"></a>01471                                                 setSetIgnore,
<a name="l01472"></a>01472                                                 mapEpsilon,
<a name="l01473"></a>01473                                                 0.0001,
<a name="l01474"></a>01474                                                 0.9999989,
<a name="l01475"></a>01475                                                 <span class="keyword">false</span>));
<a name="l01476"></a>01476 
<a name="l01477"></a>01477         <span class="keywordtype">string</span> command1 = <span class="stringliteral">&quot;./apt-copynumber-cyto &quot;</span>
<a name="l01478"></a>01478                      <span class="stringliteral">&quot; --out-dir &quot;</span> + m_testDir + <span class="stringliteral">&quot;/twoSevenCancer&quot;</span>
<a name="l01479"></a>01479                      <span class="stringliteral">&quot; --xml-file-append-only ../../../regression-data/data/lib/Cytogenetics_Array/NA30.2/ChasInputNA30.2&quot;</span>;
<a name="l01480"></a>01480 
<a name="l01481"></a>01481 
<a name="l01482"></a>01482         <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test1(<span class="stringliteral">&quot;twoSevenCancer&quot;</span>, command1.c_str(), checks1, <span class="keyword">false</span>);
<a name="l01483"></a>01483         test1.setSuite(*<span class="keyword">this</span>,  outdir1, outdir1 + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir1 + <span class="stringliteral">&quot;/valgrind.twoSevenCancer.log&quot;</span>);
<a name="l01484"></a>01484 
<a name="l01485"></a>01485         <span class="keywordtype">bool</span> bPassed=test1.pass();
<a name="l01486"></a>01486         <span class="keywordflow">if</span>(!bPassed)
<a name="l01487"></a>01487         {
<a name="l01488"></a>01488             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: twoSevenCancer &quot;</span> + test1.getErrorMsg());
<a name="l01489"></a>01489         }
<a name="l01490"></a>01490         <span class="keywordflow">else</span>
<a name="l01491"></a>01491         {
<a name="l01492"></a>01492             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: twoSevenCancer &quot;</span> + test1.getErrorMsg());            
<a name="l01493"></a>01493         }
<a name="l01494"></a>01494 
<a name="l01495"></a>01495         {
<a name="l01496"></a>01496             m_numPassed += test1.getNumPassed();
<a name="l01497"></a>01497             m_numFailed += test1.getNumFailed();
<a name="l01498"></a>01498         }
<a name="l01499"></a>01499 
<a name="l01500"></a>01500 }
<a name="l01501"></a>01501 
<a name="l01502"></a>01502 <span class="keywordtype">void</span> CytoScanUse()
<a name="l01503"></a>01503 {
<a name="l01504"></a>01504         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on CytoScanUse part 1.&quot;</span>);
<a name="l01505"></a>01505         std::string  goldDirectory=<span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/CytoScanUse&quot;</span>;
<a name="l01506"></a>01506         std::string  newDirectory=<a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;CytoScanUsePart1&quot;</span>);
<a name="l01507"></a>01507         std::string  infix=<span class="stringliteral">&quot;CytoScanHD&quot;</span>;
<a name="l01508"></a>01508 
<a name="l01509"></a>01509         std::set&lt;std::string&gt; setSetIgnore;
<a name="l01510"></a>01510 
<a name="l01511"></a>01511 <span class="preprocessor">#ifdef _WIN64</span>
<a name="l01512"></a>01512 <span class="preprocessor"></span>            {
<a name="l01513"></a>01513                 std::string ignoreString = <span class="stringliteral">&quot;ProbeSets.AllelePeaks&quot;</span>;
<a name="l01514"></a>01514                 setSetIgnore.clear(); 
<a name="l01515"></a>01515                 setSetIgnore.insert(ignoreString);
<a name="l01516"></a>01516             }
<a name="l01517"></a>01517 <span class="preprocessor">#endif</span>
<a name="l01518"></a>01518 <span class="preprocessor"></span>
<a name="l01519"></a>01519         std::map&lt;std::string, float&gt; mapEpsilon;
<a name="l01520"></a>01520 
<a name="l01521"></a>01521         <span class="keywordtype">string</span> outdir1 = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;CytoScanUsePart1&quot;</span>);
<a name="l01522"></a>01522         <a class="code" href="classFs.html#a1206f6cf8f445461466f224e6cd1bc52" title="///">Fs::rmdirPath</a>(outdir1, <span class="keyword">false</span>);
<a name="l01523"></a>01523         <a class="code" href="classFs.html#abf363360d88071c2f4a93a7e50d61c9f" title="Make one directory. ///.">Fs::mkdir</a>(outdir1, <span class="keyword">false</span>);
<a name="l01524"></a>01524         vector&lt;RegressionCheck *&gt; checks1;
<a name="l01525"></a>01525 
<a name="l01526"></a>01526         checks1.push_back(<span class="keyword">new</span> <a class="code" href="classCychpCheck.html" title="Class for testing that two cychp files are the same +/- some epsilon.">CychpCheck</a>(       m_vCytoScan_ThreeFiles,
<a name="l01527"></a>01527                                                 goldDirectory, 
<a name="l01528"></a>01528                                                 newDirectory,
<a name="l01529"></a>01529                                                 infix,
<a name="l01530"></a>01530                                                 setIgnore,
<a name="l01531"></a>01531                                                 setSetIgnore,
<a name="l01532"></a>01532                                                 mapEpsilon,
<a name="l01533"></a>01533                                                 0.0001,
<a name="l01534"></a>01534                                                 0.9999989,
<a name="l01535"></a>01535                                                 <span class="keyword">false</span>));       
<a name="l01536"></a>01536 
<a name="l01537"></a>01537         <span class="keywordtype">string</span> command1 = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l01538"></a>01538 <span class="stringliteral">            --text-output true \</span>
<a name="l01539"></a>01539 <span class="stringliteral">            --doDualNormalization true \</span>
<a name="l01540"></a>01540 <span class="stringliteral">            --keep-intermediate-data false \</span>
<a name="l01541"></a>01541 <span class="stringliteral">            --run-geno-qc true \</span>
<a name="l01542"></a>01542 <span class="stringliteral">            --qca-file ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.r1.qca \</span>
<a name="l01543"></a>01543 <span class="stringliteral">            --qcc-file ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.r1.qcc \</span>
<a name="l01544"></a>01544 <span class="stringliteral">            --snp-qc-use-contrast true \</span>
<a name="l01545"></a>01545 <span class="stringliteral">            --snp-qc-snp-list ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.snplist.txt \</span>
<a name="l01546"></a>01546 <span class="stringliteral">            --force true \</span>
<a name="l01547"></a>01547 <span class="stringliteral">            --adapter-type-normalization false \</span>
<a name="l01548"></a>01548 <span class="stringliteral">            --cnchp-output false \</span>
<a name="l01549"></a>01549 <span class="stringliteral">            --cychp-output true \</span>
<a name="l01550"></a>01550 <span class="stringliteral">            --set-analysis-name CytoScanHD \</span>
<a name="l01551"></a>01551 <span class="stringliteral">            --cdf-file ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.cdf \</span>
<a name="l01552"></a>01552 <span class="stringliteral">            --chrX-probes ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.chrXprobes \</span>
<a name="l01553"></a>01553 <span class="stringliteral">            --chrY-probes ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.chrYprobes \</span>
<a name="l01554"></a>01554 <span class="stringliteral">            --annotation-file ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.na31.annot.db \</span>
<a name="l01555"></a>01555 <span class="stringliteral">            --reference-input ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.na31.v1.REF_MODEL \</span>
<a name="l01556"></a>01556 <span class="stringliteral">            --out-dir &quot;</span> + outdir1 + <span class="stringliteral">&quot; \</span>
<a name="l01557"></a>01557 <span class="stringliteral">            --male-gender-ratio-cutoff 1.5 \</span>
<a name="l01558"></a>01558 <span class="stringliteral">            --female-gender-ratio-cutoff 0.9 \</span>
<a name="l01559"></a>01559 <span class="stringliteral">            --xx-cutoff 0.61 \</span>
<a name="l01560"></a>01560 <span class="stringliteral">            --xx-cutoff-high 0.95 \</span>
<a name="l01561"></a>01561 <span class="stringliteral">            --y-cutoff 0.58 \</span>
<a name="l01562"></a>01562 <span class="stringliteral">            --local-gc-background-intensity-adjustment-method none \</span>
<a name="l01563"></a>01563 <span class="stringliteral">            --image-correction-intensity-adjustment-method none \</span>
<a name="l01564"></a>01564 <span class="stringliteral">            --wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method.bandwidth=101.bin-count=25.wave-count=6.wave-smooth=true \</span>
<a name="l01565"></a>01565 <span class="stringliteral">            --analysis genotype \</span>
<a name="l01566"></a>01566 <span class="stringliteral">            --analysis log2-ratio.gc-correction=false.median-autosome-median-normalization=true.median-smooth-marker-count=5 \</span>
<a name="l01567"></a>01567 <span class="stringliteral">            --analysis allelic-difference-CytoScan.outlier-trim=3.0.step=20.window=100.point-count=128.bandwidth=0.25.cutoff=0.05.clean-threshold=0.35.symmetry=true \</span>
<a name="l01568"></a>01568 <span class="stringliteral">            --analysis kernel-smooth.sigma_span=50 \</span>
<a name="l01569"></a>01569 <span class="stringliteral">            --analysis cn-cyto2.hmmCN_state=\&quot;\&#39;\&quot;0,1,2,3,4\&quot;\&#39;\&quot;.hmmCN_mu=\&quot;\&#39;\&quot;-2,-0.45,0,0.3,0.51\&quot;\&#39;\&quot;.hmmCN_sigma=\&quot;\&#39;\&quot;0.35,0.35,0.25,0.25,0.25\&quot;\&#39;\&quot;.hmmCN_state-X=\&quot;\&#39;\&quot;0,1,2,3,4\&quot;\&#39;\&quot;.hmmCN_mu-X=\&quot;\&#39;\&quot;-2,-0.47,0,0.33,0.53\&quot;\&#39;\&quot;.hmmCN_sigma-X=\&quot;\&#39;\&quot;0.35,0.35,0.25,0.25,0.25\&quot;\&#39;\&quot;.hmmCN_state-Y=\&quot;\&#39;\&quot;0,1,2,3,4\&quot;\&#39;\&quot;.hmmCN_mu-Y=\&quot;\&#39;\&quot;-2,-0.45,0,0.3,0.51\&quot;\&#39;\&quot;.hmmCN_sigma-Y=\&quot;\&#39;\&quot;0.35,0.35,0.25,0.25,0.25\&quot;\&#39;\&quot;.diagonal-weight-Y=0.995.mapd-weight-Y=0.min-segment-size-Y=5.hmm-confidence-weight-Y=0.6.diagonal-weight=0.995.mapd-weight=0.min-segment-size=5.hmm-confidence-weight=0.6.diagonal-weight-X=0.995.mapd-weight-X=0.min-segment-size-X=5.hmm-confidence-weight-X=0.6 \</span>
<a name="l01570"></a>01570 <span class="stringliteral">             --analysis cn-cyto2-gender.cutoff=0.5 \</span>
<a name="l01571"></a>01571 <span class="stringliteral">             --analysis cn-segment \</span>
<a name="l01572"></a>01572 <span class="stringliteral">             --analysis lohCytoScan.lohCS_errorrate=0.05.lohCS_beta=0.001.lohCS_alpha=0.01.lohCS_separation=1000000.lohCS_nMinMarkers=10.lohCS_NoCallThreshold=0.05.lohCS_minGenomicSpan=1000000 \</span>
<a name="l01573"></a>01573 <span class="stringliteral">            --analysis loh-segment \</span>
<a name="l01574"></a>01574 <span class="stringliteral">            --analysis cn-neutral-loh \</span>
<a name="l01575"></a>01575 <span class="stringliteral">            ../../../regression-data/data/cel/CytoScan/Ref103_A1_Alpha_CytoScan_DF_20101512.CEL \</span>
<a name="l01576"></a>01576 <span class="stringliteral">            ../../../regression-data/data/cel/CytoScan/NA16595_B6_Alpha_CytoScan_JS_20101512.CEL \</span>
<a name="l01577"></a>01577 <span class="stringliteral">            ../../../regression-data/data/cel/CytoScan/3x_A4_Alpha_CytoScan_DF_20101512.CEL \</span>
<a name="l01578"></a>01578 <span class="stringliteral">        &quot;</span>;
<a name="l01579"></a>01579 
<a name="l01580"></a>01580         <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test1(<span class="stringliteral">&quot;CytoScanUse part 1&quot;</span>, command1.c_str(), checks1, <span class="keyword">false</span>);
<a name="l01581"></a>01581         test1.setSuite(*<span class="keyword">this</span>,  outdir1, outdir1 + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir1 + <span class="stringliteral">&quot;/valgrind.CytoScanUsePart1.log&quot;</span>);
<a name="l01582"></a>01582 
<a name="l01583"></a>01583         <span class="keywordtype">bool</span> bPassed=test1.pass();
<a name="l01584"></a>01584         <span class="keywordflow">if</span>(!bPassed) 
<a name="l01585"></a>01585         {
<a name="l01586"></a>01586             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: CytoScanUse part 1 &quot;</span> + test1.getErrorMsg());
<a name="l01587"></a>01587         }
<a name="l01588"></a>01588         <span class="keywordflow">else</span>
<a name="l01589"></a>01589         {
<a name="l01590"></a>01590             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: CytoScanUse part 1 &quot;</span> + test1.getErrorMsg());            
<a name="l01591"></a>01591         }
<a name="l01592"></a>01592 
<a name="l01593"></a>01593         {
<a name="l01594"></a>01594             m_numPassed += test1.getNumPassed();
<a name="l01595"></a>01595             m_numFailed += test1.getNumFailed();
<a name="l01596"></a>01596         }
<a name="l01597"></a>01597  
<a name="l01598"></a>01598 
<a name="l01599"></a>01599         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on CytoScanUse part 2.&quot;</span>);  
<a name="l01600"></a>01600         goldDirectory=<span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/CytoScanUse&quot;</span>;
<a name="l01601"></a>01601         newDirectory=<a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;CytoScanUsePart2&quot;</span>);
<a name="l01602"></a>01602         infix=<span class="stringliteral">&quot;CytoScanHD&quot;</span>;
<a name="l01603"></a>01603 
<a name="l01604"></a>01604         <span class="keywordtype">string</span> outdir2 = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;CytoScanUsePart2&quot;</span>);
<a name="l01605"></a>01605         <a class="code" href="classFs.html#a1206f6cf8f445461466f224e6cd1bc52" title="///">Fs::rmdirPath</a>(outdir2, <span class="keyword">false</span>);
<a name="l01606"></a>01606         <a class="code" href="classFs.html#abf363360d88071c2f4a93a7e50d61c9f" title="Make one directory. ///.">Fs::mkdir</a>(outdir2, <span class="keyword">false</span>);
<a name="l01607"></a>01607         vector&lt;RegressionCheck *&gt; checks2;
<a name="l01608"></a>01608 
<a name="l01609"></a>01609         checks2.push_back(<span class="keyword">new</span> <a class="code" href="classCychpCheck.html" title="Class for testing that two cychp files are the same +/- some epsilon.">CychpCheck</a>(       m_vCytoScan_ThreeFiles,
<a name="l01610"></a>01610                                                 goldDirectory,
<a name="l01611"></a>01611                                                 newDirectory,
<a name="l01612"></a>01612                                                 infix,
<a name="l01613"></a>01613                                                 setIgnore,
<a name="l01614"></a>01614                                                 setSetIgnore,
<a name="l01615"></a>01615                                                 mapEpsilon,
<a name="l01616"></a>01616                                                 0.0001,
<a name="l01617"></a>01617                                                 0.9999989,
<a name="l01618"></a>01618                                                 <span class="keyword">false</span>));
<a name="l01619"></a>01619 
<a name="l01620"></a>01620 
<a name="l01621"></a>01621 
<a name="l01622"></a>01622         <span class="keywordtype">string</span> command2 = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l01623"></a>01623 <span class="stringliteral">                --text-output true \</span>
<a name="l01624"></a>01624 <span class="stringliteral">                --doDualNormalization true \</span>
<a name="l01625"></a>01625 <span class="stringliteral">                --keep-intermediate-data false \</span>
<a name="l01626"></a>01626 <span class="stringliteral">                --run-geno-qc true \</span>
<a name="l01627"></a>01627 <span class="stringliteral">                --qca-file ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.r1.qca \</span>
<a name="l01628"></a>01628 <span class="stringliteral">                --qcc-file ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.r1.qcc \</span>
<a name="l01629"></a>01629 <span class="stringliteral">                --snp-qc-use-contrast true \</span>
<a name="l01630"></a>01630 <span class="stringliteral">                --snp-qc-snp-list ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.snplist.txt \</span>
<a name="l01631"></a>01631 <span class="stringliteral">                --force true \</span>
<a name="l01632"></a>01632 <span class="stringliteral">                --adapter-type-normalization false \</span>
<a name="l01633"></a>01633 <span class="stringliteral">                --cnchp-output false \</span>
<a name="l01634"></a>01634 <span class="stringliteral">                --cychp-output true \</span>
<a name="l01635"></a>01635 <span class="stringliteral">                --set-analysis-name CytoScanHD \</span>
<a name="l01636"></a>01636 <span class="stringliteral">                --cdf-file ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.cdf \</span>
<a name="l01637"></a>01637 <span class="stringliteral">                --chrX-probes ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.chrXprobes \</span>
<a name="l01638"></a>01638 <span class="stringliteral">                --chrY-probes ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.chrYprobes \</span>
<a name="l01639"></a>01639 <span class="stringliteral">                --annotation-file ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.na31.annot.db \</span>
<a name="l01640"></a>01640 <span class="stringliteral">                --reference-input ../../../regression-data/data/lib/CytoScan/Version.mar21.11/CytoScanHD_Array.na31.v1.REF_MODEL \</span>
<a name="l01641"></a>01641 <span class="stringliteral">                --out-dir &quot;</span> + outdir2 + <span class="stringliteral">&quot; \</span>
<a name="l01642"></a>01642 <span class="stringliteral">                --male-gender-ratio-cutoff 1.5 \</span>
<a name="l01643"></a>01643 <span class="stringliteral">                --female-gender-ratio-cutoff 0.9 \</span>
<a name="l01644"></a>01644 <span class="stringliteral">                --xx-cutoff 0.61 \</span>
<a name="l01645"></a>01645 <span class="stringliteral">                --xx-cutoff-high 0.95 \</span>
<a name="l01646"></a>01646 <span class="stringliteral">                --y-cutoff 0.58 \</span>
<a name="l01647"></a>01647 <span class="stringliteral">                --local-gc-background-intensity-adjustment-method none \</span>
<a name="l01648"></a>01648 <span class="stringliteral">                --image-correction-intensity-adjustment-method none \</span>
<a name="l01649"></a>01649 <span class="stringliteral">                --wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method.bandwidth=101.bin-count=25.wave-count=6.wave-smooth=true \</span>
<a name="l01650"></a>01650 <span class="stringliteral">                --analysis genotype \</span>
<a name="l01651"></a>01651 <span class="stringliteral">                --analysis log2-ratio.gc-correction=false.median-autosome-median-normalization=true.median-smooth-marker-count=5 \</span>
<a name="l01652"></a>01652 <span class="stringliteral">                --analysis allelic-difference-CytoScan.outlier-trim=3.0.step=20.window=100.point-count=128.bandwidth=0.25.cutoff=0.05.clean-threshold=0.35.symmetry=true \</span>
<a name="l01653"></a>01653 <span class="stringliteral">                --analysis kernel-smooth.sigma_span=50 \</span>
<a name="l01654"></a>01654 <span class="stringliteral">                --analysis cn-cyto2.hmmCN_state=\&quot;\&#39;\&quot;0,1,2,3,4\&quot;\&#39;\&quot;.hmmCN_mu=\&quot;\&#39;\&quot;-2,-0.45,0,0.3,0.51\&quot;\&#39;\&quot;.hmmCN_sigma=\&quot;\&#39;\&quot;0.35,0.35,0.25,0.25,0.25\&quot;\&#39;\&quot;.hmmCN_state-X=\&quot;\&#39;\&quot;0,1,2,3,4\&quot;\&#39;\&quot;.hmmCN_mu-X=\&quot;\&#39;\&quot;-2,-0.47,0,0.33,0.53\&quot;\&#39;\&quot;.hmmCN_sigma-X=\&quot;\&#39;\&quot;0.35,0.35,0.25,0.25,0.25\&quot;\&#39;\&quot;.hmmCN_state-Y=\&quot;\&#39;\&quot;0,1,2,3,4\&quot;\&#39;\&quot;.hmmCN_mu-Y=\&quot;\&#39;\&quot;-2,-0.45,0,0.3,0.51\&quot;\&#39;\&quot;.hmmCN_sigma-Y=\&quot;\&#39;\&quot;0.35,0.35,0.25,0.25,0.25\&quot;\&#39;\&quot;.diagonal-weight-Y=0.995.mapd-weight-Y=0.min-segment-size-Y=5.hmm-confidence-weight-Y=0.6.diagonal-weight=0.995.mapd-weight=0.min-segment-size=5.hmm-confidence-weight=0.6.diagonal-weight-X=0.995.mapd-weight-X=0.min-segment-size-X=5.hmm-confidence-weight-X=0.6 \</span>
<a name="l01655"></a>01655 <span class="stringliteral">                --analysis cn-cyto2-gender.cutoff=0.5 \</span>
<a name="l01656"></a>01656 <span class="stringliteral">                --analysis cn-segment \</span>
<a name="l01657"></a>01657 <span class="stringliteral">                --analysis lohCytoScan.lohCS_errorrate=0.05.lohCS_beta=0.001.lohCS_alpha=0.01.lohCS_separation=1000000.lohCS_nMinMarkers=10.lohCS_NoCallThreshold=0.05.lohCS_minGenomicSpan=1000000 \</span>
<a name="l01658"></a>01658 <span class="stringliteral">                --analysis loh-segment \</span>
<a name="l01659"></a>01659 <span class="stringliteral">                --analysis cn-neutral-loh \</span>
<a name="l01660"></a>01660 <span class="stringliteral">                ../../../regression-data/data/cel/CytoScan/3x_A4_Alpha_CytoScan_DF_20101512.CEL \</span>
<a name="l01661"></a>01661 <span class="stringliteral">                ../../../regression-data/data/cel/CytoScan/Ref103_A1_Alpha_CytoScan_DF_20101512.CEL \</span>
<a name="l01662"></a>01662 <span class="stringliteral">                ../../../regression-data/data/cel/CytoScan/NA16595_B6_Alpha_CytoScan_JS_20101512.CEL \</span>
<a name="l01663"></a>01663 <span class="stringliteral">            &quot;</span>;
<a name="l01664"></a>01664 
<a name="l01665"></a>01665         <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test2(<span class="stringliteral">&quot;CytoScanUse part 2&quot;</span>, command2.c_str(), checks2, <span class="keyword">false</span>);
<a name="l01666"></a>01666         test2.setSuite(*<span class="keyword">this</span>,  outdir2, outdir2 + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir2 + <span class="stringliteral">&quot;/valgrind.CytoScanUsePart2.log&quot;</span>);
<a name="l01667"></a>01667 
<a name="l01668"></a>01668         bPassed=test2.pass();
<a name="l01669"></a>01669         <span class="keywordflow">if</span>(!bPassed) 
<a name="l01670"></a>01670         {
<a name="l01671"></a>01671             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: CytoScanUse part 2 &quot;</span> + test2.getErrorMsg());
<a name="l01672"></a>01672         }
<a name="l01673"></a>01673         <span class="keywordflow">else</span>
<a name="l01674"></a>01674         {
<a name="l01675"></a>01675             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: CytoScanUse part 2 &quot;</span> + test2.getErrorMsg());            
<a name="l01676"></a>01676         }
<a name="l01677"></a>01677 
<a name="l01678"></a>01678         {
<a name="l01679"></a>01679             m_numPassed += test2.getNumPassed();
<a name="l01680"></a>01680             m_numFailed += test2.getNumFailed();
<a name="l01681"></a>01681         }
<a name="l01682"></a>01682 
<a name="l01683"></a>01683 }
<a name="l01684"></a>01684 
<a name="l01685"></a>01685 
<a name="l01686"></a>01686 <span class="keywordtype">void</span> batch_snp6_noAdapter()
<a name="l01687"></a>01687 {
<a name="l01688"></a>01688         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on batch_snp6_noAdapter part 1.&quot;</span>);
<a name="l01689"></a>01689         <span class="keywordtype">string</span> outdir = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;batch_snp6_noAdapter&quot;</span>);
<a name="l01690"></a>01690         <a class="code" href="classFs.html#a1206f6cf8f445461466f224e6cd1bc52" title="///">Fs::rmdirPath</a>(outdir, <span class="keyword">false</span>);
<a name="l01691"></a>01691         <a class="code" href="classFs.html#abf363360d88071c2f4a93a7e50d61c9f" title="Make one directory. ///.">Fs::mkdir</a>(outdir, <span class="keyword">false</span>);
<a name="l01692"></a>01692         vector&lt;RegressionCheck *&gt; checks;
<a name="l01693"></a>01693 
<a name="l01694"></a>01694         <span class="keywordtype">string</span> command = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l01695"></a>01695 <span class="stringliteral">        --verbose 1 \</span>
<a name="l01696"></a>01696 <span class="stringliteral">        --cyto2 false \</span>
<a name="l01697"></a>01697 <span class="stringliteral">        --doDualNormalization false \</span>
<a name="l01698"></a>01698 <span class="stringliteral">        --keep-intermediate-data false \</span>
<a name="l01699"></a>01699 <span class="stringliteral">        --force true \</span>
<a name="l01700"></a>01700 <span class="stringliteral">        --adapter-type-normalization false \</span>
<a name="l01701"></a>01701 <span class="stringliteral">        --analysis log2-ratio.gc-correction=false \</span>
<a name="l01702"></a>01702 <span class="stringliteral">        --analysis gaussian-smooth \</span>
<a name="l01703"></a>01703 <span class="stringliteral">        --analysis allelic-difference \</span>
<a name="l01704"></a>01704 <span class="stringliteral">        --analysis cn-state \</span>
<a name="l01705"></a>01705 <span class="stringliteral">        --analysis cn-segment \</span>
<a name="l01706"></a>01706 <span class="stringliteral">        --analysis lohCytoScan \</span>
<a name="l01707"></a>01707 <span class="stringliteral">        --analysis loh-segment  \</span>
<a name="l01708"></a>01708 <span class="stringliteral">        --analysis cn-neutral-loh \</span>
<a name="l01709"></a>01709 <span class="stringliteral">        --analysis normal-diploid \</span>
<a name="l01710"></a>01710 <span class="stringliteral">         --run-geno-qc true \</span>
<a name="l01711"></a>01711 <span class="stringliteral">         --cychp-output true \</span>
<a name="l01712"></a>01712 <span class="stringliteral">         --cnchp-output false \</span>
<a name="l01713"></a>01713 <span class="stringliteral">         --text-output true \</span>
<a name="l01714"></a>01714 <span class="stringliteral">         --reference-output &quot;</span> + m_testDir + <span class="stringliteral">&quot;/batch_snp6_noAdapter/CNReference.cyto.snp6.a5.ref \</span>
<a name="l01715"></a>01715 <span class="stringliteral">         --set-analysis-name CN5 \</span>
<a name="l01716"></a>01716 <span class="stringliteral">         --out-dir &quot;</span> + outdir + <span class="stringliteral">&quot; \</span>
<a name="l01717"></a>01717 <span class="stringliteral">         --cdf-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.cdf \</span>
<a name="l01718"></a>01718 <span class="stringliteral">         --chrX-probes ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.chrXprobes \</span>
<a name="l01719"></a>01719 <span class="stringliteral">         --chrY-probes ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.chrYprobes \</span>
<a name="l01720"></a>01720 <span class="stringliteral">         --annotation-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.na30.2.annot.db \</span>
<a name="l01721"></a>01721 <span class="stringliteral">         --qca-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.qca \</span>
<a name="l01722"></a>01722 <span class="stringliteral">         --qcc-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.qcc \</span>
<a name="l01723"></a>01723 <span class="stringliteral">         --male-gender-ratio-cutoff 0.71 \</span>
<a name="l01724"></a>01724 <span class="stringliteral">         --female-gender-ratio-cutoff 0.48 \</span>
<a name="l01725"></a>01725 <span class="stringliteral">         --xx-cutoff 0.8 \</span>
<a name="l01726"></a>01726 <span class="stringliteral">         --xx-cutoff-high 1.07 \</span>
<a name="l01727"></a>01727 <span class="stringliteral">         --y-cutoff 0.65 \</span>
<a name="l01728"></a>01728 <span class="stringliteral">         --local-gc-background-correction-reference-method none \</span>
<a name="l01729"></a>01729 <span class="stringliteral">         --local-gc-background-intensity-adjustment-method none \</span>
<a name="l01730"></a>01730 <span class="stringliteral">         --image-correction-intensity-adjustment-method none \</span>
<a name="l01731"></a>01731 <span class="stringliteral">         --wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method.bandwidth=101.bin-count=25.wave-count=7 \</span>
<a name="l01732"></a>01732 <span class="stringliteral">         --wave-correction-reference-method wave-correction-reference-method.wave-count=7.trim=2.0.percentile=0.75.demean=false \</span>
<a name="l01733"></a>01733 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA12004_GW6_C.CEL \</span>
<a name="l01734"></a>01734 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA12003_GW6_C.CEL \</span>
<a name="l01735"></a>01735 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA11993_GW6_C.CEL \</span>
<a name="l01736"></a>01736 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA11882_GW6_C.CEL \</span>
<a name="l01737"></a>01737 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA11881_GW6_C.CEL \</span>
<a name="l01738"></a>01738 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA10863_GW6_C.CEL \</span>
<a name="l01739"></a>01739 <span class="stringliteral">         &quot;</span>;
<a name="l01740"></a>01740 
<a name="l01741"></a>01741         std::string strFileName1 = m_testDir + <span class="stringliteral">&quot;/batch_snp6_noAdapter/CNReference.cyto.snp6.a5.ref&quot;</span>;
<a name="l01742"></a>01742         std::string strFileName2 = <span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/batch_snp6_noAdapter/CNReference.cyto.snp6.a5.ref&quot;</span>;
<a name="l01743"></a>01743         std::set&lt;std::string&gt; setLocalIgnore;
<a name="l01744"></a>01744 
<a name="l01745"></a>01745         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;AntigenomicProbes&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01746"></a>01746         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;CN5.plier.feature-response&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01747"></a>01747         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;CN5.snp-posteriors&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01748"></a>01748         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;target-sketch&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01749"></a>01749         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;Reference&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01750"></a>01750         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;WaveCorrection&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01751"></a>01751 
<a name="l01752"></a>01752         <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test(<span class="stringliteral">&quot;batch_snp6_noAdapter part 1&quot;</span>, command.c_str(), checks);
<a name="l01753"></a>01753         test.setSuite(*<span class="keyword">this</span>,  outdir, outdir + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir + <span class="stringliteral">&quot;/valgrind.log&quot;</span>);
<a name="l01754"></a>01754 
<a name="l01755"></a>01755         <span class="keywordtype">bool</span> bPassed = <span class="keyword">false</span>;
<a name="l01756"></a>01756         bPassed = test.pass();
<a name="l01757"></a>01757         <span class="keywordflow">if</span>(!bPassed) {
<a name="l01758"></a>01758             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in:  batch_snp6_noAdapter part 1&quot;</span> + test.getErrorMsg());
<a name="l01759"></a>01759         }
<a name="l01760"></a>01760         <span class="keywordflow">else</span>
<a name="l01761"></a>01761         {
<a name="l01762"></a>01762             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: batch_snp6_noAdapter part 1 &quot;</span> + test.getErrorMsg());            
<a name="l01763"></a>01763         }
<a name="l01764"></a>01764 
<a name="l01765"></a>01765         {
<a name="l01766"></a>01766             m_numPassed += test.getNumPassed();
<a name="l01767"></a>01767             m_numFailed += test.getNumFailed();
<a name="l01768"></a>01768 
<a name="l01769"></a>01769         }
<a name="l01770"></a>01770         <span class="keywordflow">if</span> (bPassed)
<a name="l01771"></a>01771         {
<a name="l01772"></a>01772             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on batch_snp6_noAdapter part 2.&quot;</span>);
<a name="l01773"></a>01773             <span class="keywordtype">string</span> outdir2 = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir , <span class="stringliteral">&quot;/batch_snp6_noAdapter&quot;</span>);
<a name="l01774"></a>01774 
<a name="l01775"></a>01775             std::set&lt;std::string&gt; setSetIgnore;
<a name="l01776"></a>01776 
<a name="l01777"></a>01777             std::string  goldDirectory=<span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/batch_snp6_noAdapter&quot;</span>;
<a name="l01778"></a>01778             std::string  newDirectory= <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;batch_snp6_noAdapter&quot;</span>);
<a name="l01779"></a>01779             <span class="keywordtype">string</span> infix = <span class="stringliteral">&quot;CN5&quot;</span>;
<a name="l01780"></a>01780 
<a name="l01781"></a>01781             std::map&lt;std::string, float&gt; mapEpsilon;
<a name="l01782"></a>01782 
<a name="l01783"></a>01783             vector&lt;RegressionCheck *&gt; checks2;
<a name="l01784"></a>01784 
<a name="l01785"></a>01785             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classCychpCheck.html" title="Class for testing that two cychp files are the same +/- some epsilon.">CychpCheck</a>(
<a name="l01786"></a>01786                                                 m_vCytogenetics_SNP6_SixFiles,
<a name="l01787"></a>01787                                                 goldDirectory,
<a name="l01788"></a>01788                                                 newDirectory,
<a name="l01789"></a>01789                                                 infix,
<a name="l01790"></a>01790                                                 setIgnore,
<a name="l01791"></a>01791                                                 setSetIgnore,
<a name="l01792"></a>01792                                                 mapEpsilon,
<a name="l01793"></a>01793                                                 0.0001,
<a name="l01794"></a>01794                                                 0.9999989,
<a name="l01795"></a>01795                                                 <span class="keyword">false</span>));
<a name="l01796"></a>01796 
<a name="l01797"></a>01797 
<a name="l01798"></a>01798             <span class="keywordtype">string</span> command2 = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l01799"></a>01799 <span class="stringliteral">            -verbose 1 \</span>
<a name="l01800"></a>01800 <span class="stringliteral">            --cyto2 false \</span>
<a name="l01801"></a>01801 <span class="stringliteral">            --force true \</span>
<a name="l01802"></a>01802 <span class="stringliteral">            --adapter-type-normalization false \</span>
<a name="l01803"></a>01803 <span class="stringliteral">            --doDualNormalization false \</span>
<a name="l01804"></a>01804 <span class="stringliteral">            --keep-intermediate-data false \</span>
<a name="l01805"></a>01805 <span class="stringliteral">            --analysis genotype \</span>
<a name="l01806"></a>01806 <span class="stringliteral">            --analysis log2-ratio.gc-correction=false \</span>
<a name="l01807"></a>01807 <span class="stringliteral">            --analysis gaussian-smooth \</span>
<a name="l01808"></a>01808 <span class="stringliteral">            --analysis allelic-difference \</span>
<a name="l01809"></a>01809 <span class="stringliteral">            --analysis cn-state \</span>
<a name="l01810"></a>01810 <span class="stringliteral">            --analysis cn-segment \</span>
<a name="l01811"></a>01811 <span class="stringliteral">            --analysis lohCytoScan \</span>
<a name="l01812"></a>01812 <span class="stringliteral">            --analysis loh-segment  \</span>
<a name="l01813"></a>01813 <span class="stringliteral">            --analysis cn-neutral-loh \</span>
<a name="l01814"></a>01814 <span class="stringliteral">            --analysis normal-diploid \</span>
<a name="l01815"></a>01815 <span class="stringliteral">            --run-geno-qc true \</span>
<a name="l01816"></a>01816 <span class="stringliteral">            --cychp-output true \</span>
<a name="l01817"></a>01817 <span class="stringliteral">            --cnchp-output false \</span>
<a name="l01818"></a>01818 <span class="stringliteral">            --text-output true \</span>
<a name="l01819"></a>01819 <span class="stringliteral">            --reference-input &quot;</span> + m_testDir + <span class="stringliteral">&quot;/batch_snp6_noAdapter/CNReference.cyto.snp6.a5.ref \</span>
<a name="l01820"></a>01820 <span class="stringliteral">            --set-analysis-name CN5 \</span>
<a name="l01821"></a>01821 <span class="stringliteral">            --out-dir &quot;</span> + outdir2 + <span class="stringliteral">&quot; \</span>
<a name="l01822"></a>01822 <span class="stringliteral">            --cdf-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.cdf \</span>
<a name="l01823"></a>01823 <span class="stringliteral">            --chrX-probes ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.chrXprobes \</span>
<a name="l01824"></a>01824 <span class="stringliteral">            --chrY-probes ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.chrYprobes \</span>
<a name="l01825"></a>01825 <span class="stringliteral">            --annotation-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.na30.2.annot.db \</span>
<a name="l01826"></a>01826 <span class="stringliteral">            --qca-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.qca \</span>
<a name="l01827"></a>01827 <span class="stringliteral">            --qcc-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.qcc \</span>
<a name="l01828"></a>01828 <span class="stringliteral">            --male-gender-ratio-cutoff 0.71 \</span>
<a name="l01829"></a>01829 <span class="stringliteral">            --female-gender-ratio-cutoff 0.48 \</span>
<a name="l01830"></a>01830 <span class="stringliteral">            --xx-cutoff 0.8 \</span>
<a name="l01831"></a>01831 <span class="stringliteral">            --xx-cutoff-high 1.07 \</span>
<a name="l01832"></a>01832 <span class="stringliteral">            --y-cutoff 0.65 \</span>
<a name="l01833"></a>01833 <span class="stringliteral">            --local-gc-background-correction-reference-method none \</span>
<a name="l01834"></a>01834 <span class="stringliteral">            --local-gc-background-intensity-adjustment-method none \</span>
<a name="l01835"></a>01835 <span class="stringliteral">            --image-correction-intensity-adjustment-method none \</span>
<a name="l01836"></a>01836 <span class="stringliteral">            --wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method.bandwidth=101.bin-count=25.wave-count=7 \</span>
<a name="l01837"></a>01837 <span class="stringliteral">            --wave-correction-reference-method wave-correction-reference-method.wave-count=7.trim=2.0.percentile=0.75.demean=false \</span>
<a name="l01838"></a>01838 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA12004_GW6_C.CEL \</span>
<a name="l01839"></a>01839 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA12003_GW6_C.CEL \</span>
<a name="l01840"></a>01840 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA11993_GW6_C.CEL \</span>
<a name="l01841"></a>01841 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA11882_GW6_C.CEL \</span>
<a name="l01842"></a>01842 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA11881_GW6_C.CEL \</span>
<a name="l01843"></a>01843 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA10863_GW6_C.CEL \</span>
<a name="l01844"></a>01844 <span class="stringliteral">            &quot;</span>;
<a name="l01845"></a>01845  
<a name="l01846"></a>01846             <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test2(<span class="stringliteral">&quot;batch_snp6_noAdapter part 2&quot;</span>, command2.c_str(), checks2, <span class="keyword">false</span>);
<a name="l01847"></a>01847             test2.setSuite(*<span class="keyword">this</span>,  outdir2, outdir2 + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir2 + <span class="stringliteral">&quot;/valgrind.batch2.log&quot;</span>);
<a name="l01848"></a>01848             <span class="keywordflow">if</span>(!test2.pass()) {
<a name="l01849"></a>01849                 <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: batch_snp6_noAdapter part 2&quot;</span> + test2.getErrorMsg());
<a name="l01850"></a>01850             } 
<a name="l01851"></a>01851             <span class="keywordflow">else</span>
<a name="l01852"></a>01852             {
<a name="l01853"></a>01853                 <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: batch_snp6_noAdapter part 2 &quot;</span> + test.getErrorMsg());            
<a name="l01854"></a>01854             }
<a name="l01855"></a>01855 
<a name="l01856"></a>01856             {
<a name="l01857"></a>01857                 m_numPassed += test2.getNumPassed();
<a name="l01858"></a>01858                 m_numFailed += test2.getNumFailed();
<a name="l01859"></a>01859             }
<a name="l01860"></a>01860 
<a name="l01861"></a>01861         }
<a name="l01862"></a>01862 }
<a name="l01863"></a>01863 
<a name="l01864"></a>01864 
<a name="l01865"></a>01865 
<a name="l01866"></a>01866 
<a name="l01867"></a>01867 <span class="keywordtype">void</span> batch_snp6_dual_noAdapter()
<a name="l01868"></a>01868 {
<a name="l01869"></a>01869         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on batch_snp6_dual_noAdapter part 1.&quot;</span>);
<a name="l01870"></a>01870         <span class="keywordtype">string</span> outdir = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;batch_snp6_dual_noAdapter&quot;</span>);
<a name="l01871"></a>01871         <a class="code" href="classFs.html#a1206f6cf8f445461466f224e6cd1bc52" title="///">Fs::rmdirPath</a>(outdir, <span class="keyword">false</span>);
<a name="l01872"></a>01872         <a class="code" href="classFs.html#abf363360d88071c2f4a93a7e50d61c9f" title="Make one directory. ///.">Fs::mkdir</a>(outdir, <span class="keyword">false</span>);
<a name="l01873"></a>01873         vector&lt;RegressionCheck *&gt; checks;
<a name="l01874"></a>01874 
<a name="l01875"></a>01875         <span class="keywordtype">string</span> command = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l01876"></a>01876 <span class="stringliteral">        --verbose 1 \</span>
<a name="l01877"></a>01877 <span class="stringliteral">        --cyto2 false \</span>
<a name="l01878"></a>01878 <span class="stringliteral">        --doDualNormalization true \</span>
<a name="l01879"></a>01879 <span class="stringliteral">        --keep-intermediate-data false \</span>
<a name="l01880"></a>01880 <span class="stringliteral">        --force true \</span>
<a name="l01881"></a>01881 <span class="stringliteral">        --adapter-type-normalization false \</span>
<a name="l01882"></a>01882 <span class="stringliteral">        --analysis log2-ratio.gc-correction=false \</span>
<a name="l01883"></a>01883 <span class="stringliteral">        --analysis gaussian-smooth \</span>
<a name="l01884"></a>01884 <span class="stringliteral">        --analysis allelic-difference \</span>
<a name="l01885"></a>01885 <span class="stringliteral">        --analysis cn-state \</span>
<a name="l01886"></a>01886 <span class="stringliteral">        --analysis cn-segment \</span>
<a name="l01887"></a>01887 <span class="stringliteral">        --analysis lohCytoScan \</span>
<a name="l01888"></a>01888 <span class="stringliteral">        --analysis loh-segment  \</span>
<a name="l01889"></a>01889 <span class="stringliteral">        --analysis cn-neutral-loh \</span>
<a name="l01890"></a>01890 <span class="stringliteral">        --analysis normal-diploid \</span>
<a name="l01891"></a>01891 <span class="stringliteral">         --run-geno-qc true \</span>
<a name="l01892"></a>01892 <span class="stringliteral">         --cychp-output true \</span>
<a name="l01893"></a>01893 <span class="stringliteral">         --cnchp-output false \</span>
<a name="l01894"></a>01894 <span class="stringliteral">         --text-output true \</span>
<a name="l01895"></a>01895 <span class="stringliteral">         --reference-output &quot;</span> + m_testDir + <span class="stringliteral">&quot;/batch_snp6_dual_noAdapter/CNReference.cyto.snp6.a5.ref \</span>
<a name="l01896"></a>01896 <span class="stringliteral">         --set-analysis-name CN5 \</span>
<a name="l01897"></a>01897 <span class="stringliteral">         --out-dir &quot;</span> + outdir + <span class="stringliteral">&quot; \</span>
<a name="l01898"></a>01898 <span class="stringliteral">         --cdf-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.cdf \</span>
<a name="l01899"></a>01899 <span class="stringliteral">         --chrX-probes ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.chrXprobes \</span>
<a name="l01900"></a>01900 <span class="stringliteral">         --chrY-probes ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.chrYprobes \</span>
<a name="l01901"></a>01901 <span class="stringliteral">         --annotation-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.na30.2.annot.db \</span>
<a name="l01902"></a>01902 <span class="stringliteral">         --qca-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.qca \</span>
<a name="l01903"></a>01903 <span class="stringliteral">         --qcc-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.qcc \</span>
<a name="l01904"></a>01904 <span class="stringliteral">         --male-gender-ratio-cutoff 0.71 \</span>
<a name="l01905"></a>01905 <span class="stringliteral">         --female-gender-ratio-cutoff 0.48 \</span>
<a name="l01906"></a>01906 <span class="stringliteral">         --xx-cutoff 0.8 \</span>
<a name="l01907"></a>01907 <span class="stringliteral">         --xx-cutoff-high 1.07 \</span>
<a name="l01908"></a>01908 <span class="stringliteral">         --y-cutoff 0.55 \</span>
<a name="l01909"></a>01909 <span class="stringliteral">         --local-gc-background-correction-reference-method none \</span>
<a name="l01910"></a>01910 <span class="stringliteral">         --local-gc-background-intensity-adjustment-method none \</span>
<a name="l01911"></a>01911 <span class="stringliteral">         --image-correction-intensity-adjustment-method none \</span>
<a name="l01912"></a>01912 <span class="stringliteral">         --wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method.bandwidth=101.bin-count=25.wave-count=7 \</span>
<a name="l01913"></a>01913 <span class="stringliteral">         --wave-correction-reference-method wave-correction-reference-method.wave-count=7.trim=2.0.percentile=0.75.demean=false \</span>
<a name="l01914"></a>01914 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA12004_GW6_C.CEL \</span>
<a name="l01915"></a>01915 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA12003_GW6_C.CEL \</span>
<a name="l01916"></a>01916 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA11993_GW6_C.CEL \</span>
<a name="l01917"></a>01917 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA11882_GW6_C.CEL \</span>
<a name="l01918"></a>01918 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA11881_GW6_C.CEL \</span>
<a name="l01919"></a>01919 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA10863_GW6_C.CEL \</span>
<a name="l01920"></a>01920 <span class="stringliteral">         &quot;</span>;
<a name="l01921"></a>01921 
<a name="l01922"></a>01922         std::string strFileName1 = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;batch_snp6_dual_noAdapter/CNReference.cyto.snp6.a5.ref&quot;</span>);
<a name="l01923"></a>01923         std::string strFileName2 = <span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/batch_snp6_dual_noAdapter/CNReference.cyto.snp6.a5.ref&quot;</span>;
<a name="l01924"></a>01924         std::set&lt;std::string&gt; setLocalIgnore;
<a name="l01925"></a>01925 
<a name="l01926"></a>01926         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;AntigenomicProbes&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01927"></a>01927         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;CN5.plier.feature-response&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01928"></a>01928         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;CN5.snp-posteriors&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01929"></a>01929         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;target-sketch&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01930"></a>01930         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;Reference&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01931"></a>01931         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;WaveCorrection&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01932"></a>01932         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;SketchSNP&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01933"></a>01933         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;SketchCN&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l01934"></a>01934 
<a name="l01935"></a>01935         <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test(<span class="stringliteral">&quot;batch_snp6_dual_noAdapter part 1&quot;</span>, command.c_str(), checks);
<a name="l01936"></a>01936         test.setSuite(*<span class="keyword">this</span>,  outdir, outdir + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir + <span class="stringliteral">&quot;/valgrind.log&quot;</span>);
<a name="l01937"></a>01937         <span class="keywordtype">bool</span> bPassed = <span class="keyword">false</span>;
<a name="l01938"></a>01938         bPassed = test.pass();
<a name="l01939"></a>01939         {
<a name="l01940"></a>01940             m_numPassed += test.getNumPassed();
<a name="l01941"></a>01941             m_numFailed += test.getNumFailed();
<a name="l01942"></a>01942         }
<a name="l01943"></a>01943 
<a name="l01944"></a>01944         <span class="keywordflow">if</span>(!bPassed) {
<a name="l01945"></a>01945             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in:  batch_snp6_dual_noAdapter part 1&quot;</span> + test.getErrorMsg());
<a name="l01946"></a>01946         }
<a name="l01947"></a>01947         <span class="keywordflow">else</span>
<a name="l01948"></a>01948         {
<a name="l01949"></a>01949             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: batch_snp6_dual_noAdapter part 1 &quot;</span> + test.getErrorMsg());            
<a name="l01950"></a>01950         }
<a name="l01951"></a>01951 
<a name="l01952"></a>01952         <span class="keywordflow">if</span> (bPassed)
<a name="l01953"></a>01953         {
<a name="l01954"></a>01954             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on batch_snp6_dual_noAdapter part 2.&quot;</span>);
<a name="l01955"></a>01955             <span class="keywordtype">string</span> outdir2 = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;batch_snp6_dual_noAdapter&quot;</span>);
<a name="l01956"></a>01956 
<a name="l01957"></a>01957             std::set&lt;std::string&gt; setSetIgnore;
<a name="l01958"></a>01958 
<a name="l01959"></a>01959             std::string  goldDirectory=<span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/batch_snp6_dual_noAdapter&quot;</span>;
<a name="l01960"></a>01960             std::string  newDirectory= <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;batch_snp6_dual_noAdapter&quot;</span>);
<a name="l01961"></a>01961             <span class="keywordtype">string</span> infix = <span class="stringliteral">&quot;CN5&quot;</span>;
<a name="l01962"></a>01962 
<a name="l01963"></a>01963             std::map&lt;std::string, float&gt; mapEpsilon;
<a name="l01964"></a>01964 
<a name="l01965"></a>01965             vector&lt;RegressionCheck *&gt; checks2;
<a name="l01966"></a>01966 
<a name="l01967"></a>01967             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classCychpCheck.html" title="Class for testing that two cychp files are the same +/- some epsilon.">CychpCheck</a>(
<a name="l01968"></a>01968                                                 m_vCytogenetics_SNP6_SixFiles,
<a name="l01969"></a>01969                                                 goldDirectory,
<a name="l01970"></a>01970                                                 newDirectory,
<a name="l01971"></a>01971                                                 infix,
<a name="l01972"></a>01972                                                 setIgnore,
<a name="l01973"></a>01973                                                 setSetIgnore,
<a name="l01974"></a>01974                                                 mapEpsilon,
<a name="l01975"></a>01975                                                 0.0001,
<a name="l01976"></a>01976                                                 0.9999989,
<a name="l01977"></a>01977                                                 <span class="keyword">false</span>));
<a name="l01978"></a>01978 
<a name="l01979"></a>01979             <span class="keywordtype">string</span> command2 = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l01980"></a>01980 <span class="stringliteral">            -verbose 1 \</span>
<a name="l01981"></a>01981 <span class="stringliteral">            --cyto2 false \</span>
<a name="l01982"></a>01982 <span class="stringliteral">            --force true \</span>
<a name="l01983"></a>01983 <span class="stringliteral">            --doDualNormalization true \</span>
<a name="l01984"></a>01984 <span class="stringliteral">            --adapter-type-normalization false \</span>
<a name="l01985"></a>01985 <span class="stringliteral">            --analysis genotype \</span>
<a name="l01986"></a>01986 <span class="stringliteral">            --analysis log2-ratio.gc-correction=false \</span>
<a name="l01987"></a>01987 <span class="stringliteral">            --analysis gaussian-smooth \</span>
<a name="l01988"></a>01988 <span class="stringliteral">            --analysis allelic-difference \</span>
<a name="l01989"></a>01989 <span class="stringliteral">            --analysis cn-state \</span>
<a name="l01990"></a>01990 <span class="stringliteral">            --analysis cn-segment \</span>
<a name="l01991"></a>01991 <span class="stringliteral">            --analysis lohCytoScan \</span>
<a name="l01992"></a>01992 <span class="stringliteral">            --analysis loh-segment  \</span>
<a name="l01993"></a>01993 <span class="stringliteral">            --analysis cn-neutral-loh \</span>
<a name="l01994"></a>01994 <span class="stringliteral">            --analysis normal-diploid \</span>
<a name="l01995"></a>01995 <span class="stringliteral">            --run-geno-qc true \</span>
<a name="l01996"></a>01996 <span class="stringliteral">            --cychp-output true \</span>
<a name="l01997"></a>01997 <span class="stringliteral">            --cnchp-output false \</span>
<a name="l01998"></a>01998 <span class="stringliteral">            --text-output true \</span>
<a name="l01999"></a>01999 <span class="stringliteral">            --reference-input &quot;</span> + m_testDir + <span class="stringliteral">&quot;/batch_snp6_dual_noAdapter/CNReference.cyto.snp6.a5.ref \</span>
<a name="l02000"></a>02000 <span class="stringliteral">            --set-analysis-name CN5 \</span>
<a name="l02001"></a>02001 <span class="stringliteral">            --out-dir &quot;</span> + outdir2 + <span class="stringliteral">&quot; \</span>
<a name="l02002"></a>02002 <span class="stringliteral">            --cdf-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.cdf \</span>
<a name="l02003"></a>02003 <span class="stringliteral">            --chrX-probes ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.chrXprobes \</span>
<a name="l02004"></a>02004 <span class="stringliteral">            --chrY-probes ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.chrYprobes \</span>
<a name="l02005"></a>02005 <span class="stringliteral">            --annotation-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.na30.2.annot.db \</span>
<a name="l02006"></a>02006 <span class="stringliteral">            --qca-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.qca \</span>
<a name="l02007"></a>02007 <span class="stringliteral">            --qcc-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.qcc \</span>
<a name="l02008"></a>02008 <span class="stringliteral">            --male-gender-ratio-cutoff 0.71 \</span>
<a name="l02009"></a>02009 <span class="stringliteral">            --female-gender-ratio-cutoff 0.48 \</span>
<a name="l02010"></a>02010 <span class="stringliteral">            --xx-cutoff 0.8 \</span>
<a name="l02011"></a>02011 <span class="stringliteral">            --xx-cutoff-high 1.07 \</span>
<a name="l02012"></a>02012 <span class="stringliteral">            --y-cutoff 0.55 \</span>
<a name="l02013"></a>02013 <span class="stringliteral">            --local-gc-background-correction-reference-method none \</span>
<a name="l02014"></a>02014 <span class="stringliteral">            --local-gc-background-intensity-adjustment-method none \</span>
<a name="l02015"></a>02015 <span class="stringliteral">            --image-correction-intensity-adjustment-method none \</span>
<a name="l02016"></a>02016 <span class="stringliteral">            --wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method.bandwidth=101.bin-count=25.wave-count=7 \</span>
<a name="l02017"></a>02017 <span class="stringliteral">            --wave-correction-reference-method wave-correction-reference-method.wave-count=7.trim=2.0.percentile=0.75.demean=false \</span>
<a name="l02018"></a>02018 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA12004_GW6_C.CEL \</span>
<a name="l02019"></a>02019 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA12003_GW6_C.CEL \</span>
<a name="l02020"></a>02020 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA11993_GW6_C.CEL \</span>
<a name="l02021"></a>02021 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA11882_GW6_C.CEL \</span>
<a name="l02022"></a>02022 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA11881_GW6_C.CEL \</span>
<a name="l02023"></a>02023 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA10863_GW6_C.CEL \</span>
<a name="l02024"></a>02024 <span class="stringliteral">             &quot;</span>;
<a name="l02025"></a>02025 
<a name="l02026"></a>02026              <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test2(<span class="stringliteral">&quot;batch_snp6_dual_noAdapter part 2&quot;</span>, command2.c_str(), checks2, <span class="keyword">false</span>);
<a name="l02027"></a>02027              test2.setSuite(*<span class="keyword">this</span>,  outdir2, outdir2 + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir2 + <span class="stringliteral">&quot;/valgrind.batch_snp6_dual_noAdapter2.log&quot;</span>);
<a name="l02028"></a>02028              <span class="keywordflow">if</span>(!test2.pass()) {
<a name="l02029"></a>02029                  <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: batch_snp6_dual_noAdapter part 2&quot;</span> + test2.getErrorMsg());
<a name="l02030"></a>02030              }
<a name="l02031"></a>02031              <span class="keywordflow">else</span>
<a name="l02032"></a>02032              {
<a name="l02033"></a>02033                  <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: batch_snp6_dual_noAdapter part 2 &quot;</span> + test.getErrorMsg());            
<a name="l02034"></a>02034              }
<a name="l02035"></a>02035 
<a name="l02036"></a>02036              {
<a name="l02037"></a>02037                  m_numPassed += test2.getNumPassed();
<a name="l02038"></a>02038                  m_numFailed += test2.getNumFailed();
<a name="l02039"></a>02039              }
<a name="l02040"></a>02040 
<a name="l02041"></a>02041         }
<a name="l02042"></a>02042     
<a name="l02043"></a>02043 
<a name="l02044"></a>02044 }
<a name="l02045"></a>02045 
<a name="l02046"></a>02046 
<a name="l02047"></a>02047 <span class="keywordtype">void</span> batch_snp6()
<a name="l02048"></a>02048 {
<a name="l02049"></a>02049         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on batch_snp6 part 1.&quot;</span>);
<a name="l02050"></a>02050         <span class="keywordtype">string</span> outdir = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;batch_snp6&quot;</span>);
<a name="l02051"></a>02051         <a class="code" href="classFs.html#a1206f6cf8f445461466f224e6cd1bc52" title="///">Fs::rmdirPath</a>(outdir, <span class="keyword">false</span>);
<a name="l02052"></a>02052         <a class="code" href="classFs.html#abf363360d88071c2f4a93a7e50d61c9f" title="Make one directory. ///.">Fs::mkdir</a>(outdir, <span class="keyword">false</span>);
<a name="l02053"></a>02053         vector&lt;RegressionCheck *&gt; checks;
<a name="l02054"></a>02054 
<a name="l02055"></a>02055         <span class="keywordtype">string</span> command = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l02056"></a>02056 <span class="stringliteral">        --verbose 1 \</span>
<a name="l02057"></a>02057 <span class="stringliteral">        --cyto2 false \</span>
<a name="l02058"></a>02058 <span class="stringliteral">        --doDualNormalization false \</span>
<a name="l02059"></a>02059 <span class="stringliteral">        --keep-intermediate-data false \</span>
<a name="l02060"></a>02060 <span class="stringliteral">        --force true \</span>
<a name="l02061"></a>02061 <span class="stringliteral">        --adapter-type-normalization true \</span>
<a name="l02062"></a>02062 <span class="stringliteral">        --analysis log2-ratio.gc-correction=false \</span>
<a name="l02063"></a>02063 <span class="stringliteral">        --analysis gaussian-smooth \</span>
<a name="l02064"></a>02064 <span class="stringliteral">        --analysis allelic-difference \</span>
<a name="l02065"></a>02065 <span class="stringliteral">        --analysis cn-state \</span>
<a name="l02066"></a>02066 <span class="stringliteral">        --analysis cn-segment \</span>
<a name="l02067"></a>02067 <span class="stringliteral">        --analysis lohCytoScan \</span>
<a name="l02068"></a>02068 <span class="stringliteral">        --analysis loh-segment  \</span>
<a name="l02069"></a>02069 <span class="stringliteral">        --analysis cn-neutral-loh \</span>
<a name="l02070"></a>02070 <span class="stringliteral">        --analysis normal-diploid \</span>
<a name="l02071"></a>02071 <span class="stringliteral">         --run-geno-qc true \</span>
<a name="l02072"></a>02072 <span class="stringliteral">         --cychp-output true \</span>
<a name="l02073"></a>02073 <span class="stringliteral">         --cnchp-output false \</span>
<a name="l02074"></a>02074 <span class="stringliteral">         --text-output true \</span>
<a name="l02075"></a>02075 <span class="stringliteral">         --reference-output &quot;</span> + m_testDir + <span class="stringliteral">&quot;/batch_snp6/CNReference.cyto.snp6.a5.ref \</span>
<a name="l02076"></a>02076 <span class="stringliteral">         --set-analysis-name CN5 \</span>
<a name="l02077"></a>02077 <span class="stringliteral">         --out-dir &quot;</span> + outdir + <span class="stringliteral">&quot; \</span>
<a name="l02078"></a>02078 <span class="stringliteral">         --cdf-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.cdf \</span>
<a name="l02079"></a>02079 <span class="stringliteral">         --chrX-probes ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.chrXprobes \</span>
<a name="l02080"></a>02080 <span class="stringliteral">         --chrY-probes ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.chrYprobes \</span>
<a name="l02081"></a>02081 <span class="stringliteral">         --annotation-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.na30.2.annot.db \</span>
<a name="l02082"></a>02082 <span class="stringliteral">         --qca-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.qca \</span>
<a name="l02083"></a>02083 <span class="stringliteral">         --qcc-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.qcc \</span>
<a name="l02084"></a>02084 <span class="stringliteral">         --male-gender-ratio-cutoff 0.71 \</span>
<a name="l02085"></a>02085 <span class="stringliteral">         --female-gender-ratio-cutoff 0.48 \</span>
<a name="l02086"></a>02086 <span class="stringliteral">         --xx-cutoff 0.8 \</span>
<a name="l02087"></a>02087 <span class="stringliteral">         --xx-cutoff-high 1.07 \</span>
<a name="l02088"></a>02088 <span class="stringliteral">         --y-cutoff 0.65 \</span>
<a name="l02089"></a>02089 <span class="stringliteral">         --local-gc-background-correction-reference-method none \</span>
<a name="l02090"></a>02090 <span class="stringliteral">         --local-gc-background-intensity-adjustment-method none \</span>
<a name="l02091"></a>02091 <span class="stringliteral">         --image-correction-intensity-adjustment-method none \</span>
<a name="l02092"></a>02092 <span class="stringliteral">         --wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method.bandwidth=101.bin-count=25.wave-count=7 \</span>
<a name="l02093"></a>02093 <span class="stringliteral">         --wave-correction-reference-method wave-correction-reference-method.wave-count=7.trim=2.0.percentile=0.75.demean=false \</span>
<a name="l02094"></a>02094 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA12004_GW6_C.CEL \</span>
<a name="l02095"></a>02095 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA12003_GW6_C.CEL \</span>
<a name="l02096"></a>02096 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA11993_GW6_C.CEL \</span>
<a name="l02097"></a>02097 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA11882_GW6_C.CEL \</span>
<a name="l02098"></a>02098 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA11881_GW6_C.CEL \</span>
<a name="l02099"></a>02099 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA10863_GW6_C.CEL \</span>
<a name="l02100"></a>02100 <span class="stringliteral">         &quot;</span>;
<a name="l02101"></a>02101 
<a name="l02102"></a>02102         std::string strFileName1 = m_testDir + <span class="stringliteral">&quot;/batch_snp6/CNReference.cyto.snp6.a5.ref&quot;</span>;
<a name="l02103"></a>02103         std::string strFileName2 = <span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/batch_snp6/CNReference.cyto.snp6.a5.ref&quot;</span>;
<a name="l02104"></a>02104         std::set&lt;std::string&gt; setLocalIgnore;
<a name="l02105"></a>02105 
<a name="l02106"></a>02106         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;AntigenomicProbes&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l02107"></a>02107         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;CN5.plier.feature-response&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l02108"></a>02108         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;CN5.snp-posteriors&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l02109"></a>02109         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;target-sketch&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l02110"></a>02110         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;Reference&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l02111"></a>02111         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;WaveCorrection&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l02112"></a>02112 
<a name="l02113"></a>02113         <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test(<span class="stringliteral">&quot;batch_snp6 part 1&quot;</span>, command.c_str(), checks);
<a name="l02114"></a>02114         test.setSuite(*<span class="keyword">this</span>,  outdir, outdir + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir + <span class="stringliteral">&quot;/valgrind.log&quot;</span>);
<a name="l02115"></a>02115         <span class="keywordtype">bool</span> bPassed = <span class="keyword">false</span>;
<a name="l02116"></a>02116         bPassed = test.pass();
<a name="l02117"></a>02117         {
<a name="l02118"></a>02118 
<a name="l02119"></a>02119         <span class="keywordflow">if</span>(!bPassed) {
<a name="l02120"></a>02120             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: batch_snp6 part 1 &quot;</span> + test.getErrorMsg());
<a name="l02121"></a>02121         }
<a name="l02122"></a>02122         <span class="keywordflow">else</span>
<a name="l02123"></a>02123         {
<a name="l02124"></a>02124             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: batch_snp6 part 1 &quot;</span> + test.getErrorMsg());            
<a name="l02125"></a>02125         }
<a name="l02126"></a>02126             m_numPassed += test.getNumPassed();
<a name="l02127"></a>02127             m_numFailed += test.getNumFailed();
<a name="l02128"></a>02128         }
<a name="l02129"></a>02129 
<a name="l02130"></a>02130 
<a name="l02131"></a>02131        <span class="keywordflow">if</span> (bPassed)
<a name="l02132"></a>02132         {
<a name="l02133"></a>02133             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on batch_snp6 part 2.&quot;</span>);
<a name="l02134"></a>02134             <span class="keywordtype">string</span> outdir2 = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;batch_snp6&quot;</span>);
<a name="l02135"></a>02135 
<a name="l02136"></a>02136             std::set&lt;std::string&gt; setSetIgnore;
<a name="l02137"></a>02137 
<a name="l02138"></a>02138             std::string  goldDirectory=<span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/batch_snp6&quot;</span>;
<a name="l02139"></a>02139             std::string  newDirectory=<a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;batch_snp6&quot;</span>);
<a name="l02140"></a>02140             <span class="keywordtype">string</span> infix = <span class="stringliteral">&quot;CN5&quot;</span>;
<a name="l02141"></a>02141 
<a name="l02142"></a>02142             std::map&lt;std::string, float&gt; mapEpsilon;
<a name="l02143"></a>02143 
<a name="l02144"></a>02144             vector&lt;RegressionCheck *&gt; checks2;
<a name="l02145"></a>02145 
<a name="l02146"></a>02146             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classCychpCheck.html" title="Class for testing that two cychp files are the same +/- some epsilon.">CychpCheck</a>(
<a name="l02147"></a>02147                                                 m_vCytogenetics_SNP6_SixFiles,
<a name="l02148"></a>02148                                                 goldDirectory,
<a name="l02149"></a>02149                                                 newDirectory,
<a name="l02150"></a>02150                                                 infix,
<a name="l02151"></a>02151                                                 setIgnore,
<a name="l02152"></a>02152                                                 setSetIgnore,
<a name="l02153"></a>02153                                                 mapEpsilon,
<a name="l02154"></a>02154                                                 0.0001,
<a name="l02155"></a>02155                                                 0.9999989,
<a name="l02156"></a>02156                                                 <span class="keyword">false</span>));
<a name="l02157"></a>02157 
<a name="l02158"></a>02158 
<a name="l02159"></a>02159 
<a name="l02160"></a>02160             <span class="keywordtype">string</span> command2 = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l02161"></a>02161 <span class="stringliteral">            -verbose 1 \</span>
<a name="l02162"></a>02162 <span class="stringliteral">            --cyto2 false \</span>
<a name="l02163"></a>02163 <span class="stringliteral">            --force true \</span>
<a name="l02164"></a>02164 <span class="stringliteral">            --adapter-type-normalization true \</span>
<a name="l02165"></a>02165 <span class="stringliteral">            --doDualNormalization false \</span>
<a name="l02166"></a>02166 <span class="stringliteral">            --keep-intermediate-data false \</span>
<a name="l02167"></a>02167 <span class="stringliteral">            --analysis genotype \</span>
<a name="l02168"></a>02168 <span class="stringliteral">            --analysis log2-ratio.gc-correction=false \</span>
<a name="l02169"></a>02169 <span class="stringliteral">            --analysis gaussian-smooth \</span>
<a name="l02170"></a>02170 <span class="stringliteral">            --analysis allelic-difference \</span>
<a name="l02171"></a>02171 <span class="stringliteral">            --analysis cn-state \</span>
<a name="l02172"></a>02172 <span class="stringliteral">            --analysis cn-segment \</span>
<a name="l02173"></a>02173 <span class="stringliteral">            --analysis lohCytoScan \</span>
<a name="l02174"></a>02174 <span class="stringliteral">            --analysis loh-segment  \</span>
<a name="l02175"></a>02175 <span class="stringliteral">            --analysis cn-neutral-loh \</span>
<a name="l02176"></a>02176 <span class="stringliteral">            --analysis normal-diploid \</span>
<a name="l02177"></a>02177 <span class="stringliteral">            --run-geno-qc true \</span>
<a name="l02178"></a>02178 <span class="stringliteral">            --cychp-output true \</span>
<a name="l02179"></a>02179 <span class="stringliteral">            --cnchp-output false \</span>
<a name="l02180"></a>02180 <span class="stringliteral">            --text-output true \</span>
<a name="l02181"></a>02181 <span class="stringliteral">            --reference-input &quot;</span> + m_testDir + <span class="stringliteral">&quot;/batch_snp6/CNReference.cyto.snp6.a5.ref \</span>
<a name="l02182"></a>02182 <span class="stringliteral">            --set-analysis-name CN5 \</span>
<a name="l02183"></a>02183 <span class="stringliteral">            --out-dir &quot;</span> + outdir2 + <span class="stringliteral">&quot; \</span>
<a name="l02184"></a>02184 <span class="stringliteral">            --cdf-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.cdf \</span>
<a name="l02185"></a>02185 <span class="stringliteral">            --chrX-probes ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.chrXprobes \</span>
<a name="l02186"></a>02186 <span class="stringliteral">            --chrY-probes ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.chrYprobes \</span>
<a name="l02187"></a>02187 <span class="stringliteral">            --annotation-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.na30.2.annot.db \</span>
<a name="l02188"></a>02188 <span class="stringliteral">            --qca-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.qca \</span>
<a name="l02189"></a>02189 <span class="stringliteral">            --qcc-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.qcc \</span>
<a name="l02190"></a>02190 <span class="stringliteral">            --male-gender-ratio-cutoff 0.71 \</span>
<a name="l02191"></a>02191 <span class="stringliteral">            --female-gender-ratio-cutoff 0.48 \</span>
<a name="l02192"></a>02192 <span class="stringliteral">            --xx-cutoff 0.8 \</span>
<a name="l02193"></a>02193 <span class="stringliteral">            --xx-cutoff-high 1.07 \</span>
<a name="l02194"></a>02194 <span class="stringliteral">            --y-cutoff 0.65 \</span>
<a name="l02195"></a>02195 <span class="stringliteral">            --local-gc-background-correction-reference-method none \</span>
<a name="l02196"></a>02196 <span class="stringliteral">            --local-gc-background-intensity-adjustment-method none \</span>
<a name="l02197"></a>02197 <span class="stringliteral">            --image-correction-intensity-adjustment-method none \</span>
<a name="l02198"></a>02198 <span class="stringliteral">            --wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method.bandwidth=101.bin-count=25.wave-count=7 \</span>
<a name="l02199"></a>02199 <span class="stringliteral">            --wave-correction-reference-method wave-correction-reference-method.wave-count=7.trim=2.0.percentile=0.75.demean=false \</span>
<a name="l02200"></a>02200 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA12004_GW6_C.CEL \</span>
<a name="l02201"></a>02201 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA12003_GW6_C.CEL \</span>
<a name="l02202"></a>02202 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA11993_GW6_C.CEL \</span>
<a name="l02203"></a>02203 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA11882_GW6_C.CEL \</span>
<a name="l02204"></a>02204 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA11881_GW6_C.CEL \</span>
<a name="l02205"></a>02205 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA10863_GW6_C.CEL \</span>
<a name="l02206"></a>02206 <span class="stringliteral">            &quot;</span>;
<a name="l02207"></a>02207              <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test2(<span class="stringliteral">&quot;batch_snp6 part 2&quot;</span>, command2.c_str(), checks2, <span class="keyword">false</span>);
<a name="l02208"></a>02208              test2.setSuite(*<span class="keyword">this</span>,  outdir2, outdir2 + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir2 + <span class="stringliteral">&quot;/valgrind.batch_snp62.log&quot;</span>);
<a name="l02209"></a>02209              <span class="keywordflow">if</span>(!test2.pass()) {
<a name="l02210"></a>02210                  <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: batch_snp6  part 2&quot;</span> + test2.getErrorMsg());
<a name="l02211"></a>02211              }
<a name="l02212"></a>02212              <span class="keywordflow">else</span>
<a name="l02213"></a>02213              {
<a name="l02214"></a>02214                   <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: batch_snp6 part 2 &quot;</span> + test.getErrorMsg());            
<a name="l02215"></a>02215              }
<a name="l02216"></a>02216 
<a name="l02217"></a>02217              {
<a name="l02218"></a>02218                  m_numPassed += test2.getNumPassed();
<a name="l02219"></a>02219                  m_numFailed += test2.getNumFailed();
<a name="l02220"></a>02220              }
<a name="l02221"></a>02221 
<a name="l02222"></a>02222         }
<a name="l02223"></a>02223  
<a name="l02224"></a>02224 }
<a name="l02225"></a>02225 
<a name="l02226"></a>02226 <span class="keywordtype">void</span> batch_snp6_dual()
<a name="l02227"></a>02227 {
<a name="l02228"></a>02228         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on batch_snp6_dual part 1.&quot;</span>);
<a name="l02229"></a>02229         <span class="keywordtype">string</span> outdir = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;batch_snp6_dual&quot;</span>);
<a name="l02230"></a>02230         <a class="code" href="classFs.html#a1206f6cf8f445461466f224e6cd1bc52" title="///">Fs::rmdirPath</a>(outdir, <span class="keyword">false</span>);
<a name="l02231"></a>02231         <a class="code" href="classFs.html#abf363360d88071c2f4a93a7e50d61c9f" title="Make one directory. ///.">Fs::mkdir</a>(outdir, <span class="keyword">false</span>);
<a name="l02232"></a>02232         vector&lt;RegressionCheck *&gt; checks;
<a name="l02233"></a>02233 
<a name="l02234"></a>02234         <span class="keywordtype">string</span> command = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l02235"></a>02235 <span class="stringliteral">        --verbose 1 \</span>
<a name="l02236"></a>02236 <span class="stringliteral">        --cyto2 false \</span>
<a name="l02237"></a>02237 <span class="stringliteral">        --doDualNormalization true \</span>
<a name="l02238"></a>02238 <span class="stringliteral">        --keep-intermediate-data false \</span>
<a name="l02239"></a>02239 <span class="stringliteral">        --force true \</span>
<a name="l02240"></a>02240 <span class="stringliteral">        --adapter-type-normalization true \</span>
<a name="l02241"></a>02241 <span class="stringliteral">        --analysis log2-ratio.gc-correction=false \</span>
<a name="l02242"></a>02242 <span class="stringliteral">        --analysis gaussian-smooth \</span>
<a name="l02243"></a>02243 <span class="stringliteral">        --analysis allelic-difference \</span>
<a name="l02244"></a>02244 <span class="stringliteral">        --analysis cn-state \</span>
<a name="l02245"></a>02245 <span class="stringliteral">        --analysis cn-segment \</span>
<a name="l02246"></a>02246 <span class="stringliteral">        --analysis lohCytoScan \</span>
<a name="l02247"></a>02247 <span class="stringliteral">        --analysis loh-segment  \</span>
<a name="l02248"></a>02248 <span class="stringliteral">        --analysis cn-neutral-loh \</span>
<a name="l02249"></a>02249 <span class="stringliteral">        --analysis normal-diploid \</span>
<a name="l02250"></a>02250 <span class="stringliteral">         --run-geno-qc true \</span>
<a name="l02251"></a>02251 <span class="stringliteral">         --cychp-output true \</span>
<a name="l02252"></a>02252 <span class="stringliteral">         --cnchp-output false \</span>
<a name="l02253"></a>02253 <span class="stringliteral">         --text-output true \</span>
<a name="l02254"></a>02254 <span class="stringliteral">         --reference-output &quot;</span> + m_testDir + <span class="stringliteral">&quot;/batch_snp6_dual/CNReference.cyto.snp6.a5.ref \</span>
<a name="l02255"></a>02255 <span class="stringliteral">         --set-analysis-name CN5 \</span>
<a name="l02256"></a>02256 <span class="stringliteral">         --out-dir &quot;</span> + outdir + <span class="stringliteral">&quot; \</span>
<a name="l02257"></a>02257 <span class="stringliteral">         --cdf-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.cdf \</span>
<a name="l02258"></a>02258 <span class="stringliteral">         --chrX-probes ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.chrXprobes \</span>
<a name="l02259"></a>02259 <span class="stringliteral">         --chrY-probes ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.chrYprobes \</span>
<a name="l02260"></a>02260 <span class="stringliteral">         --annotation-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.na30.2.annot.db \</span>
<a name="l02261"></a>02261 <span class="stringliteral">         --qca-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.qca \</span>
<a name="l02262"></a>02262 <span class="stringliteral">         --qcc-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.qcc \</span>
<a name="l02263"></a>02263 <span class="stringliteral">         --male-gender-ratio-cutoff 0.71 \</span>
<a name="l02264"></a>02264 <span class="stringliteral">         --female-gender-ratio-cutoff 0.48 \</span>
<a name="l02265"></a>02265 <span class="stringliteral">         --xx-cutoff 0.8 \</span>
<a name="l02266"></a>02266 <span class="stringliteral">         --xx-cutoff-high 1.07 \</span>
<a name="l02267"></a>02267 <span class="stringliteral">         --y-cutoff 0.65 \</span>
<a name="l02268"></a>02268 <span class="stringliteral">         --local-gc-background-correction-reference-method none \</span>
<a name="l02269"></a>02269 <span class="stringliteral">         --local-gc-background-intensity-adjustment-method none \</span>
<a name="l02270"></a>02270 <span class="stringliteral">         --image-correction-intensity-adjustment-method none \</span>
<a name="l02271"></a>02271 <span class="stringliteral">         --wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method.bandwidth=101.bin-count=25.wave-count=7 \</span>
<a name="l02272"></a>02272 <span class="stringliteral">         --wave-correction-reference-method wave-correction-reference-method.wave-count=7.trim=2.0.percentile=0.75.demean=false \</span>
<a name="l02273"></a>02273 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA12004_GW6_C.CEL \</span>
<a name="l02274"></a>02274 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA12003_GW6_C.CEL \</span>
<a name="l02275"></a>02275 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA11993_GW6_C.CEL \</span>
<a name="l02276"></a>02276 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA11882_GW6_C.CEL \</span>
<a name="l02277"></a>02277 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA11881_GW6_C.CEL \</span>
<a name="l02278"></a>02278 <span class="stringliteral">         ../../../regression-data/data/cel/GenomeWideSNP_6/NA10863_GW6_C.CEL \</span>
<a name="l02279"></a>02279 <span class="stringliteral">         &quot;</span>;
<a name="l02280"></a>02280 
<a name="l02281"></a>02281         <span class="keywordtype">bool</span> bPassed = <span class="keyword">false</span>;
<a name="l02282"></a>02282         std::string strFileName1 = m_testDir + <span class="stringliteral">&quot;/batch_snp6_dual/CNReference.cyto.snp6.a5.ref&quot;</span>;
<a name="l02283"></a>02283         std::string strFileName2 = <span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/batch_snp6_dual/CNReference.cyto.snp6.a5.ref&quot;</span>;
<a name="l02284"></a>02284         std::set&lt;std::string&gt; setLocalIgnore;
<a name="l02285"></a>02285 
<a name="l02286"></a>02286         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;AntigenomicProbes&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l02287"></a>02287         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;CN5.plier.feature-response&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l02288"></a>02288         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;CN5.snp-posteriors&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l02289"></a>02289         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;target-sketch&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l02290"></a>02290         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;Reference&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l02291"></a>02291         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;WaveCorrection&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l02292"></a>02292         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;SketchSNP&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l02293"></a>02293         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;SketchCN&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l02294"></a>02294 
<a name="l02295"></a>02295         <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test(<span class="stringliteral">&quot;batch_snp6_dual part 1&quot;</span>, command.c_str(), checks);
<a name="l02296"></a>02296         test.setSuite(*<span class="keyword">this</span>,  outdir, outdir + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir + <span class="stringliteral">&quot;/valgrind.log&quot;</span>);
<a name="l02297"></a>02297         bPassed = test.pass();
<a name="l02298"></a>02298 
<a name="l02299"></a>02299         <span class="keywordflow">if</span>(!bPassed) {
<a name="l02300"></a>02300             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in:  batch_snp6_dual part 1&quot;</span> + test.getErrorMsg());
<a name="l02301"></a>02301         }
<a name="l02302"></a>02302         <span class="keywordflow">else</span>
<a name="l02303"></a>02303         {
<a name="l02304"></a>02304             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: batch_snp6_dual part 1 &quot;</span> + test.getErrorMsg());            
<a name="l02305"></a>02305         }
<a name="l02306"></a>02306         {
<a name="l02307"></a>02307             m_numPassed += test.getNumPassed();
<a name="l02308"></a>02308             m_numFailed += test.getNumFailed();
<a name="l02309"></a>02309  
<a name="l02310"></a>02310         } 
<a name="l02311"></a>02311         <span class="keywordflow">if</span> (bPassed)
<a name="l02312"></a>02312         {
<a name="l02313"></a>02313             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on batch_snp6_dual part 2.&quot;</span>);
<a name="l02314"></a>02314             <span class="keywordtype">string</span> outdir2 = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;/batch_snp6_dual&quot;</span>);
<a name="l02315"></a>02315 
<a name="l02316"></a>02316             std::set&lt;std::string&gt; setSetIgnore;
<a name="l02317"></a>02317 
<a name="l02318"></a>02318             std::string  goldDirectory=<span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/batch_snp6_dual&quot;</span>;
<a name="l02319"></a>02319             std::string  newDirectory=<a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;batch_snp6_dual&quot;</span>);
<a name="l02320"></a>02320             <span class="keywordtype">string</span> infix = <span class="stringliteral">&quot;CN5&quot;</span>;
<a name="l02321"></a>02321 
<a name="l02322"></a>02322             std::map&lt;std::string, float&gt; mapEpsilon;
<a name="l02323"></a>02323 
<a name="l02324"></a>02324             vector&lt;RegressionCheck *&gt; checks2;
<a name="l02325"></a>02325 
<a name="l02326"></a>02326             checks2.push_back(<span class="keyword">new</span> <a class="code" href="classCychpCheck.html" title="Class for testing that two cychp files are the same +/- some epsilon.">CychpCheck</a>(
<a name="l02327"></a>02327                                                 m_vCytogenetics_SNP6_SixFiles,
<a name="l02328"></a>02328                                                 goldDirectory,
<a name="l02329"></a>02329                                                 newDirectory,
<a name="l02330"></a>02330                                                 infix,
<a name="l02331"></a>02331                                                 setIgnore,
<a name="l02332"></a>02332                                                 setSetIgnore,
<a name="l02333"></a>02333                                                 mapEpsilon,
<a name="l02334"></a>02334                                                 0.0001,
<a name="l02335"></a>02335                                                 0.9999989,
<a name="l02336"></a>02336                                                 <span class="keyword">false</span>));
<a name="l02337"></a>02337 
<a name="l02338"></a>02338             <span class="keywordtype">string</span> command2 = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l02339"></a>02339 <span class="stringliteral">            -verbose 1 \</span>
<a name="l02340"></a>02340 <span class="stringliteral">            --cyto2 false \</span>
<a name="l02341"></a>02341 <span class="stringliteral">            --force true \</span>
<a name="l02342"></a>02342 <span class="stringliteral">            --doDualNormalization true \</span>
<a name="l02343"></a>02343 <span class="stringliteral">            --adapter-type-normalization true \</span>
<a name="l02344"></a>02344 <span class="stringliteral">            --analysis genotype \</span>
<a name="l02345"></a>02345 <span class="stringliteral">            --analysis log2-ratio.gc-correction=false \</span>
<a name="l02346"></a>02346 <span class="stringliteral">            --analysis gaussian-smooth \</span>
<a name="l02347"></a>02347 <span class="stringliteral">            --analysis allelic-difference \</span>
<a name="l02348"></a>02348 <span class="stringliteral">            --analysis cn-state \</span>
<a name="l02349"></a>02349 <span class="stringliteral">            --analysis cn-segment \</span>
<a name="l02350"></a>02350 <span class="stringliteral">            --analysis lohCytoScan \</span>
<a name="l02351"></a>02351 <span class="stringliteral">            --analysis loh-segment  \</span>
<a name="l02352"></a>02352 <span class="stringliteral">            --analysis cn-neutral-loh \</span>
<a name="l02353"></a>02353 <span class="stringliteral">            --analysis normal-diploid \</span>
<a name="l02354"></a>02354 <span class="stringliteral">            --run-geno-qc true \</span>
<a name="l02355"></a>02355 <span class="stringliteral">            --cychp-output true \</span>
<a name="l02356"></a>02356 <span class="stringliteral">            --cnchp-output false \</span>
<a name="l02357"></a>02357 <span class="stringliteral">            --text-output true \</span>
<a name="l02358"></a>02358 <span class="stringliteral">            --reference-input &quot;</span> + m_testDir + <span class="stringliteral">&quot;/batch_snp6_dual/CNReference.cyto.snp6.a5.ref \</span>
<a name="l02359"></a>02359 <span class="stringliteral">            --set-analysis-name CN5 \</span>
<a name="l02360"></a>02360 <span class="stringliteral">            --out-dir &quot;</span> + outdir2 + <span class="stringliteral">&quot; \</span>
<a name="l02361"></a>02361 <span class="stringliteral">            --cdf-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.cdf \</span>
<a name="l02362"></a>02362 <span class="stringliteral">            --chrX-probes ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.chrXprobes \</span>
<a name="l02363"></a>02363 <span class="stringliteral">            --chrY-probes ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.chrYprobes \</span>
<a name="l02364"></a>02364 <span class="stringliteral">            --annotation-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.na30.2.annot.db \</span>
<a name="l02365"></a>02365 <span class="stringliteral">            --qca-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.qca \</span>
<a name="l02366"></a>02366 <span class="stringliteral">            --qcc-file ../../../regression-data/data/lib/GenomeWideSNP_6/GenomeWideSNP_6.qcc \</span>
<a name="l02367"></a>02367 <span class="stringliteral">            --male-gender-ratio-cutoff 0.71 \</span>
<a name="l02368"></a>02368 <span class="stringliteral">            --female-gender-ratio-cutoff 0.48 \</span>
<a name="l02369"></a>02369 <span class="stringliteral">            --xx-cutoff 0.8 \</span>
<a name="l02370"></a>02370 <span class="stringliteral">            --xx-cutoff-high 1.07 \</span>
<a name="l02371"></a>02371 <span class="stringliteral">            --y-cutoff 0.65 \</span>
<a name="l02372"></a>02372 <span class="stringliteral">            --local-gc-background-correction-reference-method none \</span>
<a name="l02373"></a>02373 <span class="stringliteral">            --local-gc-background-intensity-adjustment-method none \</span>
<a name="l02374"></a>02374 <span class="stringliteral">            --image-correction-intensity-adjustment-method none \</span>
<a name="l02375"></a>02375 <span class="stringliteral">            --wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method.bandwidth=101.bin-count=25.wave-count=7 \</span>
<a name="l02376"></a>02376 <span class="stringliteral">            --wave-correction-reference-method wave-correction-reference-method.wave-count=7.trim=2.0.percentile=0.75.demean=false \</span>
<a name="l02377"></a>02377 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA12004_GW6_C.CEL \</span>
<a name="l02378"></a>02378 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA12003_GW6_C.CEL \</span>
<a name="l02379"></a>02379 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA11993_GW6_C.CEL \</span>
<a name="l02380"></a>02380 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA11882_GW6_C.CEL \</span>
<a name="l02381"></a>02381 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA11881_GW6_C.CEL \</span>
<a name="l02382"></a>02382 <span class="stringliteral">            ../../../regression-data/data/cel/GenomeWideSNP_6/NA10863_GW6_C.CEL \</span>
<a name="l02383"></a>02383 <span class="stringliteral">             &quot;</span>;
<a name="l02384"></a>02384 
<a name="l02385"></a>02385              <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test2(<span class="stringliteral">&quot;batch_snp6_dual part2&quot;</span>, command2.c_str(), checks2, <span class="keyword">false</span>);
<a name="l02386"></a>02386              test2.setSuite(*<span class="keyword">this</span>,  outdir2, outdir2 + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir2 + <span class="stringliteral">&quot;/valgrind.batch_snp6_dual2.log&quot;</span>);
<a name="l02387"></a>02387              <span class="keywordflow">if</span>(!test2.pass()) {
<a name="l02388"></a>02388                  <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: batch_snp6_dual part 2&quot;</span> + test2.getErrorMsg());
<a name="l02389"></a>02389              }
<a name="l02390"></a>02390              <span class="keywordflow">else</span>
<a name="l02391"></a>02391              {
<a name="l02392"></a>02392                  <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: batch_snp6_dual part 2 &quot;</span> + test.getErrorMsg());            
<a name="l02393"></a>02393              }
<a name="l02394"></a>02394              {
<a name="l02395"></a>02395                  m_numPassed += test2.getNumPassed();
<a name="l02396"></a>02396                  m_numFailed += test2.getNumFailed();
<a name="l02397"></a>02397              }
<a name="l02398"></a>02398 
<a name="l02399"></a>02399         }
<a name="l02400"></a>02400 }
<a name="l02401"></a>02401 
<a name="l02402"></a>02402 <span class="keywordtype">void</span> CytoScanBadAnno()
<a name="l02403"></a>02403 {
<a name="l02404"></a>02404         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Working on CytoScanBadAnno.&quot;</span>);
<a name="l02405"></a>02405         <span class="keywordtype">string</span> outdir = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(m_testDir, <span class="stringliteral">&quot;CytoScanBadAnno&quot;</span>);
<a name="l02406"></a>02406         <a class="code" href="classFs.html#a1206f6cf8f445461466f224e6cd1bc52" title="///">Fs::rmdirPath</a>(outdir, <span class="keyword">false</span>);
<a name="l02407"></a>02407         <a class="code" href="classFs.html#abf363360d88071c2f4a93a7e50d61c9f" title="Make one directory. ///.">Fs::mkdir</a>(outdir, <span class="keyword">false</span>);
<a name="l02408"></a>02408         vector&lt;RegressionCheck *&gt; checks;
<a name="l02409"></a>02409 
<a name="l02410"></a>02410         <span class="keywordtype">string</span> command = <span class="stringliteral">&quot;./apt-copynumber-cyto \</span>
<a name="l02411"></a>02411 <span class="stringliteral">                    -verbose 4 \</span>
<a name="l02412"></a>02412 <span class="stringliteral">                    --cyto2 false \</span>
<a name="l02413"></a>02413 <span class="stringliteral">                    --force false \</span>
<a name="l02414"></a>02414 <span class="stringliteral">                    --keep-intermediate-data false \</span>
<a name="l02415"></a>02415 <span class="stringliteral">                    --doDualNormalization true \</span>
<a name="l02416"></a>02416 <span class="stringliteral">                    --adapter-type-normalization false \</span>
<a name="l02417"></a>02417 <span class="stringliteral">                    --probe-file ../../../regression-data/data/lib/CytoScan750K/CytoScan750K_Array.probe_tab \</span>
<a name="l02418"></a>02418 <span class="stringliteral">                    --run-geno-qc true \</span>
<a name="l02419"></a>02419 <span class="stringliteral">                    --qca-file ../../../regression-data/data/lib/CytoScan750K/CytoScan750K_Array.v1.qca \</span>
<a name="l02420"></a>02420 <span class="stringliteral">                    --qcc-file ../../../regression-data/data/lib/CytoScan750K/CytoScan750K_Array.v4.qcc \</span>
<a name="l02421"></a>02421 <span class="stringliteral">                    --text-output false \</span>
<a name="l02422"></a>02422 <span class="stringliteral">                    --cnchp-output false \</span>
<a name="l02423"></a>02423 <span class="stringliteral">                    --cychp-output true \</span>
<a name="l02424"></a>02424 <span class="stringliteral">                    --reference-output &quot;</span> + m_testDir + <span class="stringliteral">&quot;/CytoScanBadAnno/CNReferenceOutput.a5 \</span>
<a name="l02425"></a>02425 <span class="stringliteral">                    --cdf-file ../../../regression-data/data/lib/CytoScan750K/CytoScan750K_Array.cdf \</span>
<a name="l02426"></a>02426 <span class="stringliteral">                    --chrX-probes ../../../regression-data/data/lib/CytoScan750K/CytoScan750K_Array.v3.chrXprobes \</span>
<a name="l02427"></a>02427 <span class="stringliteral">                    --chrY-probes ../../../regression-data/data/lib/CytoScan750K/CytoScan750K_Array.v3.chrYprobes \</span>
<a name="l02428"></a>02428 <span class="stringliteral">                    --annotation-file ../../../regression-data/data/lib/CytoScan750K/ProbeSetsReorderedAndBadAnnosAdded.CytoScan750K_Array.na32.v8.annot.db \</span>
<a name="l02429"></a>02429 <span class="stringliteral">                    --male-gender-ratio-cutoff 1.5 \</span>
<a name="l02430"></a>02430 <span class="stringliteral">                    --female-gender-ratio-cutoff 0.9 \</span>
<a name="l02431"></a>02431 <span class="stringliteral">                    --xx-cutoff 0.61 \</span>
<a name="l02432"></a>02432 <span class="stringliteral">                    --xx-cutoff-high 0.95 \</span>
<a name="l02433"></a>02433 <span class="stringliteral">                    --y-cutoff 0.58 \</span>
<a name="l02434"></a>02434 <span class="stringliteral">                    --out-dir &quot;</span> + outdir + <span class="stringliteral">&quot; \</span>
<a name="l02435"></a>02435 <span class="stringliteral">                    --covariates-file  ../../../regression-data/data/lib/CytoScan750K/CytoScan750K_Array.hg19.v4.refcovar \</span>
<a name="l02436"></a>02436 <span class="stringliteral">                    --signal-adjustment-covariates covariate-signal-adjuster.order=fragment-adapter-type,fragment-length.bin-type=discrete,equally-populated.bin-count=0,100 \</span>
<a name="l02437"></a>02437 <span class="stringliteral">                    --lr-adjustment-covariates covariate-lr-adjuster.order=SuperGC,median-signal,marker-class.bin-type=discrete,equally-populated,discrete.bin-count=0,100,0.iqr-scaling=on,on,on.subtract-from-XY=on,off,on \</span>
<a name="l02438"></a>02438 <span class="stringliteral">                    --wave-correction-reference-method wave-correction-reference-method.trim=2.0.percentile=0.75.wave-count=6.demean=false \</span>
<a name="l02439"></a>02439 <span class="stringliteral">                    --local-gc-background-correction-reference-method none \</span>
<a name="l02440"></a>02440 <span class="stringliteral">                    --local-gc-background-intensity-adjustment-method none \</span>
<a name="l02441"></a>02441 <span class="stringliteral">                    --image-correction-intensity-adjustment-method none \</span>
<a name="l02442"></a>02442 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan750K/NA07345_C05_PlateD_CytoScanLD_WW_20110830.CEL \</span>
<a name="l02443"></a>02443 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan750K/NA07348_C06_PlateD_CytoScanLD_WW_20110830.CEL \</span>
<a name="l02444"></a>02444 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan750K/NA07357_C07_PlateD_CytoScanLD_WW_20110830.CEL \</span>
<a name="l02445"></a>02445 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan750K/NA10830_C08_PlateD_CytoScanLD_WW_20110830.CEL \</span>
<a name="l02446"></a>02446 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan750K/NA10831_A09_PlateDb_CytoScanLD_CC_20110824.CEL \</span>
<a name="l02447"></a>02447 <span class="stringliteral">                    ../../../regression-data/data/cel/CytoScan750K/NA10835_A10_PlateDb_CytoScanLD_CC_20110824.CEL \</span>
<a name="l02448"></a>02448 <span class="stringliteral">         &quot;</span>;
<a name="l02449"></a>02449 
<a name="l02450"></a>02450         std::string strFileName1 = m_testDir + <span class="stringliteral">&quot;/CytoScanBadAnno/CNReferenceOutput.a5&quot;</span>;
<a name="l02451"></a>02451         std::string strFileName2 = <span class="stringliteral">&quot;../../../regression-data/data/copynumber/copynumber-cyto/CytoScanBadAnno/CNReferenceOutput.a5&quot;</span>;
<a name="l02452"></a>02452         std::set&lt;std::string&gt; setLocalIgnore;
<a name="l02453"></a>02453 
<a name="l02454"></a>02454         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CN5&quot;</span>, <span class="stringliteral">&quot;CN5.snp-posteriors&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l02455"></a>02455         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;Reference&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l02456"></a>02456         checks.push_back(<span class="keyword">new</span> <a class="code" href="classFile5EquivalentCheck.html">File5EquivalentCheck</a>(strFileName1, strFileName2, <span class="stringliteral">&quot;CopyNumber&quot;</span>, <span class="stringliteral">&quot;Covariates&quot;</span>, setLocalIgnore, 0.0001, 1.0));
<a name="l02457"></a>02457 
<a name="l02458"></a>02458         <a class="code" href="classRegressionTest.html" title="Utility class for helping to do regression testing.">RegressionTest</a> test(<span class="stringliteral">&quot;CytoScanBadAnno&quot;</span>, command.c_str(), checks);
<a name="l02459"></a>02459         test.setSuite(*<span class="keyword">this</span>,  outdir, outdir + <span class="stringliteral">&quot;/apt-copynumber-cyto.log&quot;</span>, outdir + <span class="stringliteral">&quot;/valgrind.log&quot;</span>);
<a name="l02460"></a>02460         <span class="keywordtype">bool</span> bPassed = <span class="keyword">false</span>;
<a name="l02461"></a>02461         bPassed = test.pass();
<a name="l02462"></a>02462         <span class="keywordflow">if</span>(!bPassed) {
<a name="l02463"></a>02463             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Error in: CytoScanBadAnno &quot;</span> + test.getErrorMsg());
<a name="l02464"></a>02464         }
<a name="l02465"></a>02465         <span class="keywordflow">else</span>
<a name="l02466"></a>02466         {
<a name="l02467"></a>02467             <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Passed: CytoScanBadAnno &quot;</span> + test.getErrorMsg());
<a name="l02468"></a>02468         }
<a name="l02469"></a>02469 
<a name="l02470"></a>02470         {
<a name="l02471"></a>02471             m_numPassed += test.getNumPassed();
<a name="l02472"></a>02472             m_numFailed += test.getNumFailed();
<a name="l02473"></a>02473         }
<a name="l02474"></a>02474 }
<a name="l02475"></a>02475 
<a name="l02476"></a>02476 <span class="keywordtype">int</span> getNumPassed()
<a name="l02477"></a>02477 {
<a name="l02478"></a>02478     <span class="keywordflow">return</span> m_numPassed;
<a name="l02479"></a>02479 }
<a name="l02480"></a>02480 
<a name="l02481"></a>02481 <span class="keywordtype">int</span> getNumFailed()
<a name="l02482"></a>02482 {
<a name="l02483"></a>02483     <span class="keywordflow">return</span> m_numFailed;
<a name="l02484"></a>02484 }
<a name="l02485"></a>02485 
<a name="l02486"></a>02486 };
<a name="l02487"></a>02487 
<a name="l02488"></a>02488 
<a name="l02489"></a>02489 <span class="keywordtype">int</span> <a class="code" href="canary-test_8cpp.html#a0ddf1224851353fc92bfbff6f499fa97" title="Everybody&amp;#39;s favorite function.">main</a>(<span class="keywordtype">int</span> argc, <span class="keywordtype">char</span>* argv[]) {
<a name="l02490"></a>02490     <span class="keywordflow">try</span> {
<a name="l02491"></a>02491 
<a name="l02492"></a>02492         <a class="code" href="classFsTestDir.html">FsTestDir</a> testDir;
<a name="l02493"></a>02493         <span class="comment">// Full regression takes 26GB.</span>
<a name="l02494"></a>02494         int64_t byteSize = (30.0 * 1073741824);
<a name="l02495"></a>02495         testDir.<a class="code" href="classFsTestDir.html#ac14fb44eb179d1094ef5c9d127a39bbb">setTestDir</a>(<span class="stringliteral">&quot;copynumber/copynumber-cyto&quot;</span>, <span class="keyword">true</span>, byteSize ); 
<a name="l02496"></a>02496         ofstream logOut;
<a name="l02497"></a>02497         <span class="keywordtype">string</span> logName;
<a name="l02498"></a>02498         logName = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(testDir.asString(), <span class="stringliteral">&quot;test-regression-copynumber-cyto.log&quot;</span>);
<a name="l02499"></a>02499         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Log file: &quot;</span> + logName);
<a name="l02500"></a>02500         <a class="code" href="classFs.html#a1cc6e82074c84dd92bd7cdcf7d374b7c" title="Open an ofstream for writing to.">Fs::mustOpenToWrite</a>(logOut, logName.c_str());
<a name="l02501"></a>02501         <a class="code" href="classLogStream.html" title="Log messages with timestamps.">LogStream</a> log(3, &amp;logOut);
<a name="l02502"></a>02502 
<a name="l02503"></a>02503         Verbose::pushMsgHandler(&amp;log);
<a name="l02504"></a>02504         <a class="code" href="classVerbose.html#af20e052bc19f95d3a949d1daa1647fa8" title="Functions to add and remove handlers for communcation functions.">Verbose::pushProgressHandler</a>(&amp;log);
<a name="l02505"></a>02505         Verbose::pushWarnHandler(&amp;log);
<a name="l02506"></a>02506 
<a name="l02507"></a>02507         <a class="code" href="classVerbose.html#a416deece7d5098dc2e8759ef5fb134e3" title="Set the level of verbosity desired.">Verbose::setLevel</a>(3);
<a name="l02508"></a>02508         <a class="code" href="classtest__regression__copynumber__cyto.html">test_regression_copynumber_cyto</a> test;
<a name="l02509"></a>02509         test.m_testDir = testDir.asString();
<a name="l02510"></a>02510         test.parseArgv(argv);
<a name="l02511"></a>02511         test.doValgrind();
<a name="l02512"></a>02512         <span class="keywordtype">string</span> database = test.getDatabase();
<a name="l02513"></a>02513 
<a name="l02514"></a>02514         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;Execute regression test: test-regression-copynumber-cyto&quot;</span>);
<a name="l02515"></a>02515 
<a name="l02516"></a>02516         test.run();
<a name="l02517"></a>02517 
<a name="l02518"></a>02518         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1, <span class="stringliteral">&quot;NumPassed: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(test.getNumPassed()) + <span class="stringliteral">&quot; NumFailed: &quot;</span> + <a class="code" href="Convert_8h.html#a9e68e488e4da0371aea30596313537df" title="Little template function to make string conversion easy.">ToStr</a>(test.getNumFailed()) + <span class="stringliteral">&quot; for test-regression-copynumber-cyto&quot;</span>);
<a name="l02519"></a>02519         logOut.close();
<a name="l02520"></a>02520         <span class="keywordflow">return</span> test.getNumFailed() != 0;
<a name="l02521"></a>02521     }
<a name="l02522"></a>02522     <span class="keywordflow">catch</span>(...) {
<a name="l02523"></a>02523         <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(1,<span class="stringliteral">&quot;Unexpected Error: uncaught exception.&quot;</span>);
<a name="l02524"></a>02524         <span class="keywordflow">return</span> 1;
<a name="l02525"></a>02525     }
<a name="l02526"></a>02526     <span class="keywordflow">return</span> 1;
<a name="l02527"></a>02527 }
<a name="l02528"></a>02528 
<a name="l02529"></a>02529 
<a name="l02530"></a>02530 
<a name="l02531"></a>02531 
<a name="l02532"></a>02532 
<a name="l02533"></a>02533 
<a name="l02534"></a>02534 
</pre></div></div>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:38 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
