<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: translation/TranslationExperimentReport.cpp Source File</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li class="current"><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="tabs2">
    <ul class="tablist">
      <li><a href="files.html"><span>File&nbsp;List</span></a></li>
      <li><a href="globals.html"><span>File&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath">
    <ul>
      <li><a class="el" href="dir_1668076c0e919fc49032c4fb42e3bb3d.html">translation</a>      </li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>TranslationExperimentReport.cpp</h1>  </div>
</div>
<div class="contents">
<a href="TranslationExperimentReport_8cpp.html">Go to the documentation of this file.</a><div class="fragment"><pre class="fragment"><a name="l00001"></a>00001 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00002"></a>00002 <span class="comment"></span><span class="comment">//</span>
<a name="l00003"></a>00003 <span class="comment">// Copyright (C) 2008 Affymetrix, Inc.</span>
<a name="l00004"></a>00004 <span class="comment">//</span>
<a name="l00005"></a>00005 <span class="comment">// This program is free software; you can redistribute it and/or modify </span>
<a name="l00006"></a>00006 <span class="comment">// it under the terms of the GNU General Public License (version 2) as </span>
<a name="l00007"></a>00007 <span class="comment">// published by the Free Software Foundation.</span>
<a name="l00008"></a>00008 <span class="comment">// </span>
<a name="l00009"></a>00009 <span class="comment">// This program is distributed in the hope that it will be useful, </span>
<a name="l00010"></a>00010 <span class="comment">// but WITHOUT ANY WARRANTY; without even the implied warranty of </span>
<a name="l00011"></a>00011 <span class="comment">// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU </span>
<a name="l00012"></a>00012 <span class="comment">// General Public License for more details.</span>
<a name="l00013"></a>00013 <span class="comment">// </span>
<a name="l00014"></a>00014 <span class="comment">// You should have received a copy of the GNU General Public License </span>
<a name="l00015"></a>00015 <span class="comment">// along with this program;if not, write to the </span>
<a name="l00016"></a>00016 <span class="comment">// </span>
<a name="l00017"></a>00017 <span class="comment">// Free Software Foundation, Inc., </span>
<a name="l00018"></a>00018 <span class="comment">// 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA</span>
<a name="l00019"></a>00019 <span class="comment">//</span><span class="comment"></span>
<a name="l00020"></a>00020 <span class="comment">////////////////////////////////////////////////////////////////</span>
<a name="l00021"></a>00021 <span class="comment"></span><span class="comment">/**</span>
<a name="l00022"></a>00022 <span class="comment"> * @file   TranslationExperimentReport.cpp</span>
<a name="l00023"></a>00023 <span class="comment"> * @author Mybrid Spalding</span>
<a name="l00024"></a>00024 <span class="comment"> * @date   Mon Jul 14 10:02:58 PDT 2008</span>
<a name="l00025"></a>00025 <span class="comment"> * @brief  Single view object class for the three DMET3 reports </span>
<a name="l00026"></a>00026 <span class="comment"> * (comprehensive, summary and uncalled) given their similarity. </span>
<a name="l00027"></a>00027 <span class="comment"> */</span>
<a name="l00028"></a>00028 
<a name="l00029"></a>00029 <span class="preprocessor">#ifdef _MSC_VER</span>
<a name="l00030"></a>00030 <span class="preprocessor"></span><span class="preprocessor">#define _CRT_SECURE_NO_WARNINGS</span>
<a name="l00031"></a>00031 <span class="preprocessor"></span><span class="preprocessor">#endif</span>
<a name="l00032"></a>00032 <span class="preprocessor"></span>
<a name="l00033"></a>00033 <span class="comment">//</span>
<a name="l00034"></a>00034 <span class="preprocessor">#include &quot;<a class="code" href="TranslationExperimentReport_8h.html" title="Single view object class for the three DMET3 reports (comprehensive, summary and uncalled) given thei...">translation/TranslationExperimentReport.h</a>&quot;</span>
<a name="l00035"></a>00035 <span class="comment">//</span>
<a name="l00036"></a>00036 <span class="preprocessor">#include &quot;<a class="code" href="ExperimentResults_8h.html" title="Wrapper class for ExperimentGeneResults, all gene experiment results. There should ever only ever be ...">translation/ExperimentResults.h</a>&quot;</span>
<a name="l00037"></a>00037 <span class="preprocessor">#include &quot;<a class="code" href="GenotypeOverrideTableModel_8h.html" title="Class to supplant Genotype data (CHP) with user supplied data.">translation/GenotypeOverrideTableModel.h</a>&quot;</span>
<a name="l00038"></a>00038 <span class="preprocessor">#include &quot;<a class="code" href="SampleInfoTableModel_8h.html" title="A single instance class that wraps sample information either passed in by the console as a std::vecto...">translation/SampleInfoTableModel.h</a>&quot;</span>
<a name="l00039"></a>00039 <span class="preprocessor">#include &quot;<a class="code" href="TranslationTableModel_8h.html" title="Single instance data model class that wraps the translation file in the TranslationInputTsvModel.">translation/TranslationTableModel.h</a>&quot;</span>
<a name="l00040"></a>00040 <span class="comment">//</span>
<a name="l00041"></a>00041 <span class="preprocessor">#include &quot;<a class="code" href="Fs_8h.html" title="///">util/Fs.h</a>&quot;</span>
<a name="l00042"></a>00042 <span class="preprocessor">#include &quot;<a class="code" href="Util_8h.html" title="General Utilities.">util/Util.h</a>&quot;</span>
<a name="l00043"></a>00043 <span class="comment">//</span>
<a name="l00044"></a>00044 <span class="preprocessor">#include &quot;pcrecpp.h&quot;</span>
<a name="l00045"></a>00045 <span class="comment">//</span>
<a name="l00046"></a>00046 <span class="preprocessor">#include &lt;stdio.h&gt;</span>
<a name="l00047"></a>00047 
<a name="l00048"></a>00048 <span class="keyword">using namespace </span>std;
<a name="l00049"></a>00049 <span class="keyword">using namespace </span>affx;
<a name="l00050"></a>00050 
<a name="l00051"></a>00051 std::map&lt; std::string, std::string &gt; geneInterpretationCode;
<a name="l00052"></a>00052 pcrecpp::RE RE_VAR(<span class="stringliteral">&quot;Var&quot;</span>);
<a name="l00053"></a>00053 pcrecpp::RE RE_WILDCARD(<span class="stringliteral">&quot;NA|NoCall|NC|PRA|PossibleRareAllele|NotAvailable&quot;</span>);
<a name="l00054"></a>00054 
<a name="l00055"></a>00055 <span class="keyword">const</span> std::string COMPREHENSIVE_REPORT_FILE_EXT = <span class="stringliteral">&quot;_comprehensive&quot;</span>;
<a name="l00056"></a>00056 <span class="keyword">const</span> std::string SUMMARY_REPORT_FILE_EXT       = <span class="stringliteral">&quot;_summary&quot;</span>;
<a name="l00057"></a>00057 <span class="keyword">const</span> std::string UNCALLED_REPORT_FILE_EXT      = <span class="stringliteral">&quot;_uncalled&quot;</span>;
<a name="l00058"></a>00058 <span class="keyword">const</span> std::string COMPLETE_REPORT_FILE_EXT      = <span class="stringliteral">&quot;.rpt&quot;</span>;
<a name="l00059"></a>00059 <span class="keyword">const</span> std::string TEMP_REPORT_FILE_EXT          = <span class="stringliteral">&quot;.tmp&quot;</span>;
<a name="l00060"></a>00060 
<a name="l00061"></a>00061 <span class="keyword">const</span> std::string COMPREHENSIVE_TSV_COLUMNS[]   = { <span class="stringliteral">&quot;Index&quot;</span>, <span class="stringliteral">&quot;CHP Filename&quot;</span>, <span class="stringliteral">&quot;Gene&quot;</span>, <span class="stringliteral">&quot;Known Call&quot;</span>, <span class="stringliteral">&quot;Unknown Call&quot;</span>,
<a name="l00062"></a>00062     <span class="stringliteral">&quot;Interpretation Code&quot;</span>, <span class="stringliteral">&quot;Summary Flag&quot;</span>, <span class="stringliteral">&quot;Relevant Alleles&quot;</span>,
<a name="l00063"></a>00063     <span class="stringliteral">&quot;Common Name&quot;</span>, <span class="stringliteral">&quot;Probe Set ID&quot;</span>, <span class="stringliteral">&quot;Basecall&quot;</span>, <span class="stringliteral">&quot;Reference Base&quot;</span>,
<a name="l00064"></a>00064     <span class="stringliteral">&quot;Variant Base&quot;</span>,  <span class="stringliteral">&quot;Call&quot;</span>, <span class="stringliteral">&quot;Haplotype Marker&quot;</span>, <span class="stringliteral">&quot;Change for Variant&quot;</span>,
<a name="l00065"></a>00065     <span class="stringliteral">&quot;cDNA Change&quot;</span>, <span class="stringliteral">&quot;Genome Position&quot;</span>, <span class="stringliteral">&quot;dbSNP RS ID&quot;</span>, <span class="stringliteral">&quot;Original Basecall&quot;</span>,
<a name="l00066"></a>00066      <span class="stringliteral">&quot;Override Comment&quot;</span> };
<a name="l00067"></a>00067 
<a name="l00068"></a>00068 
<a name="l00069"></a>00069 <span class="keyword">const</span> std::string UNCALLED_TSV_COLUMNS[] = { <span class="stringliteral">&quot;CHP Filename&quot;</span>, <span class="stringliteral">&quot;Gene&quot;</span>, <span class="stringliteral">&quot;Common Name&quot;</span>, <span class="stringliteral">&quot;Probe Set ID&quot;</span>, <span class="stringliteral">&quot;Basecall&quot;</span>,
<a name="l00070"></a>00070                                         <span class="stringliteral">&quot;Override Comment&quot;</span>, <span class="stringliteral">&quot;Reference Allele&quot;</span>, <span class="stringliteral">&quot;Variant Allele&quot;</span>
<a name="l00071"></a>00071                                       };
<a name="l00072"></a>00072 
<a name="l00073"></a>00073 <span class="keyword">const</span> <a class="code" href="classTranslationInterpretationCodeMap.html">TranslationInterpretationCodeMap</a> INTERPRETATION_CODES[] = { {std::string(<span class="stringliteral">&quot;NoHAP&quot;</span>)}, {std::string(<span class="stringliteral">&quot;UNIQ&quot;</span>)}, {std::string(<span class="stringliteral">&quot;NC/PRA/NA&quot;</span>)}, {std::string(<span class="stringliteral">&quot;UNIQ+UNK&quot;</span>)}, {std::string(<span class="stringliteral">&quot;MULT&quot;</span>)}, {std::string(<span class="stringliteral">&quot;MULT+UNK&quot;</span>)}, {std::string(<span class="stringliteral">&quot;UNDH&quot;</span>) } };
<a name="l00074"></a>00074 
<a name="l00075"></a>00075 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00076"></a>00076 <span class="comment">/**</span>
<a name="l00077"></a>00077 <span class="comment"> * TranslationRow::dump</span>
<a name="l00078"></a>00078 <span class="comment"> * Synopsis:</span>
<a name="l00079"></a>00079 <span class="comment"> *  Debug routine only.</span>
<a name="l00080"></a>00080 <span class="comment"> */</span>
<a name="l00081"></a>00081 <span class="comment">/*****************************************************************************/</span>
<a name="l00082"></a><a class="code" href="classTranslationRow.html#a8c7fef7a2e7be12b35599381ada4d004">00082</a> <span class="keywordtype">void</span> <a class="code" href="classTranslationRow.html#a8c7fef7a2e7be12b35599381ada4d004" title="TranslationRow::dump Synopsis: Debug routine only.">TranslationRow::dump</a>() {
<a name="l00083"></a>00083   
<a name="l00084"></a>00084       cerr &lt;&lt; <span class="stringliteral">&quot;m_index:               &quot;</span> &lt;&lt; m_index &lt;&lt; endl;
<a name="l00085"></a>00085       cerr &lt;&lt; <span class="stringliteral">&quot;m_chpFileName:         &quot;</span> &lt;&lt; m_chpFileName &lt;&lt; endl; 
<a name="l00086"></a>00086       cerr &lt;&lt; <span class="stringliteral">&quot;m_gene:                &quot;</span> &lt;&lt; m_gene &lt;&lt; endl;
<a name="l00087"></a>00087       cerr &lt;&lt; <span class="stringliteral">&quot;m_knownCall:           &quot;</span> &lt;&lt; m_knownCall &lt;&lt; endl;
<a name="l00088"></a>00088       cerr &lt;&lt; <span class="stringliteral">&quot;m_unknownCall:         &quot;</span> &lt;&lt; m_unknownCall &lt;&lt; endl;
<a name="l00089"></a>00089       cerr &lt;&lt; <span class="stringliteral">&quot;m_interpretationCode:  &quot;</span> &lt;&lt; m_interpretationCode &lt;&lt; endl;
<a name="l00090"></a>00090       cerr &lt;&lt; <span class="stringliteral">&quot;m_summaryFlag :        &quot;</span> &lt;&lt; m_summaryFlag &lt;&lt; endl;
<a name="l00091"></a>00091       cerr &lt;&lt; <span class="stringliteral">&quot;m_relevantAlleles:     &quot;</span> &lt;&lt; m_relevantAlleles &lt;&lt; endl;
<a name="l00092"></a>00092       cerr &lt;&lt; <span class="stringliteral">&quot;m_markerName:          &quot;</span> &lt;&lt; m_markerName &lt;&lt; endl;
<a name="l00093"></a>00093       cerr &lt;&lt; <span class="stringliteral">&quot;m_probeSet:            &quot;</span> &lt;&lt; m_probeSet &lt;&lt; endl;
<a name="l00094"></a>00094       cerr &lt;&lt; <span class="stringliteral">&quot;m_baseCall:            &quot;</span> &lt;&lt; m_baseCall &lt;&lt; endl;
<a name="l00095"></a>00095       cerr &lt;&lt; <span class="stringliteral">&quot;m_referenceBase:       &quot;</span> &lt;&lt; m_referenceBase &lt;&lt; endl;
<a name="l00096"></a>00096       cerr &lt;&lt; <span class="stringliteral">&quot;m_referenceReportBase: &quot;</span> &lt;&lt; m_referenceReportBase &lt;&lt; endl;
<a name="l00097"></a>00097       cerr &lt;&lt; <span class="stringliteral">&quot;m_variantBase:         &quot;</span> &lt;&lt; m_variantBase &lt;&lt; endl;
<a name="l00098"></a>00098       cerr &lt;&lt; <span class="stringliteral">&quot;m_variantReportBase:   &quot;</span> &lt;&lt; m_variantReportBase &lt;&lt; endl;
<a name="l00099"></a>00099       cerr &lt;&lt; <span class="stringliteral">&quot;m_variantUncalledBase: &quot;</span> &lt;&lt; m_variantUncalledBase &lt;&lt; endl;
<a name="l00100"></a>00100       cerr &lt;&lt; <span class="stringliteral">&quot;m_call :               &quot;</span> &lt;&lt; m_call &lt;&lt; endl;
<a name="l00101"></a>00101       cerr &lt;&lt; <span class="stringliteral">&quot;m_haplotypeMarker :    &quot;</span> &lt;&lt; m_haplotypeMarker &lt;&lt; endl;
<a name="l00102"></a>00102       cerr &lt;&lt; <span class="stringliteral">&quot;m_changeForVariant :   &quot;</span> &lt;&lt; m_changeForVariant &lt;&lt; endl;
<a name="l00103"></a>00103       cerr &lt;&lt; <span class="stringliteral">&quot;m_CDNAChange :         &quot;</span> &lt;&lt; m_CDNAChange &lt;&lt; endl;
<a name="l00104"></a>00104       cerr &lt;&lt; <span class="stringliteral">&quot;m_genomicPosition :    &quot;</span> &lt;&lt; m_genomicPosition &lt;&lt; endl;
<a name="l00105"></a>00105       cerr &lt;&lt; <span class="stringliteral">&quot;m_dbSNPId :            &quot;</span> &lt;&lt; m_dbSNPId &lt;&lt; endl;
<a name="l00106"></a>00106       cerr &lt;&lt; <span class="stringliteral">&quot;m_validated :          &quot;</span> &lt;&lt; m_validated &lt;&lt; endl;
<a name="l00107"></a>00107       cerr &lt;&lt; <span class="stringliteral">&quot;m_originalBasecall :   &quot;</span> &lt;&lt; m_originalBasecall &lt;&lt; endl;
<a name="l00108"></a>00108       cerr &lt;&lt; <span class="stringliteral">&quot;m_overrideComment :    &quot;</span> &lt;&lt; m_overrideComment &lt;&lt; endl;
<a name="l00109"></a>00109 
<a name="l00110"></a>00110 }
<a name="l00111"></a>00111 <span class="comment">// end TranslationRow::dump</span>
<a name="l00112"></a>00112 <span class="comment">/*****************************************************************************/</span>
<a name="l00113"></a>00113 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00114"></a>00114 <span class="comment">/**</span>
<a name="l00115"></a>00115 <span class="comment"> * SummaryRowIndexSort::operator():</span>
<a name="l00116"></a>00116 <span class="comment"> * Synopsis:</span>
<a name="l00117"></a>00117 <span class="comment"> *  An STL class operator used to insert std::map records for the summary report in</span>
<a name="l00118"></a>00118 <span class="comment"> *  sorted order. </span>
<a name="l00119"></a>00119 <span class="comment"> *</span>
<a name="l00120"></a>00120 <span class="comment"> *  Sort by business rules.</span>
<a name="l00121"></a>00121 <span class="comment"> *  1.) Copy number designation goes at the very top under all circumstances.</span>
<a name="l00122"></a>00122 <span class="comment"> *  2.) Gene haplotype or non-haplotype markers with only references</span>
<a name="l00123"></a>00123 <span class="comment"> *      are at the bottom.</span>
<a name="l00124"></a>00124 <span class="comment"> *  3.) Gene haplotype or non-haplotype markers with variants are reported</span>
<a name="l00125"></a>00125 <span class="comment"> *      second.</span>
<a name="l00126"></a>00126 <span class="comment"> *</span>
<a name="l00127"></a>00127 <span class="comment"> *  Rows within the same group as determined by the rules above are to be</span>
<a name="l00128"></a>00128 <span class="comment"> *  sorted in gene order,  and if where genes are also equivalent</span>
<a name="l00129"></a>00129 <span class="comment"> *  then further sort by probe set id. </span>
<a name="l00130"></a>00130 <span class="comment"> * </span>
<a name="l00131"></a>00131 <span class="comment"> * @param b - The object being compared with. </span>
<a name="l00132"></a>00132 <span class="comment"> * </span>
<a name="l00133"></a>00133 <span class="comment"> * @return - true if this object is less than b. </span>
<a name="l00134"></a>00134 <span class="comment"> *</span>
<a name="l00135"></a>00135 <span class="comment"> */</span>
<a name="l00136"></a>00136 <span class="comment">/*****************************************************************************/</span>
<a name="l00137"></a><a class="code" href="classSummaryRowIndex.html#afba1886d4d01e88cf89d206e08e10569">00137</a> <span class="keywordtype">bool</span> <a class="code" href="classSummaryRowIndex.html#afba1886d4d01e88cf89d206e08e10569" title="SummaryRowIndexSort::operator(): Synopsis: An STL class operator used to insert std::map records for ...">SummaryRowIndex::operator&lt;</a>( <span class="keyword">const</span> <a class="code" href="classSummaryRowIndex.html">SummaryRowIndex</a> &amp;b)<span class="keyword"> const</span>
<a name="l00138"></a>00138 <span class="keyword"></span>{
<a name="l00139"></a>00139 
<a name="l00140"></a>00140 
<a name="l00141"></a>00141   <span class="comment">// COPY NUMBER FIRST</span>
<a name="l00142"></a>00142   <span class="keywordflow">if</span> ( (m_copyNumber &lt; 2)  &amp;&amp;  ( b.m_copyNumber &gt;= 2 ) ) {
<a name="l00143"></a>00143      <span class="keywordflow">return</span> <span class="keyword">true</span>;
<a name="l00144"></a>00144   }
<a name="l00145"></a>00145   <span class="keywordflow">else</span> <span class="keywordflow">if</span> ((m_copyNumber &gt;= 2) &amp;&amp; ( b.m_copyNumber &lt; 2) ) {
<a name="l00146"></a>00146      <span class="keywordflow">return</span> <span class="keyword">false</span>;
<a name="l00147"></a>00147   }
<a name="l00148"></a>00148   <span class="keywordflow">else</span> <span class="keywordflow">if</span> (  m_isGeneSummary || b.m_isGeneSummary ) {
<a name="l00149"></a>00149     <span class="comment">// REFERENCE LAST</span>
<a name="l00150"></a>00150     
<a name="l00151"></a>00151     <span class="keywordflow">if</span> (!( m_isGeneSummary &amp;&amp; b.m_isGeneSummary)) {
<a name="l00152"></a>00152 
<a name="l00153"></a>00153       <span class="keywordflow">if</span> (m_isGeneSummary) {
<a name="l00154"></a>00154         <span class="keywordflow">return</span> <span class="keyword">false</span>;
<a name="l00155"></a>00155       }
<a name="l00156"></a>00156       <span class="keywordflow">return</span> <span class="keyword">true</span>;
<a name="l00157"></a>00157     }
<a name="l00158"></a>00158     
<a name="l00159"></a>00159   } <span class="keywordflow">else</span> <span class="keywordflow">if</span> ( (m_knownCall.empty() || b.m_knownCall.empty()) &amp;&amp;
<a name="l00160"></a>00160               !(m_knownCall.empty() &amp;&amp; b.m_knownCall.empty())) {
<a name="l00161"></a>00161 
<a name="l00162"></a>00162     <span class="comment">// VARIANT CALLS, the filling</span>
<a name="l00163"></a>00163     <span class="keywordflow">if</span> ( m_knownCall.empty() ) {
<a name="l00164"></a>00164       <span class="keywordflow">return</span> <span class="keyword">false</span>;
<a name="l00165"></a>00165     }
<a name="l00166"></a>00166     <span class="keywordflow">return</span> <span class="keyword">true</span>;
<a name="l00167"></a>00167     
<a name="l00168"></a>00168   }
<a name="l00169"></a>00169 
<a name="l00170"></a>00170   <span class="comment">// Objects are of the same copy number, reference or variant group</span>
<a name="l00171"></a>00171   <span class="comment">// so sort by gene/probe set id.</span>
<a name="l00172"></a>00172 
<a name="l00173"></a>00173   <span class="keywordflow">if</span> ( m_gene == b.m_gene ) {
<a name="l00174"></a>00174     <span class="keywordflow">return</span>(  m_probeSet &lt; b.m_probeSet );
<a name="l00175"></a>00175   }
<a name="l00176"></a>00176 
<a name="l00177"></a>00177   <span class="keywordflow">return</span>( m_gene &lt; b.m_gene);
<a name="l00178"></a>00178 
<a name="l00179"></a>00179   
<a name="l00180"></a>00180 }
<a name="l00181"></a>00181 <span class="comment">// end SummaryRowIndex::operator&lt;</span>
<a name="l00182"></a>00182 <span class="comment">/*****************************************************************************/</span>
<a name="l00183"></a>00183 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00184"></a>00184 <span class="comment">/**</span>
<a name="l00185"></a>00185 <span class="comment"> * TranslationExperimentReport::TranslationExperimentReport:</span>
<a name="l00186"></a>00186 <span class="comment"> * Synopsis:</span>
<a name="l00187"></a>00187 <span class="comment"> *</span>
<a name="l00188"></a>00188 <span class="comment"> *  Default constructor.</span>
<a name="l00189"></a>00189 <span class="comment"> *</span>
<a name="l00190"></a>00190 <span class="comment"> * </span>
<a name="l00191"></a>00191 <span class="comment"> * @param sitm  - sample info single instance data model cached for conviencance</span>
<a name="l00192"></a>00192 <span class="comment"> * @param gotm  - genotype override single instance data model cached for convienance</span>
<a name="l00193"></a>00193 <span class="comment"> *</span>
<a name="l00194"></a>00194 <span class="comment"> */</span>
<a name="l00195"></a>00195 <span class="comment">/*****************************************************************************/</span>
<a name="l00196"></a><a class="code" href="classTranslationExperimentReport.html#a67351330f927690928cf1246ca963e83">00196</a> <a class="code" href="classTranslationExperimentReport.html#a67351330f927690928cf1246ca963e83" title="TranslationExperimentReport::TranslationExperimentReport: Synopsis:">TranslationExperimentReport::TranslationExperimentReport</a>(<a class="code" href="classSampleInfoTableModel.html">SampleInfoTableModel</a> *sitm, <a class="code" href="classGenotypeOverrideTableModel.html">GenotypeOverrideTableModel</a> *gotm) : m_gotm(gotm), m_sitm(sitm)
<a name="l00197"></a>00197 {
<a name="l00198"></a>00198 
<a name="l00199"></a>00199   m_openCalled = <span class="keyword">false</span>;
<a name="l00200"></a>00200   m_experimentFileCount = 0;
<a name="l00201"></a>00201   m_experimentCount = 0;
<a name="l00202"></a>00202 
<a name="l00203"></a>00203   m_numFixedColumns = (<span class="keyword">sizeof</span>(COMPREHENSIVE_TSV_COLUMNS) / <span class="keyword">sizeof</span>(COMPREHENSIVE_TSV_COLUMNS[0])) ;
<a name="l00204"></a>00204 
<a name="l00205"></a>00205   <span class="keywordflow">if</span> ((sitm != NULL)) {
<a name="l00206"></a>00206     m_totalColumnsWithSampleInfo = _getNumColumns() + sitm-&gt;m_rows.size();
<a name="l00207"></a>00207   } <span class="keywordflow">else</span> {
<a name="l00208"></a>00208     m_totalColumnsWithSampleInfo = 0;
<a name="l00209"></a>00209   }
<a name="l00210"></a>00210 
<a name="l00211"></a>00211 }
<a name="l00212"></a>00212 <span class="comment">// end TranslationExperimentReport::TranslationExperimentReport</span>
<a name="l00213"></a>00213 <span class="comment">/*****************************************************************************/</span>
<a name="l00214"></a>00214 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00215"></a>00215 <span class="comment">/**</span>
<a name="l00216"></a>00216 <span class="comment"> * TranslationExperimentReport::close:</span>
<a name="l00217"></a>00217 <span class="comment"> * Synopsis:</span>
<a name="l00218"></a>00218 <span class="comment"> *</span>
<a name="l00219"></a>00219 <span class="comment"> *   Close the report files and set the open state.</span>
<a name="l00220"></a>00220 <span class="comment"> *</span>
<a name="l00221"></a>00221 <span class="comment"> * </span>
<a name="l00222"></a>00222 <span class="comment"> * @param abort - if the close is happening as a resulf of an abort exception then do not message via Verbose due to a bug in Windows. </span>
<a name="l00223"></a>00223 <span class="comment"> *</span>
<a name="l00224"></a>00224 <span class="comment"> */</span>
<a name="l00225"></a>00225 <span class="comment">/*****************************************************************************/</span>
<a name="l00226"></a><a class="code" href="classTranslationExperimentReport.html#a9b179327af3a67d9692e558f612e7fde">00226</a> <span class="keywordtype">bool</span> <a class="code" href="classTranslationExperimentReport.html#a9b179327af3a67d9692e558f612e7fde" title="TranslationExperimentReport::close: Synopsis:">TranslationExperimentReport::close</a>(<span class="keywordtype">bool</span> abort)
<a name="l00227"></a>00227 {
<a name="l00228"></a>00228 
<a name="l00229"></a>00229   <span class="keywordflow">if</span> (! abort) {
<a name="l00230"></a>00230     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(ADT_VERBOSE_NORMAL, <span class="stringliteral">&quot;TranslationExperimentReport::generate called: &quot;</span>);
<a name="l00231"></a>00231     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(ADT_VERBOSE_EXCEPTION, <span class="stringliteral">&quot;dmet3-report-guid=&quot;</span> + m_guid);
<a name="l00232"></a>00232     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(ADT_VERBOSE_NORMAL,  m_comprehensiveReportName + COMPLETE_REPORT_FILE_EXT);
<a name="l00233"></a>00233     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(ADT_VERBOSE_NORMAL,  m_summaryReportName + COMPLETE_REPORT_FILE_EXT);
<a name="l00234"></a>00234     <a class="code" href="classVerbose.html#ac4034f68f4c8d2b49cd6340984b940ce" title="Print a message to the stream.">Verbose::out</a>(ADT_VERBOSE_NORMAL,  m_uncalledReportName + COMPLETE_REPORT_FILE_EXT);
<a name="l00235"></a>00235   }
<a name="l00236"></a>00236 
<a name="l00237"></a>00237   <span class="keywordtype">bool</span> okClose = <span class="keyword">true</span>;
<a name="l00238"></a>00238   <span class="keywordflow">if</span> (m_openCalled) {
<a name="l00239"></a>00239     okClose = _closeComprehensiveFile(abort) &amp;&amp; okClose;
<a name="l00240"></a>00240     okClose = _closeSummaryFile(abort) &amp;&amp; okClose;
<a name="l00241"></a>00241     okClose = _closeUncalledFile(abort) &amp;&amp; okClose;
<a name="l00242"></a>00242     m_openCalled = <span class="keyword">false</span>;
<a name="l00243"></a>00243     m_experimentFileCount = 0;
<a name="l00244"></a>00244     m_experimentCount = 0;
<a name="l00245"></a>00245     m_guid.clear();
<a name="l00246"></a>00246   }
<a name="l00247"></a>00247 
<a name="l00248"></a>00248   <span class="keywordflow">return</span> okClose;
<a name="l00249"></a>00249 }
<a name="l00250"></a>00250 <span class="comment">// end TranslationExperimentReport::close</span>
<a name="l00251"></a>00251 <span class="comment">/*****************************************************************************/</span>
<a name="l00252"></a>00252 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00253"></a>00253 <span class="comment">/**</span>
<a name="l00254"></a>00254 <span class="comment"> * _generateOriginalBaseCall</span>
<a name="l00255"></a>00255 <span class="comment"> * Synopsis:</span>
<a name="l00256"></a>00256 <span class="comment"> *</span>
<a name="l00257"></a>00257 <span class="comment"> *   Helper function to generate that take the report alleles stored</span>
<a name="l00258"></a>00258 <span class="comment"> * in the translation table and applies them to the original basecall as</span>
<a name="l00259"></a>00259 <span class="comment"> * input by the override file. </span>
<a name="l00260"></a>00260 <span class="comment"> *</span>
<a name="l00261"></a>00261 <span class="comment"> * </span>
<a name="l00262"></a>00262 <span class="comment"> * @param probeSet - The unique probe set in the translation table.</span>
<a name="l00263"></a>00263 <span class="comment"> * @param originalBasecall - the original basecall to get the report allele for</span>
<a name="l00264"></a>00264 <span class="comment"> * @param ttm - the single instance TranslationTableModel</span>
<a name="l00265"></a>00265 <span class="comment"> *</span>
<a name="l00266"></a>00266 <span class="comment"> * @return - the input originalBasecall or a translated originalBasecall if</span>
<a name="l00267"></a>00267 <span class="comment"> * report alleles exist. </span>
<a name="l00268"></a>00268 <span class="comment"> */</span>
<a name="l00269"></a>00269 <span class="comment">/*****************************************************************************/</span>
<a name="l00270"></a>00270 <span class="keyword">static</span> std::string _generateOriginalBasecall(<span class="keyword">const</span> std::string probeSet, std::string originalBasecall, <a class="code" href="classTranslationTableModel.html">TranslationTableModel</a> &amp; ttm  ) {
<a name="l00271"></a>00271 
<a name="l00272"></a>00272   std::string first, second, translatedBasecall;
<a name="l00273"></a>00273 
<a name="l00274"></a>00274   <span class="keywordflow">if</span> ( originalBasecall.empty() ) {
<a name="l00275"></a>00275     <span class="keywordflow">return</span> originalBasecall;
<a name="l00276"></a>00276   }
<a name="l00277"></a>00277 
<a name="l00278"></a>00278   <span class="keywordflow">if</span> (!pcrecpp::RE(<span class="stringliteral">&quot;^([^/]+)/?$&quot;</span>).FullMatch(originalBasecall, &amp;first)) {
<a name="l00279"></a>00279 
<a name="l00280"></a>00280     <span class="keywordflow">if</span> (! pcrecpp::RE(<span class="stringliteral">&quot;^([^/]+)/([^/]+)$&quot;</span>).FullMatch(originalBasecall, &amp;first, &amp;second)) {
<a name="l00281"></a>00281       <a class="code" href="Err_8h.html#a7e26a673d8901b5bc880dfab285f5309" title="Calls Err::apt_err_abort with the filename and linenumber set.">APT_ERR_ABORT</a>(probeSet + <span class="stringliteral">&quot;: unknown call? &quot;</span> + originalBasecall);
<a name="l00282"></a>00282     }
<a name="l00283"></a>00283   }
<a name="l00284"></a>00284 
<a name="l00285"></a>00285   
<a name="l00286"></a>00286   <span class="keywordflow">if</span> ( second.empty() ) {
<a name="l00287"></a>00287     <span class="keywordflow">if</span> ( first == <span class="stringliteral">&quot;0&quot;</span> ) {
<a name="l00288"></a>00288       translatedBasecall = <span class="stringliteral">&quot;ZeroCopyNumber&quot;</span>;
<a name="l00289"></a>00289     }
<a name="l00290"></a>00290     <span class="keywordflow">else</span> {
<a name="l00291"></a>00291       translatedBasecall = ttm.<a class="code" href="classTranslationTableModel.html#a0d5cf32bb7c3a56543767bf15b51834b" title="TranslationTableModel::getReportAllele Synopsis:">getReportAllele</a>( probeSet, first );
<a name="l00292"></a>00292     }
<a name="l00293"></a>00293   }
<a name="l00294"></a>00294   <span class="keywordflow">else</span> {
<a name="l00295"></a>00295     translatedBasecall = ttm.<a class="code" href="classTranslationTableModel.html#a0d5cf32bb7c3a56543767bf15b51834b" title="TranslationTableModel::getReportAllele Synopsis:">getReportAllele</a>( probeSet, first ) + <span class="charliteral">&#39;/&#39;</span> +
<a name="l00296"></a>00296       ttm.<a class="code" href="classTranslationTableModel.html#a0d5cf32bb7c3a56543767bf15b51834b" title="TranslationTableModel::getReportAllele Synopsis:">getReportAllele</a>( probeSet, second );
<a name="l00297"></a>00297   }
<a name="l00298"></a>00298 
<a name="l00299"></a>00299   <span class="keywordflow">return</span> translatedBasecall;
<a name="l00300"></a>00300   
<a name="l00301"></a>00301 }
<a name="l00302"></a>00302 <span class="comment">// end _generateOriginalBaseCall</span>
<a name="l00303"></a>00303 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00304"></a>00304 <span class="comment">/**</span>
<a name="l00305"></a>00305 <span class="comment"> * TranslationExperimentReport::generate</span>
<a name="l00306"></a>00306 <span class="comment"> * Synopsis:</span>
<a name="l00307"></a>00307 <span class="comment"> *</span>
<a name="l00308"></a>00308 <span class="comment"> *   Creates the following reports in tandem due to the similarity in</span>
<a name="l00309"></a>00309 <span class="comment"> * in requirements:</span>
<a name="l00310"></a>00310 <span class="comment"> * 1.) Comprehensive</span>
<a name="l00311"></a>00311 <span class="comment"> * 2.) Summary</span>
<a name="l00312"></a>00312 <span class="comment"> * 3.) Uncalled</span>
<a name="l00313"></a>00313 <span class="comment"> * </span>
<a name="l00314"></a>00314 <span class="comment"> * Reporting is accomplished on a per experiment basis in order</span>
<a name="l00315"></a>00315 <span class="comment"> * to avoid using excessive memory. There is no reporting done</span>
<a name="l00316"></a>00316 <span class="comment"> * across experiments. All data collected during the generate</span>
<a name="l00317"></a>00317 <span class="comment"> * process is expunged before the next call to generate. </span>
<a name="l00318"></a>00318 <span class="comment"> * </span>
<a name="l00319"></a>00319 <span class="comment"> * The translation table (file) contains annotation data that is</span>
<a name="l00320"></a>00320 <span class="comment"> * not used during process and hence not found in the ExperimentResults</span>
<a name="l00321"></a>00321 <span class="comment"> * object passed in. Therefore the translation table is passed in so</span>
<a name="l00322"></a>00322 <span class="comment"> * that the annotation data can be &quot;passed through&quot; to the reports.</span>
<a name="l00323"></a>00323 <span class="comment"> * </span>
<a name="l00324"></a>00324 <span class="comment"> * Some pass through data from the translation table to the the reports</span>
<a name="l00325"></a>00325 <span class="comment"> * does in fact have some processng or business logic swirling the data. The</span>
<a name="l00326"></a>00326 <span class="comment"> * relevant alleles are one such example.</span>
<a name="l00327"></a>00327 <span class="comment"> *</span>
<a name="l00328"></a>00328 <span class="comment"> * </span>
<a name="l00329"></a>00329 <span class="comment"> * @param rte - the single instance run time environment</span>
<a name="l00330"></a>00330 <span class="comment"> * @param ttm - the single instance translation table with pass through data.</span>
<a name="l00331"></a>00331 <span class="comment"> * @param er  - the ExperimentResuls container of ExperimentGeneResults for per gene results. </span>
<a name="l00332"></a>00332 <span class="comment"> * </span>
<a name="l00333"></a>00333 <span class="comment"> * @return true - the report was generated without error. </span>
<a name="l00334"></a>00334 <span class="comment"> *</span>
<a name="l00335"></a>00335 <span class="comment"> */</span>
<a name="l00336"></a>00336 <span class="comment">/*****************************************************************************/</span>
<a name="l00337"></a><a class="code" href="classTranslationExperimentReport.html#a940186e996b278ea65194cfb5bbf7f01">00337</a> <span class="keywordtype">bool</span> <a class="code" href="classTranslationExperimentReport.html#a940186e996b278ea65194cfb5bbf7f01" title="TranslationExperimentReport::generate Synopsis:">TranslationExperimentReport::generate</a>(<a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp; rte,
<a name="l00338"></a>00338     <a class="code" href="classTranslationTableModel.html">TranslationTableModel</a> &amp; ttm,
<a name="l00339"></a>00339     std::map&lt;std::string, ExperimentResults *&gt;&amp;er)
<a name="l00340"></a>00340 {
<a name="l00341"></a>00341 
<a name="l00342"></a>00342   <span class="keywordtype">bool</span> okReport = <span class="keyword">true</span>;
<a name="l00343"></a>00343 
<a name="l00344"></a>00344   pcrecpp::RE reUnknown(<span class="stringliteral">&quot;UNK&quot;</span>);
<a name="l00345"></a>00345   <span class="keywordflow">if</span> (! m_openCalled) {
<a name="l00346"></a>00346     _open(rte);
<a name="l00347"></a>00347   }
<a name="l00348"></a>00348 
<a name="l00349"></a>00349   std::map&lt; std::string, GeneHaplotypeCallMapElement&gt; ghcm;
<a name="l00350"></a>00350   std::map&lt; std::string, int &gt; markerIndex;
<a name="l00351"></a>00351   std::map&lt; std::string, bool &gt; geneSummaryWritten;
<a name="l00352"></a>00352 
<a name="l00353"></a>00353   _generateGeneHaplotypeCallMap(rte, er, ghcm);
<a name="l00354"></a>00354 
<a name="l00355"></a>00355   <span class="keywordtype">int</span> count = 0;
<a name="l00356"></a>00356 
<a name="l00357"></a>00357   std::map&lt;std::string, ExperimentResults*&gt;::const_iterator itSER;
<a name="l00358"></a>00358   std::vector&lt; std::string &gt; sampleInfo;
<a name="l00359"></a>00359 
<a name="l00360"></a>00360   <span class="comment">// For each experiment, recapitulate the translation table.</span>
<a name="l00361"></a>00361 
<a name="l00362"></a>00362 
<a name="l00363"></a>00363 
<a name="l00364"></a>00364   <span class="keywordflow">for</span> (itSER = er.begin(), geneInterpretationCode.clear();
<a name="l00365"></a>00365        (itSER != er.end()) &amp;&amp; okReport ;  itSER++) {
<a name="l00366"></a>00366 
<a name="l00367"></a>00367     m_experimentCount++;
<a name="l00368"></a>00368 
<a name="l00369"></a>00369     <span class="comment">// IGNORE Experiments with no marker calls for DMET2.</span>
<a name="l00370"></a>00370     <span class="keywordflow">if</span> (rte.m_adtOpts.m_dmet2Calling &amp;&amp; (itSER-&gt;second-&gt;m_markerCallCount == 0)) {
<a name="l00371"></a>00371       <span class="keywordflow">continue</span>;
<a name="l00372"></a>00372     }
<a name="l00373"></a>00373 
<a name="l00374"></a>00374     <span class="keywordflow">if</span> (m_sitm != NULL) {
<a name="l00375"></a>00375       sampleInfo = m_sitm-&gt;<a class="code" href="classSampleInfoTableModel.html#aff9b946d286d870f19dd8dc3a88fd022" title="SampleInfoTableModel::getSampleInfo Synopsis:">getSampleInfo</a>(itSER-&gt;second-&gt;m_experimentGuid.empty() ? itSER-&gt;second-&gt;m_experiment : itSER-&gt;second-&gt;m_experimentGuid);
<a name="l00376"></a>00376     }
<a name="l00377"></a>00377 
<a name="l00378"></a>00378 
<a name="l00379"></a>00379     std::vector&lt;TranslationRow&gt; cRows;
<a name="l00380"></a>00380     std::map&lt;std::string, int&gt; probeSetRow;
<a name="l00381"></a>00381     <span class="keywordtype">int</span> geneCount = 0;
<a name="l00382"></a>00382     <span class="keywordtype">int</span> markerCount = 0;
<a name="l00383"></a>00383     std::string prevGene;
<a name="l00384"></a>00384     <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 1; i &lt; ttm.size(); i++, count++) {
<a name="l00385"></a>00385 
<a name="l00386"></a>00386       <span class="comment">// Skip commented out rows as well as header rows.</span>
<a name="l00387"></a>00387       <span class="keywordflow">if</span> (ttm.<a class="code" href="classTranslationTableModel.html#a590375d07f5f5bf813275858d0466114" title="TranslationTableModel::ignoreRow Synopsis: A setter and getter for ignoring rows in the translation t...">ignoreRow</a>(i) || ( !((ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_HAPLOTYPE)] == <span class="stringliteral">&quot;Y&quot;</span>) || (ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_HAPLOTYPE)] == <span class="stringliteral">&quot;N&quot;</span>)))) {
<a name="l00388"></a>00388         <span class="keywordflow">continue</span>;
<a name="l00389"></a>00389       }
<a name="l00390"></a>00390 
<a name="l00391"></a>00391 
<a name="l00392"></a>00392       <span class="comment">// Initialize</span>
<a name="l00393"></a>00393 
<a name="l00394"></a>00394       std::string gene = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_GENE)];
<a name="l00395"></a>00395 
<a name="l00396"></a>00396       <span class="keywordflow">if</span> (prevGene.empty() || (prevGene != gene)) {
<a name="l00397"></a>00397         geneCount++;
<a name="l00398"></a>00398         prevGene = gene;
<a name="l00399"></a>00399         markerCount = 0;
<a name="l00400"></a>00400         geneSummaryWritten[gene] = <span class="keyword">false</span>;
<a name="l00401"></a>00401       }
<a name="l00402"></a>00402 
<a name="l00403"></a>00403       <a class="code" href="classExperimentGeneResults.html">ExperimentGeneResults</a> *egr = NULL;
<a name="l00404"></a>00404 
<a name="l00405"></a>00405       <span class="keywordflow">if</span> (itSER-&gt;second-&gt;m_geneResults.find(gene) != itSER-&gt;second-&gt;m_geneResults.end()) {
<a name="l00406"></a>00406         egr = itSER-&gt;second-&gt;m_geneResults[gene];
<a name="l00407"></a>00407       }
<a name="l00408"></a>00408 
<a name="l00409"></a>00409 
<a name="l00410"></a>00410       <a class="code" href="classTranslationRow.html">TranslationRow</a> cRowNew;
<a name="l00411"></a>00411       <a class="code" href="classTranslationRow.html">TranslationRow</a> * cRow = &amp;cRowNew;
<a name="l00412"></a>00412 
<a name="l00413"></a>00413 
<a name="l00414"></a>00414       std::string probeSet = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_PROBE_SET_ID)];
<a name="l00415"></a>00415 
<a name="l00416"></a>00416       <span class="comment">// Multiallelic Probe Set?</span>
<a name="l00417"></a>00417       <span class="keywordtype">bool</span> isMultiAllelicRow = probeSetRow.count(probeSet) &gt; 0;
<a name="l00418"></a>00418       <span class="keywordflow">if</span> (isMultiAllelicRow) {
<a name="l00419"></a>00419         <a class="code" href="Err_8h.html#ad024208f4ba6a4e75df4499ea1639a05" title="Calls Err::apt_err_assert with the filename and linenumber set. /// We want to avoid evaluating the m...">APT_ERR_ASSERT</a>(probeSetRow[probeSet] &lt; cRows.size(), <span class="stringliteral">&quot;&quot;</span>);
<a name="l00420"></a>00420         cRow = &amp;(cRows[probeSetRow[probeSet]]);
<a name="l00421"></a>00421         <a class="code" href="Err_8h.html#ad024208f4ba6a4e75df4499ea1639a05" title="Calls Err::apt_err_assert with the filename and linenumber set. /// We want to avoid evaluating the m...">APT_ERR_ASSERT</a>(probeSet == cRow-&gt;m_probeSet, <span class="stringliteral">&quot;&quot;</span>);
<a name="l00422"></a>00422       } <span class="keywordflow">else</span> {
<a name="l00423"></a>00423         markerCount++;
<a name="l00424"></a>00424         probeSetRow[probeSet] = cRows.size();
<a name="l00425"></a>00425         markerIndex[probeSet] = markerCount;
<a name="l00426"></a>00426         cRow-&gt;m_isSummaryRow = <span class="keyword">false</span>;
<a name="l00427"></a>00427         cRow-&gt;m_isGeneSummary = <span class="keyword">false</span>;
<a name="l00428"></a>00428       }
<a name="l00429"></a>00429 
<a name="l00430"></a>00430       <span class="keywordtype">bool</span> isProbeSetPresent = <span class="keyword">false</span>;
<a name="l00431"></a>00431       <span class="keywordflow">if</span> (egr != NULL) {
<a name="l00432"></a>00432         isProbeSetPresent
<a name="l00433"></a>00433         = egr-&gt;m_probeSetToMarkerCallResults.count(probeSet) &gt; 0;
<a name="l00434"></a>00434       }
<a name="l00435"></a>00435 
<a name="l00436"></a>00436       <span class="keywordtype">int</span> markerCallResultsIndex = 0;
<a name="l00437"></a>00437       <span class="keywordtype">bool</span> hasAlleleCalls = <span class="keyword">false</span>;
<a name="l00438"></a>00438       <span class="keywordflow">if</span> (isProbeSetPresent &amp;&amp; (egr != NULL)) {
<a name="l00439"></a>00439         markerCallResultsIndex = egr-&gt;m_probeSetToMarkerCallResults[probeSet];
<a name="l00440"></a>00440         hasAlleleCalls =  egr-&gt;m_callResults[markerCallResultsIndex].m_alleleCalls.size() &gt; 0;
<a name="l00441"></a>00441       }
<a name="l00442"></a>00442 
<a name="l00443"></a>00443       <span class="comment">// Report dat</span>
<a name="l00444"></a>00444 
<a name="l00445"></a>00445       <span class="comment">// Index</span>
<a name="l00446"></a>00446 
<a name="l00447"></a>00447       <span class="keywordtype">char</span> buf[BUFSIZ];
<a name="l00448"></a>00448       sprintf(buf, <span class="stringliteral">&quot;%4.4d-%4.4d-%2.2d&quot;</span>, m_experimentCount, geneCount, markerIndex[probeSet]);
<a name="l00449"></a>00449 
<a name="l00450"></a>00450       cRow-&gt;m_index       = buf;
<a name="l00451"></a>00451 
<a name="l00452"></a>00452       <span class="comment">// CHP file name</span>
<a name="l00453"></a>00453       cRow-&gt;m_chpFileName = <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(itSER-&gt;second-&gt;m_experiment);
<a name="l00454"></a>00454 
<a name="l00455"></a>00455       <span class="comment">// Gene</span>
<a name="l00456"></a>00456       cRow-&gt;m_gene       = gene;
<a name="l00457"></a>00457       cRow-&gt;m_geneCopyNumberIndicator = ttm.m_geneCopyNumberIndicator[gene];
<a name="l00458"></a>00458 
<a name="l00459"></a>00459       <span class="comment">// Haplotype Marker</span>
<a name="l00460"></a>00460 
<a name="l00461"></a>00461       cRow-&gt;m_haplotypeMarker = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_HAPLOTYPE)];
<a name="l00462"></a>00462 
<a name="l00463"></a>00463       <span class="comment">//bool isHaplotypeMarker = cRow-&gt;m_haplotypeMarker == &quot;Y&quot;;</span>
<a name="l00464"></a>00464 
<a name="l00465"></a>00465       <span class="comment">// Known Calls, Unknown Calls</span>
<a name="l00466"></a>00466       std::map&lt;std::string, GeneHaplotypeCallMapElement&gt;::iterator itGHCM;
<a name="l00467"></a>00467       itGHCM = ghcm.find(gene);
<a name="l00468"></a>00468 
<a name="l00469"></a>00469       <span class="keywordtype">int</span> numKnownCalls = 0;
<a name="l00470"></a>00470       <span class="keywordtype">int</span> numUnknownCalls = 0;
<a name="l00471"></a>00471       <span class="keywordtype">int</span> geneCopyNumber = 2;
<a name="l00472"></a>00472 
<a name="l00473"></a>00473       <span class="keywordflow">if</span> ( itGHCM != ghcm.end() ) {
<a name="l00474"></a>00474         geneCopyNumber = itGHCM-&gt;second.m_copyNumber;
<a name="l00475"></a>00475       }
<a name="l00476"></a>00476       
<a name="l00477"></a>00477       <span class="keywordflow">if</span> (!isMultiAllelicRow &amp;&amp; (itGHCM != ghcm.end())) {
<a name="l00478"></a>00478         numKnownCalls = itGHCM-&gt;second.m_knownCall.size();
<a name="l00479"></a>00479         <span class="keywordflow">for</span> (<span class="keywordtype">size_t</span> i = 0; i &lt;  itGHCM-&gt;second.m_knownCall.size(); i++) {
<a name="l00480"></a>00480           <span class="keywordflow">if</span> (cRow-&gt;m_knownCall.empty()) {
<a name="l00481"></a>00481             cRow-&gt;m_knownCall = itGHCM-&gt;second.m_knownCall[i];
<a name="l00482"></a>00482           } <span class="keywordflow">else</span> {
<a name="l00483"></a>00483             cRow-&gt;m_knownCall = cRow-&gt;m_knownCall  + <span class="stringliteral">&quot;,&quot;</span> + itGHCM-&gt;second.m_knownCall[i];
<a name="l00484"></a>00484           }
<a name="l00485"></a>00485         }
<a name="l00486"></a>00486         numUnknownCalls = itGHCM-&gt;second.m_unknownCall.size();
<a name="l00487"></a>00487         <span class="keywordflow">for</span> (<span class="keywordtype">size_t</span> i = 0; i &lt;  itGHCM-&gt;second.m_unknownCall.size(); i++) {
<a name="l00488"></a>00488           <span class="keywordflow">if</span> (cRow-&gt;m_unknownCall.empty()) {
<a name="l00489"></a>00489             cRow-&gt;m_unknownCall = itGHCM-&gt;second.m_unknownCall[i];
<a name="l00490"></a>00490           } <span class="keywordflow">else</span> {
<a name="l00491"></a>00491             cRow-&gt;m_unknownCall = cRow-&gt;m_unknownCall  + <span class="stringliteral">&quot;,&quot;</span> + itGHCM-&gt;second.m_unknownCall[i];
<a name="l00492"></a>00492           }
<a name="l00493"></a>00493         }
<a name="l00494"></a>00494       }
<a name="l00495"></a>00495 
<a name="l00496"></a>00496       <span class="comment">// Summary Flag</span>
<a name="l00497"></a>00497       <span class="keywordflow">if</span> (isMultiAllelicRow) {
<a name="l00498"></a>00498         <span class="keywordflow">if</span> (cRow-&gt;m_summaryFlag.empty()) {
<a name="l00499"></a>00499           cRow-&gt;m_summaryFlag = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_DEFINING)];
<a name="l00500"></a>00500         } <span class="keywordflow">else</span>  <span class="keywordflow">if</span> (!ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_DEFINING)].empty() &amp;&amp; (cRow-&gt;m_summaryFlag != ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_DEFINING)])) {
<a name="l00501"></a>00501           cRow-&gt;m_summaryFlag = cRow-&gt;m_summaryFlag + <span class="stringliteral">&quot;,&quot;</span> + ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_DEFINING)];
<a name="l00502"></a>00502         }
<a name="l00503"></a>00503       } <span class="keywordflow">else</span> {
<a name="l00504"></a>00504         cRow-&gt;m_summaryFlag = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_DEFINING)];
<a name="l00505"></a>00505 
<a name="l00506"></a>00506       }
<a name="l00507"></a>00507 
<a name="l00508"></a>00508       <span class="comment">// Relevant Alelles</span>
<a name="l00509"></a>00509       _generateRelevantAlleles(ttm, cRow, i, isMultiAllelicRow);
<a name="l00510"></a>00510 
<a name="l00511"></a>00511       <span class="comment">// Marker Name (External Id)</span>
<a name="l00512"></a>00512       cRow-&gt;m_markerName = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_COMMON_NAME)];
<a name="l00513"></a>00513 
<a name="l00514"></a>00514       <span class="comment">// Probe Set</span>
<a name="l00515"></a>00515       cRow-&gt;m_probeSet = probeSet;
<a name="l00516"></a>00516 
<a name="l00517"></a>00517       <span class="comment">// Reference Base</span>
<a name="l00518"></a>00518       _generateReferenceBase( i, cRow, ttm);
<a name="l00519"></a>00519 
<a name="l00520"></a>00520       <span class="comment">// Variant Base</span>
<a name="l00521"></a>00521       _generateVariantBase( i, cRow, ttm, isMultiAllelicRow);
<a name="l00522"></a>00522 
<a name="l00523"></a>00523       <span class="comment">// Call</span>
<a name="l00524"></a>00524       cRow-&gt;m_call = <span class="stringliteral">&quot;&quot;</span>;
<a name="l00525"></a>00525       <span class="keywordflow">if</span> (isProbeSetPresent &amp;&amp; (egr != NULL) &amp;&amp; (egr-&gt;m_callResults[markerCallResultsIndex].m_alleleCalls.size() &gt; 0)) {
<a name="l00526"></a>00526         <a class="code" href="Err_8h.html#ad024208f4ba6a4e75df4499ea1639a05" title="Calls Err::apt_err_assert with the filename and linenumber set. /// We want to avoid evaluating the m...">APT_ERR_ASSERT</a>(markerCallResultsIndex &lt; egr-&gt;m_callResults.size(), <span class="stringliteral">&quot;&quot;</span>);
<a name="l00527"></a>00527         cRow-&gt;m_call = egr-&gt;m_callResults[markerCallResultsIndex].m_alleleCalls[0].m_allele;
<a name="l00528"></a>00528 
<a name="l00529"></a>00529       }
<a name="l00530"></a>00530 
<a name="l00531"></a>00531       <span class="comment">// Base Call</span>
<a name="l00532"></a>00532       _generateBaseCall(egr, i, markerCallResultsIndex, cRow, ttm, isMultiAllelicRow, isProbeSetPresent);
<a name="l00533"></a>00533 
<a name="l00534"></a>00534 
<a name="l00535"></a>00535       <span class="comment">/* HACK: m_copyNumber is set in _generateBaseCall */</span>
<a name="l00536"></a>00536       <span class="keywordflow">if</span> (( egr != NULL) &amp;&amp; (cRow-&gt;m_copyNumber == 0) &amp;&amp;
<a name="l00537"></a>00537           (egr-&gt;m_callResults[markerCallResultsIndex].m_alleleCalls.size() &gt; 0)) {
<a name="l00538"></a>00538         cRow-&gt;m_call = egr-&gt;m_callResults[markerCallResultsIndex].m_alleleCalls[0].m_allele;
<a name="l00539"></a>00539       }
<a name="l00540"></a>00540       <span class="keywordflow">else</span> <span class="keywordflow">if</span> (cRow-&gt;m_baseCall == <span class="stringliteral">&quot;NotAvailable&quot;</span>) {
<a name="l00541"></a>00541         cRow-&gt;m_call = <span class="stringliteral">&quot;&quot;</span>;
<a name="l00542"></a>00542       }
<a name="l00543"></a>00543 
<a name="l00544"></a>00544       <span class="comment">// Interpretation Code</span>
<a name="l00545"></a>00545 
<a name="l00546"></a>00546       <span class="keywordflow">if</span> ((egr != NULL) &amp;&amp; geneInterpretationCode.count(gene) == 0) {
<a name="l00547"></a>00547         _generateInterpretationCode(*egr, ttm, cRow, i, numKnownCalls, numUnknownCalls, geneInterpretationCode, geneCopyNumber);
<a name="l00548"></a>00548       }
<a name="l00549"></a>00549 
<a name="l00550"></a>00550       <span class="keywordflow">if</span> ( geneInterpretationCode.count(gene ) &gt; 0 ) {
<a name="l00551"></a>00551         cRow-&gt;m_interpretationCode = geneInterpretationCode[gene];
<a name="l00552"></a>00552       }
<a name="l00553"></a>00553       <span class="keywordflow">else</span> {
<a name="l00554"></a>00554         cRow-&gt;m_interpretationCode = <span class="stringliteral">&quot;NC/PRA/NA&quot;</span>;
<a name="l00555"></a>00555       }
<a name="l00556"></a>00556 
<a name="l00557"></a>00557       <span class="keywordflow">if</span> (isMultiAllelicRow) {
<a name="l00558"></a>00558         <span class="keywordflow">if</span> (cRow-&gt;m_changeForVariant.empty()) {
<a name="l00559"></a>00559           cRow-&gt;m_changeForVariant = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_CHANGE)];
<a name="l00560"></a>00560         } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (! ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_CHANGE)].empty() &amp;&amp; (cRow-&gt;m_changeForVariant != ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_CHANGE)])) {
<a name="l00561"></a>00561 
<a name="l00562"></a>00562           cRow-&gt;m_changeForVariant = cRow-&gt;m_changeForVariant + <span class="stringliteral">&quot;,&quot;</span> + ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_CHANGE)];
<a name="l00563"></a>00563         }
<a name="l00564"></a>00564 
<a name="l00565"></a>00565       } <span class="keywordflow">else</span> {
<a name="l00566"></a>00566         cRow-&gt;m_changeForVariant = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_CHANGE)];
<a name="l00567"></a>00567       }
<a name="l00568"></a>00568 
<a name="l00569"></a>00569       <span class="comment">// CDNA Change</span>
<a name="l00570"></a>00570       <span class="keywordflow">if</span> (isMultiAllelicRow) {
<a name="l00571"></a>00571         <span class="keywordflow">if</span> (cRow-&gt;m_CDNAChange.empty()) {
<a name="l00572"></a>00572           cRow-&gt;m_CDNAChange = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_CDNA)];
<a name="l00573"></a>00573         } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (!ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_CDNA)].empty()) {
<a name="l00574"></a>00574           cRow-&gt;m_CDNAChange = cRow-&gt;m_CDNAChange + <span class="stringliteral">&quot;,&quot;</span> + ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_CDNA)];
<a name="l00575"></a>00575         }
<a name="l00576"></a>00576       } <span class="keywordflow">else</span>  {
<a name="l00577"></a>00577         cRow-&gt;m_CDNAChange = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_CDNA)];
<a name="l00578"></a>00578       }
<a name="l00579"></a>00579 
<a name="l00580"></a>00580       <span class="comment">// Genomic Position</span>
<a name="l00581"></a>00581       <span class="keywordflow">if</span> (isMultiAllelicRow) {
<a name="l00582"></a>00582         <span class="keywordflow">if</span> (cRow-&gt;m_genomicPosition.empty()) {
<a name="l00583"></a>00583           cRow-&gt;m_genomicPosition = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_GENOME_POSITION)];
<a name="l00584"></a>00584         } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (!ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_GENOME_POSITION)].empty()) {
<a name="l00585"></a>00585           cRow-&gt;m_genomicPosition = cRow-&gt;m_genomicPosition + <span class="stringliteral">&quot;,&quot;</span> + ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_GENOME_POSITION)];
<a name="l00586"></a>00586         }
<a name="l00587"></a>00587       } <span class="keywordflow">else</span> {
<a name="l00588"></a>00588         cRow-&gt;m_genomicPosition = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_GENOME_POSITION)];
<a name="l00589"></a>00589       }
<a name="l00590"></a>00590 
<a name="l00591"></a>00591       <span class="comment">// dbSNP Id</span>
<a name="l00592"></a>00592       cRow-&gt;m_dbSNPId = ttm.m_rows[i][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_DBSNP)];
<a name="l00593"></a>00593 
<a name="l00594"></a>00594       <span class="comment">// Override</span>
<a name="l00595"></a>00595       <span class="comment">// Check egr</span>
<a name="l00596"></a>00596       <span class="keywordflow">if</span> ((m_gotm != NULL) &amp;&amp; (egr != NULL)) {
<a name="l00597"></a>00597         std::vector&lt; std::string &gt; overrideInfo = m_gotm-&gt;<a class="code" href="classGenotypeOverrideTableModel.html#aa9a3cafd9eca1946d0fae5d666a7b590" title="GenotypeOverrideTableModel::getOverrideInfo: Synopsis:">getOverrideInfo</a>(egr-&gt;m_experimentName, cRow-&gt;m_probeSet);
<a name="l00598"></a>00598 
<a name="l00599"></a>00599         <span class="keywordflow">if</span> (overrideInfo.size() == 2){
<a name="l00600"></a>00600           cRow-&gt;m_originalBasecall = _generateOriginalBasecall( cRow-&gt;m_probeSet, overrideInfo[0], ttm);
<a name="l00601"></a>00601           cRow-&gt;m_overrideComment  = overrideInfo[1];
<a name="l00602"></a>00602           std::string invalidOriginalBasecall =  m_gotm-&gt;<a class="code" href="classGenotypeOverrideTableModel.html#aa9a3cafd9eca1946d0fae5d666a7b590" title="GenotypeOverrideTableModel::getOverrideInfo: Synopsis:">getOverrideInfo</a>(egr-&gt;m_experimentName, cRow-&gt;m_probeSet, <span class="keyword">true</span>)[0];
<a name="l00603"></a>00603           <span class="keywordflow">if</span> ( !(cRow-&gt;m_overrideComment.find(<span class="stringliteral">&quot;ZeroCopyNumber&quot;</span>) &gt; 0) &amp;&amp; ((cRow-&gt;m_baseCall == invalidOriginalBasecall) || (cRow-&gt;m_originalBasecall == cRow-&gt;m_baseCall )) ){
<a name="l00604"></a>00604             cRow-&gt;m_originalBasecall.clear();
<a name="l00605"></a>00605             cRow-&gt;m_overrideComment.clear();
<a name="l00606"></a>00606           }
<a name="l00607"></a>00607         }
<a name="l00608"></a>00608       }
<a name="l00609"></a>00609       <span class="comment">// Fix up ZeroCopyNumber </span>
<a name="l00610"></a>00610       <span class="keywordflow">if</span> ( (egr != NULL) &amp;&amp; (geneCopyNumber == 0 ) ) {
<a name="l00611"></a>00611         <span class="keywordflow">if</span> ( cRow-&gt;m_haplotypeMarker == <span class="stringliteral">&quot;N&quot;</span> )  {
<a name="l00612"></a>00612           <span class="keywordflow">if</span> ( cRow-&gt;m_interpretationCode == INTERPRETATION_CODES[TranslationInterpretationCodeMap::NoHAP].m_reportString) {
<a name="l00613"></a>00613             cRow-&gt;m_interpretationCode = INTERPRETATION_CODES[TranslationInterpretationCodeMap::UNIQ].m_reportString;
<a name="l00614"></a>00614           }
<a name="l00615"></a>00615           cRow-&gt;m_knownCall = egr-&gt;m_geneZeroCopyHaplotype + <span class="stringliteral">&quot;/&quot;</span> + egr-&gt;m_geneZeroCopyHaplotype;
<a name="l00616"></a>00616         }
<a name="l00617"></a>00617         <span class="keywordflow">else</span> <span class="keywordflow">if</span> ( cRow-&gt;m_knownCall.empty() ) {
<a name="l00618"></a>00618           cRow-&gt;m_knownCall = egr-&gt;m_geneZeroCopyHaplotype + <span class="stringliteral">&quot;/&quot;</span> + egr-&gt;m_geneZeroCopyHaplotype;
<a name="l00619"></a>00619         }
<a name="l00620"></a>00620       }
<a name="l00621"></a>00621 
<a name="l00622"></a>00622       <span class="comment">// Summary Report determination.</span>
<a name="l00623"></a>00623       <span class="keywordflow">if</span> (!cRow-&gt;m_summaryFlag.empty() &amp;&amp; !cRow-&gt;m_call.empty() &amp;&amp;
<a name="l00624"></a>00624           (cRow-&gt;m_summaryFlag != <span class="stringliteral">&quot;N&quot;</span>) &amp;&amp; 
<a name="l00625"></a>00625           RE_VAR.PartialMatch(cRow-&gt;m_call)) {
<a name="l00626"></a>00626         cRow-&gt;m_isSummaryRow = <span class="keyword">true</span>;
<a name="l00627"></a>00627       } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (RE_WILDCARD.FullMatch(cRow-&gt;m_baseCall)) {
<a name="l00628"></a>00628         cRow-&gt;m_isSummaryRow = <span class="keyword">true</span>;
<a name="l00629"></a>00629       }
<a name="l00630"></a>00630       <span class="keywordflow">if</span> (cRow-&gt;m_isSummaryRow) {
<a name="l00631"></a>00631         geneSummaryWritten[gene] = <span class="keyword">true</span>;
<a name="l00632"></a>00632       }
<a name="l00633"></a>00633 
<a name="l00634"></a>00634       <span class="comment">// Uncalled Report determination</span>
<a name="l00635"></a>00635       _setIsUncalledRow(rte, *cRow);
<a name="l00636"></a>00636 
<a name="l00637"></a>00637       <span class="keywordflow">if</span> (!isMultiAllelicRow) {
<a name="l00638"></a>00638         cRows.push_back(*cRow);
<a name="l00639"></a>00639       }
<a name="l00640"></a>00640 
<a name="l00641"></a>00641 
<a name="l00642"></a>00642     } <span class="comment">//for each translation table model marker</span>
<a name="l00643"></a>00643 
<a name="l00644"></a>00644     <span class="keywordflow">for</span> (<span class="keywordtype">size_t</span> j = 0; j &lt; cRows.size(); j++) {
<a name="l00645"></a>00645 
<a name="l00646"></a>00646       <span class="keywordflow">if</span> (!geneSummaryWritten[cRows[j].m_gene]) {
<a name="l00647"></a>00647         cRows[j].m_isGeneSummary = <span class="keyword">true</span>;
<a name="l00648"></a>00648         geneSummaryWritten[cRows[j].m_gene] = <span class="keyword">true</span>;
<a name="l00649"></a>00649       }
<a name="l00650"></a>00650       _report(rte,cRows[j], sampleInfo);
<a name="l00651"></a>00651     }
<a name="l00652"></a>00652 
<a name="l00653"></a>00653   } <span class="comment">// for each gene, experiment result</span>
<a name="l00654"></a>00654 
<a name="l00655"></a>00655   <span class="keywordflow">if</span> ( rte.m_adtOpts.m_summaryReportSort ) {
<a name="l00656"></a>00656     _reportSummarySorted(sampleInfo);
<a name="l00657"></a>00657   }
<a name="l00658"></a>00658 
<a name="l00659"></a>00659   <span class="keywordflow">return</span> okReport;
<a name="l00660"></a>00660 
<a name="l00661"></a>00661 }
<a name="l00662"></a>00662 <span class="comment">// end TranslationExperimentReport::generate</span>
<a name="l00663"></a>00663 <span class="comment">/*****************************************************************************/</span>
<a name="l00664"></a>00664 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00665"></a>00665 <span class="comment">/**</span>
<a name="l00666"></a>00666 <span class="comment"> * TranslationExperimentReport::getComprehensiveReportName</span>
<a name="l00667"></a>00667 <span class="comment"> * Synopsis:</span>
<a name="l00668"></a>00668 <span class="comment"> *</span>
<a name="l00669"></a>00669 <span class="comment"> * Create a Translation report file name from various constant and</span>
<a name="l00670"></a>00670 <span class="comment"> * runtime environment settings. The name is cached in a private</span>
<a name="l00671"></a>00671 <span class="comment"> * variable after the first call.</span>
<a name="l00672"></a>00672 <span class="comment"> *</span>
<a name="l00673"></a>00673 <span class="comment"> * Note that the temporary extension of &quot;.tmp&quot; is not added here but</span>
<a name="l00674"></a>00674 <span class="comment"> * rather a requirement for the callee. </span>
<a name="l00675"></a>00675 <span class="comment"> *</span>
<a name="l00676"></a>00676 <span class="comment"> * @param  rte       - the single instance RunTimeEnvironment with name options in rte.m_adtOpts that were passed in from the console or command line. </span>
<a name="l00677"></a>00677 <span class="comment"> * </span>
<a name="l00678"></a>00678 <span class="comment"> * @return comprehensivefileName - the name of the comprehensive report file.</span>
<a name="l00679"></a>00679 <span class="comment"> *</span>
<a name="l00680"></a>00680 <span class="comment"> */</span>
<a name="l00681"></a>00681 <span class="comment">/*****************************************************************************/</span>
<a name="l00682"></a><a class="code" href="classTranslationExperimentReport.html#ad531a53fc9e67cc0a5e4f1541cc7171e">00682</a> std::string <a class="code" href="classTranslationExperimentReport.html#ad531a53fc9e67cc0a5e4f1541cc7171e" title="TranslationExperimentReport::getComprehensiveReportName Synopsis:">TranslationExperimentReport::getComprehensiveReportName</a>(<span class="keyword">const</span> <a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp; rte)
<a name="l00683"></a>00683 {
<a name="l00684"></a>00684 
<a name="l00685"></a>00685 
<a name="l00686"></a>00686   <span class="keywordflow">if</span> (!m_comprehensiveReportName.empty()) {
<a name="l00687"></a>00687     <span class="keywordflow">return</span> m_comprehensiveReportName;
<a name="l00688"></a>00688   }
<a name="l00689"></a>00689 
<a name="l00690"></a>00690   std::string comprehensiveFileName;
<a name="l00691"></a>00691   <span class="keywordflow">if</span> (!rte.m_adtOpts.m_outputReportPrefix.empty()) {
<a name="l00692"></a>00692     comprehensiveFileName = rte.m_adtOpts.m_outputReportPrefix;
<a name="l00693"></a>00693   } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (rte.m_adtOpts.m_inputExperimentFiles.size() &gt; 0) {
<a name="l00694"></a>00694     comprehensiveFileName = _getFirstCHPFileRoot(rte);
<a name="l00695"></a>00695   } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (rte.m_adtOpts.m_inputGenoFile != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00696"></a>00696     comprehensiveFileName = <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(rte.m_adtOpts.m_inputGenoFile);
<a name="l00697"></a>00697     comprehensiveFileName = comprehensiveFileName.substr(0, comprehensiveFileName.find(<span class="charliteral">&#39;.&#39;</span>));
<a name="l00698"></a>00698   } <span class="keywordflow">else</span> {
<a name="l00699"></a>00699     comprehensiveFileName = name();
<a name="l00700"></a>00700   }
<a name="l00701"></a>00701 
<a name="l00702"></a>00702   comprehensiveFileName = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(rte.m_adtOpts.m_outputDir,
<a name="l00703"></a>00703                                        comprehensiveFileName+COMPREHENSIVE_REPORT_FILE_EXT);
<a name="l00704"></a>00704 
<a name="l00705"></a>00705   m_comprehensiveReportName = comprehensiveFileName;
<a name="l00706"></a>00706 
<a name="l00707"></a>00707   <span class="keywordflow">return</span> comprehensiveFileName;
<a name="l00708"></a>00708 
<a name="l00709"></a>00709 }
<a name="l00710"></a>00710 <span class="comment">// end TranslationExperimentReport::getComprehensiveReportName</span>
<a name="l00711"></a>00711 <span class="comment">/*****************************************************************************/</span>
<a name="l00712"></a>00712 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00713"></a>00713 <span class="comment">/**</span>
<a name="l00714"></a>00714 <span class="comment"> * TranslationExperimentReport::getSummaryReportName</span>
<a name="l00715"></a>00715 <span class="comment"> * Synopsis:</span>
<a name="l00716"></a>00716 <span class="comment"> *</span>
<a name="l00717"></a>00717 <span class="comment"> * Create a Translation report file name from various constant and</span>
<a name="l00718"></a>00718 <span class="comment"> * runtime environment settings. The name is cached in a private</span>
<a name="l00719"></a>00719 <span class="comment"> * variable after the first call.</span>
<a name="l00720"></a>00720 <span class="comment"> *</span>
<a name="l00721"></a>00721 <span class="comment"> * Note that the temporary extension of &quot;.tmp&quot; is not added here but</span>
<a name="l00722"></a>00722 <span class="comment"> * rather a requirement for the callee. </span>
<a name="l00723"></a>00723 <span class="comment"> *</span>
<a name="l00724"></a>00724 <span class="comment"> * @param  rte      - the single instance RunTimeEnvironment which contains name options in rte.m_adtOpts passed in from the console or command line.</span>
<a name="l00725"></a>00725 <span class="comment"> * </span>
<a name="l00726"></a>00726 <span class="comment"> * @return summaryfileName - the name of the summary report file.</span>
<a name="l00727"></a>00727 <span class="comment"> *</span>
<a name="l00728"></a>00728 <span class="comment"> */</span>
<a name="l00729"></a>00729 <span class="comment">/*****************************************************************************/</span>
<a name="l00730"></a><a class="code" href="classTranslationExperimentReport.html#a0b9c78be8ed1c94edab2ecac262e3fb3">00730</a> std::string <a class="code" href="classTranslationExperimentReport.html#a0b9c78be8ed1c94edab2ecac262e3fb3" title="TranslationExperimentReport::getSummaryReportName Synopsis:">TranslationExperimentReport::getSummaryReportName</a>(<span class="keyword">const</span> <a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp; rte)
<a name="l00731"></a>00731 {
<a name="l00732"></a>00732 
<a name="l00733"></a>00733   <span class="keywordflow">if</span> (!m_summaryReportName.empty()) {
<a name="l00734"></a>00734     <span class="keywordflow">return</span> m_summaryReportName;
<a name="l00735"></a>00735   }
<a name="l00736"></a>00736 
<a name="l00737"></a>00737   std::string summaryFileName;
<a name="l00738"></a>00738   <span class="keywordflow">if</span> (! rte.m_adtOpts.m_outputReportPrefix.empty()) {
<a name="l00739"></a>00739     summaryFileName = rte.m_adtOpts.m_outputReportPrefix;
<a name="l00740"></a>00740   } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (rte.m_adtOpts.m_inputExperimentFiles.size() &gt; 0) {
<a name="l00741"></a>00741     summaryFileName = _getFirstCHPFileRoot(rte);
<a name="l00742"></a>00742   } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (!rte.m_adtOpts.m_inputGenoFile.empty()) {
<a name="l00743"></a>00743     summaryFileName = <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(rte.m_adtOpts.m_inputGenoFile);
<a name="l00744"></a>00744     summaryFileName = <a class="code" href="classFs.html#a155921ba8dbce32539855fda92b23e98" title="chop off ONE of the exts. (&amp;quot;path/foo.a.b.c&amp;quot; =&amp;gt; &amp;quot;path/foo.a.b&amp;quot;)">Fs::noextname1</a>(summaryFileName);
<a name="l00745"></a>00745   } <span class="keywordflow">else</span> {
<a name="l00746"></a>00746     summaryFileName = name();
<a name="l00747"></a>00747   }
<a name="l00748"></a>00748 
<a name="l00749"></a>00749   summaryFileName = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(rte.m_adtOpts.m_outputDir,
<a name="l00750"></a>00750                                  summaryFileName+SUMMARY_REPORT_FILE_EXT);
<a name="l00751"></a>00751 
<a name="l00752"></a>00752   m_summaryReportName = summaryFileName;
<a name="l00753"></a>00753 
<a name="l00754"></a>00754   <span class="keywordflow">return</span> summaryFileName;
<a name="l00755"></a>00755 
<a name="l00756"></a>00756 }
<a name="l00757"></a>00757 <span class="comment">// end TranslationExperimentReport::getSummaryReportName</span>
<a name="l00758"></a>00758 <span class="comment">/*****************************************************************************/</span>
<a name="l00759"></a>00759 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00760"></a>00760 <span class="comment">/**</span>
<a name="l00761"></a>00761 <span class="comment"> * TranslationExperimentReport::getUncalledReportName</span>
<a name="l00762"></a>00762 <span class="comment"> * Syopsis:</span>
<a name="l00763"></a>00763 <span class="comment"> *</span>
<a name="l00764"></a>00764 <span class="comment"> * Create a Translation report file name from various constant and</span>
<a name="l00765"></a>00765 <span class="comment"> * runtime environment settings. The name is cached in a private</span>
<a name="l00766"></a>00766 <span class="comment"> * variable after the first call.</span>
<a name="l00767"></a>00767 <span class="comment"> *</span>
<a name="l00768"></a>00768 <span class="comment"> * Note that the temporary extension of &quot;.tmp&quot; is not added here but</span>
<a name="l00769"></a>00769 <span class="comment"> * rather a requirement for the callee. </span>
<a name="l00770"></a>00770 <span class="comment"> *</span>
<a name="l00771"></a>00771 <span class="comment"> * @param  rte      - the single instance RunTimeEnvironment which contains name options in rte.m_adtOpts passed in from the console or command line.</span>
<a name="l00772"></a>00772 <span class="comment"> * </span>
<a name="l00773"></a>00773 <span class="comment"> * @return uncalledfileName - the name of the uncalled report file.</span>
<a name="l00774"></a>00774 <span class="comment"> *</span>
<a name="l00775"></a>00775 <span class="comment"> */</span>
<a name="l00776"></a>00776 <span class="comment">/*****************************************************************************/</span>
<a name="l00777"></a><a class="code" href="classTranslationExperimentReport.html#abdcacbd2a0ddff3e0191177322b3a848">00777</a> std::string <a class="code" href="classTranslationExperimentReport.html#abdcacbd2a0ddff3e0191177322b3a848" title="TranslationExperimentReport::getUncalledReportName Syopsis:">TranslationExperimentReport::getUncalledReportName</a>(<span class="keyword">const</span> <a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp; rte)
<a name="l00778"></a>00778 {
<a name="l00779"></a>00779 
<a name="l00780"></a>00780   <span class="keywordflow">if</span> (!m_uncalledReportName.empty()) {
<a name="l00781"></a>00781     <span class="keywordflow">return</span> m_uncalledReportName;
<a name="l00782"></a>00782   }
<a name="l00783"></a>00783 
<a name="l00784"></a>00784   std::string uncalledFileName;
<a name="l00785"></a>00785   <span class="keywordflow">if</span> (! rte.m_adtOpts.m_outputReportPrefix.empty()) {
<a name="l00786"></a>00786     uncalledFileName = rte.m_adtOpts.m_outputReportPrefix;
<a name="l00787"></a>00787   } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (rte.m_adtOpts.m_inputExperimentFiles.size() &gt; 0) {
<a name="l00788"></a>00788     uncalledFileName = _getFirstCHPFileRoot(rte);
<a name="l00789"></a>00789   } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (rte.m_adtOpts.m_inputGenoFile != <span class="stringliteral">&quot;&quot;</span>) {
<a name="l00790"></a>00790     uncalledFileName = <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(rte.m_adtOpts.m_inputGenoFile);
<a name="l00791"></a>00791     uncalledFileName = <a class="code" href="classFs.html#a155921ba8dbce32539855fda92b23e98" title="chop off ONE of the exts. (&amp;quot;path/foo.a.b.c&amp;quot; =&amp;gt; &amp;quot;path/foo.a.b&amp;quot;)">Fs::noextname1</a>(uncalledFileName);
<a name="l00792"></a>00792   } <span class="keywordflow">else</span> {
<a name="l00793"></a>00793     uncalledFileName = name();
<a name="l00794"></a>00794   }
<a name="l00795"></a>00795 
<a name="l00796"></a>00796   uncalledFileName = <a class="code" href="classFs.html#af44351abee4cbc47f94b83cb2f62899f" title="join strings with the path sep. /// while we would want people to keep the path with FsPath...">Fs::join</a>(rte.m_adtOpts.m_outputDir,
<a name="l00797"></a>00797                                   uncalledFileName+UNCALLED_REPORT_FILE_EXT);
<a name="l00798"></a>00798 
<a name="l00799"></a>00799   m_uncalledReportName = uncalledFileName;
<a name="l00800"></a>00800 
<a name="l00801"></a>00801   <span class="keywordflow">return</span> uncalledFileName;
<a name="l00802"></a>00802 
<a name="l00803"></a>00803 }
<a name="l00804"></a>00804 <span class="comment">// end TranslationExperimentReport::getUncalledReportName</span>
<a name="l00805"></a>00805 <span class="comment">/*****************************************************************************/</span>
<a name="l00806"></a>00806 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00807"></a>00807 <span class="comment">/**</span>
<a name="l00808"></a>00808 <span class="comment"> * TranslationExperimentReport::_closeComprehensiveFile</span>
<a name="l00809"></a>00809 <span class="comment"> * Synopsis:</span>
<a name="l00810"></a>00810 <span class="comment"> * </span>
<a name="l00811"></a>00811 <span class="comment"> * Close a TSV file of comprehensive report results.</span>
<a name="l00812"></a>00812 <span class="comment"> * </span>
<a name="l00813"></a>00813 <span class="comment"> * !!!WARNING!!! under no circumstances should messaging via Verbose</span>
<a name="l00814"></a>00814 <span class="comment"> * take place when abort is passed as true. This will crash the</span>
<a name="l00815"></a>00815 <span class="comment"> * Windows console due to a bug in the Windows COM code. </span>
<a name="l00816"></a>00816 <span class="comment"> * </span>
<a name="l00817"></a>00817 <span class="comment"> * @param abort - if this close is happening in the midst of an abort exception.</span>
<a name="l00818"></a>00818 <span class="comment"> *</span>
<a name="l00819"></a>00819 <span class="comment"> * @return true - if the temporary file is moved to the non-temporary file name.</span>
<a name="l00820"></a>00820 <span class="comment"> */</span>
<a name="l00821"></a>00821 <span class="comment">/*****************************************************************************/</span>
<a name="l00822"></a>00822 <span class="keywordtype">bool</span> TranslationExperimentReport::_closeComprehensiveFile(<span class="keywordtype">bool</span> abort)
<a name="l00823"></a>00823 {
<a name="l00824"></a>00824 
<a name="l00825"></a>00825 
<a name="l00826"></a>00826   <span class="keywordtype">bool</span> okMove = <span class="keyword">true</span>;
<a name="l00827"></a>00827 
<a name="l00828"></a>00828   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#ae7932be2156c04a562f327b53833aed7" title="Close the file.">close</a>();
<a name="l00829"></a>00829   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a6a397412f48d477d0e87b00f914d06f7" title="Closes the file and clears bindings and other info.">clear</a>();
<a name="l00830"></a>00830 
<a name="l00831"></a>00831   <span class="keywordflow">if</span> (!abort) {
<a name="l00832"></a>00832     std::string tmpFile = m_comprehensiveReportName + TEMP_REPORT_FILE_EXT;
<a name="l00833"></a>00833     std::string completeFile = m_comprehensiveReportName + COMPLETE_REPORT_FILE_EXT ;
<a name="l00834"></a>00834     okMove = <a class="code" href="classFs.html#a57352b91aec25124a8f5b2db1cababeb" title="Return true on success.">Fs::fileRename</a>(tmpFile,  completeFile, <span class="keyword">false</span>);
<a name="l00835"></a>00835 
<a name="l00836"></a>00836     <span class="keywordflow">if</span> (!okMove) {
<a name="l00837"></a>00837       <a class="code" href="classVerbose.html#a34e1d936178741eeb78de93c5497e1b8" title="Print a warning message.">Verbose::warn</a>(ADT_VERBOSE_NORMAL, <span class="stringliteral">&quot;Can&#39;t rename file &quot;</span> + tmpFile + <span class="stringliteral">&quot; as &quot;</span> + completeFile + <span class="stringliteral">&quot;.&quot;</span>);
<a name="l00838"></a>00838     }
<a name="l00839"></a>00839   }
<a name="l00840"></a>00840 
<a name="l00841"></a>00841   <span class="keywordflow">return</span> okMove;
<a name="l00842"></a>00842 }
<a name="l00843"></a>00843 <span class="comment">// end TranslationExperimentReport::_closeComprehensiveFile</span>
<a name="l00844"></a>00844 <span class="comment">/*****************************************************************************/</span>
<a name="l00845"></a>00845 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00846"></a>00846 <span class="comment">/**</span>
<a name="l00847"></a>00847 <span class="comment"> * TranslationExperimentReport::_closeSummaryFile</span>
<a name="l00848"></a>00848 <span class="comment"> * Synopsis:</span>
<a name="l00849"></a>00849 <span class="comment"> * </span>
<a name="l00850"></a>00850 <span class="comment"> * Close a TSV file for outputing summary report results.</span>
<a name="l00851"></a>00851 <span class="comment"> *</span>
<a name="l00852"></a>00852 <span class="comment"> * !!!WARNING!!! under no circumstances should messaging via Verbose</span>
<a name="l00853"></a>00853 <span class="comment"> * take place when abort is passed as true. This will crash the</span>
<a name="l00854"></a>00854 <span class="comment"> * Windows console due to a bug in the Windows COM code. </span>
<a name="l00855"></a>00855 <span class="comment"> *</span>
<a name="l00856"></a>00856 <span class="comment"> * @param abort - if this close is happening in the midst of an abort exception.</span>
<a name="l00857"></a>00857 <span class="comment"> *</span>
<a name="l00858"></a>00858 <span class="comment"> * @return true - if the temporary file is moved to the non-temporary file name.</span>
<a name="l00859"></a>00859 <span class="comment"> */</span>
<a name="l00860"></a>00860 <span class="comment">/*****************************************************************************/</span>
<a name="l00861"></a>00861 <span class="keywordtype">bool</span> TranslationExperimentReport::_closeSummaryFile(<span class="keywordtype">bool</span> abort)
<a name="l00862"></a>00862 {
<a name="l00863"></a>00863 
<a name="l00864"></a>00864   <span class="keywordtype">bool</span> okMove = <span class="keyword">true</span>;
<a name="l00865"></a>00865 
<a name="l00866"></a>00866   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#ae7932be2156c04a562f327b53833aed7" title="Close the file.">close</a>();
<a name="l00867"></a>00867   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a6a397412f48d477d0e87b00f914d06f7" title="Closes the file and clears bindings and other info.">clear</a>();
<a name="l00868"></a>00868 
<a name="l00869"></a>00869   <span class="keywordflow">if</span> (!abort) {
<a name="l00870"></a>00870     std::string tmpFile = m_summaryReportName + TEMP_REPORT_FILE_EXT;
<a name="l00871"></a>00871     std::string completeFile = <a class="code" href="classTranslationExperimentReport.html#a54c4debab5d50f4eee012ec4f0e70921" title="TranslationExperimentReport::getSummaryFileName Synopsis:">getSummaryFileName</a>();
<a name="l00872"></a>00872     okMove = <a class="code" href="classFs.html#a57352b91aec25124a8f5b2db1cababeb" title="Return true on success.">Fs::fileRename</a>(tmpFile,  completeFile, <span class="keyword">false</span>);
<a name="l00873"></a>00873 
<a name="l00874"></a>00874     <span class="keywordflow">if</span> (!okMove) {
<a name="l00875"></a>00875       <a class="code" href="classVerbose.html#a34e1d936178741eeb78de93c5497e1b8" title="Print a warning message.">Verbose::warn</a>(ADT_VERBOSE_NORMAL, <span class="stringliteral">&quot;Can&#39;t rename file &quot;</span> + tmpFile + <span class="stringliteral">&quot; as &quot;</span> + completeFile + <span class="stringliteral">&quot;.&quot;</span>);
<a name="l00876"></a>00876     }
<a name="l00877"></a>00877   }
<a name="l00878"></a>00878 
<a name="l00879"></a>00879 
<a name="l00880"></a>00880   <span class="keywordflow">return</span> okMove;
<a name="l00881"></a>00881 
<a name="l00882"></a>00882 }
<a name="l00883"></a>00883 <span class="comment">// end TranslationExperimentReport::_closeTranslationFile</span>
<a name="l00884"></a>00884 
<a name="l00885"></a>00885 <span class="comment">/*****************************************************************************/</span>
<a name="l00886"></a>00886 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00887"></a>00887 <span class="comment">/**</span>
<a name="l00888"></a>00888 <span class="comment"> * TranslationExperimentReport::getSummaryFileName</span>
<a name="l00889"></a>00889 <span class="comment"> * Synopsis:</span>
<a name="l00890"></a>00890 <span class="comment"> *</span>
<a name="l00891"></a>00891 <span class="comment"> * Returns the name of the file storing the summary results.</span>
<a name="l00892"></a>00892 <span class="comment"> *</span>
<a name="l00893"></a>00893 <span class="comment"> * @return the file name.</span>
<a name="l00894"></a>00894 <span class="comment"> *</span>
<a name="l00895"></a>00895 <span class="comment"> */</span>
<a name="l00896"></a>00896 <span class="comment">/*****************************************************************************/</span>
<a name="l00897"></a><a class="code" href="classTranslationExperimentReport.html#a54c4debab5d50f4eee012ec4f0e70921">00897</a> std::string <a class="code" href="classTranslationExperimentReport.html#a54c4debab5d50f4eee012ec4f0e70921" title="TranslationExperimentReport::getSummaryFileName Synopsis:">TranslationExperimentReport::getSummaryFileName</a>()
<a name="l00898"></a>00898 {
<a name="l00899"></a>00899     std::string completeFile = m_summaryReportName + COMPLETE_REPORT_FILE_EXT ;
<a name="l00900"></a>00900         <span class="keywordflow">return</span> completeFile;
<a name="l00901"></a>00901 }
<a name="l00902"></a>00902 <span class="comment">// end TranslationExperimentReport::getSummaryFileName</span>
<a name="l00903"></a>00903 
<a name="l00904"></a>00904 <span class="comment">/*****************************************************************************/</span>
<a name="l00905"></a>00905 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00906"></a>00906 <span class="comment">/**</span>
<a name="l00907"></a>00907 <span class="comment"> * TranslationExperimentReport::getComprehensiveFileName</span>
<a name="l00908"></a>00908 <span class="comment"> * Synopsis:</span>
<a name="l00909"></a>00909 <span class="comment"> *</span>
<a name="l00910"></a>00910 <span class="comment"> * Returns the name of the file storing the summary results.</span>
<a name="l00911"></a>00911 <span class="comment"> *</span>
<a name="l00912"></a>00912 <span class="comment"> * @return the file name.</span>
<a name="l00913"></a>00913 <span class="comment"> *</span>
<a name="l00914"></a>00914 <span class="comment"> */</span>
<a name="l00915"></a>00915 <span class="comment">/*****************************************************************************/</span>
<a name="l00916"></a><a class="code" href="classTranslationExperimentReport.html#aa06a48a5411498a2fe538d92e321deb0">00916</a> std::string <a class="code" href="classTranslationExperimentReport.html#aa06a48a5411498a2fe538d92e321deb0" title="TranslationExperimentReport::getComprehensiveFileName Synopsis:">TranslationExperimentReport::getComprehensiveFileName</a>()
<a name="l00917"></a>00917 {
<a name="l00918"></a>00918     std::string completeFile = m_comprehensiveReportName + COMPLETE_REPORT_FILE_EXT ;
<a name="l00919"></a>00919         <span class="keywordflow">return</span> completeFile;
<a name="l00920"></a>00920 }
<a name="l00921"></a>00921 <span class="comment">// end TranslationExperimentReport::getComprehensiveFileName</span>
<a name="l00922"></a>00922 
<a name="l00923"></a>00923 <span class="comment">/*****************************************************************************/</span>
<a name="l00924"></a>00924 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00925"></a>00925 <span class="comment">/**</span>
<a name="l00926"></a>00926 <span class="comment"> * TranslationExperimentReport::_closeUncalledFile</span>
<a name="l00927"></a>00927 <span class="comment"> * Synopsis:</span>
<a name="l00928"></a>00928 <span class="comment"> *</span>
<a name="l00929"></a>00929 <span class="comment"> * Close a TSV file for outputing uncalled marker report results.</span>
<a name="l00930"></a>00930 <span class="comment"> *</span>
<a name="l00931"></a>00931 <span class="comment"> * !!!WARNING!!! under no circumstances should messaging via Verbose</span>
<a name="l00932"></a>00932 <span class="comment"> * take place when abort is passed as true. This will crash the</span>
<a name="l00933"></a>00933 <span class="comment"> * Windows console due to a bug in the Windows COM code. </span>
<a name="l00934"></a>00934 <span class="comment"> *</span>
<a name="l00935"></a>00935 <span class="comment"> * @param abort - if this close is happening in the midst of an abort exception.</span>
<a name="l00936"></a>00936 <span class="comment"> *</span>
<a name="l00937"></a>00937 <span class="comment"> * @return true - if the temporary file is moved to the non-temporary file name.</span>
<a name="l00938"></a>00938 <span class="comment"> *</span>
<a name="l00939"></a>00939 <span class="comment"> */</span>
<a name="l00940"></a>00940 <span class="comment">/*****************************************************************************/</span>
<a name="l00941"></a>00941 <span class="keywordtype">bool</span> TranslationExperimentReport::_closeUncalledFile(<span class="keywordtype">bool</span> abort)
<a name="l00942"></a>00942 {
<a name="l00943"></a>00943 
<a name="l00944"></a>00944   <span class="keywordtype">bool</span> okMove = <span class="keyword">true</span>;
<a name="l00945"></a>00945 
<a name="l00946"></a>00946   m_uncalledTsv.<a class="code" href="classaffx_1_1TsvFile.html#ae7932be2156c04a562f327b53833aed7" title="Close the file.">close</a>();
<a name="l00947"></a>00947   m_uncalledTsv.<a class="code" href="classaffx_1_1TsvFile.html#a6a397412f48d477d0e87b00f914d06f7" title="Closes the file and clears bindings and other info.">clear</a>();
<a name="l00948"></a>00948 
<a name="l00949"></a>00949   <span class="keywordflow">if</span> (!abort) {
<a name="l00950"></a>00950     std::string tmpFile = m_uncalledReportName + TEMP_REPORT_FILE_EXT;
<a name="l00951"></a>00951     std::string completeFile = m_uncalledReportName + COMPLETE_REPORT_FILE_EXT ;
<a name="l00952"></a>00952     okMove = <a class="code" href="classFs.html#a57352b91aec25124a8f5b2db1cababeb" title="Return true on success.">Fs::fileRename</a>(tmpFile,  completeFile, <span class="keyword">false</span>);
<a name="l00953"></a>00953 
<a name="l00954"></a>00954     <span class="keywordflow">if</span> (!okMove) {
<a name="l00955"></a>00955       <a class="code" href="classVerbose.html#a34e1d936178741eeb78de93c5497e1b8" title="Print a warning message.">Verbose::warn</a>(ADT_VERBOSE_NORMAL, <span class="stringliteral">&quot;Can&#39;t rename file &quot;</span> + tmpFile + <span class="stringliteral">&quot; as &quot;</span> + completeFile + <span class="stringliteral">&quot;.&quot;</span>);
<a name="l00956"></a>00956     }
<a name="l00957"></a>00957   }
<a name="l00958"></a>00958 
<a name="l00959"></a>00959   <span class="keywordflow">return</span> okMove;
<a name="l00960"></a>00960 
<a name="l00961"></a>00961 }
<a name="l00962"></a>00962 <span class="comment">// end TranslationExperimentReport::_closeUncalledFile</span>
<a name="l00963"></a>00963 <span class="comment">/*****************************************************************************/</span>
<a name="l00964"></a>00964 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l00965"></a>00965 <span class="comment">/**</span>
<a name="l00966"></a>00966 <span class="comment"> * TranslationExperimentReport::_generateBaseCall</span>
<a name="l00967"></a>00967 <span class="comment"> * Synopsis:</span>
<a name="l00968"></a>00968 <span class="comment"> * </span>
<a name="l00969"></a>00969 <span class="comment"> * Helper function to TranslationExperimentReport::generate, broken</span>
<a name="l00970"></a>00970 <span class="comment"> * out for readability only and not for reuse. </span>
<a name="l00971"></a>00971 <span class="comment"> *</span>
<a name="l00972"></a>00972 <span class="comment"> * Set the m_baseCall field in the TranslationRow structure to something like</span>
<a name="l00973"></a>00973 <span class="comment"> * &quot;*1/ *1&quot;. </span>
<a name="l00974"></a>00974 <span class="comment"> *</span>
<a name="l00975"></a>00975 <span class="comment"> * </span>
<a name="l00976"></a>00976 <span class="comment"> * @param egr - the ExperimentGeneResult for the cRow being processed. </span>
<a name="l00977"></a>00977 <span class="comment"> * @param ttmRow - the translation table model row index. </span>
<a name="l00978"></a>00978 <span class="comment"> * @param markerCallResultsIndex - egr-&gt;m_callResults index for the cRow being passed. </span>
<a name="l00979"></a>00979 <span class="comment"> * @param cRow - the output row being reported on, prepopulated with gene and  probe set id</span>
<a name="l00980"></a>00980 <span class="comment"> * @param ttm - the single instance translation table model</span>
<a name="l00981"></a>00981 <span class="comment"> * @param isMultiAllelicRow - if this probe set id being processed was previously processed. </span>
<a name="l00982"></a>00982 <span class="comment"> * @param isProbeSetPresent - if the genotype data contained this probe set. </span>
<a name="l00983"></a>00983 <span class="comment"> *</span>
<a name="l00984"></a>00984 <span class="comment"> * @return - cRow.m_baseCall filled in.</span>
<a name="l00985"></a>00985 <span class="comment"> */</span>
<a name="l00986"></a>00986 <span class="comment">/*****************************************************************************/</span>
<a name="l00987"></a>00987 <span class="keywordtype">void</span> TranslationExperimentReport::_generateBaseCall(<a class="code" href="classExperimentGeneResults.html">ExperimentGeneResults</a> *egr,  <span class="keywordtype">int</span> ttmRow,  <span class="keywordtype">int</span> markerCallResultsIndex,  <a class="code" href="classTranslationRow.html">TranslationRow</a> * cRow, <a class="code" href="classTranslationTableModel.html">TranslationTableModel</a> &amp; ttm, <span class="keywordtype">bool</span> isMultiAllelicRow, <span class="keywordtype">bool</span> isProbeSetPresent)
<a name="l00988"></a>00988 {
<a name="l00989"></a>00989 
<a name="l00990"></a>00990 
<a name="l00991"></a>00991   <span class="keywordflow">if</span> (!isMultiAllelicRow) {
<a name="l00992"></a>00992     cRow-&gt;m_copyNumber = 2;
<a name="l00993"></a>00993     cRow-&gt;m_baseCall = std::string(<span class="stringliteral">&quot;NotAvailable&quot;</span>);
<a name="l00994"></a>00994   }
<a name="l00995"></a>00995 
<a name="l00996"></a>00996   <span class="keywordflow">if</span> (isProbeSetPresent &amp;&amp; (egr != NULL)) {
<a name="l00997"></a>00997     <span class="keywordflow">if</span> ((egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_ceSet.size() &gt; 0) &amp;&amp;
<a name="l00998"></a>00998         (egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_ceSet.size() &gt; 0) &amp;&amp;
<a name="l00999"></a>00999         pcrecpp::RE(<span class="stringliteral">&quot;PRA|PossibleRareAllele&quot;</span>).FullMatch(egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_ceSet.begin()-&gt;second.m_bases[0])) {
<a name="l01000"></a>01000       cRow-&gt;m_baseCall = <span class="stringliteral">&quot;PossibleRareAllele&quot;</span>;
<a name="l01001"></a>01001     } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_copyNumber == 0) {
<a name="l01002"></a>01002       <span class="keywordflow">if</span> ((egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_ceSet.size() == 0) ||
<a name="l01003"></a>01003           (pcrecpp::RE(<span class="stringliteral">&quot;NA|NotAvailable&quot;</span>).FullMatch(egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_ceSet.begin()-&gt;second.m_bases[0]))) {
<a name="l01004"></a>01004         cRow-&gt;m_baseCall = std::string(<span class="stringliteral">&quot;NotAvailable&quot;</span>);
<a name="l01005"></a>01005       } <span class="keywordflow">else</span> {
<a name="l01006"></a>01006         cRow-&gt;m_baseCall = std::string(<span class="stringliteral">&quot;ZeroCopyNumber&quot;</span>);
<a name="l01007"></a>01007         cRow-&gt;m_copyNumber = 0;
<a name="l01008"></a>01008       }
<a name="l01009"></a>01009     } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_copyNumber == 1) {
<a name="l01010"></a>01010 
<a name="l01011"></a>01011       cRow-&gt;m_copyNumber = 1;
<a name="l01012"></a>01012       std::string definingValue = cRow-&gt;m_call.substr(cRow-&gt;m_call.find(<span class="charliteral">&#39;/&#39;</span>) + 1);
<a name="l01013"></a>01013 
<a name="l01014"></a>01014       <span class="keywordflow">if</span> (definingValue  == <span class="stringliteral">&quot;Ref&quot;</span>) {
<a name="l01015"></a>01015         cRow-&gt;m_baseCall = cRow-&gt;m_referenceReportBase;
<a name="l01016"></a>01016       } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (definingValue  == <span class="stringliteral">&quot;Var&quot;</span>) {
<a name="l01017"></a>01017         cRow-&gt;m_baseCall = egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_ceSet.begin()-&gt;second.m_bases[0];
<a name="l01018"></a>01018       } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (pcrecpp::RE(<span class="stringliteral">&quot;NC|NoCall&quot;</span>).FullMatch(egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_ceSet.begin()-&gt;second.m_bases[0])) {
<a name="l01019"></a>01019         cRow-&gt;m_baseCall = <span class="stringliteral">&quot;NoCall&quot;</span>;
<a name="l01020"></a>01020       }
<a name="l01021"></a>01021     } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (cRow-&gt;m_call == <span class="stringliteral">&quot;Ref/Ref&quot;</span>) {
<a name="l01022"></a>01022       cRow-&gt;m_baseCall = cRow-&gt;m_referenceReportBase + <span class="stringliteral">&quot;/&quot;</span> + cRow-&gt;m_referenceReportBase;
<a name="l01023"></a>01023     } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (cRow-&gt;m_call == <span class="stringliteral">&quot;Ref/Var&quot;</span>) {
<a name="l01024"></a>01024       std::string variance =  egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_ceSet.begin()-&gt;second.m_bases[0];
<a name="l01025"></a>01025       <span class="keywordflow">if</span> (variance ==  cRow-&gt;m_referenceBase) {
<a name="l01026"></a>01026         variance =  egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid2.m_ceSet.begin()-&gt;second.m_bases[0];
<a name="l01027"></a>01027       }
<a name="l01028"></a>01028       <span class="keywordflow">if</span> (variance == ttm.m_rows[ttmRow][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_VARIANT)]) {
<a name="l01029"></a>01029         cRow-&gt;m_baseCall = cRow-&gt;m_referenceReportBase + <span class="stringliteral">&quot;/&quot;</span> + ttm.<a class="code" href="classTranslationTableModel.html#a0d5cf32bb7c3a56543767bf15b51834b" title="TranslationTableModel::getReportAllele Synopsis:">getReportAllele</a>( cRow-&gt;m_probeSet, variance);
<a name="l01030"></a>01030       }
<a name="l01031"></a>01031     } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (cRow-&gt;m_call == <span class="stringliteral">&quot;Var/Var&quot;</span>) {
<a name="l01032"></a>01032       std::string variance1 =  egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_ceSet.begin()-&gt;second.m_bases[0];
<a name="l01033"></a>01033       <span class="keywordflow">if</span> (variance1 == ttm.m_rows[ttmRow][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_VARIANT)]) {
<a name="l01034"></a>01034         std::string variance2 =  egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid2.m_ceSet.begin()-&gt;second.m_bases[0];
<a name="l01035"></a>01035         cRow-&gt;m_baseCall = ttm.<a class="code" href="classTranslationTableModel.html#a0d5cf32bb7c3a56543767bf15b51834b" title="TranslationTableModel::getReportAllele Synopsis:">getReportAllele</a>( cRow-&gt;m_probeSet, variance1) + <span class="stringliteral">&quot;/&quot;</span> + ttm.<a class="code" href="classTranslationTableModel.html#a0d5cf32bb7c3a56543767bf15b51834b" title="TranslationTableModel::getReportAllele Synopsis:">getReportAllele</a>(cRow-&gt;m_probeSet, variance2) ;
<a name="l01036"></a>01036       }
<a name="l01037"></a>01037     } <span class="keywordflow">else</span> <span class="keywordflow">if</span> (pcrecpp::RE(<span class="stringliteral">&quot;NC|NoCall&quot;</span>).FullMatch(egr-&gt;m_callResults[markerCallResultsIndex].m_experimentChromatid1.m_ceSet.begin()-&gt;second.m_bases[0])) {
<a name="l01038"></a>01038       cRow-&gt;m_baseCall = <span class="stringliteral">&quot;NoCall&quot;</span>;
<a name="l01039"></a>01039     }
<a name="l01040"></a>01040   }
<a name="l01041"></a>01041 
<a name="l01042"></a>01042   <span class="keywordflow">return</span>;
<a name="l01043"></a>01043 }
<a name="l01044"></a>01044 <span class="comment">// end TranslationExperimentReport::_generateBaseCall</span>
<a name="l01045"></a>01045 <span class="comment">/*****************************************************************************/</span>
<a name="l01046"></a>01046 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l01047"></a>01047 <span class="comment">/**</span>
<a name="l01048"></a>01048 <span class="comment"> * TranslationExperimentReport::_generateGeneHaplotypeCallMap:</span>
<a name="l01049"></a>01049 <span class="comment"> * Synopsis:</span>
<a name="l01050"></a>01050 <span class="comment"> *</span>
<a name="l01051"></a>01051 <span class="comment"> * Helper function to TranslationExperimentReport::generate, broken</span>
<a name="l01052"></a>01052 <span class="comment"> * out for readability only and not for reuse. To be called once per gene</span>
<a name="l01053"></a>01053 <span class="comment"> * and not once per row. </span>
<a name="l01054"></a>01054 <span class="comment"> *</span>
<a name="l01055"></a>01055 <span class="comment"> *  Haplotype calls are per gene, not per marker. Therefore we need to compile</span>
<a name="l01056"></a>01056 <span class="comment"> *  a list of known and unkown calls per gene. These calls then get repeated</span>
<a name="l01057"></a>01057 <span class="comment"> *  per every assay id in the haplotype group for that gene.</span>
<a name="l01058"></a>01058 <span class="comment"> *</span>
<a name="l01059"></a>01059 <span class="comment"> * </span>
<a name="l01060"></a>01060 <span class="comment"> * @param rte  - the single instance run time environment.</span>
<a name="l01061"></a>01061 <span class="comment"> * @param er   - the ExperimentResults container of all the ExperimentGeneResults</span>
<a name="l01062"></a>01062 <span class="comment"> * @param geneHaplotypeMap - the index of entire gene saved for all rows of geno data for this </span>
<a name="l01063"></a>01063 <span class="comment"> *</span>
<a name="l01064"></a>01064 <span class="comment"> * @return - geneHaplotypeMap created for this gene</span>
<a name="l01065"></a>01065 <span class="comment"> * </span>
<a name="l01066"></a>01066 <span class="comment"> */</span>
<a name="l01067"></a>01067 <span class="comment">/*****************************************************************************/</span>
<a name="l01068"></a>01068 <span class="keywordtype">void</span> TranslationExperimentReport::_generateGeneHaplotypeCallMap(<a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp; rte, std::map&lt;std::string, ExperimentResults *&gt;&amp;er, std::map&lt; std::string, GeneHaplotypeCallMapElement &gt; &amp; geneHaplotypeMap)
<a name="l01069"></a>01069 {
<a name="l01070"></a>01070 
<a name="l01071"></a>01071   pcrecpp::RE reUnknown(<span class="stringliteral">&quot;UNK&quot;</span>);
<a name="l01072"></a>01072 
<a name="l01073"></a>01073   std::map&lt;std::string, ExperimentResults*&gt;::iterator itSER;
<a name="l01074"></a>01074 
<a name="l01075"></a>01075   <span class="keywordflow">for</span> (itSER = er.begin(); (itSER != er.end()) ; itSER++) {
<a name="l01076"></a>01076 
<a name="l01077"></a>01077     <span class="keywordflow">if</span> (itSER-&gt;second-&gt;m_markerCallCount == 0) {
<a name="l01078"></a>01078       <span class="keywordflow">continue</span>;
<a name="l01079"></a>01079     }
<a name="l01080"></a>01080 
<a name="l01081"></a>01081     std::map&lt;std::string, ExperimentGeneResults*&gt;::iterator itSEGR;
<a name="l01082"></a>01082 
<a name="l01083"></a>01083     <span class="keywordflow">for</span> (itSEGR = itSER-&gt;second-&gt;m_geneResults.begin();
<a name="l01084"></a>01084          itSEGR != itSER-&gt;second-&gt;m_geneResults.end(); itSEGR++) {
<a name="l01085"></a>01085 
<a name="l01086"></a>01086       <a class="code" href="classExperimentGeneResults.html">ExperimentGeneResults</a> &amp; egr = *(itSEGR-&gt;second) ;
<a name="l01087"></a>01087 
<a name="l01088"></a>01088       <span class="keywordflow">for</span> (<span class="keywordtype">size_t</span> i = 0; i &lt; egr.size(); i++) {
<a name="l01089"></a>01089 
<a name="l01090"></a>01090         std::string gene        = egr.m_callResults[i].m_geneName;
<a name="l01091"></a>01091         geneHaplotypeMap[gene].m_gene = gene;
<a name="l01092"></a>01092         <span class="keywordflow">if</span> ( egr.m_callResults[i].m_experimentChromatid1.m_copyNumber == 0 ) {
<a name="l01093"></a>01093           geneHaplotypeMap[gene].m_copyNumber = 0;
<a name="l01094"></a>01094         }
<a name="l01095"></a>01095         <span class="keywordflow">if</span> (egr.m_callResults[i].getCallType() != ADT_CALL_TYPE_HAPLOTYPE_GROUP) {
<a name="l01096"></a>01096           <span class="keywordflow">continue</span>;
<a name="l01097"></a>01097         }
<a name="l01098"></a>01098 
<a name="l01099"></a>01099         <span class="keywordflow">for</span> (<span class="keywordtype">int</span> j = 0; j &lt; egr.m_callResults[i].size() ; j++) {
<a name="l01100"></a>01100 
<a name="l01101"></a>01101           std::string call =  egr.m_callResults[i].m_alleleCalls[j].m_allele;
<a name="l01102"></a>01102 
<a name="l01103"></a>01103           geneHaplotypeMap[gene].m_gene = gene;
<a name="l01104"></a>01104           <span class="keywordflow">if</span> (reUnknown.PartialMatch(call)) {
<a name="l01105"></a>01105             geneHaplotypeMap[gene].m_unknownCall.push_back(call);
<a name="l01106"></a>01106           } <span class="keywordflow">else</span> {
<a name="l01107"></a>01107             geneHaplotypeMap[gene].m_knownCall.push_back(call);
<a name="l01108"></a>01108           }
<a name="l01109"></a>01109 
<a name="l01110"></a>01110         } <span class="comment">// Foreach  call, if they are more than one (wild cards )</span>
<a name="l01111"></a>01111 
<a name="l01112"></a>01112       } <span class="comment">// foreach experiment gene call list</span>
<a name="l01113"></a>01113 
<a name="l01114"></a>01114     } <span class="comment">// for each experiment gene</span>
<a name="l01115"></a>01115 
<a name="l01116"></a>01116   } <span class="comment">// foreach experiment</span>
<a name="l01117"></a>01117 
<a name="l01118"></a>01118   <span class="keywordflow">return</span>;
<a name="l01119"></a>01119 
<a name="l01120"></a>01120 }
<a name="l01121"></a>01121 <span class="comment">// end TranslationExperimentReport::_generateGeneHaplotypeCallMap</span>
<a name="l01122"></a>01122 <span class="comment">/*****************************************************************************/</span>
<a name="l01123"></a>01123 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l01124"></a>01124 <span class="comment">/**</span>
<a name="l01125"></a>01125 <span class="comment"> * TranslationExperimentReport::_generateInterpretationCode:</span>
<a name="l01126"></a>01126 <span class="comment"> * Synopsis:</span>
<a name="l01127"></a>01127 <span class="comment"> *</span>
<a name="l01128"></a>01128 <span class="comment"> * Helper function to TranslationExperimentReport::generate, broken</span>
<a name="l01129"></a>01129 <span class="comment"> * out for readability only and not for reuse. </span>
<a name="l01130"></a>01130 <span class="comment"> *</span>
<a name="l01131"></a>01131 <span class="comment"> *  Contains the busines logic for generating the interpretation code</span>
<a name="l01132"></a>01132 <span class="comment"> *  Report field</span>
<a name="l01133"></a>01133 <span class="comment"> *</span>
<a name="l01134"></a>01134 <span class="comment"> * Interpretation Code Description:</span>
<a name="l01135"></a>01135 <span class="comment"> * UNIQ: Unique haplotype pair ( 1 known call, 0 unknown calls )</span>
<a name="l01136"></a>01136 <span class="comment"> * UNIQ+UNK: Unique annotated haplotype pair other haplotype pairs involving</span>
<a name="l01137"></a>01137 <span class="comment"> * anannotated haplotypes also possible</span>
<a name="l01138"></a>01138 <span class="comment"> *       (1 known call, 1+ unkown calls )</span>
<a name="l01139"></a>01139 <span class="comment"> * MULT: Multiple haplotype pairs possible due to phase ambiguity</span>
<a name="l01140"></a>01140 <span class="comment"> *           ( 2+ known calls, 0 unkowns )</span>
<a name="l01141"></a>01141 <span class="comment"> * MULT+UNK: Multiple annotated haplotype calls possible other haplotype</span>
<a name="l01142"></a>01142 <span class="comment"> * pairs involving unannotated haplotypes also possible</span>
<a name="l01143"></a>01143 <span class="comment"> *           (2+ known calls, 1+ unkowns)</span>
<a name="l01144"></a>01144 <span class="comment"> * UNDH: Only haplotype pairs with undefined haplotypes possible</span>
<a name="l01145"></a>01145 <span class="comment"> *         (0 known calls, 1+ unkown calls)</span>
<a name="l01146"></a>01146 <span class="comment"> * NC/PRA: NoCall or PossibleRareAllele call for one or more markers</span>
<a name="l01147"></a>01147 <span class="comment"> * resulting in multiple haplotype pairs</span>
<a name="l01148"></a>01148 <span class="comment"> *         (1+ assay id is NC or PCRA)</span>
<a name="l01149"></a>01149 <span class="comment"> * NoHAP: No haplotypes defined at this gene</span>
<a name="l01150"></a>01150 <span class="comment"> *        ( translation table &quot;Haplotype&quot; column is all &#39;N&#39; for assay ids)</span>
<a name="l01151"></a>01151 <span class="comment"> * NA: No genotyping data available</span>
<a name="l01152"></a>01152 <span class="comment"> *        (Markers ignored by override or filter operations).</span>
<a name="l01153"></a>01153 <span class="comment"> *</span>
<a name="l01154"></a>01154 <span class="comment"> * @param egr - all call results for the experiment gene.</span>
<a name="l01155"></a>01155 <span class="comment"> * @param ttm - the translation table</span>
<a name="l01156"></a>01156 <span class="comment"> * @param cRow - has the already created known and unkown haplotype calls</span>
<a name="l01157"></a>01157 <span class="comment"> * @param ttmRow - the row of the current assay id in question.</span>
<a name="l01158"></a>01158 <span class="comment"> * @param numKnownCalls - the number of calls in cRow-&gt;m_knownCall std::string.</span>
<a name="l01159"></a>01159 <span class="comment"> * @param numUnknownCalls - the number of calls in cRow-&gt;m_unknownCall std::string.</span>
<a name="l01160"></a>01160 <span class="comment"> * @param geneInterpretationCode - returned.</span>
<a name="l01161"></a>01161 <span class="comment"> _</span>
<a name="l01162"></a>01162 <span class="comment"> * @return - geneInterpretationCode is filled in.</span>
<a name="l01163"></a>01163 <span class="comment"> */</span>
<a name="l01164"></a>01164 <span class="comment">/*****************************************************************************/</span>
<a name="l01165"></a>01165 <span class="keywordtype">void</span>  TranslationExperimentReport::_generateInterpretationCode(
<a name="l01166"></a>01166   <span class="keyword">const</span> <a class="code" href="classExperimentGeneResults.html">ExperimentGeneResults</a> &amp; egr,
<a name="l01167"></a>01167   <a class="code" href="classTranslationTableModel.html">TranslationTableModel</a> &amp; ttm,
<a name="l01168"></a>01168   <a class="code" href="classTranslationRow.html">TranslationRow</a> * cRow,
<a name="l01169"></a>01169   <span class="keywordtype">int</span> ttmRow,
<a name="l01170"></a>01170   <span class="keywordtype">int</span> numKnownCalls,
<a name="l01171"></a>01171   <span class="keywordtype">int</span> numUnknownCalls,
<a name="l01172"></a>01172   std::map&lt; std::string, std::string&gt; &amp; geneInterpretationCode,
<a name="l01173"></a>01173   <span class="keywordtype">int</span> geneCopyNumber)
<a name="l01174"></a>01174 {
<a name="l01175"></a>01175 
<a name="l01176"></a>01176   <span class="keywordtype">int</span> headerRow = ttm.<a class="code" href="classTranslationTableModel.html#abd5a952b613ca007ed4fc4ac4e12f524" title="TranslationTableModel::getHeaderRowgetHeaderRow: Synopsis:">getHeaderRow</a>(ttmRow);
<a name="l01177"></a>01177   std::map&lt; std::string , std::vector&lt;std::string&gt; &gt; haplotypeDefiningBases;
<a name="l01178"></a>01178   std::map&lt; std::string, std::vector&lt;std::string&gt; &gt;  probeSetDefiningBases;
<a name="l01179"></a>01179 
<a name="l01180"></a>01180   <span class="comment">// NoHAP, scan the translation table for &#39;Haplotype&#39; = Y.</span>
<a name="l01181"></a>01181   <span class="keywordtype">bool</span> geneHasHaplotype = <span class="keyword">false</span>;
<a name="l01182"></a>01182   <span class="keywordtype">bool</span> hasNCOrPRA       = <span class="keyword">false</span>;
<a name="l01183"></a>01183   <span class="keywordtype">bool</span> isCopyNumberZero = (geneCopyNumber == 0);
<a name="l01184"></a>01184   <span class="keywordtype">bool</span> zeroCopyCallHasNotAvailable  = <span class="keyword">false</span>;
<a name="l01185"></a>01185   
<a name="l01186"></a>01186   <span class="keywordflow">for</span> (<span class="keywordtype">int</span> row = headerRow + 1;
<a name="l01187"></a>01187        (row &lt; ttm.size()) &amp;&amp; (ttm.m_rows[row][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_GENE)] == cRow-&gt;m_gene); row++) {
<a name="l01188"></a>01188 
<a name="l01189"></a>01189     <span class="keywordflow">if</span> ( ttm.<a class="code" href="classTranslationTableModel.html#a590375d07f5f5bf813275858d0466114" title="TranslationTableModel::ignoreRow Synopsis: A setter and getter for ignoring rows in the translation t...">ignoreRow</a>(row) ) {
<a name="l01190"></a>01190       <span class="keywordflow">continue</span>;
<a name="l01191"></a>01191     }
<a name="l01192"></a>01192     std::string probeSet = ttm.m_rows[row][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_PROBE_SET_ID)];
<a name="l01193"></a>01193     std::map&lt; std::string, int&gt;::const_iterator itSI = egr.m_probeSetToMarkerCallResults.find(probeSet);
<a name="l01194"></a>01194     
<a name="l01195"></a>01195     <span class="keywordflow">if</span> (ttm.m_rows[row][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_HAPLOTYPE)] == <span class="stringliteral">&quot;Y&quot;</span>) {
<a name="l01196"></a>01196 
<a name="l01197"></a>01197       geneHasHaplotype = <span class="keyword">true</span>;
<a name="l01198"></a>01198 
<a name="l01199"></a>01199       <span class="keywordflow">if</span> (itSI !=  egr.m_probeSetToMarkerCallResults.end()) {
<a name="l01200"></a>01200 
<a name="l01201"></a>01201         <span class="keywordtype">int</span> callIndex = itSI-&gt;second;
<a name="l01202"></a>01202 
<a name="l01203"></a>01203         <span class="comment">// NOTE: Only need to check one chromatid because if one is</span>
<a name="l01204"></a>01204         <span class="comment">// a wild card it is the case they will both be wildcard.</span>
<a name="l01205"></a>01205         <span class="comment">// Wildcard is defined as either No Call, Not Available or Possible Rare Allele.</span>
<a name="l01206"></a>01206 
<a name="l01207"></a>01207         zeroCopyCallHasNotAvailable = zeroCopyCallHasNotAvailable ||
<a name="l01208"></a>01208           egr.m_callResults[callIndex].m_experimentChromatid1.m_hasZeroCopyNumberNotAvailable;
<a name="l01209"></a>01209         <span class="keywordflow">if</span> (egr.m_callResults[callIndex].m_experimentChromatid1.m_ceSet.size() &gt; 0) {
<a name="l01210"></a>01210           <span class="keywordflow">if</span> (zeroCopyCallHasNotAvailable || <a class="code" href="classCallElement.html#aae6a8be2dfee570fe4ddd451cb8a74c3" title="CallElement::isWildCardBase Synopsis:">CallElement::isWildCardBase</a>(egr.m_callResults[callIndex].m_experimentChromatid1.m_ceSet.begin()-&gt;second.m_bases[0])) {
<a name="l01211"></a>01211             hasNCOrPRA = <span class="keyword">true</span>;
<a name="l01212"></a>01212           }
<a name="l01213"></a>01213         }
<a name="l01214"></a>01214       }
<a name="l01215"></a>01215     }
<a name="l01216"></a>01216   }
<a name="l01217"></a>01217 
<a name="l01218"></a>01218   <span class="keywordflow">if</span> (!geneHasHaplotype &amp;&amp; !isCopyNumberZero){
<a name="l01219"></a>01219     geneInterpretationCode[cRow-&gt;m_gene] =  INTERPRETATION_CODES[TranslationInterpretationCodeMap::NoHAP].m_reportString;
<a name="l01220"></a>01220     <span class="keywordflow">return</span>;
<a name="l01221"></a>01221   }
<a name="l01222"></a>01222 
<a name="l01223"></a>01223   <span class="comment">// NC/PCRA, overrides all other codes. If there is a No Call or</span>
<a name="l01224"></a>01224   <span class="comment">// Possible Rare Allele then that wins.</span>
<a name="l01225"></a>01225   <span class="keywordflow">if</span> (hasNCOrPRA ) {
<a name="l01226"></a>01226     <span class="comment">//    if ( !geneHasHaplotype &amp;&amp; isCopyNumberZero ) {</span>
<a name="l01227"></a>01227     <span class="keywordflow">if</span> ( isCopyNumberZero ) {
<a name="l01228"></a>01228       geneInterpretationCode[cRow-&gt;m_gene] = INTERPRETATION_CODES[TranslationInterpretationCodeMap::UNIQ].m_reportString;
<a name="l01229"></a>01229     }
<a name="l01230"></a>01230     <span class="keywordflow">else</span> {
<a name="l01231"></a>01231       geneInterpretationCode[cRow-&gt;m_gene] = INTERPRETATION_CODES[TranslationInterpretationCodeMap::NC_PRA_NA].m_reportString;
<a name="l01232"></a>01232 
<a name="l01233"></a>01233     }
<a name="l01234"></a>01234     <span class="keywordflow">return</span>;
<a name="l01235"></a>01235   }
<a name="l01236"></a>01236 
<a name="l01237"></a>01237 
<a name="l01238"></a>01238   <span class="comment">// Call count. The rest of the codes are dependent on the count of</span>
<a name="l01239"></a>01239   <span class="comment">// known and unkown calls.</span>
<a name="l01240"></a>01240 
<a name="l01241"></a>01241   <span class="comment">// UNIQ</span>
<a name="l01242"></a>01242 
<a name="l01243"></a>01243   <span class="keywordflow">if</span> (((numKnownCalls == 1) &amp;&amp; (numUnknownCalls == 0)) ||
<a name="l01244"></a>01244       (!geneHasHaplotype &amp;&amp; isCopyNumberZero)) {
<a name="l01245"></a>01245     geneInterpretationCode[cRow-&gt;m_gene] = INTERPRETATION_CODES[TranslationInterpretationCodeMap::UNIQ].m_reportString;
<a name="l01246"></a>01246     <span class="keywordflow">return</span>;
<a name="l01247"></a>01247   }
<a name="l01248"></a>01248 
<a name="l01249"></a>01249   <span class="comment">// UNIQ+UNK</span>
<a name="l01250"></a>01250 
<a name="l01251"></a>01251   <span class="keywordflow">if</span> (numKnownCalls == 1)  {
<a name="l01252"></a>01252     geneInterpretationCode[cRow-&gt;m_gene] = INTERPRETATION_CODES[TranslationInterpretationCodeMap::UNIQ_UNK].m_reportString;
<a name="l01253"></a>01253    <span class="keywordflow">return</span>;
<a name="l01254"></a>01254   }
<a name="l01255"></a>01255 
<a name="l01256"></a>01256   <span class="comment">//MULTI</span>
<a name="l01257"></a>01257   <span class="keywordflow">if</span> ((numKnownCalls &gt; 1) &amp;&amp; (numUnknownCalls == 0)) {
<a name="l01258"></a>01258     geneInterpretationCode[cRow-&gt;m_gene] = INTERPRETATION_CODES[TranslationInterpretationCodeMap::MULT].m_reportString;
<a name="l01259"></a>01259     <span class="keywordflow">return</span>;
<a name="l01260"></a>01260   }
<a name="l01261"></a>01261 
<a name="l01262"></a>01262   <span class="comment">//MULTI+UNK</span>
<a name="l01263"></a>01263 
<a name="l01264"></a>01264   <span class="keywordflow">if</span> ((numKnownCalls &gt; 1)) {
<a name="l01265"></a>01265     geneInterpretationCode[cRow-&gt;m_gene] = INTERPRETATION_CODES[TranslationInterpretationCodeMap::MULT_UNK].m_reportString;
<a name="l01266"></a>01266     <span class="keywordflow">return</span>;
<a name="l01267"></a>01267   }
<a name="l01268"></a>01268 
<a name="l01269"></a>01269   <span class="comment">// UNDH</span>
<a name="l01270"></a>01270 
<a name="l01271"></a>01271   <span class="keywordflow">if</span> (numUnknownCalls &gt; 0) {
<a name="l01272"></a>01272     geneInterpretationCode[cRow-&gt;m_gene] = INTERPRETATION_CODES[TranslationInterpretationCodeMap::UNDH].m_reportString;
<a name="l01273"></a>01273     <span class="keywordflow">return</span>;
<a name="l01274"></a>01274   }
<a name="l01275"></a>01275 
<a name="l01276"></a>01276 
<a name="l01277"></a>01277   <span class="keywordflow">return</span>;
<a name="l01278"></a>01278 
<a name="l01279"></a>01279 }
<a name="l01280"></a>01280 <span class="comment">// end TranslationExperimentReport::_generateInterpretationCode</span>
<a name="l01281"></a>01281 <span class="comment">/*****************************************************************************/</span>
<a name="l01282"></a>01282 <span class="comment">/*****************************************************************************/</span>
<a name="l01283"></a>01283 <span class="comment">/***</span>
<a name="l01284"></a>01284 <span class="comment"> * TranslationExperimentReport::_generateReferenceBase:</span>
<a name="l01285"></a>01285 <span class="comment"> * Synopsis:</span>
<a name="l01286"></a>01286 <span class="comment"> *</span>
<a name="l01287"></a>01287 <span class="comment"> * Helper function to TranslationExperimentReport::generate, broken</span>
<a name="l01288"></a>01288 <span class="comment"> * out for readability only and not for reuse. </span>
<a name="l01289"></a>01289 <span class="comment"> *</span>
<a name="l01290"></a>01290 <span class="comment"> * The reference base is simply the translation table base in the</span>
<a name="l01291"></a>01291 <span class="comment"> * &quot;Reference&quot; column except when a report allele doesn&#39;t agree.</span>
<a name="l01292"></a>01292 <span class="comment"> *</span>
<a name="l01293"></a>01293 <span class="comment"> *</span>
<a name="l01294"></a>01294 <span class="comment"> * @param ttmRow - the row in question being reported in.</span>
<a name="l01295"></a>01295 <span class="comment"> * @param ttm - the in memory copy of the Translation Table</span>
<a name="l01296"></a>01296 <span class="comment"> * @param cRow - to be returned output row data structure</span>
<a name="l01297"></a>01297 <span class="comment"> *</span>
<a name="l01298"></a>01298 <span class="comment"> * @return - sets cRow-&gt;m_referenceBase, cRow-&gt;m_referenceReportBase</span>
<a name="l01299"></a>01299 <span class="comment"> */</span>
<a name="l01300"></a>01300 <span class="comment">/*****************************************************************************/</span>
<a name="l01301"></a>01301 <span class="keywordtype">void</span> TranslationExperimentReport::_generateReferenceBase(
<a name="l01302"></a>01302     <span class="keywordtype">int</span> ttRow,
<a name="l01303"></a>01303     <a class="code" href="classTranslationRow.html">TranslationRow</a> * cRow,
<a name="l01304"></a>01304     <a class="code" href="classTranslationTableModel.html">TranslationTableModel</a> &amp; ttm )
<a name="l01305"></a>01305 {
<a name="l01306"></a>01306 
<a name="l01307"></a>01307   cRow-&gt;m_referenceBase = ttm.m_rows[ttRow][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_REFERENCE)];
<a name="l01308"></a>01308 
<a name="l01309"></a>01309 
<a name="l01310"></a>01310 
<a name="l01311"></a>01311   cRow-&gt;m_referenceReportBase = ttm.<a class="code" href="classTranslationTableModel.html#a0d5cf32bb7c3a56543767bf15b51834b" title="TranslationTableModel::getReportAllele Synopsis:">getReportAllele</a>( cRow-&gt;m_probeSet, cRow-&gt;m_referenceBase);
<a name="l01312"></a>01312 
<a name="l01313"></a>01313   <span class="keywordflow">return</span>;
<a name="l01314"></a>01314 
<a name="l01315"></a>01315 }
<a name="l01316"></a>01316 <span class="comment">// end TranslationExperimentReport::_generateReferenceBase</span>
<a name="l01317"></a>01317 <span class="comment">/*****************************************************************************/</span>
<a name="l01318"></a>01318 <span class="comment">/*****************************************************************************/</span>
<a name="l01319"></a>01319 <span class="comment">/***</span>
<a name="l01320"></a>01320 <span class="comment"> * TranslationExperimentReport::_getFirstCHPFileRoot</span>
<a name="l01321"></a>01321 <span class="comment"> * Synopsis:</span>
<a name="l01322"></a>01322 <span class="comment"> *</span>
<a name="l01323"></a>01323 <span class="comment"> * Helper function for the various &quot;getXXXXReportName&quot; methods.</span>
<a name="l01324"></a>01324 <span class="comment"> *</span>
<a name="l01325"></a>01325 <span class="comment"> * If the base name option &quot;-b&quot; is not passed in then the</span>
<a name="l01326"></a>01326 <span class="comment"> * first CHP file name root is used as the root for all report files.</span>
<a name="l01327"></a>01327 <span class="comment"> * </span>
<a name="l01328"></a>01328 <span class="comment"> * This method is used to generate report base names when no report</span>
<a name="l01329"></a>01329 <span class="comment"> * base name is provided.</span>
<a name="l01330"></a>01330 <span class="comment"> *</span>
<a name="l01331"></a>01331 <span class="comment"> * @param rte - contains the option &#39;inputExperimentListFiles&#39;.</span>
<a name="l01332"></a>01332 <span class="comment"> *</span>
<a name="l01333"></a>01333 <span class="comment"> *</span>
<a name="l01334"></a>01334 <span class="comment"> * @return - the root of the file, or the empty std::string if not found.</span>
<a name="l01335"></a>01335 <span class="comment"> */</span>
<a name="l01336"></a>01336 <span class="comment">/*****************************************************************************/</span>
<a name="l01337"></a>01337 std::string TranslationExperimentReport::_getFirstCHPFileRoot(<span class="keyword">const</span> <a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp; rte)
<a name="l01338"></a>01338 {
<a name="l01339"></a>01339 
<a name="l01340"></a>01340   std::string firstCHPFileRoot = <span class="stringliteral">&quot;&quot;</span>;
<a name="l01341"></a>01341 
<a name="l01342"></a>01342   <span class="keywordflow">if</span> (rte.m_adtOpts.m_inputExperimentFiles.size() &gt; 0) {
<a name="l01343"></a>01343 
<a name="l01344"></a>01344     pcrecpp::RE(<span class="stringliteral">&quot;^(.*)(?:\\.[Cc][Hh][Pp]$)&quot;</span>).FullMatch(<a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(rte.m_adtOpts.m_inputExperimentFiles[0]), &amp;firstCHPFileRoot);
<a name="l01345"></a>01345   }
<a name="l01346"></a>01346 
<a name="l01347"></a>01347 
<a name="l01348"></a>01348   <span class="keywordflow">return</span> firstCHPFileRoot;
<a name="l01349"></a>01349 
<a name="l01350"></a>01350 
<a name="l01351"></a>01351 }
<a name="l01352"></a>01352 <span class="comment">//end TranslationExperimentReport::_getFirstCHPFileRoot</span>
<a name="l01353"></a>01353 <span class="comment">/*****************************************************************************/</span>
<a name="l01354"></a>01354 <span class="comment">/*****************************************************************************/</span>
<a name="l01355"></a>01355 <span class="comment">/***</span>
<a name="l01356"></a>01356 <span class="comment"> * TranslationExperimentReport::_generateVariantBase:</span>
<a name="l01357"></a>01357 <span class="comment"> * Synopsis:</span>
<a name="l01358"></a>01358 <span class="comment"> *</span>
<a name="l01359"></a>01359 <span class="comment"> * Helper function to TranslationExperimentReport::generate, broken</span>
<a name="l01360"></a>01360 <span class="comment"> * out for readability only and not for reuse.</span>
<a name="l01361"></a>01361 <span class="comment"> * </span>
<a name="l01362"></a>01362 <span class="comment"> * The variant base starts as the &quot;Variant&quot; column in the translation</span>
<a name="l01363"></a>01363 <span class="comment"> * table. For multi-allelic probe sets then the variant base is</span>
<a name="l01364"></a>01364 <span class="comment"> * a comma separated list. In addition report alleles override</span>
<a name="l01365"></a>01365 <span class="comment"> * the translation table. </span>
<a name="l01366"></a>01366 <span class="comment"> *</span>
<a name="l01367"></a>01367 <span class="comment"> *</span>
<a name="l01368"></a>01368 <span class="comment"> * </span>
<a name="l01369"></a>01369 <span class="comment"> * </span>
<a name="l01370"></a>01370 <span class="comment"> * @param ttmRow - the row in question being reported in.</span>
<a name="l01371"></a>01371 <span class="comment"> * @param ttm - the in memory copy of the Translation Table</span>
<a name="l01372"></a>01372 <span class="comment"> * @param cRow - to be returned output row data structure</span>
<a name="l01373"></a>01373 <span class="comment"> * @param isMultiAllelicRow - if this is not the first time this probe set has been seen. </span>
<a name="l01374"></a>01374 <span class="comment"> *</span>
<a name="l01375"></a>01375 <span class="comment"> * @return - sets cRow-&gt;m_variantBase, cRow-&gt;m_variantReportBase</span>
<a name="l01376"></a>01376 <span class="comment"> */</span>
<a name="l01377"></a>01377 <span class="comment">/*****************************************************************************/</span>
<a name="l01378"></a>01378 <span class="keywordtype">void</span> TranslationExperimentReport::_generateVariantBase(
<a name="l01379"></a>01379   <span class="keywordtype">int</span> ttRow,
<a name="l01380"></a>01380   <a class="code" href="classTranslationRow.html">TranslationRow</a> * cRow,
<a name="l01381"></a>01381   <a class="code" href="classTranslationTableModel.html">TranslationTableModel</a> &amp; ttm,
<a name="l01382"></a>01382   <span class="keywordtype">bool</span> isMultiAllelicRow )
<a name="l01383"></a>01383 {
<a name="l01384"></a>01384 
<a name="l01385"></a>01385 
<a name="l01386"></a>01386   std::string variantBase = ttm.m_rows[ttRow][ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_VARIANT)];
<a name="l01387"></a>01387 
<a name="l01388"></a>01388   <span class="keywordflow">if</span> (isMultiAllelicRow) {
<a name="l01389"></a>01389     cRow-&gt;m_variantBase = cRow-&gt;m_variantBase + <span class="stringliteral">&quot;,&quot;</span> + variantBase;
<a name="l01390"></a>01390   } <span class="keywordflow">else</span> {
<a name="l01391"></a>01391     <a class="code" href="Err_8h.html#ad024208f4ba6a4e75df4499ea1639a05" title="Calls Err::apt_err_assert with the filename and linenumber set. /// We want to avoid evaluating the m...">APT_ERR_ASSERT</a>(cRow-&gt;m_variantBase.empty(), <span class="stringliteral">&quot;&quot;</span>);
<a name="l01392"></a>01392     cRow-&gt;m_variantBase = variantBase;
<a name="l01393"></a>01393   }
<a name="l01394"></a>01394 
<a name="l01395"></a>01395   cRow-&gt;m_variantUncalledBase = variantBase;
<a name="l01396"></a>01396 
<a name="l01397"></a>01397   variantBase = ttm.<a class="code" href="classTranslationTableModel.html#a0d5cf32bb7c3a56543767bf15b51834b" title="TranslationTableModel::getReportAllele Synopsis:">getReportAllele</a>( cRow-&gt;m_probeSet, variantBase );
<a name="l01398"></a>01398 
<a name="l01399"></a>01399   <span class="keywordflow">if</span> (isMultiAllelicRow) {
<a name="l01400"></a>01400     cRow-&gt;m_variantReportBase = cRow-&gt;m_variantReportBase + <span class="stringliteral">&quot;,&quot;</span> + variantBase;
<a name="l01401"></a>01401   } <span class="keywordflow">else</span> {
<a name="l01402"></a>01402     <a class="code" href="Err_8h.html#ad024208f4ba6a4e75df4499ea1639a05" title="Calls Err::apt_err_assert with the filename and linenumber set. /// We want to avoid evaluating the m...">APT_ERR_ASSERT</a>(cRow-&gt;m_variantReportBase.empty(), <span class="stringliteral">&quot;&quot;</span>);
<a name="l01403"></a>01403     cRow-&gt;m_variantReportBase = variantBase;
<a name="l01404"></a>01404   }
<a name="l01405"></a>01405 
<a name="l01406"></a>01406   <span class="keywordflow">return</span>;
<a name="l01407"></a>01407 
<a name="l01408"></a>01408 }
<a name="l01409"></a>01409 <span class="comment">// end TranslationExperimentReport::_generateVariantBase</span>
<a name="l01410"></a>01410 <span class="comment">/*****************************************************************************/</span>
<a name="l01411"></a>01411 <span class="comment">/***</span>
<a name="l01412"></a>01412 <span class="comment"> * TranslationExperimentReport::_generateRelevantAlleles:</span>
<a name="l01413"></a>01413 <span class="comment"> * Synopsis:</span>
<a name="l01414"></a>01414 <span class="comment"> *</span>
<a name="l01415"></a>01415 <span class="comment"> * Set the &quot;m_relevantAlleles&quot; attribute of the TranslationRow object.</span>
<a name="l01416"></a>01416 <span class="comment"> * Broken out of &quot;generate&quot; for readability purposes only.</span>
<a name="l01417"></a>01417 <span class="comment"> *</span>
<a name="l01418"></a>01418 <span class="comment"> * The relevant alleles is the list of all alleles that this probe set</span>
<a name="l01419"></a>01419 <span class="comment"> * does not participate in. </span>
<a name="l01420"></a>01420 <span class="comment"> *</span>
<a name="l01421"></a>01421 <span class="comment"> * @param ttmRow - the row in question being reported in.</span>
<a name="l01422"></a>01422 <span class="comment"> * @param ttm - the in memory copy of the Translation Table</span>
<a name="l01423"></a>01423 <span class="comment"> * @param cRow - to be returned output row data structure</span>
<a name="l01424"></a>01424 <span class="comment"> * @param isMultiAllelicRow - if this is not the first time this probe set has been seen. </span>
<a name="l01425"></a>01425 <span class="comment"> *</span>
<a name="l01426"></a>01426 <span class="comment"> * @return - cRow.m_relevantAlleles filled in.</span>
<a name="l01427"></a>01427 <span class="comment"> */</span>
<a name="l01428"></a>01428 <span class="comment">/*****************************************************************************/</span>
<a name="l01429"></a>01429 <span class="keywordtype">void</span> TranslationExperimentReport::_generateRelevantAlleles(
<a name="l01430"></a>01430   <a class="code" href="classTranslationTableModel.html">TranslationTableModel</a> &amp; ttm,
<a name="l01431"></a>01431   <a class="code" href="classTranslationRow.html">TranslationRow</a> * cRow,
<a name="l01432"></a>01432   <span class="keywordtype">int</span> ttmRow,
<a name="l01433"></a>01433   <span class="keywordtype">bool</span> isMultiAllelicRow)
<a name="l01434"></a>01434 {
<a name="l01435"></a>01435   <span class="comment">// RELEVANT ALLELES</span>
<a name="l01436"></a>01436   <span class="keywordtype">int</span> headerRow = ttm.<a class="code" href="classTranslationTableModel.html#abd5a952b613ca007ed4fc4ac4e12f524" title="TranslationTableModel::getHeaderRowgetHeaderRow: Synopsis:">getHeaderRow</a>(ttmRow);
<a name="l01437"></a>01437 
<a name="l01438"></a>01438   std::string relevantAlleles;
<a name="l01439"></a>01439   <span class="keywordflow">for</span> (<span class="keywordtype">int</span> j = ttm.<a class="code" href="classTranslationTableModel.html#a8e519cfda65a7d4399dc0de52f7a4041" title="TranslationTableModel::getColumnIndex: Synopsis: DMET2 &amp;amp; DMET3 columns can be retrieved via this ...">getColumnIndex</a>(ADT_DMET3_TT_ALLELE_START) + 1; j &lt; ttm.m_rows[ttmRow].size(); j++) {
<a name="l01440"></a>01440     <span class="keywordflow">if</span> (!ttm.m_rows[ttmRow][j].empty() &amp;&amp; ((cRow-&gt;m_haplotypeMarker == <span class="stringliteral">&quot;N&quot;</span>) || (ttm.m_rows[headerRow][j].c_str())[0] != <span class="charliteral">&#39;#&#39;</span>)) {
<a name="l01441"></a>01441       relevantAlleles = relevantAlleles + ttm.m_rows[headerRow][j] + <span class="stringliteral">&quot;,&quot;</span>;
<a name="l01442"></a>01442     }
<a name="l01443"></a>01443   }
<a name="l01444"></a>01444   <span class="comment">// Chop off the trailing comma.</span>
<a name="l01445"></a>01445   <span class="keywordflow">if</span> (!relevantAlleles.empty()) {
<a name="l01446"></a>01446     relevantAlleles.resize(relevantAlleles.size() - 1);
<a name="l01447"></a>01447   }
<a name="l01448"></a>01448   <span class="keywordflow">if</span> (isMultiAllelicRow) {
<a name="l01449"></a>01449     <span class="keywordflow">if</span> (! relevantAlleles.empty()) {
<a name="l01450"></a>01450       cRow-&gt;m_relevantAlleles = cRow-&gt;m_relevantAlleles  + <span class="stringliteral">&quot;,&quot;</span> + relevantAlleles;
<a name="l01451"></a>01451     }
<a name="l01452"></a>01452   } <span class="keywordflow">else</span> {
<a name="l01453"></a>01453     cRow-&gt;m_relevantAlleles = relevantAlleles;
<a name="l01454"></a>01454   }
<a name="l01455"></a>01455 
<a name="l01456"></a>01456   <span class="keywordflow">if</span> (!cRow-&gt;m_relevantAlleles.empty() &amp;&amp;
<a name="l01457"></a>01457       cRow-&gt;m_relevantAlleles[ cRow-&gt;m_relevantAlleles.size() - 1 ] == <span class="charliteral">&#39;,&#39;</span>) {
<a name="l01458"></a>01458     cRow-&gt;m_relevantAlleles.resize(cRow-&gt;m_relevantAlleles.size() - 1);
<a name="l01459"></a>01459   }
<a name="l01460"></a>01460   <span class="keywordflow">return</span>;
<a name="l01461"></a>01461 
<a name="l01462"></a>01462 }
<a name="l01463"></a>01463 <span class="comment">// end TranslationExperimentReport::_generateRelevantAlleles</span>
<a name="l01464"></a>01464 <span class="comment">/*****************************************************************************/</span>
<a name="l01465"></a>01465 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l01466"></a>01466 <span class="comment">/**</span>
<a name="l01467"></a>01467 <span class="comment"> * TranslationExperimentReport::_getNumColumns:</span>
<a name="l01468"></a>01468 <span class="comment"> * Synopsis:</span>
<a name="l01469"></a>01469 <span class="comment"> *</span>
<a name="l01470"></a>01470 <span class="comment"> *  Helper function to various other helper functions.</span>
<a name="l01471"></a>01471 <span class="comment"> *</span>
<a name="l01472"></a>01472 <span class="comment"> *  Returns either the number of fixed columns for the comprehensive or</span>
<a name="l01473"></a>01473 <span class="comment"> *  summary reports only. Not to be used for the uncalled report.</span>
<a name="l01474"></a>01474 <span class="comment"> *  Pass a boolean as true to  include the number of sample columns.</span>
<a name="l01475"></a>01475 <span class="comment"> *</span>
<a name="l01476"></a>01476 <span class="comment"> * </span>
<a name="l01477"></a>01477 <span class="comment"> * @param includeSampleColumns - boolean indicates whether to include the sample info columsn. </span>
<a name="l01478"></a>01478 <span class="comment"> * </span>
<a name="l01479"></a>01479 <span class="comment"> * @return - the number of columns</span>
<a name="l01480"></a>01480 <span class="comment"> */</span>
<a name="l01481"></a>01481 <span class="comment">/*****************************************************************************/</span>
<a name="l01482"></a>01482 <span class="keywordtype">int</span> TranslationExperimentReport::_getNumColumns(<span class="keywordtype">bool</span> includeSampleColumns)
<a name="l01483"></a>01483 {
<a name="l01484"></a>01484 
<a name="l01485"></a>01485   <span class="keywordflow">if</span> (includeSampleColumns) {
<a name="l01486"></a>01486     <a class="code" href="Err_8h.html#ad024208f4ba6a4e75df4499ea1639a05" title="Calls Err::apt_err_assert with the filename and linenumber set. /// We want to avoid evaluating the m...">APT_ERR_ASSERT</a>(m_totalColumnsWithSampleInfo &gt; 0, <span class="stringliteral">&quot;&quot;</span>);
<a name="l01487"></a>01487     <span class="keywordflow">return</span> m_totalColumnsWithSampleInfo;
<a name="l01488"></a>01488   }
<a name="l01489"></a>01489 
<a name="l01490"></a>01490   <span class="keywordflow">return</span> m_numFixedColumns;
<a name="l01491"></a>01491 }
<a name="l01492"></a>01492 <span class="comment">// end TranslationExperimentReport::_getNumColumns</span>
<a name="l01493"></a>01493 <span class="comment">/*****************************************************************************/</span>
<a name="l01494"></a>01494 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l01495"></a>01495 <span class="comment">/**</span>
<a name="l01496"></a>01496 <span class="comment"> * TranslationExperimentReport::_open:</span>
<a name="l01497"></a>01497 <span class="comment"> * Synopsis:</span>
<a name="l01498"></a>01498 <span class="comment"> *</span>
<a name="l01499"></a>01499 <span class="comment"> * Helper function to TranslationExperimentReport::generate that wraps</span>
<a name="l01500"></a>01500 <span class="comment"> * the open call to summary, comprehensive and the uncalled reports.</span>
<a name="l01501"></a>01501 <span class="comment"> *</span>
<a name="l01502"></a>01502 <span class="comment"> * The private variable &quot;m_openCalled&quot; protects against calling this</span>
<a name="l01503"></a>01503 <span class="comment"> * multiple times. </span>
<a name="l01504"></a>01504 <span class="comment"> * </span>
<a name="l01505"></a>01505 <span class="comment"> *</span>
<a name="l01506"></a>01506 <span class="comment"> * @param rte - the single instance RunTimeEnvironment. </span>
<a name="l01507"></a>01507 <span class="comment"> *</span>
<a name="l01508"></a>01508 <span class="comment"> * @return - opens the report files or throws exception.</span>
<a name="l01509"></a>01509 <span class="comment"> */</span>
<a name="l01510"></a>01510 <span class="comment">/*****************************************************************************/</span>
<a name="l01511"></a>01511 <span class="keywordtype">void</span> TranslationExperimentReport::_open(<span class="keyword">const</span> <a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp; rte)
<a name="l01512"></a>01512 {
<a name="l01513"></a>01513 
<a name="l01514"></a>01514   <span class="keywordflow">if</span> (m_openCalled) {
<a name="l01515"></a>01515     <span class="keywordflow">return</span>;
<a name="l01516"></a>01516   }
<a name="l01517"></a>01517 
<a name="l01518"></a>01518   std::vector&lt; std::string &gt; sampleHeaders;
<a name="l01519"></a>01519 
<a name="l01520"></a>01520   <span class="keywordflow">if</span> (m_sitm != NULL) {
<a name="l01521"></a>01521     sampleHeaders = m_sitm-&gt;<a class="code" href="classTranslationInputTsvTableModel.html#ad98c3dde1f638ee717d3a7c72cfdddc9" title="TranslationInputTsvTableModel::getHeaderAsVector Synopsis:">getHeaderAsVector</a>() ;
<a name="l01522"></a>01522   }
<a name="l01523"></a>01523 
<a name="l01524"></a>01524   m_guid = <a class="code" href="classaffxutil_1_1Guid.html#a2929c982cb9f3b5f6c5d7d818a8e6ac4">affxutil::Guid::GenerateNewGuid</a>();
<a name="l01525"></a>01525 
<a name="l01526"></a>01526   _openComprehensiveFile(rte, sampleHeaders);
<a name="l01527"></a>01527 
<a name="l01528"></a>01528   _openSummaryFile(rte, sampleHeaders);
<a name="l01529"></a>01529 
<a name="l01530"></a>01530   _openUncalledFile(rte);
<a name="l01531"></a>01531 
<a name="l01532"></a>01532   m_openCalled = <span class="keyword">true</span>;
<a name="l01533"></a>01533 
<a name="l01534"></a>01534   m_experimentFileCount++;
<a name="l01535"></a>01535 
<a name="l01536"></a>01536   <span class="keywordflow">return</span>;
<a name="l01537"></a>01537 }
<a name="l01538"></a>01538 <span class="comment">// end TranslationExperimentReport::_open</span>
<a name="l01539"></a>01539 <span class="comment">/*****************************************************************************/</span>
<a name="l01540"></a>01540 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l01541"></a>01541 <span class="comment">/**</span>
<a name="l01542"></a>01542 <span class="comment"> * TranslationExperimentReport::_openComprehensiveFile</span>
<a name="l01543"></a>01543 <span class="comment"> * Synopsis:</span>
<a name="l01544"></a>01544 <span class="comment"> * </span>
<a name="l01545"></a>01545 <span class="comment"> * Helper function to TranslationExperimentReport::generate broken out</span>
<a name="l01546"></a>01546 <span class="comment"> * for readability.</span>
<a name="l01547"></a>01547 <span class="comment"> *</span>
<a name="l01548"></a>01548 <span class="comment"> * Open a TSV file for outputing comprehensive report results.</span>
<a name="l01549"></a>01549 <span class="comment"> * Opens the TSV file, writes file headers and column headers.</span>
<a name="l01550"></a>01550 <span class="comment"> *</span>
<a name="l01551"></a>01551 <span class="comment"> *</span>
<a name="l01552"></a>01552 <span class="comment"> * @param rte - the single instance RunTimeEnvironment.</span>
<a name="l01553"></a>01553 <span class="comment"> * @param sampleInfoHeaders - the dynamic sample headers that are not parsed</span>
<a name="l01554"></a>01554 <span class="comment"> *</span>
<a name="l01555"></a>01555 <span class="comment"> */</span>
<a name="l01556"></a>01556 <span class="comment">/*****************************************************************************/</span>
<a name="l01557"></a>01557 <span class="keywordtype">void</span> TranslationExperimentReport::_openComprehensiveFile(<span class="keyword">const</span> <a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp; rte, std::vector&lt; std::string &gt; &amp; sampleInfoHeaders)
<a name="l01558"></a>01558 {
<a name="l01559"></a>01559 
<a name="l01560"></a>01560   std::string completeName = <a class="code" href="classTranslationExperimentReport.html#ad531a53fc9e67cc0a5e4f1541cc7171e" title="TranslationExperimentReport::getComprehensiveReportName Synopsis:">getComprehensiveReportName</a>(rte) + COMPLETE_REPORT_FILE_EXT;
<a name="l01561"></a>01561   <span class="keywordflow">if</span> (Fs::fileExists(completeName))
<a name="l01562"></a>01562     <a class="code" href="classFs.html#a6b8f232817f933a42c21edde6ea55a3a" title="An error if it doesnt exist.">Fs::rm</a>(completeName, <span class="keyword">false</span>);
<a name="l01563"></a>01563 
<a name="l01564"></a>01564   <span class="comment">// describe our columns.</span>
<a name="l01565"></a>01565   <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; _getNumColumns(); i++) {
<a name="l01566"></a>01566     m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#aadc22863a63b9c6588b0f2dc61875809" title="User methods.">defineColumn</a>(0, i, COMPREHENSIVE_TSV_COLUMNS[i]);
<a name="l01567"></a>01567   }
<a name="l01568"></a>01568 
<a name="l01569"></a>01569 
<a name="l01570"></a>01570   m_totalColumnsWithSampleInfo = _getNumColumns() + sampleInfoHeaders.size();
<a name="l01571"></a>01571 
<a name="l01572"></a>01572   std::vector&lt; std::string &gt;::iterator itS  = sampleInfoHeaders.begin();
<a name="l01573"></a>01573   <span class="keywordflow">if</span> (itS != sampleInfoHeaders.end()) {
<a name="l01574"></a>01574     itS++;
<a name="l01575"></a>01575   }
<a name="l01576"></a>01576 
<a name="l01577"></a>01577   <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = _getNumColumns();
<a name="l01578"></a>01578        itS != sampleInfoHeaders.end(); i++, itS++) {
<a name="l01579"></a>01579     m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#aadc22863a63b9c6588b0f2dc61875809" title="User methods.">defineColumn</a>(0, i, *itS);
<a name="l01580"></a>01580   }
<a name="l01581"></a>01581 
<a name="l01582"></a>01582   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot; For research use only. Not for diagnostic purposes.&quot;</span>);
<a name="l01583"></a>01583   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;dmet3-report-guid&quot;</span>, m_guid);
<a name="l01584"></a>01584   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;Program&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(rte.m_programName));
<a name="l01585"></a>01585   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;Version&quot;</span>, ADT_VERSION);
<a name="l01586"></a>01586   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;Date&quot;</span>, <a class="code" href="classUtil.html#accf4f30492f3726aac3197a37b3f77cc" title="Get a timestamp string.">Util::getTimeStamp</a>());
<a name="l01587"></a>01587   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;TranslationFile&quot;</span>, rte.m_adtOpts.m_inputTTableFile);
<a name="l01588"></a>01588   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;MarkerList&quot;</span>, rte.m_adtOpts.m_inputMarkerListFile);
<a name="l01589"></a>01589   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;HaplotypeReportOption&quot;</span>, rte.m_adtOpts.m_useFirstDupAlleleDefHeaderText);
<a name="l01590"></a>01590 
<a name="l01591"></a>01591   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;GenotypeOverrideFile&quot;</span>, rte.m_adtOpts.m_inputGenotypeOverrideFile);
<a name="l01592"></a>01592   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot;Interpretation Code Description:&quot;</span>);
<a name="l01593"></a>01593   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot;UNIQ: Unique haplotype pair&quot;</span>);
<a name="l01594"></a>01594   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot;MULT: Multiple haplotype pairs possible due to phase ambiguity&quot;</span>);
<a name="l01595"></a>01595   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot;UNIQ+UNK: Unique annotated haplotype pair, with other haplotype pairs requiring unannotated haplotypes also possible&quot;</span>);
<a name="l01596"></a>01596   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot;MULT+UNK: Multiple annotated haplotype calls possible, with other haplotype pairs requiring unannotated haplotypes also possible&quot;</span>);
<a name="l01597"></a>01597   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot;UNDH: Only haplotype pairs with undefined haplotypes possible&quot;</span>);
<a name="l01598"></a>01598   <span class="comment">//  m_comprehensiveTsv.addHeaderComment(&quot;6_CN: Copy Number state indicates deletion or duplication alleles&quot;);</span>
<a name="l01599"></a>01599 
<a name="l01600"></a>01600   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot;NC/PRA/NA: NoCall, PossibleRareAllele, or NotAvailable call for one or more markers resulting in multiple haplotype pairs&quot;</span>);
<a name="l01601"></a>01601   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot;NoHAP: No haplotypes defined at this gene&quot;</span>);
<a name="l01602"></a>01602   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a1903489904b4e2f8f2a262b8f10cdb0e" title="Set the options up for &amp;quot;tsv&amp;quot; format and write the tsv-v1 header ///.">writeTsv_v1</a>(<a class="code" href="classTranslationExperimentReport.html#ad531a53fc9e67cc0a5e4f1541cc7171e" title="TranslationExperimentReport::getComprehensiveReportName Synopsis:">getComprehensiveReportName</a>(rte) + TEMP_REPORT_FILE_EXT);
<a name="l01603"></a>01603 
<a name="l01604"></a>01604 
<a name="l01605"></a>01605   <span class="keywordflow">return</span>;
<a name="l01606"></a>01606 
<a name="l01607"></a>01607 }
<a name="l01608"></a>01608 <span class="comment">// end TranslationExperimentReport::_openComprehensiveFile</span>
<a name="l01609"></a>01609 <span class="comment">/*****************************************************************************/</span>
<a name="l01610"></a>01610 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l01611"></a>01611 <span class="comment">/**</span>
<a name="l01612"></a>01612 <span class="comment"> * TranslationExperimentReport::_openSummaryFile</span>
<a name="l01613"></a>01613 <span class="comment"> * Synopsis:</span>
<a name="l01614"></a>01614 <span class="comment"> *</span>
<a name="l01615"></a>01615 <span class="comment"> * Helper function to TranslationExperimentReport::generate broken out</span>
<a name="l01616"></a>01616 <span class="comment"> * for readability.</span>
<a name="l01617"></a>01617 <span class="comment"> *</span>
<a name="l01618"></a>01618 <span class="comment"> * Open a TSV file for outputing summary report results.</span>
<a name="l01619"></a>01619 <span class="comment"> * Opens the TSV file, writes file headers and column headers.</span>
<a name="l01620"></a>01620 <span class="comment"> *</span>
<a name="l01621"></a>01621 <span class="comment"> *</span>
<a name="l01622"></a>01622 <span class="comment"> * @param rte - the single instance RunTimeEnvironment.</span>
<a name="l01623"></a>01623 <span class="comment"> * @param sampleInfoHeaders - the dynamic sample headers that are not parsed.</span>
<a name="l01624"></a>01624 <span class="comment"> */</span>
<a name="l01625"></a>01625 <span class="comment">/*****************************************************************************/</span>
<a name="l01626"></a>01626 <span class="keywordtype">void</span> TranslationExperimentReport::_openSummaryFile(<span class="keyword">const</span> <a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp; rte, std::vector&lt; std::string &gt; &amp; sampleInfoHeaders)
<a name="l01627"></a>01627 {
<a name="l01628"></a>01628 
<a name="l01629"></a>01629   std::string completeName = <a class="code" href="classTranslationExperimentReport.html#a0b9c78be8ed1c94edab2ecac262e3fb3" title="TranslationExperimentReport::getSummaryReportName Synopsis:">getSummaryReportName</a>(rte) + COMPLETE_REPORT_FILE_EXT;
<a name="l01630"></a>01630   <span class="keywordflow">if</span> (Fs::fileExists(completeName))
<a name="l01631"></a>01631     <a class="code" href="classFs.html#a6b8f232817f933a42c21edde6ea55a3a" title="An error if it doesnt exist.">Fs::rm</a>(completeName, <span class="keyword">false</span>);
<a name="l01632"></a>01632 
<a name="l01633"></a>01633   <span class="comment">// describe our columns.</span>
<a name="l01634"></a>01634   <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; _getNumColumns(); i++) {
<a name="l01635"></a>01635     m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#aadc22863a63b9c6588b0f2dc61875809" title="User methods.">defineColumn</a>(0, i, COMPREHENSIVE_TSV_COLUMNS[i]);
<a name="l01636"></a>01636   }
<a name="l01637"></a>01637 
<a name="l01638"></a>01638 
<a name="l01639"></a>01639   m_totalColumnsWithSampleInfo = _getNumColumns() + sampleInfoHeaders.size();
<a name="l01640"></a>01640 
<a name="l01641"></a>01641   std::vector&lt; std::string &gt;::iterator itS  = sampleInfoHeaders.begin();
<a name="l01642"></a>01642   <span class="keywordflow">if</span> (itS != sampleInfoHeaders.end()) {
<a name="l01643"></a>01643     itS++;
<a name="l01644"></a>01644   }
<a name="l01645"></a>01645 
<a name="l01646"></a>01646   <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = _getNumColumns();
<a name="l01647"></a>01647        itS != sampleInfoHeaders.end(); i++, itS++) {
<a name="l01648"></a>01648     m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#aadc22863a63b9c6588b0f2dc61875809" title="User methods.">defineColumn</a>(0, i, *itS);
<a name="l01649"></a>01649   }
<a name="l01650"></a>01650 
<a name="l01651"></a>01651   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot; For research use only. Not for diagnostic purposes.&quot;</span>);
<a name="l01652"></a>01652   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;dmet3-report-guid&quot;</span>, m_guid);
<a name="l01653"></a>01653   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;Program&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(rte.m_programName));
<a name="l01654"></a>01654   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;Version&quot;</span>, ADT_VERSION);
<a name="l01655"></a>01655   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;Date&quot;</span>, <a class="code" href="classUtil.html#accf4f30492f3726aac3197a37b3f77cc" title="Get a timestamp string.">Util::getTimeStamp</a>());
<a name="l01656"></a>01656   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;TranslationFile&quot;</span>, rte.m_adtOpts.m_inputTTableFile);
<a name="l01657"></a>01657   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;MarkerList&quot;</span>, rte.m_adtOpts.m_inputMarkerListFile);
<a name="l01658"></a>01658   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;HaplotypeReportOption&quot;</span>, rte.m_adtOpts.m_useFirstDupAlleleDefHeaderText);
<a name="l01659"></a>01659   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;GenotypeOverrideFile&quot;</span>, rte.m_adtOpts.m_inputGenotypeOverrideFile);
<a name="l01660"></a>01660   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot;Interpretation Code Description:&quot;</span>);
<a name="l01661"></a>01661   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot;UNIQ: Unique haplotype pair&quot;</span>);
<a name="l01662"></a>01662   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot;MULT: Multiple haplotype pairs possible due to phase ambiguity&quot;</span>);
<a name="l01663"></a>01663   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot;UNIQ+UNK: Unique annotated haplotype pair, with other haplotype pairs requiring unannotated haplotypes also possible&quot;</span>);
<a name="l01664"></a>01664   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot;MULT+UNK: Multiple annotated haplotype calls possible, with other haplotype pairs requiring unannotated haplotypes also possible&quot;</span>);
<a name="l01665"></a>01665   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot;UNDH: Only haplotype pairs with undefined haplotypes possible&quot;</span>);
<a name="l01666"></a>01666   <span class="comment">//  m_summaryTsv.addHeaderComment(&quot;6_CN: Copy Number state indicates deletion or duplication alleles&quot;);</span>
<a name="l01667"></a>01667 
<a name="l01668"></a>01668   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot;NC/PRA/NotAvailable: NoCall, PossibleRareAllele, or NotAvailable call for one or more markers resulting in multiple haplotype pairs&quot;</span>);
<a name="l01669"></a>01669   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot;NoHAP: No haplotypes defined at this gene&quot;</span>);
<a name="l01670"></a>01670   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a1903489904b4e2f8f2a262b8f10cdb0e" title="Set the options up for &amp;quot;tsv&amp;quot; format and write the tsv-v1 header ///.">writeTsv_v1</a>(<a class="code" href="classTranslationExperimentReport.html#a0b9c78be8ed1c94edab2ecac262e3fb3" title="TranslationExperimentReport::getSummaryReportName Synopsis:">getSummaryReportName</a>(rte) + TEMP_REPORT_FILE_EXT);
<a name="l01671"></a>01671 
<a name="l01672"></a>01672 
<a name="l01673"></a>01673   <span class="keywordflow">return</span>;
<a name="l01674"></a>01674 
<a name="l01675"></a>01675 }
<a name="l01676"></a>01676 <span class="comment">// end RegressionExperimentReport::_openSummaryFile</span>
<a name="l01677"></a>01677 <span class="comment">/*****************************************************************************/</span>
<a name="l01678"></a>01678 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l01679"></a>01679 <span class="comment">/**</span>
<a name="l01680"></a>01680 <span class="comment"> * TranslationExperimentReport::_openUncalledFile</span>
<a name="l01681"></a>01681 <span class="comment"> *</span>
<a name="l01682"></a>01682 <span class="comment"> * Helper function to TranslationExperimentReport::generate broken out</span>
<a name="l01683"></a>01683 <span class="comment"> * for readability.</span>
<a name="l01684"></a>01684 <span class="comment"> *</span>
<a name="l01685"></a>01685 <span class="comment"> * Open a TSV file for outputing uncalled report results.</span>
<a name="l01686"></a>01686 <span class="comment"> * Opens the TSV file, writes file headers and column headers.</span>
<a name="l01687"></a>01687 <span class="comment"> *</span>
<a name="l01688"></a>01688 <span class="comment"> *</span>
<a name="l01689"></a>01689 <span class="comment"> * @param rte - the single instance RunTimeEnvironment.</span>
<a name="l01690"></a>01690 <span class="comment"> * @param sampleInfoHeaders - the dynamic sample headers that are not parsed.</span>
<a name="l01691"></a>01691 <span class="comment"> *</span>
<a name="l01692"></a>01692 <span class="comment"> */</span>
<a name="l01693"></a>01693 <span class="comment">/*****************************************************************************/</span>
<a name="l01694"></a>01694 <span class="keywordtype">void</span> TranslationExperimentReport::_openUncalledFile(<span class="keyword">const</span> <a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp; rte)
<a name="l01695"></a>01695 {
<a name="l01696"></a>01696 
<a name="l01697"></a>01697 
<a name="l01698"></a>01698   std::string completeName = <a class="code" href="classTranslationExperimentReport.html#abdcacbd2a0ddff3e0191177322b3a848" title="TranslationExperimentReport::getUncalledReportName Syopsis:">getUncalledReportName</a>(rte) + COMPLETE_REPORT_FILE_EXT;
<a name="l01699"></a>01699   <span class="keywordflow">if</span> (Fs::fileExists(completeName))
<a name="l01700"></a>01700     <a class="code" href="classFs.html#a6b8f232817f933a42c21edde6ea55a3a" title="An error if it doesnt exist.">Fs::rm</a>(completeName, <span class="keyword">false</span>);
<a name="l01701"></a>01701 
<a name="l01702"></a>01702   <span class="comment">// describe our columns.</span>
<a name="l01703"></a>01703   <span class="keywordtype">int</span> numUncalledColumns = <span class="keyword">sizeof</span>(UNCALLED_TSV_COLUMNS) / <span class="keyword">sizeof</span>(UNCALLED_TSV_COLUMNS[0]);
<a name="l01704"></a>01704 
<a name="l01705"></a>01705   <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; numUncalledColumns; i++) {
<a name="l01706"></a>01706     m_uncalledTsv.<a class="code" href="classaffx_1_1TsvFile.html#aadc22863a63b9c6588b0f2dc61875809" title="User methods.">defineColumn</a>(0, i, UNCALLED_TSV_COLUMNS[i]);
<a name="l01707"></a>01707   }
<a name="l01708"></a>01708 
<a name="l01709"></a>01709   m_uncalledTsv.<a class="code" href="classaffx_1_1TsvFile.html#a15037d55f98a66bce872281120cd0489" title="Add a comment to the headers ///.">addHeaderComment</a>(<span class="stringliteral">&quot; For research use only. Not for diagnostic purposes.&quot;</span>);
<a name="l01710"></a>01710   m_uncalledTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;dmet3-report-guid&quot;</span>, m_guid);
<a name="l01711"></a>01711   m_uncalledTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;Program&quot;</span>, <a class="code" href="classFs.html#a16925724086de1c12ea58bad4f8df75c" title="like the unix function of the same name. ///">Fs::basename</a>(rte.m_programName));
<a name="l01712"></a>01712   m_uncalledTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;Version&quot;</span>, ADT_VERSION);
<a name="l01713"></a>01713   m_uncalledTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;Date&quot;</span>, <a class="code" href="classUtil.html#accf4f30492f3726aac3197a37b3f77cc" title="Get a timestamp string.">Util::getTimeStamp</a>());
<a name="l01714"></a>01714   m_uncalledTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;TranslationFile&quot;</span>, rte.m_adtOpts.m_inputTTableFile);
<a name="l01715"></a>01715   m_uncalledTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;MarkerList&quot;</span>, rte.m_adtOpts.m_inputMarkerListFile);
<a name="l01716"></a>01716   m_uncalledTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;HaplotypeReportOption&quot;</span>, rte.m_adtOpts.m_useFirstDupAlleleDefHeaderText);
<a name="l01717"></a>01717   m_uncalledTsv.<a class="code" href="classaffx_1_1TsvFile.html#a11a4bb7b3c796316bc5b758f42229557" title="Add a key-value to the headers plus its sort order. ///.">addHeader</a>(<span class="stringliteral">&quot;GenotypeOverrideFile&quot;</span>, rte.m_adtOpts.m_inputGenotypeOverrideFile);
<a name="l01718"></a>01718 
<a name="l01719"></a>01719   m_uncalledTsv.<a class="code" href="classaffx_1_1TsvFile.html#a1903489904b4e2f8f2a262b8f10cdb0e" title="Set the options up for &amp;quot;tsv&amp;quot; format and write the tsv-v1 header ///.">writeTsv_v1</a>(<a class="code" href="classTranslationExperimentReport.html#abdcacbd2a0ddff3e0191177322b3a848" title="TranslationExperimentReport::getUncalledReportName Syopsis:">getUncalledReportName</a>(rte) + TEMP_REPORT_FILE_EXT);
<a name="l01720"></a>01720 
<a name="l01721"></a>01721   <span class="keywordflow">return</span>;
<a name="l01722"></a>01722 
<a name="l01723"></a>01723 }
<a name="l01724"></a>01724 <span class="comment">// end RegressionExperimentReport::_openUncalledFile</span>
<a name="l01725"></a>01725 <span class="comment">/*****************************************************************************/</span>
<a name="l01726"></a>01726 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l01727"></a>01727 <span class="comment">/**</span>
<a name="l01728"></a>01728 <span class="comment"> * TranslationExperimentReport::_report</span>
<a name="l01729"></a>01729 <span class="comment"> * Synopsis:</span>
<a name="l01730"></a>01730 <span class="comment"> *</span>
<a name="l01731"></a>01731 <span class="comment"> * Helper function to TranslationExperimentReport::generate broken out</span>
<a name="l01732"></a>01732 <span class="comment"> * for readability.</span>
<a name="l01733"></a>01733 <span class="comment"> *</span>
<a name="l01734"></a>01734 <span class="comment"> * Write or accumulate a tab delimited, output TSV file record</span>
<a name="l01735"></a>01735 <span class="comment"> * of marker data. The comprehensive report and uncalled report</span>
<a name="l01736"></a>01736 <span class="comment"> * are output in translation table marker order and therefore</span>
<a name="l01737"></a>01737 <span class="comment"> * are not further sorted.</span>
<a name="l01738"></a>01738 <span class="comment"> *</span>
<a name="l01739"></a>01739 <span class="comment"> * The summary report is sorted and therefore the records are accumulated</span>
<a name="l01740"></a>01740 <span class="comment"> * for a separate _report call after all the records have been accumulated</span>
<a name="l01741"></a>01741 <span class="comment"> * for sort. </span>
<a name="l01742"></a>01742 <span class="comment"> *</span>
<a name="l01743"></a>01743 <span class="comment"> * @param rte - the single instance run time environment.</span>
<a name="l01744"></a>01744 <span class="comment"> * @param cRow - the current row to emit, complete will all data populated. </span>
<a name="l01745"></a>01745 <span class="comment"> * @param sampleInfo -the same for all rows, appended to the row. </span>
<a name="l01746"></a>01746 <span class="comment"> *</span>
<a name="l01747"></a>01747 <span class="comment"> * @return - writes the record to disk or accumulates the summary row. </span>
<a name="l01748"></a>01748 <span class="comment"> */</span>
<a name="l01749"></a>01749 <span class="comment">/*****************************************************************************/</span>
<a name="l01750"></a>01750 <span class="keywordtype">void</span> TranslationExperimentReport::_report(<span class="keyword">const</span> <a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp; rte, <span class="keyword">const</span> <a class="code" href="classTranslationRow.html">TranslationRow</a> &amp; cRow, std::vector&lt; std::string &gt; &amp; sampleInfo)
<a name="l01751"></a>01751 {
<a name="l01752"></a>01752 
<a name="l01753"></a>01753   <span class="comment">// here we expect things to be in the order defined above.</span>
<a name="l01754"></a>01754   <span class="comment">// otherwise we could say: m_tsv.set(0,&quot;experiment&quot;,m_experiment);</span>
<a name="l01755"></a>01755   m_comprehensiveTsv.set(0, 0,  cRow.m_index);
<a name="l01756"></a>01756   m_comprehensiveTsv.set(0, 1,  cRow.m_chpFileName);
<a name="l01757"></a>01757   m_comprehensiveTsv.set(0, 2,  cRow.m_gene);
<a name="l01758"></a>01758   m_comprehensiveTsv.set(0, 3,  cRow.m_knownCall);
<a name="l01759"></a>01759   m_comprehensiveTsv.set(0, 4,  cRow.m_unknownCall);
<a name="l01760"></a>01760   m_comprehensiveTsv.set(0, 5,  cRow.m_interpretationCode);
<a name="l01761"></a>01761   m_comprehensiveTsv.set(0, 6,  cRow.m_summaryFlag);
<a name="l01762"></a>01762   m_comprehensiveTsv.set(0, 7,  cRow.m_relevantAlleles);
<a name="l01763"></a>01763   m_comprehensiveTsv.set(0, 8,  cRow.m_markerName);
<a name="l01764"></a>01764   m_comprehensiveTsv.set(0, 9,  cRow.m_probeSet);
<a name="l01765"></a>01765   m_comprehensiveTsv.set(0, 10, cRow.m_baseCall);
<a name="l01766"></a>01766   m_comprehensiveTsv.set(0, 11, cRow.m_referenceReportBase);
<a name="l01767"></a>01767   m_comprehensiveTsv.set(0, 12, cRow.m_variantReportBase);
<a name="l01768"></a>01768   m_comprehensiveTsv.set(0, 13, cRow.m_call);
<a name="l01769"></a>01769   m_comprehensiveTsv.set(0, 14, cRow.m_haplotypeMarker);
<a name="l01770"></a>01770   m_comprehensiveTsv.set(0, 15, cRow.m_changeForVariant);
<a name="l01771"></a>01771   m_comprehensiveTsv.set(0, 16, cRow.m_CDNAChange);
<a name="l01772"></a>01772   m_comprehensiveTsv.set(0, 17, cRow.m_genomicPosition);
<a name="l01773"></a>01773   m_comprehensiveTsv.set(0, 18, cRow.m_dbSNPId);
<a name="l01774"></a>01774   m_comprehensiveTsv.set(0, 19, cRow.m_originalBasecall);
<a name="l01775"></a>01775   m_comprehensiveTsv.set(0, 20, cRow.m_overrideComment);
<a name="l01776"></a>01776 
<a name="l01777"></a>01777   <span class="keywordflow">if</span> (cRow.m_isUncalledRow) {
<a name="l01778"></a>01778     m_uncalledTsv.set(0, 0, cRow.m_chpFileName);
<a name="l01779"></a>01779     m_uncalledTsv.set(0, 1, cRow.m_gene);
<a name="l01780"></a>01780     m_uncalledTsv.set(0, 2, cRow.m_markerName);
<a name="l01781"></a>01781     m_uncalledTsv.set(0, 3, cRow.m_probeSet);
<a name="l01782"></a>01782     m_uncalledTsv.set(0, 4, cRow.m_baseCall);
<a name="l01783"></a>01783     m_uncalledTsv.set(0, 5, <span class="stringliteral">&quot;&quot;</span>); <span class="comment">// Override</span>
<a name="l01784"></a>01784     m_uncalledTsv.set(0, 6, cRow.m_referenceBase);
<a name="l01785"></a>01785     m_uncalledTsv.set(0, 7, cRow.m_variantBase);
<a name="l01786"></a>01786 
<a name="l01787"></a>01787   }
<a name="l01788"></a>01788 
<a name="l01789"></a>01789   <span class="keywordflow">if</span> (cRow.m_isSummaryRow || cRow.m_isGeneSummary) {
<a name="l01790"></a>01790     <span class="comment">// Save the summary rows so they can be sorted before output.</span>
<a name="l01791"></a>01791     <span class="keywordflow">if</span> ( rte.m_adtOpts.m_summaryReportSort ) {
<a name="l01792"></a>01792       <a class="code" href="classSummaryRowIndex.html">SummaryRowIndex</a> sri;
<a name="l01793"></a>01793       sri.m_gene             = cRow.m_gene;
<a name="l01794"></a>01794       sri.m_probeSet         = cRow.m_probeSet;
<a name="l01795"></a>01795       sri.m_knownCall        = cRow.m_knownCall;
<a name="l01796"></a>01796       sri.m_summaryFlag      = cRow.m_summaryFlag;
<a name="l01797"></a>01797       sri.m_markerName       = cRow.m_markerName;
<a name="l01798"></a>01798       sri.m_isSummaryRow     = cRow.m_isSummaryRow;
<a name="l01799"></a>01799       sri.m_isGeneSummary    = cRow.m_isGeneSummary;
<a name="l01800"></a>01800       sri.m_copyNumber       = cRow.m_copyNumber;
<a name="l01801"></a>01801       m_summaryRows[sri]     = cRow;
<a name="l01802"></a>01802     }
<a name="l01803"></a>01803     <span class="keywordflow">else</span> {
<a name="l01804"></a>01804       _reportSummary( cRow, sampleInfo );
<a name="l01805"></a>01805     }
<a name="l01806"></a>01806   }
<a name="l01807"></a>01807 
<a name="l01808"></a>01808   <span class="keywordflow">for</span> (<span class="keywordtype">size_t</span> i = 0; i &lt; sampleInfo.size() ; i++) {
<a name="l01809"></a>01809     m_comprehensiveTsv.set(0, 21 + i, sampleInfo[i]);
<a name="l01810"></a>01810   }
<a name="l01811"></a>01811 
<a name="l01812"></a>01812   <span class="comment">// Comprehensive file write.</span>
<a name="l01813"></a>01813   m_comprehensiveTsv.<a class="code" href="classaffx_1_1TsvFile.html#a2a459be42bea9ea3f180773370ec0de7" title="Write a level of data to the file. ///.">writeLevel</a>(0);
<a name="l01814"></a>01814 
<a name="l01815"></a>01815 
<a name="l01816"></a>01816   <span class="comment">// Uncalled file write.</span>
<a name="l01817"></a>01817   <span class="keywordflow">if</span> (cRow.m_isUncalledRow) {
<a name="l01818"></a>01818     m_uncalledTsv.<a class="code" href="classaffx_1_1TsvFile.html#a2a459be42bea9ea3f180773370ec0de7" title="Write a level of data to the file. ///.">writeLevel</a>(0);
<a name="l01819"></a>01819   }
<a name="l01820"></a>01820 
<a name="l01821"></a>01821   <span class="keywordflow">return</span>;
<a name="l01822"></a>01822 }
<a name="l01823"></a>01823 <span class="comment">// end TranslationExperimentReport::_report</span>
<a name="l01824"></a>01824 <span class="comment">/*****************************************************************************/</span>
<a name="l01825"></a>01825 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l01826"></a>01826 <span class="comment">/**</span>
<a name="l01827"></a>01827 <span class="comment"> * TranslationExperimentReport::_reportSummary</span>
<a name="l01828"></a>01828 <span class="comment"> *</span>
<a name="l01829"></a>01829 <span class="comment"> * Synopsis:</span>
<a name="l01830"></a>01830 <span class="comment"> *</span>
<a name="l01831"></a>01831 <span class="comment"> * Helper function to TranslationExperimentReport::_reportSummarySorted</span>
<a name="l01832"></a>01832 <span class="comment"> * broken out for readability.</span>
<a name="l01833"></a>01833 <span class="comment"> </span>
<a name="l01834"></a>01834 <span class="comment"> * </span>
<a name="l01835"></a>01835 <span class="comment"> * @param rte - the single instance run time environment.</span>
<a name="l01836"></a>01836 <span class="comment"> * @param cRow - the current row to emit, complete will all data populated. </span>
<a name="l01837"></a>01837 <span class="comment"> * @param sampleInfo -the same for all rows, appended to the row. </span>
<a name="l01838"></a>01838 <span class="comment"> *</span>
<a name="l01839"></a>01839 <span class="comment"> * @return - writes the summary record to disk in a TSV file. </span>
<a name="l01840"></a>01840 <span class="comment"> *</span>
<a name="l01841"></a>01841 <span class="comment"> */</span>
<a name="l01842"></a>01842 <span class="comment">/*****************************************************************************/</span>
<a name="l01843"></a>01843 <span class="keywordtype">void</span> TranslationExperimentReport::_reportSummary(<span class="keyword">const</span> <a class="code" href="classTranslationRow.html">TranslationRow</a> &amp; cRow, std::vector&lt; std::string &gt; &amp; sampleInfo) {
<a name="l01844"></a>01844 
<a name="l01845"></a>01845   
<a name="l01846"></a>01846   <a class="code" href="Err_8h.html#ad024208f4ba6a4e75df4499ea1639a05" title="Calls Err::apt_err_assert with the filename and linenumber set. /// We want to avoid evaluating the m...">APT_ERR_ASSERT</a>(cRow.m_isSummaryRow || cRow.m_isGeneSummary , <span class="stringliteral">&quot;_reportSummary got a non-summary record.&quot;</span>);
<a name="l01847"></a>01847 
<a name="l01848"></a>01848   <span class="keywordflow">if</span> (cRow.m_isSummaryRow ) {
<a name="l01849"></a>01849     
<a name="l01850"></a>01850     m_summaryTsv.set(0, 0,  cRow.m_index);
<a name="l01851"></a>01851     m_summaryTsv.set(0, 1,  cRow.m_chpFileName);
<a name="l01852"></a>01852     m_summaryTsv.set(0, 2,  cRow.m_gene);
<a name="l01853"></a>01853     m_summaryTsv.set(0, 3,  cRow.m_knownCall);
<a name="l01854"></a>01854     m_summaryTsv.set(0, 4,  cRow.m_unknownCall);
<a name="l01855"></a>01855     m_summaryTsv.set(0, 5,  cRow.m_interpretationCode);
<a name="l01856"></a>01856     m_summaryTsv.set(0, 6,  cRow.m_summaryFlag);
<a name="l01857"></a>01857     m_summaryTsv.set(0, 7,  cRow.m_relevantAlleles);
<a name="l01858"></a>01858     m_summaryTsv.set(0, 8,  cRow.m_markerName);
<a name="l01859"></a>01859     m_summaryTsv.set(0, 9,  cRow.m_probeSet);
<a name="l01860"></a>01860     m_summaryTsv.set(0, 10, cRow.m_baseCall);
<a name="l01861"></a>01861     m_summaryTsv.set(0, 11, cRow.m_referenceReportBase);
<a name="l01862"></a>01862     m_summaryTsv.set(0, 12, cRow.m_variantReportBase);
<a name="l01863"></a>01863     m_summaryTsv.set(0, 13, cRow.m_call);
<a name="l01864"></a>01864     m_summaryTsv.set(0, 14, cRow.m_haplotypeMarker);
<a name="l01865"></a>01865     m_summaryTsv.set(0, 15, cRow.m_changeForVariant);
<a name="l01866"></a>01866     m_summaryTsv.set(0, 16, cRow.m_CDNAChange);
<a name="l01867"></a>01867     m_summaryTsv.set(0, 17, cRow.m_genomicPosition);
<a name="l01868"></a>01868     m_summaryTsv.set(0, 18, cRow.m_dbSNPId);
<a name="l01869"></a>01869     m_summaryTsv.set(0, 19, cRow.m_originalBasecall);
<a name="l01870"></a>01870     m_summaryTsv.set(0, 20, cRow.m_overrideComment);
<a name="l01871"></a>01871   } <span class="keywordflow">else</span>  {
<a name="l01872"></a>01872     m_summaryTsv.set(0, 0,  cRow.m_index);
<a name="l01873"></a>01873     m_summaryTsv.set(0, 1,  cRow.m_chpFileName);
<a name="l01874"></a>01874     m_summaryTsv.set(0, 2,  cRow.m_gene);
<a name="l01875"></a>01875     m_summaryTsv.set(0, 3,  cRow.m_knownCall);
<a name="l01876"></a>01876     m_summaryTsv.set(0, 4,  cRow.m_unknownCall);
<a name="l01877"></a>01877     m_summaryTsv.set(0, 5,  cRow.m_interpretationCode);
<a name="l01878"></a>01878     <span class="keywordflow">if</span> ( cRow.m_copyNumber == 0 ) {
<a name="l01879"></a>01879       m_summaryTsv.set(0, 6, <span class="stringliteral">&quot;Deletion alleles have been detected for this gene, CN=0&quot;</span>);
<a name="l01880"></a>01880     }
<a name="l01881"></a>01881     <span class="keywordflow">else</span> {
<a name="l01882"></a>01882       m_summaryTsv.set(0, 6, <span class="stringliteral">&quot;All markers responsible for functional changes are Ref/Ref&quot;</span>);
<a name="l01883"></a>01883     }
<a name="l01884"></a>01884     <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 7; i &lt; 21; i++) {
<a name="l01885"></a>01885       m_summaryTsv.set(0, i, <span class="stringliteral">&quot;&quot;</span>);
<a name="l01886"></a>01886     }
<a name="l01887"></a>01887 
<a name="l01888"></a>01888   }
<a name="l01889"></a>01889 
<a name="l01890"></a>01890   <span class="keywordflow">for</span> (<span class="keywordtype">size_t</span> i = 0; i &lt; sampleInfo.size() ; i++) {
<a name="l01891"></a>01891     m_summaryTsv.set(0, 21 + i, sampleInfo[i]);
<a name="l01892"></a>01892   }
<a name="l01893"></a>01893 
<a name="l01894"></a>01894   m_summaryTsv.<a class="code" href="classaffx_1_1TsvFile.html#a2a459be42bea9ea3f180773370ec0de7" title="Write a level of data to the file. ///.">writeLevel</a>(0);
<a name="l01895"></a>01895 
<a name="l01896"></a>01896 }
<a name="l01897"></a>01897 <span class="comment">// end TranslationExperimentReport::_reportSummary</span>
<a name="l01898"></a>01898 <span class="comment">/*****************************************************************************/</span>
<a name="l01899"></a>01899 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l01900"></a>01900 <span class="comment">/**</span>
<a name="l01901"></a>01901 <span class="comment"> * TranslationExperimentReport::_reportSummarySorted   </span>
<a name="l01902"></a>01902 <span class="comment"> *</span>
<a name="l01903"></a>01903 <span class="comment"> * Synopsis:</span>
<a name="l01904"></a>01904 <span class="comment"> * </span>
<a name="l01905"></a>01905 <span class="comment"> * Helper function to TranslationExperimentReport::generate broken out</span>
<a name="l01906"></a>01906 <span class="comment"> * for readability.</span>
<a name="l01907"></a>01907 <span class="comment"> *</span>
<a name="l01908"></a>01908 <span class="comment"> * Separate from _report() because the sumary records need to be accumlated</span>
<a name="l01909"></a>01909 <span class="comment"> * and then sorted for output. This method outputs all the summary records</span>
<a name="l01910"></a>01910 <span class="comment"> * previously accumulated. </span>
<a name="l01911"></a>01911 <span class="comment"> * </span>
<a name="l01912"></a>01912 <span class="comment"> * @param sampleInfo - the sample info for all rows.</span>
<a name="l01913"></a>01913 <span class="comment"> *</span>
<a name="l01914"></a>01914 <span class="comment"> *</span>
<a name="l01915"></a>01915 <span class="comment"> */</span>
<a name="l01916"></a>01916 <span class="comment">/*****************************************************************************/</span>
<a name="l01917"></a>01917 <span class="keywordtype">void</span> TranslationExperimentReport::_reportSummarySorted(std::vector&lt; std::string &gt; &amp; sampleInfo) {
<a name="l01918"></a>01918 
<a name="l01919"></a>01919   std::map&lt; SummaryRowIndex, TranslationRow&gt;::iterator itSRI;
<a name="l01920"></a>01920 
<a name="l01921"></a>01921   <span class="keywordflow">for</span> ( itSRI = m_summaryRows.begin();
<a name="l01922"></a>01922         itSRI != m_summaryRows.end();
<a name="l01923"></a>01923         itSRI ++ ) {
<a name="l01924"></a>01924     _reportSummary( itSRI-&gt;second, sampleInfo);
<a name="l01925"></a>01925     
<a name="l01926"></a>01926   }
<a name="l01927"></a>01927 
<a name="l01928"></a>01928   m_summaryRows.clear();
<a name="l01929"></a>01929   
<a name="l01930"></a>01930 
<a name="l01931"></a>01931 }
<a name="l01932"></a>01932 <span class="comment">// end TranslationExperimentReport::_reportSummarySorted</span>
<a name="l01933"></a>01933 <span class="comment">/*****************************************************************************/</span>
<a name="l01934"></a>01934 <span class="comment">/*****************************************************************************/</span><span class="comment"></span>
<a name="l01935"></a>01935 <span class="comment">/**</span>
<a name="l01936"></a>01936 <span class="comment"> * TranslationExperimentReport::_setIsUncalledRow:</span>
<a name="l01937"></a>01937 <span class="comment"> * Synopsis:</span>
<a name="l01938"></a>01938 <span class="comment"> *</span>
<a name="l01939"></a>01939 <span class="comment"> * Helper function to TranslationExperimentReport::generate broken out</span>
<a name="l01940"></a>01940 <span class="comment"> * for readability.</span>
<a name="l01941"></a>01941 <span class="comment"> * </span>
<a name="l01942"></a>01942 <span class="comment"> * Determines if a row should be reported on in the uncalled report.</span>
<a name="l01943"></a>01943 <span class="comment"> * From the SRS:</span>
<a name="l01944"></a>01944 <span class="comment"> *</span>
<a name="l01945"></a>01945 <span class="comment"> *  Conditions:</span>
<a name="l01946"></a>01946 <span class="comment"> *  basecall = NoCall, NotAvailaable or PRA </span>
<a name="l01947"></a>01947 <span class="comment"> *</span>
<a name="l01948"></a>01948 <span class="comment"> *</span>
<a name="l01949"></a>01949 <span class="comment"> * @param cRow -  With baseCall already filled in data structure of row data.</span>
<a name="l01950"></a>01950 <span class="comment"> *</span>
<a name="l01951"></a>01951 <span class="comment"> * @return - void, cRow.m_isUncalledRow is set.</span>
<a name="l01952"></a>01952 <span class="comment"> */</span>
<a name="l01953"></a>01953 <span class="comment">/*****************************************************************************/</span>
<a name="l01954"></a>01954 <span class="keywordtype">void</span> TranslationExperimentReport::_setIsUncalledRow(<span class="keyword">const</span> <a class="code" href="classRunTimeEnvironment.html">RunTimeEnvironment</a> &amp; rte, <a class="code" href="classTranslationRow.html">TranslationRow</a> &amp; cRow)
<a name="l01955"></a>01955 {
<a name="l01956"></a>01956 
<a name="l01957"></a>01957   cRow.m_isUncalledRow = <span class="keyword">false</span>;
<a name="l01958"></a>01958 
<a name="l01959"></a>01959   <span class="comment">// No Call or PRA</span>
<a name="l01960"></a>01960   <span class="keywordflow">if</span> ( rte.m_adtOpts.m_uncalledReportAllMarkers || pcrecpp::RE(<span class="stringliteral">&quot;NoCall|NC|PRA|PossibleRareAllele|NotAvailable&quot;</span>).FullMatch(cRow.m_baseCall)) {
<a name="l01961"></a>01961     cRow.m_isUncalledRow = <span class="keyword">true</span>;
<a name="l01962"></a>01962 
<a name="l01963"></a>01963   }
<a name="l01964"></a>01964 
<a name="l01965"></a>01965   <span class="keywordflow">return</span>;
<a name="l01966"></a>01966 
<a name="l01967"></a>01967 }
<a name="l01968"></a>01968 <span class="comment">//end TranslationExperimentReport::_setIsUncalledRow</span>
<a name="l01969"></a>01969 <span class="comment">/*****************************************************************************/</span>
</pre></div></div>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:48 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
