<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: VIGNETTES: SNP6 Copy Number Workflows</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li class="current"><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>VIGNETTES: SNP6 Copy Number Workflows </h1>  </div>
</div>
<div class="contents">
<dl class="date"><dt><b>Date:</b></dt><dd>2007-06-27</dd></dl>
<h2><a class="anchor" id="content"></a>
Contents</h2>
<ul>
<li><a href="#workflow1a">Workflow 1a: 1.8M Copy Number Estimates</a></li>
<li><a href="#workflow1b">Workflow 1b: 2.7M Copy Number Estimates (treat A and B alleles separately)</a></li>
<li><a href="#workflow2">Workflow 2: Summarization of CN Probes Over Known Copy Number Regions [experimental]</a></li>
<li><a href="#faq">FAQ - Frequently Asked Questions.</a></li>
</ul>
<p><a class="anchor" id="introCnOld"></a> </p>
<h2><a class="anchor" id="introCnOld"></a>
Introduction</h2>
<p>[2008-06-18] This vignette describes a low level workflow for extracting probe level data off of SNP6 with an eye toward copynumber analysis. More recently, apt-copynumber-workflow has been released which provides a more complete copynumber analysis solution. Most people should probably use apt-copynumber-workflow rather than the workflows described here.</p>
<p><a class="anchor" id="workflow1a"></a> </p>
<h2><a class="anchor" id="workflow1a"></a>
Workflow 1a: 1.8M Copy Number Estimates</h2>
<p>APT supports output of Copy Number (CN) estimates on a per SNP or per CN probeset basis. The default analysis recommends quantile normalization (quant-norm) across all chips.</p>
<p>The default analysis string is </p>
<div class="fragment"><pre class="fragment">
    quant-norm.sketch=50000,pm-sum,med-polish,expr.genotype=true.allele-a=true
</pre></div><p>For SNP 6 CN probesets (which contain a single probe) no further summarization is done after normalization. log2 values are reported.</p>
<p>For SNPs, in SNP 6 chips SNP probes are spatially arranged in pairs where each A-allele probe is paired with a B-allele probe. "expr.genotype-true" causes an expression probeset to be created using one allele as the PM probes and the other allele as the MM probes. "pm-sum" then causes those values to be added and then the log2 of the sums (for all allele probe pairs in the same SNP) are median polished to produce a single value for the SNP. For SNPs, there are 2 alleles that each will have identical values (given this analysis string). "expr.allele-a=true" forces output of only the A allele.</p>
<p>The output file contains log2 transformed signal estimates arranged in a table with SNP probeset names and CN probeset names labeling rows and Chip file names labeling columns.</p>
<p>You can use "apt-probeset-summarize --help" for more information on command line arguments and any of the following to get more information about the various analysis stream components:</p>
<div class="fragment"><pre class="fragment">
    apt-probeset-summarize --explain quant-norm
    apt-probeset-summarize --explain pm-sum
    apt-probeset-summarize --explain med-polish
    apt-probeset-summarize --explain expr
</pre></div><p>This is an example of how to a do CN run using APT:</p>
<div class="fragment"><pre class="fragment">
    apt-probeset-summarize \
        --cdf-file lib_directory_name/GenomeWideSNP_6.cdf \
        --analysis quant-norm.sketch=50000,pm-sum,med-polish,expr.genotype=true.allele-a=true \
        --out-dir  output_directory_name \
        celfile_directory/*.CEL
</pre></div><p>Note that the use of wild cards (*.CEL) is dependent on the shell from which the command is run. For example, the default shell on Windows does not support wild card expansions, so you will probably want to use the "--cel-files" option instead.</p>
<p>Also note that the output file will also contain other control probesets (including some control SNPs) that can be removed (in UNIX) using the following commands:</p>
<div class="fragment"><pre class="fragment">
    grep SNP quant-norm.pm-sum.med-polish.expr.summary.txt | grep -v AFFX &gt; snp-and-cn-only.txt
    grep CN quant-norm.pm-sum.med-polish.expr.summary.txt &gt;&gt; snp-and-cn-only.txt
</pre></div><p><a class="anchor" id="workflow1b"></a> </p>
<h2><a class="anchor" id="workflow1b"></a>
Workflow 1b: 2.7M Copy Number Estimates (treat A and B alleles separately)</h2>
<p>If SNP-A and SNP-B alleles are desired to be summarized separately, use the analysis string:</p>
<div class="fragment"><pre class="fragment">
    quant-norm.sketch=50000,pm-only,med-polish,expr.genotype=true
</pre></div><p>to obtain 2 output lines per SNP, with each allele summarized separately. There are two changes here. First we drop "allele-a=true" so that there will be results for both the A and B allele. Secondly we use "pm-only" rather than "pm-sum". In short, we only want values based on the particular allele; we do not want to add in the other allele which was converted to a mismatch probe.</p>
<p>This is an example of how to a do CN run summarizing SNP alleles separately using APT:</p>
<div class="fragment"><pre class="fragment">
    apt-probeset-summarize \
        --cdf-file lib_directory_name/GenomeWideSNP_6.cdf \
        --analysis quant-norm.sketch=50000,pm-only,med-polish,expr.genotype=true \
        --out-dir  output_directory_name \
        celfile_directory/*.CEL
</pre></div><p>Note that the use of wildcards (*.CEL) is dependent on the shell from which the command is run. For example, the default shell on Windows does not support wild card expansions, so you will probably want to use the "--cel-files" option instead.</p>
<p>The output file will also contain CN probeset names and other control probesets (including some control SNPs) which can be removed (in UNIX) using a command such as:</p>
<div class="fragment"><pre class="fragment">
    grep SNP quant-norm.pm-sum.med-polish.expr.summary.txt | grep -v AFFX &gt; snp-only.txt
</pre></div><p><a class="anchor" id="workflow2"></a> </p>
<h2><a class="anchor" id="workflow2"></a>
Workflow 2: Summarization of CN Probes Over Known Copy Number Regions [experimental]</h2>
<p><b>This is an experimental workflow for the analysis of known copy number regions. Do your own due diligence in using this workflow and interpreting the results.</b></p>
<p>APT supports output of Copy Number (CN) estimates for known copy number regions that are specified in a meta-probeset file. A meta-probeset file allows the definition of new probesets (in the probeset_id column of the file) as a group of existing probesets (space delimited in the probeset_list column). The meta-probeset file we are supplying in this release contains only CN probes. The default analysis recommends quantile normalization across all chips and med-polish on probes selected as most informative with no background correction.</p>
<p>The CN output file contains log2 transformed signal estimates arranged in a table with CNV regions labeling rows and Chip file names labeling columns.</p>
<p>The default analysis string is </p>
<div class="fragment"><pre class="fragment">
    quant-norm.sketch=50000,pm-only,med-polish,pca-select
</pre></div><p>The pca-select is an <a class="el" href="classAnalysis.html">Analysis</a> Stream method which does feature selection prior to computing the summaries. In short, the first principal component of the quantile normalized probe intensities are used to select out those probes which are near the first principal component. Summarization (median polish in this case) is applied to only those probes. An extra report files (with pca-select listed twice in the name) will be generated indicating which probes were used.</p>
<p>This is an example of how to do a CNV specific run using APT.</p>
<div class="fragment"><pre class="fragment">
    apt-probeset-summarize \
        --cdf-file lib_directory_name/GenomeWideSNP_6.cdf \
        --analysis quant-norm.sketch=50000,pm-only,med-polish,pca-select \
        --meta-probesets lib_directory_name/GenomeWideSNP_6.na22.dgv-cnvMay07.mps \
        --out-dir output_directory_name \
        celfile_directory/*.CEL
</pre></div><p>The MPS file is available from the SNP6 array support page (<a href="http://www.affymetrix.com/support/technical/byproduct.affx?product=genomewidesnp_6">http://www.affymetrix.com/support/technical/byproduct.affx?product=genomewidesnp_6</a>);</p>
<p>Note that the use of wildcards (*.CEL) is dependent on the shell from which the command is run. For example, the default shell on Windows does not support wild card expansions, so you will probably want to use the "--cel-files" option instead.</p>
<p><a class="anchor" id="faq"></a> </p>
<h2><a class="anchor" id="manFaq"></a>
Frequently Asked Questions</h2>
<p>Q. Are there similar recommendations for snp array 5?</p>
<p>A. Please note that the copy number workflow for SNP 6.0 in APT is experimental and unsupported and users need to do their own due-dilligence. Keeping that in mind, it is possible to leverage the SNP 6.0 copy number APT workflow for SNP 5.0 (experimental and unsupported) with the following caveats:</p>
<ul>
<li>For SNP 5.0 you will ONLY get the CN probes included if you use the .Full version of the CDF</li>
<li>Unlike the workflow for SNP 6.0, the .Full CDF for SNP 5.0 represents CN probes as probesets of all CN probes on the same Nsp restriction fragment, and so while there are ~420K CN probes the workflow will report ~320K CN probesets (most of which consist of just one probe).</li>
<li>The way the CN probes are represented in the .Full CDF for SNP 5.0 is considered experimental and may change to be consistent with SNP 6.0 in future</li>
</ul>
<p>Affymetrix has no immediate plans to develop and validate a copy number solution in APT or any other software for SNP 5.0. If you are interested in copy number analysis software that can be used for SNP 5.0 arrays, please contact Partek <a href="http://www.partek.com/">http://www.partek.com/</a> or the Ogawa group at Univ of Tokyo working on CNAG <a href="http://www.genome.umin.jp/.">http://www.genome.umin.jp/.</a></p>
<p><b></p>
<pre>Affymetrix Power Tools (APT) Release 1.16.1</pre><p></b> </p>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:57 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
