<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: VIGNETTES: Analyzing SNP6 Arrays with Axiom &tm; Analysis</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li class="current"><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>VIGNETTES: Analyzing SNP6 Arrays with Axiom &trade; <a class="el" href="classAnalysis.html">Analysis</a> </h1>  </div>
</div>
<div class="contents">
<dl class="date"><dt><b>Date:</b></dt><dd>2014-6-23</dd></dl>
<h2><a class="anchor" id="content"></a>
Contents</h2>
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#setup">Setup</a></li>
<li><a href="#sample-qc">Running Sample QC</a></li>
<li><a href="#step1">Running Step 1 Genotyping</a></li>
<li><a href="#step2">Running Step 2 Genotyping</a></li>
</ul>
<h2><a class="anchor" id="introoduction"></a>
Introduction</h2>
<p>With the gaining popularity of Axiom, customers have inquired if it is possible to reanalyze their SNP6 data using the same algorithm. This is possible to do and as with all Axiom analysis, the Best Practices Workflow should be followed.</p>
<h2><a class="anchor" id="setup"></a>
Setup</h2>
<ol type="1">
<li>The standard library files can be found on the Affymetrix website here: <a href="http://www.affymetrix.com/Auth/support/downloads/library_files/genomewidesnp6_libraryfile.zip">http://www.affymetrix.com/Auth/support/downloads/library_files/genomewidesnp6_libraryfile.zip</a>. Download these files and unzip them.</li>
<li>Please also download the additional files needed here: <a href="http://www.affymetrix.com/Auth/analysis/downloads/lf/genotyping/GenomeWideSNP_6/SNP6_supplemental_axiom_analysis_files.zip">http://www.affymetrix.com/Auth/analysis/downloads/lf/genotyping/GenomeWideSNP_6/SNP6_supplemental_axiom_analysis_files.zip</a>. These files should be unzipped and put in the sub folder /CD_GenomeWideSNP_6_rev3/Full/GenomeWideSNP_6/LibFiles/ of the library files downloaded. If these files are not in the same folder as the .cdf and other library files, the analysis will not work.</li>
<li>This vignette assumes familiarity with running APT on Axiom arrays. Please review the vignettes for sample QC and genotyping as necessary.</li>
</ol>
<p><b>NOTE:</b> In these code examples, the “lib_directory_name” refers to your specific path where you unzipped the library files and the additional analysis files. So if you unzipped the full library file package in /mylib/SNP6/ you would use /mylib/SNP6/CD_GenomeWideSNP_6_rev3/Full/GenomeWideSNP_6/LibFiles/ in place of “lib_directory_name”.</p>
<h2><a class="anchor" id="sample-qc"></a>
Running Sample QC</h2>
<p>QC should be run on all the CEL files. As typical with any call to apt, the cel-files should be a list of the full path of the CEL files with the header “cel_files”. Any samples failing QC should not be clustered.</p>
<div class="fragment"><pre class="fragment">
  apt-geno-qc \
    --cdf-file /lib_directory_name/ GenomeWideSNP_6.cdf \
    --qca-file /lib_directory_name /GenomeWideSNP_6.r2.qca \
    --qcc-file /lib_directory_name /GenomeWideSNP_6.r2.qcc \
    --cel-files cels_list.txt \
    --out-file qc.txt
</pre></div><h2><a class="anchor" id="step1"></a>
Running Step 1 Genotyping</h2>
<p>Step 1 genotyping should be run on all CEL files passing QC over 20K SNPs. The code below allows this to be done. This example uses generic priors.</p>
<div class="fragment"><pre class="fragment">
  apt-probeset-genotype \
    --feat-effects \
    --read-models-brlmmp /lib_directory_name /GenomeWideSNP_6.generic_prior.txt \
    --probeset-ids/ lib_directory_name /GenomeWideSNP_6.random.20K.intersect.txt \
    --log-file ./axiom_outliers/apt-probeset-genotype.log \
    --write-models \
    --xml-file /lib_directory_name /GenomeWideSNP_6.apt-probeset-genotype.AxiomGT1.xml \
    --analysis-files-path lib_directory_name/ \
     --out-dir ./axiom_outliers \
    --cel-files cels_list.txt \
    --summaries \
    --force
</pre></div><h2><a class="anchor" id="step2"></a>
Running Step 2 Genotyping</h2>
<p>For samples with a call rate greater than 97% in the Step 1 genotyping, the Step 2 genotyping can be done. To run with generic priors the following script can be used:</p>
<div class="fragment"><pre class="fragment">
  apt-probeset-genotype \
    --feat-effects \
    --read-models-brlmmp /lib_directory_name/GenomeWideSNP_6.generic_prior.txt \
    --log-file ./axiom_inliers/apt-probeset-genotype.log \
    --write-models \
    --xml-file /lib_directory_name/GenomeWideSNP_6.apt-probeset-genotype.AxiomGT1.xml \
    --analysis-files-path lib_directory_name/ \
    --out-dir ./axiom_inliers \
    --cel-files cels_list.txt \
    --summaries \
    --force
</pre></div><p><b>Notes</b></p>
<p>SNP polisher should be run after this as per best practices workflow.</p>
<p>If you want to generate CHP files, the following line can be added to the Step 2 genotyping </p>
<div class="fragment"><pre class="fragment">
    --cc-chp-output
</pre></div><p><b></p>
<pre>Affymetrix Power Tools (APT) Release 1.16.1</pre><p></b> </p>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:57 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
