<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: VIGNETTES: 2.7M Cytogenetics array</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li class="current"><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>VIGNETTES: 2.7M Cytogenetics array </h1>  </div>
</div>
<div class="contents">
<dl class="date"><dt><b>Date:</b></dt><dd>2011-06-20</dd></dl>
<h2><a class="anchor" id="content"></a>
Contents</h2>
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#libraryfiles">Library Files</a></li>
<li><a href="#ref">Command line example for reference creation:</a></li>
<li><a href="#ssa">Command line example for single sample mode processing</a></li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2><a class="anchor" id="intro"></a>
Introduction</h2>
<p>This vignette describes the process for using apt-copynumber-cyto to process 2.7M Cytogenetics_Array under reference creation mode and single sample mode.</p>
<p>apt-copynumber-cyto implements two distinct workflows, a reference workflow that uses as a a set of CEL files that are input to outputs one or more reference files including copynumber and SNP information; and conceptually, a single-sample workflow that compares each CEL file to a pre-computed reference for analysis. For efficiency of computation the "single-sample workflow" operates on a set of input CEL files at a time, but the output for any CEL file is unaffected by any of the other CEL files.</p>
<p><a class="anchor" id="libraryfiles"></a> </p>
<h2><a class="anchor" id="libraryfiles"></a>
Library Files</h2>
<p>The following library files are required for analysis of 2.7M data.</p>
<div class="fragment"><pre class="fragment">
Cytogenetics_Array.na32.annot.db:    NetAffx Annotation database file. It is a SQLite 3.x database.

Cytogenetics_Array.chrXprobes:       An ASCII file that contains probe_id (1-based) of 
                                     probes on chrX. Used for copy number probe chrX/Y ratio gender calling.

Cytogenetics_Array.v2.chrYprobes:    An ASCII file that contains probe_id (1-based) of 
                                     probes on chrY as column 1 and probeset_id as column 2. Probe_id is used for 
                                     copy number probe chrX/Y ratio gender calling and probeset_id is used for
                                     hasY gender call and Y-Gender call. 

Cytogenetics_Array.CDF:              Chip definition file (binary file).

Cytogenetics_Array.probe_tab:        An ASCII file that contains probe sequence information. (Reference creation only).

Genders_ReferenceSetD.txt:           An ASCII file that contains gender information from the input reference. (SNP reference creation only).

HapMapCalls_ReferenceSetD.txt:       An ASCII file that contains genotype calls from the hapmaps. (SNP reference creation only)

Cytogenetics_Array.snplist.txt:      An ASCII file that contains SNP probeset_id that are used to compute processed SNPQC.
                                     The first row of the file must be "probeset_id". (Single sample mode processing only).

Cytogenetics_Array.na32.v1.REF_MODEL:  Affymetrix copy number reference generated from a set of hapmaps and normal blood samples (hdf5 format file).

Cytogenetics_Array.na32.v1.snpref.a5:  Affymetrix snp reference generated from a set of hapmaps and normal blood samples (hdf5 format file).
</pre></div><p><a class="anchor" id="ref"></a> </p>
<h2><a class="anchor" id="ref"></a>
Command line example for reference creation:</h2>
<div class="fragment"><pre class="fragment">
apt-copynumber-cyto -v 4 \     #verbose output level in the log file ranging from 1 (least details) to 4 (most details)
	--cyto2 true \                 #true: Cytogenetics_Array; false: CytoScanHD_Array
	--force true \                 #true: Disable various checks including chip types; false: otherwise.
	--run-geno-qc false \          #true: Run the GenoQC engine; false: otherwise. Should always be false for Cytogenetics_Array.
	--adapter-type-normalization false \ #true: perform adapter type normalization; false: otherwise. Should always be false for Cytogenetics_Array.
	--text-output true \                 #true: Output data in ASCII text format in addition to calvin format; false: otherwise
	--cnchp-output false \               #true: report CNCHP file(s); false: otherwise. Either cnchp-output or cychp-output has to be true
	--cychp-output true \                #true: report CYCHP file(s); false: otherwise. Either cnchp-output or cychp-output has to be true
	--reference-output CnNewRef.REF_MODEL \  #specify the name of the new copy number reference file with file extension "REF_MODEL"
	--snp-reference-input-file SnpRef.existing.snpref.a5 \  #specify an existing snp reference if you want to create new snp reference, otherwise do not specify this option
	--snp-reference-output-file SnpNewRef.snpref.a5 \       #specify the name of the new snp reference file with file extension "snpref.a5"
	--gender-override-file Genders_ReferenceSetD.txt \      #specify the gender information corresponding to the reference set if you want to create new snp reference, otherwise do not specify this option
	--genotype-call-override-file HapMapCalls_ReferenceSetD.txt \  #specify the genotype calls correspoinding to the reference set if you want to create new snp reference, otherwise do not specify this option
	--set-analysis-name Cytogenetics \                      #Analysis name to use as prefix for output files.
	--probe-file Cytogenetics_Array.probe_tab \             #specify the probe sequence file
	--cdf-file Cytogenetics_Array.CDF \                     #specify the chp definition file
	--chrX-probes Cytogenetics_Array.chrXprobes \           #specify the chrX probe file
	--chrY-probes Cytogenetics_Array.v2.chrYprobes \        #specify the chrY probe file
	--annotation-file Cytogenetics_Array.na32.annot.db \    #specify the NetAffx database file
	--out-dir output_dir \                                  #specify the output directory 
	--male-gender-ratio-cutoff=1.3 \                        #male gender call threshold for chrX/Y ratio gender calling
	--female-gender-ratio-cutoff=1.0 \                      #female gender call threshold for chrX/Y ratio gender calling
	--xx-cutoff=0.8 \                                       #lower limit of the gender call threshold for hasXX gender call
	--xx-cutoff-high=1.07 \                                 #upper limit of the gender call threshold for hasXX gender call 
	--y-cutoff=0.65 \                                       #gender call threshold for hasY gender call
	--local-gc-background-correction-reference-method pdnn-reference-method \ #create pdnn predicted intensities using sequence information
	--wave-correction-reference-method wave-correction-reference-method.\     #estimate waves using the reference samples
  	   wave-count=6.\                                       #number of waves to calculate
	   trim=2.0\                                            #trim parameter for absolute adjusted log2 ratios after each wave
	   .percentile=0.75.\                                   #percentile to use for each probeset in finding wave. E.g., 0.75 means 75-th percentile.
	   demean=false \                                       #true: demean prior to finding each wave; false: otherwise
	--local-gc-background-intensity-adjustment-method pdnn-intensity-adjustment-method.\ #apply pdnn and local gc adjustment on the sample intensities for each reference samples (Cytogenetics_Array only)
	   predicted-intensity-bin-count=20.\                   #number of bins to be used for pdnn adjustment
	   gc-bin-count=20.\                                    #number of bins to be used for gc-correction
	   residual-trim=2.0 \                                  #trim parameter for the probe level residual normalized intensities for high pass filter (Cytogenetics_Array only)
	--image-correction-intensity-adjustment-method high-pass-filter-intensity-adjustment-method.\ #probe level image correction using high pass filter (Cytogenetics_Array only)
	   data-block-rows=320.\                                #number of rows for Cytogenetics_Array chip
	   data-block-cols=2015.\                               #number of columns for Cytogenetics_Array chip
	   mini-block-rows=8.\                                  #parameter for high pass filter
	   mini-block-cols=8.\                                  #parameter for high pass filter
	   global-smooth-weight=256.\                           #parameter for high pass filter
	   local-smooth-weight=64.\                             #parameter for high pass filter
	   converged=0.0001 \                                   #parameter for high pass filter
	--cel-files CELFileList.txt                        #an ASCII file that contains the full path of input CEL files for reference creation, where the first row of the file is "cel_files" and each subsequent row corresponds to each CEL file
</pre></div><p><a class="anchor" id="ssa"></a> </p>
<h2><a class="anchor" id="ssa"></a>
Command line example for single sample mode processing</h2>
<div class="fragment"><pre class="fragment">
apt-copynumber-cyto  -v 4 \    #verbose output level in the log file ranging from 1 (least details) to 4 (most details)
	--cyto2 true \                 #true: Cytogenetics_Array; false: CytoScanHD_Array
	--force true                   #true: Disable various checks including chip types; false: otherwise.
	--run-geno-qc false \          #true: Run the GenoQC engine; false: otherwise. Should always be false for Cytogenetics_Array.
	--adapter-type-normalization false \ #true: perform adapter type normalization; false: otherwise. Should always be false for Cytogenetics_Array.
	--text-output false \                #true: Output data in ASCII text format in addition to calvin format; false: otherwise
	--cnchp-output false \               #true: report CNCHP file(s); false: otherwise. Either cnchp-output or cychp-output has to be true. Should be false for Cytogenetics_Array.
	--cychp-output true \                #true: report CYCHP file(s); false: otherwise. Either cnchp-output or cychp-output has to be true. Should be true for Cytogenetics_Array.
	--reference-input Cytogenetics_Array.na32.v1.REF_MODEL \ #specify the input copy number reference 
	--snp-reference-input-file Cytogenetics_Array.na32.v1.snpref.a5 \  #specify the input SNP reference 
	--set-analysis-name Cytogenetics \                       #Analysis name to use as prefix for output files.
	--cdf-file Cytogenetics_Array.CDF \                      #specify the chp definition file 
	--chrX-probes Cytogenetics_Array.chrXprobes \            #specify the chrX probe file
	--chrY-probes Cytogenetics_Array.v2.chrYprobes \         #specify the chrY probe file
	--annotation-file Cytogenetics_Array.na32.annot.db \     #specify the NetAffx database file
	--out-dir output_dir \                                   #specify the output directory
	--snp-qc-snp-list Cytogenetics_Array.snplist.txt \ #specify the snp list to compute the processed SNPQC. 
	--male-gender-ratio-cutoff=1.3 \                        #male gender call threshold for chrX/Y ratio gender calling 
	--female-gender-ratio-cutoff=1.0 \                      #female gender call threshold for chrX/Y ratio gender calling 
	--xx-cutoff=0.8 \                                       #lower limit of the gender call threshold for hasXX gender call 
	--xx-cutoff-high=1.07 \                                 #upper limit of the gender call threshold for hasXX gender call  
	--y-cutoff=0.65 \                                       #gender call threshold for hasY gender call 
	--waviness-block-size 50 \                              #control waviness segment count (QC)
	--waviness-genomic-span 0 \                             #control waviness segment count (QC)
	--analysis log2-ratio-cyto2.\                           #log2 ratio calculation analysis module
	   gc-correction=false.\                                #true: apply gc-correction; false: otherwise.
	   median-autosome-median-normalization=true.\          #true: apply median autosome median correction; false: otherwise.
	   median-smooth-marker-count=5.\                       #running median marker count for weighted log2 ratios (Cytogenetics_Array only)
	   trim-high=2.0.\                                      #trim parameter before any wave correction
	   trim-low=-2.5 \                                      #trim parameter before any wave correction
	--analysis kernel-smooth.\                              #apply kernel smooth to the log2 ratios
	   sigma_span=50 \                                      #bandwidth parameter for Gaussian smooth
	--analysis cn-cyto2.\                                   #Hidden Markove Model (HMM) analysis module
	   hmmCN_state=\'0,1,2,3,4,5\'.\                        #supported CN states
	   hmmCN_mu=\'-1.63,-0.58,0,0.41,0.65,0.9\'\.\          #HMM state means
	   hmmCN_sigma=\'0.3,0.3,0.3,0.3,0.3,0.3\'.\            #HMM state sigmas
	   diagonal-weight=0.995.\                              #transition probability matrix diagnal
	   mapd-weight=0.22.\                                   #per sample adjustment parameter to HMM state sigmas using MAPD
	   min-segment-size=5.\                                 #require min segment size (number of markers), except for the par regions
	   hmm-confidence-weight=0.6 \                          #parameter for HMM confidence
	--analysis cn-cyto2-gender.\                            #Y-Gender call analysis module
	   cutoff=0.5 \                                         #cutoff threshold for Y-Gender call
	--analysis cn-segment \                                 #generate HMM copy number call segment table. Should always specify this for Cytogenetics_Array.
	--analysis loh-cyto2.\                                  #LOH analysis module for Cytogenetics_Array
	   lohCNSegSeparation=1000000.\                         #LOH CN Separation
	   minInformation=100.\                                 #parameter that controls the size of the windows used by the algorithm
	   lambdaCritical=8.0000 \                              #the cut-off value for the test statistic used to decide between LOH and non-LOH.
	--analysis mosaicism.\                                  #mosaicism analysis module for Cytogenetics_Array
	   gains-boundries=\'0.08764945,0.15380349,0.21465931,0.27100300\'.\       #Mosaicism gains boundries
	   losses-boundries=\'-0.08293345,-0.17551812,-0.28048196,-0.40165383\'.\  #Mosaicism losses boundries
	   marker-bandwidth=6000.\                                                 #Mosaicism marker bandwitdth
	   confidence-window=251 \                                                 #Mosaicism confidence running median window size
	--analysis allele-peaks.\                    #allelic peak calculation module for Cytogenetics_Array
	   step=50.\                                 #step size for another density estimation across the genome
	   window=250.\                              #number of SNPs used by the kernel density estimator
	   point-count=128.\                         #number of output grids for the kernel density estimator
	   bandwidth=0.35.\                          #adjustment to the data-adaptive bandwidth for the kernel density estimator. Must be 0 &lt; bandwidth &lt;= 1.
	   cutoff=0.05.\                             #parameter to remove density peaks that are likely due to noise
	   clean-threshold=0.3.\                     #controls with noisy markers are removed from the visualization because they are indeterminately far from adjacent peaks.
	   symmetry=true \                           #true: the data is mirrored about the X axis before fitting the density curves; false: otherwise.
	--local-gc-background-intensity-adjustment-method pdnn-intensity-adjustment-method.\ #apply pdnn and local gc adjustment on the sample intensities (Cytogenetics_Array only)
	   predicted-intensity-bin-count=20.\        #number of bins to be used for pdnn adjustment
	   gc-bin-count=20.\                         #number of bins to be used for gc-correction
	   residual-trim=2.0 \                       #trim parameter for the probe level residual normalized intensities for high pass filter (Cytogenetics_Array only)
	--image-correction-intensity-adjustment-method high-pass-filter-intensity-adjustment-method.\ #probe level image correction using high pass filter (Cytogenetics_Array only) 
	   data-block-rows=320.\                     #number of rows for Cytogenetics_Array chip
	   data-block-cols=2015.\                    #number of columns for Cytogenetics_Array chip
	   mini-block-rows=8.\                       #parameter for high pass filter
	   mini-block-cols=8.\                       #parameter for high pass filter
	   global-smooth-weight=256.\                #parameter for high pass filter
	   local-smooth-weight=64.\                  #parameter for high pass filter
	   converged=0.0001 \                        #parameter for high pass filter
	--wave-correction-log2ratio-adjustment-method wave-correction-log2ratio-adjustment-method.\  #apply wave correction
	   bandwidth=101.\                           #used in non-parametric wave smoothing
	   bin-count=25.\                            #used in non-parametric wave smoothing
	   wave-count=6.\                            #must be less than or equal to the number of waves specified in wave-correction-reference-method
	   wave-smooth=true \                        #true: apply additional non-parametric wave smoothing; false: otherwise.
	--cel-files CELFileList.txt         #an ASCII file that contains the full path of input CEL files for single sample mode processing, where the first row of the file is "cel_files" and each subsequent row corresponds to each CEL file
</pre></div> </div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:56 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
