<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: VIGNETTES: Use of APT to Analyze WT Based Exon and Gene Expression Arrays</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li class="current"><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>VIGNETTES: Use of APT to Analyze WT Based Exon and Gene Expression Arrays </h1>  </div>
</div>
<div class="contents">
<dl class="date"><dt><b>Date:</b></dt><dd>2007-05-31</dd></dl>
<h2><a class="anchor" id="content"></a>
Contents</h2>
<ul>
<li><a href="#intro">Introduction</a><ul>
<li><a href="#introFiles">Analysis Library Files</a></li>
</ul>
</li>
<li><a href="#quickStartExon">Quick Start for Exon Arrays</a><ul>
<li><a href="#quickStartExonFiles">Download the analysis library files</a></li>
<li><a href="#quickStartExonGeneAnalysis">Use apt-probeset-summarize to compute <b>gene level estimates</b></a></li>
<li><a href="#quickStartExonExonAnalysis">Use apt-probeset-summarize to compute <b>exon level estimates</b></a></li>
</ul>
</li>
<li><a href="#quickStartGene">Quick Start for Gene Arrays</a><ul>
<li><a href="#quickStartGeneFiles">Download the analysis library files</a></li>
<li><a href="#quickStartGeneGeneAnalysis">Use apt-probeset-summarize to compute <b>gene level estimates</b></a></li>
<li><a href="#quickStartGeneExonAnalysis">Use apt-probeset-summarize to compute <b>exon level estimates</b></a></li>
</ul>
</li>
<li><a href="#otherOptions">Other Command Line Options</a></li>
<li><a href="#metaProbesetAlternatives">Annotation Quality and Meta Probeset Alternatives</a></li>
<li><a href="#analysisMethods">Analysis Methods</a></li>
<li><a href="#faq">FAQ - Frequently Asked Questions.</a></li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2><a class="anchor" id="intro"></a>
Introduction</h2>
<p>You can use the APT program, apt-probeset-summarize, to compute gene and exon level signal estimates from Exon and Gene array data.</p>
<p><a class="anchor" id="introFiles"></a> </p>
<h3><a class="anchor" id="introFiles"></a>
Analysis Library Files</h3>
<p>To use apt-probeset-summarize to analyze WT-based Gene and Exon expression arrays you first need to obtain the necessary library files:</p>
<ul>
<li>PGF: The <a class="el" href="classProbe.html" title="Represents an individual feature on a chip.">Probe</a> Group File indicates which probes are grouped together into a probeset.</li>
<li>CLF: The Cel Layout File indicates where the probes are located in the CEL file.</li>
<li>BGP: The BackGround <a class="el" href="classProbe.html" title="Represents an individual feature on a chip.">Probe</a> file indicates which probes are to be used for computing background.</li>
<li>QCC: Quality Control Contents file indicates which probesets should be treated as controls and how they are group together for quality assessment reporting purposes.</li>
<li>MPS: The <a class="el" href="classMeta.html" title="Helper class for meta probesets in apt-data-subset.cpp.">Meta</a> Probeset File is needed for gene level analysis. The MPS file defines a probeset as a collection of existing probesets. Typically these are different probesets for the same gene. Note that control probesets are also listed in the MPS file. This ensures that they are included in the quality assessment reports.</li>
<li>PS: The <a class="el" href="classProbeSet.html" title="Collection of atoms that measure target that should be absent or present at the same time...">ProbeSet</a> list file is only needed when an analysis on a subset of the probesets is desired. Note that control probesets are also listed in the PS file to ensure that they are included in the quality assessment reports.</li>
</ul>
<p>Note that you can specify either a MPS or PS file, not both in a single analysis.</p>
<p><a class="anchor" id="quickStartExon"></a> </p>
<h2><a class="anchor" id="quickStartExon"></a>
Quick Start Exon Array</h2>
<p><a class="anchor" id="quickStartExonFiles"></a> </p>
<h3><a class="anchor" id="quickStartExonFiles"></a>
Step 1: Download the analysis library files</h3>
<p>For Exon Arrays (ie Human Exon 1.0 ST Array), all of the files needed are in the analysis library file package (zip archive file) that can be downloaded from the respective <a href="http://www.affymetrix.com/support/technical/byproduct.affx?cat=exparrays">array support page</a>. (For example, the <a class="el" href="classAnalysis.html">Analysis</a> zip file under the "Library Files" section of the <a href="http://www.affymetrix.com/support/technical/byproduct.affx?product=huexon-st">Human Exon 1.0 ST Array - Support Materials</a> page.)</p>
<p>Direct links for various exon array analysis library file packages:</p>
<ul>
<li><a href="http://www.affymetrix.com/Auth/support/downloads/library_files/HuEx-1_0-st-v2.r2.zip">http://www.affymetrix.com/Auth/support/downloads/library_files/HuEx-1_0-st-v2.r2.zip</a></li>
<li><a href="http://www.affymetrix.com/Auth/support/downloads/library_files/MoEx-1_0-st-v1_libraryfile.zip">http://www.affymetrix.com/Auth/support/downloads/library_files/MoEx-1_0-st-v1_libraryfile.zip</a></li>
<li><a href="http://www.affymetrix.com/Auth/support/downloads/library_files/RaEx-1_0-st-v1_libraryfile.zip">http://www.affymetrix.com/Auth/support/downloads/library_files/RaEx-1_0-st-v1_libraryfile.zip</a></li>
</ul>
<p>NOTE: If you have Expression Console (EC) installed on a windows box, you can use EC to download the library files and then simply copy the library files you need from the EC library file folder.</p>
<p>NOTE: Earlier versions (ie prior to June 2007) of the analysis library file zip archive did not include the MPS files.</p>
<p><a class="anchor" id="quickStartExonGeneAnalysis"></a> </p>
<h3><a class="anchor" id="quickStartExonGeneAnalysis"></a>
Step 2: Use apt-probeset-summarize to compute &lt;b&gt;gene level estimates&lt;/b&gt;</h3>
<p>Here is a basic example you might run from the bash *NIX shell: </p>
<div class="fragment"><pre class="fragment">
    apt-probeset-summarize \
        -p HuEx-1_0-st-v2.r2.pgf \
        -c HuEx-1_0-st-v2.r2.clf \
        -b HuEx-1_0-st-v2.r2.antigenomic.bgp \
        --qc-probesets HuEx-1_0-st-v2.r2.qcc \
        -m HuEx-1_0-st-v2.r2.dt1.hg18.core.mps \
        -a rma-sketch \
        -o output-gene \
        *.CEL
</pre></div><p>This assumes that apt-probeset-summarize is in your PATH. One might wonder why a bgp file is specified when we are not using the pm-gcbg pm adjustor (rma-sketch uses pm-only). The reason is that APT will still use the bgp information to compute the bgrd mean quality metric in the report file.</p>
<p>Note under Windows you need to make a couple changes compared to the *NIX command line. First everything must be on the same line (no "\" continuations). Second, the wild card "*.CEL" will not work. You need to either list every CEL file on the command line or use the --cel-files option. Here is an example using the later: </p>
<div class="fragment"><pre class="fragment">
    ./apt-probeset-summarize -p HuEx-1_0-st-v2.r2.pgf -c HuEx-1_0-st-v2.r2.clf -b HuEx-1_0-st-v2.r2.antigenomic.bgp --qc-probesets HuEx-1_0-st-v2.r2.qcc -m HuEx-1_0-st-v2.r2.dt1.hg18.core.mps -a rma-sketch -o output-gene --cel-files celfiles.txt
</pre></div><p>The format of the file specified with the --cel-files option is a tab-separated file with a header line containing "cel_files". For example: </p>
<div class="fragment"><pre class="fragment">
cel_files	rep	tissue
heart1.CEL	1	heart
heart2.CEL	2	heart
heart3.CEL	3	heart
brain1.CEL	1	brain
brain2.CEL	2	brain
brain3.CEL	3	brain
</pre></div><p>The notable option that makes this a gene level analysis is the inclusion of a meta probeset file, HuEx-1_0-st-v2.r2.dt1.hg18.core.mps, with the "-m" option.</p>
<p>The analysis is specified with the "-a" option. This can be either a full analysis specification or the name of a pre-canned analysis specification. For example the pre-canned "rma-sketch" analysis method could also be specified using the full analysis specification "rma-bg,quant-norm.sketch=-1.usepm=true.bioc=true,pm-only,med-polish". See the <a href="#analysisMethods">Analysis Methods</a> section for other analysis options.</p>
<p>Output files will be generated in the "output-gene" subfolder. <b>WARNING:</b> apt-probeset-summarize will overwrite previously generated results. With the example command above, the following output files will be created:</p>
<ul>
<li>rma-sketch.summary.txt: This tab separated text file contains a row for each probeset (corresponds to a transcript_cluster_id in this example) and a row for each chip. The results are the gene level signal estimates.</li>
<li>rma-sketch.report.txt: This tab separated test file contains a row for each chip and a column for each quality assessment metric. See the <a href="http://www.affymetrix.com/support/technical/whitepapers/exon_gene_arrays_qa_whitepaper.pdf">Quality Assessment of Exon and Gene Arrays</a> white paper for more information on the various quality metrics.</li>
<li>apt-probeset-summarize.log: This is a log file with various runtime information. When contacting Affymetrix for help, please send the contents of this file along with the command line that your running.</li>
</ul>
<p>Here is an example of running PLIER with the GC-bin background correction under the bash *NIX shell: </p>
<div class="fragment"><pre class="fragment">
    apt-probeset-summarize \
        -p HuEx-1_0-st-v2.r2.pgf \
        -c HuEx-1_0-st-v2.r2.clf \
        -b HuEx-1_0-st-v2.r2.antigenomic.bgp \
        --qc-probesets HuEx-1_0-st-v2.r2.qcc \
        -m HuEx-1_0-st-v2.r2.dt1.hg18.core.mps \
        -a plier-gcbg-sketch \
        -o output-gene \
        *.CEL
</pre></div><p>In this case the same files will be generated, but with a "plier-gcbg-sketch" prefix rather than "rma-sketch". Also note that the bgp file is required due to the use of GC-bin background correction. In the <a class="el" href="classRMA.html" title="Holds methods for computing the popular Robust Muliarray Average of Microarrays (RMA).">RMA</a> case above the bgp file was option and only included to ensure the production of the bgrd mean quality assessment metric.</p>
<p>Note that these two analysis could be performed in a single APT run by including both "-a" parameters. For example: </p>
<div class="fragment"><pre class="fragment">
    apt-probeset-summarize \
        -p HuEx-1_0-st-v2.r2.pgf \
        -c HuEx-1_0-st-v2.r2.clf \
        -b HuEx-1_0-st-v2.r2.antigenomic.bgp \
        --qc-probesets HuEx-1_0-st-v2.r2.qcc \
        -m HuEx-1_0-st-v2.r2.dt1.hg18.core.mps \
        -a plier-gcbg-sketch \
        -a rma-sketch \
        -o output-gene \
        *.CEL
</pre></div><p><a class="anchor" id="quickStartExonExonAnalysis"></a> </p>
<h3><a class="anchor" id="quickStartExonExonAnalysis"></a>
Step 3: Use apt-probeset-summarize to compute &lt;b&gt;exon level estimates&lt;/b&gt;</h3>
<p>Here is a basic example you might run from the bash *NIX shell: </p>
<div class="fragment"><pre class="fragment">
    apt-probeset-summarize \
        -p HuEx-1_0-st-v2.r2.pgf \
        -c HuEx-1_0-st-v2.r2.clf \
        -b HuEx-1_0-st-v2.r2.antigenomic.bgp \
        --qc-probesets HuEx-1_0-st-v2.r2.qcc \
        -s HuEx-1_0-st-v2.r2.dt1.hg18.core.ps \
        -a rma-sketch \
        -o output-gene \
        *.CEL
</pre></div><p>The only change compared to the gene level example above is the use of a probeset list instead of a meta probeset file. By specifying the probeset list HuEx-1_0-st-v2.r2.dt1.hg18.core.ps with the "-s" option, we are restricting the analysis to just the "core" supported exons rather than the whole chip. Note that this probeset list file includes controls, so they will be present in the output as well. One could omit the "-s" in which case all the probesets in the PGF file would be processed. This would also result in an exon level result file with the inclusion of various controls.</p>
<p><a class="anchor" id="quickStartGene"></a> </p>
<h2><a class="anchor" id="quickStartGene"></a>
Quick Start Gene Array</h2>
<p><a class="anchor" id="quickStartGeneFiles"></a> </p>
<h3><a class="anchor" id="quickStartGeneFiles"></a>
Step 1: Download the analysis library files</h3>
<p>For Gene Arrays (ie Human Gene 1.0 ST Array), all the needed files are in the analysis library file package (zip archive file) that can be downloaded from the respective <a href="http://www.affymetrix.com/support/technical/byproduct.affx?cat=exparrays">array support page</a>. (For example, the <a class="el" href="classAnalysis.html">Analysis</a> zip file under the "Library Files" section of the <a href="http://www.affymetrix.com/support/technical/byproduct.affx?product=hugene-1_0-st-v1">Human Gene 1.0 ST Array - Support Materials</a> page.)</p>
<p>Direct links for various gene array analysis library file packages:</p>
<ul>
<li><a href="http://www.affymetrix.com/Auth/support/downloads/library_files/HuGene-1_0-st-v1.r4.analysis_libraryfile.zip">http://www.affymetrix.com/Auth/support/downloads/library_files/HuGene-1_0-st-v1.r4.analysis_libraryfile.zip</a></li>
</ul>
<p>NOTE: If you have Expression Console (EC) installed on a windows box, you can use EC to download the library files and then simply copy the library files you need from the EC library file folder.</p>
<p>NOTE: The original Human, Mouse, Rat 1.0 ST Gene array library files did not support exon level analysis, so be sure that you have the latest library file versions for these arrays (r4 or later).</p>
<p><a class="anchor" id="quickStartGeneGeneAnalysis"></a> </p>
<h3><a class="anchor" id="quickStartGeneGeneAnalysis"></a>
Step 2: Use apt-probeset-summarize to compute gene level estimates</h3>
<p>Here is a basic example you might run from the bash *NIX shell: </p>
<div class="fragment"><pre class="fragment">
    apt-probeset-summarize \
        -p HuGene-1_0-st-v1.r4.pgf \
        -c HuGene-1_0-st-v1.r4.clf \
        -b HuGene-1_0-st-v1.r4.bgp \
        --qc-probesets HuGene-1_0-st-v1.r4.qcc \
        -m HuGene-1_0-st-v1.r4.mps \
        -a rma-sketch \
        -o output-gene \
        *.CEL
</pre></div><p>This assumes that apt-probeset-summarize is in your PATH.</p>
<p>Note under Windows you need to make a couple changes compared to the *NIX command line. First everything must be on the same line (no "\" continuations). Second, the wild card "*.CEL" will not work. You need to either list every CEL file on the command line or use the --cel-files option. Here is an example using the later: </p>
<div class="fragment"><pre class="fragment">
    ./apt-probeset-summarize -p HuGene-1_0-st-v1.r4.pgf -c HuGene-1_0-st-v1.r4.clf -b HuGene-1_0-st-v1.r4.bgp --qc-probesets HuGene-1_0-st-v1.r4.qcc -m HuGene-1_0-st-v1.r4.mps -a rma-sketch -o output-gene --cel-files celfiles.txt
</pre></div><p>The celfiles.txt file is a tab separated file with a header line. There must be a column "cel_files". Here is an example: </p>
<div class="fragment"><pre class="fragment">
cel_files
heart1.CEL
heart2.CEL
heart3.CEL
brain1.CEL
brain2.CEL
brain3.CEL
</pre></div><p>The notable option that makes this a gene level analysis is the inclusion of a meta probeset file, HuGene-1_0-st-v1.r4.mps, with the "-m" option.</p>
<p>The analysis is specified with the "-a" option. This can be either a full analysis specification or the name of a pre-canned analysis specification. For example the pre-canned "rma-sketch" analysis method could also be specified using the full analysis specification "rma-bg,quant-norm.sketch=-1.usepm=true.bioc=true,pm-only,med-polish". See the <a href="#analysisMethods">Analysis Methods</a> section for other analysis options.</p>
<p>Output files will be generated in the "output-gene" subfolder. <b>WARNING:</b> apt-probeset-summarize will overwrite previously generated results. With the example command above, the following output files will be created:</p>
<ul>
<li>rma-sketch.summary.txt: This tab separated text file contains a row for each probeset (corresponds to a transcript_cluster_id in this example) and a row for each chip. The results are the gene level signal estimates.</li>
<li>rma-sketch.report.txt: This tab separated test file contains a row for each chip and a column for each quality assessment metric. See the <a href="http://www.affymetrix.com/support/technical/whitepapers/exon_gene_arrays_qa_whitepaper.pdf">Quality Assessment of Exon and Gene Arrays</a> white paper for more information on the various quality metrics.</li>
<li>apt-probeset-summarize.log: This is a log file with various runtime information. When contacting Affymetrix for help, please send the contents of this file along with the command line that your running.</li>
</ul>
<p><a class="anchor" id="quickStartGeneExonAnalysis"></a> </p>
<h3><a class="anchor" id="quickStartGeneExonAnalysis"></a>
Step 3: Use apt-probeset-summarize to compute &lt;b&gt;exon level estimates&lt;/b&gt;</h3>
<p>Here is a basic example you might run from the bash *NIX shell: </p>
<div class="fragment"><pre class="fragment">
    apt-probeset-summarize \
        -p HuGene-1_0-st-v1.r4.pgf \
        -c HuGene-1_0-st-v1.r4.clf \
        -b HuGene-1_0-st-v1.r4.bgp \
        --qc-probesets HuGene-1_0-st-v1.r4.qcc \
        -s HuGene-1_0-st-v1.r4.all.ps \
        -a rma-sketch \
        -o output-gene \
        *.CEL
</pre></div><p>The only change compared to the gene level example above is the use of a probeset list instead of a meta probeset file. By specifying the probeset list HuGene-1_0-st-v1.r4.all.ps with the "-s" option, we are restricting the analysis to just the "core" supported exons rather than the whole chip. Note that this probeset list file includes controls, so they will be present in the output as well. One could omit the "-s" in which case all the probesets in the PGF file would be processed. This would also result in an exon level result file with the inclusion of various controls.</p>
<p>NOTE: With older versions (r3) of the Human, Mouse, and Rat 1.0 ST gene array library files omission of the meta probeset file will still result in a gene level analysis. This is in part due to the fact that probes were already grouped by gene in the PGF file. The reason to include the meta probeset file is it will exclude certain control probesets from the analysis (ie we do not want to compute <a class="el" href="classRMA.html" title="Holds methods for computing the popular Robust Muliarray Average of Microarrays (RMA).">RMA</a> estimates for the various GC background probe bins) and it will propagate information into the output files that a meta probeset file was used. There may be downstream analysis software that uses this information to distinguish a gene level analysis from an exon level analysis.</p>
<p><a class="anchor" id="#otherOptions"></a> </p>
<h2><a class="anchor" id="otherOptions"></a>
Other Command Line Options</h2>
<p>apt-probeset-summarize also has a number of other options. As you get more comfortable using APT, you should take some time to familiarize yourself with these options. Some of the options that you may want to use include:</p>
<ul>
<li>cc-chp-output: This option enable AGCC CHP output files (in addition to the text summary.txt file). Note that you cannot use the --xda-chp-output option with WT based arrays. This is because some of the assumptions made in the XDA file format are not valid for these arrays.</li>
<li>feature-effects: This option will create a tab separated output file with the computed feature effects when using a model based estimator such as the med-polish (as used by <a class="el" href="classRMA.html" title="Holds methods for computing the popular Robust Muliarray Average of Microarrays (RMA).">RMA</a>) or PLIER.</li>
<li>feature-details: This option will output the residuals for all the features (or in the case of DABG it will output the probe level P-values). Note that these files can be huge as there is a data point for every array by every probe.</li>
</ul>
<p>See the <a href="apt-probeset-summarize.html">apt-probeset-summarize Manual</a> and the online help (run "apt-probeset-summarize -h") for more information.</p>
<p><a class="anchor" id="#metaProbesetAlternatives"></a> </p>
<h2><a class="anchor" id="metaProbesetAlternatives"></a>
Annotation Quality and Meta Probeset Alternatives</h2>
<p>The use of more speculative content in the meta probeset files will pull in more genes, but it may also attenuate the signal values reported for all genes depending on the analysis method. In general, new analysis methods (ie IterPLIER or PCA Feature Selection) are recommended when dealing with more speculative content. See the <a href="#analysisMethods">Analysis Methods</a> section for more information about alternative analysis methods.</p>
<p>Two key resources for more information on this topic:</p>
<ul>
<li>A detailed description of exon array annotations and how exons are grouped into genes is discussed in the <a href="http://www.affymetrix.com/support/technical/whitepapers/exon_probeset_trans_clust_whitepaper.pdf">Exon Probeset Annotations and Transcript Cluster Groupings</a> white paper.</li>
<li>Implications for the use of more speculative exons in gene level estimates is discussed in the <a href="http://www.affymetrix.com/support/technical/whitepapers/exon_gene_signal_estimate_whitepaper.pdf">Gene Signal Estimates from Exon Arrays</a> white paper.</li>
</ul>
<p><a class="anchor" id="#analysisMethods"></a> </p>
<h2><a class="anchor" id="analysisMethods"></a>
Analysis Methods</h2>
<p>A variety of analysis methods are implemented in apt-probeset-summarize and you can combine various analysis components to create new methods as well. More information is in the <a href="apt-probeset-summarize.html">apt-probeset-summarize Manual</a>.</p>
<p>With regard to gene level analysis, the following two sources contain important information:</p>
<ul>
<li>Performance comparison of PLIER and IterPLIER with different annotation groupings is described in the <a href="http://www.affymetrix.com/support/technical/whitepapers/exon_gene_signal_estimate_whitepaper.pdf">Gene Signal Estimates from Exon Arrays</a> white paper.</li>
<li>The PCA Feature Selection method is described in a talk by Chuck Sugnet given at the Affymetrix Low Level <a class="el" href="classAnalysis.html">Analysis</a> Workshop. See <a href="https://www.affymetrix.com/corporate/events/low_level_workshop/2007_04_low_level_workshop.affx">Dynamically Selecting Probes for Gene-Level Expression Estimates Using a Principal Components Based Approach</a> for more information on this method.</li>
</ul>
<p>Some of the methods you may want to consider:</p>
<ul>
<li>rma-sketch<br/>
 This is like the original <a class="el" href="classRMA.html" title="Holds methods for computing the popular Robust Muliarray Average of Microarrays (RMA).">RMA</a> method, however a more memory efficient and fast approximation is used instead of a full quantile normalization.</li>
<li>plier-gcbg-sketch<br/>
 This is a PLIER based method using the sketch approximation of quantile normalization and it includes a GC content based background correction using the probes listed in the bgp file.</li>
<li>quant-norm.sketch=-1.bioc=false,pm-gcbg,iter-plier<br/>
 This is the fully specified analysis string for running IterPLIER. Use of this method has been explored when dealing with very speculative content (ie gene level estimates computed on the "full" meta probeset file). You may be better off using the PCA based feature selection methods listed below.</li>
<li>rma-bg,quant-norm.sketch=-1.usepm=true.bioc=true,pm-only,med-polish,pca-select<br/>
 This is the fully specified analysis string for "rma-sketch" with the addition of a PCA based feature selection prior to the median polish step.</li>
<li>quant-norm.sketch=-1.bioc=false,pm-gcbg,plier,pca-select<br/>
 This is the fully specified analysis string for "plier-gcbg-sketch" with the addition of a PCA based feature selection prior to the median polish step.</li>
</ul>
<p><a class="anchor" id="faq"></a> </p>
<h2><a class="anchor" id="manFaq"></a>
Frequently Asked Questions</h2>
<p>Q. The "full" meta-probeset file for the Human Exon 1.0 ST Array does not use all 1.4 million probesets in the probeset_list column. Why not?</p>
<p>A. Probesets that either do not map uniquely to the genome or are classified as bounded probesets are not included in the MPS file and thus are not used in the calculation of gene level signals. Bounded probesets are probesets hitting an exon that falls within a larger gene structure, but for which there is no annotation evidence that the exon is spliced to other parts of the gene structure. For example, there may be genscan single exon sub-optimal prediction that falls within a known gene, but this single exon prediction is not associated with the other parts of the gene structure by other spliced annotations (See the <a href="http://www.affymetrix.com/support/technical/whitepapers/exon_probeset_trans_clust_whitepaper.pdf">Exon Probeset Annotations and Transcript Cluster Groupings</a> whitepaper for more details.)</p>
<p>Q. A core exon analysis using PLIER in Expression Console produces signal estimates for 287,329 exon probeset IDs. However, if I look at the HuEx-1_0-st-probeset-annot.csv file or the HuEx-1_0-st-v2.r2.dt1.hg18.csv there are only 284,258 core exon probeset IDs in these two files. Why are there approximately 3,000 more probeset IDs in the exon core summary that are not included in the annotation files?</p>
<p>A. The library files for the exon array include the various control probesets in the probeset list files (*.ps) and the meta probeset files (*.mps). This ensures that the various controls are processed and that the QC report file is fully populated regardless of the analysis type. So the additional probesets are the various control probesets. The majority of these are the intron/exon controls (listed in QC report file as positive and negative controls) that are used to compute the positive/negative area under the curve (AUC) QC metric.</p>
<p>Q. The number of results reported from a Gene Array are greater than the number of probesets listed in the CSV annotation file. Why?</p>
<p>A. There are a number of controls that are included in the MPS files to ensure that all of the control metrics can be calculated. Some of these controls are not listed in the CSV annotation file.</p>
<p>Q. Where can I get meta probeset files?</p>
<p>A. For the Exon and Gene Arrays, the default meta probeset files are provided in the analysis library file zip archive.</p>
<p>For the Human, Mouse, and Rat Exon 1.0 ST Arrays, additional meta probeset files can be obtained from the respective array support page. You need to download one of the "Design Time Annotation Files". Specifically, one of the "Probeset" (not "Full") annotation files. For example, MoEx-1_0-st-v1 Annotations, Probeset, Mm6, CSV.</p>
<p>Note that prior to around June 2007 there were no meta probeset files in the Exon Array analysis library file zip archive.</p>
<p><b></p>
<pre>Affymetrix Power Tools (APT) Release 1.16.1</pre><p></b> </p>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Tue Jul 15 2014 10:37:57 for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
