<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Affymetrix Power Tools: VIGNETTES: OncoScan Single Sample Analysis</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">Affymetrix Power Tools
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li class="current"><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<div class="title">VIGNETTES: OncoScan Single Sample Analysis </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><dl class="section date"><dt>Date</dt><dd>2013-10-23</dd></dl>
<h1><a class="anchor" id="content"></a>
Contents</h1>
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#libraryfiles">Library Files</a></li>
<li><a href="#ssa">Command line example for single sample copynumber processing</a></li>
<li><a href="#som">Command line example for single sample somatic mutation processing</a></li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h1><a class="anchor" id="intro"></a>
Introduction</h1>
<p>This vignette describes the process for using apt-copynumber-onco-ssa and apt-copynumber-onco-som-ssa to process FFPE samples run on OncoScan arrays in matched-normal or single sample mode.</p>
<p><a class="anchor" id="libraryfiles"></a> </p>
<h1><a class="anchor" id="libraryfiles"></a>
Library Files</h1>
<p>The following library files are required for analysis of OncoScan data.</p>
<pre class="fragment">  OncoScan.r1.snplist.txt               - SNP probes list
  OncoScan.FFPE.na33.r1.REF_MODEL       - Reference model file for apt-copynumber-onco-ssa
  OncoScan.Ref103.na33.r1.SOM_REF_MODEL - Reference model file for apt-copynumber-onco-som-ssa
  OncoScan.na33.r1.annot.db             - Annotation file for OncoScan arrays
  OncoScan.r1.chrXprobes                - X chromosome probes
  OncoScan.r1.chrYprobes                - Y chromosome probes
  OncoScan.Som1.0.r1.Som_thresh.txt     - Thresholds for somatic mutation calls
</pre><p><a class="anchor" id="ssa"></a> </p>
<h2>Command line example for Copynumber Single Sample Analysis </h2>
<p>The following assumes that the environment variables <code>$OUTDIR</code> and <code>$LIBDIR</code> point to the locations of the output directory and the library files directory respectively.</p>
<pre class="fragment">apt-copynumber-onco-ssa \
    --log-file $OUTDIR/apt-copynumber-onco-ssa.log \
    --out-dir $OUTDIR \
    --temp-dir $OUTDIR/temp \
    --force false \
    --keep-intermediate-data false \
    --keep-combined-cels false \
    --snp-qc-snp-list $LIBDIR/OncoScan.r1.snplist.txt \
    --reference-file $LIBDIR/OncoScan.FFPE.na33.r1.REF_MODEL \
    --y-target -0.8 \
    --nddqc-min-count 2000 \
    --set-analysis-name mip240k \
    --brlmmp-HARD 3 \
    --brlmmp-SB 0.750000 \
    --brlmmp-CM 2 \
    --brlmmp-bins 100 \
    --brlmmp-mix 1 \
    --brlmmp-bic 2 \
    --brlmmp-CSepPen 0.1 \
    --brlmmp-CSepThr 16 \
    --brlmmp-lambda 1.0 \
    --brlmmp-wobble 0.20 \
    --brlmmp-copyqc 0.000000 \
    --brlmmp-copytype -1 \
    --brlmmp-ocean 0.0000000000000001 \
    --brlmmp-clustertype 2 \
    --brlmmp-transform mva \
    --brlmmp-MS 0.06 \
    --annotation-file $LIBDIR/OncoScan.na33.r1.annot.db \
    --x-probes-file $LIBDIR/OncoScan.r1.chrXprobes \
    --y-probes-file $LIBDIR/OncoScan.r1.chrYprobes \
    --cel-pairs-file CELPairsFile.txt \
    --alpha-cn-calibrate 0.780931 \
    --alpha-X-cn-calibrate 0.800507 \
    --alpha-Y-cn-calibrate 0.800507 \
    --beta-cn-calibrate 1 \
    --beta-X-cn-calibrate 1 \
    --beta-Y-cn-calibrate 1 \
    --agr-denominator-source SNP \
    --agr-denominator-percentile 0.75 \
    --igender-female-threshold 0.44 \
    --igender-male-threshold 0.63 \
    --nddicov-min-nd-probe-count-per-bin 30 \
    --nddicov-use-nd-signal-markers off \
    --dual-channel-normalization false \
    --ad-outlier-trim 3.0 \
    --normal-diploid-detection-node:enable=true \
    --ndd-weights-type FLD \
    --ndd-point-count 128 \
    --ndd-bandwidth 0.25 \
    --ndd-cutoff 0.05 \
    --ndd-step 40 \
    --ndd-window 275 \
    --ndd-min-markers 10 \
    --ndd-log2-threshold 0.28 \
    --ndd-th1 0,0.4 \
    --ndd-th2 0.7,1.3 \
    --ndd-th3 0.7,1.3 \
    --ndd-th4 0,0.4 \
    --ndd-min-percent-nd 0 \
    --ndd-min-number-nd 2000 \
    --ndd-med-l2r-low-percentile-ND 0.015 \
    --ndd-med-l2r-threshold-ND 0.20 \
    --ndd-iqr-factor-coarse 5.0 \
    --ndd-iqr-factor-fine 3.0 \
    --sig-gender-reference-chromosome 2 \
    --sig-gender-xx-cutoff 0.85 \
    --sig-gender-xx-cutoff-high 1.22 \
    --sig-gender-y-cutoff 0.31 \
    --hpf-mini-block-rows 8 \
    --hpf-mini-block-cols 8 \
    --hpf-global-smooth-weight 256 \
    --hpf-local-smooth-weight 64 \
    --hpf-converged 0.000100 \
    --nddlrcov-min-nd-marker-count-per-bin 30 \
    --nddlrcov-use-nd-lr-markers on \
    --log2-ratio-adjuster-wave-node:enable=true \
    --wave-bandwidth 101 \
    --wave-bin-count 25 \
    --wave-count 6 \
    --wave-smooth true \
    --median-autosome-median-normalization true \
    --shrinkage-v2-node:enable=true \
    --shrink-lprec 0.25,2.0 \
    --outlier-shrink-lprec 0.25,2.0,8.0 \
    --shrink-converge 0.000001 \
    --outlier-shrink-converge 0.1 \
    --shrink-downweight-maxiter 25 \
    --outlier-shrink-large-stdev 125.0 \
    --cn-gender-cutoff 0.5 \
    --cn-state-threshold 0.0 \
    --baf-transform linear \
    --baf-trim 10.0 \
    --kernel-sigma-span 50 \
    --sig-snp-min-call-rate 0.6 \
    --sig-snp-min-compare-rate 0.65 \
    --sig-snp-min-concordance 0.60 \
    --sig-snp-min-strength-sketch-rank 0.305 \
    --sig-snp-report-filename $OUTDIR/OncoScan.CelPairCheckReport.txt \
    --ap-ad-step 20 \
    --ap-ad-window 100 \
    --ap-ad-point-count 128 \
    --ap-ad-bandwidth 0.250000 \
    --ap-ad-cutoff 0.050000 \
    --ap-ad-threshold 0.350000 \
    --ap-ad-height-threshold 0.000000 \
    --ap-ad-height-threshold-bound 1024.000000 \
    --ap-ad-symmetry true \
    --ap-baf-step 20 \
    --ap-baf-window 100 \
    --ap-baf-point-count 129 \
    --ap-baf-bandwidth 0.2 \
    --ap-baf-cutoff 0.00000 \
    --ap-baf-threshold 0.30000 \
    --ap-baf-height-threshold 3.000000 \
    --ap-baf-height-threshold-bound 1.476981 \
    --ap-baf-symmetry true \
    --loh-error-rate 0.05 \
    --loh-beta 0.001 \
    --loh-alpha 0.01 \
    --loh-separation 1000000 \
    --loh-min-marker-count 10 \
    --loh-no-call-threshold 0.05 \
    --loh-min-genomic-span 2500000 \
    --loh-ss-thresh 3.59 \
    --loh-ss-begin-slope 1.6416 \
    --loh-baf-upper-thresh 0.5 \
    --loh-baf-lower-thresh 0.5 \
    --loh-slope-thresh 0.25 \
    --loh-alpha-cn-calibrate 0.780931 \
    --loh-alpha-X-cn-calibrate 0.800507 \
    --loh-alpha-Y-cn-calibrate 0.800507 \
    --loh-beta-cn-calibrate 1 \
    --loh-beta-X-cn-calibrate 1 \
    --loh-beta-Y-cn-calibrate 1 \
    --cn-neutral-loh-node:enable=false \
    --tuscan-m3-low-cost-band 0.030000 \
    --tuscan-nd-snp-qc-threshold 19.000000 \
    --tuscan-mapd-qc-threshold 0.4000000 \
    --tuscan-m1-shift 0.000000 \
    --tuscan-m1-trim 0.010000 \
    --tuscan-m1-threshold 0.001050 \
    --tuscan-m2-lr-cost 0.500000 \
    --tuscan-m2-min-seg-size 151 \
    --tuscan-m2-min-alt-seg-size 19 \
    --tuscan-m2-min-fine-seg-size 19 \
    --tuscan-m2-exclude-chr 24,25 \
    --tuscan-consecutive-hom-count 100 \
    --tuscan-no-break-vicinity 19 \
    --lambda 50.000000 \
    --tuscan-m3-rho-min 0.200000 \
    --tuscan-m3-rho-max 1.000000 \
    --tuscan-m3-rho-N 100 \
    --tuscan-m3-psi-min 1.000000 \
    --tuscan-m3-psi-max 6.000000 \
    --tuscan-m3-psi-N 100 \
    --tuscan-m3-low-cost-weight 0.010000 \
    --tuscan-m3-max-percent-hom 85.000000 \
    --tuscan-m3-min-count-het 10 \
    --tuscan-m3-ploidy-factor 3.000000 \
    --tuscan-m3-ploidy-sigma 0.350000 \
    --alt-mode-bin-size 0.333333 \
    --nGoF-bin-size 0.500000 \
    --MedL2Rcutoff 0.080000 \
    --MedBAFcutoff 0.000000 \
    --min-nGoF-length 1001 \
    --nGoF4cutoff 1000.000000 \
    --gamma 0.800000 \
    --tuscan-round-ploidy true \
    --tuscan-m1-use-raw-baf false \
    --infMarkL2RMin 0.06 \
    --infMarkerThreshold 1000 \
    --MinChrYthreshold -3 \
    --MedL2Rcutoff-no-cn2 0.08 \
    --MedBAFcutoff-no-cn2 0 \
    --min-nGoF-length-no-cn2 51 \
    --nGoF4cutoff-no-cn2 5000 \
    --tuscan-fp-exponent 2.0 \
    --tuscan-fp-min-med-l2r 0.1 \
    --tuscan-fp-max-med-l2r 0.25 \
    --tuscan-use-fp-exponent-in-normal-mode false \
    --tuscan-cn-bin '[0,6)[6,10)[10,20)[20,50)[50,10000]' \
    --tuscan-cn-bin-values 'as-is-2,6-10,10-20,20-50,50+' \
    --tuscan-run-once false \
    --tuscan-maxCN2Weight 0.025 \
    --tuscan-maxNonCN2Increase 1 \
    --tuscan-FN_l2r_floor 0.095 \
    --tuscan-FN_exp 2.46 \
    --tuscan-FN_low_l2r_threshold 0.08 \
    --tuscan-FN_low_BAF_threshold 0.48 \
    --tuscan-REL_l2r_weight 0.5 \
    --tuscan-REL_CN2_gauss_P 1.1 \
    --tuscan-REL_CN2_gauss_sigma 0.35 \
    --tuscan-REL_CN2_gauss_mu 2 \
    --tuscan-REL_CN4_gauss_P 1.023 \
    --tuscan-REL_CN4_gauss_sigma 0.2 \
    --tuscan-REL_CN4_gauss_mu 4 \
    --tuscan-REL_threshold 0.1 \
    --tuscan-REL_threshold1 0.76 \
    --tuscan-REL_threshold2_hi 0.9 \
    --tuscan-REL_threshold2_low 0.7 \
    --tuscan-REL_maxMinimumIndex 2 \
    --tuscan-REL_noSearchIfminWeight true \
    --tuscan-infMarkTrimBand 0.05 \
    --tuscan-m3-ploidy4-factor 1.25 \
    --tuscan-m3-ploidy4-sigma 0.2 \
    --tuscan-m3-ploidy4-mu 4.0 \
    --tuscan-lowPloidy1 0 \
    --tuscan-lowPloidy2 1.76 \
    --tuscan-lowPloidyTB 1 \
    --tuscan-lowPloidy_nGoF4 200 \
    --tuscan-hiPloidy1 2.3 \
    --tuscan-hiPloidy2 2.5 \
    --tuscan-hiPloidyTB 1 \
    --tuscan-hiPloidy_nGoF4 200 \
    --tuscan-centromere-breakpoints 1,121535434,2,92326171,3,90504854,4,49660117,5,46405641,6,58830166,7,58054331,8,43838887,9,47367679,10,39254935,11,51644205,12,34856694,13,-1,14,-1,15,-1,16,35335801,17,22263006,18,15460898,19,24681782,20,26369569,21,-1,22,-1,24,58632012,25,10104553 \
    --tuscan-centromere-min-seg-size 19 \
    --tuscan-report-internal-ploidy true \
    --report-file $OUTDIR/QCReport.txt \
    --cychp-output false \
    --text-output false \
    --oschp-output true \
    --mangle-probeset-names false \
    --waviness-block-size 50 \
    --waviness-genomic-span 0 
</pre><p>Note: This example is in Unix/bash shell syntax. For Windows command prompt adjustments must be made to the file path syntax, and the '\' character can not be used to escape the end of a line.</p>
<p><a class="anchor" id="som"></a> </p>
<h2>Command line example for Somatic Mutation Single Sample Analysis </h2>
<p>The following assumes that the environment variables <code>$OUTDIR</code> and <code>$LIBDIR</code> point to the locations of the output directory and the library files directory respectively.</p>
<pre class="fragment">apt-copynumber-onco-som-ssa \
  --log-file $OUTDIR/apt-copynumber-onco-som-ssa.log \
  --out-dir $OUTDIR \
  --temp-dir $OUTDIR/temp \
  --force false \
  --keep-intermediate-data false \
  --keep-combined-cels false \
  --reference-file $LIBDIR/OncoScan.Ref103.na33.r1.SOM_REF_MODEL \
  --set-analysis-name mip240k \
  --annotation-file $LIBDIR/OncoScan.na33.r1.annot.db \
  --x-probes-file $LIBDIR/OncoScan.r1.chrXprobes \
  --y-probes-file $LIBDIR/OncoScan.r1.chrYprobes \
  --cel-pairs-file CELPairsFile.txt \
  --som-thresholds-file $LIBDIR/OncoScan.Som1.0.r1.Som_thresh.txt \
  --agr-denominator-source SNP \
  --agr-denominator-percentile 0.75 \
  --igender-female-threshold 0.44 \
  --igender-male-threshold 0.63 \
  --nddicov-min-nd-probe-count-per-bin 30 \
  --nddicov-use-nd-signal-markers off \
  --dual-channel-normalization true \
  --norm-use-autosomal false \
  --norm-use-autosomal-nd true \
  --norm-use-som false 
</pre> </div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Mon May 19 2014 23:25:13 for Affymetrix Power Tools by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
