<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1" />
    <title>
      Affymetrix Developer Network
    </title>
    <link href="css/styles.css" rel="stylesheet" type="text/css" />
  </head>
  <body>
	<div id="wrapper">
		<div id="page">
			<div id="header">
				<img alt="Affymetrix.com" src="images/header_banner.jpg" />
			</div>
			<div id="content">
			
			<h1>Affymetrix® Chromosome Analysis Suite CYCHP Data File Format</h1>
			
			<h2>CYCHP FILES</h2>
			
			 <p><b>Description</b></p>

			<p>CYCHP files contain analysis results generated by the Chromosome Analysis Suite (ChAS).</p>

			<p><b>Format</b></p>

			<p>The format of the CYCHP files generated by the CHAS software uses the Command Console <a href="generic.html">binary</a> data format.</p>

			<p>The data type identifier is set to: "affymetrix-multi-data-type-analysis"</p>

			<p>The parameters stored in the header of the file include algorithm parameters (those whose names are prefixed with "affymetrix-algorithm-param-" and summary statistics (those whose names are prefixed with "affymetrix-chipsummary-". These parameters are algorithm specific.</p>

			<p>Analysis results are retained in four data groups.  Each data group contains one or more data sets as shown below.</p>

			
	<table>
	<tr>
        <th>
        <p><b>Data Group Name</b></p>
        </th>
        <th>
        <p><b>Description</b></p>
        </th>
        <th>
        <p><b>Number of Data Sets</b></p>
        </th>
       </tr>
       <tr>
        <td>
        <p>Chromosomes</p>
        </td>
        <td>
        <p>Summary analysis results for
        each chromosome</p>
        </td>
        <td>
        <p>1</p>
        </td>
       </tr>
       <tr>
        <td>
        <p>ProbeSets</p>
        </td>
        <td>
        <p>Analysis
        results for each probeset</p>
        </td>
        <td>
        <p>2</p>
        </td>
       </tr>
       <tr>
        <td>
        <p>Algorithm Data</p>
        </td>
        <td>
        <p>Data produced during algorithm development</p>
        </td>
        <td>
        <p>1</p>
        </td>
       </tr>
       <tr>
        <td>
        <p>Segments</p>
        </td>
        <td>
        <p>Analysis results for the segments found in the cychp</p>
        file
        </td>
        <td>
        <p>Maximum of *3 or **5 </p>
        </td>
       </tr>
	    <tr>
        <td>
        <p>*Genotyping</p>
        </td>
        <td>
        <p>Genotyping analysis results from BRLMM-P-PLUS</p>
        </td>
        <td>
        <p>1</p>
        </td>
       </tr>
	   </table>	  
		<p>*  CytoScan HD Array</p>
		<p>** 2.7M Cytogenetic Array</p>
	   
	    <p>Each data set header will contain a set of parameters to define the column labels for the data set.</p>

	    <p>The following describes the data groups and data sets stored in the file.</p>

	    <p><b>Chromosomes Data Group</b></p>

	    <p>The Chromosomes data group consists of s single data set. This data group has one data set.</p>

	    <p>The <b><i>Summary data set</i></b>  contains summary information for each chromosome.  This data set  will contain the following columns:</p>
			
			
		<table>
       <tbody><tr>
        <th>
        Column Name
        </th>
        <th>
        Column Type
        </th>
        <th>
        Description
        </th>
       </tr>
       <tr>
        <td>
        Chromosome
        </td>
        <td>
        UByte
        </td>
        <td>
        The chromosome display values (1-22, X, Y, MT) are stored
        in the data set header. The values in this column are defined as
        1-22, 24 (for X), 25 (for Y) and 26
        (for MT) and 255 (for no value)
        </td>
       </tr>
       <tr>
        <td>
        Display
        </td>
        <td>
        Ascii
        </td>
        <td>
        Text label for the chromosome
        </td>
       </tr>
       <tr>
        <td>
        StartIndex
        </td>
        <td>
        UInt
        </td>
        <td>
        Zero-based reference to the row in the CopyNumber data
        set of the ProbeSets data group containing results for this chromosome
        </td>
       </tr>
       <tr>
        <td>
        MarkerCount
        </td>
        <td>
        UInt
        </td>
        <td>
        The number of markers in the chromosome
        </td>
       </tr>
       <tr>
        <td>
        MinSignal
        </td>
        <td>
        Float
        </td>
        <td>
        Minimum log2 ratio
        value found in the chromosome*
        </td>
       </tr>
       <tr>
        <td>
        MaxSignal
        </td>
        <td>
        Float
        </td>
        <td>
        Maximum log2
        ratio value found in the chromosome*
        </td>
       </tr>	   
       <tr>
        <td>
        MedianCNState
        </td>
        <td>
        Float
        </td>
        <td>
        Median
        calibrated log2 ratio
        </td>
       </tr>
       <tr>
        <td>
        HomFrequency
        </td>
        <td>
        Float
        </td>
        <td>
        Frequency of
        Homozygosity
        </td>
       </tr>
       <tr>
        <td>
        HetFrequency
        </td>
        <td>
        Float
        </td>
        <td>
        Frequency of
        Heterozygosity
        </td>
       </tr>
       <tr>
        <td>
        Mosaicism
        </td>
        <td>
        Float
        </td>
        <td>
        Median
        mosaicism mixture value
        </td>
       </tr>
       <tr>
        <td>
        LOH
        </td>
        <td>
        Float
        </td>
        <td>
        Proportion of
        genomic distance of LOH calls per chromosome
        </td>
       </tr>
	   <tr>
        <td>
        MedianSignal
        </td>
        <td>
        Float
        </td>
        <td>
        Median log2
        ratio value found in the chromosome*
        </td>
       </tr>
      </tbody></table>
	  
	  <p>* The log2 ratios are trimmed log2 ratios by default, but the values might go beyond the trimmed values due to GC correction and/or wave correction step.</p>

<p><b>Probesets Data Group</b></p>

<p>The Probesets data group consists of 2 data sets, containing analysis results specific to the probesets.  The data sets included in this group are</p>

    <ul>
	<li>CopyNumber</li>
    <li>AllelePeaks</li>
	</ul>

<p>The <b><i>CopyNumber data set</i></b> will contain the following columns:</p>
	

<table>
       <tbody><tr>
        <th>
        Column Name
        </th>
        <th>
        Column Type
        </th>
        <th>
        Description
        </th>
       </tr>
       <tr>
        <td>
        ProbeSetName
        </td>
        <td>
        Ascii
        </td>
        <td>
		The name of the probeset.  The probeset names begin with "C" (copy number probes) or "S" (SNP probes).
        </td>
       </tr>
       <tr>
        <td>
        Chromosome
        </td>
        <td>
        UByte
        </td>
        <td>
        The chromosome display values (1-22, X, Y, MT) are stored
        in the data set header. The values in this column are defined as
        1-22, 24 (for X), 25 (for Y) and 26
        (for MT) and 255 (for no value).  The data rows for each chromosome are consecutive, in a single block, 
		starting at the index given in the "Chromosomes/Summary" data set in the StartIndex column.  The number 
		of data rows for each chromosome is exactly equal to the value given in the "Chromosomes/Summary" data set in the MarkerCount column.
        </td>
       </tr>
       <tr>
        <td>
        Position
        </td>
        <td>
        UInt
        </td>
        <td>
		Genomic position in the chromosome.
		The Position values for each chromosome are strictly increasing. 
		Each position can be assosciated with one probe only.
        </td>
       </tr>
       <tr>
        <td>
        Log2Ratio
        </td>
        <td>
        Float
        </td>
        <td>
        Log2 ratio (signal / median of Reference signal)
        </td>
       </tr>
       <tr>
        <td>
        WeightedLog2Ratio
        </td>
        <td>
        Float
        </td>
        <td>
        Signal restored log2 ratio value
        </td>
       </tr>
       <tr>
        <td>
        SmoothSignal
        </td>
        <td>
        Float
        </td>
        <td>
        Each value is the
        result of a kernel smooth over a window of genomically contiguous log2
        ratios.  By default this is converted to a copy number value using a log-linear model that fits
        each CN state to its expected log2 ratio. This is useful for seeing mosaics.
        </td>
       </tr>
      </tbody></table>

<p>The <b><i>AllelePeaks data set</i></b> houses data summarizing allele patterns and contains the following columns: </p>

<table cellpadding="0" border="1" width="600" id="table8" style="width: 6.25in;" class="MsoNormalTable">
       <tbody><tr>
        <th>
        Column Name
        </th>
        <th>
        Column Type
        </th>
        <th>
        Description
        </th>
       </tr>
       <tr>
        <td>
        ProbesetName
        </td>
        <td>
        Ascii
        </td>
        <td>
        The name of the probeset
        </td>
       </tr>
       <tr>
        <td>
        Chromosome
        </td>
        <td>
        UByte
        </td>
        <td>
        The chromosome display values (1-22, X, Y, MT) are
        stored in the data set header. The values in this column are defined as
        1-22, 24 (for X), 25 (for Y) and 26
        (for MT) and 255 (for no value)
        </td>
       </tr>
       <tr>
        <td>
        Position
        </td>
        <td>
        UInt
        </td>
        <td>
        Genomic position in the chromosome
        </td>
       </tr>
       <tr>
        <td>
        AllelePeaks0
        </td>
        <td>
        UInt
        </td>
        <td>
        First three encoded allele peak values
        </td>
       </tr>
       <tr>
        <td>
        AllelePeaks1
        </td>
        <td>
        UInt
        </td>
        <td>
        Next three encoded allele peak values
        </td>
       </tr>
      </tbody></table>
	  
	  
	  <p><b>AlgorithmData Data Group</b></p>

	<p>The AlgorithmData data group contains data produced during algorithm development.  It is not intended for external use and is not read or used by the Chromosome Analysis Suite User Interface (the Browser). The data is subject to change and/or removal at any time in future releases of APT and/or Chromosome Analysis Suite.</p>

	<p>There is one data set in the AlgorithmData data group called MarkerABSignal.</p>

	<p>The <b><i>MarkerABSignal data set</i></b> will contain the following columns:</p>
	  
	<table>
       <tbody><tr>
        <th>
        Column Name
        </th>
        <th>
        Column Type
        </th>
        <th>
        Description
        </th>
       </tr>
       <tr>
        <td>
        Index
        </td>
        <td>
        UInt
        </td>
        <td>
        This column contains the index of a SNP probeset in the CopyNumber data set of the ProbeSet data group (0-based)
        </td>
       </tr>
       <tr>
        <td>
        *ASignal
        </td>
        <td>
        Float
        </td>
        <td>
        A Signal value
        from PLIER
        </td>
       </tr>
       <tr>
        <td>
        *BSignal
        </td>
        <td>
        Float
        </td>
        <td>
        B Signal value from PLIER
        </td>
       </tr>
       <tr>
        <td>
        SCAR
        </td>
        <td>
        Float
        </td>
        <td>
        Scaled centered allelic ratio values, LOH metric
        </td>
       </tr>
      </tbody></table>
	  <p>* These columns are present in this table for cychp files generated from the 2.7M Cytogenetic Array.  They can be found in the Genotyping Data Group's Calls data set for cychp files generated from the CytoScan array.</p>
	 <p><b>Segments Data Group</b></p>

	<p>The Segments data group contains data sets housing segments data.  The Segments data group is optional and will be included in the CYCHP file 
	if segments are reported by an analysis.  This data group will not be present if there are no segments to report.</p>
	<p>Segments data group can contain a variable number of data sets.  Current analysis produces up to five data sets, but future analyses may produce additional data sets.</p>
	<p>Currently, the presence or absence of any data set in the "Segments" data group
	indicates whether or not the analysis generating that data was performed.  
	The following uses the Mosaicism data set as an example:</p>
	<ul>
		<li>If the "Mosaicism" data set is in the file then a "Mosaicism" analysis was performed.</li>  
		<li>An empty "Mosaicism" data set signals that analysis for "Mosaicism" was performed and none was found.</li>
		<li>A missing data set for "Mosaisicism" indicates that the analysis was not performed.</li>
		<li>If the Segments data group is not present, no analyses were performed that resulted in segments to report.</li>
	</ul>

	<p>In each data set in this data group:</p>
	<ul>
	<li>The StartPosition column is strictly increasing.</li>
    <li>The StopPosition is always greater than the StartPosition in the same row and less than the StartPosition in the next row.</li>
    <li>Each StartPosition and StopPosition corresponds exactly to one Position in the "ProbeSets/CopyNumber" dataset</li>  
	</ul>

<p>The data sets included in this data group are:</p>

    <ul>
	<li>CN</li>
    <li>LOH</li>
    <li>CNNeutralLOH</li>
    <li>*NormalDiploid</li>
    <li>*Mosaicism</li>
	</ul>	
	
	<p>* These data sets can be present in files generated from the 2.7M Cytogenetic Array.  They are not present in files generated from the CytoScan array.</p>
	<p>The <b><i>CN data set</i></b>  will contain the following columns:</p>
	  
	  <table>
       <tbody><tr>
        <th>
        Column Name
        </th>
        <th>
        Column Type
        </th>
        <th>
        Description
        </th>
       </tr>
       <tr>
        <td>
        SegmentID
        </td>
        <td>
        UInt
        </td>
        <td>
        Segment identifier unique to the CYCHP file
        </td>
       </tr>
       <tr>
        <td>
        Chromosome
        </td>
        <td>
        UByte
        </td>
        <td>
        The chromosome display values (1-22, X, Y, MT) are
        stored in the data set header. The values in this column are defined as
        1-22, 24 (for X), 25 (for Y) and 26
        (for MT) and 255 (for no value)
        </td>
       </tr>
       <tr>
        <td>
        StartPosition
        </td>
        <td>
        UInt
        </td>
        <td>
        Start genomic
        position of the segment
        </td>
       </tr>
       <tr>
        <td>
        StopPosition
        </td>
        <td>
        UInt
        </td>
        <td>
        Ending genomic
        position of the segment
        </td>
       </tr>
       <tr>
        <td>
        MarkerCount
        </td>
        <td>
        Int
        </td>
        <td>
        Number of
        markers in the segment
        </td>
       </tr>
       <tr>
        <td>
        MeanMarkerDistance
        </td>
        <td>
        UInt
        </td>
        <td>
        Mean distance
        between markers in the segment
        </td>
       </tr>
       <tr>
        <td>
        State
        </td>
        <td>
        Float
        </td>
        <td>
        Copy Number
        State 
        </td>
       </tr>
       <tr>
        <td>
        Confidence
        </td>
        <td>
        Float
        </td>
        <td>
        Indicator score for non-normal copy number
        </td>
       </tr>
      </tbody></table>
	  
	  
	  <p>The <b><i>LOH data set</i></b> will contain the following columns:</p>
	  
	  <table>
       <tbody><tr>
        <th>
        Column Name
        </th>
        <th>
        Column Type
        </th>
        <th>
        Description
        </th>
       </tr>
       <tr>
        <td>
        SegmentID
        </td>
        <td>
        UInt
        </td>
        <td>
        Segment identifier unique to the CYCHP file
        </td>
       </tr>
       <tr>
        <td>
        Chromosome
        </td>
        <td>
        UByte
        </td>
        <td>
        The chromosome display values (1-22, X, Y, MT) are
        stored in the data set header. The values in this column are defined as
        1-22, 24 (for X), 25 (for Y) and 26
        (for MT) and 255 (for no value)
        </td>
       </tr>
       <tr>
        <td>
        StartPosition
        </td>
        <td>
        UInt
        </td>
        <td>
        Start genomic
        position of the segment
        </td>
       </tr>
       <tr>
        <td>
        StopPosition
        </td>
        <td>
        UInt
        </td>
        <td>
        Ending genomic
        position of the segment
        </td>
       </tr>
       <tr>
        <td>
        MarkerCount
        </td>
        <td>
        Int
        </td>
        <td>
        Number of
        markers in the segment
        </td>
       </tr>
       <tr>
        <td>
        MeanMarkerDistance
        </td>
        <td>
        UInt
        </td>
        <td>
        Mean distance between
        markers in the segment
        </td>
       </tr>
       <tr>
        <td>
        LOH
        </td>
        <td>
        UByte
        </td>
        <td>
        The value of
        this column is 1 when a Loss of Heterozygosity is found, 0 when not
        found.
        </td>
       </tr>
       <tr>
        <td>
        Confidence
        </td>
        <td>
        Float
        </td>
        <td>
        The ratio of
        the probability of the SCAR measurements under the LOH model to the sum
        of the probability under each of the LOH and non-LOH models. 
        </td>
       </tr>
      </tbody></table>
	  
	  <p>The <b><i>CNNeutralLOH data set</i></b> will contain the following columns:</p>
	  
	  <table>
       <tbody><tr>
        <th>
        Column Name
        </th>
        <th>
        Column Type
        </th>
        <th>
        Description
        </th>
       </tr>
       <tr>
        <td>
        SegmentID
        </td>
        <td>
        UInt
        </td>
        <td>
        &nbsp;Segment identifier unique to the CYCHP file
        </td>
       </tr>
       <tr>
        <td>
        Chromosome
        </td>
        <td>
        UByte
        </td>
        <td>
        The chromosome display values (1-22, X, Y, MT) are
        stored in the data set header. The values in this column are defined as
        1-22, 24 (for X), 25 (for Y) and 26
        (for MT) and 255 (for no value)
        </td>
       </tr>
       <tr>
        <td>
        StartPosition
        </td>
        <td>
        UInt
        </td>
        <td>
        Start genomic position
        of the segment
        </td>
       </tr>
       <tr>
        <td>
        StopPosition
        </td>
        <td>
        UInt
        </td>
        <td>
        Ending genomic
        position of the segment
        </td>
       </tr>
       <tr>
        <td>
        MarkerCount
        </td>
        <td>
        Int
        </td>
        <td>
        Number of
        markers in the segment
        </td>
       </tr>
       <tr>
        <td>
        MeanMarkerDistance
        </td>
        <td>
        UInt
        </td>
        <td>
        Mean distance
        between markers in the segment
        </td>
       </tr>
       <tr>
        <td>
        CNNeutralLOH
        </td>
        <td>
        UByte
        </td>
        <td>
        The value of
        this column is 1 when CN State is 2 and LOH is 1. &nbsp; Otherwise, the value of this column
        is 0.
        </td>
       </tr>
       <tr>
        <td>
        Confidence
        </td>
        <td>
        Float
        </td>
        <td>
        The ratio of
        the probability of the SCAR measurements under the LOH model to the sum
        of the probability under each of the LOH and non-LOH models. 
        </td>
       </tr>
      </tbody></table>
	  
	  <p>The <b><i>NormalDiploid data set</i></b> will contain the following columns:</p>
	  
	  <table>
       <tbody><tr>
        <th>
        Column Name
        </th>
        <th>
        Column Type
        </th>
        <th>
        Description
        </th>
       </tr>
       <tr>
        <td>
        SegmentID
        </td>
        <td>
        UInt
        </td>
        <td>
        Segment identifier unique to the CYCHP file
        </td>
       </tr>
       <tr>
        <td>
        Chromosome
        </td>
        <td>
        UByte
        </td>
        <td>
        The chromosome display values (1-22, X, Y, MT) are
        stored in the data set header. The values in this column are defined as
        1-22, 24 (for X), 25 (for Y) and 26
        (for MT) and 255 (for no value)
        </td>
       </tr>
       <tr>
        <td>
        StartPosition
        </td>
        <td>
        UInt
        </td>
        <td>
        Start genomic position
        of the segment
        </td>
       </tr>
       <tr>
        <td>
        StopPosition
        </td>
        <td>
        UInt
        </td>
        <td>
        Ending genomic
        position of the segment
        </td>
       </tr>
       <tr>
        <td>
        MarkerCount
        </td>
        <td>
        Int
        </td>
        <td>
        Number of
        markers in the segment
        </td>
       </tr>
       <tr>
        <td>
        MeanMarkerDistance
        </td>
        <td>
        UInt
        </td>
        <td>
        Mean distance
        between markers in the segment
        </td>
       </tr>
       <tr>
        <td>
        NormalDiploid
        </td>
        <td>
        UByte
        </td>
        <td>
        The value of
        this column is 1 when CN State is 2 and LOH is 0. Otherwise, the value of this column
        is 0.
        </td>
       </tr>
       <tr>
        <td>
        Confidence
        </td>
        <td>
        Float
        </td>
        <td>
        Confidence
        </td>
       </tr>
      </tbody></table>
	  
	  
	  <p>The <b><i>Mosaicism data set</i></b> will contain the following columns:</p>
	  
	  <table>
       <tbody><tr>
        <th>
        Column Name
        </th>
        <th>
        Column Type
        </th>
        <th>
        Description
        </th>
       </tr>
       <tr>
        <td>
        SegmentID
        </td>
        <td>
        UInt
        </td>
        <td>
        &nbsp;Segment identifier unique to the CYCHP file
        </td>
       </tr>
       <tr>
        <td>
        Chromosome
        </td>
        <td>
        UByte
        </td>
        <td>
        The chromosome display values (1-22, X, Y, MT) are
        stored in the data set header. The values in this column are defined as
        1-22, 24 (for X), 25 (for Y) and 26
        (for MT) and 255 (for no value)
        </td>
       </tr>
       <tr>
        <td>
        StartPosition
        </td>
        <td>
        UInt
        </td>
        <td>
        Start genomic
        position of the segment
        </td>
       </tr>
       <tr>
        <td>
        StopPosition
        </td>
        <td>
        UInt
        </td>
        <td>
        Ending genomic
        position of the segment
        </td>
       </tr>
       <tr>
        <td>
        MarkerCount
        </td>
        <td>
        Int
        </td>
        <td>
        Number of
        markers in the segment
        </td>
       </tr>
       <tr>
        <td>
        MeanMarkerDistance
        </td>
        <td>
        UInt
        </td>
        <td>
        Mean distance
        between markers in the segment
        </td>
       </tr>
       <tr>
        <td>
        Mosaicism
        </td>
        <td>
        UByte
        </td>
        <td>
        The value of this column is 1 when there is mosaicism present.  Otherwise, the value of this column is 0.&nbsp; 
        </td>
       </tr>
       <tr>
        <td>
        Confidence
        </td>
        <td>
        Float
        </td>
        <td>
        Proportion of
        markers that are above or below the thresholds required to make a CN
        change call for a running median segment size of 251.
        </td>
       </tr>
       <tr>
        <td>
        State
        </td>
        <td>
        Float
        </td>
        <td>
        CN State
        </td>
       </tr>
       <tr>
        <td>
        Mixture
        </td>
        <td>
        Float
        </td>
        <td>
        Estimated
        mosaicism mixture (i.e. typically one of -1, -0.7, -0.5, -0.3, 0, 0.3,
        0.5, 0.7, 1.0)
        </td>
       </tr>
      </tbody></table>
	 
	  <p><b>The Genotyping Data Group (CytoScan HD arrays only).</b></p>

	  <p>There is one data set in the Genotyping data group called Calls.</p>

      <p>The <b><i>Calls data set</i></b> will contain the following columns:</p>	  
       <table>
       <tbody><tr>
        <th>
        Column Name
        </th>
        <th>
        Column Type
        </th>
        <th>
        Description
        </th>
       </tr>
       <tr>
        <td>
        Index
        </td>
        <td>
        UInt
        </td>
        <td>
        This column contains the index of a SNP probeset in the CopyNumber data set of the ProbeSet data group (0-based).
		The Index values are strictly increasing.
        </td>
       </tr>
       <tr>
        <td>
        Call
        </td>
        <td>
        UByte
        </td>
        <td>
        6 = AA, 7=BB, 8=AB, no call otherwise (no
        call displayed beyond confidence threshold)
        </td>
       </tr>
       <tr>
        <td>
        Confidence
        </td>
        <td>
        Float
        </td>
        <td>
        Call confidence from BRLMM-P-PLUS
        </td>
       </tr>
	    <tr>
        <td>
        ForcedCall
        </td>
        <td>
        UByte
        </td>
        <td>
        6 = AA, 7=BB, 8=AB
        </td>
       </tr>
	    <tr>
        <td>
        ASignal
        </td>
        <td>
        Float
        </td>
        <td>
        A Signal value 
        </td>
       </tr>
	    <tr>
        <td>
        BSignal
        </td>
        <td>
        Float
        </td>
        <td>
        B Signal value 
        </td>
       </tr>
	    <tr>
        <td>
        SignalStrength
        </td>
        <td>
        Float
        </td>
        <td>
        Strength value from BRLMM-P-PLUS (transform=MvA)
        </td>
       </tr>
	    <tr>
        <td>
        Contrast
        </td>
        <td>
        Float
        </td>
        <td>
        Contrast value
        from BRLMM-P-PLUS (transform=MvA)
        </td>
       </tr>
      </tbody></table>
	
			</div>
			<div id="footer">
				&#169; 2011 Affymetrix, Inc. All rights reserved.
			</div>
		</div>
	</div>
  </body>
</html>